Index: node-v10.12.0/deps/uv/src/unix/linux-core.c
===================================================================
--- node-v10.12.0.orig/deps/uv/src/unix/linux-core.c
+++ node-v10.12.0/deps/uv/src/unix/linux-core.c
@@ -213,6 +213,8 @@ void uv__io_poll(uv_loop_t* loop, int ti
   int op;
   int i;
 
+  memset(&e, 0, sizeof(e));
+
   if (loop->nfds == 0) {
     assert(QUEUE_EMPTY(&loop->watcher_queue));
     return;
