Index: scikit-umfpack-0.3.3/scikits/umfpack/tests/test_interface.py
===================================================================
--- scikit-umfpack-0.3.3.orig/scikits/umfpack/tests/test_interface.py
+++ scikit-umfpack-0.3.3/scikits/umfpack/tests/test_interface.py
@@ -3,7 +3,7 @@ from __future__ import division, print_f
 import warnings
 import unittest
 
-from numpy.testing import assert_allclose, run_module_suite
+from numpy.testing import assert_allclose
 from numpy.linalg import norm as dense_norm
 
 from scipy.sparse import csc_matrix, spdiags, SparseEfficiencyWarning
@@ -133,4 +133,4 @@ class TestSolvers(unittest.TestCase):
         assert_allclose(A2, A.A, atol=1e-13)
 
 if __name__ == "__main__":
-    run_module_suite()
+    unittest.main()
Index: scikit-umfpack-0.3.3/scikits/umfpack/tests/test_umfpack.py
===================================================================
--- scikit-umfpack-0.3.3.orig/scikits/umfpack/tests/test_umfpack.py
+++ scikit-umfpack-0.3.3/scikits/umfpack/tests/test_umfpack.py
@@ -8,7 +8,7 @@ import random
 import unittest
 import warnings
 
-from numpy.testing import assert_array_almost_equal, run_module_suite
+from numpy.testing import assert_array_almost_equal
 
 from scipy import rand, matrix, diag, eye
 from scipy.sparse import csc_matrix, linalg, spdiags, SparseEfficiencyWarning
@@ -239,4 +239,4 @@ class TestFactorization(_DeprecationAcce
 
 
 if __name__ == "__main__":
-    run_module_suite()
+    unittest.main()
Index: scikit-umfpack-0.3.3/scikits/umfpack/__init__.py
===================================================================
--- scikit-umfpack-0.3.3.orig/scikits/umfpack/__init__.py
+++ scikit-umfpack-0.3.3/scikits/umfpack/__init__.py
@@ -22,5 +22,3 @@ if __doc__ is not None:
     del _umfpack_doc, _interface_doc
 
 __all__ = [s for s in dir() if not s.startswith('_')]
-from numpy.testing import Tester
-test = Tester().test
