Index: scilab-6.0.2/modules/helptools/macros/help.sci
===================================================================
--- scilab-6.0.2.orig/modules/helptools/macros/help.sci
+++ scilab-6.0.2/modules/helptools/macros/help.sci
@@ -58,7 +58,8 @@ function help(varargin)
             error(msprintf(gettext("%s: The help browser is disabled in %s mode.\n"), "help", getscilabmode()));
         end
     else
-        error(msprintf(gettext("%s: help file(.jar) is not installed.\n"), "help"));
+        warning(msprintf(gettext("%s: help file(.jar) is not installed.\n"), "help"));
+        unix_w("pkcon install scilab-doc");
     end
 
 endfunction

