Index: scilab-6.0.2/modules/graphic_objects/src/java/org/scilab/modules/graphic_objects/xmlloader/CSSParser.java
===================================================================
--- scilab-6.0.2.orig/modules/graphic_objects/src/java/org/scilab/modules/graphic_objects/xmlloader/CSSParser.java
+++ scilab-6.0.2/modules/graphic_objects/src/java/org/scilab/modules/graphic_objects/xmlloader/CSSParser.java
@@ -14,7 +14,6 @@ import java.io.StringReader;
 import java.util.HashMap;
 import java.util.Map;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 
 /**
Index: scilab-6.0.2/modules/helptools/src/java/jflex/c.jflex
===================================================================
--- scilab-6.0.2.orig/modules/helptools/src/java/jflex/c.jflex
+++ scilab-6.0.2/modules/helptools/src/java/jflex/c.jflex
@@ -6,7 +6,6 @@ import java.io.Reader;
 import java.io.StringReader;
 import java.io.IOException;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 %%
 
Index: scilab-6.0.2/modules/helptools/src/java/jflex/function.jflex
===================================================================
--- scilab-6.0.2.orig/modules/helptools/src/java/jflex/function.jflex
+++ scilab-6.0.2/modules/helptools/src/java/jflex/function.jflex
@@ -5,7 +5,6 @@ package org.scilab.modules.helptools ;
 import java.io.IOException;
 import java.io.StringReader;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 %%
 
Index: scilab-6.0.2/modules/helptools/src/java/jflex/java.jflex
===================================================================
--- scilab-6.0.2.orig/modules/helptools/src/java/jflex/java.jflex
+++ scilab-6.0.2/modules/helptools/src/java/jflex/java.jflex
@@ -8,7 +8,6 @@ import java.io.IOException;
 
 import org.scilab.modules.helptools.c.AbstractCCodeHandler;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 %%
 
Index: scilab-6.0.2/modules/helptools/src/java/jflex/scilab.jflex
===================================================================
--- scilab-6.0.2.orig/modules/helptools/src/java/jflex/scilab.jflex
+++ scilab-6.0.2/modules/helptools/src/java/jflex/scilab.jflex
@@ -15,7 +15,6 @@ import java.io.Reader;
 import java.io.StringReader;
 import java.io.IOException;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 %%
 
Index: scilab-6.0.2/modules/helptools/src/java/jflex/xml.jflex
===================================================================
--- scilab-6.0.2.orig/modules/helptools/src/java/jflex/xml.jflex
+++ scilab-6.0.2/modules/helptools/src/java/jflex/xml.jflex
@@ -6,7 +6,6 @@ import java.io.Reader;
 import java.io.StringReader;
 import java.io.IOException;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 %%
 
Index: scilab-6.0.2/modules/helptools/src/java/org/scilab/modules/helptools/SynopsisLexer.java
===================================================================
--- scilab-6.0.2.orig/modules/helptools/src/java/org/scilab/modules/helptools/SynopsisLexer.java
+++ scilab-6.0.2/modules/helptools/src/java/org/scilab/modules/helptools/SynopsisLexer.java
@@ -7,7 +7,6 @@ package org.scilab.modules.helptools ;
 import java.io.IOException;
 import java.io.StringReader;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 
 /**
Index: scilab-6.0.2/modules/helptools/src/java/org/scilab/modules/helptools/XML/XMLLexer.java
===================================================================
--- scilab-6.0.2.orig/modules/helptools/src/java/org/scilab/modules/helptools/XML/XMLLexer.java
+++ scilab-6.0.2/modules/helptools/src/java/org/scilab/modules/helptools/XML/XMLLexer.java
@@ -8,7 +8,6 @@ import java.io.Reader;
 import java.io.StringReader;
 import java.io.IOException;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 
 /**
Index: scilab-6.0.2/modules/helptools/src/java/org/scilab/modules/helptools/c/CLexer.java
===================================================================
--- scilab-6.0.2.orig/modules/helptools/src/java/org/scilab/modules/helptools/c/CLexer.java
+++ scilab-6.0.2/modules/helptools/src/java/org/scilab/modules/helptools/c/CLexer.java
@@ -8,7 +8,6 @@ import java.io.Reader;
 import java.io.StringReader;
 import java.io.IOException;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 
 /**
Index: scilab-6.0.2/modules/helptools/src/java/org/scilab/modules/helptools/java/JavaLexer.java
===================================================================
--- scilab-6.0.2.orig/modules/helptools/src/java/org/scilab/modules/helptools/java/JavaLexer.java
+++ scilab-6.0.2/modules/helptools/src/java/org/scilab/modules/helptools/java/JavaLexer.java
@@ -10,7 +10,6 @@ import java.io.IOException;
 
 import org.scilab.modules.helptools.c.AbstractCCodeHandler;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 
 /**
Index: scilab-6.0.2/modules/helptools/src/java/org/scilab/modules/helptools/scilab/ScilabLexer.java
===================================================================
--- scilab-6.0.2.orig/modules/helptools/src/java/org/scilab/modules/helptools/scilab/ScilabLexer.java
+++ scilab-6.0.2/modules/helptools/src/java/org/scilab/modules/helptools/scilab/ScilabLexer.java
@@ -14,7 +14,6 @@ import java.io.Reader;
 import java.io.StringReader;
 import java.io.IOException;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 
 /**
Index: scilab-6.0.2/modules/scinotes/src/java/jflex/function.jflex
===================================================================
--- scilab-6.0.2.orig/modules/scinotes/src/java/jflex/function.jflex
+++ scilab-6.0.2/modules/scinotes/src/java/jflex/function.jflex
@@ -8,7 +8,6 @@ import java.util.ArrayList;
 import javax.swing.text.BadLocationException;
 import javax.swing.text.Element;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 %%
 
Index: scilab-6.0.2/modules/scinotes/src/java/jflex/indentation.jflex
===================================================================
--- scilab-6.0.2.orig/modules/scinotes/src/java/jflex/indentation.jflex
+++ scilab-6.0.2/modules/scinotes/src/java/jflex/indentation.jflex
@@ -6,7 +6,6 @@ import java.io.IOException;
 import javax.swing.text.Element;
 import javax.swing.text.BadLocationException;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 %% 
 
Index: scilab-6.0.2/modules/scinotes/src/java/jflex/matchingblock.jflex
===================================================================
--- scilab-6.0.2.orig/modules/scinotes/src/java/jflex/matchingblock.jflex
+++ scilab-6.0.2/modules/scinotes/src/java/jflex/matchingblock.jflex
@@ -5,7 +5,6 @@ package org.scilab.modules.scinotes;
 import java.io.IOException;
 import javax.swing.text.Element;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 %%
 
Index: scilab-6.0.2/modules/scinotes/src/java/jflex/scilab.jflex
===================================================================
--- scilab-6.0.2.orig/modules/scinotes/src/java/jflex/scilab.jflex
+++ scilab-6.0.2/modules/scinotes/src/java/jflex/scilab.jflex
@@ -12,7 +12,6 @@ import javax.swing.text.BadLocationExcep
 import javax.swing.text.Element;
 import org.scilab.modules.commons.ScilabCommonsUtils;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 
 %%
Index: scilab-6.0.2/modules/scinotes/src/java/org/scilab/modules/scinotes/FunctionScanner.java
===================================================================
--- scilab-6.0.2.orig/modules/scinotes/src/java/org/scilab/modules/scinotes/FunctionScanner.java
+++ scilab-6.0.2/modules/scinotes/src/java/org/scilab/modules/scinotes/FunctionScanner.java
@@ -10,7 +10,6 @@ import java.util.ArrayList;
 import javax.swing.text.BadLocationException;
 import javax.swing.text.Element;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 
 /**
Index: scilab-6.0.2/modules/scinotes/src/java/org/scilab/modules/scinotes/IndentScanner.java
===================================================================
--- scilab-6.0.2.orig/modules/scinotes/src/java/org/scilab/modules/scinotes/IndentScanner.java
+++ scilab-6.0.2/modules/scinotes/src/java/org/scilab/modules/scinotes/IndentScanner.java
@@ -8,7 +8,6 @@ import java.io.IOException;
 import javax.swing.text.Element;
 import javax.swing.text.BadLocationException;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 
 /**
Index: scilab-6.0.2/modules/scinotes/src/java/org/scilab/modules/scinotes/MatchingBlockScanner.java
===================================================================
--- scilab-6.0.2.orig/modules/scinotes/src/java/org/scilab/modules/scinotes/MatchingBlockScanner.java
+++ scilab-6.0.2/modules/scinotes/src/java/org/scilab/modules/scinotes/MatchingBlockScanner.java
@@ -7,7 +7,6 @@ package org.scilab.modules.scinotes;
 import java.io.IOException;
 import javax.swing.text.Element;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 
 /**
Index: scilab-6.0.2/modules/scinotes/src/java/org/scilab/modules/scinotes/ScilabLexer.java
===================================================================
--- scilab-6.0.2.orig/modules/scinotes/src/java/org/scilab/modules/scinotes/ScilabLexer.java
+++ scilab-6.0.2/modules/scinotes/src/java/org/scilab/modules/scinotes/ScilabLexer.java
@@ -14,7 +14,6 @@ import javax.swing.text.BadLocationExcep
 import javax.swing.text.Element;
 import org.scilab.modules.commons.ScilabCommonsUtils;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 
 
