diff -Nur warzone2100/lib/ivis_opengl/gfx_api_vk.cpp new/lib/ivis_opengl/gfx_api_vk.cpp
--- warzone2100/lib/ivis_opengl/gfx_api_vk.cpp	2021-11-28 17:36:15.000000000 +0100
+++ new/lib/ivis_opengl/gfx_api_vk.cpp	2021-12-04 07:18:29.172562922 +0100
@@ -477,7 +477,7 @@
 	if (result != vk::Result::eSuccess)
 	{
 		// Failed to allocate memory!
-		throwResultException( result, "vmaCreateBuffer" );
+		vk::throwResultException( result, "vmaCreateBuffer" );
 	}
 
 	if (autoMap)
@@ -1304,7 +1304,7 @@
 			object = std::move(result.value);
 			break;
 		default:
-			throwResultException(result.result, "createGraphicsPipeline");
+			vk::throwResultException(result.result, "createGraphicsPipeline");
 	}
 }
 
@@ -1368,7 +1368,7 @@
 	if (result != vk::Result::eSuccess)
 	{
 		// Failed to allocate memory!
-		throwResultException( result, "vmaCreateBuffer" );
+		vk::throwResultException( result, "vmaCreateBuffer" );
 	}
 
 	buffer_size = size;
@@ -1486,7 +1486,7 @@
 	if (result != vk::Result::eSuccess)
 	{
 		// Failed to allocate memory!
-		throwResultException( result, "vmaCreateImage" );
+		vk::throwResultException( result, "vmaCreateImage" );
 	}
 
 	const auto imageViewCreateInfo = vk::ImageViewCreateInfo()
