From 996a638e04766df43cb8026673f93927b1047639 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Fri, 3 Jan 2020 23:46:51 +0100
Subject: [PATCH 31/35] configure: fix new libatopology check

---
 configure.ac | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 4bee49be81ba..b7ed81a0d32c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -49,6 +49,7 @@ AC_CHECK_LIB([asound], [snd_seq_client_info_get_pid], [HAVE_SEQ_CLIENT_INFO_GET_
 if test "$HAVE_SEQ_CLIENT_INFO_GET_PID" = "yes" ; then
     AC_DEFINE([HAVE_SEQ_CLIENT_INFO_GET_PID], 1, [alsa-lib supports snd_seq_client_info_get_pid])
 fi
+AC_CHECK_LIB([atopology], [snd_tplg_save], [have_topology="no"])
 
 #
 # NOTE: The library 'libffado' (at least v2.4.1) executes ctor/dtor of instances
@@ -70,11 +71,6 @@ AM_CONDITIONAL(HAVE_TOPOLOGY, test "$have_topology" = "yes")
 AM_CONDITIONAL(HAVE_SAMPLERATE, test "$have_samplerate" = "yes")
 AM_CONDITIONAL(HAVE_FFADO, test "$have_ffado" = "yes")
 
-# old libasound with the topology routines in the main library
-if test "x$have_topology" = "xyes" -a "x$ALSA_TOPOLOGY_LIBS" = "x"; then
-  ALSA_TOPOLOGY_LIBS="$ALSA_LIBS"
-fi
-
 dnl Use tinyalsa
 alsabat_backend_tiny=
 AC_ARG_ENABLE(alsabat_backend_tiny,
-- 
2.16.4

