From 1983e82b7e6dbdd83fb8825db5f685a5d6fcf364 Mon Sep 17 00:00:00 2001
From: Antonio Terceiro <terceiro@debian.org>
Date: Sat, 11 Oct 2014 16:54:58 -0300
Subject: [PATCH 01/10] bin/vagrant: silence warning about installer

Signed-off-by: Johannes Kastl <kastl@b1-systems.de>
---
 bin/vagrant | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/bin/vagrant b/bin/vagrant
index 7ca30b391..d3f4ea61a 100755
--- a/bin/vagrant
+++ b/bin/vagrant
@@ -221,11 +221,6 @@ begin
       end
     end
 
-    if !Vagrant.in_installer? && !Vagrant.very_quiet?
-      # If we're not in the installer, warn.
-      env.ui.warn(I18n.t("vagrant.general.not_in_installer") + "\n", prefix: false)
-    end
-
     # Acceptable experimental flag values include:
     #
     # Unset  - Disables experimental features
-- 
2.38.1

