From 51af48f8f2ae39ef682d4310970c87b17bb72780 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20=C4=8Cerm=C3=A1k?= <dcermak@suse.com>
Date: Mon, 11 May 2020 11:32:01 +0200
Subject: [PATCH 10/11] Bump version of net-ssh to ~> 6.0 and net-sftp to ~>
 3.0

---
 vagrant.gemspec | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vagrant.gemspec b/vagrant.gemspec
index 49fdf2e8e..78bb2b8b5 100644
--- a/vagrant.gemspec
+++ b/vagrant.gemspec
@@ -23,8 +23,8 @@ Gem::Specification.new do |s|
   s.add_dependency "listen", "~> 3.1.5"
   s.add_dependency "hashicorp-checkpoint", "~> 0.1.5"
   s.add_dependency "log4r", "~> 1.1.9", "< 1.1.11"
-  s.add_dependency "net-ssh", "~> 5.2.0"
-  s.add_dependency "net-sftp", "~> 2.1"
+  s.add_dependency "net-ssh", "~> 6.0"
+  s.add_dependency "net-sftp", "~> 3.0"
   s.add_dependency "net-scp", "~> 1.2.0"
   s.add_dependency "rb-kqueue", "~> 0.2.0"
   s.add_dependency "rest-client", ">= 1.6.0", "< 3.0"
-- 
2.26.2

