# HG changeset patch
# Parent  a5fe45b73fb205e3c5a0734db3885c37afcc706b

Index: firefox-145.0/gfx/skia/skia/include/codec/SkEncodedOrigin.h
===================================================================
--- firefox-145.0.orig/gfx/skia/skia/include/codec/SkEncodedOrigin.h
+++ firefox-145.0/gfx/skia/skia/include/codec/SkEncodedOrigin.h
@@ -41,6 +41,7 @@ static inline SkMatrix SkEncodedOriginTo
         case  kLeftBottom_SkEncodedOrigin: return SkMatrix::MakeAll( 0,  1, 0, -1,  0, h, 0, 0, 1);
     }
     SK_ABORT("Unexpected origin");
+    SkUNREACHABLE;
 }
 
 /**
Index: firefox-145.0/gfx/skia/skia/include/private/gpu/ganesh/GrTypesPriv.h
===================================================================
--- firefox-145.0.orig/gfx/skia/skia/include/private/gpu/ganesh/GrTypesPriv.h
+++ firefox-145.0/gfx/skia/skia/include/private/gpu/ganesh/GrTypesPriv.h
@@ -301,6 +301,7 @@ static inline bool GrTextureTypeHasRestr
         default:
             SK_ABORT("Unexpected texture type");
     }
+    SkUNREACHABLE;
 }
 
 //////////////////////////////////////////////////////////////////////////////
Index: firefox-145.0/gfx/skia/skia/src/core/SkDescriptor.cpp
===================================================================
--- firefox-145.0.orig/gfx/skia/skia/src/core/SkDescriptor.cpp
+++ firefox-145.0/gfx/skia/skia/src/core/SkDescriptor.cpp
@@ -26,6 +26,7 @@ std::unique_ptr<SkDescriptor> SkDescript
 void SkDescriptor::operator delete(void* p) { ::operator delete(p); }
 void* SkDescriptor::operator new(size_t) {
     SK_ABORT("Descriptors are created with placement new.");
+    SkUNREACHABLE;
 }
 
 void SkDescriptor::flatten(SkWriteBuffer& buffer) const {
Index: firefox-145.0/gfx/skia/skia/src/core/SkGeometry.h
===================================================================
--- firefox-145.0.orig/gfx/skia/skia/src/core/SkGeometry.h
+++ firefox-145.0/gfx/skia/skia/src/core/SkGeometry.h
@@ -282,6 +282,7 @@ static inline bool SkCubicIsDegenerate(S
             return true;
     }
     SK_ABORT("Invalid SkCubicType");
+    SkUNREACHABLE;
 }
 
 static inline const char* SkCubicTypeName(SkCubicType type) {
@@ -294,6 +295,7 @@ static inline const char* SkCubicTypeNam
         case SkCubicType::kLineOrPoint: return "kLineOrPoint";
     }
     SK_ABORT("Invalid SkCubicType");
+    SkUNREACHABLE;
 }
 
 /** Returns the cubic classification.
Index: firefox-145.0/gfx/skia/skia/src/core/SkTextBlob.cpp
===================================================================
--- firefox-145.0.orig/gfx/skia/skia/src/core/SkTextBlob.cpp
+++ firefox-145.0/gfx/skia/skia/src/core/SkTextBlob.cpp
@@ -213,6 +213,7 @@ void SkTextBlob::operator delete(void* p
 
 void* SkTextBlob::operator new(size_t) {
     SK_ABORT("All blobs are created by placement new.");
+    SkUNREACHABLE;
 }
 
 void* SkTextBlob::operator new(size_t, void* p) {
Index: firefox-145.0/gfx/skia/skia/src/core/SkTypeface_remote.h
===================================================================
--- firefox-145.0.orig/gfx/skia/skia/src/core/SkTypeface_remote.h
+++ firefox-145.0/gfx/skia/skia/src/core/SkTypeface_remote.h
@@ -108,12 +108,14 @@ public:
     bool isLogging() const {return fIsLogging;}
 
 protected:
-    int onGetUPEM() const override { SK_ABORT("Should never be called."); }
+    int onGetUPEM() const override { SK_ABORT("Should never be called."); SkUNREACHABLE; }
     std::unique_ptr<SkStreamAsset> onOpenStream(int* ttcIndex) const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
     sk_sp<SkTypeface> onMakeClone(const SkFontArguments& args) const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
     bool onGlyphMaskNeedsCurrentColor() const override {
         return fGlyphMaskNeedsCurrentColor;
@@ -121,10 +123,12 @@ protected:
     int onGetVariationDesignPosition(SkFontArguments::VariationPosition::Coordinate coordinates[],
                                      int coordinateCount) const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
     int onGetVariationDesignParameters(SkFontParameters::Variation::Axis parameters[],
                                        int parameterCount) const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
     void onGetFamilyName(SkString* familyName) const override {
         // Used by SkStrikeCache::DumpMemoryStatistics.
@@ -132,15 +136,19 @@ protected:
     }
     bool onGetPostScriptName(SkString*) const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
     SkTypeface::LocalizedStrings* onCreateFamilyNameIterator() const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
     int onGetTableTags(SkFontTableTag tags[]) const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
     size_t onGetTableData(SkFontTableTag, size_t offset, size_t length, void* data) const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
     std::unique_ptr<SkScalerContext> onCreateScalerContext(
         const SkScalerContextEffects& effects, const SkDescriptor* desc) const override
@@ -154,20 +162,25 @@ protected:
     }
     void onGetFontDescriptor(SkFontDescriptor*, bool*) const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
     void getGlyphToUnicodeMap(SkUnichar*) const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
 
     void getPostScriptGlyphNames(SkString*) const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
 
     std::unique_ptr<SkAdvancedTypefaceMetrics> onGetAdvancedMetrics() const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
     void onCharsToGlyphs(const SkUnichar* chars, int count, SkGlyphID glyphs[]) const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
     int onCountGlyphs() const override {
         return this->glyphCount();
@@ -175,6 +188,7 @@ protected:
 
     void* onGetCTFontRef() const override {
         SK_ABORT("Should never be called.");
+        SkUNREACHABLE;
     }
 
 private:
Index: firefox-145.0/gfx/skia/skia/src/sksl/SkSLOperator.cpp
===================================================================
--- firefox-145.0.orig/gfx/skia/skia/src/sksl/SkSLOperator.cpp
+++ firefox-145.0/gfx/skia/skia/src/sksl/SkSLOperator.cpp
@@ -51,7 +51,7 @@ OperatorPrecedence Operator::getBinaryPr
         case Kind::BITWISEXOREQ: // fall through
         case Kind::BITWISEOREQ:  return OperatorPrecedence::kAssignment;
         case Kind::COMMA:        return OperatorPrecedence::kSequence;
-        default: SK_ABORT("unsupported binary operator");
+        default: SkUNREACHABLE;
     }
 }
 
Index: firefox-145.0/gfx/skia/skia/src/sksl/ir/SkSLType.h
===================================================================
--- firefox-145.0.orig/gfx/skia/skia/src/sksl/ir/SkSLType.h
+++ firefox-145.0/gfx/skia/skia/src/sksl/ir/SkSLType.h
@@ -468,6 +468,7 @@ public:
 
     virtual SkSpan<const Field> fields() const {
         SK_ABORT("Internal error: not a struct");
+        SkUNREACHABLE;
     }
 
     /**
Index: firefox-145.0/gfx/skia/skia/src/utils/SkShadowUtils.cpp
===================================================================
--- firefox-145.0.orig/gfx/skia/skia/src/utils/SkShadowUtils.cpp
+++ firefox-145.0/gfx/skia/skia/src/utils/SkShadowUtils.cpp
@@ -142,6 +142,7 @@ struct SpotVerticesFactory {
                 return true;
         }
         SK_ABORT("Uninitialized occluder type?");
+        SkUNREACHABLE;
     }
 
     sk_sp<SkVertices> makeVertices(const SkPath& path, const SkMatrix& ctm,
Index: firefox-145.0/intl/icu/source/i18n/number_rounding.cpp
===================================================================
--- firefox-145.0.orig/intl/icu/source/i18n/number_rounding.cpp
+++ firefox-145.0/intl/icu/source/i18n/number_rounding.cpp
@@ -287,6 +287,7 @@ FractionPrecision Precision::constructFr
     settings.fMaxFrac = static_cast<digits_t>(maxFrac);
     settings.fMinSig = -1;
     settings.fMaxSig = -1;
+    settings.fPriority = UNUM_ROUNDING_PRIORITY_RELAXED;
     PrecisionUnion union_{};
     union_.fracSig = settings;
     return {RND_FRACTION, union_};
@@ -298,6 +299,7 @@ Precision Precision::constructSignifican
     settings.fMaxFrac = -1;
     settings.fMinSig = static_cast<digits_t>(minSig);
     settings.fMaxSig = static_cast<digits_t>(maxSig);
+    settings.fPriority = UNUM_ROUNDING_PRIORITY_RELAXED;
     PrecisionUnion union_{};
     union_.fracSig = settings;
     return {RND_SIGNIFICANT, union_};
Index: firefox-145.0/js/src/irregexp/imported/regexp-parser.cc
===================================================================
--- firefox-145.0.orig/js/src/irregexp/imported/regexp-parser.cc
+++ firefox-145.0/js/src/irregexp/imported/regexp-parser.cc
@@ -2799,6 +2799,7 @@ bool MayContainStrings(ClassSetOperandTy
       if (operand->IsClassRanges()) return false;
       return operand->AsClassSetExpression()->may_contain_strings();
   }
+  UNREACHABLE();
 }
 
 }  // namespace
Index: firefox-145.0/third_party/libwebrtc/api/adaptation/resource.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/api/adaptation/resource.cc
+++ firefox-145.0/third_party/libwebrtc/api/adaptation/resource.cc
@@ -22,6 +22,7 @@ const char* ResourceUsageStateToString(R
       return "kUnderuse";
   }
   RTC_CHECK_NOTREACHED();
+  return nullptr;
 }
 
 ResourceListener::~ResourceListener() {}
Index: firefox-145.0/third_party/libwebrtc/api/rtp_parameters.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/api/rtp_parameters.cc
+++ firefox-145.0/third_party/libwebrtc/api/rtp_parameters.cc
@@ -37,6 +37,7 @@ const char* DegradationPreferenceToStrin
       return "balanced";
   }
   RTC_CHECK_NOTREACHED();
+  return "";
 }
 
 const double kDefaultBitratePriority = 1.0;
Index: firefox-145.0/third_party/libwebrtc/api/video/video_frame_buffer.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/api/video/video_frame_buffer.cc
+++ firefox-145.0/third_party/libwebrtc/api/video/video_frame_buffer.cc
@@ -114,6 +114,8 @@ const char* VideoFrameBufferTypeToString
     default:
       RTC_DCHECK_NOTREACHED();
   }
+  RTC_DCHECK_NOTREACHED();
+  return nullptr;
 }
 
 int I420BufferInterface::ChromaWidth() const {
Index: firefox-145.0/third_party/libwebrtc/api/video_codecs/video_codec.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/api/video_codecs/video_codec.cc
+++ firefox-145.0/third_party/libwebrtc/api/video_codecs/video_codec.cc
@@ -158,6 +158,7 @@ const char* CodecTypeToPayloadString(Vid
       return kPayloadNameH265;
   }
   RTC_CHECK_NOTREACHED();
+  return "";
 }
 
 VideoCodecType PayloadStringToCodecType(const std::string& name) {
Index: firefox-145.0/third_party/libwebrtc/api/video_codecs/video_encoder_software_fallback_wrapper.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/api/video_codecs/video_encoder_software_fallback_wrapper.cc
+++ firefox-145.0/third_party/libwebrtc/api/video_codecs/video_encoder_software_fallback_wrapper.cc
@@ -193,6 +193,7 @@ class VideoEncoderSoftwareFallbackWrappe
         return fallback_encoder_.get();
     }
     RTC_CHECK_NOTREACHED();
+    return nullptr;
   }
 
   // Updates encoder with last observed parameters, such as callbacks, rates,
@@ -391,6 +392,7 @@ int32_t VideoEncoderSoftwareFallbackWrap
       return fallback_encoder_->Encode(frame, frame_types);
   }
   RTC_CHECK_NOTREACHED();
+  return WEBRTC_VIDEO_CODEC_ERROR;
 }
 
 int32_t VideoEncoderSoftwareFallbackWrapper::EncodeWithMainEncoder(
Index: firefox-145.0/third_party/libwebrtc/call/adaptation/video_stream_adapter.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/call/adaptation/video_stream_adapter.cc
+++ firefox-145.0/third_party/libwebrtc/call/adaptation/video_stream_adapter.cc
@@ -177,6 +177,7 @@ const char* Adaptation::StatusToString(A
       return "kRejectedByConstraint";
   }
   RTC_CHECK_NOTREACHED();
+  return "";
 }
 
 Adaptation::Adaptation(int validation_id,
@@ -399,6 +400,7 @@ VideoStreamAdapter::RestrictionsOrState
       return Adaptation::Status::kAdaptationDisabled;
   }
   RTC_CHECK_NOTREACHED();
+  return Adaptation::Status::kAdaptationDisabled;
 }
 
 Adaptation VideoStreamAdapter::GetAdaptationDown() {
@@ -479,6 +481,7 @@ VideoStreamAdapter::GetAdaptationDownSte
       return Adaptation::Status::kAdaptationDisabled;
   }
   RTC_CHECK_NOTREACHED();
+  return Adaptation::Status::kAdaptationDisabled;
 }
 
 VideoStreamAdapter::RestrictionsOrState VideoStreamAdapter::DecreaseResolution(
@@ -632,6 +635,8 @@ Adaptation VideoStreamAdapter::GetAdaptD
     }
   }
   RTC_CHECK_NOTREACHED();
+  return RestrictionsOrStateToAdaptation(
+         Adaptation::Status::kAdaptationDisabled, input_state);
 }
 
 VideoStreamAdapter::RestrictionsOrState
Index: firefox-145.0/third_party/libwebrtc/call/rtp_payload_params.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/call/rtp_payload_params.cc
+++ firefox-145.0/third_party/libwebrtc/call/rtp_payload_params.cc
@@ -430,7 +430,8 @@ std::optional<FrameDependencyStructure>
     case VideoCodecType::kVideoCodecH265:
       return std::nullopt;
   }
-  RTC_DCHECK_NOTREACHED() << "Unsupported codec.";
+  //RTC_DCHECK_NOTREACHED() << "Unsupported codec.";
+  RTC_CHECK_NOTREACHED();
 }
 
 void RtpPayloadParams::GenericToGeneric(int64_t frame_id,
Index: firefox-145.0/third_party/libwebrtc/call/video_send_stream.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/call/video_send_stream.cc
+++ firefox-145.0/third_party/libwebrtc/call/video_send_stream.cc
@@ -34,6 +34,7 @@ const char* StreamTypeToString(VideoSend
       return "flexfec";
   }
   RTC_CHECK_NOTREACHED();
+  return "";
 }
 
 }  // namespace
Index: firefox-145.0/third_party/libwebrtc/media/base/codec_comparators.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/media/base/codec_comparators.cc
+++ firefox-145.0/third_party/libwebrtc/media/base/codec_comparators.cc
@@ -330,6 +330,7 @@ bool MatchesWithCodecRules(const Codec&
         return IsSameCodecSpecific(left_codec.name, left_codec.params,
                                    right_codec.name, right_codec.params);
     }
+    return false;
   };
 
   return matches_id && matches_type_specific();
Index: firefox-145.0/third_party/libwebrtc/modules/audio_processing/agc2/clipping_predictor.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/modules/audio_processing/agc2/clipping_predictor.cc
+++ firefox-145.0/third_party/libwebrtc/modules/audio_processing/agc2/clipping_predictor.cc
@@ -385,7 +385,7 @@ std::unique_ptr<ClippingPredictor> Creat
           config.reference_window_delay, config.clipping_threshold,
           /*adaptive_step_estimation=*/false);
   }
-  RTC_DCHECK_NOTREACHED();
+  RTC_CHECK_NOTREACHED();
 }
 
 }  // namespace webrtc
Index: firefox-145.0/third_party/libwebrtc/modules/audio_processing/agc2/input_volume_stats_reporter.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/modules/audio_processing/agc2/input_volume_stats_reporter.cc
+++ firefox-145.0/third_party/libwebrtc/modules/audio_processing/agc2/input_volume_stats_reporter.cc
@@ -48,6 +48,7 @@ constexpr absl::string_view MetricNamePr
     case InputVolumeType::kRecommended:
       return "WebRTC.Audio.Apm.RecommendedInputVolume.";
   }
+  RTC_CHECK_NOTREACHED();
 }
 
 metrics::Histogram* CreateVolumeHistogram(InputVolumeType input_volume_type) {
Index: firefox-145.0/third_party/libwebrtc/modules/audio_processing/agc2/rnn_vad/rnn_fc.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/modules/audio_processing/agc2/rnn_vad/rnn_fc.cc
+++ firefox-145.0/third_party/libwebrtc/modules/audio_processing/agc2/rnn_vad/rnn_fc.cc
@@ -65,6 +65,8 @@ FunctionView<float(float)> GetActivation
     case ActivationFunction::kSigmoidApproximated:
       return ::rnnoise::SigmoidApproximated;
   }
+  // supposed to be never reached apparently therefore returning bogus
+  return ::rnnoise::TansigApproximated;
 }
 
 }  // namespace
Index: firefox-145.0/third_party/libwebrtc/modules/audio_processing/audio_processing_impl.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/modules/audio_processing/audio_processing_impl.cc
+++ firefox-145.0/third_party/libwebrtc/modules/audio_processing/audio_processing_impl.cc
@@ -124,6 +124,7 @@ GainControl::Mode Agc1ConfigModeToInterf
       return GainControl::kFixedDigital;
   }
   RTC_CHECK_NOTREACHED();
+  return GainControl::kAdaptiveAnalog;
 }
 
 bool MinimizeProcessingForUnusedOutput(const FieldTrialsView& field_trials) {
@@ -191,7 +192,7 @@ int AudioFormatValidityToErrorCode(Audio
     case AudioFormatValidity::kInvalidChannelCount:
       return AudioProcessing::kBadNumberChannelsError;
   }
-  RTC_DCHECK(false);
+  RTC_CHECK_NOTREACHED();
 }
 
 // Returns an AudioProcessing::Error together with the best possible option for
@@ -2102,6 +2103,7 @@ void AudioProcessingImpl::InitializeNois
               return NsConfig::SuppressionLevel::k21dB;
           }
           RTC_CHECK_NOTREACHED();
+	  return NsConfig::SuppressionLevel::k6dB;
         };
 
     NsConfig cfg;
Index: firefox-145.0/third_party/libwebrtc/modules/congestion_controller/goog_cc/goog_cc_network_control.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/modules/congestion_controller/goog_cc/goog_cc_network_control.cc
+++ firefox-145.0/third_party/libwebrtc/modules/congestion_controller/goog_cc/goog_cc_network_control.cc
@@ -83,6 +83,8 @@ BandwidthLimitedCause GetBandwidthLimite
     case LossBasedState::kDelayBasedEstimate:
       return BandwidthLimitedCause::kDelayBasedLimited;
   }
+  // just return something by default
+  return BandwidthLimitedCause::kLossLimitedBwe;
 }
 
 }  // namespace
Index: firefox-145.0/third_party/libwebrtc/modules/desktop_capture/linux/wayland/screencast_portal.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/modules/desktop_capture/linux/wayland/screencast_portal.cc
+++ firefox-145.0/third_party/libwebrtc/modules/desktop_capture/linux/wayland/screencast_portal.cc
@@ -56,6 +56,7 @@ ScreenCastPortal::CaptureSourceType Scre
     case CaptureType::kAnyScreenContent:
       return ScreenCastPortal::CaptureSourceType::kAnyScreenContent;
   }
+  RTC_CHECK_NOTREACHED();
 }
 
 ScreenCastPortal::ScreenCastPortal(CaptureType type, PortalNotifier* notifier)
Index: firefox-145.0/third_party/libwebrtc/modules/pacing/bitrate_prober.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/modules/pacing/bitrate_prober.cc
+++ firefox-145.0/third_party/libwebrtc/modules/pacing/bitrate_prober.cc
@@ -91,6 +91,7 @@ bool BitrateProber::ReadyToSetActiveStat
       return packet_size >=
              std::min(RecommendedMinProbeSize(), config_.min_packet_size.Get());
   }
+  RTC_CHECK_NOTREACHED();
 }
 
 void BitrateProber::OnIncomingPacket(DataSize packet_size) {
Index: firefox-145.0/third_party/libwebrtc/modules/rtp_rtcp/source/create_video_rtp_depacketizer.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/modules/rtp_rtcp/source/create_video_rtp_depacketizer.cc
+++ firefox-145.0/third_party/libwebrtc/modules/rtp_rtcp/source/create_video_rtp_depacketizer.cc
@@ -47,6 +47,7 @@ std::unique_ptr<VideoRtpDepacketizer> Cr
       return std::make_unique<VideoRtpDepacketizerGeneric>();
   }
   RTC_CHECK_NOTREACHED();
+  return nullptr;
 }
 
 }  // namespace webrtc
Index: firefox-145.0/third_party/libwebrtc/modules/rtp_rtcp/source/rtcp_packet/congestion_control_feedback.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/modules/rtp_rtcp/source/rtcp_packet/congestion_control_feedback.cc
+++ firefox-145.0/third_party/libwebrtc/modules/rtp_rtcp/source/rtcp_packet/congestion_control_feedback.cc
@@ -113,6 +113,7 @@ uint16_t To2BitEcn(EcnMarking ecn_markin
     case EcnMarking::kCe:
       return kEcnCe << 13;
   }
+  return 0; // should not be reached
 }
 
 EcnMarking ToEcnMarking(uint16_t receive_info) {
Index: firefox-145.0/third_party/libwebrtc/modules/rtp_rtcp/source/rtp_sender.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/modules/rtp_rtcp/source/rtp_sender.cc
+++ firefox-145.0/third_party/libwebrtc/modules/rtp_rtcp/source/rtp_sender.cc
@@ -149,6 +149,7 @@ bool IsNonVolatile(RTPExtensionType type
 #endif
   }
   RTC_CHECK_NOTREACHED();
+  return false;
 }
 
 bool HasBweExtension(const RtpHeaderExtensionMap& extensions_map) {
Index: firefox-145.0/third_party/libwebrtc/modules/video_coding/codecs/vp8/default_temporal_layers.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/modules/video_coding/codecs/vp8/default_temporal_layers.cc
+++ firefox-145.0/third_party/libwebrtc/modules/video_coding/codecs/vp8/default_temporal_layers.cc
@@ -120,6 +120,7 @@ size_t BufferToIndex(Vp8BufferReference
     case Vp8FrameConfig::Vp8BufferReference::kNone:
       RTC_CHECK_NOTREACHED();
   }
+  RTC_CHECK_NOTREACHED();
 }
 
 }  // namespace
Index: firefox-145.0/third_party/libwebrtc/modules/video_coding/codecs/vp8/temporal_layers_checker.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/modules/video_coding/codecs/vp8/temporal_layers_checker.cc
+++ firefox-145.0/third_party/libwebrtc/modules/video_coding/codecs/vp8/temporal_layers_checker.cc
@@ -34,6 +34,7 @@ TemporalLayersChecker::CreateTemporalLay
       return std::make_unique<TemporalLayersChecker>(num_temporal_layers);
   }
   RTC_CHECK_NOTREACHED();
+  return nullptr;
 }
 
 TemporalLayersChecker::TemporalLayersChecker(int num_temporal_layers)
Index: firefox-145.0/third_party/libwebrtc/modules/video_coding/h26x_packet_buffer.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/modules/video_coding/h26x_packet_buffer.cc
+++ firefox-145.0/third_party/libwebrtc/modules/video_coding/h26x_packet_buffer.cc
@@ -67,6 +67,7 @@ bool BeginningOfIdr(const H26xPacketBuff
       return contains_idr_nalu && IsFirstPacketOfFragment(h264_header);
     }
   }
+  return false;
 }
 
 bool HasSps(const H26xPacketBuffer::Packet& packet) {
Index: firefox-145.0/third_party/libwebrtc/net/dcsctp/packet/parameter/reconfiguration_response_parameter.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/net/dcsctp/packet/parameter/reconfiguration_response_parameter.cc
+++ firefox-145.0/third_party/libwebrtc/net/dcsctp/packet/parameter/reconfiguration_response_parameter.cc
@@ -61,6 +61,7 @@ absl::string_view ToString(Reconfigurati
     case ReconfigurationResponseParameter::Result::kInProgress:
       return "In progress";
   }
+  return "Should never be reached";
 }
 
 std::optional<ReconfigurationResponseParameter>
Index: firefox-145.0/third_party/libwebrtc/net/dcsctp/public/dcsctp_handover_state.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/net/dcsctp/public/dcsctp_handover_state.cc
+++ firefox-145.0/third_party/libwebrtc/net/dcsctp/public/dcsctp_handover_state.cc
@@ -43,6 +43,7 @@ constexpr absl::string_view HandoverUnre
     case HandoverUnreadinessReason::kPendingStreamResetRequest:
       return "PENDING_STREAM_RESET_REQUEST";
   }
+  return "NOTREACHED";
 }
 }  // namespace
 
Index: firefox-145.0/third_party/libwebrtc/net/dcsctp/rx/data_tracker.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/net/dcsctp/rx/data_tracker.cc
+++ firefox-145.0/third_party/libwebrtc/net/dcsctp/rx/data_tracker.cc
@@ -359,6 +359,7 @@ absl::string_view DataTracker::ToString(
     case AckState::kImmediate:
       return "IMMEDIATE";
   }
+  return "NOTREACHED";
 }
 
 HandoverReadinessStatus DataTracker::GetHandoverReadiness() const {
Index: firefox-145.0/third_party/libwebrtc/net/dcsctp/socket/dcsctp_socket.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/net/dcsctp/socket/dcsctp_socket.cc
+++ firefox-145.0/third_party/libwebrtc/net/dcsctp/socket/dcsctp_socket.cc
@@ -255,6 +255,7 @@ bool DcSctpSocket::IsConsistent() const
       return (tcb_ != nullptr && !t1_init_->is_running() &&
               !t1_cookie_->is_running() && t2_shutdown_->is_running());
   }
+  return false; // notreached
 }
 
 constexpr absl::string_view DcSctpSocket::ToString(DcSctpSocket::State state) {
@@ -276,6 +277,7 @@ constexpr absl::string_view DcSctpSocket
     case DcSctpSocket::State::kShutdownAckSent:
       return "SHUTDOWN_ACK_SENT";
   }
+  return "NOTREACHED";
 }
 
 void DcSctpSocket::SetState(State state, absl::string_view reason) {
@@ -594,6 +596,7 @@ SocketState DcSctpSocket::state() const
     case State::kShutdownAckSent:
       return SocketState::kShuttingDown;
   }
+  return SocketState::kShuttingDown; // notreached
 }
 
 void DcSctpSocket::SetMaxMessageSize(size_t max_message_size) {
Index: firefox-145.0/third_party/libwebrtc/net/dcsctp/socket/packet_sender.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/net/dcsctp/socket/packet_sender.cc
+++ firefox-145.0/third_party/libwebrtc/net/dcsctp/socket/packet_sender.cc
@@ -44,5 +44,6 @@ bool PacketSender::Send(SctpPacket::Buil
       return false;
     }
   }
+  return false; // not reached
 }
 }  // namespace dcsctp
Index: firefox-145.0/third_party/libwebrtc/net/dcsctp/timer/timer.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/net/dcsctp/timer/timer.cc
+++ firefox-145.0/third_party/libwebrtc/net/dcsctp/timer/timer.cc
@@ -49,6 +49,7 @@ TimeDelta GetBackoffDuration(const Timer
       return TimeDelta(std::min(duration, Timer::kMaxTimerDuration));
     }
   }
+  return base_duration; // fake nonreached
 }
 }  // namespace
 
Index: firefox-145.0/third_party/libwebrtc/video/adaptation/video_stream_encoder_resource_manager.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/video/adaptation/video_stream_encoder_resource_manager.cc
+++ firefox-145.0/third_party/libwebrtc/video/adaptation/video_stream_encoder_resource_manager.cc
@@ -91,6 +91,7 @@ std::string ToString(VideoAdaptationReas
       return "cpu";
   }
   RTC_CHECK_NOTREACHED();
+  return "";
 }
 
 std::vector<bool> GetActiveLayersFlags(const VideoCodec& codec) {
Index: firefox-145.0/third_party/libwebrtc/video/config/encoder_stream_factory.cc
===================================================================
--- firefox-145.0.orig/third_party/libwebrtc/video/config/encoder_stream_factory.cc
+++ firefox-145.0/third_party/libwebrtc/video/config/encoder_stream_factory.cc
@@ -118,6 +118,7 @@ int GetDefaultMaxQp(VideoCodecType codec
     case kVideoCodecAV1:
       return kDefaultVideoMaxQpAv1;
   }
+  return kDefaultVideoMaxQpAv1;
 }
 
 // Round size to nearest simulcast-friendly size.
