From 010f6e617eb8d7b1ff7e9fe5f27953e3f8b58cb4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Klaus=20K=C3=A4mpf?= <kkaempf@gmail.com>
Date: Tue, 5 Nov 2024 16:53:28 +0100
Subject: [PATCH 3/6] fix bashisms

---
 provider-register.sh | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/provider-register.sh b/provider-register.sh
index cd44462..0b60a9a 100755
--- a/provider-register.sh
+++ b/provider-register.sh
@@ -146,12 +146,12 @@ EOFP
     done
 
 #produce Capabilities
-    let serial=0
+    serial=0
     for rf in $regfiles
     do
       cat $rf | grep -v '^[[:space:]]*#.*' | while read CLASSNAME NAMESPACE PROVIDERNAME PROVIDERMODULE CAPS
       do
-	let serial=serial+1
+	serial=$((serial + 1))
 	numcap=
 	for cap in $CAPS
 	do
@@ -792,14 +792,14 @@ mofs=$*
 if [ "$help" = "1" ]
 then
     usage
-    echo -e "\t-h display help message"
-    echo -e "\t-v verbose mode"
-    echo -e "\t-d deregister provider and uninstall schema"
-    echo -e "\t-t specify cimserver type (pegasus|sfcb|openwbem|sniacimom)"
-    echo -e "\t-r specify registration files"
-    echo -e "\t-m specify schema mof files"
-    echo -e "\t-X create repository for alternate platform (sfcb only at the moment)."
-    echo -e "\t-n target namespace definition (default: root/cimv2)."
+    echo "	-h display help message"
+    echo "	-v verbose mode"
+    echo "	-d deregister provider and uninstall schema"
+    echo "	-t specify cimserver type (pegasus|sfcb|openwbem|sniacimom)"
+    echo "	-r specify registration files"
+    echo "	-m specify schema mof files"
+    echo "	-X create repository for alternate platform (sfcb only at the moment)."
+    echo "	-n target namespace definition (default: root/cimv2)."
     echo
     echo Use this command to install schema mofs and register providers.
     echo CIM Server Type is required as well as at least one registration file and one mof.
-- 
2.47.0

