--- Imakefile
+++ Imakefile	2001/06/20 00:17:51
@@ -4,10 +4,10 @@
   SYS_LIBRARIES = -lm
     CDEBUGFLAGS = -g
 
-           SRCS = xfish.c medcut.c read.c gifread.c
-           OBJS = xfish.o medcut.o read.o gifread.o
+           SRCS = xfish.c medcut.c read.c
+           OBJS = xfish.o medcut.o read.o
 
-ComplexProgramTarget(xfishtank)
+ComplexProgramTargetNoMan(xfishtank)
 
 xfishy.h: makeh FishList
 	./makeh > xfishy.h
--- makeh.c
+++ makeh.c	2001/06/20 00:17:30
@@ -1,5 +1,7 @@
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 
 main()
 {
--- read.c
+++ read.c	2001/06/20 00:17:30
@@ -1,5 +1,6 @@
 #include <X11/Xlib.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include "compact.h"
 #include "medcut.h"
 
@@ -676,12 +677,12 @@
 			return(bit_data);
 		}
 		rewind(fp);
-		bit_data = ReadGifBitmap(fp, w, h, colrs);
+/*		bit_data = ReadGifBitmap(fp, w, h, colrs);
 		if (bit_data != NULL)
 		{
 			return(bit_data);
 		}
-		rewind(fp);
+		rewind(fp); */
 		bit_data = ReadXbmBitmap(fp, w, h, colrs);
 		if (bit_data != NULL)
 		{
--- xfish.c
+++ xfish.c	2001/06/20 00:17:30
@@ -41,6 +41,7 @@
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <malloc.h>
+#include <stdlib.h>
 
 #include "vroot.h"
 #include "xfishy.h"
