Index: sphinxcontrib-htmlhelp-2.0.0/tests/test_htmlhelp.py
===================================================================
--- sphinxcontrib-htmlhelp-2.0.0.orig/tests/test_htmlhelp.py
+++ sphinxcontrib-htmlhelp-2.0.0/tests/test_htmlhelp.py
@@ -21,7 +21,7 @@ from sphinx.config import Config
 def test_build_htmlhelp(app, status, warning):
     app.build()
 
-    hhp = (app.outdir / 'pythondoc.hhp').text()
+    hhp = (app.outdir / 'pythondoc.hhp').read_text()
     assert 'Compiled file=pythondoc.chm' in hhp
     assert 'Contents file=pythondoc.hhc' in hhp
     assert 'Default Window=pythondoc' in hhp
@@ -84,7 +84,7 @@ def test_htmlhelp_hhc(app):
         assert node[1].attrib == {'name': 'Local', 'value': filename}
 
     # .hhc file
-    hhc = (app.outdir / 'pythondoc.hhc').text()
+    hhc = (app.outdir / 'pythondoc.hhc').read_text()
     tree = HTMLParser(namespaceHTMLElements=False).parse(hhc)
     items = tree.find('.//body/ul')
     assert len(items) == 4
