diff --git a/src/makefile b/src/makefile
index 494baab..fb56ba4 100644
--- a/src/makefile
+++ b/src/makefile
@@ -169,9 +169,9 @@ DEF_linux=-DLUASOCKET_$(DEBUG) \
 	-DLUASOCKET_API='__attribute__((visibility("default")))' \
 	-DUNIX_API='__attribute__((visibility("default")))' \
 	-DMIME_API='__attribute__((visibility("default")))'
-CFLAGS_linux= -I$(LUAINC) $(DEF) -Wall -Wshadow -Wextra \
-	-Wimplicit -O2 -ggdb3 -fpic -fvisibility=hidden
-LDFLAGS_linux=-O -shared -fpic -o 
+CFLAGS_linux= -I$(LUAINC) $(DEF) $(OPTFLAGS) -Wall -Wshadow -Wextra \
+	-Wimplicit -fpic -fvisibility=hidden
+LDFLAGS_linux=-O $(OPTFLAGS) -shared -fpic -o 
 LD_linux=gcc
 SOCKET_linux=usocket.o
 
