Index: rdiff-backup-1.2.8/_librsyncmodule.c
===================================================================
--- rdiff-backup-1.2.8.orig/_librsyncmodule.c
+++ rdiff-backup-1.2.8/_librsyncmodule.c
@@ -59,8 +59,14 @@ _librsync_new_sigmaker(PyObject* self, P
   if (sm == NULL) return NULL;
   sm->x_attr = NULL;
 
+#ifdef RS_DEFAULT_STRONG_LEN /* librsync < 1.0.0 */
   sm->sig_job = rs_sig_begin((size_t)blocklen,
 							 (size_t)RS_DEFAULT_STRONG_LEN);
+#else /* librsync >= 1.0.0 */
+  sm->sig_job = rs_sig_begin((size_t)blocklen,
+                             (size_t)8, RS_MD4_SIG_MAGIC);
+#endif
+	  
   return (PyObject*)sm;
 }
 
