diff -ru scim-tomoe-0.6.0//src/scim_tomoe_prefs_widget.cpp scim-tomoe-0.6.0.new//src/scim_tomoe_prefs_widget.cpp
--- scim-tomoe-0.6.0//src/scim_tomoe_prefs_widget.cpp	2007-05-07 09:49:15.000000000 +0200
+++ scim-tomoe-0.6.0.new//src/scim_tomoe_prefs_widget.cpp	2008-08-11 19:10:38.000000000 +0200
@@ -20,7 +20,7 @@
 #ifdef HAVE_CONFIG_H
   #include <config.h>
 #endif
-
+#include <string.h>
 #include <glib/gi18n-lib.h>
 
 #include "scim_tomoe_prefs_widget.h"
