diff --git a/CMakeLists.txt b/CMakeLists.txt
index 00dff0d..52ea70e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -229,11 +229,6 @@ endif()
 
 target_link_libraries(redumper ${libs})
 
-if(REDUMPER_TARGET_LINUX)
-    set(REDUMPER_LINKER_FLAGS "-static" CACHE STRING "Linker flags")
-    target_link_options(redumper PRIVATE "${REDUMPER_LINKER_FLAGS}")
-endif()
-
 install(TARGETS redumper DESTINATION "bin")
 
 # bundle LLVM libc++ for C++20 support (system libc++ is too old, static linking not allowed on macOS)
