diff --git a/src/antTask/edu/umd/cs/findbugs/anttask/AbstractFindBugsTask.java b/src/antTask/edu/umd/cs/findbugs/anttask/AbstractFindBugsTask.java
index 70af8ef..4919d09 100644
--- a/src/antTask/edu/umd/cs/findbugs/anttask/AbstractFindBugsTask.java
+++ b/src/antTask/edu/umd/cs/findbugs/anttask/AbstractFindBugsTask.java
@@ -28,6 +28,7 @@ import org.apache.tools.ant.BuildException;
 import org.apache.tools.ant.Task;
 import org.apache.tools.ant.taskdefs.Java;
 import org.apache.tools.ant.types.Path;
+import org.apache.tools.ant.types.FileSet;
 import org.apache.tools.ant.types.Reference;
 
 /**
@@ -311,7 +312,11 @@ public abstract class AbstractFindBugsTask extends Task {
             File findbugsLibFindBugs = new File(findbugsLib, "findbugs.jar");
             // log("executing using home dir [" + homeDir + "]");
             if (findbugsLibFindBugs.exists()) {
-                findbugsEngine.setClasspath(new Path(getProject(), findbugsLibFindBugs.getPath()));
+                FileSet findbugsLibFileSet = new FileSet();
+                findbugsLibFileSet.setProject(getProject());
+                findbugsLibFileSet.setDir(findbugsLib);
+                findbugsLibFileSet.setIncludes("*.jar");
+                findbugsEngine.createClasspath().addFileset(findbugsLibFileSet);
             } else {
                 throw new IllegalArgumentException("Can't find findbugs.jar in " + findbugsLib);
             }
