Index: i2pd-2.32.1/contrib/apparmor/usr.sbin.i2pd
===================================================================
--- i2pd-2.32.1.orig/contrib/apparmor/usr.sbin.i2pd
+++ i2pd-2.32.1/contrib/apparmor/usr.sbin.i2pd
@@ -18,9 +18,7 @@
   /etc/hosts r,
   /etc/nsswitch.conf r,
   /etc/resolv.conf r,
-  /run/resolvconf/resolv.conf r,
   /run/systemd/resolve/resolv.conf r,
-  /run/systemd/resolve/stub-resolv.conf r,
 
   # path specific (feel free to modify if you have another paths)
   /etc/i2pd/** r,
