Index: node-git.96a986d675/test/sequential/test-buffer-creation-regression.js
===================================================================
--- node-git.96a986d675.orig/test/sequential/test-buffer-creation-regression.js
+++ node-git.96a986d675/test/sequential/test-buffer-creation-regression.js
@@ -2,6 +2,7 @@
 
 const common = require('../common');
 const assert = require('assert');
+const os = require('os');
 
 function test(arrayBuffer, offset, length) {
   const uint8Array = new Uint8Array(arrayBuffer, offset, length);
@@ -25,6 +26,10 @@ const offset = 4294967296; /* 1 << 32 */
 const size = offset + length;
 let arrayBuffer;
 
+if (os.totalmem()/(1<<30) < 7) {
+  common.skip('Skipping test on low-memory build machine');
+}
+
 try {
   arrayBuffer = new ArrayBuffer(size);
 } catch (e) {
