Index: VirtualBox-6.1.12/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.cpp
===================================================================
--- VirtualBox-6.1.12.orig/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.cpp
+++ VirtualBox-6.1.12/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.cpp
@@ -18,6 +18,7 @@
 /* Qt includes: */
 #include <QApplication>
 #include <QLabel>
+#include <QPainterPath>
 #include <QPaintEvent>
 #include <QStyle>
 #include <QVBoxLayout>
Index: VirtualBox-6.1.12/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.h
===================================================================
--- VirtualBox-6.1.12.orig/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.h
+++ VirtualBox-6.1.12/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.h
@@ -37,7 +37,6 @@ class QIcon;
 class QLabel;
 class QMouseEvent;
 class QObject;
-class QPainterPath;
 class QPaintEvent;
 class QResizeEvent;
 class QString;
