Index: VirtualBox-7.0.4/src/VBox/Additions/common/VBoxGuest/VBoxGuest-haiku.c
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Additions/common/VBoxGuest/VBoxGuest-haiku.c
+++ VirtualBox-7.0.4/src/VBox/Additions/common/VBoxGuest/VBoxGuest-haiku.c
@@ -140,7 +140,7 @@ static struct vboxguest_module_info g_VB
     RTMemAllocExTag,
     RTMemContAlloc,
     RTMemContFree,
-    RTMemFreeEx,
+    RTMemFreeExG,
     RTMpIsCpuPossible,
     RTMpNotificationDeregister,
     RTMpNotificationRegister,
