Index: VirtualBox-7.0.4/src/VBox/Frontends/VirtualBox/src/precomp_vcc.h
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Frontends/VirtualBox/src/precomp_vcc.h
+++ VirtualBox-7.0.4/src/VBox/Frontends/VirtualBox/src/precomp_vcc.h
@@ -97,6 +97,7 @@
 #include <QtWidgets/qgraphicsitem.h>
 #include <QtWidgets/qgraphicslayoutitem.h>
 #include <QtWidgets/qgraphicswidget.h>
+#include <QtWidgets/qbuttongroup.h>
 #include <QtCore/QMetaType>
 #include <QtGui/qevent.h>
 #include <QtGui/qtouchdevice.h>
Index: VirtualBox-7.0.4/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp
+++ VirtualBox-7.0.4/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp
@@ -27,6 +27,7 @@
 
 /* Qt includes: */
 #include <QVBoxLayout>
+#include <QButtonGroup>
 
 /* GUI includes: */
 #include "UIGlobalSettingsProxy.h"
