Index: VirtualBox-7.0.18/src/libs/libxml2-2.12.6/include/vboxconfig.h
===================================================================
--- VirtualBox-7.0.18.orig/src/libs/libxml2-2.12.6/include/vboxconfig.h
+++ VirtualBox-7.0.18/src/libs/libxml2-2.12.6/include/vboxconfig.h
@@ -387,7 +387,7 @@
 #define PACKAGE_TARNAME ""
 
 /* Define to the version of this package. */
-#define PACKAGE_VERSION "2.6.30"
+#define PACKAGE_VERSION "2.12.6"
 
 /* Define to 1 if the C compiler supports function prototypes. */
 #define PROTOTYPES 1
@@ -399,7 +399,7 @@
 #undef SUPPORT_IP6
 
 /* Version number of package */
-#define VERSION "2.6.30"
+#define VERSION "2.12.6"
 
 /* Determine what socket length (socklen_t) data type is */
 #if defined(RT_OS_WINDOWS) || defined(RT_OS_OS2)
