Index: VirtualBox-7.0.4/Makefile.kmk
===================================================================
--- VirtualBox-7.0.4.orig/Makefile.kmk
+++ VirtualBox-7.0.4/Makefile.kmk
@@ -823,7 +823,6 @@ VBOX_CORE_DOXYFILE_INPUT_DIRS = \
 	src/VBox/Additions/x11/vboxmouse \
 	src/VBox/Additions/x11/vboxmouse/xorg70 \
 	src/VBox/Additions/x11/vboxmouse/xorg71 \
-	src/VBox/Additions/x11/vboxvideo \
 	src/VBox/NetworkServices \
 	src/VBox/NetworkServices/Dhcpd \
 	src/VBox/NetworkServices/NAT \
Index: VirtualBox-7.0.4/src/VBox/Additions/x11/vboxvideo/Makefile.kmk
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Additions/x11/vboxvideo/Makefile.kmk
+++ VirtualBox-7.0.4/src/VBox/Additions/x11/vboxvideo/Makefile.kmk
@@ -407,7 +407,8 @@ vboxvideo_drv_118_INCS += $(PATH_ROOT)/s
 vboxvideo_drv_118_SOURCES := $(vboxvideo_drv_17_SOURCES)
 vboxvideo_drv_118_LIBS += $(vboxvideo_drv_70_LIBS)
 
-ifdef VBOX_USE_SYSTEM_XORG_HEADERS
+# Fix to never build
+ifdef XX_VBOX_USE_SYSTEM_XORG_HEADERS
  # Build using local X.Org headers.  We assume X.Org Server 1.7 or later.
  DLLS    := $(filter-out vboxvideo_drv_%,$(DLLS)) vboxvideo_drv_system
  SYSMODS := $(filter-out vboxvideo_drv%,$(SYSMODS))
Index: VirtualBox-7.0.4/src/VBox/Additions/x11/Makefile.kmk
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Additions/x11/Makefile.kmk
+++ VirtualBox-7.0.4/src/VBox/Additions/x11/Makefile.kmk
@@ -32,7 +32,7 @@ include $(KBUILD_PATH)/subheader.kmk
 if1of ($(KBUILD_TARGET), freebsd linux netbsd openbsd solaris)
  include $(PATH_SUB_CURRENT)/VBoxClient/Makefile.kmk
  ifndef VBOX_NO_LEGACY_XORG_X11
-  include $(PATH_SUB_CURRENT)/vboxvideo/Makefile.kmk
+#  include $(PATH_SUB_CURRENT)/vboxvideo/Makefile.kmk
   ifn1of ($(KBUILD_TARGET), netbsd solaris)
    include $(PATH_SUB_CURRENT)/vboxmouse/Makefile.kmk
   endif
