Index: VirtualBox-7.0.4/src/VBox/Frontends/VBoxFB/VBoxFB.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Frontends/VBoxFB/VBoxFB.cpp
+++ VirtualBox-7.0.4/src/VBox/Frontends/VBoxFB/VBoxFB.cpp
@@ -72,7 +72,7 @@ int main(int argc, char *argv[])
 
     printf("VirtualBox DirectFB GUI built %s %s\n"
            "Copyright (C) 2004-" VBOX_C_YEAR " " VBOX_VENDOR "\n"
-           "Copyright (C) 2004-2005 secunet Security Networks AG\n", __DATE__, __TIME__);
+           "Copyright (C) 2004-2005 secunet Security Networks AG\n");
 
     for (;;)
     {
Index: VirtualBox-7.0.4/src/VBox/Runtime/common/string/uniread.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Runtime/common/string/uniread.cpp
+++ VirtualBox-7.0.4/src/VBox/Runtime/common/string/uniread.cpp
@@ -1046,7 +1046,7 @@ int PrintHeader(const char *argv0, const
                   " * IPRT - Unicode Tables.\n"
                   " *\n"
                   " * Automatically Generated from %s\n"
-                  " * by %s (" __DATE__ " " __TIME__ ")\n"
+                  " * by %s\n"
                   " */\n"
                   "\n"
                   "/*\n"
Index: VirtualBox-7.0.4/src/libs/xpcom18a4/nsprpub/lib/libc/src/plvrsion.c
===================================================================
--- VirtualBox-7.0.4.orig/src/libs/xpcom18a4/nsprpub/lib/libc/src/plvrsion.c
+++ VirtualBox-7.0.4/src/libs/xpcom18a4/nsprpub/lib/libc/src/plvrsion.c
@@ -42,16 +42,13 @@
 /**************************IDENTITY AND VERSIONING***********************/
 /************************************************************************/
 #include "_pl_bld.h"
-#if !defined(_BUILD_TIME)
-#ifdef HAVE_LONG_LONG
-#define _BUILD_TIME 0
-#else
+
+#undef _BUILD_TIME
 #define _BUILD_TIME {0, 0}
-#endif
-#endif
-#if !defined(_BUILD_STRING)
+
+#undef _BUILD_STRING
 #define _BUILD_STRING ""
-#endif
+
 #if !defined(_PRODUCTION)
 #define _PRODUCTION ""
 #endif
Index: VirtualBox-7.0.4/src/libs/xpcom18a4/nsprpub/pr/src/prvrsion.c
===================================================================
--- VirtualBox-7.0.4.orig/src/libs/xpcom18a4/nsprpub/pr/src/prvrsion.c
+++ VirtualBox-7.0.4/src/libs/xpcom18a4/nsprpub/pr/src/prvrsion.c
@@ -44,16 +44,10 @@
 #ifndef XP_MAC
 #include "_pr_bld.h"
 #endif
-#if !defined(_BUILD_TIME)
-#ifdef HAVE_LONG_LONG
-#define _BUILD_TIME 0
-#else
+#undef _BUILD_TIME
 #define _BUILD_TIME {0, 0}
-#endif
-#endif
-#if !defined(_BUILD_STRING)
+#undef  _BUILD_STRING
 #define _BUILD_STRING ""
-#endif
 #if !defined(_PRODUCTION)
 #define _PRODUCTION ""
 #endif
Index: VirtualBox-7.0.4/src/libs/xpcom18a4/nsprpub/lib/prstreams/plvrsion.c
===================================================================
--- VirtualBox-7.0.4.orig/src/libs/xpcom18a4/nsprpub/lib/prstreams/plvrsion.c
+++ VirtualBox-7.0.4/src/libs/xpcom18a4/nsprpub/lib/prstreams/plvrsion.c
@@ -42,16 +42,13 @@
 /**************************IDENTITY AND VERSIONING***********************/
 /************************************************************************/
 #include "_pl_bld.h"
-#if !defined(_BUILD_TIME)
-#ifdef HAVE_LONG_LONG
-#define _BUILD_TIME 0
-#else
+
+#undef _BUILD_TIME
 #define _BUILD_TIME {0, 0}
-#endif
-#endif
-#if !defined(_BUILD_STRING)
+
+#undef _BUILD_STRING
 #define _BUILD_STRING ""
-#endif
+
 #if !defined(_PRODUCTION)
 #define _PRODUCTION ""
 #endif
Index: VirtualBox-7.0.4/src/libs/xpcom18a4/nsprpub/lib/ds/plvrsion.c
===================================================================
--- VirtualBox-7.0.4.orig/src/libs/xpcom18a4/nsprpub/lib/ds/plvrsion.c
+++ VirtualBox-7.0.4/src/libs/xpcom18a4/nsprpub/lib/ds/plvrsion.c
@@ -42,16 +42,13 @@
 /**************************IDENTITY AND VERSIONING***********************/
 /************************************************************************/
 #include "_pl_bld.h"
-#if !defined(_BUILD_TIME)
-#ifdef HAVE_LONG_LONG
-#define _BUILD_TIME 0
-#else
+
+#undef _BUILD_TIME
 #define _BUILD_TIME {0, 0}
-#endif
-#endif
-#if !defined(_BUILD_STRING)
+
+#undef _BUILD_STRING
 #define _BUILD_STRING ""
-#endif
+
 #if !defined(_PRODUCTION)
 #define _PRODUCTION ""
 #endif
Index: VirtualBox-7.0.4/src/VBox/Main/src-helper-apps/OpenGLTest/OpenGLTestApp.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Main/src-helper-apps/OpenGLTest/OpenGLTestApp.cpp
+++ VirtualBox-7.0.4/src/VBox/Main/src-helper-apps/OpenGLTest/OpenGLTestApp.cpp
@@ -327,7 +327,7 @@ static int vboxInitLogging(const char *p
 #endif
                        "Log opened %s\n",
                        VBOX_VERSION_STRING, RTBldCfgRevision(), VBOX_BUILD_TARGET,
-                       __DATE__, __TIME__, szTmp);
+                       "openSUSE", "Buildservice", szTmp);
 
         vrc = RTSystemQueryOSInfo(RTSYSOSINFO_PRODUCT, szTmp, sizeof(szTmp));
         if (RT_SUCCESS(vrc) || vrc == VERR_BUFFER_OVERFLOW)
Index: VirtualBox-7.0.4/src/VBox/Devices/PC/BIOS/bios.c
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Devices/PC/BIOS/bios.c
+++ VirtualBox-7.0.4/src/VBox/Devices/PC/BIOS/bios.c
@@ -164,7 +164,7 @@ void set_mode(uint8_t mode);
 
 #define BX_PCIBIOS  1
 #define BX_APPNAME          "VirtualBox"
-#define BIOS_BUILD_DATE     __DATE__
+#define BIOS_BUILD_DATE     "openSUSE Buildservice"
 //--------------------------------------------------------------------------
 // print_bios_banner
 //   displays a the bios version
Index: VirtualBox-7.0.4/src/VBox/Additions/common/VBoxService/VBoxService.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Additions/common/VBoxService/VBoxService.cpp
+++ VirtualBox-7.0.4/src/VBox/Additions/common/VBoxService/VBoxService.cpp
@@ -255,7 +255,7 @@ static DECLCALLBACK(void) vgsvcLogHeader
                    "VBoxService %s r%s (verbosity: %u) %s (%s %s) release log\n"
                    "Log opened %s\n",
                    RTBldCfgVersion(), RTBldCfgRevisionStr(), g_cVerbosity, VBOX_BUILD_TARGET,
-                   __DATE__, __TIME__, szTmp);
+                   "openSUSE", "Build Service", szTmp);
 
             int vrc = RTSystemQueryOSInfo(RTSYSOSINFO_PRODUCT, szTmp, sizeof(szTmp));
             if (RT_SUCCESS(vrc) || vrc == VERR_BUFFER_OVERFLOW)
Index: VirtualBox-7.0.4/src/VBox/ExtPacks/BusMouseSample/Makefile.kmk
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/ExtPacks/BusMouseSample/Makefile.kmk
+++ VirtualBox-7.0.4/src/VBox/ExtPacks/BusMouseSample/Makefile.kmk
@@ -193,7 +193,7 @@ $(VBOX_PATH_PACKAGES)/$(VBOX_BUSMOUSE_MA
 		$(VBoxBusMouseIns_0_OUTDIR)/Stage/ExtPack.manifest \
 		$(VBoxBusMouseIns_0_OUTDIR)/Stage/ExtPack.signature
 # Tar it up.
-	tar -cvf - -C $(VBoxBusMouseIns_0_OUTDIR)/Stage/ . | gzip -9c > $@
+	tar -cvf - -C $(VBoxBusMouseIns_0_OUTDIR)/Stage/ . | gzip -9nc > $@
 # Clean up
 	$(RM) -Rf $(VBoxBusMouseIns_0_OUTDIR)/Stage/
 
Index: VirtualBox-7.0.4/src/VBox/ExtPacks/Skeleton/Makefile.kmk
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/ExtPacks/Skeleton/Makefile.kmk
+++ VirtualBox-7.0.4/src/VBox/ExtPacks/Skeleton/Makefile.kmk
@@ -161,7 +161,7 @@ $(VBOX_PATH_PACKAGES)/$(VBOX_SKELETON_MA
 		$(VBoxSkeletonIns_0_OUTDIR)/Stage/ExtPack.manifest \
 		$(VBoxSkeletonIns_0_OUTDIR)/Stage/ExtPack.signature
 # Tar it up.
-	tar -cvf - -C $(VBoxSkeletonIns_0_OUTDIR)/Stage/ . | gzip -9c > $@
+	tar -cvf - -C $(VBoxSkeletonIns_0_OUTDIR)/Stage/ . | gzip -9nc > $@
 # Clean up
 	$(RM) -Rf $(VBoxSkeletonIns_0_OUTDIR)/Stage/
 
Index: VirtualBox-7.0.4/src/VBox/ExtPacks/VBoxDTrace/Makefile.kmk
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/ExtPacks/VBoxDTrace/Makefile.kmk
+++ VirtualBox-7.0.4/src/VBox/ExtPacks/VBoxDTrace/Makefile.kmk
@@ -354,7 +354,7 @@ if defined(VBOX_WITH_EXTPACK_VBOXDTRACE)
 		$(VBoxDTraceIns_0_OUTDIR)/Stage/ExtPack.manifest \
 		$(VBoxDTraceIns_0_OUTDIR)/Stage/ExtPack.signature
  # Tar it up.
-	tar -cvf - -C $(VBoxDTraceIns_0_OUTDIR)/Stage/ . | gzip -9c > $@
+	tar -cvf - -C $(VBoxDTraceIns_0_OUTDIR)/Stage/ . | gzip -9nc > $@
  # Clean up
 	$(RM) -Rf $(VBoxDTraceIns_0_OUTDIR)/Stage/
 
Index: VirtualBox-7.0.4/src/VBox/ExtPacks/VNC/Makefile.kmk
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/ExtPacks/VNC/Makefile.kmk
+++ VirtualBox-7.0.4/src/VBox/ExtPacks/VNC/Makefile.kmk
@@ -164,7 +164,7 @@ $(VBOX_PATH_PACKAGES)/$(VBOX_VNC_MANGLED
 		$(VBoxVNCIns_0_OUTDIR)/Stage/ExtPack.manifest \
 		$(VBoxVNCIns_0_OUTDIR)/Stage/ExtPack.signature
 # Tar it up.
-	tar -cvf - -C $(VBoxVNCIns_0_OUTDIR)/Stage/ . | gzip -9c > $@
+	tar -cvf - -C $(VBoxVNCIns_0_OUTDIR)/Stage/ . | gzip -9nc > $@
 # Clean up
 	$(RM) -Rf $(VBoxVNCIns_0_OUTDIR)/Stage/
 
Index: VirtualBox-7.0.4/src/VBox/Additions/common/VBoxGuest/VBoxGuest-win.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Additions/common/VBoxGuest/VBoxGuest-win.cpp
+++ VirtualBox-7.0.4/src/VBox/Additions/common/VBoxGuest/VBoxGuest-win.cpp
@@ -329,7 +329,7 @@ NTSTATUS DriverEntry(PDRIVER_OBJECT pDrv
     }
     VGDrvCommonInitLoggers();
 
-    LogFunc(("Driver built: %s %s\n", __DATE__, __TIME__));
+    LogFunc(("Driver built: %s %s\n", "no date", "no time"));
 
     /*
      * Check if the NT version is supported and initialize g_enmVGDrvNtVer.
Index: VirtualBox-7.0.4/src/VBox/Additions/linux/lightdm-greeter/vbox-greeter.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Additions/linux/lightdm-greeter/vbox-greeter.cpp
+++ VirtualBox-7.0.4/src/VBox/Additions/linux/lightdm-greeter/vbox-greeter.cpp
@@ -882,7 +882,7 @@ static DECLCALLBACK(void) vboxGreeterLog
                    "vbox-greeter %s r%s (verbosity: %d) %s (%s %s) release log\n"
                    "Log opened %s\n",
                    RTBldCfgVersion(), RTBldCfgRevisionStr(), g_iVerbosity, VBOX_BUILD_TARGET,
-                   __DATE__, __TIME__, szTmp);
+                   "no date", "no time", szTmp);
 
             int vrc = RTSystemQueryOSInfo(RTSYSOSINFO_PRODUCT, szTmp, sizeof(szTmp));
             if (RT_SUCCESS(vrc) || vrc == VERR_BUFFER_OVERFLOW)
Index: VirtualBox-7.0.4/src/VBox/Additions/solaris/Mouse/vboxms.c
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Additions/solaris/Mouse/vboxms.c
+++ VirtualBox-7.0.4/src/VBox/Additions/solaris/Mouse/vboxms.c
@@ -266,7 +266,7 @@ static VBMSSTATE            g_OpenNodeSt
 int _init(void)
 {
     int rc;
-    LogRelFlow((DEVICE_NAME ": built on " __DATE__ " at " __TIME__ "\n"));
+    LogRelFlow((DEVICE_NAME ": built on " "no date" " at " "no time" "\n"));
     mutex_init(&g_OpenNodeState.InitMtx, NULL, MUTEX_DRIVER, NULL);
     /*
      * Prevent module autounloading.
Index: VirtualBox-7.0.4/src/VBox/Devices/EFI/Firmware/BaseTools/Source/C/VolInfo/VolInfo.c
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Devices/EFI/Firmware/BaseTools/Source/C/VolInfo/VolInfo.c
+++ VirtualBox-7.0.4/src/VBox/Devices/EFI/Firmware/BaseTools/Source/C/VolInfo/VolInfo.c
@@ -251,10 +251,10 @@ Returns:
     UTILITY_NAME,
     UTILITY_MAJOR_VERSION,
     UTILITY_MINOR_VERSION,
-    __BUILD_VERSION
+    __BUILD_VERSION,
+    "no date"
     );
 
-  if (argc == 1) {
     Usage ();
     return -1;
   }
Index: VirtualBox-7.0.4/src/VBox/HostDrivers/VBoxUSB/win/dev/VBoxUsbDev.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/HostDrivers/VBoxUSB/win/dev/VBoxUsbDev.cpp
+++ VirtualBox-7.0.4/src/VBox/HostDrivers/VBoxUSB/win/dev/VBoxUsbDev.cpp
@@ -128,7 +128,7 @@ static NTSTATUS vboxUsbDdiAddDevice(PDRI
 static VOID vboxUsbDdiUnload(PDRIVER_OBJECT pDriverObject)
 {
     RT_NOREF1(pDriverObject);
-    LogRel(("VBoxUsb::DriverUnload. Built Date (%s) Time (%s)\n", __DATE__, __TIME__));
+    LogRel(("VBoxUsb::DriverUnload. Built Date (%s) Time (%s)\n", "no date", "no time"));
     VBoxDrvToolStrFree(&g_VBoxUsbGlobals.RegPath);
 
     vboxUsbRtGlobalsTerm();
@@ -284,7 +284,7 @@ RT_C_DECLS_END
 
 NTSTATUS DriverEntry(IN PDRIVER_OBJECT pDriverObject, IN PUNICODE_STRING pRegistryPath)
 {
-    LogRel(("VBoxUsb::DriverEntry. Built Date (%s) Time (%s)\n", __DATE__, __TIME__));
+    LogRel(("VBoxUsb::DriverEntry. Built Date (%s) Time (%s)\n", "no date", "no time"));
 
     NTSTATUS Status = vboxUsbRtGlobalsInit();
     Assert(Status == STATUS_SUCCESS);
Index: VirtualBox-7.0.4/src/VBox/HostDrivers/VBoxUSB/win/mon/VBoxUsbMon.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/HostDrivers/VBoxUSB/win/mon/VBoxUsbMon.cpp
+++ VirtualBox-7.0.4/src/VBox/HostDrivers/VBoxUSB/win/mon/VBoxUsbMon.cpp
@@ -1518,7 +1518,7 @@ NTSTATUS _stdcall DriverEntry(PDRIVER_OB
     RTLogDestinations(0, "debugger");
 #endif
 
-    LOGREL(("Built %s %s", __DATE__, __TIME__));
+    LOGREL(("Built %s %s", "no date", "no time"));
 
     memset (&g_VBoxUsbMonGlobals, 0, sizeof (g_VBoxUsbMonGlobals));
 
Index: VirtualBox-7.0.4/src/VBox/Main/glue/VBoxLogRelCreate.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Main/glue/VBoxLogRelCreate.cpp
+++ VirtualBox-7.0.4/src/VBox/Main/glue/VBoxLogRelCreate.cpp
@@ -71,7 +71,7 @@ static DECLCALLBACK(void) vboxHeaderFoot
 #endif
                    "Log opened %s\n",
                    g_pszLogEntity, VBOX_VERSION_STRING, RTBldCfgRevision(),
-                   RTBldCfgTargetDotArch(), __DATE__, __TIME__, szTmp);
+                   RTBldCfgTargetDotArch(), "no date", "no time", szTmp);
 
             pfnLog(pReleaseLogger, "Build Type: %s\n", KBUILD_TYPE);
             int vrc = RTSystemQueryOSInfo(RTSYSOSINFO_PRODUCT, szTmp, sizeof(szTmp));
Index: VirtualBox-7.0.4/src/libs/xpcom18a4/nsprpub/pr/tests/depend.c
===================================================================
--- VirtualBox-7.0.4.orig/src/libs/xpcom18a4/nsprpub/pr/tests/depend.c
+++ VirtualBox-7.0.4/src/libs/xpcom18a4/nsprpub/pr/tests/depend.c
@@ -141,7 +141,7 @@ int main(int argc, char **argv)
 {
     PRIntn tab = 0;
     const PRVersionInfo *info = DummyLibVersion();
-    const char *buildDate = __DATE__, *buildTime = __TIME__;
+    const char *buildDate = "no date", *buildTime = "no time";
 
     printf("Depend.c build time is %s %s\n", buildDate, buildTime);
     
Index: VirtualBox-7.0.4/src/libs/xpcom18a4/xpcom/tests/StringFactoringTests/profile_main.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/libs/xpcom18a4/xpcom/tests/StringFactoringTests/profile_main.cpp
+++ VirtualBox-7.0.4/src/libs/xpcom18a4/xpcom/tests/StringFactoringTests/profile_main.cpp
@@ -458,7 +458,7 @@ int
 main()
   {
 
-    cout << "String performance profiling.  Compiled " __DATE__ " " __TIME__ << endl;
+    cout << "String performance profiling.  Compiled nodate no time" << endl;
 #ifdef TEST_STD_STRING
     cout << "Testing std::string." << endl;
 #else
Index: VirtualBox-7.0.4/src/libs/xpcom18a4/xpcom/tests/StringFactoringTests/test_main.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/libs/xpcom18a4/xpcom/tests/StringFactoringTests/test_main.cpp
+++ VirtualBox-7.0.4/src/libs/xpcom18a4/xpcom/tests/StringFactoringTests/test_main.cpp
@@ -425,7 +425,7 @@ int
 main()
   {	
     int tests_failed = 0;
-  	cout << "String unit tests.  Compiled " __DATE__ " " __TIME__ << endl;
+  	cout << "String unit tests.  Compiled no date no time" << endl;
 
 #if 0
     {
Index: VirtualBox-7.0.4/src/libs/xpcom18a4/xpcom/tests/TestCOMPtr.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/libs/xpcom18a4/xpcom/tests/TestCOMPtr.cpp
+++ VirtualBox-7.0.4/src/libs/xpcom18a4/xpcom/tests/TestCOMPtr.cpp
@@ -102,7 +102,7 @@ class test_message
     public:
       test_message()
         {
-          printf("BEGIN unit tests for |nsCOMPtr|, compiled " __DATE__ "\n");
+          printf("BEGIN unit tests for |nsCOMPtr|, compiled no date\n");
         }
 
      ~test_message()
Index: VirtualBox-7.0.4/src/VBox/Additions/3D/mesa/mesa-21.3.8/src/util/build_id.c
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Additions/3D/mesa/mesa-21.3.8/src/util/build_id.c
+++ VirtualBox-7.0.4/src/VBox/Additions/3D/mesa/mesa-21.3.8/src/util/build_id.c
@@ -30,9 +30,8 @@
 #include "build_id.h"
 #include "macros.h"
 
-#ifndef NT_GNU_BUILD_ID
+#undef NT_GNU_BUILD_ID
 #define NT_GNU_BUILD_ID 3
-#endif
 
 #ifndef ElfW
 #define ElfW(type) Elf_##type
Index: VirtualBox-7.0.4/src/VBox/Additions/x11/VBoxClient/logging.cpp
===================================================================
--- VirtualBox-7.0.4.orig/src/VBox/Additions/x11/VBoxClient/logging.cpp
+++ VirtualBox-7.0.4/src/VBox/Additions/x11/VBoxClient/logging.cpp
@@ -203,10 +203,10 @@ static DECLCALLBACK(void) vbClLogHeaderF
         case RTLOGPHASE_BEGIN:
         {
             pfnLog(pLoggerRelease,
-                   "VBoxClient %s r%s (verbosity: %u) %s (%s %s) release log\n"
+                   "VBoxClient %s r%s (verbosity: %u) %s release log\n"
                    "Log opened %s\n",
                    RTBldCfgVersion(), RTBldCfgRevisionStr(), g_cVerbosity, VBOX_BUILD_TARGET,
-                   __DATE__, __TIME__, szTmp);
+                   szTmp);
 
             int vrc = RTSystemQueryOSInfo(RTSYSOSINFO_PRODUCT, szTmp, sizeof(szTmp));
             if (RT_SUCCESS(vrc) || vrc == VERR_BUFFER_OVERFLOW)
