Index: elasticsearch-py-8.3.3/test_elasticsearch/test_helpers.py
===================================================================
--- elasticsearch-py-8.3.3.orig/test_elasticsearch/test_helpers.py
+++ elasticsearch-py-8.3.3/test_elasticsearch/test_helpers.py
@@ -19,7 +19,7 @@
 import threading
 import time
 
-import mock
+from unittest import mock
 import pytest
 
 from elasticsearch import Elasticsearch, helpers
Index: elasticsearch-py-8.3.3/test_elasticsearch/test_server/test_helpers.py
===================================================================
--- elasticsearch-py-8.3.3.orig/test_elasticsearch/test_server/test_helpers.py
+++ elasticsearch-py-8.3.3/test_elasticsearch/test_server/test_helpers.py
@@ -21,7 +21,7 @@ from datetime import datetime, timedelta
 import pytest
 from dateutil import tz
 from elastic_transport import ApiResponseMeta, ObjectApiResponse
-from mock import call, patch
+from unittest.mock import call, patch
 
 from elasticsearch import ApiError, helpers
 from elasticsearch.helpers import ScanError
Index: elasticsearch-py-8.3.3/test_elasticsearch/test_async/test_server/test_helpers.py
===================================================================
--- elasticsearch-py-8.3.3.orig/test_elasticsearch/test_async/test_server/test_helpers.py
+++ elasticsearch-py-8.3.3/test_elasticsearch/test_async/test_server/test_helpers.py
@@ -20,7 +20,7 @@ from datetime import datetime, timedelta
 
 import pytest
 from elastic_transport import ApiResponseMeta, ObjectApiResponse
-from mock import MagicMock, call, patch
+from unittest.mock import MagicMock, call, patch
 
 from elasticsearch import helpers
 from elasticsearch.exceptions import ApiError
