Index: b/lib-src/portaudio-v19/qa/loopback/src/paqa.c
===================================================================
--- a/lib-src/portaudio-v19/qa/loopback/src/paqa.c	2021-03-23 08:54:14.352557632 +0200
+++ b/lib-src/portaudio-v19/qa/loopback/src/paqa.c	2021-03-23 08:54:38.785492888 +0200
@@ -1460,7 +1460,7 @@ int main( int argc, char **argv )
 	int justMath = 0;
     char *executableName = argv[0];
 
-	printf("PortAudio LoopBack Test built " __DATE__ " at " __TIME__ "\n");
+	printf("PortAudio LoopBack Test built " "17 March 2017" " at " "23:15" "\n");
 
 	if( argc > 1 ){
 		printf("running with arguments:");
Index: b/src/AboutDialog.cpp
===================================================================
--- a/src/AboutDialog.cpp	2021-03-23 08:54:14.352557632 +0200
+++ b/src/AboutDialog.cpp	2021-03-23 08:54:38.785492888 +0200
@@ -69,7 +69,7 @@ hold information about one contributor t
 #endif
 
 #ifdef REV_LONG
-#define REV_IDENT wxString( "[[https://github.com/audacity/audacity/commit/" )+ REV_LONG + "|" + wxString( REV_LONG ).Left(6) + "]] of " +  REV_TIME 
+#define REV_IDENT wxString( "Official openSUSE Build" )+ REV_LONG + "|" + wxString( REV_LONG ).Left(6) + "]] of " +  REV_TIME
 #else
 #define REV_IDENT (XO("No revision identifier was provided").Translation())
 #endif
@@ -750,8 +750,8 @@ void AboutDialog::PopulateInformationPag
       << XO("Build Information")
       << wxT("</h3>\n<table>");
 
-   // Current date
-   AddBuildinfoRow(&informationStr, XO("Program build date:"), __TDATE__);
+   /*/ Current date
+   AddBuildinfoRow(&informationStr, XO("Program build date:"), __TDATE__);*/
    AddBuildinfoRow(&informationStr, XO("Commit Id:"), REV_IDENT );
 
    auto buildType =
