Index: jmespath.py-0.10.0/extra/test_hypothesis.py
===================================================================
--- jmespath.py-0.10.0.orig/extra/test_hypothesis.py
+++ jmespath.py-0.10.0/extra/test_hypothesis.py
@@ -6,7 +6,6 @@ import os
 import sys
 import numbers
 
-from nose.plugins.skip import SkipTest
 from hypothesis import given, settings, assume, HealthCheck
 import hypothesis.strategies as st
 
Index: jmespath.py-0.10.0/tests/test_compliance.py
===================================================================
--- jmespath.py-0.10.0.orig/tests/test_compliance.py
+++ jmespath.py-0.10.0/tests/test_compliance.py
@@ -3,8 +3,6 @@ from pprint import pformat
 from tests import OrderedDict
 from tests import json
 
-from nose.tools import assert_equal
-
 from jmespath.visitor import Options
 
 
@@ -80,7 +78,7 @@ def _test_expression(given, expression,
                      actual_repr, pformat(parsed.parsed),
                      json.dumps(given, indent=4)))
     error_msg = error_msg.replace(r'\n', '\n')
-    assert_equal(actual, expected, error_msg)
+    assert actual == expected, error_msg
 
 
 def _test_error_expression(given, expression, error, filename):
