diff -ur Finance-Quote-1.50/lib/Finance/Quote/Cdnfundlibrary.pm Finance-Quote-1.50_fix/lib/Finance/Quote/Cdnfundlibrary.pm
--- Finance-Quote-1.50/lib/Finance/Quote/Cdnfundlibrary.pm	2021-06-27 05:52:21.000000000 +0200
+++ Finance-Quote-1.50_fix/lib/Finance/Quote/Cdnfundlibrary.pm	2021-08-25 18:33:24.262128339 +0200
@@ -102,7 +102,7 @@
 
       @rows = $te->rows;
       if(@rows) {
-        $row = $rows[1];
+        $row = $rows[0];
 
         $fundquote {$mutual, "name"} = trim($$row[0]);
         $fundquote {$mutual, "symbol"} = $mutual;
