From 3a56481f8d13b6874a28752502a58520b9139dc7 Mon Sep 17 00:00:00 2001
From: Amaan Qureshi <amaanq12@gmail.com>
Date: Sun, 10 Mar 2024 12:53:33 -0400
Subject: [PATCH] fix: escape braces in regex

---
 grammar.js       |   4 ++--
1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/grammar.js b/grammar.js
index 4119447..4df6abd 100644
--- a/grammar.js
+++ b/grammar.js
@@ -1476,7 +1476,7 @@ module.exports = grammar({
         seq('\\', choice(
           /[^xu]/,
           /u[0-9a-fA-F]{4}/,
-          /u{[0-9a-fA-F]+}/,
+          /u\{[0-9a-fA-F]+\}/,
           /x[0-9a-fA-F]{2}/,
         )),
         /[^\\']/,
@@ -1489,7 +1489,7 @@ module.exports = grammar({
         choice(
           /[^xu]/,
           /u[0-9a-fA-F]{4}/,
-          /u{[0-9a-fA-F]+}/,
+          /u\{[0-9a-fA-F]+\}/,
           /x[0-9a-fA-F]{2}/,
         ),
       )),

-- 
2.44.0

