Index: sqlparse-0.3.1/tests/test_cli.py
===================================================================
--- sqlparse-0.3.1.orig/tests/test_cli.py
+++ sqlparse-0.3.1/tests/test_cli.py
@@ -1,5 +1,6 @@
 # -*- coding: utf-8 -*-
 
+import io
 import subprocess
 import sys
 
@@ -78,7 +79,10 @@ def test_script():
 def test_encoding_utf8_stdout(filepath, load_file, capfd):
     path = filepath('encoding_utf8.sql')
     expected = load_file('encoding_utf8.sql', 'utf-8')
-    sys.stdout.encoding = 'utf-8'
+    if isinstance(sys.stdout, io.TextIOWrapper):
+        sys.stdout.reconfigure(encoding='utf-8')
+    else:
+        sys.stdout.encoding = 'utf-8'
     sqlparse.cli.main([path])
     out, _ = capfd.readouterr()
     assert out == expected
@@ -96,7 +100,10 @@ def test_encoding_utf8_output_file(filep
 def test_encoding_gbk_stdout(filepath, load_file, capfd):
     path = filepath('encoding_gbk.sql')
     expected = load_file('encoding_gbk.sql', 'gbk')
-    sys.stdout.encoding = 'gbk'
+    if isinstance(sys.stdout, io.TextIOWrapper):
+        sys.stdout.reconfigure(encoding='gbk')
+    else:
+        sys.stdout.encoding = 'gbk'
     sqlparse.cli.main([path, '--encoding', 'gbk'])
     out, _ = capfd.readouterr()
     assert out == expected
@@ -117,7 +124,10 @@ def test_encoding_stdin_utf8(filepath, l
     old_stdin = sys.stdin
     with open(path, 'r') as f:
         sys.stdin = f
-        sys.stdout.encoding = 'utf-8'
+        if isinstance(sys.stdout, io.TextIOWrapper):
+            sys.stdout.reconfigure(encoding='utf-8')
+        else:
+            sys.stdout.encoding = 'utf-8'
         sqlparse.cli.main(['-'])
     sys.stdin = old_stdin
     out, _ = capfd.readouterr()
@@ -130,7 +140,10 @@ def test_encoding_stdin_gbk(filepath, lo
     old_stdin = sys.stdin
     with open(path, 'r') as stream:
         sys.stdin = stream
-        sys.stdout.encoding = 'gbk'
+        if isinstance(sys.stdout, io.TextIOWrapper):
+            sys.stdout.reconfigure(encoding='gbk')
+        else:
+            sys.stdout.encoding = 'gbk'
         sqlparse.cli.main(['-', '--encoding', 'gbk'])
         sys.stdin = old_stdin
     out, _ = capfd.readouterr()
