Index: opal-3.6.1/plugins/audio/iLBC/iLBC/iCBSearch.c
===================================================================
--- opal-3.6.1.orig/plugins/audio/iLBC/iLBC/iCBSearch.c
+++ opal-3.6.1/plugins/audio/iLBC/iLBC/iCBSearch.c
@@ -121,7 +121,8 @@
                *ppe=0.0;
                pp=buf+LPC_FILTERORDER+lMem-lTarget;
                for (j=0; j<lTarget; j++) {
-                   *ppe+=(*pp)*(*pp++);
+                   *ppe+=(*pp)*(*pp);
+                   *pp++;
                }
                
                if (*ppe>0.0) {
