From 8f22f67ecdce4991ce8b5a8f271e6db3e0e5dc35 Mon Sep 17 00:00:00 2001
From: therandomstring <bal.horv.98@gmail.com>
Date: Thu, 28 Aug 2025 14:14:59 +0200
Subject: [PATCH 4/4] Merge pull request #5447 from
 therandomstring/s3-fix-missing-upload-trigger

S3: Fix missing upload trigger
(cherry picked from commit e0b765ef9b82abc9127c3e71c3971279450b614e)
---
 modules/python-modules/syslogng/modules/s3/s3_destination.py | 2 ++
 news/bugfix-5447.md                                          | 1 +
 2 files changed, 3 insertions(+)
 create mode 100644 news/bugfix-5447.md

diff --git a/modules/python-modules/syslogng/modules/s3/s3_destination.py b/modules/python-modules/syslogng/modules/s3/s3_destination.py
index cd78b6db2..26b39c779 100644
--- a/modules/python-modules/syslogng/modules/s3/s3_destination.py
+++ b/modules/python-modules/syslogng/modules/s3/s3_destination.py
@@ -371,6 +371,8 @@ class S3Destination(LogDestination):
                     with self.__indices_lock:
                         last_index = self.__indices.pop(s3_object.object_key)
                         self.__indices[s3_object.object_key] = last_index
+        if not self.s3_object_ready_queue.queue.empty():
+            self.__session_handler.trigger_upload()
 
         self.__start_flush_poll_timer()
 
diff --git a/news/bugfix-5447.md b/news/bugfix-5447.md
new file mode 100644
index 000000000..7bb8cdde7
--- /dev/null
+++ b/news/bugfix-5447.md
@@ -0,0 +1 @@
+s3: Fixed bug where in certain conditions finished object buffers would fail to upload.
-- 
2.51.0

