--- T1Hndlr.c
+++ T1Hndlr.c
@@ -149,7 +149,7 @@
   UCHAR ucRTemp[MAX_BUFFER_SIZE];
   UCHAR ucRHeader[5];
   DWORD dwCycles, dwRemaining, dwOffset;
-  DWORD dwRBufferLen, dwRCount;
+  DWORD dwRBufferLen, dwRCount = 0;
   int i;
   static UCHAR ucSChainNum  = 0;
   static UCHAR ucRChainNum  = 1;
--- ctapi.c
+++ ctapi.c
@@ -128,7 +128,7 @@
   unsigned char BrdrResp[3];                  /* Reader Response */
   unsigned char cRdrIrsp[5];                  /* Ack + Size Resp */
   unsigned int  lctb;                         /* CTBCS Size      */
-  int IretVal;                                /* Return Value    */
+  int IretVal = 0;                            /* Return Value    */
   int i;
 
 
--- ifdhandler.c
+++ ifdhandler.c
@@ -244,7 +244,7 @@
 
 RESPONSECODE IFD_Set_Capabilities ( DWORD Tag, BYTE Value[] ) {
 
-  RESPONSECODE lRetVal;
+  RESPONSECODE lRetVal=0;
   DWORD HighNibble;
   DWORD LowNibble;
 
