--- ifdhandler.c
+++ ifdhandler.c
@@ -425,7 +425,7 @@
 /* Print Transmit Error Messaging */
 
 #ifdef PCSC_DEBUG
-  printf("[%04x] -> ", lc );
+  printf("[%04x] -> ", (int)lc );
 
   for ( i=0; i < lc; i++ ) {
     printf("%02x ", CommandData[i]);
@@ -483,7 +483,7 @@
 /* Print Receive Error Messaging */
 
 #ifdef PCSC_DEBUG
-  printf("[%04x] <- ", *ResponseSize );
+  printf("[%04x] <- ", (int)*ResponseSize );
 
   for ( i=0; i < lr; i++ ) {
     printf("%02x ", ResponseData[i]);
