--- monitor/scpk11monitor-threads.c.orig	2007-05-23 15:48:44.000000000 +0000
+++ monitor/scpk11monitor-threads.c	2008-03-15 17:56:58.000000000 +0000
@@ -96,7 +96,7 @@
     ScPk11MonitorState state;
 };
 
-#define ASSERT_LOCKED(m) (g_assert (!g_mutex_trylock ((m)->priv->mutex)))
+#define ASSERT_LOCKED(m) g_assert (!g_mutex_trylock ((m)->priv->mutex));
 
 typedef struct {
     char       *token_name;
Only in scmon-0.3: scmon
