Index: cadabra2-2.5.4/core/packages/CMakeLists.txt
===================================================================
--- cadabra2-2.5.4.orig/core/packages/CMakeLists.txt
+++ cadabra2-2.5.4/core/packages/CMakeLists.txt
@@ -67,7 +67,7 @@ foreach(PACKAGE ${COMPILED_PACKAGES})
   set_target_properties(${PNAME} PROPERTIES SUFFIX ".${PYTHON_MOD_SUFFIX}")
   set_target_properties(${PNAME} PROPERTIES INSTALL_RPATH "$ORIGIN/../../:$ORIGIN/../../../../")
   
-  target_link_libraries(${PNAME} PUBLIC cadabra2 ${GMPXX_LIBRARIES})
+  target_link_libraries(${PNAME} PUBLIC cadabra2 ${GMPXX_LIBRARIES} ${GMP_LIBRARIES})
   target_include_directories(${PNAME} PUBLIC ${CADABRA_CORE_DIR})
 
   if(IPO_SUPPORTED)

