diff -Nru aws-checksums-0.1.10.orig/CMakeLists.txt aws-checksums-0.1.10/CMakeLists.txt
--- aws-checksums-0.1.10.orig/CMakeLists.txt	2020-09-12 01:26:35.000000000 +0200
+++ aws-checksums-0.1.10/CMakeLists.txt	2020-11-19 13:33:13.773965519 +0100
@@ -118,8 +118,8 @@
 
 aws_add_sanitizers(${PROJECT_NAME})
 
-# We are not ABI stable yet
 set_target_properties(${PROJECT_NAME} PROPERTIES VERSION 1.0.0)
+set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES SOVERSION 1)
 
 target_include_directories(${PROJECT_NAME} PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
