--- config.c
+++ config.c
@@ -170,7 +170,7 @@
     Token *vp, *vp2;
     ParamData *ppt;
     SeqnData *sd;
-    void *dp;
+    void *dp = NULL;
     extern void dumpToken(Token *, FILE *);
     
     ppt = newpdata();
@@ -209,7 +209,7 @@
 		default:
 		    if (strncmp(vp->text, "0x", 2) == 0)
 		    {
-		        sscanf(vp->text, "%x", (unsigned int *) &dp);
+		        sscanf(vp->text, "%x", (unsigned int *) dp);
 		    }
 		    else
 		    {
--- configure.ac
+++ configure.ac
@@ -6,7 +6,7 @@
 # Checks for programs.
 AC_PROG_CC
 
-CFLAGS="-g"
+AM_CFLAGS="-g"
 
 AC_PROG_INSTALL
 AC_PROG_AWK
