Index: orage-4.10.0/doc/C/Makefile.am
===================================================================
--- orage-4.10.0.orig/doc/C/Makefile.am
+++ orage-4.10.0/doc/C/Makefile.am
@@ -4,12 +4,12 @@ SUBDIRS = \
 noinst_DATA = \
 	orage.xml
 
-docdir=$(datadir)/orage/doc/C
+cdocdir=$(docdir)/C
 
-doc_DATA= \
+cdoc_DATA= \
 	orage.html
 
-EXTRA_DIST = $(noinst_DATA) $(doc_DATA)
+EXTRA_DIST = $(noinst_DATA) $(cdoc_DATA)
 
 html: $(srcdir)/orage.xml
 	xsltproc $(top_srcdir)/doc/xfce-nochunk.xsl \
Index: orage-4.10.0/doc/C/images/Makefile.am
===================================================================
--- orage-4.10.0.orig/doc/C/images/Makefile.am
+++ orage-4.10.0/doc/C/images/Makefile.am
@@ -1,5 +1,5 @@
-imagesdir = $(datadir)/orage/doc/C/images
-images_DATA = \
+cimagesdir = $(docdir)/C/images
+cimages_DATA = \
 	default_calendar.png \
 	orage_calendar_edit_menu.png \
 	orage_calendar_file_menu.png \
@@ -18,4 +18,4 @@ images_DATA = \
 	orage_preferences_main.png \
 	orage_preferences_extra.png 
 
-EXTRA_DIST = $(images_DATA)
+EXTRA_DIST = $(cimages_DATA)
Index: orage-4.10.0/src/Makefile.am
===================================================================
--- orage-4.10.0.orig/src/Makefile.am
+++ orage-4.10.0/src/Makefile.am
@@ -39,7 +39,8 @@ orage_SOURCES =					\
 orage_CFLAGS =							\
     $(LIBGTK_CFLAGS)                    \
 	-DPACKAGE_DATA_DIR=\""$(datadir)"\"	\
-	-DPACKAGE_LOCALE_DIR=\""$(localedir)"\"
+	-DPACKAGE_LOCALE_DIR=\""$(localedir)"\" \
+	-DPACKAGE_DOC_DIR=\""$(docdir)"\"
 
 orage_LDADD =							\
     $(LIBGTK_LIBS)                      \
Index: orage-4.10.0/src/mainbox.c
===================================================================
--- orage-4.10.0.orig/src/mainbox.c
+++ orage-4.10.0/src/mainbox.c
@@ -196,9 +196,7 @@ static void mHelp_help_activate_cb(GtkMe
 #ifdef ORAGE_DEBUG
     orage_message(-100, P_N);
 #endif
-    helpdoc = g_strconcat("exo-open ", PACKAGE_DATA_DIR
-           , G_DIR_SEPARATOR_S, "orage"
-           , G_DIR_SEPARATOR_S, "doc"
+    helpdoc = g_strconcat("exo-open ", PACKAGE_DOC_DIR
            , G_DIR_SEPARATOR_S, "C"
            , G_DIR_SEPARATOR_S, "orage.html"
            , NULL);
@@ -207,9 +205,7 @@ static void mHelp_help_activate_cb(GtkMe
                                     , error->message);
         g_clear_error(&error);
         g_free(helpdoc);
-        helpdoc = g_strconcat("firefox ", PACKAGE_DATA_DIR
-               , G_DIR_SEPARATOR_S, "orage"
-               , G_DIR_SEPARATOR_S, "doc"
+        helpdoc = g_strconcat("firefox ", PACKAGE_DOC_DIR
                , G_DIR_SEPARATOR_S, "C"
                , G_DIR_SEPARATOR_S, "orage.html"
                , NULL);
Index: orage-4.10.0/src/parameters.c
===================================================================
--- orage-4.10.0.orig/src/parameters.c
+++ orage-4.10.0/src/parameters.c
@@ -97,9 +97,7 @@ static void dialog_response(GtkWidget *d
 
     if (response_id == GTK_RESPONSE_HELP) {
         /* Needs to be in " to keep # */
-        helpdoc = g_strconcat("exo-open \"file://", PACKAGE_DATA_DIR
-                , G_DIR_SEPARATOR_S, "orage"
-                , G_DIR_SEPARATOR_S, "doc"
+        helpdoc = g_strconcat("exo-open \"file://", PACKAGE_DOC_DIR
                 , G_DIR_SEPARATOR_S, "C"
                 , G_DIR_SEPARATOR_S, "orage.html#orage-preferences-window\""
                 , NULL);
@@ -108,9 +106,7 @@ static void dialog_response(GtkWidget *d
                     , error->message);
             g_clear_error(&error);
             g_free(helpdoc);
-            helpdoc = g_strconcat("firefox \"file://", PACKAGE_DATA_DIR
-                    , G_DIR_SEPARATOR_S, "orage"
-                    , G_DIR_SEPARATOR_S, "doc"
+            helpdoc = g_strconcat("firefox \"file://", PACKAGE_DOC_DIR
                     , G_DIR_SEPARATOR_S, "C"
                     , G_DIR_SEPARATOR_S, "orage.html#orage-preferences-window\""
                     , NULL);
