# use system sqlite3
Index: redeclipse-1.5.5/src/Makefile
===================================================================
--- redeclipse-1.5.5.orig/src/Makefile	2016-07-01 19:48:00.000000000 +0200
+++ redeclipse-1.5.5/src/Makefile	2016-07-01 21:07:49.223921890 +0200
@@ -83,7 +83,7 @@
 endif
 endif
 CLIENT_INCLUDES= $(INCLUDES) -I/usr/X11R6/include `sdl2-config --cflags`
-CLIENT_LIBS= -Lenet -lenet -L/usr/X11R6/lib -lX11 `sdl2-config --libs` -lSDL2_image -lSDL2_mixer -lz -lGL
+CLIENT_LIBS= -Lenet -lenet -L/usr/X11R6/lib -lX11 `sdl2-config --libs` -lSDL2_image -lSDL2_mixer -lz -lGL -lsqlite3
 endif
 ifneq (,$(findstring linux,$(PLATFORM)))
 CLIENT_LIBS+= -lrt
@@ -99,7 +99,6 @@
 	shared/stream.o \
 	shared/tools.o \
 	shared/zip.o \
-	support/sqlite3.o \
 	engine/bih.o \
 	engine/blend.o \
 	engine/blob.o \
@@ -161,7 +160,7 @@
 SERVER_LIBS= -mwindows $(STD_LIBS) -L$(WINDLL) -L$(WINLIB) -lzlib1 -lenet -lws2_32 -lwinmm
 else
 SERVER_INCLUDES= -DSTANDALONE $(INCLUDES)
-SERVER_LIBS= -Lenet -lenet -lz
+SERVER_LIBS= -Lenet -lenet -lz -lsqlite3
 endif
 SERVER_OBJS= \
 	shared/crypto-standalone.o \
@@ -169,7 +168,6 @@
 	shared/stream-standalone.o \
 	shared/tools-standalone.o \
 	shared/zip-standalone.o \
-	support/sqlite3.o \
 	engine/command-standalone.o \
 	engine/irc-standalone.o \
 	engine/master-standalone.o \
@@ -205,9 +203,6 @@
 %-standalone.o: %.cpp
 	$(CXX) $(CXXFLAGS) -c -o $@ $(subst -standalone.o,.cpp,$@)
 
-support/sqlite3.o: support/sqlite3.c
-	$(CC) -DSQLITE_THREADSAFE=0 -DSQLITE_OMIT_LOAD_EXTENSION -c -o support/sqlite3.o support/sqlite3.c
-
 $(CLIENT_OBJS): CXXFLAGS += $(CLIENT_INCLUDES)
 $(filter shared/%,$(CLIENT_OBJS)): $(filter shared/%,$(CLIENT_PCH))
 $(filter engine/%,$(CLIENT_OBJS)): $(filter engine/%,$(CLIENT_PCH))
@@ -387,7 +382,6 @@
 engine/master.o: shared/glemu.h shared/iengine.h shared/igame.h engine/irc.h
 engine/master.o: engine/sound.h engine/world.h engine/octa.h
 engine/master.o: engine/lightmap.h engine/bih.h engine/texture.h
-engine/master.o: engine/model.h support/sqlite3.h
 engine/material.o: engine/engine.h engine/version.h shared/cube.h shared/tools.h
 engine/material.o: shared/command.h shared/geom.h shared/ents.h
 engine/material.o: shared/glexts.h shared/glemu.h shared/iengine.h
@@ -682,7 +676,6 @@
 engine/master-standalone.o: engine/engine.h engine/version.h shared/cube.h
 engine/master-standalone.o: shared/tools.h shared/command.h shared/geom.h
 engine/master-standalone.o: shared/ents.h shared/iengine.h shared/igame.h
-engine/master-standalone.o: engine/irc.h engine/sound.h support/sqlite3.h
 engine/server-standalone.o: engine/engine.h engine/version.h shared/cube.h
 engine/server-standalone.o: shared/tools.h shared/command.h shared/geom.h
 engine/server-standalone.o: shared/ents.h shared/iengine.h shared/igame.h
