From a25849e4d54a23ae1294b129d5696ca7e144ec8b Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Sun, 24 Apr 2022 18:46:24 +0200
Subject: [PATCH] configure.ac: Fix inline function test

fixes #1171
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7af4dc971..05a4a4024 100644
--- a/configure.ac
+++ b/configure.ac
@@ -97,7 +97,7 @@ AC_LANG_PUSH(C)
 AC_MSG_CHECKING([if $CC supports C99 without any flags])
 AC_COMPILE_IFELSE(
 	[AC_LANG_PROGRAM(
-		[[static inline int c99_restrict(int *restrict c99_restrict) {}]],
+		[[static inline void c99_restrict(int *restrict c99_restrict) {}]],
 		[[]])],
 	[AC_MSG_RESULT([yes])
 	 have_c99_native=yes],
@@ -111,7 +111,7 @@ CFLAGS="$CFLAGS -std=c99"
 AC_MSG_CHECKING([if $CC supports C99 with the -std=c99 flag])
 AC_COMPILE_IFELSE(
 	[AC_LANG_PROGRAM(
-		[[static inline int c99_restrict(int *restrict c99_restrict) {}]],
+		[[static inline void c99_restrict(int *restrict c99_restrict) {}]],
 		[[]])],
 	[AC_MSG_RESULT([yes])
 	 have_c99_c99=yes],
