Index: napalm-3.4.0/napalm/pyIOSXR/iosxr.py
===================================================================
--- napalm-3.4.0.orig/napalm/pyIOSXR/iosxr.py
+++ napalm-3.4.0/napalm/pyIOSXR/iosxr.py
@@ -35,8 +35,8 @@ from xml.sax.saxutils import escape as e
 # third party lib
 from lxml import etree as ET
 from netmiko import ConnectHandler
-from netmiko.ssh_exception import NetMikoTimeoutException
-from netmiko.ssh_exception import NetMikoAuthenticationException
+from netmiko.exceptions import NetMikoTimeoutException
+from netmiko.exceptions import NetMikoAuthenticationException
 
 # local modules
 from napalm.pyIOSXR.exceptions import LockError
