Index: src/common.h
===================================================================
--- src/common.h.orig
+++ src/common.h
@@ -282,7 +282,7 @@ extern char *getenv __P_ ((const char *e
 
 /*! Default length of a "string" text buffer. */
 # ifndef MAXLEN_MAX
-#  define MAXLEN_MAX       1024
+#  define MAXLEN_MAX       2048
 # endif
 
 /*! Default number of command line arguments in `my_argv[]' table. */
