Index: src/lp5250d.c
===================================================================
--- src/lp5250d.c.orig	2002-06-13 12:58:56.000000000 -0400
+++ src/lp5250d.c	2010-08-29 15:39:47.127123473 -0400
@@ -63,8 +63,7 @@ int main(int argc, char *argv[])
 #ifndef NDEBUG
    if(tn5250_config_get (config, "trace")) {
       tn5250_log_open(tn5250_config_get (config, "trace"));
-      TN5250_LOG(("lp5250d version %s, built on %s\n", version_string, 
-            __DATE__));
+      TN5250_LOG(("lp5250d version %s\n", version_string));
       TN5250_LOG(("host = %s\n", tn5250_config_get (config, "host")));
    }
 #endif 
