--- src/tn5250.c
+++ src/tn5250.c
@@ -178,9 +178,9 @@
 Syntax:\n\
   tn5250 [options] HOST[:PORT]\n");
 #ifdef HAVE_LIBSSL
-   printf("\
-   To connect using ssl prefix HOST with 'ssl:'.  Example:
-      tn5250 +ssl_verify_server ssl:as400.example.com\n");
+   printf("\n"
+"   To connect using ssl prefix HOST with 'ssl:'.  Example:\n"
+"      tn5250 +ssl_verify_server ssl:as400.example.com\n");
 #endif
    printf ("\n\
 Options:\n\
