Index: tn5250-0.18.0/lp5250d/lp5250d.c
===================================================================
--- tn5250-0.18.0.orig/lp5250d/lp5250d.c
+++ tn5250-0.18.0/lp5250d/lp5250d.c
@@ -101,7 +101,7 @@ int main(int argc, char* argv[]) {
     if (tn5250_config_get(config, "trace")) {
         tn5250_log_open(tn5250_config_get(config, "trace"));
         TN5250_LOG(
-            ("lp5250d version %s, built on %s\n", version_string, __DATE__));
+            ("lp5250d version %s\n", version_string));
         TN5250_LOG(("host = %s\n", tn5250_config_get(config, "host")));
     }
 #endif
