Index: invocations-1.4.0/tests/autodoc/base.py
===================================================================
--- invocations-1.4.0.orig/tests/autodoc/base.py
+++ invocations-1.4.0/tests/autodoc/base.py
@@ -1,4 +1,5 @@
 from os.path import join, dirname
+import codecs
 import re
 import shutil
 
@@ -28,7 +29,7 @@ class autodoc_:
     def setup_class(self):
         # Build once, introspect many...for now
         self.build_dir = _build()
-        with open(join(self.build_dir, "api.html")) as fd:
+        with codecs.open(join(self.build_dir, "api.html"), 'r', encoding='utf-8') as fd:
             self.api_docs = fd.read()
 
     @classmethod
