Index: nut-2.7.4/scripts/systemd/nut-server.service.in
===================================================================
--- nut-2.7.4.orig/scripts/systemd/nut-server.service.in
+++ nut-2.7.4/scripts/systemd/nut-server.service.in
@@ -18,6 +18,17 @@
 PartOf=nut.target
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+ProtectHostname=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 EnvironmentFile=-@CONFPATH@/nut.conf
 SyslogIdentifier=%N
 # Note: foreground mode by default skips writing a PID file (and
