Index: nut-2.8.5/conf/hosts.conf.sample
===================================================================
--- nut-2.8.5.orig/conf/hosts.conf.sample
+++ nut-2.8.5/conf/hosts.conf.sample
@@ -44,6 +44,8 @@
 # MONITOR su2200@10.64.1.1 "Finance department"
 # MONITOR matrix@shs-server.example.edu "Sierra High School data room #1"
 
+MONITOR myups@localhost "Local UPS"
+
 # -----------------------------------------------------------------------
 #
 # Allowed custom template file (adapted copy of upsstats.html) for listing
Index: nut-2.8.5/conf/ups.conf.sample
===================================================================
--- nut-2.8.5.orig/conf/ups.conf.sample
+++ nut-2.8.5/conf/ups.conf.sample
@@ -251,3 +251,8 @@ maxretry = 3
 #
 # To find out if your driver supports any extra settings, start it with
 # the -h option and/or read the driver's documentation.
+
+[myups]
+       driver = undefined
+       port = /dev/undefined
+       desc = "Local UPS"
Index: nut-2.8.5/conf/upsd.conf.sample
===================================================================
--- nut-2.8.5.orig/conf/upsd.conf.sample
+++ nut-2.8.5/conf/upsd.conf.sample
@@ -217,3 +217,11 @@
 # or CLI options, regardless of older logging level being higher or lower
 # than the newly found number; a missing (or commented away) value however
 # does not change the previously active logging verbosity.
+
+[upsmaster]
+       password = @UPSD_INITIAL_MASTER_PASSWORD@
+       upsmon master
+
+[upsslave]
+       password = @UPSD_INITIAL_SLAVE_PASSWORD@
+       upsmon slave
Index: nut-2.8.5/conf/upsmon.conf.sample.in
===================================================================
--- nut-2.8.5.orig/conf/upsmon.conf.sample.in
+++ nut-2.8.5/conf/upsmon.conf.sample.in
@@ -128,6 +128,8 @@
 # MONITOR su700@server.example.com 1 monuser secretpass secondary
 # MONITOR myups@localhost 1 monuser pass primary	# (or secondary)
 
+MONITOR myups@localhost 1 upsmaster @UPSD_INITIAL_MASTER_PASSWORD@ master
+
 # --------------------------------------------------------------------------
 # MINSUPPLIES <num>
 #
