Index: libreoffice-7.4.1.2/vcl/unx/gtk3/gtkinst.cxx
===================================================================
--- libreoffice-7.4.1.2.orig/vcl/unx/gtk3/gtkinst.cxx
+++ libreoffice-7.4.1.2/vcl/unx/gtk3/gtkinst.cxx
@@ -22552,7 +22552,11 @@ public:
         }
 #endif
 
-        gtk_popover_popup(m_pPopover);
+        // in GTK 3.22 you can just call gtk_popover_popup() to get it to show with transitions,
+        // but on SLE-12-SP5 we just have GTK 3.20, so enable transitions and show it to get the
+        // same effect
+        gtk_popover_set_transitions_enabled(m_pPopover, true);
+        gtk_widget_show(GTK_WIDGET(m_pPopover));
     }
 
 #if !GTK_CHECK_VERSION(4, 0, 0)
@@ -22593,7 +22597,7 @@ public:
         }
 #endif
 
-        gtk_popover_popdown(m_pPopover);
+        gtk_widget_hide(GTK_WIDGET(m_pPopover));
     }
 
     void PopdownAndFlushClosedSignal()
Index: libreoffice-7.4.1.2/libreofficekit/qa/gtktiledviewer/gtv-signal-handlers.cxx
===================================================================
--- libreoffice-7.4.1.2.orig/libreofficekit/qa/gtktiledviewer/gtv-signal-handlers.cxx
+++ libreoffice-7.4.1.2/libreofficekit/qa/gtktiledviewer/gtv-signal-handlers.cxx
@@ -342,7 +342,7 @@ void changeDateContentControl(GtkWidget*
         GtkPopover* pPopover = GTK_POPOVER(gtk_widget_get_parent(gtk_widget_get_parent(pSelector)));
         guint nYear, nMonth, nDay;
         gtk_calendar_get_date(GTK_CALENDAR(pSelector), &nYear, &nMonth, &nDay);
-        gtk_popover_popdown(pPopover);
+        gtk_widget_hide(GTK_WIDGET(pPopover));
 
         std::stringstream aDate;
         aDate << std::setfill('0') << std::setw(4) << nYear;
