Index: libreoffice-7.4.3.2/configure.ac
===================================================================
--- libreoffice-7.4.3.2.orig/configure.ac
+++ libreoffice-7.4.3.2/configure.ac
@@ -10594,7 +10594,7 @@ if test "$with_system_libfixmath" = "yes
     AC_LANG_PUSH([C++])
     AC_CHECK_HEADER([libfixmath/fix16.hpp], [],
        [AC_MSG_ERROR([libfixmath/fix16.hpp not found. install libfixmath])], [])
-    AC_CHECK_LIB([libfixmath], [fix16_mul], [:], [AC_MSG_ERROR(libfixmath lib not found or functional)], [])
+    AC_SEARCH_LIBS([fix16_mul], [fixmath libfixmath], [:], [AC_MSG_ERROR(libfixmath lib not found or functional)], [])
     AC_LANG_POP([C++])
 else
     AC_MSG_RESULT([internal])
