From 1e7d610b93fe575de776342f5cb84b8ba668cdb3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20=C4=8Cerm=C3=A1k?= <dcermak@suse.com>
Date: Fri, 30 Aug 2019 15:37:39 +0200
Subject: [PATCH] Use /usr/bin/bash instead of env as the shebang

---
 bin/setup | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/setup b/bin/setup
index 54e9ac36d..b3617a583 100755
--- a/bin/setup
+++ b/bin/setup
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/usr/bin/bash
 
 set -euo pipefail
 IFS=$'\n\t'
-- 
2.23.0

