Index: libopensync-0.22/tools/osyncdump.c
===================================================================
--- libopensync-0.22.orig/tools/osyncdump.c
+++ libopensync-0.22/tools/osyncdump.c
@@ -49,14 +49,17 @@ static void dump_map(OSyncEnv *osync, ch
 	while ((change = changes[i])) {
 		OSyncMember *member = osync_change_get_member(change);
 		int memberid = 0;
-    	if (member)
+    	if (member) {
     		memberid = osync_member_get_id(member);
-		const char *formatname = NULL;
-    	if (osync_change_get_objformat(change))
+	}
+	const char *formatname = NULL;
+    	if (osync_change_get_objformat(change)) {
     		formatname = osync_objformat_get_name(osync_change_get_objformat(change));
+	}
     	const char *objname = NULL;
-    	if (osync_change_get_objtype(change))
+    	if (osync_change_get_objtype(change)) {
     		objname = osync_objtype_get_name(osync_change_get_objtype(change));
+	}
     	printf("ID: %lli UID: %s MEMBER: %i\n\tOBJTYPE: %s OBJFORMAT: %s MAPPINGID: %lli\n", osync_change_get_id(change), osync_change_get_uid(change), memberid, objname, formatname, osync_change_get_mappingid(change));
 		i++;
 	}
Index: libopensync-0.22/tools/osyncstress.c
===================================================================
--- libopensync-0.22.orig/tools/osyncstress.c
+++ libopensync-0.22/tools/osyncstress.c
@@ -184,8 +184,9 @@ void change_content(OSyncEngine *engine)
 					case 2:
 						if (osync_member_delete_data(meminfo->member, chinfo->change)) {
 							printf("Deleting data %s for member %lli. Objtype: %s Format: %s\n", osync_change_get_uid(chinfo->change), osync_member_get_id(meminfo->member), osync_objtype_get_name(osync_change_get_objtype(chinfo->change)), osync_objformat_get_name(osync_change_get_objformat(chinfo->change)));
-							if (!osync_group_get_slow_sync(engine->group, osync_objtype_get_name(osync_change_get_objtype(chinfo->change))));
+							if (!osync_group_get_slow_sync(engine->group, osync_objtype_get_name(osync_change_get_objtype(chinfo->change)))) {
 								chinfo->type = CHANGE_DELETED;
+							}
 						}
 						break;
 					case 3:
