Index: Bear-3.1.4/source/citnames/source/semantic/Parsers.h
===================================================================
--- Bear-3.1.4.orig/source/citnames/source/semantic/Parsers.h
+++ Bear-3.1.4/source/citnames/source/semantic/Parsers.h
@@ -30,6 +30,7 @@
 #include <optional>
 
 #include <fmt/format.h>
+#include <fmt/ranges.h>
 
 namespace cs::semantic {
 
Index: Bear-3.1.4/source/citnames/source/semantic/Semantic.cc
===================================================================
--- Bear-3.1.4.orig/source/citnames/source/semantic/Semantic.cc
+++ Bear-3.1.4/source/citnames/source/semantic/Semantic.cc
@@ -21,6 +21,7 @@
 #include "semantic/Semantic.h"
 
 #include <fmt/format.h>
+#include <fmt/ranges.h>
 #ifdef HAVE_FMT_STD_H
 #include <fmt/std.h>
 #else
@@ -133,4 +134,4 @@ namespace cs::semantic {
         }
         return results;
     }
-}
\ No newline at end of file
+}
