Drop the build date from our binary

Index: acpica-unix-20240321/source/compiler/aslutils.c
===================================================================
--- acpica-unix-20240321.orig/source/compiler/aslutils.c
+++ acpica-unix-20240321/source/compiler/aslutils.c
@@ -637,8 +637,8 @@ UtDisplayOneSummary (
     {
         /* Compiler name and version number */
 
-        FlPrintFile (FileId, "%s version %X [%s]\n\n",
-            ASL_COMPILER_NAME, (UINT32) ACPI_CA_VERSION, __DATE__);
+        FlPrintFile (FileId, "%s version %X\n\n",
+            ASL_COMPILER_NAME, (UINT32) ACPI_CA_VERSION);
     }
 
     /* Summary of main input and output files */
