Index: python-efl-1.25.0/efl/evas/efl.evas.c
===================================================================
--- python-efl-1.25.0.orig/efl/evas/efl.evas.c
+++ python-efl-1.25.0/efl/evas/efl.evas.c
@@ -83766,15 +83766,15 @@ static int __pyx_pf_3efl_4evas_18SmartCb
   /* "efl/evas/efl.evas_object_smart.pxi":70
  *     def __init__(self, name, types):
  *         cdef Evas_Smart_Cb_Description *tmp
- *         tmp = <Evas_Smart_Cb_Description *>malloc(sizeof(Evas_Smart_Cb_Description*))             # <<<<<<<<<<<<<<
+ *         tmp = <Evas_Smart_Cb_Description *>malloc(sizeof(Evas_Smart_Cb_Description))             # <<<<<<<<<<<<<<
  *         if isinstance(name, unicode): name = PyUnicode_AsUTF8String(name)
  *         tmp.name = strdup(name)
  */
-  __pyx_v_tmp = ((Evas_Smart_Cb_Description *)malloc((sizeof(Evas_Smart_Cb_Description *))));
+  __pyx_v_tmp = ((Evas_Smart_Cb_Description *)malloc((sizeof(Evas_Smart_Cb_Description))));
 
   /* "efl/evas/efl.evas_object_smart.pxi":71
  *         cdef Evas_Smart_Cb_Description *tmp
- *         tmp = <Evas_Smart_Cb_Description *>malloc(sizeof(Evas_Smart_Cb_Description*))
+ *         tmp = <Evas_Smart_Cb_Description *>malloc(sizeof(Evas_Smart_Cb_Description))
  *         if isinstance(name, unicode): name = PyUnicode_AsUTF8String(name)             # <<<<<<<<<<<<<<
  *         tmp.name = strdup(name)
  *         if isinstance(types, unicode): types = PyUnicode_AsUTF8String(types)
@@ -83789,7 +83789,7 @@ static int __pyx_pf_3efl_4evas_18SmartCb
   }
 
   /* "efl/evas/efl.evas_object_smart.pxi":72
- *         tmp = <Evas_Smart_Cb_Description *>malloc(sizeof(Evas_Smart_Cb_Description*))
+ *         tmp = <Evas_Smart_Cb_Description *>malloc(sizeof(Evas_Smart_Cb_Description))
  *         if isinstance(name, unicode): name = PyUnicode_AsUTF8String(name)
  *         tmp.name = strdup(name)             # <<<<<<<<<<<<<<
  *         if isinstance(types, unicode): types = PyUnicode_AsUTF8String(types)
@@ -83838,7 +83838,7 @@ static int __pyx_pf_3efl_4evas_18SmartCb
  * 
  *     def __init__(self, name, types):             # <<<<<<<<<<<<<<
  *         cdef Evas_Smart_Cb_Description *tmp
- *         tmp = <Evas_Smart_Cb_Description *>malloc(sizeof(Evas_Smart_Cb_Description*))
+ *         tmp = <Evas_Smart_Cb_Description *>malloc(sizeof(Evas_Smart_Cb_Description))
  */
 
   /* function exit code */
Index: python-efl-1.25.0/efl/evas/efl.evas_object_smart.pxi
===================================================================
--- python-efl-1.25.0.orig/efl/evas/efl.evas_object_smart.pxi
+++ python-efl-1.25.0/efl/evas/efl.evas_object_smart.pxi
@@ -67,7 +67,7 @@ cdef class SmartCbDescription:
 
     def __init__(self, name, types):
         cdef Evas_Smart_Cb_Description *tmp
-        tmp = <Evas_Smart_Cb_Description *>malloc(sizeof(Evas_Smart_Cb_Description*))
+        tmp = <Evas_Smart_Cb_Description *>malloc(sizeof(Evas_Smart_Cb_Description))
         if isinstance(name, unicode): name = PyUnicode_AsUTF8String(name)
         tmp.name = strdup(name)
         if isinstance(types, unicode): types = PyUnicode_AsUTF8String(types)
