diff -rc alpine-2.21/alpine/imap.c alpine-2.21.pwdfilebug/alpine/imap.c
*** alpine-2.21/alpine/imap.c	2017-02-05 17:06:22.523218671 -0700
--- alpine-2.21.pwdfilebug/alpine/imap.c	2019-04-27 16:48:22.433116057 -0600
***************
*** 2636,2642 ****
         if(ps_global->pwdcert == NULL){
  	  q_status_message(SM_ORDER, 3, 3, "Attempting to encrypt password file");
  	  i = setup_pwdcert(&ps_global->pwdcert);
! 	  if(i == 0 && ps_global->pwdcert == NULL)
  	    ps_global->pwdcert = (void *) ALPINE_self_signed_certificate(NULL, 0, ps_global->pwdcertdir, MASTERNAME);
         }
         if(ps_global->pwdcert == NULL){	/* we tried but failed	*/
--- 2636,2642 ----
         if(ps_global->pwdcert == NULL){
  	  q_status_message(SM_ORDER, 3, 3, "Attempting to encrypt password file");
  	  i = setup_pwdcert(&ps_global->pwdcert);
! 	  if((i == 0 || i == -5) && ps_global->pwdcert == NULL)
  	    ps_global->pwdcert = (void *) ALPINE_self_signed_certificate(NULL, 0, ps_global->pwdcertdir, MASTERNAME);
         }
         if(ps_global->pwdcert == NULL){	/* we tried but failed	*/
