Index: sc-7.16/cmds.c
===================================================================
--- sc-7.16.orig/cmds.c
+++ sc-7.16/cmds.c
@@ -2563,7 +2563,7 @@ openfile(char *fname, int *rpid, int *rf
 	    (void) dup(pipefd[3]);	/* connect to second pipe */
 	}
 	(void) signal(SIGINT, SIG_DFL);	/* reset */
-	(void) execl("/bin/sh", "sh", "-c", efname, 0);
+	(void) execl("/bin/sh", "sh", "-c", efname, NULL);
 	exit (-127);
     } else {				/* else parent */
 	*rpid = pid;
