Index: postlogin.c
===================================================================
--- postlogin.c.orig	2012-04-10 16:09:50.440384915 +0200
+++ postlogin.c	2012-04-10 16:10:01.193753389 +0200
@@ -1053,6 +1053,11 @@
     {
       do_truncate = 1;
     }
+    if (new_file_fd >= 0) {
+      vsf_sysutil_fstat(new_file_fd, &s_p_statbuf);
+      if (vsf_sysutil_statbuf_is_regfile(s_p_statbuf))
+        vsf_sysutil_deactivate_noblock(new_file_fd);
+    } 
   }
   if (vsf_sysutil_retval_is_error(new_file_fd))
   {
