From 13f7016d219546c0d3367b5fa2bfa28f774b2761 Mon Sep 17 00:00:00 2001
From: "Albert Y. Shih" <ayshih@gmail.com>
Date: Mon, 19 Dec 2022 11:18:35 -0500
Subject: [PATCH 1/4] Ignore a deprecation warning in NumPy 1.24 that is
 triggered by scikit-image

Index: sunpy-4.1.0/setup.cfg
===================================================================
--- sunpy-4.1.0.orig/setup.cfg
+++ sunpy-4.1.0/setup.cfg
@@ -163,6 +163,8 @@ filterwarnings =
 	ignore:unable to download valid IERS file, using local IERS-B
 	ignore:.*will attempt to set the values inplace instead of always setting a new array:FutureWarning
 	ignore:'cgi' is deprecated and slated for removal in Python 3.13:DeprecationWarning:zeep.utils
+    # Deprecation warning added in NumPy 1.24, triggered by scikit-image
+    ignore:.*np.bool8.*is a deprecated alias.*:DeprecationWarning
 
 [pycodestyle]
 max_line_length = 110
