From 4b31846f3f90aa24f883ceed80e91f204c0a9389 Mon Sep 17 00:00:00 2001
From: Xiao Ni <xni@redhat.com>
Date: Fri, 29 Nov 2019 17:14:47 +0800
Subject: [PATCH] Remove unused code
Git-commit: 4b31846f3f90aa24f883ceed80e91f204c0a9389
Patch-mainline: mdadm-4.1+
References: jsc#SLE-10078, jsc#SLE-9348

Signed-off-by: Xiao Ni <xni@redhat.com>
Signed-off-by: Jes Sorensen <jsorensen@fb.com>
Signed-off-by: Coly Li <colyli@suse.de>

---
 platform-intel.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/platform-intel.h b/platform-intel.h
index 29c85f1..7cb370e 100644
--- a/platform-intel.h
+++ b/platform-intel.h
@@ -169,7 +169,6 @@ static inline int fls(int x)
 		r -= 2;
 	}
 	if (!(x & 0x80000000u)) {
-		x <<= 1;
 		r -= 1;
 	}
 	return r;
-- 
2.25.0

