From 37339bd211623574efd2f6f5e5133a3247e3bbde Mon Sep 17 00:00:00 2001
From: Martin Wilck <mwilck@suse.com>
Date: Thu, 27 Feb 2025 23:24:16 +0100
Subject: [PATCH 1001/1005] mdadm -D: display timeout status

---
 Detail.c | 2 ++
 md_p.h   | 1 +
 2 files changed, 3 insertions(+)

diff --git a/Detail.c b/Detail.c
index 5819ced..1e2e4c0 100644
--- a/Detail.c
+++ b/Detail.c
@@ -745,6 +745,8 @@ int Detail(char *dev, struct context *c)
 				    disk.raid_disk >= 0)
 					failed++;
 			}
+			if (disk.state & (1<<MD_DISK_TIMEOUT))
+				printf(" timeout");
 			if (disk.state & (1 << MD_DISK_ACTIVE))
 				printf(" active");
 			if (disk.state & (1 << MD_DISK_SYNC)) {
diff --git a/md_p.h b/md_p.h
index 358a28c..9c66dc6 100644
--- a/md_p.h
+++ b/md_p.h
@@ -90,6 +90,7 @@
 				   * dire need
 				   */
 #define	MD_DISK_FAILFAST	10 /* Fewer retries, more failures */
+#define	MD_DISK_TIMEOUT		11 /* disk is faulty due to timeout */
 
 #define MD_DISK_REPLACEMENT	17
 #define MD_DISK_JOURNAL		18 /* disk is used as the write journal in RAID-5/6 */
-- 
2.48.1

