Index: protobuf-21.4/src/google/protobuf/port_def.inc
===================================================================
--- protobuf-21.4.orig/src/google/protobuf/port_def.inc
+++ protobuf-21.4/src/google/protobuf/port_def.inc
@@ -652,7 +652,11 @@
 #define PROTOBUF_CONSTINIT [[clang::require_constant_initialization]]
 #define PROTOBUF_CONSTEXPR constexpr
 #elif PROTOBUF_GNUC_MIN(12, 2)
+#if PROTOBUF_CPLUSPLUS_MIN(201703L)
 #define PROTOBUF_CONSTINIT __constinit
+#else
+#define PROTOBUF_CONSTINIT
+#endif
 #define PROTOBUF_CONSTEXPR constexpr
 // MSVC 17 currently seems to raise an error about constant-initialized pointers.
 #elif defined(_MSC_VER) && _MSC_VER >= 1930
