--- src/main/org/exolab/adaptx/jaxp/transform/IdentityTransformer.java
+++ src/main/org/exolab/adaptx/jaxp/transform/IdentityTransformer.java
@@ -424,9 +424,9 @@
         }
         else {
             _outputProperties = new Properties();
-            Enumeration enum = properties.keys();
-            while (enum.hasMoreElements()) {
-                String name = (String)enum.nextElement();
+            Enumeration enum2 = properties.keys();
+            while (enum2.hasMoreElements()) {
+                String name = (String)enum2.nextElement();
                 String value = properties.getProperty(name);
                 _outputProperties.setProperty(name, value);
             }
--- src/main/org/exolab/adaptx/jaxp/transform/TransformerFactoryImpl.java
+++ src/main/org/exolab/adaptx/jaxp/transform/TransformerFactoryImpl.java
@@ -513,6 +513,9 @@
     } //-- newXMLFilter
 
 
+    //JDK5 build
+    public void setFeature(String name, boolean value)
+        throws javax.xml.transform.TransformerConfigurationException    {}
     
     
 } //-- TransformerFactoryImpl
--- src/main/org/exolab/adaptx/jaxp/transform/TransformerImpl.java
+++ src/main/org/exolab/adaptx/jaxp/transform/TransformerImpl.java
@@ -454,9 +454,9 @@
         else {
             _outputProperties = new Properties();
             XSLOutput output = new XSLOutput();
-            Enumeration enum = properties.keys();
-            while (enum.hasMoreElements()) {
-                String name = (String)enum.nextElement();
+            Enumeration enum2 = properties.keys();
+            while (enum2.hasMoreElements()) {
+                String name = (String)enum2.nextElement();
                 String value = properties.getProperty(name);
                 _outputProperties.setProperty(name, value);
                 try {
--- src/main/org/exolab/adaptx/xslt/RuleProcessor.java
+++ src/main/org/exolab/adaptx/xslt/RuleProcessor.java
@@ -262,7 +262,7 @@
         
         ResultFormatter rf = ps.getResultFormatter();
             
-        Enumeration enum = null;
+        Enumeration enum2 = null;
         //-- copy top-level declared namespaces
         /*
         Hashtable namespaces = stylesheet.getNamespaces();
@@ -304,9 +304,9 @@
         
         // Process top-level parameters
         ScopedVariableSet vars = ps.getVariables();
-        enum = stylesheet.getParameters();
-        while (enum.hasMoreElements()) {
-            Param param = (Param)enum.nextElement();
+        enum2 = stylesheet.getParameters();
+        while (enum2.hasMoreElements()) {
+            Param param = (Param)enum2.nextElement();
             //-- first check passed in parameters
             String value = getParameter(param.getName());
             if (value != null) {
@@ -322,9 +322,9 @@
         }
         
         //-- Process top-level variables
-        enum = stylesheet.getVariables();
-        while (enum.hasMoreElements()) {
-            Variable variable = (Variable)enum.nextElement();
+        enum2 = stylesheet.getVariables();
+        while (enum2.hasMoreElements()) {
+            Variable variable = (Variable)enum2.nextElement();
             XPathResult value = processVariable(variable, ps);
             if (value != null) {
                 vars.setVariable(variable.getName(), value);
--- src/main/org/exolab/adaptx/xslt/XSLTProcessor.java
+++ src/main/org/exolab/adaptx/xslt/XSLTProcessor.java
@@ -1266,9 +1266,9 @@
 	 * @param ruleProcessor the RuleProcessor to copy parameters to
 	**/
 	private void copyParams(RuleProcessor ruleProcessor) {
-	    Enumeration enum = params.keys();
-	    while (enum.hasMoreElements()) {
-	        String name = (String)enum.nextElement();
+	    Enumeration enum2 = params.keys();
+	    while (enum2.hasMoreElements()) {
+	        String name = (String)enum2.nextElement();
 	        ruleProcessor.setParameter(name, params.getProperty(name));
 	    }
 	} //-- copyParams
@@ -1635,4 +1635,4 @@
   } //-- readXSLStylesheet
   
  
-} //-- XSLProcessor
\ No newline at end of file
+} //-- XSLProcessor
--- src/main/org/exolab/adaptx/xslt/XSLTStylesheet.java
+++ src/main/org/exolab/adaptx/xslt/XSLTStylesheet.java
@@ -569,10 +569,10 @@
 	    if (namespace == null) return null;
 	    String qns = (String) quotedns.get(namespace);
 	    if (qns != null) {
-	        Enumeration enum = namespaces.keys();
+	        Enumeration enum2 = namespaces.keys();
     	    String key;
-	        while (enum.hasMoreElements()) {
-	            key = (String)enum.nextElement();
+	        while (enum2.hasMoreElements()) {
+	            key = (String)enum2.nextElement();
 	            if (qns.equals(namespaces.get(key)))
 	                return key;
 	        }
@@ -1094,16 +1094,16 @@
         
         // combine global variables and parameters
         try {
-            Enumeration enum = xsl.getVariables();
-            while (enum.hasMoreElements()) {
-                Variable v = (Variable) enum.nextElement();
+            Enumeration enum2 = xsl.getVariables();
+            while (enum2.hasMoreElements()) {
+                Variable v = (Variable) enum2.nextElement();
                 Variable tmp = getVariable(v.getName());
                 if (tmp != null) variables.remove(tmp);
                 addVariable(v, true);
             }
-            enum = xsl.getParameters();
-            while (enum.hasMoreElements()) {
-                Param p = (Param) enum.nextElement();
+            enum2 = xsl.getParameters();
+            while (enum2.hasMoreElements()) {
+                Param p = (Param) enum2.nextElement();
                 Param tmp = getParameter(p.getName());
                 if (tmp != null) parameters.remove(tmp);
                 addParam(p);
--- src/main/org/exolab/adaptx/xslt/handlers/ResultHandlerAdapter2.java
+++ src/main/org/exolab/adaptx/xslt/handlers/ResultHandlerAdapter2.java
@@ -200,9 +200,9 @@
             _handler.endElement(uri, name, name);
         }
         
-        Enumeration enum = _namespaces.getLocalNamespacePrefixes();
-        while (enum.hasMoreElements()) {
-            _handler.endPrefixMapping((String)enum.nextElement());
+        Enumeration enum2 = _namespaces.getLocalNamespacePrefixes();
+        while (enum2.hasMoreElements()) {
+            _handler.endPrefixMapping((String)enum2.nextElement());
         }
         _namespaces = _namespaces.getParent();
         
