--- src/doc/style/exolab.xsl.sav	2006-06-19 18:25:07.000000000 +0200
+++ src/doc/style/exolab.xsl	2006-06-19 18:28:57.000000000 +0200
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="ISO-8859-1"?>
-<xsl:stylesheet version="1.0">
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
 
   <xsl:output method="html" indent="no"/>
 
--- src/doc/style/keywords.xsl.sav	2006-06-19 18:30:23.000000000 +0200
+++ src/doc/style/keywords.xsl	2006-06-19 18:30:38.000000000 +0200
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="ISO-8859-1"?>
-<xsl:stylesheet version="1.0">
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
   <xsl:template match="keywords">
     <xsl:element name="meta">
       <xsl:attribute name="http-equiv">Keywords</xsl:attribute>
--- src/doc/style/leftNav.xsl.sav	2006-06-19 18:31:37.000000000 +0200
+++ src/doc/style/leftNav.xsl	2006-06-19 18:31:51.000000000 +0200
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="ISO-8859-1"?>
-<xsl:stylesheet version="1.0">
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
 
   <xsl:template match="$project/navSections">
     <xsl:for-each select="section">
--- src/doc/style/topNav.xsl.sav	2006-06-19 18:33:16.000000000 +0200
+++ src/doc/style/topNav.xsl	2006-06-19 18:33:32.000000000 +0200
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="ISO-8859-1"?>
-<xsl:stylesheet version="1.0">
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
 
 <xsl:variable name="leftTopNavPos" select="5"/>
 <xsl:template match="$project/topNav">
--- src/doc/style/searchForm.xsl.sav	2006-06-19 18:32:38.000000000 +0200
+++ src/doc/style/searchForm.xsl	2006-06-19 18:32:53.000000000 +0200
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="ISO-8859-1"?>
-<xsl:stylesheet version="1.0">
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
 
 <xsl:template match="searchForm">
   <table width="95" border="0" cellpadding="0" cellspacing="0">
