diff -ruN gcad3d-2.35+git20170419.c50f21f.orig/src/APP/doxy_help1.c gcad3d-2.35+git20170419.c50f21f/src/APP/doxy_help1.c
--- gcad3d-2.35+git20170419.c50f21f.orig/src/APP/doxy_help1.c	2017-04-19 15:28:44.000000000 +0300
+++ gcad3d-2.35+git20170419.c50f21f/src/APP/doxy_help1.c	2017-04-19 18:10:52.573782794 +0300
@@ -652,7 +652,7 @@
   if ((fpi = fopen (fnam, "rb")) == NULL) {
     TX_Error ("MEM_get_file: Error open %s",fnam);
     *fSiz = -2;
-    return;
+    return 42;
   }
 
   fread (txbuf, *fSiz, 1, fpi);
@@ -663,7 +663,7 @@
 
     // printf(" _get_file|%s|\n",txbuf);
 
-  return;
+  return 42;
 
 }
 
diff -ruN gcad3d-2.35+git20170419.c50f21f.orig/src/gui_gtk3/gtk_base.c gcad3d-2.35+git20170419.c50f21f/src/gui_gtk3/gtk_base.c
--- gcad3d-2.35+git20170419.c50f21f.orig/src/gui_gtk3/gtk_base.c	2017-04-19 15:28:44.000000000 +0300
+++ gcad3d-2.35+git20170419.c50f21f/src/gui_gtk3/gtk_base.c	2017-04-19 18:08:45.834592903 +0300
@@ -2101,7 +2101,7 @@
   Obj_Unknown   *go;
 
   go = GUI_obj_pos (mo);
-  if(!go) return;
+  if(!go) return 42;
 
   return (go->gio_src);
 
diff -ruN gcad3d-2.35+git20170419.c50f21f.orig/src/gui_gtk3/gtk_menu.c gcad3d-2.35+git20170419.c50f21f/src/gui_gtk3/gtk_menu.c
--- gcad3d-2.35+git20170419.c50f21f.orig/src/gui_gtk3/gtk_menu.c	2017-04-19 15:28:44.000000000 +0300
+++ gcad3d-2.35+git20170419.c50f21f/src/gui_gtk3/gtk_menu.c	2017-04-19 18:09:23.898325094 +0300
@@ -200,7 +200,7 @@
   void   *go, *wo;
 
   go = GUI_obj_pos (mo);
-  if(!go) return;
+  if(!go) return 42;
   wo = ((Obj_Unknown*)go)->widget;
 
   return gtk_check_menu_item_get_active (wo);
diff -ruN gcad3d-2.35+git20170419.c50f21f.orig/src/gui_gtk3/gtk_multichoice.c gcad3d-2.35+git20170419.c50f21f/src/gui_gtk3/gtk_multichoice.c
--- gcad3d-2.35+git20170419.c50f21f.orig/src/gui_gtk3/gtk_multichoice.c	2017-04-19 15:28:44.000000000 +0300
+++ gcad3d-2.35+git20170419.c50f21f/src/gui_gtk3/gtk_multichoice.c	2017-04-19 18:10:07.230046578 +0300
@@ -257,7 +257,7 @@
   // gtk_widget_set_size_request (GTK_WIDGET(menu), 144, 0);
   // gtk_widget_set_hexpand (menu, FALSE);
 
-  return;
+  return 42;
 
 }
 
@@ -880,7 +880,7 @@
   void   *go, *wo;
 
   go = GUI_obj_pos (mo);
-  if(!go) return;
+  if(!go) return 42;
   wo = ((Obj_Unknown*)go)->widget;
 
   return gtk_toggle_button_get_active (wo);
diff -ruN gcad3d-2.35+git20170419.c50f21f.orig/src/xa/xa_ed.c gcad3d-2.35+git20170419.c50f21f/src/xa/xa_ed.c
--- gcad3d-2.35+git20170419.c50f21f.orig/src/xa/xa_ed.c	2017-04-19 15:28:44.000000000 +0300
+++ gcad3d-2.35+git20170419.c50f21f/src/xa/xa_ed.c	2017-04-19 18:11:45.785506139 +0300
@@ -2741,7 +2741,7 @@
   long   il;
 
   ED_cpos = UTF_GetPosLnr (&il, ED_lnr_act);
-
+return 0;  
 }
 
 
