From fe9f359db1cc0f59e4703c69c97ed17f88916f5a Mon Sep 17 00:00:00 2001
From: Ben Greiner <code@bnavigator.de>
Date: Fri, 29 Mar 2024 11:26:58 +0100
Subject: [PATCH] Bump pylint to 3.1

---
 pyproject.toml | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 4665dcbe..89ffab66 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -33,7 +33,7 @@ all = [
     "pycodestyle>=2.11.0,<2.12.0",
     "pydocstyle>=6.3.0,<6.4.0",
     "pyflakes>=3.2.0,<3.3.0",
-    "pylint>=2.5.0,<3.1",
+    "pylint>=3.1,<4",
     "rope>=1.11.0",
     "yapf>=0.33.0",
     "whatthepatch>=1.0.2,<2.0.0"
@@ -44,12 +44,12 @@ mccabe = ["mccabe>=0.7.0,<0.8.0"]
 pycodestyle = ["pycodestyle>=2.11.0,<2.12.0"]
 pydocstyle = ["pydocstyle>=6.3.0,<6.4.0"]
 pyflakes = ["pyflakes>=3.2.0,<3.3.0"]
-pylint = ["pylint>=2.5.0,<3.1"]
+pylint = ["pylint>=3.1,<4"]
 rope = ["rope>=1.11.0"]
 yapf = ["yapf>=0.33.0", "whatthepatch>=1.0.2,<2.0.0"]
 websockets = ["websockets>=10.3"]
 test = [
-    "pylint>=2.5.0,<3.1",
+    "pylint>=3.1,<4",
     "pytest",
     "pytest-cov",
     "coverage",
