--- configure.in.orig
+++ configure.in
@@ -27,7 +27,9 @@ AM_CONDITIONAL(FREEBSD, test x$os = xfre
 dnl Checks for programs.
 AC_PREREQ(2.52)
 AC_PROG_AWK
-AC_PROG_CC
+AC_PROG_CC_STDC
+AC_USE_SYSTEM_EXTENSIONS
+AC_SYS_LARGEFILE
 AC_LIBLTDL_CONVENIENCE
 AM_PROG_LIBTOOL
 AC_SUBST(LIBTOOL_DEPS)  
@@ -61,7 +63,9 @@ AC_SUBST(EXPAT_LIBS)
 
 dnl libcurl check
 AC_CHECK_LIB(curl, curl_global_init,
-	[CURL_LIBS="-lcurl"], [
+	[CURL_LIBS="-lcurl"
+	AC_CHECK_HEADERS([curl/types.h])
+	], [
 	echo "*"
 	echo "*  libcurl is needed to build this library. It is either not"
 	echo "*  installed on your system or it is too old."
--- examples/protocol.cpp.orig
+++ examples/protocol.cpp
@@ -6,13 +6,17 @@
    No rights reserved.
 
 -------------------------------------------------------------------*/
+
+#include "config.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <cstring>
 #include <map>
 #include <expat.h>
 #include <curl/curl.h>
+#ifdef HAVE_CURL_TYPES_H
 #include <curl/types.h>
+#endif
 #include <curl/easy.h>
 
 using namespace std;
