From fb1708b95898aa626c7e2a465cd336021487199c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andrius=20=C5=A0tikonas?= <andrius@stikonas.eu>
Date: Tue, 15 Feb 2022 19:06:12 +0000
Subject: [PATCH] Fix a typo in definition of MiB constant.

---
 src/util/externalcommandhelper.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/util/externalcommandhelper.h b/src/util/externalcommandhelper.h
index de10de7..bb1995f 100644
--- a/src/util/externalcommandhelper.h
+++ b/src/util/externalcommandhelper.h
@@ -21,7 +21,7 @@
 #include <QString>
 
 class QDBusServiceWatcher;
-constexpr qint64 MiB = 1 << 30;
+constexpr qint64 MiB = 1 << 20;
 
 class ExternalCommandHelper : public QObject, public QDBusContext
 {
-- 
2.35.1

