Index: gstreamer-1.28.8/tests/check/meson.build
===================================================================
--- gstreamer-1.28.8.orig/tests/check/meson.build
+++ gstreamer-1.28.8/tests/check/meson.build
@@ -134,7 +134,7 @@ if build_gstgl
   test_defines += ['-DTEST_GST_GL_ABI_CHECK']
 endif
 
-test_deps = [gst_dep, gst_base_dep, gst_net_dep, gst_check_dep, audio_dep,
+test_deps = [gst_dep, gst_base_dep, gst_net_dep, gst_check_dep, wayland_client_dep, audio_dep,
   video_dep, pbutils_dep, rtp_dep, rtsp_dep, tag_dep, allocators_dep, app_dep,
   fft_dep, riff_dep, sdp_dep, gio_dep, valgrind_dep]
 
