Workaround incomplete gstreamer-gl.pc, missing dependency on wayland-client.
This patch is not suitable for upstream, as it does not cater for the possible
alternative implementations being built (e.g. without wayland support)

https://gitlab.freedesktop.org/gstreamer/gst-plugins-base/issues/735


Index: gst-plugins-base-1.16.2/pkgconfig/gstreamer-gl.pc.in
===================================================================
--- gst-plugins-base-1.16.2.orig/pkgconfig/gstreamer-gl.pc.in
+++ gst-plugins-base-1.16.2/pkgconfig/gstreamer-gl.pc.in
@@ -10,7 +10,7 @@ gl_apis=@GL_APIS@
 Name: GStreamer OpenGL Plugins Libraries
 Description: Streaming media framework, OpenGL plugins libraries
 Version: @VERSION@
-Requires: gstreamer-video-@GST_API_VERSION@ gstreamer-base-@GST_API_VERSION@ gstreamer-@GST_API_VERSION@
+Requires: gstreamer-video-@GST_API_VERSION@ gstreamer-base-@GST_API_VERSION@ gstreamer-@GST_API_VERSION@ wayland-client
 
 Libs: -L${libdir} -lgstgl-@GST_API_VERSION@
 Cflags: -I${includedir} -I${libdir}/gstreamer-@GST_API_VERSION@/include
