--- Makefile
+++ Makefile
@@ -6,7 +6,7 @@
 #
 
 
-DATA_DIR=`ispell -vv | grep LIBDIR | sed s/"LIBDIR = "// | sed s/\"//g`
+DATA_DIR=`ispell -vv | grep LIBDIR | sed s/".*LIBDIR = "// | sed s/\"//g`
 
 
 all:
@@ -23,9 +23,9 @@
 
 install:
 	@echo -n "Copying bulgarian.hash and bulgarian.aff to "
-	@echo ${DATA_DIR}
-	@install -D -m 644 data/bulgarian.hash ${DATA_DIR}/bulgarian.hash
-	@install -D -m 644 data/bulgarian.aff ${DATA_DIR}/bulgarian.aff
+	@echo ${DESTDIR}${DATA_DIR}
+	@install -D -m 644 data/bulgarian.hash ${DESTDIR}${DATA_DIR}/bulgarian.hash
+	@install -D -m 644 data/bulgarian.aff ${DESTDIR}${DATA_DIR}/bulgarian.aff
 
 	@echo ""
 	@echo "It seems that everything is OK. To use type following command:"
