diff -Nru azure-cognitiveservices-search-entitysearch-1.0.0.orig/setup.py azure-cognitiveservices-search-entitysearch-1.0.0/setup.py
--- azure-cognitiveservices-search-entitysearch-1.0.0.orig/setup.py	2018-05-02 19:33:22.000000000 +0200
+++ azure-cognitiveservices-search-entitysearch-1.0.0/setup.py	2018-09-03 11:45:07.874496646 +0200
@@ -78,7 +78,7 @@
     packages=find_packages(exclude=["tests"]),
     install_requires=[
         'msrest>=0.4.28,<2.0.0',
-        'azure-common~=1.1',
+        'azure-common>=1.1',
     ],
     cmdclass=cmdclass
 )
