--- efont-unicode-bdf-0.4.2-src/tools/hex2bdf.in_orig	2019-11-21 19:52:54.341747143 +0100
+++ efont-unicode-bdf-0.4.2-src/tools/hex2bdf.in	2019-11-21 19:53:57.469747085 +0100
@@ -14,7 +14,7 @@
 
 
 while (<>) { $glyph{$1} = $2 if /(.{4,}):(.+)\n/; }
-@chars = sort keys %glyph; $[ = 1;
+@chars = sort keys %glyph;
 #dbmopen (%charname, "/usr/share/unicode/unidata/charname.db", 0);
 
 print "STARTFONT 2.1\n";
@@ -55,7 +55,7 @@
 FONT_DESCENT 2
 DEFAULT_CHAR $default_char
 ENDPROPERTIES
-CHARS $#chars\n";
+CHARS " . scalar(@chars) . "\n";
 
 foreach $character (@chars) {
   $encoding = hex($character); $glyph = $glyph{$character};
