Index: libdc1394-2.2.3/dc1394/conversions.c
===================================================================
--- libdc1394-2.2.3.orig/dc1394/conversions.c
+++ libdc1394-2.2.3/dc1394/conversions.c
@@ -20,13 +20,12 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#define _XOPEN_SOURCE
 #include <string.h>
 #include <stdlib.h>
+#include <unistd.h>
 #include "conversions.h"
 
-// this should disappear...
-extern void swab();
-
 /**********************************************************************
  *
  *  CONVERSION FUNCTIONS TO YUV422
