Index: init.h
===================================================================
---
 init.h |    1 +
 1 file changed, 1 insertion(+)

--- init.h
+++ init.h	2020-01-14 13:28:27.408149074 +0000
@@ -4423,6 +4423,7 @@ struct option_t MuttVars[] = {
   ** printing, or replying to messages.
   */
   { "wrap",             DT_NUM,  R_PAGER, {.p=&Wrap}, {.l=0} },
+  { "wrapcolumn",       DT_SYN,  R_NONE, {.p="wrap"}, {.p=0} },
   /*
   ** .pp
   ** When set to a positive value, mutt will wrap text at $$wrap characters.
