---
 init.h |    7 +++++++
 mutt.h |    1 +
 send.c |    9 ++++++---
 3 files changed, 14 insertions(+), 3 deletions(-)

--- init.h
+++ init.h	2020-01-14 13:12:26.725982952 +0000
@@ -3155,6 +3155,13 @@ struct option_t MuttVars[] = {
   ** .pp
   ** Also see the $$force_name variable.
   */
+  { "send_group_reply_to",	DT_BOOL, R_NONE, {.l=OPTSENDGROUPREPLYTO}, {.l=0} },
+  /*
+  ** .pp
+  ** This variable controls how group replies are done.
+  ** When set, all recepients listet in "To:" are set in the
+  ** "To:" header again, else in the "CC", which is the default.
+  */
   { "score", 		DT_BOOL, R_NONE, {.l=OPTSCORE}, {.l=1} },
   /*
   ** .pp
--- mutt.h
+++ mutt.h	2020-01-14 13:07:33.727426389 +0000
@@ -515,6 +515,7 @@ enum
   OPTSAVEADDRESS,
   OPTSAVEEMPTY,
   OPTSAVENAME,
+  OPTSENDGROUPREPLYTO,
   OPTSCORE,
 #ifdef USE_SIDEBAR
   OPTSIDEBAR,
--- send.c
+++ send.c	2020-01-14 13:07:33.727426389 +0000
@@ -681,9 +681,12 @@ int mutt_fetch_recips (ENVELOPE *out, EN
         (!in->mail_followup_to || hmfupto != MUTT_YES))
     {
       /* if (!mutt_addr_is_user(in->to)) */
-      if (flags & SENDGROUPREPLY)
-        rfc822_append (&out->cc, in->to, 1);
-      else
+      if (flags & SENDGROUPREPLY) {
+	if (option (OPTSENDGROUPREPLYTO))
+          rfc822_append (&out->to, in->to, 1);
+	else
+          rfc822_append (&out->cc, in->to, 1);
+      } else
         rfc822_append (&out->to, in->to, 1);
       rfc822_append (&out->cc, in->cc, 1);
     }
