diff -Nru s2n-tls-1.3.7.orig/CMakeLists.txt s2n-tls-1.3.7/CMakeLists.txt
--- s2n-tls-1.3.7.orig/CMakeLists.txt	2022-02-16 18:38:44.000000000 +0000
+++ s2n-tls-1.3.7/CMakeLists.txt	2022-03-01 12:08:54.471824612 +0000
@@ -636,14 +636,14 @@
 endif()
 
 install(EXPORT "${PROJECT_NAME}-targets"
-        DESTINATION "${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake/${TARGET_DIR}"
+        DESTINATION "${LIB_INSTALL_DIR}/cmake/${CMAKE_PROJECT_NAME}/${TARGET_DIR}/"
         NAMESPACE AWS::
         COMPONENT Development)
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake"
-        DESTINATION "${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake/"
+        DESTINATION "${LIB_INSTALL_DIR}/cmake/${CMAKE_PROJECT_NAME}"
         COMPONENT Development)
 
 install(FILES "cmake/modules/Findcrypto.cmake"
-        DESTINATION "${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake/modules/"
+        DESTINATION "${LIB_INSTALL_DIR}/cmake/${CMAKE_PROJECT_NAME}/modules"
         COMPONENT Development)
