diff -Ndur xlogin-0.2/xtelnet.script xlogin-0.2-fix-bashisms/xtelnet.script
--- xlogin-0.2/xtelnet.script	2006-08-21 20:33:49.000000000 +0300
+++ xlogin-0.2-fix-bashisms/xtelnet.script	2014-11-19 01:02:18.361062183 +0200
@@ -24,7 +24,8 @@
 #
 usage ()
 {
-    echo -e "$0: Usage\n  $0 [-iconic] [-a] [-q] host [host ...]" 1>&2
+    echo "$0: Usage" >&2
+    echo "  $0 [-iconic] [-a] [-q] host [host ...]" >&2
     exit 1
 }
 #
@@ -43,11 +44,13 @@
 #
 # And start the xterm
 #
-test -z "$quiet" && echo -e "$0:
+if [ -z "$quiet" ]; then
+    echo "$0:
   If you have problems with the backspace key you should execute
   the command line:
-\tstty erase '$kbs'
+	stty erase '$kbs'
   after successful login."
+fi
 for h in $hosts; do
     ${term} -T "telnet ${h}" -n ${h} ${iconic} ${tn} -tm "erase ${kbs}" \
     -e ${telnet} ${h} &
