Avoid colorization of test output in obs runners

Index: gevent-24.2.1/src/gevent/testing/util.py
===================================================================
--- gevent-24.2.1.orig/src/gevent/testing/util.py
+++ gevent-24.2.1/src/gevent/testing/util.py
@@ -98,6 +98,8 @@ def _color(what):
     return _color_code(_colorscheme[what])
 
 def _colorize(what, message, normal='normal'):
+    if os.environ.get("TEST_NOCOLOR", False):
+        return message
     return _color(what) + message + _color(normal)
 
 def log(message, *args, **kwargs):
