Subject: init: Fix link to renamed kernel documentation
From: Richard W.M. Jones rjones@redhat.com Mon Nov 13 10:13:26 2023 +0000
Date: Mon Nov 13 10:13:26 2023 +0000:
Git: c8e79b7d62730f850f9e5d46c3747f414198b902


diff --git a/init/init.c b/init/init.c
index bc28c69..4cc72f7 100644
--- a/init/init.c
+++ b/init/init.c
@@ -279,8 +279,8 @@ main ()
     fprintf (stderr, "supermin: deleting initramfs files\n");
   delete_initramfs_files ();
 
-  /* Note that pivot_root won't work.  See the note in
-   * Documentation/filesystems/ramfs-rootfs-initramfs.txt
+  /* Note that pivot_root won't work.  See the note in Linux
+   * Documentation/filesystems/ramfs-rootfs-initramfs.rst
    */
   if (!quiet)
     fprintf (stderr, "supermin: chroot\n");
