diff -urEbw gpicview-0.2.5/src/main-win.c gpicview-0.2.5.new/src/main-win.c
--- gpicview-0.2.5/src/main-win.c	2016-02-20 23:38:31.000000000 +0100
+++ gpicview-0.2.5.new/src/main-win.c	2019-10-11 21:57:48.501711945 +0200
@@ -378,7 +378,7 @@
         image_list_sort_by_name( mw->img_list, GTK_SORT_DESCENDING );
         if (image_list_get_first(mw->img_list))
             main_win_open(mw, image_list_get_current_file_path(mw->img_list), zoom);
-        return;
+        return FALSE;
     }
 
 
