--- a/configure.tests/quazip/main.cpp
+++ b/configure.tests/quazip/main.cpp
@@ -33,8 +33,8 @@
  *   along with this program.  If not, see <http://www.gnu.org/licenses/>. *
  ***************************************************************************/
 
-#include "quazip/quazip.h"
-#include "quazip/quazipfile.h"
+#include "quazip5/quazip.h"
+#include "quazip5/quazipfile.h"
 #include <QFile>
 
 int main()
--- a/qonf/test.rb
+++ b/qonf/test.rb
@@ -103,7 +103,7 @@ class Test
                          qmakeLine = "'LIBS += #{extraLib}'"
                          qmakeLine += " 'INCLUDEPATH += #{extraInclude}'"
                       else
-                         extraLib = "-lquazip-qt5"
+                         extraLib = "-lquazip5"
                          qmakeLine = "'LIBS += #{extraLib}'"
                       end
                    else
@@ -176,7 +176,7 @@ class Test
         if conf.hasArgument?("with-quazip")
            config.addLib("-lquazip")
         else
-           config.addLib("-lquazip-qt5")
+           config.addLib("-lquazip5")
         end
 
         # config.addLib("-lquazip-qt5")
--- a/src/libtupi/tuppackagehandler.h
+++ b/src/libtupi/tuppackagehandler.h
@@ -38,8 +38,8 @@
 
 #include "tglobal.h"
 #include "tapplicationproperties.h"
-#include "quazip/quazip.h"
-#include "quazip/quazipfile.h"
+#include "quazip5/quazip.h"
+#include "quazip5/quazipfile.h"
 
 // SQA: Pending to enable in the future
 // #include "quazip/JlCompress.h"
