--- src/sisctrl_gui.c.orig
+++ src/sisctrl_gui.c
@@ -5885,8 +5885,8 @@ vi_layout_configure(GtkWidget *widget, G
 {
    /*fprintf(stderr, "conf\n");*/
    if(gl.noxvdemo) return TRUE;
-   if(gui.notebook_page != SIS_PAGE_VIDEO) return;
-   if(!gui.xvwin) return;
+   if(gui.notebook_page != SIS_PAGE_VIDEO) return TRUE;
+   if(!gui.xvwin) return TRUE;
    if(!gui.xvinit) return TRUE;
    resizexvwindow();
    showxvimage();
--- src/sisctrl.c.orig
+++ src/sisctrl.c
@@ -419,7 +419,7 @@ newsetrealxvval(unsigned int cmd, Atom a
 
 /* Get/Set gamma brightness. Does proper unlocking/locking */
 
-static Bool
+static void
 sd_writegammabr(unsigned int br, unsigned int bg, unsigned int bb)
 {
     if(UnlockSiSDirect(1)) {
@@ -433,7 +433,7 @@ sd_writegammabr(unsigned int br, unsigne
     } else fprintf(stderr, xverrorstring);
 }
 
-static Bool
+static void
 sd_writenewgammabr(unsigned int br, unsigned int bg, unsigned int bb,
 		   unsigned int cr, unsigned int cg, unsigned int cb)
 {
@@ -1169,6 +1169,7 @@ checkmodeforgivencrt2(int index, unsigne
 #else
    return 0x80;
 #endif
+   return 0x80;
 }
 
 static int
--- configure.ac.orig
+++ configure.ac
@@ -9,7 +9,10 @@ AC_CONFIG_HEADER([config.h])
 AM_INIT_AUTOMAKE
 
 # Checks for programs.
-AC_PROG_CC
+AC_PROG_CC_STDC
+AC_USE_SYSTEM_EXTENSIONS
+AC_SYS_LARGEFILE
+AM_PROG_CC_C_O
 
 # Checks for libraries.
 
@@ -211,6 +214,8 @@ AM_PATH_GTK_2_0(2.4.0,,AC_MSG_ERROR([
 *** GTK+ 2.4 or later is required to build sisctrl; please make sure you have the GTK+
 *** development headers installed.]))
 
+AC_SEARCH_LIBS(sin, m)
+
 AC_OUTPUT([Makefile],)
 
 
