Index: mitmproxy-5.1.1/setup.py
===================================================================
--- mitmproxy-5.1.1.orig/setup.py
+++ mitmproxy-5.1.1/setup.py
@@ -62,29 +62,29 @@ setup(
     # https://packaging.python.org/en/latest/requirements/#install-requires
     # It is not considered best practice to use install_requires to pin dependencies to specific versions.
     install_requires=[
-        "blinker>=1.4, <1.5",
-        "Brotli>=1.0,<1.1",
+        "blinker>=1.4",
+        "Brotli>=1.0",
         "certifi>=2019.9.11",  # no semver here - this should always be on the last release!
-        "click>=7.0,<8",
-        "cryptography>=2.9,<3.0",
-        "flask>=1.1.1,<1.2",
-        "h2>=3.2.0,<4",
-        "hyperframe>=5.1.0,<6",
-        "kaitaistruct>=0.7,<0.9",
-        "ldap3>=2.6.1,<2.8",
-        "passlib>=1.6.5, <1.8",
-        "protobuf>=3.6.0, <3.12",
-        "pyasn1>=0.3.1,<0.5",
-        "pyOpenSSL>=19.1.0,<19.2",
-        "pyparsing>=2.4.2,<2.5",
-        "pyperclip>=1.6.0,<1.9",
-        "ruamel.yaml>=0.16,<0.17",
-        "sortedcontainers>=2.1.0,<2.2",
-        "tornado>=4.3,<7",
-        "urwid>=2.1.0,<2.2",
-        "wsproto>=0.14,<0.16",
-        "publicsuffix2>=2.20190812,<3",
-        "zstandard>=0.11,<0.14",
+        "click>=7.0",
+        "cryptography>=2.9",
+        "flask>=1.1.1",
+        "h2>=3.2.0",
+        "hyperframe>=5.1.0",
+        "kaitaistruct>=0.7",
+        "ldap3>=2.6.1",
+        "passlib>=1.6.5",
+        "protobuf>=3.6.0",
+        "pyasn1>=0.3.1",
+        "pyOpenSSL>=19.1.0",
+        "pyparsing>=2.4.2",
+        "pyperclip>=1.6.0",
+        "ruamel.yaml>=0.16",
+        "sortedcontainers>=2.1.0",
+        "tornado>=4.3",
+        "urwid>=2.1.0",
+        "wsproto>=0.14",
+        "publicsuffix2>=2.20190812",
+        "zstandard>=0.11",
     ],
     extras_require={
         ':sys_platform == "win32"': [
@@ -104,7 +104,7 @@ setup(
             "tox>=3.5,<3.15",
         ],
         'examples': [
-            "beautifulsoup4>=4.4.1,<4.9"
+            "beautifulsoup4>=4.4.1"
         ]
     }
 )
