Index: b/src/calibre/headless/CMakeLists.txt
===================================================================
--- a/src/calibre/headless/CMakeLists.txt
+++ b/src/calibre/headless/CMakeLists.txt
@@ -2,6 +2,11 @@ cmake_minimum_required(VERSION 3.21)
 project(headless)
 set(CMAKE_AUTOMOC ON)
 find_package(Qt6Gui REQUIRED)
+if(Qt6Gui_VERSION VERSION_GREATER_EQUAL "6.10.0")
+    set(QT_NO_PRIVATE_MODULE_WARNING ON)
+    find_package(Qt6GuiPrivate REQUIRED NO_MODULE)
+    find_package(Qt6CorePrivate REQUIRED NO_MODULE)
+endif()
 add_library(headless MODULE main.cpp headless_backingstore.cpp headless_integration.cpp)
 set_property(TARGET headless PROPERTY QT_PLUGIN_TYPE "platforms")
 set_property(TARGET headless PROPERTY QT_PLUGIN_CLASS_NAME "HeadlessIntegrationPlugin")
