--- ctlwords.c
+++ ctlwords.c
@@ -23,6 +23,7 @@
 #endif
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 
 main(int argc, char *argv[])
--- image/cmuwmraster.c
+++ image/cmuwmraster.c
@@ -108,7 +108,7 @@
       {
 	  fprintf(stderr,"CMU WM raster %s is of depth %d, must be 1",
 		  name,
-		  (int) header.depth);
+		  (int) memToVal(header.depth, sizeof(short)));
 	  return(NULL);
       }
 
--- image/image.h
+++ image/image.h
@@ -10,6 +10,7 @@
 
 #include "copyright.h"
 #include <stdio.h>
+#include <stdlib.h>
 
 #if defined(SYSV) || defined(VMS) || 1
 #include <string.h>
--- mgp.c
+++ mgp.c
@@ -494,7 +494,7 @@
 	char *childdebug;
 	char *convdb[][3] = {{ "jpg", "cjpeg", "djpeg" },
 				{ "png", "pnmtopng", "pngtopnm" },
-				{ NULL, NULL, NULL }};
+				{ { 0 }, { 0 }, { 0 } }};
 	int inum = 0;
 
 	/* check image type */
--- print.c
+++ print.c
@@ -1242,7 +1242,7 @@
 
 	paintit = (painticon || colorps);
 
-	switch ((int)tp->font) {	/*XXX*/
+	switch ((long int)tp->font) {	/*XXX*/
 	case 0:
 		/* XXX: image is not supported yet */
 		break;
@@ -1599,7 +1599,7 @@
 	textpool[ntextpool].xoffset = offset;
 	textpool[ntextpool].xsiz = char_size[0];
 	textpool[ntextpool].size = fontsize;
-	textpool[ntextpool].font = (struct fontmap *)icon;	/*XXX*/
+	textpool[ntextpool].font = (struct fontmap *)(long int )icon;	/*XXX*/
 	textpool[ntextpool].text = NULL;
 	textpool[ntextpool].fore = color;
 	textpool[ntextpool].back = back;	/*XXX*/
