--- mp3_check-1.98/support_functions.h.orig	2011-05-17 20:59:31.000000000 +0200
+++ mp3_check-1.98/support_functions.h	2011-05-17 20:59:43.000000000 +0200
@@ -680,7 +680,7 @@
 	printf("%-20s%d\n", "BitRate", mp3_i->BIT_RATE);
 	printf("%-20s%d\n", "SampRate", mp3_i->SAMPLE_FREQ);
 	printf("%-20s%s\n", "BinString", mp3_i->BIN_STRING);
-	printf("%-20s%d\n", "BinLen", strlen(mp3_i->BIN_STRING));
+	printf("%-20s%zu\n", "BinLen", strlen(mp3_i->BIN_STRING));
 	printf("%-20s%u\n", "IntHeader", mp3_i->INT_HEADER);
 	printf("%-20s0x%x\n", "CRC16Value", mp3_i->CRC16_VALUE);
 	printf("%-20s0x%x\n", "Correct_CRC16Value", mp3_i->CORRECT_CRC16_VALUE);
