--- gramps-5.1.0_orig/gramps/gui/grampsgui.py	2019-08-25 07:55:51.438443839 +0200
+++ gramps-5.1.0/gramps/gui/grampsgui.py	2019-08-25 07:57:09.600932670 +0200
@@ -618,7 +618,7 @@
                 and not gettext.find(GTK_GETTEXT_DOMAIN)):
             _display_gtk_gettext_message(parent=self._vm.window)
 
-        _display_welcome_message(parent=self._vm.window)
+        #_display_welcome_message(parent=self._vm.window)
 
         _display_translator_message(parent=self._vm.window)
 
