From: Andreas Stieger <astieger@suse.com>
Date: Fri, 03 Nov 2017 20:12:29 +0000
Subject: fix missing return in nonvoid function

Index: complexity-1.10/src/score.c
===================================================================
--- complexity-1.10.orig/src/score.c
+++ complexity-1.10/src/score.c
@@ -245,6 +245,7 @@ fiddle_subexpr_score(subexpr_seen_t * se
 
     default:
         CX_ASSERT(which == 0);
+        return NULL;
     }
 }
 
