Index: lazarus/Makefile.fpc
===================================================================
--- lazarus.orig/Makefile.fpc
+++ lazarus/Makefile.fpc
@@ -20,7 +20,7 @@ files=$(wildcard *$(OEXT)) $(wildcard *$
 
 [install]
 fpcpackage=n
-basedir=share/lazarus
+basedir=$(_LIB)/lazarus
 
 [dist]
 destdir=$(BASEDIR)/dist
@@ -29,7 +29,7 @@ destdir=$(BASEDIR)/dist
 RCPP?=$(strip $(firstword cpp$(SRCEXEEXT)))
 #
 ifndef LAZARUS_INSTALL_DIR
-LAZARUS_INSTALL_DIR=$(INSTALL_PREFIX)/share/lazarus
+LAZARUS_INSTALL_DIR=$(INSTALL_PREFIX)/$(_LIB)/lazarus
 INSTALL_MAN_DIR=$(INSTALL_PREFIX)/share/man
 ifneq ($(findstring $(OS_TARGET),win32 win64),)
 LAZARUS_INSTALL_DIR=C:\lazarus
@@ -249,8 +249,8 @@ installbase:
 ifneq ($(findstring $(OS_TARGET),win32 win64),)
 
 else
-	$(MKDIR) $(INSTALL_PREFIX)/share
-	$(MKDIR) $(INSTALL_PREFIX)/share/lazarus
+	$(MKDIR) $(INSTALL_PREFIX)/$(_LIB)
+	$(MKDIR) $(INSTALL_PREFIX)/$(_LIB)/lazarus
 	$(MKDIR) $(INSTALL_PREFIX)/share/applications
 	$(MKDIR) $(INSTALL_PREFIX)/share/pixmaps
 	$(MKDIR) $(INSTALL_PREFIX)/share/mime/packages
