From: Mattias Gaertner <mattias@freepascal.org>
Date: 2020-07-07 09:25:48 UTC
Subject: fixed compile fpc 3.0.4
References: r63523

chmhelp: fixed compile fpc 3.0.4

--- trunk/components/chmhelp/lhelp/chmspecialparser.pas	2020/07/07 03:59:57	63522
+++ trunk/components/chmhelp/lhelp/chmspecialparser.pas	2020/07/07 09:25:48	63523
@@ -136,6 +136,7 @@
     // Add new child node
     fLastNode := AParentNode;
     NewNode := TContentTreeNode(fTreeView.Items.AddChild(AParentNode, txt));
+    {$IF FPC_FULLVERSION>=30200}
     URL:='';
     for x:=0 to AItem.SubItemcount-1 do
     begin
@@ -146,6 +147,9 @@
       if URL<>'' then
         break;
     end;
+    {$ELSE}
+    URL:=AItem.URL;
+    {$ENDIF}
     NewNode.Url := FixURL('/'+URL);
     NewNode.Data := fChm;
     if fTreeView.Images <> nil then
