Index: libchipcard-5.0.0/Makefile.am
===================================================================
--- libchipcard-5.0.0.orig/Makefile.am
+++ libchipcard-5.0.0/Makefile.am
@@ -47,7 +47,7 @@ win32:
 	 rm -Rf win32-bin
 
 listdoc.h:
-	$(gwenhywfar_bindir)/mklistdoc -v `find chipcard -name '*.h' ` >listdoc.h
+	$(gwenhywfar_bindir)/mklistdoc -v `find . -name '*.h' ` >listdoc.h
 
 
 srcdoc: Doxyfile listdoc.h
Index: libchipcard-5.0.0/Makefile.in
===================================================================
--- libchipcard-5.0.0.orig/Makefile.in
+++ libchipcard-5.0.0/Makefile.in
@@ -915,7 +915,7 @@ win32:
 	 rm -Rf win32-bin
 
 listdoc.h:
-	$(gwenhywfar_bindir)/mklistdoc -v `find chipcard -name '*.h' ` >listdoc.h
+	$(gwenhywfar_bindir)/mklistdoc -v `find . -name '*.h' ` >listdoc.h
 
 srcdoc: Doxyfile listdoc.h
 	if test -d apidoc; then rm -Rf apidoc; fi
