Index: pidgin-2.14.4/configure.ac
===================================================================
--- pidgin-2.14.4.orig/configure.ac
+++ pidgin-2.14.4/configure.ac
@@ -259,8 +259,8 @@ dnl # Disable creation and installation
 dnl #######################################################################
 AC_ARG_ENABLE(nls, AC_HELP_STRING([--disable-nls], [disable installation of translation files]), enable_i18n="$enableval", enable_i18n=yes)
 
-if test x$enable_i18n = xyes; then
 AC_PROG_INTLTOOL
+if test x$enable_i18n = xyes; then
 	GETTEXT_PACKAGE=pidgin
 	AC_SUBST(GETTEXT_PACKAGE)
 
