--- contrib/xwintoppm/dsimple.c
+++ contrib/xwintoppm/dsimple.c
@@ -503,6 +503,7 @@
 	fprintf(stderr, msg, arg0, arg1, arg2, arg3, arg4, arg5, arg6);
 	fprintf(stderr, "\n");
 	fflush(stderr);
+	return 0;
 }
 
 
--- contrib/xwintoppm/xwintoppm.c
+++ contrib/xwintoppm/xwintoppm.c
@@ -604,6 +604,7 @@
      * Free image
      */
     XDestroyImage(image);
+    return 0;
 }
 
 /*
@@ -729,6 +730,7 @@
 	bp++;
 	*bp++ = c;
     }
+    return 0;
 }
 
 _swaplong (bp, n)
@@ -750,4 +752,5 @@
 	*bp++ = c;
 	bp += 2;
     }
+    return 0;
 }
