--- magicpoint-1.11b/Imakefile.in.xx	2005-09-28 14:32:11.000000000 +0200
+++ magicpoint-1.11b/Imakefile.in	2005-09-28 14:32:51.000000000 +0200
@@ -87,7 +87,7 @@
 print.o: $(srcdir)/print.c ctlwords.h
 
 ctlwords.h: $(srcdir)/ctlwords.c $(srcdir)/globals.c
-	$(CC) -o ./ctlwords $(srcdir)/ctlwords.c
+	$(CC) $(CFLAGS) -o ./ctlwords $(srcdir)/ctlwords.c
 	./ctlwords < $(srcdir)/globals.c > ctlwords.h
 
 clean::
@@ -104,7 +104,7 @@
 
 # -O2 will work badly...
 unimap.o:
-	$(CC) -c $(CCOPTIONS) $(THREADS_CFLAGS) $(ALLDEFINES) \
+	$(CC) -c $(CFLAGS) $(THREADS_CFLAGS) $(ALLDEFINES) \
 		-I$(srcdir) $(DEFS) $(OPTFLAGS) unimap.c
 
 # missing libraries
