--- translation-update-upstream-embedded.sh	2018-12-10 22:58:17.369805285 +0100
+++ translation-update-upstream-embedded.sh	2018-12-10 23:01:32.309607787 +0100
@@ -74,6 +74,7 @@
 	fi
     fi
 fi
+BASE_DIR=$PWD
 DIR=${1:-po}
 
 if test -z "$3" ; then
@@ -202,7 +203,7 @@
 		echo "WARNING: Missing $DOMAIN in translation-update-upstream configuration! No updates available."
 	    fi
 	else
-	    echo "ERROR: Package translation-update-upstream is not installed. Please update your BuildRequires!"
+	    echo "ERROR: Incorrect use of translation-update-upstream-embedded!"
 	    exit 1
 	fi
     fi
