diff -Pdpru mame-mame0288.orig/src/osd/sdl/sdlprefix.h mame-mame0288/src/osd/sdl/sdlprefix.h
--- mame-mame0288.orig/src/osd/sdl/sdlprefix.h	2026-05-27 22:03:56.000000000 +0300
+++ mame-mame0288/src/osd/sdl/sdlprefix.h	2026-05-28 19:44:22.241006879 +0300
@@ -57,9 +57,4 @@ struct _IO_FILE {};  //_IO_FILE is an op
 #define SDLMAME_ANDROID 1
 #endif
 
-// fix for Ubuntu 8.10
-#ifdef _FORTIFY_SOURCE
-#undef _FORTIFY_SOURCE
-#endif
-
 #endif /* SDLMAME_UNIX */
