--- pipenv-2018.11.26.1543215385.742ae1b9-orig/pipenv/patched/notpip/_internal/utils/ui.py	2019-04-04 10:40:03.384546366 +0700
+++ pipenv-2018.11.26.1543215385.742ae1b9/pipenv/patched/notpip/_internal/utils/ui.py	2019-04-04 10:40:51.640970428 +0700
@@ -7,13 +7,13 @@
 import time
 from signal import SIGINT, default_int_handler, signal
 
-from pipenv.patched.notpip._vendor import six
-from pipenv.patched.notpip._vendor.progress import HIDE_CURSOR, SHOW_CURSOR
-from pipenv.patched.notpip._vendor.progress.bar import (
+import six
+from progress import HIDE_CURSOR, SHOW_CURSOR
+from progress.bar import (
     Bar, ChargingBar, FillingCirclesBar, FillingSquaresBar, IncrementalBar,
     ShadyBar,
 )
-from pipenv.patched.notpip._vendor.progress.spinner import Spinner
+from progress.spinner import Spinner
 
 from pipenv.patched.notpip._internal.utils.compat import WINDOWS
 from pipenv.patched.notpip._internal.utils.logging import get_indentation
@@ -24,7 +24,7 @@
     from typing import Any  # noqa: F401
 
 try:
-    from pipenv.patched.notpip._vendor import colorama
+    import colorama
 # Lots of different errors can come from this, including SystemError and
 # ImportError.
 except Exception:
