Index: plplot-5.15.0/cmake/modules/pdf.cmake
===================================================================
--- plplot-5.15.0.orig/cmake/modules/pdf.cmake
+++ plplot-5.15.0/cmake/modules/pdf.cmake
@@ -47,6 +47,7 @@ if(PLD_pdf)
     # Determine that version in cross-platform way.
     set(check_libharu_version_source "
 #include \"hpdf.h\"
+#include \"hpdf_version.h\"
 int main(void)
 {
 // True if version is 2.3.0 or later....
