Index: plplot-5.15.0/pkgcfg/plplot-template.pc.in
===================================================================
--- plplot-5.15.0.orig/pkgcfg/plplot-template.pc.in
+++ plplot-5.15.0/pkgcfg/plplot-template.pc.in
@@ -1,6 +1,6 @@
-libdir=@LIB_DIR@
-includedir=@INCLUDE_DIR@
-drvdir=@LIB_DIR@/plplot@PLPLOT_VERSION@/drivers
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+drvdir=@CMAKE_INSTALL_FULL_LIBDIR@/plplot@PLPLOT_VERSION@/drivers
 
 Name: PLplot @PC_SHORT_NAME@
 Description: Scientific plotting library (@PC_PRECISION@@PC_LONG_NAME@)
Index: plplot-5.15.0/CMakeLists.txt
===================================================================
--- plplot-5.15.0.orig/CMakeLists.txt
+++ plplot-5.15.0/CMakeLists.txt
@@ -95,6 +95,8 @@ endif(POLICY CMP0086)
 # install trees.
 set(PROJECT_NAMESPACE PLPLOT::)
 
+include(GNUInstallDirs)
+
 option(USE_WRITEABLE_TARGET_TEST_PREFIX "Option to test library and executable targets which have both writeable and (ALIASed namespaced) read-only versions of their targets in the build tree and namespaced read-only targets in the install tree.  This test prepends a prefix to the writeable version of the target name so as a result any such target referred to in the build tree without a writeable (\"writeable_target_test_prefix_)\") or read-only (\"PLPLOT::\") prefix should be discovered by this test (N.B. unless the unprefixed target reference occurs in CMake logic branches that are unexecuted by the particular configuration options that are chosen which makes this test necessary but not sufficient)." OFF)
 if(USE_WRITEABLE_TARGET_TEST_PREFIX)
   set(WRITEABLE_TARGET writeable_target_test_prefix_)
