From b1d0296a937fe278239bdfac840a3fd0e93b3ee9 Mon Sep 17 00:00:00 2001
From: "Yukihiro \"Matz\" Matsumoto" <matz@ruby.or.jp>
Date: Sat, 9 Apr 2022 09:25:35 +0900
Subject: [PATCH] class.c: clear method cache after `remove_method`.

---
 src/class.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/class.c b/src/class.c
index 37fc4e68a4..68a0ff0843 100644
--- a/src/class.c
+++ b/src/class.c
@@ -2361,7 +2361,10 @@ mrb_remove_method(mrb_state *mrb, struct RClass *c, mrb_sym mid)
   MRB_CLASS_ORIGIN(c);
   h = c->mt;
 
-  if (h && mt_del(mrb, h, mid)) return;
+  if (h && mt_del(mrb, h, mid)) {
+    mrb_mc_clear_by_class(mrb, c);
+    return;
+  }
   mrb_name_error(mrb, mid, "method '%n' not defined in %C", mid, c);
 }
 
