diff -urEbwB antlr-maven-plugin-2.2.orig/src/main/java/org/codehaus/mojo/antlr/AbstractAntlrMojo.java antlr-maven-plugin-2.2/src/main/java/org/codehaus/mojo/antlr/AbstractAntlrMojo.java
--- antlr-maven-plugin-2.2.orig/src/main/java/org/codehaus/mojo/antlr/AbstractAntlrMojo.java	2025-10-24 10:50:58.787477162 +0200
+++ antlr-maven-plugin-2.2/src/main/java/org/codehaus/mojo/antlr/AbstractAntlrMojo.java	2025-10-24 11:22:33.315255452 +0200
@@ -337,7 +337,7 @@
             oldSm = System.getSecurityManager();
             System.setSecurityManager( NoExitSecurityManager.INSTANCE );
         }
-        catch ( SecurityException ex )
+        catch ( UnsupportedOperationException | SecurityException ex )
         {
             // ANTLR-12
             oldSm = null;
