diff --git a/src/main.c b/src/main.c
index 64aa7da..d06f5b0 100644
--- a/src/main.c
+++ b/src/main.c
@@ -373,7 +373,7 @@ copy_thumbnail_from_file (GnomeDesktopThumbnailFactory *factory, const gchar *fi
 				g_setenv ("GNOME_DIRECTORY_THUMBNAILER_RECURSION_LIMIT", new_recursion_limit_str, TRUE);
 				g_free (new_recursion_limit_str);
 
-				pixbuf = gnome_desktop_thumbnail_factory_generate_thumbnail (factory, file_uri, file_mime_type);
+				pixbuf = gnome_desktop_thumbnail_factory_generate_thumbnail (factory, file_uri, file_mime_type, NULL, NULL);
 				if (pixbuf == NULL) {
 					/* gnome-desktop doesn't set an error so we have to. */
 					g_debug ("Error generating thumbnail.");
