Index: img2pdf-0.3.3/src/tests/__init__.py
===================================================================
--- img2pdf-0.3.3.orig/src/tests/__init__.py	2018-12-25 12:28:33.000000000 +0100
+++ img2pdf-0.3.3/src/tests/__init__.py	2019-03-27 13:43:28.471760192 +0100
@@ -55,12 +55,12 @@ def compare_pdf(outx, outy):
         x = recompress_last_stream(outx)
         y = recompress_last_stream(outy)
         if x != y:
-            print('original outx:\n{}\nouty:\n{}\n'.format(outx, outy), file=sys.stderr)
-            print('recompressed outx:\n{}\nouty:\n{}\n'.format(x, y), file=sys.stderr)
+            print >>sys.stderr, 'original outx:\n{}\nouty:\n{}\n'.format(outx, outy)
+            print >>sys.stderr, 'recompressed outx:\n{}\nouty:\n{}\n'.format(x, y)
             return False
     else:
         if outx != outy:
-            print('original outx:\n{}\nouty:\n{}\n'.format(outx, outy), file=sys.stderr)
+            print >>sys.stderr, 'original outx:\n{}\nouty:\n{}\n'.format(outx, outy)
     return True
 
 # convert +set date:create +set date:modify -define png:exclude-chunk=time
