From 763079de767df0e3ae42f51b65f2f25d7ea7b8b8 Mon Sep 17 00:00:00 2001
From: Adrian Knagg-Baugh <aje.baugh@gmail.com>
Date: Wed, 31 Jul 2024 09:16:14 +0000
Subject: [PATCH] Refactor deconvolution code (fixes #1202)

---
 src/core/processing.h                         |    8 +
 src/gui/progress_and_log.h                    |    8 +

diff --git a/src/core/processing.h b/src/core/processing.h
index 07a067503a..cb1773378e 100644
--- a/src/core/processing.h
+++ b/src/core/processing.h
@@ -4,6 +4,10 @@
 #include "sequence_filtering.h"
 #include "io/fits_sequence.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /**
  * \file processing.h
  * \brief Manages background computation and parallel image processing.
@@ -187,4 +191,8 @@ gpointer generic_sequence_metadata_worker(gpointer args);
 
 void kill_child_process(gboolean on_exit);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif
diff --git a/src/gui/progress_and_log.h b/src/gui/progress_and_log.h
index a584bb8ccb..9572d47dee 100644
--- a/src/gui/progress_and_log.h
+++ b/src/gui/progress_and_log.h
@@ -10,6 +10,10 @@
 #define PROGRESS_DONE 1.0		// fill the progress bar
 #define PROGRESS_TEXT_RESET ""		// reset the progress bar's text
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 void initialize_log_tags();
 void gui_log_message(const char* msg, const char* color);
 
@@ -17,4 +21,8 @@ void set_progress_bar_data(const char *text, double percent);
 void set_cursor_waiting(gboolean waiting);
 void set_cursor(const gchar* cursor_name);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif

