Index: gnome-control-center-41.0/panels/power/cc-power-panel.c
===================================================================
--- gnome-control-center-41.0.orig/panels/power/cc-power-panel.c
+++ gnome-control-center-41.0/panels/power/cc-power-panel.c
@@ -796,7 +796,8 @@ populate_power_button_row (HdyComboRow *
     { N_("Suspend"), GSD_POWER_BUTTON_ACTION_SUSPEND },
     { N_("Power Off"), GSD_POWER_BUTTON_ACTION_INTERACTIVE },
     { N_("Hibernate"), GSD_POWER_BUTTON_ACTION_HIBERNATE },
-    { N_("Nothing"), GSD_POWER_BUTTON_ACTION_NOTHING }
+    { N_("Nothing"), GSD_POWER_BUTTON_ACTION_NOTHING },
+    { N_("Force Off"), GSD_POWER_BUTTON_ACTION_SHUTDOWN }
   };
   guint i;
 
