Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/gpu/vaapi/vaapi_wrapper.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/media/gpu/vaapi/vaapi_wrapper.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/gpu/vaapi/vaapi_wrapper.cc
@@ -418,6 +418,8 @@ static VAEntrypoint GetVaEntryPoint(Vaap
     case VaapiWrapper::kCodecModeMax:
       NOTREACHED();
       return VAEntrypointVLD;
+    default:
+      return 0;
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/cc/input/snap_fling_controller.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/cc/input/snap_fling_controller.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/cc/input/snap_fling_controller.cc
@@ -26,6 +26,8 @@ bool SnapFlingController::FilterEventFor
     case GestureScrollType::kEnd: {
       return state_ == State::kActive || state_ == State::kFinished;
     }
+    default:
+      return false;
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/chrome/browser/ui/webui/discards/discards_ui.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/chrome/browser/ui/webui/discards/discards_ui.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/chrome/browser/ui/webui/discards/discards_ui.cc
@@ -56,6 +56,8 @@ mojom::LifecycleUnitVisibility GetLifecy
       return mojom::LifecycleUnitVisibility::OCCLUDED;
     case content::Visibility::VISIBLE:
       return mojom::LifecycleUnitVisibility::VISIBLE;
+    default:
+      return mojom::LifecycleUnitVisibility::VISIBLE;
   }
 #if defined(COMPILER_MSVC)
   NOTREACHED();
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/components/autofill/core/browser/autofill_manager.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/components/autofill/core/browser/autofill_manager.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/components/autofill/core/browser/autofill_manager.cc
@@ -276,6 +276,7 @@ PopupType AutofillManager::GetPopupType(
 
     default:
       NOTREACHED();
+      return PopupType::kUnspecified;
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/components/data_reduction_proxy/core/browser/data_reduction_proxy_util.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/components/data_reduction_proxy/core/browser/data_reduction_proxy_util.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/components/data_reduction_proxy/core/browser/data_reduction_proxy_util.cc
@@ -322,6 +322,8 @@ PageloadMetrics_ConnectionType ProtoConn
       return PageloadMetrics_ConnectionType_CONNECTION_NONE;
     case net::NetworkChangeNotifier::CONNECTION_BLUETOOTH:
       return PageloadMetrics_ConnectionType_CONNECTION_BLUETOOTH;
+    default:
+      return PageloadMetrics_ConnectionType_CONNECTION_UNKNOWN;
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/browser/cache_storage/cache_storage_quota_client.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/content/browser/cache_storage/cache_storage_quota_client.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/browser/cache_storage/cache_storage_quota_client.cc
@@ -98,6 +98,8 @@ storage::QuotaClient::ID CacheStorageQuo
       return kServiceWorkerCache;
     case CacheStorageOwner::kBackgroundFetch:
       return kBackgroundFetch;
+    default:
+      return kBackgroundFetch;
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/capture/video/video_capture_device_descriptor.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/media/capture/video/video_capture_device_descriptor.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/capture/video/video_capture_device_descriptor.cc
@@ -94,6 +94,8 @@ const char* VideoCaptureDeviceDescriptor
       return "Virtual Device";
     case VideoCaptureApi::UNKNOWN:
       return "Unknown";
+    default:
+      return "Unknown";
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/net/nqe/network_quality_estimator.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/net/nqe/network_quality_estimator.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/net/nqe/network_quality_estimator.cc
@@ -1162,6 +1162,8 @@ base::TimeDelta NetworkQualityEstimator:
     case nqe::internal::OBSERVATION_CATEGORY_COUNT:
       NOTREACHED();
       return base::TimeDelta();
+    default:
+      return base::TimeDelta();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/audio/public/cpp/audio_system_to_service_adapter.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/services/audio/public/cpp/audio_system_to_service_adapter.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/audio/public/cpp/audio_system_to_service_adapter.cc
@@ -66,6 +66,8 @@ const char* GetTraceEvent(Action action)
       return "AudioSystemToServiceAdapter::GetAssociatedOutputDeviceID";
     case kGetInputDeviceInfo:
       return "AudioSystemToServiceAdapter::GetInputDeviceInfo";
+    default:
+      return "Unknown";
   }
   NOTREACHED();
 }
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/network/cross_origin_read_blocking.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/services/network/cross_origin_read_blocking.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/network/cross_origin_read_blocking.cc
@@ -861,6 +861,8 @@ bool CrossOriginReadBlocking::ResponseAn
       return sniffers_.empty() && !found_blockable_content_;
     case kBlock:
       return false;
+    default:
+      return true;
   }
 }
 
@@ -872,6 +874,8 @@ bool CrossOriginReadBlocking::ResponseAn
       return sniffers_.empty() && found_blockable_content_;
     case kBlock:
       return true;
+    default:
+      return false;
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/webaudio/audio_param.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/modules/webaudio/audio_param.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/webaudio/audio_param.cc
@@ -146,6 +146,7 @@ String AudioParamHandler::GetParamName()
       return custom_param_name_;
     default:
       NOTREACHED();
+      return "Unknown";
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/ui/accessibility/platform/ax_platform_node_auralinux.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/ui/accessibility/platform/ax_platform_node_auralinux.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/ui/accessibility/platform/ax_platform_node_auralinux.cc
@@ -1863,6 +1863,8 @@ AtkRole AXPlatformNodeAuraLinux::GetAtkR
     case ax::mojom::Role::kPresentational:
     case ax::mojom::Role::kUnknown:
       return ATK_ROLE_REDUNDANT_OBJECT;
+    default:
+      return ATK_ROLE_REDUNDANT_OBJECT;
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/components/password_manager/core/browser/browser_save_password_progress_logger.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/components/password_manager/core/browser/browser_save_password_progress_logger.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/components/password_manager/core/browser/browser_save_password_progress_logger.cc
@@ -74,6 +74,8 @@ std::string VoteTypeToString(
       return "HTML classifier";
     case AutofillUploadContents::Field::FIRST_USE:
       return "First use";
+    default:
+      return "No information";
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/components/previews/core/previews_black_list.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/components/previews/core/previews_black_list.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/components/previews/core/previews_black_list.cc
@@ -34,6 +34,9 @@ PreviewsEligibilityReason BlacklistReaso
       return PreviewsEligibilityReason::ALLOWED;
     case blacklist::BlacklistReason::kAllowed:
       return PreviewsEligibilityReason::ALLOWED;
+    default:
+      NOTREACHED() << "Previews does not support type-base blacklisting";
+      return PreviewsEligibilityReason::ALLOWED;
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/browser/frame_host/frame_tree_node.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/content/browser/frame_host/frame_tree_node.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/browser/frame_host/frame_tree_node.cc
@@ -608,6 +608,7 @@ bool FrameTreeNode::UpdateUserActivation
       return ClearUserActivation();
   }
   NOTREACHED() << "Invalid update_type.";
+  return false;
 }
 
 void FrameTreeNode::OnSetHasReceivedUserGestureBeforeNavigation(bool value) {
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/renderer/input/input_event_prediction.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/content/renderer/input/input_event_prediction.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/renderer/input/input_event_prediction.cc
@@ -108,6 +108,8 @@ std::unique_ptr<ui::InputPredictor> Inpu
       return std::make_unique<ui::LeastSquaresPredictor>();
     case PredictorType::kKalman:
       return std::make_unique<ui::KalmanPredictor>();
+    default:
+      return std::make_unique<ui::EmptyPredictor>();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/renderer/media/webrtc/peer_connection_tracker.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/content/renderer/media/webrtc/peer_connection_tracker.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/renderer/media/webrtc/peer_connection_tracker.cc
@@ -118,6 +118,8 @@ static const char* SerializeDirection(
       return "'recvonly'";
     case webrtc::RtpTransceiverDirection::kInactive:
       return "'inactive'";
+    default:
+      return "'unknown'";
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/device/bluetooth/bluetooth_remote_gatt_characteristic.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/device/bluetooth/bluetooth_remote_gatt_characteristic.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/device/bluetooth/bluetooth_remote_gatt_characteristic.cc
@@ -396,6 +396,8 @@ bool BluetoothRemoteGattCharacteristic::
       return hasNotify;
     case NotificationType::kIndication:
       return hasIndicate;
+    default:
+      return false;
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/blink/watch_time_reporter.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/media/blink/watch_time_reporter.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/blink/watch_time_reporter.cc
@@ -571,6 +571,8 @@ WatchTimeKey WatchTimeReporter::GetDispl
       return DISPLAY_TYPE_KEY(DisplayFullscreen);
     case DisplayType::kPictureInPicture:
       return DISPLAY_TYPE_KEY(DisplayPictureInPicture);
+    default:
+      return DISPLAY_TYPE_KEY(DisplayInline);
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/core/css/style_environment_variables.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/core/css/style_environment_variables.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/core/css/style_environment_variables.cc
@@ -67,6 +67,7 @@ const AtomicString StyleEnvironmentVaria
   }
 
   NOTREACHED();
+  return "unknown";
 }
 
 // static
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/core/html/media/media_controls.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/core/html/media/media_controls.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/core/html/media/media_controls.cc
@@ -36,6 +36,9 @@ AtomicString MediaControls::GetSizingCSS
       return kMediaControlsSizingMediumCSSClass;
     case MediaControlsSizingClass::kLarge:
       return kMediaControlsSizingLargeCSSClass;
+    default:
+      NOTREACHED();
+      return kMediaControlsSizingMediumCSSClass;
   }
 
   NOTREACHED();
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/cookie_store/cookie_change_event.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/modules/cookie_store/cookie_change_event.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/cookie_store/cookie_change_event.cc
@@ -53,6 +53,9 @@ String ToCookieListItemSameSite(network:
       return "lax";
     case network::mojom::CookieSameSite::NO_RESTRICTION:
       return "unrestricted";
+    default:
+      NOTREACHED();
+      return "unknown";
   }
 
   NOTREACHED();
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/platform/heap/heap_stats_collector.h
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/platform/heap/heap_stats_collector.h
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/platform/heap/heap_stats_collector.h
@@ -94,6 +94,8 @@ class PLATFORM_EXPORT ThreadHeapStatsCol
         return "BlinkGC.VisitPersistents";
       case Id::kVisitStackRoots:
         return "BlinkGC.VisitStackRoots";
+      default:
+        return "BlinkGC.Unknown";
     }
   }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/platform/scheduler/main_thread/page_scheduler_impl.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/platform/scheduler/main_thread/page_scheduler_impl.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/platform/scheduler/main_thread/page_scheduler_impl.cc
@@ -679,6 +679,9 @@ PageSchedulerImpl::PageLifecycleStateTra
           NOTREACHED();
           return base::nullopt;
       }
+    default:
+      NOTREACHED();
+      return base::nullopt;
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/ui/base/mojo/clipboard_struct_traits.h
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/ui/base/mojo/clipboard_struct_traits.h
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/ui/base/mojo/clipboard_struct_traits.h
@@ -20,6 +20,9 @@ struct EnumTraits<ui::mojom::ClipboardTy
         return ui::mojom::ClipboardType::SELECTION;
       case ui::CLIPBOARD_TYPE_DRAG:
         return ui::mojom::ClipboardType::DRAG;
+      default:
+        NOTREACHED();
+        return ui::mojom::ClipboardType::SELECTION;
     }
   }
 
@@ -34,6 +37,8 @@ struct EnumTraits<ui::mojom::ClipboardTy
       case ui::mojom::ClipboardType::DRAG:
         *out = ui::CLIPBOARD_TYPE_DRAG;
         return true;
+      default:
+        return false;
     }
     NOTREACHED();
     return false;
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/peerconnection/rtc_rtp_transceiver.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/modules/peerconnection/rtc_rtp_transceiver.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/peerconnection/rtc_rtp_transceiver.cc
@@ -28,6 +28,8 @@ String TransceiverDirectionToString(
       return "recvonly";
     case webrtc::RtpTransceiverDirection::kInactive:
       return "inactive";
+    default:
+      return "unknown";
   }
   NOTREACHED();
 }
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/core/loader/mixed_content_checker.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/core/loader/mixed_content_checker.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/core/loader/mixed_content_checker.cc
@@ -235,6 +235,7 @@ bool MixedContentChecker::IsMixedContent
     const FetchClientSettingsObject& settings,
     const KURL& url) {
   switch (settings.GetHttpsState()) {
+    default:
     case HttpsState::kNone:
       return false;
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/browser/media/audio_output_stream_broker.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/content/browser/media/audio_output_stream_broker.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/browser/media/audio_output_stream_broker.cc
@@ -53,6 +53,8 @@ StreamBrokerDisconnectReason GetDisconne
                  : StreamBrokerDisconnectReason::kDocumentDestroyed;
     case DisconnectReason::kDefault:
       return StreamBrokerDisconnectReason::kDefault;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/renderer/media/stream/media_stream_constraints_util_video_device.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/content/renderer/media/stream/media_stream_constraints_util_video_device.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/renderer/media/stream/media_stream_constraints_util_video_device.cc
@@ -554,6 +554,8 @@ blink::WebMediaStreamTrack::DisplayCaptu
           kApplication;
     case media::mojom::DisplayCaptureSurfaceType::BROWSER:
       return blink::WebMediaStreamTrack::DisplayCaptureSurfaceType::kBrowser;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -566,6 +568,8 @@ blink::WebMediaStreamTrack::CursorCaptur
       return blink::WebMediaStreamTrack::CursorCaptureType::kAlways;
     case media::mojom::CursorCaptureType::MOTION:
       return blink::WebMediaStreamTrack::CursorCaptureType::kMotion;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/extensions/browser/url_loader_factory_manager.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/extensions/browser/url_loader_factory_manager.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/extensions/browser/url_loader_factory_manager.cc
@@ -330,6 +330,8 @@ bool IsSpecialURLLoaderFactoryRequired(c
              DoExtensionPermissionsCoverCorsOrCorbRelatedOrigins(extension);
     case FactoryUser::kExtensionProcess:
       return DoExtensionPermissionsCoverCorsOrCorbRelatedOrigins(extension);
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/gin/v8_isolate_memory_dump_provider.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/gin/v8_isolate_memory_dump_provider.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/gin/v8_isolate_memory_dump_provider.cc
@@ -131,6 +131,8 @@ std::string IsolateTypeString(IsolateHol
       return "test";
     case IsolateHolder::IsolateType::kUtility:
       return "utility";
+    default:
+      __builtin_abort ();
   }
   LOG(FATAL) << "Unreachable code";
 }
@@ -147,6 +149,8 @@ bool CanHaveMultipleIsolates(IsolateHold
     case IsolateHolder::IsolateType::kUtility:
       // PDFium and ProxyResolver create one isolate per process.
       return false;
+    default:
+      __builtin_abort ();
   }
   LOG(FATAL) << "Unreachable code";
 }
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/gpu/config/gpu_info.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/gpu/config/gpu_info.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/gpu/config/gpu_info.cc
@@ -84,6 +84,8 @@ const char* OverlayFormatToString(Overla
       return "YUY2";
     case OverlayFormat::kNV12:
       return "NV12";
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/audio/audio_processing.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/media/audio/audio_processing.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/audio/audio_processing.cc
@@ -19,6 +19,8 @@ std::string AudioProcessingSettings::ToS
         return "experimental";
       case AutomaticGainControlType::kHybridExperimental:
         return "hybrid experimental";
+      default:
+        __builtin_abort ();
     }
   };
 
@@ -32,6 +34,8 @@ std::string AudioProcessingSettings::ToS
         return "aec3";
       case EchoCancellationType::kSystemAec:
         return "system aec";
+      default:
+        __builtin_abort ();
     }
   };
 
@@ -43,6 +47,8 @@ std::string AudioProcessingSettings::ToS
         return "default";
       case NoiseSuppressionType::kExperimental:
         return "experimental";
+      default:
+        __builtin_abort ();
     }
   };
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/base/decode_status.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/media/base/decode_status.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/base/decode_status.cc
@@ -18,6 +18,8 @@ const char* GetDecodeStatusString(Decode
       return "DecodeStatus::ABORTED";
     case DecodeStatus::DECODE_ERROR:
       return "DecodeStatus::DECODE_ERROR";
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/base/decryptor.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/media/base/decryptor.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/base/decryptor.cc
@@ -17,6 +17,8 @@ const char* Decryptor::GetStatusName(Sta
       return "need_more_data";
     case kError:
       return "error";
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/base/demuxer_stream.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/media/base/demuxer_stream.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/base/demuxer_stream.cc
@@ -17,6 +17,8 @@ const char* DemuxerStream::GetTypeName(T
       return "text";
     case DemuxerStream::UNKNOWN:
       return "unknown";
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -31,6 +33,8 @@ const char* DemuxerStream::GetStatusName
       return "config_changed";
     case DemuxerStream::kError:
       return "error";
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/blink/webmediaplayer_impl.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/media/blink/webmediaplayer_impl.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/blink/webmediaplayer_impl.cc
@@ -199,6 +199,8 @@ EncryptionMode DetermineEncryptionMode(
       return EncryptionMode::kCenc;
     case EncryptionScheme::CIPHER_MODE_AES_CBC:
       return EncryptionMode::kCbcs;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/capture/video/video_capture_device.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/media/capture/video/video_capture_device.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/capture/video/video_capture_device.cc
@@ -84,6 +84,8 @@ VideoCaptureFrameDropReason ConvertReser
       return VideoCaptureFrameDropReason::kBufferPoolMaxBufferCountExceeded;
     case VideoCaptureDevice::Client::ReserveResult::kAllocationFailed:
       return VideoCaptureFrameDropReason::kBufferPoolBufferAllocationFailed;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/cdm/cdm_helpers.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/media/cdm/cdm_helpers.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/cdm/cdm_helpers.cc
@@ -25,6 +25,8 @@ gfx::ColorSpace::RangeID ToGfxColorSpace
       return gfx::ColorSpace::RangeID::FULL;
     case cdm::ColorRange::kDerived:
       return gfx::ColorSpace::RangeID::DERIVED;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/filters/decoder_stream.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/media/filters/decoder_stream.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/filters/decoder_stream.cc
@@ -89,6 +89,8 @@ const char* GetStatusString(typename Dec
       return "demuxer_read_aborted";
     case DecoderStream<StreamType>::DECODE_ERROR:
       return "decode_error";
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/network/public/cpp/cors/cors.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/services/network/public/cpp/cors/cors.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/network/public/cpp/cors/cors.cc
@@ -539,6 +539,8 @@ bool IsCorsSameOriginResponseType(mojom:
     case mojom::FetchResponseType::kOpaque:
     case mojom::FetchResponseType::kOpaqueRedirect:
       return false;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -552,6 +554,8 @@ bool IsCorsCrossOriginResponseType(mojom
     case mojom::FetchResponseType::kOpaque:
     case mojom::FetchResponseType::kOpaqueRedirect:
       return true;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -569,6 +573,8 @@ bool CalculateCredentialsFlag(mojom::Fet
       return response_tainting == network::mojom::FetchResponseType::kBasic;
     case network::mojom::FetchCredentialsMode::kInclude:
       return true;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/network/public/cpp/host_resolver_mojom_traits.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/services/network/public/cpp/host_resolver_mojom_traits.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/network/public/cpp/host_resolver_mojom_traits.cc
@@ -40,6 +40,8 @@ base::Optional<bool> FromTristate(DnsCon
       return true;
     case DnsConfigOverrides::Tristate::TRISTATE_FALSE:
       return false;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -232,6 +234,8 @@ DnsQueryType EnumTraits<DnsQueryType, ne
       return DnsQueryType::PTR;
     case net::DnsQueryType::SRV:
       return DnsQueryType::SRV;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -258,6 +262,8 @@ bool EnumTraits<DnsQueryType, net::DnsQu
     case DnsQueryType::SRV:
       *output = net::DnsQueryType::SRV;
       return true;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -274,6 +280,8 @@ EnumTraits<ResolveHostParameters::Source
       return ResolveHostParameters::Source::DNS;
     case net::HostResolverSource::MULTICAST_DNS:
       return ResolveHostParameters::Source::MULTICAST_DNS;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -294,6 +302,8 @@ bool EnumTraits<ResolveHostParameters::S
     case ResolveHostParameters::Source::MULTICAST_DNS:
       *output = net::HostResolverSource::MULTICAST_DNS;
       return true;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -309,6 +319,8 @@ EnumTraits<MdnsListenClient::UpdateType,
       return MdnsListenClient::UpdateType::CHANGED;
     case net::HostResolver::MdnsListener::Delegate::UpdateType::REMOVED:
       return MdnsListenClient::UpdateType::REMOVED;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -327,6 +339,8 @@ bool EnumTraits<MdnsListenClient::Update
     case MdnsListenClient::UpdateType::REMOVED:
       *output = net::HostResolver::MdnsListener::Delegate::UpdateType::REMOVED;
       return true;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/ws/public/cpp/gpu/command_buffer_metrics.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/services/ws/public/cpp/gpu/command_buffer_metrics.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/ws/public/cpp/gpu/command_buffer_metrics.cc
@@ -96,6 +96,8 @@ std::string ContextTypeToString(ContextT
       return "ForTesting";
     case ContextType::XR_COMPOSITING:
       return "XRCompositing";
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/core/layout/layout_flexible_box.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/core/layout/layout_flexible_box.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/core/layout/layout_flexible_box.cc
@@ -639,6 +639,8 @@ LayoutUnit LayoutFlexibleBox::FlowAwareC
       return ContentLeft();
     case TransformedWritingMode::kRightToLeftWritingMode:
       return ContentInsetRight();
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
@@ -654,6 +656,8 @@ LayoutUnit LayoutFlexibleBox::FlowAwareC
       return ContentInsetRight();
     case TransformedWritingMode::kRightToLeftWritingMode:
       return ContentLeft();
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.cc
@@ -18,7 +18,7 @@ WTF::String OverscrollBehaviorTypeToStri
     case OverscrollBehavior::kOverscrollBehaviorTypeContain:
       return "contain";
     default:
-      NOTREACHED();
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/platform/scheduler/common/metrics_helper.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/platform/scheduler/common/metrics_helper.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/platform/scheduler/common/metrics_helper.cc
@@ -44,6 +44,8 @@ scheduling_metrics::ThreadType ConvertBl
     case WebThreadType::kCount:
       NOTREACHED();
       return scheduling_metrics::ThreadType::kCount;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/ui/aura/mus/in_flight_change.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/ui/aura/mus/in_flight_change.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/ui/aura/mus/in_flight_change.cc
@@ -57,6 +57,8 @@ std::string ChangeTypeToString(ChangeTyp
       return "TRANSFORM";
     case ChangeType::VISIBLE:
       return "VISIBLE";
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/ui/gfx/skia_vector_animation.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/ui/gfx/skia_vector_animation.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/ui/gfx/skia_vector_animation.cc
@@ -163,6 +163,8 @@ float SkiaVectorAnimation::GetCurrentPro
       // yet
       DCHECK(timer_control_);
       return timer_control_->GetNormalizedCurrentCycleProgress();
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc
@@ -1237,6 +1237,8 @@ ax::mojom::TextPosition AXLayoutObject::
       return ax::mojom::TextPosition::kSubscript;
     case EVerticalAlign::kSuper:
       return ax::mojom::TextPosition::kSuperscript;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/accessibility/ax_position.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/modules/accessibility/ax_position.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/accessibility/ax_position.cc
@@ -768,6 +768,8 @@ const AXObject* AXPosition::FindNeighbor
       }
       return nullptr;
     }
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/background_fetch/background_fetch_registration.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/modules/background_fetch/background_fetch_registration.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/background_fetch/background_fetch_registration.cc
@@ -351,6 +351,8 @@ const String BackgroundFetchRegistration
       return "failure";
     case mojom::BackgroundFetchResult::UNSET:
       return "";
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
@@ -371,6 +373,8 @@ const String BackgroundFetchRegistration
       return "quota-exceeded";
     case mojom::BackgroundFetchFailureReason::DOWNLOAD_TOTAL_EXCEEDED:
       return "download-total-exceeded";
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/storage/cached_storage_area.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/modules/storage/cached_storage_area.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/storage/cached_storage_area.cc
@@ -643,6 +643,8 @@ Vector<uint8_t> CachedStorageArea::Strin
                   input.length() * sizeof(UChar));
       return result;
     }
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/renderer/media/stream/media_stream_audio_processor_options.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/content/renderer/media/stream/media_stream_audio_processor_options.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/renderer/media/stream/media_stream_audio_processor_options.cc
@@ -98,6 +98,8 @@ AudioProcessingProperties::ToAudioProces
         return media::EchoCancellationType::kAec3;
       case EchoCancellationType::kEchoCancellationSystem:
         return media::EchoCancellationType::kSystemAec;
+      default:
+        __builtin_abort ();
     }
   };
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/webrtc/modules/video_coding/codecs/vp8/temporal_layers_checker.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/webrtc/modules/video_coding/codecs/vp8/temporal_layers_checker.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/webrtc/modules/video_coding/codecs/vp8/temporal_layers_checker.cc
@@ -26,6 +26,8 @@ TemporalLayersChecker::CreateTemporalLay
     case Vp8TemporalLayersType::kBitrateDynamic:
       // Conference mode temporal layering for screen content in base stream.
       return absl::make_unique<TemporalLayersChecker>(num_temporal_layers);
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/base/task/common/operations_controller.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/base/task/common/operations_controller.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/base/task/common/operations_controller.cc
@@ -52,6 +52,8 @@ OperationsController::OperationToken Ope
     case State::kShuttingDown:
       DecrementBy(1);
       return OperationToken(nullptr);
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -101,4 +103,4 @@ void OperationsController::DecrementBy(u
 }
 
 }  // namespace internal
-}  // namespace base
\ No newline at end of file
+}  // namespace base
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/components/viz/common/gl_scaler.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/components/viz/common/gl_scaler.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/components/viz/common/gl_scaler.cc
@@ -692,6 +692,8 @@ const char* GLScaler::GetShaderName(GLSc
     CASE_RETURN_SHADER_STR(PLANAR_CHANNEL_3);
     CASE_RETURN_SHADER_STR(I422_NV61_MRT);
     CASE_RETURN_SHADER_STR(DEINTERLEAVE_PAIRWISE_MRT);
+    default:
+      __builtin_abort ();
 #undef CASE_RETURN_SHADER_STR
   }
 }
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/components/viz/service/display/skia_renderer.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/components/viz/service/display/skia_renderer.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/components/viz/service/display/skia_renderer.cc
@@ -1305,6 +1305,8 @@ GrContext* SkiaRenderer::GetGrContext()
       return context_provider_->GrContext();
     case DrawMode::SKPRECORD:
       return nullptr;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/browser/cache_storage/cache_storage_cache_entry_handler.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/content/browser/cache_storage/cache_storage_cache_entry_handler.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/browser/cache_storage/cache_storage_cache_entry_handler.cc
@@ -162,6 +162,8 @@ CacheStorageCacheEntryHandler::CreateCac
     case CacheStorageOwner::kBackgroundFetch:
       return std::make_unique<background_fetch::CacheEntryHandlerImpl>(
           std::move(blob_context));
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/browser/cache_storage/cache_storage_histogram_utils.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/content/browser/cache_storage/cache_storage_histogram_utils.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/browser/cache_storage/cache_storage_histogram_utils.cc
@@ -33,6 +33,8 @@ base::StringPiece UMAToName(CacheStorage
       RETURN_LITERAL_STRING_PIECE("QueueDuration2");
     case CacheStorageSchedulerUMA::kQueueLength:
       RETURN_LITERAL_STRING_PIECE("QueueLength");
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -44,6 +46,8 @@ base::StringPiece ClientToName(CacheStor
       RETURN_LITERAL_STRING_PIECE("Cache");
     case CacheStorageSchedulerClient::kStorage:
       RETURN_LITERAL_STRING_PIECE("CacheStorage");
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -88,6 +92,8 @@ base::StringPiece OpToName(CacheStorageS
       RETURN_LITERAL_STRING_PIECE("WriteIndex");
     case CacheStorageSchedulerOp::kWriteSideData:
       RETURN_LITERAL_STRING_PIECE("WriteSideData");
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/browser/media/hardware_key_media_controller.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/content/browser/media/hardware_key_media_controller.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/browser/media/hardware_key_media_controller.cc
@@ -182,6 +182,8 @@ HardwareKeyMediaController::MediaSession
     case MediaSessionAction::kSeekForward:
     case MediaSessionAction::kSkipAd:
       return base::nullopt;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/common/navigation_params.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/content/common/navigation_params.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/common/navigation_params.cc
@@ -47,6 +47,8 @@ bool IsNavigationDownloadAllowed(Navigat
     case NavigationDownloadPolicy::kAllowOpenerCrossOrigin:
     case NavigationDownloadPolicy::kAllowOpenerCrossOriginNoGesture:
       return true;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/renderer/media/stream/media_stream_constraints_util_audio.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/content/renderer/media/stream/media_stream_constraints_util_audio.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/content/renderer/media/stream/media_stream_constraints_util_audio.cc
@@ -417,6 +417,8 @@ class EchoCancellationContainer {
         return Score::EcModeScore::kAec3;
       case EchoCancellationType::kEchoCancellationAec2:
         return Score::EcModeScore::kAec2;
+      default:
+        __builtin_abort ();
     }
   }
 
@@ -968,6 +970,8 @@ class ProcessingBasedContainer {
         return DoubleRangeSet::FromValue(allowed_latency);
       case ProcessingType::kUnprocessed:
         return DoubleRangeSet::FromValue(allowed_latency);
+      default:
+        __builtin_abort ();
     }
   }
 
@@ -988,6 +992,8 @@ class ProcessingBasedContainer {
         return 2;
       case ProcessingType::kApmProcessed:
         return use_processing_by_default ? 3 : 1;
+      default:
+        __builtin_abort ();
     }
   }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/google_apis/gaia/google_service_auth_error.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/google_apis/gaia/google_service_auth_error.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/google_apis/gaia/google_service_auth_error.cc
@@ -29,6 +29,8 @@ const char* InvalidCredentialsReasonToSt
     case InvalidGaiaCredentialsReason::NUM_REASONS:
       NOTREACHED();
       return "";
+    default:
+      __builtin_abort ();
   }
 }
 }  // namespace
@@ -238,6 +240,8 @@ std::string GoogleServiceAuthError::ToSt
     case NUM_STATES:
       NOTREACHED();
       return std::string();
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/audio/audio_input_device.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/media/audio/audio_input_device.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/audio/audio_input_device.cc
@@ -52,6 +52,8 @@ base::ThreadPriority ThreadPriorityFromP
       return base::ThreadPriority::REALTIME_AUDIO;
     case AudioInputDevice::Purpose::kLoopback:
       return base::ThreadPriority::NORMAL;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/base/supported_types.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/media/base/supported_types.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/media/base/supported_types.cc
@@ -162,6 +162,8 @@ bool IsAudioCodecProprietary(AudioCodec
     case media::kCodecPCM_ALAW:
     case media::kUnknownAudioCodec:
       return false;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -220,6 +222,8 @@ bool IsVideoCodecProprietary(VideoCodec
     case kCodecVP9:
     case kCodecAV1:
       return false;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/net/dns/dns_util.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/net/dns/dns_util.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/net/dns/dns_util.cc
@@ -246,6 +246,8 @@ uint16_t DnsQueryTypeToQtype(DnsQueryTyp
       return dns_protocol::kTypePTR;
     case DnsQueryType::SRV:
       return dns_protocol::kTypeSRV;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/net/dns/host_resolver_mdns_listener_impl.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/net/dns/host_resolver_mdns_listener_impl.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/net/dns/host_resolver_mdns_listener_impl.cc
@@ -24,6 +24,8 @@ HostResolver::MdnsListener::Delegate::Up
       return HostResolver::MdnsListener::Delegate::UpdateType::CHANGED;
     case net::MDnsListener::RECORD_REMOVED:
       return HostResolver::MdnsListener::Delegate::UpdateType::REMOVED;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/net/dns/host_resolver_mdns_task.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/net/dns/host_resolver_mdns_task.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/net/dns/host_resolver_mdns_task.cc
@@ -218,6 +218,8 @@ HostCache::Entry HostResolverMdnsTask::P
     case DnsQueryType::SRV:
       return ParseHostnameResult(parsed->rdata<SrvRecordRdata>()->target(),
                                  parsed->rdata<SrvRecordRdata>()->port());
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/net/third_party/quic/core/qpack/qpack_instruction_decoder.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/net/third_party/quic/core/qpack/qpack_instruction_decoder.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/net/third_party/quic/core/qpack/qpack_instruction_decoder.cc
@@ -169,6 +169,8 @@ size_t QpackInstructionDecoder::DoVarint
     case http2::DecodeStatus::kDecodeError:
       OnError("Encoded integer too large.");
       return bytes_consumed;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -194,6 +196,8 @@ size_t QpackInstructionDecoder::DoVarint
     case http2::DecodeStatus::kDecodeError:
       OnError("Encoded integer too large.");
       return bytes_consumed;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/net/third_party/quic/core/tls_handshaker.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/net/third_party/quic/core/tls_handshaker.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/net/third_party/quic/core/tls_handshaker.cc
@@ -111,6 +111,8 @@ EncryptionLevel TlsHandshaker::QuicEncry
       return ENCRYPTION_INITIAL;
     case ssl_encryption_application:
       return ENCRYPTION_FORWARD_SECURE;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/media_session/audio_focus_manager.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/services/media_session/audio_focus_manager.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/media_session/audio_focus_manager.cc
@@ -520,6 +520,8 @@ bool AudioFocusManager::ShouldSessionBeS
     case mojom::EnforcementMode::kDefault:
       NOTIMPLEMENTED();
       return false;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -537,6 +539,8 @@ bool AudioFocusManager::ShouldSessionBeD
     case mojom::EnforcementMode::kDefault:
       NOTIMPLEMENTED();
       return false;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/network/mdns_responder.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/services/network/mdns_responder.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/network/mdns_responder.cc
@@ -439,6 +439,8 @@ class MdnsResponderManager::SocketHandle
       case MdnsResponseSendOption::ResponseClass::UNSPECIFIED:
         NOTREACHED();
         return RateLimitScheme::PER_RESPONSE;
+      default:
+        __builtin_abort ();
     }
   }
   // Returns null if the computed delay exceeds kMaxScheduledDelay and the next
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/network/resource_scheduler.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/services/network/resource_scheduler.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/services/network/resource_scheduler.cc
@@ -79,6 +79,8 @@ const char* RequestStartTriggerString(Re
       return "REQUEST_REPRIORITIZED";
     case RequestStartTrigger::LONG_QUEUED_REQUESTS_TIMER_FIRED:
       return "LONG_QUEUED_REQUESTS_TIMER_FIRED";
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/core/css/properties/longhands/custom_property.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/core/css/properties/longhands/custom_property.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/core/css/properties/longhands/custom_property.cc
@@ -101,6 +101,8 @@ const CSSValue* CustomProperty::ParseSin
       if (registration_ && !ParseTyped(range, context, local_context))
         return nullptr;
       return ParseUntyped(range, context, local_context);
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/core/trustedtypes/trusted_types_util.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/core/trustedtypes/trusted_types_util.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/core/trustedtypes/trusted_types_util.cc
@@ -122,6 +122,8 @@ String GetStringFromSpecificTrustedType(
       return GetStringFromTrustedURL(string_or_trusted_url, doc,
                                      exception_state);
     }
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/idle/idle_status.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/modules/idle/idle_status.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/idle/idle_status.cc
@@ -87,6 +87,8 @@ String IdleStatus::state() const {
       return "idle";
     case IdleState::LOCKED:
       return "locked";
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/peerconnection/rtc_peer_connection.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/modules/peerconnection/rtc_peer_connection.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/peerconnection/rtc_peer_connection.cc
@@ -604,6 +604,8 @@ RTCSetSessionDescriptionOperation GetRTC
         return RTCSetSessionDescriptionOperation::kSetRemoteDescriptionAnswer;
       return RTCSetSessionDescriptionOperation::
           kSetRemoteDescriptionInvalidType;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -629,6 +631,8 @@ SdpUsageCategory DeduceSdpUsageCategory(
               sdp_semantics == webrtc::SdpSemantics::kUnifiedPlan)
                  ? SdpUsageCategory::kSafe
                  : SdpUsageCategory::kUnsafe;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/service_worker/wait_until_observer.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/modules/service_worker/wait_until_observer.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/service_worker/wait_until_observer.cc
@@ -188,6 +188,8 @@ bool WaitUntilObserver::IsEventActive(Sc
     case EventDispatchState::kDispatched:
     case EventDispatchState::kFailed:
       return false;
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/worklet/animation_and_paint_worklet_thread.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/modules/worklet/animation_and_paint_worklet_thread.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/modules/worklet/animation_and_paint_worklet_thread.cc
@@ -92,6 +92,8 @@ AnimationAndPaintWorkletThread::CreateWo
       return AnimationWorkletGlobalScope::Create(std::move(creation_params),
                                                  this);
     }
+    default:
+      __builtin_abort ();
     case WorkletType::PAINT_WORKLET:
       TRACE_EVENT0(TRACE_DISABLED_BY_DEFAULT("paint-worklet"),
                    "AnimationAndPaintWorkletThread::CreateWorkerGlobalScope");
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/platform/graphics/canvas_resource_provider.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/blink/renderer/platform/graphics/canvas_resource_provider.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/blink/renderer/platform/graphics/canvas_resource_provider.cc
@@ -509,6 +509,8 @@ const std::vector<CanvasResourceType>& G
       return kAcceleratedCompositedFallbackList;
     case CanvasResourceProvider::kAcceleratedDirectResourceUsage:
       return kAcceleratedDirectFallbackList;
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/webrtc/api/video_codecs/create_vp8_temporal_layers.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/webrtc/api/video_codecs/create_vp8_temporal_layers.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/webrtc/api/video_codecs/create_vp8_temporal_layers.cc
@@ -27,6 +27,8 @@ std::unique_ptr<Vp8TemporalLayers> Creat
       // Conference mode temporal layering for screen content in base stream.
       return absl::make_unique<ScreenshareLayers>(num_temporal_layers,
                                                   Clock::GetRealTimeClock());
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/webrtc/audio/channel_receive.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/webrtc/audio/channel_receive.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/webrtc/audio/channel_receive.cc
@@ -68,6 +68,8 @@ webrtc::FrameType WebrtcFrameTypeForMedi
         kDiscountinuousTransmission:
       return kAudioFrameCN;
       break;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/audio_processing_impl.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/audio_processing_impl.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/audio_processing_impl.cc
@@ -121,6 +121,7 @@ NoiseSuppression::Level NsConfigLevelToI
     case NsConfig::kVeryHigh:
       return NoiseSuppression::kVeryHigh;
     default:
+      __builtin_abort ();
       RTC_NOTREACHED();
   }
 }
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/webrtc/pc/dtls_transport.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/webrtc/pc/dtls_transport.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/webrtc/pc/dtls_transport.cc
@@ -33,6 +33,8 @@ DtlsTransportState TranslateState(cricke
     case cricket::DTLS_TRANSPORT_FAILED:
       return DtlsTransportState::kFailed;
       break;
+    default:
+      __builtin_abort ();
   }
 }
 
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/webrtc/rtc_base/key_derivation.cc
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/third_party/webrtc/rtc_base/key_derivation.cc
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/third_party/webrtc/rtc_base/key_derivation.cc
@@ -25,6 +25,8 @@ std::unique_ptr<KeyDerivation> KeyDeriva
   switch (key_derivation_algorithm) {
     case KeyDerivationAlgorithm::HKDF_SHA256:
       return absl::make_unique<OpenSSLKeyDerivationHKDF>();
+    default:
+      __builtin_abort ();
   }
   RTC_NOTREACHED();
 }
Index: qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/ui/accessibility/mojom/ax_tree_id_mojom_traits.h
===================================================================
--- qtwebengine-everywhere-src-5.13.0-beta2.orig/src/3rdparty/chromium/ui/accessibility/mojom/ax_tree_id_mojom_traits.h
+++ qtwebengine-everywhere-src-5.13.0-beta2/src/3rdparty/chromium/ui/accessibility/mojom/ax_tree_id_mojom_traits.h
@@ -19,6 +19,8 @@ struct UnionTraits<ax::mojom::AXTreeIDDa
         return ax::mojom::AXTreeIDDataView::Tag::UNKNOWN;
       case ax::mojom::AXTreeIDType::kToken:
         return ax::mojom::AXTreeIDDataView::Tag::TOKEN;
+      default:
+        __builtin_abort ();
     }
   }
   static uint8_t unknown(const ui::AXTreeID& p) { return 0; }
