diff -rup a/src/hdf5/ossimHdf5ImageDataset.cpp b/src/hdf5/ossimHdf5ImageDataset.cpp
--- a/src/hdf5/ossimHdf5ImageDataset.cpp	2018-09-12 22:14:50.000000000 +0200
+++ b/src/hdf5/ossimHdf5ImageDataset.cpp	2018-09-30 14:51:06.777826706 +0200
@@ -766,7 +766,7 @@ void ossimHdf5ImageDataset::getTileBuf(v
    catch( const H5::Exception& error )
    {
       ossimNotify(ossimNotifyLevel_WARN) << MODULE << " caught H5 Exception!" << std::endl;
-      error.printError();
+      error.printErrorStack();
    }
 
    catch( ... )
