From 958cc2b0d5cc17d0e5440196a579951e4008f449 Mon Sep 17 00:00:00 2001
From: "C.A.M. Gerlach" <CAM.Gerlach@Gerlach.CAM>
Date: Mon, 2 Mar 2020 21:06:19 -0600
Subject: [PATCH 1/3] Fix, improve and cleanup runtests.py script arg parsing

---
 runtests.py | 48 +++++++++++++++++++++++-------------------------
 1 file changed, 23 insertions(+), 25 deletions(-)

diff --git a/runtests.py b/runtests.py
index 7d16848cb2..5760e9a7fe 100644
--- a/runtests.py
+++ b/runtests.py
@@ -9,18 +9,18 @@
 """
 
 # Standard library imports
-import os
-import sys
 import argparse
+import os
 
-# To activate/deactivate certain things for pytest's only
+# To activate/deactivate certain things for pytests only
 # NOTE: Please leave this before any other import here!!
 os.environ['SPYDER_PYTEST'] = 'True'
 
 # Third party imports
 # NOTE: This needs to be imported before any QApplication.
 # Don't remove it or change it to a different location!
-from qtpy import QtWebEngineWidgets
+# pylint: disable=wrong-import-position
+from qtpy import QtWebEngineWidgets  # pylint: disable=unused-import
 import pytest
 
 
@@ -29,30 +29,24 @@
 RUN_SLOW = os.environ.get('RUN_SLOW', None) == 'true'
 
 
-def main(run_slow=False, extra_args=None):
-    """
-    Run pytest tests for Spyder.
-    """
-    pytest_args = ['spyder',
-                   '-vv',
-                   '-rw',
-                   '--durations=10']
+def run_pytest(run_slow=False, extra_args=None):
+    """Run pytest tests for Spyder."""
+    pytest_args = ['-vv', '-rw', '--durations=10']
 
     if CI:
         # Exit on first failure and show coverage
         pytest_args += ['-x', '--cov=spyder', '--no-cov-on-fail']
 
-        # Run slow tests only
-        if RUN_SLOW:
-            pytest_args += ['--run-slow']
-
         # To display nice tests resume in Azure's web page
         if os.environ.get('AZURE', None) is not None:
             pytest_args += ['--cache-clear', '--junitxml=result.xml']
-    elif run_slow:
+    if run_slow or RUN_SLOW:
         pytest_args += ['--run-slow']
-    elif extra_args:
+    # Allow user to pass a custom test path to pytest to e.g. run just one test
+    if extra_args:
         pytest_args += extra_args
+    else:
+        pytest_args += ['spyder']
 
     print("Pytest Arguments: " + str(pytest_args))
     errno = pytest.main(pytest_args)
@@ -64,13 +58,17 @@ def main(run_slow=False, extra_args=None):
         raise SystemExit(errno)
 
 
-if __name__ == '__main__':
+def main():
+    """Parse args then run the pytest suite for Spyder."""
     test_parser = argparse.ArgumentParser(
-        usage='python runtests.py [--run-slow] [-- pytest_args]')
-    test_parser.add_argument('--run-slow', action='store_true',
-                             default=False,
+        description="Helper script to run Spyder's test suite")
+    test_parser.add_argument('--run-slow', action='store_true', default=False,
                              help='Run the slow tests')
-    test_parser.add_argument('pytest_args', nargs='*',
-                             help="Args to pass to pytest")
+    test_parser.add_argument('pytest_args', nargs=argparse.REMAINDER,
+                             metavar="...", help="Args to pass to pytest")
     test_args = test_parser.parse_args()
-    main(run_slow=test_args.run_slow, extra_args=test_args.pytest_args)
+    run_pytest(run_slow=test_args.run_slow, extra_args=test_args.pytest_args)
+
+
+if __name__ == '__main__':
+    main()

From b83a72e4e7db2fa173fa16b8c69ad6830ed2fefa Mon Sep 17 00:00:00 2001
From: "C.A.M. Gerlach" <CAM.Gerlach@Gerlach.CAM>
Date: Mon, 2 Mar 2020 21:07:30 -0600
Subject: [PATCH 2/3] Ignore pytest args properly in Spyder mainwindow setup

---
 spyder/app/mainwindow.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/spyder/app/mainwindow.py b/spyder/app/mainwindow.py
index 71f9f4ed94..a3847c84c0 100644
--- a/spyder/app/mainwindow.py
+++ b/spyder/app/mainwindow.py
@@ -104,7 +104,12 @@
 from spyder.app.cli_options import get_options
 
 # Get CLI options/args and make them available for future use
-CLI_OPTIONS, CLI_ARGS = get_options()
+# Ignore args if running tests or Spyder will try and fail to parse pytests's
+if bool(os.environ.get('SPYDER_PYTEST')):
+    sys_argv = [sys.argv[0]]
+else:
+    sys_argv = sys.argv
+CLI_OPTIONS, CLI_ARGS = get_options(sys_argv)
 
 # **** Set OpenGL implementation to use ****
 if CLI_OPTIONS.opengl_implementation:

From 1a0caedbc4864b5c257097a435d9ad290e7c9b15 Mon Sep 17 00:00:00 2001
From: "C.A.M. Gerlach" <CAM.Gerlach@Gerlach.CAM>
Date: Mon, 2 Mar 2020 21:10:53 -0600
Subject: [PATCH 3/3] Add .pylint.d to gitignore

---
 .gitignore | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/.gitignore b/.gitignore
index b407abc4d8..a1ef21c876 100644
--- a/.gitignore
+++ b/.gitignore
@@ -34,4 +34,7 @@ spyder_crash.log
 # Rope project folders
 .ropeproject/
 .vscode/
-result.xml
\ No newline at end of file
+result.xml
+
+# Pylint dirs/files
+.pylint.d/
