Index: v4l-utils-1.28.1/meson.build
===================================================================
--- v4l-utils-1.28.1.orig/meson.build
+++ v4l-utils-1.28.1/meson.build
@@ -386,13 +386,17 @@ man_pages = []
 
 i18n_gettext_arguments = ['--directory=' + source_root,
                           '--keyword=_', '--keyword=N_', '--keyword=P_:1,2']
+if get_option('qv4l2').disabled()
 subdir('libdvbv5-po')
 subdir('v4l-utils-po')
 
 subdir('lib')
+else
+  ir_bpf_enabled = false
+endif
 
+subdir('utils')
 if get_option('v4l-utils')
-    subdir('utils')
     subdir('contrib')
 else
     ir_bpf_enabled = false
Index: v4l-utils-1.28.1/utils/qv4l2/meson.build
===================================================================
--- v4l-utils-1.28.1.orig/utils/qv4l2/meson.build
+++ v4l-utils-1.28.1/utils/qv4l2/meson.build
@@ -23,14 +23,17 @@ qv4l2_sources = files(
     'v4l2-tpg-core.c',
     'vbi-tab.cpp',
     'vbi-tab.h',
+    '../libv4l2util/frequencies.c',
 )
 
+dep_libv4l2 = dependency('libv4l2')
+dep_libv4lconvert = dependency('libv4lconvert')
+
 qv4l2_deps = [
     dep_alsa,
     dep_gl,
     dep_libmedia_dev,
     dep_libv4l2,
-    dep_libv4l2util,
     dep_libv4lconvert,
     dep_qt,
     dep_qt_opengl,
Index: v4l-utils-1.28.1/utils/meson.build
===================================================================
--- v4l-utils-1.28.1.orig/utils/meson.build
+++ v4l-utils-1.28.1/utils/meson.build
@@ -19,6 +19,7 @@ endforeach
 # Libraries
 subdir('libcecutil')
 subdir('libmedia_dev')
+if get_option('qv4l2').disabled()
 subdir('libv4l2util')
 
 # Utils
@@ -30,8 +31,10 @@ subdir('ir-ctl')
 subdir('ivtv-ctl')
 subdir('keytable')
 subdir('media-ctl')
+endif
 subdir('qv4l2')
 subdir('qvidcap')
+if get_option('qv4l2').disabled()
 subdir('rds-ctl')
 subdir('v4l2-ctl')
 subdir('v4l2-dbg')
@@ -41,3 +44,4 @@ subdir('v4l2-tracer')
 # Compliance tools
 subdir('cec-compliance')
 subdir('v4l2-compliance')
+endif
Index: v4l-utils-1.28.1/utils/qvidcap/meson.build
===================================================================
--- v4l-utils-1.28.1.orig/utils/qvidcap/meson.build
+++ v4l-utils-1.28.1/utils/qvidcap/meson.build
@@ -20,6 +20,7 @@ qvidcap_sources = files(
     'v4l2-info.cpp',
     'v4l2-tpg-colors.c',
     'v4l2-tpg-core.c',
+    '../libv4l2util/frequencies.c',
 )
 
 qvidcap_deps = [
@@ -27,7 +28,6 @@ qvidcap_deps = [
     dep_gl,
     dep_libmedia_dev,
     dep_libv4l2,
-    dep_libv4l2util,
     dep_libv4lconvert,
     dep_qt,
     dep_qt_opengl,
