diff --git a/utils/qv4l2/Makefile.am b/utils/qv4l2/Makefile.am
index ccd1a2a..6978cf3 100644
--- a/utils/qv4l2/Makefile.am
+++ b/utils/qv4l2/Makefile.am
@@ -6,8 +6,7 @@ qv4l2_SOURCES = qv4l2.cpp general-tab.cpp ctrl-tab.cpp vbi-tab.cpp capture-win.c
   raw2sliced.cpp qv4l2.h capture-win.h general-tab.h vbi-tab.h raw2sliced.h \
   v4l2-tpg-core.c v4l2-tpg-colors.c
 nodist_qv4l2_SOURCES = moc_qv4l2.cpp moc_general-tab.cpp moc_capture-win.cpp moc_vbi-tab.cpp qrc_qv4l2.cpp
-qv4l2_LDADD = ../../lib/libv4l2/libv4l2.la ../../lib/libv4lconvert/libv4lconvert.la \
-  ../libv4l2util/libv4l2util.la ../libmedia_dev/libmedia_dev.la
+qv4l2_LDADD = ../libv4l2util/libv4l2util.la ../libmedia_dev/libmedia_dev.la
 qv4l2_CPPFLAGS = -I$(top_srcdir)/utils/common
 
 if WITH_QTGL
@@ -21,6 +20,8 @@ endif
 qv4l2_CPPFLAGS += $(ALSA_CFLAGS)
 qv4l2_LDFLAGS += $(ALSA_LIBS) -pthread
 
+qv4l2_LDFLAGS += -lv4l2 -lv4lconvert
+
 EXTRA_DIST = enterbutt.png exit.png fileopen.png qv4l2_24x24.png qv4l2_64x64.png qv4l2.png qv4l2.svg \
   snapshot.png video-television.png fileclose.png qv4l2_16x16.png qv4l2_32x32.png qv4l2.desktop \
   qv4l2.qrc saveraw.png qv4l2.pro qv4l2.1 fullscreenexit.png fullscreen.png start.png step.png
