diff -urEbw vapoursynth-R50/Makefile.am vapoursynth-R50.new/Makefile.am
--- vapoursynth-R50/Makefile.am	2020-05-08 17:10:41.000000000 +0200
+++ vapoursynth-R50.new/Makefile.am	2020-05-14 10:50:38.446478782 +0200
@@ -79,7 +79,7 @@
 
 pkgconfig_DATA += pc/vapoursynth.pc
 
-libvapoursynth_la_LDFLAGS = -no-undefined -avoid-version
+libvapoursynth_la_LDFLAGS = -no-undefined -release $(PACKAGE_VERSION)
 libvapoursynth_la_CPPFLAGS = $(ZIMG_CFLAGS) -DVS_PATH_PLUGINDIR='"$(PLUGINDIR)"'
 libvapoursynth_la_LIBADD = $(ZIMG_LIBS) $(DLOPENLIB) libexprfilter.la
 
