--- a/meson.build
+++ b/meson.build
@@ -223,7 +223,7 @@
     gnu_symbol_visibility: 'hidden',
     include_directories: incdir,
     install: true,
-    install_dir: install_dir,
+    install_dir: get_option('libdir'),
     name_prefix: '',
     soversion: soversion,
 )
@@ -262,7 +262,8 @@
     gnu_symbol_visibility: 'hidden',
     include_directories: incdir,
     install: true,
-    install_dir: install_dir,
+    install_dir: get_option('libdir'),
+    soversion: '0',
 )
 
 vsscript_dep = declare_dependency(
@@ -288,7 +289,7 @@
     dependencies: vsscript_dep,
     gnu_symbol_visibility: 'hidden',
     install: true,
-    install_dir: install_dir,
+    install_dir: get_option('bindir'),
     install_rpath: '$ORIGIN',
     link_args: link_args,
 )
@@ -494,15 +495,19 @@
     'include',
     exclude_directories: 'cython',
     exclude_files: ['VapourSynth.h', 'VSHelper.h'],
-    install_dir: install_dir,
+    strip_directory: true,
+    install_dir: get_option('includedir') / 'vapoursynth',
 )
 
 pc_data = configuration_data()
 pc_data.set('version', meson.project_version())
+pc_data.set('prefix', get_option('prefix'))
+pc_data.set('includedir', get_option('prefix') / get_option('includedir'))
+pc_data.set('libdir', get_option('prefix') / get_option('libdir'))
 configure_file(
     configuration: pc_data,
     input: 'vapoursynth.pc.in',
     install: true,
-    install_dir: install_dir / 'pkgconfig',
+    install_dir: get_option('libdir') / 'pkgconfig',
     output: 'vapoursynth.pc',
 )
--- a/vapoursynth.pc.in
+++ b/vapoursynth.pc.in
@@ -1,7 +1,9 @@
-prefix=${pcfiledir}/..
-includedir=${prefix}/include
+prefix=@prefix@
+includedir=@includedir@
+libdir=@libdir@
 
 Name: vapoursynth
 Description: A frameserver for the 21st century
 Version: @version@
-Cflags: -I${includedir}
+Cflags: -I${includedir}/vapoursynth
+Libs: -L${libdir} -lvapoursynth
