--- psion/Makefile
+++ psion/Makefile
@@ -22,7 +22,7 @@
 	$(ECMD)install $(PSION_TARGETS) $(ROOT)/usr/bin
 
 
-CFLAGS += -g -I../include -Wall -Wstrict-prototypes $(RPM_OPT_FLAGS)
+CFLAGS += -g -I../include -Wall -Wstrict-prototypes $(RPM_OPT_FLAGS) -fno-strict-aliasing
 irpsion5: 
 	$(prn_cc_o)
-	$(ECMD)$(CC) $(CFLAGS) $(PSION_SRC) -o $@
\ No newline at end of file
+	$(ECMD)$(CC) $(CFLAGS) $(PSION_SRC) -o $@
