Index: src/modules/avformat/vdpau.c
===================================================================
--- src/modules/avformat/vdpau.c.orig	2017-08-25 10:06:04.343316776 +0200
+++ src/modules/avformat/vdpau.c	2017-08-25 10:06:22.887952653 +0200
@@ -73,18 +73,7 @@ static int vdpau_init( producer_avformat
 	if ( !vdpau_init_done )
 	{
 		int flags = RTLD_NOW;
-		object = dlopen( "/usr/lib/libvdpau.so", flags );
-#ifdef ARCH_X86_64
-		if ( !object )
-			object = dlopen( "/usr/lib64/libvdpau.so", flags );
-		if ( !object )
-			object = dlopen( "/usr/lib/x86_64-linux-gnu/libvdpau.so.1", flags );
-#elif ARCH_X86
-		if ( !object )
-			object = dlopen( "/usr/lib/i386-linux-gnu/libvdpau.so.1", flags );
-#endif
-		if ( !object )
-			object = dlopen( "/usr/local/lib/libvdpau.so", flags );
+		object = dlopen( "__VDPAU_SONAME__", flags );
 		if ( object )
 			vdpau_device_create_x11 = dlsym( object, "vdp_device_create_x11" );
 		else
