Index: mod_nss-1.0.14/migrate.pl
===================================================================
--- mod_nss-1.0.14.orig/migrate.pl	2016-04-15 20:27:59.000000000 +0200
+++ mod_nss-1.0.14/migrate.pl	2016-04-16 11:50:59.588366719 +0200
@@ -6,7 +6,7 @@ use Cwd;
 use Getopt::Std;
 
 BEGIN {
-   $NSSDir = cwd();
+   $NSSDir = "/etc/apache2/mod_nss.d";
 
    $SSLCACertificatePath = "";
    $SSLCACertificateFile = "";
