diff -Nru azure-mgmt-trafficmanager-0.51.0.orig/setup.py azure-mgmt-trafficmanager-0.51.0/setup.py
--- azure-mgmt-trafficmanager-0.51.0.orig/setup.py	2019-04-01 13:14:50.048351217 +0200
+++ azure-mgmt-trafficmanager-0.51.0/setup.py	2019-04-01 13:15:03.592477035 +0200
@@ -81,7 +81,4 @@
         'msrestazure>=0.4.32,<2.0.0',
         'azure-common>=1.1',
     ],
-    extras_require={
-        ":python_version<'3.0'": ['azure-mgmt-nspkg'],
-    }
 )
