Index: x3270/Examples/peer_script.bash
===================================================================
--- x3270/Examples/peer_script.bash.orig
+++ x3270/Examples/peer_script.bash
@@ -60,9 +60,9 @@ ip=/tmp/ip.$$
 op=/tmp/op.$$
 rm -f $ip $op
 trap "rm -f $ip $op" EXIT
-trap "exit" INT QUIT HUP TERM
-mknod $ip p
-mknod $op p
+trap "exit" INT QUIT HUP BUS PIPE TERM
+mknod $ip p || exit 1
+mknod $op p || exit 1
 
 # Start x3270
 x3270 -script -model 2 <$ip >$op &
Index: x3270/Examples/peer_script.sh
===================================================================
--- x3270/Examples/peer_script.sh.orig
+++ x3270/Examples/peer_script.sh
@@ -59,9 +59,9 @@ ip=/tmp/ip.$$
 op=/tmp/op.$$
 rm -f $ip $op
 trap "rm -f $ip $op" 0
-trap "exit" 2 3 1 15
-mknod $ip p
-mknod $op p
+trap "exit" 2 3 1 7 13 15
+mknod $ip p || exit 1
+mknod $op p || exit 1
 
 # Start x3270
 x3270 -script -model 2 <$ip >$op &
