diff --git a/common-src/testutils.h b/common-src/testutils.h
index 31f89737..2e9bb0b3 100644
--- a/common-src/testutils.h
+++ b/common-src/testutils.h
@@ -75,7 +75,7 @@ typedef struct TestUtilsTest {
 #define tu_dbg(...) if (tu_debugging_enabled) { g_fprintf(stderr, __VA_ARGS__); }
 
 /* Is debugging enabled for this test run? (set internally) */
-int tu_debugging_enabled;
+extern gboolean tu_debugging_enabled;
 
 /*
  * Main loop
