---
 Documentation/asciidoc.conf.in |    2 ++
 1 file changed, 2 insertions(+)

Index: git-2.48.0/Documentation/asciidoc.conf.in
===================================================================
--- git-2.48.0.orig/Documentation/asciidoc.conf.in
+++ git-2.48.0/Documentation/asciidoc.conf.in
@@ -24,6 +24,8 @@ litdd=&#45;&#45;
 manmanual=Git Manual
 mansource=Git @GIT_VERSION@
 revdate=@GIT_DATE@
+# drops the "last-updated" footer, with asciidoc-8.6.9+
+footer-style=none
 
 ifdef::doctype-book[]
 [titles]
