---
 contrib/completion/git-completion.zsh |    1 +
 1 file changed, 1 insertion(+)

Index: git-2.30.0/contrib/completion/git-completion.zsh
===================================================================
--- git-2.30.0.orig/contrib/completion/git-completion.zsh
+++ git-2.30.0/contrib/completion/git-completion.zsh
@@ -36,6 +36,7 @@ if [ -z "$script" ]; then
 		"$(dirname ${funcsourcetrace[1]%:*})"/git-completion.bash
 		"$HOME/.local/share/bash-completion/completions/git"
 		"$bash_completion/git"
+		'/etc/bash_completion.d/git.sh' # suse
 		'/etc/bash_completion.d/git' # old debian
 		)
 	for e in $locations; do
