--- FreeWnn-1.1.1-a022.bak/makerule.mk.in	2013-06-07 03:24:38.000000000 +0900
+++ FreeWnn-1.1.1-a022/makerule.mk.in	2013-08-07 14:15:30.506662233 +0900
@@ -136,7 +136,7 @@
     exec_prefix = @exec_prefix@
          bindir = @bindir@
         sbindir = @sbindir@
-      WNNWNNDIR = $(exec_prefix)/lib/wnn
+      WNNWNNDIR = /etc/FreeWnn
       WNNBINDIR = $(bindir)
      WNNSBINDIR = @WNNSBINDIR@
       WNNINCDIR = @includedir@
@@ -157,12 +157,12 @@
   WNNCONSDICSRC = $(IMTOP)/Wnn-consortium/dic
 
          WNNTOP = $(IMTOP)/Wnn
-       JWNNLANG = ja_JP
+       JWNNLANG = ja
      JWNNWNNDIR = $(WNNWNNDIR)/$(JWNNLANG)
      JWNNBINDIR = $(WNNBINDIR)$(JWNNSUBBINDIR)
     JWNNSBINDIR = $(WNNSBINDIR)$(JWNNSUBBINDIR)
      JWNNINCDIR = $(WNNINCDIR)/wnn
-     JWNNDICDIR = $(JWNNWNNDIR)/dic
+     JWNNDICDIR = /var/lib/wnn/$(JWNNLANG)/dic
   JWNNPUBDICDIR = $(JWNNDICDIR)/pubdic
 JWNNWNNCONSDICDIR = $(JWNNDICDIR)/wnncons
   JWNNFZKSRCDIR = $(JWNNDICDIR)/src
@@ -193,7 +193,7 @@
      CWNNINCDIR = $(WNNINCDIR)/cwnn
      CWNNBINDIR = $(WNNBINDIR)$(CWNNSUBBINDIR)
     CWNNSBINDIR = $(WNNSBINDIR)$(CWNNSUBBINDIR)
-     CWNNDICDIR = $(CWNNWNNDIR)/dic
+     CWNNDICDIR = /var/lib/wnn/$(CWNNLANG)/dic
   CWNNSYSDICDIR = $(CWNNDICDIR)/sys
       CWNNCDSRC = $(CWNNTOP)/cd
     CWNNCDICSRC = $(CWNNTOP)/cdic
@@ -217,7 +217,7 @@
      KWNNINCDIR = $(WNNINCDIR)/kwnn
      KWNNBINDIR = $(WNNBINDIR)$(KWNNSUBBINDIR)
     KWNNSBINDIR = $(WNNSBINDIR)$(KWNNSUBBINDIR)
-     KWNNDICDIR = $(KWNNWNNDIR)/dic
+     KWNNDICDIR = /var/lib/wnn/$(KWNNLANG)/dic
   KWNNSYSDICDIR = $(KWNNDICDIR)/sys
       KWNNKDSRC = $(KWNNTOP)/kd
     KWNNKDICSRC = $(KWNNTOP)/kdic
@@ -237,7 +237,7 @@
 
        TWNNLANG = zh_TW
      TWNNWNNDIR = $(WNNWNNDIR)/$(TWNNLANG)
-     TWNNDICDIR = $(TWNNWNNDIR)/dic
+     TWNNDICDIR = /var/lib/wnn/$(TWNNLANG)/dic
   TWNNSYSDICDIR = $(TWNNDICDIR)/sys
       TWNNTDSRC = $(CWNNTOP)/td
     TWNNTDICSRC = $(CWNNTOP)/tdic
