Index: FreeWnn-1.1.1-a023/Contrib/dic/gerodic/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/Contrib/dic/gerodic/Makefile.in
+++ FreeWnn-1.1.1-a023/Contrib/dic/gerodic/Makefile.in
@@ -70,9 +70,7 @@ $(OBJ): $(SRC)
 install:: ${OBJ}
 	@if [ -d $(DESTDIR)$(GERODIR) ]; then set +x; \
 	else (set -x; $(MKDIRHIER) $(DESTDIR)$(GERODIR)); fi
-	chown ${WNNOWNER} $(DESTDIR)$(GERODIR)
-	$(INSTALL) $(INSTDATFLAGS) $(LOCAL_INSTFLAGS) ${OBJ} $(DESTDIR)$(GERODIR)
-	${WNNTOUCH} $(DESTDIR)$(GERODIR)/${OBJ}
+	$(INSTALL) $(INSTDATFLAGS) ${OBJ} $(DESTDIR)$(GERODIR)
 
 clean::
 	$(RM) ${OBJ}
Index: FreeWnn-1.1.1-a023/Wnn/jserver/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/Wnn/jserver/Makefile.in
+++ FreeWnn-1.1.1-a023/Wnn/jserver/Makefile.in
@@ -36,7 +36,7 @@ include @top_builddir@/makerule.mk
 top_builddir = @top_builddir@
 
 INSTALL         = @INSTALL@ $(INSTALLFLAGS)
-LOCAL_INSTFLAGS = -o $(WNNOWNER)
+LOCAL_INSTFLAGS =
 
  JSERVERDEFINES = -DJSERVER -DWRITE_CHECK
        INCLUDES = -I$(WNNINCLUDESRC) -I$(top_builddir) -I$(srcdir)
@@ -104,7 +104,7 @@ instserver:: install
 install:: jserver
 	@if [ -d $(DESTDIR)$(JWNNSBINDIR) ]; then set +x; \
 	else (set -x; $(MKDIRHIER) $(DESTDIR)$(JWNNSBINDIR)); fi
-	$(LT_INSTALL) $(INSTPGMFLAGS) $(INSTUIDFLAGS) $(LOCAL_INSTFLAGS) jserver $(DESTDIR)$(JWNNSBINDIR)/`echo jserver | sed '$(transform)'`
+	$(LT_INSTALL) $(INSTPGMFLAGS) $(INSTBINFLAGS) $(LOCAL_INSTFLAGS) jserver $(DESTDIR)$(JWNNSBINDIR)/`echo jserver | sed '$(transform)'`
 
 $(OBJ0) : $(SRC0)
 	$(CC) -c $(CFLAGS) $(SERVERINITDEF) $(SOLINGERDEFS) $(srcdir)/`basename $@ .o`.c
Index: FreeWnn-1.1.1-a023/Wnn/jutil/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/Wnn/jutil/Makefile.in
+++ FreeWnn-1.1.1-a023/Wnn/jutil/Makefile.in
@@ -37,7 +37,7 @@ top_builddir = @top_builddir@
 INSTALL         = @INSTALL@ $(INSTALLFLAGS)
 LT_INSTALL      = $(LIBTOOL) --mode=install $(INSTALL)
 LT_LD           = $(LIBTOOL) --mode=link $(CCLINK)
-LOCAL_INSTFLAGS = -o $(WNNOWNER)
+LOCAL_INSTFLAGS =
 
        INCLUDES = -I$(top_builddir) @EXTWNNJINC@ -I$(WNNINCLUDESRC)
         DEFINES = $(WNNDEFINES) $(WNNLANGDEF)
Index: FreeWnn-1.1.1-a023/Wnn/pubdicplus/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/Wnn/pubdicplus/Makefile.in
+++ FreeWnn-1.1.1-a023/Wnn/pubdicplus/Makefile.in
@@ -35,7 +35,7 @@ top_builddir = @top_builddir@
  WNNJUTILOBJ = $(top_builddir)/Wnn/jutil
 
 INSTALL         = @INSTALL@ $(INSTALLFLAGS)
-LOCAL_INSTFLAGS = -o $(WNNOWNER)
+LOCAL_INSTFLAGS =
 
        INCLUDES = -I$(top_builddir)
            ATOD = $(WNNJUTILOBJ)/atod
@@ -281,8 +281,7 @@ instserver:: install
 install::
 	@case '${MFLAGS}' in *[i]*) set +e;; esac;
 	@for i in $(JWNNDICDIR); do if [ -d $(DESTDIR)$$i ]; then \
-	set +x; else (set -x; $(MKDIRHIER) $(DESTDIR)$$i; \
-	chown $(WNNOWNER) $(DESTDIR)$$i); fi \
+	set +x; else (set -x; $(MKDIRHIER) $(DESTDIR)$$i); fi \
 	done
 
 install:: $(DIC)
Index: FreeWnn-1.1.1-a023/Wnn/wnncons/dic/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/Wnn/wnncons/dic/Makefile.in
+++ FreeWnn-1.1.1-a023/Wnn/wnncons/dic/Makefile.in
@@ -33,7 +33,7 @@ include @top_srcdir@/makerule.mk
 top_builddir = ../../..
 
 INSTALL         = @INSTALL@ $(INSTALLFLAGS)
-LOCAL_INSTFLAGS = -o $(WNNOWNER)
+LOCAL_INSTFLAGS =
 
            ATOD = $(WNNJUTILSRC)/atod
        WNNTOUCH = $(WNNJUTILSRC)/wnntouch
@@ -75,7 +75,7 @@ install::
 	@case '${MFLAGS}' in *[i]*) set +e;; esac;
 	@for i in $(JWNNDICDIR); do if [ -d $(DESTDIR)$$i ]; then \
 	set +x; else (set -x; $(MKDIRHIER) $(DESTDIR)$$i; \
-	chown $(WNNOWNER) $(DESTDIR)$$i); fi \
+	); fi \
 	done
 
 install:: $(DICS)
@@ -86,7 +86,7 @@ install:: $(DICS)
 	set +x; \
 	else (set -x; \
 	$(INSTALL) $(INSTDATFLAGS) $(LOCAL_INSTFLAGS) $$i $(DESTDIR)$(JWNNWNNCONSDICDIR); \
-	$(WNNTOUCH) $(DESTDIR)$(JWNNWNNCONSDICDIR)/$$i); fi \
+	); fi \
 	done
 
 depend::
Index: FreeWnn-1.1.1-a023/Xwnmo/xjutil/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/Xwnmo/xjutil/Makefile.in
+++ FreeWnn-1.1.1-a023/Xwnmo/xjutil/Makefile.in
@@ -26,7 +26,7 @@ include @top_srcdir@/makerule.mk
 top_builddir = ../..
 
 INSTALL         = @INSTALL@ $(INSTALLFLAGS)
-LOCAL_INSTFLAGS = -o $(WNNOWNER)
+LOCAL_INSTFLAGS =
 
        INCLUDES = -I. $(XWNMOINCLUDES) -I$(XWNMOSRC)
 
Index: FreeWnn-1.1.1-a023/Xwnmo/xwnmo/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/Xwnmo/xwnmo/Makefile.in
+++ FreeWnn-1.1.1-a023/Xwnmo/xwnmo/Makefile.in
@@ -26,7 +26,7 @@ include @top_srcdir@/makerule.mk
 top_builddir = ../..
 
 INSTALL         = @INSTALL@ $(INSTALLFLAGS)
-LOCAL_INSTFLAGS = -o $(WNNOWNER)
+LOCAL_INSTFLAGS =
 
        INCLUDES = $(XWNMOINCLUDES)
 
Index: FreeWnn-1.1.1-a023/cWnn/cdic/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/cWnn/cdic/Makefile.in
+++ FreeWnn-1.1.1-a023/cWnn/cdic/Makefile.in
@@ -32,7 +32,7 @@ top_builddir = @top_builddir@
 CWNNJUTILOBJ = $(top_builddir)/cWnn/jutil
 
 INSTALL         = @INSTALL@ $(INSTALLFLAGS)
-LOCAL_INSTFLAGS = -o $(WNNOWNER)
+LOCAL_INSTFLAGS =
 
         ATOD = $(CWNNJUTILOBJ)/catod
         ATOF = $(CWNNJUTILOBJ)/catof
@@ -86,8 +86,7 @@ instserver:: install
 install::
 	@case '${MFLAGS}' in *[i]*) set +e;; esac;
 	@for i in $(CWNNDICDIR); do if [ -d $(DESTDIR)$$i ]; then \
-	set +x; else (set -x; $(MKDIRHIER) $(DESTDIR)$$i; \
-	chown $(WNNOWNER) $(DESTDIR)$$i); fi \
+	set +x; else (set -x; $(MKDIRHIER) $(DESTDIR)$$i); fi \
 	done
 
 install:: $(ALLTGT)
Index: FreeWnn-1.1.1-a023/cWnn/jserver/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/cWnn/jserver/Makefile.in
+++ FreeWnn-1.1.1-a023/cWnn/jserver/Makefile.in
@@ -39,7 +39,7 @@ INSTALL      = @INSTALL@ $(INSTALLFLAGS)
  JSERVERDEFINES = -DJSERVER -DWRITE_CHECK
        INCLUDES = -I$(CWNNINCLUDESRC) -I$(top_builddir) -I$(srcdir)
         DEFINES = $(CWNNDEFINES) $(JSERVERDEFINES) $(WNNTERM) $(CWNNLANGDEF)
-LOCAL_INSTFLAGS = $(INSTUIDFLAGS) -o $(WNNOWNER)
+LOCAL_INSTFLAGS =
   SERVERINITDEF = -DSERVER_INIT_FILE=\"/$(CWNNLANG)/cserverrc\"
     HINSIDEFINE = -DHINSIDATA_FILE=\"/$(CWNNLANG)/cixing.data\"
  TSERVERINITDEF = -DSERVER_INIT_FILE=\"/$(TWNNLANG)/tserverrc\"
Index: FreeWnn-1.1.1-a023/cWnn/jutil/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/cWnn/jutil/Makefile.in
+++ FreeWnn-1.1.1-a023/cWnn/jutil/Makefile.in
@@ -36,7 +36,7 @@ top_builddir = @top_builddir@
 INSTALL         = @INSTALL@ $(INSTALLFLAGS)
 LT_INSTALL      = $(LIBTOOL) --mode=install $(INSTALL)
 LT_LD           = $(LIBTOOL) --mode=link $(CCLINK)
-LOCAL_INSTFLAGS = -o $(WNNOWNER)
+LOCAL_INSTFLAGS =
 
        INCLUDES = -I$(top_builddir) @EXTCWNNJINC@ -I$(CWNNINCLUDESRC)
         DEFINES = $(CWNNDEFINES) $(CWNNLANGDEF)
Index: FreeWnn-1.1.1-a023/cWnn/tdic/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/cWnn/tdic/Makefile.in
+++ FreeWnn-1.1.1-a023/cWnn/tdic/Makefile.in
@@ -32,7 +32,7 @@ top_builddir = @top_builddir@
 CWNNJUTILOBJ = $(top_builddir)/cWnn/jutil
 
 INSTALL      = @INSTALL@
-INSTALLFLAGS = -o $(WNNOWNER)
+INSTALLFLAGS =
 
         ATOD = $(CWNNJUTILOBJ)/catod
         ATOF = $(CWNNJUTILOBJ)/catof
@@ -86,8 +86,7 @@ instserver:: install
 install::
 	@case '${MFLAGS}' in *[i]*) set +e;; esac;
 	@for i in $(TWNNDICDIR); do if [ -d $(DESTDIR)$$i ]; then \
-	set +x; else (set -x; $(MKDIRHIER) $(DESTDIR)$$i; \
-	chown $(WNNOWNER) $(DESTDIR)$$i); fi \
+	set +x; else (set -x; $(MKDIRHIER) $(DESTDIR)$$i); fi \
 	done
 
 install:: $(ALLTGT)
Index: FreeWnn-1.1.1-a023/kWnn/jserver/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/kWnn/jserver/Makefile.in
+++ FreeWnn-1.1.1-a023/kWnn/jserver/Makefile.in
@@ -38,7 +38,7 @@ INSTALL      = @INSTALL@
  JSERVERDEFINES = -DJSERVER -DWRITE_CHECK
        INCLUDES = -I$(KWNNJSERVERSRC) -I$(KWNNINCLUDESRC) -I$(top_builddir)
         DEFINES = $(KWNNDEFINES) $(JSERVERDEFINES) $(WNNTERM) $(KWNNLANGDEF)
-LOCAL_INSTFLAGS = $(INSTUIDFLAGS) -o $(WNNOWNER)
+LOCAL_INSTFLAGS =
   SERVERINITDEF = -DSERVER_INIT_FILE=\"/$(KWNNLANG)/kserverrc\"
     HINSIDEFINE = -DHINSIDATA_FILE=\"/$(KWNNLANG)/hinsi.data\"
         WRAPLIB = @WRAPLIB@
Index: FreeWnn-1.1.1-a023/kWnn/kdic/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/kWnn/kdic/Makefile.in
+++ FreeWnn-1.1.1-a023/kWnn/kdic/Makefile.in
@@ -32,7 +32,7 @@ top_builddir = @top_builddir@
 KWNNJUTILOBJ = $(top_builddir)/kWnn/jutil
 
 INSTALL      = @INSTALL@
-INSTALLFLAGS = -o $(WNNOWNER)
+INSTALLFLAGS =
 
         ATOD = $(KWNNJUTILOBJ)/katod
         ATOF = $(KWNNJUTILOBJ)/katof
@@ -77,8 +77,7 @@ instserver:: install
 install::
 	@case '${MFLAGS}' in *[i]*) set +e;; esac;
 	@for i in $(KWNNDICDIR); do if [ -d $(DESTDIR)$$i ]; then \
-	set +x; else (set -x; $(MKDIRHIER) $(DESTDIR)$$i; \
-	chown $(WNNOWNER) $(DESTDIR)$$i); fi \
+	set +x; else (set -x; $(MKDIRHIER) $(DESTDIR)$$i); fi \
 	done
 
 install:: $(ALLTGT)
Index: FreeWnn-1.1.1-a023/kWnn/jutil/Makefile.in
===================================================================
--- FreeWnn-1.1.1-a023.orig/kWnn/jutil/Makefile.in
+++ FreeWnn-1.1.1-a023/kWnn/jutil/Makefile.in
@@ -39,7 +39,7 @@ LT_LD           = $(LIBTOOL) --mode=link
 
        INCLUDES = -I$(top_builddir) @EXTKWNNJINC@ -I$(KWNNINCLUDESRC)
         DEFINES = $(KWNNDEFINES) $(KWNNLANGDEF)
-LOCAL_INSTFLAGS = -o $(WNNOWNER)
+LOCAL_INSTFLAGS =
     HINSIDEFINE = -DHINSIDATA_FILE=\"/$(KWNNLANG)/hinsi.data\"
 
 CLIENTPROGRAMS = kwddel kwdreg
