--- qpid-cpp-1.36.0/BuildInstallSettings.cmake	2017-02-22 07:59:46.401899801 +0100
+++ qpid-cpp-1.36.0/BuildInstallSettings.cmake	2017-02-22 08:00:08.393411592 +0100
@@ -161,7 +161,7 @@
   set (QPID_INSTALL_SBINDIR sbin)
   set (QPID_INSTALL_TESTDIR ${LIBEXEC_INSTALL_DIR}/qpid/tests) # Directory for test executables
   set (QPID_INSTALL_CONFDIR ${SYSCONF_INSTALL_DIR}/qpid)
-  set (QPID_INSTALL_INITDDIR ${SYSCONF_INSTALL_DIR}/rc.d/init.d)
+  set (QPID_INSTALL_INITDDIR ${SYSCONF_INSTALL_DIR}/init.d)
   set (QPID_INSTALL_SASLDIR ${SYSCONF_INSTALL_DIR}/sasl2)
   set (QPID_INSTALL_DATADIR ${SHARE_INSTALL_DIR}/qpid)
   set (QPID_INSTALL_EXAMPLESDIR ${SHARE_INSTALL_DIR}/qpid/examples)
