Index: mugshot-0.4.1/mugshot/MugshotWindow.py
===================================================================
--- mugshot-0.4.1.orig/mugshot/MugshotWindow.py
+++ mugshot-0.4.1/mugshot/MugshotWindow.py
@@ -184,7 +184,7 @@ class MugshotWindow(Window):
 
         if get_has_camera_support():
             self.CameraDialog = CameraMugshotDialog
-            self.image_from_camera.set_visible(True)
+            self.image_from_camera.set_visible(False)
         else:
             self.image_from_camera.set_visible(False)
 
