Index: rssguard-4.3.3/src/librssguard/3rd-party/mimesis/quoted-printable.cpp
===================================================================
--- rssguard-4.3.3.orig/src/librssguard/3rd-party/mimesis/quoted-printable.cpp
+++ rssguard-4.3.3/src/librssguard/3rd-party/mimesis/quoted-printable.cpp
@@ -19,6 +19,8 @@
 
 #include "quoted-printable.hpp"
 
+#include <cstdint>
+
 using namespace std;
 
 string quoted_printable_decode(string_view in) {
