Index: EternalTerminal-et-v6.2.4/external_imported/sentry-native/external/breakpad/src/client/linux/handler/minidump_descriptor.h
===================================================================
--- EternalTerminal-et-v6.2.4.orig/external_imported/sentry-native/external/breakpad/src/client/linux/handler/minidump_descriptor.h
+++ EternalTerminal-et-v6.2.4/external_imported/sentry-native/external/breakpad/src/client/linux/handler/minidump_descriptor.h
@@ -37,6 +37,7 @@
 
 #include "client/linux/handler/microdump_extra_info.h"
 #include "common/using_std_string.h"
+#include <cstdint>
 
 // This class describes how a crash dump should be generated, either:
 // - Writing a full minidump to a file in a given directory (the actual path,
