Index: EternalTerminal-et-v6.2.0/src/base/Headers.hpp
===================================================================
--- EternalTerminal-et-v6.2.0.orig/src/base/Headers.hpp
+++ EternalTerminal-et-v6.2.0/src/base/Headers.hpp
@@ -353,9 +353,8 @@ inline bool waitOnSocketData(int fd) {
     } else {
       FATAL_FAIL(selectResult);
     }
-  } else {
-    return FD_ISSET(fd, &fdset);
   }
+  return FD_ISSET(fd, &fdset);
 }
 
 inline string genRandomAlphaNum(int len) {
