diff -urEbw kristall-0.3/Makefile kristall-0.3.new/Makefile
--- kristall-0.3/Makefile	2020-06-21 23:47:31.000000000 +0200
+++ kristall-0.3.new/Makefile	2020-07-06 11:50:01.584684982 +0200
@@ -17,7 +17,7 @@
 
 build/kristall: src/*
 	mkdir -p build
-	cd build && qmake ../src/kristall.pro && $(MAKE) $(MAKEFLAGS)
+	cd build && qmake-qt5 ../src/kristall.pro && $(MAKE) $(MAKEFLAGS)
 
 install: kristall
 	# Install icons
