From 18e18d4c31d39f12c87044db6d2d2f03163c8171 Mon Sep 17 00:00:00 2001
From: Johannes Kastl <kastl@b1-systems.de>
Date: Tue, 30 Jun 2020 18:51:20 +0200
Subject: [PATCH] templates/lxc-download.in: fix wrong if condition (use the
 result of the gpg command, not the result when executing the result of the
 gpg command)

Signed-off-by: Johannes Kastl <kastl@b1-systems.de>
---
 templates/lxc-download.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/templates/lxc-download.in b/templates/lxc-download.in
index 58d063f4c..f69b18fd4 100644
--- a/templates/lxc-download.in
+++ b/templates/lxc-download.in
@@ -134,8 +134,8 @@ gpg_setup() {
 
   success=
   for _ in $(seq 3); do
-    if $(gpg --keyserver "${DOWNLOAD_KEYSERVER}" ${DOWNLOAD_GPG_PROXY:-} \
-      --recv-keys "${DOWNLOAD_KEYID}" >/dev/null 2>&1); then
+    if gpg --keyserver "${DOWNLOAD_KEYSERVER}" ${DOWNLOAD_GPG_PROXY:-} \
+      --recv-keys "${DOWNLOAD_KEYID}" >/dev/null 2>&1; then
       success=1
       break
     fi
-- 
2.27.0

