--- openj9-openjdk-jdk11/omr/ddr/tools/ddrgen/Makefile	2020-04-21 15:53:10.964314332 +0200
+++ openj9-openjdk-jdk11/omr/ddr/tools/ddrgen/Makefile	2020-04-21 15:53:55.292575567 +0200
@@ -35,15 +35,9 @@
   $(top_srcdir)/include_core
 
 ifeq (linux,$(OMR_HOST_OS))
-  MODULE_SHARED_LIBS += rt pthread dwarf
-  ifeq (x86,$(OMR_HOST_ARCH))
-    MODULE_SHARED_LIBS += elf
-  endif
-  ifeq (ppc,$(OMR_HOST_ARCH))
-    MODULE_SHARED_LIBS += elf
-  endif
+  MODULE_SHARED_LIBS += rt pthread dwarf elf
   ifeq (s390,$(OMR_HOST_ARCH))
-    MODULE_SHARED_LIBS += elf z
+    MODULE_SHARED_LIBS += z
   endif
 endif
 
--- openj9-openjdk-jdk11/openj9/buildspecs/linux_aarch64_cmprssptrs.spec	2020-04-21 15:53:11.364316690 +0200
+++ openj9-openjdk-jdk11/openj9/buildspecs/linux_aarch64_cmprssptrs.spec	2020-04-21 19:13:31.087227448 +0200
@@ -223,7 +223,7 @@
 		<flag id="opt_zipSupport" value="true"/>
 		<flag id="opt_zlibCompression" value="true"/>
 		<flag id="opt_zlibSupport" value="true"/>
-		<flag id="port_omrsigSupport" value="false"/>
+		<flag id="port_omrsigSupport" value="true"/>
 		<flag id="port_signalSupport" value="true"/>
 		<flag id="prof_eventReporting" value="true"/>
 		<flag id="ras_dumpAgents" value="true"/>
--- openj9-openjdk-jdk11/openj9/buildspecs/linux_aarch64.spec	2020-04-21 15:53:11.364316690 +0200
+++ openj9-openjdk-jdk11/openj9/buildspecs/linux_aarch64.spec	2020-04-21 19:13:31.087227448 +0200
@@ -222,7 +222,7 @@
 		<flag id="opt_zipSupport" value="true"/>
 		<flag id="opt_zlibCompression" value="true"/>
 		<flag id="opt_zlibSupport" value="true"/>
-		<flag id="port_omrsigSupport" value="false"/>
+		<flag id="port_omrsigSupport" value="true"/>
 		<flag id="port_signalSupport" value="true"/>
 		<flag id="prof_eventReporting" value="true"/>
 		<flag id="ras_dumpAgents" value="true"/>
