--- a/closed/src/java.base/share/native/libjncrypto/NativeCrypto.c
+++ b/closed/src/java.base/share/native/libjncrypto/NativeCrypto.c
@@ -77,6 +77,10 @@
 # include <pthread.h>
 #endif /* defined(WINDOWS) */
 
+#if OPENSSL_VERSION_MAJOR < 3
+typedef struct ossl_lib_ctx_st OSSL_LIB_CTX;
+#endif
+
 /* Header for NativeCrypto loading methods. */
 static void * find_crypto_symbol(void *handle, const char *symname);
 static void * find_crypto_library(jboolean traceEnabled, jboolean skipBundled, const char *chomepath);
