diff -u -r xkeyboard-config-2.16.orig/rules/evdev.m_s.part xkeyboard-config-2.16/rules/evdev.m_s.part
--- xkeyboard-config-2.16.orig/rules/evdev.m_s.part	2015-10-20 16:46:14.666938000 +0200
+++ xkeyboard-config-2.16/rules/evdev.m_s.part	2015-10-20 16:47:50.094095000 +0200
@@ -1,4 +1,4 @@
-  $evdevkbds    =   +inet(evdev)+inet(%m)
-  chromebook    =   +inet(evdev)+inet(chromebook)
-  applealu_jis  =   +inet(evdev)+macintosh_vndr/jp(alujiskeys)
-  *             =   +inet(evdev)
+  $evdevkbds    =   +inet(evdev)+inet(%m)+terminate(ctrl_alt_bksp)
+  chromebook    =   +inet(evdev)+inet(chromebook)+terminate(ctrl_alt_bksp)
+  applealu_jis  =   +inet(evdev)+macintosh_vndr/jp(alujiskeys)+terminate(ctrl_alt_bksp)
+  *             =   +inet(evdev)+terminate(ctrl_alt_bksp)
