--- src/HiScore.cpp.orig	2013-05-15 07:03:14.000000000 +0200
+++ src/HiScore.cpp	2014-11-22 15:22:39.403465942 +0100
@@ -29,6 +29,8 @@
 #include <locale.h>
 #endif
 
+#define CHROMIUM_BSU_SCORE "/var/games/chromium-bsu.highscore"
+
 HiScore	*HiScore::instance = 0;
 
 static const char* header = "# Chromium B.S.U. high scores file: skill rank score date time(UTC) name (do not remove this comment line)\n";
@@ -160,7 +162,7 @@
 const char *HiScore::getFileName()
 {
 	static char	configFilename[256];
-	const char *envFile = getenv("CHROMIUM_BSU_SCORE");
+	const char *envFile = CHROMIUM_BSU_SCORE;
 	if(envFile && strlen(envFile) < 256)
 	{
 		strcpy(configFilename, envFile);
