Index: rosegarden-19.12/src/gui/widgets/FileDialog.cpp
===================================================================
--- rosegarden-19.12.orig/src/gui/widgets/FileDialog.cpp	2020-06-03 13:56:13.142007118 +0200
+++ rosegarden-19.12/src/gui/widgets/FileDialog.cpp	2020-06-03 13:56:27.638580234 +0200
@@ -56,8 +56,8 @@ FileDialog::FileDialog(QWidget *parent,
 #else
     QString home = QDesktopServices::storageLocation(QDesktopServices::HomeLocation);
 #endif
-    QString examples = home + "/.local/share/rosegarden/examples";
-    QString templates = home + "/.local/share/rosegarden/templates";
+    QString examples = "/usr/share/rosegarden/examples";
+    QString templates = "/usr/share/rosegarden/templates";
     QString rosegarden = home + "/rosegarden";
 
     RG_DEBUG  << "FileDialog::FileDialog(...)"
Index: rosegarden-19.12/src/gui/studio/BankEditorDialog.cpp
===================================================================
--- rosegarden-19.12.orig/src/gui/studio/BankEditorDialog.cpp	2020-06-03 13:56:13.142007118 +0200
+++ rosegarden-19.12/src/gui/studio/BankEditorDialog.cpp	2020-06-03 13:56:27.638580234 +0200
@@ -1553,7 +1553,7 @@ BankEditorDialog::slotImport()
 #else
     QString home = QUrl::fromLocalFile(QDesktopServices::storageLocation(QDesktopServices::HomeLocation)).path();
 #endif
-    QString deviceDir = home + "/.local/share/rosegarden/library";
+    QString deviceDir = "/usr/share/rosegarden/library";
 
     QString url_str = FileDialog::getOpenFileName(this, tr("Import Banks from Device in File"), deviceDir,
                       tr("Rosegarden Device files") + " (*.rgd *.RGD)" + ";;" +
