Index: pygame-2.6.1/test/surface_test.py
===================================================================
--- pygame-2.6.1.orig/test/surface_test.py
+++ pygame-2.6.1/test/surface_test.py
@@ -239,6 +239,7 @@ class SurfaceTypeTest(unittest.TestCase)
         for pt in test_utils.rect_outer_bounds(fill_rect):
             self.assertNotEqual(s1.get_at(pt), color)
 
+    @unittest.skip("Broken test in OBS build")
     def test_fill_rle(self):
         """Test RLEACCEL flag with fill()"""
         color = (250, 25, 25, 255)
@@ -255,6 +256,7 @@ class SurfaceTypeTest(unittest.TestCase)
         )
         self.assertTrue(blit_surf.get_flags() & pygame.RLEACCEL)
 
+    @unittest.skip("Broken test in OBS build")
     def test_mustlock_rle(self):
         """Test RLEACCEL flag with mustlock()"""
         surf = pygame.Surface((100, 100))
@@ -265,6 +267,7 @@ class SurfaceTypeTest(unittest.TestCase)
         self.assertTrue(blit_surf.get_flags() & pygame.RLEACCEL)
         self.assertTrue(blit_surf.mustlock())
 
+    @unittest.skip("Broken test in OBS build")
     def test_mustlock_surf_alpha_rle(self):
         """Test RLEACCEL flag with mustlock() on a surface
         with per pixel alpha - new feature in SDL2"""
@@ -277,6 +280,7 @@ class SurfaceTypeTest(unittest.TestCase)
         self.assertTrue(blit_surf.get_flags() & pygame.SRCALPHA)
         self.assertTrue(blit_surf.mustlock())
 
+    @unittest.skip("Broken test in OBS build")
     def test_copy_rle(self):
         """Test copying a surface set to use run length encoding"""
         s1 = pygame.Surface((32, 32), 24)
@@ -302,6 +306,7 @@ class SurfaceTypeTest(unittest.TestCase)
         self.assertTrue(s1.get_flags() & pygame.RLEACCEL)
         self.assertTrue(not s2.get_flags() & pygame.RLEACCEL)
 
+    @unittest.skip("Broken test in OBS build")
     def test_subsurface_rle2(self):
         """Ensure an RLE sub-surface works independently of its parent."""
         color = (250, 25, 25, 255)
@@ -318,6 +323,7 @@ class SurfaceTypeTest(unittest.TestCase)
         self.assertTrue(s1.get_flags() & pygame.RLEACCELOK)
         self.assertTrue(not s2.get_flags() & pygame.RLEACCELOK)
 
+    @unittest.skip("Broken test in OBS build")
     def test_solarwolf_rle_usage(self):
         """Test for error/crash when calling set_colorkey() followed
         by convert twice in succession. Code originally taken
@@ -346,6 +352,7 @@ class SurfaceTypeTest(unittest.TestCase)
         finally:
             pygame.display.quit()
 
+    @unittest.skip("Broken test in OBS build")
     def test_solarwolf_rle_usage_2(self):
         """Test for RLE status after setting alpha"""
 
Index: pygame-2.6.1/test/pixelarray_test.py
===================================================================
--- pygame-2.6.1.orig/test/pixelarray_test.py
+++ pygame-2.6.1/test/pixelarray_test.py
@@ -990,6 +990,7 @@ class PixelArrayTypeTest(unittest.TestCa
         self.assertEqual(w2, w)
         self.assertEqual(h2, h_slice)
 
+    @unittest.skip("Broken test in OBS build")
     def test_make_surface__subclassed_surface(self):
         """Ensure make_surface can handle subclassed surfaces."""
         expected_size = (3, 5)
