commit 16a19ee4c568047fb03581b23b4fcb273a615474
Author: Scott Pakin <pakin@lanl.gov>
Date:   Wed Feb 19 13:43:40 2020 -0700

    Replace deprecated MPI_Errhandler_set with newer MPI_Comm_set_errhandler
    
    Resolves #17.

diff --git src/init.c src/init.c
index cd070a785c41..46b1127e538d 100644
--- src/init.c
+++ src/init.c
@@ -77,7 +77,7 @@ mpi_init_builtin (WORD_LIST *list)
 
   /* Make MPI errors return instead of crash.  Also, store our rank
    * and number of ranks. */
-  MPI_Errhandler_set (MPI_COMM_WORLD, MPI_ERRORS_RETURN);
+  MPI_Comm_set_errhandler (MPI_COMM_WORLD, MPI_ERRORS_RETURN);
   MPI_Comm_rank (MPI_COMM_WORLD, &mpibash_rank);
   MPI_Comm_size (MPI_COMM_WORLD, &mpibash_num_ranks);
 
