Index: Makefile.in
===================================================================
--- Makefile.in.orig
+++ Makefile.in
@@ -149,7 +149,7 @@ install-program-files: mkinstalldirs
 
 compileall:
 ifneq "$(nopycompile)" "YES"
-	$(PYTHON) -c "import compileall; compileall.compile_dir('$(DESTDIR)/$(datadir)/$(PACKAGE)/solfege')"
+	$(PYTHON) -c "import compileall; compileall.compile_dir('$(DESTDIR)$(datadir)/$(PACKAGE)/solfege',10,'$(datadir)/$(PACKAGE)/solfege')"
 endif
 
 mkinstalldirs:
