Description: fix a FTBFS with e2fsprogs 1.44.1-1. (Closes: #894650)
Author: Hilko Bengen <bengen@debian.org>
Bug-Debian: https://bugs.debian.org/894650
Last-Update: 2018-04-13
Index: ext3grep/src/ext3.h
===================================================================
--- ext3grep.orig/src/ext3.h
+++ ext3grep/src/ext3.h
@@ -117,7 +117,7 @@ struct Inode : protected ext3_inode {
     __u32 const* block(void) const { return i_block; }
     __u32 generation(void) const { return i_generation; }
     __u32 file_acl(void) const { return i_file_acl; }
-    __u32 dir_acl(void) const { return i_dir_acl; }
+    __u32 size_high(void) const { return i_size_high; }
     __u32 faddr(void) const { return i_faddr; }
     __u16 uid_high(void) const { return i_uid_high; }
     __u16 gid_high(void) const { return i_gid_high; }
