Index: upmpdcli-1.5.11/systemd/upmpdcli.service
===================================================================
--- upmpdcli-1.5.11.orig/systemd/upmpdcli.service
+++ upmpdcli-1.5.11/systemd/upmpdcli.service
@@ -4,6 +4,19 @@ After=network-online.target mpd.service
 Wants=network-online.target
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+PrivateDevices=true
+ProtectHostname=true
+ProtectClock=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 Type=simple
 # Note: if start fails check with "systemctl status upmpdcli"
 ExecStart=/usr/bin/upmpdcli -c /etc/upmpdcli.conf
