Index: mailman-3.3.0/src/mailman/utilities/tests/test_interact.py
===================================================================
--- mailman-3.3.0.orig/src/mailman/utilities/tests/test_interact.py	2019-04-25 07:07:47.000000000 +0200
+++ mailman-3.3.0/src/mailman/utilities/tests/test_interact.py	2019-09-30 16:27:50.460382346 +0200
@@ -67,7 +67,7 @@ class TestInteract(unittest.TestCase):
         self._enter(hackenv('PYTHONSTARTUP', None))
         interact()
         stderr = self._stderr.getvalue().splitlines()
-        banner = 'Python {} on {} '.format(sys.version, sys.platform)
+        banner = 'Python {} on {}'.format(sys.version, sys.platform)
         self.assertEqual(stderr[0], banner.splitlines()[0])
 
     def test_interact_custom_banner(self):
