diff -wruN -x '*~' -x '*.o' -x '*.a' -x '*.so' -x '*.so.[0-9]' -x autom4te.cache -x .deps -x .libs ../orig-pyparted-3.10/src/parted/disk.py ./src/parted/disk.py
--- ../orig-pyparted-3.10/src/parted/disk.py	2013-04-09 17:31:40.000000000 +0200
+++ ./src/parted/disk.py	2014-02-09 09:49:31.693576602 +0100
@@ -486,4 +486,7 @@
     if not __flag:
         __readFlags = False
     else:
-        diskFlag[__flag] = _ped.disk_flag_get_name(__flag)
+        try:
+            diskFlag[__flag] = _ped.disk_flag_get_name(__flag)
+        except:
+            print("Bad disk flag: {}".format(__flag))
