diff --git a/buildscripts/cmake/SetupQt6.cmake b/buildscripts/cmake/SetupQt6.cmake
index 9507ae5..42c10d9 100644
--- a/buildscripts/cmake/SetupQt6.cmake
+++ b/buildscripts/cmake/SetupQt6.cmake
@@ -25,6 +25,7 @@ endif()
 set(qt_components
     Core
     Gui
+    GuiPrivate
     Widgets
     Network
     Qml
diff --git a/src/appshell/CMakeLists.txt b/src/appshell/CMakeLists.txt
index 868dd99..8ca54d4 100644
--- a/src/appshell/CMakeLists.txt
+++ b/src/appshell/CMakeLists.txt
@@ -143,7 +143,7 @@ if (NOT OS_IS_MAC)
 endif(NOT OS_IS_MAC)
 
 if (QT_SUPPORT)
-    list(APPEND MODULE_LINK Qt::Core5Compat Qt::Quick Qt::Svg Qt::Widgets)
+    list(APPEND MODULE_LINK Qt::Core5Compat Qt::GuiPrivate Qt::Quick Qt::Svg Qt::Widgets)
 endif()
 
 setup_module()
diff --git a/src/framework/accessibility/CMakeLists.txt b/src/framework/accessibility/CMakeLists.txt
index cdb8c08..ab5ce69 100644
--- a/src/framework/accessibility/CMakeLists.txt
+++ b/src/framework/accessibility/CMakeLists.txt
@@ -57,7 +57,7 @@ if (MUSE_MODULE_ACCESSIBILITY_TRACE)
 endif()
 
 if (QT_SUPPORT)
-    list(APPEND MODULE_LINK Qt::Quick)
+    list(APPEND MODULE_LINK Qt::CorePrivate Qt::Quick)
 endif()
 
 setup_module()
diff --git a/src/framework/dockwindow/thirdparty/KDDockWidgets/CMakeLists.txt b/src/framework/dockwindow/thirdparty/KDDockWidgets/CMakeLists.txt
index 1f15515..c8681e2 100644
--- a/src/framework/dockwindow/thirdparty/KDDockWidgets/CMakeLists.txt
+++ b/src/framework/dockwindow/thirdparty/KDDockWidgets/CMakeLists.txt
@@ -139,9 +139,17 @@ set(${PROJECT_NAME}_DEPS "widgets")
 if(${PROJECT_NAME}_QTQUICK)
   find_package(Qt${Qt_VERSION_MAJOR}Quick)
   find_package(Qt${Qt_VERSION_MAJOR}QuickControls2)
+  if(Qt6Core_VERSION VERSION_GREATER_EQUAL "6.10.0")
+    set(QT_NO_PRIVATE_MODULE_WARNING ON)
+    find_package(Qt6 ${QT_MIN_VERSION} NO_MODULE REQUIRED COMPONENTS QuickPrivate)
+  endif()
   add_definitions(-DKDDOCKWIDGETS_QTQUICK)
   set(${PROJECT_NAME}_DEPS "${${PROJECT_NAME}_DEPS} quick quickcontrols2")
 else()
+  if(Qt6Core_VERSION VERSION_GREATER_EQUAL "6.10.0")
+    set(QT_NO_PRIVATE_MODULE_WARNING ON)
+    find_package(Qt6 ${QT_MIN_VERSION} NO_MODULE REQUIRED COMPONENTS WidgetsPrivate)
+  endif()
   add_definitions(-DKDDOCKWIDGETS_QTWIDGETS)
 endif()
 
diff --git a/src/importexport/midi/internal/midiimport/importmidi_tuplet.cpp b/src/importexport/midi/internal/midiimport/importmidi_tuplet.cpp
index 729e928..21c0ee1 100644
--- a/src/importexport/midi/internal/midiimport/importmidi_tuplet.cpp
+++ b/src/importexport/midi/internal/midiimport/importmidi_tuplet.cpp
@@ -895,9 +895,9 @@ void addTupletEvents(std::multimap<ReducedFraction, TupletData>& tupletEvents,
                 message += QString::number(tiedTuplet.voice) + ", chord voice = ";
                 message += QString::number(midiChord.voice) + ", bar number (from 1) = ";
                 message += QString::number(midiChord.barIndex + 1);
-#endif
                 Q_ASSERT_X(tiedTuplet.voice == midiChord.voice,
                            "MidiTuplet::addTupletEvents", message.toLatin1().data());
+#endif
 
                 for (int j: tiedTuplet.tiedNoteIndexes) {
                     midiChord.notes[j].tuplet = it;
