From aefc0d09f48dc85202cbf24fac4fd40c5908f1e6 Mon Sep 17 00:00:00 2001
From: Malcolm <malcolmlewis@opensuse.org>
Date: Thu, 20 Nov 2014 20:13:37 -0600
Subject: [PATCH][BUILD] Fix renamed gettext include header reference

Signed-off-by: Malcolm <malcolmlewis@opensuse.org>
---
 barcode.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/barcode.h b/barcode.h
index ed90e0c..ff8574d 100644
--- a/barcode.h
+++ b/barcode.h
@@ -26,7 +26,7 @@
 #include "config.h"
 #include <stdio.h>
 
-#include <gettext.h>
+#include <gettext-po.h>
 
 #define _(X) gettext (X)
 #define _N(X) (X)
-- 
1.8.5.2

