diff --git a/EXAMPLE/Makefile b/EXAMPLE/Makefile
index eb961e3..0ae9052 100644
--- a/EXAMPLE/Makefile
+++ b/EXAMPLE/Makefile
@@ -39,8 +39,6 @@ SLINXEXM 	= slinsolx.o
 SLINXEXM1 	= slinsolx1.o
 SLINXEXM2	= slinsolx2.o
 SLINXEXM3	= slinsolx3.o
-SITSOL	     	= sitersol.o sfgmr.o
-SITSOL1	     	= sitersol1.o sfgmr.o
 
 DLINEXM		= dlinsol.o dreadtriple_noheader.o #dgstrf.o dcolumn_dfs.o #sp_ienv.o
 DLINEXM1	= dlinsol1.o
@@ -49,8 +47,6 @@ DLINXEXM1 	= dlinsolx1.o
 DLINXEXM2 	= dlinsolx2.o
 DLINXEXM3 	= dlinsolx3.o
 SUPERLUEXM 	= superlu.o sp_ienv.o
-DITSOL	     	= ditersol.o dfgmr.o
-DITSOL1	     	= ditersol1.o dfgmr.o
 
 CLINEXM 	= clinsol.o
 CLINEXM1 	= clinsol1.o
@@ -58,8 +54,6 @@ CLINXEXM 	= clinsolx.o
 CLINXEXM1 	= clinsolx1.o
 CLINXEXM2 	= clinsolx2.o
 CLINXEXM3 	= clinsolx3.o
-CITSOL	     	= citersol.o cfgmr.o
-CITSOL1	     	= citersol1.o cfgmr.o
 
 ZLINEXM 	= zlinsol.o
 ZLINEXM1 	= zlinsol1.o
@@ -67,20 +61,17 @@ ZLINXEXM 	= zlinsolx.o
 ZLINXEXM1 	= zlinsolx1.o
 ZLINXEXM2 	= zlinsolx2.o
 ZLINXEXM3 	= zlinsolx3.o
-ZITSOL	     	= zitersol.o zfgmr.o
-ZITSOL1	     	= zitersol1.o zfgmr.o
 
 
 all:    single double complex complex16
 
-single:    slinsol slinsol1 slinsolx slinsolx1 slinsolx2 slinsolx3 \
-		sitersol sitersol1
+single:    slinsol slinsol1 slinsolx slinsolx1 slinsolx2 slinsolx3 
+
 double:    dlinsol dlinsol1 dlinsolx dlinsolx1 dlinsolx2 dlinsolx3 \
-		superlu ditersol ditersol1
-complex:   clinsol clinsol1 clinsolx clinsolx1 clinsolx2 clinsolx3 \
-		citersol citersol1
-complex16: zlinsol zlinsol1 zlinsolx zlinsolx1 zlinsolx2 zlinsolx3 \
-		zitersol zitersol1
+		superlu
+complex:   clinsol clinsol1 clinsolx clinsolx1 clinsolx2 clinsolx3 
+
+complex16: zlinsol zlinsol1 zlinsolx zlinsolx1 zlinsolx2 zlinsolx3 
 
 slinsol: $(SLINEXM) $(SUPERLULIB)
 	$(LOADER) $(LOADOPTS) $(SLINEXM) $(LIBS) -lm -o $@
@@ -100,11 +91,6 @@ slinsolx2: $(SLINXEXM2) $(SUPERLULIB)
 slinsolx3: $(SLINXEXM3) $(SUPERLULIB)
 	$(LOADER) $(LOADOPTS) $(SLINXEXM3) $(LIBS) -lm -o $@
 
-sitersol: $(SITSOL) $(SUPERLULIB)
-	$(LOADER) $(LOADOPTS) $(SITSOL) $(LIBS) -lm -o $@
-
-sitersol1: $(SITSOL1) $(SUPERLULIB)
-	$(LOADER) $(LOADOPTS) $(SITSOL1) $(LIBS) -lm -o $@
 
 dlinsol: $(DLINEXM) $(SUPERLULIB)
 	$(LOADER) $(LOADOPTS) $(DLINEXM) $(LIBS) -lm -o $@
@@ -127,12 +113,6 @@ dlinsolx3: $(DLINXEXM3) $(SUPERLULIB)
 superlu: $(SUPERLUEXM) $(SUPERLULIB)
 	$(LOADER) $(LOADOPTS) $(SUPERLUEXM) $(LIBS) -lm -o $@
 
-ditersol: $(DITSOL) $(SUPERLULIB)
-	$(LOADER) $(LOADOPTS) $(DITSOL) $(LIBS) -lm -o $@
-
-ditersol1: $(DITSOL1) $(SUPERLULIB)
-	$(LOADER) $(LOADOPTS) $(DITSOL1) $(LIBS) -lm -o $@
-
 clinsol: $(CLINEXM) $(SUPERLULIB)
 	$(LOADER) $(LOADOPTS) $(CLINEXM) $(LIBS) -lm -o $@
 
@@ -151,12 +131,6 @@ clinsolx2: $(CLINXEXM2) $(SUPERLULIB)
 clinsolx3: $(CLINXEXM3) $(SUPERLULIB)
 	$(LOADER) $(LOADOPTS) $(CLINXEXM3) $(LIBS) -lm -o $@
 
-citersol: $(CITSOL) $(SUPERLULIB)
-	$(LOADER) $(LOADOPTS) $(CITSOL) $(LIBS) -lm -o $@
-
-citersol1: $(CITSOL1) $(SUPERLULIB)
-	$(LOADER) $(LOADOPTS) $(CITSOL1) $(LIBS) -lm -o $@
-
 zlinsol: $(ZLINEXM) $(SUPERLULIB)
 	$(LOADER) $(LOADOPTS) $(ZLINEXM) $(LIBS) -lm -o $@
 
@@ -175,11 +149,6 @@ zlinsolx2: $(ZLINXEXM2) $(SUPERLULIB)
 zlinsolx3: $(ZLINXEXM3) $(SUPERLULIB)
 	$(LOADER) $(LOADOPTS) $(ZLINXEXM3) $(LIBS) -lm -o $@
 
-zitersol: $(ZITSOL) $(SUPERLULIB)
-	$(LOADER) $(LOADOPTS) $(ZITSOL) $(LIBS) -lm -o $@
-
-zitersol1: $(ZITSOL1) $(SUPERLULIB)
-	$(LOADER) $(LOADOPTS) $(ZITSOL1) $(LIBS) -lm -o $@
 
 .c.o:
 	$(CC) $(CFLAGS) $(CDEFS) $(INCLUDEDIR) -c $< $(VERBOSE)
@@ -189,7 +158,7 @@ zitersol1: $(ZITSOL1) $(SUPERLULIB)
 
 clean:	
 	rm -f *.o *linsol *linsol1 *linsolx *linsolx1 *linsolx2 *linsolx3 \
-	    superlu *itersol *itersol1
+	    superlu
