diff -Nrua a/SRC/CMakeLists.txt b/SRC/CMakeLists.txt
--- a/SRC/CMakeLists.txt
+++ b/SRC/CMakeLists.txt
@@ -21,7 +21,6 @@
   ilu_relax_snode.c
   ilu_heap_relax_snode.c
   mark_relax.c
-  mc64ad.c
   qselect.c
   input_error.c
   dmach.c
@@ -65,9 +64,7 @@
     smemory.c
     sutil.c
     smyblas2.c
-    sgsisx.c
     sgsitrf.c
-    sldperm.c
     ilu_sdrop_row.c
     ilu_ssnode_dfs.c
     ilu_scolumn_dfs.c
@@ -115,9 +112,7 @@
     dmemory.c
     dutil.c
     dmyblas2.c
-    dgsisx.c
     dgsitrf.c
-    dldperm.c
     ilu_ddrop_row.c
     ilu_dsnode_dfs.c
     ilu_dcolumn_dfs.c
@@ -168,9 +163,7 @@
     cmemory.c
     cutil.c
     cmyblas2.c
-    cgsisx.c
     cgsitrf.c
-    cldperm.c
     ilu_cdrop_row.c
     ilu_csnode_dfs.c
     ilu_ccolumn_dfs.c
@@ -219,9 +212,7 @@
     zmemory.c
     zutil.c
     zmyblas2.c
-    zgsisx.c
     zgsitrf.c
-    zldperm.c
     ilu_zdrop_row.c
     ilu_zsnode_dfs.c
     ilu_zcolumn_dfs.c
