From 5876e8a594c0636438a9b57c3c1e9b0585cffa8d Mon Sep 17 00:00:00 2001
From: Andy Lutomirski <luto@kernel.org>
Date: Wed, 16 Oct 2019 12:34:22 -0700
Subject: [PATCH 3/9] Fix the error message for --mods misuse

Signed-off-by: Andy Lutomirski <luto@kernel.org>
---
 virtme/commands/run.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/virtme/commands/run.py b/virtme/commands/run.py
index 4c354a1..50c670a 100644
--- a/virtme/commands/run.py
+++ b/virtme/commands/run.py
@@ -225,7 +225,7 @@ def find_kernel_and_mods(arch, args) -> Kernel:
         else:
             kernel.dtb = os.path.join(args.kdir, dtb_path)
     elif args.mods is not None:
-        arg_fail("--mods must be used together with --kdir")
+        arg_fail("--mods must be used together with --kdir or --installed-kernel")
     elif args.kimg is not None:
         kernel.kimg = args.kimg
         kernel.modfiles = []
-- 
2.26.2

