From 9c7b668bdf792ca885154d4fd57984280a7a5c17 Mon Sep 17 00:00:00 2001
From: Andy Lutomirski <luto@kernel.org>
Date: Wed, 16 Oct 2019 12:49:23 -0700
Subject: [PATCH 4/9] Fix the --mods error even better.

Signed-off-by: Andy Lutomirski <luto@kernel.org>
---
 virtme/commands/run.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/virtme/commands/run.py b/virtme/commands/run.py
index 50c670a..722d18a 100644
--- a/virtme/commands/run.py
+++ b/virtme/commands/run.py
@@ -224,13 +224,13 @@ def find_kernel_and_mods(arch, args) -> Kernel:
             kernel.dtb = None
         else:
             kernel.dtb = os.path.join(args.kdir, dtb_path)
-    elif args.mods is not None:
-        arg_fail("--mods must be used together with --kdir or --installed-kernel")
     elif args.kimg is not None:
         kernel.kimg = args.kimg
         kernel.modfiles = []
         kernel.moddir = None
         kernel.dtb = None # TODO: fix this
+        if args.mods != 'use':
+            arg_fail("--mods is not currently supported properly with --kimg")
     else:
         arg_fail('You must specify a kernel to use.')
 
-- 
2.26.2

