From: Armin Novak <armin.novak@thincast.com>
Date: Thu, 8 Aug 2024 11:06:54 +0200
Subject: [server,proxy] deactivate capture module

the module does not work (and did not for a long time)

(cherry picked from commit be23ed4ba990bd39391a651444fbb9130722c93b)
---
 server/proxy/modules/capture/CMakeLists.txt | 28 +++++++++++++++-------------
 1 file changed, 15 insertions(+), 13 deletions(-)

diff --git a/server/proxy/modules/capture/CMakeLists.txt b/server/proxy/modules/capture/CMakeLists.txt
index 80ba3b7..4004aaa 100644
--- a/server/proxy/modules/capture/CMakeLists.txt
+++ b/server/proxy/modules/capture/CMakeLists.txt
@@ -17,17 +17,19 @@
 # limitations under the License.
 #
 
-set(PLUGIN_NAME "proxy-capture-plugin")
+# deactivated: does not work
 
-add_library(${PLUGIN_NAME} MODULE
-	cap_main.c
-	cap_config.c
-	cap_config.h
-	cap_protocol.c
-	cap_protocol.h
-)
-
-set_target_properties(${PLUGIN_NAME} PROPERTIES PREFIX "")
-set_target_properties(${PLUGIN_NAME} PROPERTIES NO_SONAME 1)
-set_target_properties(${PLUGIN_NAME} PROPERTIES
-LIBRARY_OUTPUT_DIRECTORY "${FREERDP_PROXY_PLUGINDIR}")
+#set(PLUGIN_NAME "proxy-capture-plugin")
+#
+#add_library(${PLUGIN_NAME} MODULE
+#	cap_main.c
+#	cap_config.c
+#	cap_config.h
+#	cap_protocol.c
+#	cap_protocol.h
+#)
+#
+#set_target_properties(${PLUGIN_NAME} PROPERTIES PREFIX "")
+#set_target_properties(${PLUGIN_NAME} PROPERTIES NO_SONAME 1)
+#set_target_properties(${PLUGIN_NAME} PROPERTIES
+#LIBRARY_OUTPUT_DIRECTORY "${FREERDP_PROXY_PLUGINDIR}")
