Index: xlockmore-5.53/configure.ac
===================================================================
--- xlockmore-5.53.orig/configure.ac
+++ xlockmore-5.53/configure.ac
@@ -1912,7 +1912,7 @@ dnl freetype FREETYPE
 AC_DEFUN([AC_PATH_FREETYPE_DIRECT],
 [test -z "$freetype_direct_test_library" && freetype_direct_test_library=freetype
 test -z "$freetype_direct_test_function" && freetype_direct_test_function=FT_Init_FreeType
-test -z "$freetype_direct_test_include" && freetype_direct_test_include=freetype2/freetype.h
+test -z "$freetype_direct_test_include" && freetype_direct_test_include=freetype.h
   for ac_dir in               \
     /usr/X11R6.5.1/include	    \
     /usr/X11R6.4/include	      \
@@ -1972,11 +1972,14 @@ test -z "$freetype_direct_test_include"
     /usr/openwin/share/include	 \
     /usr/openwin/include	      \
                               \
+    /usr/include/freetype2	      \
+    /usr/include/freetype2/freetype	      \
+                              \
     $extra_include	            \
     ; \
   do
     if test -r "$ac_dir/$freetype_direct_test_include"; then
-      no_freetype= ac_freetype_includes=$ac_dir/freetype2
+      no_freetype= ac_freetype_includes=$ac_dir
       break
     fi
   done
