---
 src/Makefile.am |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: YODA-1.7.4/src/Makefile.am
===================================================================
--- YODA-1.7.4.orig/src/Makefile.am
+++ YODA-1.7.4/src/Makefile.am
@@ -23,7 +23,7 @@ libYODA_la_SOURCES = \
     Scatter2D.cc \
     Scatter3D.cc
 
-libYODA_la_LDFLAGS = -avoid-version
+libYODA_la_LDFLAGS = -release ${PACKAGE_VERSION}
 libYODA_la_LIBADD = $(builddir)/tinyxml/libyoda-tinyxml.la $(builddir)/yamlcpp/libyoda-yaml-cpp.la
 libYODA_la_CPPFLAGS = $(AM_CPPFLAGS) -DTIXML_USE_STL -I$(srcdir)/yamlcpp -I$(srcdir) -DYAML_NAMESPACE=YODA_YAML
 
