--- a/steam-mobile/Makefile
+++ b/steam-mobile/Makefile
@@ -1,5 +1,6 @@
 
 COMPILER = gcc
+CFLAGS ?= -g
 
 LIBPURPLE_CFLAGS += $(shell pkg-config --cflags glib-2.0 json-glib-1.0 purple nss gnome-keyring-1)
 LIBPURPLE_LIBS += $(shell pkg-config --libs glib-2.0 json-glib-1.0 purple nss)
@@ -14,4 +15,4 @@ clean:
 	rm -f libsteam.so
 
 libsteam.so: ${STEAM_SOURCES}
-	${COMPILER} -Wall -I. -g -O2 -fPIC -pipe ${STEAM_SOURCES} -o $@ ${LIBPURPLE_CFLAGS} ${LIBPURPLE_LIBS} -shared
+	${COMPILER} -Wall -I. -O2 -fPIC -pipe $(CFLAGS) ${LIBPURPLE_CFLAGS} ${STEAM_SOURCES} -o $@ $(LDFLAGS) ${LIBPURPLE_LIBS} -shared
