From 7737c0446bb30ca14782b488ce2399c7c752f640 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lars=20Gr=C3=BCter?= <lagru@mailbox.org>
Date: Fri, 6 Mar 2026 00:39:28 +0100
Subject: [PATCH] Avoid deprecated assign to ndarray.shape (#8020)

We also had to update our minimum requirements to get everything running smoothly (certain versions of NumPy didn't have the `copy` flag yet, and once that version of NumPy becomes the minimum there's a cascade of dependency updates).

Pyodide now depends on a different version of SciPy than the rest of the library, since it doesn't yet have SciPy 1.15.

Closes #8019
Closes #8025
---------

Co-authored-by: Stefan van der Walt <sjvdwalt@gmail.com>
Co-authored-by: Stefan van der Walt <stefanv@berkeley.edu>
---
 .github/workflows/test-pyodide.yaml           |  4 +--
 environment.yml                               | 28 +++++++--------
 meson.options                                 |  6 ----
 pyproject.toml                                | 34 ++++++++++---------
 requirements/default.txt                      |  9 ++---
 requirements/docs.txt                         | 10 +++---
 requirements/optional.txt                     | 10 +++---
 requirements/optional_free_threaded.txt       |  8 ++---
 src/meson.build                               |  7 +---
 src/skimage/feature/sift.py                   |  4 +--
 src/skimage/graph/_mcp.pyx                    |  2 +-
 src/skimage/io/_plugins/pil_plugin.py         |  2 +-
 .../measure/_marching_cubes_lewiner.py        |  4 +--
 src/skimage/morphology/grayreconstruct.py     |  4 ++-
 src/skimage/restoration/_rolling_ball_cy.pyx  |  4 +--
 src/skimage/util/_map_array.py                |  2 +-
 tools/generate_requirements.py                |  4 +++
 17 files changed, 70 insertions(+), 72 deletions(-)
 delete mode 100644 meson.options

Index: scikit_image-0.26.0/meson.options
===================================================================
--- scikit_image-0.26.0.orig/meson.options
+++ /dev/null
@@ -1,6 +0,0 @@
-option(
-  'include-v2',
-  type: 'boolean',
-  value: true,
-  description: 'Include skimage2 namespace in the built package'
-)
Index: scikit_image-0.26.0/pyproject.toml
===================================================================
--- scikit_image-0.26.0.orig/pyproject.toml
+++ scikit_image-0.26.0/pyproject.toml
@@ -202,6 +203,10 @@ filterwarnings = [
     # Pyodide warning coming via threadpoolctl, remove when
     # https://github.com/joblib/threadpoolctl/pull/201 is released
     "ignore:.*JsProxy\\.as_object_map.*:RuntimeWarning",
+    # Warning caused by matplotlib 3.7.0 because it uses deprecated API of pyparsing 3.3.1
+    "ignore:.*(parseString|resetCache|enablePackrat|oneOf).*:DeprecationWarning:matplotlib",
+    # tifffile (v2025.12.20) uses deprecated NumPy API
+    "ignore:Setting the shape on a NumPy array:DeprecationWarning:tifffile",
 ]
 norecursedirs = ["io/_plugins"]
 
@@ -226,7 +231,6 @@ build-verbosity = 0
 enable="cpython-freethreading"
 
 [tool.cibuildwheel.config-settings]
-setup-args = "-Dinclude-v2=false"
 # Could be overwritten using CIBW_CONFIG_SETTINGS environment variable
 
 [[tool.cibuildwheel.overrides]]
Index: scikit_image-0.26.0/src/meson.build
===================================================================
--- scikit_image-0.26.0.orig/src/meson.build
+++ scikit_image-0.26.0/src/meson.build
@@ -1,7 +1,2 @@
-include_v2 = get_option('include-v2')
-
 subdir('skimage')
-
-if include_v2
-  subdir('skimage2')
-endif
+subdir('skimage2')
Index: scikit_image-0.26.0/src/skimage/feature/sift.py
===================================================================
--- scikit_image-0.26.0.orig/src/skimage/feature/sift.py
+++ scikit_image-0.26.0/src/skimage/feature/sift.py
@@ -304,7 +304,7 @@ class SIFT(FeatureDetector, DescriptorEx
             mode='reflect',
         )
 
-        # Eq. 10:  sigmas.shape = (n_octaves, n_scales + 3).
+        # Eq. 10:  sigmas.reshape((n_octaves, n_scales + 3), copy=False).
         # The three extra scales are:
         #    One for the differences needed for DoG and two auxiliary
         #    images (one at either end) for peak_local_max with exclude
@@ -317,7 +317,7 @@ class SIFT(FeatureDetector, DescriptorEx
         self.scalespace_sigmas = sigmas
 
         # Eq. 7: Gaussian smoothing depends on difference with previous sigma
-        #        gaussian_sigmas.shape = (n_octaves, n_scales + 2)
+        #        gaussian_sigmas.reshape((n_octaves, n_scales + 2), copy=False)
         var_diff = np.diff(sigmas * sigmas, axis=1)
         gaussian_sigmas = np.sqrt(var_diff) / self.deltas[:, np.newaxis]
 
Index: scikit_image-0.26.0/src/skimage/graph/_mcp.pyx
===================================================================
--- scikit_image-0.26.0.orig/src/skimage/graph/_mcp.pyx
+++ scikit_image-0.26.0/src/skimage/graph/_mcp.pyx
@@ -77,7 +77,7 @@ def _offset_edge_map(shape, offsets):
           [0, 0, 2, 1]], dtype=int8)
 
     """
-    indices = np.indices(shape)  # indices.shape = (n,)+shape
+    indices = np.indices(shape)  # indices.reshape((n,) + shape, copy=False)
 
     #get the distance from each index to the upper or lower edge in each dim
     pos_edges = (shape - indices.T).T
Index: scikit_image-0.26.0/src/skimage/io/_plugins/pil_plugin.py
===================================================================
--- scikit_image-0.26.0.orig/src/skimage/io/_plugins/pil_plugin.py
+++ scikit_image-0.26.0/src/skimage/io/_plugins/pil_plugin.py
@@ -109,7 +109,7 @@ def pil_to_ndarray(image, dtype=None, im
             if 'S' in image.mode:
                 dtype = dtype.replace('u', 'i')
             frame = np.frombuffer(frame.tobytes(), dtype)
-            frame.shape = shape[::-1]
+            frame = np.reshape(frame, shape[::-1], copy=False)
 
         else:
             frame = np.array(frame, dtype=dtype)
Index: scikit_image-0.26.0/src/skimage/measure/_marching_cubes_lewiner.py
===================================================================
--- scikit_image-0.26.0.orig/src/skimage/measure/_marching_cubes_lewiner.py
+++ scikit_image-0.26.0/src/skimage/measure/_marching_cubes_lewiner.py
@@ -210,7 +210,7 @@ def _marching_cubes_lewiner(
     normals = np.fliplr(normals)
 
     # Finishing touches to output
-    faces.shape = -1, 3
+    faces = np.reshape(faces, (-1, 3), copy=False)
     if gradient_direction == 'descent':
         # MC implementation is right-handed, but gradient_direction is
         # left-handed
@@ -234,7 +234,7 @@ def _to_array(args):
     shape, text = args
     byts = base64.decodebytes(text.encode('utf-8'))
     ar = np.frombuffer(byts, dtype='int8')
-    ar.shape = shape
+    ar = np.reshape(ar, shape, copy=False)
     return ar
 
 
Index: scikit_image-0.26.0/src/skimage/morphology/grayreconstruct.py
===================================================================
--- scikit_image-0.26.0.orig/src/skimage/morphology/grayreconstruct.py
+++ scikit_image-0.26.0/src/skimage/morphology/grayreconstruct.py
@@ -213,5 +213,7 @@ def reconstruction(seed, mask, method='d
 
     # Reshape reconstructed image to original image shape and remove padding.
     rec_img = value_map[value_rank[:image_stride]]
-    rec_img.shape = np.array(seed.shape) + (np.array(footprint.shape) - 1)
+    rec_img = np.reshape(
+        rec_img, np.array(seed.shape) + (np.array(footprint.shape) - 1), copy=False
+    )
     return rec_img[inside_slices]
Index: scikit_image-0.26.0/src/skimage/restoration/_rolling_ball_cy.pyx
===================================================================
--- scikit_image-0.26.0.orig/src/skimage/restoration/_rolling_ball_cy.pyx
+++ scikit_image-0.26.0/src/skimage/restoration/_rolling_ball_cy.pyx
@@ -112,7 +112,7 @@ def apply_kernel_nan(DTYPE_FLOAT[::1] im
     -------
     out_data : ndarray
         The array passed into ``out``, reshaped to
-        ``out_data.shape = img_shape`` (possibly a view) and filled with the
+        ``out_data.reshape(img_shape, copy=False)`` (possibly a view) and filled with the
         estimated background intensity.
 
     See Also
@@ -196,7 +196,7 @@ def apply_kernel(DTYPE_FLOAT[::1] img no
     -------
     out_data : ndarray
         The array passed into ``out``, reshaped to
-        ``out_data.shape = img_shape`` (possibly a view) and filled with the
+        ``out_data.reshape(img_shape, copy=False)`` (possibly a view) and filled with the
         estimated background intensity.
 
     See Also
Index: scikit_image-0.26.0/src/skimage/util/_map_array.py
===================================================================
--- scikit_image-0.26.0.orig/src/skimage/util/_map_array.py
+++ scikit_image-0.26.0/src/skimage/util/_map_array.py
@@ -58,7 +58,7 @@ def map_array(input_arr, input_vals, out
         )
     try:
         out_view = out.view()
-        out_view.shape = (-1,)  # no-copy reshape/ravel
+        out_view = np.reshape(out_view, (-1,), copy=False)
     except AttributeError:  # if out strides are not compatible with 0-copy
         raise ValueError(
             'If out array is provided, it should be either contiguous '
Index: scikit_image-0.26.0/tools/generate_requirements.py
===================================================================
--- scikit_image-0.26.0.orig/tools/generate_requirements.py
+++ scikit_image-0.26.0/tools/generate_requirements.py
@@ -58,6 +58,10 @@ def generate_environment_yml(req_section
 
             lines.append(f"  - {dep}")
 
+            # Strip duplicates
+            if re.split('[>=]', lines[-2])[0] == re.split('[>=]', lines[-1])[0]:
+                lines = lines[:-1]
+
     with open("environment.yml", "w") as f:
         f.writelines(f"{line}\n" for line in lines)
 
