--- openafs-1.8.0//src/libafs/MakefileProto.LINUX.in.orig	2018-06-27 07:57:10.916249040 +0200
+++ openafs-1.8.0//src/libafs/MakefileProto.LINUX.in	2018-06-27 07:57:56.492849436 +0200
@@ -320,7 +320,7 @@
 ${LINUX_LIBAFS_NAME}.ko afspag.ko: .FORCE
 	env EXTRA_CFLAGS="${EXTRA_CFLAGS}" @TOP_SRCDIR@/libafs/make_kbuild_makefile.pl ${KDIR} $@ @TOP_OBJDIR@/src/config/Makefile.config Makefile.afs Makefile.common
 	( env EXTRA_CFLAGS="${EXTRA_CFLAGS}" \
-		$(MAKE) -C ${LINUX_KERNEL_BUILD} M=@TOP_OBJDIR@/src/libafs/${KDIR} modules 2>&1 \
+		$(MAKE) -C ${LINUX_KERNEL_BUILD} ${LINUX_MAKE_ARCH} M=@TOP_OBJDIR@/src/libafs/${KDIR} modules 2>&1 \
 		|| echo "FAILURE: make exit code $$?" ) | tee .makelog
 	@if grep -E -q '^(FAILURE|WARNING)' .makelog; then \
 		grep -q '^WARNING' .makelog && echo "Error: Undefined symbols in modules" ; \
