---
 ulogd.conf.in |   29 +++++++++++------------------
 1 file changed, 11 insertions(+), 18 deletions(-)

Index: ulogd-2.0.7/ulogd.conf.in
===================================================================
--- ulogd-2.0.7.orig/ulogd.conf.in
+++ ulogd-2.0.7/ulogd.conf.in
@@ -6,9 +6,10 @@
 # GLOBAL OPTIONS
 ######################################################################
 
+user="ulogd"
 
 # logfile for status messages
-logfile="/var/log/ulogd.log"
+logfile="/var/log/ulogd/ulogd.log"
 
 # loglevel: debug(1), info(3), notice(5), error(7) or fatal(8) (default 5)
 # loglevel=1
@@ -58,9 +59,6 @@ logfile="/var/log/ulogd.log"
 # this is a stack for packet-based logging via LOGEMU
 #stack=log2:NFLOG,base1:BASE,ifi1:IFINDEX,ip2str1:IP2STR,print1:PRINTPKT,emu1:LOGEMU
 
-# this is a stack for ULOG packet-based logging via LOGEMU
-#stack=ulog1:ULOG,base1:BASE,ip2str1:IP2STR,print1:PRINTPKT,emu1:LOGEMU
-
 # this is a stack for packet-based logging via LOGEMU with filtering on MARK
 #stack=log2:NFLOG,base1:BASE,mark1:MARK,ifi1:IFINDEX,ip2str1:IP2STR,print1:PRINTPKT,emu1:LOGEMU
 
@@ -175,34 +173,29 @@ numeric_label=1 # you can label the log
 #netlink_socket_buffer_maxsize=1085440
 #bind=1
 
-[ulog1]
-# netlink multicast group (the same as the iptables --ulog-nlgroup param)
-nlgroup=1
-#numeric_label=0 # optional argument
-
 [nuauth1]
 socket_path="/tmp/nuauth_ulogd2.sock"
 
 [emu1]
-file="/var/log/ulogd_syslogemu.log"
+file="/var/log/ulogd/syslogemu.log"
 sync=1
 
 [op1]
-file="/var/log/ulogd_oprint.log"
+file="/var/log/ulogd/oprint.log"
 sync=1
 
 [gp1]
-file="/var/log/ulogd_gprint.log"
+file="/var/log/ulogd/gprint.log"
 sync=1
 timestamp=1
 
 [xml1]
-directory="/var/log/"
+directory="/var/log/ulogd/"
 sync=1
 
 [json1]
 sync=1
-#file="/var/log/ulogd.json"
+#file="/var/log/ulogd/ulogd.json"
 #timestamp=0
 # device name to be used in JSON message
 #device="My awesome Netfilter firewall"
@@ -216,7 +209,7 @@ sync=1
 
 [pcap1]
 #default file is /var/log/ulogd.pcap
-#file="/var/log/ulogd.pcap"
+#file="/var/log/ulogd/ulogd.pcap"
 sync=1
 
 [mysql1]
@@ -301,18 +294,18 @@ procedure="INSERT_PACKET_FULL"
 
 [sqlite3_ct]
 table="ulog_ct"
-db="/var/log/ulogd.sqlite3db"
+db="/var/log/ulogd/ulogd.sqlite3db"
 
 [sqlite3_pkt]
 table="ulog_pkt"
-db="/var/log/ulogd.sqlite3db"
+db="/var/log/ulogd/ulogd.sqlite3db"
 
 [sys2]
 facility=LOG_LOCAL2
 
 [nacct1]
 sync = 1
-#file = /var/log/ulogd_nacct.log
+#file = /var/log/ulogd/nacct.log
 
 [mark1]
 mark = 1
