--- setup.py.smb	2024-01-19 09:43:25.073225511 -0500
+++ setup.py	2024-01-19 10:26:09.830214118 -0500
@@ -980,7 +980,7 @@
     if args.skip_building_kitten:
         print('Skipping building of the kitten binary because of a command line option. Build is incomplete', file=sys.stderr)
         return ''
-    cmd = [go, 'build', '-v']
+    cmd = [go, 'build', '-buildmode=pie', '-v']
     vcs_rev = args.vcs_rev or get_vcs_rev()
     ld_flags: List[str] = []
     binary_data_flags = [f"-X kitty.VCSRevision={vcs_rev}"]
@@ -1165,13 +1165,14 @@
     base = Path(ddir)
     in_src_launcher = base / (f'{libdir_name}/kitty/kitty/launcher/kitty')
     launcher = base / 'bin/kitty'
-    skip_docs = False
+    skip_docs = True
     if not os.path.exists('docs/_build/html'):
         kitten_exe = os.path.join(os.path.dirname(str(launcher)), 'kitten')
         if os.path.exists(kitten_exe):
             os.environ['KITTEN_EXE_FOR_DOCS'] = kitten_exe
             make = 'gmake' if is_freebsd else 'make'
-            run_tool([make, 'docs'])
+            if not skip_docs:
+                run_tool([make, 'docs'])
         else:
             if args.skip_building_kitten:
                 skip_docs = True
