--- setup.py.smb	2023-09-21 01:30:53.606748707 -0400
+++ setup.py	2023-09-21 01:36:16.627212417 -0400
@@ -143,6 +143,7 @@
     verbose: int = 0
     sanitize: bool = False
     prefix: str = './linux-package'
+    no_docs: bool = False
     dir_for_static_binaries: str = 'build/static'
     skip_code_generation: bool = False
     clean_for_cross_compile: bool = False
@@ -1149,12 +1150,13 @@
         invalidation_mode=py_compile.PycInvalidationMode.UNCHECKED_HASH, ddir='')
 
 
-def create_linux_bundle_gunk(ddir: str, libdir_name: str) -> None:
-    if not os.path.exists('docs/_build/html'):
-        make = 'gmake' if is_freebsd else 'make'
-        run_tool([make, 'docs'])
-    copy_man_pages(ddir)
-    copy_html_docs(ddir)
+def create_linux_bundle_gunk(ddir: str, libdir_name: str, no_docs=False) -> None:
+    if not no_docs:
+        if not os.path.exists('docs/_build/html'):
+            make = 'gmake' if is_freebsd else 'make'
+            run_tool([make, 'docs'])
+        copy_man_pages(ddir)
+        copy_html_docs(ddir)
     for (icdir, ext) in {'256x256': 'png', 'scalable': 'svg'}.items():
         icdir = os.path.join(ddir, 'share', 'icons', 'hicolor', icdir, 'apps')
         safe_makedirs(icdir)
@@ -1536,7 +1538,7 @@
     if not for_freeze and not bundle_type.startswith('macos-'):
         build_static_kittens(args, launcher_dir=launcher_dir)
     if not is_macos:
-        create_linux_bundle_gunk(ddir, args.libdir_name)
+        create_linux_bundle_gunk(ddir, args.libdir_name, args.no_docs)
 
     if bundle_type.startswith('macos-'):
         create_macos_bundle_gunk(ddir, for_freeze, args)
@@ -1640,6 +1642,12 @@
         help='Where to create the linux package'
     )
     p.add_argument(
+        '--no-docs',
+        default=Options.no_docs,
+        action='store_true',
+        help='Disable building documentation when creating the linux package'
+    )
+    p.add_argument(
         '--dir-for-static-binaries',
         default=Options.dir_for_static_binaries,
         help='Where to create the static kitten binary'
