Index: lapdog-1.1/init/lapdog.service
===================================================================
--- lapdog-1.1.orig/init/lapdog.service
+++ lapdog-1.1/init/lapdog.service
@@ -2,6 +2,19 @@
 Description=A LAN device presence checker
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+PrivateDevices=true
+ProtectHostname=true
+ProtectClock=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 Type=simple
 PIDFile=/run/lapdog.pid
 ExecStart=/usr/sbin/lapdog
