Index: ckermit-9.0.302/ckcpro.c
===================================================================
--- ckermit-9.0.302.orig/ckcpro.c
+++ ckermit-9.0.302/ckcpro.c
@@ -173,7 +173,9 @@ _PROTOTYP( int cmdsrc, (void) );
   extern int wslots, success, xitsta, rprintf, discard, cdtimo, keep, fdispla;
   extern int timef, stdinf, rscapu, sendmode, epktflg, epktrcvd, epktsent;
   extern int binary, fncnv;
-  extern long speed, ffc, crc16, calibrate, dest;
+  extern long speed, crc16;
+  extern CK_OFF_T ffc, calibrate;
+  extern int dest;
 #ifdef COMMENT
   extern char *TYPCMD, *DIRCMD, *DIRCM2;
 #endif /* COMMENT */
Index: ckermit-9.0.302/ckcpro.w
===================================================================
--- ckermit-9.0.302.orig/ckcpro.w
+++ ckermit-9.0.302/ckcpro.w
@@ -152,7 +152,9 @@ _PROTOTYP( int cmdsrc, (void) );
   extern int wslots, success, xitsta, rprintf, discard, cdtimo, keep, fdispla;
   extern int timef, stdinf, rscapu, sendmode, epktflg, epktrcvd, epktsent;
   extern int binary, fncnv;
-  extern long speed, ffc, crc16, calibrate, dest;
+  extern long speed, crc16;
+  extern CK_OFF_T ffc, calibrate;
+  extern int dest;
 #ifdef COMMENT
   extern char *TYPCMD, *DIRCMD, *DIRCM2;
 #endif /* COMMENT */
