Author: Adam Majer <amajer@suse.de>
Date: Fri Mar  3 15:59:09 CET 2017
Summary: Remove pure ugliness with paths

Index: mariadb-connector-c-3.1.8-src/mariadb_config/mariadb_config.c.in
===================================================================
--- mariadb-connector-c-3.1.8-src.orig/mariadb_config/mariadb_config.c.in
+++ mariadb-connector-c-3.1.8-src/mariadb_config/mariadb_config.c.in
@@ -5,13 +5,13 @@
 
 static char *mariadb_progname;
 
-#define INCLUDE "-I@CMAKE_SYSROOT@@CMAKE_INSTALL_PREFIX@/@INSTALL_INCLUDEDIR@ -I@CMAKE_SYSROOT@@CMAKE_INSTALL_PREFIX@/@INSTALL_INCLUDEDIR@/mysql"
-#define LIBS    "-L@CMAKE_SYSROOT@@CMAKE_INSTALL_PREFIX@/@INSTALL_LIBDIR@/ -lmariadb"
+#define INCLUDE "-I@INSTALL_INCLUDEDIR@ -I@INSTALL_INCLUDEDIR@/mariadb"
+#define LIBS    "-L@INSTALL_LIBDIR@ -lmariadb"
 #define LIBS_SYS "@extra_dynamic_LDFLAGS@"
 #define CFLAGS  INCLUDE
 #define VERSION "@MARIADB_CLIENT_VERSION@"
 #define CC_VERSION "@CPACK_PACKAGE_VERSION@"
-#define PLUGIN_DIR "@CMAKE_INSTALL_PREFIX@/@INSTALL_PLUGINDIR@"
+#define PLUGIN_DIR "@INSTALL_PLUGINDIR@"
 #define SOCKET  "@MARIADB_UNIX_ADDR@"
 #define PORT "@MARIADB_PORT@"
 #define TLS_LIBRARY_VERSION "@TLS_LIBRARY_VERSION@"
Index: mariadb-connector-c-3.1.8-src/mariadb_config/libmariadb.pc.in
===================================================================
--- mariadb-connector-c-3.1.8-src.orig/mariadb_config/libmariadb.pc.in
+++ mariadb-connector-c-3.1.8-src/mariadb_config/libmariadb.pc.in
@@ -7,8 +7,8 @@
 #
 
 prefix=@CMAKE_INSTALL_PREFIX@
-includedir=${prefix}/@INSTALL_INCLUDEDIR@/@SUFFIX_INSTALL_DIR@
-libdir=${prefix}/@INSTALL_LIBDIR@/@SUFFIX_INSTALL_DIR@
+includedir=@INSTALL_INCLUDEDIR@
+libdir=@INSTALL_LIBDIR@
 
 Name: libmariadb
 Version: @CPACK_PACKAGE_VERSION@
