From 18650d56088fb65751793f24e550d5f36fb75ea9 Mon Sep 17 00:00:00 2001
From: Yuri Chornoivan <yurchor@ukr.net>
Date: Mon, 8 Oct 2018 09:07:20 +0300
Subject: [PATCH 15/35] Fix minor typo

---
 src/dialogs/projectdialogs.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dialogs/projectdialogs.cpp b/src/dialogs/projectdialogs.cpp
index 38466b6b..029e432c 100644
--- a/src/dialogs/projectdialogs.cpp
+++ b/src/dialogs/projectdialogs.cpp
@@ -154,7 +154,7 @@ bool KileProjectDialogBase::acceptUserExtensions()
             QStringList list = m_val_extensions[i-1].split(' ');
             for (it = list.constBegin(); it != list.constEnd(); ++it) {
                 if (! reg.exactMatch(*it)) {
-                    KMessageBox::error(this, i18n("Error in extension '%s':\nAll user-defined extensions should look like '.xyz'", *it), i18n("Invalid extension"));
+                    KMessageBox::error(this, i18n("Error in extension '%1':\nAll user-defined extensions should look like '.xyz'", *it), i18n("Invalid extension"));
                     return false;
                 }
             }
-- 
2.20.1

