Index: xmms2-0.8DrO_o/wscript
===================================================================
--- xmms2-0.8DrO_o.orig/wscript
+++ xmms2-0.8DrO_o/wscript
@@ -448,6 +448,9 @@ int main() { return 0; }
             enabled_optionals, disabled_optionals,
             output_plugins)
 
+    # Do not link against unused libraries (make dpkg-shlibdeps happy).
+    conf.env.LINKFLAGS += ['-Wl,--as-needed']
+
     return True
 
 ####
