Index: xmms2-0.8DrO_o/src/plugins/pulse/wscript
===================================================================
--- xmms2-0.8DrO_o.orig/src/plugins/pulse/wscript
+++ xmms2-0.8DrO_o/src/plugins/pulse/wscript
@@ -3,5 +3,5 @@ from waftools.plugin import plugin
 def plugin_configure(conf):
     conf.check_cfg(package="libpulse", args="--cflags --libs", uselib_store="pulse")
 
-configure, build = plugin("pulse", configure=plugin_configure, output_prio=25,
+configure, build = plugin("pulse", configure=plugin_configure, output_prio=45,
         libs=["pulse"])
Index: xmms2-0.8DrO_o/wscript
===================================================================
--- xmms2-0.8DrO_o.orig/wscript
+++ xmms2-0.8DrO_o/wscript
@@ -295,8 +295,8 @@ def configure(conf):
         conf.msg("uncommited changed", changed and "yes" or "no")
         conf.env.VERSION = "%s (git commit: %s%s)" % (BASEVERSION, nam, dirty)
 
-    conf.env.append_unique('CFLAGS', ['-g', '-O0'])
-    conf.env.append_unique('CXXFLAGS', ['-g', '-O0'])
+    #conf.env.append_unique('CFLAGS', ['-g', '-O0'])
+    #conf.env.append_unique('CXXFLAGS', ['-g', '-O0'])
 
     if conf.options.with_profiling:
         conf.env.with_profiling = True
