diff -Naur xmms2-0.8DrO_o.orig/src/plugins/samba/samba.c xmms2-0.8DrO_o/src/plugins/samba/samba.c
--- xmms2-0.8DrO_o.orig/src/plugins/samba/samba.c	2011-10-20 21:26:08.000000000 +0200
+++ xmms2-0.8DrO_o/src/plugins/samba/samba.c	2013-11-05 14:10:11.540534423 +0100
@@ -27,7 +27,7 @@
 #include <string.h>
 #include <unistd.h>
 
-#include "libsmbclient.h"
+#include "samba-4.0/libsmbclient.h"
 
 /*
  * libsmbclient is not threadsafe.
diff -Naur xmms2-0.8DrO_o.orig/src/plugins/samba/wscript xmms2-0.8DrO_o/src/plugins/samba/wscript
--- xmms2-0.8DrO_o.orig/src/plugins/samba/wscript	2011-10-20 21:26:08.000000000 +0200
+++ xmms2-0.8DrO_o/src/plugins/samba/wscript	2013-11-05 14:09:58.668511884 +0100
@@ -1,7 +1,7 @@
 from waftools.plugin import plugin
 
 def plugin_configure(conf):
-    conf.check_cc(header_name="libsmbclient.h")
+    conf.check_cc(header_name="samba-4.0/libsmbclient.h")
     conf.check(lib="smbclient", uselib_store="smbclient")
 
 configure, build = plugin("samba", configure=plugin_configure,
