From d792be6717b57ba8413d261d9a125b9eb93230de Mon Sep 17 00:00:00 2001
From: "Jonatan \"jaw\" Wallmander" <jonatan@vovoid.com>
Date: Sun, 10 Jan 2016 19:18:54 +0100
Subject: [PATCH] Types need undef under Linux as well

---
 lzhamdecomp/lzham_types.h | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/lzhamdecomp/lzham_types.h b/lzhamdecomp/lzham_types.h
index 47f68a3..3811e3d 100644
--- a/lzhamdecomp/lzham_types.h
+++ b/lzhamdecomp/lzham_types.h
@@ -3,26 +3,26 @@
 #pragma once
 
 // TODO
-#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__NetBSD__)
-   #undef INT8_MIN
-   #undef INT8_MAX
+#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__NetBSD__) \
+    || defined(linux) || defined(__linux) || defined(__linux__)
+
    #undef UINT8_MIN
    #undef UINT8_MAX
-
-   #undef INT16_MIN
-   #undef INT16_MAX
    #undef UINT16_MIN
    #undef UINT16_MAX
-
-   #undef INT32_MIN
-   #undef INT32_MAX
    #undef UINT32_MIN
    #undef UINT32_MAX
-
-   #undef INT64_MIN
-   #undef INT64_MAX
    #undef UINT64_MIN
    #undef UINT64_MAX
+   #undef INT8_MIN
+   #undef INT8_MAX
+   #undef INT16_MIN
+   #undef INT16_MAX
+   #undef INT32_MIN
+   #undef INT32_MAX
+   #undef INT64_MIN
+   #undef INT64_MAX
+
 #endif
 
 namespace lzham
