---
 healpy/sphtfunc.py |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

Index: healpy-1.16.6/healpy/sphtfunc.py
===================================================================
--- healpy-1.16.6.orig/healpy/sphtfunc.py
+++ healpy-1.16.6/healpy/sphtfunc.py
@@ -24,7 +24,10 @@ import numpy as np
 
 import astropy.io.fits as pf
 from .utils.deprecation import deprecated_renamed_argument
-from scipy.integrate import trapz
+try:
+    from scipy.integrate import trapezoid as trapz
+except ImportError:
+    from scipy.integrate import trapz
 from astropy.utils import data
 
 DATAURL = "https://healpy.github.io/healpy-data/"
