Index: setupmeta-3.3.0/tests/test_commands.py
===================================================================
--- setupmeta-3.3.0.orig/tests/test_commands.py
+++ setupmeta-3.3.0/tests/test_commands.py
@@ -1,7 +1,7 @@
 import os
 import re
 
-from mock import patch
+from unittest.mock import patch
 from six import StringIO
 
 import setupmeta
Index: setupmeta-3.3.0/tests/test_model.py
===================================================================
--- setupmeta-3.3.0.orig/tests/test_model.py
+++ setupmeta-3.3.0/tests/test_model.py
@@ -1,7 +1,7 @@
 import os
 import sys
 
-from mock import patch
+from unittest.mock import patch
 
 import setupmeta
 from setupmeta.model import Definition, DefinitionEntry, is_setup_py_path
Index: setupmeta-3.3.0/tests/test_versioning.py
===================================================================
--- setupmeta-3.3.0.orig/tests/test_versioning.py
+++ setupmeta-3.3.0/tests/test_versioning.py
@@ -3,7 +3,7 @@ import sys
 
 import pep440
 import pytest
-from mock import patch
+from unittest.mock import patch
 
 import setupmeta
 import setupmeta.versioning
