diff -Nur Wyrmgus-5.2.4/src/economy/resource_finder.h new/src/economy/resource_finder.h
--- Wyrmgus-5.2.4/src/economy/resource_finder.h	2021-11-17 15:34:51.000000000 +0100
+++ new/src/economy/resource_finder.h	2021-12-06 13:58:39.499619874 +0100
@@ -36,7 +36,7 @@
 {
 	bool is_valid() const
 	{
-		this->resource_pos != QPoint(-1, -1) || this->resource_unit != nullptr;
+		return this->resource_pos != QPoint(-1, -1) || this->resource_unit != nullptr;
 	}
 
 	QPoint resource_pos = QPoint(-1, -1);
