diff -upr django-tastypie-0.14.4.orig/docs/code/authentication.py django-tastypie-0.14.4/docs/code/authentication.py
--- django-tastypie-0.14.4.orig/docs/code/authentication.py	2022-05-11 14:26:13.697935887 +0200
+++ django-tastypie-0.14.4/docs/code/authentication.py	2022-05-11 14:26:13.713935987 +0200
@@ -1,4 +1,4 @@
-import mock
+from unittest import mock
 
 
 OAuth20Authentication = mock.Mock()
diff -upr django-tastypie-0.14.4.orig/docs/code/myapp/models.py django-tastypie-0.14.4/docs/code/myapp/models.py
--- django-tastypie-0.14.4.orig/docs/code/myapp/models.py	2022-05-11 14:26:13.697935887 +0200
+++ django-tastypie-0.14.4/docs/code/myapp/models.py	2022-05-11 14:26:13.713935987 +0200
@@ -1,4 +1,4 @@
-import mock
+from unittest import mock
 
 from django.contrib.auth.models import User  # noqa
 
diff -upr django-tastypie-0.14.4.orig/tests/authorization/tests.py django-tastypie-0.14.4/tests/authorization/tests.py
--- django-tastypie-0.14.4.orig/tests/authorization/tests.py	2022-05-11 14:26:13.701935911 +0200
+++ django-tastypie-0.14.4/tests/authorization/tests.py	2022-05-11 14:26:13.721936038 +0200
@@ -1,5 +1,5 @@
 import json
-import mock
+from unittest import mock
 
 from django.contrib.auth.models import User
 from django.contrib.sites.models import Site
diff -upr django-tastypie-0.14.4.orig/tests/core/tests/cache.py django-tastypie-0.14.4/tests/core/tests/cache.py
--- django-tastypie-0.14.4.orig/tests/core/tests/cache.py	2022-05-11 14:26:13.705935937 +0200
+++ django-tastypie-0.14.4/tests/core/tests/cache.py	2022-05-11 14:26:13.721936038 +0200
@@ -1,4 +1,4 @@
-import mock
+from unittest import mock
 
 from django.core.cache import cache
 from django.test import TestCase
diff -upr django-tastypie-0.14.4.orig/tests/core/tests/resources.py django-tastypie-0.14.4/tests/core/tests/resources.py
--- django-tastypie-0.14.4.orig/tests/core/tests/resources.py	2022-05-11 14:26:13.705935937 +0200
+++ django-tastypie-0.14.4/tests/core/tests/resources.py	2022-05-11 14:26:13.725936064 +0200
@@ -7,7 +7,7 @@ import copy
 import datetime
 from decimal import Decimal
 import json
-from mock import patch, Mock
+from unittest.mock import patch, Mock
 import sys
 import time
 from unittest import skipIf
diff -upr django-tastypie-0.14.4.orig/tests/core/tests/serializers.py django-tastypie-0.14.4/tests/core/tests/serializers.py
--- django-tastypie-0.14.4.orig/tests/core/tests/serializers.py	2022-05-11 14:26:13.705935937 +0200
+++ django-tastypie-0.14.4/tests/core/tests/serializers.py	2022-05-11 14:26:13.725936064 +0200
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 import datetime
 from decimal import Decimal
-import mock
+from unittest import mock
 from unittest import skipIf
 import yaml
 
diff -upr django-tastypie-0.14.4.orig/tests/core/tests/throttle.py django-tastypie-0.14.4/tests/core/tests/throttle.py
--- django-tastypie-0.14.4.orig/tests/core/tests/throttle.py	2022-05-11 14:26:13.705935937 +0200
+++ django-tastypie-0.14.4/tests/core/tests/throttle.py	2022-05-11 14:26:13.725936064 +0200
@@ -1,4 +1,4 @@
-import mock
+from unittest import mock
 import time
 
 from django.core.cache import cache
diff -upr django-tastypie-0.14.4.orig/tests/core/tests/utils.py django-tastypie-0.14.4/tests/core/tests/utils.py
--- django-tastypie-0.14.4.orig/tests/core/tests/utils.py	2022-05-11 14:26:13.705935937 +0200
+++ django-tastypie-0.14.4/tests/core/tests/utils.py	2022-05-11 14:26:13.725936064 +0200
@@ -1,5 +1,5 @@
 import datetime
-import mock
+from unittest import mock
 from pytz.reference import Pacific
 
 from django.http import HttpRequest
