Index: accountsservice-23.13.9/subprojects/mocklibc.wrap
===================================================================
--- accountsservice-23.13.9.orig/subprojects/mocklibc.wrap
+++ accountsservice-23.13.9/subprojects/mocklibc.wrap
@@ -3,7 +3,7 @@ directory = mocklibc-1.0
 
 source_url = https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/mocklibc/mocklibc-1.0.tar.gz
 source_filename = mocklibc-1.0.tar.gz
-source_hash = b2236a6af1028414783e9734a46ea051916ec226479d6a55a3bb823bff68f120
+source_hash = 48966f7a5d3f2139ffdc1e700de7deb9390f852cd4f2a7b97e55029c14097c6e
 
 patch_url = https://wrapdb.mesonbuild.com/v1/projects/mocklibc/1.0/2/get_zip
 patch_filename = mocklibc-1.0-2-wrap.zip
