Index: htop-2.2.0/htop.c
===================================================================
--- htop-2.2.0.orig/htop.c
+++ htop-2.2.0/htop.c
@@ -93,7 +93,7 @@ static CommandLineSettings parseArgument
 
    int opt, opti=0;
    /* Parse arguments */
-   while ((opt = getopt_long(argc, argv, "hvCst::d:u:p:i", long_opts, &opti))) {
+   while ((opt = getopt_long(argc, argv, "hvCs:td:u:p:i", long_opts, &opti))) {
       if (opt == EOF) break;
       switch (opt) {
          case 'h':
