diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3bb2337..5ad369e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -342,9 +342,6 @@ set_target_properties(engine PROPERTIES
 	CXX_STANDARD 98
 	OUTPUT_NAME "${ENGINE_BINARY_NAME}"
 	RUNTIME_OUTPUT_DIRECTORY "${OUTPUT_DIR}"
-	# On Linux look for any libraries we link against in the executable directory.
-	INSTALL_RPATH "\$ORIGIN"
-	BUILD_WITH_INSTALL_RPATH ON
 )
 
 # Install
