---
 contrib/systemd/osmo-stp.service |   13 +++++++++++++
 1 file changed, 13 insertions(+)

Index: libosmo-sccp-1.7.0/contrib/systemd/osmo-stp.service
===================================================================
--- libosmo-sccp-1.7.0.orig/contrib/systemd/osmo-stp.service
+++ libosmo-sccp-1.7.0/contrib/systemd/osmo-stp.service
@@ -3,6 +3,19 @@ Description=Osmocom STP (Signal Transfer
 Documentation=https://osmocom.org/projects/osmo-stp/wiki
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+PrivateDevices=true
+ProtectHostname=true
+ProtectClock=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 Type=simple
 Restart=always
 StateDirectory=osmocom
