Index: xcoral-3.49.8/Makefile.in
===================================================================
--- xcoral-3.49.8.orig/Makefile.in
+++ xcoral-3.49.8/Makefile.in
@@ -22,7 +22,7 @@ LDOPTIONS    = @LDFLAGS@
 XCORALLIBDIR = \"$(XC_LIBDIR)\"
 X_INCLUDE    = @XC_X_INCLUDES@
 
-CFLAGS       = -Wall @CFLAGS@ @DEFS@ $(X_INCLUDE) -DXCORAL_LIB_DIR=$(XCORALLIBDIR)
+CFLAGS       = -Wall -fno-strict-aliasing @CFLAGS@ @DEFS@ $(X_INCLUDE) -DXCORAL_LIB_DIR=$(XCORALLIBDIR)
 LDLIBS       = @XC_X_LDPATH@ @x_libs@ @LIBS@
 
 ifeq ($(HAVE_XFT),true)
Index: xcoral-3.49.8/tbox.c
===================================================================
--- xcoral-3.49.8.orig/tbox.c
+++ xcoral-3.49.8/tbox.c
@@ -54,6 +54,7 @@
 extern Display *dpy;
 static  Atom trombines_del;
 //static  int display_tbox = True;
+TBox trombines_box;
 
 XImage *image[3];
 XImage * LoadImage16 ();
Index: xcoral-3.49.8/Smac/Makefile.in
===================================================================
--- xcoral-3.49.8.orig/Smac/Makefile.in
+++ xcoral-3.49.8/Smac/Makefile.in
@@ -6,7 +6,7 @@ LEX        = mylex
 YACC       = @YACC@
 MAKEDEPEND = @MAKEDEPEND@
 
-CFLAGS  = @CFLAGS@ -D$(LEX) $(XC_DEFS) -DXCORAL_LIB_DIR=$(XCORALLIBDIR) @DEFS@
+CFLAGS  = @CFLAGS@ -fno-strict-aliasing -D$(LEX) $(XC_DEFS) -DXCORAL_LIB_DIR=$(XCORALLIBDIR) @DEFS@
 LDFLAGS = 
 
 
