Index: fcitx5-5.0.23/src/modules/dbus/dbusmodule.cpp
===================================================================
--- fcitx5-5.0.23.orig/src/modules/dbus/dbusmodule.cpp
+++ fcitx5-5.0.23/src/modules/dbus/dbusmodule.cpp
@@ -571,7 +571,7 @@ public:
                 }
                 ss << "] program:" << ic->program()
                    << " frontend:" << ic->frontend()
-                   << " cap:" << fmt::format("{:x}", ic->capabilityFlags())
+                   << " cap:" << fmt::format("{:x}", static_cast<uint64_t>(ic->capabilityFlags()))
                    << " focus:" << ic->hasFocus() << std::endl;
                 return true;
             });
