Index: Photini-2021.1.0/src/photini/data/linux/photini.desktop
===================================================================
--- Photini-2021.1.0.orig/src/photini/data/linux/photini.desktop
+++ Photini-2021.1.0/src/photini/data/linux/photini.desktop
@@ -3,6 +3,8 @@ Type=Application
 Name=Photini
 Comment=An easy to use digital photograph metadata (EXIF, IPTC, XMP) editing application.
 GenericName=Photini photo metadata editor
+Exec=photini
+Icon=photini
 Terminal=false
 Categories=Graphics;Photography;
 MimeType=image/jpeg;image/jpeg2000;image/tiff;image/png;image/gif;image/svg+xml;image/x-dcraw;
Index: Photini-2021.1.0/setup.py
===================================================================
--- Photini-2021.1.0.orig/setup.py
+++ Photini-2021.1.0/setup.py
@@ -57,9 +57,6 @@ setup(name = 'Photini',
       package_dir = {'' : 'src'},
       package_data = {'photini' : package_data},
       entry_points = {
-          'console_scripts' : [
-              'photini-post-install = photini.scripts:post_install',
-              ],
           'gui_scripts' : [
               'photini = photini.editor:main',
               ],
