Author: Bernhard M. Wiedemann <bwiedemann suse de>
Date: 2017-12-15

Index: arj-3.10.22/today.c
===================================================================
--- arj-3.10.22.orig/today.c
+++ arj-3.10.22/today.c
@@ -44,8 +44,8 @@ int main(int argc, char **argv)
   out_name[++l]='\0';
  }
  strcat(out_name, "date_sig.c");
- cur_unixtime=time(NULL);
- stm=localtime(&cur_unixtime);
+ cur_unixtime=1119484800;
+ stm=gmtime(&cur_unixtime);
  if(!stricmp(argv[1], "en"))
   sprintf(date_sig, "[%02d %s %04d]", stm->tm_mday, months_en[stm->tm_mon], stm->tm_year+1900);
  else if(!stricmp(argv[1], "fr"))
