--- a/belr.pc.in
+++ b/belr.pc.in
@@ -4,6 +4,7 @@ prefix=@CMAKE_INSTALL_PREFIX@
 Name:  @PROJECT_NAME@
 Description: Belr is Belledonne Communications' language recognition library
 Version: @PROJECT_VERSION@
+Requires.private: bctoolbox
 Libs: -L@CMAKE_INSTALL_FULL_LIBDIR@ -lbelr
 Libs.private: @LIBS_PRIVATE@
 Cflags: -I@CMAKE_INSTALL_FULL_INCLUDEDIR@
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -83,6 +83,11 @@ if(MSVC)
 	include_directories(${MSVC_INCLUDE_DIR})
 endif()
 
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/belr.pc.in ${CMAKE_CURRENT_BINARY_DIR}/belr.pc)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/belr.pc DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+
+
 set(STRICT_OPTIONS_CPP )
 set(STRICT_OPTIONS_C )
 set(STRICT_OPTIONS_CXX )
