Index: scintilla/gtk/makefile
===================================================================
--- scintilla.orig/gtk/makefile
+++ scintilla/gtk/makefile
@@ -50,7 +50,8 @@ ifndef windir
 BASE_FLAGS += -fPIC
 endif
 
-LDFLAGS += -dynamiclib
+# Only macOS needs/supports -dynamiclib
+# LDFLAGS += -dynamiclib
 LDFLAGS += -shared
 
 # Take care of changing Unix style '/' directory separator to '\' on Windows
@@ -66,7 +67,7 @@ DEL = del /q
 else
 DEL = rm -f
 endif
-COMPLIB=$(basedir)/bin/scintilla.a
+COMPLIB=$(basedir)/bin/libscintilla.la
 COMPONENT=$(basedir)/bin/libscintilla.$(SHAREDEXTENSION)
 
 vpath %.h $(srcdir) $(basedir)/src $(basedir)/include $(basedir)/lexlib
@@ -82,12 +83,12 @@ DEFINES += -DNO_CXX11_REGEX
 endif
 
 DEFINES += -D$(if $(DEBUG),DEBUG,NDEBUG)
-BASE_FLAGS += $(if $(DEBUG),-g,-Os)
+BASE_FLAGS += $(if $(DEBUG),-g)
 
 CXX_BASE_FLAGS =--std=c++17 $(BASE_FLAGS)
 CXX_ALL_FLAGS =$(DEFINES) $(INCLUDES) $(CXX_BASE_FLAGS) $(CONFIG_FLAGS)
 
-CONFIG_FLAGS:=$(shell $(PKG_CONFIG) --cflags $(GTK_VERSION))
+CONFIG_FLAGS:=$(shell $(PKG_CONFIG) --cflags --libs $(GTK_VERSION))
 CONFIGLIB:=$(shell $(PKG_CONFIG) --libs $(GTK_VERSION) gmodule-no-export-2.0)
 MARSHALLER=scintilla-marshal.o
 
@@ -161,8 +162,7 @@ GTK_OBJS = \
 	ScintillaGTKAccessible.o
 
 $(COMPLIB): $(SRC_OBJS) $(GTK_OBJS) $(MARSHALLER) $(LEX_OBJS)
-	$(AR) $(ARFLAGS) $@ $^
-	$(RANLIB) $@
+	libtool --mode=link gcc $(CONFIGFLAGS) -o $@ $^ -rpath $(LIBDIR) $(CONFIGLDFLAGS) -version-info $(VERSION)
 
 $(COMPONENT): $(SRC_OBJS) $(GTK_OBJS) $(MARSHALLER)
 	$(CXX) $(CXX_ALL_FLAGS) $(CXXFLAGS) $(LDFLAGS) $^ -o $@ $(CONFIGLIB)
