From 681d7041f0e9f50735bcea3b3b896e0155111950 Mon Sep 17 00:00:00 2001
From: "Friedrich W. H. Kossebau" <kossebau@kde.org>
Date: Tue, 15 Dec 2020 13:26:45 +0100
Subject: [PATCH] Add missing module identifier to qmldir

---
 declarativeimports/qml/qmldir | 1 +
 1 file changed, 1 insertion(+)

diff --git a/declarativeimports/qml/qmldir b/declarativeimports/qml/qmldir
index 79aae321..bfc17a4e 100644
--- a/declarativeimports/qml/qmldir
+++ b/declarativeimports/qml/qmldir
@@ -1,3 +1,4 @@
+module org.kde.games.core
 plugin corebindingsplugin
 
 KgItem 0.1 KgItem.qml
-- 
2.25.1

