--- mozc-2.18.2612.102/rewriter/date_rewriter.cc.org	2019-04-09 00:24:39.974142422 +0900
+++ mozc-2.18.2612.102/rewriter/date_rewriter.cc	2019-04-09 00:36:30.557247652 +0900
@@ -1501,6 +1501,11 @@
     1989,
     "\xE5\xB9\xB3\xE6\x88\x90",
     "\xE3\x81\xB8\xE3\x81\x84\xE3\x81\x9B\xE3\x81\x84",
+  }, {
+    // "令和" "れいわ"
+    2019,
+    "\xE4\xBB\xA4\xE5\x92\x8C",
+    "\xE3\x82\x8C\xE3\x81\x84\xE3\x82\x8F",
   }
 };
 
