--- a/amavisd.conf	2021-02-25 10:30:28.234737601 +0100
+++ b/amavisd.conf	2021-02-25 10:31:25.883474087 +0100
@@ -44,7 +44,7 @@ $do_syslog = 1;              # log via s
 $syslog_facility = 'mail';   # Syslog facility as a string
            # e.g.: mail, daemon, user, local0, ... local7
 
-$enable_db = 1;              # enable use of BerkeleyDB/libdb (SNMP and nanny)
+# $enable_db = 1;            # enable use of BerkeleyDB/libdb (SNMP and nanny)
 # $enable_zmq = 1;           # enable use of ZeroMQ (SNMP and nanny)
 $nanny_details_level = 2;    # nanny verbosity: 1: traditional, 2: detailed
 $enable_dkim_verification = 1;  # enable DKIM signatures verification
