--- Makefile.in
+++ Makefile.in
@@ -17,7 +17,7 @@
 #  WARRANTIES, INCLUDING, WITHOUT LIMITATION, THE IMPLIED WARRANTIES OF
 #  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
 #
-# @(#) $Id: Makefile.in,v 1.68 2000/06/15 00:39:54 leres Exp $ (LBL)
+# @(#) $Id: Makefile.in,v 1.1 2001/01/11 00:41:05 root Exp root $ (LBL)
 
 #
 # Various configurable paths (remember to edit Makefile.in, not Makefile)
@@ -110,13 +110,16 @@
 	$(CC) $(CFLAGS) -o $@ zap.o intoa.o -lutil
 
 install: force
-	$(INSTALL) -m 555 -o bin -g bin arpwatch $(DESTDIR)$(BINDEST)
-	$(INSTALL) -m 555 -o bin -g bin arpsnmp $(DESTDIR)$(BINDEST)
+	$(INSTALL) -m 555 arpwatch $(DESTDIR)$(BINDEST)
+	$(INSTALL) -m 555 arpsnmp $(DESTDIR)$(BINDEST)
+	$(INSTALL) -d -m 755 $(DESTDIR)/var/lib/arpwatch
+	$(INSTALL) -d -m 755 $(DESTDIR)/usr/share/arpwatch
+	$(INSTALL) -m 444 ethercodes.dat $(DESTDIR)/usr/share/arpwatch
 
 install-man: force
-	$(INSTALL) -m 444 -o bin -g bin $(srcdir)/arpwatch.8 \
+	$(INSTALL) -m 444 $(srcdir)/arpwatch.8 \
 	    $(DESTDIR)$(MANDEST)/man8
-	$(INSTALL) -m 444 -o bin -g bin $(srcdir)/arpsnmp.8 \
+	$(INSTALL) -m 444 $(srcdir)/arpsnmp.8 \
 	    $(DESTDIR)$(MANDEST)/man8
 
 lint:	$(GENSRC) force
--- arpwatch.h
+++ arpwatch.h
@@ -1,7 +1,7 @@
-/* @(#) $Id: arpwatch.h,v 1.29 2000/09/30 23:40:49 leres Exp $ (LBL) */
+/* @(#) $Id: arpwatch.h,v 1.1 2001/01/11 01:08:54 root Exp root $ (LBL) */
 
-#define ARPFILE "arp.dat"
-#define ETHERCODES "ethercodes.dat"
+#define ARPFILE "/var/lib/arpwatch/arp.dat"
+#define ETHERCODES "/usr/share/arpwatch/ethercodes.dat"
 #define CHECKPOINT (15*60)		/* Checkpoint time in seconds */
 
 #define MEMCMP(a, b, n) memcmp((char *)a, (char *)b, n)
