diff -Nur tinygettext-0.1.1469459657.bf66a57/CMakeLists.txt new/CMakeLists.txt
--- tinygettext-0.1.1469459657.bf66a57/CMakeLists.txt	2016-08-02 02:17:46.000000000 +0200
+++ new/CMakeLists.txt	2016-08-02 02:54:17.048663566 +0200
@@ -36,6 +36,7 @@
 
 project(tinygettext)
 set(VERSION "0.1")
+set(VERSION_MAJOR 0)
 
 ### CMake configuration
 
@@ -97,6 +98,10 @@
 ## define a target for building the library
 
 add_library(tinygettext ${TINYGETTEXT_SOURCES})
+set_target_properties(tinygettext PROPERTIES
+  VERSION ${VERSION}
+  SOVERSION ${VERSION_MAJOR}
+)
 
 ## Add tinygettext dir to search path
 
