diff -Nur dde-file-manager-5.2.18/3rdparty/doctotext/data_stream.cpp dde-file-manager-5.2.18-new/3rdparty/doctotext/data_stream.cpp
--- dde-file-manager-5.2.18/3rdparty/doctotext/data_stream.cpp	2021-04-27 16:51:31.000000000 +0800
+++ dde-file-manager-5.2.18-new/3rdparty/doctotext/data_stream.cpp	2021-05-03 16:37:16.743190858 +0800
@@ -262,6 +262,7 @@
     if (impl->m_pointer < 1)
         return false;
     --impl->m_pointer;
+    return true;
 }
 
 size_t BufferStream::size()
diff -Nur dde-file-manager-5.2.18/3rdparty/doctotext/doctotext_c_api.cpp dde-file-manager-5.2.18-new/3rdparty/doctotext/doctotext_c_api.cpp
--- dde-file-manager-5.2.18/3rdparty/doctotext/doctotext_c_api.cpp	2021-04-27 16:51:31.000000000 +0800
+++ dde-file-manager-5.2.18-new/3rdparty/doctotext/doctotext_c_api.cpp	2021-05-03 16:38:05.067016201 +0800
@@ -248,6 +248,7 @@
 size_t DOCTOTEXT_CALL doctotext_exception_error_messages_count(DocToTextException* exception)
 {
 	exception->m_exception.getErrorCount();
+	return 0;
 }
 
 const char* DOCTOTEXT_CALL doctotext_exception_get_error_message(DocToTextException* exception, size_t index_message)
diff -Nur dde-file-manager-5.2.18/3rdparty/doctotext/eml_parser.cpp dde-file-manager-5.2.18-new/3rdparty/doctotext/eml_parser.cpp
--- dde-file-manager-5.2.18/3rdparty/doctotext/eml_parser.cpp	2021-04-27 16:51:31.000000000 +0800
+++ dde-file-manager-5.2.18-new/3rdparty/doctotext/eml_parser.cpp	2021-05-03 16:40:04.054586143 +0800
@@ -380,6 +380,7 @@
 	std::string sender = header.sender().str();
 	if (!sender.empty())
 		metadata.addField("Sender", sender);
+	return metadata;
 }
 
 bool EMLParser::error()
diff -Nur dde-file-manager-5.2.18/3rdparty/doctotext/iwork_parser.cpp dde-file-manager-5.2.18-new/3rdparty/doctotext/iwork_parser.cpp
--- dde-file-manager-5.2.18/3rdparty/doctotext/iwork_parser.cpp	2021-04-27 16:51:31.000000000 +0800
+++ dde-file-manager-5.2.18-new/3rdparty/doctotext/iwork_parser.cpp	2021-05-03 16:41:02.362375401 +0800
@@ -898,6 +898,7 @@
 				m_text_body += (*it)->m_text;
 				++it;
 			}
+		return true;
 		}
 
 		//Functions responsible for parsing xml elements:
