[gdb/testsuite] Handle quotes in gdb_py_module_available

On openSUSE Leap 42.3 with python 3.4, I run into:
...
(gdb) python import pygments^M
Traceback (most recent call last):^M
  File "<string>", line 1, in <module>^M
ImportError: No module named 'pygments'^M
Error while executing Python code.^M
(gdb) FAIL: gdb.base/style.exp: python import pygments
ERROR: unexpected output from python import
...
because gdb_py_module_available doesn't handle the single quotes around the
module name in the ImportError.

Fix this by allowing the single quotes.

Tested on x86_64-linux.

---
 gdb/testsuite/lib/gdb-python.exp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdb/testsuite/lib/gdb-python.exp b/gdb/testsuite/lib/gdb-python.exp
index d5039bbd099..c7a3907dd2b 100644
--- a/gdb/testsuite/lib/gdb-python.exp
+++ b/gdb/testsuite/lib/gdb-python.exp
@@ -61,7 +61,7 @@ proc gdb_py_module_available { name } {
 	-re -wrap "ModuleNotFoundError: No module named '${name}'.*" {
 	    set available false
 	}
-	-re -wrap "ImportError: No module named ${name}.*" {
+	-re -wrap "ImportError: No module named '?${name}.*'?" {
 	    set available false
 	}
 	-re -wrap "python import ${name}" {
