diff -Nur luabind-0.9.1+20150408/src/CMakeLists.txt new/src/CMakeLists.txt
--- luabind-0.9.1+20150408/src/CMakeLists.txt	2017-04-04 20:14:43.335563532 +0200
+++ new/src/CMakeLists.txt	2017-04-04 20:24:19.791047443 +0200
@@ -137,19 +137,24 @@
 set(ALLHEADERS ${LUABIND_API} ${LUABIND_DETAIL_API} PARENT_SCOPE)
 set(APIHEADERS ${LUABIND_API} PARENT_SCOPE)
 
+if(NOT INCLUDE_DIR)
+        set(INCLUDE_DIR include)
+endif()
+if(NOT LIB_DIR)
+        set(LIB_DIR lib)
+endif()
+configure_file(luabind.pc.in luabind.pc @ONLY)
+
 if(INSTALL_LUABIND)
-	if(NOT INCLUDE_DIR)
-		set(INCLUDE_DIR include)
-	endif()
-	if(NOT LIB_DIR)
-		set(LIB_DIR lib)
-	endif()
 	if(NOT ARCH_DIR)
 		set(ARCH_DIR lib)
 	endif()
 	if(NOT BIN_DIR)
 		set(BIN_DIR bin)
 	endif()
+	install(FILES "${CMAKE_CURRENT_BINARY_DIR}/luabind.pc"
+		DESTINATION ${LIB_DIR}/pkgconfig
+		COMPONENT sdk)
 	install(FILES ${LUABIND_API}
 		DESTINATION ${INCLUDE_DIR}/luabind
 		COMPONENT sdk)
diff -Nur luabind-0.9.1+20150408/src/luabind.pc.in new/src/luabind.pc.in
--- luabind-0.9.1+20150408/src/luabind.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ new/src/luabind.pc.in	2017-04-04 20:20:18.579878429 +0200
@@ -0,0 +1,10 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+includedir=${prefix}/@INCLUDE_DIR@
+libdir=${prefix}/@LIB_DIR@
+
+Name: luabind
+Description: Luabind is a library that helps you create bindings between C++ and Lua.
+Version: @CPACK_PACKAGE_VERSION@
+Cflags: -I${includedir}
+Libs: -L${libdir} -lluabind
