diff -Nur mapserver-7.6.3/mapscript/python/CMakeLists.txt new/mapscript/python/CMakeLists.txt
--- mapserver-7.6.3/mapscript/python/CMakeLists.txt	2021-04-30 23:26:25.000000000 +0200
+++ new/mapscript/python/CMakeLists.txt	2021-05-31 20:53:16.269084059 +0200
@@ -146,7 +146,7 @@
     endif()
 
     execute_process(
-      COMMAND ${PYTHON_EXECUTABLE} setup.py install \${PYTHON_ROOT} \${PYTHON_PREFIX}
+      COMMAND ${PYTHON_EXECUTABLE} setup.py install --install-lib=${PYTHON_SITE_PACKAGES} \${PYTHON_ROOT} \${PYTHON_PREFIX}
       WORKING_DIRECTORY ${OUTPUT_FOLDER}
     )
   "
