Index: tigervnc-1.12.0/common/rfb/Security.cxx
===================================================================
--- tigervnc-1.12.0.orig/common/rfb/Security.cxx
+++ tigervnc-1.12.0/common/rfb/Security.cxx
@@ -52,7 +52,7 @@ static LogWriter vlog("Security");
 #ifdef HAVE_GNUTLS
 StringParameter Security::GnuTLSPriority("GnuTLSPriority",
   "GnuTLS priority string that controls the TLS session’s handshake algorithms",
-  "");
+  "@SYSTEM");
 #endif
 
 Security::Security()
