diff -Nru oci-cli-2.6.10.orig/src/oci_cli/cli_root.py oci-cli-2.6.10/src/oci_cli/cli_root.py
--- oci-cli-2.6.10.orig/src/oci_cli/cli_root.py	2019-10-29 21:42:32.000000000 +0100
+++ oci-cli-2.6.10/src/oci_cli/cli_root.py	2019-11-05 13:48:54.066905418 +0100
@@ -219,7 +219,7 @@
 @click.option('--profile',
               default=Sentinel(DEFAULT_PROFILE), show_default=False,
               help='The profile in the config file to load. This profile will also be used to locate any default parameter values which have been specified in the OCI CLI-specific configuration file.  [default: DEFAULT]')
-@click.option('--cli-rc-file', '--defaults-file',
+@click.option('--cli-rc-file', '--defaults-file', 'defaults_file',
               default=cli_constants.CLI_RC_DEFAULT_LOCATION, show_default=True,
               is_eager=True, callback=eager_load_cli_rc_file,
               help='The path to the OCI CLI-specific configuration file, containing parameter default values and other configuration information such as command aliases and predefined queries. The --defaults-file option is deprecated and you should use the --cli-rc-file option instead.')
