--- nls/compose.dir.pre.orig	2007-10-26 10:38:23.854146750 +0200
+++ nls/compose.dir.pre	2007-10-26 10:38:51.159853250 +0200
@@ -141,6 +141,7 @@
 georgian-ps/Compose:		ka_GE.GEORGIAN-PS
 iso8859-1/Compose:		kl_GL.ISO8859-1
 iso8859-15/Compose:		kl_GL.ISO8859-15
+en_US.UTF-8/Compose:		km_KH.UTF-8
 iso8859-1/Compose:		kw_GB.ISO8859-1
 iso8859-14/Compose:		kw_GB.ISO8859-14
 iso8859-15/Compose:		kw_GB.ISO8859-15
