--- a/Nagios-Plugin-0.36/t/Nagios-Plugin-Functions-01.t	2010-12-03 12:46:21.000000000 +0100
+++ b/Nagios-Plugin-0.36/t/Nagios-Plugin-Functions-01.t	2015-08-27 12:16:13.649140550 +0200
@@ -19,7 +19,7 @@
 is(get_shortname, "NAGIOS-PLUGIN-FUNCTIONS-01", "get_shortname ok");
 
 # Hardcoded checks of constants
-ok(defined %ERRORS, '%ERRORS defined');
+ok(%ERRORS, '%ERRORS defined');
 is(OK,          $ERRORS{OK},            "OK        => $ERRORS{OK}");
 is(WARNING,     $ERRORS{WARNING},       "WARNING   => $ERRORS{WARNING}");
 is(CRITICAL,    $ERRORS{CRITICAL},      "CRITICAL  => $ERRORS{CRITICAL}");
