Index: pyginac-1.5.5/Makefile
===================================================================
--- pyginac-1.5.5.orig/Makefile
+++ pyginac-1.5.5/Makefile
@@ -3,11 +3,12 @@
 # (add it to, or add to it, whatever you like)
 #
 # Common prefix for installation directories.
-prefix = /usr/local
+prefix = /usr
 includedir =  $(prefix)/include
-libdir = $(prefix)/lib/python3/dist-packages
+libdir = $(prefix)/lib/python3/site-packages
 datarootdir = $(prefix)/share
-docdir = $(datarootdir)/doc/python3-pyginac
+docdir = $(datarootdir)/doc/packages/python3-pyginac
+
 #======================================================================
 ifeq ($(DISTRO_NAME),stretch)
 BOOST_PYTHON_LIB=boost_python-py35
