diff --git a/zeroconf/test.py b/zeroconf/test.py
index f0d5ad4..26e6216 100644
--- a/zeroconf/test.py
+++ b/zeroconf/test.py
@@ -435,6 +435,7 @@ class Names(unittest.TestCase):
 
 
 class Framework(unittest.TestCase):
+    @unittest.skip("Does not work in an OBS chroot")
     def test_launch_and_close(self):
         rv = r.Zeroconf(interfaces=r.InterfaceChoice.All)
         rv.close()
@@ -443,6 +444,7 @@ class Framework(unittest.TestCase):
 
     @unittest.skipIf(not socket.has_ipv6, 'Requires IPv6')
     @attr('IPv6')
+    @unittest.skip("Does not work in an OBS chroot")
     def test_launch_and_close_v4_v6(self):
         rv = r.Zeroconf(interfaces=r.InterfaceChoice.All, ip_version=r.IPVersion.All)
         rv.close()
@@ -451,6 +453,7 @@ class Framework(unittest.TestCase):
 
     @unittest.skipIf(not socket.has_ipv6, 'Requires IPv6')
     @attr('IPv6')
+    @unittest.skip("Does not work in an OBS chroot")
     def test_launch_and_close_v6_only(self):
         rv = r.Zeroconf(interfaces=r.InterfaceChoice.All, ip_version=r.IPVersion.V6Only)
         rv.close()
@@ -770,6 +773,7 @@ class TestDNSCache(unittest.TestCase):
 
 
 class ServiceTypesQuery(unittest.TestCase):
+    @unittest.skip("Does not work in an OBS chroot")
     def test_integration_with_listener(self):
 
         type_ = "_test-srvc-type._tcp.local."
@@ -818,6 +822,7 @@ class ServiceTypesQuery(unittest.TestCase):
 
     @unittest.skipIf(not socket.has_ipv6, 'Requires IPv6')
     @attr('IPv6')
+    @unittest.skip("Does not work in an OBS chroot")
     def test_integration_with_listener_ipv6(self):
 
         type_ = "_test-srvc-type._tcp.local."
