--- a/docs/conf.py
+++ b/docs/conf.py
@@ -14,7 +14,10 @@ sys.path.insert(0, root_path)
 # Mock some expensive/platform-specific modules so build will work.
 # (https://read-the-docs.readthedocs.io/en/latest/faq.html#\
 #  i-get-import-errors-on-libraries-that-depend-on-c-modules)
-import mock
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
 
 
 class MockModule(mock.Mock):
@@ -77,7 +80,7 @@ html_logo = "images/banner.svg"
 
 html_theme_options = {
     "announcement": """
-        <a style=\"text-decoration: none; color: white;\" 
+        <a style=\"text-decoration: none; color: white;\"
            href=\"https://opencollective.com/urllib3\">
            <img src=\"/en/latest/_static/favicon.png\"/> Sponsor urllib3 v2.0 on Open Collective
         </a>
--- a/test/appengine/test_urlfetch.py
+++ b/test/appengine/test_urlfetch.py
@@ -5,7 +5,10 @@ Engine-patched version of httplib to mak
 import httplib
 import pytest
 import StringIO
-from mock import patch
+try:
+    from unittest.mock import patch
+except ImportError:
+    from mock import patch
 
 from ..test_no_ssl import TestWithoutSSL
 
--- a/test/contrib/test_pyopenssl.py
+++ b/test/contrib/test_pyopenssl.py
@@ -1,7 +1,10 @@
 # -*- coding: utf-8 -*-
 import os
 
-import mock
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
 import pytest
 
 try:
--- a/test/contrib/test_pyopenssl_dependencies.py
+++ b/test/contrib/test_pyopenssl_dependencies.py
@@ -1,6 +1,9 @@
 # -*- coding: utf-8 -*-
 import pytest
-from mock import Mock, patch
+try:
+    from unittest.mock import Mock, patch
+except ImportError:
+    from mock import Mock, patch
 
 try:
     from urllib3.contrib.pyopenssl import extract_from_urllib3, inject_into_urllib3
--- a/test/test_connection.py
+++ b/test/test_connection.py
@@ -1,6 +1,9 @@
 import datetime
 
-import mock
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
 import pytest
 
 from urllib3.connection import RECENT_DATE, CertificateError, _match_hostname
--- a/test/test_connectionpool.py
+++ b/test/test_connectionpool.py
@@ -6,7 +6,10 @@ from ssl import SSLError as BaseSSLError
 from test import SHORT_TIMEOUT
 
 import pytest
-from mock import Mock
+try:
+    from unittest.mock import Mock
+except ImportError:
+    from mock import Mock
 
 from dummyserver.server import DEFAULT_CA
 from urllib3._collections import HTTPHeaderDict
--- a/test/test_queue_monkeypatch.py
+++ b/test/test_queue_monkeypatch.py
@@ -1,6 +1,9 @@
 from __future__ import absolute_import
 
-import mock
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
 import pytest
 
 from urllib3 import HTTPConnectionPool
--- a/test/test_response.py
+++ b/test/test_response.py
@@ -9,7 +9,10 @@ from base64 import b64decode
 from io import BufferedReader, BytesIO, TextIOWrapper
 from test import onlyBrotlipy
 
-import mock
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
 import pytest
 import six
 
--- a/test/test_retry.py
+++ b/test/test_retry.py
@@ -1,6 +1,9 @@
 import warnings
 
-import mock
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
 import pytest
 
 from urllib3.exceptions import (
--- a/test/test_retry_deprecated.py
+++ b/test/test_retry_deprecated.py
@@ -1,7 +1,10 @@
 # This is a copy-paste of test_retry.py with extra asserts about deprecated options. It will be removed for v2.
 import warnings
 
-import mock
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
 import pytest
 
 from urllib3.exceptions import (
--- a/test/test_ssl.py
+++ b/test/test_ssl.py
@@ -1,6 +1,9 @@
 from test import notPyPy2
 
-import mock
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
 import pytest
 
 from urllib3.exceptions import SNIMissingWarning
--- a/test/test_ssltransport.py
+++ b/test/test_ssltransport.py
@@ -4,7 +4,10 @@ import socket
 import ssl
 import sys
 
-import mock
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
 import pytest
 
 from dummyserver.server import DEFAULT_CA, DEFAULT_CERTS
--- a/test/test_util.py
+++ b/test/test_util.py
@@ -9,7 +9,10 @@ from itertools import chain
 from test import notBrotlipy, onlyBrotlipy, onlyPy2, onlyPy3
 
 import pytest
-from mock import Mock, patch
+try:
+    from unittest.mock import Mock, patch
+except ImportError:
+    from mock import Mock, patch
 
 from urllib3 import add_stderr_logger, disable_warnings, util
 from urllib3.exceptions import (
--- a/test/with_dummyserver/test_connectionpool.py
+++ b/test/with_dummyserver/test_connectionpool.py
@@ -10,7 +10,10 @@ import warnings
 from test import LONG_TIMEOUT, SHORT_TIMEOUT, onlyPy2
 from threading import Event
 
-import mock
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
 import pytest
 import six
 
--- a/test/with_dummyserver/test_https.py
+++ b/test/with_dummyserver/test_https.py
@@ -19,7 +19,10 @@ from test import (
     resolvesLocalhostFQDN,
 )
 
-import mock
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
 import pytest
 import trustme
 
--- a/test/with_dummyserver/test_socketlevel.py
+++ b/test/with_dummyserver/test_socketlevel.py
@@ -52,7 +52,10 @@ from test import (
 )
 from threading import Event
 
-import mock
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
 import pytest
 import trustme
 
