diff -ru splint-3.1.1-orig/src/flags.c splint-3.1.1/src/flags.c
--- splint-3.1.1-orig/src/flags.c	2004-04-19 16:26:12.947597050 +0200
+++ splint-3.1.1/src/flags.c	2004-04-19 16:40:53.549668433 +0200
@@ -148,6 +148,7 @@
     case ARG_SPECIAL:
       BADBRANCH;
     }
+  return NULL;
 }      
 
 # ifdef WIN32
diff -ru splint-3.1.1-orig/src/stateInfo.c splint-3.1.1/src/stateInfo.c
--- splint-3.1.1-orig/src/stateInfo.c	2004-04-19 16:26:12.957595168 +0200
+++ splint-3.1.1/src/stateInfo.c	2004-04-19 16:41:44.090143739 +0200
@@ -313,6 +313,7 @@
     case NS_ABSNULL:
       return SA_BECOMESPOSSIBLYNULL;
     }
+  return SA_UNKNOWN;
 }
 
 stateAction stateAction_fromExkind (exkind ex)
@@ -405,6 +406,7 @@
     case SS_KILLED:     
     case SS_UNDEFKILLED:
     case SS_LAST:
+    default:
       llbug (message ("Unexpected sstate: %s", sstate_unparse (ss)));
       /*@notreached@*/ return SA_UNKNOWN;
     }
@@ -459,6 +461,7 @@
  
   DPRINTF (("Bad state action: %d", sa));
   BADBRANCH;
+  return NULL;
 }
 
 void stateInfo_display (stateInfo s, cstring sname)
diff -ru splint-3.1.1-orig/src/uentry.c splint-3.1.1/src/uentry.c
--- splint-3.1.1-orig/src/uentry.c	2004-04-19 16:26:12.897606463 +0200
+++ splint-3.1.1/src/uentry.c	2004-04-19 16:39:10.887015966 +0200
@@ -5804,6 +5804,7 @@
     {
       BADBRANCH;
     }
+  return sRefSet_undefined;
 }
 
 ekind
