diff --git a/examples/systemd/vnstat.service b/examples/systemd/vnstat.service
index 104082e..af29c93 100644
--- a/examples/systemd/vnstat.service
+++ b/examples/systemd/vnstat.service
@@ -11,20 +11,6 @@ ExecReload=/bin/kill -HUP $MAINPID
 Restart=on-failure
 RestartSec=2
 
-# Hardening
-ProtectSystem=strict
-ReadWritePaths=/var/lib
-StateDirectory=vnstat
-PrivateDevices=yes
-ProtectKernelTunables=yes
-ProtectControlGroups=yes
-ProtectHome=yes
-ProtectKernelModules=yes
-PrivateTmp=yes
-MemoryDenyWriteExecute=yes
-RestrictRealtime=yes
-RestrictNamespaces=yes
-
 [Install]
 WantedBy=multi-user.target
 Alias=vnstatd.service
