diff -ur a/build/cmake/FindQt5.cmake b/build/cmake/FindQt5.cmake
--- a/build/cmake/FindQt5.cmake	2022-12-13 22:58:31.000000000 +0100
+++ b/build/cmake/FindQt5.cmake	2022-12-14 20:37:54.029305951 +0100
@@ -2,7 +2,7 @@
 include(GetUtilsFunctions) # library of CMake functions ("fn__" namespace)
 
 # Print Qt version or fail the build if Qt (qmake) is not in PATH.
-fn__require_program(QMAKE Qt --version "https://musescore.org/en/handbook/developers-handbook/compilation" qmake)
+fn__require_program(QMAKE Qt --version "https://musescore.org/en/handbook/developers-handbook/compilation" qmake-qt5)
 
 set(CMAKE_AUTOUIC ON)
 set(CMAKE_AUTOMOC ON)
@@ -52,7 +52,7 @@
 
 include_directories(${QT_INCLUDES})
 
-find_program(QT_QMAKE_EXECUTABLE qmake)
+find_program(QT_QMAKE_EXECUTABLE qmake-qt5)
 set(_qmake_vars
     QT_INSTALL_ARCHDATA
     QT_INSTALL_BINS
