Index: vlc-3.0.12/modules/demux/adaptive/playlist/SegmentInformation.cpp
===================================================================
--- vlc-3.0.12.orig/modules/demux/adaptive/playlist/SegmentInformation.cpp
+++ vlc-3.0.12/modules/demux/adaptive/playlist/SegmentInformation.cpp
@@ -34,6 +34,7 @@
 
 #include <algorithm>
 #include <cassert>
+#include <limits>
 
 using namespace adaptive::playlist;
 
Index: vlc-3.0.12/modules/demux/hls/playlist/Parser.cpp
===================================================================
--- vlc-3.0.12.orig/modules/demux/hls/playlist/Parser.cpp
+++ vlc-3.0.12/modules/demux/hls/playlist/Parser.cpp
@@ -42,6 +42,7 @@
 #include <map>
 #include <cctype>
 #include <algorithm>
+#include <limits>
 
 using namespace adaptive;
 using namespace adaptive::playlist;
Index: vlc-3.0.12/modules/demux/mkv/matroska_segment.cpp
===================================================================
--- vlc-3.0.12.orig/modules/demux/mkv/matroska_segment.cpp
+++ vlc-3.0.12/modules/demux/mkv/matroska_segment.cpp
@@ -31,6 +31,7 @@
 
 #include <new>
 #include <iterator>
+#include <limits>
 
 matroska_segment_c::matroska_segment_c( demux_sys_t & demuxer, EbmlStream & estream, KaxSegment *p_seg )
     :segment(p_seg)
Index: vlc-3.0.12/modules/demux/mkv/mkv.cpp
===================================================================
--- vlc-3.0.12.orig/modules/demux/mkv/mkv.cpp
+++ vlc-3.0.12/modules/demux/mkv/mkv.cpp
@@ -34,6 +34,7 @@
 #include "stream_io_callback.hpp"
 
 #include <new>
+#include <limits>
 
 extern "C" {
     #include "../av1_unpack.h"
