Index: zhu3d.pri
===================================================================
--- zhu3d.pri.orig
+++ zhu3d.pri
@@ -23,11 +23,10 @@
 # application path. As default the texture-directory resides under the work-
 # directory.
 
-SYSDIR=
-DOCDIR=
-WORKDIR=
-TEXDIR=
-
+SYSDIR=/usr/share/zhu3d/system
+DOCDIR=/usr/share/doc/packages/zhu3d
+WORKDIR=/usr/share/zhu3d/work
+TEXDIR=/usr/share/zhu3d/work/textures
 
 # DETERMINING A DEFAULT GUI-STYLE
 #
