Index: py-1.10.0/testing/path/test_local.py
===================================================================
--- py-1.10.0.orig/testing/path/test_local.py
+++ py-1.10.0/testing/path/test_local.py
@@ -338,11 +338,11 @@ class TestLocalPath(common.CommonFSTests
         l2 = local(l)
         assert l2 == l
 
-        wc = py.path.svnwc('.')
-        l3 = local(wc)
-        assert l3 is not wc
-        assert l3.strpath == wc.strpath
-        assert not hasattr(l3, 'commit')
+        #wc = py.path.svnwc('.')
+        #l3 = local(wc)
+        #assert l3 is not wc
+        #assert l3.strpath == wc.strpath
+        #assert not hasattr(l3, 'commit')
 
     @py.test.mark.xfail(run=False, reason="unreliable est for long filenames")
     def test_long_filenames(self, tmpdir):
Index: py-1.10.0/py/__init__.py
===================================================================
--- py-1.10.0.orig/py/__init__.py
+++ py-1.10.0/py/__init__.py
@@ -57,10 +57,7 @@ apipkg.initpkg(__name__, attr={'_apipkg'
 
     'path' : {
         '__doc__'        : '._path:__doc__',
-        'svnwc'          : '._path.svnwc:SvnWCCommandPath',
-        'svnurl'         : '._path.svnurl:SvnCommandPath',
         'local'          : '._path.local:LocalPath',
-        'SvnAuth'        : '._path.svnwc:SvnAuth',
     },
 
     # python inspection/code-generation API
