diff -wruN -x '*~' -x '*.o' -x '*.a' -x '*.so' -x '*.so.[0-9]' -x autom4te.cache -x .deps -x .libs ../orig-python-lmiwbem-0.7.2/configure.ac ./configure.ac
--- ../orig-python-lmiwbem-0.7.2/configure.ac	2015-06-29 09:01:54.000000000 +0200
+++ ./configure.ac	2016-01-11 13:40:34.433720100 +0100
@@ -29,7 +29,7 @@
 case $host_os in
     linux*)
         case `uname -p` in
-            i*86)
+            athlon | i*86)
                 PEGASUS_PLATFORM=PEGASUS_PLATFORM_LINUX_IX86_GNU
                 ;;
             x86_64)
