 pdfTeX 3.14159265-2.6-1.40.18 (TeX Live 2017)  (April 27, 2017)
diff --git a/texk/web2c/pdftexdir/utils.c b/texk/web2c/pdftexdir/utils.c
index c93a87819..7f5e50416 100644
--- a/texk/web2c/pdftexdir/utils.c
+++ b/texk/web2c/pdftexdir/utils.c
@@ -33,7 +33,7 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "ptexlib.h"
 #include <png.h>
 #ifdef POPPLER_VERSION
-#include <poppler-config.h>
+/* POPPLER_VERSION should be a proper version string */
 #define xpdfVersion POPPLER_VERSION
 #define xpdfString "poppler"
 #else

diff --git a/texk/web2c/xetexdir/XeTeX_ext.c b/texk/web2c/xetexdir/XeTeX_ext.c
index 4968ee41e..0aee4ee38 100644
--- a/texk/web2c/xetexdir/XeTeX_ext.c
+++ b/texk/web2c/xetexdir/XeTeX_ext.c
@@ -38,7 +38,10 @@ authorization from the copyright holders.
 
 #include <w2c/config.h>
 
+#ifndef POPPLER_VERSION
 #include <poppler-config.h>
+#endif
+
 #include <png.h>
 #include <zlib.h>
 #include <graphite2/Font.h>

