Index: mpi-selector.csh.in
===================================================================
--- mpi-selector.csh.in.orig	2009-08-02 09:58:22.000000000 +0200
+++ mpi-selector.csh.in	2010-12-10 16:44:40.618387677 +0100
@@ -1,4 +1,3 @@
-#!/bin/csh
 # Copyright (c) 2007 Cisco Systems, Inc.  All rights reserved.
 #
 # File installed for Bourne-shell startups to select which MPI
Index: mpi-selector.sh.in
===================================================================
--- mpi-selector.sh.in.orig	2009-08-02 09:58:22.000000000 +0200
+++ mpi-selector.sh.in	2010-12-10 16:44:44.690098294 +0100
@@ -1,4 +1,3 @@
-#!/bin/sh
 # Copyright (c) 2007 Cisco Systems, Inc.  All rights reserved.
 #
 # File installed for Bourne-shell startups to select which MPI
