Index: fedora-crypto-policies-20230920.570ea89/python/policygenerators/__init__.py
===================================================================
--- fedora-crypto-policies-20230920.570ea89.orig/python/policygenerators/__init__.py
+++ fedora-crypto-policies-20230920.570ea89/python/policygenerators/__init__.py
@@ -8,7 +8,7 @@ from .gnutls import GnuTLSGenerator
 from .java import JavaGenerator
 from .java import JavaSystemGenerator
 from .krb5 import KRB5Generator
-from .libreswan import LibreswanGenerator
+# from .libreswan import LibreswanGenerator
 from .libssh import LibsshGenerator
 from .nss import NSSGenerator
 from .openssh import OpenSSHClientGenerator
@@ -16,8 +16,8 @@ from .openssh import OpenSSHServerGenera
 from .openssl import OpenSSLConfigGenerator
 from .openssl import OpenSSLGenerator
 from .openssl import OpenSSLFIPSGenerator
-from .sequoia import SequoiaGenerator
-from .sequoia import RPMSequoiaGenerator
+# from .sequoia import SequoiaGenerator
+# from .sequoia import RPMSequoiaGenerator
 
 __all__ = [
     'BindGenerator',
@@ -25,7 +25,6 @@ __all__ = [
     'JavaGenerator',
     'JavaSystemGenerator',
     'KRB5Generator',
-    'LibreswanGenerator',
     'LibsshGenerator',
     'NSSGenerator',
     'OpenSSHClientGenerator',
@@ -33,6 +32,8 @@ __all__ = [
     'OpenSSLConfigGenerator',
     'OpenSSLGenerator',
     'OpenSSLFIPSGenerator',
-    'SequoiaGenerator',
-    'RPMSequoiaGenerator',
 ]
+
+#   'LibreswanGenerator',
+#   'SequoiaGenerator',
+#   'RPMSequoiaGenerator',
