Index: fedora-crypto-policies-20250124.4d262e7/python/policygenerators/openssl.py
===================================================================
--- fedora-crypto-policies-20250124.4d262e7.orig/python/policygenerators/openssl.py
+++ fedora-crypto-policies-20250124.4d262e7/python/policygenerators/openssl.py
@@ -312,8 +312,8 @@ class OpenSSLConfigGenerator(OpenSSLGene
         'SECP256R1': 'secp256r1',
         'SECP384R1': 'secp384r1',
         'SECP521R1': 'secp521r1',
-        'X25519': 'X25519',
-        'X448': 'X448',
+        'X25519': '?X25519',
+        'X448': '?X448',
         'FFDHE-2048': 'ffdhe2048',
         'FFDHE-3072': 'ffdhe3072',
         'FFDHE-4096': 'ffdhe4096',
