diff -Nurp deepin-pw-check-5.1.16/Makefile deepin-pw-check-5.1.16-new/Makefile
--- deepin-pw-check-5.1.16/Makefile	2022-08-15 11:11:11.000000000 +0800
+++ deepin-pw-check-5.1.16-new/Makefile	2022-08-24 21:55:43.659287279 +0800
@@ -25,7 +25,7 @@ prepare:
 	@mkdir -p out/bin
 
 out/bin/%: prepare
-	env GOPATH="${GOPATH}" ${GOBUILD} -o $@ ${GOBUILD_OPTIONS} ${GOPKG_PREFIX}/*.go
+#	env GOPATH="${GOPATH}" ${GOBUILD} -o $@ ${GOBUILD_OPTIONS} ${GOPKG_PREFIX}/*.go
 
 out/${LIBRARIES}:
 	gcc lib/*.c ${SECURITY_BUILD_OPTIONS} -shared -DIN_CRACKLIB -W -Wall -Wl,-soname,libdeepin_pw_check.so.1 -o $@ $^ -lcrypt -lcrack -liniparser
@@ -48,13 +48,13 @@ out/${PAM_MODULE}: out/${LIBRARIES}
 build_tool: prepare
 	gcc tool/*.c -liniparser -W -Wall ${SECURITY_BUILD_OPTIONS} -o out/${TOOL_BINARAY}
 
-build: prepare $(addprefix out/bin/, ${BINARIES}) out/${LIBRARIES} static_lib out/${PAM_MODULE} build_tool ts_to_policy
+build: prepare out/${LIBRARIES} static_lib out/${PAM_MODULE} build_tool ts_to_policy
 
 install: translate
 	mkdir -pv ${DESTDIR}${PREFIX}/share/locale
 	- cp -rf out/locale/* ${DESTDIR}${PREFIX}/share/locale
-	mkdir -p ${DESTDIR}${PREFIX}/lib
-	cp -f out/lib* ${DESTDIR}${PREFIX}/lib
+	mkdir -p ${DESTDIR}${LIBDIR}
+	cp -f out/lib* ${DESTDIR}${LIBDIR}
 	mkdir -p ${DESTDIR}${PREFIX}/include
 	cp lib/deepin_pw_check.h ${DESTDIR}${PREFIX}/include/
 	mkdir -pv ${DESTDIR}/${PKG_FILE_DIR}
