Index: nss/tests/lowhash/lowhash.sh
===================================================================
--- nss.orig/tests/lowhash/lowhash.sh
+++ nss/tests/lowhash/lowhash.sh
@@ -61,11 +61,13 @@ lowhash_test()
        ! -f ${BINDIR}/lowhashtest${PROG_SUFFIX} ]; then
     echo "freebl lowhash not supported in this plaform."
   else
-    TESTS="MD5 SHA1 SHA224 SHA256 SHA384 SHA512"
+    TESTS_FIPS_0="MD5 SHA1 SHA224 SHA256 SHA384 SHA512"
+    TESTS_FIPS_1="SHA224 SHA256 SHA384 SHA512"
     OLD_MODE=`echo ${NSS_FIPS}`
     for fips_mode in 0 1; do
       echo "lowhashtest with fips mode=${fips_mode}"
       export NSS_FIPS=${fips_mode}
+      eval TESTS=\${TESTS_FIPS_${fips_mode}}
       for TEST in ${TESTS}
       do
         echo "lowhashtest ${TEST}"
