diff --git a/coreconf/Linux.mk b/coreconf/Linux.mk
index 956f0e4..b3a352a 100644
--- a/coreconf/Linux.mk
+++ b/coreconf/Linux.mk
@@ -108,11 +108,7 @@ LIBC_TAG		= _glibc
 endif
 
 ifdef BUILD_OPT
-ifeq (11,$(ALLOW_OPT_CODE_SIZE)$(OPT_CODE_SIZE))
-	OPTIMIZER = -Os
-else
-	OPTIMIZER = -O2
-endif
+	OPTIMIZER = $(OPT_FLAGS)
 ifdef MOZ_DEBUG_SYMBOLS
 	ifdef MOZ_DEBUG_FLAGS
 		OPTIMIZER += $(MOZ_DEBUG_FLAGS)
