--- a/test/test_backends.py	2023-05-21 17:38:00.000000000 +0200
+++ b/test/test_backends.py	2023-05-22 06:22:56.491195059 +0200
@@ -476,13 +476,13 @@
     with pytest.raises(KeyError, match="^'unknown'$"):
         runner.options_to_cli({"unknown": True})
 
-
-def test_backend_importables():
+@pytest.mark.parametrize('connection_type',
+                         testinfra.backend.BACKENDS.keys())
+def test_backend_importables(connection_type):
     # just check that all declared backend are importable and NAME is set
     # correctly
-    for connection_type in testinfra.backend.BACKENDS:
-        obj = testinfra.backend.get_backend_class(connection_type)
-        assert obj.get_connection_type() == connection_type
+    obj = testinfra.backend.get_backend_class(connection_type)
+    assert obj.get_connection_type() == connection_type
 
 
 @pytest.mark.testinfra_hosts("docker://rockylinux8", "ssh://rockylinux8")
