Index: libparserutils-0.2.3/Makefile
===================================================================
--- libparserutils-0.2.3.orig/Makefile
+++ libparserutils-0.2.3/Makefile
@@ -23,7 +23,7 @@ WARNFLAGS := -Wall -W -Wundef -Wpointer-
 	-Wmissing-declarations -Wnested-externs -pedantic
 # BeOS/Haiku standard library headers create warnings.
 ifneq ($(BUILD),i586-pc-haiku)
-  WARNFLAGS := $(WARNFLAGS) -Werror
+  WARNFLAGS := $(WARNFLAGS)
 endif
 
 CFLAGS := -D_BSD_SOURCE -D_DEFAULT_SOURCE \
