diff -Nru azure-mgmt-web-0.41.0.orig/setup.py azure-mgmt-web-0.41.0/setup.py
--- azure-mgmt-web-0.41.0.orig/setup.py	2019-02-15 18:14:58.000000000 +0100
+++ azure-mgmt-web-0.41.0/setup.py	2019-04-01 13:53:28.590135637 +0200
@@ -79,7 +79,7 @@
     install_requires=[
         'msrest>=0.5.0',
         'msrestazure>=0.4.32,<2.0.0',
-        'azure-common~=1.1',
+        'azure-common>=1.1',
     ],
     extras_require={
         ":python_version<'3.0'": ['azure-mgmt-nspkg'],
