From 533a1b6996c3d99cf27ae05f0cd8131dcd9bbf30 Mon Sep 17 00:00:00 2001
From: Pawel Wieczorkiewicz <pwieczorkiewicz@suse.de>
Date: Tue, 2 Jun 2015 13:33:24 +0000
Subject: [PATCH 07/11] networkd: make network.service an alias of
 systemd-networkd.service

NetworkManager and wicked does this already. This is needed by yast2
and other parts of the system.

[fixes boo#933092]
---
 units/systemd-networkd.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/units/systemd-networkd.service.in b/units/systemd-networkd.service.in
index ffa45ba049..6a5d0dea3c 100644
--- a/units/systemd-networkd.service.in
+++ b/units/systemd-networkd.service.in
@@ -53,6 +53,7 @@ User=systemd-network
 
 [Install]
 WantedBy=multi-user.target
+Alias=network.service
 Also=systemd-networkd.socket
 Alias=dbus-org.freedesktop.network1.service
 
-- 
2.26.2

