--- config.h
+++ config.h
@@ -30,3 +30,7 @@
 #endif
 
 #include <stdio.h>
+
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
--- configure.ac
+++ configure.ac
@@ -13,7 +13,7 @@
 
 dnl Checks for header files.
 AC_HEADER_STDC
-AC_CHECK_HEADERS(errno.h fcntl.h strings.h)
+AC_CHECK_HEADERS(errno.h fcntl.h strings.h unistd.h)
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_CHECK_FUNCS(strerror)
--- file.c
+++ file.c
@@ -141,7 +141,7 @@
 	t = fbuf;
 	bits = digits = words = i = 0;
 	if (compface_xbitmap) {
-		sprintf(t,"#define noname_width 48\n#define noname_height 48\nstatic char noname_bits[] = {\n ");
+		strcpy(t,"#define noname_width 48\n#define noname_height 48\nstatic char noname_bits[] = {\n ");
 		while (*t) t++;
 		digsperword = 2;
 		wordsperline = 15;
@@ -189,7 +189,7 @@
 		}
 	}
 	if (compface_xbitmap) {
-		sprintf(t, "};\n");
+		strcpy(t, "};\n");
 		while (*t) t++;
 	}
 	*(t++) = '\0';
