diff -Nur minisat-2.2.0+20130925/minisat-2.2.0+20130925/minisat/core/SolverTypes.h minisat-2.2.0+20130925-p/minisat-2.2.0+20130925/minisat/core/SolverTypes.h
--- a/minisat/core/SolverTypes.h	2013-09-25 14:16:18.000000000 +0200
+++ b/minisat/core/SolverTypes.h	2019-07-29 02:55:41.499558102 +0200
@@ -52,7 +52,7 @@
     int     x;
 
     // Use this as a constructor:
-    friend Lit mkLit(Var var, bool sign = false);
+    friend Lit mkLit(Var var, bool sign);
 
     bool operator == (Lit p) const { return x == p.x; }
     bool operator != (Lit p) const { return x != p.x; }
@@ -60,7 +60,7 @@
 };
 
 
-inline  Lit  mkLit     (Var var, bool sign) { Lit p; p.x = var + var + (int)sign; return p; }
+inline  Lit  mkLit     (Var var, bool sign = false) { Lit p; p.x = var + var + (int)sign; return p; }
 inline  Lit  operator ~(Lit p)              { Lit q; q.x = p.x ^ 1; return q; }
 inline  Lit  operator ^(Lit p, bool b)      { Lit q; q.x = p.x ^ (unsigned int)b; return q; }
 inline  bool sign      (Lit p)              { return p.x & 1; }
