From: Thomas Andrejak <thomas.andrejak@gmail.com>
Date: 2017-10-04 07:00:00 +0100
References: http://prelude-siem.org/issues/863
Upstream: submitted
Subject: Wrong FSF address

diff -ru ./src/libprelude-error/err-codes.h ./src/libprelude-error/err-codes.h
--- ./src/libprelude-error/err-codes.h	2016-09-15 08:51:51.964000884 +0200
+++ ./src/libprelude-error/err-codes.h	2016-12-02 07:37:52.786070647 +0100
@@ -15,10 +15,9 @@
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    Lesser General Public License for more details.
  
-   You should have received a copy of the GNU Lesser General Public
-   License along with libgpg-error; if not, write to the Free
-   Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
-   02111-1307, USA.  */
+   You should have received a copy of the GNU General Public License along
+   with this program; if not, write to the Free Software Foundation, Inc.,
+   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA. */
 
 
 /* The purpose of this complex string table is to produce
diff -ru ./src/libprelude-error/err-codes.h.in ./src/libprelude-error/err-codes.h.in
--- ./src/libprelude-error/err-codes.h.in	2016-09-15 08:49:10.805000884 +0200
+++ ./src/libprelude-error/err-codes.h.in	2016-12-02 07:38:08.252722176 +0100
@@ -14,10 +14,9 @@
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    Lesser General Public License for more details.
  
-   You should have received a copy of the GNU Lesser General Public
-   License along with libgpg-error; if not, write to the Free
-   Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
-   02111-1307, USA.  */
+   You should have received a copy of the GNU General Public License along
+   with this program; if not, write to the Free Software Foundation, Inc.,
+   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA. */
 
 # Everything up to the first line that starts with a number in the
 # first column is copied into the output verbatim.  Then, empty lines
diff -ru ./src/libprelude-error/errnos.in ./src/libprelude-error/errnos.in
--- ./src/libprelude-error/errnos.in	2016-09-15 08:49:10.806000884 +0200
+++ ./src/libprelude-error/errnos.in	2016-12-02 07:38:23.777372399 +0100
@@ -14,10 +14,9 @@
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    Lesser General Public License for more details.
  
-   You should have received a copy of the GNU Lesser General Public
-   License along with libgpg-error; if not, write to the Free
-   Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
-   02111-1307, USA.  */
+   You should have received a copy of the GNU General Public License along
+   with this program; if not, write to the Free Software Foundation, Inc.,
+   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA. */
 
 # Everything up to the first line that starts with a number in the
 # first column is copied into the output verbatim.  Then, empty lines
diff -ru ./src/libprelude-error/err-sources.h ./src/libprelude-error/err-sources.h
--- ./src/libprelude-error/err-sources.h	2016-09-15 08:51:51.961000884 +0200
+++ ./src/libprelude-error/err-sources.h	2016-12-02 07:38:32.605173508 +0100
@@ -15,10 +15,9 @@
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    Lesser General Public License for more details.
  
-   You should have received a copy of the GNU Lesser General Public
-   License along with libgpg-error; if not, write to the Free
-   Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
-   02111-1307, USA.  */
+   You should have received a copy of the GNU General Public License along
+   with this program; if not, write to the Free Software Foundation, Inc.,
+   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA. */
 
 
 /* The purpose of this complex string table is to produce
diff -ru ./src/libprelude-error/err-sources.h.in ./src/libprelude-error/err-sources.h.in
--- ./src/libprelude-error/err-sources.h.in	2016-09-15 08:49:10.806000884 +0200
+++ ./src/libprelude-error/err-sources.h.in	2016-12-02 07:39:14.072239231 +0100
@@ -14,10 +14,9 @@
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    Lesser General Public License for more details.
  
-   You should have received a copy of the GNU Lesser General Public
-   License along with libgpg-error; if not, write to the Free
-   Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
-   02111-1307, USA.  */
+   You should have received a copy of the GNU General Public License along
+   with this program; if not, write to the Free Software Foundation, Inc.,
+   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA. */
 
 # Everything up to the first line that starts with a number in the
 # first column is copied into the output verbatim.  Then, empty lines
diff -ru ./src/libprelude-error/mkerrcodes1.awk ./src/libprelude-error/mkerrcodes1.awk
--- ./src/libprelude-error/mkerrcodes1.awk	2016-09-15 08:49:10.806000884 +0200
+++ ./src/libprelude-error/mkerrcodes1.awk	2016-12-02 07:40:41.671265585 +0100
@@ -11,9 +11,9 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 # General Public License for more details.
 #
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
-# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 # As a special exception, g10 Code GmbH gives unlimited permission to
 # copy, distribute and modify the C source files that are the output
diff -ru ./src/libprelude-error/mkerrcodes2.awk ./src/libprelude-error/mkerrcodes2.awk
--- ./src/libprelude-error/mkerrcodes2.awk	2016-09-15 08:49:10.806000884 +0200
+++ ./src/libprelude-error/mkerrcodes2.awk	2016-12-02 07:40:57.327912836 +0100
@@ -11,9 +11,9 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 # General Public License for more details.
 #
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
-# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 # As a special exception, g10 Code GmbH gives unlimited permission to
 # copy, distribute and modify the C source files that are the output
diff -ru ./src/libprelude-error/mkerrcodes.awk ./src/libprelude-error/mkerrcodes.awk
--- ./src/libprelude-error/mkerrcodes.awk	2016-09-15 08:49:10.806000884 +0200
+++ ./src/libprelude-error/mkerrcodes.awk	2016-12-02 07:41:21.451369320 +0100
@@ -11,9 +11,9 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 # General Public License for more details.
 #
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
-# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 # As a special exception, g10 Code GmbH gives unlimited permission to
 # copy, distribute and modify the C source files that are the output
diff -ru ./src/libprelude-error/mkerrcodes.c ./src/libprelude-error/mkerrcodes.c
--- ./src/libprelude-error/mkerrcodes.c	2016-09-15 08:49:10.806000884 +0200
+++ ./src/libprelude-error/mkerrcodes.c	2016-12-02 07:42:27.668877410 +0100
@@ -13,10 +13,9 @@
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    Lesser General Public License for more details.
  
-   You should have received a copy of the GNU Lesser General Public
-   License along with libgpg-error; if not, write to the Free
-   Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
-   02111-1307, USA.  */
+   You should have received a copy of the GNU General Public License along
+   with this program; if not, write to the Free Software Foundation, Inc.,
+   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA. */
 
 /* This file must not include config.h, as that is for the host
    system, while this file will be run on the build system.  */
@@ -40,10 +39,9 @@
 "   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU\n"
 "   Lesser General Public License for more details.\n"
 "\n"
-"   You should have received a copy of the GNU Lesser General Public\n"
-"   License along with libgpg-error; if not, write to the Free\n"
-"   Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA\n"
-"   02111-1307, USA.  */\n"
+"   You should have received a copy of the GNU General Public License along\n"
+"   with this program; if not, write to the Free Software Foundation,\n"
+"   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.  */\n"
 "\n";
 
 int
diff -ru ./src/libprelude-error/mkerrnos.awk ./src/libprelude-error/mkerrnos.awk
--- ./src/libprelude-error/mkerrnos.awk	2016-09-15 08:49:10.806000884 +0200
+++ ./src/libprelude-error/mkerrnos.awk	2016-12-02 07:42:46.332456910 +0100
@@ -11,9 +11,9 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 # General Public License for more details.
 #
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
-# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 # As a special exception, g10 Code GmbH gives unlimited permission to
 # copy, distribute and modify the C source files that are the output
diff -ru ./src/libprelude-error/mkheader.awk ./src/libprelude-error/mkheader.awk
--- ./src/libprelude-error/mkheader.awk	2016-09-15 08:49:10.806000884 +0200
+++ ./src/libprelude-error/mkheader.awk	2016-12-02 07:42:59.250165869 +0100
@@ -11,9 +11,9 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 # General Public License for more details.
 #
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
-# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 # As a special exception, g10 Code GmbH gives unlimited permission to
 # copy, distribute and modify the C source files that are the output
diff -ru ./src/libprelude-error/mkstrtable.awk ./src/libprelude-error/mkstrtable.awk
--- ./src/libprelude-error/mkstrtable.awk	2016-09-15 08:49:10.806000884 +0200
+++ ./src/libprelude-error/mkstrtable.awk	2016-12-02 07:43:08.552956273 +0100
@@ -11,9 +11,9 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 # General Public License for more details.
 #
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
-# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 # As a special exception, g10 Code GmbH gives unlimited permission to
 # copy, distribute and modify the C source files that are the output
diff -ru ./src/libprelude-error/strerror.c ./src/libprelude-error/strerror.c
--- ./src/libprelude-error/strerror.c	2016-09-15 08:49:10.806000884 +0200
+++ ./src/libprelude-error/strerror.c	2016-12-02 07:44:11.256543536 +0100
@@ -13,10 +13,9 @@
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    Lesser General Public License for more details.
  
-   You should have received a copy of the GNU Lesser General Public
-   License along with libgpg-error; if not, write to the Free
-   Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
-   02111-1307, USA.  */
+   You should have received a copy of the GNU General Public License along
+   with this program; if not, write to the Free Software Foundation, Inc.,
+   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA. */
 
 #include "config.h"
 
diff -ru ./src/libprelude-error/strsource.c ./src/libprelude-error/strsource.c
--- ./src/libprelude-error/strsource.c	2016-09-15 08:49:10.806000884 +0200
+++ ./src/libprelude-error/strsource.c	2016-12-02 07:44:26.455201104 +0100
@@ -13,10 +13,9 @@
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    Lesser General Public License for more details.
  
-   You should have received a copy of the GNU Lesser General Public
-   License along with libgpg-error; if not, write to the Free
-   Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
-   02111-1307, USA.  */
+   You should have received a copy of the GNU General Public License along
+   with this program; if not, write to the Free Software Foundation, Inc.,
+   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA. */
 
 
 #include "config.h"
