--- src/CMakeLists.txt~	2019-07-28 13:32:49.436473952 +0200
+++ src/CMakeLists.txt	2019-07-28 13:33:52.016439550 +0200
@@ -261,9 +261,8 @@
     endif()
     include_directories(${LLVM_INCLUDE_DIRS})
     add_definitions(${LLVM_DEFINITIONS})
-    llvm_map_components_to_libnames(llvm_libs support core option)
     target_compile_definitions(doxygen PRIVATE ${LLVM_DEFINITIONS})
-    set(CLANG_LIBS libclang clangTooling ${llvm_libs})
+    set(CLANG_LIBS libclang clangTooling LLVM)
 endif()
 
 target_link_libraries(doxygen
