From 248a77c7fd2f9682c58d3d3dd6f3c2ba3ad2c111 Mon Sep 17 00:00:00 2001
From: Tanu Kaskinen <tanuk@iki.fi>
Date: Sat, 2 Sep 2017 15:35:01 +0300
Subject: [PATCH] alsa-mixer: recognize the "Speaker Jack" control

This control has been seen in the wild:
https://lists.freedesktop.org/archives/pulseaudio-discuss/2017-August/028595.html
(The pastebin link in that mail might not work anymore, but the paste
just shows that there's a Speaker Jack control).
---
 src/modules/alsa/mixer/paths/analog-output-speaker.conf | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/modules/alsa/mixer/paths/analog-output-speaker.conf b/src/modules/alsa/mixer/paths/analog-output-speaker.conf
index 9f4dac414134..6f9968e1f537 100644
--- a/src/modules/alsa/mixer/paths/analog-output-speaker.conf
+++ b/src/modules/alsa/mixer/paths/analog-output-speaker.conf
@@ -56,6 +56,9 @@ state.unplugged = unknown
 state.plugged = no
 state.unplugged = unknown
 
+[Jack Speaker]
+required-any = any
+
 [Jack Speaker Phantom]
 required-any = any
 state.plugged = unknown
-- 
2.16.4

