---
 src/pulsecore/socket-client.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/pulsecore/socket-client.c
+++ b/src/pulsecore/socket-client.c
@@ -250,7 +250,7 @@ static int sockaddr_prepare(pa_socket_cl
     c->local = pa_socket_address_is_local(sa);
 
     if ((c->fd = pa_socket_cloexec(sa->sa_family, SOCK_STREAM, 0)) < 0) {
-        pa_log("socket(): %s", pa_cstrerror(errno));
+        /*pa_log("socket(): %s", pa_cstrerror(errno));*/
         return -1;
     }
 
