Index: src/util/inet_proto.c
===================================================================
--- src/util/inet_proto.c.orig
+++ src/util/inet_proto.c
@@ -195,7 +195,6 @@ INET_PROTO_INFO *inet_proto_init(const c
 	if ((sock = socket(PF_INET6, SOCK_STREAM, 0)) >= 0) {
 	    close(sock);
 	} else if (errno == EAFNOSUPPORT || errno == EPROTONOSUPPORT) {
-	    msg_warn("%s: disabling IPv6 name/address support: %m", context);
 	    inet_proto_mask &= ~INET_PROTO_MASK_IPV6;
 	} else {
 	    msg_fatal("socket: %m");
