From 67fe9c150965b66b205dd9a0b3c1666d43d74335 Mon Sep 17 00:00:00 2001
From: Daniel Pizetta <daniel.pizetta@usp.br>
Date: Sun, 4 Apr 2021 10:54:25 -0300
Subject: [PATCH] Fix tristate enumerator, closes #275

---
 qdarkstyle/example/__main__.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/qdarkstyle/example/__main__.py b/qdarkstyle/example/__main__.py
index c9a34fa5..b3d84e49 100644
--- a/qdarkstyle/example/__main__.py
+++ b/qdarkstyle/example/__main__.py
@@ -162,9 +162,9 @@ def main():
     uic.loadUi(os.path.join(here, 'ui/dw_buttons.ui'), dw_buttons)
     window.addDockWidget(QtCore.Qt.RightDockWidgetArea, dw_buttons)
 
-    # Set state indeterminate (value=1)
+    # Set state indeterminate #275
     dw_buttons.checkBoxTristate.stateChanged.connect(dw_buttons.checkBoxTristateDis.setCheckState)
-    dw_buttons.checkBoxTristate.setCheckState(1)
+    dw_buttons.checkBoxTristate.setCheckState(QtCore.Qt.PartiallyChecked)
 
     # Add actions on popup toolbuttons
     menu = QtWidgets.QMenu()
