Index: freetds-1.4.2/configure
===================================================================
--- freetds-1.4.2.orig/configure
+++ freetds-1.4.2/configure
@@ -21580,6 +21580,7 @@ SQLRETURN  SQL_API SQLColAttribute (SQLH
 	SQLUSMALLINT ColumnNumber, SQLUSMALLINT FieldIdentifier,
 	SQLPOINTER CharacterAttribute, SQLSMALLINT BufferLength,
 	SQLSMALLINT *StringLength, SQLLEN * NumericAttribute) {
+  return 0;
 }
 int
 main (void)
