Index: freetds-1.3.18/configure
===================================================================
--- freetds-1.3.18.orig/configure
+++ freetds-1.3.18/configure
@@ -19944,6 +19944,7 @@ SQLRETURN  SQL_API SQLColAttribute (SQLH
 	SQLUSMALLINT ColumnNumber, SQLUSMALLINT FieldIdentifier,
 	SQLPOINTER CharacterAttribute, SQLSMALLINT BufferLength,
 	SQLSMALLINT *StringLength, SQLLEN * NumericAttribute) {
+  return 0;
 }
 int
 main ()
