From 8e092603ea429fb6dde3f2580d0d6d8fe79ec265 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kristoffer=20Gr=C3=B6nlund?= <krig@koru.se>
Date: Thu, 9 Jan 2014 11:46:15 +0100
Subject: [PATCH 6/8] openSUSE: Adjust man pages destination

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index b82a880038c8..70a238a3900a 100644
--- a/setup.py
+++ b/setup.py
@@ -38,6 +38,6 @@ setup(
 
     packages=['psshlib'],
     scripts = [os.path.join("bin", p) for p in ["pssh", "pnuke", "prsync", "pslurp", "pscp", "pssh-askpass"]],
-    data_files=[('man/man1', ['man/man1/pssh.1', 'man/man1/pscp.1',
+    data_files=[('share/man/man1', ['man/man1/pssh.1', 'man/man1/pscp.1',
         'man/man1/prsync.1', 'man/man1/pslurp.1', 'man/man1/pnuke.1'])],
     )
-- 
1.8.4

