diff --git a/lib/Rcompat.c b/lib/Rcompat.c
index e13d365..0788b5f 100755
--- a/lib/Rcompat.c
+++ b/lib/Rcompat.c
@@ -44,6 +44,8 @@
 
 #include "common.h"
 
+#include<stdio.h>
+
 static const char rcsid[] =
 "$Id: Rcompat.c,v 1.89.4.2.6.2.4.1 2024/11/21 10:22:42 michaels Exp $";
 
@@ -513,7 +515,7 @@ Rgets(buf)
    slog(LOG_DEBUG, "%s: fd %d", function, d);
 
    if (!gssapi_isencrypted(d))
-      return gets(buf);
+      return fgets(buf, INT_MAX, stdin);
 
    i = 0;
    while (Rread(d, buf + i, 1) == 1 && buf[i] != '\n')
