--- src/Makefile.in.orig	2010-03-13 02:58:08.000000000 +0100
+++ src/Makefile.in	2010-03-13 02:58:42.000000000 +0100
@@ -88,7 +88,7 @@
 prefix	= @prefix@
 mandir	= @mandir@
 install:
-	mkdir -p /etc/fsvs /var/spool/fsvs $(bindir) /etc/fsvs/svn/auth/svn.{simple,ssl.server,ssl.client-passphrase}
+	mkdir -p $(DESTDIR)/etc/fsvs $(DESTDIR)/var/spool/fsvs $(DESTDIR)$(bindir) $(DESTDIR)/etc/fsvs/svn/auth/svn.{simple,ssl.server,ssl.client-passphrase}
 	install -m 0755 $(DEST) $(DESTDIR)/$(bindir)
 #	install -m 0644 ../doc/fsvs.1 $(DESTDIR)/(mandir)
 
