--- configure.ac	2010-03-26 09:07:07.000000000 +0200
+++ configure.ac.new	2011-03-06 14:57:31.446178384 +0200
@@ -66,9 +66,9 @@
 else
         case $host in
         *-*-linux*)
-                DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O2 -Wall -ffast-math -fsigned-char"
-                PROFILE="-Wall -W -pg -g -O2 -ffast-math -fsigned-char"
+                DEBUG="-g -Wall"
+                CFLAGS="-O2 -Wall -ffast-math"
+                PROFILE="-Wall -W -pg -g -O2 -ffast-math"
                 ;;
         sparc-sun-*)
                 DEBUG="-g -Wall -fsigned-char -mv8"
