--- oggenc/resample.c	2010-01-22 08:28:06.000000000 +0200
+++ oggenc/resample.c.new	2011-03-06 15:24:42.925869442 +0200
@@ -174,7 +174,7 @@
                 assert(beta > 2.0);
                 break;
             default:
-                assert("arglist" == "valid");
+                assert(!"arglist not valid");
                 return -1;
             }
             op1 = va_arg(argp, res_parameter);
