Index: pecan-1.3.2/setup.py
===================================================================
--- pecan-1.3.2.orig/setup.py
+++ pecan-1.3.2/setup.py
@@ -35,16 +35,8 @@ tests_require = requirements + [
     'mock',
     'sqlalchemy'
 ]
-
-if sys.version_info < (3, 0):
-    # These don't support Python3 yet - don't run their tests
-    if platform.python_implementation() != 'PyPy':
-        # Kajiki is not pypy-compatible
-        tests_require += ['Kajiki']
-    tests_require += ['Genshi']
-else:
-    # Genshi added Python3 support in 0.7
-    tests_require += ['Genshi>=0.7']
+# Genshi added Python3 support in 0.7
+tests_require += ['Genshi>=0.7']
 
 #
 # call setup
