diff --git a/setup.py b/setup.py
index a4c6200..00d04a4 100644
--- a/setup.py
+++ b/setup.py
@@ -42,7 +42,7 @@ setup(
     long_description=open('README.rst').read() + '\n\n' + open('CHANGELOG.rst').read(),
     keywords='redmine redmineup redminecrm redminelib easyredmine',
     python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*',
-    install_requires=['requests>=2.20.0'],
+    install_requires=['requests>=1.20.0'],
     tests_require=tests_require,
     cmdclass={'test': NoseTests},
     zip_safe=False,
