--- jdk8/jdk/src/share/lib/security/java.security-linux	2014-07-15 23:08:27.000000000 +0200
+++ jdk8/jdk/src/share/lib/security/java.security-linux	2014-07-18 09:04:45.127566697 +0200
@@ -224,7 +224,9 @@
                com.ibm.oti.,\
                openj9.internal.,\
                jdk.xml.internal.,\
-               com.sun.activation.registries.
+               com.sun.activation.registries.,\
+               org.GNOME.Accessibility.,\
+               org.GNOME.Bonobo.
 
 #
 # List of comma-separated packages that start with or equal this string
@@ -274,7 +276,9 @@
                    jdk.nashorn.internal.,\
                    jdk.nashorn.tools.,\
                    jdk.xml.internal.,\
-                   com.sun.activation.registries.
+                   com.sun.activation.registries.,\
+                   org.GNOME.Accessibility.,\
+                   org.GNOME.Bonobo.
 
 #
 # Determines whether this properties file can be appended to
