diff --git a/CMakeLists.txt b/CMakeLists.txt
index fc141c2..c2f2b66 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,8 @@ project (EBOOK-TOOLS C)
 
 cmake_minimum_required(VERSION 2.4.0)
 
+find_package(PkgConfig)
+pkg_check_modules(ZIP libzip)
 #set(CMAKE_CURRENT_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
 set(LIBRARY_OUTPUT_PATH ${EBOOK-TOOLS_BINARY_DIR}/libs)
 set(EXECUTABLE_OUTPUT_PATH ${EBOOK-TOOLS_BINARY_DIR}/bin)
diff --git a/src/libepub/CMakeLists.txt b/src/libepub/CMakeLists.txt
index a3208b7..3bebea7 100644
--- a/src/libepub/CMakeLists.txt
+++ b/src/libepub/CMakeLists.txt
@@ -1,4 +1,4 @@
-include_directories (${EBOOK-TOOLS_SOURCE_DIR}/src/libepub ${LIBXML2_INCLUDE_DIR} ${LIBZIP_INCLUDE_DIR})
+include_directories (${EBOOK-TOOLS_SOURCE_DIR}/src/libepub ${LIBXML2_INCLUDE_DIR} ${LIBZIP_INCLUDE_DIR} ${ZIP_INCLUDE_DIRS} )
 add_library (epub SHARED epub.c ocf.c opf.c linklist.c list.c)
 target_link_libraries (epub ${LIBZIP_LIBRARY} ${LIBXML2_LIBRARIES})
 
