Index: Molsketch-0.7.2/generateTranslations.cmake
===================================================================
--- Molsketch-0.7.2.orig/generateTranslations.cmake
+++ Molsketch-0.7.2/generateTranslations.cmake
@@ -4,7 +4,7 @@ function(addTranslationResources _resour
 
   file(GLOB_RECURSE translationFiles *.ts)
   set_source_files_properties(${translationFiles} PROPERTIES OUTPUT_LOCATION ${qmDir})
-  qt_add_translation(qmFileList ${translationFiles})
+  qt5_add_translation(qmFileList ${translationFiles})
 
   set(qmResourcesFile "${qmDir}/${resourceFileName}.qrc")
   file(WRITE ${qmResourcesFile} "<!DOCTYPE RCC><RCC version=\"1.0\">\n  <qresource prefix=\"i18n\">\n")
