diff -Nru google_cloud_storage-3.13.1.orig/google/cloud/storage/_media/requests/__init__.py google_cloud_storage-3.13.1/google/cloud/storage/_media/requests/__init__.py
--- google_cloud_storage-3.13.1.orig/google/cloud/storage/_media/requests/__init__.py	2026-08-06 07:54:31.000000000 +0200
+++ google_cloud_storage-3.13.1/google/cloud/storage/_media/requests/__init__.py	2026-08-14 09:20:33.796177231 +0200
@@ -32,7 +32,7 @@
 
    import google.auth
    import google.auth.credentials as creds_mod
-   import mock
+   from unittest import mock
 
    def mock_default(scopes=None):
        credentials = mock.Mock(spec=creds_mod.Credentials)
@@ -68,7 +68,7 @@
 
 .. testsetup:: basic-download
 
-   import mock
+   from unittest import mock
    import requests
    import http.client
 
@@ -111,7 +111,7 @@
 
 .. testsetup:: basic-download-with-slice
 
-   import mock
+   from unittest import mock
    import requests
    import http.client
 
@@ -170,7 +170,7 @@
 
    import io
 
-   import mock
+   from unittest import mock
    import requests
    import http.client
 
@@ -230,7 +230,7 @@
 
 .. testsetup:: chunked-download-end
 
-   import mock
+   from unittest import mock
    import requests
    import http.client
 
@@ -303,7 +303,7 @@
 
    import json
 
-   import mock
+   from unittest import mock
    import requests
    import http.client
 
@@ -363,7 +363,7 @@
 
    import time
 
-   import mock
+   from unittest import mock
    import requests
    import http.client
 
@@ -438,7 +438,7 @@
 
    import json
 
-   import mock
+   from unittest import mock
    import requests
    import http.client
 
@@ -524,7 +524,7 @@
 
    import io
 
-   import mock
+   from unittest import mock
    import requests
    import http.client
 
@@ -585,7 +585,7 @@
    import io
    import json
 
-   import mock
+   from unittest import mock
    import requests
    import http.client
 
diff -Nru google_cloud_storage-3.13.1.orig/google/cloud/storage/_media/requests/upload.py google_cloud_storage-3.13.1/google/cloud/storage/_media/requests/upload.py
--- google_cloud_storage-3.13.1.orig/google/cloud/storage/_media/requests/upload.py	2026-08-06 07:54:31.000000000 +0200
+++ google_cloud_storage-3.13.1/google/cloud/storage/_media/requests/upload.py	2026-08-14 09:20:33.794086163 +0200
@@ -196,7 +196,7 @@
        import os
        import tempfile
 
-       import mock
+       from unittest import mock
        import requests
        import http.client
 
@@ -253,7 +253,7 @@
 
        import io
 
-       import mock
+       from unittest import mock
        import requests
        import http.client
 
@@ -294,7 +294,7 @@
 
        import io
 
-       import mock
+       from unittest import mock
        import requests
        import http.client
 
@@ -455,7 +455,7 @@
 
            import io
 
-           import mock
+           from unittest import mock
            import requests
            import http.client
 
diff -Nru google_cloud_storage-3.13.1.orig/tests/system/test_blob.py google_cloud_storage-3.13.1/tests/system/test_blob.py
--- google_cloud_storage-3.13.1.orig/tests/system/test_blob.py	2026-08-06 07:54:30.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/system/test_blob.py	2026-08-14 09:20:33.863698761 +0200
@@ -20,7 +20,7 @@
 import uuid
 import warnings
 
-import mock
+from unittest import mock
 import pytest
 from google.api_core import exceptions
 
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/asyncio/test_bidi_async.py google_cloud_storage-3.13.1/tests/unit/asyncio/test_bidi_async.py
--- google_cloud_storage-3.13.1.orig/tests/unit/asyncio/test_bidi_async.py	2026-08-06 07:54:34.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/asyncio/test_bidi_async.py	2026-08-14 09:20:33.916682075 +0200
@@ -19,7 +19,7 @@
 try:
     from unittest.mock import AsyncMock
 except ImportError:  # pragma: NO COVER
-    from mock import AsyncMock  # type: ignore
+    from unittest.mock import AsyncMock  # type: ignore
 
 
 import pytest
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test_acl.py google_cloud_storage-3.13.1/tests/unit/test_acl.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test_acl.py	2026-08-06 07:54:30.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test_acl.py	2026-08-14 09:20:33.960474373 +0200
@@ -14,7 +14,7 @@
 
 import unittest
 
-import mock
+from unittest import mock
 
 from google.cloud.storage.retry import (
     DEFAULT_RETRY,
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test_batch.py google_cloud_storage-3.13.1/tests/unit/test_batch.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test_batch.py	2026-08-06 07:54:30.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test_batch.py	2026-08-14 09:20:33.914709422 +0200
@@ -16,7 +16,7 @@
 import unittest
 from http.client import NO_CONTENT, SERVICE_UNAVAILABLE
 
-import mock
+from unittest import mock
 import requests
 
 from google.cloud.storage._helpers import _DEFAULT_UNIVERSE_DOMAIN
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test_blob.py google_cloud_storage-3.13.1/tests/unit/test_blob.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test_blob.py	2026-08-06 07:54:33.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test_blob.py	2026-08-14 09:20:33.903459190 +0200
@@ -24,7 +24,7 @@
 from unittest.mock import patch
 from urllib.parse import urlencode
 
-import mock
+from unittest import mock
 import pytest
 from google.cloud.exceptions import NotFound
 
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test_bucket.py google_cloud_storage-3.13.1/tests/unit/test_bucket.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test_bucket.py	2026-08-06 07:54:30.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test_bucket.py	2026-08-14 09:20:33.957374707 +0200
@@ -15,7 +15,7 @@
 import datetime
 import unittest
 
-import mock
+from unittest import mock
 import pytest
 
 from google.cloud.storage._helpers import _NOW, _UTC, _get_default_storage_base_url
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test_client.py google_cloud_storage-3.13.1/tests/unit/test_client.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test_client.py	2026-08-06 07:54:29.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test_client.py	2026-08-14 09:20:33.912962889 +0200
@@ -21,7 +21,7 @@
 import urllib
 from unittest.mock import patch
 
-import mock
+from unittest import mock
 import pytest
 import requests
 from google.api_core import exceptions
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test_fileio.py google_cloud_storage-3.13.1/tests/unit/test_fileio.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test_fileio.py	2026-08-06 07:54:33.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test_fileio.py	2026-08-14 09:20:33.910273213 +0200
@@ -18,7 +18,7 @@
 import string
 import unittest
 
-import mock
+from unittest import mock
 from google.api_core.exceptions import RequestRangeNotSatisfiable
 
 from google.cloud.storage.fileio import CHUNK_SIZE_MULTIPLE
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test__helpers.py google_cloud_storage-3.13.1/tests/unit/test__helpers.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test__helpers.py	2026-08-06 07:54:33.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test__helpers.py	2026-08-14 09:20:33.908690311 +0200
@@ -14,7 +14,7 @@
 
 import unittest
 
-import mock
+from unittest import mock
 
 from google.cloud.storage.retry import (
     DEFAULT_RETRY,
@@ -590,7 +590,7 @@
         self.assertEqual(SIGNED_CONTENT, b"kBiQqOnIz21aGlQrIp/r/w==")
 
     def test_it_with_stubs(self):
-        import mock
+        from unittest import mock
 
         class _Buffer(object):
             def __init__(self, return_vals):
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test_hmac_key.py google_cloud_storage-3.13.1/tests/unit/test_hmac_key.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test_hmac_key.py	2026-08-06 07:54:30.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test_hmac_key.py	2026-08-14 09:20:33.955793629 +0200
@@ -14,7 +14,7 @@
 
 import unittest
 
-import mock
+from unittest import mock
 
 from google.cloud.storage._helpers import _NOW, _UTC
 from google.cloud.storage.retry import DEFAULT_RETRY, DEFAULT_RETRY_IF_ETAG_IN_JSON
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test__http.py google_cloud_storage-3.13.1/tests/unit/test__http.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test__http.py	2026-08-06 07:54:30.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test__http.py	2026-08-14 09:20:33.894019171 +0200
@@ -15,7 +15,7 @@
 import unittest
 from unittest.mock import patch
 
-import mock
+from unittest import mock
 
 from google.cloud.storage import _helpers
 
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test_notification.py google_cloud_storage-3.13.1/tests/unit/test_notification.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test_notification.py	2026-08-06 07:54:29.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test_notification.py	2026-08-14 09:20:33.896269843 +0200
@@ -14,7 +14,7 @@
 
 import unittest
 
-import mock
+from unittest import mock
 
 from google.cloud.storage.retry import DEFAULT_RETRY
 
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test__opentelemetry_tracing.py google_cloud_storage-3.13.1/tests/unit/test__opentelemetry_tracing.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test__opentelemetry_tracing.py	2026-08-06 07:54:30.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test__opentelemetry_tracing.py	2026-08-14 09:20:33.901547874 +0200
@@ -16,7 +16,7 @@
 import os
 import sys
 
-import mock
+from unittest import mock
 import pytest
 from google.api_core.exceptions import GoogleAPICallError
 
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test_retry.py google_cloud_storage-3.13.1/tests/unit/test_retry.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test_retry.py	2026-08-06 07:54:33.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test_retry.py	2026-08-14 09:20:33.941949643 +0200
@@ -14,7 +14,7 @@
 
 import unittest
 
-import mock
+from unittest import mock
 
 from google.cloud.storage import _helpers
 from google.cloud.storage.exceptions import InvalidResponse
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test__signing.py google_cloud_storage-3.13.1/tests/unit/test__signing.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test__signing.py	2026-08-06 07:54:30.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test__signing.py	2026-08-14 09:20:33.907175009 +0200
@@ -23,7 +23,7 @@
 import unittest
 import urllib.parse
 
-import mock
+from unittest import mock
 import pytest
 
 from google.cloud.storage._helpers import _UTC
diff -Nru google_cloud_storage-3.13.1.orig/tests/unit/test_transfer_manager.py google_cloud_storage-3.13.1/tests/unit/test_transfer_manager.py
--- google_cloud_storage-3.13.1.orig/tests/unit/test_transfer_manager.py	2026-08-06 07:54:30.000000000 +0200
+++ google_cloud_storage-3.13.1/tests/unit/test_transfer_manager.py	2026-08-14 09:20:33.940362163 +0200
@@ -16,7 +16,7 @@
 import pickle
 import tempfile
 
-import mock
+from unittest import mock
 import pytest
 from google.api_core import exceptions
 
