Index: yafc-1.1.1/src/help.c
===================================================================
--- yafc-1.1.1.orig/src/help.c
+++ yafc-1.1.1/src/help.c
@@ -66,7 +66,6 @@ void cmd_version(int argc, char **argv)
 
 	printf(FULLVER "\n");
 
-	printf(_("Compiled " __TIME__ " " __DATE__ " (" HOSTTYPE ")\n"));
 #if (HAVE_LIBREADLINE >= 210)
 	printf(_("Using Readline version %s\n"), rl_library_version);
 #elif defined(HAVE_LIBREADLINE)
