Index: samplicator-1.3.8rc1+git.20171112/samplicator.service
===================================================================
--- samplicator-1.3.8rc1+git.20171112.orig/samplicator.service
+++ samplicator-1.3.8rc1+git.20171112/samplicator.service
@@ -11,6 +11,19 @@ Description=Samplicator
 After=network.target
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+PrivateDevices=true
+ProtectHostname=true
+ProtectClock=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 Type=forking
 ExecStart=/opt/samplicator/bin/samplicate -S -c /opt/samplicator/etc/samplicator.conf -p 162 -d 0 -f
 
