From 32f1f7b64da269735c4a15df511affe0f8d760fe Mon Sep 17 00:00:00 2001
From: mslacken <christian.goll@gmail.com>
Date: Wed, 24 Feb 2021 16:08:14 +0100
Subject: [PATCH 1/2] petsc 3.3 p2 no rpath

---
 config/BuildSystem/config/setCompilers.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/config/BuildSystem/config/setCompilers.py b/config/BuildSystem/config/setCompilers.py
index fef9b858..d17c8444 100644
--- a/config/BuildSystem/config/setCompilers.py
+++ b/config/BuildSystem/config/setCompilers.py
@@ -1691,9 +1691,9 @@ class Configure(config.base.Configure):
       self.pushLanguage(language)
       # test '-R' before '-rpath' as sun compilers [c,fortran] don't give proper errors with wrong options.
       if not Configure.isDarwin(self.log):
-        testFlags = ['-Wl,-rpath,', '-R','-rpath ' , '-Wl,-R,']
+        testFlags = []
       else:
-        testFlags = ['-Wl,-rpath,']
+        testFlags = []
       # test '-R' before '-Wl,-rpath' for SUN compilers [as cc on linux accepts -Wl,-rpath, but  f90 & CC do not.
       if self.isSun(self.framework.getCompiler(), self.log):
         testFlags.insert(0,'-R')
-- 
2.26.2

