Index: libircclient-1.9/doc/conf.py
===================================================================
--- libircclient-1.9.orig/doc/conf.py
+++ libircclient-1.9/doc/conf.py
@@ -25,7 +25,7 @@ import sys, os
 
 # Add any Sphinx extension module names here, as strings. They can be extensions
 # coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
-extensions = ['sphinx.ext.doctest', 'rst2pdf.pdfbuilder']
+extensions = ['sphinx.ext.doctest']
 
 # Add any paths that contain templates here, relative to this directory.
 templates_path = ['_templates']
