Index: mcabber-1.1.2/mcabber/screen.h
===================================================================
--- mcabber-1.1.2.orig/mcabber/screen.h
+++ mcabber-1.1.2/mcabber/screen.h
@@ -28,7 +28,7 @@ void spellcheck_deinit(void);
 #include <mcabber/logprint.h>
 #include <mcabber/roster.h>
 
-#define INPUTLINE_LENGTH  1024
+#define INPUTLINE_LENGTH  16384
 
 // Only used in screen.c; this is the maximum line number
 // in a multi-line message.  Should be < 1000
