Index: gcin-2.8.6/qt5-im/Makefile
===================================================================
--- gcin-2.8.6.orig/qt5-im/Makefile
+++ gcin-2.8.6/qt5-im/Makefile
@@ -3,7 +3,7 @@ QT=qt5
 QTIM=$(QT)/plugins/platforminputcontexts
 IMMODULES=$(libdir)/$(DEB_BUILD_MULTIARCH)/$(QTIM)
 
-INCS=-I../im-client -I/usr/include/X11 `pkg-config Qt5Core Qt5Gui --cflags`
+INCS=-I../im-client -I/usr/include/X11 `pkg-config Qt5Core Qt5Gui Qt5DBus --cflags`
 MODVERSION=`pkg-config --modversion Qt5Gui`
 INCS+=-I`pkg-config --variable=includedir Qt5Core`/QtCore/$(MODVERSION)
 INCS+=-I`pkg-config --variable=includedir Qt5Gui`/QtGui/$(MODVERSION)
@@ -13,7 +13,7 @@ INCS+=-I`pkg-config --variable=includedi
 CXXFLAGS=$(OPTFLAGS) $(INCS) -Wall -D_REENTRANT -DUNIX=1 -fPIC  -DQT5 -DQT_SHARED -DQT_IMMODULE -DPIC -DDEBUG="0$(GCIN_DEBUG)"
 OBJS= moc_gcin-qt5.o gcin-qt5.o gcin-imcontext-qt5.o moc_qgcinplatforminputcontext.o
 .SUFFIXES:	.c .cpp .a .so .E .h
-LDFLAGS=-L../im-client -lgcin-im-client `pkg-config Qt5Core Qt5Gui --libs`
+LDFLAGS=-L../im-client -lgcin-im-client `pkg-config Qt5Core Qt5Gui Qt5DBus --libs`
 
 GCIN_QT5 = libgcinplatforminputcontextplugin.so
 
@@ -23,10 +23,10 @@ all:    $(GCIN_QT5)
 	$(CXX) -E $(CFLAGS) $(INCS) $< > $@
 
 moc_gcin-qt5.cpp:	gcin-qt5.h
-	$(QT5_MOC) $< -o $@
+	$(QT5_MOC) $(INCS) $< -o $@
 
 moc_qgcinplatforminputcontext.cpp:	qgcinplatforminputcontext.h
-	$(QT5_MOC) $< -o $@
+	$(QT5_MOC) $(INCS) $< -o $@
 
 $(GCIN_QT5): $(OBJS)
 	export LD_RUN_PATH=$(gcinlibdir) ;\
