diff -rc alpine-2.25/pico/display.c alpine-2.25.insertpat/pico/display.c
*** alpine-2.25/pico/display.c	2021-09-18 09:02:36.386783561 -0600
--- alpine-2.25.insertpat/pico/display.c	2021-09-18 09:03:18.847076682 -0600
***************
*** 2200,2205 ****
--- 2200,2217 ----
  
  	    continue;
  
+ 	  case (CTRL|'N'):			/* Insert pattern */
+ 	   if (pat[0] != '\0'){
+ 		ucs4_strncpy(buf+ucs4_strlen(buf), pat, NPAT);
+ 		pputs(pat,1);
+ 		b = &buf[ucs4_strlen(buf)];
+ 		dline.vused += ucs4_strlen(pat);
+ 		changed = TRUE;
+ 		}
+ 	   else
+ 		(*term.t_beep)();
+ 	  continue;
+ 
  	  case (CTRL|'G') :			/* CTRL-G help		*/
  	    if(term.t_mrow == 0 && km_popped == 0){
  		movecursor(term.t_nrow-2, 0);
diff -rc alpine-2.25/pico/search.c alpine-2.25.insertpat/pico/search.c
*** alpine-2.25/pico/search.c	2021-09-18 09:02:36.386783561 -0600
--- alpine-2.25.insertpat/pico/search.c	2021-09-18 09:03:18.847076682 -0600
***************
*** 122,127 ****
--- 122,131 ----
  N_("~        Hitting only ~R~e~t~u~r~n or at the prompt will cause the"),
  N_("        search to be made with the default value."),
  "  ",
+ N_("~        Hitting ~^~N will reinsert the last string you searched for"),
+ N_("        so that you can edit it (in case you made a mistake entering the"),
+ N_("        search pattern the first time)."),
+ "  ",
  N_("        The text search is not case sensitive, and will examine the"),
  N_("        entire message."),
  "  ",
