---
 imap/src/c-client/mail.c |   10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

Index: alpine-2.20/imap/src/c-client/mail.c
===================================================================
--- alpine-2.20.orig/imap/src/c-client/mail.c
+++ alpine-2.20/imap/src/c-client/mail.c
@@ -2979,8 +2979,14 @@ long mail_parse_date (MESSAGECACHE *elt,
 				/* parse time */
     d = strtoul (s+1,(char **) &s,10);
     if (*s != ':') return NIL;
-    m = strtoul (++s,(char **) &s,10);
-    y = (*s == ':') ? strtoul (++s,(char **) &s,10) : 0;
+    s++;
+    m = strtoul (s,(char **) &s,10);
+    if (*s == ':') {
+	s++;
+	y = strtoul(s, (char **)&s, 10);
+    } else {
+	y = 0;
+    }
 				/* validity check time */
     if ((d > 23) || (m > 59) || (y > 60)) return NIL; 
 				/* set values in elt */
