Index: CLHEP-CLHEP_2_4_7_1/CMakeLists.txt
===================================================================
--- CLHEP-CLHEP_2_4_7_1.orig/CMakeLists.txt
+++ CLHEP-CLHEP_2_4_7_1/CMakeLists.txt
@@ -145,8 +145,9 @@ if(CLHEP_BUILD_DOCS)
     )
   add_custom_target(doc ALL DEPENDS "${PROJECT_BINARY_DIR}/Doxygen/html/index.html")
 
+  set(CLHEP_DOCDIR "share/doc/CLHEP" CACHE PATH "Directory to install documentation")
   install(DIRECTORY ${PROJECT_BINARY_DIR}/Doxygen
-    DESTINATION "share/doc/CLHEP"
+    DESTINATION ${CLHEP_DOCDIR}
     )
 endif()
 
Index: CLHEP-CLHEP_2_4_7_1/cmake/Modules/ClhepBuildTex.cmake
===================================================================
--- CLHEP-CLHEP_2_4_7_1.orig/cmake/Modules/ClhepBuildTex.cmake
+++ CLHEP-CLHEP_2_4_7_1/cmake/Modules/ClhepBuildTex.cmake
@@ -39,7 +39,7 @@ IF(PDFLATEX_COMPILER)
     )
 
  INSTALL (FILES ${CMAKE_CURRENT_BINARY_DIR}/${main_tex_file}.pdf
-          DESTINATION share/doc/CLHEP/${package} )
+   DESTINATION ${CLHEP_DOCDIR}/${package} )
 ENDIF(PDFLATEX_COMPILER)
 
 endmacro( clhep_latex )
Index: CLHEP-CLHEP_2_4_7_1/Exceptions/doc/CMakeLists.txt
===================================================================
--- CLHEP-CLHEP_2_4_7_1.orig/Exceptions/doc/CMakeLists.txt
+++ CLHEP-CLHEP_2_4_7_1/Exceptions/doc/CMakeLists.txt
@@ -15,4 +15,4 @@ INSTALL (FILES ex-design-1
 	       reply-pcanal-2
 	       whyZMthrowRethrows.txt
 	       ZMthrow_event_sequence.txt
-         DESTINATION share/doc/CLHEP/Exceptions )
+				 DESTINATION ${CLHEP_DOCDIR}/Exceptions )
Index: CLHEP-CLHEP_2_4_7_1/Random/doc/CMakeLists.txt
===================================================================
--- CLHEP-CLHEP_2_4_7_1.orig/Random/doc/CMakeLists.txt
+++ CLHEP-CLHEP_2_4_7_1/Random/doc/CMakeLists.txt
@@ -5,4 +5,4 @@ INSTALL (FILES Random.html
 	       JamesRandomSeeding.txt
 	       html/CLHEP-random.html
                html/onSavingRandomStates.html
-         DESTINATION share/doc/CLHEP/Random )
+         DESTINATION ${CLHEP_DOCDIR}/Random )
Index: CLHEP-CLHEP_2_4_7_1/Vector/doc/CMakeLists.txt
===================================================================
--- CLHEP-CLHEP_2_4_7_1.orig/Vector/doc/CMakeLists.txt
+++ CLHEP-CLHEP_2_4_7_1/Vector/doc/CMakeLists.txt
@@ -9,5 +9,5 @@ clhep_latex( Vector VectorDefs )
 INSTALL (FILES keyMergeIssues.doc
                merge-details.doc
 	       minorMergeIssues.doc
-         DESTINATION share/doc/CLHEP/Vector )
+         DESTINATION ${CLHEP_DOCDIR}/Vector )
 
