From: Jan Engelhardt <jengelh@inai.de>
Date: Tue, 2 Jan 2018 00:21:09 +0100
Subject: [PATCH] iu_client uses gprs_msgb.h

---
 configure.ac    |    1 +
 src/Makefile.am |    5 +++--
 2 files changed, 4 insertions(+), 2 deletions(-)

Index: osmo-iuh-0.3.0/configure.ac
===================================================================
--- osmo-iuh-0.3.0.orig/configure.ac
+++ osmo-iuh-0.3.0/configure.ac
@@ -34,6 +34,7 @@ fi
 PKG_PROG_PKG_CONFIG([0.20])
 
 PKG_CHECK_MODULES(OSMOCORE, libosmocore >= 0.11.0)
+PKG_CHECK_MODULES([OSMOGB], [libosmogb])
 PKG_CHECK_MODULES(OSMOGSM, libosmogsm >= 0.11.0)
 PKG_CHECK_MODULES(OSMOVTY, libosmovty >= 0.11.0)
 PKG_CHECK_MODULES(OSMOCTRL, libosmoctrl >= 0.11.0)
Index: osmo-iuh-0.3.0/src/Makefile.am
===================================================================
--- osmo-iuh-0.3.0.orig/src/Makefile.am
+++ osmo-iuh-0.3.0/src/Makefile.am
@@ -42,9 +42,10 @@ gen_ranap.stamp: $(ASN1_ROOT)/ranap/RANA
 
 ranap_decoder.c ranap_encoder.c: gen_ranap.stamp
 
-AM_CFLAGS = -Wall -I$(top_srcdir)/include -I$(top_builddir)/include \
+AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_builddir)/include \
 	    $(OSMOCORE_CFLAGS) $(OSMOVTY_CFLAGS) $(OSMOCTRL_CFLAGS) $(OSMOGSM_CFLAGS) \
-	    $(OSMONETIF_CFLAGS) $(ASN1C_CFLAGS) $(OSMOSIGTRAN_CFLAGS)
+	    $(OSMONETIF_CFLAGS) $(ASN1C_CFLAGS) $(OSMOSIGTRAN_CFLAGS) $(OSMOGB_CFLAGS)
+AM_CFLAGS = -Wall
 
 # build the shared RANAP library
 #
