Index: nls/locale.alias.pre
===================================================================
--- nls/locale.alias.pre.orig
+++ nls/locale.alias.pre
@@ -316,6 +316,7 @@ en_CA.iso88591:					en_CA.ISO8859-1
 en_CA.ISO-8859-1:				en_CA.ISO8859-1
 en_CA.ISO_8859-1:				en_CA.ISO8859-1
 en_CA.utf8:					en_CA.UTF-8
+en_DK.utf8:					en_DK.UTF-8
 en_DL.utf8:					en_DL.UTF-8
 en_GB:						en_GB.ISO8859-1
 en_GB.88591:					en_GB.ISO8859-1
