From: Some One <nobody@opensuse.org>
Date: Thu, 9 Apr 2015 14:55:37 +0200
Subject: [PATCH] suse-checks.diff

===================================================================
---
 Config.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Config.py b/Config.py
index ac3c5ea..f27607c 100644
--- a/Config.py
+++ b/Config.py
@@ -16,14 +16,13 @@ try:
 except ImportError:
     __version__ = 'devel'
 
-DEFAULT_CHECKS = ("DistributionCheck",
+DEFAULT_CHECKS = (
                   "TagsCheck",
                   "BinariesCheck",
                   "ConfigCheck",
                   "FilesCheck",
                   "DocFilesCheck",
                   "FHSCheck",
-                  "SignatureCheck",
                   "I18NCheck",
                   "MenuCheck",
                   "PostCheck",
