From: Some One <nobody@opensuse.org>
Date: Thu, 9 Apr 2015 14:55:38 +0200
Subject: [PATCH] yast-provides.diff

===================================================================
---
 TagsCheck.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: rpmlint-rpmlint-1.11/TagsCheck.py
===================================================================
--- rpmlint-rpmlint-1.11.orig/TagsCheck.py
+++ rpmlint-rpmlint-1.11/TagsCheck.py
@@ -866,7 +866,7 @@ class TagsCheck(AbstractCheck.AbstractCh
                     printWarning(pkg, 'no-url-tag')
 
         obs_names = [x[0] for x in pkg.obsoletes()]
-        prov_names = [x[0] for x in pkg.provides()]
+        prov_names = [x[0].split(':/')[0] for x in pkg.provides()]
 
         for o in (x for x in obs_names if x not in prov_names):
             printWarning(pkg, 'obsolete-not-provided', o)
