diff -up jni/GNUmakefile.patch4 jni/GNUmakefile
--- jni/GNUmakefile.patch4	2021-07-08 22:01:13.455019146 +0300
+++ jni/GNUmakefile	2021-07-08 22:10:00.737857846 +0300
@@ -265,7 +265,7 @@ debug:
 	@echo "JFFI_BUILD_DIR=$(JFFI_BUILD_DIR)"
 	@echo "OBJS=$(OBJS)"
 
-$(LIBJFFI):  $(OBJS) $(LIBFFI_LIBS)
+$(LIBJFFI):  $(OBJS)
 	$(CC) -o $@ $(LDFLAGS) $(SOFLAGS) $(OBJS) $(LIBFFI_LIBS) $(LIBS)
 	$(STRIP) $@
 
@@ -277,7 +277,9 @@ $(BUILD_DIR)/%.o : $(SRC_DIR)/%.S $(wild
 	@mkdir -p $(@D)
 	@$(CC) $(CFLAGS) -o $@ -c $<
 
+ifneq ($(USE_SYSTEM_LIBFFI),1)
 $(OBJS) : $(LIBFFI_LIBS)
+endif
 
 ifeq ($(OS), darwin)
 build_ffi = \
