--- mkelfImage-2.5/configure.ac.xx	2005-06-15 11:30:35.000000000 +0200
+++ mkelfImage-2.5/configure.ac	2005-06-15 11:30:57.000000000 +0200
@@ -170,7 +170,7 @@
 dnl ---Output variables...
 
 HOST_CC=$CC
-HOST_CFLAGS='-O2 -Wall $(HOST_CPPFLAGS)'
+HOST_CFLAGS='$(CFLAGS) $(HOST_CPPFLAGS)'
 
 dnl TODO: figure out how to set these appropriately for compilers other than gcc
 I386_CFLAGS='-Os -ffreestanding -Wall -W -Wno-format $(I386_CPPFLAGS)'
