--- jetty-alpn-alpn-project-8.1.13.v20181017/alpn-boot/src/main/java/sun/security/ssl/ClientHandshaker.java	2020-04-27 09:01:49.010193170 +0200
+++ jetty-alpn-alpn-project-8.1.13.v20181017/alpn-boot/src/main/java/sun/security/ssl/ClientHandshaker.java	2020-04-27 09:02:03.454272464 +0200
@@ -1347,7 +1347,7 @@
     @Override
     HandshakeMessage getKickstartMessage() throws SSLException {
         // session ID of the ClientHello message
-        SessionId sessionId = SSLSessionImpl.nullSession.getSessionId();
+        SessionId sessionId = new SessionId(new byte[0]);
 
         // a list of cipher suites sent by the client
         CipherSuiteList cipherSuites = getActiveCipherSuites();
