From: Thomas Andrejak <thomas.andrejak@gmail.com>
Date: 2017-02-27 23:25:00 +0100
References: http://prelude-siem.org/issues/878
Upstream: submitted
Subject: Fix Prelude Manager run dir for Systemd

--- configure	2016-04-23 00:32:48.541000362 +0200
+++ configure	2016-04-26 15:21:31.816028013 +0200
@@ -31054,7 +31054,7 @@
 
 manager_failover_dir=$LOCALSTATEDIR/spool/prelude-manager/failover
 manager_scheduler_dir=$LOCALSTATEDIR/spool/prelude-manager/scheduler
-manager_run_dir=$LOCALSTATEDIR/run/prelude-manager
+manager_run_dir=/run/prelude-manager
 
 
 cat >>confdefs.h <<_ACEOF
--- configure.ac	2016-04-22 23:27:29.451000362 +0200
+++ configure.ac	2016-04-26 15:21:41.811800293 +0200
@@ -337,7 +337,7 @@
 
 manager_failover_dir=$LOCALSTATEDIR/spool/prelude-manager/failover
 manager_scheduler_dir=$LOCALSTATEDIR/spool/prelude-manager/scheduler
-manager_run_dir=$LOCALSTATEDIR/run/prelude-manager
+manager_run_dir=/run/prelude-manager
 
 AC_DEFINE_UNQUOTED(REPORT_PLUGIN_DIR, "$report_plugin_dir", Prelude-Manager report plugin directory)
 AC_DEFINE_UNQUOTED(DECODE_PLUGIN_DIR, "$decode_plugin_dir", Prelude-Manager decode plugin directory)
