Index: phoronix-test-suite/deploy/phoromatic-systemd/phoromatic-client.service
===================================================================
--- phoronix-test-suite.orig/deploy/phoromatic-systemd/phoromatic-client.service
+++ phoronix-test-suite/deploy/phoromatic-systemd/phoromatic-client.service
@@ -4,6 +4,19 @@ After=syslog.target network-online.targe
 Documentation=man:phoronix-test-suite(1)
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+PrivateDevices=true
+ProtectHostname=true
+ProtectClock=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 TimeoutStartSec=300
 ExecStart=/usr/bin/phoronix-test-suite phoromatic.connect
 Environment=DISPLAY=:0
