Index: fluidsynth-2.2.2/fluidsynth.service.in
===================================================================
--- fluidsynth-2.2.2.orig/fluidsynth.service.in
+++ fluidsynth-2.2.2/fluidsynth.service.in
@@ -4,6 +4,17 @@ Documentation=man:fluidsynth(1)
 After=sound.target
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+ProtectHostname=true
+ProtectClock=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+# end of automatic additions 
 Type=notify
 NotifyAccess=main
 EnvironmentFile=@FLUID_DAEMON_ENV_FILE@
