diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4e1deef..c76b2e8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,6 +38,15 @@ set(CROSS_COMPILE_LIB_PATH     "/usr/${CC_ARCH}/lib")
 
 find_package(libusb-1.0 REQUIRED)
 
+########################################################################
+# Compiler setup
+########################################################################
+if(CMAKE_COMPILER_IS_GNUCC)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
+    add_definitions(-Wall)
+    add_definitions(-Wextra)
+    add_definitions(-pedantic)
+endif()
 
 set(LIBUSB3380_LIBRARY_DIR      lib${LIB_SUFFIX})
 set(LIBUSB3380_INCLUDE_DIR      include)
