Index: xen-4.8.0-testing/tools/libacpi/ssdt_s3.asl
===================================================================
--- xen-4.8.0-testing.orig/tools/libacpi/ssdt_s3.asl
+++ xen-4.8.0-testing/tools/libacpi/ssdt_s3.asl
@@ -16,13 +16,9 @@
 
 DefinitionBlock ("SSDT_S3.aml", "SSDT", 2, "Xen", "HVM", 0)
 {
-    /* Must match piix emulation */
-    Name (\_S3, Package (0x04)
-    {
-        0x01,  /* PM1a_CNT.SLP_TYP */
-        0x01,  /* PM1b_CNT.SLP_TYP */
-        0x0,   /* reserved */
-        0x0    /* reserved */
-    })
+    /*
+     * Turn off support for s3 sleep state to deal with SVVP tests.
+     * This is what MSFT does on HyperV.
+     */
 }
 
Index: xen-4.8.0-testing/tools/libacpi/ssdt_s4.asl
===================================================================
--- xen-4.8.0-testing.orig/tools/libacpi/ssdt_s4.asl
+++ xen-4.8.0-testing/tools/libacpi/ssdt_s4.asl
@@ -16,13 +16,9 @@
 
 DefinitionBlock ("SSDT_S4.aml", "SSDT", 2, "Xen", "HVM", 0)
 {
-    /* Must match piix emulation */
-    Name (\_S4, Package (0x04)
-    {
-        0x00,  /* PM1a_CNT.SLP_TYP */
-        0x00,  /* PM1b_CNT.SLP_TYP */
-        0x00,  /* reserved */
-        0x00   /* reserved */
-    })
+    /*
+     * Turn off support for s4 sleep state to deal with SVVP tests.
+     * This is what MSFT does on HyperV.
+     */
 }
 
