Index: xf86-video-intel-2.99.917/src/sna/sna_dri2.c
===================================================================
--- xf86-video-intel-2.99.917.orig/src/sna/sna_dri2.c
+++ xf86-video-intel-2.99.917/src/sna/sna_dri2.c
@@ -3707,8 +3707,10 @@ static const char *dri_driver_name(struc
 			return has_i830_dri() ? "i830" : "i915";
 		else if (sna->kgem.gen < 040)
 			return "i915";
+		else if (sna->kgem.gen < 0100)
+			return "crocus";
 		else
-			return "i965";
+			return "iris";
 	}
 
 	return s;
Index: xf86-video-intel-2.99.917/src/uxa/intel_dri.c
===================================================================
--- xf86-video-intel-2.99.917.orig/src/uxa/intel_dri.c
+++ xf86-video-intel-2.99.917/src/uxa/intel_dri.c
@@ -1540,8 +1540,10 @@ static const char *dri_driver_name(intel
 			return has_i830_dri() ? "i830" : "i915";
 		else if (INTEL_INFO(intel)->gen < 040)
 			return "i915";
+		else if (INTEL_INFO(intel)->gen < 0100)
+			return "crocus";
 		else
-			return "i965";
+			return "iris";
 	}
 
 	return s;
