--- Doom64EX-Plus-4.0.0.3.SDL.3.1.3.orig/src/engine/i_system.c	2024-10-06 08:16:07.000000000 +0200
+++ Doom64EX-Plus-4.0.0.3.SDL.3.1.3/src/engine/i_system.c	2024-10-06 10:46:14.355668407 +0200
@@ -299,6 +299,16 @@
 	}
 
 #if !defined(_WIN32)
+
+#ifdef DOOM_UNIX_INSTALL
+	if ((dir = I_GetUserDir())) {
+		snprintf(path, 511, "%s%s", dir, file);
+		if (I_FileExists(path)) {
+			return path;
+		}
+	}
+#endif    
+    
 #ifdef DOOM_UNIX_SYSTEM_DATADIR
 	snprintf(path, 511, "%s/%s", DOOM_UNIX_SYSTEM_DATADIR, file);
 	if (I_FileExists(path)) {
