diff -Nur ibus-1.5.29/ui/gtk3/panel.vala ibus-1.5.29-new/ui/gtk3/panel.vala
--- ibus-1.5.29/ui/gtk3/panel.vala	2023-11-30 23:08:12.858426343 +0800
+++ ibus-1.5.29-new/ui/gtk3/panel.vala	2023-11-30 23:08:43.015089972 +0800
@@ -1361,7 +1361,7 @@
 
     private void show_setup_dialog() {
         if (m_setup_pid != 0) {
-            if (Posix.kill(m_setup_pid, Posix.Signal.USR1) == 0)
+            if (Posix.kill(m_setup_pid, Posix.SIGUSR1) == 0)
                 return;
             m_setup_pid = 0;
         }
