Index: msm-0.7.6/msm/skill_entry.py
===================================================================
--- msm-0.7.6.orig/msm/skill_entry.py
+++ msm-0.7.6/msm/skill_entry.py
@@ -327,6 +327,8 @@ class SkillEntry(object):
         setup_script = join(self.path, "requirements.sh")
         if not exists(setup_script):
             return False
+        LOG.info("Please check manually the requirements file at " + requirements_file)
+        return False
 
         with work_dir(self.path):
             rc = subprocess.call(["bash", setup_script])
@@ -505,6 +507,8 @@ class SkillEntry(object):
         else:
             LOG.info('Nothing new for ' + self.name)
             return False
+        LOG.info("Please check manually the setup script at " + setup_script)
+        return False
 
     @staticmethod
     def find_git_url(path):
