Subject: domcaps: get list of supported panic device models
From: Lin Ma lma@suse.de Tue Dec 3 13:32:15 2024 +0800
Date: Tue Dec 10 14:05:20 2024 +0100:
Git: c859c7acec38a68a46b3ee98b1ff494fa88d508e

libvirt commit a52cd504 added support for advertisting panic device models.
Let's use it in domcapabilities.

Signed-off-by: Lin Ma <lma@suse.de>

diff --git a/virtinst/domcapabilities.py b/virtinst/domcapabilities.py
index 5d82a351e..9e1b11932 100644
--- a/virtinst/domcapabilities.py
+++ b/virtinst/domcapabilities.py
@@ -115,6 +115,7 @@ class _Devices(_CapsBlock):
     filesystem = XMLChildProperty(_make_capsblock("filesystem"), is_single=True)
     redirdev = XMLChildProperty(_make_capsblock("redirdev"), is_single=True)
     channel = XMLChildProperty(_make_capsblock("channel"), is_single=True)
+    panic = XMLChildProperty(_make_capsblock("panic"), is_single=True)
 
 
 class _Features(_CapsBlock):
@@ -508,3 +509,9 @@ class DomainCapabilities(XMLBuilder):
             return []
 
         return self.features.hyperv.get_enum("features").get_values()
+
+    def supported_panic_models(self):
+        """
+        Return list of supported panic device models.
+        """
+        return self.devices.panic.get_enum("model").get_values()
