Subject: virtManager: domain: fix indentation
From: Pavel Hrdina phrdina@redhat.com Wed Jan 29 10:50:03 2025 +0100
Date: Wed Jan 29 10:50:03 2025 +0100:
Git: 5ddd3456a0ca9836a98fc6ca4f0b2eaab268bf47

Signed-off-by: Pavel Hrdina <phrdina@redhat.com>

diff --git a/virtManager/object/domain.py b/virtManager/object/domain.py
index 5aade01d8..51aae4d8c 100644
--- a/virtManager/object/domain.py
+++ b/virtManager/object/domain.py
@@ -1157,14 +1157,13 @@ class vmmDomain(vmmLibvirtObject):
         return self._snapshot_list[:]
 
     def get_current_snapshot(self):
+        if self._backend.hasCurrentSnapshot(0):
+            rawsnap = self._backend.snapshotCurrent(0)
+            obj = vmmDomainSnapshot(self.conn, rawsnap)
+            obj.init_libvirt_state()
+            return obj
 
-       if self._backend.hasCurrentSnapshot(0):
-           rawsnap = self._backend.snapshotCurrent(0)
-           obj = vmmDomainSnapshot(self.conn, rawsnap)
-           obj.init_libvirt_state()
-           return obj
-
-       return None
+        return None
 
     @vmmLibvirtObject.lifecycle_action
     def revert_to_snapshot(self, snap):
