Subject: tests: cpio: set owner to 0:0
From: Weijia Wang 9713184+wegank@users.noreply.github.com Sat Aug 6 19:00:07 2022 +0000
Date: Tue Dec 13 13:45:16 2022 -0500:
Git: 75a25e37660c5578587f4a7a75917cf98d77cf7e


diff --git a/virtinst/install/installerinject.py b/virtinst/install/installerinject.py
index 0b2a9bc5..98d88cf8 100644
--- a/virtinst/install/installerinject.py
+++ b/virtinst/install/installerinject.py
@@ -20,7 +20,7 @@ def _run_initrd_commands(initrd, tempdir):
                                  stderr=subprocess.PIPE,
                                  cwd=tempdir)
     cpio_proc = subprocess.Popen(['cpio', '--create', '--null', '--quiet',
-                                  '--format=newc', '--owner=root:root'],
+                                  '--format=newc', '--owner=0:0'],
                                  stdin=find_proc.stdout,
                                  stdout=subprocess.PIPE,
                                  stderr=subprocess.PIPE,
