Subject: asyncjob: Remove unused import
From: Cole Robinson crobinso@redhat.com Thu Jan 19 11:59:20 2023 -0500
Date: Thu Jan 19 11:59:20 2023 -0500:
Git: fe86f4639f6e055f1e107fed9e0eb8be3fb2a66f

Signed-off-by: Cole Robinson <crobinso@redhat.com>

diff --git a/virtManager/asyncjob.py b/virtManager/asyncjob.py
index 46692ace..958320b2 100644
--- a/virtManager/asyncjob.py
+++ b/virtManager/asyncjob.py
@@ -7,7 +7,6 @@
 import threading
 import traceback
 
-from gi.repository import Gdk
 from gi.repository import GLib
 
 import libvirt
