Subject: cli: Add --features hyperv.xmm_input.state=on/off
From: Lin Ma lma@suse.de Mon Dec 30 19:49:47 2024 +0800
Date: Wed Jan 29 10:48:57 2025 +0100:
Git: 5617330513e951643d69afd4c0cfbd230f1d2983

Signed-off-by: Lin Ma <lma@suse.de>

diff --git a/tests/data/cli/compare/virt-install-many-devices.xml b/tests/data/cli/compare/virt-install-many-devices.xml
index 76e044731..eeb964620 100644
--- a/tests/data/cli/compare/virt-install-many-devices.xml
+++ b/tests/data/cli/compare/virt-install-many-devices.xml
@@ -154,6 +154,7 @@
       <ipi state="on"/>
       <evmcs state="on"/>
       <avic state="on"/>
+      <xmm_input state="on"/>
     </hyperv>
     <vmport state="off"/>
     <kvm>
diff --git a/tests/test_cli.py b/tests/test_cli.py
index 8bfccea18..76768dfe0 100644
--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -581,6 +581,7 @@ hyperv.tlbflush.state=on,\
 hyperv.ipi.state=on,\
 hyperv.evmcs.state=on,\
 hyperv.avic.state=on,\
+hyperv.xmm_input.state=on,\
 kvm.pv-ipi.state=on,\
 msrs.unknown=ignore
 
diff --git a/virtinst/cli.py b/virtinst/cli.py
index 6f633b933..add19ac09 100644
--- a/virtinst/cli.py
+++ b/virtinst/cli.py
@@ -3094,6 +3094,7 @@ class ParserFeatures(VirtCLIParser):
         cls.add_arg("hyperv.ipi.state", "hyperv_ipi", is_onoff=True)
         cls.add_arg("hyperv.evmcs.state", "hyperv_evmcs", is_onoff=True)
         cls.add_arg("hyperv.avic.state", "hyperv_avic", is_onoff=True)
+        cls.add_arg("hyperv.xmm_input.state", "hyperv_xmm_input", is_onoff=True)
 
         cls.add_arg("vmport.state", "vmport", is_onoff=True)
         cls.add_arg("kvm.hidden.state", "kvm_hidden", is_onoff=True)
diff --git a/virtinst/domain/features.py b/virtinst/domain/features.py
index 93a576360..ae3b23d98 100644
--- a/virtinst/domain/features.py
+++ b/virtinst/domain/features.py
@@ -43,6 +43,7 @@ class DomainFeatures(XMLBuilder):
     hyperv_ipi = XMLProperty("./hyperv/ipi/@state", is_onoff=True)
     hyperv_evmcs = XMLProperty("./hyperv/evmcs/@state", is_onoff=True)
     hyperv_avic = XMLProperty("./hyperv/avic/@state", is_onoff=True)
+    hyperv_xmm_input = XMLProperty("./hyperv/xmm_input/@state", is_onoff=True)
 
     vmport = XMLProperty("./vmport/@state", is_onoff=True)
     kvm_hidden = XMLProperty("./kvm/hidden/@state", is_onoff=True)
