Subject: tests: uitests: Fix window reposition on f38
From: Cole Robinson crobinso@redhat.com Fri May 5 14:54:31 2023 -0400
Date: Fri May 5 14:54:31 2023 -0400:
Git: 909c8aa880396fecb3e1fa174bdf89ce5e8b36c8

Signed-off-by: Cole Robinson <crobinso@redhat.com>

diff --git a/tests/uitests/lib/app.py b/tests/uitests/lib/app.py
index 221808d0..83628a7f 100644
--- a/tests/uitests/lib/app.py
+++ b/tests/uitests/lib/app.py
@@ -222,7 +222,7 @@ class VMMDogtailApp(object):
         # Give time for the child window to appear and possibly grab focus
         self.sleep(1)
         self.get_manager(check_active=False)
-        dogtail.rawinput.drag(childwin.title_coordinates(), (1000, 1000))
+        dogtail.rawinput.dragWithTrajectory(childwin.title_coordinates(), (1000, 1000))
         self.manager_conn_disconnect(conn_label)
         utils.check(lambda: not childwin.showing)
 
