Subject: rpm: convert license to SPDX format
From: Daniel P. Berrangé berrange@redhat.com Wed Apr 26 17:49:37 2023 +0100
Date: Sun May 7 12:40:05 2023 -0400:
Git: 6258d536895b1a64a81f111445994c35c0928d4b

Signed-off-by: Daniel P. Berrangé <berrange@redhat.com>

diff --git a/virt-manager.spec b/virt-manager.spec
index 55c24405..5b939b5d 100644
--- a/virt-manager.spec
+++ b/virt-manager.spec
@@ -12,7 +12,7 @@ Release: 1%{?dist}
 %global verrel %{version}-%{release}
 
 Summary: Desktop tool for managing virtual machines via libvirt
-License: GPLv2+
+License: GPL-2.0-or-later
 BuildArch: noarch
 URL: https://virt-manager.org/
 Source0: https://virt-manager.org/download/sources/%{name}/%{name}-%{version}.tar.gz
