Subject: fix: default start_folder to None
From: Zahid Kizmaz tech@zahid.rocks Sun May 11 22:41:44 2025 +0200
Date: Tue Jun 10 09:22:10 2025 +0200:
Git: 62f976a61b0363b9e6a0eac1fd2e8553d24d5457


--- a/virtManager/storagebrowse.py
+++ b/virtManager/storagebrowse.py
@@ -181,6 +181,7 @@ class vmmStorageBrowser(vmmGObjectUI):
         data = _BrowseReasonMetadata(self._browse_reason)
         gsettings_key = data.gsettings_key
 
+        start_folder = None
         if gsettings_key:
             start_folder = self.config.get_default_directory(gsettings_key)
 
