Subject: fix resizing of spice clients with guest resizing enabled
From: Florian Ludwig f.ludwig@greyrook.com Wed Oct 9 11:18:53 2019 +0200
Date: Tue Nov 12 11:46:35 2019 -0500:
Git: ae19d6d6ec0b48085ce1c765fb2264e27fb51c63

Fixes 1414718

diff --git a/virtManager/details/console.py b/virtManager/details/console.py
index 8ef8631a..193e79eb 100644
--- a/virtManager/details/console.py
+++ b/virtManager/details/console.py
@@ -374,7 +374,10 @@ class vmmConsolePages(vmmGObjectUI):
         if is_resizeguest:
             # With resize guest, we don't want to maintain aspect ratio,
             # since the guest can resize to arbitrary resolutions.
-            self._viewer.console_set_size_request(req.width, req.height)
+            viewer_alloc = Gdk.Rectangle()
+            viewer_alloc.width = req.width
+            viewer_alloc.height = req.height
+            self._viewer.console_size_allocate(viewer_alloc)
             return
 
         if not is_scale:
