Reference: bnc#869024
Disable graphics on s390x
Index: virt-manager-3.0.0/virtinst/guest.py
===================================================================
--- virt-manager-3.0.0.orig/virtinst/guest.py
+++ virt-manager-3.0.0/virtinst/guest.py
@@ -195,7 +195,10 @@ class Guest(XMLBuilder):
         self.skip_default_channel = False
         self.skip_default_sound = False
         self.skip_default_usbredir = False
-        self.skip_default_graphics = False
+        if self.os.is_s390x():
+            self.skip_default_graphics = True
+        else:
+            self.skip_default_graphics = False
         self.skip_default_rng = False
         self.x86_cpu_default = self.cpu.SPECIAL_MODE_APP_DEFAULT
 
@@ -337,7 +340,7 @@ class Guest(XMLBuilder):
         if not os_support:
             return False
 
-        if self.os.is_x86():
+        if self.os.is_x86() or self.os.is_s390x():
             return True
 
         return False  # pragma: no cover
@@ -854,7 +857,7 @@ class Guest(XMLBuilder):
         self.add_device(dev)
 
     def _add_default_video_device(self):
-        if self.os.is_container():
+        if self.os.is_container() or self.os.is_s390x():
             return
         if self.devices.video:
             return
