References: bsc#1180069

Index: virt-manager-4.2.0/virtinst/guest.py
===================================================================
--- virt-manager-4.2.0.orig/virtinst/guest.py
+++ virt-manager-4.2.0/virtinst/guest.py
@@ -737,6 +737,8 @@ class Guest(XMLBuilder):
             self.type != "kvm"):
             log.warning(  # pragma: no cover
                     "KVM acceleration not available, using '%s'", self.type)
+        if self.os.is_xenpv() and self.os.smbios_mode is not None:
+            raise RuntimeError(_("The --sysinfo flag (smbios) is not supported for Xen PV guests."))
 
     def refresh_machine_type(self):
         """
