Index: virt-manager-2.2.0/tests/clitest.py
===================================================================
--- virt-manager-2.2.0.orig/tests/clitest.py
+++ virt-manager-2.2.0/tests/clitest.py
@@ -300,7 +300,7 @@ class Command(object):
 
     def _run(self):
         conn = None
-        for idx in reversed(range(len(self.argv))):
+        for idx in reversed(list(range(len(self.argv)))):
             if self.argv[idx] == "--connect":
                 conn = utils.URIs.openconn(self.argv[idx + 1])
                 break
Index: virt-manager-2.2.0/tests/test_inject.py
===================================================================
--- virt-manager-2.2.0.orig/tests/test_inject.py
+++ virt-manager-2.2.0/tests/test_inject.py
@@ -15,7 +15,7 @@ FEDORA_URL = "http://dl.fedoraproject.or
 
 (WARN_RHEL5,
  WARN_DEBIAN,
- WARN_FEDORA) = range(1, 4)
+ WARN_FEDORA) = list(range(1, 4))
 
 
 def prompt():
@@ -117,7 +117,7 @@ def _make_tests():
         return lambda s: _test_distro(_d)
 
     idx = 0
-    for dname, dobj in _alldistros.items():
+    for dname, dobj in list(_alldistros.items()):
         idx += 1
         setattr(InjectTests, "testInitrd%.3d_%s" %
                 (idx, dname.replace("-", "_")), _make_check_cb(dobj))
