Index: virt-manager-3.0.0/virtManager/xmleditor.py
===================================================================
--- virt-manager-3.0.0.orig/virtManager/xmleditor.py
+++ virt-manager-3.0.0/virtManager/xmleditor.py
@@ -7,12 +7,12 @@ import gi
 from virtinst import log
 
 # We can use either gtksourceview3 or gtksourceview4
-try:
-    gi.require_version("GtkSource", "4")
-    log.debug("Using GtkSource 4")
-except ValueError:  # pragma: no cover
-    gi.require_version("GtkSource", "3.0")
-    log.debug("Using GtkSource 3.0")
+#try:
+#    gi.require_version("GtkSource", "4")
+#    log.debug("Using GtkSource 4")
+#except ValueError:  # pragma: no cover
+#    gi.require_version("GtkSource", "3.0")
+#    log.debug("Using GtkSource 3.0")
 from gi.repository import GtkSource
 
 from .lib import uiutil
