References: fate#326698 - Add pvh support to virt-manager
At this time support is disabled in this patch.

Index: virt-manager-3.0.0/virtManager/createvm.py
===================================================================
--- virt-manager-3.0.0.orig/virtManager/createvm.py
+++ virt-manager-3.0.0/virtManager/createvm.py
@@ -837,6 +837,9 @@ class vmmCreateVM(vmmGObjectUI):
                     break
             if label is None:
                 continue
+            # xenpvh is currently unsupported
+            if guest.os_type == "xenpvh":
+                continue
 
             # Determine if this is the default given by guest_lookup
             if (gtype == self._capsinfo.os_type and
Index: virt-manager-3.0.0/virtinst/domain/os.py
===================================================================
--- virt-manager-3.0.0.orig/virtinst/domain/os.py
+++ virt-manager-3.0.0/virtinst/domain/os.py
@@ -32,6 +32,8 @@ class DomainOs(XMLBuilder):
         return self.os_type == "hvm"
     def is_xenpv(self):
         return self.os_type in ["xen", "linux"]
+    def is_xenpvh(self):
+        return self.os_type in ["xenpvh", "linux"]
     def is_container(self):
         return self.os_type == "exe"
 
Index: virt-manager-3.0.0/virtinst/guest.py
===================================================================
--- virt-manager-3.0.0.orig/virtinst/guest.py
+++ virt-manager-3.0.0/virtinst/guest.py
@@ -814,7 +814,7 @@ class Guest(XMLBuilder):
 
         usb_tablet = False
         usb_keyboard = False
-        if self.os.is_x86() and not self.os.is_xenpv():
+        if self.os.is_x86() and not self.os.is_xenpv() and not self.os.is_xenpvh():
             usb_tablet = self.osinfo.supports_usbtablet()
         if (self.os.is_arm_machvirt() or
             self.os.is_riscv_virt() or
Index: virt-manager-3.0.0/virtManager/object/domain.py
===================================================================
--- virt-manager-3.0.0.orig/virtManager/object/domain.py
+++ virt-manager-3.0.0/virtManager/object/domain.py
@@ -1223,6 +1223,8 @@ class vmmDomain(vmmLibvirtObject):
         return self.get_xmlobj().os.is_xenpv()
     def is_hvm(self):
         return self.get_xmlobj().os.is_hvm()
+    def is_xenpvh(self):
+        return self.get_xmlobj().os.is_xenpvh()
 
     def get_uuid(self):
         if self._uuid is None:
Index: virt-manager-3.0.0/virtManager/connection.py
===================================================================
--- virt-manager-3.0.0.orig/virtManager/connection.py
+++ virt-manager-3.0.0/virtManager/connection.py
@@ -211,6 +211,8 @@ class vmmConnection(vmmGObject):
                 label = "xen (paravirt)"
             elif gtype == "hvm":
                 label = "xen (fullvirt)"
+            elif gtype == "xenpvh":
+                label = "xen (pvh - Technical Preview)"
         elif domtype == "test":
             if gtype == "xen":
                 label = "test (xen)"
