--- virt-manager-2.1.0/virt-clone.orig	2019-02-04 10:40:20.390102363 -0700
+++ virt-manager-2.1.0/virt-clone	2019-02-04 10:40:41.122360052 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 #
 # Copyright(c) FUJITSU Limited 2007.
 #
--- virt-manager-2.1.0/virt-convert.orig	2019-02-04 10:41:50.275219728 -0700
+++ virt-manager-2.1.0/virt-convert	2019-02-04 10:41:55.563285476 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 #
 # Copyright 2008, 2013, 2014  Red Hat, Inc.
 #
--- virt-manager-2.1.0/virt-install.orig	2019-02-04 10:41:17.258809255 -0700
+++ virt-manager-2.1.0/virt-install	2019-02-04 10:41:22.370872807 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 #
 # Copyright 2005-2014 Red Hat, Inc.
 #
--- virt-manager-2.1.0/virt-manager.orig	2019-02-04 10:41:02.902630789 -0700
+++ virt-manager-2.1.0/virt-manager	2019-02-04 10:41:11.870742273 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 #
 # Copyright (C) 2006, 2014 Red Hat, Inc.
 # Copyright (C) 2006 Daniel P. Berrange <berrange@redhat.com>
--- virt-manager-2.1.0/virt-xml.orig	2019-02-04 10:41:32.923003992 -0700
+++ virt-manager-2.1.0/virt-xml	2019-02-04 10:41:38.803077096 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 #
 # Copyright 2013-2014 Red Hat, Inc.
 #
