From ea99d89082c2473c2e22c848bff4f9f537af3fde Mon Sep 17 00:00:00 2001
From: Zong Wei <wei.zong@intel.com>
Date: Mon, 25 Oct 2021 15:52:32 +0800
Subject: [PATCH] pkg-config: check opencv module version

---
 m4/xcam-utils.m4 | 1 +
 1 file changed, 1 insertion(+)

diff --git a/m4/xcam-utils.m4 b/m4/xcam-utils.m4
index a93c45c6..236fdda8 100644
--- a/m4/xcam-utils.m4
+++ b/m4/xcam-utils.m4
@@ -62,6 +62,7 @@ AC_DEFUN([XCAM_CHECK_OPENCV],
         [
             ocv_version=`opencv_version`
             AS_IF([test -z $ocv_version], [ocv_version=`$PKG_CONFIG --modversion opencv`])
+            AS_IF([test -z $ocv_version], [ocv_version=`$PKG_CONFIG --modversion opencv4`])
             AC_MSG_NOTICE(OpenCV version: $ocv_version)
 
             ocv_module=opencv
