Index: b/src/CMakeLists.txt
===================================================================
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -26,5 +26,5 @@ add_definitions( -DLOCALEDIR=\"${CMAKE_I
 fcitx_add_addon_full(libpinyin DESC
 SOURCES ${fcitx_libpinyin_sources}
 IM_CONFIG pinyin-libpinyin.conf shuangpin-libpinyin.conf zhuyin-libpinyin.conf
-LINK_LIBS ${LIBPINYIN_LIBRARIES} ${LIBINTL_LIBRARIES} ${GLIB2_LIBRARIES} ${DBUS_LIBRARIES}
+LINK_LIBS ${LIBPINYIN_LIBRARIES} ${LIBINTL_LIBRARIES} ${GLIB2_LIBRARIES} ${DBUS_LIBRARIES} ${FCITX4_FCITX_UTILS_LIBRARIES} ${FCITX4_FCITX_CONFIG_LIBRARIES} ${FCITX4_FCITX_LIBRARIES}
 )
