Index: monitoring-plugins-2.3.1/lib/utils_cmd.c
===================================================================
--- monitoring-plugins-2.3.1.orig/lib/utils_cmd.c
+++ monitoring-plugins-2.3.1/lib/utils_cmd.c
@@ -355,10 +355,10 @@ cmd_run_array (char *const *argv, output
 	if ((fd = _cmd_open (argv, pfd_out, pfd_err)) == -1)
 		die (STATE_UNKNOWN, _("Could not open pipe: %s\n"), argv[0]);
 
-	if (out)
-		out->lines = _cmd_fetch_output (pfd_out[0], out, flags);
 	if (err)
 		err->lines = _cmd_fetch_output (pfd_err[0], err, flags);
+	if (out)
+		out->lines = _cmd_fetch_output (pfd_out[0], out, flags);
 
 	return _cmd_close (fd);
 }
