Index: monitoring-plugins-2.4.0/plugins-scripts/check_log.sh
===================================================================
--- monitoring-plugins-2.4.0.orig/plugins-scripts/check_log.sh
+++ monitoring-plugins-2.4.0/plugins-scripts/check_log.sh
@@ -112,23 +112,23 @@ while test -n "$1"; do
             exit "$STATE_UNKNOWN"
             ;;
         -F | --filename)
-            logfile=$2
+            logfile="$2"
             shift 2
             ;;
         -O | --oldlog)
-            oldlog=$2
+            oldlog="$2"
             shift 2
             ;;
         -q | --query)
-            query=$2
+            query="$2"
             shift 2
             ;;
         --exclude)
-            exclude=$2
+            exclude="$2"
             shift 2
             ;;
         -x | --exitstatus)
-            exitstatus=$2
+            exitstatus="$2"
             shift 2
             ;;
         -e | --extended-regex)
