--- monitoring-plugins-2.3.5/plugins-root/Makefile.in.orig	2023-11-28 17:23:06.400019507 -0600
+++ monitoring-plugins-2.3.5/plugins-root/Makefile.in	2023-11-28 17:49:03.832492223 -0600
@@ -1960,8 +1960,6 @@
 	p=$$f; \
 	echo " $(INSTALL_PROGRAM) $$p $(DESTDIR)$(libexecdir)/$$p"; \
 	$(INSTALL_PROGRAM) $$p $(DESTDIR)$(libexecdir)/$$p; \
-	echo " chown root $(DESTDIR)$(libexecdir)/$$p"; \
-	chown root $(DESTDIR)$(libexecdir)/$$p; \
 	echo " chmod $(setuid_root_mode) $(DESTDIR)$(libexecdir)/$$p"; \
 	chmod $(setuid_root_mode) $(DESTDIR)$(libexecdir)/$$p; \
 	done
@@ -2484,7 +2482,7 @@
 	@TMPFILE=$(DESTDIR)$(libexecdir)/.setuid-$$$$; \
 	rm -f $$TMPFILE; \
 	echo > $$TMPFILE; \
-	can_create_suid_root_executable=no; \
+	can_create_suid_root_executable=yes; \
 	chown root $$TMPFILE > /dev/null 2>&1 \
 	  && chmod $(setuid_root_mode) $$TMPFILE > /dev/null 2>&1 \
 	  && can_create_suid_root_executable=yes; \
