Index: m4/netsnmp.m4
===================================================================
--- m4/netsnmp.m4.orig	2025-04-17 08:10:29.018991005 +0200
+++ m4/netsnmp.m4	2025-04-28 00:37:37.429780123 +0200
@@ -175,7 +175,22 @@ session.securityAuthProto = usmHMAC384SH
 		],[
 		AC_MSG_RESULT(no)
 		])
-
+################## monkey patching....
+               dnl Check if MD5 auth is removed
+               AC_MSG_CHECKING(for disabled MD5 auth protocol support)
+               AC_TRY_LINK([
+#include <net-snmp/net-snmp-config.h>
+#include <net-snmp/net-snmp-includes.h>
+               ],[
+struct snmp_session session;
+session.securityAuthProto = usmHMACMD5AuthProtocol;
+               ],[
+               AC_DEFINE(HAVE_NETSNMP_NO_MD5_AUTH, 1, [Define to 1 if MD5 auth protocols is disabled.])
+               AC_MSG_RESULT(yes)
+               ],[
+               AC_MSG_RESULT(no)
+               ])
+################## end monkey patching....
 		dnl Check for AES192/256 protocol support for privacy
 		AC_MSG_CHECKING(for strong AES privacy protocol support)
 		AC_LINK_IFELSE([AC_LANG_PROGRAM([[
Index: src/libs/zbxpoller/checks_snmp.c
===================================================================
--- src/libs/zbxpoller/checks_snmp.c.orig	2025-04-22 11:06:27.189800009 +0200
+++ src/libs/zbxpoller/checks_snmp.c	2025-04-28 00:37:37.430164603 +0200
@@ -705,10 +705,12 @@ static int	zbx_snmpv3_set_auth_protocol(
 
 	switch (snmpv3_authprotocol)
 	{
+#ifdef usmHMACMD5AuthProtocol
 		case ITEM_SNMPV3_AUTHPROTOCOL_MD5:
 			session->securityAuthProto = usmHMACMD5AuthProtocol;
 			session->securityAuthProtoLen = USM_AUTH_PROTO_MD5_LEN;
 			break;
+#endif
 		case ITEM_SNMPV3_AUTHPROTOCOL_SHA1:
 			session->securityAuthProto = usmHMACSHA1AuthProtocol;
 			session->securityAuthProtoLen = USM_AUTH_PROTO_SHA_LEN;
