Index: discount-3.0.0a/configure.inc
===================================================================
--- discount-3.0.0a.orig/configure.inc
+++ discount-3.0.0a/configure.inc
@@ -1456,7 +1456,6 @@ __AC_ELF_LIBRARIAN() {
     if $AC_CC $_VFLAGS -o ngc$$.so ngc$$.c; then
 	USE_SONAME=T
     fi
-    LDCONFIG=`AC_PATH=/sbin:/usr/sbin:/usr/local/sbin acLookFor ldconfig`
 
     if [ "$LDCONFIG" ]; then
 	case `uname -s 2>/dev/null | $AC_UPPERCASE` in
@@ -1501,7 +1500,6 @@ install)$PROG_INSTALL -c \$FULLNAME "\$1
 	$PROG_LN_S -f \$FULLNAME \$1/\$LIBNAME.\$MAJOR
 	$PROG_LN_S -f \$FULLNAME \$1/\$LIBNAME
 EOF
-    test "$LDCONFIG" && echo '	'$LDCONFIG '"$1"' >> librarian.sh
     cat >> librarian.sh << EOF
 	;;
 esac
