From: Arjen de Korte <suse+build@de-korte.org>
Date: 2025-10-28 20:05:15 +0100

boost 1.89 does not have boost_system anymore.

diff -purN a/meson.build b/meson.build
--- a/meson.build	2025-08-20 10:14:44.000000000 +0200
+++ b/meson.build	2026-07-07 00:00:00.000000000 +0000
@@ -192,7 +192,7 @@

 boost_dep = dependency('boost', version: '>=1.69', required: false)
 if not boost_dep.found()
-    boost_dep = dependency('boost', version: '>=1.66', modules: ['system'])
+    boost_dep = dependency('boost', version: '>=1.66', modules: ['system'], required: false)
 endif
 dl_dep = dependency('dl', required: not meson.is_cross_build())
 threads_dep = dependency('threads')
