--- m4/reorganization/yara.m4.orig
+++ m4/reorganization/yara.m4
@@ -6,7 +6,7 @@ enable_yara=$enableval, enable_yara="yes
 
 if test "$enable_yara" = "yes"; then
     AC_DEFINE([HAVE_YARA],1,[yara sources are compiled in])
-    AC_SUBST([HAVE_YARA])
+    AC_SUBST([HAVE_YARA], 1)
 fi
 
 
--- unit_tests/check_common.sh.orig
+++ unit_tests/check_common.sh
@@ -222,6 +222,7 @@ EOF
 	scan_failed clamscan4.log "clamscan has detected spurious VI's"
     fi
 
+if test "x$HAVE_YARA" = "x1"; then
 cat <<EOF >test-db/test.yara
 rule yara_at_offset {strings: \$tar_magic = { 75 73 74 61 72 } condition: \$tar_magic at 257}
 EOF
@@ -249,6 +250,7 @@ EOF
     fi
 
     test_end $1
+fi
 }
 
 # ----------- clamd tests --------------------------------------------------------
--- configure.orig
+++ configure
@@ -28446,6 +28446,7 @@ if test "$enable_yara" = "yes"; then
 
 $as_echo "#define HAVE_YARA 1" >>confdefs.h
 
+    HAVE_YARA=1
 
 fi
 
