Index: screen-message-0.23/Makefile.am
===================================================================
--- screen-message-0.23.orig/Makefile.am	2014-12-24 02:41:16.000000000 +0800
+++ screen-message-0.23/Makefile.am	2017-01-30 13:02:37.217960440 +0800
@@ -3,11 +3,11 @@
 sm_CFLAGS = $(DEPS_CFLAGS)
 sm_LDADD = $(DEPS_LIBS)
 
-execgamesdir= $(prefix)/games
-execgames_PROGRAMS=sm
+bindir= $(prefix)/bin
+bin_PROGRAMS=sm
 
 sm.desktop: sm.desktop.in
-	sed -e s%@BINPATH@%${execgamesdir}% <$< >$@
+	sed -e s%@BINPATH@%${bindir}% <$< >$@
 
 desktopdir = $(datadir)/applications
 desktop_DATA = sm.desktop
