From 058a59814f0be1d89e2a0fa11438c2037112cd7a Mon Sep 17 00:00:00 2001
From: Jeffrey Walton <noloader@gmail.com>
Date: Fri, 19 Jul 2019 00:14:25 -0400
Subject: [PATCH] Fix TCXXFLAGS using openSUSE standard flags (GH #865)

---
 GNUmakefile       | 2 +-
 GNUmakefile-cross | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/GNUmakefile b/GNUmakefile
index 405a26fa..6fee54d3 100755
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -121,7 +121,7 @@ endif
 
 # Strip out -Wall, -Wextra and friends for feature testing
 ifeq ($(DETECT_FEATURES),1)
-  TCXXFLAGS := $(filter-out -Wall -Wextra -Werror% -Wunused -Wconversion -Wp%, $(CXXFLAGS))
+  TCXXFLAGS := $(filter-out -D_FORTIFY_SOURCE=1 -D_FORTIFY_SOURCE=2 -Wall -Wextra -Werror% -Wunused -Wconversion -Wp%, $(CXXFLAGS))
   ifneq ($(strip $(TCXXFLAGS)),)
     $(info Using testing flags: $(TCXXFLAGS))
   endif
diff --git a/GNUmakefile-cross b/GNUmakefile-cross
index c84b399a..7c151c59 100755
--- a/GNUmakefile-cross
+++ b/GNUmakefile-cross
@@ -194,8 +194,7 @@ endif
 
 # Strip out -Wall, -Wextra and friends for feature testing
 ifeq ($(DETECT_FEATURES),1)
-  TCXXFLAGS := $(filter-out -Wall -Wextra -Werror% -Wunused -Wconversion -Wp%, $(CXXFLAGS))
-  ifneq ($(strip $(TCXXFLAGS)),)
+  TCXXFLAGS := $(filter-out -D_FORTIFY_SOURCE=1 -D_FORTIFY_SOURCE=2 -Wall -Wextra -Werror% -Wunused -Wconversion -Wp%, $(CXXFLAGS))
     $(info Using testing flags: $(TCXXFLAGS))
   endif
   #TPROG = TestPrograms/test_cxx.cxx
-- 
2.16.4

