From 417fbd719ab6a40aa35d7dfe2562e56b0e425130 Mon Sep 17 00:00:00 2001
From: Jeffrey Walton <noloader@gmail.com>
Date: Fri, 19 Jul 2019 00:16:23 -0400
Subject: [PATCH] Fix missing if statement Copy/paste error from the regular
 GNUmakefile

---
 GNUmakefile-cross | 1 +
 1 file changed, 1 insertion(+)

diff --git a/GNUmakefile-cross b/GNUmakefile-cross
index 7c151c59..d32139ed 100755
--- a/GNUmakefile-cross
+++ b/GNUmakefile-cross
@@ -195,6 +195,7 @@ endif
 # Strip out -Wall, -Wextra and friends for feature testing
 ifeq ($(DETECT_FEATURES),1)
   TCXXFLAGS := $(filter-out -D_FORTIFY_SOURCE=1 -D_FORTIFY_SOURCE=2 -Wall -Wextra -Werror% -Wunused -Wconversion -Wp%, $(CXXFLAGS))
+  ifneq ($(strip $(TCXXFLAGS)),)
     $(info Using testing flags: $(TCXXFLAGS))
   endif
   #TPROG = TestPrograms/test_cxx.cxx
-- 
2.16.4

