Index: commons-daemon-1.3.4-src/src/native/unix/native/jsvc-unix.c
===================================================================
--- commons-daemon-1.3.4-src.orig/src/native/unix/native/jsvc-unix.c
+++ commons-daemon-1.3.4-src/src/native/unix/native/jsvc-unix.c
@@ -1308,7 +1308,7 @@ static int run_controller(arg_data *args
      * These will be replaced in the child process.
      */
     memset(&act, '\0', sizeof(act));
-    act.sa_handler = controller;
+    act.sa_handler = (void *) controller;
     sigemptyset(&act.sa_mask);
     act.sa_flags = SA_RESTART | SA_NOCLDSTOP | SA_SIGINFO;
 
