--- superhack.c.orig	2015-04-04 12:36:55.000000000 +0200
+++ superhack.c	2015-06-08 18:29:28.574280952 +0200
@@ -13,6 +13,9 @@
 #include <stdio.h>
 #include <ctype.h>
 #include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/socket.h>
 
 static int path[5];
 static int j, k, scratchloc, pies;
@@ -423,6 +426,7 @@
 	    break;
 	}
     }
+    return 0;
 }
 
 /* superhack.c ends here */
