diff -ru scim-uim-0.2.0.orig/src/scim_uim_imengine.cpp scim-uim-0.2.0/src/scim_uim_imengine.cpp
--- scim-uim-0.2.0.orig/src/scim_uim_imengine.cpp	2007-04-10 07:43:59.000000000 +0200
+++ scim-uim-0.2.0/src/scim_uim_imengine.cpp	2007-11-08 14:31:32.000000000 +0100
@@ -32,6 +32,8 @@
   #include <config.h>
 #endif
 
+#include <string.h>
+
 #include <scim.h>
 #include "scim_uim_imengine.h"
 #include <uim/uim-compat-scm.h>
