Index: Expr.h
===================================================================
--- Expr.h.orig	2012-06-29 16:24:40.560239703 +0200
+++ Expr.h	2012-07-02 11:43:59.435836695 +0200
@@ -133,7 +133,7 @@ public:
     typedef struct {
         int  type;        /* type of operation - internal number */
         int  pri;         /* priority of this operation          */
-        char *st;         /* string which cause this operation   */
+        const char *st;   /* string which cause this operation   */
     } table;
 
 private:
Index: Image.cpp
===================================================================
--- Image.cpp.orig	2012-06-29 16:24:40.561239679 +0200
+++ Image.cpp	2012-06-29 16:28:55.092194054 +0200
@@ -60,7 +60,7 @@ static const u_int32_t _code_crspace_off
 const int Image::NVRam::_fsizes[] = { 512*1024, 1048576, 2*1048576, 4*1048576 };
 
 static struct {
-    char           *fname;
+    const char     *fname;
     Image::FORMATS fvalue;
 } formats[] = {
     { "image",   Image::OUT_IMAGE  },
@@ -1118,7 +1118,7 @@ bool Image::write_image(FILE* fp, const
 
 bool Image::get_fw_ver (FwVer v) {
 
-    char* ver_params[] = {
+    const char *ver_params[] = {
         "FWInfo.FW_VER_MAJOR",
         "FWInfo.FW_VER_MINOR",
         "FWInfo.FW_VER_SUBMINOR"
Index: TImage.cpp
===================================================================
--- TImage.cpp.orig	2012-06-29 16:24:40.594238896 +0200
+++ TImage.cpp	2012-07-02 11:41:51.583949021 +0200
@@ -280,8 +280,8 @@ bool TImage::GpSect::CreateGuids (ParamL
     // ??? Do I need to fill this data, or burner will replace it anyway ???
 
     // Get GUIDs and related info
-    char *guid_names[NGUIDS]     = GUID_PARAM_NAMES;
-    char *guid_names_old[NGUIDS] = GUID_PARAM_NAMES_OLD;
+    const char * const guid_names[NGUIDS]     = GUID_PARAM_NAMES;
+    const char * const guid_names_old[NGUIDS] = GUID_PARAM_NAMES_OLD;
 
 
     u_int32_t           guids[NGUIDS];
@@ -427,7 +427,7 @@ bool TImage::GpSect::CreateFile (const s
     // Get the file's extension. Allowed extensions: bin, img.
     //
 
-    char* file_type_str;
+    const char* file_type_str;
     if        (file_type == H_ROM) {
 	file_type_str = "expantion rom";
     } else if (file_type == H_USER_DATA) {
@@ -1166,7 +1166,7 @@ bool TImage::binary_dump(FILE* fp, u_int
 
 bool TImage::get_fw_ver (FwVer v) {
 
-    char* ver_params[] = {
+    const char* ver_params[] = {
 	"FW.Major Rev ID",
 	"FW.Minor Rev ID",
 	"FW.SubMinor rev ID"
