Index: libdc1394-2.2.3/configure.ac
===================================================================
--- libdc1394-2.2.3.orig/configure.ac
+++ libdc1394-2.2.3/configure.ac
@@ -107,21 +107,30 @@ AM_CONDITIONAL(MAKE_EXAMPLES, test x$bui
 
 # check for Xv extensions (necessary for examples/dc1394_multiview)
 # imported from Coriander
+AC_ARG_ENABLE([xv],
+              AS_HELP_STRING([--disable-xv], [Disable the check for Xv extensions]),
+              [test_xv=yes], [test_xv=no])
+
 AC_DEFUN([AC_CHECK_XV],[
-    AC_SUBST(XV_CFLAGS)
-    AC_SUBST(XV_LIBS)
-    AC_MSG_CHECKING(for Xv extensions)
-    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-        #include <X11/Xlib.h>
-        #include <X11/extensions/Xvlib.h>]], [[
-        int main(void) { (void) XvGetPortAttribute(0, 0, 0, 0); return 0; }
-    ]])],[xv=yes],[xv=no])
-    AC_MSG_RESULT($xv)
-    if test x$xv = xyes; then
-        XV_LIBS="-lXv -lXext"
-        XV_CFLAGS=""
-    fi
-    AM_CONDITIONAL(HAVE_XV, test x$xv = xyes)
+       if test x$test_xv != "xno"; then
+            AC_SUBST(XV_CFLAGS)
+            AC_SUBST(XV_LIBS)
+            AC_MSG_CHECKING(for Xv extensions)
+            AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+            #include <X11/Xlib.h>
+            #include <X11/extensions/Xvlib.h>]], [[
+            int main(void) { (void) XvGetPortAttribute(0, 0, 0, 0); return 0; }
+            ]])],[xv=yes],[xv=no]);
+            AC_MSG_RESULT($xv)
+            if test x$xv = xyes; then
+               XV_LIBS="-lXv -lXext"
+               XV_CFLAGS=""
+               AC_DEFINE(HAVE_XV,1,[defined if XV video overlay is available])
+            fi
+        else
+            xv=no
+        fi
+        AM_CONDITIONAL(HAVE_XV, test x$xv = xyes)
 ])
 AC_CHECK_XV
 
