Index: scripts/apache2-httpd.include.conf
===================================================================
--- scripts/apache2-httpd.include.conf.orig
+++ scripts/apache2-httpd.include.conf
@@ -2,13 +2,13 @@
 # added for OTRS (http://otrs.org/)
 # --
 
-ScriptAlias /otrs/ "/opt/otrs/bin/cgi-bin/"
-Alias /otrs-web/ "/opt/otrs/var/httpd/htdocs/"
+ScriptAlias /otrs/ "@OTRS_ROOT@/bin/cgi-bin/"
+Alias /otrs-web/ "@OTRS_ROOT@/var/httpd/htdocs/"
 
 <IfModule mod_perl.c>
 
     # Setup environment and preload modules
-    Perlrequire /opt/otrs/scripts/apache2-perl-startup.pl
+    Perlrequire @OTRS_ROOT@/scripts/apache2-perl-startup.pl
 
     # Reload Perl modules when changed on disk
     PerlModule Apache2::Reload
@@ -16,7 +16,8 @@ Alias /otrs-web/ "/opt/otrs/var/httpd/ht
 
     # general mod_perl2 options
     <Location /otrs>
-#        ErrorDocument 403 /otrs/customer.pl
+
+#       ErrorDocument 403 /otrs/customer.pl
         ErrorDocument 403 /otrs/index.pl
         SetHandler  perl-script
         PerlResponseHandler ModPerl::Registry
@@ -24,19 +25,21 @@ Alias /otrs-web/ "/opt/otrs/var/httpd/ht
         PerlOptions +ParseHeaders
         PerlOptions +SetupEnv
 
-        <IfModule mod_version.c>
-            <IfVersion < 2.4>
-                Order allow,deny
-                Allow from all
-            </IfVersion>
-            <IfVersion >= 2.4>
-                Require all granted
-            </IfVersion>
-        </IfModule>
-        <IfModule !mod_version.c>
+        <IfVersion < 2.4>
             Order allow,deny
             Allow from all
-        </IfModule>
+        </IfVersion>
+
+        <IfVersion >= 2.4>
+            <IfModule !mod_access_compat.c>
+                Require all granted
+            </IfModule>
+            <IfModule mod_access_compat.c>
+                Order allow,deny
+                Allow from all
+            </IfModule>
+        </IfVersion>
+
     </Location>
 
     # mod_perl2 options for GenericInterface
@@ -46,23 +49,25 @@ Alias /otrs-web/ "/opt/otrs/var/httpd/ht
 
 </IfModule>
 
-<Directory "/opt/otrs/bin/cgi-bin/">
+<Directory "@OTRS_ROOT@/bin/cgi-bin/">
+
     AllowOverride None
     Options +ExecCGI -Includes
 
-    <IfModule mod_version.c>
-        <IfVersion < 2.4>
-            Order allow,deny
-            Allow from all
-        </IfVersion>
-        <IfVersion >= 2.4>
-            Require all granted
-        </IfVersion>
-    </IfModule>
-    <IfModule !mod_version.c>
+    <IfVersion < 2.4>
         Order allow,deny
         Allow from all
-    </IfModule>
+    </IfVersion>
+
+    <IfVersion >= 2.4>
+        <IfModule !mod_access_compat.c>
+           Require all granted
+       </IfModule>
+       <IfModule mod_access_compat.c>
+           Order allow,deny
+           Allow from all
+       </IfModule>
+    </IfVersion>
 
     <IfModule mod_filter.c>
         <IfModule mod_deflate.c>
@@ -72,22 +77,52 @@ Alias /otrs-web/ "/opt/otrs/var/httpd/ht
 
 </Directory>
 
-<Directory "/opt/otrs/var/httpd/htdocs/">
+<Directory "@OTRS_ROOT@/var/httpd/htdocs/js/thirdparty/ckeditor-4.4.8/">
+    #
+    # Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+    # For licensing, see LICENSE.html or http://ckeditor.com/license
+    #
+
+    #
+    # On some specific Linux installations you could face problems with Firefox.
+    # It could give you errors when loading the editor saying that some illegal
+    # characters were found (three strange chars in the beginning of the file).
+    # This could happen if you map the .js or .css files to PHP, for example.
+    #
+    # Those characters are the Byte Order Mask (BOM) of the Unicode encoded files.
+    # All FCKeditor files are Unicode encoded.
+    #
+
+    AddType application/x-javascript .js
+    AddType text/css .css
+
+    #
+    # If PHP is mapped to handle XML files, you could have some issues. The
+    # following will disable it.
+    #
+
+    AddType text/xml .xml
+</Directory>
+
+<Directory "@OTRS_ROOT@/var/httpd/htdocs/">
+
+    Options +FollowSymlinks
     AllowOverride None
 
-    <IfModule mod_version.c>
-        <IfVersion < 2.4>
-            Order allow,deny
-            Allow from all
-        </IfVersion>
-        <IfVersion >= 2.4>
-            Require all granted
-        </IfVersion>
-    </IfModule>
-    <IfModule !mod_version.c>
+    <IfVersion < 2.4>
         Order allow,deny
         Allow from all
-    </IfModule>
+    </IfVersion>
+
+    <IfVersion >= 2.4>
+        <IfModule !mod_access_compat.c>
+            Require all granted
+        </IfModule>
+        <IfModule mod_access_compat.c>
+            Order allow,deny
+            Allow from all
+        </IfModule>
+    </IfVersion>
 
     <IfModule mod_filter.c>
         <IfModule mod_deflate.c>
@@ -123,33 +158,35 @@ Alias /otrs-web/ "/opt/otrs/var/httpd/ht
 # </Location>
 
 <IfModule mod_headers.c>
+
     # Cache css-cache for 30 days
-    <Directory "/opt/otrs/var/httpd/htdocs/skins/*/*/css-cache">
+    <Directory "@OTRS_ROOT@/var/httpd/htdocs/skins/*/*/css-cache">
         <FilesMatch "\.(css|CSS)$">
             Header set Cache-Control "max-age=2592000 must-revalidate"
         </FilesMatch>
     </Directory>
 
     # Cache css thirdparty for 4 hours, including icon fonts
-    <Directory "/opt/otrs/var/httpd/htdocs/skins/*/*/css/thirdparty">
+    <Directory "@OTRS_ROOT@/var/httpd/htdocs/skins/*/*/css/thirdparty">
         <FilesMatch "\.(css|CSS|woff|svg)$">
             Header set Cache-Control "max-age=14400 must-revalidate"
         </FilesMatch>
     </Directory>
 
     # Cache js-cache for 30 days
-    <Directory "/opt/otrs/var/httpd/htdocs/js/js-cache">
+    <Directory "@OTRS_ROOT@/var/httpd/htdocs/js/js-cache">
         <FilesMatch "\.(js|JS)$">
             Header set Cache-Control "max-age=2592000 must-revalidate"
         </FilesMatch>
     </Directory>
 
     # Cache js thirdparty for 4 hours
-    <Directory "/opt/otrs/var/httpd/htdocs/js/thirdparty/">
+    <Directory "@OTRS_ROOT@/var/httpd/htdocs/js/thirdparty/">
         <FilesMatch "\.(js|JS)$">
             Header set Cache-Control "max-age=14400 must-revalidate"
         </FilesMatch>
     </Directory>
+
 </IfModule>
 
 # Limit the number of requests per child to avoid excessive memory usage
