Index: giada-1.3.0/CMakeLists.txt
===================================================================
--- giada-1.3.0.orig/CMakeLists.txt
+++ giada-1.3.0/CMakeLists.txt
@@ -552,7 +552,7 @@ list(APPEND INCLUDE_DIRS ${LIBRARY_RTMID
 include (FetchContent)
 FetchContent_Declare(
 	FLTK
-	SOURCE_DIR ${CMAKE_SOURCE_DIR}/src/deps/fltk)
+	SOURCE_DIR ${CMAKE_SOURCE_DIR}/src/deps/fltk EXCLUDE_FROM_ALL)
 
 set(FLTK_BUILD_GL ON CACHE BOOL "FLTK: enable OpenGL support")
 set(FLTK_BUILD_TEST OFF CACHE BOOL "FLTK: don't build tests")
@@ -754,7 +754,7 @@ endif()
 # Extra parameters for audio plug-ins support.
 # ------------------------------------------------------------------------------
 
-add_subdirectory(${CMAKE_SOURCE_DIR}/src/deps/juce)
+add_subdirectory(${CMAKE_SOURCE_DIR}/src/deps/juce EXCLUDE_FROM_ALL)
 
 list(APPEND LIBRARIES 
 	juce::juce_audio_utils
