diff -up giada-0.20.1/CMakeLists.txt.orig giada-0.20.1/CMakeLists.txt
--- giada-0.20.1/CMakeLists.txt.orig	2022-02-20 11:04:38.000000000 +0100
+++ giada-0.20.1/CMakeLists.txt	2022-03-16 21:37:45.040389627 +0100
@@ -312,7 +312,7 @@ endif()
 
 set(FLTK_SKIP_FLUID TRUE)  # Don't search for FLTK's fluid
 set(FLTK_SKIP_OPENGL TRUE) # Don't search for FLTK's OpenGL
-find_package(FLTK REQUIRED NO_MODULE)
+find_package(FLTK REQUIRED)
 list(APPEND LIBRARIES fltk)
 list(APPEND INCLUDE_DIRS ${FLTK_INCLUDE_DIRS})
 message("FLTK library found in " ${FLTK_DIR})
