Index: kbuild-0.1.9998svn3110/src/kmk/configure.in
===================================================================
--- kbuild-0.1.9998svn3110.orig/src/kmk/configure.in
+++ kbuild-0.1.9998svn3110/src/kmk/configure.in
@@ -360,10 +360,9 @@ AC_CACHE_VAL(make_cv_sys_gnu_glob, [
 #include <glob.h>
 #include <fnmatch.h>
 
-#define GLOB_INTERFACE_VERSION 1
 #if !defined _LIBC && defined __GNU_LIBRARY__ && __GNU_LIBRARY__ > 1
 # include <gnu-versions.h>
-# if _GNU_GLOB_INTERFACE_VERSION == GLOB_INTERFACE_VERSION
+# if _GNU_GLOB_INTERFACE_VERSION == 1 || _GNU_GLOB_INTERFACE_VERSION == 2
    gnu glob
 # endif
 #endif
