---

---
 src/kmk/dir.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: kbuild-0.1.9998svn3110/src/kmk/dir.c
===================================================================
--- kbuild-0.1.9998svn3110.orig/src/kmk/dir.c
+++ kbuild-0.1.9998svn3110/src/kmk/dir.c
@@ -1340,8 +1340,11 @@ void print_dir_stats (void)
 
 /* Hooks for globbing.  */
 
+#if defined(KMK) && !defined(__GLIBC__)
+# include "glob/glob.h"
+#else
 #include <glob.h>
-
+#endif
 /* Structure describing state of iterating through a directory hash table.  */
 
 struct dirstream
