diff -ru wxedid-0.0.20.orig/src/CEA_class.cpp wxedid-0.0.20/src/CEA_class.cpp
--- wxedid-0.0.20.orig/src/CEA_class.cpp	2020-06-03 11:34:20.000000000 +0200
+++ wxedid-0.0.20/src/CEA_class.cpp	2020-06-05 15:14:06.096511581 +0200
@@ -1017,7 +1017,7 @@
 
    type_id   = ID_VSD;
 
-   retU = init_fields(&fields[0], inst, fcount, orflags, Name, Desc);
+   retU = init_fields(&fields[0], inst, fcount, orflags, Name, Desc.ToUTF8());
    return retU;
 }
 
