commit e66e254e0accb6479d447548492a1db322473cbb
Author: Boris Faure <billiob@gmail.com>
Date:   Thu Jan 13 20:37:57 2022 +0100

    colors: fix typo

diff --git a/src/bin/colors.c b/src/bin/colors.c
index 01037eb1..5001741c 100644
--- a/src/bin/colors.c
+++ b/src/bin/colors.c
@@ -748,7 +748,7 @@ color_scheme_list(void)
 
    /* Search homedir first, so color classes there get used */
    snprintf(buf, sizeof(buf) - 1,
-            "%s/terminology/colorchemes",
+            "%s/terminology/colorschemes",
             efreet_config_home_get());
    search_paths = eina_list_append(search_paths, eina_stringshare_add(buf));
    snprintf(buf, sizeof(buf) - 1,
