From 24dff9f710a02ceb5bb63049ab6dfd591321ca3a Mon Sep 17 00:00:00 2001
From: Kyle Altendorf <sda@fstab.net>
Date: Tue, 15 Mar 2022 09:26:16 -0400
Subject: [PATCH] specify just pytester as the plugin

---
 testing/conftest.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/testing/conftest.py b/testing/conftest.py
index 72dffd2..c8ed3b7 100644
--- a/testing/conftest.py
+++ b/testing/conftest.py
@@ -2,7 +2,7 @@
 import pytest_twisted
 
 
-pytest_plugins = "_pytest.pytester"
+pytest_plugins = "pytester"
 
 
 @pytest.hookimpl(tryfirst=True)
