diff -up csound-6.18.1/include/plugin.h.orig csound-6.18.1/include/plugin.h
--- csound-6.18.1/include/plugin.h.orig	2022-11-23 20:53:55.000000000 +0100
+++ csound-6.18.1/include/plugin.h	2025-08-19 12:59:50.211218300 +0200
@@ -1002,7 +1002,7 @@ template <std::size_t N> struct InPlug :
   /** check if this opcode runs at perf time
   */
   bool is_perf() {
-      return this->opaddr ? true : false;
+      return this->opadr ? true : false;
   }
 
 };
