Index: datadiff-2.2.0/datadiff/tests/test_datadiff.py
===================================================================
--- datadiff-2.2.0.orig/datadiff/tests/test_datadiff.py
+++ datadiff-2.2.0/datadiff/tests/test_datadiff.py
@@ -377,8 +377,12 @@ def test_DataDiff_change_type():
 
 
 def test_unhashable_type():
+    class UnhashableType:
+        def __eq__(self, other):
+            return False
+
     a = []
-    b = [slice(1)]
+    b = [UnhashableType()]
     pytest.raises(NotHashable, diff, a, b)
 
 
