--- a/data/default.layout
+++ b/data/default.layout
@@ -18,6 +18,14 @@ locked=true
 object-type=applet
 applet-iid=NotificationAreaAppletFactory::NotificationArea
 toplevel-id=top
+position=20
+panel-right-stick=true
+locked=true
+
+[Object volume-control]
+object-type=applet
+applet-iid=GvcAppletFactory::GvcApplet
+toplevel-id=top
 position=10
 panel-right-stick=true
 locked=true
--- a/data/linuxmint.layout
+++ b/data/linuxmint.layout
@@ -28,6 +28,14 @@ locked=true
 object-type=applet
 applet-iid=NotificationAreaAppletFactory::NotificationArea
 toplevel-id=bottom
+position=20
+panel-right-stick=true
+locked=true
+
+[Object volume-control]
+object-type=applet
+applet-iid=GvcAppletFactory::GvcApplet
+toplevel-id=bottom
 position=10
 panel-right-stick=true
 locked=true
--- a/data/mageia.layout
+++ b/data/mageia.layout
@@ -39,6 +39,14 @@ locked=true
 object-type=applet
 applet-iid=NotificationAreaAppletFactory::NotificationArea
 toplevel-id=top
+position=20
+panel-right-stick=true
+locked=true
+
+[Object volume-control]
+object-type=applet
+applet-iid=GvcAppletFactory::GvcApplet
+toplevel-id=top
 position=10
 panel-right-stick=true
 locked=true
--- a/data/opensuse.layout
+++ b/data/opensuse.layout
@@ -28,6 +28,14 @@ locked=true
 object-type=applet
 applet-iid=NotificationAreaAppletFactory::NotificationArea
 toplevel-id=bottom
+position=40
+panel-right-stick=true
+locked=true
+
+[Object volume-control]
+object-type=applet
+applet-iid=GvcAppletFactory::GvcApplet
+toplevel-id=bottom
 position=30
 panel-right-stick=true
 locked=true
--- a/data/ubuntu.layout
+++ b/data/ubuntu.layout
@@ -26,6 +26,14 @@ menu-path=applications:/
 object-type=applet
 applet-iid=NotificationAreaAppletFactory::NotificationArea
 toplevel-id=top
+position=40
+panel-right-stick=true
+locked=true
+
+[Object volume-control]
+object-type=applet
+applet-iid=GvcAppletFactory::GvcApplet
+toplevel-id=top
 position=30
 panel-right-stick=true
 locked=true
