diff --git a/src/playlist/playlisttab.cpp b/src/playlist/playlisttab.cpp
index 01f1290..6ae454b 100644
--- a/src/playlist/playlisttab.cpp
+++ b/src/playlist/playlisttab.cpp
@@ -25,6 +25,7 @@
 #include <QAction>
 #include <QBoxLayout>
 #include <QFileDialog>
+#include <QToolBar>
 #include <QKeyEvent>
 #include <QListView>
 #include <QMenu>
@@ -446,33 +447,35 @@ PlaylistTab::PlaylistTab(QMenu *menu, KActionCollection *collection, MediaWidget
 	QBoxLayout *sideLayout = new QVBoxLayout(widget);
 	sideLayout->setMargin(0);
 
+	QToolBar *ToolBarLayout= new QToolBar();
 	QBoxLayout *boxLayout = new QHBoxLayout();
 
 	QToolButton *toolButton = new QToolButton(widget);
 	toolButton->setDefaultAction(newAction);
 	toolButton->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
-	boxLayout->addWidget(toolButton);
+	ToolBarLayout->addWidget(toolButton);
 
 	toolButton = new QToolButton(widget);
 	toolButton->setDefaultAction(renameAction);
 	toolButton->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
-	boxLayout->addWidget(toolButton);
+	ToolBarLayout->addWidget(toolButton);
 
 	toolButton = new QToolButton(widget);
 	toolButton->setDefaultAction(removePlaylistAction);
 	toolButton->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
-	boxLayout->addWidget(toolButton);
+	ToolBarLayout->addWidget(toolButton);
 
 	toolButton = new QToolButton(widget);
 	toolButton->setDefaultAction(savePlaylistAction);
 	toolButton->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
-	boxLayout->addWidget(toolButton);
+	ToolBarLayout->addWidget(toolButton);
 
 	toolButton = new QToolButton(widget);
 	toolButton->setDefaultAction(savePlaylistAsAction);
 	toolButton->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
-	boxLayout->addWidget(toolButton);
+	ToolBarLayout->addWidget(toolButton);
 
+	boxLayout->addWidget(ToolBarLayout);
 	boxLayout->addStretch();
 	sideLayout->addLayout(boxLayout);
 
@@ -499,28 +502,30 @@ PlaylistTab::PlaylistTab(QMenu *menu, KActionCollection *collection, MediaWidget
 	sideLayout = new QVBoxLayout(widget);
 	sideLayout->setMargin(0);
 
+	ToolBarLayout= new QToolBar();
 	boxLayout = new QHBoxLayout();
 
 	toolButton = new QToolButton(widget);
 	toolButton->setDefaultAction(repeatAction);
 	toolButton->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
-	boxLayout->addWidget(toolButton);
+	ToolBarLayout->addWidget(toolButton);
 
 	toolButton = new QToolButton(widget);
 	toolButton->setDefaultAction(randomAction);
 	toolButton->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
-	boxLayout->addWidget(toolButton);
+	ToolBarLayout->addWidget(toolButton);
 
 	toolButton = new QToolButton(widget);
 	toolButton->setDefaultAction(removeTrackAction);
 	toolButton->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
-	boxLayout->addWidget(toolButton);
+	ToolBarLayout->addWidget(toolButton);
 
 	toolButton = new QToolButton(widget);
 	toolButton->setDefaultAction(clearAction);
 	toolButton->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
-	boxLayout->addWidget(toolButton);
+	ToolBarLayout->addWidget(toolButton);
 
+	boxLayout->addWidget(ToolBarLayout);
 	boxLayout->addStretch();
 	sideLayout->addLayout(boxLayout);
 
@@ -615,6 +620,7 @@ void PlaylistTab::createFileWidget()
 	fileWidget->setFilter(MediaWidget::extensionFilter());
 	fileWidget->setMode(KFile::Files | KFile::ExistingOnly);
 	fileWidgetSplitter->setStretchFactor(1, 1);
+	fileWidget->setMinimumWidth(40);
 
 	// KFileWidget creates a QUrlComboBox without layout (!), which steals the focus:
 	//     kDebug() << QApplication::focusWidget();
