@@ -0,0 +1,24 @@
Index: b/libuuu/libcomm.h
===================================================================
--- a/libuuu/libcomm.h
+++ b/libuuu/libcomm.h
@@ -34,6 +34,7 @@
 #include <locale>
 #include <cctype>
 #include <algorithm>
+#include <cstdint>
 #include "libuuu.h"
 
 #pragma once
Index: b/uuu/buildincmd.h
===================================================================
--- a/uuu/buildincmd.h
+++ b/uuu/buildincmd.h
@@ -35,6 +35,7 @@
 #include <map>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 extern const char * g_vt_boldwhite;
 extern const char * g_vt_default;
