diff -rub pillow_heif-1.0.0/pillow_heif/_pillow_heif.c pillow_heif-1.0.0-patched/pillow_heif/_pillow_heif.c
--- pillow_heif-1.0.0/pillow_heif/_pillow_heif.c	2025-06-29 08:10:31.000000000 +0200
+++ pillow_heif-1.0.0-patched/pillow_heif/_pillow_heif.c	2025-07-10 08:17:59.174582650 +0200
@@ -2,6 +2,7 @@
 
 #include "Python.h"
 #include "libheif/heif.h"
+#include "libheif/heif_properties.h"
 #include "_ph_postprocess.h"
 
 /* =========== Common stuff ======== */
