Index: firefox-68.6.0/js/src/tests/jstests.list
===================================================================
--- firefox-68.6.0.orig/js/src/tests/jstests.list
+++ firefox-68.6.0/js/src/tests/jstests.list
@@ -542,3 +542,11 @@ skip script test262/intl402/RelativeTime
 
 # Privateuse-only tags no longer allowed: "x-u-foo"
 skip script test262/intl402/Intl/getCanonicalLocales/weird-cases.js
+
+# Skip tests expected to fail on i586
+skip script test262/language/types/number/S8.5_A2.2.js
+skip script test262/language/types/number/S8.5_A2.1.js
+skip script non262/Date/timeclip.js
+
+#Skip tests expected to fail on ppc64
+skip script test262/built-ins/TypedArray/prototype/set/typedarray-arg-set-values-same-buffer-other-type.js
