--- poi/src/java/org/apache/poi/hssf/dev/BiffViewer.java	2022-03-20 17:52:02.256845307 +0100
+++ poi/src/java/org/apache/poi/hssf/dev/BiffViewer.java	2022-03-20 18:02:01.660566286 +0100
@@ -21,6 +21,7 @@
  */
 package org.apache.poi.hssf.dev;
 
+import org.apache.poi.hssf.record.Record;
 import org.apache.poi.hssf.record.*;
 import org.apache.poi.poifs.filesystem.POIFSFileSystem;
 import org.apache.poi.util.HexDump;
--- poi/src/java/org/apache/poi/hssf/dev/EFHSSF.java	2022-03-20 17:52:02.256845307 +0100
+++ poi/src/java/org/apache/poi/hssf/dev/EFHSSF.java	2022-03-20 18:02:56.344905755 +0100
@@ -23,6 +23,7 @@
 import java.io.IOException;
 
 import org.apache.poi.poifs.filesystem.POIFSFileSystem;
+import org.apache.poi.hssf.record.Record;
 import org.apache.poi.hssf.record.*;
 import org.apache.poi.hssf.eventmodel.*;
 import org.apache.poi.hssf.eventusermodel.*;
--- poi/src/java/org/apache/poi/hssf/dev/FormulaViewer.java	2022-03-20 17:52:02.256845307 +0100
+++ poi/src/java/org/apache/poi/hssf/dev/FormulaViewer.java	2022-03-20 18:07:46.182707479 +0100
@@ -35,6 +35,7 @@
 import org.apache.poi.poifs.filesystem.POIFSFileSystem;
 import org.apache.poi.util.LittleEndian;
 import org.apache.poi.util.HexDump;
+import org.apache.poi.hssf.record.Record;
 import org.apache.poi.hssf.record.*;
 import org.apache.poi.hssf.record.formula.*;
 import org.apache.poi.hssf.model.*;
--- poi/src/java/org/apache/poi/hssf/dev/HSSF.java	2022-03-20 17:52:02.256845307 +0100
+++ poi/src/java/org/apache/poi/hssf/dev/HSSF.java	2022-03-20 18:08:49.751103009 +0100
@@ -23,6 +23,7 @@
 import java.io.FileOutputStream;
 
 import org.apache.poi.poifs.filesystem.POIFSFileSystem;
+import org.apache.poi.hssf.record.Record;
 import org.apache.poi.hssf.record.*;
 import org.apache.poi.hssf.usermodel.*;
 import org.apache.poi.hssf.util.*;
--- poi/src/java/org/apache/poi/hssf/model/Sheet.java	2022-03-20 17:52:02.256845307 +0100
+++ poi/src/java/org/apache/poi/hssf/model/Sheet.java	2022-03-20 18:00:46.556100056 +0100
@@ -18,6 +18,7 @@
 
 package org.apache.poi.hssf.model;
 
+import org.apache.poi.hssf.record.Record;
 import org.apache.poi.hssf.record.*;
 import org.apache.poi.hssf.record.aggregates.FormulaRecordAggregate;
 import org.apache.poi.hssf.record.aggregates.RowRecordsAggregate;
--- poi/src/java/org/apache/poi/hssf/model/Workbook.java	2022-03-20 17:52:02.256845307 +0100
+++ poi/src/java/org/apache/poi/hssf/model/Workbook.java	2022-03-20 17:58:25.175222395 +0100
@@ -17,6 +17,7 @@
 
 package org.apache.poi.hssf.model;
 
+import org.apache.poi.hssf.record.Record;
 import org.apache.poi.hssf.record.*;
 import org.apache.poi.hssf.util.HSSFColor;
 import org.apache.poi.hssf.util.SheetReferences;
--- poi/src/java/org/apache/poi/hssf/record/aggregates/FormulaRecordAggregate.java	2022-03-20 17:52:02.260845332 +0100
+++ poi/src/java/org/apache/poi/hssf/record/aggregates/FormulaRecordAggregate.java	2022-03-20 18:05:39.121916889 +0100
@@ -17,6 +17,7 @@
 
 package org.apache.poi.hssf.record.aggregates;
 
+import org.apache.poi.hssf.record.Record;
 import org.apache.poi.hssf.record.*;
 
 /**
--- poi/src/java/org/apache/poi/hssf/record/aggregates/ValueRecordsAggregate.java	2022-03-20 17:52:02.260845332 +0100
+++ poi/src/java/org/apache/poi/hssf/record/aggregates/ValueRecordsAggregate.java	2022-03-20 18:03:46.929219772 +0100
@@ -18,6 +18,7 @@
 
 package org.apache.poi.hssf.record.aggregates;
 
+import org.apache.poi.hssf.record.Record;
 import org.apache.poi.hssf.record.*;
 
 import java.util.Iterator;
--- poi/src/java/org/apache/poi/hssf/usermodel/HSSFSheet.java	2022-03-20 17:52:02.264845357 +0100
+++ poi/src/java/org/apache/poi/hssf/usermodel/HSSFSheet.java	2022-03-20 18:06:38.454286062 +0100
@@ -24,6 +24,7 @@
 import org.apache.poi.ddf.EscherRecord;
 import org.apache.poi.hssf.model.Sheet;
 import org.apache.poi.hssf.model.Workbook;
+import org.apache.poi.hssf.record.Record;
 import org.apache.poi.hssf.record.*;
 import org.apache.poi.hssf.util.Region;
 import org.apache.poi.util.POILogFactory;
--- poi/src/testcases/org/apache/poi/hssf/record/aggregates/TestValueRecordsAggregate.java	2022-03-20 17:52:02.280845456 +0100
+++ poi/src/testcases/org/apache/poi/hssf/record/aggregates/TestValueRecordsAggregate.java	2022-03-20 18:11:26.180076295 +0100
@@ -20,6 +20,7 @@
 import java.util.Iterator;
 import java.util.List;
 
+import org.apache.poi.hssf.record.Record;
 import org.apache.poi.hssf.record.*;
 import org.apache.poi.util.HexDump;
 import junit.framework.TestCase;
