diff -ru cedilla-0.6.orig/system.lisp cedilla-0.6/system.lisp
--- cedilla-0.6.orig/system.lisp	2006-02-15 21:52:09.000000000 +0100
+++ cedilla-0.6/system.lisp	2006-07-20 13:32:50.000000000 +0200
@@ -46,6 +46,7 @@
              (make-pathname :defaults (pathname file)
                             :directory (pathname (dir dir))))
            (install (file dir out &optional executable (name file))
+	     (format out "mkdir -p \"$TARGET~A\"~%" dir)
              (format out "cp -f '~A' \"$TARGET\"'~A'~%" file (target name dir))
              (when executable
                (format out "chmod +x \"$TARGET\"'~A'~%" (target name dir)))))
