Index: aioquic-0.9.21/tests/test_tls.py
===================================================================
--- aioquic-0.9.21.orig/tests/test_tls.py
+++ aioquic-0.9.21/tests/test_tls.py
@@ -443,7 +443,7 @@ class ContextTest(TestCase):
 
         with self.assertRaises(tls.AlertBadCertificate) as cm:
             self._handshake(client, server)
-        self.assertEqual(str(cm.exception), "unable to get local issuer certificate")
+            self.assertEqual(str(cm.exception), "unable to get local issuer certificate")
 
     def test_handshake_with_certificate_no_verify(self):
         client = self.create_client(cafile=None, verify_mode=ssl.CERT_NONE)
@@ -1282,9 +1282,9 @@ class VerifyCertificateTest(TestCase):
             # fail
             with self.assertRaises(tls.AlertBadCertificate) as cm:
                 verify_certificate(certificate=certificate, server_name="localhost")
-            self.assertEqual(
-                str(cm.exception), "unable to get local issuer certificate"
-            )
+                self.assertEqual(
+                    str(cm.exception), "unable to get local issuer certificate"
+                )
 
             # ok
             verify_certificate(
