Index: lldb-8.0.0rc3.src/source/lldb.cpp
===================================================================
--- lldb-8.0.0rc3.src.orig/source/lldb.cpp
+++ lldb-8.0.0rc3.src/source/lldb.cpp
@@ -23,11 +23,7 @@ using namespace lldb_private;
 #endif
 
 static const char *GetLLDBRevision() {
-#ifdef LLDB_REVISION
   return LLDB_REVISION;
-#else
-  return NULL;
-#endif
 }
 
 static const char *GetLLDBRepository() {
Index: lldb-8.0.0rc3.src/source/Core/IOHandler.cpp
===================================================================
--- lldb-8.0.0rc3.src.orig/source/Core/IOHandler.cpp
+++ lldb-8.0.0rc3.src/source/Core/IOHandler.cpp
@@ -10,8 +10,8 @@
 #include "lldb/Core/IOHandler.h"
 
 #ifndef LLDB_DISABLE_CURSES
-#include <curses.h>
-#include <panel.h>
+#include <ncurses/curses.h>
+#include <ncurses/panel.h>
 #endif
 
 #if defined(__APPLE__)
Index: lldb-8.0.0rc3.src/source/Host/CMakeLists.txt
===================================================================
--- lldb-8.0.0rc3.src.orig/source/Host/CMakeLists.txt
+++ lldb-8.0.0rc3.src/source/Host/CMakeLists.txt
@@ -143,6 +143,8 @@ endif()
 set(EXTRA_LIBS)
 if (CMAKE_SYSTEM_NAME MATCHES "NetBSD")
   list(APPEND EXTRA_LIBS kvm)
+elseif (CMAKE_SYSTEM_NAME MATCHES "Linux")
+  list(APPEND EXTRA_LIBS dl pthread)
 endif ()
 if (APPLE)
   list(APPEND EXTRA_LIBS xml2)
