diff -urEbwB junit-r4.11/src/test/java/org/junit/tests/running/core/MainRunner.java junit-r4.11/src/test/java/org/junit/tests/running/core/MainRunner.java
--- junit-r4.11/src/test/java/org/junit/tests/running/core/MainRunner.java	2012-11-13 21:10:09.000000000 +0100
+++ junit-r4.11/src/test/java/org/junit/tests/running/core/MainRunner.java	2017-12-18 10:32:05.822083468 +0100
@@ -40,11 +40,6 @@
         }
 
         @Override
-        public boolean getInCheck() {
-            return (originalSecurityManager != null) && originalSecurityManager.getInCheck();
-        }
-
-        @Override
         public Object getSecurityContext() {
             return (originalSecurityManager == null) ? super.getSecurityContext() : originalSecurityManager.getSecurityContext();
         }
