---
 xvkbd.c |   16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

Index: xvkbd-4.0/xvkbd.c
===================================================================
--- xvkbd-4.0.orig/xvkbd.c
+++ xvkbd-4.0/xvkbd.c
@@ -2578,7 +2578,9 @@ static void MenuSelected(Widget w, char
   Widget form;
   
   if (strcmp(key, "man") == 0) {
-    if (!appres.secure) system(appres.show_manual_command);
+    if (!appres.secure) {
+      if (system(appres.show_manual_command) == -1) fprintf(stderr, "Error call: %s\n", appres.show_manual_command);
+    }
   } else if (strcmp(key, "about") == 0) {
     if (about_panel == None) {
       about_panel = XtVaCreatePopupShell("about_panel", transientShellWidgetClass,
@@ -2776,7 +2778,9 @@ static void KeyPressed(Widget w, char *k
     if (appres.debug) fprintf(stderr, "  Assigned string: %s\n", value);
     if (value[0] == '!') {
       if (appres.debug) fprintf(stderr, "  Launching: %s\n", value + 1);
-      if (!appres.secure) system(value + 1);
+      if (!appres.secure) {
+        if (system(value + 1) == -1) fprintf(stderr, "Could not Launch: %s\n", value + 1);
+      }
     } else {
       if (value[0] == '\\') value = value + 1;
       if (appres.debug) fprintf(stderr, "  Sending: %s\n", value);
@@ -2956,9 +2960,13 @@ static void KeyPressed(Widget w, char *k
       if ((cur_shift & ControlMask) && (cur_shift & alt_mask)) {
         if (strstr(XServerVendor(dpy), "XFree86") != NULL) {
           if (strcmp(key1, "KP_Add") == 0) {
-            if (!appres.secure) system("xvidtune -next");
+            if (!appres.secure) {
+              if (system("xvidtune -next") == -1) fprintf(stderr, "Error call system(xvidtune -next)\n");
+            }
           } else if (strcmp(key1, "KP_Subtract") == 0) {
-            if (!appres.secure) system("xvidtune -prev");
+            if (!appres.secure) {
+	      if (system("xvidtune -prev") == -1) fprintf(stderr, "Error call system(xvidtune -prev)\n");
+	    }
           }
         }
       }
