From: Bruce Rogers <brogers@suse.com>
Date: Wed, 2 Oct 2019 07:28:04 -0600
Subject: tests: Disable some block tests for now

For a very long time we've experienced intermittent failure of some
block tests. There doesn't seem to be an obvious cause. Given that we
now build qemu using multibuild, the qemu-testsuite failure is now more
problematic to just ignore. Until we can address these failures, which
seem to be tied to the build environment in some way, disable the tests
which are randomly failing.

Currently, this patch is based on these failures seen on 02OCT2019 on
v4.1.0 qemu:
on x86_64: 130, 177, 204, 205, 246, 250
on i586: 161, 250
on ppc64le: 161, 182, 250, 250
on aarch64: 235
(other architectures not evaluated at this time)

Signed-off-by: Bruce Rogers <brogers@suse.com>
---
 tests/qemu-iotests/group | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/tests/qemu-iotests/group b/tests/qemu-iotests/group
index f13e5f2e23417b94bd84d820bcf6..d0da6ec0c8daba541b39ae920f68 100644
--- a/tests/qemu-iotests/group
+++ b/tests/qemu-iotests/group
@@ -151,7 +151,7 @@
 127 rw backing quick
 128 rw quick
 129 rw quick
-130 rw auto quick
+#DISABLE FOR NOW 130 rw auto quick
 131 rw quick
 132 rw quick
 133 auto quick
@@ -182,7 +182,7 @@
 158 rw auto quick
 159 rw auto quick
 160 rw quick
-161 rw auto quick
+#DISABLE FOR NOW 161 rw auto quick
 162 quick
 163 rw
 165 rw quick
@@ -194,11 +194,11 @@
 174 auto
 175 quick
 176 rw auto backing
-177 rw auto quick
+#DISABLE FOR NOW 177 rw auto quick
 178 img
 179 rw auto quick
 181 rw migration
-182 rw quick
+#DISABLE FOR NOW 182 rw quick
 183 rw migration
 184 rw auto quick
 185 rw
@@ -219,8 +219,8 @@
 201 rw migration
 202 rw quick
 203 rw migration
-204 rw quick
-205 rw quick
+#DISABLE FOR NOW 204 rw quick
+#DISABLE FOR NOW 205 rw quick
 206 rw
 207 rw
 208 rw quick
@@ -249,7 +249,7 @@
 232 quick
 233 quick
 234 quick migration
-235 quick
+#DISABLE FOR NOW 235 quick
 236 quick
 237 rw quick
 238 quick
@@ -260,11 +260,11 @@
 243 rw quick
 244 rw auto quick
 245 rw
-246 rw quick
+#DISABLE FOR NOW 246 rw quick
 247 rw quick
 248 rw quick
 249 rw auto quick
-250 rw quick
+#DISABLE FOR NOW 250 rw quick
 251 rw auto quick
 252 rw auto backing quick
 253 rw quick
