From 316c086e79d66b62c0c4bc66229ee894e4fdb7d1 Mon Sep 17 00:00:00 2001
From: Hayaki Saito <saitoha@me.com>
Date: Tue, 12 Aug 2025 21:30:24 +0900
Subject: [PATCH] Fix a heap-buffer-overflow problem reported by @err2zero

--- a/src/encoder.c
+++ b/src/encoder.c
@@ -744,9 +744,9 @@ sixel_debug_print_palette(
     fprintf(stderr, "palette:\n");
     for (i = 0; i < sixel_dither_get_num_of_palette_colors(dither); ++i) {
         fprintf(stderr, "%d: #%02x%02x%02x\n", i,
+                palette[i * 3 + 0],
                 palette[i * 3 + 1],
-                palette[i * 3 + 2],
-                palette[i * 3 + 3]);
+                palette[i * 3 + 2]);
     }
 }
