diff -Nur woff2-1.0.2/CMakeLists.txt new/CMakeLists.txt
--- woff2-1.0.2/CMakeLists.txt	2017-11-13 19:31:28.000000000 +0100
+++ new/CMakeLists.txt	2023-01-06 12:19:53.666418912 +0100
@@ -260,12 +260,10 @@
   LIBRARIES woff2enc)
 
 # Installation
-if (NOT BUILD_SHARED_LIBS)
-  install(
-    TARGETS woff2_decompress woff2_compress woff2_info
-    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
-  )
-endif()
+install(
+  TARGETS woff2_decompress woff2_compress woff2_info
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+)
 
 install(
   TARGETS woff2common woff2dec woff2enc
