Index: libofa.pc.in
===================================================================
--- libofa.pc.in.orig	2006-05-10 19:58:51.000000000 +0200
+++ libofa.pc.in	2007-11-06 07:42:53.000000000 +0100
@@ -7,6 +7,7 @@ Name: libofa
 Description: The Open Fingerprint Architecture Library
 URL: http://www.musicdns.org/
 Version: @VERSION@
-Requires: fftw3
-Libs: -L${libdir} -lofa -lexpat -lm
+Requires.private: fftw3
+Libs: -L${libdir} -lofa
+Libs.private: -lexpat -lm
 Cflags: -I${includedir}
