rewritten upstream - probably reproducible

Index: qpid-proton-0.17.0/proton-c/bindings/python/setup.py.in
===================================================================
--- qpid-proton-0.17.0.orig/proton-c/bindings/python/setup.py.in
+++ qpid-proton-0.17.0/proton-c/bindings/python/setup.py.in
@@ -147,7 +147,7 @@ class Configure(build_ext):
                        'extra', 'message', 'reactor', 'messenger', 'handlers',
                        'platform', 'reactor/io/posix']:
 
-            sources.extend(glob.glob(os.path.join(proton_src, subdir, '*.c')))
+            sources.extend(sorted(glob.glob(os.path.join(proton_src, subdir, '*.c'))))
 
         sources.extend(filter(lambda x: not x.endswith('dump.c'),
                        glob.iglob(os.path.join(proton_src, '*.c'))))
