From 7a4ab7ac5fb9a044fae9da0e3bdda1bfe0ecf6af Mon Sep 17 00:00:00 2001
From: =?utf8?q?Tom=C3=A1=C5=A1=20Chv=C3=A1tal?= <tomas.chvatal@gmail.com>
Date: Wed, 28 Feb 2018 19:09:14 +0100
Subject: [PATCH 3/5] Add content of lftp-completion.patch

---
 src/complete.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/complete.cc b/src/complete.cc
index 02e3ce6c..cfeaef28 100644
--- a/src/complete.cc
+++ b/src/complete.cc
@@ -480,6 +480,9 @@ static completion_type cmd_completion_type(const char *cmd,int start)
       w=skip_word(w);
    }
 
+   if (!strcmp(buf, "edit"))
+	   return REMOTE_FILE;
+
    if(!strcmp(buf,"get")
    || !strcmp(buf,"pget")
    || !strcmp(buf,"get1"))
-- 
2.16.2

