diff -ru libmysofa-0.7_/CMakeLists.txt libmysofa-0.7/CMakeLists.txt
--- libmysofa-0.7_/CMakeLists.txt	2019-03-31 17:16:24.000000000 +0200
+++ libmysofa-0.7/CMakeLists.txt	2019-04-02 21:18:09.989815090 +0200
@@ -67,5 +67,5 @@
 )
 install(FILES
   "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc"
-  DESTINATION "lib/pkgconfig"
+  DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
 )
diff -ru libmysofa-0.7_/libmysofa.pc.cmake libmysofa-0.7/libmysofa.pc.cmake
--- libmysofa-0.7_/libmysofa.pc.cmake	2019-03-31 17:16:24.000000000 +0200
+++ libmysofa-0.7/libmysofa.pc.cmake	2019-04-02 21:27:18.581456912 +0200
@@ -2,9 +2,8 @@
 Description: @CPACK_PACKAGE_DESCRIPTION@
 Version: @PROJECT_VERSION@
 Requires: @PKG_CONFIG_REQUIRES@
-prefix=@CMAKE_INSTALL_PREFIX@
-includedir=${prefix}/include
-libdir=${prefix}/lib
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 Libs: -L${libdir} -lmysofa
 Cflags: -I${includedir}
 
