Index: pytest_check-2.6.0/pyproject.toml
===================================================================
--- pytest_check-2.6.0.orig/pyproject.toml
+++ pytest_check-2.6.0/pyproject.toml
@@ -15,8 +15,6 @@ classifiers = [
     'Programming Language :: Python :: 3.10',
     'Programming Language :: Python :: 3.11',
     'Programming Language :: Python :: 3.12',
-    'Programming Language :: Python :: 3.13',
-    'Programming Language :: Python :: 3.14',
     'Topic :: Software Development :: Testing',
     'Topic :: Utilities',
 ]
