* Signedness fix for 0.9.8 version (denis@fateyev.com)
--- a/mimetic/codec/base64.cxx	2016-03-03 07:05:58.587129195 +0600
+++ b/mimetic/codec/base64.cxx	2016-03-03 07:06:32.213128743 +0600
@@ -8,12 +8,12 @@
 
 using namespace mimetic;
 
-const char Base64::sEncTable[] = 
+const signed char Base64::sEncTable[] = 
     "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
     "abcdefghijklmnopqrstuvwxyz"
     "0123456789+/=";
 
-const char Base64::sDecTable[] = {
+const signed char Base64::sDecTable[] = {
         -1,-1,-1,-1,-1,-1,-1,-1,-1,-1,
         -1,-1,-1,-1,-1,-1,-1,-1,-1,-1,
         -1,-1,-1,-1,-1,-1,-1,-1,-1,-1,
--- a/mimetic/codec/base64.h	2016-03-03 07:13:26.405123140 +0600
+++ b/mimetic/codec/base64.h	2016-03-03 07:14:34.629121732 +0600
@@ -19,8 +19,8 @@
     enum { LF = 0xA, CR = 0xD, NL = '\n' };
     enum { default_maxlen = 76 };
     enum { eq_sign = 100 };
-    static const char sEncTable[];
-    static const char sDecTable[];
+    static const signed char sEncTable[];
+    static const signed char sDecTable[];
     static const int sDecTableSz;
 public:
     class Encoder; class Decoder;
