From 7ebbba9d7766e20786a6f0de728c60ba64cabe05 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Klaus=20K=C3=A4mpf?= <kkaempf@suse.de>
Date: Wed, 27 Dec 2017 08:34:28 +0100
Subject: [PATCH 3/6] Fix nf2ff load

---
 matlab/CalcNF2FF.m | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/matlab/CalcNF2FF.m b/matlab/CalcNF2FF.m
index c80b52e4a9ff..d782248821a4 100644
--- a/matlab/CalcNF2FF.m
+++ b/matlab/CalcNF2FF.m
@@ -99,9 +99,7 @@ m_filename = mfilename('fullpath');
 dir_name = fileparts( m_filename );
 
 if isunix
-    nf2ff_bin = searchBinary('nf2ff', ...
-    {[dir_name filesep '..' filesep 'nf2ff' filesep], ...
-     [dir_name filesep '..' filesep '..' filesep '..' filesep 'bin' filesep]}, 0);
+    nf2ff_bin = 'nf2ff';
 else
     nf2ff_bin = searchBinary('nf2ff.exe',[dir_name filesep '..' filesep], 0);
 end
-- 
2.15.1

