From 0cf33c4bf04596e368978e663aa4a2ea42289651 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Sun, 21 Nov 2021 19:14:37 +0100
Subject: [PATCH] Consider VTK_PYTHON_SITE_PACKAGES_SUFFIX for Python Metadata
 files

When building wheels, VTK_PYTHON_SITE_PACKAGES_SUFFIX is ".", so in this
case nothing is changed. For all other builds, e.g. setup.py should be
located in the same directory as e.g. vtk.py, otherwise setup.py fails.
---
 CMake/vtkPythonMetadata.cmake    | 8 ++++----
 CMake/vtkWheelFinalization.cmake | 2 +-
 CMakeLists.txt                   | 2 +-
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/CMake/vtkPythonMetadata.cmake b/CMake/vtkPythonMetadata.cmake
index 44fb591321..f88f9e8aaa 100644
--- a/CMake/vtkPythonMetadata.cmake
+++ b/CMake/vtkPythonMetadata.cmake
@@ -1,17 +1,17 @@
 configure_file(
   "${CMAKE_CURRENT_LIST_DIR}/setup.py.in"
-  "${CMAKE_BINARY_DIR}/setup.py"
+  "${CMAKE_BINARY_DIR}/${VTK_PYTHON_SITE_PACKAGES_SUFFIX}/setup.py"
   @ONLY)
 configure_file(
   "${CMAKE_CURRENT_LIST_DIR}/MANIFEST.in.in"
-  "${CMAKE_BINARY_DIR}/MANIFEST.in"
+  "${CMAKE_BINARY_DIR}/${VTK_PYTHON_SITE_PACKAGES_SUFFIX}/MANIFEST.in"
   @ONLY)
 configure_file(
   "${CMAKE_SOURCE_DIR}/Copyright.txt"
-  "${CMAKE_BINARY_DIR}/LICENSE"
+  "${CMAKE_BINARY_DIR}/${VTK_PYTHON_SITE_PACKAGES_SUFFIX}/LICENSE"
   COPYONLY)
 configure_file(
   "${CMAKE_SOURCE_DIR}/README.md"
-  "${CMAKE_BINARY_DIR}/README.md"
+  "${CMAKE_BINARY_DIR}/${VTK_PYTHON_SITE_PACKAGES_SUFFIX}/README.md"
   COPYONLY)
 
diff --git a/CMake/vtkWheelFinalization.cmake b/CMake/vtkWheelFinalization.cmake
index 23c17fd977..269b31e7d0 100644
--- a/CMake/vtkWheelFinalization.cmake
+++ b/CMake/vtkWheelFinalization.cmake
@@ -62,5 +62,5 @@ foreach (vtk_feature IN LISTS vtk_features)
   string(APPEND vtk_feature_entries
     "    '${vtk_feature}': [],\n")
 endforeach ()
-file(WRITE "${CMAKE_BINARY_DIR}/vtk_features.py"
+file(WRITE "${CMAKE_BINARY_DIR}/${VTK_PYTHON_SITE_PACKAGES_SUFFIX}/vtk_features.py"
   "FEATURES = {\n${vtk_feature_entries}}\n")
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 90c90f9d46..03f5836a0f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -450,15 +450,15 @@ if (VTK_WRAP_PYTHON)
     list(REMOVE_DUPLICATES vtk_required_python_modules)
   endif ()
   string(REPLACE ";" "\n" vtk_required_python_modules "${vtk_required_python_modules}")
-  file(WRITE "${CMAKE_BINARY_DIR}/requirements.txt"
+  file(WRITE "${CMAKE_BINARY_DIR}/${VTK_PYTHON_SITE_PACKAGES_SUFFIX}/requirements.txt"
     "${vtk_required_python_modules}\n")
   get_property(vtk_web_python_modules GLOBAL
     PROPERTY  vtk_web_python_modules)
   if (vtk_web_python_modules)
     list(REMOVE_DUPLICATES vtk_web_python_modules)
   endif ()
   string(REPLACE ";" "\n" vtk_web_python_modules "${vtk_web_python_modules}")
-  file(WRITE "${CMAKE_BINARY_DIR}/requirements_web.txt"
+  file(WRITE "${CMAKE_BINARY_DIR}/${VTK_PYTHON_SITE_PACKAGES_SUFFIX}/requirements_web.txt"
     "${vtk_web_python_modules}\n")
   get_property(vtk_soabi GLOBAL
     PROPERTY  _vtk_python_soabi)
-- 
2.33.1

