--- a/Rendering/CellGrid/vtkDGOpenGLRenderer.cxx_orig	2023-12-28 06:22:40.696501865 +0100
+++ b/Rendering/CellGrid/vtkDGOpenGLRenderer.cxx	2023-12-28 06:23:24.633470522 +0100
@@ -1046,11 +1046,13 @@
   }
 
   // III. Render draw
+#ifndef GL_ES_VERSION_3_0
   if (renderer->GetUseImageBasedLighting() && renderer->GetEnvironmentTexture())
   {
     vtkOpenGLState* ostate = oglRenWin->GetState();
     ostate->vtkglEnable(GL_TEXTURE_CUBE_MAP_SEAMLESS);
   }
+#endif
   //      a. Update shaders
   state->CellBO.VAO->Bind();
   // state->LastBoundBO = &state->CellBO; // We only bind the one...
