From ac01ac54fed5a719002792f8458e9a82493ca14f Mon Sep 17 00:00:00 2001
From: Shawn W Dunn <sfalken@opensuse.org>
Date: Wed, 15 Oct 2025 07:00:36 -0700
Subject: [PATCH] fix for Qt 6.10

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8e4113e..2408b84 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,9 +20,11 @@ set(SHELLQT_MINIMUM_VERSION "6.0.0")
 set(QTXDG_MINIMUM_VERSION "4.2.0")
 set(LXQTBT_MINIMUM_VERSION "2.2.0")
 
+find_package(Qt6GuiPrivate ${QT_MINIMUM_VERSION} REQUIRED)
 find_package(Qt6Network ${QT_MINIMUM_VERSION} REQUIRED)
 find_package(Qt6Widgets ${QT_MINIMUM_VERSION} REQUIRED)
 find_package(Qt6WaylandClient ${QT_MINIMUM_VERSION} REQUIRED)
+find_package(Qt6WaylandClientPrivate ${QT_MINIMUM_VERSION} REQUIRED)
 find_package(Qt6LinguistTools  ${QT_MINIMUM_VERSION} REQUIRED)
 find_package(LayerShellQt ${SHELLQT_MINIMUM_VERSION} REQUIRED)
 find_package(KF6WindowSystem ${KF6_MINIMUM_VERSION} REQUIRED)
-- 
2.51.0

