From 175c72d97d073a64714cebef05bd37f0221f94fa Mon Sep 17 00:00:00 2001
From: Elliott Sales de Andrade <quantum.analyst@gmail.com>
Date: Mon, 22 Jul 2019 02:03:20 -0400
Subject: [PATCH] Fix tests with Octave 5.

---
 test_octave_kernel.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test_octave_kernel.py b/test_octave_kernel.py
index 5ef93d3..750d1a8 100644
--- a/test_octave_kernel.py
+++ b/test_octave_kernel.py
@@ -19,8 +19,8 @@ class OctaveKernelTests(jkt.KernelTests):
 
     completion_samples = [
         {
-            'text': 'one',
-            'matches': {'ones', 'onenormest'},
+            'text': 'acos',
+            'matches': {'acos', 'acosd', 'acosh'},
         },
     ]
 
