Index: xpra-4.0/setup.py
===================================================================
--- xpra-4.0.orig/setup.py
+++ xpra-4.0/setup.py
@@ -1477,7 +1477,7 @@ else:
         add_data_files("%s/man1" % man_path,  ["man/xpra.1", "man/xpra_launcher.1"])
         add_data_files("share/applications",  glob.glob("xdg/*.desktop"))
         add_data_files("share/mime/packages", ["xdg/application-x-xpraconfig.xml"])
-        add_data_files("share/icons",         ["xdg/xpra.png", "xdg/xpra-mdns.png", "xdg/xpra-shadow.png"])
+        add_data_files("share/pixmaps",         ["xdg/xpra.png", "xdg/xpra-mdns.png", "xdg/xpra-shadow.png"])
         add_data_files("share/metainfo",      ["xdg/xpra.appdata.xml"])
 
     #here, we override build and install so we can
@@ -1553,10 +1553,10 @@ else:
             if pam_ENABLED:
                 copytodir("etc/pam.d/xpra", "/etc/pam.d")
 
-            systemd_dir = "/lib/systemd/system"
+            systemd_dir = "__UNITDIR__"
             if service_ENABLED:
                 #Linux init service:
-                if os.path.exists("/bin/systemctl"):
+                if os.path.exists("/usr/bin/systemctl"):
                     if sd_listen_ENABLED:
                         copytodir("service/xpra.service", systemd_dir)
                     else:
@@ -1564,7 +1564,9 @@ else:
                 else:
                     copytodir("service/xpra", "/etc/init.d")
                 if os.path.exists("/etc/sysconfig"):
-                    copytodir("etc/sysconfig/xpra", "/etc/sysconfig")
+                    shutil.copy("etc/sysconfig/xpra", "etc/sysconfig/sysconfig.xpra")
+                    os.chmod("etc/sysconfig/sysconfig.xpra", 0o644)
+                    copytodir("etc/sysconfig/sysconfig.xpra", "__FILLUPDIR__")
                 elif os.path.exists("/etc/default"):
                     copytodir("etc/sysconfig/xpra", "/etc/default")
             if sd_listen_ENABLED:
@@ -1676,7 +1678,6 @@ if scripts_ENABLED:
 toggle_modules(WIN32, "xpra/scripts/win32_service")
 
 if data_ENABLED:
-    add_data_files(share_xpra,                      ["README", "COPYING"])
     add_data_files(share_xpra,                      ["bell.wav"])
     add_data_files("%s/http-headers" % share_xpra,   glob.glob("http-headers/*"))
     add_data_files("%s/icons" % share_xpra,          glob.glob("icons/*png"))
