Index: xpra-4.2/setup.py
===================================================================
--- xpra-4.2.orig/setup.py
+++ xpra-4.2/setup.py
@@ -1527,7 +1527,7 @@ else:
             add_data_files("%s/xpra/" % libexec, libexec_scripts)
     if data_ENABLED:
         man_path = "share/man"
-        icons_dir = "icons"
+        icons_dir = "pixmaps"
         if OPENBSD or FREEBSD:
             man_path = "man"
             icons_dir = "pixmaps"
@@ -1623,7 +1623,7 @@ else:
             if pam_ENABLED:
                 copytodir("fs/etc/pam.d/xpra", "/etc/pam.d")
 
-            systemd_dir = "/lib/systemd/system"
+            systemd_dir = "__UNITDIR__"
             if service_ENABLED:
                 #Linux init service:
                 subs = {}
@@ -1631,15 +1631,15 @@ else:
                     cdir = "/etc/sysconfig"
                 elif is_Debian() or is_Ubuntu():
                     cdir = "/etc/default"
-                elif os.path.exists("/etc/sysconfig"):
-                    cdir = "/etc/sysconfig"
                 else:
-                    cdir = "/etc/default"
-                copytodir("fs/etc/sysconfig/xpra", cdir)
+                    cdir = "__FILLUPDIR__"
+                shutil.copy("fs/etc/sysconfig/xpra", "fs/etc/sysconfig/sysconfig.xpra")
+                os.chmod("fs/etc/sysconfig/sysconfig.xpra", 0o644)
+                copytodir("fs/etc/sysconfig/sysconfig.xpra", cdir)
                 if cdir!="/etc/sysconfig":
                     #also replace the reference to it in the service file below
                     subs[b"/etc/sysconfig"] = cdir.encode()
-                if os.path.exists("/bin/systemctl"):
+                if os.path.exists("/usr/bin/systemctl"):
                     if sd_listen_ENABLED:
                         copytodir("fs/lib/systemd/system/xpra.service", systemd_dir,
                                   subs=subs)
@@ -1767,7 +1767,6 @@ if scripts_ENABLED:
 toggle_modules(WIN32, "xpra/scripts/win32_service")
 
 if data_ENABLED:
-    add_data_files(share_xpra,                      ["README.md", "COPYING"])
     add_data_files(share_xpra,                      ["fs/share/xpra/bell.wav"])
     ICONS = glob.glob("fs/share/xpra/icons/*.png")
     if OSX:
