diff -U 3 -dHrN -- a/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.h b/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.h
--- a/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.h	2020-09-24 15:23:00.000000000 +0200
+++ b/Source/MediaStorageAndFileFormat/gdcmImageChangePhotometricInterpretation.h	2021-06-04 09:01:27.948756274 +0200
@@ -16,6 +16,7 @@
 
 #include "gdcmImageToImageFilter.h"
 #include "gdcmPhotometricInterpretation.h"
+#include <limits>
 
 namespace gdcm
 {
diff -U 3 -dHrN -- a/Source/MediaStorageAndFileFormat/gdcmImage.cxx b/Source/MediaStorageAndFileFormat/gdcmImage.cxx
--- a/Source/MediaStorageAndFileFormat/gdcmImage.cxx	2020-09-24 15:23:00.000000000 +0200
+++ b/Source/MediaStorageAndFileFormat/gdcmImage.cxx	2021-06-04 08:57:06.857815911 +0200
@@ -20,6 +20,7 @@
 #include "gdcmFragment.h"
 
 #include <iostream>
+#include <limits>
 
 namespace gdcm
 {
