--- ./Testing/Source/Common/Cxx/TestString2.cxx.orig	2019-02-26 22:01:26.738230841 +0000
+++ ./Testing/Source/Common/Cxx/TestString2.cxx	2019-02-26 22:59:07.751826144 +0000
@@ -16,7 +16,6 @@
 #include <iostream>
 
 #include <string.h> // strlen
-#include <stdio.h> // EOF
 
 int TestString2(int , char *[])
 {
@@ -26,24 +25,24 @@ int TestString2(int , char *[])
   gdcm::String<> s2 = "coucou!";
   std::cout << s2 << " -> " << s2.size() << std::endl;
 
-  gdcm::String<EOF,64,0> s3 = "coucou";
+  gdcm::String<'\0',64,0> s3 = "coucou";
   std::cout << s3.c_str() << " -> " << s3.size() << std::endl;
 
-  gdcm::String<EOF,64,0> s4 = "coucou!";
+  gdcm::String<'\0',64,0> s4 = "coucou!";
   std::cout << s4.c_str() << " -> " << s4.size() << std::endl;
 
   const char *s = "coucou!";
-  gdcm::String<EOF,64,0> s5( s, strlen(s) );
+  gdcm::String<'\0',64,0> s5( s, strlen(s) );
   std::cout << s5.c_str() << " -> " << s5.size() << std::endl;
 
   std::string ss = "coucou!";
-  gdcm::String<EOF,64,0> s6( ss );
+  gdcm::String<'\0',64,0> s6( ss );
   std::cout << s6.c_str() << " -> " << s6.size() << std::endl;
 
-  gdcm::String<EOF,64,0> s7( ss, 1, 5 );
+  gdcm::String<'\0',64,0> s7( ss, 1, 5 );
   std::cout << s7.c_str() << " -> " << s7.size() << std::endl;
 
-  gdcm::String<EOF,64,0> s8( ss, 1, 6 );
+  gdcm::String<'\0',64,0> s8( ss, 1, 6 );
   std::cout << s8.c_str() << " -> " << s8.size() << std::endl;
 
   return 0;
