From 171d0cab39579541e8d167b31acd93d0286556e2 Mon Sep 17 00:00:00 2001
From: Robert Munteanu <rombert@apache.org>
Date: Tue, 4 Jan 2022 14:37:07 +0100
Subject: [PATCH] Remove shebang from non-executable files

---
 linodecli/__init__.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/linodecli/__init__.py b/linodecli/__init__.py
index 3af9a36..8702e39 100755
--- a/linodecli/__init__.py
+++ b/linodecli/__init__.py
@@ -1,4 +1,3 @@
-#!/usr/local/bin/python3
 from __future__ import print_function
 
 import argparse
-- 
2.34.1

