Index: e16-1.0.24/configure.ac
===================================================================
--- e16-1.0.24.orig/configure.ac
+++ e16-1.0.24/configure.ac
@@ -261,6 +261,7 @@ AC_ARG_ENABLE(pango,
   enable_pango=yes)
 if test "x$enable_pango" = "xyes"; then
   PKG_CHECK_MODULES(PANGO, pangoxft gobject-2.0, AC_DEFINE(USE_PANGO, 1, [pango-xft support]), enable_pango=no)
+  PKG_CHECK_MODULES(GOBJECT2, gobject-2.0, AC_DEFINE(USE_PANGO, 1, [pango-xft support]), enable_pango=no)
 fi
 AM_CONDITIONAL(USE_LIBPANGO, test "x$enable_pango" = "xyes")
 
Index: e16-1.0.24/src/Makefile.am
===================================================================
--- e16-1.0.24.orig/src/Makefile.am
+++ e16-1.0.24/src/Makefile.am
@@ -134,7 +134,8 @@ e16_CPPFLAGS = \
 	$(XI_CFLAGS) \
 	$(XRANDR_CFLAGS) \
 	$(XPRESENT_CFLAGS) \
-	$(X_CFLAGS)
+	$(X_CFLAGS) \
+	$(GOBJECT2_CFLAGS)
 
 e16_LDADD = \
 	$(LTLIBINTL)		\
@@ -160,6 +161,7 @@ e16_LDADD = \
 	$(DBUS_LIBS)		\
 	$(MODULE_LIBS)		\
 	$(CLOCK_LIBS)		\
+	$(GOBJECT2_LIBS) \
 	-lX11 -lm
 
 SNDLDR_CFLAGS = $(AUDIOFILE_CFLAGS) $(SNDFILE_CFLAGS)
@@ -233,8 +235,8 @@ endif
 if USE_LIBPANGO
 LIBFNT_PANGO = libfont_pango.la
 libfont_pango_la_SOURCES  = text_pango.c
-libfont_pango_la_CFLAGS   = $(PANGO_CFLAGS)
-libfont_pango_la_LIBADD   = $(PANGO_LIBS) libfont_xft.la
+libfont_pango_la_CFLAGS   = $(PANGO_CFLAGS) $(GOBJECT2_CFLAGS)
+libfont_pango_la_LIBADD   = $(PANGO_LIBS) $(GOBJECT2_LIBS) libfont_xft.la
 libfont_pango_la_LDFLAGS  = -module -avoid-version
 endif
 
