From 79f02581982773c7d0ab54570a944e35f9e4ecd3 Mon Sep 17 00:00:00 2001
From: Christophe Giboudeaux <christophe@krop.fr>
Date: Sat, 26 May 2018 14:34:29 +0200
Subject: [PATCH] Fix build.

---
 src/src.pro | 21 +--------------------
 1 file changed, 1 insertion(+), 20 deletions(-)

diff --git a/src/src.pro b/src/src.pro
index ae4222a..fb56346 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -108,30 +108,11 @@ unix:!macx: {
     HEADERS += dbus_interface.h
     SOURCES += dbus_interface.cpp
     CONFIG += dbus
-    LIBS += ../lib/libebook/libebook.a
+    LIBS += ../lib/libebook/libebook.a -lchm -lzip
     POST_TARGETDEPS += ../lib/libebook/libebook.a
 }
 
-greaterThan(QT_MAJOR_VERSION, 4) {
-    # Qt 5
-    greaterThan(QT_MINOR_VERSION, 5) {
-        # Qt 5.6+
-        error("You use Qt5.6+ - QWebEngine is not yet suitable for kchmviewer and is not supported")
-        QT += webengine webenginewidgets
-        DEFINES += USE_WEBENGINE
-        SOURCES += viewwindow_webengine.cpp dataprovider_qwebengine.cpp
-        HEADERS += dataprovider_qwebengine.h viewwindow_webengine.h
-    } else {
-        # Qt 5.0-5.5
-        QT += webkit webkitwidgets
-        DEFINES += USE_WEBKIT
-        SOURCES += viewwindow_webkit.cpp dataprovider_qwebkit.cpp
-        HEADERS += dataprovider_qwebkit.h viewwindow_webkit.h
-    }
-} else {
-    message("Qt4 is not supported anymore, please do not report any errors")
     QT += webkit webkitwidgets
     DEFINES += USE_WEBKIT
     SOURCES += viewwindow_webkit.cpp dataprovider_qwebkit.cpp
     HEADERS += dataprovider_qwebkit.h viewwindow_webkit.h
-}
-- 
2.17.0

