diff -Nru azure-storage-file-1.4.0.orig/setup.py azure-storage-file-1.4.0/setup.py
--- azure-storage-file-1.4.0.orig/setup.py	2019-04-02 13:23:22.348692702 +0200
+++ azure-storage-file-1.4.0/setup.py	2019-04-02 13:23:31.424779470 +0200
@@ -74,7 +74,4 @@
         'azure-common>=1.1.5',
         'azure-storage-common>=1.4'
     ],
-    extras_require={
-        ":python_version<'3.0'": ['futures'],
-    },
 )
