Index: hashlink-1.14/CMakeLists.txt
===================================================================
--- hashlink-1.14.orig/CMakeLists.txt
+++ hashlink-1.14/CMakeLists.txt
@@ -181,7 +181,7 @@ if(INTERPRETER_ENABLED)
     )
 
     if (UNIX AND NOT APPLE)
-        set_target_properties(hl PROPERTIES INSTALL_RPATH "$ORIGIN;${CMAKE_INSTALL_PREFIX}/lib")
+        set_target_properties(hl PROPERTIES INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
     endif()
 
     target_link_libraries(hl libhl)
