diff -upr django-request-1.6.1.orig/tests/test_command.py django-request-1.6.1/tests/test_command.py
--- django-request-1.6.1.orig/tests/test_command.py	2022-05-11 13:48:09.679479134 +0200
+++ django-request-1.6.1/tests/test_command.py	2022-05-11 13:48:09.683479158 +0200
@@ -1,7 +1,7 @@
 from datetime import timedelta
 from io import StringIO
 
-import mock
+from unittest import mock
 from django.core.management.base import CommandError
 from django.test import TestCase
 from django.utils.timezone import now
diff -upr django-request-1.6.1.orig/tests/test_middlewares.py django-request-1.6.1/tests/test_middlewares.py
--- django-request-1.6.1.orig/tests/test_middlewares.py	2022-05-11 13:48:09.679479134 +0200
+++ django-request-1.6.1/tests/test_middlewares.py	2022-05-11 13:48:09.683479158 +0200
@@ -1,4 +1,4 @@
-import mock
+from unittest import mock
 from django.contrib.auth import get_user_model
 from django.http import HttpResponse, HttpResponseServerError
 from django.test import RequestFactory, TestCase
diff -upr django-request-1.6.1.orig/tests/test_models.py django-request-1.6.1/tests/test_models.py
--- django-request-1.6.1.orig/tests/test_models.py	2022-05-11 13:48:09.679479134 +0200
+++ django-request-1.6.1/tests/test_models.py	2022-05-11 13:48:09.683479158 +0200
@@ -1,7 +1,7 @@
 import socket
 from datetime import datetime
 
-import mock
+from unittest import mock
 from django.contrib.auth import get_user_model
 from django.http import HttpRequest, HttpResponse
 from django.test import TestCase
diff -upr django-request-1.6.1.orig/tests/test_plugins.py django-request-1.6.1/tests/test_plugins.py
--- django-request-1.6.1.orig/tests/test_plugins.py	2022-05-11 13:48:09.679479134 +0200
+++ django-request-1.6.1/tests/test_plugins.py	2022-05-11 13:48:09.683479158 +0200
@@ -1,4 +1,4 @@
-import mock
+from unittest import mock
 from django.core import exceptions
 from django.test import TestCase
 
diff -upr django-request-1.6.1.orig/tests/test_traffic.py django-request-1.6.1/tests/test_traffic.py
--- django-request-1.6.1.orig/tests/test_traffic.py	2022-05-11 13:48:09.679479134 +0200
+++ django-request-1.6.1/tests/test_traffic.py	2022-05-11 13:48:09.687479185 +0200
@@ -1,4 +1,4 @@
-import mock
+from unittest import mock
 from django.core import exceptions
 from django.test import TestCase
 
