---
 src/qxl_kms.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/src/qxl_kms.c
+++ b/src/qxl_kms.c
@@ -593,6 +593,9 @@ qxl_kms_surface_create(qxl_screen_t *qxl
 		       int height,
 		       int bpp)
 {
+#if 1 /* XXX surfaces support is broken on KMS */
+    return NULL;
+#else
     SpiceSurfaceFmt format;
     qxl_surface_t *surface;
     int stride;
@@ -669,6 +672,7 @@ qxl_kms_surface_create(qxl_screen_t *qxl
     surface->bpp = bpp;
 
     return surface;
+#endif /* XXX */
 }
 
 static void qxl_kms_surface_destroy(qxl_surface_t *surf)
