Index: uwsgi-2.0.25.1/uwsgiconfig.py
===================================================================
--- uwsgi-2.0.25.1.orig/uwsgiconfig.py
+++ uwsgi-2.0.25.1/uwsgiconfig.py
@@ -1444,10 +1444,7 @@ def build_plugin(path, uc, cflags, ldfla
     except:
         pass
 
-    if uc:
-        plugin_dest = uc.get('plugin_build_dir', uc.get('plugin_dir')) + '/' + name + '_plugin'
-    else:
-        plugin_dest = name + '_plugin'
+    plugin_dest = name + '_plugin'
 
     shared_flag = '-shared'
 
