---
 uwsgiconfig.py |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

Index: uwsgi-2.0.31/uwsgiconfig.py
===================================================================
--- uwsgi-2.0.31.orig/uwsgiconfig.py	2025-10-11 21:15:54.000000000 +0200
+++ uwsgi-2.0.31/uwsgiconfig.py	2025-10-20 11:57:37.196618275 +0200
@@ -1444,10 +1444,7 @@
     except:
         pass
 
-    if uc:
-        plugin_dest = uc.get('plugin_build_dir', uc.get('plugin_dir')) + '/' + name + '_plugin'
-    else:
-        plugin_dest = name + '_plugin'
+    plugin_dest = name + '_plugin'
 
     shared_flag = '-shared'
 
