diff -wruN -x '*~' ../orig-sblim-cmpi-ssh_service_profile-1.0.0/src/Resource.c ./src/Resource.c
--- ../orig-sblim-cmpi-ssh_service_profile-1.0.0/src/Resource.c	2008-08-07 18:31:58.000000000 +0200
+++ ./src/Resource.c	2012-01-24 09:40:14.000000000 +0100
@@ -252,7 +252,7 @@
         if (!ptr)
         {
             free(data);
-            fclose(is);
+            pclose(is);
             return -1;
         }
 
@@ -260,6 +260,7 @@
         memcpy(&data[size++], &sess, sizeof(sess));
     }
 
+    pclose(is);
     *sessions = data;
     *num_sessions = size;
     return 0;
