diff -Nur deepin-editor-1.2.9/src/jumplinebar.cpp deepin-editor-1.2.9-new/src/jumplinebar.cpp
--- deepin-editor-1.2.9/src/jumplinebar.cpp	2019-06-11 17:03:48.000000000 +0800
+++ deepin-editor-1.2.9-new/src/jumplinebar.cpp	2020-06-25 10:54:09.358941794 +0800
@@ -24,6 +24,7 @@
 #include "jumplinebar.h"
 
 #include <QDebug>
+#include <QPainterPath>
 
 JumpLineBar::JumpLineBar(QWidget *parent)
     : QWidget(parent)
diff -Nur deepin-editor-1.2.9/src/thememodule/themepanel.cpp deepin-editor-1.2.9-new/src/thememodule/themepanel.cpp
--- deepin-editor-1.2.9/src/thememodule/themepanel.cpp	2019-06-11 17:03:48.000000000 +0800
+++ deepin-editor-1.2.9-new/src/thememodule/themepanel.cpp	2020-06-25 10:43:11.199045484 +0800
@@ -23,6 +23,7 @@
 #include <QScroller>
 #include <QVBoxLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QTimer>
 
 ThemePanel::ThemePanel(QWidget *parent)
diff -Nur deepin-editor-1.2.9/src/utils.h deepin-editor-1.2.9-new/src/utils.h
--- deepin-editor-1.2.9/src/utils.h	2019-06-11 17:03:48.000000000 +0800
+++ deepin-editor-1.2.9-new/src/utils.h	2020-06-25 10:41:24.975089100 +0800
@@ -24,6 +24,7 @@
 #include <QKeyEvent>
 #include <QObject>
 #include <QPainter>
+#include <QPainterPath>
 #include <QString>
 #include <QImage>
 
