diff -Nurp net-snmp-5.9.5.2-orig/dist/snmpd.service net-snmp-5.9.5.2/dist/snmpd.service
--- net-snmp-5.9.5.2-orig/dist/snmpd.service	2026-01-08 14:05:58.027152850 +0000
+++ net-snmp-5.9.5.2/dist/snmpd.service	2026-01-08 14:08:59.335152850 +0000
@@ -10,6 +10,15 @@ Description=Simple Network Management Pr
 After=network.target
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHostname=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 # Type=notify is also supported. It should be set when snmpd.socket is not used.
 Type=simple
 ExecStart=/usr/sbin/snmpd -f
