Index: libcaca-0.99.beta19/caca/driver/ncurses.c
===================================================================
--- libcaca-0.99.beta19.orig/caca/driver/ncurses.c
+++ libcaca-0.99.beta19/caca/driver/ncurses.c
@@ -262,11 +262,7 @@ static int ncurses_init_graphics(caca_di
     mouseinterval(-1); /* No click emulation */
 
     /* Set the escape delay to a ridiculously low value */
-#if defined set_escdelay
     set_escdelay(10);
-#else
-    ESCDELAY = 10;
-#endif
 
     /* Activate colour */
     start_color();
