diff --git a/src/MODULE.bazel b/src/MODULE.bazel
index 0c1866b30..a8bae14d4 100644
--- a/src/MODULE.bazel
+++ b/src/MODULE.bazel
@@ -192,6 +192,23 @@ pkg_config_repository(
     ],
 )
 
+# Fcitx
+new_local_repository(
+    name = "fcitx",
+    build_file = "BUILD.fcitx.bazel",
+    # This path should be updated per the environment.
+    path = "/usr",  # For Debian
+)
+
+# Fcitx 5
+pkg_config_repository(
+    name = "fcitx5",
+    packages = [
+        "Fcitx5Core",
+        "Fcitx5Module",
+    ],
+)
+
 # Qt for Linux
 pkg_config_repository(
     name = "qt_linux",
diff --git a/src/session/BUILD.bazel b/src/session/BUILD.bazel
index 090eae44e..8da0728ce 100644
--- a/src/session/BUILD.bazel
+++ b/src/session/BUILD.bazel
@@ -155,6 +155,7 @@ mozc_cc_library(
     visibility = [
         "//android/jni:__pkg__",
         "//ios:__pkg__",
+        "//unix/fcitx5:__pkg__",
     ],
     deps = [
         ":keymap",
@@ -351,6 +352,7 @@ mozc_cc_library(
     hdrs = ["key_info_util.h"],
     visibility = [
         "//client:__pkg__",
+        "//unix/fcitx5:__pkg__",
         "//win32/base:__pkg__",
     ],
     deps = [
