diff -ruN a/src/BUILD.fcitx.bazel b/src/BUILD.fcitx.bazel
--- a/src/BUILD.fcitx.bazel	1970-01-01 09:00:00.000000000 +0900
+++ b/src/BUILD.fcitx.bazel	2023-04-13 23:45:01.175952717 +0900
@@ -0,0 +1,29 @@
+# revert 2154ebe
+
+package(
+    default_visibility = ["//visibility:public"],
+)
+
+# This rule is created from pkg-config
+#
+# packages="fcitx"
+# copts, hdrs, indludes: pkg-config --cflags $packages
+# linkopts: pkg-config --libs-only-l $packages
+
+cc_library(
+    name = "fcitx",
+    hdrs = glob([
+        "include/fcitx/**",
+        "include/fcitx-config/**",
+        "include/fcitx-utils/**",
+    ]),
+    copts = ["-pthread"],
+    includes = [
+        "include",
+    ],
+    linkopts = [
+        "-lfcitx-core",
+        "-lfcitx-config",
+        "-lfcitx-utils",
+    ],
+)
diff -ruN a/src/WORKSPACE.bazel b/src/WORKSPACE.bazel
--- a/src/WORKSPACE.bazel	2023-04-13 23:40:49.927487544 +0900
+++ b/src/WORKSPACE.bazel	2023-04-13 23:41:42.676424993 +0900
@@ -70,6 +70,21 @@
   packages = ["glib-2.0", "gobject-2.0", "ibus-1.0"],
 )
 
+# Fcitx
+# pkg_config_repository does not work because pkgconfig --cflags-only-I fcitx returns nothing
+new_local_repository(
+  name = "fcitx",
+  # This path should be updated per the environment.
+  path = "/usr",  # For Debian
+  build_file = "BUILD.fcitx.bazel",
+)
+
+# Fcitx 5
+pkg_config_repository(
+  name = "fcitx5",
+  packages = ["Fcitx5Module", "Fcitx5Core"],
+)
+
 # Japanese Usage Dictionary
 new_local_repository(
     name = "ja_usage_dict",
