diff --git a/src/unix/ibus/gen_mozc_xml.py b/src/unix/ibus/gen_mozc_xml.py
index 1e2180c..5cb1e50 100644
--- a/src/unix/ibus/gen_mozc_xml.py
+++ b/src/unix/ibus/gen_mozc_xml.py
@@ -183,14 +183,20 @@ def main():
       'description': '%(product_name)s (Japanese Input Method)',
       'language': 'ja',
       'icon': '%(ibus_mozc_icon_path)s',
-      'rank': '80',
+      'symbol': '&#x3042;',
   }
 
   # DO NOT change the engine name 'mozc-jp'. The names is referenced by
   # unix/ibus/mozc_engine.cc.
+  # openSUSE provides four engines like ChromeOS
   engines_props = {
-      'name': ['mozc-jp'],
-      'longname': ['%(product_name)s'],
+      'name': ['mozc-jp', 'mozc-jp-jp', 'mozc-us', 'mozc-dv'],
+      'longname': ['%(product_name)s',
+                   '%(product_name)s - JP layout',
+                   '%(product_name)s - US layout',
+                   '%(product_name)s - US Dvorak layout'],
+      'layout': ['default', 'jp', 'us', 'us(dvorak)'],
+      'rank': ['80', '0', '0', '0'],
   }
 
   # IBus 1.5.11 and greater supports 'icon_prop_key'.
@@ -202,7 +208,6 @@ def main():
 
   if CheckIBusVersion(options, '1.5.0'):
     engine_common_props['symbol'] = '&#x3042;'
-    engines_props['layout'] = ['default']
   else:
     engines_props['layout'] = ['jp']
 
