diff --git a/include/builddefs.in b/include/builddefs.in
index 0bb364310d78..07775ac2d5e7 100644
--- a/include/builddefs.in
+++ b/include/builddefs.in
@@ -54,7 +54,7 @@ PKG_LIB_DIR	= @libdir@@libdirsuffix@
 PKG_INC_DIR	= @includedir@/xfs
 DK_INC_DIR	= @includedir@/disk
 PKG_MAN_DIR	= @mandir@
-PKG_DOC_DIR	= @datadir@/doc/@pkg_name@
+PKG_DOC_DIR	= @datadir@/doc/packages/@pkg_name@
 PKG_LOCALE_DIR	= @datadir@/locale
 PKG_DATA_DIR	= @datadir@/@pkg_name@
 MKFS_CFG_DIR	= @datadir@/@pkg_name@/mkfs
