Index: julia/contrib/build_sysimg.jl
===================================================================
--- julia.orig/contrib/build_sysimg.jl
+++ julia/contrib/build_sysimg.jl
@@ -1,4 +1,4 @@
-#!/usr/bin/env julia
+#!/usr/bin/julia
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 # Build a system image binary at sysimg_path.dlext. Allow insertion of a userimg via
Index: julia/contrib/julia-config.jl
===================================================================
--- julia.orig/contrib/julia-config.jl
+++ julia/contrib/julia-config.jl
@@ -1,4 +1,4 @@
-#!/usr/bin/env julia
+#!/usr/bin/julia
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 import Libdl
