Index: wannier90-4.0.2/test-suite/tests/CMakeLists.txt
===================================================================
--- wannier90-4.0.2.orig/test-suite/tests/CMakeLists.txt
+++ wannier90-4.0.2/test-suite/tests/CMakeLists.txt
@@ -1,7 +1,7 @@
 ## List tests
 set(test_names
                 checkpoint0_write
-                checkpoint1_read
+#               checkpoint1_read   # Disabled because fails to del files on chroot
 #               partestw90_mpierr  # Disabled because test intended to fail
                 testpostw90_boltzwann
                 testpostw90_example04_dos
