--- libvpl-2.10.1.orig/tools/legacy/sample_common/src/vaapi_utils.cpp	2024-01-20 09:55:42.622941953 +0000
+++ libvpl-2.10.1/tools/legacy/sample_common/src/vaapi_utils.cpp	2024-01-20 09:57:05.012571878 +0000
@@ -348,7 +348,7 @@ VAStatus CLibVA::AcquireVASurface(void**
     MSDK_ZERO_MEMORY(extsrf);
     MSDK_ZERO_MEMORY(bufferInfo);
     extsrf.num_buffers = 1;
-    extsrf.buffers     = &handle;
+    extsrf.buffers     = (uintptr_t*) &handle;
 
     attribs[0].type          = (VASurfaceAttribType)VASurfaceAttribMemoryType;
     attribs[0].flags         = VA_SURFACE_ATTRIB_SETTABLE;
