---
 tests/user_commands/tests.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/tests/user_commands/tests.py
+++ b/tests/user_commands/tests.py
@@ -334,7 +334,7 @@ class CommandTests(SimpleTestCase):
         self.assertIn('bar', out.getvalue())
 
     def test_subparser_invalid_option(self):
-        msg = "Error: invalid choice: 'test' (choose from 'foo')"
+        msg = "invalid choice: 'test' (choose from 'foo')"
         with self.assertRaisesMessage(CommandError, msg):
             management.call_command('subparser', 'test', 12)
         if PY37:
