From: Petr Uzel <petr.uzel@suse.cz>
Subject: libparted: Fix NVMe device naming
References: bsc#982169
Patch-mainline: no, required to fix custom SUSE device naming
---
 libparted/arch/linux.c | 1 +
 1 file changed, 1 insertion(+)

Index: parted-3.3/libparted/arch/linux.c
===================================================================
--- parted-3.3.orig/libparted/arch/linux.c
+++ parted-3.3/libparted/arch/linux.c
@@ -2488,6 +2488,7 @@ _device_get_part_path (PedDevice const *
                 if (dev->type == PED_DEVICE_ATARAID ||
                     dev->type == PED_DEVICE_CPQARRAY ||
                     dev->type == PED_DEVICE_DAC960 ||
+                    dev->type == PED_DEVICE_NVME ||
                     dev->type == PED_DEVICE_SDMMC)
                         p = "p";
                 else if (dev->type == PED_DEVICE_DM)
