Index: django-4.2.28/tests/serializers/test_xml.py
===================================================================
--- django-4.2.28.orig/tests/serializers/test_xml.py
+++ django-4.2.28/tests/serializers/test_xml.py
@@ -1,6 +1,7 @@
 import gc
 import time
 from xml.dom import minidom
+import unittest
 
 from django.core import serializers
 from django.core.serializers.xml_serializer import Deserializer, DTDForbidden
@@ -93,6 +94,7 @@ class XmlSerializerTestCase(SerializersT
         with self.assertRaises(DTDForbidden):
             next(serializers.deserialize("xml", xml))
 
+    @unittest.skip("flaky by design, OBS makes it even worse")
     def test_crafted_xml_performance(self):
         """The time to process invalid inputs is not quadratic."""
 
