Index: makefile
===================================================================
--- makefile.orig	2009-11-24 19:46:08.143710000 +0100
+++ makefile	2009-11-24 19:46:08.754711000 +0100
@@ -21,7 +21,7 @@ MAKE = make	      # needed for some syst
 CC = gcc
 CFLAGS =
 MODEL =
-EXTRA = -DBIG_MEM -DNDEBUG
+EXTRA = -DBIG_MEM -DNDEBUG -DANSI_PROTO
 LINTFLAGS = -DLINT
 OPTIM = -O
 DESTDIR = /usr/bin
Index: options.h
===================================================================
--- options.h.orig	2009-11-24 19:46:08.151711000 +0100
+++ options.h	2009-11-24 19:46:08.758710000 +0100
@@ -104,7 +104,7 @@ For documentation about this file, see o
 #define GETTZ
 #define FATTR
 #define T_SIGNAL	int
-#define VARARGS
+#define STDARG
 #define NEED_MEMMOVE
 #define T_UINT16		unsigned short		/* must be 16 bit unsigned */
 #define HAVE_ISATTY
