Index: libspectre-0.2.7/libspectre/spectre-document.c
===================================================================
--- libspectre-0.2.7.orig/libspectre/spectre-document.c
+++ libspectre-0.2.7/libspectre/spectre-document.c
@@ -67,7 +67,7 @@ spectre_document_load (SpectreDocument *
 		document->doc = NULL;
 	}
 	
-	document->doc = psscan (filename, SCANSTYLE_NORMAL);
+	document->doc = psscan (filename, SCANSTYLE_IGNORE_EOF);
 	if (!document->doc) {
 		document->status = SPECTRE_STATUS_LOAD_ERROR;
 		return;
