Index: node-v19.9.0/deps/npm/lib/commands/help-search.js
===================================================================
--- node-v19.9.0.orig/deps/npm/lib/commands/help-search.js
+++ node-v19.9.0/deps/npm/lib/commands/help-search.js
@@ -16,7 +16,7 @@ class HelpSearch extends BaseCommand {
       throw this.usageError()
     }
 
-    const docPath = path.resolve(this.npm.npmRoot, 'docs/content')
+    const docPath = '/usr/share/doc/packages/nodejs'
     let files = await glob(`${globify(docPath)}/*/*.md`)
     // preserve glob@8 behavior
     files = files.sort((a, b) => a.localeCompare(b, 'en'))
Index: node-v19.9.0/deps/npm/lib/npm.js
===================================================================
--- node-v19.9.0.orig/deps/npm/lib/npm.js
+++ node-v19.9.0/deps/npm/lib/npm.js
@@ -408,7 +408,13 @@ class Npm extends EventEmitter {
   }
 
   get globalPrefix () {
-    return this.config.globalPrefix
+    let prefix = this.config.globalPrefix
+
+    // don't poop all over distro territory - use /usr/local instead
+    if (prefix === '/usr')
+      return '/usr/local'
+
+    return prefix;
   }
 
   set globalPrefix (r) {
