diff --git a/src/common.rs b/src/common.rs
index 724d7ed..4c0c66c 100644
--- a/src/common.rs
+++ b/src/common.rs
@@ -101,7 +101,7 @@ impl<M: From<Message> + Send + 'static> Common<M> {
             core,
             error_opt: None,
             fallback_background: widget::image::Handle::from_bytes(
-                include_bytes!("../res/background.jpg").as_slice(),
+                include_bytes!("../res/background.png").as_slice(),
             ),
             layouts_opt,
             network_icon_opt: None,
