Index: clusterssh-4.18/bin_PL/_build_docs
===================================================================
--- clusterssh-4.18.orig/bin_PL/_build_docs
+++ clusterssh-4.18/bin_PL/_build_docs
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 use 5.008.004;
 use strict;
 use warnings;
Index: clusterssh-4.18/bin_PL/ccon
===================================================================
--- clusterssh-4.18.orig/bin_PL/ccon
+++ clusterssh-4.18/bin_PL/ccon
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 use 5.008.004;
 use strict;
 use warnings;
Index: clusterssh-4.18/bin_PL/crsh
===================================================================
--- clusterssh-4.18.orig/bin_PL/crsh
+++ clusterssh-4.18/bin_PL/crsh
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 use 5.008.004;
 use strict;
 use warnings;
Index: clusterssh-4.18/bin_PL/csftp
===================================================================
--- clusterssh-4.18.orig/bin_PL/csftp
+++ clusterssh-4.18/bin_PL/csftp
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 use 5.008.004;
 use strict;
 use warnings;
Index: clusterssh-4.18/bin_PL/cssh
===================================================================
--- clusterssh-4.18.orig/bin_PL/cssh
+++ clusterssh-4.18/bin_PL/cssh
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 use 5.008.004;
 use strict;
 use warnings;
Index: clusterssh-4.18/bin_PL/ctel
===================================================================
--- clusterssh-4.18.orig/bin_PL/ctel
+++ clusterssh-4.18/bin_PL/ctel
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 use 5.008.004;
 use strict;
 use warnings;
