diff --git a/bin_PL/_build_docs b/bin_PL/_build_docs
index 30b4b09..fab4d1d 100755
--- a/bin_PL/_build_docs
+++ b/bin_PL/_build_docs
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 use 5.008.004;
 use strict;
 use warnings;
diff --git a/bin_PL/ccon b/bin_PL/ccon
index 0a49778..eef4ddb 100755
--- a/bin_PL/ccon
+++ b/bin_PL/ccon
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 use 5.008.004;
 use strict;
 use warnings;
diff --git a/bin_PL/crsh b/bin_PL/crsh
index 3a95cfc..12a74ff 100755
--- a/bin_PL/crsh
+++ b/bin_PL/crsh
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 use 5.008.004;
 use strict;
 use warnings;
diff --git a/bin_PL/csftp b/bin_PL/csftp
index 3a95cfc..12a74ff 100755
--- a/bin_PL/csftp
+++ b/bin_PL/csftp
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 use 5.008.004;
 use strict;
 use warnings;
diff --git a/bin_PL/cssh b/bin_PL/cssh
index 3a95cfc..12a74ff 100755
--- a/bin_PL/cssh
+++ b/bin_PL/cssh
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 use 5.008.004;
 use strict;
 use warnings;
diff --git a/bin_PL/ctel b/bin_PL/ctel
index 9c3d11c..c0490e3 100755
--- a/bin_PL/ctel
+++ b/bin_PL/ctel
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 use 5.008.004;
 use strict;
 use warnings;
