diff -Ndur sblim-cmpi-base-1.6.4/provider-register.sh sblim-cmpi-base-1.6.4-fix-bashisms/provider-register.sh
--- sblim-cmpi-base-1.6.4/provider-register.sh	2014-10-23 17:54:13.000000000 +0300
+++ sblim-cmpi-base-1.6.4-fix-bashisms/provider-register.sh	2014-11-20 01:21:36.147815454 +0200
@@ -92,12 +92,12 @@
     done
 
 #produce Capabilities
-    let serial=0
+    serial=0
     for rf in $regfiles
     do
       cat $rf | grep -v '^[[:space:]]*#.*' | while read CLASSNAME NAMESPACE PROVIDERNAME PROVIDERMODULE CAPS
       do
-	let serial=serial+1
+	serial=$((serial + 1))
 	numcap=
 	for cap in $CAPS
 	do
@@ -703,14 +703,14 @@
 if [ "$help" = "1" ]
 then
     usage
-    echo -e "\t-h display help message"
-    echo -e "\t-v verbose mode"
-    echo -e "\t-d deregister provider and uninstall schema"
-    echo -e "\t-t specify cimserver type (pegasus|sfcb|openwbem|sniacimom)"
-    echo -e "\t-r specify registration files"
-    echo -e "\t-m specify schema mof files"
-    echo -e "\t-X create repository for alternate platform (sfcb only at the moment)."
-    echo -e "\t-n target namespace definition (default: root/cimv2)."
+    echo "	-h display help message"
+    echo "	-v verbose mode"
+    echo "	-d deregister provider and uninstall schema"
+    echo "	-t specify cimserver type (pegasus|sfcb|openwbem|sniacimom)"
+    echo "	-r specify registration files"
+    echo "	-m specify schema mof files"
+    echo "	-X create repository for alternate platform (sfcb only at the moment)."
+    echo "	-n target namespace definition (default: root/cimv2)."
     echo
     echo Use this command to install schema mofs and register providers.
     echo CIM Server Type is required as well as at least one registration file and one mof.
diff -Ndur sblim-cmpi-base-1.6.4/test/system/linux/createKeyFiles.sh sblim-cmpi-base-1.6.4-fix-bashisms/test/system/linux/createKeyFiles.sh
--- sblim-cmpi-base-1.6.4/test/system/linux/createKeyFiles.sh	2014-10-23 17:54:13.000000000 +0300
+++ sblim-cmpi-base-1.6.4-fix-bashisms/test/system/linux/createKeyFiles.sh	2014-11-20 01:22:32.313815532 +0200
@@ -13,7 +13,7 @@
 
 # Name
 
-if [[ -n $DOMAIN ]]; then
+if [ -n "$DOMAIN" ]; then
     echo `hostname` | grep $DOMAIN >/dev/null\
 	&& echo `hostname` >> ComputerSystem.keys \
 	|| echo `hostname`.$DOMAIN >> ComputerSystem.keys
@@ -35,7 +35,7 @@
 
 # Name
 
-if [[ -n $DOMAIN ]]; then
+if [ -n "$DOMAIN" ]; then
     echo `hostname` | grep $DOMAIN >/dev/null\
 	&& echo `hostname` >> OperatingSystem.keys \
 	|| echo `hostname`.$DOMAIN >> OperatingSystem.keys
diff -Ndur sblim-cmpi-base-1.6.4/test/system/linux/Linux_OperatingSystem.version.sh sblim-cmpi-base-1.6.4-fix-bashisms/test/system/linux/Linux_OperatingSystem.version.sh
--- sblim-cmpi-base-1.6.4/test/system/linux/Linux_OperatingSystem.version.sh	2014-10-23 17:54:13.000000000 +0300
+++ sblim-cmpi-base-1.6.4-fix-bashisms/test/system/linux/Linux_OperatingSystem.version.sh	2014-11-20 01:23:21.329815600 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 files=(`find /etc/ -type f -maxdepth 1 -name *release* 2>/dev/null`)
 
