--- libslirp-4.7.0+44/src/ip6_icmp.h.orig	2022-11-11 09:18:31.486198471 +0000
+++ libslirp-4.7.0+44/src/ip6_icmp.h	2022-11-11 09:18:44.050294018 +0000
@@ -59,7 +59,7 @@ struct ndp_na { /* Neighbor Advertisemen
     uint8_t R : 1, /* Router Flag */
         S : 1, /* Solicited Flag */
         O : 1, /* Override Flag */
-        reserved_1 : 5
+        reserved_1 : 5;
 #else
     uint8_t reserved_1 : 5, O : 1, S : 1, R : 1;
 #endif
