Index: libfdata-20211023/include/libfdata.h
===================================================================
--- libfdata-20211023.orig/include/libfdata.h
+++ libfdata-20211023/include/libfdata.h
@@ -1411,7 +1411,7 @@ int libfdata_vector_get_number_of_elemen
  * Returns 1 if successful or -1 on error
  */
 LIBFDATA_EXTERN \
-int libfdata_vector_get_element_index_at_offset(
+int64_t libfdata_vector_get_element_index_at_offset(
      libfdata_vector_t *vector,
      off64_t element_value_offset,
      int *element_index,
Index: libfdata-20211023/libfdata/libfdata_vector.c
===================================================================
--- libfdata-20211023.orig/libfdata/libfdata_vector.c
+++ libfdata-20211023/libfdata/libfdata_vector.c
@@ -979,7 +979,7 @@ int libfdata_vector_get_number_of_elemen
 /* Retrieves the element index for a specific offset
  * Returns 1 if successful or -1 on error
  */
-int libfdata_vector_get_element_index_at_offset(
+int64_t libfdata_vector_get_element_index_at_offset(
      libfdata_vector_t *vector,
      off64_t element_value_offset,
      int *element_index,
Index: libfdata-20211023/libfdata/libfdata_vector.h
===================================================================
--- libfdata-20211023.orig/libfdata/libfdata_vector.h
+++ libfdata-20211023/libfdata/libfdata_vector.h
@@ -236,7 +236,7 @@ int libfdata_vector_get_number_of_elemen
 /* Mapped range functions
  */
 LIBFDATA_EXTERN \
-int libfdata_vector_get_element_index_at_offset(
+int64_t libfdata_vector_get_element_index_at_offset(
      libfdata_vector_t *vector,
      off64_t element_value_offset,
      int *element_index,
