diff --git a/copy-firmware.sh b/copy-firmware.sh
index 1ae1e37c37e5..68fbf83ea650 100755
--- a/copy-firmware.sh
+++ b/copy-firmware.sh
@@ -66,6 +66,9 @@ done
 
 # shellcheck disable=SC2162 # file/folder name can include escaped symbols
 grep -E '^(RawFile|File):' WHENCE | sed -E -e 's/^(RawFile|File): */\1 /;s/"//g' | while read k f; do
+    case "$f" in
+        */README) continue;;
+    esac
     test -f "$f" || continue
     install -d "$destdir/$(dirname "$f")"
     $verbose "copying/compressing file $f$compext"
