diff --git a/WHENCE b/WHENCE
index 07a68814c23b..d7ef4a4d3f7d 100644
--- a/WHENCE
+++ b/WHENCE
@@ -5848,9 +5848,9 @@ Driver: wfx - Silicon Labs Wi-Fi Transceiver
 File: wfx/wfm_wf200_C0.sec
 Version: 3.14
 
-File: wfx/brd4001a.pds not listed in WHENCE
-File: wfx/brd8022a.pds not listed in WHENCE
-File: wfx/brd8023a.pds not listed in WHENCE
+File: wfx/brd4001a.pds
+File: wfx/brd8022a.pds
+File: wfx/brd8023a.pds
 
 Licence: Redistributable. See wfx/LICENCE.wf200 for details.
 
