--- src/transfor.c
+++ src/transfor.c
@@ -1632,6 +1632,11 @@
     for( i=0; i<LastShade; i++ )
         if( !Shade[i].refcount )
             ScaleCount++;
+    /* the i variable must be decremented becasue after for cycle the value is
+       equal LastShade, what is beyond the range af array ScaleRef used below.
+       Or the author just forgot to give brackets along the for cycle?
+    */
+    i--;
 
     /* If there are no shades free! */
     if( !ScaleCount ) ScaleCount = LastShade;
