Index: Session-Token-1.503/rand.h
===================================================================
--- Session-Token-1.503.orig/rand.h
+++ Session-Token-1.503/rand.h
@@ -36,9 +36,9 @@ typedef  struct randctx  randctx;
  If (flag==TRUE), then use the contents of randrsl[0..RANDSIZ-1] as the seed.
 ------------------------------------------------------------------------------
 */
-void randinit(/*_ randctx *r, word flag _*/);
+void randinit(randctx *r, word flag);
 
-void isaac(/*_ randctx *r _*/);
+void isaac(randctx *r);
 
 
 /*
