Index: python-vlc-3.0.101/README.module
===================================================================
--- python-vlc-3.0.101.orig/README.module
+++ python-vlc-3.0.101/README.module
@@ -68,10 +68,12 @@ Libvlc methods are available as attribut
 
     >>> import vlc
     >>> vlc.libvlc_get_version()
-    '3.0.0-rc2 Vetinari'
-    >>> e=vlc.VLCException()
-    >>> i=vlc.libvlc_new(0, [], e)
+    '3.0.0 Vetinari'
+    >>> i=vlc.libvlc_new(0, [])
     >>> i
     <vlc.Instance object at 0x8384a4c>
-    >>> vlc.libvlc_audio_get_volume(i,e)
+    >>> p=vlc.libvlc_media_player_new(i)
+    >>> p
+    <vlc.MediaPlayer object at 0x7f7a40d0af28>
+    >>> vlc.libvlc_audio_get_volume(p)
     50
