Index: five-or-more-3.32.2/src/preferences-dialog.vala
===================================================================
--- five-or-more-3.32.2.orig/src/preferences-dialog.vala
+++ five-or-more-3.32.2/src/preferences-dialog.vala
@@ -27,10 +27,10 @@ private class PreferencesDialog : Gtk.Di
     private Settings settings;
 
     [GtkChild]
-    private Gtk.ComboBoxText theme_box;
+    private unowned Gtk.ComboBoxText theme_box;
 
     [GtkChild]
-    private Gtk.ColorButton color_button;
+    private unowned Gtk.ColorButton color_button;
 
     private void theme_set_cb (Gtk.ComboBox self)
     {
Index: five-or-more-3.32.2/src/window.vala
===================================================================
--- five-or-more-3.32.2.orig/src/window.vala
+++ five-or-more-3.32.2/src/window.vala
@@ -25,16 +25,16 @@
 private class GameWindow : Gtk.ApplicationWindow
 {
     [GtkChild]
-    private Gtk.HeaderBar headerbar;
+    private unowned Gtk.HeaderBar headerbar;
 
     [GtkChild]
-    private Gtk.Box preview_hbox;
+    private unowned Gtk.Box preview_hbox;
 
     [GtkChild]
-    private Gtk.MenuButton primary_menu_button;
+    private unowned Gtk.MenuButton primary_menu_button;
 
     [GtkChild]
-    private Games.GridFrame grid_frame;
+    private unowned Games.GridFrame grid_frame;
 
     private Settings? settings = null;
     private bool window_tiled;
