diff -ruN seamonkey-2.53.17b1.orig/security/sandbox/linux/moz.build seamonkey-2.53.17b1/security/sandbox/linux/moz.build
--- seamonkey-2.53.17b1.orig/security/sandbox/linux/moz.build	2023-06-10 12:41:21.000000000 +0200
+++ seamonkey-2.53.17b1/security/sandbox/linux/moz.build	2023-07-12 09:48:50.099448143 +0200
@@ -101,7 +101,7 @@
 # forcing there to be only one partition.
 for f in CONFIG['OS_CXXFLAGS']:
     if f.startswith('-flto') and CONFIG['CC_TYPE'] != 'clang':
-        LDFLAGS += ['--param lto-partitions=1']
+        LDFLAGS += ['--param=lto-partitions=1']
 
 DEFINES['NS_NO_XPCOM'] = True
 DisableStlWrapping()
