--- seamonkey-2.53.21.old/python/mach/mach/commands/settings.py	2025-04-20 11:29:48.000000000 -0500
+++ seamonkey-2.53.21/python/mach/mach/commands/settings.py	2025-08-27 14:09:03.800204529 -0500
@@ -27,7 +27,7 @@
              description='Show available config settings.')
     @CommandArgument('-l', '--list', dest='short', action='store_true',
                      help='Show settings in a concise list')
-    def settings(self, command_context, short=None):
+    def run_settings(self, command_context, short=None):
         """List available settings."""
         types = {v: k for k, v in TYPE_CLASSES.items()}
         wrapper = TextWrapper(initial_indent='# ', subsequent_indent='# ')
