From 44bb53709096f9a7c3cb171638e437b29249df93 Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa13@gmail.com>
Date: Sun, 1 Dec 2019 11:17:43 -0500
Subject: [PATCH] Switch default reposdir to /etc/dnf/repos.d

---
 libdnf/conf/ConfigMain.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libdnf/conf/ConfigMain.cpp b/libdnf/conf/ConfigMain.cpp
index d147e3bd..1fb04887 100644
--- a/libdnf/conf/ConfigMain.cpp
+++ b/libdnf/conf/ConfigMain.cpp
@@ -187,7 +187,7 @@ class ConfigMain::Impl {
     OptionNumber<std::int32_t> log_rotate{4, 0};
     OptionBool log_compress{false};
     OptionStringList varsdir{VARS_DIRS};
-    OptionStringList reposdir{{"/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d"}};
+    OptionStringList reposdir{{"/etc/dnf/repos.d", "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d"}};
     OptionBool debug_solver{false};
     OptionStringList installonlypkgs{INSTALLONLYPKGS};
     OptionStringList group_package_types{GROUP_PACKAGE_TYPES};
-- 
2.28.0

