--- maven-jar-plugin-3.4.2/src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java	2025-06-06 13:29:29.658765179 +0200
+++ maven-jar-plugin-3.4.2/src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java	2025-06-06 13:29:55.193773784 +0200
@@ -302,6 +302,14 @@
                             getLog().info("Set Build-Jdk-Spec based on toolchain in maven-jar-plugin " + toolchain);
                         }));
 
+        if (outputTimestamp == null ||
+            outputTimestamp.length() < 1 ||
+            ((outputTimestamp.length() == 1)
+                && !Character.isDigit(outputTimestamp.charAt(0))))
+        {
+            outputTimestamp = System.getenv("SOURCE_DATE_EPOCH");
+        }
+
         // configure for Reproducible Builds based on outputTimestamp value
         archiver.configureReproducibleBuild(outputTimestamp);
 
