Index: tevent-0.17.1/wscript
===================================================================
--- tevent-0.17.1.orig/wscript
+++ tevent-0.17.1/wscript
@@ -27,6 +27,8 @@ def options(opt):
     opt.PRIVATE_EXTENSION_DEFAULT('tevent', noextension='tevent')
     opt.RECURSE('lib/replace')
     opt.RECURSE('lib/talloc')
+    opt.load('python') # options for disabling pyc or pyo compilation
+    # enable options related to building python extensions
 
 
 def configure(conf):
