Index: buildsystem-1.9/makefiles/Makefile.tools
===================================================================
--- buildsystem-1.9.orig/makefiles/Makefile.tools
+++ buildsystem-1.9/makefiles/Makefile.tools
@@ -135,7 +135,7 @@ else
   endif
 
   # Search the path for the compiler
-  toolpath_ := $(shell /bin/which $(CC__))
+  toolpath_ := $(shell which $(CC__))
   ifeq ($(toolpath_),)
     toolpath_ := /opt/netsurf/$(HOST)/cross/bin/
     CC__  := $(toolpath_)$(HOST)-gcc
