diff --git a/lekha/app.py b/lekha/app.py
index b580341..519b63e 100644
--- a/lekha/app.py
+++ b/lekha/app.py
@@ -296,11 +296,12 @@ class Document(Table):
         n.show()
 
         p = self.ol_p = Panel(
-            self, orient=ELM_PANEL_ORIENT_LEFT, hidden=True,
+            self, orient=ELM_PANEL_ORIENT_LEFT,
             size_hint_weight=EXPAND_BOTH, size_hint_align=FILL_BOTH,
-            scrollable=True, scrollable_content_size=0.35)
+            ) #scrollable=True, scrollable_content_size=0.35)
+        p.hidden = True
         scr.on_move_add(lambda x: p.move(*x.pos))
-        scr.on_resize_add(lambda x: p.resize(*x.size))
+        scr.on_resize_add(lambda x: p.resize(x.size[0] * 0.35, x.size[1]))
 
         ol_gl = self.ol_gl = Genlist(
             p, size_hint_weight=EXPAND_BOTH, size_hint_align=FILL_BOTH,
