Index: pygal-2.4.0/pygal/test/conftest.py
===================================================================
--- pygal-2.4.0.orig/pygal/test/conftest.py
+++ pygal-2.4.0/pygal/test/conftest.py
@@ -50,9 +50,9 @@ def pytest_generate_tests(metafunc):
     if hasattr(sys, 'pypy_version_info'):
         etree.to_etree()
 
-    if "Chart" in metafunc.funcargnames:
+    if "Chart" in metafunc.fixturenames:
         metafunc.parametrize("Chart", pygal.CHARTS)
-    if "datas" in metafunc.funcargnames:
+    if "datas" in metafunc.fixturenames:
         metafunc.parametrize(
             "datas",
             [
