--- a/src/core/keybindings.c
+++ b/src/core/keybindings.c
@@ -32,7 +32,7 @@
 #ifndef _GNU_SOURCE
 #define _GNU_SOURCE
 #endif
-#define _SVID_SOURCE /* for putenv() */
+#define _DEFAULT_SOURCE /* for putenv() */
 
 #include <config.h>
 #include "keybindings-private.h"
--- a/src/core/main.c
+++ b/src/core/main.c
@@ -45,7 +45,7 @@
 #ifndef _GNU_SOURCE
 #define _GNU_SOURCE
 #endif
-#define _SVID_SOURCE /* for putenv() and some signal-related functions */
+#define _DEFAULT_SOURCE /* for putenv() and some signal-related functions */
 
 #include <config.h>
 #include <meta/main.h>
--- a/src/core/window-props.c
+++ b/src/core/window-props.c
@@ -39,7 +39,7 @@
 #ifndef _GNU_SOURCE
 #define _GNU_SOURCE
 #endif
-#define _SVID_SOURCE /* for gethostname() */
+#define _DEFAULT_SOURCE /* for gethostname() */
 
 #include <config.h>
 #include "window-props.h"
