Index: vim-8.1.1467/runtime/doc/Makefile
===================================================================
--- vim-8.1.1467.orig/runtime/doc/Makefile
+++ vim-8.1.1467/runtime/doc/Makefile
@@ -321,6 +321,7 @@ all: tags vim.man evim.man vimdiff.man v
 # Use Vim to generate the tags file.  Can only be used when Vim has been
 # compiled and installed.  Supports multiple languages.
 vimtags: $(DOCS)
+	export VIMRUNTIME=../; \
 	$(VIMEXE) -u NONE -esX -c "helptags ++t ." -c quit
 
 # Use "doctags" to generate the tags file.  Only works for English!
