--- a2ps-h.py.orig	Fri Jan 26 12:12:32 2001
+++ a2ps-h.py	Fri Jan 26 11:41:50 2001
@@ -24,15 +24,15 @@
         def tmpnam():
                 return "/tmp/a2ps-h-tmp"
 
-hps = '/Gulim-Medium-KSC-EUC-H'
-hbps = '/Gulim-Medium-KSC-EUC-H'
-hbops = '/Gulim-Medium-KSC-EUC-H'
-hops = '/Gulim-Medium-KSC-EUC-H'
+hps = '/Munhwa-Regular-KSC-EUC-H'
+hbps = '/Munhwa-Regular-KSC-EUC-H'
+hbops = '/Munhwa-Regular-KSC-EUC-H'
+hops = '/Munhwa-Regular-KSC-EUC-H'
 
-hps = '/Gulim-Regular'
-hbps = '/Gulim-Regular'
-hbops = '/Gulim-Regular'
-hops = '/Gulim-Regular'
+hps = '/SMgoJ'
+hbps = '/SMgoJ'
+hbops = '/SMgoJ'
+hops = '/SMgoJ'
 
 def startswith(s, needle):
         if len(needle) > len(s): return 0
