diff -Nur Img-1.4.13/libpng/configure new/libpng/configure
--- Img-1.4.13/libpng/configure	2020-08-04 20:59:17.744532500 +0200
+++ new/libpng/configure	2022-05-22 12:15:12.031999739 +0200
@@ -5536,6 +5536,7 @@
 
 
     PKG_CFLAGS="$PKG_CFLAGS -DZLIBTCL_VERSION=\\\"${zlibtcl_VERSION}\\\""
+    PKG_CFLAGS="$PKG_CFLAGS -DPNG_ARM_NEON_IMPLEMENTATION=0 -DPNG_ARM_NEON_OPT=0"
 
 
 
diff -Nur Img-1.4.13/libpng/configure.ac new/libpng/configure.ac
--- Img-1.4.13/libpng/configure.ac	2020-08-04 20:59:30.597417200 +0200
+++ new/libpng/configure.ac	2022-05-22 12:06:16.251999803 +0200
@@ -78,6 +78,9 @@
 TEA_ADD_INCLUDES([-I\"`\${CYGPATH} \${srcdir}`\"])
 TEA_ADD_LIBS([-lm])
 TEA_ADD_CFLAGS([-DZLIBTCL_VERSION=\\\"${zlibtcl_VERSION}\\\"])
+
+# Disable NEON optimizations as needed source code is not included 
+TEA_ADD_CFLAGS([-DPNG_ARM_NEON_IMPLEMENTATION=0 -DPNG_ARM_NEON_OPT=0])
 TEA_ADD_STUB_SOURCES([pngtclStubLib.c])
 TEA_ADD_SOURCES([pngtclStubLib.c])
 TEA_ADD_TCL_SOURCES([])
