--- base/Makefile.in.orig
+++ base/Makefile.in
@@ -139,7 +139,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- bmp/Makefile.in.orig
+++ bmp/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- dted/Makefile.in.orig
+++ dted/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- flir/Makefile.in.orig
+++ flir/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- gif/Makefile.in.orig
+++ gif/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- ico/Makefile.in.orig
+++ ico/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- jpeg/Makefile.in.orig
+++ jpeg/Makefile.in
@@ -149,7 +149,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- libjpeg/Makefile.in.orig
+++ libjpeg/Makefile.in
@@ -138,7 +138,7 @@ EXTRA_PATH	= $(top_builddir):$(TCL_BIN_D
 TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
 SHARED_BUILD	= @SHARED_BUILD@
--- libpng/Makefile.in.orig
+++ libpng/Makefile.in
@@ -142,7 +142,7 @@ EXTRA_PATH	= $(top_builddir):$(TCL_BIN_D
 TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
 SHARED_BUILD	= @SHARED_BUILD@
--- libtiff/Makefile.in.orig
+++ libtiff/Makefile.in
@@ -140,7 +140,7 @@ EXTRA_PATH	= $(top_builddir):$(TCL_BIN_D
 TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
 SHARED_BUILD	= @SHARED_BUILD@
--- pcx/Makefile.in.orig
+++ pcx/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- pixmap/Makefile.in.orig
+++ pixmap/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- png/Makefile.in.orig
+++ png/Makefile.in
@@ -153,7 +153,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- ppm/Makefile.in.orig
+++ ppm/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- ps/Makefile.in.orig
+++ ps/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- raw/Makefile.in.orig
+++ raw/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- sgi/Makefile.in.orig
+++ sgi/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- sun/Makefile.in.orig
+++ sun/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- tga/Makefile.in.orig
+++ tga/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- tiff/Makefile.in.orig
+++ tiff/Makefile.in
@@ -157,7 +157,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- window/Makefile.in.orig
+++ window/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- xbm/Makefile.in.orig
+++ xbm/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- xpm/Makefile.in.orig
+++ xpm/Makefile.in
@@ -145,7 +145,7 @@ TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL
 		  TK_LIBRARY=`@CYGPATH@ $(TK_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 WISH_PROG	= @WISH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
--- zlib/Makefile.in.orig
+++ zlib/Makefile.in
@@ -138,7 +138,7 @@ EXTRA_PATH	= $(top_builddir):$(TCL_BIN_D
 TCLSH_ENV	= TCL_LIBRARY=`@CYGPATH@ $(TCL_SRC_DIR)/library` \
 		  @LD_LIBRARY_PATH_VAR@="$(EXTRA_PATH):$(@LD_LIBRARY_PATH_VAR@)" \
 		  PATH="$(EXTRA_PATH):$(PATH)" \
-		  TCLLIBPATH="$(libdir)"
+		  TCLLIBPATH="$(DESTDIR)$(libdir)"
 TCLSH_PROG	= @TCLSH_PROG@
 TCLSH		= $(TCLSH_ENV) $(TCLSH_PROG)
 SHARED_BUILD	= @SHARED_BUILD@
