--- python-whois-60adc3acc2d59627b2e32dc36d27802942e0820d/setup.py	2014-07-23 01:22:50.000000000 +0200
+++ python-whois-60adc3acc2d59627b2e32dc36d27802942e0820d/setup.py	2018-02-18 16:57:09.284672900 +0100
@@ -9,7 +9,6 @@
       packages=['pythonwhois'],
       package_dir={"pythonwhois":"pythonwhois"},
       package_data={"pythonwhois":["*.dat"]},
-      install_requires=['argparse'],
       provides=['pythonwhois'],
       scripts=["pwhois"],
       license="WTFPL"
