--- openvino-2026.3.1-origin/src/plugins/intel_gpu/src/runtime/ocl/ocl_wrapper.hpp	2026-08-27 22:14:16.524973350 -0300
+++ openvino-2026.3.1/src/plugins/intel_gpu/src/runtime/ocl/ocl_wrapper.hpp	2026-08-27 23:02:49.666799469 -0300
@@ -72,7 +72,8 @@
 
 #include "ocl_ext.hpp"
 
-#if CL_HPP_TARGET_OPENCL_VERSION >= 200 && CL_HPP_MINIMUM_OPENCL_VERSION >= 200
+#if defined(CL_HPP_PARAM_NAME_INFO_1_1_DEPRECATED_IN_2_0_) && \
+    CL_HPP_TARGET_OPENCL_VERSION >= 200 && CL_HPP_MINIMUM_OPENCL_VERSION >= 200
 namespace cl {
 namespace detail {
 CL_HPP_PARAM_NAME_INFO_1_1_DEPRECATED_IN_2_0_(CL_HPP_DECLARE_PARAM_TRAITS_)
