diff -uNr openvino-2024.6.0.orig/cmake/developer_package/packaging/archive.cmake openvino-2024.6.0/cmake/developer_package/packaging/archive.cmake
--- openvino-2024.6.0.orig/cmake/developer_package/packaging/archive.cmake	2024-12-27 17:04:54.520685198 -0300
+++ openvino-2024.6.0/cmake/developer_package/packaging/archive.cmake	2024-12-27 17:02:57.644273948 -0300
@@ -25,14 +25,18 @@
 macro(ov_archive_cpack_set_dirs)
     # common "archive" package locations
     # TODO: move current variables to OpenVINO specific locations
-    set(OV_CPACK_INCLUDEDIR runtime/include)
-    set(OV_CPACK_OPENVINO_CMAKEDIR runtime/cmake)
-    set(OV_CPACK_DOCDIR docs)
-    set(OV_CPACK_LICENSESDIR licenses)
-    set(OV_CPACK_SAMPLESDIR samples)
-    set(OV_CPACK_WHEELSDIR wheels)
-    set(OV_CPACK_DEVREQDIR tools)
-    set(OV_CPACK_PYTHONDIR python)
+    set(OV_CPACK_INCLUDEDIR include)
+    set(OV_CPACK_OPENVINO_CMAKEDIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+    set(OV_CPACK_DOCDIR ${CMAKE_INSTALL_DOCDIR})
+    set(OV_CPACK_LICENSESDIR ${CMAKE_INSTALL_DATAROOTDIR}/licenses/${PROJECT_NAME})
+    set(OV_CPACK_SAMPLESDIR ${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/samples)
+    if (ENABLE_PYTHON)
+        find_package(Python3 QUIET COMPONENTS Interpreter)
+        file(RELATIVE_PATH OV_PYTHON_MODPATH ${CMAKE_INSTALL_PREFIX} ${Python3_SITEARCH})
+        set(OV_CPACK_WHEELSDIR tools)
+        set(OV_CPACK_DEVREQDIR tools)
+        set(OV_CPACK_PYTHONDIR ${OV_PYTHON_MODPATH})
+    endif()
 
     if(USE_BUILD_TYPE_SUBFOLDER)
         set(build_type ${CMAKE_BUILD_TYPE})
@@ -49,11 +53,12 @@
         set(OV_CPACK_RUNTIMEDIR runtime/lib/${ARCH_FOLDER}/${build_type})
         set(OV_CPACK_ARCHIVEDIR runtime/lib/${ARCH_FOLDER}/${build_type})
     else()
-        set(OV_CPACK_LIBRARYDIR runtime/lib/${ARCH_FOLDER})
-        set(OV_CPACK_RUNTIMEDIR runtime/lib/${ARCH_FOLDER})
-        set(OV_CPACK_ARCHIVEDIR runtime/lib/${ARCH_FOLDER})
+        set(OV_CPACK_LIBRARYDIR ${CMAKE_INSTALL_LIBDIR})
+        set(OV_CPACK_RUNTIMEDIR ${CMAKE_INSTALL_LIBDIR})
+        set(OV_CPACK_ARCHIVEDIR ${CMAKE_INSTALL_LIBDIR})
     endif()
-    set(OV_CPACK_PLUGINSDIR ${OV_CPACK_RUNTIMEDIR})
+    set(OV_CPACK_PLUGINSDIR ${OV_CPACK_RUNTIMEDIR}/${PROJECT_NAME})
+
 endmacro()
 
 ov_archive_cpack_set_dirs()
diff -uNr openvino-2024.6.0.orig/src/cmake/openvino.cmake openvino-2024.6.0/src/cmake/openvino.cmake
--- openvino-2024.6.0.orig/src/cmake/openvino.cmake	2024-12-27 17:04:55.240687724 -0300
+++ openvino-2024.6.0/src/cmake/openvino.cmake	2024-12-27 17:03:50.176459053 -0300
@@ -267,6 +267,7 @@
 
     # define relative paths
     file(RELATIVE_PATH PKGCONFIG_OpenVINO_PREFIX "/${OV_CPACK_RUNTIMEDIR}/pkgconfig" "/")
+    cmake_path(NORMAL_PATH PKGCONFIG_OpenVINO_PREFIX)
 
     set(pkgconfig_in "${OpenVINO_SOURCE_DIR}/cmake/templates/openvino.pc.in")
     if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.20 AND OV_GENERATOR_MULTI_CONFIG)
diff -uNr openvino-2024.6.0.orig/src/plugins/intel_npu/tools/compile_tool/cmake/standalone.cmake openvino-2024.6.0/src/plugins/intel_npu/tools/compile_tool/cmake/standalone.cmake
--- openvino-2024.6.0.orig/src/plugins/intel_npu/tools/compile_tool/cmake/standalone.cmake	2024-12-27 17:04:56.868693438 -0300
+++ openvino-2024.6.0/src/plugins/intel_npu/tools/compile_tool/cmake/standalone.cmake	2024-12-28 00:55:18.661614722 -0300
@@ -43,5 +43,5 @@
 endif()
 
 install(TARGETS ${TARGET_NAME}
-        DESTINATION "tools/${TARGET_NAME}"
+        DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/tools/${TARGET_NAME}"
         COMPONENT npu_tools)
diff -uNr openvino-2024.6.0.orig/src/plugins/intel_npu/tools/compile_tool/CMakeLists.txt openvino-2024.6.0/src/plugins/intel_npu/tools/compile_tool/CMakeLists.txt
--- openvino-2024.6.0.orig/src/plugins/intel_npu/tools/compile_tool/CMakeLists.txt	2024-12-27 17:04:56.868693438 -0300
+++ openvino-2024.6.0/src/plugins/intel_npu/tools/compile_tool/CMakeLists.txt	2024-12-28 02:18:52.768816190 -0300
@@ -41,13 +41,13 @@
 #
 
 install(TARGETS ${TARGET_NAME}
-        RUNTIME DESTINATION "tools/${TARGET_NAME}"
+        RUNTIME DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/tools/${TARGET_NAME}"
         COMPONENT ${NPU_INTERNAL_COMPONENT}
         ${OV_CPACK_COMP_NPU_INTERNAL_EXCLUDE_ALL})
 
 if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/README.md")
     install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/README.md"
-            DESTINATION "tools/${TARGET_NAME}"
+	    DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/tools/${TARGET_NAME}"
             COMPONENT ${NPU_INTERNAL_COMPONENT}
             ${OV_CPACK_COMP_NPU_INTERNAL_EXCLUDE_ALL})
 endif()
