Index: cobbler-3.0.1+git20191008.21acab67/cobbler/cli.py
===================================================================
--- cobbler-3.0.1+git20191008.21acab67.orig/cobbler/cli.py	2019-10-08 16:08:04.000000000 +0200
+++ cobbler-3.0.1+git20191008.21acab67/cobbler/cli.py	2019-10-08 16:37:26.492555510 +0200
@@ -56,7 +56,7 @@ OBJECT_TYPES = list(OBJECT_ACTIONS_MAP.k
 OBJECT_ACTIONS = []
 for actions in list(OBJECT_ACTIONS_MAP.values()):
     OBJECT_ACTIONS += actions
-DIRECT_ACTIONS = "aclsetup buildiso import list replicate report reposync sync validate-autoinstalls version signature get-loaders hardlink".split()
+DIRECT_ACTIONS = "aclsetup buildiso import list replicate report reposync sync validate-autoinstalls version signature hardlink".split()
 
 ####################################################
 
@@ -580,9 +580,7 @@ class CobblerCLI(object):
             (options, args) = self.parser.parse_args()
             task_id = self.start_task("validate_autoinstall_files", options)
         elif action_name == "get-loaders":
-            self.parser.add_option("--force", dest="force", action="store_true", help="overwrite any existing content in /var/lib/cobbler/loaders")
-            (options, args) = self.parser.parse_args()
-            task_id = self.start_task("dlcontent", options)
+            print ("Command 'get-loaders' is not supported in this version of cobbler")
         elif action_name == "import":
             self.parser.add_option("--arch", dest="arch", help="OS architecture being imported")
             self.parser.add_option("--breed", dest="breed", help="the breed being imported")
