--- java-atk-wrapper-0.38.0/configure.ac	2022-03-07 20:15:51.171967671 +0100
+++ java-atk-wrapper-0.38.0/configure.ac	2020-03-05 23:17:04.000000000 +0100
@@ -46,7 +46,7 @@
 AS_IF([test "x$enable_modular_jar" = "xyes"], [
   JAVA_REQUIRED=9.0.0
 ], [
-  JAVA_REQUIRED=1.7.0
+  JAVA_REQUIRED=1.8.0
 ])
 JAVA_ERROR_MESSAGE="Java $JAVA_REQUIRED or later is required to build java-access-bridge"
 AM_CONDITIONAL([MODULAR_JAR], [test "x$enable_modular_jar" = "xyes"])
--- java-atk-wrapper-0.38.0/wrapper/Makefile.am	2019-04-01 16:22:10.000000000 +0200
+++ java-atk-wrapper-0.38.0/wrapper/Makefile.am	2022-03-07 19:47:41.281675042 +0100
@@ -19,7 +19,7 @@
 if MODULAR_JAR
 
 module-info.class : $(srcdir)/module-info.java
-	$(JAVAC) -cp $(top_builddir)/wrapper -sourcepath $(top_srcdir)/wrapper:$(top_builddir)/wrapper $(JAVACFLAGS) -d $(top_builddir)/wrapper $<
+	$(JAVAC) -cp $(top_builddir)/wrapper -sourcepath $(top_srcdir)/wrapper:$(top_builddir)/wrapper $(JAVACFLAGS) -source 9 -target 9 -d $(top_builddir)/wrapper $<
 
 $(JARFILES) : $(DEP_CLASSES) module-info.class
 	$(JAR) cfm $(JARFILES) $(srcdir)/manifest.txt org/GNOME/Accessibility/*.class module-info.class -C $(srcdir) META-INF/services/javax.accessibility.AccessibilityProvider
--- java-atk-wrapper-0.38.0/wrapper/org/GNOME/Accessibility/Makefile.am	2019-05-20 20:54:30.000000000 +0200
+++ java-atk-wrapper-0.38.0/wrapper/org/GNOME/Accessibility/Makefile.am	2022-03-07 20:12:54.986888891 +0100
@@ -12,7 +12,7 @@
 all-local: $(BUILD_CLASSES)
 
 %.class : %.java AtkWrapper.java
-	$(JAVAC) -cp $(top_builddir)/wrapper -sourcepath $(top_srcdir)/wrapper:$(top_builddir)/wrapper $(JAVACFLAGS) -d $(top_builddir)/wrapper $<
+	$(JAVAC) -cp $(top_builddir)/wrapper -sourcepath $(top_srcdir)/wrapper:$(top_builddir)/wrapper $(JAVACFLAGS) -source 8 -target 8 -d $(top_builddir)/wrapper $(JAVA_SRC) AtkWrapper.java
 
 clean-local:
 	$(RM) *.class
