Index: flatpak-1.15.8/subprojects/libglnx/meson.build
===================================================================
--- flatpak-1.15.8.orig/subprojects/libglnx/meson.build
+++ flatpak-1.15.8/subprojects/libglnx/meson.build
@@ -40,7 +40,7 @@ foreach check_function : check_functions
     #include <linux/random.h>
     #include <sys/mman.h>
 
-    int func (void) {
+    void func (void) {
       (void) ''' + check_function + ''';
     }
     ''',
