diff -Nru azure-mgmt-botservice-0.1.0.orig/setup.py azure-mgmt-botservice-0.1.0/setup.py
--- azure-mgmt-botservice-0.1.0.orig/setup.py	2018-08-07 16:41:54.000000000 +0200
+++ azure-mgmt-botservice-0.1.0/setup.py	2018-09-03 16:09:08.646356313 +0200
@@ -80,7 +80,7 @@
     install_requires=[
         'msrest>=0.5.0',
         'msrestazure>=0.4.32,<2.0.0',
-        'azure-common~=1.1',
+        'azure-common>=1.1',
     ],
     cmdclass=cmdclass
 )
