From: Jan Engelhardt <jengelh@inai.de>
Date: 2016-03-31 10:22:47.574917659 +0200

Attempt to fix really bad code (return-without-value and pre-ANSI function heads).

lrsnashlib.c:55:5: warning: 'return' with no value, in function returning non-void
	return; [multiple instances]
lrslib.cpp:1485:54: error: invalid conversion from 'void (*)()' to '__sighandler_t {aka void (*)(int)}' [-fpermissive]
        errcheck ("signal", signal (SIGALRM, timecheck)); [multiple]

---
 lrslib.c |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

Index: lrslib-071/lrslib.c
===================================================================
--- lrslib-071.orig/lrslib.c
+++ lrslib-071/lrslib.c
@@ -72,10 +72,10 @@ static double get_time(void);
 /* signals handling            */
 /*******************************/
 #ifndef SIGNALS
-static void checkpoint ();
-static void die_gracefully ();
+static void checkpoint (int);
+static void die_gracefully (int);
 static void setup_signals (void);
-static void timecheck ();
+static void timecheck (int);
 #endif
 
 /*******************************/
@@ -5365,7 +5365,7 @@ setup_signals ()
 }
 
 static void
-timecheck ()
+timecheck (int s)
 {
   lrs_dump_state ();
   errcheck ("signal", signal (SIGALRM, timecheck));
@@ -5373,14 +5373,14 @@ timecheck ()
 }
 
 static void
-checkpoint ()
+checkpoint (int s)
 {
   lrs_dump_state ();
   errcheck ("signal", signal (SIGUSR1, checkpoint));
 }
 
 static void
-die_gracefully ()
+die_gracefully (int s)
 {
   lrs_dump_state ();
 
