---
 datovka.pro |   26 ++++++++------------------
 1 file changed, 8 insertions(+), 18 deletions(-)

Index: datovka-4.13.0/datovka.pro
===================================================================
--- datovka-4.13.0.orig/datovka.pro
+++ datovka-4.13.0/datovka.pro
@@ -46,7 +46,6 @@ isEmpty(UI_DIR) {
 CONFIG += object_parallel_to_source
 
 DEFINES += \
-	DEBUG=1 \
 	VERSION=\\\"$${VERSION}\\\" \
 	APP_NAME=\\\"$${APP_NAME}\\\" \
 	APP_ORG_DOMAIN=\\\"cz.nic\\\" \
@@ -100,11 +99,6 @@ unix:!macx {
 	localisation.files += locale/datovka_cs.qm \
 		locale/datovka_en.qm
 
-	additional.path = "$${TEXT_FILES_INST_DIR}"
-	additional.files = \
-		AUTHORS \
-		COPYING
-
 	DEFINES += DATADIR=\\\"$$DATADIR\\\" \
 		PKGDATADIR=\\\"$$PKGDATADIR\\\" \
 		LOCALE_INST_DIR="\"\\\"$${LOCALE_INST_DIR}\\\"\"" \
@@ -122,15 +116,13 @@ unix:!macx {
 			icon64 \
 			icon128 \
 			icon256 \
-			localisation \
-			additional
+			localisation
 	}
 }
 
 QMAKE_CXXFLAGS = \
-	-g -O0 -std=c++11 \
-	-Wall -Wextra -pedantic \
-	-Wdate-time -Wformat -Werror=format-security
+	-g -std=c++11 $$(CXXFLAGS) \
+	-Wall -Wextra -pedantic
 
 #INCLUDEPATH +=
 
@@ -251,15 +243,8 @@ macx {
 		locale/datovka_en.qm
 	localisation.files += locale/qtbase_cs.qm
 
-	additional.path = "Contents/Resources"
-	additional.files = \
-		AUTHORS \
-		COPYING \
-		ChangeLog
-
 	QMAKE_BUNDLE_DATA +=\
-		localisation \
-		additional
+		localisation
 }
 
 win32 {
