--- jersey-2.28/containers/jetty-http/src/main/java/org/glassfish/jersey/jetty/JettyHttpContainerFactory.java	2024-10-30 04:43:55.503938128 +0100
+++ jersey-2.28/containers/jetty-http/src/main/java/org/glassfish/jersey/jetty/JettyHttpContainerFactory.java	2024-10-30 04:44:57.714388706 +0100
@@ -293,7 +293,7 @@
                 .build();
 
         @Override
-        protected Thread newThread(Runnable runnable) {
+        public Thread newThread(Runnable runnable) {
             return threadFactory.newThread(runnable);
         }
     }
