Index: vdirsyncer-0.16.7/tests/conftest.py
===================================================================
--- vdirsyncer-0.16.7.orig/tests/conftest.py	2018-02-16 19:48:40.000000000 +0100
+++ vdirsyncer-0.16.7/tests/conftest.py	2019-05-07 07:42:51.601117380 +0200
@@ -7,7 +7,7 @@ import os
 
 import click_log
 
-from hypothesis import HealthCheck, Verbosity, settings
+from hypothesis import Verbosity, settings
 
 import pytest
 
@@ -27,15 +27,12 @@ else:
     del pytest_benchmark
 
 
-settings.suppress_health_check = [HealthCheck.too_slow]
-
 settings.register_profile("ci", settings(
     max_examples=1000,
     verbosity=Verbosity.verbose,
 ))
 settings.register_profile("deterministic", settings(
     derandomize=True,
-    perform_health_check=False
 ))
 
 if os.environ.get('DETERMINISTIC_TESTS', 'false').lower() == 'true':
Index: vdirsyncer-0.16.7/tests/unit/test_repair.py
===================================================================
--- vdirsyncer-0.16.7.orig/tests/unit/test_repair.py	2018-07-19 21:07:31.000000000 +0200
+++ vdirsyncer-0.16.7/tests/unit/test_repair.py	2019-05-07 07:37:15.015521334 +0200
@@ -11,7 +11,6 @@ from vdirsyncer.vobject import Item
 
 
 @given(uid=uid_strategy)
-@settings(perform_health_check=False)  # Using the random module for UIDs
 def test_repair_uids(uid):
     s = MemoryStorage()
     s.items = {
@@ -35,7 +34,6 @@ def test_repair_uids(uid):
 
 
 @given(uid=uid_strategy.filter(lambda x: not href_safe(x)))
-@settings(perform_health_check=False)  # Using the random module for UIDs
 def test_repair_unsafe_uids(uid):
     s = MemoryStorage()
     item = Item(u'BEGIN:VCARD\nUID:{}\nEND:VCARD'.format(uid))
