--- icons/viruskiller.desktop.orig	2010-02-12 13:59:38.000000000 +0100
+++ icons/viruskiller.desktop	2015-06-27 02:07:48.572515478 +0200
@@ -1,9 +1,10 @@
 [Desktop Entry]
 Encoding=UTF-8
-Categories=Application;Game;ArcadeGame;
-X-Desktop-File-Install-Version=1.0
 Name=Virus Killer
-Icon=viruskiller.png
-Exec=/usr/games/viruskiller
-Terminal=0
+GenericName=Virus
+Comment=Arcade shoot 'em up game about virus invasion
 Type=Application
+Icon=viruskiller
+Exec=viruskiller
+Categories=Application;Game;ArcadeGame;
+Terminal=false
