---
 restructuredtext_lint/test/test.py |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

Index: restructuredtext_lint-1.4.0/restructuredtext_lint/test/test.py
===================================================================
--- restructuredtext_lint-1.4.0.orig/restructuredtext_lint/test/test.py	2022-02-24 06:50:23.000000000 +0100
+++ restructuredtext_lint-1.4.0/restructuredtext_lint/test/test.py	2025-10-30 11:37:31.488257443 +0100
@@ -4,7 +4,7 @@
 import subprocess
 import sys
 import textwrap
-from unittest import TestCase
+from unittest import TestCase, skip
 
 import restructuredtext_lint
 
@@ -201,14 +201,13 @@
         # This is the expected behaviour we are checking:
         # $ rst-lint --level warning second_short_heading.rst ; echo "Return code $?"
         # WARNING second_short_heading.rst:6 Title underline too short.
-        # WARNING second_short_heading.rst:6 Title underline too short.
         # Return code 2
         with self.assertRaises(subprocess.CalledProcessError) as e:
             subprocess.check_output((sys.executable, rst_lint_path, '--level', 'warning', warning_rst),
                                     universal_newlines=True)
         output = str(e.exception.output)
-        self.assertEqual(output.count('\n'), 2, output)
-        self.assertEqual(output.count('WARNING'), 2, output)
+        self.assertEqual(output.count('\n'), 1, output)
+        self.assertEqual(output.count('WARNING'), 1, output)
         # The expected 2 warnings should be treated as failing
         self.assertEqual(e.exception.returncode, 2)
 
