--- micropipenv-1.10.0/tests/test_micropipenv.py
+++ micropipenv-1.10.0/tests/test_micropipenv.py
@@ -45,7 +45,7 @@
 BIN_DIR = "Scripts" if WIN else "bin"
 SP_DIR = ("lib", "site-packages") if WIN else ("lib", "python*", "site-packages")
 PY_LE_39 = sys.version_info[:2] <= (3, 9)
-MANYLINUX_2014 = PIP_VERSION.major >= 19 and PIP_VERSION.minor >= 3
+MANYLINUX_2014 = PIP_VERSION and PIP_VERSION.major >= 19 and PIP_VERSION.minor >= 3
 
 
 @contextmanager
@@ -213,7 +213,7 @@
 @pytest.mark.online
 @pytest.mark.skipif(WIN, reason="Fails to remove .git folder on Windows")
 @pytest.mark.skipif(
-    PIP_VERSION.major < 21,
+    PIP_VERSION and PIP_VERSION.major < 21,
     reason="Old pip versions do not work with new setuptools and deprecated setup.py develop command",
 )
 def test_install_pipenv_vcs_editable(venv):
