diff -ru scim-skk-0.5.2.orig/src/scim_skk_imengine_setup.cpp scim-skk-0.5.2/src/scim_skk_imengine_setup.cpp
--- scim-skk-0.5.2.orig/src/scim_skk_imengine_setup.cpp	2005-12-18 13:15:53.000000000 +0100
+++ scim-skk-0.5.2/src/scim_skk_imengine_setup.cpp	2007-11-07 15:59:44.000000000 +0100
@@ -31,6 +31,8 @@
 #  include <config.h>
 #endif
 
+#include <string.h>
+
 #include <vector>
 
 #include <scim.h>
