--- univocity-parsers-2.5.5/src/main/java/com/univocity/parsers/common/AbstractParser.java	2022-03-23 12:59:51.808169163 +0100
+++ univocity-parsers-2.5.5/src/main/java/com/univocity/parsers/common/AbstractParser.java	2022-03-23 13:00:53.628543705 +0100
@@ -21,6 +21,7 @@
 import com.univocity.parsers.common.processor.*;
 import com.univocity.parsers.common.processor.core.*;
 import com.univocity.parsers.common.record.*;
+import com.univocity.parsers.common.record.Record;
 
 import java.io.*;
 import java.nio.charset.*;
--- univocity-parsers-2.5.5/src/main/java/com/univocity/parsers/common/iterators/RecordIterator.java	2022-03-23 12:59:51.812169188 +0100
+++ univocity-parsers-2.5.5/src/main/java/com/univocity/parsers/common/iterators/RecordIterator.java	2022-03-23 13:03:36.293529188 +0100
@@ -17,6 +17,7 @@
 
 import com.univocity.parsers.common.*;
 import com.univocity.parsers.common.record.*;
+import com.univocity.parsers.common.record.Record;
 
 import java.io.*;
 
