From: Lee Duncan <lduncan@suse.com>
Date: Sat Mar 22 10:52:22 AM PDT 2025
Subject: [PATCH] blah

Blah
---
--- a/systemd/target.service	2019-01-31 11:11:28.517558290 -0800
+++ b/systemd/target.service	2020-10-16 09:34:28.888091013 -0700
@@ -6,9 +6,12 @@ After=sys-kernel-config.mount network.ta
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=/usr/bin/targetctl restore
-ExecStop=/usr/bin/targetctl clear
+Environment=CONFIG_FILE=/etc/target/saveconfig.json
+EnvironmentFile=-/etc/sysconfig/target
+ExecStart=/usr/bin/targetctl restore $CONFIG_FILE
+ExecStop=/usr/bin/targetctl save $CONFIG_FILE
 SyslogIdentifier=target
 
 [Install]
 WantedBy=multi-user.target
+Alias=targetcli.service

