Index: spotifyd-0.3.2/contrib/spotifyd.service
===================================================================
--- spotifyd-0.3.2.orig/contrib/spotifyd.service
+++ spotifyd-0.3.2/contrib/spotifyd.service
@@ -7,6 +7,19 @@ Wants=network-online.target
 After=network-online.target
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=read-only
+PrivateDevices=true
+ProtectHostname=true
+ProtectClock=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 ExecStart=/usr/bin/spotifyd --no-daemon
 Restart=always
 RestartSec=12
