From ce0fedf25dea7eb63ccff8f9b90a9a35571a5f9d Mon Sep 17 00:00:00 2001
From: Mihai Dinca <mdinca@suse.de>
Date: Thu, 7 Nov 2019 15:11:49 +0100
Subject: [PATCH] Read repo info without using interpolation
 (bsc#1135656)

---
 salt/modules/zypperpkg.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/salt/modules/zypperpkg.py b/salt/modules/zypperpkg.py
index 6adf5f9aa3..d8220a1fdd 100644
--- a/salt/modules/zypperpkg.py
+++ b/salt/modules/zypperpkg.py
@@ -1155,7 +1155,9 @@ def _get_repo_info(alias, repos_cfg=None, root=None):
     Get one repo meta-data.
     """
     try:
-        meta = dict((repos_cfg or _get_configured_repos(root=root)).items(alias))
+        meta = dict(
+            (repos_cfg or _get_configured_repos(root=root)).items(alias, raw=True)
+        )
         meta["alias"] = alias
         for key, val in meta.items():
             if val in ["0", "1"]:
-- 
2.39.2


