 firstboot/fde | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: fde-tools-0.7.3/firstboot/fde
===================================================================
--- fde-tools-0.7.3.orig/firstboot/fde
+++ fde-tools-0.7.3/firstboot/fde
@@ -342,8 +342,8 @@ function fde_systemd_firstboot {
     fi
 
     if [ -z "$fde_root_passphrase" ]; then
-	display_errorbox "Cannot find the initial FDE password for the root file system"
-	return 1
+	# HACK: we just know it's 1234 for the ALP kvm_encrypted image, so go with that
+	fde_root_passphrase=1234
     fi
 
     if [ ! -s "$KIWI_ROOT_KEYFILE" ]; then
