Index: dd_rhelp-0.3.0/dd_rhelp
===================================================================
--- dd_rhelp-0.3.0.orig/dd_rhelp
+++ dd_rhelp-0.3.0/dd_rhelp
@@ -1746,7 +1746,7 @@ function process_log() {
 
     ## Remove first line of summaries that had 0.0k transfered. (They are useless)
     # XXXVaab : bad if file is less that 0.1Ko length
-    data=$(echo "$data" | "${grep}" -v "xferd: \+0.0k$")
+    data=$(echo "$data" | "${grep}" -v "xferd: \+0.0k$" | "${grep}" -v "$string")
     [ "$DEBUG" == "on" ] && echo -n "."
 
 
