From: Mel Gorman <mgorman@techsingularity.net>
Subject: [PATCH] Obey package guidelines for bash completions
Upstream: No

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index a78e3a277499..feaf09a944ba 100644
--- a/Makefile
+++ b/Makefile
@@ -91,7 +91,7 @@ HELP_DIR = -DHELP_DIR=$(html_install)
 HELP_DIR_SQ = '$(subst ','\'',$(HELP_DIR))'
 #' emacs highlighting gets confused by the above escaped quote.
 
-BASH_COMPLETE_DIR ?= /etc/bash_completion.d
+BASH_COMPLETE_DIR ?= $(prefix)/share/bash-completion/completions
 
 export PLUGIN_DIR
 export PYTHON_DIR
