From: Dave Plater <davejplater@gmail.com>
Date: 2018-01-23 14:27:22 +0200
Subject: input: Adapt a52_init () to new a52dec api

xine-lib uses wrong prototype for a52_init.
--- a/src/audio_dec/xine_a52_decoder.c
+++ b/src/audio_dec/xine_a52_decoder.c
@@ -715,7 +715,7 @@ static audio_decoder_t *open_plugin (aud
 	 capabilities themselves?
       */
 #warning "Enabling all external liba52 capabilities."
-      a52_init (0xFFFFFFFF)
+      a52_init ()
 #else
       a52_init (xine_mm_accel())
 #endif
