Index: libnet-1.2-rc3/acinclude.m4
===================================================================
--- libnet-1.2-rc3.orig/acinclude.m4
+++ libnet-1.2-rc3/acinclude.m4
@@ -71,13 +71,8 @@ main(int argc, char **argv)
     fd = socket(PF_PACKET, SOCK_RAW, htons(ETH_P_ALL));
     if (fd == -1)
     {
-        if (errno == EPERM)
-        {
-            /* user's UID != 0 */
-            printf("probably");
-            exit (EXIT_FAILURE);
-        }
-        printf("no");
+        /* user's UID != 0 */
+        printf("probably");
         exit (EXIT_FAILURE);
     }
     printf("yes");
