From 00d7573da739e6d3f623bae3025797f15c019ec3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bj=C3=B6rn=20Kettunen?= <bjorn.kettunen@thaodan.de>
Date: Mon, 16 Mar 2026 16:20:13 +0200
Subject: [PATCH] fix(binding.gyp): use correct grammar source paths

---
 binding.gyp |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

Index: tree-sitter-rpmspec-0+20260218.12ca618/binding.gyp
===================================================================
--- tree-sitter-rpmspec-0+20260218.12ca618.orig/binding.gyp	2026-02-18 00:10:58.000000000 +0100
+++ tree-sitter-rpmspec-0+20260218.12ca618/binding.gyp	2026-03-19 11:53:52.928270474 +0100
@@ -6,11 +6,15 @@
         "<!(node -p \"require('node-addon-api').targets\"):node_addon_api_except",
       ],
       "include_dirs": [
-        "src",
+        "rpmspec/src",
+        "rpmbash/src",
       ],
       "sources": [
+        "rpmspec/src/parser.c",
+        "rpmspec/src/scanner.c",
+        "rpmbash/src/parser.c",
+        "rpmbash/src/scanner.c",
         "bindings/node/binding.cc",
-        "src/parser.c",
         # NOTE: if your language has an external scanner, add it here.
       ],
       "cflags_c": [
