Index: zhu3d-4.2.4/src/tsc.h
===================================================================
--- zhu3d-4.2.4.orig/src/tsc.h
+++ zhu3d-4.2.4/src/tsc.h
@@ -134,12 +134,12 @@ inline TSC_UINT TSC::fetch() {
 inline TSC_UINT TSC::fetch(void) {
 	unsigned int high, low, tmp;
 	__asm__ volatile(
-		"loop:           \n\t"
+		"0:              \n\t"
 		"mftbu   %0      \n\t"
 		"mftb    %1      \n\t"
 		"mftbu   %2      \n\t"
 		"cmpw    %2,%0   \n\t"
-		"bne     loop    \n\t"
+		"bne     0       \n\t"
 		: "=r"(high), "=r"(low), "=r"(tmp)
 	);
 	return ((TSC_UINT)high << 32) | low;
