--- zhu3d-4.2.4.orig/src/mainwindow.cpp	2012-01-10 20:23:23.000000000 +0400
+++ zhu3d-4.2.4/src/mainwindow.cpp	2012-09-12 20:31:22.697126326 +0400
@@ -418,7 +418,6 @@
 void MaiWindow::aboutSlot() {
 
 	/** Get compilation date */
-	const QString tmp = (QString)(__DATE__);
 
 	QMessageBox::about(
 		this,
@@ -427,7 +426,7 @@
 		/** This is never translated */
 		QString("<b>") + NAME + QString("</b> ") +
         QString("is an OpenGL-based function-viewer and solver.") +
-		QString("<p>Version ")+VERS+QString("  compiled ")+tmp+QString("</p>") +
+		QString("<p>Version ")+VERS+QString("</p>") +
         QString("<p>Check updates on: "
 		"<a href=\"http://www.kde-apps.org/content/show.php?content=43071/\""
 		">www.kde-apps.org</a><BR>"
