Index: asl-current/sysdefs.h
===================================================================
--- asl-current.orig/sysdefs.h
+++ asl-current/sysdefs.h
@@ -738,8 +738,8 @@ typedef unsigned short Card16;
 #define HAS16
 typedef signed int Integ32;
 typedef unsigned int Card32;
-typedef signed long Integ64;
-typedef unsigned long Card64;
+typedef signed long long Integ64;
+typedef unsigned long long Card64;
 #define HAS64
 #define LOCALE_NLS
 #endif
