From e97d040fa49a9c6e347bfe51fdf6f3d2a00763f8 Mon Sep 17 00:00:00 2001
From: Dylan McCall <dylanmccall@ubuntu.com>
Date: Thu, 21 Nov 2013 02:46:20 +0000
Subject: Fixed build error with Vala 0.22.1

---
diff --git a/helper/UIManager.vala b/helper/UIManager.vala
index 6a9f114..5dc9229 100644
--- a/helper/UIManager.vala
+++ b/helper/UIManager.vala
@@ -155,7 +155,7 @@ public class UIManager : SimpleFocusManager {
 	protected ScreenOverlay? screen_overlay;
 	protected Notify.Notification? notification;
 
-	protected unowned List<string> notify_capabilities;
+	protected List<string> notify_capabilities;
 
 	// The desktop-entry notification hint wants our desktop ID without the
 	// ".desktop" part, so we need to trim it accordingly
--
cgit v0.9.2

