---
 configure    |    6 ++++++
 configure.ac |    6 ++++++
 2 files changed, 12 insertions(+)

Index: ptlib-2.18.8/configure
===================================================================
--- ptlib-2.18.8.orig/configure
+++ ptlib-2.18.8/configure
@@ -7047,6 +7047,12 @@ fi
         LIB64=1
 	;;
 
+   powerpc64le)
+	MACHTYPE=ppc64le
+	P_64BIT=1
+        LIB64=1
+	;;
+
    * )
       { printf "%s\n" "$as_me:${as_lineno-$LINENO}: WARNING: Operating system $target_os not recognized - proceed with caution!" >&5
 printf "%s\n" "$as_me: WARNING: Operating system $target_os not recognized - proceed with caution!" >&2;}
Index: ptlib-2.18.8/configure.ac
===================================================================
--- ptlib-2.18.8.orig/configure.ac
+++ ptlib-2.18.8/configure.ac
@@ -178,6 +178,12 @@ case "$target_os" in
         LIB64=1
 	;;
 
+   powerpc64le)
+	MACHTYPE=ppc64le
+	P_64BIT=1
+        LIB64=1
+	;;
+
    * )
       OS_TAG="P_TARGET_OS"
 esac
