--- a/configure.ac
+++ b/configure.ac
@@ -104,7 +104,7 @@ AC_ARG_ENABLE(shape,
 ## try definining HAVE_BACKTRACE
 AC_CHECK_HEADERS(execinfo.h, [AC_CHECK_FUNCS(backtrace)])
 
-PKG_CHECK_MODULES(ALL, glib-2.0 >= 2.58.0)
+PKG_CHECK_MODULES(ALL, glib-2.0 >= 2.54.0)
 PKG_CHECK_MODULES(MARCO_MESSAGE, gtk+-3.0 >= $GTK_MIN_VERSION)
 PKG_CHECK_MODULES(MARCO_WINDOW_DEMO, gtk+-3.0 >= $GTK_MIN_VERSION)
 
--- a/src/core/eventqueue.c
+++ b/src/core/eventqueue.c
@@ -71,7 +71,14 @@ meta_event_queue_new (Display *display,
   g_source_add_poll (source, &eq->poll_fd);
   g_source_set_can_recurse (source, TRUE);
 
+#if GLIB_CHECK_VERSION (2, 58, 0)
   g_source_set_callback (source, G_SOURCE_FUNC (func), data, NULL);
+#else
+  g_source_set_callback (source,
+                         (GSourceFunc) (void (*)(void)) func,
+                         data,
+                         NULL);
+#endif
 
   g_source_attach (source, NULL);
   g_source_unref (source);
