diff -ru netdata-v1.12.1.orig/collectors/python.d.plugin/smartd_log/smartd_log.conf netdata-v1.12.1/collectors/python.d.plugin/smartd_log/smartd_log.conf
--- netdata-v1.12.1.orig/collectors/python.d.plugin/smartd_log/smartd_log.conf	2019-02-21 19:43:46.764671505 +0100
+++ netdata-v1.12.1/collectors/python.d.plugin/smartd_log/smartd_log.conf	2019-02-22 06:15:55.813201050 +0100
@@ -65,3 +65,5 @@
 #    exclude_disks: 'PATTERN1 PATTERN2'  # space separated patterns. If the pattern is in the drive name, the module will not collect data for it.
 #
 # ----------------------------------------------------------------------
+
+log_path: '/var/lib/smartmontools'
