Index: mediastreamer2-5.3.4/cmake/FindSRTP.cmake
===================================================================
--- mediastreamer2-5.3.4.orig/cmake/FindSRTP.cmake
+++ mediastreamer2-5.3.4/cmake/FindSRTP.cmake
@@ -61,7 +61,7 @@ else()
 	set(_SRTP_ROOT_PATHS ${CMAKE_INSTALL_PREFIX})
 
 	find_path(_SRTP_INCLUDE_DIRS
-		NAMES srtp2/srtp.h
+		NAMES srtp2-linphone/srtp.h
 		HINTS ${_SRTP_ROOT_PATHS}
 		PATH_SUFFIXES include
 	)
@@ -69,7 +69,7 @@ else()
 	if(_SRTP_INCLUDE_DIRS)
 		set(SRTP_VERSION 2)
 		find_library(_SRTP_LIBRARY
-			NAMES srtp2
+			NAMES srtp2-linphone
 			HINTS ${_SRTP_ROOT_PATHS}
 			PATH_SUFFIXES bin lib
 		)
Index: mediastreamer2-5.3.4/src/utils/srtp_prefix.h
===================================================================
--- mediastreamer2-5.3.4.orig/src/utils/srtp_prefix.h
+++ mediastreamer2-5.3.4/src/utils/srtp_prefix.h
@@ -30,7 +30,7 @@
 #elif SRTP_VERSION == 1
 #include <srtp/srtp.h>
 #else
-#include <srtp2/srtp.h>
+#include <srtp2-linphone/srtp.h>
 #endif
 
 #if SRTP_VERSION == 2
