--- mediastreamer2-5.2.49/CMakeLists.txt.orig	2023-04-07 11:21:45.535261082 +0200
+++ mediastreamer2-5.2.49/CMakeLists.txt	2023-04-07 11:21:50.571286441 +0200
@@ -29,7 +29,7 @@
     cmake_policy(SET CMP0077 NEW)
 endif()
 
-project(mediastreamer2 VERSION 5.2.0 LANGUAGES C CXX)
+project(mediastreamer2 VERSION 5.2.49 LANGUAGES C CXX)
 
 if(WIN32)
     if( NOT CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
