diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0750edb..3a6323f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -177,6 +177,8 @@ macro(GR_OSMOSDR_APPEND_LIBS lib)
     list(APPEND osmo_libraries ${lib})
 endmacro(GR_OSMOSDR_APPEND_LIBS)
 
+GR_OSMOSDR_APPEND_LIBS("-lpthread")
+
 IF(HAVE_CLOCK_GETTIME)
     MESSAGE(STATUS "  High resolution timing supported through clock_gettime.")
     SET(TIME_SPEC_DEFS HAVE_CLOCK_GETTIME)
