From df18e7f683b610e198f8d359d2036655a1891795 Mon Sep 17 00:00:00 2001
From: Alberto Planas <aplanas@suse.com>
Date: Tue, 17 Jan 2023 11:35:59 +0100
Subject: [PATCH] Cargo.toml: tss-esapi bindings

Generate the tss-esapi bindings during compilation time.  This will
extend the amount of supported architectures.

This is a regression done after the crate split, that was fixed before
in #459.

Signed-off-by: Alberto Planas <aplanas@suse.com>
---
 keylime-ima-emulator/Cargo.toml | 2 +-
 keylime/Cargo.toml              | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/keylime-ima-emulator/Cargo.toml b/keylime-ima-emulator/Cargo.toml
index ee3d5c1..333a192 100644
--- a/keylime-ima-emulator/Cargo.toml
+++ b/keylime-ima-emulator/Cargo.toml
@@ -12,4 +12,4 @@ keylime = { path = "../keylime" }
 log = "0.4"
 openssl = "0.10.15"
 thiserror = "1.0"
-tss-esapi = "7.1.0"
+tss-esapi = {version = "7.1.0", features = ["generate-bindings"]}
diff --git a/keylime/Cargo.toml b/keylime/Cargo.toml
index 9693912..c343d40 100644
--- a/keylime/Cargo.toml
+++ b/keylime/Cargo.toml
@@ -14,7 +14,7 @@ serde = "1.0.80"
 serde_derive = "1.0.80"
 static_assertions = "1"
 thiserror = "1.0"
-tss-esapi = "7.1.0"
+tss-esapi = {version = "7.1.0", features = ["generate-bindings"]}
 
 [dev-dependencies]
 tempfile = "3.0.4"
-- 
2.39.0

