diff --git a/Makefile.am b/Makefile.am
index 26f2143..a549d21 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -949,6 +949,7 @@ EXTRA_DIST = next_build.sh run_tests.sh ABOUT RELEASE-NOTES \
 	modules/reflection/docs/mainpage.dox.tmpl \
 	modules/reflection/CMakeLists.txt \
 	modules/reflection/cmake/cmake_uninstall.cmake.in \
+    doxygen/lang/mainpage.dox.tmpl \
 	doxygen/lang/Doxyfile.in \
 	doxygen/qlib/Doxyfile.in \
 	doxygen/modules/Doxyfile.in \
diff --git a/modules/astparser/CMakeLists.txt b/modules/astparser/CMakeLists.txt
index 6fdc146..0d8a55f 100644
--- a/modules/astparser/CMakeLists.txt
+++ b/modules/astparser/CMakeLists.txt
@@ -49,7 +49,7 @@ set(ASTPARSER_CPP_SRC
     ${CMAKE_CURRENT_BINARY_DIR}/ast_scanner.cpp
 )
 
-set(QORE_DOX_TMPL_SRC
+set(QORE_DOX_ASTPARSER_TMPL_SRC
     docs/mainpage.dox.tmpl
 )
 
@@ -77,7 +77,7 @@ set(MODULE_DOX_INPUT ${CMAKE_BINARY_DIR}/modules/astparser/mainpage.dox ${_dox_s
 qore_binary_module_qore(${astparser_module_name} "${VERSION}" "/${VERSION}")
 
 if (DOXYGEN_FOUND)
-    qore_wrap_dox(QORE_DOX_SRC ${QORE_DOX_TMPL_SRC})
-    add_custom_target(QORE_ASTPARSER_MOD_DOX_FILES DEPENDS ${QORE_DOX_SRC})
+    qore_wrap_dox(QORE_ASTPARSER_DOX_SRC ${QORE_DOX_ASTPARSER_TMPL_SRC})
+    add_custom_target(QORE_ASTPARSER_MOD_DOX_FILES DEPENDS ${QORE_DOX_ASTPARSER_SRC})
     add_dependencies(docs-astparser QORE_ASTPARSER_MOD_DOX_FILES)
 endif()
diff --git a/modules/reflection/CMakeLists.txt b/modules/reflection/CMakeLists.txt
index afe31f2..b198013 100644
--- a/modules/reflection/CMakeLists.txt
+++ b/modules/reflection/CMakeLists.txt
@@ -101,7 +101,7 @@ set(REFLECTION_H_SRC
     src/QC_Type.h
 )
 
-set(QORE_DOX_TMPL_SRC
+set(QORE_DOX_REFLECTION_TMPL_SRC
     docs/mainpage.dox.tmpl
 )
 
@@ -129,7 +129,7 @@ set(MODULE_DOX_INPUT ${CMAKE_BINARY_DIR}/modules/reflection/mainpage.dox ${_dox_
 qore_binary_module_qore(${reflection_module_name} "${VERSION}" "/${VERSION}")
 
 if (DOXYGEN_FOUND)
-    qore_wrap_dox(QORE_DOX_SRC ${QORE_DOX_TMPL_SRC})
-    add_custom_target(QORE_REFLECTION_MOD_DOX_FILES DEPENDS ${QORE_DOX_SRC})
+    qore_wrap_dox(QORE_DOX_REFLECTION_SRC ${QORE_DOX_REFLECTION_TMPL_SRC})
+    add_custom_target(QORE_REFLECTION_MOD_DOX_FILES DEPENDS ${QORE_DOX_REFLECTION_SRC})
     add_dependencies(docs-reflection QORE_REFLECTION_MOD_DOX_FILES)
 endif()
