diff -Nur qore-release-1.0.10/cmake/QoreMacros.cmake new/cmake/QoreMacros.cmake
--- qore-release-1.0.10/cmake/QoreMacros.cmake	2021-10-16 17:14:58.000000000 +0200
+++ new/cmake/QoreMacros.cmake	2021-10-28 14:13:51.791708661 +0200
@@ -187,7 +187,7 @@
     target_link_libraries(${_module_name} ${_libs})
 
     # ensure that modules use dynamic lookups; works with g++ & clang++
-    set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-undefined -Wl,dynamic_lookup")
+    set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,-undefined -Wl,dynamic_lookup")
 
     install( TARGETS ${_module_name} DESTINATION ${_mod_target_dir})
 
