Index: djvulibre-3.5.27/desktopfiles/Makefile.am
===================================================================
--- djvulibre-3.5.27.orig/desktopfiles/Makefile.am
+++ djvulibre-3.5.27/desktopfiles/Makefile.am
@@ -61,7 +61,7 @@ else
 endif
 
 svg_process =\
-cat $< | gzip >$@
+cat $< | gzip -n >$@
 
 svg_verbose = $(svg_verbose_@AM_V@)
 svg_verbose_ = $(svg_verbose_@AM_DEFAULT_V@)
Index: djvulibre-3.5.27/desktopfiles/Makefile.in
===================================================================
--- djvulibre-3.5.27.orig/desktopfiles/Makefile.in
+++ djvulibre-3.5.27/desktopfiles/Makefile.in
@@ -321,7 +321,7 @@ png_verbose = $(png_verbose_@AM_V@)
 png_verbose_ = $(png_verbose_@AM_DEFAULT_V@)
 png_verbose_0 = @echo "  PNG     " $@;
 svg_process = \
-cat $< | gzip >$@
+cat $< | gzip -n >$@
 
 svg_verbose = $(svg_verbose_@AM_V@)
 svg_verbose_ = $(svg_verbose_@AM_DEFAULT_V@)
