diff --git a/tools/gpick.py b/tools/gpick.py
index 8538af1..22631e8 100644
--- a/tools/gpick.py
+++ b/tools/gpick.py
@@ -156,7 +156,7 @@ class GpickEnvironment(SConsEnvironment):
 			(version, revision, hash, date) = getVersionInfo()
 		except:
 			try:
-				with open("../.version", "r", encoding = 'utf-8') as version_file:
+				with open(self.File('#.version').abspath, "r", encoding = 'utf-8') as version_file:
 					(version, revision, hash, date) = version_file.read().splitlines()
 			except:
 				print("Version file \".version\" is required when GIT can not be used to get version information.")
