diff -uNr libsmi-0.4.8-orig/lib/parser-sming.y libsmi-0.4.8/lib/parser-sming.y
--- libsmi-0.4.8-orig/lib/parser-sming.y	2008-04-18 12:42:50.000000000 +0200
+++ libsmi-0.4.8/lib/parser-sming.y	2010-03-23 10:44:11.777122974 +0100
@@ -2161,8 +2161,8 @@
 							  classPtr, thisParserPtr);
 			    setAttributeParentType($$, smiHandle->typeBitsPtr);
 			    if ($1) {
-				setAttributeList($$, $1);
-				for (p = $1; p; p = p->nextPtr)
+				setAttributeList($$, (void *)$1);
+				for (p = (void *)$1; p; p = (void *)p->nextPtr)
 				    ((NamedNumber *)(p->ptr))->typePtr = (Type*)$$;
 			    }
 			    
