Index: WeasyPrint-52.5/weasyprint/tests/test_api.py
===================================================================
--- WeasyPrint-52.5.orig/weasyprint/tests/test_api.py
+++ WeasyPrint-52.5/weasyprint/tests/test_api.py
@@ -785,7 +785,7 @@ def test_links():
             <div style="-weasy-link: url(../lipsum);
                         display: block; margin: 10px 5px">
         ''', [[]], [{}], [([], [])], base_url=None, warnings=[
-            'WARNING: Ignored `-weasy-link: url(../lipsum)` at 1:1, '
+            'WARNING: Ignored `-weasy-link: url("../lipsum")` at 1:1, '
             'Relative URI reference without a base URI'])
 
     # Internal or absolute URI reference without a base URI: OK
Index: WeasyPrint-52.5/weasyprint/tests/test_pdf.py
===================================================================
--- WeasyPrint-52.5.orig/weasyprint/tests/test_pdf.py
+++ WeasyPrint-52.5/weasyprint/tests/test_pdf.py
@@ -528,7 +528,7 @@ def test_relative_links_missing_base():
     with pytest.raises(AttributeError):
         pdf_file.pages[0].get_indirect_dict_array('Annots', pdf_file)
     assert len(logs) == 1
-    assert 'WARNING: Ignored `-weasy-link: url(../lipsum)`' in logs[0]
+    assert 'WARNING: Ignored `-weasy-link: url("../lipsum")`' in logs[0]
     assert 'Relative URI reference without a base URI' in logs[0]
 
 
