Index: asciimatics-1.13.0/tests/test_effects.py
===================================================================
--- asciimatics-1.13.0.orig/tests/test_effects.py	2020-09-18 16:37:11.000000000 +0200
+++ asciimatics-1.13.0/tests/test_effects.py	2022-03-07 13:53:31.271205331 +0100
@@ -1,7 +1,7 @@
 from builtins import chr
 import unittest
 from datetime import datetime
-from mock.mock import MagicMock, patch
+from unittest.mock import MagicMock, patch
 from random import randint
 import os
 import sys
Index: asciimatics-1.13.0/tests/test_particles.py
===================================================================
--- asciimatics-1.13.0.orig/tests/test_particles.py	2020-09-18 16:37:11.000000000 +0200
+++ asciimatics-1.13.0/tests/test_particles.py	2022-03-07 13:54:14.599457606 +0100
@@ -1,5 +1,5 @@
 import unittest
-from mock.mock import MagicMock
+from unittest.mock import MagicMock
 from asciimatics.particles import ShootScreen, DropScreen, Explosion, Rain, \
     StarFirework, PalmFirework, RingFirework, SerpentFirework
 from asciimatics.screen import Screen, Canvas
Index: asciimatics-1.13.0/tests/test_screen.py
===================================================================
--- asciimatics-1.13.0.orig/tests/test_screen.py	2021-03-06 18:11:13.000000000 +0100
+++ asciimatics-1.13.0/tests/test_screen.py	2022-03-07 13:54:48.895656238 +0100
@@ -5,7 +5,7 @@ from __future__ import print_function
 from __future__ import unicode_literals
 
 import os
-from mock import MagicMock
+from unittest.mock import MagicMock
 from random import randint
 import unittest
 import sys
Index: asciimatics-1.13.0/tests/test_widgets.py
===================================================================
--- asciimatics-1.13.0.orig/tests/test_widgets.py	2021-04-05 16:20:35.000000000 +0200
+++ asciimatics-1.13.0/tests/test_widgets.py	2022-03-07 13:58:16.396856253 +0100
@@ -5,13 +5,12 @@ from __future__ import print_function
 from __future__ import unicode_literals
 from datetime import date, time
 from time import sleep
-from mock import patch
+from unittest.mock import MagicMock,patch
 from builtins import ord
 from builtins import chr
 from builtins import str
 import unittest
 import sys
-from mock.mock import MagicMock
 from asciimatics.event import KeyboardEvent, MouseEvent
 from asciimatics.exceptions import NextScene, StopApplication, InvalidFields
 from asciimatics.scene import Scene
