--- examples/chrony.conf.example3.orig
+++ examples/chrony.conf.example3
@@ -27,12 +27,38 @@
 # you can access at http://support.ntp.org/bin/view/Servers/WebHome or
 # you can use servers from the pool.ntp.org project.
 
-! server ntp1.example.net iburst
-! server ntp2.example.net iburst
-! server ntp3.example.net iburst
-
 ! pool pool.ntp.org iburst
  
+# for Europe:
+! server 0.europe.pool.ntp.org
+! server 1.europe.pool.ntp.org
+! server 2.europe.pool.ntp.org
+! server 3.europe.pool.ntp.org
+
+# for Asia:
+! server 0.asia.pool.ntp.org
+! server 1.asia.pool.ntp.org
+! server 2.asia.pool.ntp.org
+! server 3.asia.pool.ntp.org
+
+# for North America:
+! server 0.north-america.pool.ntp.org
+! server 1.north-america.pool.ntp.org
+! server 2.north-america.pool.ntp.org
+! server 3.north-america.pool.ntp.org
+
+# for South America:
+! server 0.south-america.pool.ntp.org
+! server 1.south-america.pool.ntp.org
+! server 2.south-america.pool.ntp.org
+! server 3.south-america.pool.ntp.org
+
+# for Oceania:
+! server 0.oceania.pool.ntp.org
+! server 1.oceania.pool.ntp.org
+! server 2.oceania.pool.ntp.org
+! server 3.oceania.pool.ntp.org
+
 #######################################################################
 ### AVOIDING POTENTIALLY BOGUS CHANGES TO YOUR CLOCK
 #
@@ -79,7 +105,7 @@
 # immediately so that it doesn't gain or lose any more time.  You
 # generally want this, so it is uncommented.
 
-driftfile /var/lib/chrony/drift
+driftfile /var/lib/chrony/chrony.drift
 
 # If you want to enable NTP authentication with symmetric keys, you will need
 # to uncomment the following line and edit the file to set up the keys.
@@ -165,8 +191,8 @@ ntsdumpdir /var/lib/chrony
 # produce some graphs of your system's timekeeping performance, or you
 # need help in debugging a problem.
 
-! logdir /var/log/chrony
-! log measurements statistics tracking
+logdir /var/log/chrony
+log measurements statistics tracking
 
 # If you have real time clock support enabled (see below), you might want
 # this line instead:
