diff -U 3 -dHrN a/tr/etc/probemodem.sh.in b/tr/etc/probemodem.sh.in
--- a/tr/etc/probemodem.sh.in	2020-07-28 03:29:41.000000000 +0200
+++ b/tr/etc/probemodem.sh.in	2021-10-14 11:07:07.454848194 +0200
@@ -79,10 +79,10 @@
 # directory.
 #
 if [ -e $SPOOL/etc/setup.cache ] && [ ! -e $DIR_LIBDATA/setup.cache ]; then
-    ln $SPOOL/etc/setup.cache $DIR_LIBDATA/setup.cache
+    ln $SPOOL/etc/setup.cache $DIR_LIBDATA/setup.cache || ln -s $SPOOL/etc/setup.cache $DIR_LIBDATA/setup.cache
 fi
 if [ -e $SPOOL/etc/setup.modem ] && [ ! -e $DIR_LIBDATA/setup.modem ]; then
-    ln $SPOOL/etc/setup.modem $DIR_LIBDATA/setup.modem
+    ln $SPOOL/etc/setup.modem $DIR_LIBDATA/setup.modem || ln -s $SPOOL/etc/setup.modem $DIR_LIBDATA/setup.modem
 fi
 
 #
--- a/trunk/etc/faxaddmodem.sh.in
+++ b/trunk/etc/faxaddmodem.sh.in
@@ -92,10 +92,10 @@
 # directory.
 #
 if [ -e $SPOOL/etc/setup.cache ] && [ ! -e $DIR_LIBDATA/setup.cache ]; then
-    ln $SPOOL/etc/setup.cache $DIR_LIBDATA/setup.cache
+    ln $SPOOL/etc/setup.cache $DIR_LIBDATA/setup.cache || ln -s $SPOOL/etc/setup.cache $DIR_LIBDATA/setup.cache
 fi
 if [ -e $SPOOL/etc/setup.modem ] && [ ! -e $DIR_LIBDATA/setup.modem ]; then
-    ln $SPOOL/etc/setup.modem $DIR_LIBDATA/setup.modem
+    ln $SPOOL/etc/setup.modem $DIR_LIBDATA/setup.modem || ln -s $SPOOL/etc/setup.modem $DIR_LIBDATA/setup.modem
 fi
 
 test -f $DIR_LIBDATA/setup.cache || {
--- a/trunk/etc/faxsetup.sh.in
+++ b/trunk/etc/faxsetup.sh.in
@@ -778,10 +778,10 @@
 # directory.
 #
 if [ -e $SPOOL/etc/setup.cache ] && [ ! -e $DIR_LIBDATA/setup.cache ]; then
-    ln $SPOOL/etc/setup.cache $DIR_LIBDATA/setup.cache
+    ln $SPOOL/etc/setup.cache $DIR_LIBDATA/setup.cache || ln -s $SPOOL/etc/setup.cache $DIR_LIBDATA/setup.cache
 fi
 if [ -e $SPOOL/etc/setup.modem ] && [ ! -e $DIR_LIBDATA/setup.modem ]; then
-    ln $SPOOL/etc/setup.modem $DIR_LIBDATA/setup.modem
+    ln $SPOOL/etc/setup.modem $DIR_LIBDATA/setup.modem || ln -s $SPOOL/etc/setup.modem $DIR_LIBDATA/setup.modem
 fi
 
 #
@@ -2204,7 +2204,7 @@
     $RM $DIR_SPOOL/etc/setup.modem
     $MV $DIR_LIBDATA/setup.tmp $DIR_LIBDATA/setup.modem
     $CHMOD 444 $DIR_LIBDATA/setup.modem
-    $LN $DIR_LIBDATA/setup.modem $DIR_SPOOL/etc/setup.modem
+    $LN $DIR_LIBDATA/setup.modem $DIR_SPOOL/etc/setup.modem || $LN -s $DIR_LIBDATA/setup.modem $DIR_SPOOL/etc/setup.modem
 
     $RM $DIR_LIBDATA/setup.cache
     $RM $DIR_SPOOL/etc/setup.cache
@@ -2212,7 +2212,7 @@
      echo '# on' `date` "for ${USER:-$euid}"
      dumpvals |sort)> $DIR_LIBDATA/setup.cache
     $CHMOD 444 $DIR_LIBDATA/setup.cache
-    $LN $DIR_LIBDATA/setup.cache $DIR_SPOOL/etc/setup.cache
+    $LN $DIR_LIBDATA/setup.cache $DIR_SPOOL/etc/setup.cache || $LN -s $DIR_LIBDATA/setup.cache $DIR_SPOOL/etc/setup.cache
 
     Note ""
     Note "Modem support functions written to $DIR_LIBDATA/setup.modem."
--- a/trunk/etc/hylafax.in
+++ b/trunk/etc/hylafax.in
@@ -59,10 +59,10 @@
 # directory.
 #
 if [ -e $SPOOL/etc/setup.cache ] && [ ! -e $DIR_LIBDATA/setup.cache ]; then
-    ln $SPOOL/etc/setup.cache $DIR_LIBDATA/setup.cache
+    ln $SPOOL/etc/setup.cache $DIR_LIBDATA/setup.cache || ln -s $SPOOL/etc/setup.cache $DIR_LIBDATA/setup.cache
 fi
 if [ -e $SPOOL/etc/setup.modem ] && [ ! -e $DIR_LIBDATA/setup.modem ]; then
-    ln $SPOOL/etc/setup.modem $DIR_LIBDATA/setup.modem
+    ln $SPOOL/etc/setup.modem $DIR_LIBDATA/setup.modem || ln -s $SPOOL/etc/setup.modem $DIR_LIBDATA/setup.modem
 fi
 
 test -f $DIR_LIBDATA/setup.cache || {
