diff -U 3 -dHrN -- a/configure b/configure
--- a/configure	2021-07-25 23:16:27.000000000 +0200
+++ b/configure	2022-06-07 13:08:28.354137542 +0200
@@ -2583,7 +2583,7 @@
 				echo '#define TIFFSTRIPBYTECOUNTS uint32_t'
 				echo '#define TIFFVERSION TIFF_VERSION'
 				echo '#define TIFFHEADER TIFFHeader';;
-		4.[0123])	tiff_runlen_t="uint32_t"
+		4.[01234])	tiff_runlen_t="uint32_t"
 				tiff_offset_t="uint64_t"
 				echo '#define TIFFSTRIPBYTECOUNTS uint64_t'
 				echo '#define TIFFVERSION TIFF_VERSION_CLASSIC'
