---
 mcelog.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/mcelog.c
+++ b/mcelog.c
@@ -888,7 +888,6 @@
 "--ping              Send ping command to the currently running mcelog daemon\n"
 "--ignorenodev       Exit silently when the device cannot be opened\n"
 "--file filename     With --ascii read machine check log from filename instead of stdin\n"
-"--logfile filename  Log decoded machine checks in file filename\n"
 "--syslog            Log decoded machine checks in syslog (default stdout or syslog for daemon)\n"
 "--syslog-error      Log decoded machine checks in syslog with error level\n"
 "--no-syslog         Never log anything to syslog\n"
@@ -1113,7 +1112,7 @@
 
 static void modifier_finish(void)
 {
-	if(!foreground && daemon_mode && !logfile && !(syslog_opt & SYSLOG_LOG)) {
+	if(daemon_mode && !logfile && !(syslog_opt & SYSLOG_LOG)) {
 		logfile = logfile_default;
 	}
 	if (logfile) {
