---
 memdb.c |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

Index: mcelog-198/memdb.c
===================================================================
--- mcelog-198.orig/memdb.c
+++ mcelog-198/memdb.c
@@ -440,11 +440,11 @@ void prefill_memdb(int do_dmi)
 		md->location = xstrdup(bl);
 		md->name = xstrdup(dmi_getstring(&d->header, d->device_locator));
 	}
-	if (missed) { 
-		static int warned;
-		if (!warned) {
-			Eprintf("failed to prefill DIMM database from DMI data");
-			warned = 1;
+	if (!missed) {
+		static int db_rill_msg;
+		if (!db_rill_msg) {
+			Gprintf("Prefilled DIMM database from DMI data");
+			db_rill_msg = 1;
 		}
 	}
 }
