diff -ruN libvips-8.4.2.orig/configure.ac libvips-8.4.2/configure.ac
--- libvips-8.4.2.orig/configure.ac	2016-10-12 19:43:25.000000000 +0300
+++ libvips-8.4.2/configure.ac	2016-11-19 19:30:14.183632731 +0300
@@ -925,8 +925,7 @@
   # cppflags not cflags because we want the preproc to see the -I as well
   save_CPPFLAGS="$CPPFLAGS"
   CPPFLAGS="$EXIF_CFLAGS $CPPFLAGS"
-  AC_CHECK_HEADER(exif-data.h,
-    AC_DEFINE(UNTAGGED_EXIF,1,[libexif includes don't need libexif prefix]))
+  AC_CHECK_HEADER(libexif/exif-data.h)
   CPPFLAGS="$save_CPPFLAGS"
 fi
 
