Index: xscreensaver-5.40/driver/windows.c
===================================================================
--- xscreensaver-5.40.orig/driver/windows.c
+++ xscreensaver-5.40/driver/windows.c
@@ -898,7 +898,11 @@ saver_exit (saver_info *si, int status,
       if (bugp)
 	fprintf(real_stderr,
 		"%s: see https://www.jwz.org/xscreensaver/bugs.html\n"
-		"\t\t\tfor bug reporting information.\n\n",
+		"\t\t\tfor bug reporting information.\n\n"
+		"Before reporting any bugs to the upstream,\n"
+		"please check it in the latest version.\n"
+		"If it is fixed there, please report it to\n"
+		"https://bugzilla.opensuse.org/.\n\n",
 		blurb());
 
 # if defined(HAVE_GETCWD)
Index: xscreensaver-5.40/driver/xscreensaver.c
===================================================================
--- xscreensaver-5.40.orig/driver/xscreensaver.c
+++ xscreensaver-5.40/driver/xscreensaver.c
@@ -431,6 +431,10 @@ saver_ehandler (Display *dpy, XErrorEven
    "    The more information you can provide, the better.  But please\n"
    "    report this bug, regardless!\n"
    "\n"
+   "    Before reporting any bugs to the upstream, please check it in the\n"
+   "    latest version.  If it is fixed there, please report it to\n"
+   "    https://bugzilla.opensuse.org/.\n"
+   "\n"
    "#######################################################################\n"
    "\n"
    "\n");
Index: xscreensaver-5.40/driver/xscreensaver.man
===================================================================
--- xscreensaver-5.40.orig/driver/xscreensaver.man
+++ xscreensaver-5.40/driver/xscreensaver.man
@@ -373,6 +373,9 @@ and
 Bugs?  There are no bugs.  Ok, well, maybe.  If you find one, please let
 me know.  https://www.jwz.org/xscreensaver/bugs.html explains how to
 construct the most useful bug reports.
+
+Before reporting any bugs to the upstream, please check it in the latest
+version. If it is fixed there, please report it to https://bugzilla.opensuse.org/.
 .PP
 .TP 4
 .B Locking and root logins
