Index: hacks/webcollage
===================================================================
--- a/hacks/webcollage
+++ b/hacks/webcollage
@@ -827,6 +827,7 @@ sub pick_image_from_rss($$) {
 
 sub pick_dictionary() {
   my @dicts = ("/usr/dict/words",
+               "/var/lib/dict/words",
                "/usr/share/dict/words",
                "/usr/share/lib/dict/words",
                "/usr/share/dict/cracklib-small",
