Index: hacks/Makefile.in
===================================================================
--- a/hacks/Makefile.in
+++ b/hacks/Makefile.in
@@ -18,8 +18,8 @@ datarootdir	= @datarootdir@
 datadir		= @datadir@
 mandir		= @mandir@
 libexecdir	= @libexecdir@
-mansuffix	= 6
-manNdir		= $(mandir)/man$(mansuffix)
+mansuffix	= 6x
+manNdir		= $(mandir)/man6
 
 HACKDIR		= @HACKDIR@
 HACK_CONF_DIR	= @HACK_CONF_DIR@
Index: hacks/glx/Makefile.in
===================================================================
--- a/hacks/glx/Makefile.in
+++ b/hacks/glx/Makefile.in
@@ -19,8 +19,8 @@ datarootdir	= @datarootdir@
 datadir		= @datadir@
 mandir		= @mandir@
 libexecdir	= @libexecdir@
-mansuffix	= 6
-manNdir		= $(mandir)/man$(mansuffix)
+mansuffix	= 6x
+manNdir		= $(mandir)/man6
 
 HACKDIR		= @HACKDIR@
 HACK_CONF_DIR	= @HACK_CONF_DIR@
