Index: xscreensaver-6.01/driver/xscreensaver.c
===================================================================
--- xscreensaver-6.01.orig/driver/xscreensaver.c
+++ xscreensaver-6.01/driver/xscreensaver.c
@@ -811,6 +811,10 @@ error_handler (Display *dpy, XErrorEvent
    "    The more information you can provide, the better.  But please\n"
    "    report this bug, regardless!\n"
    "\n"
+   "    Before reporting any bugs to the upstream, please check it in the\n"
+   "    latest version.  If it is fixed there, please report it to\n"
+   "    https://bugzilla.opensuse.org/.\n"
+   "\n"
    "#######################################################################\n"
    "\n"
    "\n");
Index: xscreensaver-6.01/driver/xscreensaver.man
===================================================================
--- xscreensaver-6.01.orig/driver/xscreensaver.man
+++ xscreensaver-6.01/driver/xscreensaver.man
@@ -985,6 +985,9 @@ power-management-related activity detect
 https://www.jwz.org/xscreensaver/bugs.html explains how to write the most
 useful bug reports.  If you find a bug, please let me know!
 
+Before reporting any bugs to the upstream, please check it in the latest
+version. If it is fixed there, please report it to https://bugzilla.opensuse.org/.
+
 .SH ENVIRONMENT
 .PP
 .TP 8
