change default background to black

================================================================================
Index: utils/resources.c
===================================================================
--- a/utils/resources.c
+++ b/utils/resources.c
@@ -176,7 +176,7 @@ get_pixel_resource (Display *dpy, Colorm
   if (s) free (s);
 
   {
-    Bool black_p = (strlen(res_class) >= 10 &&
+    Bool black_p = ! (strlen(res_class) >= 10 &&
                     !strcasecmp ("Background",
                                  res_class + strlen(res_class) - 10));
     if (!ok)
