diff --git a/scripts/build-udev-rules.sh b/scripts/build-udev-rules.sh
index ca3967d5..18940c6a 100755
--- a/scripts/build-udev-rules.sh
+++ b/scripts/build-udev-rules.sh
@@ -23,8 +23,7 @@ UDEV_LINE='#---------------------------------------------------------------#\n'
 #-----------------------------------------------------------------------------#
 echo -e "Adding Static Headers"
 UDEV_HEADER=${UDEV_LINE}'#  OpenRGB udev rules - Git Commit: '${GIT_SHORT_HASH:0:8}'                    #\n'${UDEV_LINE}'\n'
-UDEV_HEADER+=${UDEV_LINE}'#  User I2C/SMBus Access                                        #\n'${UDEV_LINE}'KERNEL=="i2c-[0-99]*", TAG+="uaccess"\n\n'
-UDEV_HEADER+=${UDEV_LINE}'#  Super I/O Access                                             #\n'${UDEV_LINE}'KERNEL=="port", TAG+="uaccess"\n\n'
+UDEV_HEADER+=${UDEV_LINE}'#  User I2C/SMBus Access                                        #\n'${UDEV_LINE}'KERNEL=="i2c-[0-99]*", RUN="/usr/bin/setfacl -m g:openrgb:rw /dev/%k"\n\n'
 
 # Faustus rules
 ASUS_TUF_DEVICES=('blue' 'flags' 'green' 'mode' 'red' 'set' 'speed')
