 Index: OpenRGB-release_0.8/dependencies/hueplusplus-1.0.0/src/EntertainmentMode.cpp
===================================================================
--- OpenRGB-release_0.8.orig/dependencies/hueplusplus-1.0.0/src/EntertainmentMode.cpp
+++ OpenRGB-release_0.8/dependencies/hueplusplus-1.0.0/src/EntertainmentMode.cpp
@@ -21,7 +21,6 @@
 **/
 
 #include "hueplusplus/EntertainmentMode.h"
-#include "mbedtls/certs.h"
 #include "mbedtls/ctr_drbg.h"
 #include "mbedtls/debug.h"
 #include "mbedtls/entropy.h"

