Subject: use openssl for sha1
Upstream: no

===================================================================

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,6 +87,7 @@ ELSEIF(libdiscid_OS STREQUAL "generic")
 ELSE()
     # unix platforms are the standard/default case
     SET(libdiscid_OSDEP_SRCS src/toc.c src/unix.c src/disc_${libdiscid_OS}.c)
+    FIND_PACKAGE(OpenSSL REQUIRED)
     IF(libdiscid_OS STREQUAL "darwin") # Extra libraries needed
         FIND_LIBRARY(COREFOUNDATION_LIBRARY CoreFoundation)
         FIND_LIBRARY(IOKIT_LIBRARY IOKit)
@@ -104,8 +105,8 @@ IF(NOT MUSICBRAINZ5_FOUND)
     SET(MUSICBRAINZ5_INCLUDE_DIRS "")
 ENDIF()

-ADD_LIBRARY(libdiscid SHARED ${libdiscid_OSDEP_SRCS} ${libdiscid_RCS} src/base64.c src/disc.c src/sha1.c)
-TARGET_LINK_LIBRARIES(libdiscid ${libdiscid_OSDEP_LIBS})
+ADD_LIBRARY(libdiscid SHARED ${libdiscid_OSDEP_SRCS} ${libdiscid_RCS} src/base64.c src/disc.c)
+TARGET_LINK_LIBRARIES(libdiscid ${libdiscid_OSDEP_LIBS} OpenSSL::Crypto)
 SET_TARGET_PROPERTIES(libdiscid PROPERTIES
     OUTPUT_NAME discid
     VERSION ${libdiscid_VERSION}

==================================================================

--- a/src/disc.c
+++ b/src/disc.c
@@ -33,8 +33,8 @@
 #include <string.h>
 #include <assert.h>
 #include <limits.h>
+#include <openssl/evp.h>

-#include "sha1.h"
 #include "base64.h"

 #include "discid/discid.h"
@@ -361,31 +361,35 @@ char *discid_get_version_string(void) {
  * The DiscID is placed in the provided string buffer.
  */
 static void create_disc_id(mb_disc_private *d, char buf[]) {
-	SHA_INFO	sha;
-	unsigned char	digest[20], *base64;
-	unsigned long	size;
+	unsigned char	*base64;
+	unsigned long 	size;
+	unsigned char	digest[EVP_MAX_MD_SIZE];
 	char		tmp[17]; /* for 8 hex digits (16 to avoid trouble) */
 	int		i;
+    EVP_MD_CTX 	*sha;
+	unsigned int 	sha_len;

 	assert(d != NULL);
 	assert(d->success);

-	sha_init(&sha);
+	sha = EVP_MD_CTX_create();
+	EVP_DigestInit_ex(sha, EVP_sha1(), NULL);

 	sprintf(tmp, "%02X", d->first_track_num);
-	sha_update(&sha, (unsigned char *) tmp, strlen(tmp));
+	EVP_DigestUpdate(sha, (unsigned char *) tmp, strlen(tmp));

 	sprintf(tmp, "%02X", d->last_track_num);
-	sha_update(&sha, (unsigned char *) tmp, strlen(tmp));
+	EVP_DigestUpdate(sha, (unsigned char *) tmp, strlen(tmp));

 	for (i = 0; i < 100; i++) {
 		sprintf(tmp, "%08X", d->track_offsets[i]);
-		sha_update(&sha, (unsigned char *) tmp, strlen(tmp));
+		EVP_DigestUpdate(sha, (unsigned char *) tmp, strlen(tmp));
 	}

-	sha_final(digest, &sha);
-
-	base64 = rfc822_binary(digest, sizeof(digest), &size);
+	EVP_DigestFinal_ex(sha, digest, &sha_len);
+	EVP_MD_CTX_destroy(sha);
+
+	base64 = rfc822_binary(digest, sha_len , &size);

 	memcpy(buf, base64, size);
 	buf[size] = '\0';
