Author: Bernhard M. Wiedemann <bwiedemann suse de>
Date: 2020-07-30

https://bugzilla.opensuse.org/show_bug.cgi?id=1100677

packages do not build reproducibly from compile-time CPU-detection

Index: InsightToolkit-5.1.0/CMake/ITKSetStandardCompilerFlags.cmake
===================================================================
--- InsightToolkit-5.1.0.orig/CMake/ITKSetStandardCompilerFlags.cmake
+++ InsightToolkit-5.1.0/CMake/ITKSetStandardCompilerFlags.cmake
@@ -241,7 +241,7 @@ function(check_compiler_optimization_fla
     set(InstructionSetOptimizationFlags
        # https://gcc.gnu.org/onlinedocs/gcc-4.8.0/gcc/i386-and-x86_002d64-Options.html
        # NOTE the corei7 release date was 2008
-       -mtune=native # Tune the code for the computer used compile ITK, but allow running on generic cpu archetectures
+       #-mtune=native # boo#1100677 - disabled to not consider build machine CPU for reproducible results
        -march=corei7 # Use ABI settings to support corei7 (circa 2008 ABI feature sets, core-avx circa 2013)
        )
   endif()
