---
 pam/crond |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: cronie-cronie-1.7.0/pam/crond
===================================================================
--- cronie-cronie-1.7.0.orig/pam/crond
+++ cronie-cronie-1.7.0/pam/crond
@@ -10,4 +10,4 @@ auth     include        common-auth
 account  include        common-account
 password include        common-password
 session  optional       pam_keyinit.so force revoke
-session  include        common-session
+session  include        common-session-nonlogin
