--- aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-cognitosync/src/main/java/com/amazonaws/services/cognitosync/model/transform/ListRecordsResultJsonUnmarshaller.java	2022-03-21 19:35:10.061186250 +0100
+++ aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-cognitosync/src/main/java/com/amazonaws/services/cognitosync/model/transform/ListRecordsResultJsonUnmarshaller.java	2022-03-21 19:52:54.219667109 +0100
@@ -21,6 +21,7 @@
 import java.math.*;
 import java.nio.ByteBuffer;
 
+import com.amazonaws.services.cognitosync.model.Record;
 import com.amazonaws.services.cognitosync.model.*;
 import com.amazonaws.transform.SimpleTypeJsonUnmarshallers.*;
 import com.amazonaws.transform.*;
--- aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-cognitosync/src/main/java/com/amazonaws/services/cognitosync/model/transform/RecordJsonMarshaller.java	2022-03-21 19:35:10.061186250 +0100
+++ aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-cognitosync/src/main/java/com/amazonaws/services/cognitosync/model/transform/RecordJsonMarshaller.java	2022-03-21 19:50:20.562729618 +0100
@@ -20,6 +20,7 @@
 import java.util.List;
 
 import com.amazonaws.AmazonClientException;
+import com.amazonaws.services.cognitosync.model.Record;
 import com.amazonaws.services.cognitosync.model.*;
 import com.amazonaws.transform.Marshaller;
 import com.amazonaws.util.BinaryUtils;
--- aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-cognitosync/src/main/java/com/amazonaws/services/cognitosync/model/transform/RecordJsonUnmarshaller.java	2022-03-21 19:35:10.061186250 +0100
+++ aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-cognitosync/src/main/java/com/amazonaws/services/cognitosync/model/transform/RecordJsonUnmarshaller.java	2022-03-21 19:51:33.159172537 +0100
@@ -21,6 +21,7 @@
 import java.math.*;
 import java.nio.ByteBuffer;
 
+import com.amazonaws.services.cognitosync.model.Record;
 import com.amazonaws.services.cognitosync.model.*;
 import com.amazonaws.transform.SimpleTypeJsonUnmarshallers.*;
 import com.amazonaws.transform.*;
--- aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-cognitosync/src/main/java/com/amazonaws/services/cognitosync/model/transform/UpdateRecordsResultJsonUnmarshaller.java	2022-03-21 19:35:10.061186250 +0100
+++ aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-cognitosync/src/main/java/com/amazonaws/services/cognitosync/model/transform/UpdateRecordsResultJsonUnmarshaller.java	2022-03-21 19:54:09.372125626 +0100
@@ -21,6 +21,7 @@
 import java.math.*;
 import java.nio.ByteBuffer;
 
+import com.amazonaws.services.cognitosync.model.Record;
 import com.amazonaws.services.cognitosync.model.*;
 import com.amazonaws.transform.SimpleTypeJsonUnmarshallers.*;
 import com.amazonaws.transform.*;
--- aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-dynamodb/src/main/java/com/amazonaws/services/dynamodbv2/model/transform/GetRecordsResultJsonUnmarshaller.java	2022-03-21 19:35:10.329187881 +0100
+++ aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-dynamodb/src/main/java/com/amazonaws/services/dynamodbv2/model/transform/GetRecordsResultJsonUnmarshaller.java	2022-03-21 19:48:59.754236598 +0100
@@ -21,6 +21,7 @@
 import java.math.*;
 import java.nio.ByteBuffer;
 
+import com.amazonaws.services.dynamodbv2.model.Record;
 import com.amazonaws.services.dynamodbv2.model.*;
 import com.amazonaws.transform.SimpleTypeJsonUnmarshallers.*;
 import com.amazonaws.transform.*;
--- aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-dynamodb/src/main/java/com/amazonaws/services/dynamodbv2/model/transform/RecordJsonMarshaller.java	2022-03-21 19:35:10.329187881 +0100
+++ aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-dynamodb/src/main/java/com/amazonaws/services/dynamodbv2/model/transform/RecordJsonMarshaller.java	2022-03-21 19:45:21.080904784 +0100
@@ -20,6 +20,7 @@
 import java.util.List;
 
 import com.amazonaws.AmazonClientException;
+import com.amazonaws.services.dynamodbv2.model.Record;
 import com.amazonaws.services.dynamodbv2.model.*;
 import com.amazonaws.transform.Marshaller;
 import com.amazonaws.util.BinaryUtils;
--- aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-dynamodb/src/main/java/com/amazonaws/services/dynamodbv2/model/transform/RecordJsonUnmarshaller.java	2022-03-21 19:35:10.329187881 +0100
+++ aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-dynamodb/src/main/java/com/amazonaws/services/dynamodbv2/model/transform/RecordJsonUnmarshaller.java	2022-03-21 19:36:09.749549498 +0100
@@ -21,6 +21,7 @@
 import java.math.*;
 import java.nio.ByteBuffer;
 
+import com.amazonaws.services.dynamodbv2.model.Record;
 import com.amazonaws.services.dynamodbv2.model.*;
 import com.amazonaws.transform.SimpleTypeJsonUnmarshallers.*;
 import com.amazonaws.transform.*;
--- aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-kinesis/src/main/java/com/amazonaws/services/kinesis/model/transform/GetRecordsResultJsonUnmarshaller.java	2022-03-21 19:35:10.049186176 +0100
+++ aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-kinesis/src/main/java/com/amazonaws/services/kinesis/model/transform/GetRecordsResultJsonUnmarshaller.java	2022-03-21 20:27:11.176216227 +0100
@@ -21,6 +21,7 @@
 import java.math.*;
 import java.nio.ByteBuffer;
 
+import com.amazonaws.services.kinesis.model.Record;
 import com.amazonaws.services.kinesis.model.*;
 import com.amazonaws.transform.SimpleTypeJsonUnmarshallers.*;
 import com.amazonaws.transform.*;
--- aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-kinesis/src/main/java/com/amazonaws/services/kinesis/model/transform/RecordJsonMarshaller.java	2022-03-21 19:35:10.053186202 +0100
+++ aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-kinesis/src/main/java/com/amazonaws/services/kinesis/model/transform/RecordJsonMarshaller.java	2022-03-21 19:55:44.348705090 +0100
@@ -20,6 +20,7 @@
 import java.util.List;
 
 import com.amazonaws.AmazonClientException;
+import com.amazonaws.services.kinesis.model.Record;
 import com.amazonaws.services.kinesis.model.*;
 import com.amazonaws.transform.Marshaller;
 import com.amazonaws.util.BinaryUtils;
--- aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-kinesis/src/main/java/com/amazonaws/services/kinesis/model/transform/RecordJsonUnmarshaller.java	2022-03-21 19:35:10.053186202 +0100
+++ aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-kinesis/src/main/java/com/amazonaws/services/kinesis/model/transform/RecordJsonUnmarshaller.java	2022-03-21 19:58:14.637622016 +0100
@@ -21,6 +21,7 @@
 import java.math.*;
 import java.nio.ByteBuffer;
 
+import com.amazonaws.services.kinesis.model.Record;
 import com.amazonaws.services.kinesis.model.*;
 import com.amazonaws.transform.SimpleTypeJsonUnmarshallers.*;
 import com.amazonaws.transform.*;
--- aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-kinesis/src/main/java/com/amazonaws/services/kinesisfirehose/model/transform/PutRecordBatchRequestMarshaller.java	2022-03-21 19:35:10.053186202 +0100
+++ aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-kinesis/src/main/java/com/amazonaws/services/kinesisfirehose/model/transform/PutRecordBatchRequestMarshaller.java	2022-03-21 20:27:41.132399224 +0100
@@ -26,6 +26,7 @@
 import com.amazonaws.Request;
 import com.amazonaws.DefaultRequest;
 import com.amazonaws.http.HttpMethodName;
+import com.amazonaws.services.kinesisfirehose.model.Record;
 import com.amazonaws.services.kinesisfirehose.model.*;
 import com.amazonaws.transform.Marshaller;
 import com.amazonaws.util.BinaryUtils;
--- aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-kinesis/src/main/java/com/amazonaws/services/kinesisfirehose/model/transform/RecordJsonMarshaller.java	2022-03-21 19:35:10.053186202 +0100
+++ aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-kinesis/src/main/java/com/amazonaws/services/kinesisfirehose/model/transform/RecordJsonMarshaller.java	2022-03-21 20:26:44.896055688 +0100
@@ -20,6 +20,7 @@
 import java.util.List;
 
 import com.amazonaws.AmazonClientException;
+import com.amazonaws.services.kinesisfirehose.model.Record;
 import com.amazonaws.services.kinesisfirehose.model.*;
 import com.amazonaws.transform.Marshaller;
 import com.amazonaws.util.BinaryUtils;
--- aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-kinesis/src/main/java/com/amazonaws/services/kinesisfirehose/model/transform/RecordJsonUnmarshaller.java	2022-03-21 19:35:10.053186202 +0100
+++ aws-sdk-java-9883b981ab5103cc6944fbf8f3b973994777350f/aws-java-sdk-kinesis/src/main/java/com/amazonaws/services/kinesisfirehose/model/transform/RecordJsonUnmarshaller.java	2022-03-21 19:57:15.677262296 +0100
@@ -21,6 +21,7 @@
 import java.math.*;
 import java.nio.ByteBuffer;
 
+import com.amazonaws.services.kinesisfirehose.model.Record;
 import com.amazonaws.services.kinesisfirehose.model.*;
 import com.amazonaws.transform.SimpleTypeJsonUnmarshallers.*;
 import com.amazonaws.transform.*;
