Index: gkrellm-2.5.1/src/gkrellm.h
===================================================================
--- gkrellm-2.5.1.orig/src/gkrellm.h
+++ gkrellm-2.5.1/src/gkrellm.h
@@ -118,6 +118,9 @@ G_BEGIN_DECLS
 #if !defined(SYSTEM_THEMES_DIR)
 #define SYSTEM_THEMES_DIR "/usr/share/gkrellm2/themes"
 #endif
+#if !defined(SYSTEM64_PLUGINS_DIR)
+#define	SYSTEM64_PLUGINS_DIR		"/usr/lib64/gkrellm2/plugins"
+#endif
 
 #if !defined(LOCAL_PLUGINS_DIR)
 #define LOCAL_PLUGINS_DIR "/usr/local/lib/gkrellm2/plugins"
Index: gkrellm-2.5.1/src/plugins.c
===================================================================
--- gkrellm-2.5.1.orig/src/plugins.c
+++ gkrellm-2.5.1/src/plugins.c
@@ -1254,6 +1254,10 @@ gkrellm_plugins_load(void)
 #if defined(SYSTEM_PLUGINS_DIR)
 	scan_for_plugins(SYSTEM_PLUGINS_DIR);
 #endif
+
+#if defined(SYSTEM64_PLUGINS_DIR)
+	scan_for_plugins(SYSTEM64_PLUGINS_DIR);
+#endif
 	}
 
 #if 0
