From 0efb2331f1c39c9665fb6e92e83c7d080b877de1 Mon Sep 17 00:00:00 2001
From: Jan Kara <jack@suse.cz>
Date: Mon, 1 Apr 2019 17:28:34 +0200
Subject: [PATCH] warnquota: Improve examples in warnquota.conf

Signed-off-by: Jan Kara <jack@suse.cz>
---
 warnquota.conf | 104 ++++++++++++++++++++++++++++++++++---------------
 1 file changed, 72 insertions(+), 32 deletions(-)

diff --git a/warnquota.conf b/warnquota.conf
index b06f81f..60b0672 100644
--- a/warnquota.conf
+++ b/warnquota.conf
@@ -1,21 +1,37 @@
-# this is an example warnquota.conf
+###################################################################
+# Configuration file for the warnquota utility
+# File Format:
+# ^^^^^^^^^^^^
+# (1) lines begining with # or ; are comments
+# (2) blank lines are ignored
+# (3) other lines have the form 'value = string'
+# (4) strings may be quoted (double quotes) but they don't have to
+# (5) strings may end with backslash in order to continue
+#     on the next line
+# (6) line breaks are marked with '|' character
+###################################################################
+
+#
+# Comment this out or remove it once you have edited this config file
 #
-; ; and # type comments are allowed
-# and even blank lines
+FAIL        = "configure /etc/warnquota.conf before running warnquota"
 
-# values can be quoted:
-#MAIL_CMD 	= "/usr/my/sendmail/instead/sendmail -t"
-MAIL_CMD 	= "/bin/echo"
-FROM 		= "bas@example.com"
-# but they don't have to be:
-SUBJECT 	= Hey, user, clean up your account!
-CC_TO 		= "sysadm@example.com"
+#
+# command used for sending mails
+#
+MAIL_CMD = "/usr/lib/sendmail -t"
+
+#
+# Standard mail fields
+FROM        = "root@localhost"
+SUBJECT 	= "Your account quota has exceeded!"
+CC_TO 		= "root@localhost"
 # If you set this variable CC will be used only when user has less than
 # specified grace time left (examples of possible times: 5 seconds, 1 minute,
 # 12 hours, 5 days)
 # CC_BEFORE = 2 days
-SUPPORT 	= "support@example.com"
-PHONE 		= "(123) 456-1111 or (222) 333-4444"
+SUPPORT 	= "root@localhost"
+PHONE 		= "123 456 789"
 # Text in the beginning of the mail (if not specified, default text is used)
 # This way text can be split to more lines
 # Line breaks are done by '|' character
@@ -45,33 +61,57 @@ GROUP_SIGNATURE	= See you!|			Your admin|
 #
 #Here you can set a charset for emails sent by warnquota (e.g. UTF-8)
 #CHARSET	= 
-# If you are using LDAP mail lookups.
+
+##############################################################
+# Configuration for LDAP (if you are using LDAP mail lookups)
 # host, port, tls, binddn, and bindpw are straight forward.
-# LDAP_BASEDN is your search base dn
-# LDAP_SEARCH_ATTRIBUTE is the attr for the value you are looking for
-# LDAP_MAIL_ATTRIBUTE is the attribute you want used for the mail address
-# LDAP_DEFAULT_MAIL_DOMAIN is the default domain
-#    if the attribute isn't found
-# if binddn and bindpw are blank or left out, an anonymous bind is used
-# 
-# LDAP_MAIL = false # or true if you want to use it
-# If you have at least LDAP 2.3 installed, you should use LDAP_URI
+##############################################################
+# Your search base dn
+#
+# LDAP_BASEDN
+
+# Your search bind dn
+#
+# LDAP_BINDDN
+
+# Your search bind password
+#
+# LDAP_BINDPW
+
+# The attr for the value you are looking for
+#
+# LDAP_SEARCH_ATTRIBUTE
+
+# The attribute you want used for the mail address
+#
+# LDAP_MAIL_ATTRIBUTE
+
+# The default domain if the attribute isn't found
+#
+# LDAP_DEFAULT_MAIL_DOMAIN
+
+# Whether LDAP support should be used
+#
+# LDAP_MAIL = false
+
+# Ldap server. For LDAP >= 2.3 use
+#
 # LDAP_URI = ldaps://my.server:389
-# Otherwise you should specify LDAP_HOST and LDAP_PORT
-# LDAP_HOST = ldap
+#
+# For older LDAP libraries use
+#
+# LDAP_HOST = my.server
 # LDAP_PORT = 389
-# LDAP_TLS = false (false|never|allow|try|demand) use StarTLS
-#   false - don't use starTLS
+
+# TLS handling for the connection
+#
+# LDAP_TLS = (false|never|allow|try|demand)
+#   false - don't use start TLS
 #   never - don't ask for a certificate
 #   allow - request certificate, proceed even if not verified
 #   try - request certificate, terminate if bad, proceed if not sent
 #   demand - request certificate, proceed only if verified
-# LDAP_BINDDN = uid=ReadOnlyUser,o=YourOrg
-# LDAP_BINDPW = YourReadOnlyUserPassword
-# LDAP_BASEDN = YourSearchBase
-# LDAP_SEARCH_ATTRIBUTE = uid
-# LDAP_MAIL_ATTRIBUTE = mailLocalAddress
-# LDAP_DEFAULT_MAIL_DOMAIN = YourDefaultMailDomain.com
+
 #
 # end of example warnquota.conf file
 #
-- 
2.22.0

