From 4f2257d533bd461bda6ce58ac21c7dc6995cf7c7 Mon Sep 17 00:00:00 2001
From: Martin Herkt <lachs0r@srsfckn.biz>
Date: Fri, 29 Jun 2018 02:50:20 +0200
Subject: Do not disable _FORTIFY_SOURCE

---
 build/cmake/aom_configure.cmake | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/build/cmake/aom_configure.cmake b/build/cmake/aom_configure.cmake
index 5d782aaf9..cb77ff4d9 100644
--- a/build/cmake/aom_configure.cmake
+++ b/build/cmake/aom_configure.cmake
@@ -290,9 +290,6 @@ else()
     add_compiler_flag_if_supported("-Werror")
   endif()
 
-  if("${CMAKE_BUILD_TYPE}" MATCHES "Rel")
-    add_compiler_flag_if_supported("-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0")
-  endif()
   add_compiler_flag_if_supported("-D_LARGEFILE_SOURCE")
   add_compiler_flag_if_supported("-D_FILE_OFFSET_BITS=64")
 endif()
-- 
2.18.0

