Index: xemacs-21.5.29/etc/Emacs.ad
===================================================================
--- xemacs-21.5.29/etc/Emacs.ad
+++ xemacs-21.5.29/etc/Emacs.ad
@@ -65,6 +65,19 @@
 ! Note that by default, the pointer foreground and background are the same
 ! as the default face.
 
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+! Fonts for Xft ONLY:
+XEmacs*Tabs.fcFontName: sans-serif
+XEmacs*menubar.fcFontName: sans-serif:size=10
+XEmacs.modeline.attributeFont: sans-serif
+XEmacs.default.attributeFont: sans-serif
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+!! FontSet for menus when you use --with-xfs or --with-xim=xlib.
+XEmacs*FontSet			-*-helvetica-bold-r-*--*-120-*-*-*-*-*-*, \
+				-*-helvetica-bold-r-*--*-120-*-*-*-*-iso10646-1, \
+				*
+
 ! Set the menubar colors.  This overrides the default foreground and
 ! background colors specified above.
 *menubar*Foreground:			Gray30
@@ -222,36 +235,36 @@
 ! *menubar*FontSet:	-*-helvetica-bold-r-*-*-*-120-*-*-*-*-iso8859-*, \
 !			-*-*-*-*-*-*-*-120-*-jisx0208.1983-0
 !
-*menubar*Font:			-*-helvetica-bold-r-*-*-*-120-*-*-*-*-iso8859-*
-*popup*Font:			-*-helvetica-bold-r-*-*-*-120-*-*-*-*-iso8859-*
-*menubar*FontSet:		-*-helvetica-bold-r-*-*-*-120-*-*-*-*-iso8859-*, \
- 			        -*-*-*-*-*-*-*-120-*-iso10646-1, \
- 			        -*-*-*-*-*-*-*-120-*-jisx0208.1983-0, \
- 			        -*-*-*-*-*-*-*-120-*-jisx0201.1976-0
-*popup*FontSet:			-*-helvetica-bold-r-*-*-*-120-*-*-*-*-iso8859-*, \
-   			        -*-*-*-*-*-*-*-120-*-iso10646-1, \
- 			        -*-*-*-*-*-*-*-120-*-jisx0208.1983-0, \
- 			        -*-*-*-*-*-*-*-120-*-jisx0201.1976-0
+*menubar*Font:			-*-helvetica-bold-r-*--*-120-*-*-*-*-*-*
+*popup*Font:			-*-helvetica-bold-r-*--*-120-*-*-*-*-*-*
+*menubar*FontSet:		-*-helvetica-bold-r-*--*-120-*-*-*-*-*-*, \
+ 			        -*-*-*-*-*--*-120-*-iso10646-1, \
+ 			        -*-*-*-*-*--*-120-*-jisx0208.1983-0, \
+ 			        -*-*-*-*-*--*-120-*-jisx0201.1976-0
+*popup*FontSet:			-*-helvetica-bold-r-*--*-120-*-*-*-*-*-*, \
+   			        -*-*-*-*-*--*-120-*-iso10646-1, \
+ 			        -*-*-*-*-*--*-120-*-jisx0208.1983-0, \
+ 			        -*-*-*-*-*--*-120-*-jisx0201.1976-0
 
 ! Gui elements share this font
 !
-Emacs.gui-element.attributeFont:  -*-helvetica-medium-r-*-*-*-120-*-*-*-*-iso8859-*
+Emacs.gui-element.attributeFont:  -*-helvetica-medium-r-*--*-120-*-*-*-*-*-*
 
 ! Font in the Motif dialog boxes.
 ! (Motif uses `fontList' while most other things use `font' - if you don't
 ! know why you probably don't want to.)
 ! 
-*XmDialogShell*FontList:	-*-helvetica-bold-r-*-*-*-120-*-*-*-*-iso8859-*
-*XmTextField*FontList:		-*-courier-medium-r-*-*-*-120-*-*-*-*-iso8859-*
-*XmText*FontList:		-*-courier-medium-r-*-*-*-120-*-*-*-*-iso8859-*
-*XmList*FontList:		-*-courier-medium-r-*-*-*-120-*-*-*-*-iso8859-*
+*XmDialogShell*FontList:	-*-helvetica-bold-r-*--*-120-*-*-*-*-*-*
+*XmTextField*FontList:		-*-fixed-medium-r-*--*-120-*-*-*-*-*-*
+*XmText*FontList:		-*-fixed-medium-r-*--*-120-*-*-*-*-*-*
+*XmList*FontList:		-*-fixed-medium-r-*--*-120-*-*-*-*-*-*
 
 ! Font in the Athena dialog boxes.
 ! I think 14-point looks nicer than 12-point.
 ! Some people use 12-point anyway because you get more text, but
 ! there's no purpose at all in doing this for dialog boxes.
 
-*Dialog*Font:			-*-helvetica-bold-r-*-*-*-140-*-*-*-*-iso8859-*
+*Dialog*Font:			-*-helvetica-bold-r-*--*-140-*-*-*-*-*-*
 
 ! Dialog box translations.
 ! =======================
Index: xemacs-21.5.29/etc/sample.Xresources
===================================================================
--- xemacs-21.5.29/etc/sample.Xresources
+++ xemacs-21.5.29/etc/sample.Xresources
@@ -238,32 +238,35 @@ XEmacs*toolBarShadowThickness:		2
 ! There is no harm in having both resources set, except for the confusion
 ! you suffer.  Sorry; that's the price of backward compatibility.
 !
-*menubar*Font:		-*-helvetica-bold-r-*-*-*-120-*-*-*-*-iso8859-*
-*popup*Font:		-*-helvetica-bold-r-*-*-*-120-*-*-*-*-iso8859-*
-*menubar*FontSet:	-*-helvetica-bold-r-*-*-*-120-*-*-*-*-iso8859-*, \
-			-*-*-*-*-*-*-*-120-*-iso10646-1, \
-			-*-*-*-*-*-*-*-120-*-jisx0208.1983-0, \
-			-*-*-*-*-*-*-*-120-*-jisx0201.1976-0
-*popup*FontSet:		-*-helvetica-bold-r-*-*-*-120-*-*-*-*-iso8859-*, \
-			-*-*-*-*-*-*-*-120-*-iso10646-1, \
-			-*-*-*-*-*-*-*-120-*-jisx0208.1983-0, \
-			-*-*-*-*-*-*-*-120-*-jisx0201.1976-0
+XEmacs*menubar*Font:		-*-helvetica-bold-r-*--*-120-*-*-*-*-*-*
+XEmacs*popup*Font:		-*-helvetica-bold-r-*--*-120-*-*-*-*-*-*
+XEmacs*FontSet			-*-helvetica-bold-r-*--*-120-*-*-*-*-*-*, \
+				-*-helvetica-bold-r-*--*-120-*-*-*-*-iso10646-1, \
+				*
+XEmacs*menubar*FontSet:		-*-helvetica-bold-r-*--*-120-*-*-*-*-*-*, \
+				-*-*-*-*-*--*-120-*-iso10646-1, \
+				-*-*-*-*-*--*-120-*-jisx0208.1983-0, \
+				-*-*-*-*-*--*-120-*-jisx0201.1976-0
+XEmacs*popup*FontSet:		-*-helvetica-bold-r-*--*-120-*-*-*-*-*-*, \
+				-*-*-*-*-*--*-120-*-iso10646-1, \
+				-*-*-*-*-*--*-120-*-jisx0208.1983-0, \
+				-*-*-*-*-*--*-120-*-jisx0201.1976-0
 
 ! Font in the Motif dialog boxes.
 ! (Motif uses `fontList' while most other things use `font' - if you don't
 ! know why you probably don't want to.)
 ! 
-XEmacs*XmDialogShell*FontList:	-*-helvetica-bold-r-*-*-*-120-*-*-*-*-iso8859-*
-XEmacs*XmTextField*FontList:	-*-courier-medium-r-*-*-*-120-*-*-*-*-iso8859-*
-XEmacs*XmText*FontList:		-*-courier-medium-r-*-*-*-120-*-*-*-*-iso8859-*
-XEmacs*XmList*FontList:		-*-courier-medium-r-*-*-*-120-*-*-*-*-iso8859-*
+XEmacs*XmDialogShell*FontList:	-*-helvetica-bold-r-*--*-120-*-*-*-*-*-*
+XEmacs*XmTextField*FontList:	-*-courier-medium-r-*--*-120-*-*-*-*-*-*
+XEmacs*XmText*FontList:		-*-courier-medium-r-*--*-120-*-*-*-*-*-*
+XEmacs*XmList*FontList:		-*-courier-medium-r-*--*-120-*-*-*-*-*-*
 
 ! Font in the Athena dialog boxes.
 ! I think 14-point looks nicer than 12-point.
 ! Some people use 12-point anyway because you get more text, but
 ! there's no purpose at all in doing this for dialog boxes.
 
-XEmacs*Dialog*Font:		-*-helvetica-bold-r-*-*-*-140-*-*-*-*-iso8859-*
+XEmacs*Dialog*Font:		-*-helvetica-bold-r-*--*-140-*-*-*-*-*-*
 
 ! Dialog box translations.
 ! =======================
