--- a/CMakeLists.txt.orig	2020-03-03 10:41:54.921061882 +0100
+++ b/CMakeLists.txt	2020-03-03 10:42:22.933104452 +0100
@@ -29,7 +29,7 @@
 set(ORTP_MICRO_VERSION ${PROJECT_VERSION_PATCH})
 set(ORTP_VERSION ${PROJECT_VERSION})
 set(ORTP_SO_VERSION "14") # incremented on September 17, 2019
-set(ORTP_DOC_VERSION "${ORTP_VERSION_MAJOR}.${ORTP_VERSION_MINOR}")
+set(ORTP_DOC_VERSION "${ORTP_MAJOR_VERSION}.${ORTP_MINOR_VERSION}")
 
 
 option(ENABLE_SHARED "Build shared library." YES)
@@ -177,7 +177,7 @@
 		)
 		add_custom_target(ortp-html-doc ALL DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/doc/html/index.html")
 		install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/doc/html"
-			DESTINATION "${CMAKE_INSTALL_DATADIR}/doc/ortp-${ORTP_VERSION}")
+			DESTINATION "${CMAKE_INSTALL_DATADIR}/doc/ortp-${ORTP_DOC_VERSION}")
 	endif()
 endif()
 
