--- tamago/eggrc	Sat Jan 27 19:53:13 2001
+++ tamago/eggrc.new	Wed Oct 17 22:31:57 2001
@@ -193,6 +193,16 @@
   (canna-add-dict "fuzokugo" nil)
   (canna-add-dict "hojomwd" nil)
   (canna-add-dict "hojoswd" nil)
+  (canna-add-dict "keishiki" nil)
+  (canna-add-dict "chimei" nil)
+  (canna-add-dict "kanasample" nil)
+  (canna-add-dict "katakana" nil)
+  (canna-add-dict "number" nil)
+  (canna-add-dict "software" nil)
+  (canna-add-dict "suffix" nil)
+  (canna-add-dict "gcanna" nil)
+  (canna-add-dict "gcannaf" nil)
+  (canna-add-dict "yubin7" nil)
   (canna-add-dict "user" t)
 
   (canna-define-environment "Bushu" 0 t)
