diff -Nura plymouth-0.9.5+git20200921+20778f2/configure.ac plymouth-0.9.5+git20200921+20778f2_new/configure.ac
--- plymouth-0.9.5+git20200921+20778f2/configure.ac	2020-07-18 15:27:12.000000000 +0800
+++ plymouth-0.9.5+git20200921+20778f2_new/configure.ac	2020-10-02 21:41:46.998308963 +0800
@@ -154,7 +154,7 @@
   plymouthclientdir=$bindir
   plymouthdaemondir=$sbindir
   if (test -z "${plymouthruntimedir}"); then
-    plymouthruntimedir=$localstatedir/run/plymouth
+    plymouthruntimedir=/run/plymouth
   fi
 fi
 AC_SUBST(plymouthclientdir)
