diff -Nura plymouth-22.02.122+180.b1d5aa9/themes/spinfinity/meson.build plymouth-22.02.122+180.b1d5aa9_new/themes/spinfinity/meson.build
--- plymouth-22.02.122+180.b1d5aa9/themes/spinfinity/meson.build	2022-11-27 00:45:52.000000000 +0800
+++ plymouth-22.02.122+180.b1d5aa9_new/themes/spinfinity/meson.build	2023-07-12 18:00:50.140486797 +0800
@@ -53,8 +53,3 @@
   'throbber-33.png',
   install_dir: plymouth_theme_path / 'spinfinity',
 )
-
-install_symlink('header-image.png',
-  install_dir: plymouth_theme_path / 'spinfinity',
-  pointing_to: plymouth_logo_file,
-)
