diff -rup a/install.sh b/install.sh
--- a/install.sh	2018-03-01 17:47:59.000000000 +0100
+++ b/install.sh	2018-03-05 17:47:03.045587938 +0100
@@ -1227,6 +1227,13 @@ if [ -n "$SYSCONFFILE" -a -f "$SYSCONFFI
 	make_parent_directory ${DESTDIR}${SYSCONFDIR} 0755
     fi
 
+    if [ $HOST = suse ]; then
+    mkdir -p ${DESTDIR}/${FILLUPDIR}
+    run_install $OWNERSHIP -m 0644 ${SYSCONFFILE} ${DESTDIR}/${FILLUPDIR}/sysconfig.${PRODUCT}
+    else
+    run_install $OWNERSHIP -m 0644 ${SYSCONFFILE} ${DESTDIR}${SYSCONFDIR}/$PRODUCT
+    fi
+
     run_install $OWNERSHIP -m 0644 ${SYSCONFFILE} ${DESTDIR}${SYSCONFDIR}/${PRODUCT}
     echo "$SYSCONFFILE file installed in ${DESTDIR}${SYSCONFDIR}/${PRODUCT}"
 fi
