diff -Nru azure-mgmt-batch-6.0.0.orig/setup.py azure-mgmt-batch-6.0.0/setup.py
--- azure-mgmt-batch-6.0.0.orig/setup.py	2019-04-11 14:22:39.147677310 +0200
+++ azure-mgmt-batch-6.0.0/setup.py	2019-04-11 14:22:55.443812815 +0200
@@ -81,7 +81,4 @@
         'msrestazure>=0.4.32,<2.0.0',
         'azure-common>=1.1',
     ],
-    extras_require={
-        ":python_version<'3.0'": ['azure-mgmt-nspkg'],
-    }
 )
