diff -Nru azure-mgmt-batch-6.0.0.orig/setup.py azure-mgmt-batch-6.0.0/setup.py
--- azure-mgmt-batch-6.0.0.orig/setup.py	2019-01-23 16:43:48.000000000 +0100
+++ azure-mgmt-batch-6.0.0/setup.py	2019-03-25 12:29:55.296255828 +0100
@@ -79,7 +79,7 @@
     install_requires=[
         'msrest>=0.5.0',
         'msrestazure>=0.4.32,<2.0.0',
-        'azure-common~=1.1',
+        'azure-common>=1.1',
     ],
     extras_require={
         ":python_version<'3.0'": ['azure-mgmt-nspkg'],
