Index: qtdeclarative-everywhere-src-5.13.0-beta1/src/qml/jsruntime/jsruntime.pri
===================================================================
--- qtdeclarative-everywhere-src-5.13.0-beta1.orig/src/qml/jsruntime/jsruntime.pri
+++ qtdeclarative-everywhere-src-5.13.0-beta1/src/qml/jsruntime/jsruntime.pri
@@ -174,6 +174,11 @@ qmldevtools_build {
 }
 
 
+linux-g++*:isEqual(QT_ARCH,i386):!no_sse2 {
+    QMAKE_CFLAGS += -msse2 -mfpmath=sse
+    QMAKE_CXXFLAGS += -msse2 -mfpmath=sse
+}
+
 valgrind {
     DEFINES += V4_USE_VALGRIND
 }
Index: qtdeclarative-everywhere-src-5.13.0-beta1/src/qml/jsruntime/qv4global_p.h
===================================================================
--- qtdeclarative-everywhere-src-5.13.0-beta1.orig/src/qml/jsruntime/qv4global_p.h
+++ qtdeclarative-everywhere-src-5.13.0-beta1/src/qml/jsruntime/qv4global_p.h
@@ -115,7 +115,7 @@ inline double trunc(double d) { return d
 
 // Black list some platforms
 #if defined(V4_ENABLE_JIT)
-#if defined(Q_OS_IOS) || defined(Q_OS_TVOS)
+#if defined(Q_OS_IOS) || defined(Q_OS_TVOS) || (defined(Q_PROCESSOR_X86) && !defined(__SSE2__))
 #    undef V4_ENABLE_JIT
 #endif
 #endif
Index: qtdeclarative-everywhere-src-5.13.0-beta1/src/qml/qml/v8/qv8engine.cpp
===================================================================
--- qtdeclarative-everywhere-src-5.13.0-beta1.orig/src/qml/qml/v8/qv8engine.cpp
+++ qtdeclarative-everywhere-src-5.13.0-beta1/src/qml/qml/v8/qv8engine.cpp
@@ -141,7 +141,7 @@ QV8Engine::QV8Engine(QV4::ExecutionEngin
 #ifndef Q_OS_WASM // wasm does not have working simd QTBUG-63924
 #ifdef Q_PROCESSOR_X86_32
     if (!qCpuHasFeature(SSE2)) {
-        qFatal("This program requires an X86 processor that supports SSE2 extension, at least a Pentium 4 or newer");
+        qDebug("This program requires an X86 processor that supports SSE2 extension, at least a Pentium 4 or newer, processors missing the extension are NOT supported to run QML2 code!");
     }
 #endif
 #endif
