diff -rub a/lib/libxfce4sensors-1.0.pc.in b/lib/libxfce4sensors-1.0.pc.in
--- a/lib/libxfce4sensors-1.0.pc.in	2012-04-16 16:29:57.000000000 +0800
+++ b/lib/libxfce4sensors-1.0.pc.in	2021-01-12 16:33:43.998223936 +0800
@@ -9,7 +9,7 @@
 
 Name: libxfce4sensors
 Description: Library for the Xfce 4 Sensors Plugin and Viewer
-Requires: libxfce4util-1.0 libxfce4ui-1
+Requires: libxfce4util-1.0 libxfce4ui-2
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -lxfce4sensors
 Cflags: -I${includedir}/xfce4/
