From d4f6cdfad2572dd6493f404a4a02db40c4a91ea0 Mon Sep 17 00:00:00 2001
From: "Bernhard M. Wiedemann" <bwiedemann@suse.de>
Date: Wed, 2 Dec 2020 20:36:14 +0100
Subject: [PATCH] api/project: Fix handling of "false"

---
 pagure/api/project.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pagure/api/project.py b/pagure/api/project.py
index 654c008a..64f413b7 100644
--- a/pagure/api/project.py
+++ b/pagure/api/project.py
@@ -2828,7 +2828,7 @@ def _check_value(value):
     if str(value).lower() in ["true"]:
         value = True
     elif str(value).lower() in ["false"]:
-        value = True
+        value = False
     elif str(value).isdigit():
         value = int(value)
     return value
-- 
2.28.0

