Index: packETH-1.8.1/src/function_send.c
===================================================================
--- packETH-1.8.1.orig/src/function_send.c
+++ packETH-1.8.1/src/function_send.c
@@ -54,7 +54,7 @@ extern long li_packets_sent;
 extern long li_last_packets_sent;
 extern long li_packets_sent_lastsec;
 extern long sentstream[10];
-char iftext[20];
+extern char iftext[20];
 
 struct params  {
         long long del;
