diff -Nur openvswitch-2.17.0/ovsdb/automake.mk new/ovsdb/automake.mk
--- openvswitch-2.17.0/ovsdb/automake.mk	2022-02-17 23:15:55.240680736 +0100
+++ new/ovsdb/automake.mk	2022-02-27 20:21:59.358719903 +0100
@@ -88,8 +88,9 @@
 MAN_ROOTS += ovsdb/ovsdb-server.1.in
 
 # ovsdb-idlc
-noinst_SCRIPTS += ovsdb/ovsdb-idlc
+bin_SCRIPTS += ovsdb/ovsdb-idlc
 EXTRA_DIST += ovsdb/ovsdb-idlc.in
+man_MANS += ovsdb/ovsdb-idlc.1
 MAN_ROOTS += ovsdb/ovsdb-idlc.1
 CLEANFILES += ovsdb/ovsdb-idlc
 SUFFIXES += .ovsidl .ovsschema
@@ -112,14 +113,18 @@
 # at least for now.
 $(OVSIDL_BUILT): ovsdb/ovsdb-idlc.in python/ovs/dirs.py
 
+# Some internal tools, but installed for e.g. depending projects like OVN
+ovsdbdir = $(pkgdatadir)/ovsdb
+ovsdb_SCRIPTS =
 # ovsdb-doc
+ovsdb_SCRIPTS += ovsdb/ovsdb-doc
 EXTRA_DIST += ovsdb/ovsdb-doc
 OVSDB_DOC = $(run_python) $(srcdir)/ovsdb/ovsdb-doc
 ovsdb/ovsdb-doc: python/ovs/dirs.py
 
 # ovsdb-dot
 EXTRA_DIST += ovsdb/ovsdb-dot.in ovsdb/dot2pic
-noinst_SCRIPTS += ovsdb/ovsdb-dot
+ovsdb_SCRIPTS += ovsdb/ovsdb-dot
 CLEANFILES += ovsdb/ovsdb-dot
 OVSDB_DOT = $(run_python) $(srcdir)/ovsdb/ovsdb-dot.in
 
