---
 Makefile.in |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: subversion-1.14.2/Makefile.in
===================================================================
--- subversion-1.14.2.orig/Makefile.in
+++ subversion-1.14.2/Makefile.in
@@ -124,8 +124,8 @@ EXEEXT = @EXEEXT@
 
 SHELL = @SHELL@
 LIBTOOL = @SVN_LIBTOOL@
-LTFLAGS = --tag=CC --silent
-LTCXXFLAGS = --tag=CXX --silent
+LTFLAGS = --tag=CC
+LTCXXFLAGS = --tag=CXX
 LT_CFLAGS = @LT_CFLAGS@
 LT_LDFLAGS = @LT_LDFLAGS@
 LT_SO_VERSION = @SVN_LT_SOVERSION@
