Index: pdns-4.9.2/pdns/pdns.service.in
===================================================================
--- pdns-4.9.2.orig/pdns/pdns.service.in
+++ pdns-4.9.2/pdns/pdns.service.in
@@ -17,8 +17,8 @@ StartLimitInterval=0
 RuntimeDirectory=pdns
 
 # Sandboxing
-CapabilityBoundingSet=CAP_NET_BIND_SERVICE CAP_CHOWN
-AmbientCapabilities=CAP_NET_BIND_SERVICE CAP_CHOWN
+CapabilityBoundingSet=CAP_NET_BIND_SERVICE CAP_CHOWN CAP_DAC_OVERRIDE
+AmbientCapabilities=CAP_NET_BIND_SERVICE CAP_CHOWN CAP_DAC_OVERRIDE
 LockPersonality=true
 NoNewPrivileges=true
 PrivateDevices=true
