From 48a195a1c09771d6bf4db022ddce56fac04c4deb Mon Sep 17 00:00:00 2001
From: Pekka Vuorela <pekka.vuorela@jollamobile.com>
Date: Wed, 7 Mar 2018 15:50:41 +0200
Subject: [PATCH 2/3] Fix some documentation typos

---
 src/daemon.h      | 2 +-
 src/transaction.h | 8 ++++----
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/daemon.h b/src/daemon.h
index 607d32c..be89c72 100644
--- a/src/daemon.h
+++ b/src/daemon.h
@@ -184,7 +184,7 @@ public:
      * Allows a client to find out if it would be allowed to authorize an action.
      * The action ID, e.g. org.freedesktop.packagekit.system-network-proxy-configure
      * specified in \p actionId
-     * Returm might be either yes, no or interactive \sa Authorize.
+     * Return might be either yes, no or interactive \sa Authorize.
      */
     static QDBusPendingReply<Authorize> canAuthorize(const QString &actionId);
 
diff --git a/src/transaction.h b/src/transaction.h
index 71c9fcc..6f9e312 100644
--- a/src/transaction.h
+++ b/src/transaction.h
@@ -508,9 +508,9 @@ public:
     bool allowCancel() const;
 
     /**
-     * Indicates weither the transaction caller is active or not
+     * Indicates whether the transaction caller is active or not
      *
-     * The caller can be inactive if it has quitted before the transaction finished.
+     * The caller can be inactive if it has quit before the transaction finished.
      *
      * \return true if the caller is active, false else
      */
@@ -583,7 +583,7 @@ public:
     QDateTime timespec() const;
 
     /**
-     * Returns weither the transaction succeded or not
+     * Returns whether the transaction succeded or not
      * \return true if the transaction succeeded, false else
      * \note This function only returns a real value for old transactions returned by getOldTransactions
      */
@@ -597,7 +597,7 @@ public:
     uint duration() const;
 
     /**
-     * Returns some data set by the backend to pass additionnal information
+     * Returns some data set by the backend to pass additional information
      * \return a string set by the backend
      * \note This function only returns a real value for old transactions returned by getOldTransactions
      */
-- 
2.19.0

