Index: doc/doxygen.cfg.in
===================================================================
--- doc/doxygen.cfg.in.orig	2010-01-03 04:54:32.000000000 +0200
+++ doc/doxygen.cfg.in	2010-11-20 07:32:05.000000000 +0200
@@ -1663,7 +1663,7 @@
 # The default value is: NO.
 # This tag requires that the tag GENERATE_LATEX is set to YES.
 
-LATEX_BATCHMODE        = YES
+LATEX_BATCHMODE        = NO
 
 # If the LATEX_HIDE_INDICES tag is set to YES then doxygen will not include the
 # index chapters (such as File Index, Compound Index, etc.) in the output.
