From efb43ef813d90dca48aa01880f097a2161005db1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marco=20Trevisan=20=28Trevi=C3=B1o=29?= <mail@3v1n0.net>
Date: Sun, 23 Oct 2022 17:10:59 +0200
Subject: [PATCH] gio/gdesktopappinfo: Free the wrapped argv array on launch
 failure

We create an array that we never free, ensure this is the case.
The previous commit gives CI a chance to check this with valgrind job.

Found as part of another review:
 - https://gitlab.gnome.org/GNOME/glib/-/merge_requests/2839#note_1524922
---
 gio/gdesktopappinfo.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gio/gdesktopappinfo.c b/gio/gdesktopappinfo.c
index f8cccca025..c29c309478 100644
--- a/gio/gdesktopappinfo.c
+++ b/gio/gdesktopappinfo.c
@@ -2963,6 +2963,7 @@ g_desktop_app_info_launch_uris_with_spawn (GDesktopAppInfo            *info,
 
           g_free (sn_id);
           g_list_free (launched_uris);
+          g_clear_pointer (&wrapped_argv, g_strfreev);
 
           goto out;
         }
-- 
GitLab

