diff --git a/gio/gdbus-2.0/codegen/codegen.py b/gio/gdbus-2.0/codegen/codegen.py
index 2e8ef8e2..772dad77 100644
--- a/gio/gdbus-2.0/codegen/codegen.py
+++ b/gio/gdbus-2.0/codegen/codegen.py
@@ -98,8 +98,7 @@ class HeaderCodeGenerator:
     # ----------------------------------------------------------------------------------------------------
 
     def generate_header_preamble(self):
-        basenames = ", ".join(self.input_files_basenames)
-        self.outfile.write(LICENSE_STR.format(config.VERSION, basenames))
+        self.outfile.write(LICENSE_STR)
         self.outfile.write("\n")
 
         if self.use_pragma:
@@ -1043,8 +1042,7 @@ class InterfaceInfoHeaderCodeGenerator:
     # ----------------------------------------------------------------------------------------------------
 
     def generate_header_preamble(self):
-        basenames = ", ".join(self.input_files_basenames)
-        self.outfile.write(LICENSE_STR.format(config.VERSION, basenames))
+        self.outfile.write(LICENSE_STR)
         self.outfile.write("\n")
 
         if self.use_pragma:
@@ -1115,8 +1113,7 @@ class InterfaceInfoBodyCodeGenerator:
     # ----------------------------------------------------------------------------------------------------
 
     def generate_body_preamble(self):
-        basenames = ", ".join(self.input_files_basenames)
-        self.outfile.write(LICENSE_STR.format(config.VERSION, basenames))
+        self.outfile.write(LICENSE_STR)
 
         if self.symbol_decoration_define is not None:
             self.outfile.write("\n")
@@ -1469,7 +1466,7 @@ class CodeGenerator:
 
     def generate_body_preamble(self):
         basenames = ", ".join(self.input_files_basenames)
-        self.outfile.write(LICENSE_STR.format(config.VERSION, basenames))
+        self.outfile.write(LICENSE_STR)
 
         if self.symbol_decoration_define is not None:
             self.outfile.write("\n")
