Index: glib-2.66.5/gio/gdbusaddress.c
===================================================================
--- glib-2.66.5.orig/gio/gdbusaddress.c
+++ glib-2.66.5/gio/gdbusaddress.c
@@ -1331,7 +1331,7 @@ g_dbus_address_get_for_bus_sync (GBusTyp
 
       if (ret == NULL)
         {
-          ret = g_strdup ("unix:path=/var/run/dbus/system_bus_socket");
+          ret = g_strdup ("unix:path=/run/dbus/system_bus_socket");
         }
       break;
 
