commit 0143b0b5eb14291cbd9e0b8328eeec93c4871ba9
Author: Mike Wallio <michael.wallio@outlook.com>
Date:   Tue Mar 16 16:01:34 2021 -0400

    feat: Add support for GNOME 40

diff --git a/metadata.json b/metadata.json
index 2d13be6..d877ff0 100644
--- a/metadata.json
+++ b/metadata.json
@@ -5,6 +5,7 @@
     "uuid": "pop-shell@system76.com",
     "settings-schema": "org.gnome.shell.extensions.pop-shell",
     "shell-version": [
-        "3.36"
+        "3.36",
+        "40"
     ]
 }
\ No newline at end of file
diff --git a/src/mod.d.ts b/src/mod.d.ts
index 4e2b67e..96fae99 100644
--- a/src/mod.d.ts
+++ b/src/mod.d.ts
@@ -82,8 +82,8 @@ declare namespace Gtk {
 
     export class Widget {
         constructor();
-
-        show_all(): void;
+        show_all?: () => void;
+        show(): void;
     }
 }
 
diff --git a/src/prefs.ts b/src/prefs.ts
index 15d0b4a..c4793de 100644
--- a/src/prefs.ts
+++ b/src/prefs.ts
@@ -70,7 +70,6 @@ function settings_dialog_view(): [AppWidgets, Gtk.Container] {
     let grid = new Gtk.Grid({
         column_spacing: 12,
         row_spacing: 12,
-        border_width: 12
     });
 
     let win_label = new Gtk.Label({
@@ -182,6 +181,10 @@ function logging_combo(grid: any, top_index: number) {
 // @ts-ignore
 function buildPrefsWidget() {
     let dialog = settings_dialog_new();
-    dialog.show_all();
+    if (dialog.show_all) {
+        dialog.show_all()
+    } else {
+        dialog.show();
+    }
     return dialog;
 }
