From 8be10fa999b6631382c4f71826ee78ea9993f459 Mon Sep 17 00:00:00 2001
From: Michael Aaron Murphy <mmstick@pm.me>
Date: Tue, 31 Aug 2021 16:29:01 +0200
Subject: [PATCH] fix: TypeScript 4.4 compiler error

---
 src/settings.ts | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/settings.ts b/src/settings.ts
index 1ae463f2..46ecfbdb 100644
--- a/src/settings.ts
+++ b/src/settings.ts
@@ -20,9 +20,9 @@ interface Settings extends GObject.Object {
 function settings_new_id(schema_id: string): Settings | null {
     try {
         return new Gio.Settings({ schema_id });
-    } catch (err) {
+    } catch (why) {
         if (schema_id !== "org.gnome.shell.extensions.user-theme") {
-            global.log(err)
+            global.log(`failed to get settings for ${schema_id}: ${why}`)
         }
 
         return null
