From 74a4eff529e6b71f6a822841d9112d9b71b44d24 Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Sat, 13 Jan 2018 15:16:39 +0100
Subject: [PATCH] Warning for FlatHub

---
 .../backends/FlatpakBackend/FlatpakSourcesBackend.cpp      | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

Index: discover-%{_plasma6_version}git.20240314T013804~34410c0/libdiscover/backends/FlatpakBackend/FlatpakSourcesBackend.cpp
===================================================================
--- discover-%{_plasma6_version}git.20240314T013804~34410c0.orig/libdiscover/backends/FlatpakBackend/FlatpakSourcesBackend.cpp
+++ discover-%{_plasma6_version}git.20240314T013804~34410c0/libdiscover/backends/FlatpakBackend/FlatpakSourcesBackend.cpp
@@ -18,6 +18,8 @@
 
 #include <QStandardPaths>
 #include <QTemporaryFile>
+#include <QAbstractButton>
+#include <QMessageBox>
 #include <glib.h>
 #include <resources/DiscoverAction.h>
 #include <resources/StoredResultsStream.h>
@@ -99,7 +101,17 @@ FlatpakSourcesBackend::FlatpakSourcesBac
     m_flathubAction->setObjectName(QStringLiteral("flathub"));
     m_flathubAction->setToolTip(i18n("Makes it possible to easily install the applications listed in https://flathub.org"));
     connect(m_flathubAction, &DiscoverAction::triggered, this, [this]() {
-        addSource(QStringLiteral("https://dl.flathub.org/repo/flathub.flatpakrepo"));
+        auto *warnBox = new QMessageBox(QMessageBox::Information, i18n("Adding a third-party source"),
+                            i18n("FlatHub is not part of the openSUSE distribution.\n"
+                                 "It contains packages of unknown quality which might not be tested to work correctly "
+                                 "on openSUSE."),
+                            QMessageBox::Ok | QMessageBox::Abort);
+        connect(warnBox, &QMessageBox::buttonClicked, this, [this, warnBox](QAbstractButton *button) {
+            if (warnBox->standardButton(button) == QMessageBox::Ok)
+                addSource(QStringLiteral("https://dl.flathub.org/repo/flathub.flatpakrepo"));
+        });
+        warnBox->setAttribute(Qt::WA_DeleteOnClose);
+        warnBox->show();
     });
 
     m_noSourcesItem->setEnabled(false);
