--- "a/bundles/org.eclipse.swt/Eclipse SWT WebKit/gtk/library/webkitgtk.h"	2018-09-05 01:01:31.000000000 +0200
+++ "b/bundles/org.eclipse.swt/Eclipse SWT WebKit/gtk/library/webkitgtk.h"	2026-04-13 19:39:33.049418961 +0200
@@ -41,20 +41,15 @@
 	static void *var = NULL; \
 	if (!initialized) { \
 		void* handle = 0; \
-		char *gtk3 = getenv("SWT_GTK3"); \
-		if (gtk3 == NULL || strcmp(gtk3, "1") == 0) { \
-			char *webkit2 = getenv("SWT_WEBKIT2"); \
-			if (webkit2 != NULL && strcmp(webkit2, "0") == 0) { \
-				handle = dlopen("libwebkitgtk-3.0.so.0", LOAD_FLAGS); /* webkitgtk >= 3.x lib */ \
+		char *gtk4 = getenv("SWT_GTK4"); \
+		if (gtk4 != NULL && strcmp(gtk4, "1") == 0) { \
+			handle = dlopen("libwebkit2gtk-5.0.so.0", LOAD_FLAGS); \
 			} else { \
-				handle = dlopen("libwebkit2gtk-4.0.so.37", LOAD_FLAGS); /* webkit2 */ \
+			handle = dlopen("libwebkit2gtk-4.0.so.37", LOAD_FLAGS); /* webkit2/libsoup2 */ \
 				if (!handle) { \
-					handle = dlopen("libwebkitgtk-3.0.so.0", LOAD_FLAGS); /* webkitgtk 3.x  as machine doesn't have webkit2gtk */ \
+				handle = dlopen("libwebkit2gtk-4.1.so.0", LOAD_FLAGS); /* webkit2/libsoup3 */ \
 				} \
 			} \
-		} else { \
-			handle = dlopen("libwebkitgtk-1.0.so.0", LOAD_FLAGS); /* webkitgtk >= 1.4.x lib */ \
-		} \
 		if (handle) { \
 			var = dlsym(handle, #name); \
 		} \
