--- "eclipse-platform-sources-I20180906-0745/eclipse.platform.swt/bundles/org.eclipse.swt/Eclipse SWT PI/gtk/library/build.sh"	2018-08-21 14:03:55.000000000 +0200
+++ "eclipse-platform-sources-I20180906-0745/eclipse.platform.swt/bundles/org.eclipse.swt/Eclipse SWT PI/gtk/library/build.sh"	2019-11-23 18:56:22.469583236 +0100
@@ -195,13 +195,13 @@
 
 
 # For 64-bit CPUs, we have a switch
-if [ ${MODEL} = 'x86_64' -o ${MODEL} = 'ia64' -o ${MODEL} = 's390x' -o ${MODEL} = 'ppc64le' -o ${MODEL} = 'aarch64' ]; then
+if [ ${MODEL} = 'x86_64' -o ${MODEL} = 'ia64' -o ${MODEL} = 's390x' -o ${MODEL} = 'ppc64le' -o ${MODEL} = 'ppc64' -o ${MODEL} = 'aarch64' ]; then
 	SWT_PTR_CFLAGS=-DJNI64
 	if [ -d /lib64 ]; then
 		XLIB64=-L/usr/X11R6/lib64
 		export XLIB64
 	fi
-	if [ ${MODEL} = 'ppc64le' ]; then
+	if [ ${MODEL} = 'ppc64le'  -o ${MODEL} = 'ppc64']; then
 		SWT_PTR_CFLAGS="${SWT_PTR_CFLAGS} -m64"
 		XLIB64="${XLIB64} -L/usr/lib64"
 		SWT_LFLAGS=-m64
