--- libvisio-0.1.8/src/lib/VSDStyles.h	2024-10-22 17:15:20.000000000 +0200
+++ libvisio-0.1.8/src/lib/VSDStyles.h	2025-09-11 03:10:32.788117037 +0200
@@ -179,9 +179,9 @@
     {
       // Quick Style Colour 100 is special. It is the default,
       // and it is not saved explicitely in the VSDX file.
-      ASSIGN_OPTIONAL(theme->getThemeColour(style.qsFillColour.value_or(100)), fgColour);
-      ASSIGN_OPTIONAL(theme->getThemeColour(style.qsFillColour.value_or(100)), bgColour);
-      ASSIGN_OPTIONAL(theme->getThemeColour(style.qsShadowColour.value_or(100)), shadowFgColour);
+      ASSIGN_OPTIONAL(theme->getThemeColour(boost::get_optional_value_or(style.qsFillColour, 100)), fgColour);
+      ASSIGN_OPTIONAL(theme->getThemeColour(boost::get_optional_value_or(style.qsFillColour, 100)), bgColour);
+      ASSIGN_OPTIONAL(theme->getThemeColour(boost::get_optional_value_or(style.qsShadowColour, 100)), shadowFgColour);
       if (!!style.qsFillMatrix && style.qsFillMatrix.get() >= 0)
         ASSIGN_OPTIONAL(theme->getFillStyleColour(style.qsFillMatrix.get()), fgColour);
     }
