--- maven-reporting-impl-3.1.0/pom.xml	2022-03-10 17:08:29.894389771 +0100
+++ maven-reporting-impl-3.1.0/pom.xml	2022-03-10 17:09:12.654649105 +0100
@@ -140,12 +140,6 @@
       <version>4.13.1</version>
       <scope>test</scope>
     </dependency>
-    <dependency>
-      <groupId>junit-addons</groupId>
-      <artifactId>junit-addons</artifactId>
-      <version>1.4</version>
-      <scope>test</scope>
-    </dependency>
   </dependencies>
 
   <build>
--- maven-reporting-impl-3.1.0/src/main/java/org/apache/maven/reporting/AbstractMavenReportRenderer.java	2022-03-10 17:08:29.894389771 +0100
+++ maven-reporting-impl-3.1.0/src/main/java/org/apache/maven/reporting/AbstractMavenReportRenderer.java	2022-03-10 17:09:12.654649105 +0100
@@ -599,7 +599,7 @@
      * @param text a text with or without the pattern <code>{text, url}</code>
      * @return a map of text/href
      */
-    private static List<String> applyPattern( String text )
+    static List<String> applyPattern( String text )
     {
         if ( StringUtils.isEmpty( text ) )
         {
--- maven-reporting-impl-3.1.0/src/test/java/org/apache/maven/reporting/AbstractMavenReportRendererTest.java	2022-03-10 17:08:29.898389795 +0100
+++ maven-reporting-impl-3.1.0/src/test/java/org/apache/maven/reporting/AbstractMavenReportRendererTest.java	2022-03-10 17:09:12.654649105 +0100
@@ -24,7 +24,8 @@
 
 import junit.framework.Assert;
 import junit.framework.TestCase;
-import junitx.util.PrivateAccessor;
+
+import static org.apache.maven.reporting.AbstractMavenReportRenderer.applyPattern;
 
 /**
  * Test case for some public method in AbstractMavenReportRenderer.
@@ -32,13 +33,6 @@
 public class AbstractMavenReportRendererTest
     extends TestCase
 {
-    private static List<String> applyPattern( String pattern )
-        throws Throwable
-    {
-        return (List<String>) PrivateAccessor.invoke( AbstractMavenReportRenderer.class, "applyPattern",
-                                              new Class[] { String.class }, new Object[] { pattern } );
-    }
-
     private static void checkPattern( String pattern, String[] expectedResult ) throws Throwable
     {
         List<String> result = applyPattern( pattern );
