diff -Naur a/src/llua.cc b/src/llua.cc
--- a/src/llua.cc	2021-03-02 09:39:20.000000000 -0600
+++ b/src/llua.cc	2021-03-04 10:43:51.393815765 -0600
@@ -157,8 +157,7 @@
   lua_setfield(lua_L, -3, "cpath");
   lua_pop(lua_L, 2);
 
-  lua_pushstring(lua_L, PACKAGE_NAME " " VERSION " compiled " BUILD_DATE
-                                     " for " BUILD_ARCH);
+  lua_pushstring(lua_L, PACKAGE_NAME " " VERSION " compiled for " BUILD_ARCH);
   lua_setglobal(lua_L, "conky_build_info");
 
   lua_pushstring(lua_L, VERSION);
diff -Naur a/src/main.cc b/src/main.cc
--- a/src/main.cc	2021-03-02 09:39:20.000000000 -0600
+++ b/src/main.cc	2021-03-04 10:44:34.869260595 -0600
@@ -56,8 +56,7 @@
 #endif /* BUILD_BUILTIN_CONFIG */
 
 static void print_version() {
-  std::cout << _(PACKAGE_NAME " " VERSION " compiled " BUILD_DATE
-                              " for " BUILD_ARCH
+  std::cout << _(PACKAGE_NAME " " VERSION " compiled for " BUILD_ARCH
                               "\n"
                               "\nCompiled in features:\n\n"
                               "System config file: " SYSTEM_CONFIG_FILE
