Index: ndindex-1.9.2/conftest.py
===================================================================
--- ndindex-1.9.2.orig/conftest.py
+++ ndindex-1.9.2/conftest.py
@@ -6,7 +6,7 @@ try:
 except ImportError:
     from distutils.version import LooseVersion
 
-from hypothesis import settings
+from hypothesis import settings, HealthCheck
 
 # Make sure a new enough version of NumPy is installed for the tests
 import numpy
@@ -59,4 +59,10 @@ def pytest_configure(config):
 
 
 settings.register_profile('ndindex_hypothesis_profile', deadline=800)
-settings.load_profile('ndindex_hypothesis_profile')
+#settings.load_profile('ndindex_hypothesis_profile')
+
+settings.register_profile(
+     'obs',
+     deadline=5000,
+     suppress_health_check=[HealthCheck.too_slow, HealthCheck.filter_too_much]
+)
Index: ndindex-1.9.2/pytest.ini
===================================================================
--- ndindex-1.9.2.orig/pytest.ini
+++ ndindex-1.9.2/pytest.ini
@@ -1,3 +1,4 @@
 [pytest]
-addopts = --cov=ndindex/ --cov-report=term-missing --flakes
 filterwarnings = error
+markers =
+  no_cover : we don't use coverage but there are some markers for it
