diff --git a/src/libQtAV.pro b/src/libQtAV.pro
index 27c1570..6056a9a 100644
--- a/src/libQtAV.pro
+++ b/src/libQtAV.pro
@@ -304,7 +304,7 @@ config_vaapi* {
     HEADERS += vaapi/SurfaceInteropVAAPI.h
     SOURCES += vaapi/SurfaceInteropVAAPI.cpp
   #}
-    LIBS *= -lva -lX11 #dynamic load va-glx va-x11 using dllapi. -lX11: used by tfp
+    LIBS *= -lva -lX11 -ldl #dynamic load va-glx va-x11 using dllapi. -lX11: used by tfp
 }
 config_libcedarv {
     DEFINES *= QTAV_HAVE_CEDARV=1
