diff --git connexions/Makefile.am connexions/Makefile.am
index 9b30310..a944253 100644
--- connexions/Makefile.am
+++ connexions/Makefile.am
@@ -26,6 +26,7 @@ connexiondatadir   = $(pkgdatadir)
 connexion_ldflags  = $(AM_LDFLAGS)
 connexion_ldflags += -export-dynamic
 connexion_ldflags += -export-symbols-regex
+connexion_ldflags += -module -avoid-version
 
 connexionlib_LTLIBRARIES  = libcnx-usb.la
 connexionlib_LTLIBRARIES += libcnx-hexdump.la
diff --git filters/Makefile.am filters/Makefile.am
index 203628d..2480623 100644
--- filters/Makefile.am
+++ filters/Makefile.am
@@ -28,6 +28,7 @@ filterlibdir    = $(pkglibdir)
 filterdatadir   = $(pkgdatadir)
 filter_ldflags  = $(AM_LDFLAGS)
 filter_ldflags += -export-dynamic
+filter_ldflags += -module -avoid-version
 
 dist_pkglibexec_SCRIPTS =
 pkglibexec_PROGRAMS     =
diff --git gtkmm/Makefile.am gtkmm/Makefile.am
index 5acd292..86dc036 100644
--- gtkmm/Makefile.am
+++ gtkmm/Makefile.am
@@ -48,6 +48,7 @@ libutsushi_gtkmm_la_SOURCES += preview.cpp
 libutsushi_gtkmm_la_SOURCES += preview.hpp
 libutsushi_gtkmm_la_SOURCES += pump.cpp
 libutsushi_gtkmm_la_SOURCES += pump.hpp
+libutsushi_gtkmm_la_LDFLAGS  = -module -avoid-version
 
 AM_CPPFLAGS += -DGLIB_DISABLE_DEPRECATION_WARNINGS
 AM_CXXFLAGS += $(LIBGTKMM_CFLAGS)
diff --git lib/Makefile.am lib/Makefile.am
index 7830f30..4415873 100644
--- lib/Makefile.am
+++ lib/Makefile.am
@@ -77,6 +77,7 @@ libutsushi_la_SOURCES += scanner.cpp
 libutsushi_la_SOURCES += $(streams)
 libutsushi_la_SOURCES += $(settings)
 libutsushi_la_SOURCES += $(outputs)
+libutsushi_la_LDFLAGS  = -module -avoid-version
 
 if have_libtiff
 libutsushi_la_LIBADD  += $(LIBTIFF_LIBS)
