--- chat/Makefile.linux
+++ chat/Makefile.linux
@@ -11,7 +11,7 @@
 CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
 
 COPTS=	-O2 -g -pipe
-CFLAGS=	$(COPTS) $(CDEFS)
+CFLAGS=	$(MY_CFLAGS) $(COPTS) $(CDEFS)
 
 INSTALL= install
 
@@ -25,8 +25,8 @@
 
 install: chat
 	mkdir -p $(BINDIR) $(MANDIR)
-	$(INSTALL) -s -c chat $(BINDIR)
-	$(INSTALL) -c -m 644 chat.8 $(MANDIR)
+	$(INSTALL) chat $(BINDIR)
+	$(INSTALL) -m 644 chat.8 $(MANDIR)
 
 clean:
 	rm -f chat.o chat *~
--- pppd/Makefile.linux
+++ pppd/Makefile.linux
@@ -32,8 +32,9 @@
 
 # CC = gcc
 #
-COPTS = -O2 -pipe -Wall -g
-LIBS =
+CC = gcc
+COPTS = $(MY_CFLAGS) -O2 -pipe -Wall -fno-strict-aliasing
+LIBS = -lutil
 
 # Uncomment the next 2 lines to include support for Microsoft's
 # MS-CHAP authentication protocol.  Also, edit plugins/radius/Makefile.linux.
@@ -208,10 +209,8 @@
 install: pppd
 	mkdir -p $(BINDIR) $(MANDIR)
 	$(EXTRAINSTALL)
-	$(INSTALL) -s -c -m 555 pppd $(BINDIR)/pppd
-	if chgrp pppusers $(BINDIR)/pppd 2>/dev/null; then \
-	  chmod o-rx,u+s $(BINDIR)/pppd; fi
-	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)
+	$(INSTALL) -m 755 pppd $(BINDIR)/pppd
+	$(INSTALL) -m 644 pppd.8 $(MANDIR)
 
 pppd: $(PPPDOBJS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o pppd $(PPPDOBJS) $(LIBS)
--- pppd/plugins/Makefile.linux
+++ pppd/plugins/Makefile.linux
@@ -1,5 +1,5 @@
-#CC	= gcc
-COPTS	= -O2 -g
+CC	= gcc
+COPTS	= $(MY_CFLAGS) -O2 -g -Wall
 CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
 LDFLAGS	= -shared
 INSTALL	= install
--- pppd/plugins/pppoatm/Makefile.linux
+++ pppd/plugins/pppoatm/Makefile.linux
@@ -1,5 +1,5 @@
 #CC	= gcc
-COPTS	= -O2 -g
+COPTS	= $(MY_CFLAGS) -O2 -g
 CFLAGS	= $(COPTS) -I../.. -I../../../include -fPIC
 LDFLAGS	= -shared
 INSTALL	= install
--- pppd/plugins/pppol2tp/Makefile.linux
+++ pppd/plugins/pppol2tp/Makefile.linux
@@ -1,5 +1,5 @@
-#CC	= gcc
-COPTS	= -O2 -g
+CC	= gcc
+COPTS	= $(MY_CFLAGS) -O2 -pipe -Wall
 CFLAGS	= $(COPTS) -I. -I../.. -I../../../include -fPIC
 LDFLAGS	= -shared
 INSTALL	= install
@@ -20,7 +20,7 @@
 
 install: all
 	$(INSTALL) -d -m 755 $(LIBDIR)
-	$(INSTALL) -c -m 755 $(PLUGINS) $(LIBDIR)
+	$(INSTALL) -m 755 $(PLUGINS) $(LIBDIR)
 
 clean:
 	rm -f *.o *.so
--- pppd/plugins/radius/Makefile.linux
+++ pppd/plugins/radius/Makefile.linux
@@ -12,7 +12,8 @@
 INSTALL	= install
 
 PLUGIN=radius.so radattr.so radrealms.so
-CFLAGS=-I. -I../.. -I../../../include -O2 -fPIC -DRC_LOG_FACILITY=LOG_DAEMON
+COPTS  = $(MY_CFLAGS) -O2 -fPIC
+CFLAGS= $(COPTS) -I. -I../.. -I../../../include -DRC_LOG_FACILITY=LOG_DAEMON -fno-strict-aliasing
 
 # Uncomment the next line to include support for Microsoft's
 # MS-CHAP authentication protocol.
--- pppd/plugins/rp-pppoe/Makefile.linux
+++ pppd/plugins/rp-pppoe/Makefile.linux
@@ -25,7 +25,7 @@
 # Version is set ONLY IN THE MAKEFILE!  Don't delete this!
 RP_VERSION=3.8p
 
-COPTS=-O2 -g
+COPTS= $(MY_CFLAGS) -O2 -pipe -Wall 
 CFLAGS=$(COPTS) -I../../../include '-DRP_VERSION="$(RP_VERSION)"'
 all: rp-pppoe.so pppoe-discovery
 
@@ -43,7 +43,7 @@
 
 install: all
 	$(INSTALL) -d -m 755 $(LIBDIR)
-	$(INSTALL) -s -c -m 4550 rp-pppoe.so $(LIBDIR)
+	$(INSTALL) rp-pppoe.so $(LIBDIR)
 	$(INSTALL) -d -m 755 $(BINDIR)
 	$(INSTALL) -s -c -m 555 pppoe-discovery $(BINDIR)
 
--- pppdump/Makefile.linux
+++ pppdump/Makefile.linux
@@ -2,7 +2,7 @@
 BINDIR = $(DESTDIR)/sbin
 MANDIR = $(DESTDIR)/share/man/man8
 
-CFLAGS= -O -I../include/net
+CFLAGS = $(MY_CFLAGS) -pipe -Wall -O -I../include/net
 OBJS = pppdump.o bsd-comp.o deflate.o zlib.o
 
 INSTALL= install
@@ -17,5 +17,5 @@
 
 install:
 	mkdir -p $(BINDIR) $(MANDIR)
-	$(INSTALL) -s -c pppdump $(BINDIR)
-	$(INSTALL) -c -m 444 pppdump.8 $(MANDIR)
+	$(INSTALL) pppdump $(BINDIR)
+	$(INSTALL) -m 444 pppdump.8 $(MANDIR)
--- pppstats/Makefile.linux
+++ pppstats/Makefile.linux
@@ -9,8 +9,8 @@
 PPPSTATSRCS = pppstats.c
 PPPSTATOBJS = pppstats.o
 
-#CC = gcc
-COPTS = -O
+CC = gcc
+COPTS = $(MY_CFLAGS) -O2 -pipe -Wall
 COMPILE_FLAGS = -I../include
 LIBS =
 
@@ -22,8 +22,8 @@
 
 install: pppstats
 	-mkdir -p $(MANDIR)
-	$(INSTALL) -s -c pppstats $(BINDIR)
-	$(INSTALL) -c -m 444 pppstats.8 $(MANDIR)
+	$(INSTALL) pppstats $(BINDIR)
+	$(INSTALL) -m 444 pppstats.8 $(MANDIR)
 
 pppstats: $(PPPSTATSRCS)
 	$(CC) $(CFLAGS) -o pppstats pppstats.c $(LIBS)
