Index: pppd/sys-linux.c
===================================================================
--- pppd/sys-linux.c.orig
+++ pppd/sys-linux.c
@@ -875,6 +875,24 @@ struct speed {
 #ifdef B4000000
     { 4000000, B4000000 },
 #endif
+#ifdef B1500000
+    { 1500000, B1500000 },
+#endif
+#ifdef B2000000
+    { 2000000, B2000000 },
+#endif
+#ifdef B2500000
+    { 2500000, B2500000 },
+#endif
+#ifdef B3000000
+    { 3000000, B3000000 },
+#endif
+#ifdef B3500000
+    { 3500000, B3500000 },
+#endif
+#ifdef B4000000
+    { 4000000, B4000000 },
+#endif
     { 0, 0 }
 };
 
