Index: hugin-2021.0.0/CMakeModules/FindFLANN.cmake
===================================================================
--- hugin-2021.0.0.orig/CMakeModules/FindFLANN.cmake
+++ hugin-2021.0.0/CMakeModules/FindFLANN.cmake
@@ -11,6 +11,7 @@
 if(UNIX)
   find_package(PkgConfig QUIET)
   pkg_check_modules(FLANN flann)
+  set(FLANN_LIBRARIES "lz4;${FLANN_LIBRARIES}")
 else()
   find_path(FLANN_INCLUDE_DIRS flann/flann.hpp)
 
