Index: hugin-2020.0.0/CMakeLists.txt
===================================================================
--- hugin-2020.0.0.orig/CMakeLists.txt
+++ hugin-2020.0.0/CMakeLists.txt
@@ -256,7 +256,7 @@ IF(NOT HUGIN_SHARED OR NOT WIN32)
   ADD_DEFINITIONS(-DGLEW_STATIC)
 ENDIF()
 
-IF(NOT APPLE AND NOT WIN32 AND NOT HAVE_EGL)
+IF(NOT APPLE AND NOT WIN32)
 MESSAGE(STATUS "Searching X11 header")
 FIND_PACKAGE(X11 REQUIRED)
 ENDIF()
