Index: seafile-client-9.0.1/src/ui/event-details-tree.cpp
===================================================================
--- seafile-client-9.0.1.orig/src/ui/event-details-tree.cpp
+++ seafile-client-9.0.1/src/ui/event-details-tree.cpp
@@ -127,6 +127,8 @@ QString EventDetailsFileItem::etype_desc
       case DIR_DELETED:
         return QObject::tr("Deleted");
     };
+
+    return NULL;
 }
 
 QIcon EventDetailsFileItem::etype_icon() const
@@ -145,6 +147,8 @@ QIcon EventDetailsFileItem::etype_icon()
       case DIR_DELETED:
         return awesome->icon(icon_minus, QColor("#BD2C00"));
     };
+
+    return awesome->icon(icon_minus, QColor("#000000"));
 }
 
 const char* EventDetailsFileItem::etype_color() const
@@ -163,6 +167,8 @@ const char* EventDetailsFileItem::etype_
       case DIR_DELETED:
         return "#BD2C00";
     };
+
+    return NULL;
 }
 
 bool EventDetailsFileItem::isFileOpenable() const
