Index: keepalived-2.0.7/keepalived/etc/init.d/keepalived.suse.init.in
===================================================================
--- keepalived-2.0.7.orig/keepalived/etc/init.d/keepalived.suse.init.in
+++ keepalived-2.0.7/keepalived/etc/init.d/keepalived.suse.init.in
@@ -1,23 +1,24 @@
 #! /bin/sh
 
 ### BEGIN INIT INFO
-# Provides:	 keepalived
+# Provides:	      keepalived
 # Required-Start: $remote_fs $syslog
-# Required-Stop : $remote_fs $syslog
-# Default-Start : 3 5
-# Default-Stop  : 0 1 2 6
-# Description   : Start keepalived to allow XY and provide YZ
-# continued on second line by '#<TAB>'
+# Required-Stop:  $remote_fs $syslog
+# Default-Start:  3 5
+# Default-Stop:   0 1 2 6
+# Short-Description:  Start keepalived to manage IPVS & LVS
+# Description:  Start keepalived to manage IPVS & LVS
 ### END INIT INFO
 
 
-DAEMON="Keepalived daemon"
-DAEMON_BIN="@sbindir@/keepalived"
-DAEMON_CONF="@sysconfdir@/keepalived/keepalived.conf"
-DAEMON_PIDFILE="@localstatedir@/run/keepalived.pid"
-DAEMON_OPT="-d"
+DAEMON='Keepalived daemon'
+DAEMON_BIN='@sbindir@/keepalived'
+DAEMON_CONF='@sysconfdir@/keepalived/keepalived.conf'
+DAEMON_PIDFILE='@localstatedir@/run/keepalived.pid'
+DAEMON_SYSCONF='/etc/sysconfig/keepalived'
+DAEMON_OPT='-d'
 #DAEMON_USER="root"
-SUPPORTS_HUP="yes"
+SUPPORTS_HUP='yes'
 #       PidFile @localstatedir@/run/keepalived.pid
 #       DatabaseOwner root
 
@@ -25,6 +26,18 @@ pid_par=${DAEMON_PIDFILE:+"-p $DAEMON_PI
 usr_par=${DAEMON_USER:+"-u $DAEMON_USER"}
 
 test -x $DAEMON_BIN || exit 5
+test -x $DAEMON_BIN || { echo "$DAEMON_BIN not installed or executable";
+    if [ "$1" = "stop" ]; then exit 0;
+    else exit 5; fi; }
+
+test -r $DAEMON_CONF || { echo "$DAEMON_CONF not existing or readable";
+    if [ "$1" = "stop" ]; then exit 0;
+    else exit 6; fi; }
+
+# read sysconfig variables that might overwrite the daemon options
+if [ -r "$DAEMON_SYSCONF" ]; then
+   . "$DAEMON_SYSCONF"
+fi
 
 # Shell functions sourced from /etc/rc.status:
 #      rc_check         check and set local and overall rc status
@@ -72,8 +85,8 @@ case "$1" in
 		    0) echo -n "- Warning: daemon already running. " ;;
 		    1) echo -n "- Warning: ${DAEMON_PIDFILE} exists. " ;;
 		esac
-#		echo "startproc $usr_par $pid_par ${DAEMON_BIN} ${DAEMON_OPT}"
-		startproc $usr_par $pid_par ${DAEMON_BIN} ${DAEMON_OPT}
+#		echo "startproc $usr_par $pid_par ${DAEMON_BIN} ${DAEMON_OPT} ${KEEPALIVED_OPTIONS}"
+		startproc $usr_par $pid_par ${DAEMON_BIN} ${DAEMON_OPT} ${KEEPALIVED_OPTIONS}
 		rc_status -v
 		;;
 	stop)
