--- minicom-2.2.orig/src/updown.c	2005-08-14 22:39:30.000000000 +0200
+++ minicom-2.2/src/updown.c	2006-10-18 16:48:34.000000000 +0200
@@ -355,6 +355,22 @@
   mcd("");
   timer_update();
 
+  /* return code == 1 if exeve failed, see #111060 et al */
+  if (win && status == 0x0001) {
+#if VC_MUSIC
+	if (P_SOUND[0] == 'Y') {
+		wprintf(win, _("\n Failure executing protocol. Press any key to continue..."));
+		music();
+	} else
+		sleep(1);
+#else
+	/* MARK updated 02/17/94 - If there was no VC_MUSIC capability, */
+	/* then at least make some beeps! */
+	if (P_SOUND[0] == 'Y') wprintf(win, "\007\007\007");
+	sleep(1);
+#endif
+  } else
+
   /* If we got interrupted, status != 0 */
   if (win && (status & 0xFF00) == 0) {
 #if VC_MUSIC
