diff --git a/CMakeLists.txt b/CMakeLists.txt
index acfcd3a..d696e22 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,7 +38,7 @@ set(tinyobjloader-examples-objsticher
 #Install destinations
 include(GNUInstallDirs)
 
-set(TINYOBJLOADER_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake)
+set(TINYOBJLOADER_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake)
 set(TINYOBJLOADER_DOC_DIR ${CMAKE_INSTALL_DOCDIR})
 set(TINYOBJLOADER_INCLUDE_DIR ${CMAKE_INSTALL_INCLUDEDIR})
 set(TINYOBJLOADER_LIBRARY_DIR ${CMAKE_INSTALL_LIBDIR})
