Index: SHERPA-MC-2.2.12/m4/ax_python_devel.m4
===================================================================
--- SHERPA-MC-2.2.12.orig/m4/ax_python_devel.m4
+++ SHERPA-MC-2.2.12/m4/ax_python_devel.m4
@@ -136,7 +136,7 @@ variable to configure. See ``configure -
 	# Check if you have distutils, else fail
 	#
 	AC_MSG_CHECKING([for the distutils Python package])
-	ac_distutils_result=`$PYTHON -c "import distutils" 2>&1`
+	ac_distutils_result=`$PYTHON -c "import warnings; warnings.simplefilter('ignore'); import distutils" 2>&1`
 	if test -z "$ac_distutils_result"; then
 		AC_MSG_RESULT([yes])
 	else
