--- libetonyek-0.1.10.orig/src/lib/IWAParser.cpp	2021-09-08 08:13:58.425324040 +0200
+++ libetonyek-0.1.10/src/lib/IWAParser.cpp	2021-09-08 08:34:25.497038861 +0200
@@ -2355,10 +2355,10 @@
     IWORKPropertyMap props;
     // normally yellow, but blue may be better in LO
     props.put<property::FontColor>(IWORKColor(0,0,1,1));
-    spans[0]=std::make_shared<IWORKStyle>(props, boost::none, nullptr);
+    spans[0]=std::make_shared<IWORKStyle>(props, boost::none, IWORKStylePtr_t());
     // reset color to default, if not, comment will be blue colored
     props.put<property::FontColor>(IWORKColor(0,0,0,1));
-    spans[unsigned(len)]=std::make_shared<IWORKStyle>(props, boost::none, nullptr);
+    spans[unsigned(len)]=std::make_shared<IWORKStyle>(props, boost::none, IWORKStylePtr_t());
     text.setSpans(spans);
     text.parse(*m_currentText);
   }
@@ -3403,7 +3403,7 @@
   }
   else
   {
-    formula.reset(new IWORKFormula(0));
+    formula.reset(new IWORKFormula(boost::make_optional(0u)));
     formula->parse(stack[0]);
   }
   return ok;
--- libetonyek-0.1.10.orig/src/lib/libetonyek_utils.cpp	2021-09-08 08:13:58.421324015 +0200
+++ libetonyek-0.1.10/src/lib/libetonyek_utils.cpp	2021-09-08 08:37:10.586089369 +0200
@@ -12,6 +12,7 @@
 #include <cmath>
 #include <cstdarg>
 #include <cstdio>
+#include <cstring>
 #include <limits>
 #include <stdexcept>
 
