diff -auNr med-4.1.0/include/2.3.6/med_utils.h.in med-4.1.0-new/include/2.3.6/med_utils.h.in
--- med-4.1.0/include/2.3.6/med_utils.h.in	2020-03-11 17:36:33.000000000 +0800
+++ med-4.1.0-new/include/2.3.6/med_utils.h.in	2021-01-04 14:30:43.727414400 +0800
@@ -235,7 +235,11 @@
 					fprintf(stderr,"%s = %llu\n",#entier,entier) ;\
 					fflush(stderr) ;\
 				}
-# define ISCRUTE_id(entier) ISCRUTE_long(entier)
+# define ISCRUTE_id(entier)    {\
+                    ICI ;\
+                    fprintf(stderr,"%s = %"PRId64"\n",#entier,entier) ;\
+                    fflush(stderr) ;\
+                }
 # define RSCRUTE(reel)		{\
 					ICI ;\
 					fprintf(stderr,"%s = %f\n",#reel,reel) ;\
diff -auNr med-4.1.0/include/med_utils.h.in med-4.1.0-new/include/med_utils.h.in
--- med-4.1.0/include/med_utils.h.in	2020-03-11 17:36:32.000000000 +0800
+++ med-4.1.0-new/include/med_utils.h.in	2021-01-04 14:29:40.759652392 +0800
@@ -258,7 +258,11 @@
 					fprintf(stderr,"%s = %llu\n",#entier,entier) ;\
 					fflush(stderr) ;\
 				}
-# define ISCRUTE_id(entier) ISCRUTE_long(entier)
+# define ISCRUTE_id(entier)    {\
+                    ICI ;\
+                    fprintf(stderr,"%s = %"PRId64"\n",#entier,entier) ;\
+                    fflush(stderr) ;\
+                }
 # define RSCRUTE(reel)		{\
 					ICI ;\
 					fprintf(stderr,"%s = %f\n",#reel,reel) ;\
