Index: GraphicsMagick-1.3.46/PerlMagick/Makefile.PL.in
===================================================================
--- GraphicsMagick-1.3.46.orig/PerlMagick/Makefile.PL.in
+++ GraphicsMagick-1.3.46/PerlMagick/Makefile.PL.in
@@ -78,7 +78,7 @@ WriteMakefile
    'INSTALLBIN' => $magick_BIN_DIR,
 
    # Library specification
-   'LIBS'       => ["-L$magick_LIB_DIR $magick_LDFLAGS $magick_API_LIBS"],
+   'LIBS'       => ["-L$magick_LIB_DIR  -L../magick/.libs $magick_LDFLAGS $magick_API_LIBS"],
 
    # Perl binary name (if a Perl binary is built)
    'MAP_TARGET' => 'PerlMagick',
