Index: GraphicsMagick-1.3.40/coders/cineon.c
===================================================================
--- GraphicsMagick-1.3.40.orig/coders/cineon.c
+++ GraphicsMagick-1.3.40/coders/cineon.c
@@ -1138,7 +1138,7 @@ static void GenerateCineonTimeStamp(char
   (void) memset(date_str,0,date_str_length);
   (void) strlcpy(date_str,timestamp,11);
   (void) memset(time_str,0,time_str_length);
-  (void) strlcpy(time_str,timestamp+11,15);
+  (void) strlcpy(time_str,timestamp+11,time_str_length);
 }
 
 
Index: GraphicsMagick-1.3.40/magick/attribute.c
===================================================================
--- GraphicsMagick-1.3.40.orig/magick/attribute.c
+++ GraphicsMagick-1.3.40/magick/attribute.c
@@ -3301,7 +3301,7 @@ SetImageAttribute(Image *image,const cha
                     { /* nada */};
               MagickReallocMemory(char *,p->value,realloc_l);
               if (p->value != (char *) NULL)
-                (void) strlcat(p->value+p->length,attribute->value,min_l);
+                (void) memcpy(p->value+p->length,attribute->value,attribute->length+1);
               p->length += attribute->length;
               DestroyImageAttribute(attribute);
             }
