From fae0dd2f4bd0e74b8a928e19407fd4167f4b2295 Mon Sep 17 00:00:00 2001
From: Frank Hoffmann <15r10nk-git@polarbit.de>
Date: Wed, 18 Jun 2025 22:20:25 +0200
Subject: [PATCH] fix: prevent pytest from thinking that Tester() could be a
 test function

---
 pyproject.toml | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index d5cf1ab..226e061 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -14,4 +14,7 @@ warn_redundant_casts=true
 
 [[tool.mypy.overrides]]
 module = "astroid"
-ignore_missing_imports = true
\ No newline at end of file
+ignore_missing_imports = true
+
+[tool.pytest.ini_options]
+python_functions = "test_"
