Index: mariadb-connector-odbc-3.1.1-ga-src/CMakeLists.txt
===================================================================
--- mariadb-connector-odbc-3.1.1-ga-src.orig/CMakeLists.txt
+++ mariadb-connector-odbc-3.1.1-ga-src/CMakeLists.txt
@@ -324,7 +324,7 @@ ELSE()
   IF(NOT DEFINED INSTALL_LIB_SUFFIX)
     SET(INSTALL_LIB_SUFFIX "lib")
     IF("${CMAKE_SIZEOF_VOID_P}" EQUAL "8" AND EXISTS "/usr/lib64/")
-      SET(INSTALL_LIB_SUFFIX "lib")
+      SET(INSTALL_LIB_SUFFIX "lib64")
     ENDIF()
   ENDIF()
 
