Index: mariadb-connector-odbc-3.1.1-ga-src/CMakeLists.txt
===================================================================
--- mariadb-connector-odbc-3.1.1-ga-src.orig/CMakeLists.txt
+++ mariadb-connector-odbc-3.1.1-ga-src/CMakeLists.txt
@@ -159,7 +159,7 @@ IF(WIN32)
   ENDIF()
 ELSE()
   SEARCH_LIBRARY(LIB_MATH floor m)
-  SET(PLATFORM_DEPENDENCIES ${LIB_MATH})
+  SET(PLATFORM_DEPENDENCIES ${LIB_MATH} pthread)
   SET (MARIADB_ODBC_SOURCES ${MARIADB_ODBC_SOURCES} ma_platform_posix.c)
 ENDIF()
 
