Index: bzrtp-5.3.7/CMakeLists.txt
===================================================================
--- bzrtp-5.3.7.orig/CMakeLists.txt
+++ bzrtp-5.3.7/CMakeLists.txt
@@ -22,7 +22,7 @@
 
 cmake_minimum_required(VERSION 3.22)
 
-project(BZRTP VERSION 5.3.0 LANGUAGES C CXX)
+project(BZRTP VERSION 5.3.105 LANGUAGES C CXX)
 
 option(ENABLE_ZIDCACHE "Turn on compilation of ZID cache, request sqlite" YES)
 option(ENABLE_STRICT "Build with strict compile options." YES)
