Index: grantlee-5.3.1/templates/lib/CMakeLists.txt
===================================================================
--- grantlee-5.3.1.orig/templates/lib/CMakeLists.txt
+++ grantlee-5.3.1/templates/lib/CMakeLists.txt
@@ -137,7 +137,7 @@ install(TARGETS Grantlee_Templates EXPOR
   RUNTIME DESTINATION bin COMPONENT Templates
   LIBRARY DESTINATION ${LIB_INSTALL_DIR} COMPONENT Templates
   ARCHIVE DESTINATION ${LIB_INSTALL_DIR} COMPONENT Templates
-  INCLUDES DESTINATION include
+  INCLUDES DESTINATION include/grantlee5
 )
 
 install(FILES
@@ -164,10 +164,10 @@ install(FILES
   token.h
   util.h
   variable.h
-  DESTINATION include/grantlee COMPONENT Templates
+  DESTINATION include/grantlee5/grantlee COMPONENT Templates
 )
 
 install(FILES
   grantlee_templates.h
-  DESTINATION include COMPONENT Templates
+  DESTINATION include/grantlee5 COMPONENT Templates
 )
Index: grantlee-5.3.1/textdocument/lib/CMakeLists.txt
===================================================================
--- grantlee-5.3.1.orig/textdocument/lib/CMakeLists.txt
+++ grantlee-5.3.1/textdocument/lib/CMakeLists.txt
@@ -54,7 +54,7 @@ install(TARGETS Grantlee_TextDocument EX
   RUNTIME DESTINATION bin COMPONENT TextDocument
   LIBRARY DESTINATION ${LIB_INSTALL_DIR} COMPONENT TextDocument
   ARCHIVE DESTINATION ${LIB_INSTALL_DIR} COMPONENT TextDocument
-  INCLUDES DESTINATION include
+  INCLUDES DESTINATION include/grantlee5
 )
 
 install(FILES
@@ -65,10 +65,10 @@ install(FILES
   texthtmlbuilder.h
   mediawikimarkupbuilder.h
   ${CMAKE_CURRENT_BINARY_DIR}/grantlee_textdocument_export.h
-  DESTINATION include/grantlee COMPONENT TextDocument
+  DESTINATION include/grantlee5/grantlee COMPONENT TextDocument
 )
 
 install(FILES
   grantlee_textdocument.h
-  DESTINATION include COMPONENT TextDocument
+  DESTINATION include/grantlee5 COMPONENT TextDocument
 )
