Index: src/osip2/port_sema.c
===================================================================
--- src/osip2/port_sema.c.orig
+++ src/osip2/port_sema.c
@@ -300,6 +300,16 @@ osip_sem_trywait (struct osip_sem *_sem)
 #elif defined (HAVE_SYS_SEM_H)
 /* support for semctl, semop, semget */
 
+#ifdef _SEM_SEMUN_UNDEFINED
+union semun
+{
+  int val;                           /* value for SETVAL */
+  struct semid_ds *buf;              /* buffer for IPC_STAT & IPC_SET */
+  unsigned short int *array;         /* array for GETALL & SETALL */
+  struct seminfo *__buf;             /* buffer for IPC_INFO */
+};
+#endif
+
 #define SEM_PERM 0600
 
 struct osip_sem *
