--- sonatype-sisu-build-api-f1f8849/src/main/java/org/sonatype/plexus/build/incremental/DefaultBuildContext.java	2022-05-19 08:15:44.882357964 +0200
+++ sonatype-sisu-build-api-f1f8849/src/main/java/org/sonatype/plexus/build/incremental/DefaultBuildContext.java	2022-05-19 08:16:29.470635918 +0200
@@ -30,7 +30,7 @@
  * hasDelta returns <code>true</code> for all paths
  * newScanner returns Scanner that scans all files under provided basedir
  * newDeletedScanner always returns empty scanner.
- * isIncremental returns <code>false</code<
+ * isIncremental returns <code>false</code>
  * getValue always returns <code>null</code>
  * 
  * @plexus.component role="org.sonatype.plexus.build.incremental.BuildContext"
