--- osu-micro-benchmarks-6.1.orig/c/util/osu_util_mpi.c	2022-09-19 15:48:03.000000000 +0200
+++ osu-micro-benchmarks-6.1/c/util/osu_util_mpi.c	2022-09-30 10:51:18.683964260 +0200
@@ -211,7 +211,7 @@
         bad_usage.message = "Please pass a ddt"
             " type[cont,vect,indx]\n";
         bad_usage.optarg = optarg;
-        return PO_BAD_USAGE;
+        return;
     }
     option = strtok(optarg, ":");
     if (0 == strncasecmp(optarg, "vect", 4)) {
@@ -236,7 +236,7 @@
         bad_usage.message = "Invalid ddt type. Valid ddt"
             " types[cont,vect,indx]\n";
         bad_usage.optarg = optarg;
-        return PO_BAD_USAGE;
+        return;
     }
 }
 
@@ -2006,7 +2006,7 @@
     int displacements[OMB_DDT_INDEXED_MAX_LENGTH] = {0};
 
     if (0 == options.omb_enable_ddt) {
-        return;
+        return 0;
     }
     switch (options.ddt_type) {
         case OMB_DDT_CONTIGUOUS:
