Index: festival/src/scripts/festival_server.sh
===================================================================
--- festival.orig/src/scripts/festival_server.sh
+++ festival/src/scripts/festival_server.sh
@@ -210,14 +210,19 @@ trap "handle_term" 0
 
 if $show
     then
-    create_server_startup $port $server_log /tmp/$$ 3>/dev/null
+    tmpfile=`mktemp -q`
+    if test $? -ne 0; then
+        echo "Error while getting configuration."
+        exit 1
+    fi
+    create_server_startup $port $server_log "$tmpfile" 3>/dev/null
     fl=false
     while read l
 	do
 	if $fl ; then echo $l ; fi
 	if [ "$l" = ";---" ] ; then fl=true ; fi
-    done </tmp/$$ 
-    /bin/rm -f /tmp/$$
+    done < "$tmpfile"
+    /bin/rm -f "$tmpfile"
     exit 0
 fi
 
