Index: mac-robber.c
===================================================================
--- mac-robber.c.orig	2010-02-13 02:21:24.000000000 +0100
+++ mac-robber.c	2011-08-17 16:27:31.277572859 +0200
@@ -70,7 +70,7 @@ usage(char *prog)
 }
 
 static void
-print_version()
+print_version(void)
 {
     printf("mac-robber information:\n");
     printf("   version: %s \n", VER);
@@ -84,7 +84,7 @@ print_version()
  * print the header information to stdout
  */
 static void
-print_header()
+print_header(void)
 {
     char hostnamebuf[BUFSIZ];
     unsigned long now;
