--- apache-ant-1.10.14/src/main/org/apache/tools/ant/taskdefs/Zip.java	2024-09-18 09:25:56.006778808 +0200
+++ apache-ant-1.10.14/src/main/org/apache/tools/ant/taskdefs/Zip.java	2024-09-18 11:06:39.405766871 +0200
@@ -846,6 +846,14 @@
                 archiveType);
         }
 
+        if (fixedModTime == null) {
+            String sde = System.getenv("SOURCE_DATE_EPOCH");
+            if (sde != null) {
+                // the DateUtils.parseLenientDateTime requires miliseconds
+                fixedModTime = sde + "000";
+            }
+        }
+
         if (fixedModTime != null) {
             try {
                 modTimeMillis = DateUtils.parseLenientDateTime(fixedModTime).getTime();
