Index: radare2-6.1.2/pkgcfg/r_muta.pc.acr
===================================================================
--- radare2-6.1.2.orig/pkgcfg/r_muta.pc.acr
+++ radare2-6.1.2/pkgcfg/r_muta.pc.acr
@@ -8,4 +8,4 @@ Description: radare foundation libraries
 Version: @VERSION@
 Requires: r_util
 Libs: -L${libdir} -lr_muta  
-Cflags: -I${includedir}/libr 
+Cflags: -I${includedir}/libr -I${includedir}/libr/r_muta
Index: radare2-6.1.2/pkgcfg/r_util.pc.acr
===================================================================
--- radare2-6.1.2.orig/pkgcfg/r_util.pc.acr
+++ radare2-6.1.2/pkgcfg/r_util.pc.acr
@@ -9,4 +9,4 @@ Version: @VERSION@
 Requires: 
 Libs: -L${libdir} -lr_util  @DL_LIBS@ @LZ4_LDFLAGS@
 Libs.private: @PKGCFG_ZLIBS@
-Cflags: -I${includedir}/libr 
+Cflags: -I${includedir}/libr -I${includedir}/libr/r_util
