From e8d2bf94bccd1a4884b519af203f3741ad2121a7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kristoffer=20Gr=C3=B6nlund?= <krig@koru.se>
Date: Wed, 1 Feb 2017 14:52:50 +0100
Subject: [PATCH 3/5] ldirectord: don't create subsys lock

---
 ldirectord/init.d/ldirectord.in | 2 --
 1 file changed, 2 deletions(-)

diff --git a/ldirectord/init.d/ldirectord.in b/ldirectord/init.d/ldirectord.in
index 9d8083e4..f7ab5525 100755
--- a/ldirectord/init.d/ldirectord.in
+++ b/ldirectord/init.d/ldirectord.in
@@ -60,11 +60,9 @@ action() {
 case "$1" in
   start)
         action "Starting ldirectord" $DAEMON start
-	touch /var/lock/subsys/ldirectord
 	;;
   stop)
         action "Stopping ldirectord" $DAEMON stop
-	rm -f /var/lock/subsys/ldirectord
 	;;
   restart)
         action "Restarting ldirectord" $DAEMON restart
-- 
2.13.2

