diff -wruN -x '*~' -x '*.o' -x '*.a' -x '*.so' -x '*.so.[0-9]' -x autom4te.cache -x .deps -x .libs ../orig-openlmi-networking-0.3.0/src/CMakeLists.txt ./src/CMakeLists.txt
--- ../orig-openlmi-networking-0.3.0/src/CMakeLists.txt	2014-09-04 11:09:32.000000000 +0200
+++ ./src/CMakeLists.txt	2014-10-07 13:46:02.735491746 +0200
@@ -104,7 +104,7 @@
 cim_registration(${PROVIDER_NAME} ${LIBRARY_NAME} ${MOF} share/openlmi-networking "${TARGET_MOF}")
 
 install(TARGETS ${LIBRARY_NAME} DESTINATION lib${LIB_SUFFIX}/cmpi/)
-install(PROGRAMS ${CIMPROVAGT_SCRIPT} DESTINATION libexec/pegasus)
+install(PROGRAMS ${CIMPROVAGT_SCRIPT} DESTINATION lib/pegasus)
 install(FILES ${TARGET_MOF} DESTINATION share/openlmi-networking/)
 
 add_subdirectory(unittest)
diff -wruN -x '*~' -x '*.o' -x '*.a' -x '*.so' -x '*.so.[0-9]' -x autom4te.cache -x .deps -x .libs ../orig-openlmi-networking-0.3.0/src/cmpiLMI_Networking-cimprovagt ./src/cmpiLMI_Networking-cimprovagt
--- ../orig-openlmi-networking-0.3.0/src/cmpiLMI_Networking-cimprovagt	2014-09-04 11:09:32.000000000 +0200
+++ ./src/cmpiLMI_Networking-cimprovagt	2014-10-07 13:46:02.735491746 +0200
@@ -20,4 +20,4 @@
 #
 
 
-/usr/libexec/pegasus/cimprovagt "$@"
+/usr/sbin/cimprovagt "$@"
