Index: folks-0.11.4/backends/tracker/lib/FolksTracker-0.6.gir
===================================================================
--- folks-0.11.4.orig/backends/tracker/lib/FolksTracker-0.6.gir
+++ folks-0.11.4/backends/tracker/lib/FolksTracker-0.6.gir
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <repository version="1.2" xmlns="http://www.gtk.org/introspection/core/1.0" xmlns:c="http://www.gtk.org/introspection/c/1.0" xmlns:glib="http://www.gtk.org/introspection/glib/1.0">
 <include name="Folks" version="0.6"/>
-<include name="Tracker" version="1.0"/>
+<include name="Tracker" version="2.0"/>
 <include name="Gee" version="0.8"/>
 <include name="GObject" version="2.0"/>
 <package name="folks-tracker"/>
