From b3dea479b766213fa7e6c00af67eb7d1b757b295 Mon Sep 17 00:00:00 2001
From: bee-keeper <ricbottomley@gmail.com>
Date: Sat, 3 Feb 2018 15:08:32 +0000
Subject: [PATCH] Devel (#97)

 test_settings.py |  6 ++--

diff --git a/test_settings.py b/test_settings.py
index e44606d..d64b685 100644
--- a/test_settings.py
+++ b/test_settings.py
@@ -40,15 +40,15 @@
         "django.contrib.messages.context_processors.messages",
     )
 
-MIDDLEWARE_CLASSES = (
+MIDDLEWARE = [
+    'django.middleware.security.SecurityMiddleware',
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.middleware.common.CommonMiddleware',
     'django.middleware.csrf.CsrfViewMiddleware',
     'django.contrib.auth.middleware.AuthenticationMiddleware',
-    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
-)
+]
 
 EMAIL_BACKEND = 'django.core.mail.backends.locmem.EmailBackend'
 
