From 7cff6237dacf97be00204de0455c15733948bba9 Mon Sep 17 00:00:00 2001
From: dghart <dghart@users.sourceforge.net>
Date: Tue, 29 Mar 2022 16:56:54 +0100
Subject: [PATCH]  Compilation fix for wxWidgets >3.1.5

---
 Configure.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Configure.cpp b/Configure.cpp
index b6af5dd..471414f 100644
--- a/Configure.cpp
+++ b/Configure.cpp
@@ -141,7 +141,12 @@ Configure::DetectFixedDevices();  // Now there's an ini, add to it any detectabl
   wxDEFINE_EVENT(WizardCommandEvent, wxCommandEvent);
 #endif
 
-NoConfigWizard::NoConfigWizard(wxWindow* parent, int id, const wxString& title, const wxString& configFPath)  :  wxWizard(parent, id, title, wizardbitmap), m_configFPath(configFPath)
+NoConfigWizard::NoConfigWizard(wxWindow* parent, int id, const wxString& title, const wxString& configFPath)
+#if wxVERSION_NUMBER > 3105
+    : wxWizard(parent, id, title,  wxBitmapBundle::FromBitmap(wizardbitmap)), m_configFPath(configFPath)
+#else
+    : wxWizard(parent, id, title, wizardbitmap), m_configFPath(configFPath)
+#endif
 {
 m_Next = m_btnNext;
 
-- 
2.37.2

