--- garlic-1.3/register.c
+++ garlic-1.3/register.c
@@ -33,6 +33,8 @@
 
 int Register_ (void)
 {
+	return 1;
+
 char		outfile_nameA[SHORTSTRINGSIZE];
 char		command_stringA[STRINGSIZE];
 FILE		*outfileP;
--- garlic-1.3/typedefs.h
+++ garlic-1.3/typedefs.h
@@ -38,7 +38,7 @@
 
 
 /* Configuration data from  $HOME/.garlicrc , or - if this file */
-/* does not exist - from  /usr/local/lib/garlic/.garlicrc file: */
+/* does not exist - from  /etc/garlicrc file: */
 /*----------------------------------*/
 /* _atomic_ = atomic world units    */
 /* _real_   = real world units      */
