Index: gtkam-1.0/src/gtkam-preview.c
===================================================================
--- gtkam-1.0.orig/src/gtkam-preview.c
+++ gtkam-1.0/src/gtkam-preview.c
@@ -44,6 +44,8 @@
 
 #include <gdk-pixbuf/gdk-pixbuf-loader.h>
 
+#include <gphoto2/gphoto2-setting.h>
+
 #include "gdk-pixbuf-hacks.h"
 
 #include "gtkam-config.h"
@@ -561,7 +563,7 @@ gtkam_preview_new (GtkamCamera *camera)
             G_CALLBACK (on_button_file_selection_changed), preview);
 
 	if (gp_setting_get ("gtkam-preview", "download_folder", buf) == GP_OK) {
-		gtk_file_chooser_set_filename (preview->priv->button_file, buf);
+		gtk_file_chooser_set_filename (GTK_FILE_CHOOSER (preview->priv->button_file), buf);
 	}
 	if (gp_setting_get ("gtkam-preview", "direct_download", buf) == GP_OK) {
 		gtk_toggle_button_set_active (
Index: gtkam-1.0/src/gtkam-main.c
===================================================================
--- gtkam-1.0.orig/src/gtkam-main.c
+++ gtkam-1.0/src/gtkam-main.c
@@ -452,12 +452,12 @@ action_about (GtkAction *action, gpointe
 
 	for (n = 0; module_versions[n].name != NULL; n++) {
 		gchar *features;
-		const char **v = NULL;
+		char **v = NULL;
 		char *name = module_versions[n].name;
 		GPVersionFunc func = module_versions[n].version_func;
 		CHECK_NULL (name);
 		CHECK_NULL (func);
-		v = func(GP_VERSION_SHORT);
+		v = (char**)func(GP_VERSION_SHORT);
 		CHECK_NULL (v);
 		CHECK_NULL (v[0]);
 		CHECK_NULL (v[1]);
Index: gtkam-1.0/src/gtkam-gimp.c
===================================================================
--- gtkam-1.0.orig/src/gtkam-gimp.c
+++ gtkam-1.0/src/gtkam-gimp.c
@@ -457,7 +457,7 @@ run_load (gchar *name, gint nparams, Gim
 }
 
 static void
-run (gchar *name, gint nparams, GimpParam *params, gint *nreturn_vals,
+run (const gchar *name, gint nparams, const GimpParam *params, gint *nreturn_vals,
      GimpParam **return_vals)
 {
 	static GimpParam values[1];
