From: Egbert Eich <eich@suse.com>
Date: Sun Nov 29 20:38:47 2020 +0100
Subject: Add return value to non-void function
Patch-mainline: Not yet
Git-commit: 98cc1a7453a2f55dfe2477d62e8ce3f049e0ed1f
References: 

Signed-off-by: Egbert Eich <eich@suse.com>
---
 testing/mt/mt_common.c | 1 +
 1 file changed, 1 insertion(+)
diff --git a/testing/mt/mt_common.c b/testing/mt/mt_common.c
index ff92d9e..bda59c3 100644
--- a/testing/mt/mt_common.c
+++ b/testing/mt/mt_common.c
@@ -310,6 +310,7 @@ void* _mt_common_proxy_thread(void *id_ptr)
 {
   test_thr_data_t *tdata = (test_thr_data_t*)id_ptr;
   tdata->fptr(tdata->tid);
+  return NULL;
 }
 
 void mt_common_exec(mt_common_thrptr_t *workers)
