--- src/scanner.l
+++ src/scanner.l
@@ -691,7 +691,7 @@
   return strcmp (first->msgid, second->msgid) == 0;
 }
 
-static bool
+static void
 prepare_for_rescan (void)
 {
   char *copy = alloca (yyleng);
--- src/xpot.c
+++ src/xpot.c
@@ -4,6 +4,7 @@
 
 #include "xpot.h"
 #include <time.h>
+#include <string.h>
 
 #include <errno.h>
 #ifndef errno
@@ -16,9 +17,9 @@
 /* Name of executing program.  */
 const char *program_name;
 /* Display usage information and exit.  */
-static bool show_help = false;
+static int show_help = false;
 /* Print the version and exit.  */
-static bool show_version = false;
+static int show_version = false;
 
 /* Extract documentation strings.  */
 bool doc_strings_option = false;
@@ -477,14 +478,14 @@
 {
   {"doc-strings", no_argument, NULL, 'd'},
   {"domain", required_argument, NULL, 'p'},
-  {"help", no_argument, (int *) &show_help, 1},
+  {"help", no_argument, &show_help, 1},
   {"keyword", required_argument, NULL, 'k'},
   {"no-header", no_argument, NULL, 'n'},
   {"output_file", required_argument, NULL, 'o'},
   {"parser-debug", 0, NULL, 'y'},
   {"release", required_argument, NULL, 'v'},
   {"scanner-debug", 0, NULL, 'l'},
-  {"version", no_argument, (int *) &show_version, 1},
+  {"version", no_argument, &show_version, 1},
   {0, 0, 0, 0},
 };
 
