Index: spew-1.0.8/configure.ac
===================================================================
--- spew-1.0.8.orig/configure.ac
+++ spew-1.0.8/configure.ac
@@ -21,6 +21,7 @@ AC_PROG_LN_S
 # Checks for libraries.
 AC_CHECK_LIB(popt, poptGetContext, ,AC_MSG_ERROR([PACKAGE] requires popt))
 AC_CHECK_LIB(ncurses, wresize, ,AC_MSG_ERROR([PACKAGE] requires ncurses))
+PKG_CHECK_MODULES(NCURSES, [ncurses])
 
 # Checks for header files.
 AC_HEADER_STDC
Index: spew-1.0.8/src/Makefile.am
===================================================================
--- spew-1.0.8.orig/src/Makefile.am
+++ spew-1.0.8/src/Makefile.am
@@ -24,7 +24,7 @@ libcommon_a_SOURCES = \
 libtui_a_SOURCES = \
         Tui.cpp Tui.h
 
-LDADD = libcommon.a libtui.a
+LDADD = libcommon.a libtui.a $(NCURSES_LIBS)
 
 bin_PROGRAMS = spew
 spew_SOURCES = spew.cpp \
@@ -37,6 +37,7 @@ spew_SOURCES = spew.cpp \
                SpewTuiStatisticsWindow.cpp SpewTuiStatisticsWindow.h \
                SpewTuiStatusWindow.cpp SpewTuiStatusWindow.h
 
+spew_CFLAGS = $(NCURSES_CFLAGS)
 spew_LDFLAGS = $(STATIC_LINK)
 
 install-exec-local:
