--- joda-time-2.9.9/src/test/java/org/joda/time/TestDateTimeZone.java	2017-03-23 14:33:18.000000000 +0100
+++ joda-time-2.9.9/src/test/java/org/joda/time/TestDateTimeZone.java	2018-11-20 21:45:41.269188883 +0100
@@ -20,6 +20,7 @@
 import java.io.FilePermission;
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
+import java.lang.Integer;
 import java.lang.reflect.Modifier;
 import java.security.AllPermission;
 import java.security.CodeSource;
@@ -759,7 +760,7 @@
         boolean jdk9 = true;
         try {
             String str = System.getProperty("java.version");
-            jdk9 = str.startsWith("9");
+            jdk9 = (9 <= Integer.parseInt((str.trim().split("\\s+|-\\s*|\\.\\s*"))[0]));
         } catch (Exception ex) {
             jdk9 = false;
         }
