Index: wptools-0.4.15/setup.py
===================================================================
--- wptools-0.4.15.orig/setup.py
+++ wptools-0.4.15/setup.py
@@ -2,8 +2,6 @@
 
 from setuptools import setup, find_packages
 
-with open('README.rst') as f:
-    readme = f.read()
 with open('HISTORY.rst') as f:
     history = f.read()
 
@@ -12,7 +10,16 @@ setup(
     name='wptools',
     version='0.4.17',
     description='Wikipedia tools (for Humans)',
-    long_description=readme + '\n\n' + history,
+    long_description=
+'''Python and command-line MediaWiki access for Humans.
+
+Features:
+* get an HTML or plain text "extract" (lead or summary)
+* get a representative image (pageimage, thumb, etc.)
+* get an Infobox as a python dictionary
+* get any/all Wikidata by title
+* get info in any language
+* get random info''' + '\n\n' + history,
     url='https://github.com/siznax/wptools/',
     license='MIT',
     author='Steve @siznax',
