diff --git a/XS/PropertySheetDialog.xsp b/XS/PropertySheetDialog.xsp
index 5164a13..4e72673 100644
--- a/XS/PropertySheetDialog.xsp
+++ b/XS/PropertySheetDialog.xsp
@@ -49,7 +49,13 @@
     void SetBookCtrl( wxBookCtrlBase* book );
     wxBookCtrlBase* GetBookCtrl() const;
 
+#if WXPERL_W_VERSION_GE( 3, 0, 3 )
+    void SetInnerSizer( wxSizer* sizer );
+    %name{SetInnerSize} void SetInnerSizer( wxSizer* sizer );
+#else
     void SetInnerSize( wxSizer* sizer );
+    %name{SetInnerSizer} void SetInnerSize( wxSizer* sizer );
+#endif
     wxSizer* GetInnerSizer() const;
 
 #if WXPERL_W_VERSION_GE( 2, 7, 2 )
