commit f9196c3e0c7b03bedfedeb981987e976f3c59aea
Author: Ignaz Forster <iforster@suse.com>
Date:   Mon Aug 10 15:33:18 2020 +0200
References: (gh#coreos/ignition#1069)
Upstream: Submitted

    Add support for building on 32 bit x86 archticture

diff --git a/Makefile b/Makefile
index 35b0deda..d80e0870 100644
--- a/Makefile
+++ b/Makefile
@@ -6,6 +6,8 @@ ifeq ($(GOARCH),x86_64)
 	GOARCH=amd64
 else ifeq ($(GOARCH),aarch64)
 	GOARCH=arm64
+else ifeq ($(patsubst i%86,386,$(GOARCH)),386)
+	GOARCH=386
 endif
 
 .PHONY: all
