--- chromium-147.0.7727.3/third_party/blink/renderer/platform/media/BUILD.gn	2026/03/18 13:00:37	1.1
+++ chromium-147.0.7727.3/third_party/blink/renderer/platform/media/BUILD.gn	2026/03/18 13:03:25
@@ -114,6 +114,13 @@
     "//third_party/blink/renderer/platform:allow_discouraged_type",
   ]
 
+  if (media_use_ffmpeg) {
+    deps += [
+      "//third_party/ffmpeg",
+      "//third_party/ffmpeg:ffmpeg_features",
+    ]
+  }
+
   if (enable_hls_demuxer) {
     sources += [
       "multi_buffer_data_source_factory.cc",
