diff -up chromium-122.0.6261.57/base/allocator/partition_allocator/partition_alloc.gni.me chromium-122.0.6261.57/base/allocator/partition_allocator/partition_alloc.gni
--- chromium-122.0.6261.57/base/allocator/partition_allocator/partition_alloc.gni.me	2024-02-22 20:22:44.115269845 +0100
+++ chromium-122.0.6261.57/base/allocator/partition_allocator/partition_alloc.gni	2024-02-22 20:23:07.067681890 +0100
@@ -29,8 +29,7 @@ if (is_nacl) {
   assert(false, "Unknown CPU: $current_cpu")
 }
 
-has_memory_tagging =
-    current_cpu == "arm64" && is_clang && !is_asan && (is_linux || is_android)
+has_memory_tagging = false
 
 declare_args() {
   # Causes all the allocations to be routed via allocator_shim.cc. Usually,
