diff --git a/media_switches.cc b/media_switches.cc.new
index 9cffb93..bd79b68 100644
--- a/media/base/media_switches.cc
+++ b/media/base/media_switches.cc
@@ -1649,7 +1649,7 @@ BASE_FEATURE(kUseSharedImagesForPepperVideo,
 // Enables FFmpeg allow lists for supported codecs / containers.
 BASE_FEATURE(kFFmpegAllowLists,
              "FFmpegAllowLists",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 #if BUILDFLAG(ENABLE_FFMPEG_VIDEO_DECODERS)
 // Allows decoding of theora / vp3 content.
