--- chromium-122.0.6261.57/chrome/browser/ui/views/bubble/bubble_contents_wrapper.h	2024/03/06 11:19:44	1.1
+++ chromium-122.0.6261.57/chrome/browser/ui/views/bubble/bubble_contents_wrapper.h	2024/03/06 11:23:50
@@ -148,10 +148,7 @@
                               webui_resizes_host,
                               esc_closes_ui,
                               T::GetWebUIName()),
-        webui_url_(webui_url) {
-    static_assert(
-        views_metrics::IsValidWebUINameVariant("." + T::GetWebUIName()));
-  }
+        webui_url_(webui_url) {}
 
   void ReloadWebContents() override {
     web_contents()->GetController().LoadURL(webui_url_, content::Referrer(),
