--- chromium-138.0.7180.2/build/rust/std/rules/BUILD.gn	2025/05/20 14:59:34	1.1
+++ chromium-138.0.7180.2/build/rust/std/rules/BUILD.gn	2025/05/20 15:01:08
@@ -75,23 +75,23 @@
   output_dir =
       "$root_out_dir/local_rustc_sysroot/lib/rustlib/$rust_abi_target/lib/"
 }
-cargo_crate("adler2") {
+cargo_crate("adler") {
   crate_type = "rlib"
-  crate_root = "//third_party/rust-toolchain/lib/rustlib/src/rust/library/vendor/adler2-2.0.0/src/lib.rs"
+  crate_root = "//third_party/rust-toolchain/lib/rustlib/src/rust/library/vendor/adler-1.0.2/src/lib.rs"
   enabled = !is_win
   sources = [
-    "//third_party/rust-toolchain/lib/rustlib/src/rust/library/vendor/adler2-2.0.0/src/algo.rs",
-    "//third_party/rust-toolchain/lib/rustlib/src/rust/library/vendor/adler2-2.0.0/src/lib.rs",
+    "//third_party/rust-toolchain/lib/rustlib/src/rust/library/vendor/adler-1.0.2/src/algo.rs",
+    "//third_party/rust-toolchain/lib/rustlib/src/rust/library/vendor/adler-1.0.2/src/lib.rs",
   ]
   inputs = []
   no_std = true
 
   # Unit tests skipped. Generate with --with-tests to include them.
   build_native_rust_unit_tests = false
-  edition = "2021"
-  cargo_pkg_version = "2.0.0"
-  cargo_pkg_authors = "Jonas Schievink <jonasschievink@gmail.com>, oyvindln <oyvindln@users.noreply.github.com>"
-  cargo_pkg_name = "adler2"
+  edition = "2015"
+  cargo_pkg_version = "1.0.2"
+  cargo_pkg_authors = "Jonas Schievink <jonasschievink@gmail.com>"
+  cargo_pkg_name = "adler"
   cargo_pkg_description =
       "A simple clean-room implementation of the Adler-32 checksum"
   library_configs -= [
@@ -1719,7 +1719,7 @@
   ]
   if (!is_win) {
     deps += [
-      ":adler2",
+      ":adler",
       ":compiler_builtins",
       ":rustc_std_workspace_alloc",
       ":rustc_std_workspace_core",
--- chromium-138.0.7180.2/build/rust/std/BUILD.gn	2025/05/20 15:09:01	1.1
+++ chromium-138.0.7180.2/build/rust/std/BUILD.gn	2025/05/20 15:09:08
@@ -50,7 +50,7 @@
     # These are no longer present in the Windows toolchain.
     stdlib_files += [
       "addr2line",
-      "adler2",
+      "adler",
       "gimli",
       "libc",
       "memchr",
