--- chromium-137.0.7151.15/third_party/pdfium/core/fxcodec/png/png_decoder.cpp	2025/05/12 14:04:10	1.1
+++ chromium-137.0.7151.15/third_party/pdfium/core/fxcodec/png/png_decoder.cpp	2025/05/12 14:04:27
@@ -52,8 +52,8 @@
                              CFX_DIBAttribute* pAttribute) {
   if (pAttribute) {
 #if defined(PNG_pHYs_SUPPORTED)
-    pAttribute->xdpi_ = png_get_x_pixels_per_meter(png_ptr, info_ptr);
-    pAttribute->ydpi_ = png_get_y_pixels_per_meter(png_ptr, info_ptr);
+    pAttribute->x_dpi_ = png_get_x_pixels_per_meter(png_ptr, info_ptr);
+    pAttribute->y_dpi_ = png_get_y_pixels_per_meter(png_ptr, info_ptr);
     png_uint_32 res_x, res_y;
     int unit_type;
     png_get_pHYs(png_ptr, info_ptr, &res_x, &res_y, &unit_type);
