diff --git a/chrome/browser/safe_browsing/download_protection/file_analyzer.cc b/chrome/browser/safe_browsing/download_protection/file_analyzer.cc
index 2a930f44908f4..095a1ac913334 100644
--- a/chrome/browser/safe_browsing/download_protection/file_analyzer.cc
+++ b/chrome/browser/safe_browsing/download_protection/file_analyzer.cc
@@ -78,11 +78,6 @@ void FileAnalyzer::Start(const base::FilePath& target_file_name,
     return;
   }
 
-  if (inspection_type == DownloadFileType::RAR) {
-    StartExtractRarFeatures();
-    return;
-  }
-
 #if BUILDFLAG(IS_MAC)
   if (inspection_type == DownloadFileType::DMG) {
     StartExtractDmgFeatures();
