commit bbd4b7752f0a9e5f486fa55c9f2b80071ef99d01
Author: lauren n. liberda <lauren@selfisekai.rocks>
Date:   Thu Apr 25 04:24:47 2024 +0200

    iwyu: missing #include <memory> for std::unique_ptr

--- a/third_party/vulkan-deps/vulkan-utility-libraries/src/include/vulkan/utility/vk_small_containers.hpp
+++ b/third_party/vulkan-deps/vulkan-utility-libraries/src/include/vulkan/utility/vk_small_containers.hpp
@@ -9,6 +9,7 @@
 #pragma once
 
 #include <cassert>
+#include <memory>
 #include <unordered_map>
 #include <unordered_set>
 
