Index: chromium-126.0.6478.8/gpu/command_buffer/service/shared_image/shared_image_backing.h
===================================================================
--- chromium-126.0.6478.8.orig/gpu/command_buffer/service/shared_image/shared_image_backing.h
+++ chromium-126.0.6478.8/gpu/command_buffer/service/shared_image/shared_image_backing.h
@@ -386,7 +386,7 @@ class GPU_GLES2_EXPORT SharedImageBackin
  private:
   class ScopedWriteUMA {
    public:
-    ScopedWriteUMA() = default;
+    ScopedWriteUMA() : content_consumed_(false) {}
 
     ScopedWriteUMA(const ScopedWriteUMA&) = delete;
     ScopedWriteUMA& operator=(const ScopedWriteUMA&) = delete;
@@ -400,7 +400,7 @@ class GPU_GLES2_EXPORT SharedImageBackin
     void SetConsumed() { content_consumed_ = true; }
 
    private:
-    bool content_consumed_ = false;
+    bool content_consumed_;
   };
 
   const Mailbox mailbox_;
Index: chromium-126.0.6478.8/third_party/blink/renderer/core/loader/history_item.h
===================================================================
--- chromium-126.0.6478.8.orig/third_party/blink/renderer/core/loader/history_item.h
+++ chromium-126.0.6478.8/third_party/blink/renderer/core/loader/history_item.h
@@ -77,13 +77,13 @@ class CORE_EXPORT HistoryItem final : pu
     DISALLOW_NEW();
 
    public:
-    ViewState() = default;
+    ViewState() : page_scale_factor_(0) {}
     ViewState(const ViewState&) = default;
     ViewState& operator=(const ViewState&) = default;
 
     ScrollOffset visual_viewport_scroll_offset_;
     ScrollOffset scroll_offset_;
-    float page_scale_factor_ = 0;
+    float page_scale_factor_;
     ScrollAnchorData scroll_anchor_data_;
   };
 
Index: chromium-126.0.6478.8/third_party/blink/public/common/interest_group/auction_config.h
===================================================================
--- chromium-126.0.6478.8.orig/third_party/blink/public/common/interest_group/auction_config.h
+++ chromium-126.0.6478.8/third_party/blink/public/common/interest_group/auction_config.h
@@ -247,9 +247,11 @@ struct BLINK_COMMON_EXPORT AuctionConfig
 
     // Povides whether debug mode is enabled and, if it is, any debug key.
     struct BLINK_COMMON_EXPORT AuctionReportBuyerDebugModeConfig {
-      bool is_enabled = false;
+      bool is_enabled;
       std::optional<uint64_t> debug_key;
 
+      AuctionReportBuyerDebugModeConfig() : is_enabled(false) {}
+
       friend BLINK_COMMON_EXPORT bool operator==(
           const AuctionReportBuyerDebugModeConfig&,
           const AuctionReportBuyerDebugModeConfig&);
@@ -463,7 +465,7 @@ struct BLINK_COMMON_EXPORT AuctionConfig
   // Origin for the Coordinator to be used for Private Aggregation.
   std::optional<url::Origin> aggregation_coordinator_origin;
 
-  static_assert(__LINE__ == 466, R"(
+  static_assert(__LINE__ == 468, R"(
 If modifying AuctionConfig fields, please make sure to also modify:
 
 * third_party/blink/public/mojom/interest_group/interest_group_types.mojom
