Index: chromium-148.0.7778.5/third_party/blink/renderer/platform/media/BUILD.gn
===================================================================
--- chromium-148.0.7778.5.orig/third_party/blink/renderer/platform/media/BUILD.gn
+++ chromium-148.0.7778.5/third_party/blink/renderer/platform/media/BUILD.gn
@@ -25,6 +25,13 @@ blink_platform_sources("media_final") {
     "//url",
   ]
   deps = [ "//third_party/blink/renderer/platform:allow_discouraged_type" ]
+
+  if (media_use_ffmpeg) {
+    deps += [
+      "//third_party/ffmpeg",
+      "//third_party/ffmpeg:ffmpeg_features",
+    ]
+  }
 }
 
 component("media") {
@@ -112,6 +119,12 @@
     "//third_party/blink/renderer/platform",
     "//third_party/blink/renderer/platform:allow_discouraged_type",
   ]
+  if (media_use_ffmpeg) {
+    deps += [
+      "//third_party/ffmpeg",
+      "//third_party/ffmpeg:ffmpeg_features",
+    ]
+  }
 }
 
 source_set("unit_tests") {
