Index: chromium-144.0.7524.0/build/config/BUILDCONFIG.gn
===================================================================
--- chromium-144.0.7524.0.orig/build/config/BUILDCONFIG.gn
+++ chromium-144.0.7524.0/build/config/BUILDCONFIG.gn
@@ -135,6 +135,8 @@ declare_args() {
   # set "is_official_build" to true for any build intended to ship to end-users.
   is_official_build = false
 
+  gcc_lto = false
+
   # Set to true when compiling with the Clang compiler.
   is_clang = current_os != "linux" ||
              (current_cpu != "mips" && current_cpu != "mips64")
@@ -370,6 +372,12 @@ default_compiler_configs = [
   "//build/config:feature_flags",
 ]
 
+if (gcc_lto) {
+  default_compiler_configs += [
+    "//build/config/compiler:gcc_lto",
+  ]
+}
+
 if (is_win) {
   default_compiler_configs += [
     "//build/config/win:default_cfg_compiler",
Index: chromium-144.0.7524.0/build/config/compiler/BUILD.gn
===================================================================
--- chromium-144.0.7524.0.orig/build/config/compiler/BUILD.gn
+++ chromium-144.0.7524.0/build/config/compiler/BUILD.gn
@@ -2750,6 +2750,10 @@ if (is_win) {
   }
 }
 
+config("gcc_lto") {
+  cflags = [ "-flto" ]
+}
+
 config("default_stack_frames") {
   if (!is_win) {
     if (enable_frame_pointers) {
Index: chromium-144.0.7524.0/sandbox/linux/BUILD.gn
===================================================================
--- chromium-144.0.7524.0.orig/sandbox/linux/BUILD.gn
+++ chromium-144.0.7524.0/sandbox/linux/BUILD.gn
@@ -255,6 +255,9 @@ component("seccomp_bpf") {
     "//base",
     "//sandbox:sandbox_buildflags",
   ]
+  if (gcc_lto) {
+    configs -= [ "//build/config/compiler:gcc_lto" ]
+  }
 
   if (is_android) {
     sources += [
Index: chromium-144.0.7524.0/base/BUILD.gn
===================================================================
--- chromium-144.0.7524.0.orig/base/BUILD.gn
+++ chromium-144.0.7524.0/base/BUILD.gn
@@ -3748,6 +3748,10 @@ test("base_unittests") {
     sources += [ "immediate_crash_unittest.cc" ]
   }
 
+  if (gcc_lto) {
+    configs -= [ "//build/config/compiler:gcc_lto" ]
+  }
+
   sources += [
     "containers/span_rust_unittest.cc",
     "strings/string_piece_rust_unittest.cc",
