--- chromium-142.0.7444.52/third_party/highway/src/hwy/targets.cc	2025/10/24 14:08:16	1.1
+++ chromium-142.0.7444.52/third_party/highway/src/hwy/targets.cc	2025/10/24 14:09:30
@@ -26,7 +26,7 @@
 #if HWY_ARCH_X86
 #include <xmmintrin.h>
 
-#elif (HWY_ARCH_ARM || HWY_ARCH_PPC || HWY_ARCH_S390X || HWY_ARCH_RISCV || \
+#elif (HWY_ARCH_ARM || HWY_ARCH_S390X || HWY_ARCH_RISCV || \
        HWY_ARCH_LOONGARCH) &&                                              \
     HWY_OS_LINUX
 // sys/auxv.h does not always include asm/hwcap.h, or define HWCAP*, hence we
