--- a/third_party/highway/src/hwy/targets.cc
+++ b/third_party/highway/src/hwy/targets.cc
@@ -35,7 +35,7 @@
     HWY_OS_LINUX
 // sys/auxv.h does not always include asm/hwcap.h, or define HWCAP*, hence we
 // still include this directly. See #1199.
-#ifndef TOOLCHAIN_MISS_ASM_HWCAP_H
+#if !defined(TOOLCHAIN_MISS_ASM_HWCAP_H) && !defined(HWY_ARCH_PPC)
 #include <asm/hwcap.h>
 #endif
 #if HWY_HAVE_AUXV
