possible crash, strlen doesnt take NULL
================================================================================
--- pdisk/layout_dump.c
+++ pdisk/layout_dump.c
@@ -75,6 +75,10 @@
     
     max_name = 0;
     for (entry = desc; entry->format != kEnd; entry++) {
+	if (NULL == entry->name) {
+		fprintf(stderr, "pdisk debug: %s(%u) entry->name NULL\n", __FUNCTION__, __LINE__);
+		continue;
+	}
 	value = strlen(entry->name);
 	if (value > max_name) {
 	    max_name = value;
