diff --git a/__init__.py b/__init__.py
index f85c808..a9b37ac 100644
--- a/__init__.py
+++ b/__init__.py
@@ -25,8 +25,13 @@ from mycroft.util.parse import extract_datetime
 from mycroft.util.format import nice_number
 from pyowm import OWM
 from pyowm.webapi25.forecaster import Forecaster
-from pyowm.webapi25.forecastparser import ForecastParser
-from pyowm.webapi25.observationparser import ObservationParser
+try:
+    from pyowm.webapi25.parsers.forecastparser import ForecastParser
+    from pyowm.webapi25.parsers.observationparser import ObservationParser
+except ImportError:
+    from pyowm.webapi25.forecastparser import ForecastParser
+    from pyowm.webapi25.observationparser import ObservationParser
+
 from requests import HTTPError
 
 from mycroft.util.format import nice_time
