From 4451cc3350b8e2f611061bd968a380106650e30c Mon Sep 17 00:00:00 2001
From: Callum Farmer <callumjfarmer13@gmail.com>
Date: Thu, 10 Sep 2020 17:19:11 +0100
Subject: [PATCH] Fix build with vala 0.49

Signed-off-by: Callum Farmer <callumjfarmer13@gmail.com>
---
 src/appsys/AppSystem.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/appsys/AppSystem.vala b/src/appsys/AppSystem.vala
index 4224ec64..70d47601 100644
--- a/src/appsys/AppSystem.vala
+++ b/src/appsys/AppSystem.vala
@@ -70,7 +70,7 @@ public class AppSystem : GLib.Object
     }
 
     private void signal_received(GLib.DBusConnection connection,
-                                 string sender,
+                                 string? sender,
                                  string object_path,
                                  string interface_name,
                                  string signal_name,
-- 


