From e93a42f612fcfc1c5a8c18e8dcd3219c7bea5925 Mon Sep 17 00:00:00 2001
From: Tamas Levai <levait@tmit.bme.hu>
Date: Wed, 7 Oct 2020 15:46:32 +0200
Subject: [PATCH] Fix compilation for Python 3.9

---
 src/objects/bandsplitmodule.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: pyo-1.0.3/src/objects/bandsplitmodule.c
===================================================================
--- pyo-1.0.3.orig/src/objects/bandsplitmodule.c
+++ pyo-1.0.3/src/objects/bandsplitmodule.c
@@ -1636,7 +1636,7 @@ MultiBandMain_new(PyTypeObject *type, Py
 static PyObject *
 MultiBandMain_setFrequencies(MultiBandMain* self, PyObject *arg) {
     int i, bounds = self->nbands - 1;
-    if PyList_Check(arg) {
+    if (PyList_Check(arg)) {
         if (PyList_Size(arg) == bounds) {
             for (i=0; i<bounds; i++) {
                 MultiBandMain_compute_variables(self, PyFloat_AsDouble(PyList_GetItem(arg, i)), i);
