diff -ruN a/vendor/avb/aftltool.py b/vendor/avb/aftltool.py
--- a/vendor/avb/aftltool.py
+++ b/vendor/avb/aftltool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 
 # Copyright 2020, The Android Open Source Project
 #
@@ -47,7 +47,7 @@
 
 # pylint: disable=wrong-import-position,import-error
 import avbtool
-import api_pb2
+from google.protobuf import api_pb2
 # pylint: enable=wrong-import-position,import-error
 
 
diff -ruN a/vendor/avb/avbtool.py b/vendor/avb/avbtool.py
--- a/vendor/avb/avbtool.py
+++ b/vendor/avb/avbtool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 
 # Copyright 2016, The Android Open Source Project
 #
diff -ruN a/vendor/CMakeLists.txt b/vendor/CMakeLists.txt
--- a/vendor/CMakeLists.txt
+++ b/vendor/CMakeLists.txt
@@ -68,6 +68,11 @@
 install(TARGETS adb fastboot "${ANDROID_MKE2FS_NAME}"
 	simg2img img2simg append2simg e2fsdroid ext2simg DESTINATION bin)
 
+# Install some python3 tools
+install(PROGRAMS avb/aftltool avb/aftltool.py DESTINATION bin)
+install(PROGRAMS avb/avbtool avb/avbtool.py DESTINATION bin)
+install(PROGRAMS mkbootimg/mkbootimg.py RENAME mkbootimg DESTINATION bin)
+
 # Install common completion files.
 install(FILES adb/adb.bash RENAME adb DESTINATION "${COMPLETION_COMMON_DIR}")
 install(FILES core/fastboot/fastboot.bash RENAME fastboot DESTINATION "${COMPLETION_COMMON_DIR}")
diff -ruN a/vendor/mkbootimg/mkbootimg.py b/vendor/mkbootimg/mkbootimg.py
--- a/vendor/mkbootimg/mkbootimg.py
+++ b/vendor/mkbootimg/mkbootimg.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 #
 # Copyright 2015, The Android Open Source Project
 #
