--- a/platforms/suse/Makefile
+++ b/platforms/suse/Makefile
@@ -11,14 +11,8 @@
    # unlink old rc script (if not doing a DESTDIR install)
 	$(call CHKCFG,del,/etc/rc.d/apcupsd)
    # install new rc script
-	$(call MKDIR,/etc/rc.d)
-	$(call INSTDATA,744,apcupsd,/etc/rc.d)
-   # save old halt script
-	$(call COPY,/etc/rc.d/halt,/etc/rc.d/halt.old)
-   # insert apcupsd callout into halt script
-	@echo "  PATCH" $(DESTDIR)/etc/rc.d/halt
-	$(V)awk -f awkhaltprog $(DESTDIR)/etc/rc.d/halt.old >$(DESTDIR)/etc/rc.d/halt
-	$(V)chmod 744 $(DESTDIR)/etc/rc.d/halt
+	$(call MKDIR,/etc/init.d)
+	$(call INSTDATA,744,apcupsd,/etc/init.d)
    # link new rc script (if not doing a DESTDIR install)
 	$(call CHKCFG,add,/etc/rc.d/apcupsd)
 	@echo "================================================="
@@ -39,7 +33,5 @@
 
 uninstall-suse:
    # unlink old rc script (if not doing a DESTDIR install)
-	-$(call CHKCFG,del,/etc/rc.d/apcupsd)
-	-$(call UNINST,/etc/rc.d/apcupsd)
-	-$(call COPY,$(DESTDIR)/etc/rc.d/halt.old,/etc/rc.d/halt)
-	-$(call UNINST,/etc/rc.d/halt.old)
+	-$(call CHKCFG,del,/etc/init.d/apcupsd)
+	-$(call UNINST,/etc/init.d/apcupsd)
