Index: libxspf-1.1.0/test/ProjectOpus/ProjectOpusSuite.cpp
===================================================================
--- libxspf-1.1.0.orig/test/ProjectOpus/ProjectOpusSuite.cpp
+++ libxspf-1.1.0/test/ProjectOpus/ProjectOpusSuite.cpp
@@ -28,6 +28,7 @@
 #include <xspf/XspfExtensionReaderFactory.h>
 #include <xspf/ProjectOpus/ProjectOpusPlaylistExtension.h>
 #include <xspf/ProjectOpus/ProjectOpusPlaylistExtensionReader.h>
+#include <cstdio>
 using namespace Xspf;
 using namespace Xspf::ProjectOpus;
 
Index: libxspf-1.1.0/test/OfficialSuite.cpp
===================================================================
--- libxspf-1.1.0.orig/test/OfficialSuite.cpp
+++ libxspf-1.1.0/test/OfficialSuite.cpp
@@ -23,6 +23,7 @@
 
 #include "OfficialSuite.h"
 #include <xspf/XspfReaderCallback.h>
+#include <cstdio>
 using namespace Xspf;
 
 
