From ddce6e86d87810225b81c40841cf16bda304adc7 Mon Sep 17 00:00:00 2001
From: StefanBruens <stefan.bruens@rwth-aachen.de>
Date: Tue, 6 Aug 2019 23:00:59 +0200
Subject: [PATCH] Add explicit CheckSymbolExists include

---
 src/cmake/modules/FindCEPHFS.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/cmake/modules/FindCEPHFS.cmake b/src/cmake/modules/FindCEPHFS.cmake
index e7266fb5ee..706b498b48 100644
--- a/src/cmake/modules/FindCEPHFS.cmake
+++ b/src/cmake/modules/FindCEPHFS.cmake
@@ -127,6 +127,7 @@ else (NOT CEPH_FS)
   endif(NOT CEPH_FS_GET_FS_CID)
 
   set(CMAKE_REQUIRED_INCLUDES ${CEPHFS_INCLUDE_DIR})
+  include(CheckSymbolExists)
   check_symbol_exists(CEPH_STATX_INO "cephfs/libcephfs.h" CEPH_FS_CEPH_STATX)
   if(NOT CEPH_FS_CEPH_STATX)
     message("Cannot find CEPH_STATX_INO. Enabling backward compatibility for pre-ceph_statx APIs.")
