From 8313cd28950a529adfe670b7072b31aabc8fc961 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Thu, 14 Dec 2017 15:14:34 +0100
Subject: [PATCH] fix build with libgsf

---
 src/lib/OutputFileHelper.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/OutputFileHelper.cxx b/src/lib/OutputFileHelper.cxx
index 739ab22..7f5e4b8 100644
--- a/src/lib/OutputFileHelper.cxx
+++ b/src/lib/OutputFileHelper.cxx
@@ -57,7 +57,7 @@ private:
 OutputFileHelperImpl::~OutputFileHelperImpl()
 {
 #ifdef USE_GSF_OUTPUT
-	if (mpOutfile && !gsf_output_close((GsfOutput *) mpOutfile.get()))
+	if (mpOutfile && !gsf_output_close((GsfOutput *) mpOutfile))
 		fprintf(stderr, "ERROR : Couldn't close outfile\n");
 
 	if (mpOutfile)
-- 
2.14.3

