Index: netpbm-10.88.1/converter/pbm/pbmtozinc.c
===================================================================
--- netpbm-10.88.1.orig/converter/pbm/pbmtozinc.c	2019-12-20 12:48:08.981181384 +0100
+++ netpbm-10.88.1/converter/pbm/pbmtozinc.c	2019-12-20 12:48:58.121462453 +0100
@@ -110,7 +110,10 @@ packer_putitem(Packer *      const packe
 
     ++packerP->itemsperline;
 
-    printf ("0x%02x%02x", hi, lo);
+    if (BYTE_ORDER == LITTLE_ENDIAN)
+        printf ("0x%02x%02x", hi, lo);
+    else
+        printf ("0x%02x%02x", lo, hi);
 
 }
 
