Index: gnutls-3.6.14/configure
===================================================================
--- gnutls-3.6.14.orig/configure	2020-06-09 11:01:15.306654318 +0200
+++ gnutls-3.6.14/configure	2020-06-09 12:40:08.262985909 +0200
@@ -66054,12 +66054,12 @@ LIBS="$LIBS $GMP_LIBS"
 $as_echo_n "checking gmp soname... " >&6; }
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
+#include <gmp.h>
 int
 main ()
 {
-
-  ;
+  mpz_t n;
+  mpz_init(n);
   return 0;
 }
 _ACEOF
@@ -66088,12 +66088,12 @@ LIBS="$LIBS $NETTLE_LIBS"
 $as_echo_n "checking nettle soname... " >&6; }
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
+#include <nettle/sha2.h>
 int
 main ()
 {
-
-  ;
+  struct sha256_ctx ctx;
+  sha256_init (&ctx);
   return 0;
 }
 _ACEOF
@@ -66122,12 +66122,12 @@ LIBS="$LIBS $HOGWEED_LIBS"
 $as_echo_n "checking hogweed soname... " >&6; }
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
+#include <nettle/rsa.h>
 int
 main ()
 {
-
-  ;
+  struct rsa_private_key priv;
+  nettle_rsa_private_key_init(&priv);
   return 0;
 }
 _ACEOF
