Index: gnutls-3.6.15/guile/Makefile.in
===================================================================
--- gnutls-3.6.15.orig/guile/Makefile.in	2020-09-08 10:24:09.581998087 +0200
+++ gnutls-3.6.15/guile/Makefile.in	2020-09-08 10:24:30.046134403 +0200
@@ -1857,7 +1857,7 @@ CLEANFILES = modules/gnutls.scm $(am__ap
 TESTS = tests/anonymous-auth.scm tests/session-record-port.scm \
 	tests/pkcs-import-export.scm tests/errors.scm \
 	tests/x509-certificates.scm tests/x509-auth.scm \
-	tests/reauth.scm tests/priorities.scm $(am__append_2)
+	tests/priorities.scm $(am__append_2)
 TESTS_ENVIRONMENT = \
   GUILE_AUTO_COMPILE=0				\
   GUILE_WARN_DEPRECATED=detailed
