Index: gnutls-3.8.10/lib/fips.c
===================================================================
--- gnutls-3.8.10.orig/lib/fips.c
+++ gnutls-3.8.10/lib/fips.c
@@ -702,11 +702,6 @@ int _gnutls_fips_perform_self_checks2(vo
 	}
 
 	/* MAC (includes message digest test) */
-	ret = gnutls_mac_self_test(0, GNUTLS_MAC_SHA1);
-	if (ret < 0) {
-		return gnutls_assert_val(GNUTLS_E_SELF_TEST_ERROR);
-	}
-
 	ret = gnutls_mac_self_test(0, GNUTLS_MAC_SHA224);
 	if (ret < 0) {
 		return gnutls_assert_val(GNUTLS_E_SELF_TEST_ERROR);
