Index: gnutls-3.6.12/lib/fips.c
===================================================================
--- gnutls-3.6.12.orig/lib/fips.c	2019-06-27 06:40:43.000000000 +0200
+++ gnutls-3.6.12/lib/fips.c	2020-03-16 09:29:39.056332128 +0100
@@ -136,7 +136,7 @@ void _gnutls_fips_mode_reset_zombie(void
 }
 
 #define GNUTLS_LIBRARY_NAME "libgnutls.so.30"
-#define NETTLE_LIBRARY_NAME "libnettle.so.6"
+#define NETTLE_LIBRARY_NAME "libnettle.so.7"
 #define HOGWEED_LIBRARY_NAME "libhogweed.so.4"
 #define GMP_LIBRARY_NAME "libgmp.so.10"
 
