Index: hvac-0.11.2/tests/unit_tests/api/secrets_engines/test_kv.py
===================================================================
--- hvac-0.11.2.orig/tests/unit_tests/api/secrets_engines/test_kv.py
+++ hvac-0.11.2/tests/unit_tests/api/secrets_engines/test_kv.py
@@ -1,6 +1,6 @@
 from unittest import TestCase
 
-from mock import MagicMock
+from unittest.mock import MagicMock
 from parameterized import parameterized
 
 from hvac.api.secrets_engines.kv import Kv
Index: hvac-0.11.2/tests/unit_tests/api/test_vault_api_category.py
===================================================================
--- hvac-0.11.2.orig/tests/unit_tests/api/test_vault_api_category.py
+++ hvac-0.11.2/tests/unit_tests/api/test_vault_api_category.py
@@ -1,6 +1,6 @@
 from unittest import TestCase
 
-from mock import create_autospec
+from unittest.mock import create_autospec
 
 from hvac.adapters import Adapter
 from hvac.api.vault_api_base import VaultApiBase
Index: hvac-0.11.2/tests/unit_tests/v1/test_aws_iam_methods.py
===================================================================
--- hvac-0.11.2.orig/tests/unit_tests/v1/test_aws_iam_methods.py
+++ hvac-0.11.2/tests/unit_tests/v1/test_aws_iam_methods.py
@@ -3,7 +3,7 @@ from base64 import b64decode
 from datetime import datetime
 from unittest import TestCase
 
-import mock
+from unittest import mock
 
 from hvac import Client
 
