Index: mathgl-8.0.3/CMakeLists.txt
===================================================================
--- mathgl-8.0.3.orig/CMakeLists.txt
+++ mathgl-8.0.3/CMakeLists.txt
@@ -313,7 +313,7 @@ set(MGL_DEF_FONT "STIX2" CACHE STRING "S
 
 #	set(MGL_CGI_PATH "${CMAKE_INSTALL_PREFIX}/share/mathgl")
 	set(MGL_DATA_PATH "${CMAKE_INSTALL_PREFIX}/share/mathgl")
-	set(MGL_DOC_PATH "${CMAKE_INSTALL_PREFIX}/share/doc/mathgl")
+	set(MGL_DOC_PATH "${CMAKE_INSTALL_PREFIX}/share/doc/packages/mathgl")
 	set(MGL_MAN_PATH "${CMAKE_INSTALL_PREFIX}/share/man")
 	set(MGL_INFO_PATH "${CMAKE_INSTALL_PREFIX}/share/info")
 if(NOT WIN32)
