Index: brlemu-0.1/brlemu.c
===================================================================
--- brlemu-0.1.orig/brlemu.c
+++ brlemu-0.1/brlemu.c
@@ -70,8 +70,14 @@ int main (int argc, char **argv)
       cfsetispeed (&tiodata, B19200) ||
       cfsetospeed (&tiodata, B19200) || tcsetattr (devfd, TCSANOW, &tiodata))
    {
-     fprintf (stderr, "%s: init faild\n", argv[0]);
-     return 3;
+     /* --fake is e.g. for qemu that provides a "fake" serial port
+      * do not complain in this case
+      */
+     if (!(argc > 2 && strcmp(argv[2], "--fake") == 0))
+      {
+        fprintf (stderr, "%s: init failed\n", argv[0]);
+        return 3;
+      }
    }
 
   /* wait - and send the braille ID */
@@ -89,7 +95,7 @@ int main (int argc, char **argv)
 
   wprint (36, 2, "braille star 80 emulator");
   wprint (0, 8,
-	  "up/donw = brl up/down - pgup/pgdown = first/last line - enter = jump to focus");
+	  "up/down = brl up/down - pgup/pgdown = first/last line - enter = jump to focus");
   wprint (0, 9, "a = atributes off - q = quit");
 
   while (processkeys ())
Index: brlemu-0.1/README
===================================================================
--- brlemu-0.1.orig/README
+++ brlemu-0.1/README
@@ -19,3 +19,13 @@ if you see garbage on the brlemu screen
 attribute-underline for braille-displays
 the attribute-underline can't be recoded to usual characters
 
+Using brlemu with Qemu
+----------------------
+You can also use it together with "qemu" on one physical machine - in that
+case you start qemu with all your usual options, and add "-serial pty" to
+them. Qemu will show a line "char device redirected to /dev/pts/11". That's
+the device you need to feed to brlemu. Start it like that:
+
+   brlemu /dev/pts/11 --fake
+
+replacing /dev/pts/11 with what you got from Qemu. Voila - there you go.
