From 392743b95ab0aaff13d5c91eaaab5ae24388384f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Christoph=20B=C3=B6hmwalder?=
 <christoph.boehmwalder@linbit.com>
Date: Tue, 13 Aug 2024 12:29:35 +0200
Subject: [PATCH 12/32] drbd: remove const from function return type

Linux 6.10 enables some compiler flag that errors out with:

error: type qualifiers ignored on function return type
---
 drbd/drbd_receiver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drbd/drbd_receiver.c b/drbd/drbd_receiver.c
index 985fac8e69b3..1b9f80de6f41 100644
--- a/drbd/drbd_receiver.c
+++ b/drbd/drbd_receiver.c
@@ -272,7 +272,7 @@ static const char *drbd_sync_rule_str(enum sync_rule rule)
 	return sync_rule_names[rule];
 }
 
-static const struct sync_descriptor strategy_descriptor(enum sync_strategy strategy)
+static struct sync_descriptor strategy_descriptor(enum sync_strategy strategy)
 {
 	if (strategy < 0 || strategy > ARRAY_SIZE(sync_descriptors)) {
 		WARN_ON(true);
-- 
2.35.3

