From b14af25da949b88fb09339fd84ac97e67aa9109f Mon Sep 17 00:00:00 2001
From: Ben Greiner <code@bnavigator.de>
Date: Wed, 10 Mar 2021 13:22:50 +0100
Subject: [PATCH] Check if JPEGLS is present when testing that codec

---
 tests/test_imagecodecs.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/test_imagecodecs.py b/tests/test_imagecodecs.py
index a057e15..c70a155 100644
--- a/tests/test_imagecodecs.py
+++ b/tests/test_imagecodecs.py
@@ -1535,7 +1535,7 @@ def test_jpeg2k_ycbc():
     assert tuple(decoded[-1, -1]) == (0, 0, 0)
 
 
-@pytest.mark.skipif(not imagecodecs.JPEGXL, reason='jpegls missing')
+@pytest.mark.skipif(not imagecodecs.JPEGLS, reason='jpegls missing')
 @pytest.mark.parametrize('output', ['new', 'out', 'bytearray'])
 def test_jpegls_decode(output):
     """Test JPEG LS decoder with RGBA32 image."""
@@ -1889,7 +1889,7 @@ def encode(data, *args, **kwargs):
                 )
 
     elif codec == 'jpegls':
-        if not imagecodecs.JPEGXL:
+        if not imagecodecs.JPEGLS:
             pytest.skip(f'{codec} missing')
         if itype in ('view', 'graya') or deout == 'view':
             pytest.xfail('jpegls does not support this case')
