--- imagecodecs-2024.1.1.orig/setup.py
+++ imagecodecs-2024.1.1/setup.py
@@ -246,7 +246,7 @@ def customize_build_default(EXTENSIONS,
     del EXTENSIONS['sperr']  # sperr not commonly available
     del EXTENSIONS['zlibng']  # zlib-ng library not commonly available
 
-    if 'arch' not in platform.platform():
+    if True:
         del EXTENSIONS['jpegls']  # CharLS 2.1 library not commonly available
         del EXTENSIONS['jpegxl']  # jpeg-xl library not commonly available
         del EXTENSIONS['brunsli']  # Brunsli library not commonly available
