---
 bash_completion.sh.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- bash_completion.sh.in
+++ bash_completion.sh.in	2017-07-04 07:42:39.171304553 +0000
@@ -1,5 +1,5 @@
 # Check for interactive bash and that we haven't already been sourced.
-if [ -n "${BASH_VERSION-}" -a -n "${PS1-}" -a -z "${BASH_COMPLETION_VERSINFO-}" ]; then
+if [[ -n "${BASH_VERSION-}" && $- = *i* && -z "${BASH_COMPLETION_VERSINFO-}" ]]; then
 
     # Check for recent enough version of bash.
     if [ ${BASH_VERSINFO[0]} -gt 4 ] || \
