diff -Nur 0ad-0.0.25b-alpha/source/scriptinterface/ScriptTypes.h 0ad-0.0.25b-alpha-patched/source/scriptinterface/ScriptTypes.h
--- 0ad-0.0.25b-alpha/source/scriptinterface/ScriptTypes.h	2021-07-27 23:56:58.000000000 +0200
+++ 0ad-0.0.25b-alpha-patched/source/scriptinterface/ScriptTypes.h	2021-12-30 17:18:03.719142315 +0100
@@ -81,6 +81,7 @@
 include paths.
 #endif
 
+#if 0
 #if MOZJS_MINOR_VERSION != 6
 #error Your compiler is trying to use an untested minor version of the \
 SpiderMonkey library. If you are a package maintainer, please make sure \
@@ -93,6 +94,7 @@
 final hash (check trac.wildfiregames.com/wiki/Debugging#Replaymode). \
 For more information check this link: trac.wildfiregames.com/wiki/Debugging#Outofsync
 #endif
+#endif
 
 class ScriptInterface;
 
