diff -Nur deepin-movie-reborn-5.9.14/src/backends/mpv/mpv_proxy.cpp deepin-movie-reborn-5.9.14-new/src/backends/mpv/mpv_proxy.cpp
--- deepin-movie-reborn-5.9.14/src/backends/mpv/mpv_proxy.cpp	2022-03-15 14:10:26.000000000 +0800
+++ deepin-movie-reborn-5.9.14-new/src/backends/mpv/mpv_proxy.cpp	2022-04-14 16:55:15.895534382 +0800
@@ -661,12 +661,6 @@
             // caused by seek or just playing
             break;
 
-        case MPV_EVENT_TRACKS_CHANGED:
-            qInfo() << m_eventName(pEvent->event_id);
-            updatePlayingMovieInfo();
-            emit tracksChanged();
-            break;
-
         case MPV_EVENT_FILE_LOADED: {
             qInfo() << m_eventName(pEvent->event_id);
 
