Index: flask_security-5.6.0/pytest.ini
===================================================================
--- flask_security-5.6.0.orig/pytest.ini
+++ flask_security-5.6.0/pytest.ini
@@ -34,3 +34,4 @@ filterwarnings =
     ignore::DeprecationWarning:pony:0
     ignore:.*'sms' was enabled in SECURITY_US_ENABLED_METHODS;.*:UserWarning:flask_security:0
     ignore:.*'get_token_status' is deprecated.*:DeprecationWarning:flask_security:0
+    ignore::ResourceWarning:
