--- configure.in.orig
+++ configure.in
@@ -13,7 +13,8 @@ dnl
 dnl Initialization.
 AC_INIT
 AC_CONFIG_SRCDIR([pubtext])
-AM_INIT_AUTOMAKE(OpenSP, 1.5.2, no-define)
+AC_CONFIG_MACRO_DIR([m4])
+AM_INIT_AUTOMAKE(OpenSP, 1.5.2, no-define, foreign)
 AM_MAINTAINER_MODE
 AC_PREREQ(2.53)
 AM_CONFIG_HEADER(config.h)
@@ -226,7 +227,9 @@ AC_SUBST(SP_LIBOSP_REV, 0)
 AC_SUBST(SP_LIBOSP_AGE, 0)
 
 dnl Check for programs.
-AC_PROG_CC
+AC_PROG_CC_STDC
+AC_USE_SYSTEM_EXTENSIONS
+AC_SYS_LARGEFILE
 AC_PROG_CPP
 AC_PROG_CXX
 AC_PROG_CXXCPP
