--- aelfred-7.0/net/sf/saxon/aelfred/SAXDriver.java	2022-03-19 13:47:57.460297985 +0100
+++ aelfred-7.0/net/sf/saxon/aelfred/SAXDriver.java	2022-03-19 13:58:48.212214516 +0100
@@ -60,13 +60,14 @@
  * no longer public.  See the package level documentation for more
  * information.
  *
- * <table border="1" width='100%' cellpadding='3' cellspacing='0'>
- * <tr bgcolor='#ccccff'>
- *	<th><font size='+1'>Name</font></th>
- *	<th><font size='+1'>Notes</font></th></tr>
+ * <table border="1" style="width:100%">
+ * <caption></caption>
+ * <tr style="background-color:#ccccff">
+ *	<th>Name</th>
+ *	<th>Notes</th></tr>
  *
- * <tr><td colspan=2><center><em>Features ... URL prefix is
- * <b>http://xml.org/sax/features/</b></em></center></td></tr>
+ * <tr><td colspan=2 style="align:center"><em>Features ... URL prefix is
+ * <b>http://xml.org/sax/features/</b></em></td></tr>
  *
  * <tr><td>(URL)/external-general-entities</td>
  *	<td>Value is fixed at <em>true</em></td></tr>
@@ -82,8 +83,8 @@
  * <tr><td>(URL)/validation</td>
  *	<td>Value is fixed at <em>false</em></td></tr>
  *
- * <tr><td colspan=2><center><em>Handler Properties ... URL prefix is
- * <b>http://xml.org/sax/properties/</b></em></center></td></tr>
+ * <tr><td colspan=2 style="align:center"><em>Handler Properties ... URL prefix is
+ * <b>http://xml.org/sax/properties/</b></em></td></tr>
  *
  * <tr><td>(URL)/declaration-handler</td>
  *	<td>A declaration handler may be provided.  Declaration of general
@@ -186,7 +187,7 @@
 
     /**
      * <b>SAX1, SAX2</b>: Set the entity resolver for this parser.
-     * @param handler The object to receive entity events.
+     * @param resolver The object to receive entity events.
      */
     public void setEntityResolver (EntityResolver resolver)
     {
