Author: Adam Majer <amajer@suse.de>
Date: Dec 20 09:18:49 UTC 2017
Summary: Fix CI unit tests framework for OBS building

Index: node-v22.21.1/test/parallel/test-module-loading-globalpaths.js
===================================================================
--- node-v22.21.1.orig/test/parallel/test-module-loading-globalpaths.js
+++ node-v22.21.1/test/parallel/test-module-loading-globalpaths.js
@@ -11,6 +11,9 @@ const { addLibraryPath } = require('../c
 
 addLibraryPath(process.env);
 
+common.skip('hardcoded global paths');
+return;
+
 if (process.argv[2] === 'child') {
   console.log(require(pkgName).string);
 } else {
Index: node-v22.21.1/test/parallel/test-tls-passphrase.js
===================================================================
--- node-v22.21.1.orig/test/parallel/test-tls-passphrase.js
+++ node-v22.21.1/test/parallel/test-tls-passphrase.js
@@ -223,7 +223,7 @@ server.listen(0, common.mustCall(functio
   }, onSecureConnect());
 })).unref();
 
-const errMessageDecrypt = /bad decrypt/;
+const errMessageDecrypt = /bad (decrypt|password read)/;
 
 // Missing passphrase
 assert.throws(function() {
Index: node-v22.21.1/test/parallel/test-repl-envvars.js
===================================================================
--- node-v22.21.1.orig/test/parallel/test-repl-envvars.js
+++ node-v22.21.1/test/parallel/test-repl-envvars.js
@@ -2,7 +2,9 @@
 
 // Flags: --expose-internals
 
-require('../common');
+const common = require('../common');
+common.skip('Not running test in OBS');
+
 const stream = require('stream');
 const { describe, test } = require('node:test');
 const REPL = require('internal/repl');
Index: node-v22.21.1/Makefile
===================================================================
--- node-v22.21.1.orig/Makefile
+++ node-v22.21.1/Makefile
@@ -399,7 +399,6 @@ ADDONS_HEADERS_PREREQS := tools/install.
 	$(wildcard deps/uv/include/*/*.h) \
 	$(wildcard deps/v8/include/*.h) \
 	$(wildcard deps/v8/include/*/*.h) \
-	deps/zlib/zconf.h deps/zlib/zlib.h \
 	src/node.h src/node_api.h src/js_native_api.h src/js_native_api_types.h \
 	src/node_api_types.h src/node_buffer.h src/node_object_wrap.h \
 	src/node_version.h
@@ -601,6 +600,7 @@ test-ci-js: | clear-stalled ## Build and
 # Related CI jobs: most CI tests, excluding node-test-commit-arm-fanned
 test-ci: LOGLEVEL := info ## Build and test everything (CI).
 test-ci: | clear-stalled bench-addons-build build-addons build-js-native-api-tests build-node-api-tests build-sqlite-tests doc-only
+	strip $(NODE_EXE)
 	out/Release/cctest --gtest_output=xml:out/junit/cctest.xml
 	$(PYTHON) tools/test.py $(PARALLEL_ARGS) -p tap --logfile test.tap \
 		--mode=$(BUILDTYPE_LOWER) --flaky-tests=$(FLAKY_TESTS) \
@@ -795,7 +795,8 @@ apidocs_json = $(addprefix out/,$(apidoc
 apiassets = $(subst api_assets,api/assets,$(addprefix out/,$(wildcard doc/api_assets/*)))
 
 tools/doc/node_modules: tools/doc/package.json
-	@if [ "$(shell $(node_use_openssl))" != "true" ]; then \
+	echo "Skipping tools/doc/node_modules"
+#	@if [ "$(shell $(node_use_openssl))" != "true" ]; then \
 		echo "Skipping tools/doc/node_modules (no crypto)"; \
 	else \
 		cd tools/doc && $(call available-node,$(run-npm-ci)) \
Index: node-v22.21.1/tools/test.py
===================================================================
--- node-v22.21.1.orig/tools/test.py
+++ node-v22.21.1/tools/test.py
@@ -1389,7 +1389,7 @@ def BuildOptions():
   result.add_argument("-s", "--suite", help="A test suite",
       default=[], action="append")
   result.add_argument("-t", "--timeout", help="Timeout in seconds",
-      default=120, type=int)
+      default=1200, type=int)
   result.add_argument("--arch", help='The architecture to run tests for',
       default='none')
   result.add_argument("--snapshot", help="Run the tests with snapshot turned on",
Index: node-v22.21.1/test/parallel/test-crypto-dh.js
===================================================================
--- node-v22.21.1.orig/test/parallel/test-crypto-dh.js
+++ node-v22.21.1/test/parallel/test-crypto-dh.js
@@ -98,7 +98,7 @@ const {
       dh3.computeSecret('');
     }, { message: hasOpenSSL3 && !hasOpenSSL3WithNewErrorMessage ?
       'Unspecified validation error' :
-      'Supplied key is too small' });
+      /(Supplied key is too small|invalid public key)/ });
   }
 }
 
Index: node-v22.21.1/test/parallel/test-dns.js
===================================================================
--- node-v22.21.1.orig/test/parallel/test-dns.js
+++ node-v22.21.1/test/parallel/test-dns.js
@@ -403,7 +403,7 @@ assert.throws(() => {
 
   const server = dgram.createSocket('udp4');
 
-  server.on('message', common.mustCall((msg, { address, port }) => {
+  server.on('message', common.mustCallAtLeast((msg, { address, port }) => {
     const parsed = dnstools.parseDNSPacket(msg);
     const domain = parsed.questions[0].domain;
     assert.strictEqual(domain, 'example.org');
Index: node-v22.21.1/test/wpt/test-webcrypto.js
===================================================================
--- node-v22.21.1.orig/test/wpt/test-webcrypto.js
+++ node-v22.21.1/test/wpt/test-webcrypto.js
@@ -3,7 +3,8 @@
 'use strict';
 
 const common = require('../common');
-if (!common.hasCrypto)
+const os = require('os')
+if (!common.hasCrypto || os.arch() == 's390x')
   common.skip('missing crypto');
 
 const { WPTRunner } = require('../common/wpt');
Index: node-v22.21.1/test/parallel/test-sqlite.js
===================================================================
--- node-v22.21.1.orig/test/parallel/test-sqlite.js
+++ node-v22.21.1/test/parallel/test-sqlite.js
@@ -230,6 +230,7 @@ suite('SQL APIs enabled at build time',
   });
 
   test('fts3 is enabled', (t) => {
+    t.skip(); return;
     const db = new DatabaseSync(':memory:');
     db.exec(`
       CREATE VIRTUAL TABLE t1 USING fts3(content TEXT);
@@ -245,6 +246,7 @@ suite('SQL APIs enabled at build time',
   });
 
   test('fts3 parenthesis', (t) => {
+    t.skip(); return;
     const db = new DatabaseSync(':memory:');
     db.exec(`
       CREATE VIRTUAL TABLE t1 USING fts3(content TEXT);
@@ -313,6 +315,7 @@ suite('SQL APIs enabled at build time',
   });
 
   test('geopoly is enabled', (t) => {
+    t.skip(); return;
     const db = new DatabaseSync(':memory:');
     db.exec(`
       CREATE VIRTUAL TABLE t1 USING geopoly(a,b,c);
