Index: sgrep-1.94a/sample.sgreprc
===================================================================
--- sgrep-1.94a.orig/sample.sgreprc
+++ sgrep-1.94a/sample.sgreprc
@@ -33,7 +33,7 @@ define(UPPER,("A"or"B"or"C"or"D"or"E"or"
 # Lines
 define(LEND,( "\n" or end ))
 define(LSTART,( "\n" or start))
-define(LINES,( (start .. LEND) or (("\n") _. (LEND)) ))
+define(LINE,( (start .. LEND) or (("\n") _. (LEND)) ))
 # Empty lines
 define(ELINE,( LINE in concat(BLANK)))
 
