Index: thermal_daemon-1.8/src/thd_zone.cpp
===================================================================
--- thermal_daemon-1.8.orig/src/thd_zone.cpp	2018-09-12 22:58:08.000000000 +0200
+++ thermal_daemon-1.8/src/thd_zone.cpp	2018-11-02 18:34:30.651800655 +0100
@@ -102,9 +102,13 @@ int cthd_zone::read_user_set_psv_temp()
 	return temp;
 }
 
+#define __STDC_FORMAT_MACROS
+
+#include <inttypes.h>
+
 void cthd_zone::sort_and_update_poll_trip() {
-	thd_log_debug("sort_and_update_poll_trip: trip_points_size =%lu\n",
-			trip_points.size());
+	thd_log_debug("sort_and_update_poll_trip: trip_points_size =%"
+                      PRIuPTR, trip_points.size());
 	if (trip_points.size()) {
 		unsigned int polling_trip = 0;
 
