Index: thermal_daemon-2.5.4.0.git+63b290f/tools/thermal_monitor/ThermalMonitor.pro
===================================================================
--- thermal_daemon-2.5.4.0.git+63b290f.orig/tools/thermal_monitor/ThermalMonitor.pro
+++ thermal_daemon-2.5.4.0.git+63b290f/tools/thermal_monitor/ThermalMonitor.pro
@@ -14,7 +14,7 @@ CONFIG(release): DEFINES += QT_NO_DEBUG_
 TARGET = ThermalMonitor
 TEMPLATE = app
 
-LIBS += -lQCustomPlot
+LIBS += -lqcustomplot
 
 SOURCES += main.cpp\
         mainwindow.cpp \
