---
 src/meson.build |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/meson.build
+++ b/src/meson.build
@@ -123,7 +123,7 @@
 
 # On Linux, we can run the additional integration test;
 # defined here as we would have a circular dependency otherwise.
-if os_backend == 'linux' and gobject_introspection.found()
+if false and os_backend == 'linux' and gobject_introspection.found()
     env = environment()
     env.prepend('top_builddir', join_paths(meson.project_build_root()))
     env.prepend('LD_LIBRARY_PATH', join_paths(meson.project_build_root(), 'libupower-glib'))
