Index: pytest-json-report-1.2.4/setup.py
===================================================================
--- pytest-json-report-1.2.4.orig/setup.py
+++ pytest-json-report-1.2.4/setup.py
@@ -25,7 +25,7 @@ setup(
     url='https://github.com/numirias/pytest-json-report',
     license='MIT',
     install_requires=[
-        'pytest>=4.2.0',
+        'pytest',
         'pytest-metadata',
     ],
     entry_points={
Index: pytest-json-report-1.2.4/tests/test_jsonreport.py
===================================================================
--- pytest-json-report-1.2.4.orig/tests/test_jsonreport.py
+++ pytest-json-report-1.2.4/tests/test_jsonreport.py
@@ -358,7 +358,10 @@ def test_warnings(make_json, num_process
     assert set(warnings[0]) == {
         'category', 'filename', 'lineno', 'message', 'when'
     }
-    assert warnings[0]['category'] == 'PytestCollectionWarning'
+    assert warnings[0]['category'] in (
+        'PytestCollectionWarning',
+        'PytestWarning'
+    )
     assert warnings[0]['filename'].endswith('.py')
     assert warnings[0]['lineno'] == 1
     assert warnings[0]['when'] == 'collect'
@@ -488,15 +491,15 @@ def test_xdist(make_json, match_reports)
 
 def test_bug_31(make_json):
     data = make_json('''
-        import pytest
+        from flaky import flaky
 
         FLAKY_RUNS = 0
 
-        @pytest.mark.flaky
+        @flaky
         def test_flaky_pass():
             assert 1 + 1 == 2
 
-        @pytest.mark.flaky
+        @flaky
         def test_flaky_fail():
             global FLAKY_RUNS
             FLAKY_RUNS += 1
