From a679ca31b8dbd412e5f379b624200e3a96dda0ce Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@suse.de>
Date: Fri, 5 Feb 2021 12:39:40 +0100
Subject: [PATCH] linux: No symver for riscV

---
 src/brickd/symver.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/brickd/symver.h b/src/brickd/symver.h
index e82f240..dbdb68f 100644
--- a/src/brickd/symver.h
+++ b/src/brickd/symver.h
@@ -28,7 +28,7 @@
 
 #ifdef __GLIBC__
 
-#if defined __aarch64__
+#if defined __aarch64__ || defined __riscv
 // do nothing, because arm64 requires glibc >= 2.17 anyway
 #elif defined __arm__
 __asm__(".symver memcpy,memcpy@GLIBC_2.4");
