From fdf3b58fd700dbb57cda48db87eb1cf9197434d0 Mon Sep 17 00:00:00 2001
From: Frank Kunz <mailinglists@kunz-im-inter.net>
Date: Thu, 4 Jul 2024 19:20:43 +0200
Subject: [PATCH 1/2] harden_brickd-resume.service

Signed-off-by: Frank Kunz <mailinglists@kunz-im-inter.net>
---
 .../lib/systemd/system/brickd-resume.service          | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/build_data/linux/installer/lib/systemd/system/brickd-resume.service b/src/build_data/linux/installer/lib/systemd/system/brickd-resume.service
index 6e142c9..58922dc 100644
--- a/src/build_data/linux/installer/lib/systemd/system/brickd-resume.service
+++ b/src/build_data/linux/installer/lib/systemd/system/brickd-resume.service
@@ -4,6 +4,17 @@ After=suspend.target
 Requisite=brickd.service
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+ProtectHostname=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions
 User=root
 Type=oneshot
 ExecStart=/usr/bin/pkill -F /run/brickd.pid -L -USR1
-- 
2.45.2

