--- pam-python-1.0.6/src/Makefile  2016-08-27 14:27:10.000000000 +0200
+++ pam-python-1.0.6/src/Makefile  2017-10-05 14:06:51.377228092 +0200
@@ -1,6 +1,6 @@
 all:	ctest pam_python.so test-pam_python.pam
 
-WARNINGS=-Wall -Wextra -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wsign-compare -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Werror
+WARNINGS=-Wall -Wextra -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wsign-compare -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations
 #WARNINGS=-Wunreachable-code 	# Gcc 4.1 .. 4.4 are too buggy to make this useful
 
 LIBDIR ?= /lib/security
