--- azurelinuxagent/common/osutil/suse.py.orig
+++ azurelinuxagent/common/osutil/suse.py
@@ -82,6 +82,11 @@ class SUSEOSUtil(SUSE11OSUtil):
         super(SUSEOSUtil, self).__init__()
         self.dhclient_name = 'wickedd-dhcp4'
 
+    def set_hostname(self, hostname):
+        shellutil.run(
+            "hostnamectl set-hostname {0}".format(hostname), chk_err=False
+        )
+
     def stop_dhcp_service(self):
         cmd = "systemctl stop {0}".format(self.dhclient_name)
         return shellutil.run(cmd, chk_err=False)
