Index: rmw-0.9.2/test/test_restore.sh
===================================================================
--- rmw-0.9.2.orig/test/test_restore.sh
+++ rmw-0.9.2/test/test_restore.sh
@@ -108,8 +108,25 @@ done
 cmp_substr "$(${RMW_TEST_CMD_STRING} -z ${PRIMARY_WASTE_DIR}/files/. && exit 1)" \
   "refusing to process"
 
-if [ -n "${TERM}" ] && [ "${TERM}" != "dumb" ]; then
+# I don't want to force anyone to install Xvfb for this single test
+# so I'll only run it if it's already installed
+if [ -n "$(command -v Xvfb)" ]; then
+  # Start Xvfb on display :99
+  Xvfb :99 &
+  XVFB_PID=$!
+
+  # Set the DISPLAY environment variable to use the virtual display
+  OLD_DISPLAY="$DISPLAY"
+  export DISPLAY=:99
+
+  # No visual test here, but when used with llvm sanitize or valgrind,
+  # the chances of spotting any memory leaks are pretty good.
   echo q | ${RMW_TEST_CMD_STRING} -s
+
+  # Stop Xvfb
+  kill $XVFB_PID
+
+  DISPLAY="$OLD_DISPLAY"
 fi
 
 # This test will only work on Andy's workstation.
