diff --git a/Keyboard/ckbcomp b/Keyboard/ckbcomp
index b0df60695362ae2930c1f8f3a9e2035b416c4288..47ec9224bde6f7c1416c07095de96819d93de803 100755
--- a/Keyboard/ckbcomp
+++ b/Keyboard/ckbcomp
@@ -2581,7 +2581,7 @@ if ($freebsd) {
          'any' => 'NoSymbol', # Is this recognised by X ?
          'VoidSymbol' => 'VoidSymbol',
          'voidsymbol' => 'VoidSymbol', # Is this recognised by X ?
-         'ISO_Left_Tab' => 'Tab',
+         'ISO_Left_Tab' => 'Meta_Tab',
          'Clear' => 'VoidSymbol',
          'Pause' => 'Pause',
          'Scroll_Lock' => 'Scroll_Lock',
diff --git a/Keyboard/compose_translator b/Keyboard/compose_translator
index d478def19335cd6637fc63812fd2c111dff67983..3af492f771c16e220d61687fe356d780526d531d 100755
--- a/Keyboard/compose_translator
+++ b/Keyboard/compose_translator
@@ -1827,7 +1827,7 @@ my %xkbsym_table = (
      'any' => 'NoSymbol', # Is this recognised by X ?
      'VoidSymbol' => 'VoidSymbol',
      'voidsymbol' => 'VoidSymbol', # Is this recognised by X ?
-     'ISO_Left_Tab' => 'Tab',
+     'ISO_Left_Tab' => 'Meta_Tab',
      'Clear' => 'VoidSymbol',
      'Pause' => 'Pause',
      'Scroll_Lock' => 'Scroll_Lock',
