--- conf/gpm-syn.conf
+++ conf/gpm-syn.conf
@@ -1,49 +1,107 @@
+/* generated from src/synaptics.c, gpm-1.20.1 */
 /* enabling configuration parameters */
-[edge_motion_enabled]        TRUE
-[edge_motion_speed_enabled]  TRUE
-[corner_taps_enabled]        TRUE
-[tap_gesture_enabled]        TRUE
-[pressure_speed_enabled]     TRUE
-[tossing_enabled]            TRUE
-[does_toss_use_static_speed] TRUE
-/* pressure induced speed related configuration parameters */
-[low_pressure]               60
-[speed_up_pressure]          60
-[pressure_factor]            0.10
-[standard_speed_factor]      0.10
-/* toss/catch related parameters */
-[min_toss_time]              100
-[max_toss_time]              300
-[prevent_toss_time]          300
-[min_toss_dist]              2
-[static_toss_speed]          70
-[toss_speed_factor]          0.5
-/* edge motion related configuration parameters */
-[edge_speed]                 20
-/* corner tap actions */
-[upper_left_action]          0 (none)
-[upper_right_action]         2 (middle)
-[lower_left_action]          0 (none)
-[lower_right_action]         3 (right)
-/* wmode parameters */
-[wmode_enabled]              TRUE
-[drag_lock_enabled]	     FALSE
-[finger_threshold]           30
-[tap_lower_limit]            5
-[tap_upper_limit]	     200
-[tap_range]		     100
-[tap_interval]		     200
-[pads_tap_interval]	     8
-[multiple_tap_delay]	     30
-[palm_detect_enabled]	     TRUE
-[palm_detect_level]	     12
-[multi_finger_tap_enable]    TRUE
-[multi_finger_stop_enabled]  TRUE
-[multi_finger_stop_delay]    8
-  /* Additional options */   
-[touch_pad_enabled]	     TRUE
-[stick_enabled]		     TRUE
-[scrolling_enabled]	     TRUE
-[scrolling_edge]	     2
-[auto_scrolling_enabled]     TRUE
-
+[edge_motion_enabled]		TRUE
+[edge_motion_speed_enabled]	TRUE
+[corner_taps_enabled]		TRUE
+[tap_gesture_enabled]		TRUE
+[pressure_speed_enabled]	TRUE
+[tossing_enabled]		TRUE
+[does_toss_use_static_speed]	TRUE
+[tap_hold_edge_motion_enabled]	TRUE
+/* pressure induced speed related configuration parameters */		     
+[low_pressure]			60
+[speed_up_pressure]		60
+[pressure_factor]		0.05
+[standard_speed_factor]		0.08
+/* toss/catch related parameters */		 			     
+[min_toss_time]			100
+[max_toss_time]			300
+[prevent_toss_time]		300
+[min_toss_dist]			2
+[static_toss_speed]		70
+[toss_speed_factor]		0.5
+/* edge motion related configuration parameters */			     
+[x_min_center]			1632
+[x_max_center]			5312
+[y_min_center]			1408
+[y_max_center]			4108
+[edge_speed]			20
+/* use wmode */				 			     
+[wmode_enabled]			TRUE
+[drag_lock_enabled]		TRUE
+[finger_threshold]		30
+[tap_lower_limit]		5
+[tap_upper_limit]		200
+[tap_range]			100
+[tap_interval]			200
+[multiple_tap_delay]		30
+[pads_tap_interval]		8
+/* Additional wmode parameters */				 			     
+[palm_detect_enabled]		TRUE
+[palm_detect_level]		12
+[multi_finger_tap_enable]	TRUE
+[multi_finger_stop_enabled]	TRUE
+[multi_finger_stop_delay]	8
+[fake_finger_layer_enabled]	1
+/* Additional options*/
+/* Values for [scrolling_edge]:
+	LEFT_EDGE           0x01
+	RIGHT_EDGE          0x02
+	TOP_EDGE            0x04
+	BOTTOM_EDGE         0x08
+	UPPER_LEFT_CORNER   (LEFT_EDGE  | TOP_EDGE)
+	LOWER_LEFT_CORNER   (LEFT_EDGE  | BOTTOM_EDGE)
+	UPPER_RIGHT_CORNER  (RIGHT_EDGE | TOP_EDGE)
+	LOWER_RIGHT_CORNER  (RIGHT_EDGE | BOTTOM_EDGE) */
+[touchpad_enabled]		TRUE
+[stick_enabled]			TRUE
+[stick_pressure_enabled]	FALSE
+[four_way_button_enabled]	TRUE
+[four_way_button_is_mouse]	TRUE
+[scrolling_enabled]		TRUE
+[auto_scrolling_enabled]	TRUE
+[scrolling_edge]		0x02
+[scrolling_speed]		10
+[scrolling_button_factor]	0.5
+[auto_scrolling_factor]		2.0
+/* Action id's:
+	No_Action			= 0
+	Left_Button_Action		= 1
+	Middle_Button_Action		= 2
+	Right_Button_Action		= 3
+	Fourth_Button_Action		= 4
+	Up_Button_Action		= 5
+	Down_Button_Action		= 6
+	Turn_On_Off_Action		= 7
+	Debug_On_Off_Action		= 8
+	Reset_Touchpad_Action		= 9
+	Toggle_Four_Way_Button_Action	= 10
+	Toggle_Stick_Pressure_Action	= 11
+	Toggle_Scrolling_Action		= 12
+	Left_Double_Click_Action	= 13
+*/
+/* corner tap actions */			 			     
+[upper_left_action]		0
+[lower_left_action]		0	
+[upper_right_action]		2
+[lower_right_action]		3
+/* no. of fingers tap actions */
+[one_finger_tap_action]		1
+[two_fingers_tap_action]	3
+[three_fingers_tap_action]	2
+/* normal button actions */
+[left_button_action]		1
+[middle_button_action]		2
+[right_button_action]		3
+[fourth_button_action]		4
+[up_button_action]		5
+[down_button_action]		6
+/* 4 way button actions */
+[four_way_left_button_action]	2
+[four_way_right_button_action]	4
+[four_way_up_button_action]	5
+[four_way_down_button_action]	6
+/* Synaptic Stick (passthrugh,stick) actions */
+[stick_left_button_action]	1
+[stick_middle_button_action]	2
+[stick_right_button_action]	3
