From 26bc48c9f3bec3671fcc39a60e2381ba4214c638 Mon Sep 17 00:00:00 2001
From: Benjamin Greiner <code@bnavigator.de>
Date: Fri, 10 Apr 2020 00:46:03 +0200
Subject: [PATCH] fix references test

---
 test/plugins/test_references.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/plugins/test_references.py b/test/plugins/test_references.py
index 6caab4c4..7e7cbe75 100644
--- a/test/plugins/test_references.py
+++ b/test/plugins/test_references.py
@@ -73,7 +73,7 @@ def test_references_builtin(tmp_workspace):  # pylint: disable=redefined-outer-n
     doc2 = Document(doc2_uri)
 
     refs = pyls_references(doc2, position)
-    assert len(refs) == 1
+    assert len(refs) >= 1
 
     assert refs[0]['range']['start'] == {'line': 4, 'character': 7}
     assert refs[0]['range']['end'] == {'line': 4, 'character': 19}
