Index: fontforge/start.c
===================================================================
--- fontforge/start.c.orig	2015-04-30 22:29:11.000000000 +0200
+++ fontforge/start.c	2015-05-11 12:45:19.923253423 +0200
@@ -133,9 +133,14 @@
 }
 
 void doversion(const char *source_version_str) {
+    /* not to be upstreamed ! */
     if ( source_version_str!=NULL )
+    {
 	printf( "fontforge %s\n", source_version_str );
+ 	printf( "libfontforge %s\n", source_version_str );
+   } else {
     printf( "libfontforge %d\n",
 	    FONTFORGE_VERSIONDATE_RAW );
+   }
 exit(0);
 }
Index: fontforgeexe/startnoui.c
===================================================================
--- fontforgeexe/startnoui.c.orig	2015-04-30 22:29:11.000000000 +0200
+++ fontforgeexe/startnoui.c	2015-05-11 11:54:33.442771227 +0200
@@ -116,7 +116,7 @@
 	else if ( strcmp(pt,"-help")==0 )
 	    doscripthelp();
 	else if ( strcmp(pt,"-version")==0 || strcmp(pt,"-v")==0 || strcmp(pt,"-V")==0 )
-	    doversion(FONTFORGE_MODTIME_STR);
+	    doversion(source_version_str);
     }
 #  if defined(_NO_PYTHON)
     ProcessNativeScript(argc, argv,stdin);
