--- skinlf-6.7/src/com/l2fprod/gui/nativeskin/win32/Win32NativeSkin.java	2017-09-20 09:40:46.096375168 +0200
+++ skinlf-6.7/src/com/l2fprod/gui/nativeskin/win32/Win32NativeSkin.java	2017-09-20 10:07:02.226568456 +0200
@@ -200,7 +200,7 @@
    * @return        The HWND value
    */
   private static int getHWND(Window window) {
-    if (window.getPeer() == null) {
+    if (!window.isDisplayable()) {
       window.addNotify();
     }
 
@@ -217,7 +217,7 @@
       drawingSurfaceInfo =
         Class.forName("sun.awt.DrawingSurface").
         getMethod("getDrawingSurfaceInfo", null).
-        invoke(window.getPeer(), null);
+        invoke(window.getClass().getMethod("getPeer").invoke(window), null);
 
       if (null != drawingSurfaceInfo) {
         // drawingSurfaceInfo.lock();
