Index: gedit-3.27.92/plugins/externaltools/data/send-to-fpaste.tool.in
===================================================================
--- gedit-3.27.92.orig/plugins/externaltools/data/send-to-fpaste.tool.in	2018-02-01 14:58:38.000000000 +0100
+++ gedit-3.27.92/plugins/externaltools/data/send-to-fpaste.tool.in	2018-03-06 22:46:20.502284840 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 
 import os, urllib, json, sys, urllib.request
 from gi.repository import Gtk, Gdk
