--- orig-gedit-3.31.3/data/org.gnome.gedit.desktop.in.in	2018-10-21 15:48:49.000000000 +0200
+++ gedit-3.31.3/data/org.gnome.gedit.desktop.in.in	2019-01-03 14:24:10.125418509 +0100
@@ -5,7 +5,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-MimeType=text/plain;
+MimeType=application/x-awk;application/x-csh;application/x-perl;application/x-php;application/x-shellscript;text/plain;text/x-c;text/x-c++;text/x-c++hdr;text/x-chdr;text/x-c++src;text/x-csrc;text/x-fortran;text/x-java;text/x-log;text/x-makefile;text/x-moc;text/x-pascal;text/x-patch;text/x-python;text/x-readme;text/x-scheme;text/x-tcl;text/x-tex;
 # TRANSLATORS: Do NOT translate or transliterate this text!
 #              This is an icon file name.
 Icon=org.gnome.gedit

