--- GNUmakefile.am.orig
+++ GNUmakefile.am
@@ -1,3 +1,5 @@
+AUTOMAKE_OPTIONS=foreign
+
 DIST_SUBDIRS = xml help wingdingfont patterns
 
 SUBDIRS = xml help wingdingfont patterns .
@@ -8,7 +10,7 @@
 
 CFLAGS = @CFLAGS@ @ANSI_CFLAGS@
 
-CPPFLAGS = @CPPFLAGS@ @DEPS_CFLAGS@ @OPT_CFLAGS@
+CPPFLAGS = $(filter-out -I/usr/include,@CPPFLAGS@ @DEPS_CFLAGS@ @OPT_CFLAGS@)
 
 WVLIBS = @DEPS_LIBS@ @OPT_LIBS@ -lm
 
