--- scheduler/client.c.orig	2024-06-18 13:11:05.000000000 +0200
+++ scheduler/client.c	2024-07-08 15:44:58.336923944 +0200
@@ -1145,8 +1145,8 @@ cupsdReadClient(cupsd_client_t *con)	/*
 		  {
 		    unsigned int i = 0;	// Array index
 
-		    for (char *start = con->uri + 9; *start && *start != '?' && i < sizeof(name);)
-		      name[i++] = *start++;
+		    for (ptr = con->uri + 9; *ptr && *ptr != '?' && i < sizeof(name);)
+		      name[i++] = *ptr++;
 
 		    name[i] = '\0';
 
@@ -1185,8 +1185,8 @@ cupsdReadClient(cupsd_client_t *con)	/*
 		  {
 		    unsigned int i = 0;	// Array index
 
-		    for (char *start = con->uri + 10; *start && *start != '?' && i < sizeof(name);)
-		      name[i++] = *start++;
+    		    for (ptr = con->uri + 10; *ptr && *ptr != '?' && i < sizeof(name);)
+		      name[i++] = *ptr++;
 
 		    name[i] = '\0';
 
