diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7311294..45d9840 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,7 +40,7 @@ if(CMAKE_BUILD_TYPE STREQUAL "Debug")
 endif()
 
 set_target_properties(freesrp PROPERTIES VERSION ${VERSION} SOVERSION ${VERSION_MAJOR})
-target_link_libraries(freesrp ${LIBUSB_1_LIBRARIES})
+target_link_libraries(freesrp ${LIBUSB_1_LIBRARIES} pthread)
 
 # Examples
 include_directories(${PROJECT_SOURCE_DIR}/include)
