Index: Stockfish-sf_12/src/misc.cpp
===================================================================
--- Stockfish-sf_12.orig/src/misc.cpp
+++ Stockfish-sf_12/src/misc.cpp
@@ -140,18 +140,10 @@ public:
 
 const string engine_info(bool to_uci) {
 
-  const string months("Jan Feb Mar Apr May Jun Jul Aug Sep Oct Nov Dec");
-  string month, day, year;
-  stringstream ss, date(__DATE__); // From compiler, format is "Sep 21 2008"
+  stringstream ss;
 
   ss << "Stockfish " << Version << setfill('0');
 
-  if (Version.empty())
-  {
-      date >> month >> day >> year;
-      ss << setw(2) << day << setw(2) << (1 + months.find(month) / 4) << year.substr(2);
-  }
-
   ss << (to_uci  ? "\nid author ": " by ")
      << "the Stockfish developers (see AUTHORS file)";
 
