From 086ad336ad1e29c6260251e287de8994397a7dc7 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Tue, 15 Nov 2022 16:29:46 +0100
Subject: [PATCH] audioconvert: redo setup when format changes

---
 spa/plugins/audioconvert/audioconvert.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/spa/plugins/audioconvert/audioconvert.c b/spa/plugins/audioconvert/audioconvert.c
index 7fe62228f..ba2ea4178 100644
--- a/spa/plugins/audioconvert/audioconvert.c
+++ b/spa/plugins/audioconvert/audioconvert.c
@@ -1989,6 +1989,7 @@ static int port_set_format(void *object,
 			}
 			this->dir[direction].format = info;
 			this->dir[direction].have_format = true;
+			this->setup = false;
 		}
 		port->format = info;
 		port->have_format = true;
-- 
GitLab

