Index: pipewire-0.3.8/spa/plugins/alsa/meson.build
===================================================================
--- pipewire-0.3.8.orig/spa/plugins/alsa/meson.build
+++ pipewire-0.3.8/spa/plugins/alsa/meson.build
@@ -1,5 +1,5 @@
 subdir('acp')
-subdir('mixer')
+#subdir('mixer')
 
 spa_alsa_sources = ['alsa.c',
                 'alsa-udev.c',
@@ -26,8 +26,8 @@ alsa_udevrules = [
   install : true,
 )
 
-if libudev_dep.found()
-  install_data(alsa_udevrules,
-    install_dir : udevrulesdir,
-  )
-endif
+#if libudev_dep.found()
+#  install_data(alsa_udevrules,
+#    install_dir : udevrulesdir,
+#  )
+#endif
