diff -Nru azure-mgmt-sql-0.11.0.orig/setup.py azure-mgmt-sql-0.11.0/setup.py
--- azure-mgmt-sql-0.11.0.orig/setup.py	2019-04-01 11:46:00.221990345 +0200
+++ azure-mgmt-sql-0.11.0/setup.py	2019-04-01 11:46:17.182179738 +0200
@@ -81,7 +81,4 @@
         'msrestazure>=0.4.32,<2.0.0',
         'azure-common>=1.1',
     ],
-    extras_require={
-        ":python_version<'3.0'": ['azure-mgmt-nspkg'],
-    }
 )
