Index: dask_image-2025.11.0/tests/test_dask_image/test_ndmeasure/test_core.py
===================================================================
--- dask_image-2025.11.0.orig/tests/test_dask_image/test_ndmeasure/test_core.py
+++ dask_image-2025.11.0/tests/test_dask_image/test_ndmeasure/test_core.py
@@ -142,7 +142,7 @@ def test_measure_props(funcname, shape,
     if (
         funcname == "median" and
         ind is not None and
-        not np.in1d(np.atleast_1d(ind), lbls).all()
+        not np.isin(np.atleast_1d(ind), lbls).all()
     ):
         pytest.skip("SciPy's `median` mishandles missing labels.")
 
