diff -u -r xf86-input-wacom-0.37.0.orig/conf/70-wacom.conf xf86-input-wacom-0.37.0/conf/70-wacom.conf
--- xf86-input-wacom-0.37.0.orig/conf/70-wacom.conf	2020-07-16 12:25:53.259337000 +0200
+++ xf86-input-wacom-0.37.0/conf/70-wacom.conf	2020-07-16 12:26:52.602097000 +0200
@@ -38,13 +38,13 @@
 	Driver "wacom"
 EndSection
 
-Section "InputClass"
-	Identifier "Wacom touchscreen class"
-	MatchProduct "Wacom|WACOM|PTK-540WL|ISD-V4"
-	MatchDevicePath "/dev/input/event*"
-	MatchIsTouchscreen "true"
-	Driver "wacom"
-EndSection
+#Section "InputClass"
+#	Identifier "Wacom touchscreen class"
+#	MatchProduct "Wacom|WACOM|PTK-540WL|ISD-V4"
+#	MatchDevicePath "/dev/input/event*"
+#	MatchIsTouchscreen "true"
+#	Driver "wacom"
+#EndSection
 
 Section "InputClass"
 	Identifier "Wacom touchpad class"
