Index: grpc-1.70.1/src/ruby/ext/grpc/extconf.rb
===================================================================
--- grpc-1.70.1.orig/src/ruby/ext/grpc/extconf.rb
+++ grpc-1.70.1/src/ruby/ext/grpc/extconf.rb
@@ -209,6 +209,7 @@ $CFLAGS << ' -std=c11 '
 $CFLAGS << ' -Wall '
 $CFLAGS << ' -Wextra '
 $CFLAGS << ' -pedantic '
+$CFLAGS << ' -Wno-error=incompatible-pointer-types '
 $LDFLAGS << ' -lgrpc '
 
 output = File.join('grpc', 'grpc_c')
