Index: restic-0.18.0/build.go
===================================================================
--- restic-0.18.0.orig/build.go
+++ restic-0.18.0/build.go
@@ -58,7 +58,7 @@ var config = Config{
 	Main:      "./cmd/restic",             // package name for the main package
 	// disable_grpc_modules is necessary to reduce the binary size since cloud.google.com/go/storage v1.44.0
 	// see https://github.com/googleapis/google-cloud-go/issues/11448
-	DefaultBuildTags: []string{"selfupdate", "disable_grpc_modules"}, // specify build tags which are always used
+	DefaultBuildTags: []string{"disable_grpc_modules"}, // specify build tags which are always used
 	Tests:            []string{"./..."},                              // tests to run
 	MinVersion:       GoVersion{Major: 1, Minor: 23, Patch: 0},       // minimum Go version supported
 }
