--- old/sys/simsys.cc   2022-01-11 11:20:33.991101481 +0100
+++ new/sys/simsys.cc   2022-01-11 11:01:19.637291267 +0100
@@ -395,7 +395,7 @@
 #elif defined __ANDROID__
 	tstrncpy(buffer,SDL_GetPrefPath("Simutrans Team","simutrans"),lengthof(buffer));
 #else
-	sprintf(buffer, "%s/simutrans", getenv("HOME"));
+	sprintf(buffer, "%s/.simutrans", getenv("HOME"));
 #endif
 
 	// create directory and subdirectories
