diff --git a/fix_includes.py b/fix_includes.py
index 6bfdf77..d9ad5b4 100755
--- a/fix_includes.py
+++ b/fix_includes.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 ##===--- fix_includes.py - rewrite source files based on iwyu output ------===##
 #
diff --git a/iwyu_tool.py b/iwyu_tool.py
index 19de45b..8faefd6 100755
--- a/iwyu_tool.py
+++ b/iwyu_tool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 ##===--- iwyu_tool.py -----------------------------------------------------===##
 #
