Index: jikes-1.22/src/stream.cpp
===================================================================
--- src/stream.cpp
+++ src/stream.cpp
@@ -271,13 +271,20 @@ wchar_t Stream::DecodeNextCharacter()
     size_t chl = sizeof(wchar_t);
     size_t srcl = 1;
 
+    char* c_chp = (char *) chp;
+#  ifdef HAVE_ERROR_CALL_ICONV_CONST
+    char* c_source_ptr = (char *) source_ptr;
+#  endif // HAVE_ERROR_CALL_ICONV_CONST
  try_it_again:
     size_t n = iconv(_decoder,
 #  ifdef HAVE_ERROR_CALL_ICONV_CONST
-                     (char**)
-#  endif // HAVE_ERROR_CALL_ICONV_CONST
-                     &source_ptr, &srcl,
-                     (char**) &chp, &chl);
+                     &c_source_ptr,
+#  else // HAVE_ERROR_CALL_ICONV_CONST
+                     &source_ptr,
+#  endif
+                     &srcl,
+                     &c_chp, 
+                     &chl);
 
     if (n == (size_t) -1)
     {
