--- build/build.c.orig	2024-10-07 09:35:46.000000000 +0000
+++ build/build.c	2024-12-16 09:50:14.728138966 +0000
@@ -385,8 +385,10 @@ static int buildSpec(rpmts ts, BTA_t bui
 	    setenv("SOURCE_DATE_EPOCH", sdestr, 0);
 	    rpmtdFreeData(&td);
 	} else {
-	    rpmlog(RPMLOG_WARNING, _("%%source_date_epoch_from_changelog is set, but "
+	    rpmlog(RPMLOG_ERR, _("%%source_date_epoch_from_changelog is set, but "
 	        "%%changelog has no entries to take a date from\n"));
+	    rc = RPMRC_FAIL;
+	    goto exit;
 	}
     }
 
