--- ./scripts/brp-compress.orig	2020-09-30 13:09:28.846434576 +0000
+++ ./scripts/brp-compress	2020-09-30 13:09:34.514422459 +0000
@@ -51,6 +51,7 @@ do
     find $d -type f ! -name dir | while read f
     do
         [ -f "$f" ] || continue
+	case $(file "$f") in *"image data"*) continue;; esac
 
 	case "$f" in
 	 *.gz|*.Z)    gunzip  -f "$f" || check_for_hard_link $d "$f"; b=`echo $f | sed -e 's/\.\(gz\|Z\)$//'`;;
