--- scripts/brp-strip-comment-note.orig	2024-10-07 09:35:46.000000000 +0000
+++ scripts/brp-strip-comment-note	2024-12-16 09:15:41.572425334 +0000
@@ -15,7 +15,7 @@ esac
 
 # Strip .comment and .note sections (the latter only if it is not allocated)
 # for already stripped elf files in the build root
-for f in `find "$RPM_BUILD_ROOT" -type f \( -perm -0100 -or -perm -0010 -or -perm -0001 \) -print0 | xargs -0 -r -P$NCPUS -n32 sh -c "file \"\\$@\" | grep -v \"^${RPM_BUILD_ROOT}/\?usr/lib/debug\" | sed -n -e 's/^\(.*\):[ 	]*ELF.*, stripped.*/\1/p'" ARG0`; do
+for f in `find "$RPM_BUILD_ROOT" -type f \( -perm -0100 -or -perm -0010 -or -perm -0001 \) -print0 | xargs -0 -r -P$NCPUS -n32 sh -c "file \"\\$@\" | grep -v \"^${RPM_BUILD_ROOT}/\?usr/lib/debug\" | grep -v ' shared object,' | grep -v '/lib/modules/' | sed -n -e 's/^\(.*\):[ 	]*ELF.*, stripped.*/\1/p'" ARG0`; do
 	note="-R .note"
 	if $OBJDUMP -h $f | grep '^[ 	]*[0-9]*[ 	]*.note[ 	]' -A 1 | \
 		grep ALLOC >/dev/null; then
--- scripts/brp-strip.orig	2024-10-07 09:35:46.000000000 +0000
+++ scripts/brp-strip	2024-12-16 09:15:41.572425334 +0000
@@ -37,6 +37,7 @@ strip_elf_binaries()
     ! -name "*.py" ! -name "*.js" ! -name "*.rb" \
     ! -name "*.go" -links "${nlinks}" -print0 | \
     xargs -0 -r -P${nprocs} -n${MAX_ARGS} sh -c "file \"\$@\" | \
+    grep -v ' shared object,' | grep -v '/lib/modules/ | \
     sed -n -e 's/^\(.*\):[ 	]*ELF.*, not stripped.*/\1/p' | \
     grep -v 'no machine' | \
     xargs -I\{\} $STRIP -g \{\}" ARG0
