--- scripts/brp-strip-comment-note.orig	2022-07-01 09:02:17.586292400 +0000
+++ scripts/brp-strip-comment-note	2022-07-21 14:03:57.119809772 +0000
@@ -16,6 +16,8 @@ esac
 # for already stripped elf files in the build root
 for f in `find "$RPM_BUILD_ROOT" -type f \( -perm -0100 -or -perm -0010 -or -perm -0001 \) -exec file {} \; | \
         grep -v "^${RPM_BUILD_ROOT}/\?usr/lib/debug"  | \
+	grep -v ' shared object,' | \
+	grep -v '/lib/modules/' | \
 	sed -n -e 's/^\(.*\):[ 	]*ELF.*, stripped.*/\1/p'`; do
 	note="-R .note"
 	if $OBJDUMP -h $f | grep '^[ 	]*[0-9]*[ 	]*.note[ 	]' -A 1 | \
--- scripts/brp-strip.orig	2022-07-01 10:31:45.997506545 +0000
+++ scripts/brp-strip	2022-07-21 14:06:12.155573458 +0000
@@ -14,4 +14,4 @@ esac
 
 # Strip ELF binaries
 find "$RPM_BUILD_ROOT" -type f \! -regex "${RPM_BUILD_ROOT}/*usr/lib/debug.*" \! -name "*.go" -print0 | \
-    xargs -0 -r -P$NCPUS -n32 sh -c "file \"\$@\" | sed -n -e 's/^\(.*\):[ 	]*ELF.*, not stripped.*/\1/p' | grep -v 'no machine' | xargs -I\{\} $STRIP -g \{\}" ARG0
+    xargs -0 -r -P$NCPUS -n32 sh -c "file \"\$@\" | grep -v ' shared object,' | grep -v '/lib/modules/ | sed -n -e 's/^\(.*\):[ 	]*ELF.*, not stripped.*/\1/p' | grep -v 'no machine' | xargs -I\{\} $STRIP -g \{\}" ARG0
