--- ./scripts/brp-compress.orig	2019-10-02 09:59:40.403995534 +0000
+++ ./scripts/brp-compress	2019-10-02 10:03:27.075537047 +0000
@@ -5,6 +5,9 @@ if [ -z "$RPM_BUILD_ROOT" ] || [ "$RPM_B
 	exit 0
 fi
 
+LC_ALL=
+LANG=
+LC_TIME=POSIX
 PREFIX=${1:-/usr}
 
 cd "$RPM_BUILD_ROOT"
@@ -13,6 +16,29 @@ cd "$RPM_BUILD_ROOT"
 COMPRESS=${COMPRESS:-gzip -9 -n}
 COMPRESS_EXT=${COMPRESS_EXT:-.gz}
 
+check_for_hard_link()
+{
+    dir=$1
+    b=$2
+
+    inode=`ls -i $b | awk '{ print $1 }'`
+    others=`find $dir -type f -inum $inode`
+    for afile in $others ; do
+	[ "$afile" != "$b" ] && rm -f "$afile"
+    done
+
+    case $b in
+	*.Z|*.gz)     gunzip  $b ;;
+	*.bz2)        bunzip2 $b ;;
+	*.xz|*.lzma)  unxz    $b ;;
+    esac
+
+    type=${b##*.}
+    for afile in $others ; do
+	[ "${afile%.$type}" != "${b%.$type}" ] && ln ${b%.$type} ${afile%.$type}
+    done
+}
+
 for d in .${PREFIX}/man/man* .${PREFIX}/man/*/man* .${PREFIX}/info \
 	.${PREFIX}/share/man/man* .${PREFIX}/share/man/*/man* \
 	.${PREFIX}/share/info .${PREFIX}/kerberos/man \
@@ -21,39 +47,39 @@ for d in .${PREFIX}/man/man* .${PREFIX}/
 	.${PREFIX}/share/fish/man/man*
 do
     [ -d $d ] || continue
-    for f in `find $d -type f ! -name dir`
+    find $d -type f ! -name dir | while read f
     do
         [ -f "$f" ] || continue
 
 	case "$f" in
-	 *.gz|*.Z)    gunzip  -f $f; b=`echo $f | sed -e 's/\.\(gz\|Z\)$//'`;;
-	 *.bz2)       bunzip2 -f $f; b=`echo $f | sed -e 's/\.bz2$//'`;;
-	 *.xz|*.lzma) unxz    -f $f; b=`echo $f | sed -e 's/\.\(xz\|lzma\)$//'`;;
+	 *.gz|*.Z)    gunzip  -f "$f" || check_for_hard_link $d "$f"; b=`echo $f | sed -e 's/\.\(gz\|Z\)$//'`;;
+	 *.bz2)       bunzip2 -f "$f" || check_for_hard_link $d "$f"; b=`echo $f | sed -e 's/\.bz2$//'`;;
+	 *.xz|*.lzma) unxz    -f "$f" || check_for_hard_link $d "$f"; b=`echo $f | sed -e 's/\.\(xz\|lzma\)$//'`;;
 	 *) b=$f;;
 	esac
 
-	$COMPRESS $b </dev/null 2>/dev/null || {
+	$COMPRESS "$b" </dev/null 2>/dev/null || {
 	    inode=`ls -i $b | awk '{ print $1 }'`
 	    others=`find $d -type f -inum $inode`
 	    if [ -n "$others" ]; then
 		for afile in $others ; do
-		    [ "$afile" != "$b" ] && rm -f $afile
+		    [ "$afile" != "$b" ] && rm -f "$afile"
 		done
-		$COMPRESS -f $b
+		$COMPRESS -f "$b"
 		for afile in $others ; do
-		    [ "$afile" != "$b" ] && ln $b$COMPRESS_EXT $afile$COMPRESS_EXT
+		    [ "$afile" != "$b" ] && ln "$b$COMPRESS_EXT" "$afile$COMPRESS_EXT"
 		done
 	    else
-		$COMPRESS -f $b
+		$COMPRESS -f "$b"
 	    fi
 	}
     done
 
-    for f in `find $d -type l`
+    find $d -type l | while read f
     do
-	l=`ls -l $f | sed -e 's/.* -> //' -e 's/\.\(gz\|Z\|bz2\|xz\|lzma\)$//'`
-	rm -f $f
-	b=`echo $f | sed -e 's/\.\(gz\|Z\|bz2\|xz\|lzma\)$//'`
-	ln -sf $l$COMPRESS_EXT $b$COMPRESS_EXT
+	l="`ls -l "$f" | sed -e 's/.* -> //' -e 's/\.\(gz\|Z\|bz2\|xz\|lzma\)$//'`"
+	rm -f "$f"
+	b="`echo "$f" | sed -e 's/\.\(gz\|Z\|bz2\|xz\|lzma\)$//'`"
+	ln -sf "$l$COMPRESS_EXT" "$b$COMPRESS_EXT"
     done
 done
