--- ./lib/rpmrc.c.orig	2019-06-26 14:17:31.416985688 +0000
+++ ./lib/rpmrc.c	2019-10-02 09:52:05.076903733 +0000
@@ -79,11 +79,13 @@ struct rpmOption {
     int localize;
 };
 
+#if defined(__linux__) && (defined(__powerpc__) || defined(__sparc__) || (defined(__arm__) && defined(__ARM_PCS_VFP)))
 static struct rpmat_s {
     const char *platform;
     uint64_t hwcap;
     uint64_t hwcap2;
 } rpmat;
+#endif
 
 typedef struct defaultEntry_s {
     char * name;
@@ -950,7 +952,7 @@ static int is_geode(void)
 #endif
 
 
-#if defined(__linux__)
+#if defined(__linux__) && (defined(__powerpc__) || defined(__sparc__) || (defined(__arm__) && defined(__ARM_PCS_VFP)))
 #ifndef AT_HWCAP2 /* glibc < 2.18 */
 #define AT_HWCAP2 26
 #endif
@@ -1013,7 +1015,7 @@ static void defaultMachine(rpmrcCtx ctx,
     canonEntry canon;
     int rc;
 
-#if defined(__linux__)
+#if defined(__linux__) && (defined(__powerpc__) || defined(__sparc__) || (defined(__arm__) && defined(__ARM_PCS_VFP)))
     /* Populate rpmat struct with hw info */
     read_auxv();
 #endif
