From 8763969a4a5f19231c0d851e5d50eff614cf5940 Mon Sep 17 00:00:00 2001
Date: Thu, 23 Jan 2020 13:43:36 +0100
Subject: [PATCH] Do not unset $MALLOC_CHECK_

How do we even _know_ user wants to debug malloc in rpmbuild,
maybe user wants to debug it in _the child_?

Signed-off-by: Denys Vlasenko <dvlasenk@redhat.com>
---
 build/rpmfc.c    | 1 -
 lib/rpmscript.c  | 3 ---
 sign/rpmgensig.c | 1 -
 3 files changed, 5 deletions(-)

--- build/rpmfc.c
+++ build/rpmfc.c
@@ -319,7 +319,6 @@ static int getOutputFrom(ARGV_t argv,
 	if (buildRoot)
 	    setenv("RPM_BUILD_ROOT", buildRoot, 1);
 
-	unsetenv("MALLOC_CHECK_");
 	execvp(argv[0], (char *const *)argv);
 	rpmlog(RPMLOG_ERR, _("Couldn't exec %s: %s\n"),
 		argv[0], strerror(errno));
--- lib/rpmscript.c
+++ lib/rpmscript.c
@@ -245,9 +245,6 @@ static void doScriptExec(ARGV_const_t argv, ARGV_const_t prefixes,
     }
 	
     if (chdir("/") == 0) {
-	/* XXX Don't mtrace into children. */
-	unsetenv("MALLOC_CHECK_");
-
 	xx = execv(argv[0], argv);
 	if (xx) {
 	    rpmlog(RPMLOG_ERR,
--- sign/rpmgensig.c
+++ sign/rpmgensig.c
@@ -216,7 +216,6 @@ static int runGPG(sigTarget sigt, const char *sigfile)
 	dup2(pipefd[0], STDIN_FILENO);
 	close(pipefd[1]);
 
-	unsetenv("MALLOC_CHECK_");
 	cmd = rpmExpand("%{?__gpg_sign_cmd}", NULL);
 	rc = poptParseArgvString(cmd, NULL, (const char ***)&av);
 	if (!rc)
