diff -Nru azure-mgmt-servermanager-2.0.0.orig/setup.py azure-mgmt-servermanager-2.0.0/setup.py
--- azure-mgmt-servermanager-2.0.0.orig/setup.py	2018-05-29 19:49:26.000000000 +0200
+++ azure-mgmt-servermanager-2.0.0/setup.py	2018-09-05 14:25:51.691050949 +0200
@@ -78,7 +78,7 @@
     packages=find_packages(exclude=["tests"]),
     install_requires=[
         'msrestazure>=0.4.27,<2.0.0',
-        'azure-common~=1.1',
+        'azure-common>=1.1',
     ],
     cmdclass=cmdclass
 )
