diff -Naur a/variety_lib/varietyconfig.py b/variety_lib/varietyconfig.py
--- a/variety_lib/varietyconfig.py	2018-10-23 10:56:25.000000000 -0500
+++ b/variety_lib/varietyconfig.py	2018-11-23 20:40:49.710196312 -0600
@@ -27,7 +27,7 @@
 
 # Where your project will look for your data (for instance, images and ui
 # files). By default, this is ../data, relative your trunk layout
-__variety_data_directory__ = '../data'
+__variety_data_directory__ = '/usr/share/variety'
 __license__ = 'GPL-3'
 __version__ = '0.7.1'
 
