--- httpd-2.4.6.orig/server/request.c
+++ httpd-2.4.6/server/request.c
@@ -46,10 +46,13 @@
 #include "util_script.h"
 #include "ap_expr.h"
 #include "mod_request.h"
-
+#include "http_connection.h"
 #include "mod_core.h"
 #include "mod_auth.h"
 
+#include <unistd.h>
+#include <sys/types.h>
+
 #if APR_HAVE_STDARG_H
 #include <stdarg.h>
 #endif
--- httpd-2.4.6.orig/server/config.c
+++ httpd-2.4.6/server/config.c
@@ -48,10 +48,14 @@
 #include "http_request.h"  /* for default_handler (see invoke_handler) */
 #include "http_main.h"
 #include "http_vhost.h"
+#include "http_connection.h"
 #include "util_cfgtree.h"
 #include "util_varbuf.h"
 #include "mpm_common.h"
 
+#include <unistd.h>
+#include <sys/types.h>
+
 #define APLOG_UNSET   (APLOG_NO_MODULE - 1)
 /* we know core's module_index is 0 */
 #undef APLOG_MODULE_INDEX
