diff --git a/CMakeLists.txt b/CMakeLists.txt
index e18fab718..98ecbd76e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -118,6 +118,9 @@ include(${PDAL_CMAKE_DIR}/lazperf.cmake)
 include(${PDAL_CMAKE_DIR}/gcs.cmake) # Optional
 include(${PDAL_CMAKE_DIR}/utfcpp.cmake)
 
+find_package(PkgConfig REQUIRED)
+PKG_CHECK_MODULES(EIGEN REQUIRED eigen3)
+
 #------------------------------------------------------------------------------
 # generate the pdal_features.hpp header
 #------------------------------------------------------------------------------
@@ -153,7 +156,6 @@ endif()
 add_subdirectory(plugins)
 
 if (WITH_TESTS)
-    include (${PDAL_CMAKE_DIR}/gtest.cmake)
     add_subdirectory(test)
 endif()
 add_subdirectory(dimbuilder)
@@ -219,7 +221,7 @@ add_library(PDAL::PDAL ALIAS ${PDAL_LIB_NAME})
 #
 target_include_directories(${PDAL_LIB_NAME}
     SYSTEM BEFORE PRIVATE
-        ${PDAL_VENDOR_DIR}/eigen
+    	${EIGEN_INCLUDE_DIRS}
         ${PDAL_VENDOR_DIR}/kazhdan
         ${PDAL_VENDOR_DIR}/h3/include
 )
diff --git a/plugins/cpd/CMakeLists.txt b/plugins/cpd/CMakeLists.txt
index ba85e27a7..dcf936502 100644
--- a/plugins/cpd/CMakeLists.txt
+++ b/plugins/cpd/CMakeLists.txt
@@ -13,7 +13,7 @@ find_package(Fgt REQUIRED CONFIG)
 option(BUILD_PLUGIN_CPD "Build Coherent Point Drift support" ${Cpd_FOUND})
 
 set(files filters/CpdFilter.cpp)
-set(include_dirs "${CMAKE_CURRENT_LIST_DIR}" "${PDAL_VENDOR_DIR}/eigen")
+set(include_dirs "${CMAKE_CURRENT_LIST_DIR}" "${EIGEN_INCLUDE_DIRS}")
 
 PDAL_ADD_PLUGIN(filter_libname filter cpd
     FILES
@@ -22,7 +22,7 @@ PDAL_ADD_PLUGIN(filter_libname filter cpd
         Cpd::Library-C++
         ${PDAL_LIBRARIES}
     INCLUDES
-        ${PDAL_VENDOR_DIR}/eigen
+        ${EIGEN_INCLUDE_DIRS}
         ${CMAKE_CURRENT_LIST_DIR}
     )
 
diff --git a/plugins/rdb/CMakeLists.txt b/plugins/rdb/CMakeLists.txt
index 0ddb9a24e..49a74a3e8 100644
--- a/plugins/rdb/CMakeLists.txt
+++ b/plugins/rdb/CMakeLists.txt
@@ -25,7 +25,6 @@ PDAL_ADD_PLUGIN(libname reader rdb
         ${RDB_INCLUDE_C_DIR}
         ${PDAL_INCLUDE_DIRS}
 )
-target_include_directories(${libname} PRIVATE "${PDAL_VENDOR_DIR}/eigen")
 target_include_directories(${libname} PUBLIC "${CMAKE_CURRENT_LIST_DIR}/../../vendor/nlohmann/")
 target_include_directories(${libname} PUBLIC "${CMAKE_CURRENT_LIST_DIR}/../../vendor/eigen/")
 
@@ -54,7 +53,7 @@ if (BUILD_RDBLIB_TESTS)
             ${PROJECT_SOURCE_DIR}/plugins/rdb/io
             ${RDB_INCLUDE_CPP_DIR}
             ${RDB_INCLUDE_C_DIR}
-            ${PDAL_VENDOR_DIR}/eigen
+	    ${EIGEN_INCLUDE_DIRS}
 
     )
 endif()
diff --git a/plugins/teaser/CMakeLists.txt b/plugins/teaser/CMakeLists.txt
index cb1cb0e4e..c4017cab8 100644
--- a/plugins/teaser/CMakeLists.txt
+++ b/plugins/teaser/CMakeLists.txt
@@ -67,6 +67,6 @@ if (WITH_TESTS)
         LINK_WITH
             ${filter_libname}
         INCLUDES
-            "${PDAL_VENDOR_DIR}/eigen"
+	    ${EIGEN_INCLUDE_DIRS}
     )
 endif()
diff --git a/plugins/trajectory/CMakeLists.txt b/plugins/trajectory/CMakeLists.txt
index 31fc9942f..6ec207d95 100644
--- a/plugins/trajectory/CMakeLists.txt
+++ b/plugins/trajectory/CMakeLists.txt
@@ -18,7 +18,7 @@ PDAL_ADD_PLUGIN(libname filter trajectory
         ${CERES_LIBRARIES}
         ${PDAL_LIBRARIES}
     INCLUDES
-        ${PDAL_VENDOR_DIR}/eigen
+        ${EIGEN_INCLUDE_DIRS}
 )
 
 if (WITH_TRAJECTORY_TEST)
diff --git a/test/unit/CMakeLists.txt b/test/unit/CMakeLists.txt
index 7ac5a1ede..a7e3bedd2 100644
--- a/test/unit/CMakeLists.txt
+++ b/test/unit/CMakeLists.txt
@@ -34,7 +34,7 @@ PDAL_ADD_TEST(pdal_eigen_test
         ${PDAL_SRC_DIR}/private/MathUtils.cpp
 
     INCLUDES
-        ${PDAL_VENDOR_DIR}/eigen
+        ${EIGEN_INCLUDE_DIRS}
 )
 PDAL_ADD_TEST(pdal_file_utils_test FILES FileUtilsTest.cpp)
 PDAL_ADD_TEST(pdal_georeference_test FILES GeoreferenceTest.cpp)
@@ -43,7 +43,7 @@ PDAL_ADD_TEST(pdal_kdindex_test
         KDIndexTest.cpp
     INCLUDES
         ${PDAL_VENDOR_DIR}
-        ${PDAL_VENDOR_DIR}/eigen
+        ${EIGEN_INCLUDE_DIRS}
 )
 PDAL_ADD_TEST(pdal_kernel_test FILES KernelTest.cpp)
 PDAL_ADD_TEST(pdal_log_test FILES LogTest.cpp)
@@ -52,7 +52,7 @@ PDAL_ADD_TEST(pdal_math_utils_test
         MathUtilsTest.cpp
         ${PDAL_SRC_DIR}/private/MathUtils.cpp
     INCLUDES
-        ${PDAL_VENDOR_DIR}/eigen
+        ${EIGEN_INCLUDE_DIRS}
 )
 PDAL_ADD_TEST(pdal_metadata_test FILES MetadataTest.cpp)
 PDAL_ADD_TEST(pdal_oldpclblock_test FILES OldPCLBlockTest.cpp)
@@ -84,7 +84,7 @@ PDAL_ADD_TEST(pdal_point_view_test
     FILES
         PointViewTest.cpp
     INCLUDES
-        ${PDAL_VENDOR_DIR}/eigen
+        ${EIGEN_INCLUDE_DIRS}
 )
 PDAL_ADD_TEST(pdal_point_table_test FILES PointTableTest.cpp)
 
@@ -227,13 +227,13 @@ PDAL_ADD_TEST(pdal_io_pcd_reader_test
     FILES
         io/PcdReaderTest.cpp
     INCLUDES
-        ${PDAL_VENDOR_DIR}/eigen
+        ${EIGEN_INCLUDE_DIRS}
 )
 PDAL_ADD_TEST(pdal_io_pcd_writer_test
     FILES
         io/PcdWriterTest.cpp
     INCLUDES
-        ${PDAL_VENDOR_DIR}/eigen
+        ${EIGEN_INCLUDE_DIRS}
 )
 PDAL_ADD_TEST(pdal_io_ply_reader_test
     FILES
@@ -279,7 +279,7 @@ PDAL_ADD_TEST(pdal_i3s_obb_test
     INCLUDES
         ${NLOHMANN_INCLUDE_DIR}
         ${PDAL_INCLUDE_DIR}
-        ${PDAL_VENDOR_DIR}/eigen
+        ${EIGEN_INCLUDE_DIRS}
 )
 
 if (BUILD_I3S_TESTS)
@@ -300,7 +300,7 @@ PDAL_ADD_TEST(pdal_filters_chipper_test
         filters/ChipperTest.cpp
     INCLUDES
         ${NLOHMANN_INCLUDE_DIR}
-        ${PDAL_VENDOR_DIR}/eigen
+        ${EIGEN_INCLUDE_DIRS}
 )
 PDAL_ADD_TEST(pdal_filters_nndistance_test
     FILES
@@ -359,7 +359,7 @@ PDAL_ADD_TEST(pdal_filters_icp_test
     FILES
         filters/IcpFilterTest.cpp
     INCLUDES
-        ${PDAL_VENDOR_DIR}/eigen
+        ${EIGEN_INCLUDE_DIRS}
 )
 PDAL_ADD_TEST(pdal_filters_info_test FILES filters/InfoFilterTest.cpp)
 PDAL_ADD_TEST(pdal_filters_lloydkmeans_test FILES filters/LloydKMeansFilterTest.cpp)
@@ -380,7 +380,7 @@ PDAL_ADD_TEST(pdal_filters_planefit_test
     FILES
         filters/PlaneFitFilterTest.cpp
     INCLUDES
-        ${PDAL_VENDOR_DIR}/eigen)
+        ${EIGEN_INCLUDE_DIRS})
 PDAL_ADD_TEST(pdal_filters_pmf_test FILES filters/PMFFilterTest.cpp)
 PDAL_ADD_TEST(pdal_filters_reprojection_test FILES
     filters/ReprojectionFilterTest.cpp)
@@ -408,7 +408,7 @@ PDAL_ADD_TEST(pdal_filters_splitter_test
     FILES
         filters/SplitterTest.cpp
     INCLUDES
-        ${PDAL_VENDOR_DIR}/eigen)
+        ${EIGEN_INCLUDE_DIRS})
 PDAL_ADD_TEST(pdal_filters_stats_test FILES filters/StatsFilterTest.cpp)
 PDAL_ADD_TEST(pdal_filters_transformation_test FILES
     filters/TransformationFilterTest.cpp)
