From a1ff5e22f6a1260a2edbda90d31f2ad8e9a02635 Mon Sep 17 00:00:00 2001
From: Luigi Baldoni <aloisio@gmx.com>
Date: Tue, 15 Feb 2022 18:02:02 +0100
Subject: [PATCH] Use correct path for manpage

---
 StandAlone/ManPages/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/StandAlone/ManPages/CMakeLists.txt b/StandAlone/ManPages/CMakeLists.txt
index 6da3dfeb8..7b2c669ec 100644
--- a/StandAlone/ManPages/CMakeLists.txt
+++ b/StandAlone/ManPages/CMakeLists.txt
@@ -35,5 +35,5 @@ if (GZIP_TOOL)
                       VERBATIM
                       SOURCES src/webcamoid.1.in)
 
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/src/webcamoid.1.gz DESTINATION ${CMAKE_INSTALL_MANDIR})
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/src/webcamoid.1.gz DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
 endif ()
