Index: argcomplete-3.0.8/setup.py
===================================================================
--- argcomplete-3.0.8.orig/setup.py
+++ argcomplete-3.0.8/setup.py
@@ -5,7 +5,7 @@ import glob
 from setuptools import find_packages, setup
 
 install_requires = []
-tests_require = ["coverage", "pexpect", "wheel", "ruff", "mypy"]
+tests_require = ["pexpect"]
 importlib_backport_requires = ["importlib-metadata >= 0.23, < 7"]
 
 setup(
