diff -rub a/scripts/startxfce4.in b/scripts/startxfce4.in
--- a/scripts/startxfce4.in	2019-05-13 13:08:40.936637683 +0800
+++ b/scripts/startxfce4.in	2019-05-13 13:16:44.378397419 +0800
@@ -76,6 +76,10 @@
 then
   XDG_DATA_DIRS="$XDG_DATA_DIRS:@_datadir_@"
 fi
+if test -d "@_datadir_@/xfce"
+then
+  XDG_DATA_DIRS="@_datadir_@/xfce:$XDG_DATA_DIRS"
+fi
 export XDG_DATA_DIRS
 
 if test "x$XDG_CONFIG_DIRS" = "x"
