---
 src/base/ftobjs.c |    5 +++++
 1 file changed, 5 insertions(+)

Index: freetype-2.10.4/src/base/ftobjs.c
===================================================================
--- freetype-2.10.4.orig/src/base/ftobjs.c
+++ freetype-2.10.4/src/base/ftobjs.c
@@ -2602,6 +2602,11 @@
 
     if ( FT_IS_SCALABLE( face ) )
     {
+      if ( face->family_name && strncmp(face->family_name, "CMEX", 4 ) == 0){
+	face->underline_position = (FT_Short)( -face->units_per_EM / 10 );
+	face->underline_thickness = (FT_Short)( face->units_per_EM / 30 );
+      }
+
       if ( face->height < 0 )
         face->height = (FT_Short)-face->height;
 
