Index: ft2demos-2.10.0/src/ttdebug.c
===================================================================
--- ft2demos-2.10.0.orig/src/ttdebug.c
+++ ft2demos-2.10.0/src/ttdebug.c
@@ -2024,11 +2024,11 @@
 
       FT_Library_Version( library, &major, &minor, &patch );
 
-      offset = snprintf( version_string, 64,
+      offset = snprintf( version_string, sizeof(version_string),
                          "ttdebug (FreeType) %d.%d",
                          major, minor );
       if ( patch )
-        offset = snprintf( version_string + offset, (size_t)( 64 - offset ),
+        offset = snprintf( version_string + offset, (size_t)( sizeof(version_string) - offset ),
                            ".%d",
                            patch );
     }
