diff -urp gst-plugins-base-1.16.0.orig/docs/libs/Makefile.am gst-plugins-base-1.16.0/docs/libs/Makefile.am
--- gst-plugins-base-1.16.0.orig/docs/libs/Makefile.am	2019-01-26 12:51:01.000000000 -0600
+++ gst-plugins-base-1.16.0/docs/libs/Makefile.am	2019-06-15 11:56:25.524212304 -0500
@@ -104,7 +104,7 @@ extra_files =
 
 # CFLAGS and LDFLAGS for compiling scan program. Only needed if your app/lib
 # contains GtkObjects/GObjects and you want to document signals and properties.
-GTKDOC_CFLAGS = $(GST_PLUGINS_BASE_CFLAGS) $(GST_BASE_CFLAGS)
+GTKDOC_CFLAGS = $(GST_PLUGINS_BASE_CFLAGS) $(GST_BASE_CFLAGS) $(GL_CFLAGS)
 GTKDOC_LIBS = \
 	$(top_builddir)/gst-libs/gst/allocators/libgstallocators-@GST_API_VERSION@.la \
 	$(top_builddir)/gst-libs/gst/audio/libgstaudio-@GST_API_VERSION@.la \
diff -urp gst-plugins-base-1.16.0.orig/docs/libs/Makefile.in gst-plugins-base-1.16.0/docs/libs/Makefile.in
--- gst-plugins-base-1.16.0.orig/docs/libs/Makefile.in	2019-04-18 18:16:04.000000000 -0500
+++ gst-plugins-base-1.16.0/docs/libs/Makefile.in	2019-06-15 11:57:19.124524248 -0500
@@ -634,7 +634,7 @@ extra_files =
 
 # CFLAGS and LDFLAGS for compiling scan program. Only needed if your app/lib
 # contains GtkObjects/GObjects and you want to document signals and properties.
-GTKDOC_CFLAGS = $(GST_PLUGINS_BASE_CFLAGS) $(GST_BASE_CFLAGS)
+GTKDOC_CFLAGS = $(GST_PLUGINS_BASE_CFLAGS) $(GST_BASE_CFLAGS) $(GL_CFLAGS)
 GTKDOC_LIBS = \
 	$(top_builddir)/gst-libs/gst/allocators/libgstallocators-@GST_API_VERSION@.la \
 	$(top_builddir)/gst-libs/gst/audio/libgstaudio-@GST_API_VERSION@.la \
