diff --git a/configure b/configure
index 58c2986..32ca4bf 100755
--- a/configure
+++ b/configure
@@ -24206,7 +24206,7 @@ cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 
 
   #include <arm_neon.h>
-  int32x4_t testfunc(int16_t *a, int16_t *b) {
+  int32x4_t __attribute__((used)) testfunc(int16_t *a, int16_t *b) {
       asm volatile ("vmull.s16 q0, d0, d0" : : : "q0");
       return vmull_s16(vld1_s16(a), vld1_s16(b));
   }

