From c2b959b7ce05d5a6a9b486b78f2aaac7b4e874a9 Mon Sep 17 00:00:00 2001
From: Mel Gorman <mgorman@suse.de>
Date: Thu, 23 May 2013 00:49:36 +0100
Subject: [PATCH] Version binary according to openSUSE guidelines

Subject says is all. rpmlint will complain about date in the name and
the binary version should match the rpm string.

Signed-off-by: Mel Gorman <mgorman@suse.de>

diff --git a/numad.c b/numad.c
index c9c86f5..d08b622 100644
--- a/numad.c
+++ b/numad.c
@@ -54,7 +54,7 @@ Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 #include <values.h>
 
 
-#define VERSION_STRING "20121130"
+#define VERSION_STRING "0.5.20130522"
 
 
 #define VAR_RUN_FILE "/var/run/numad.pid"
@@ -671,7 +671,7 @@ void shut_down_numad() {
 
 
 void print_version_and_exit(char *prog_name) {
-    fprintf(stdout, "%s version: %s: compiled %s\n", prog_name, VERSION_STRING, __DATE__);
+    fprintf(stdout, "%s version: %s\n", prog_name, VERSION_STRING);
     exit(EXIT_SUCCESS);
 }
 
