From fa3542ae702884c59dc2a1df9da22b9b939d5db8 Mon Sep 17 00:00:00 2001
From: Mamoru TASAKA <mtasaka@fedoraproject.org>
Date: Wed, 8 Nov 2017 15:00:05 +0900
Subject: [PATCH] Properly set return value for lxterminal_socket_initialize

Fixes #40 #41
---
 src/unixsocket.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/unixsocket.c b/src/unixsocket.c
index b813f52..1874399 100644
--- a/src/unixsocket.c
+++ b/src/unixsocket.c
@@ -263,5 +263,5 @@ static gboolean handle_request(GIOChannel* gio, GIOCondition condition, ClientIn
 }
 
 gboolean lxterminal_socket_initialize(LXTermWindow* lxtermwin, gint argc, gchar** argv) {
-    init(lxtermwin, argc, argv);
+    return init(lxtermwin, argc, argv);
 }
