Index: ParaView-v5.7.0/VTK/ThirdParty/exodusII/vtkexodusII/CMakeLists.txt
===================================================================
--- ParaView-v5.7.0.orig/VTK/ThirdParty/exodusII/vtkexodusII/CMakeLists.txt
+++ ParaView-v5.7.0/VTK/ThirdParty/exodusII/vtkexodusII/CMakeLists.txt
@@ -301,6 +301,8 @@ vtk_module_add_module(VTK::exodusII
   SOURCES ${sources}
   HEADERS ${headers}
   HEADERS_SUBDIR "vtkexodusII/include")
+vtk_module_link(VTK::exodusII PUBLIC pthread)
+vtk_module_link(VTK::exodusII PUBLIC hdf5)
 target_compile_definitions(exodusII
   PRIVATE
     exoIIc_EXPORTS)
