Index: ParaView-v5.8.0/Clients/ParaView/org.paraview.ParaView.desktop.in
===================================================================
--- ParaView-v5.8.0.orig/Clients/ParaView/org.paraview.ParaView.desktop.in
+++ ParaView-v5.8.0/Clients/ParaView/org.paraview.ParaView.desktop.in
@@ -6,3 +6,5 @@ Comment=Parallel visualization applicati
 Exec=paraview %f
 TryExec=paraview
 Icon=paraview
+Categories=Graphics;2DGraphics;3DGraphics;
+StartupNotify=true
