diff --git a/MyGUIEngine/include/MyGUI_Singleton.h b/MyGUIEngine/include/MyGUI_Singleton.h
index 5067da9..e12287d 100644
--- a/MyGUIEngine/include/MyGUI_Singleton.h
+++ b/MyGUIEngine/include/MyGUI_Singleton.h
@@ -58,8 +58,14 @@ namespace MyGUI
 		}
 
 	private:
+#if MYGUI_COMPILER == MYGUI_COMPILER_MSVC || MYGUI_PLATFORM == MYGUI_PLATFORM_APPLE
 		static T* msInstance;
 		static const char* mClassTypeName;
+#else
+		static MYGUI_EXPORT T* msInstance;
+
+		static MYGUI_EXPORT const char* mClassTypeName;
+#endif
 	};
 
 } // namespace MyGUI
