diff --git a/Common/CMakeLists.txt b/Common/CMakeLists.txt
index 785dded..a232443 100644
--- a/Common/CMakeLists.txt
+++ b/Common/CMakeLists.txt
@@ -121,6 +121,7 @@ elseif(MYGUI_SAMPLES_INPUT EQUAL 4)
 endif()
 
 add_library(${PROJECTNAME} ${HEADER_FILES} ${SOURCE_FILES})
+set_target_properties(${PROJECTNAME} PROPERTIES VERSION ${MYGUI_VERSION} SOVERSION ${MYGUI_VERSION_MAJOR})
 
 if(MYGUI_RENDERSYSTEM EQUAL 1)
 	add_dependencies(${PROJECTNAME} MyGUI.DummyPlatform)
