From aeaaecbbaa5c2eac245abf2229a51844384bb5f9 Mon Sep 17 00:00:00 2001
From: PhilippSelenium <31542906+PhilippSelenium@users.noreply.github.com>
Date: Wed, 1 Apr 2020 10:12:09 +0200
Subject: [PATCH] Remove deprecated pytest_namespace

---
 nose2pytest/assert_tools.py | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/nose2pytest/assert_tools.py b/nose2pytest/assert_tools.py
index 9f796ed..8468bdb 100644
--- a/nose2pytest/assert_tools.py
+++ b/nose2pytest/assert_tools.py
@@ -10,6 +10,7 @@
 module may be sufficient to decrease your test suite's third-party dependencies by 1.
 """
 
+import pytest
 import unittest
 
 
@@ -96,13 +97,10 @@ def do_nothing(self):
 
 # Use similar trick as Nose to bring in bound methods from unittest.TestCase as free functions:
 
-def pytest_namespace() -> {str: callable}:
-    namespace = {}
+def pytest_configure():
     for name, obj in globals().items():
         if name.startswith('assert_'):
-            namespace[name] = obj
-
-    return namespace
+            setattr(pytest, name, obj)
 
 
 # licensing
