---
 gnulib-tests/gnulib.mk |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

Index: gnulib-tests/gnulib.mk
===================================================================
--- gnulib-tests/gnulib.mk.orig
+++ gnulib-tests/gnulib.mk
@@ -1115,10 +1115,10 @@ EXTRA_DIST += test-getloadavg.c signatur
 
 ## begin gnulib module getlogin-tests
 
-TESTS += test-getlogin
-check_PROGRAMS += test-getlogin
-test_getlogin_LDADD = $(LDADD) $(GETLOGIN_LIB)
-EXTRA_DIST += test-getlogin.c test-getlogin.h signature.h macros.h
+# TESTS += test-getlogin
+# check_PROGRAMS += test-getlogin
+# test_getlogin_LDADD = $(LDADD) $(GETLOGIN_LIB)
+# EXTRA_DIST += test-getlogin.c test-getlogin.h signature.h macros.h
 
 ## end   gnulib module getlogin-tests
 
