diff -Nru azure-mgmt-hdinsight-0.2.1.orig/setup.py azure-mgmt-hdinsight-0.2.1/setup.py
--- azure-mgmt-hdinsight-0.2.1.orig/setup.py	2019-03-27 14:48:10.823325468 +0100
+++ azure-mgmt-hdinsight-0.2.1/setup.py	2019-03-27 14:48:26.915475968 +0100
@@ -81,7 +81,4 @@
         'msrestazure>=0.4.32,<2.0.0',
         'azure-common>=1.1',
     ],
-    extras_require={
-        ":python_version<'3.0'": ['azure-mgmt-nspkg'],
-    }
 )
