Index: util/texi2pdf
===================================================================
--- util/texi2pdf.orig	2005-01-28 03:52:04.000000000 +0200
+++ util/texi2pdf	2015-07-18 15:48:10.240544056 +0200
@@ -16,4 +16,4 @@ unset RUNNING_BSH
 # hack around a bug in zsh:
 test -n "${ZSH_VERSION+set}" && alias -g '${1+"$@"}'='"$@"'
 
-texi2dvi --pdf ${1+"$@"}
+texi2dvi4 --pdf ${1+"$@"}
Index: util/pdftexi2dvi
===================================================================
--- util/pdftexi2dvi.orig	2008-01-02 03:15:37.000000000 +0200
+++ util/pdftexi2dvi	2015-07-18 15:48:10.240544056 +0200
@@ -16,4 +16,4 @@ unset RUNNING_BSH
 # hack around a bug in zsh:
 test -n "${ZSH_VERSION+set}" && alias -g '${1+"$@"}'='"$@"'
 
-texi2dvi --pdf ${1+"$@"}
+texi2dvi4 --pdf ${1+"$@"}
Index: util/texi2dvi
===================================================================
--- util/texi2dvi.orig	2015-07-18 15:47:54.000000000 +0200
+++ util/texi2dvi	2015-07-18 15:57:20.551314521 +0200
@@ -96,7 +96,7 @@ CDPATH=${ZSH_VERSION+.}$path_sep
 # If $TEX is set to a directory, don't use it.
 test -n "$TEX" && test -d "$TEX" && unset TEX
 
-# 
+#
 ## --------------------- ##
 ## Auxiliary functions.  ##
 ## --------------------- ##
@@ -1017,7 +1017,7 @@ run_index ()
 {
   case $in_lang in
     latex)   texindex=${MAKEINDEX:-makeindex};;
-    texinfo) texindex=${TEXINDEX:-texindex};;
+    texinfo) texindex=${TEXINDEX:-texindex4};;
   esac
   index_files=`generated_files_get $in_noext index_file_p`
   if test -n "$texindex" && test -n "$index_files"; then
@@ -1483,7 +1483,7 @@ cleanup ()
 arg_sep="$$--$$"
 set dummy ${1+"$@"} "$arg_sep"; shift
 
-# 
+#
 # Parse command line arguments.
 while test x"$1" != x"$arg_sep"; do
 
@@ -1639,7 +1639,7 @@ else
   exec 6>/dev/null
 fi
 
-# 
+#
 
 # input_file_name_decode
 # ----------------------
