Index: tvtime-1.0.11/src/tvtime.c
===================================================================
--- tvtime-1.0.11.orig/src/tvtime.c
+++ tvtime-1.0.11/src/tvtime.c
@@ -28,7 +28,8 @@
 #include <string.h>
 #include <locale.h>
 #include <stdlib.h>
-#include <termio.h>
+#include <sys/ioctl.h>
+#include <termios.h>
 #include <ctype.h>
 #if defined (__SVR4) && defined (__sun)
 # include <sys/int_types.h>
