diff -Ppru cyrus-imapd-2.4.18.orig/lib/strarray.c cyrus-imapd-2.4.18.impl/lib/strarray.c
--- cyrus-imapd-2.4.18.orig/lib/strarray.c	2015-07-06 05:38:29.000000000 +0200
+++ cyrus-imapd-2.4.18.impl/lib/strarray.c	2015-08-31 17:28:35.256727763 +0200
@@ -43,6 +43,7 @@
  * Start Date: 2011/01/11
  */
 
+#include <ctype.h>
 #include "strarray.h"
 #include <memory.h>
 #include "util.h"   /* for strcmpsafe et al */
diff -Ppru cyrus-imapd-2.4.18.orig/sieve/flags.c cyrus-imapd-2.4.18.impl/sieve/flags.c
--- cyrus-imapd-2.4.18.orig/sieve/flags.c	2015-07-06 05:38:29.000000000 +0200
+++ cyrus-imapd-2.4.18.impl/sieve/flags.c	2015-08-31 17:29:11.323443879 +0200
@@ -8,6 +8,7 @@
 #include "flags.h"
 #include "imparse.h"
 #include "strarray.h"
+#include <string.h>
 #include "util.h"
 #include "xmalloc.h"
 
diff -Ppru cyrus-imapd-2.4.19.orig/ptclient/ldap.c cyrus-imapd-2.4.19.ldap/ptclient/ldap.c
--- cyrus-imapd-2.4.19.orig/ptclient/ldap.c	2017-05-15 07:43:30.000000000 +0200
+++ cyrus-imapd-2.4.19.ldap/ptclient/ldap.c	2017-06-21 17:56:18.831138260 +0200
@@ -77,6 +77,7 @@
 #include "auth_pts.h"
 #include "strhash.h"
 #include "xmalloc.h"
+#include "xstrlcat.h"
 
 /* xxx this just uses the UNIX canonicalization semantics, which is
  * most likely wrong */
