From: Dave Plater <davejplater@gmail.com>
Date: Wed  3 Jun 10:50:09 SAST 2020
Subject: QT 5.15 changes

(obsolete) 	QSplashScreen

Index: hydrogen-0.99+git20200527/src/gui/src/SplashScreen.cpp
===================================================================
--- hydrogen-0.99+git20200527.orig/src/gui/src/SplashScreen.cpp	2020-06-04 13:39:17.180544956 +0200
+++ hydrogen-0.99+git20200527/src/gui/src/SplashScreen.cpp	2020-06-04 13:46:01.503097637 +0200
@@ -32,7 +32,7 @@
 const char* SplashScreen::__class_name = "SplashScreen";
 
 SplashScreen::SplashScreen()
- : QSplashScreen( nullptr )
+: QSplashScreen((QWidget*) nullptr )
  , Object( __class_name )
 {
 	//INFOLOG( "SplashScreen" );
