--- a/daemon/gdm-session.c
+++ b/daemon/gdm-session.c
@@ -43,6 +43,8 @@
 #include <glib-object.h>
 #include <gio/gio.h>
 
+#include "gdm-sysconfig.h"
+
 #include "gdm-session.h"
 #include "gdm-session-glue.h"
 #include "gdm-dbus-util.h"
@@ -589,6 +591,14 @@ get_fallback_session_name (GdmSession *s
                 }
         }
 
+        name = gdm_sysconfig_load_value ("/etc/sysconfig/windowmanager", "DEFAULT_WM");
+        if (name && get_session_command_for_name (self, name, NULL, NULL)) {
+                g_free (self->fallback_session_name);
+                self->fallback_session_name = name;
+                goto out;
+        }
+        g_free (name);
+
         name = g_strdup ("gnome");
         if (get_session_command_for_name (self, name, NULL, NULL)) {
                 g_free (self->fallback_session_name);
