Index: nagios-4.4.2/cgi/archivejson.c
===================================================================
--- nagios-4.4.2.orig/cgi/archivejson.c
+++ nagios-4.4.2/cgi/archivejson.c
@@ -761,7 +761,7 @@ int main(void) {
 				json_result(query_time, THISCGI, 
 				svm_get_string_from_value(cgi_data.query, valid_queries), 
 				get_query_status(query_status, cgi_data.query),
-				compile_time(__DATE__, __TIME__), &current_authdata,
+				compile_time("Jan  1 1970", "00:00:00"), &current_authdata,
 				RESULT_SUCCESS, ""));
 		json_object_append_object(json_root, "data", 
 				json_help(archive_json_help));
Index: nagios-4.4.2/cgi/objectjson.c
===================================================================
--- nagios-4.4.2.orig/cgi/objectjson.c
+++ nagios-4.4.2/cgi/objectjson.c
@@ -1130,7 +1130,7 @@ int main(void) {
 				json_result(query_time, THISCGI, 
 				svm_get_string_from_value(cgi_data.query, valid_queries), 
 				get_query_status(query_status, cgi_data.query),
-				compile_time(__DATE__, __TIME__), &current_authdata,
+				compile_time("Jan  1 1970", "00:00:00"), &current_authdata,
 				RESULT_SUCCESS, ""));
 		json_object_append_object(json_root, "data", json_help(object_json_help));
 		break;
Index: nagios-4.4.2/cgi/statusjson.c
===================================================================
--- nagios-4.4.2.orig/cgi/statusjson.c
+++ nagios-4.4.2/cgi/statusjson.c
@@ -1135,7 +1135,7 @@ int main(void) {
 				json_result(query_time, THISCGI, 
 				svm_get_string_from_value(cgi_data.query, valid_queries), 
 				get_query_status(query_status, cgi_data.query),
-				compile_time(__DATE__, __TIME__), &current_authdata,
+				compile_time("Jan  1 1970", "00:00:00"), &current_authdata,
 				RESULT_SUCCESS, ""));
 		json_object_append_object(json_root, "data", json_help(status_json_help));
 		break;
