--- a/Makefile
+++ b/Makefile
@@ -21,15 +21,15 @@
 
 isutf8: is_utf8/*.c is_utf8/*.h
 	$(MAKE) -C is_utf8/
-	cp is_utf8/isutf8 .
+	cp -p is_utf8/isutf8 .
 
 install:
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
-	$(INSTALL_BIN) $(BINS) $(DESTDIR)$(PREFIX)/bin
-	install $(PERLSCRIPTS) $(DESTDIR)$(PREFIX)/bin
+	install -pm755 $(BINS) $(DESTDIR)$(PREFIX)/bin
+	install -pm755 $(PERLSCRIPTS) $(DESTDIR)$(PREFIX)/bin
 
 	mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1
-	install -m 644 $(MANS) $(DESTDIR)$(PREFIX)/share/man/man1
+	install -pm644 $(MANS) $(DESTDIR)$(PREFIX)/share/man/man1
 
 uninstall:
 	for i in $(BINS) $(PERLSCRIPTS); do rm -f $(DESTDIR)$(PREFIX)/bin/$$i; done
