---
 parsetime.y |    5 +++++
 1 file changed, 5 insertions(+)

Index: at-3.2.0/parsetime.y
===================================================================
--- at-3.2.0.orig/parsetime.y
+++ at-3.2.0/parsetime.y
@@ -518,6 +518,11 @@ parsetime(time_t currtime, int argc, cha
 	    else
 		unsetenv("TZ");
 	}
+	/* exectime zeroes its seconds, thus we need +60,
+	 * else "now" will be scheduled to tomorrow */
+	if (currtime > exectime + 60) {
+	    exectime += 24*3600;
+	}
 	if (exectime < currtime)
 		panic("refusing to create job destined in the past");
         return exectime;
