diff -Nur deepin-turbo-0.0.3/src/invoker/invoker.c deepin-turbo-0.0.3-new/src/invoker/invoker.c
--- deepin-turbo-0.0.3/src/invoker/invoker.c	2018-10-13 23:44:23.000000000 +0800
+++ deepin-turbo-0.0.3-new/src/invoker/invoker.c	2019-08-02 22:01:40.515235000 +0800
@@ -468,7 +468,7 @@
     } else {
         info("Failed to connect to the DBus session bus: %s", error.message);
         dbus_error_free(&error);
-        return 1;
+        return;
     }
 }
 
