--- a/src/dictionary.cpp
+++ b/src/dictionary.cpp
@@ -3533,7 +3533,7 @@
 		say_as = option_sayas;
 		option_sayas = 0;   // don't speak replacement word as letter names
 		text[0] = 0;
-		strncpy0(&text[1], word1, sizeof(text));
+		strncpy0(&text[1], word1, sizeof(text)-strlen(text)-1);
 		found = TranslateWord(tr, &text[1], 0, NULL, NULL);
 		strcpy(ph_out, word_phonemes);
 		option_sayas = say_as;
