--- src/wmpinboard.c
+++ src/wmpinboard.c
@@ -165,7 +165,7 @@
   char t[STRING_BUF_SIZE];
   FILE *file;
   int pid = (int) getpid();
-  static int sizes[6];
+  static int sizes[7];
   sizes[0] = size_0;
   sizes[1] = size_1;
   sizes[2] = size_2;
--- src/xmisc.c
+++ src/xmisc.c
@@ -204,7 +204,7 @@
     return;
   }
 
-  if (XGetIMValues(xim, XNQueryInputStyle, &xim_styles, 0) || !xim_styles) {
+  if (XGetIMValues(xim, XNQueryInputStyle, &xim_styles, (char *)NULL) || !xim_styles) {
     XCloseIM(xim);
 #ifdef DEBUG_IC
     fprintf(stderr, "Input method doesn't support any style.\n");
@@ -258,7 +258,7 @@
   }
 
   InputContext = XCreateIC(xim, XNInputStyle, input_style,
-    XNClientWindow, win, XNFocusWindow, win, 0);
+    XNClientWindow, win, XNFocusWindow, win, (char *)NULL);
 
   if (!InputContext) {
     XCloseIM(xim);
