Index: libreadline-java-0.8.2/Makefile
===================================================================
--- libreadline-java-0.8.2.orig/Makefile
+++ libreadline-java-0.8.2/Makefile
@@ -37,7 +37,7 @@ T_LIBS    = JavaReadline
 
 ## normal javac
 JAVAC = $(JAVA_HOME)/bin/javac
-JC_FLAGS += -target 1.5 -source 1.5
+JC_FLAGS += -target 1.6 -source 1.6
 
 # installation directories   --------------------------------------------------
 
