Index: cmd2-2.5.11/pyproject.toml
===================================================================
--- cmd2-2.5.11.orig/pyproject.toml
+++ cmd2-2.5.11/pyproject.toml
@@ -104,12 +104,6 @@ warn_unused_ignores = false
 
 [tool.pytest.ini_options]
 testpaths = ["tests"]
-addopts = [
-    "--cov=cmd2",
-    "--cov-append",
-    "--cov-report=term",
-    "--cov-report=html",
-]
 
 [tool.ruff]
 # Exclude a variety of commonly ignored directories.
