Index: cmd2-3.2.0/pyproject.toml
===================================================================
--- cmd2-3.2.0.orig/pyproject.toml
+++ cmd2-3.2.0/pyproject.toml
@@ -97,12 +97,6 @@ warn_unused_ignores = false
 
 [tool.pytest.ini_options]
 testpaths = ["tests"]
-addopts = [
-    "--cov=cmd2",
-    "--cov-append",
-    "--cov-report=term",
-    "--cov-report=html",
-]
 
 [tool.setuptools]
 packages = ["cmd2"]
