---
 pyproject.toml |    6 ------
 1 file changed, 6 deletions(-)

Index: cmd2-4.0.0/pyproject.toml
===================================================================
--- cmd2-4.0.0.orig/pyproject.toml
+++ cmd2-4.0.0/pyproject.toml
@@ -93,12 +93,6 @@ warn_unused_ignores = false
 
 [tool.pytest.ini_options]
 testpaths = ["tests"]
-addopts = [
-    "--cov=cmd2",
-    "--cov-append",
-    "--cov-report=term",
-    "--cov-report=html",
-]
 
 [tool.setuptools]
 packages = ["cmd2"]
