From f6dc7cca6246cc61571a652fae0ea73171fc8d7b Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Mon, 11 Aug 2014 03:14:41 +0200
Subject: [PATCH] src: avoid relying on implicitly-defined setgroups

---
 src/arping.c | 1 +
 1 file changed, 1 insertion(+)

Index: arping-2.16/src/arping.c
===================================================================
--- arping-2.16.orig/src/arping.c
+++ arping-2.16/src/arping.c
@@ -95,6 +95,7 @@
 
 #if HAVE_PWD_H
 #include <pwd.h>
+#include <grp.h>
 #endif
 
 #if HAVE_SYS_CAPABILITY_H
