Index: xrdp-0.9.26/instfiles/xrdp-sesman.service.in
===================================================================
--- xrdp-0.9.26.orig/instfiles/xrdp-sesman.service.in
+++ xrdp-0.9.26/instfiles/xrdp-sesman.service.in
@@ -6,12 +6,9 @@ StopWhenUnneeded=true
 BindsTo=xrdp.service
 
 [Service]
-Type=forking
-PIDFile=@localstatedir@/run/xrdp-sesman.pid
 EnvironmentFile=-@sysconfdir@/sysconfig/xrdp
 EnvironmentFile=-@sysconfdir@/default/xrdp
-ExecStart=@sbindir@/xrdp-sesman $SESMAN_OPTIONS
-ExecStop=@sbindir@/xrdp-sesman $SESMAN_OPTIONS --kill
+ExecStart=@sbindir@/xrdp-sesman $SESMAN_OPTIONS --nodaemon
 
 [Install]
 WantedBy=multi-user.target
Index: xrdp-0.9.26/instfiles/xrdp.service.in
===================================================================
--- xrdp-0.9.26.orig/instfiles/xrdp.service.in
+++ xrdp-0.9.26/instfiles/xrdp.service.in
@@ -5,12 +5,9 @@ Requires=xrdp-sesman.service
 After=network-online.target xrdp-sesman.service
 
 [Service]
-Type=forking
-PIDFile=@localstatedir@/run/xrdp.pid
 EnvironmentFile=-@sysconfdir@/sysconfig/xrdp
 EnvironmentFile=-@sysconfdir@/default/xrdp
-ExecStart=@sbindir@/xrdp $XRDP_OPTIONS
-ExecStop=@sbindir@/xrdp $XRDP_OPTIONS --kill
+ExecStart=@sbindir@/xrdp $XRDP_OPTIONS --nodaemon
 
 [Install]
 WantedBy=multi-user.target
