Index: b/sesman/sesman.ini
===================================================================
--- a/sesman/sesman.ini	2020-09-14 13:14:33.645606549 +0800
+++ b/sesman/sesman.ini	2020-09-14 13:14:43.105484131 +0800
@@ -27,7 +27,7 @@ RestrictOutboundClipboard=false
 ;; X11DisplayOffset - x11 display number offset
 ; Type: integer
 ; Default: 10
-X11DisplayOffset=10
+X11DisplayOffset=200
 
 ;; MaxSessions - maximum number of connections to an xrdp server
 ; Type: integer
@@ -65,9 +65,9 @@ Policy=Default
 
 [Logging]
 LogFile=xrdp-sesman.log
-LogLevel=DEBUG
-EnableSyslog=1
-SyslogLevel=DEBUG
+LogLevel=ERROR
+EnableSyslog=0
+SyslogLevel=ERROR
 
 ;
 ; Session definitions - startup command-line parameters for each session type
Index: b/xrdp/xrdp.ini
===================================================================
--- a/xrdp/xrdp.ini	2020-09-14 13:14:33.645606549 +0800
+++ b/xrdp/xrdp.ini	2020-09-14 13:14:43.105484131 +0800
@@ -101,7 +101,7 @@ grey=dedede
 #ls_title=My Login Title
 
 ; top level window background color in RGB format
-ls_top_window_bg_color=009cb5
+ls_top_window_bg_color=000000
 
 ; width and height of login screen
 ls_width=350
@@ -144,9 +144,9 @@ ls_btn_cancel_height=30
 
 [Logging]
 LogFile=xrdp.log
-LogLevel=DEBUG
-EnableSyslog=true
-SyslogLevel=DEBUG
+LogLevel=ERROR
+EnableSyslog=false
+SyslogLevel=ERROR
 ; LogLevel and SysLogLevel could by any of: core, error, warning, info or debug
 
 [Channels]
