Index: lm-sensors-3-6-0/prog/init/sensord.service
===================================================================
--- lm-sensors-3-6-0.orig/prog/init/sensord.service
+++ lm-sensors-3-6-0/prog/init/sensord.service
@@ -3,6 +3,16 @@ Description=Log hardware monitoring data
 After=lm_sensors.service
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+ProtectHostname=true
+ProtectKernelTunables=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 EnvironmentFile=/etc/sysconfig/sensord
 Type=forking
 PIDFile=/run/sensord.pid
