Index: hiawatha-10.11/extra/debian/hiawatha.service
===================================================================
--- hiawatha-10.11.orig/extra/debian/hiawatha.service
+++ hiawatha-10.11/extra/debian/hiawatha.service
@@ -3,6 +3,19 @@ Description=Hiawatha webserver
 After=network.target
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=read-only
+PrivateDevices=true
+ProtectHostname=true
+ProtectClock=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 Type=forking
 PIDFile=/var/run/hiawatha.pid
 ExecStartPre=/usr/sbin/wigwam -q
