--- configure.ac.orig	2020-06-07 00:44:49.780581237 +0200
+++ configure.ac	2020-06-07 00:46:14.617504085 +0200
@@ -340,23 +340,23 @@
 AC_SUBST(FFTW_CFLAGS)
 AC_SUBST(FFTW_LIBS)
 
-PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.56.0])
+PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.54.0])
 AC_SUBST(GLIB_CFLAGS)
 AC_SUBST(GLIB_LIBS)
 
-PKG_CHECK_MODULES([GOBJECT], [gobject-2.0 >= 2.56.0 glib-2.0 >= 2.56.0 gmodule-2.0 >= 2.56.0 gthread-2.0 >= 2.56.0])
+PKG_CHECK_MODULES([GOBJECT], [gobject-2.0 >= 2.54.0 glib-2.0 >= 2.54.0 gmodule-2.0 >= 2.54.0 gthread-2.0 >= 2.54.0])
 AC_SUBST(GOBJECT_CFLAGS)
 AC_SUBST(GOBJECT_LIBS)
 
 AC_DEFINE([HAVE_GLIB_2_6], [1], [GLib 2.6 available])
 AC_DEFINE([HAVE_GLIB_2_44], [1], [GLib 2.44 available])
 AC_DEFINE([HAVE_GLIB_2_54], [1], [GLib 2.54 available])
-AC_DEFINE([HAVE_GLIB_2_56], [1], [GLib 2.56 available])
+AC_DEFINE([HAVE_GLIB_2_56], [0], [GLib 2.56 available])
 AC_DEFINE([HAVE_GLIB_2_62], [0], [GLib 2.62 available])
 
 GOBJECT_INTROSPECTION_CHECK([1.30.0])
 
-PKG_CHECK_MODULES([GIO], [gio-2.0 >= 2.56.0 glib-2.0 >= 2.56.0 gmodule-2.0 >= 2.56.0 gthread-2.0 >= 2.56.0])
+PKG_CHECK_MODULES([GIO], [gio-2.0 >= 2.54.0 glib-2.0 >= 2.54.0 gmodule-2.0 >= 2.54.0 gthread-2.0 >= 2.54.0])
 AC_SUBST(GIO_CFLAGS)
 AC_SUBST(GIO_LIBS)
 
