Index: newlib-2.1.0/libgloss/epiphany/Makefile.in
===================================================================
--- newlib-2.1.0.orig/libgloss/epiphany/Makefile.in
+++ newlib-2.1.0/libgloss/epiphany/Makefile.in
@@ -184,7 +184,7 @@ distclean maintainer-clean realclean: cl
 install:
 	@for outputs in ${OUTPUTS}; do\
 	 mkdir -p $(DESTDIR)$(tooldir)/lib${MULTISUBDIR}; \
-	 $(INSTALL_PROGRAM) $${outputs} $(DESTDIR)$(tooldir)/lib${MULTISUBDIR}; \
+	 $(INSTALL_DATA) $${outputs} $(DESTDIR)$(tooldir)/lib${MULTISUBDIR}; \
 	done
 info:
 install-info:
--- newlib-4.3.0.20230120/libgloss/epiphany/crt0.S.orig	2023-03-08 13:57:15.671910169 +0100
+++ newlib-4.3.0.20230120/libgloss/epiphany/crt0.S	2023-03-08 13:57:27.135910057 +0100
@@ -121,7 +121,7 @@
 #error "not implemented"
 #else  /* !__STRUCT_ALIGN_64__ */
 	str	r1, [r2, 0]	; __atexit = &__atexit0
-	movr	r0, 1
+	mov	r0, 1
 	str	r0, [r1, 4]	; __atexit0._ind = 1
 	mov	r0,%low(fini)
 	movt	r0,%high(fini)
