diff -Nru aws-cli-1.42.50.orig/setup.py aws-cli-1.42.50/setup.py
--- aws-cli-1.42.50.orig/setup.py	2025-10-10 20:54:09.000000000 +0200
+++ aws-cli-1.42.50/setup.py	2025-10-24 09:22:29.248403986 +0200
@@ -24,12 +24,12 @@
 
 
 install_requires = [
-    'botocore==1.40.50',
-    'docutils>=0.18.1,<=0.19',
-    's3transfer>=0.14.0,<0.15.0',
-    'PyYAML>=3.10,<6.1',
-    'colorama>=0.2.5,<0.4.7',
-    'rsa>=3.1.2,<4.8',
+    'botocore>=1.40.50',
+    'docutils>=0.18.1',
+    's3transfer>=0.14.0',
+    'PyYAML>=3.10',
+    'colorama>=0.2.5',
+    'rsa>=3.1.2',
 ]
 
 
