diff -Nru aws-cli-1.32.105.orig/requirements-docs.txt aws-cli-1.32.105/requirements-docs.txt
--- aws-cli-1.32.105.orig/requirements-docs.txt	2024-05-14 20:05:47.000000000 +0200
+++ aws-cli-1.32.105/requirements-docs.txt	2024-05-16 13:42:46.446914441 +0200
@@ -3,7 +3,7 @@
 jinja2<3.1
 
 
-docutils>=0.10,<0.17
+docutils>=0.10
 Sphinx==1.4.9
 -e .
 # alabaster version 0.7.13 requires Sphinx>1.6
diff -Nru aws-cli-1.32.105.orig/setup.cfg aws-cli-1.32.105/setup.cfg
--- aws-cli-1.32.105.orig/setup.cfg	2024-05-14 20:05:47.000000000 +0200
+++ aws-cli-1.32.105/setup.cfg	2024-05-16 13:43:08.640300938 +0200
@@ -4,7 +4,7 @@
 [metadata]
 requires_dist =
         botocore==1.34.105
-        docutils>=0.10,<0.17
+        docutils>=0.10
         s3transfer>=0.10.0,<0.11.0
         PyYAML>=3.10,<6.1
         colorama>=0.2.5,<0.4.7
diff -Nru aws-cli-1.32.105.orig/setup.py aws-cli-1.32.105/setup.py
--- aws-cli-1.32.105.orig/setup.py	2024-05-14 20:05:47.000000000 +0200
+++ aws-cli-1.32.105/setup.py	2024-05-16 13:42:57.993608767 +0200
@@ -25,7 +25,7 @@
 
 install_requires = [
     'botocore==1.34.105',
-    'docutils>=0.10,<0.17',
+    'docutils>=0.10',
     's3transfer>=0.10.0,<0.11.0',
     'PyYAML>=3.10,<6.1',
     'colorama>=0.2.5,<0.4.7',
