--- docbook-sgml-4.3/docbook.dcl.~1~	2004-03-31 13:44:16.000000000 +0200
+++ docbook-sgml-4.3/docbook.dcl	2004-06-06 09:07:41.722095720 +0200
@@ -61,8 +61,8 @@
 	NAMING
 		LCNMSTRT ""
 		UCNMSTRT ""
-		LCNMCHAR ".-_"
-		UCNMCHAR ".-_"
+		LCNMCHAR ".-_:"
+		UCNMCHAR ".-_:"
 		NAMECASE
 			GENERAL YES
 			ENTITY  NO

--- docbook-sgml-4.3/htmltblx.mod.~1~	2004-03-31 13:43:24.000000000 +0200
+++ docbook-sgml-4.3/htmltblx.mod	2004-06-06 09:45:55.033459400 +0200
@@ -213,8 +213,10 @@
     bgcolor     CDATA         #IMPLIED
 '>
 
+<!-- HTML 4.01 defines TFrame as
+     "(void|above|below|hsides|lhs|rhs|vsides|box|border)" -->
 <!ENTITY % tbl.frame.attval "void|above|below|hsides|lhs|rhs|vsides|box|border|
-top|bottom|topbot|all|sides|none">
+top|bottom|topbot|sides">
 
 <!-- Allow either objects or inlines; beware of REs between elements. -->
 <!ENTITY % tbl.entry.mdl "%para.char.mix; | %tabentry.mix;">
