
---
 configure.ac |    6 ++++++
 1 file changed, 6 insertions(+)

Index: autotrace-0.31.10/configure.ac
===================================================================
--- autotrace-0.31.10.orig/configure.ac
+++ autotrace-0.31.10/configure.ac
@@ -88,6 +88,8 @@ else
       [HAVE_MAGICK=yes
        HAVE_GRAPHICSMAGICK=yes
        MAGICK_LIBRARY=GraphicsMagick
+       AC_SUBST([MAGICK_CFLAGS], [${GRAPHICSMAGICK_CFLAGS}])
+       AC_SUBST([MAGICK_LIBS], [${GRAPHICSMAGICK_LIBS}])
        AC_DEFINE(HAVE_GRAPHICSMAGICK, 1, [GraphicsMagick library is available.])
 	 ],
       AC_MSG_WARN(GraphicsMagick >= 1.3.40 is not available. Falling back to ImageMagick...)
@@ -98,6 +100,8 @@ else
       PKG_CHECK_MODULES([IMAGEMAGICK7],
           [ImageMagick >= 7.0.1],
 	 [HAVE_IMAGEMAGICK7=yes
+    AC_SUBST([MAGICK_CFLAGS], [${IMAGEMAGICK7_CFLAGS}])
+    AC_SUBST([MAGICK_LIBS], [${IMAGEMAGICK7_LIBS}])
 	  AC_DEFINE(HAVE_IMAGEMAGICK7, 1, [ImageMagick version 7 or higher is available.])
 	 ],[]
 	)
@@ -106,6 +110,8 @@ else
 	 [HAVE_MAGICK=yes
 	  HAVE_IMAGEMAGICK=yes
 	  MAGICK_LIBRARY=ImageMagick
+    AC_SUBST([MAGICK_CFLAGS], [${IMAGEMAGICK_CFLAGS}])
+    AC_SUBST([MAGICK_LIBS], [${IMAGEMAGICK_LIBS}])
 	  AC_DEFINE(HAVE_IMAGEMAGICK, 1, [ImageMagick library is available.])
 	 ],
 	AC_MSG_ERROR([Nor GraphicsMagick neither ImageMagick are available. Try recompiling with --without-magick.])
