--- publicsuffixlist-1.0.2.20240903/publicsuffixlist/__init__.py	2024-09-03 05:27:39.000000000 +0200
+++ publicsuffixlist-1.0.2.20240903/publicsuffixlist/__init__.py	2024-09-04 11:59:54.896862467 +0200
@@ -18,7 +18,7 @@
 
 PSLURL = "https://publicsuffix.org/list/public_suffix_list.dat"
 
-PSLFILE = os.path.join(os.path.dirname(__file__), "public_suffix_list.dat")
+PSLFILE = "/usr/share/publicsuffix/public_suffix_list.dat"
 
 BytesTuple = Tuple[bytes, ...]
 ByteString = Union[bytes, bytearray]
