Index: node-git.2070d5b524/deps/npm/lib/help-search.js
===================================================================
--- node-git.2070d5b524.orig/deps/npm/lib/help-search.js
+++ node-git.2070d5b524/deps/npm/lib/help-search.js
@@ -21,7 +21,7 @@ const helpSearch = async args => {
   if (!args.length)
     throw usage
 
-  const docPath = path.resolve(__dirname, '..', 'docs/content')
+  const docPath = '/usr/share/doc/packages/nodejs';
 
   const files = await glob(`${docPath}/*/*.md`)
   const data = await readFiles(files)
Index: node-git.2070d5b524/deps/npm/lib/npm.js
===================================================================
--- node-git.2070d5b524.orig/deps/npm/lib/npm.js
+++ node-git.2070d5b524/deps/npm/lib/npm.js
@@ -231,7 +231,13 @@ const npm = module.exports = new class e
   }
 
   get globalPrefix () {
-    return this.config.globalPrefix
+    let prefix = this.config.globalPrefix
+
+    // don't poop all over distro territory - use /usr/local instead
+    if (prefix === '/usr')
+      return '/usr/local'
+
+    return prefix;
   }
 
   set globalPrefix (r) {
