diff -Nur antimicrox-3.2.2.orig/CMakeLists.txt antimicrox-3.2.2/CMakeLists.txt
--- antimicrox-3.2.2.orig/CMakeLists.txt	2022-02-24 15:01:41.000000000 +0100
+++ antimicrox-3.2.2/CMakeLists.txt	2022-02-25 18:27:31.537083011 +0100
@@ -492,6 +492,7 @@
     if(WITH_X11)
         LIST(APPEND X11_LIBS ${X11_X11_LIB})
         LIST(APPEND X11_LIBS ${X11_Xi_LIB})
+        LIST(APPEND LIBS -lxcb)
     endif(WITH_X11)
 
     if(WITH_XTEST)
