From bb41035870ccca27a4eee8b07c0c6c2ec9c0ce38 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Wed, 19 Jul 2023 09:52:27 +0200
Subject: [PATCH] build: resolve Permission Denied error during link
References: https://github.com/intel/thunderbolt-utils/pull/4

gcc -g -Wall -W -O2 -o /usr/bin/lstbt lstbt.o lstbt_t.o lstbt_r.o lstbt_v.o router.o adapter.o helpers.o ../utils.o -lm
/usr/lib64/gcc/x86_64-suse-linux/13/../../../../x86_64-suse-linux/bin/ld: cannot open output file /usr/bin/lstbt: Permission denied
collect2: error: ld returned 1 exit status
---
 lib/Makefile | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/lib/Makefile b/lib/Makefile
index d29f0e0..a40b72b 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -5,7 +5,7 @@
 # Copyright (C) 2023 Rajat Khandelwal <rajat.khandelwal@intel.com>
 # Copyright (C) 2023 Intel Corporation
 
-LIBTBT_EXEC = /usr/bin/lstbt
+LIBTBT_EXEC = lstbt
 
 CC = gcc
 RM = rm -f
@@ -26,4 +26,3 @@ $(LIBTBT_EXEC): $(O_FILES)
 
 clean:
 	-$(RM) $(LIBTBT_EXEC) $(O_FILES)
-
-- 
2.41.0

