Index: django-pyscss-2.0.2/testproject/testproject/urls.py
===================================================================
--- django-pyscss-2.0.2.orig/testproject/testproject/urls.py
+++ django-pyscss-2.0.2/testproject/testproject/urls.py
@@ -8,5 +8,5 @@ urlpatterns = [
     # url(r'^$', 'testproject.views.home', name='home'),
     # url(r'^blog/', include('blog.urls')),
 
-    url(r'^admin/', include(admin.site.urls)),
+    url(r'^admin/', admin.site.urls),
 ]
