From 82a527a2791a5b8ca9c1d5c53e7817c2222064da Mon Sep 17 00:00:00 2001
From: Jeremy Epstein <jazepstein@gmail.com>
Date: Tue, 29 Jun 2021 19:18:43 +1000
Subject: [PATCH] Thread.is_alive as isAlive removed in Python 3.9

---
 socketpool/backend_thread.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/socketpool/backend_thread.py
+++ b/socketpool/backend_thread.py
@@ -51,7 +51,7 @@ class ConnectionReaper(threading.Thread)
             self.pool.murder_connections()
 
     def ensure_started(self):
-        if not self.running and not self.isAlive():
+        if not self.running and not self.is_alive():
             self.start()
 
 
