Index: sc-7.16/vi.c
===================================================================
--- sc-7.16.orig/vi.c
+++ sc-7.16/vi.c
@@ -42,7 +42,7 @@ void gotobottom();
 
 #define istext(a) (isalnum(a) || ((a) == '_'))
 
-#define bool	int
+#define boolean	int
 #define true	1
 #define false	0
 
@@ -74,7 +74,7 @@ static void replace_mode();
 static void restore_it();
 static void savedot(int c);
 static void save_hist();
-static void search_again(bool reverse);
+static void search_again(boolean reverse);
 static void search_hist();
 static void search_mode(char sind);
 static void stop_edit();
@@ -1565,7 +1565,7 @@ search_hist()
 }
 
 static void
-search_again(bool reverse)
+search_again(boolean reverse)
 {
     int prev_match;
     int found_it;
