--- a/makefiles/Makefile.linux
+++ b/makefiles/Makefile.linux
@@ -1,5 +1,5 @@
 # $Id: Makefile.linux,v 1.14 2006/10/26 02:20:16 tringali Exp $
-CC=cc
+CC=gcc
 AR=ar
 
 # For editres, add -DEDITRES to CFLAGS and -lXmu to LIBS
@@ -17,7 +17,7 @@
 # To test if the Motif library exports the runtime version
 # add -DHAVE__XMVERSIONSTRING to CFLAGS
 #
-CFLAGS=-O -I/usr/X11R6/include -DUSE_DIRENT -DUSE_LPR_PRINT_CMD -ansi -U__STRICT_ANSI__
+CFLAGS=-I/usr/X11R6/include $(RPM_OPT_FLAGS) -DUSE_DIRENT -DUSE_LPR_PRINT_CMD -ansi -U__STRICT_ANSI__
 
 ARFLAGS=-urs
 
