Index: postgres_exporter-0.4.7/magefile.go
===================================================================
--- postgres_exporter-0.4.7.orig/magefile.go
+++ postgres_exporter-0.4.7/magefile.go
@@ -97,6 +97,9 @@ func (p *Platform) ReleaseBase() string
 var platforms []Platform = []Platform{
 	{"linux", "amd64", ""},
 	{"linux", "386", ""},
+	{"linux", "s390x", ""},
+	{"linux", "ppc64le", ""},
+	{"linux", "arm64", ""},
 	{"darwin", "amd64", ""},
 	{"darwin", "386", ""},
 	{"windows", "amd64", ".exe"},
