Index: html5lib-1.1/html5lib/tests/tokenizer.py
===================================================================
--- html5lib-1.1.orig/html5lib/tests/tokenizer.py
+++ html5lib-1.1/html5lib/tests/tokenizer.py
@@ -246,7 +246,7 @@ class TokenizerTest(pytest.Item):
     def repr_failure(self, excinfo):
         traceback = excinfo.traceback
         ntraceback = traceback.cut(path=__file__)
-        excinfo.traceback = ntraceback.filter()
+        excinfo.traceback = ntraceback.filter(excinfo)
 
         return excinfo.getrepr(funcargs=True,
                                showlocals=False,
Index: html5lib-1.1/html5lib/tests/tree_construction.py
===================================================================
--- html5lib-1.1.orig/html5lib/tests/tree_construction.py
+++ html5lib-1.1/html5lib/tests/tree_construction.py
@@ -135,7 +135,7 @@ class ParserTest(pytest.Item):
     def repr_failure(self, excinfo):
         traceback = excinfo.traceback
         ntraceback = traceback.cut(path=__file__)
-        excinfo.traceback = ntraceback.filter()
+        excinfo.traceback = ntraceback.filter(excinfo)
 
         return excinfo.getrepr(funcargs=True,
                                showlocals=False,
