Index: atop-2.8.0/atop-rotate.service
===================================================================
--- atop-2.8.0.orig/atop-rotate.service
+++ atop-2.8.0/atop-rotate.service
@@ -3,5 +3,18 @@ Description=Restart atop daemon to rotat
 Documentation=man:atop(1)
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+PrivateDevices=true
+ProtectHostname=true
+ProtectClock=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 Type=oneshot
 ExecStart=/usr/bin/systemctl try-restart atop.service
