Index: emacs-25.2/configure.ac
===================================================================
---
 emacs-26.1/configure.ac |    2 +-
 emacs-26.1/src/image.c  |   12 ++++++++++++
 2 files changed, 13 insertions(+), 1 deletion(-)

--- emacs-27.1/configure.ac
+++ emacs-27.1/configure.ac	2020-08-11 09:59:04.349950601 +0000
@@ -2603,7 +2603,7 @@ if test "${HAVE_X11}" = "yes" || test "$
     else
        ## 6.3.5 is the earliest version known to work; see Bug#17339.
        ## 6.8.2 makes Emacs crash; see Bug#13867.
-       EMACS_CHECK_MODULES([IMAGEMAGICK], [Wand >= 6.3.5 Wand != 6.8.2])
+       EMACS_CHECK_MODULES([IMAGEMAGICK], [MagickWand >= 6.3.5 MagickWand != 6.8.2])
     fi
 
     if test $HAVE_IMAGEMAGICK = yes; then
--- emacs-27.1/src/image.c
+++ emacs-27.1/src/image.c	2020-08-11 09:49:35.500181432 +0000
@@ -9005,7 +9005,11 @@ imagemagick_compute_animated_image (Magi
       PixelWand **source, **dest;
       size_t source_width, source_height;
       ssize_t source_left, source_top;
+#if MagickLibVersion >= 0x700
+      PixelInfo pixel;  
+#else
       MagickPixelPacket pixel;
+#endif
       DisposeType dispose;
       ptrdiff_t lines = 0;
 
@@ -9070,7 +9074,11 @@ imagemagick_compute_animated_image (Magi
 	      if (dispose == BackgroundDispose || PixelGetAlpha (source[x]))
 		{
 		  PixelGetMagickColor (source[x], &pixel);
+#if MagickLibVersion >= 0x700
+		  PixelSetPixelColor (dest[x + source_left], &pixel);
+#else
 		  PixelSetMagickColor (dest[x + source_left], &pixel);
+#endif
 		}
 	    }
 	  PixelSyncIterator (dest_iterator);
@@ -9115,7 +9123,11 @@ imagemagick_load_image (struct frame *f,
   MagickWand *image_wand;
   PixelIterator *iterator;
   PixelWand **pixels, *bg_wand = NULL;
+#if MagickLibVersion >= 0x700
+  PixelInfo pixel;
+#else
   MagickPixelPacket  pixel;
+#endif
   Lisp_Object image;
   Lisp_Object value;
   Lisp_Object crop;
