--- ditrace/ditrace.c
+++ ditrace/ditrace.c
@@ -394,7 +394,7 @@
 int ditrace_main (int argc, char** argv) {
   diva_cmd_line_parameter_t* e;
 
-	dprintf = do_dprintf;
+	my_dprintf = do_dprintf;
 	VisualStream = stdout;
 	DebugStream  = VisualStream;
   OutStream    = VisualStream;
--- divactrl.c
+++ divactrl.c
@@ -77,7 +77,7 @@
 {
   /* dummy debug function */
 }
-DIVA_DI_PRINTF dprintf = no_printf;
+DIVA_DI_PRINTF my_dprintf = no_printf;
 
 char *cmd;
 
--- include/kernel/debuglib.h
+++ include/kernel/debuglib.h
@@ -218,7 +218,7 @@
 #define DbgDeregister() do { } while(0)
 #define DbgSetLevel(mask) do { } while(0)
 #else
-extern DIVA_DI_PRINTF dprintf;
+extern DIVA_DI_PRINTF my_dprintf;
 extern int  DbgRegister (char *drvName, char *drvTag, unsigned long dbgMask) ;
 extern void DbgDeregister (void) ;
 extern void DbgSetLevel (unsigned long dbgMask) ;
--- include/kernel/di_dbg.h
+++ include/kernel/di_dbg.h
@@ -32,6 +32,6 @@
 #define dbug(a)
 #endif
 #if !defined USE_EXTENDED_DEBUGS
-extern void (*dprintf)(char*, ...);
+extern void (*my_dprintf)(char*, ...);
 #endif
 #endif
--- include/platform.h
+++ include/platform.h
@@ -138,10 +138,10 @@
 typedef struct _ISDN_ADAPTER* PISDN_ADAPTER;
 
 typedef void (* DIVA_DI_PRINTF) (char *, ...);
-extern DIVA_DI_PRINTF dprintf;
+extern DIVA_DI_PRINTF my_dprintf;
 
 #ifndef __NO_STANDALONE_DBG_INC__
-#define DBG_TEST(func,args) dprintf args ;
+#define DBG_TEST(func,args) my_dprintf args ;
 
 #define DBG_LOG(args)  DBG_TEST(LOG, args)
 #define DBG_FTL(args)  DBG_TEST(FTL, args)
--- load/divaload.c
+++ load/divaload.c
@@ -566,7 +566,7 @@
 	diva_cmd_line_parameter_t* e;
 	char property[256], serial[64];
 
-	dprintf = do_dprintf;
+	my_dprintf = do_dprintf;
 	VisualStream = stdout;
 
 	diva_process_cmd_line (argc, argv);
