Index: b/capifax/configure.in
===================================================================
--- a/capifax/configure.in
+++ b/capifax/configure.in
@@ -41,7 +41,7 @@ CS_TEST_ALERT
 CS_TEST_GLOBALCONFIG
 
 AC_SUBST(INSTALL)
 AC_SUBST(CONFIG_SBINDIR)
 AC_SUBST(CONFIG_MANDIR)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_OUTPUT(Makefile)
Index: b/capiinfo/configure.in
===================================================================
--- a/capiinfo/configure.in
+++ b/capiinfo/configure.in
@@ -40,7 +40,7 @@ AC_ARG_WITH(man,
 AC_SUBST(INSTALL)
 AC_SUBST(I4LCONFDIR)
 AC_SUBST(I4LVERSION)
 AC_SUBST(CONFIG_SBINDIR)
 AC_SUBST(CONFIG_MANDIR)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_OUTPUT(Makefile)
Index: b/capiinit/configure.in
===================================================================
--- a/capiinit/configure.in
+++ b/capiinit/configure.in
@@ -22,7 +22,7 @@ dnl Checks for typedefs, structures, and
 
 AC_SUBST(INSTALL)
 AC_SUBST(CONFIG_SBINDIR)
 AC_SUBST(CONFIG_MANDIR)
 AC_SUBST(CONFIG_FIRMWAREDIR)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_OUTPUT(Makefile)
