Index: aranym-1.1.0/configure
===================================================================
--- aranym-1.1.0.orig/configure
+++ aranym-1.1.0/configure
@@ -12354,6 +12354,8 @@ struct possibility table [] =
   C(-5.22995989424860458374e+10)  /* IBMHEXFP - s/390 format, EBCDIC */
 };
 EOF
+ac_save_CFLAGS=$CFLAGS
+CFLAGS="$CFLAGS -fno-lto"
 if { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$ac_compile\""; } >&5
   (eval $ac_compile) 2>&5
   ac_status=$?
@@ -12389,6 +12391,7 @@ if { { eval echo "\"\$as_me\":${as_linen
 else
   as_fn_error $? "compile failed" "$LINENO" 5
 fi
+CFLAGS=$ac_save_CFLAGS
 rm -rf conftest*
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
Index: aranym-1.1.0/m4/float.m4
===================================================================
--- aranym-1.1.0.orig/m4/float.m4
+++ aranym-1.1.0/m4/float.m4
@@ -23,6 +23,8 @@ cat > conftest.$ac_ext <<EOF
 #include "confdefs.h"
 $1
 ]EOF
+ac_save_CFLAGS=$CFLAGS
+CFLAGS="$CFLAGS -fno-lto"
 if AC_TRY_EVAL(ac_compile); then
   od -c conftest.o |
     sed ['s/^[0-7]*[ 	]*/ /
@@ -39,6 +41,7 @@ ifelse($3, , , else
   $3
 )dnl
 fi
+CFLAGS=$ac_save_CFLAGS
 rm -rf conftest*
 AC_LANG_RESTORE])
 
