diff --git a/helpdev/__init__.py b/helpdev/__init__.py
index ca739c8ee30e8899a33ef98a60a59d458b376261..41bf220cf890fae1eb1aae557c75e8d404dd9a13 100644
--- a/helpdev/__init__.py
+++ b/helpdev/__init__.py
@@ -26,7 +26,11 @@ import sys
 import time
 import warnings
 
-import importlib_metadata
+try:
+    import importlib.metadata as importlib_metadata
+except ImportError:
+    # python < 3.8
+    import importlib_metadata
 
 if sys.version_info >= (3, 4):
     import importlib.util
