From d3b136268a7017cf1db298443024ec3477e1ca2d Mon Sep 17 00:00:00 2001
From: Egbert Eich <eich@suse.com>
Date: Wed, 30 Nov 2022 20:14:53 +0100
Subject: [PATCH] Handle s390 correctly

Signed-off-by: Egbert Eich <eich@suse.com>
Signed-off-by: Egbert Eich <eich@suse.de>
---
 c_check | 2 +-
 ctest.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/c_check b/c_check
index 9801b8778..05ef8df9f 100755
--- a/c_check
+++ b/c_check
@@ -14,7 +14,7 @@ case "$hostarch" in
     arm*) [ "$hostarch" = "arm64" ] || hostarch='arm' ;;
     aarch64) hostarch=arm64 ;;
     powerpc*|ppc*) hostarch=power ;;
-    s390x) hostarch=zarch ;;
+    s390*) hostarch=zarch ;;
 esac
 
 makefile="$1"
diff --git a/ctest.c b/ctest.c
index cbc15326f..3ac5c335f 100644
--- a/ctest.c
+++ b/ctest.c
@@ -121,7 +121,7 @@ ARCH_X86_64
 ARCH_POWER
 #endif
 
-#if defined(__s390x__) || defined(__zarch__)
+#if defined(__s390x__) || defined(__s390__) || defined(__zarch__)
 ARCH_ZARCH
 #endif
 
-- 
2.46.1

