diff --git a/test/test_autopep8.py b/test/test_autopep8.py
index 5f4dbed8..c6319cfa 100755
--- a/test/test_autopep8.py
+++ b/test/test_autopep8.py
@@ -40,13 +40,14 @@
 )
 
 if 'AUTOPEP8_COVERAGE' in os.environ and int(os.environ['AUTOPEP8_COVERAGE']):
-    AUTOPEP8_CMD_TUPLE = ('coverage', 'run', '--branch', '--parallel',
+    AUTOPEP8_CMD_TUPLE = (sys.executable, '-Wignore::DeprecationWarning',
+                          '-m', 'coverage', 'run', '--branch', '--parallel',
                           '--omit=*/site-packages/*',
                           os.path.join(ROOT_DIR, 'autopep8.py'),)
 else:
     # We need to specify the executable to make sure the correct Python
     # interpreter gets used.
-    AUTOPEP8_CMD_TUPLE = (sys.executable,
+    AUTOPEP8_CMD_TUPLE = (sys.executable, '-Wignore::DeprecationWarning',
                           os.path.join(ROOT_DIR,
                                        'autopep8.py'),)  # pragma: no cover
 
