---
 tests/test_ansi2html.py |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

Index: ansi2html-1.9.5/tests/test_ansi2html.py
===================================================================
--- ansi2html-1.9.5.orig/tests/test_ansi2html.py
+++ ansi2html-1.9.5/tests/test_ansi2html.py
@@ -50,6 +50,7 @@ _here = dirname(abspath(__file__))
 
 class TestAnsi2HTML:
     maxDiff = None
+    cmd_exe = "ansi2html-%d.%d" % sys.version_info[:2]
 
     def test_linkify(self) -> None:
         ansi = "http://threebean.org#foobar"
@@ -486,7 +487,7 @@ class TestAnsi2HTML:
         assert target in html
 
     def test_command_script(self) -> None:
-        result = run(["ansi2html", "--version"], check=True)
+        result = run([self.cmd_exe, "--version"], check=True)
         assert result.returncode == 0
 
     def test_command_input_output_encoding(self) -> None:
@@ -501,7 +502,7 @@ class TestAnsi2HTML:
 
         with Popen(
             [
-                "ansi2html",
+                self.cmd_exe,
                 "--inline",
                 f"--input-encoding={input_encoding}",
                 f"--output-encoding={output_encoding}",
