Index: grub-2.02~beta2/grub-core/osdep/linux/getroot.c
===================================================================
--- grub-2.02~beta2.orig/grub-core/osdep/linux/getroot.c
+++ grub-2.02~beta2/grub-core/osdep/linux/getroot.c
@@ -713,6 +713,10 @@ grub_util_part_to_disk (const char *os_d
   if (! realpath (os_dev, path))
     return NULL;
 
+#ifdef __s390x__
+  return path;
+#endif
+
   if (strncmp ("/dev/", path, 5) == 0)
     {
       char *p = path + 5;
