Index: pytest-codestyle-2.0.0/pytest_pycodestyle.py
===================================================================
--- pytest-codestyle-2.0.0.orig/pytest_pycodestyle.py
+++ pytest-codestyle-2.0.0/pytest_pycodestyle.py
@@ -27,7 +27,7 @@ class Item(pytest.Item, pytest.File):
     CACHE_KEY = 'pycodestyle/mtimes'
 
     def __init__(self, path, parent, options):
-        super().__init__(path, parent)
+        super(Item, self).__init__(path, parent)
         self.add_marker('pycodestyle')
         self.options = options
         # https://github.com/pytest-dev/pytest/blob/92d6a0500b9f528a9adcd6bbcda46ebf9b6baf03/src/_pytest/nodes.py#L380
