diff -Nur orig-gnome-maps-41.2/data/meson.build gnome-maps-41.2/data/meson.build
--- orig-gnome-maps-41.2/data/meson.build	2022-02-08 18:52:33.109927983 +0100
+++ gnome-maps-41.2/data/meson.build	2022-02-08 18:53:13.382375495 +0100
@@ -33,7 +33,6 @@
 appdata = app_id + '.appdata.xml'
 
 appdata_file = i18n.merge_file(
-	appdata,
 	input: appdata + '.in',
 	output: appdata,
 	po_dir: po_dir,
@@ -77,4 +76,4 @@
 	)
 endif
 
-subdir('icons')
\ Intet linjeskift i slutten av fila
+subdir('icons')
