--- orig-gnome-maps-43.0/meson.build	2022-09-17 22:44:30.000000000 +0200
+++ gnome-maps-43.0/meson.build	2022-09-18 11:54:43.164613308 +0200
@@ -42,7 +42,7 @@
 geocodeglib = dependency('geocode-glib-2.0', version: '>= 3.15.2')
 
 libmaps_deps = [
-	dependency('shumate-1.0', version: '>= 1.0.0.beta'),
+	dependency('shumate-1.0', version: '>= 1.0.0'),
 	dependency('libxml-2.0'),
 	dependency('rest-1.0')
 ]
