--- plexus-ant-factory/src/main/java/org/codehaus/plexus/component/factory/ant/AntComponentFactory.java	2024-05-15 19:54:52.747188171 +0200
+++ plexus-ant-factory/src/main/java/org/codehaus/plexus/component/factory/ant/AntComponentFactory.java	2024-05-15 19:55:53.817620623 +0200
@@ -21,7 +21,7 @@
         }
         catch ( IOException e )
         {
-            throw new ComponentInstantiationException( "Failed to extract Ant script for: " + componentDescriptor.getHumanReadableKey(), e );
+            throw new ComponentInstantiationException( "Failed to extract Ant script for: " + componentDescriptor.toString(), e );
         }
     }
 
--- plexus-ant-factory/src/main/java/org/codehaus/plexus/component/factory/ant/AntScriptInvoker.java	2024-05-15 19:54:52.747188171 +0200
+++ plexus-ant-factory/src/main/java/org/codehaus/plexus/component/factory/ant/AntScriptInvoker.java	2024-05-15 19:55:26.287425676 +0200
@@ -118,7 +118,7 @@
     {
         if ( !descriptor.getRequirements().contains( rd ) )
         {
-            throw new ComponentConfigurationException( "Requirement: " + rd.getHumanReadableKey()
+            throw new ComponentConfigurationException( rd.toString()
                 + " is not listed in this component's descriptor." );
         }
 
