Index: ujson-2.0.1/setup.py
===================================================================
--- ujson-2.0.1.orig/setup.py	2020-03-08 09:02:52.000000000 +0100
+++ ujson-2.0.1/setup.py	2020-03-10 15:01:45.077782242 +0100
@@ -33,9 +33,9 @@ module1 = Extension(
         "./lib/ultrajsonenc.c",
         "./lib/ultrajsondec.c",
     ],
-    include_dirs=["./python", "./lib", "./deps/double-conversion/double-conversion"],
+    include_dirs=["./python", "./lib", "/usr/include/double-conversion"],
     extra_compile_args=["-D_GNU_SOURCE"],
-    extra_link_args=["-lstdc++", "-lm"],
+    extra_link_args=["-lstdc++", "-lm", "-ldouble-conversion"],
 )
 
 
