Index: ujson-4.3.0/setup.py
===================================================================
--- ujson-4.3.0.orig/setup.py
+++ ujson-4.3.0/setup.py
@@ -15,9 +15,9 @@ module1 = Extension(
         "./lib/ultrajsonenc.c",
         "./lib/ultrajsondec.c",
     ],
-    include_dirs=["./python", "./lib", "./deps/double-conversion/double-conversion"],
+    include_dirs=["./python", "./lib", "/usr/include/double-conversion"],
     extra_compile_args=["-D_GNU_SOURCE"],
-    extra_link_args=["-lstdc++", "-lm"],
+    extra_link_args=["-lstdc++", "-lm", "-ldouble-conversion"],
 )
 
 with open("python/version_template.h") as f:
