http://bugzilla.suse.com/show_bug.cgi?id=1047218


Index: ddskk-20150816/SKK-MK
===================================================================
--- ddskk-20150816.orig/SKK-MK
+++ ddskk-20150816/SKK-MK
@@ -427,8 +427,7 @@ This version of SKK requires MULE featur
 				 (point))))
       (when (search-forward ";;; skk-setup.el --- initial setup for SKK")
 	(forward-line 1)
-	(insert ";; This file was generated automatically by SKK-MK at "
-		(current-time-string)
+	(insert ";; This file was generated automatically by SKK-MK "
 		"\n"))
       (while spec
 	(goto-char (point-min))
@@ -911,8 +910,7 @@ SKK tutorials:
       (insert "\
 ;;; skk-autoloads.el --- autoload settings for SKK  -*- coding: iso-2022-jp -*-
 
-;; This file was generated automatically by SKK-MK at "
-	      (current-time-string)
+;; This file was generated automatically by SKK-MK "
 	      ".
 
 ;; This file is part of Daredevil SKK.
