From 7064f5f0b8094877459f4b6ed2eff72627a19750 Mon Sep 17 00:00:00 2001
From: PerchunPak <git@perchun.it>
Date: Sun, 28 Apr 2024 17:42:10 +0200
Subject: [PATCH] Keep up with changes in geventhttpclient

---
 dulwich/contrib/swift.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: dulwich-dulwich-0.22.1/dulwich/contrib/swift.py
===================================================================
--- dulwich-dulwich-0.22.1.orig/dulwich/contrib/swift.py
+++ dulwich-dulwich-0.22.1/dulwich/contrib/swift.py
@@ -257,7 +257,7 @@ class SwiftConnector:
         if ret.status_code < 200 or ret.status_code >= 300:
             raise SwiftException(
                 "AUTH v1.0 request failed on "
-                + f"{str(auth_httpclient.get_base_url()) + path} with error code {ret.status_code} ({ret.items()!s})"
+                + f"{self.auth_url} with error code {ret.status_code} ({ret.items()!s})"
             )
         storage_url = ret["X-Storage-Url"]
         token = ret["X-Auth-Token"]
Index: dulwich-dulwich-0.22.1/tests/test_greenthreads.py
===================================================================
--- dulwich-dulwich-0.22.1.orig/tests/test_greenthreads.py
+++ dulwich-dulwich-0.22.1/tests/test_greenthreads.py
@@ -35,7 +35,7 @@ except ImportError:
     gevent_support = False
 
 if gevent_support:
-    from ..greenthreads import GreenThreadsMissingObjectFinder
+    from dulwich.greenthreads import GreenThreadsMissingObjectFinder
 
 skipmsg = "Gevent library is not installed"
 
