Index: vim-8.2.2411/src/po/Make_all.mak
===================================================================
--- vim-8.2.2411.orig/src/po/Make_all.mak
+++ vim-8.2.2411/src/po/Make_all.mak
@@ -24,7 +24,6 @@ LANGUAGES = \
 		lv \
 		nb \
 		nl \
-		no \
 		pl \
 		pl.UTF-8 \
 		pl.cp1250 \
@@ -109,7 +108,6 @@ MOFILES = \
 		lv.mo \
 		nb.mo \
 		nl.mo \
-		no.mo \
 		pl.mo \
 		pt_BR.mo \
 		ru.mo \
@@ -159,7 +157,6 @@ CHECKFILES = \
 		lv.ck \
 		nb.ck \
 		nl.ck \
-		no.ck \
 		pl.UTF-8.ck \
 		pl.ck \
 		pl.cp1250.ck \
