Index: vim-8.0.1568/runtime/defaults.vim
===================================================================
--- vim-8.0.1568.orig/runtime/defaults.vim
+++ vim-8.0.1568/runtime/defaults.vim
@@ -75,9 +75,11 @@ inoremap <C-U> <C-G>u<C-U>
 
 " In many terminal emulators the mouse works just fine.  By enabling it you
 " can position the cursor, Visually select and scroll with the mouse.
-if has('mouse')
-  set mouse=a
-endif
+" disabled in (open)SUSE to make copy&paste work as most people
+" expect it bsc#1079185
+"if has('mouse')
+"  set mouse=a
+"endif
 
 " Switch syntax highlighting on when the terminal has colors or when using the
 " GUI (which always has colors).
