Index: vim-9.1.1918/runtime/filetype.vim
===================================================================
--- vim-9.1.1918.orig/runtime/filetype.vim
+++ vim-9.1.1918/runtime/filetype.vim
@@ -124,6 +124,10 @@ au BufNewFile,BufRead *.demo,*.dm{1,2,3,
 " *.[sS], *.[aA] usually Assembly - GNU
 au BufNewFile,BufRead *.asm,*.[sS],*.[aA],*.mac,*.lst	call dist#ft#FTasm()
 
+" AppArmor
+au BufNewFile,BufRead */etc/apparmor.d/*       setf apparmor
+au BufNewFile,BufRead */etc/apparmor/profiles/*        setf apparmor
+
 " BASIC or Visual Basic
 au BufNewFile,BufRead *.bas			call dist#ft#FTbas()
 au BufNewFile,BufRead *.bi,*.bm			call dist#ft#FTbas()
