From: Luigi Baldoni <aloisio@gmx.com>
Date: 2020-09-25 09:46:36 +0200
Subject: Use system libraries when possible

Praat includes a number of external libraries that then builds
internally.
Some (gsl, glpk) can be replaced by system ones, other like espeak,
and FLAC are too heavily patched for the program to be compatible
with upstream ones.
Praat also supplies an unstable upstream portaudio version that's
not recommendable for distro inclusion.

Index: praat-6.1.22/melder/NUMlinprog.cpp
===================================================================
--- praat-6.1.22.orig/melder/NUMlinprog.cpp
+++ praat-6.1.22/melder/NUMlinprog.cpp
@@ -17,7 +17,7 @@
  */
 
 #include "melder.h"
-#include "../external/glpk/glpk.h"
+#include <glpk.h>
 
 struct structNUMlinprog {
 	glp_prob *linearProgram;
Index: praat-6.1.22/melder/NUMspecfunc.cpp
===================================================================
--- praat-6.1.22.orig/melder/NUMspecfunc.cpp
+++ praat-6.1.22/melder/NUMspecfunc.cpp
@@ -31,10 +31,10 @@
 #include "melder.h"
 #include "../dwsys/NUM2.h"   // NUMridders
 
-#include "../external/gsl/gsl_errno.h"
-#include "../external/gsl/gsl_sf_bessel.h"
-#include "../external/gsl/gsl_sf_gamma.h"
-#include "../external/gsl/gsl_sf_erf.h"
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_sf_bessel.h>
+#include <gsl/gsl_sf_gamma.h>
+#include <gsl/gsl_sf_erf.h>
 
 double NUMlnGamma (double x) {
 	gsl_sf_result result;
Index: praat-6.1.22/melder/melder.cpp
===================================================================
--- praat-6.1.22.orig/melder/melder.cpp
+++ praat-6.1.22/melder/melder.cpp
@@ -24,7 +24,7 @@
 #include "melder_enums.h"
 
 #include "../dwsys/NUMmachar.h"
-#include "../external/gsl/gsl_errno.h"
+#include <gsl/gsl_errno.h>
 #ifdef macintosh
 	#include <Carbon/Carbon.h>   // Gestalt
 #endif
Index: praat-6.1.22/dwsys/Makefile
===================================================================
--- praat-6.1.22.orig/dwsys/Makefile
+++ praat-6.1.22/dwsys/Makefile
@@ -4,7 +4,7 @@
 
 include ../makefile.defs
 
-CPPFLAGS = -I ../melder -I ../stat -I ../sys -I ../fon -I ../external/gsl -I ../external/clapack -I ../kar
+CPPFLAGS = -I ../melder -I ../stat -I ../sys -I ../fon -I /usr/include/gsl -I ../external/clapack -I ../kar
 
 all: libdwsys.a
 
@@ -37,5 +37,5 @@ libdwsys.a: $(OBJECTS) NUMmachar.o
 	$(AR) cq libdwsys.a $(OBJECTS)
 	$(RANLIB) libdwsys.a
 
-$(OBJECTS): *.h ../melder/*.h ../stat/*.h ../sys/*.h ../fon/*.h ../external/gsl/*.h ../dwsys/*.h ../kar/*.h
+$(OBJECTS): *.h ../melder/*.h ../stat/*.h ../sys/*.h ../fon/*.h /usr/include/gsl/*.h ../dwsys/*.h ../kar/*.h
 
Index: praat-6.1.22/Makefile
===================================================================
--- praat-6.1.22.orig/Makefile
+++ praat-6.1.22/Makefile
@@ -14,8 +14,6 @@ include makefile.defs
 # Makes the Praat executable in the source directory.
 all:
 	$(MAKE) -C external/clapack
-	$(MAKE) -C external/gsl
-	$(MAKE) -C external/glpk
 	$(MAKE) -C external/mp3
 	$(MAKE) -C external/flac
 	$(MAKE) -C external/portaudio
@@ -42,9 +40,8 @@ all:
 		external/espeak/libespeak.a \
 		external/portaudio/libportaudio.a \
 		external/flac/libflac.a external/mp3/libmp3.a \
-		external/glpk/libglpk.a \
+		`pkg-config --libs libpulse` -lglpk `pkg-config --libs gsl` \
 		external/clapack/libclapack.a \
-		external/gsl/libgsl.a \
 		$(LIBS)
 
 clean:
