Index: praat-6.1.41/melder/melder_search.cpp
===================================================================
--- praat-6.1.41.orig/melder/melder_search.cpp
+++ praat-6.1.41/melder/melder_search.cpp
@@ -145,7 +145,7 @@ bool Melder_stringMatchesCriterion (cons
 			return !! place;
 		}
 	}
-	//return false;   // should not occur
+	return false;   // should not occur
 }
 
 /* End of file melder_search.cpp */
