Index: VirtualGL-2.5.2/server/CMakeLists.txt
===================================================================
--- VirtualGL-2.5.2.orig/server/CMakeLists.txt
+++ VirtualGL-2.5.2/server/CMakeLists.txt
@@ -114,12 +114,12 @@ foreach(fakerlib ${VGL_FAKER_NAME} ${VGL
 endforeach()
 
 add_library(${VGL_DLFAKER_NAME} SHARED dlfaker.c)
-target_link_libraries(${VGL_DLFAKER_NAME} ${LIBDL})
+target_link_libraries(${VGL_DLFAKER_NAME} ${LIBDL} vglfaker)
 install(TARGETS ${VGL_DLFAKER_NAME} DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 add_library(${VGL_GEFAKER_NAME} SHARED gefaker.c)
 set_target_properties(${VGL_GEFAKER_NAME} PROPERTIES LINK_FLAGS "${MINUSZ}defs")
-target_link_libraries(${VGL_GEFAKER_NAME} ${LIBDL})
+target_link_libraries(${VGL_GEFAKER_NAME} ${LIBDL} vglfaker)
 install(TARGETS ${VGL_GEFAKER_NAME} DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 set(HEADERS ../common/rr.h rrtransport.h)
@@ -183,13 +183,14 @@ if(VGL_USESSL AND CMAKE_SYSTEM_NAME STRE
 	set_target_properties(vgltrans_test PROPERTIES LINK_FLAGS "-Wl,-Bsymbolic")
 endif()
 target_link_libraries(vgltrans_test vglcommon ${FBXFAKERLIB} ${TJPEG_LIBRARY}
-	vglsocket)
+	vglsocket vglfaker)
 if(CMAKE_SYSTEM_NAME STREQUAL "SunOS" AND GNUCXX)
 	target_link_libraries(vgltrans_test stdc++)
 endif()
 
 add_library(vgltrans_test2 SHARED testplugin2.cpp X11Trans.cpp)
-target_link_libraries(vgltrans_test2 vglcommon ${FBXFAKERLIB} ${TJPEG_LIBRARY})
+target_link_libraries(vgltrans_test2 vglcommon ${FBXFAKERLIB} ${TJPEG_LIBRARY}
+	vglfaker)
 if(CMAKE_SYSTEM_NAME STREQUAL "SunOS" AND GNUCXX)
 	target_link_libraries(vgltrans_test2 stdc++)
 endif()
