From f2e4dfeb6f374fa891ceee911eb5a11b1b7bdf78 Mon Sep 17 00:00:00 2001
From: Dominik Schmidt <domme@tomahawk-player.org>
Date: Wed, 27 Aug 2014 19:05:24 +0200
Subject: [PATCH] Add cmake script to build host tool

---
 CMakeLists.txt | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..647bfea
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,23 @@
+include_directories(src)
+add_definitions(-DNO_STABS_SUPPORT)
+
+add_executable(dump_syms
+    src/tools/windows/dump_syms_dwarf/dump_syms.cc
+    src/common/pecoff/dump_symbols.cc
+    src/common/dwarf_cfi_to_module.cc
+    src/common/module.cc
+    src/common/pecoff/pecoffutils.cc
+    src/common/pecoff/pecoff_file_id.cc
+    src/common/stabs_to_module.cc
+    src/common/dwarf_cu_to_module.cc
+    src/common/language.cc
+    src/common/dwarf/dwarf2diehandler.cc
+    src/common/dwarf_line_to_module.cc
+    src/common/dwarf/dwarf2reader.cc
+    src/common/dwarf/bytereader.cc
+    src/common/pecoff/pecoff_file_id.cc
+)
+
+target_link_libraries(dump_syms
+    stdc++
+)
-- 
1.8.4.5

