Index: stdeb-0.10.0/stdeb/util.py
===================================================================
--- stdeb-0.10.0.orig/stdeb/util.py
+++ stdeb-0.10.0/stdeb/util.py
@@ -1214,7 +1214,10 @@ class DebianInfo:
         else:
             self.override_dh_python3 = ''
 
-        sequencer_options = ['--with '+','.join(sequencer_with)]
+        # These depend on the dh_python helpers in the Debian python packages
+        # https://salsa.debian.org/cpython-team/python-defaults
+        # sequencer_options = ['--with '+','.join(sequencer_with)]
+        sequencer_options = []
 
         if with_dh_virtualenv:
             if with_python2:
