From: Jesse Becker <hawson@gmail.com>
Date: Mon, 2 May 2016 21:53:36 -0400
Git-commit: 6887ef08f151ec7fc303b0d9c6cb15614d0f1ca9
References: bsc#1087487
Subject: [PATCH] Fixes #245

Acked-by: John Jolly <jjolly@suse.de>
---
 gmond/g25_config.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gmond/g25_config.c b/gmond/g25_config.c
index d295ab64..98721817 100644
--- a/gmond/g25_config.c
+++ b/gmond/g25_config.c
@@ -373,5 +373,6 @@ print_ganglia_25_config( char *path )
    dotconf_cleanup(configfile);
 
    print_config(path, &gmond_config);
+   fclose(fp);
    return 0;
 }
-- 
2.13.6

