Index: smartmontools-7.4/smartd.service.in
===================================================================
--- smartmontools-7.4.orig/smartd.service.in
+++ smartmontools-7.4/smartd.service.in
@@ -7,6 +7,16 @@ Documentation=man:smartd(8) man:smartd.c
 ConditionVirtualization=no
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+ProtectHostname=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 Type=notify
 EnvironmentFile=-/usr/local/etc/sysconfig/smartmontools
 ExecStart=/usr/local/sbin/smartd -n -q never $smartd_opts
