--- smartd.conf
+++ smartd.conf
@@ -14,6 +14,15 @@
 # You can usually identify which hard disks are on your system by
 # looking in /proc/ide and in /proc/scsi.
 
+# SUSE default:
+#   -d removable: Prevent error messages after disconnecting of
+#                 monitored removable discs.
+#   -s S/:        Run Short Self Test every day in the deep night.
+#                 (Takes several minutes.)
+#   -s L/:        Run Extended Self Test every first Sunday in the
+#                 month. (Start earlier, it could take tens of hours.)
+DEFAULT -d removable -s (S/../.././03|L/../(01|02|03|04|05|06|07)/7/01)
+
 # The word DEVICESCAN will cause any remaining lines in this
 # configuration file to be ignored: it tells smartd to scan for all
 # ATA and SCSI devices.  DEVICESCAN may be followed by any of the
