From: Andreas Stieger <Andreas.Stieger@gmx.de>
Date: Wed, 26 Jul 2023 21:53:09 +0200
Subject: object_paint_properties_sparse.h:258:12: error: no template named 'variant' in namespace 'std'

[12507s] In file included from ../third_party/blink/renderer/core/paint/object_paint_properties.cc:8:
[12507s] ../third_party/blink/renderer/core/paint/object_paint_properties_sparse.h:258:12: error: no template named 'variant' in namespace 'std'
[12507s]       std::variant<scoped_refptr<TransformPaintPropertyNode>,
[12507s]       ~~~~~^

Index: chromium-116.0.5845.42/third_party/blink/renderer/core/paint/object_paint_properties_sparse.h
===================================================================
--- chromium-116.0.5845.42.orig/third_party/blink/renderer/core/paint/object_paint_properties_sparse.h
+++ chromium-116.0.5845.42/third_party/blink/renderer/core/paint/object_paint_properties_sparse.h
@@ -8,6 +8,7 @@
 #include <array>
 #include <memory>
 #include <utility>
+#include <variant>
 
 #include "base/dcheck_is_on.h"
 #include "base/memory/ptr_util.h"
