Index: aiocontextvars-0.2.2/setup.py
===================================================================
--- aiocontextvars-0.2.2.orig/setup.py
+++ aiocontextvars-0.2.2/setup.py
@@ -15,15 +15,6 @@ requirements = [
     'contextvars==2.4;python_version<"3.7"',
 ]
 
-setup_requirements = [
-    'pytest-runner',
-]
-
-test_requirements = [
-    'pytest',
-    'pytest-asyncio==0.8.0',
-]
-
 setup(
     name='aiocontextvars',
     version='0.2.2',
@@ -48,7 +39,4 @@ setup(
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
     ],
-    test_suite='tests',
-    tests_require=test_requirements,
-    setup_requires=setup_requirements,
 )
