--- openjdk/hotspot/src/cpu/aarch64/vm/c1_LIRAssembler_aarch64.cpp	2019-05-24 21:46:57.000000000 +0200
+++ openjdk/hotspot/src/cpu/aarch64/vm/c1_LIRAssembler_aarch64.cpp	2019-10-17 12:04:06.232646618 +0200
@@ -3281,8 +3281,8 @@ void LIR_Assembler::atomic_op(LIR_Code c
           __ prfm(Address(tmp), PSTL1STRM);
         __ bind(again);
         (_masm->*lda)(dst, tmp);
-        (_masm->*stl)(rscratch2, obj, tmp);
-        __ cbnzw(rscratch2, again);
+        (_masm->*stl)(rscratch1, obj, tmp);
+        __ cbnzw(rscratch1, again);
       }
       if (is_oop && UseCompressedOops) {
 	__ decode_heap_oop(dst);
