--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -4,19 +4,9 @@ Fixtures for pytest.
 
 import locale
 
-import hypothesis
 import pytest
 
 
-# This disables the "too slow" hypothesis heath check globally.
-# For some reason it thinks that the text/binary generation is too
-# slow then causes the tests to fail.
-hypothesis.settings.register_profile(
-    "slow-tests",
-    suppress_health_check=[hypothesis.HealthCheck.too_slow],
-)
-
-
 def load_locale(x):
     """Convenience to load a locale, trying ISO8859-1 first."""
     try:
