From: Ruediger Oertel <ro at suse dot de>
Subject: fix deprecated tail call syntax (-1)
Reported-Upstream: Yes
Bugtracker: bugs.exim.org 1080

Index: scripts/Configure-config.h
===================================================================
--- scripts/Configure-config.h.orig
+++ scripts/Configure-config.h
@@ -47,7 +47,7 @@ fi
 
 # Double-check that config.h is complete.
 
-if [ "`tail -1 config.h`" != "/* End of config.h */" ] ; then
+if [ "`tail -n 1 config.h`" != "/* End of config.h */" ] ; then
   echo "*** config.h appears to be incomplete"
   echo "*** unexpected failure in buildconfig program"
   exit 1
