Index: liferea-2.0.1/opml/feedlist_bg.opml
===================================================================
--- liferea-2.0.1.orig/opml/feedlist_bg.opml
+++ liferea-2.0.1/opml/feedlist_bg.opml
@@ -11,6 +11,10 @@
 
 	<outline text="International">		<!-- translate this title for localized feed list -->
 		<!-- start of default block: lets keep this default English block in sync over all feed lists! -->
+                <outline text="openSUSE">
+                        <outline text="openSUSE News" htmlUrl="https://news.opensuse.org/" xmlUrl="https://news.opensuse.org/feed.xml" />
+                        <outline text="Planet openSUSE" htmlUrl="https://planet.opensuse.org/" xmlUrl="https://planet.opensuse.org/global/rss20.xml" />
+                </outline>
 		<outline text="News">
 			<outline text="Ars Technica" htmlUrl="https://arstechnica.com" xmlUrl="https://feeds.arstechnica.com/arstechnica/index"/>
 			<outline text="Slashdot" htmlUrl="https://slashdot.org"  xmlUrl="https://rss.slashdot.org/Slashdot/slashdotMain" />
Index: liferea-2.0.1/opml/feedlist_ca.opml
===================================================================
--- liferea-2.0.1.orig/opml/feedlist_ca.opml
+++ liferea-2.0.1/opml/feedlist_ca.opml
@@ -17,6 +17,10 @@
 
 	<outline text="Example Feeds">		<!-- translate this title for localized feed list -->
 		<!-- start of default block: lets keep this default English block in sync over all feed lists! -->
+                <outline text="openSUSE">
+                        <outline text="openSUSE News" htmlUrl="https://news.opensuse.org/" xmlUrl="https://news.opensuse.org/feed.xml" />
+                        <outline text="Planet openSUSE" htmlUrl="https://planet.opensuse.org/" xmlUrl="https://planet.opensuse.org/global/rss20.xml" />
+                </outline>
 		<outline text="News">
 			<outline text="Ars Technica" htmlUrl="https://arstechnica.com" xmlUrl="https://feeds.arstechnica.com/arstechnica/index"/>
 			<outline text="Slashdot" htmlUrl="https://slashdot.org"  xmlUrl="https://rss.slashdot.org/Slashdot/slashdotMain" />
Index: liferea-2.0.1/opml/feedlist_de.opml
===================================================================
--- liferea-2.0.1.orig/opml/feedlist_de.opml
+++ liferea-2.0.1/opml/feedlist_de.opml
@@ -18,6 +18,10 @@
  
 		<outline text="International">		<!-- translate this title for localized feed list -->
 		<!-- start of default block: lets keep this default English block in sync over all feed lists! -->
+                <outline text="openSUSE">
+                        <outline text="openSUSE News" htmlUrl="https://news.opensuse.org/" xmlUrl="https://news.opensuse.org/feed.xml" />
+                        <outline text="Planet openSUSE" htmlUrl="https://planet.opensuse.org/" xmlUrl="https://planet.opensuse.org/global/rss20.xml" />
+                </outline>
 		<outline text="News">
 			<outline text="Ars Technica" htmlUrl="https://arstechnica.com" xmlUrl="https://feeds.arstechnica.com/arstechnica/index"/>
 			<outline text="Slashdot" htmlUrl="https://slashdot.org"  xmlUrl="https://rss.slashdot.org/Slashdot/slashdotMain" />
Index: liferea-2.0.1/opml/feedlist_en.opml
===================================================================
--- liferea-2.0.1.orig/opml/feedlist_en.opml
+++ liferea-2.0.1/opml/feedlist_en.opml
@@ -7,6 +7,10 @@
 
 	<outline text="Example Feeds">		<!-- translate this title for localized feed list -->
 		<!-- start of default block: lets keep this default English block in sync over all feed lists! -->
+                <outline text="openSUSE">
+                        <outline text="openSUSE News" htmlUrl="https://news.opensuse.org/" xmlUrl="https://news.opensuse.org/feed.xml" />
+                        <outline text="Planet openSUSE" htmlUrl="https://planet.opensuse.org/" xmlUrl="https://planet.opensuse.org/global/rss20.xml" />
+                </outline>
 		<outline text="News">
 			<outline text="Ars Technica" htmlUrl="https://arstechnica.com" xmlUrl="https://feeds.arstechnica.com/arstechnica/index"/>
 			<outline text="Slashdot" htmlUrl="https://slashdot.org"  xmlUrl="https://rss.slashdot.org/Slashdot/slashdotMain" />
Index: liferea-2.0.1/opml/feedlist_es.opml
===================================================================
--- liferea-2.0.1.orig/opml/feedlist_es.opml
+++ liferea-2.0.1/opml/feedlist_es.opml
@@ -15,6 +15,10 @@
 
 	<outline text="Fuentes de ejemplo (inglés)">
 		<!-- start of default block: lets keep this default English block in sync over all feed lists! -->
+                <outline text="openSUSE">
+                        <outline text="openSUSE News" htmlUrl="https://news.opensuse.org/" xmlUrl="https://news.opensuse.org/feed.xml" />
+                        <outline text="Planet openSUSE" htmlUrl="https://planet.opensuse.org/" xmlUrl="https://planet.opensuse.org/global/rss20.xml" />
+                </outline>
 		<outline text="News">
 			<outline text="Ars Technica" htmlUrl="https://arstechnica.com" xmlUrl="https://feeds.arstechnica.com/arstechnica/index"/>
 			<outline text="Slashdot" htmlUrl="https://slashdot.org"  xmlUrl="https://rss.slashdot.org/Slashdot/slashdotMain" />
Index: liferea-2.0.1/opml/feedlist_eu.opml
===================================================================
--- liferea-2.0.1.orig/opml/feedlist_eu.opml
+++ liferea-2.0.1/opml/feedlist_eu.opml
@@ -17,6 +17,10 @@
 
 	<outline text="Nazioarteko jarioak">		<!-- translate this title for localized feed list -->
 		<!-- start of default block: lets keep this default English block in sync over all feed lists! -->
+                <outline text="openSUSE">
+                        <outline text="openSUSE News" htmlUrl="https://news.opensuse.org/" xmlUrl="https://news.opensuse.org/feed.xml" />
+                        <outline text="Planet openSUSE" htmlUrl="https://planet.opensuse.org/" xmlUrl="https://planet.opensuse.org/global/rss20.xml" />
+                </outline>
 		<outline text="News">
 			<outline text="Ars Technica" htmlUrl="https://arstechnica.com" xmlUrl="https://feeds.arstechnica.com/arstechnica/index"/>
 			<outline text="Slashdot" htmlUrl="https://slashdot.org"  xmlUrl="https://rss.slashdot.org/Slashdot/slashdotMain" />
Index: liferea-2.0.1/opml/feedlist_fa.opml
===================================================================
--- liferea-2.0.1.orig/opml/feedlist_fa.opml
+++ liferea-2.0.1/opml/feedlist_fa.opml
@@ -13,6 +13,10 @@
 		
 		<outline text="International">		
 		<!-- start of default block: lets keep this default English block in sync over all feed lists! -->
+                <outline text="openSUSE">
+                        <outline text="openSUSE News" htmlUrl="https://news.opensuse.org/" xmlUrl="https://news.opensuse.org/feed.xml" />
+                        <outline text="Planet openSUSE" htmlUrl="https://planet.opensuse.org/" xmlUrl="https://planet.opensuse.org/global/rss20.xml" />
+                </outline>
 		<outline text="News">
 			<outline text="Ars Technica" htmlUrl="https://arstechnica.com" xmlUrl="https://feeds.arstechnica.com/arstechnica/index"/>
 			<outline text="Slashdot" htmlUrl="https://slashdot.org"  xmlUrl="https://rss.slashdot.org/Slashdot/slashdotMain" />
Index: liferea-2.0.1/opml/feedlist_fr.opml
===================================================================
--- liferea-2.0.1.orig/opml/feedlist_fr.opml
+++ liferea-2.0.1/opml/feedlist_fr.opml
@@ -17,6 +17,10 @@
 
       <outline text="International">		<!-- translate this title for localized feed list -->
 		<!-- start of default block: lets keep this default English block in sync over all feed lists! -->
+                <outline text="openSUSE">
+                        <outline text="openSUSE News" htmlUrl="https://news.opensuse.org/" xmlUrl="https://news.opensuse.org/feed.xml" />
+                        <outline text="Planet openSUSE" htmlUrl="https://planet.opensuse.org/" xmlUrl="https://planet.opensuse.org/global/rss20.xml" />
+                </outline>
 		<outline text="News">
 			<outline text="Ars Technica" htmlUrl="https://arstechnica.com" xmlUrl="https://feeds.arstechnica.com/arstechnica/index"/>
 			<outline text="Slashdot" htmlUrl="https://slashdot.org"  xmlUrl="https://rss.slashdot.org/Slashdot/slashdotMain" />
Index: liferea-2.0.1/opml/feedlist_gl.opml
===================================================================
--- liferea-2.0.1.orig/opml/feedlist_gl.opml
+++ liferea-2.0.1/opml/feedlist_gl.opml
@@ -13,6 +13,10 @@
 	
 	<outline text="Exemplos de fontes (inglés)">
 		<!-- start of default block: lets keep this default English block in sync over all feed lists! -->
+                <outline text="openSUSE">
+                        <outline text="openSUSE News" htmlUrl="https://news.opensuse.org/" xmlUrl="https://news.opensuse.org/feed.xml" />
+                        <outline text="Planet openSUSE" htmlUrl="https://planet.opensuse.org/" xmlUrl="https://planet.opensuse.org/global/rss20.xml" />
+                </outline>
 		<outline text="News">
 			<outline text="Ars Technica" htmlUrl="https://arstechnica.com" xmlUrl="https://feeds.arstechnica.com/arstechnica/index"/>
 			<outline text="Slashdot" htmlUrl="https://slashdot.org"  xmlUrl="https://rss.slashdot.org/Slashdot/slashdotMain" />
Index: liferea-2.0.1/opml/feedlist_he.opml
===================================================================
--- liferea-2.0.1.orig/opml/feedlist_he.opml
+++ liferea-2.0.1/opml/feedlist_he.opml
@@ -29,6 +29,10 @@
 
 		<outline text="מהעולם">		<!-- translate this for localized feed list -->
 		<!-- start of default block: lets keep this default English block in sync over all feed lists! -->
+                <outline text="openSUSE">
+                        <outline text="openSUSE News" htmlUrl="https://news.opensuse.org/" xmlUrl="https://news.opensuse.org/feed.xml" />
+                        <outline text="Planet openSUSE" htmlUrl="https://planet.opensuse.org/" xmlUrl="https://planet.opensuse.org/global/rss20.xml" />
+                </outline>
 		<outline text="News">
 			<outline text="Ars Technica" htmlUrl="https://arstechnica.com" xmlUrl="https://feeds.arstechnica.com/arstechnica/index"/>
 			<outline text="Slashdot" htmlUrl="https://slashdot.org"  xmlUrl="https://rss.slashdot.org/Slashdot/slashdotMain" />
Index: liferea-2.0.1/opml/feedlist_hu.opml
===================================================================
--- liferea-2.0.1.orig/opml/feedlist_hu.opml
+++ liferea-2.0.1/opml/feedlist_hu.opml
@@ -6,6 +6,10 @@
 <body>
 	<outline text="Példa hírforrások">		<!-- translate this for localized feed list -->
 		<!-- start of default block: lets keep this default English block in sync over all feed lists! -->
+                <outline text="openSUSE">
+                        <outline text="openSUSE News" htmlUrl="https://news.opensuse.org/" xmlUrl="https://news.opensuse.org/feed.xml" />
+                        <outline text="Planet openSUSE" htmlUrl="https://planet.opensuse.org/" xmlUrl="https://planet.opensuse.org/global/rss20.xml" />
+                </outline>
 		<outline text="News">
 			<outline text="Ars Technica" htmlUrl="https://arstechnica.com" xmlUrl="https://feeds.arstechnica.com/arstechnica/index"/>
 			<outline text="Slashdot" htmlUrl="https://slashdot.org"  xmlUrl="https://rss.slashdot.org/Slashdot/slashdotMain" />
Index: liferea-2.0.1/opml/feedlist_id.opml
===================================================================
--- liferea-2.0.1.orig/opml/feedlist_id.opml
+++ liferea-2.0.1/opml/feedlist_id.opml
@@ -6,6 +6,10 @@
 <body>
 	<outline text="Contoh Feed">		<!-- translate this title for localized feed list -->
 		<!-- start of default block: lets keep this default English block in sync over all feed lists! -->
+                <outline text="openSUSE">
+                        <outline text="openSUSE News" htmlUrl="https://news.opensuse.org/" xmlUrl="https://news.opensuse.org/feed.xml" />
+                        <outline text="Planet openSUSE" htmlUrl="https://planet.opensuse.org/" xmlUrl="https://planet.opensuse.org/global/rss20.xml" />
+                </outline>
 		<outline text="News">
 			<outline text="Ars Technica" htmlUrl="https://arstechnica.com" xmlUrl="https://feeds.arstechnica.com/arstechnica/index"/>
 			<outline text="Slashdot" htmlUrl="https://slashdot.org"  xmlUrl="https://rss.slashdot.org/Slashdot/slashdotMain" />
