From: Fabian Vogt <fabian@ritter-vogt.de>
Subject: Disable plugins which cause automatic network requests by default
References: boo#1070899

Index: amarok-2.8.0/src/services/lastfm/amarok_service_lastfm.desktop
===================================================================
--- amarok-2.8.0.orig/src/services/lastfm/amarok_service_lastfm.desktop
+++ amarok-2.8.0/src/services/lastfm/amarok_service_lastfm.desktop
@@ -119,6 +119,6 @@ X-KDE-PluginInfo-Email=kde@dontletsstart
 X-KDE-PluginInfo-Version=1.0
 X-KDE-PluginInfo-Category=Service
 X-KDE-PluginInfo-License=GPL
-X-KDE-PluginInfo-EnabledByDefault=true
+X-KDE-PluginInfo-EnabledByDefault=false
 X-KDE-Library=amarok_service_lastfm
 X-KDE-PluginInfo-Name=amarok_service_lastfm
Index: amarok-2.8.0/src/services/magnatune/amarok_service_magnatunestore.desktop
===================================================================
--- amarok-2.8.0.orig/src/services/magnatune/amarok_service_magnatunestore.desktop
+++ amarok-2.8.0/src/services/magnatune/amarok_service_magnatunestore.desktop
@@ -126,7 +126,7 @@ X-KDE-PluginInfo-Email=nhnFreespirit@gma
 X-KDE-PluginInfo-Version=2.0
 X-KDE-PluginInfo-Category=Service
 X-KDE-PluginInfo-License=GPL
-X-KDE-PluginInfo-EnabledByDefault=true
+X-KDE-PluginInfo-EnabledByDefault=false
 X-KDE-Library=amarok_service_magnatunestore
 X-KDE-PluginInfo-Name=amarok_service_magnatunestore
 
