diff --git a/platform/x11/detect.py b/platform/x11/detect.py
index 28f5335..35892d8 100644
--- a/platform/x11/detect.py
+++ b/platform/x11/detect.py
@@ -160,7 +160,7 @@ def configure(env):
         version = get_compiler_version(env)
         if version != None and version[0] >= '6':
             env.Append(CCFLAGS=['-fpie'])
-            env.Append(LINKFLAGS=['-no-pie'])
+
     # Do the same for clang should be fine with Clang 4 and higher
     if using_clang(env):
         version = get_compiler_version(env)
