Index: hdparm-9.60/wiper/wiper.sh
===================================================================
--- hdparm-9.60.orig/wiper/wiper.sh
+++ hdparm-9.60/wiper/wiper.sh
@@ -48,6 +48,9 @@ function usage_error(){
 
 echo
 echo "${0##*/}: Linux SATA SSD TRIM utility, version $VERSION, by Mark Lord."
+echo "${0##*/}: This tool is DANGEROUS! Please read and understand"
+echo "${0##*/}: /usr/share/doc/packages/hdparm/README.wiper"
+echo "${0##*/}: before going any further."
 
 export verbose=0
 commit=""
