Index: Makefile
===================================================================
--- Makefile.orig
+++ Makefile
@@ -11,7 +11,6 @@ mandir = $(manprefix)/share/man
 oldmandir = $(manprefix)/man
 
 CC ?= gcc
-STRIP ?= strip
 
 CFLAGS := -O2 -W -Wall -Wbad-function-cast -Wcast-align -Wpointer-arith -Wcast-qual -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -fkeep-inline-functions -Wwrite-strings -Waggregate-return -Wnested-externs -Wtrigraphs $(CFLAGS)
 
@@ -30,7 +29,6 @@ static:
 
 hdparm: Makefile hdparm.h sgio.h $(OBJS)
 	$(CC) $(LDFLAGS) -o hdparm $(OBJS)
-	$(STRIP) hdparm
 
 hdparm.o:	hdparm.h sgio.h
 
