From 69de82b56bed9675c1acda293b5cdb4b1c1dcc7a Mon Sep 17 00:00:00 2001
From: Joffrey Bienvenu <joffreybvn@gmail.com>
Date: Tue, 31 May 2022 22:22:32 +0200
Subject: [PATCH 1/2] Fix package version

---
 traitlets/_version.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/traitlets/_version.py b/traitlets/_version.py
index ecff6d1d..a1b6b2c6 100644
--- a/traitlets/_version.py
+++ b/traitlets/_version.py
@@ -1,5 +1,5 @@
 version_info = (5, 2, 2, "final")
-__version__ = "5.2.2"
+__version__ = "5.2.2.final"
 
 # unlike `.dev`, alpha, beta and rc _must not_ have dots,
 # or the wheel and tgz won't look to pip like the same version.

From ad10c659385a59c5d015ecae123b9b7a31a108cf Mon Sep 17 00:00:00 2001
From: Joffrey Bienvenu <joffreybvn@gmail.com>
Date: Tue, 31 May 2022 22:57:03 +0200
Subject: [PATCH 2/2] Remove the 'final' keyword in version

---
 traitlets/_version.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/traitlets/_version.py b/traitlets/_version.py
index a1b6b2c6..f75993c7 100644
--- a/traitlets/_version.py
+++ b/traitlets/_version.py
@@ -1,5 +1,5 @@
-version_info = (5, 2, 2, "final")
-__version__ = "5.2.2.final"
+version_info = (5, 2, 2)
+__version__ = "5.2.2"
 
 # unlike `.dev`, alpha, beta and rc _must not_ have dots,
 # or the wheel and tgz won't look to pip like the same version.
