diff -upr django-tables2-2.4.1.orig/tests/test_models.py django-tables2-2.4.1/tests/test_models.py
--- django-tables2-2.4.1.orig/tests/test_models.py	2022-05-12 09:08:59.126006528 +0200
+++ django-tables2-2.4.1/tests/test_models.py	2022-05-12 09:08:59.138006605 +0200
@@ -1,6 +1,6 @@
 from collections import defaultdict
 
-import mock
+from unittest import mock
 from django.contrib.contenttypes.models import ContentType
 from django.db import models
 from django.db.models.functions import Length
