commit a8f809faddff9a71adcee565b3644fa66ebfe4bc (origin/f22, origin/f21)
Author: Peter Robinson <pbrobinson@gmail.com>
Date:   Wed Nov 26 14:52:05 2014 +0000
References: https://src.fedoraproject.org/rpms/latrace/blob/f30/f/latrace-ppc64-basic-audit.patch
Notes: Cherry-picked to openSUSE by fstrba, 2022-04-21T16:11:09

[ The latrace-ppc64-basic-audit.patch from F30 contains
what has become upstream commit fe46504f854e28ad83b17b085564d1d7eb5c67d4;
the following hunk with lt_stack_framesize did not make it upstream still. -jengelh]

--- a/src/stack.c
+++ b/src/stack.c
@@ -6,6 +6,13 @@
 
 #include "config.h"
 
+#if defined __powerpc__
+/* check the current stack pointer and check its boundaries */
+int lt_stack_framesize(struct lt_config_audit *cfg, La_regs *regs)
+{
+	return lt_sh(cfg, framesize);
+}
+#else
 static __thread void *stack_start = NULL;
 static __thread void *stack_end   = NULL;
 
@@ -124,3 +131,4 @@ int lt_stack_framesize(struct lt_config_audit *cfg, La_regs *regs)
 
 	return framesize;
 }
+#endif


