Index: open-gpu-kernel-modules-550.135/kernel-open/Makefile
===================================================================
--- open-gpu-kernel-modules-550.135.orig/kernel-open/Makefile
+++ open-gpu-kernel-modules-550.135/kernel-open/Makefile
@@ -83,7 +83,7 @@ else
 
   KERNEL_ARCH = $(ARCH)
 
-  ifneq ($(filter $(ARCH),i386 x86_64),)
+  ifneq ($(filter $(ARCH),i386 x86_64 x86),)
     KERNEL_ARCH = x86
   else
     ifeq ($(filter $(ARCH),arm64 powerpc),)
