diff -ru dde-file-manager-5.6.4.orig/3rdparty/fsearch/database_search.c dde-file-manager-5.6.4/3rdparty/fsearch/database_search.c
--- dde-file-manager-5.6.4.orig/3rdparty/fsearch/database_search.c	2025-04-10 20:32:40.061029999 +0200
+++ dde-file-manager-5.6.4/3rdparty/fsearch/database_search.c	2025-04-10 20:32:57.781025345 +0200
@@ -24,7 +24,6 @@
 #include <string.h>
 #include <assert.h>
 #include <ctype.h>
-#include <pcre.h>
 #include <fnmatch.h>
 
 #include "database_search.h"
