diff -Nur dde-file-manager-5.5.10/src/dde-file-manager-lib/dde-file-manager-lib.pro dde-file-manager-5.5.10-new/src/dde-file-manager-lib/dde-file-manager-lib.pro
--- dde-file-manager-5.5.10/src/dde-file-manager-lib/dde-file-manager-lib.pro	2022-02-25 11:23:25.000000000 +0800
+++ dde-file-manager-5.5.10-new/src/dde-file-manager-lib/dde-file-manager-lib.pro	2022-04-18 15:32:07.986501937 +0800
@@ -93,10 +93,8 @@
 include(interfaces/vfs/vfs.pri)
 include(interfaces/customization/customization.pri)
 include(src.pri)
+include(search/dfsearch.pri)
 
-isEqual(ARCH, sw_64) | isEqual(ARCH, mips64) | isEqual(ARCH, mips32) | isEqual(ARCH, aarch64) | isEqual(ARCH, loongarch64) {
-    include(search/dfsearch.pri)
-}
 
 APPSHAREDIR = $$PREFIX/share/$$TARGET
 ICONDIR = $$PREFIX/share/icons/hicolor/scalable/apps
