--- sisu-inject/org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/inject/Implementations.java	2024-04-09 19:53:29.797479006 +0200
+++ sisu-inject/org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/inject/Implementations.java	2024-04-09 19:59:23.506494426 +0200
@@ -39,17 +39,6 @@
 
     static
     {
-        boolean hasGuiceServlet;
-        try
-        {
-            hasGuiceServlet = BindingTargetVisitor.class.isInstance( ServletFinder.THIS );
-        }
-        catch ( final LinkageError e )
-        {
-            hasGuiceServlet = false;
-        }
-        HAS_GUICE_SERVLET = hasGuiceServlet;
-
         boolean hasJsr250Priority;
         try
         {
@@ -66,8 +55,6 @@
     // Constants
     // ----------------------------------------------------------------------
 
-    private static final boolean HAS_GUICE_SERVLET;
-
     private static final boolean HAS_JSR250_PRIORITY;
 
     // ----------------------------------------------------------------------
@@ -107,8 +94,7 @@
 
         final Class<?> annotationSource =
             // when looking for @Priority also consider annotations on providers (and servlets/filters if available)
-            binding.acceptTargetVisitor( isPriority ? ( HAS_GUICE_SERVLET ? ServletFinder.THIS : ProviderFinder.THIS )
-                                                    : ClassFinder.THIS );
+            binding.acceptTargetVisitor( isPriority ? ProviderFinder.THIS : ClassFinder.THIS );
 
         T annotation = null;
         if ( null != annotationSource )
@@ -253,45 +239,4 @@
             return binding.getProviderKey().getTypeLiteral().getRawType();
         }
     }
-
-    /**
-     * {@link ProviderFinder} that also returns servlet/filter implementations.
-     */
-    static final class ServletFinder
-        extends ProviderFinder
-        implements com.google.inject.servlet.ServletModuleTargetVisitor<Object, Class<?>>
-    {
-        // ----------------------------------------------------------------------
-        // Constants
-        // ----------------------------------------------------------------------
-
-        @SuppressWarnings( "hiding" )
-        static final BindingTargetVisitor<Object, Class<?>> THIS = new ServletFinder();
-
-        // ----------------------------------------------------------------------
-        // Public methods
-        // ----------------------------------------------------------------------
-
-        public Class<?> visit( final com.google.inject.servlet.InstanceFilterBinding binding )
-        {
-            return binding.getFilterInstance().getClass();
-        }
-
-        public Class<?> visit( final com.google.inject.servlet.InstanceServletBinding binding )
-        {
-            return binding.getServletInstance().getClass();
-        }
-
-        public Class<?> visit( final com.google.inject.servlet.LinkedFilterBinding binding )
-        {
-            // this assumes only one level of indirection: api-->impl
-            return binding.getLinkedKey().getTypeLiteral().getRawType();
-        }
-
-        public Class<?> visit( final com.google.inject.servlet.LinkedServletBinding binding )
-        {
-            // this assumes only one level of indirection: api-->impl
-            return binding.getLinkedKey().getTypeLiteral().getRawType();
-        }
-    }
 }
