--- a/org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/space/AbstractSisuIndex.java	2026-02-11 14:36:37.906142024 +0100
+++ b/org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/space/AbstractSisuIndex.java	2026-02-11 14:38:32.910020678 +0100
@@ -139,7 +139,7 @@
     private void writeTable(final Object name, final Set<String> table) {
         try {
             try (BufferedWriter writer = new BufferedWriter(getWriter(INDEX_FOLDER + name))) {
-                for (final String line : table) {
+                for (final String line : new TreeSet<String>(table)) {
                     writer.write(line);
                     writer.newLine();
                 }
