--- a/org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/inject/Implementations.java	2026-02-11 14:25:02.494361564 +0100
+++ b/org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/inject/Implementations.java	2026-02-11 14:28:52.321660010 +0100
@@ -36,14 +36,6 @@
     // ----------------------------------------------------------------------
 
     static {
-        boolean hasGuiceServlet;
-        try {
-            hasGuiceServlet = BindingTargetVisitor.class.isInstance(ServletFinder.THIS);
-        } catch (final LinkageError e) {
-            hasGuiceServlet = false;
-        }
-        HAS_GUICE_SERVLET = hasGuiceServlet;
-
         boolean hasJsr250Priority;
         try {
             hasJsr250Priority = javax.annotation.Priority.class.isAnnotation();
@@ -57,8 +49,6 @@
     // Constants
     // ----------------------------------------------------------------------
 
-    private static final boolean HAS_GUICE_SERVLET;
-
     private static final boolean HAS_JSR250_PRIORITY;
 
     // ----------------------------------------------------------------------
@@ -96,7 +86,7 @@
         final Class<?> annotationSource =
                 // when looking for @Priority also consider annotations on providers (and servlets/filters if available)
                 binding.acceptTargetVisitor(
-                        isPriority ? (HAS_GUICE_SERVLET ? ServletFinder.THIS : ProviderFinder.THIS) : ClassFinder.THIS);
+                        isPriority ? ProviderFinder.THIS : ClassFinder.THIS);
 
         T annotation = null;
         if (null != annotationSource) {
@@ -223,43 +213,4 @@
             return binding.getProviderKey().getTypeLiteral().getRawType();
         }
     }
-
-    /**
-     * {@link ProviderFinder} that also returns servlet/filter implementations.
-     */
-    static final class ServletFinder extends ProviderFinder
-            implements com.google.inject.servlet.ServletModuleTargetVisitor<Object, Class<?>> {
-        // ----------------------------------------------------------------------
-        // Constants
-        // ----------------------------------------------------------------------
-
-        @SuppressWarnings("hiding")
-        static final BindingTargetVisitor<Object, Class<?>> THIS = new ServletFinder();
-
-        // ----------------------------------------------------------------------
-        // Public methods
-        // ----------------------------------------------------------------------
-
-        @Override
-        public Class<?> visit(final com.google.inject.servlet.InstanceFilterBinding binding) {
-            return binding.getFilterInstance().getClass();
-        }
-
-        @Override
-        public Class<?> visit(final com.google.inject.servlet.InstanceServletBinding binding) {
-            return binding.getServletInstance().getClass();
-        }
-
-        @Override
-        public Class<?> visit(final com.google.inject.servlet.LinkedFilterBinding binding) {
-            // this assumes only one level of indirection: api-->impl
-            return binding.getLinkedKey().getTypeLiteral().getRawType();
-        }
-
-        @Override
-        public Class<?> visit(final com.google.inject.servlet.LinkedServletBinding binding) {
-            // this assumes only one level of indirection: api-->impl
-            return binding.getLinkedKey().getTypeLiteral().getRawType();
-        }
-    }
 }
