Index: grub-2.02~beta2/util/grub-mkconfig.in
===================================================================
--- grub-2.02~beta2.orig/util/grub-mkconfig.in
+++ grub-2.02~beta2/util/grub-mkconfig.in
@@ -311,6 +311,11 @@ and /etc/grub.d/* files or please file a
   else
     # none of the children aborted with error, install the new grub.cfg
     mv -f ${grub_cfg}.new ${grub_cfg}
+    # check if default entry need to be corrected for updated distributor version 
+    # and/or use fallback entry if default kernel entry removed
+    if test -x /usr/sbin/grub2-check-default; then
+       /usr/sbin/grub2-check-default >&2
+    fi
   fi
 fi
 
