Index: grub-2.02/util/grub-mkconfig.in
===================================================================
--- grub-2.02.orig/util/grub-mkconfig.in
+++ grub-2.02/util/grub-mkconfig.in
@@ -268,7 +268,8 @@ export GRUB_DEFAULT \
   GRUB_CMDLINE_LINUX_RECOVERY \
   GRUB_USE_LINUXEFI \
   SUSE_BTRFS_SNAPSHOT_BOOTING \
-  SUSE_CMDLINE_XENEFI
+  SUSE_CMDLINE_XENEFI \
+  SUSE_REMOVE_LINUX_ROOT_PARAM
 
 if test "x${grub_cfg}" != "x"; then
   rm -f "${grub_cfg}.new"
Index: grub-2.02/util/grub.d/10_linux.in
===================================================================
--- grub-2.02.orig/util/grub.d/10_linux.in
+++ grub-2.02/util/grub.d/10_linux.in
@@ -64,7 +64,7 @@ case x"$GRUB_FS" in
 	else
 	rootsubvol="`make_system_path_relative_to_its_root /`"
 	rootsubvol="${rootsubvol#/}"
-	if [ "x${rootsubvol}" != x ]; then
+	if [ "x${rootsubvol}" != x ] && [ "x$SUSE_REMOVE_LINUX_ROOT_PARAM" != "xtrue" ]; then
 	    GRUB_CMDLINE_LINUX="rootflags=subvol=${rootsubvol} ${GRUB_CMDLINE_LINUX}"
 	fi
 	fi;;
@@ -75,6 +75,10 @@ case x"$GRUB_FS" in
 	;;
 esac
 
+if [ "x$SUSE_REMOVE_LINUX_ROOT_PARAM" = "xtrue" ]; then
+  LINUX_ROOT_DEVICE=""
+fi
+
 title_correction_code=
 
 hotkey=1
Index: grub-2.02/util/grub.d/20_linux_xen.in
===================================================================
--- grub-2.02.orig/util/grub.d/20_linux_xen.in
+++ grub-2.02/util/grub.d/20_linux_xen.in
@@ -78,7 +78,7 @@ case x"$GRUB_FS" in
 	else
 	rootsubvol="`make_system_path_relative_to_its_root /`"
 	rootsubvol="${rootsubvol#/}"
-	if [ "x${rootsubvol}" != x ]; then
+	if [ "x${rootsubvol}" != x ] && [ "x$SUSE_REMOVE_LINUX_ROOT_PARAM" != "xtrue" ]; then
 	    GRUB_CMDLINE_LINUX="rootflags=subvol=${rootsubvol} ${GRUB_CMDLINE_LINUX}"
 	fi
 	fi;;
@@ -89,6 +89,10 @@ case x"$GRUB_FS" in
 	;;
 esac
 
+if [ "x$SUSE_REMOVE_LINUX_ROOT_PARAM" = "xtrue" ]; then
+  LINUX_ROOT_DEVICE=""
+fi
+
 title_correction_code=
 
 grub2_unquote ()
Index: grub-2.02/util/s390x/zipl2grub.pl.in
===================================================================
--- grub-2.02.orig/util/s390x/zipl2grub.pl.in
+++ grub-2.02/util/s390x/zipl2grub.pl.in
@@ -367,9 +367,13 @@ while ( <IN> ) {
     } else {
       $v = "";
     }
-    if ($k eq "GRUB_DEVICE" && $v !~ /^UUID/ && ! -e $v) {
-      s{root=\@$k\@}{}g;
-      next;
+    if ($k eq "GRUB_DEVICE") {
+      if (($v !~ /^UUID/ && ! -e $v) ||
+          (exists( $C{SUSE_REMOVE_LINUX_ROOT_PARAM}) &&
+          $C{SUSE_REMOVE_LINUX_ROOT_PARAM} eq "true")) {
+        s{root=\@$k\@}{}g;
+        next;
+      }
     }
     s{\@$k\@}{$v}g;
   }
