--- a/util/grub-mkconfig_lib.in
+++ b/util/grub-mkconfig_lib.in
@@ -193,6 +193,12 @@
       *.rpmsave|*.rpmnew) return 1 ;;
       README*|*/README*)  return 1 ;; # documentation
       *.sig) return 1 ;; # signatures
+      # Skip zfcpdump kernel from the grub boot menu (bsc#1166513) The zfcpdump
+      # kernel image is used by zipl to prepare a SCSI dump disc and is only
+      # intended to boot from that disk for creating kernel crash dumps,
+      # therefore booting it from grub is not making sense and also will result
+      # in unbootable system.
+      *-zfcpdump) return 1 ;; # s390 zfcpdump image
     esac
   else
     return 1
