Index: libksba-1.6.7/autogen.sh
===================================================================
--- libksba-1.6.7.orig/autogen.sh
+++ libksba-1.6.7/autogen.sh
@@ -244,7 +244,7 @@ if [ "$myhost" = "find-version" ]; then
     fi
 
     beta=no
-    if [ -e .git ]; then
+    if false; then
       ingit=yes
       tmp=$(git describe --match "${matchstr1}" --long 2>/dev/null)
       if [ -n "$tmp" ]; then
@@ -269,8 +269,8 @@ if [ "$myhost" = "find-version" ]; then
       rvd=$((0x$(echo ${rev} | dd bs=1 count=4 2>/dev/null)))
     else
       ingit=no
-      beta=yes
-      tmp="-unknown"
+      beta=no
+      tmp=""
       rev="0000000"
       rvd="0"
     fi
