diff -Nru azure-mgmt-logic-4.0.0rc2.orig/setup.py azure-mgmt-logic-4.0.0rc2/setup.py
--- azure-mgmt-logic-4.0.0rc2.orig/setup.py	2019-01-15 19:04:34.000000000 +0100
+++ azure-mgmt-logic-4.0.0rc2/setup.py	2019-03-28 10:54:20.524761957 +0100
@@ -79,7 +79,7 @@
     install_requires=[
         'msrest>=0.5.0',
         'msrestazure>=0.4.32,<2.0.0',
-        'azure-common~=1.1',
+        'azure-common>=1.1',
     ],
     extras_require={
         ":python_version<'3.0'": ['azure-mgmt-nspkg'],
