diff -Nru icewm-2.0.0.orig/lib/menu.in icewm-2.0.0/lib/menu.in
--- icewm-2.0.0.orig/lib/menu.in	2020-12-14 02:14:27.000000000 +0100
+++ icewm-2.0.0/lib/menu.in	2020-12-20 18:55:09.026453502 +0100
@@ -11,5 +11,6 @@
 prog Hexchat xchat hexchat
 prog Gimp gimp gimp
 includeprog icewm-menu-fdo --sep-before --no-sep-others
-menufile Programs folder programs
+#menufile Programs folder programs
 menufile Tool_bar folder toolbar
+menuprog SUSE folder xdg_menu --format icewm
diff -Nru icewm-2.0.0.orig/src/wmmenu.cc icewm-2.0.0/src/wmmenu.cc
--- icewm-2.0.0.orig/src/wmmenu.cc	2020-12-14 02:14:27.000000000 +0100
+++ icewm-2.0.0/src/wmmenu.cc	2020-12-20 18:59:54.401885022 +0100
@@ -294,6 +294,46 @@
     return p;
 }
 
+char* MenuLoader::parseSuseProg(char *p, ObjectContainer *container)
+{
+    Argument name;
+
+    p = YConfig::getArgument(&name, p);
+    if (p == nullptr) return p;
+
+    Argument icons;
+
+    p = YConfig::getArgument(&icons, p);
+    if (p == nullptr) return p;
+
+    Argument command;
+    YStringArray args;
+
+    p = getCommandArgs(p, &command, args);
+    if (p == nullptr) {
+        msg(_("Error at menuprog '%s'"), name.cstr());
+        return p;
+    }
+
+    ref<YIcon> icon;
+#ifndef LITE
+    if (icons[0] != '-')
+        icon = YIcon::getIcon(icons);
+#endif
+    MSG(("suseprog %s %s", name.cstr(), command.cstr()));
+
+    csmart path(path_lookup(command.cstr()));
+    if (path) {
+        ObjectMenu *progmenu = new MenuProgMenu(
+                app, smActionListener, wmActionListener,
+                name.cstr(), command.cstr(), args);
+        if (progmenu)
+            container->addContainer(name.cstr(), icon, progmenu);
+    }
+
+    return p;
+}
+
 char* MenuLoader::parseMenuProgReload(char *p, ObjectContainer *container)
 {
     Argument name;
@@ -395,6 +435,9 @@
         else if (!strcmp(word, "menuprogreload")) {
             p = parseMenuProgReload(p, container);
         }
+        else if (!strcmp(word, "suseprog")) {
+            p = parseSuseProg(p, container);
+        }
         else if (!strcmp(word, "include")) {
             p = parseIncludeStatement(p, container);
         }
diff -Nru icewm-2.0.0.orig/src/wmprog.h icewm-2.0.0/src/wmprog.h
--- icewm-2.0.0.orig/src/wmprog.h	2020-12-14 02:14:27.000000000 +0100
+++ icewm-2.0.0/src/wmprog.h	2020-12-20 19:01:08.201727478 +0100
@@ -30,6 +30,7 @@
     char* parseAMenu(char *data, ObjectContainer *container);
     char* parseMenuFile(char *data, ObjectContainer *container);
     char* parseMenuProg(char *data, ObjectContainer *container);
+    char* parseSuseProg(char *p, ObjectContainer *container);
     char* parseMenuProgReload(char *data, ObjectContainer *container);
     char* parseKey(char *word, char *p);
     char* parseProgram(char *word, char *p, ObjectContainer *container);
