Index: mx4j-3.0.1/src/docs/xsl/mx4j-chunk.xsl
===================================================================
--- mx4j-3.0.1.orig/src/docs/xsl/mx4j-chunk.xsl
+++ mx4j-3.0.1/src/docs/xsl/mx4j-chunk.xsl
@@ -1,6 +1,6 @@
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version='1.0'>
 
-  <xsl:import href="http://docbook.sourceforge.net/release/xsl/current/html/chunk.xsl"/>
+  <xsl:import href="/usr/share/xml/docbook/stylesheet/nwalsh/current/html/chunk.xsl"/>
   <xsl:import href="autotoc.xsl"/>
 
   <xsl:param name="html.stylesheet">styles.css</xsl:param>
