Index: node-git.6dbcd0a0f3/test/parallel/test-repl-mode.js
===================================================================
--- node-git.6dbcd0a0f3.orig/test/parallel/test-repl-mode.js
+++ node-git.6dbcd0a0f3/test/parallel/test-repl-mode.js
@@ -1,9 +1,11 @@
 'use strict';
-require('../common');
+const common = require('../common');
 const assert = require('assert');
 const Stream = require('stream');
 const repl = require('repl');
 
+common.skipIfDumbTerminal();
+
 const tests = [
   testSloppyMode,
   testStrictMode,
