Index: rarian-0.8.1/librarian/rarian-reg-utils.c
===================================================================
--- rarian-0.8.1.orig/librarian/rarian-reg-utils.c
+++ rarian-0.8.1/librarian/rarian-reg-utils.c
@@ -170,7 +170,7 @@ rrn_reg_parse_suse_manual (char *filenam
 		reg->identifier = malloc (sizeof (char)*(strlen(basename)+11));
 		sprintf (reg->identifier, "org.other.%s", basename);
 		free (basename);
-		printf ("identifier is %s\n", reg->identifier);
+		/* printf ("identifier is %s\n", reg->identifier); */
 
 	}
 	if (reg)
@@ -321,7 +321,7 @@ rrn_reg_parse_file (char *filename)
 		reg->identifier = malloc (sizeof (char)*(strlen(basename)+11));
 		sprintf (reg->identifier, "org.other.%s", basename);
 		free (basename);
-		printf ("identifier is %s\n", reg->identifier);
+		/* printf ("identifier is %s\n", reg->identifier); */
 
 	}
 	if (reg)
