Index: grpcio-1.72.0/src/core/lib/promise/party.h
===================================================================
--- grpcio-1.72.0.orig/src/core/lib/promise/party.h
+++ grpcio-1.72.0/src/core/lib/promise/party.h
@@ -487,6 +487,7 @@ class Party : public Activity, private W
           Crash(
               "unreachable: promises should not be repolled after completion");
       }
+      return false;
     }
 
     // Outside party poll: check whether the spawning party has completed this
@@ -499,6 +500,7 @@ class Party : public Activity, private W
         case State::kResult:
           return std::move(result_);
       }
+      return Pending{};
     }
 
     void Destroy() override { this->Unref(); }
Index: grpcio-1.72.0/src/core/call/call_state.h
===================================================================
--- grpcio-1.72.0.orig/src/core/call/call_state.h
+++ grpcio-1.72.0/src/core/call/call_state.h
@@ -155,6 +155,7 @@ class CallState {
       case ClientToServerPullState::kTerminated:
         return "Terminated";
     }
+    return "";
   }
   template <typename Sink>
   friend void AbslStringify(Sink& out, ClientToServerPullState state) {
@@ -185,6 +186,7 @@ class CallState {
       case ClientToServerPushState::kFinished:
         return "Finished";
     }
+    return "";
   }
   template <typename Sink>
   friend void AbslStringify(Sink& out, ClientToServerPushState state) {
@@ -235,6 +237,7 @@ class CallState {
       case ServerToClientPullState::kTerminated:
         return "Terminated";
     }
+    return "";
   }
   template <typename Sink>
   friend void AbslStringify(Sink& out, ServerToClientPullState state) {
@@ -275,6 +278,7 @@ class CallState {
       case ServerToClientPushState::kFinished:
         return "Finished";
     }
+    return "";
   }
   template <typename Sink>
   friend void AbslStringify(Sink& out, ServerToClientPushState state) {
@@ -305,6 +309,7 @@ class CallState {
       case ServerTrailingMetadataState::kPulledCancel:
         return "PulledCancel";
     }
+    return "";
   }
   template <typename Sink>
   friend void AbslStringify(Sink& out, ServerTrailingMetadataState state) {
