--- ./libmissing/tests/Makefile.in	2018-12-23 22:01:04.381000000 +0100
+++ ./libmissing/tests/Makefile.in 	2019-02-27 00:25:12.529345408 +0100
@@ -129,22 +129,21 @@
 	test-langinfo$(EXEEXT) test-limits-h$(EXEEXT) \
 	test-listen$(EXEEXT) test-locale$(EXEEXT) \
 	test-localeconv$(EXEEXT) test-localename$(EXEEXT) \
-	test-rwlock1$(EXEEXT) test-lock$(EXEEXT) test-once1$(EXEEXT) \
-	test-once2$(EXEEXT) test-lseek.sh test-lstat$(EXEEXT) \
-	test-malloca$(EXEEXT) test-mbrtowc1.sh test-mbrtowc2.sh \
-	test-mbrtowc3.sh test-mbrtowc4.sh test-mbrtowc5.sh \
-	test-mbrtowc-w32-1.sh test-mbrtowc-w32-2.sh \
-	test-mbrtowc-w32-3.sh test-mbrtowc-w32-4.sh \
-	test-mbrtowc-w32-5.sh test-mbsinit.sh test-memchr$(EXEEXT) \
-	test-memmem$(EXEEXT) test-nanosleep$(EXEEXT) \
-	test-netdb$(EXEEXT) test-netinet_in$(EXEEXT) \
-	test-nl_langinfo.sh test-open$(EXEEXT) test-pathmax$(EXEEXT) \
-	test-perror.sh test-perror2$(EXEEXT) test-pipe$(EXEEXT) \
-	test-poll-h$(EXEEXT) test-poll$(EXEEXT) \
-	test-pthread_sigmask1$(EXEEXT) test-pthread_sigmask2$(EXEEXT) \
-	test-raise$(EXEEXT) test-regex$(EXEEXT) test-select$(EXEEXT) \
-	test-select-in.sh test-select-out.sh test-setenv$(EXEEXT) \
-	test-setlocale1.sh test-setlocale2.sh test-setsockopt$(EXEEXT) \
+	test-lseek.sh test-lstat$(EXEEXT) test-malloca$(EXEEXT) \
+	test-mbrtowc1.sh test-mbrtowc2.sh test-mbrtowc3.sh \
+	test-mbrtowc4.sh test-mbrtowc5.sh test-mbrtowc-w32-1.sh \
+	test-mbrtowc-w32-2.sh test-mbrtowc-w32-3.sh \
+	test-mbrtowc-w32-4.sh test-mbrtowc-w32-5.sh test-mbsinit.sh \
+	test-memchr$(EXEEXT) test-memmem$(EXEEXT) \
+	test-nanosleep$(EXEEXT) test-netdb$(EXEEXT) \
+	test-netinet_in$(EXEEXT) test-nl_langinfo.sh \
+	test-open$(EXEEXT) test-pathmax$(EXEEXT) test-perror.sh \
+	test-perror2$(EXEEXT) test-pipe$(EXEEXT) test-poll-h$(EXEEXT) \
+	test-poll$(EXEEXT) test-pthread_sigmask1$(EXEEXT) \
+	test-pthread_sigmask2$(EXEEXT) test-raise$(EXEEXT) \
+	test-regex$(EXEEXT) test-select$(EXEEXT) test-select-in.sh \
+	test-select-out.sh test-setenv$(EXEEXT) test-setlocale1.sh \
+	test-setlocale2.sh test-setsockopt$(EXEEXT) \
 	test-sigaction$(EXEEXT) test-signal-h$(EXEEXT) \
 	test-sigprocmask$(EXEEXT) test-sleep$(EXEEXT) \
 	test-snprintf$(EXEEXT) test-sockets$(EXEEXT) \
@@ -196,9 +195,8 @@
 	test-isblank$(EXEEXT) test-langinfo$(EXEEXT) \
 	test-limits-h$(EXEEXT) test-listen$(EXEEXT) \
 	test-locale$(EXEEXT) test-localeconv$(EXEEXT) \
-	test-localename$(EXEEXT) test-rwlock1$(EXEEXT) \
-	test-lock$(EXEEXT) test-once1$(EXEEXT) test-once2$(EXEEXT) \
-	test-lseek$(EXEEXT) test-lstat$(EXEEXT) test-malloca$(EXEEXT) \
+	test-localename$(EXEEXT) test-lseek$(EXEEXT) \
+	test-lstat$(EXEEXT) test-malloca$(EXEEXT) \
 	test-mbrtowc$(EXEEXT) test-mbrtowc-w32$(EXEEXT) \
 	test-mbsinit$(EXEEXT) test-memchr$(EXEEXT) \
 	test-memmem$(EXEEXT) test-nanosleep$(EXEEXT) \
@@ -774,9 +772,6 @@
 test_localename_OBJECTS = test-localename.$(OBJEXT)
 test_localename_DEPENDENCIES = $(am__DEPENDENCIES_2) \
 	$(am__DEPENDENCIES_1)
-test_lock_SOURCES = test-lock.c
-test_lock_OBJECTS = test-lock.$(OBJEXT)
-test_lock_DEPENDENCIES = $(am__DEPENDENCIES_2)
 test_lseek_SOURCES = test-lseek.c
 test_lseek_OBJECTS = test-lseek.$(OBJEXT)
 test_lseek_LDADD = $(LDADD)
@@ -839,12 +834,6 @@
 test_nl_langinfo_DEPENDENCIES = libtests.a \
 	../../libmissing/libmissing.la libtests.a \
 	$(am__DEPENDENCIES_1)
-am_test_once1_OBJECTS = test-once.$(OBJEXT)
-test_once1_OBJECTS = $(am_test_once1_OBJECTS)
-test_once1_DEPENDENCIES = $(am__DEPENDENCIES_2)
-am_test_once2_OBJECTS = test-once.$(OBJEXT)
-test_once2_OBJECTS = $(am_test_once2_OBJECTS)
-test_once2_DEPENDENCIES = $(am__DEPENDENCIES_2)
 test_open_SOURCES = test-open.c
 test_open_OBJECTS = test-open.$(OBJEXT)
 test_open_LDADD = $(LDADD)
@@ -893,9 +882,6 @@
 test_regex_SOURCES = test-regex.c
 test_regex_OBJECTS = test-regex.$(OBJEXT)
 test_regex_DEPENDENCIES = $(am__DEPENDENCIES_2) $(am__DEPENDENCIES_1)
-test_rwlock1_SOURCES = test-rwlock1.c
-test_rwlock1_OBJECTS = test-rwlock1.$(OBJEXT)
-test_rwlock1_DEPENDENCIES = $(am__DEPENDENCIES_2)
 test_select_SOURCES = test-select.c
 test_select_OBJECTS = test-select.$(OBJEXT)
 test_select_DEPENDENCIES = $(am__DEPENDENCIES_2) $(am__DEPENDENCIES_1)
@@ -1207,30 +1193,28 @@
 	test-inet_ntop.c test-inet_pton.c test-intprops.c \
 	test-inttypes.c test-ioctl.c test-isblank.c test-langinfo.c \
 	test-limits-h.c test-listen.c test-localcharset.c \
-	test-locale.c test-localeconv.c test-localename.c test-lock.c \
-	test-lseek.c test-lstat.c test-malloca.c test-mbrtowc.c \
-	test-mbrtowc-w32.c test-mbsinit.c test-memchr.c test-memmem.c \
-	test-nanosleep.c test-netdb.c test-netinet_in.c \
-	test-nl_langinfo.c $(test_once1_SOURCES) $(test_once2_SOURCES) \
-	test-open.c test-pathmax.c test-perror.c test-perror2.c \
-	test-pipe.c test-poll.c test-poll-h.c test-pthread_sigmask1.c \
+	test-locale.c test-localeconv.c test-localename.c test-lseek.c \
+	test-lstat.c test-malloca.c test-mbrtowc.c test-mbrtowc-w32.c \
+	test-mbsinit.c test-memchr.c test-memmem.c test-nanosleep.c \
+	test-netdb.c test-netinet_in.c test-nl_langinfo.c test-open.c \
+	test-pathmax.c test-perror.c test-perror2.c test-pipe.c \
+	test-poll.c test-poll-h.c test-pthread_sigmask1.c \
 	test-pthread_sigmask2.c test-raise.c test-regex.c \
-	test-rwlock1.c test-select.c test-select-fd.c \
-	test-select-stdin.c test-setenv.c test-setlocale1.c \
-	test-setlocale2.c test-setsockopt.c test-sigaction.c \
-	test-signal-h.c test-sigprocmask.c test-sleep.c \
-	test-snprintf.c test-sockets.c test-stat.c test-stat-time.c \
-	test-stdalign.c test-stdbool.c test-stddef.c test-stdint.c \
-	test-stdio.c test-stdlib.c test-strcasestr.c test-strerror.c \
-	test-strerror_r.c test-string.c test-strings.c test-strnlen.c \
-	test-symlink.c test-sys_ioctl.c test-sys_select.c \
-	test-sys_socket.c test-sys_stat.c test-sys_time.c \
-	test-sys_types.c test-sys_uio.c test-sys_utsname.c \
-	test-thread_create.c test-thread_self.c test-time.c test-tls.c \
-	test-uname.c test-unistd.c test-unsetenv.c test-usleep.c \
-	test-vasnprintf.c test-verify.c test-verify-try.c \
-	test-vsnprintf.c test-wchar.c test-wcrtomb.c \
-	test-wcrtomb-w32.c test-wctype-h.c test-write.c
+	test-select.c test-select-fd.c test-select-stdin.c \
+	test-setenv.c test-setlocale1.c test-setlocale2.c \
+	test-setsockopt.c test-sigaction.c test-signal-h.c \
+	test-sigprocmask.c test-sleep.c test-snprintf.c test-sockets.c \
+	test-stat.c test-stat-time.c test-stdalign.c test-stdbool.c \
+	test-stddef.c test-stdint.c test-stdio.c test-stdlib.c \
+	test-strcasestr.c test-strerror.c test-strerror_r.c \
+	test-string.c test-strings.c test-strnlen.c test-symlink.c \
+	test-sys_ioctl.c test-sys_select.c test-sys_socket.c \
+	test-sys_stat.c test-sys_time.c test-sys_types.c \
+	test-sys_uio.c test-sys_utsname.c test-thread_create.c \
+	test-thread_self.c test-time.c test-tls.c test-uname.c \
+	test-unistd.c test-unsetenv.c test-usleep.c test-vasnprintf.c \
+	test-verify.c test-verify-try.c test-vsnprintf.c test-wchar.c \
+	test-wcrtomb.c test-wcrtomb-w32.c test-wctype-h.c test-write.c
 DIST_SOURCES = $(libtests_a_SOURCES) $(EXTRA_libtests_a_SOURCES) \
 	test-accept.c test-alloca-opt.c test-arpa_inet.c \
 	test-binary-io.c test-bind.c test-btowc.c test-c-ctype.c \
@@ -1249,30 +1233,28 @@
 	test-inet_ntop.c test-inet_pton.c test-intprops.c \
 	test-inttypes.c test-ioctl.c test-isblank.c test-langinfo.c \
 	test-limits-h.c test-listen.c test-localcharset.c \
-	test-locale.c test-localeconv.c test-localename.c test-lock.c \
-	test-lseek.c test-lstat.c test-malloca.c test-mbrtowc.c \
-	test-mbrtowc-w32.c test-mbsinit.c test-memchr.c test-memmem.c \
-	test-nanosleep.c test-netdb.c test-netinet_in.c \
-	test-nl_langinfo.c $(test_once1_SOURCES) $(test_once2_SOURCES) \
-	test-open.c test-pathmax.c test-perror.c test-perror2.c \
-	test-pipe.c test-poll.c test-poll-h.c test-pthread_sigmask1.c \
+	test-locale.c test-localeconv.c test-localename.c test-lseek.c \
+	test-lstat.c test-malloca.c test-mbrtowc.c test-mbrtowc-w32.c \
+	test-mbsinit.c test-memchr.c test-memmem.c test-nanosleep.c \
+	test-netdb.c test-netinet_in.c test-nl_langinfo.c test-open.c \
+	test-pathmax.c test-perror.c test-perror2.c test-pipe.c \
+	test-poll.c test-poll-h.c test-pthread_sigmask1.c \
 	test-pthread_sigmask2.c test-raise.c test-regex.c \
-	test-rwlock1.c test-select.c test-select-fd.c \
-	test-select-stdin.c test-setenv.c test-setlocale1.c \
-	test-setlocale2.c test-setsockopt.c test-sigaction.c \
-	test-signal-h.c test-sigprocmask.c test-sleep.c \
-	test-snprintf.c test-sockets.c test-stat.c test-stat-time.c \
-	test-stdalign.c test-stdbool.c test-stddef.c test-stdint.c \
-	test-stdio.c test-stdlib.c test-strcasestr.c test-strerror.c \
-	test-strerror_r.c test-string.c test-strings.c test-strnlen.c \
-	test-symlink.c test-sys_ioctl.c test-sys_select.c \
-	test-sys_socket.c test-sys_stat.c test-sys_time.c \
-	test-sys_types.c test-sys_uio.c test-sys_utsname.c \
-	test-thread_create.c test-thread_self.c test-time.c test-tls.c \
-	test-uname.c test-unistd.c test-unsetenv.c test-usleep.c \
-	test-vasnprintf.c test-verify.c test-verify-try.c \
-	test-vsnprintf.c test-wchar.c test-wcrtomb.c \
-	test-wcrtomb-w32.c test-wctype-h.c test-write.c
+	test-select.c test-select-fd.c test-select-stdin.c \
+	test-setenv.c test-setlocale1.c test-setlocale2.c \
+	test-setsockopt.c test-sigaction.c test-signal-h.c \
+	test-sigprocmask.c test-sleep.c test-snprintf.c test-sockets.c \
+	test-stat.c test-stat-time.c test-stdalign.c test-stdbool.c \
+	test-stddef.c test-stdint.c test-stdio.c test-stdlib.c \
+	test-strcasestr.c test-strerror.c test-strerror_r.c \
+	test-string.c test-strings.c test-strnlen.c test-symlink.c \
+	test-sys_ioctl.c test-sys_select.c test-sys_socket.c \
+	test-sys_stat.c test-sys_time.c test-sys_types.c \
+	test-sys_uio.c test-sys_utsname.c test-thread_create.c \
+	test-thread_self.c test-time.c test-tls.c test-uname.c \
+	test-unistd.c test-unsetenv.c test-usleep.c test-vasnprintf.c \
+	test-verify.c test-verify-try.c test-vsnprintf.c test-wchar.c \
+	test-wcrtomb.c test-wcrtomb-w32.c test-wctype-h.c test-write.c
 RECURSIVE_TARGETS = all-recursive check-recursive cscopelist-recursive \
 	ctags-recursive dvi-recursive html-recursive info-recursive \
 	install-data-recursive install-dvi-recursive \
@@ -2672,9 +2654,8 @@
 	test-langinfo.c test-limits-h.c test-listen.c signature.h \
 	macros.h test-localcharset.c test-locale.c test-localeconv.c \
 	signature.h macros.h localename.h test-localename.c macros.h \
-	test-rwlock1.c test-lock.c test-once.c test-lseek.c \
-	test-lseek.sh signature.h macros.h lstat.c test-lstat.h \
-	test-lstat.c signature.h macros.h test-malloca.c \
+	test-lseek.c test-lseek.sh signature.h macros.h lstat.c \
+	test-lstat.h test-lstat.c signature.h macros.h test-malloca.c \
 	test-mbrtowc1.sh test-mbrtowc2.sh test-mbrtowc3.sh \
 	test-mbrtowc4.sh test-mbrtowc5.sh test-mbrtowc.c \
 	test-mbrtowc-w32-1.sh test-mbrtowc-w32-2.sh \
@@ -2767,12 +2748,6 @@
 test_inet_pton_LDADD = $(LDADD) @INET_PTON_LIB@
 test_listen_LDADD = $(LDADD) @LIBSOCKET@
 test_localename_LDADD = $(LDADD) @INTL_MACOSX_LIBS@ $(LIBTHREAD)
-test_rwlock1_LDADD = $(LDADD) @LIBMULTITHREAD@ @YIELD_LIB@
-test_lock_LDADD = $(LDADD) @LIBMULTITHREAD@ @YIELD_LIB@
-test_once1_SOURCES = test-once.c
-test_once1_LDADD = $(LDADD) @LIBTHREAD@
-test_once2_SOURCES = test-once.c
-test_once2_LDADD = $(LDADD) @LIBMULTITHREAD@
 test_nanosleep_LDADD = $(LDADD) $(LIB_NANOSLEEP)
 test_poll_LDADD = $(LDADD) $(LIB_POLL) @LIBSOCKET@ $(INET_PTON_LIB)
 test_pthread_sigmask1_LDADD = $(LDADD) @LIB_PTHREAD_SIGMASK@
@@ -3135,10 +3110,6 @@
 	@rm -f test-localename$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(test_localename_OBJECTS) $(test_localename_LDADD) $(LIBS)
 
-test-lock$(EXEEXT): $(test_lock_OBJECTS) $(test_lock_DEPENDENCIES) $(EXTRA_test_lock_DEPENDENCIES) 
-	@rm -f test-lock$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(test_lock_OBJECTS) $(test_lock_LDADD) $(LIBS)
-
 test-lseek$(EXEEXT): $(test_lseek_OBJECTS) $(test_lseek_DEPENDENCIES) $(EXTRA_test_lseek_DEPENDENCIES) 
 	@rm -f test-lseek$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(test_lseek_OBJECTS) $(test_lseek_LDADD) $(LIBS)
@@ -3187,14 +3158,6 @@
 	@rm -f test-nl_langinfo$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(test_nl_langinfo_OBJECTS) $(test_nl_langinfo_LDADD) $(LIBS)
 
-test-once1$(EXEEXT): $(test_once1_OBJECTS) $(test_once1_DEPENDENCIES) $(EXTRA_test_once1_DEPENDENCIES) 
-	@rm -f test-once1$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(test_once1_OBJECTS) $(test_once1_LDADD) $(LIBS)
-
-test-once2$(EXEEXT): $(test_once2_OBJECTS) $(test_once2_DEPENDENCIES) $(EXTRA_test_once2_DEPENDENCIES) 
-	@rm -f test-once2$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(test_once2_OBJECTS) $(test_once2_LDADD) $(LIBS)
-
 test-open$(EXEEXT): $(test_open_OBJECTS) $(test_open_DEPENDENCIES) $(EXTRA_test_open_DEPENDENCIES) 
 	@rm -f test-open$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(test_open_OBJECTS) $(test_open_LDADD) $(LIBS)
@@ -3239,10 +3202,6 @@
 	@rm -f test-regex$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(test_regex_OBJECTS) $(test_regex_LDADD) $(LIBS)
 
-test-rwlock1$(EXEEXT): $(test_rwlock1_OBJECTS) $(test_rwlock1_DEPENDENCIES) $(EXTRA_test_rwlock1_DEPENDENCIES) 
-	@rm -f test-rwlock1$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(test_rwlock1_OBJECTS) $(test_rwlock1_LDADD) $(LIBS)
-
 test-select$(EXEEXT): $(test_select_OBJECTS) $(test_select_DEPENDENCIES) $(EXTRA_test_select_DEPENDENCIES) 
 	@rm -f test-select$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(test_select_OBJECTS) $(test_select_LDADD) $(LIBS)
@@ -3547,7 +3506,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-locale.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-localeconv.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-localename.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-lock.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-lseek.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-lstat.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-malloca.Po@am__quote@
@@ -3560,7 +3518,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-netdb.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-netinet_in.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-nl_langinfo.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-once.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-open.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-pathmax.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-perror.Po@am__quote@
@@ -3572,7 +3529,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-pthread_sigmask2.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-raise.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-regex.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-rwlock1.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-select-fd.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-select-stdin.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-select.Po@am__quote@
@@ -4366,34 +4322,6 @@
 	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
 	--log-file $$b.log --trs-file $$b.trs \
 	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
-test-rwlock1.log: test-rwlock1$(EXEEXT)
-	@p='test-rwlock1$(EXEEXT)'; \
-	b='test-rwlock1'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
-test-lock.log: test-lock$(EXEEXT)
-	@p='test-lock$(EXEEXT)'; \
-	b='test-lock'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
-test-once1.log: test-once1$(EXEEXT)
-	@p='test-once1$(EXEEXT)'; \
-	b='test-once1'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
-test-once2.log: test-once2$(EXEEXT)
-	@p='test-once2$(EXEEXT)'; \
-	b='test-once2'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
 	"$$tst" $(AM_TESTS_FD_REDIRECT)
 test-lseek.sh.log: test-lseek.sh
 	@p='test-lseek.sh'; \
--- ./libmissing/tests/Makefile.am	2018-12-23 21:59:17.836000000 +0100
+++ ./libmissing/tests/Makefile.am	2019-02-27 00:25:24.928094253 +0100
@@ -770,20 +770,6 @@
 
 ## end   gnulib module localename-tests
 
-## begin gnulib module lock-tests
-
-TESTS += test-rwlock1 test-lock test-once1 test-once2
-check_PROGRAMS += test-rwlock1 test-lock test-once1 test-once2
-test_rwlock1_LDADD = $(LDADD) @LIBMULTITHREAD@ @YIELD_LIB@
-test_lock_LDADD = $(LDADD) @LIBMULTITHREAD@ @YIELD_LIB@
-test_once1_SOURCES = test-once.c
-test_once1_LDADD = $(LDADD) @LIBTHREAD@
-test_once2_SOURCES = test-once.c
-test_once2_LDADD = $(LDADD) @LIBMULTITHREAD@
-EXTRA_DIST += test-rwlock1.c test-lock.c test-once.c
-
-## end   gnulib module lock-tests
-
 ## begin gnulib module lseek-tests
 
 TESTS += test-lseek.sh
