From: Thomas Andrejak <thomas.andrejak@gmail.com>
Date: 2017-10-04 07:00:00 +0100
References: http://prelude-siem.org/issues/859
Upstream: submitted
Subject: Correctly link for Perl bindings

diff -up ./bindings/perl/Makefile.PL.in.orig ./bindings/perl/Makefile.PL.in
--- ./bindings/perl/Makefile.PL.in.orig	2016-10-16 17:13:27.006400921 +0300
+++ ./bindings/perl/Makefile.PL.in	2016-10-16 17:14:39.521628420 +0300
@@ -26,8 +26,8 @@ use Cwd "abs_path";
 
 my %attributs = (NAME => 'Prelude',
 		 INC => '-I@top_builddir@ -I@top_builddir@/src/include -I@top_srcdir@/src/include -I@top_builddir@/src/libprelude-error -I@top_srcdir@/bindings/c++/include',
-		 LIBS => ["-L@top_builddir@/src/.libs -lprelude -L@top_builddir@/bindings/c++/.libs -lpreludecpp", "-L$ENV{LIBDIR} -lprelude @LIBPRELUDE_LIBS@ @LIBADD_DL@ @LTLIBTHREAD@" ],
-		 LDDLFLAGS => "-L@top_builddir@/src/.libs $Config{lddlflags} @LDFLAGS@,-rpath");
+		 LIBS => ["-L@top_builddir@/src/.libs -lprelude -L@top_builddir@/bindings/c++/.libs -lpreludecpp -L$ENV{LIBDIR} -lstdc++ -lm @LIBPRELUDE_LIBS@ @LIBADD_DL@ @LTLIBTHREAD@" ],
+		 LDDLFLAGS => "-L@top_builddir@/src/.libs $Config{lddlflags} @LDFLAGS@");
 
 if ( abs_path("@top_srcdir@") ne abs_path("@top_builddir@") ) {
     foreach my $filename ( qw/Prelude.cxx Prelude.pm/ ) {
