--- a/libcxxabi/test/forced_unwind3.pass.cpp
+++ b/libcxxabi/test/forced_unwind3.pass.cpp
@@ -47,7 +47,7 @@ struct Stop<R (*)(Args...)> {
   // libunwind while _Unwind_Exception_Class in libgcc.
   typedef typename std::tuple_element<2, std::tuple<Args...>>::type type;
 
-  static _Unwind_Reason_Code stop(int, _Unwind_Action actions, type, struct _Unwind_Exception*, struct _Unwind_Context*,
+  static _Unwind_Reason_Code stop(int, _Unwind_Action actions, type, _Unwind_Exception*, struct _Unwind_Context*,
                                   void*) {
     if (actions & _UA_END_OF_STACK) {
       assert(destructorCalled == true);
