Index: lld-15.0.0rc2.src/ELF/Options.td
===================================================================
--- lld-15.0.0rc2.src.orig/ELF/Options.td
+++ lld-15.0.0rc2.src/ELF/Options.td
@@ -52,7 +52,7 @@ def Bstatic: F<"Bstatic">, HelpText<"Do
 
 def build_id: J<"build-id=">, HelpText<"Generate build ID note">,
   MetaVarName<"[fast,md5,sha1,uuid,0x<hexstring>]">;
-def : F<"build-id">, Alias<build_id>, AliasArgs<["fast"]>, HelpText<"Alias for --build-id=fast">;
+def : F<"build-id">, Alias<build_id>, AliasArgs<["sha1"]>, HelpText<"Alias for --build-id=sha1">;
 
 defm check_sections: B<"check-sections",
     "Check section addresses for overlaps (default)",
