From 305813b6039fe2ef5d5f4b246355078c2bb620da Mon Sep 17 00:00:00 2001
From: tjuhasz <tjuhasz@redhat.com>
Date: Wed, 13 May 2026 15:26:14 +0200
Subject: [PATCH] deps: fix V8 build failures by adding climits include

Add explicit #include <climits> to memcopy.h to fix build failures
caused by recent GCC/Clang versions removing transient inclusions
of the climits header.

Fixes compilation errors when using newer toolchain versions.
---
 deps/v8/src/base/memcopy.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/deps/v8/src/base/memcopy.h b/deps/v8/src/base/memcopy.h
index b1d8a36b4cfe..82e9a75c24a8 100644
--- a/deps/v8/src/base/memcopy.h
+++ b/deps/v8/src/base/memcopy.h
@@ -5,6 +5,7 @@
 #ifndef V8_BASE_MEMCOPY_H_
 #define V8_BASE_MEMCOPY_H_
 
+#include <climits>
 #include <stdlib.h>
 
 #include <atomic>
