Index: node-v26.6.0/deps/npm/lib/base-cmd.js
===================================================================
--- node-v26.6.0.orig/deps/npm/lib/base-cmd.js
+++ node-v26.6.0/deps/npm/lib/base-cmd.js
@@ -104,7 +104,7 @@ class BaseCommand {
         fullUsage.push('')
         for (const def of cmdDefs) {
           if (def.description) {
-            const desc = def.description.trim().split('\n')[0]
+            const desc = def.description.split('\n\n')[0].trim().replace(/\s+/g, ' ')
             const shortcuts = def.short ? `-${def.short}` : ''
             const aliases = (def.alias || []).map(v => `--${v}`).join('|')
             const mainFlag = `--${def.key}`
