diff -ur pcc-libs-1.0.0.orig/csu/linux/Makefile pcc-libs-1.0.0/csu/linux/Makefile
--- pcc-libs-1.0.0.orig/csu/linux/Makefile	2009-08-13 11:16:46.000000000 +0200
+++ pcc-libs-1.0.0/csu/linux/Makefile	2011-04-14 16:48:02.776999991 +0200
@@ -4,7 +4,7 @@
 PCCLIBDIR ?= /usr/local/lib/pcc/$(TARGET)/$(VERSION)/lib
 SRCDIR ?= .
 
-CFLAGS = -I$(SRCDIR)/$(MACH) -I$(SRCDIR) -Wall -Wstrict-prototypes -Wmissing-prototypes
+CFLAGS := -I$(SRCDIR)/$(MACH) -I$(SRCDIR) $(filter-out -g,$(CFLAGS))
 
 OBJS = crt0.o crt1.o gcrt1.o crti.o crtn.o crtbegin.o crtend.o
 
diff -ur pcc-libs-1.0.0.orig/libF77/Makefile pcc-libs-1.0.0/libF77/Makefile
--- pcc-libs-1.0.0.orig/libF77/Makefile	2008-02-28 17:48:50.000000000 +0100
+++ pcc-libs-1.0.0/libF77/Makefile	2011-04-14 16:48:21.876999993 +0200
@@ -1,6 +1,6 @@
 # compile, then strip unnecessary symbols
 
-CFLAGS=-Wall -Werror -Wstrict-prototypes -Wmissing-prototypes
+#CFLAGS=-Wall -Werror -Wstrict-prototypes -Wmissing-prototypes
 
 .c.o :
 	cc -O $(CFLAGS) -c $<
diff -ur pcc-libs-1.0.0.orig/libI77/Makefile pcc-libs-1.0.0/libI77/Makefile
--- pcc-libs-1.0.0.orig/libI77/Makefile	2008-03-01 14:44:12.000000000 +0100
+++ pcc-libs-1.0.0/libI77/Makefile	2011-04-14 16:48:33.166999976 +0200
@@ -2,7 +2,7 @@
 # Simple makefile for the F77 I/O library.
 #
 
-CFLAGS=-Wall -Werror -Wstrict-prototypes -Wmissing-prototypes
+#CFLAGS=-Wall -Werror -Wstrict-prototypes -Wmissing-prototypes
 
 .c.o :
 	cc -O $(CFLAGS) -c $<
