Index: thinkfan-1.3.0/rcscripts/systemd/thinkfan-sleep.service
===================================================================
--- thinkfan-1.3.0.orig/rcscripts/systemd/thinkfan-sleep.service
+++ thinkfan-1.3.0/rcscripts/systemd/thinkfan-sleep.service
@@ -3,6 +3,16 @@ Description=Notify thinkfan of imminent
 Before=sleep.target
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+ProtectHostname=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+RestrictRealtime=true
+# end of automatic additions 
 Type=oneshot
 ExecStart=/usr/bin/pkill -x -winch thinkfan
 # Hack: Since the signal handler races with the sleep, we need to delay a bit
