Index: thinkfan-1.3.0/rcscripts/systemd/thinkfan-wakeup.service
===================================================================
--- thinkfan-1.3.0.orig/rcscripts/systemd/thinkfan-wakeup.service
+++ thinkfan-1.3.0/rcscripts/systemd/thinkfan-wakeup.service
@@ -7,6 +7,15 @@ After=hybrid-sleep.target
 After=hibernate.target
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+ProtectHostname=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+RestrictRealtime=true
+# end of automatic additions 
 Type=oneshot
 ExecStart=/usr/bin/pkill -x -usr2 thinkfan
 
