Index: libharu-RELEASE_2_3_0/CMakeLists.txt
===================================================================
--- libharu-RELEASE_2_3_0.orig/CMakeLists.txt
+++ libharu-RELEASE_2_3_0/CMakeLists.txt
@@ -99,6 +99,8 @@ if(PNG_FOUND)
   set(ADDITIONAL_LIBRARIES ${ADDITIONAL_LIBRARIES} ${PNG_LIBRARIES})
 endif(PNG_FOUND)
 
+find_library(M_LIB m)
+set(ADDITIONAL_LIBRARIES ${ADDITIONAL_LIBRARIES} ${M_LIB})
 
 # =======================================================================
 # configure header files, add compiler flags
