From 576248aa816e42204451b1918644fc07db6aa95b Mon Sep 17 00:00:00 2001
From: Ludwig Nussel <ludwig.nussel@suse.de>
Date: Wed, 19 May 2021 16:38:21 +0200
Subject: [PATCH] Gracefully handle unavailable libreport

---
 framework/src/setroubleshoot/browser.py | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

Index: setroubleshoot-3.3.26/src/setroubleshoot/browser.py
===================================================================
--- setroubleshoot-3.3.26.orig/src/setroubleshoot/browser.py
+++ setroubleshoot-3.3.26/src/setroubleshoot/browser.py
@@ -65,10 +65,13 @@ from setroubleshoot.util import *
 from setroubleshoot.html_util import html_to_text
 import re
 import dbus
-import report
-import report.io
-import report.io.GTKIO
-import report.accountmanager
+try:
+    import report
+    import report.io
+    import report.io.GTKIO
+    import report.accountmanager
+except ImportError:
+    report = None
 
 from gi.repository import Gio
 
@@ -279,7 +282,8 @@ class BrowserApplet:
         self.make_treeview()
         self.troubleshoot_visible = False
         self.current_alert = -1
-        self.accounts = report.accountmanager.AccountManager()
+        if report is not None:
+            self.accounts = report.accountmanager.AccountManager()
 
     def get_current_alert(self):
         try:
@@ -539,7 +543,7 @@ class BrowserApplet:
             report_button.connect("clicked", self.fix_bug, alert.local_id, plugin.analysis_id)
             vbox.add(report_button)
 
-        elif plugin.report_bug:
+        elif plugin.report_bug and report is not None:
             report_button = Gtk.Button()
             report_button.set_label(_("Report\nBug"))
             report_button.show()
