Index: devscripts-v2.26.9/scripts/debcommit.pl
===================================================================
--- devscripts-v2.26.9.orig/scripts/debcommit.pl
+++ devscripts-v2.26.9/scripts/debcommit.pl
@@ -686,7 +686,9 @@ sub commit {
     } else {
         die "debcommit: unknown program $prog";
     }
-    die "debcommit: commit failed\n" if (!$action_rc);
+    # mercurial 1.6 returns failure if nothing is to be committed
+    # but we still want to continue with tagging
+    die "debcommit: commit failed\n" if (!$action_rc) && (!$release);
 }
 
 sub tag {
