--- a/scripts/Makefile
+++ b/scripts/Makefile
@@ -121,7 +120,7 @@ clean:
 test: test_pl test_sh test_py
 
 install: all
-	python3 setup.py install --root="$(DESTDIR)" --no-compile --install-layout=deb
+	python3 setup.py install --root="$(DESTDIR)" --prefix="$(PREFIX)" --no-compile
 	cp $(SCRIPTS) $(DESTDIR)$(BINDIR)
 	ln -sf edit-patch $(DESTDIR)$(BINDIR)/add-patch
 	install -d $(DESTDIR)$(COMPL_DIR)
