--- tvm-0.21.0/python/tvm/libinfo.py.orig	2025-07-07 16:48:21.000000000 +0200
+++ tvm-0.21.0/python/tvm/libinfo.py	2025-09-24 09:12:48.885453735 +0200
@@ -67,8 +67,8 @@ def get_dll_directories():
     # Pip lib directory
     dll_path.append(ffi_dir)
     # Default cmake build directory
-    dll_path.append(os.path.join(source_dir, "build"))
-    dll_path.append(os.path.join(source_dir, "build", "Release"))
+    dll_path.append(os.path.join(source_dir, "build_cmake"))
+    dll_path.append(os.path.join(source_dir, "build_cmake", "Release"))
     # Default make build directory
     dll_path.append(os.path.join(source_dir, "lib"))
 
