From 028d750249ab00c8c362598c838b15e8aa6a2951 Mon Sep 17 00:00:00 2001
From: Michael Black W9MDB <mdblack98@yahoo.com>
Date: Sun, 29 Dec 2024 10:43:39 -0600
Subject: [PATCH 5/6] Fix rig_list_foreach in testmW2power.c

---
 tests/testmW2power.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/testmW2power.c b/tests/testmW2power.c
index 88c38b6ed..9e910e95c 100644
--- a/tests/testmW2power.c
+++ b/tests/testmW2power.c
@@ -13,7 +13,7 @@
 
 int nrigs = 0;
 
-int callback(struct rig_caps *caps, rig_ptr_t rigp)
+int callback(const struct rig_caps *caps, rig_ptr_t rigp)
 {
     RIG *rig = (RIG *) rigp;
 
-- 
2.47.1

