From 56ad12fe568675957e87c88a7b9f318b52b07fc4 Mon Sep 17 00:00:00 2001
From: Henry Schreiner <henryschreineriii@gmail.com>
Date: Tue, 2 Jul 2024 08:23:03 -0400
Subject: [PATCH] tests: fix for distutils change

Signed-off-by: Henry Schreiner <henryschreineriii@gmail.com>
---
 tests/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/__init__.py b/tests/__init__.py
index 00d27289..4f3c9543 100644
--- a/tests/__init__.py
+++ b/tests/__init__.py
@@ -215,7 +215,7 @@ def execute_setup_py(project_dir, setup_args, disable_languages_test=False):
     """
 
     # See https://stackoverflow.com/questions/9160227/dir-util-copy-tree-fails-after-shutil-rmtree
-    distutils.dir_util._path_created = {}  # type: ignore[attr-defined]
+    distutils.dir_util._path_created.clear()  # type: ignore[attr-defined]
 
     # Clear _PYTHON_HOST_PLATFORM to ensure value sets in skbuild.setuptools_wrap.setup() does not
     # influence other tests.
