diff --git a/unix/fbiterm/src/term.c b/unix/fbiterm/src/term.c
index 18c9945..7adc3b5 100644
--- a/unix/fbiterm/src/term.c
+++ b/unix/fbiterm/src/term.c
@@ -17,7 +17,9 @@ void
 init_tty ()
 {
   struct termios t;
-  write(STDOUT_FILENO, "\x1b=",2); /* to switch keypad application modes */
+
+  // don't do this (bsc#1158271)
+  // write(STDOUT_FILENO, "\x1b=",2); /* to switch keypad application modes */
 
   /* save current parameters */
   tcgetattr (STDIN_FILENO, &(pIterm->ntio));
