Index: b/channels/video/client/CMakeLists.txt
===================================================================
--- a/channels/video/client/CMakeLists.txt
+++ b/channels/video/client/CMakeLists.txt
@@ -26,7 +26,10 @@ include_directories(..)
 add_channel_client_library(${MODULE_PREFIX} ${MODULE_NAME} ${CHANNEL_NAME} TRUE "DVCPluginEntry")
 
 
-
+if (NOT BUILTIN_CHANNELS)
+	set(${MODULE_PREFIX}_LIBS ${${MODULE_PREFIX}_LIBS} freerdp-client)
+	set(${MODULE_PREFIX}_LIBS ${${MODULE_PREFIX}_LIBS} rdpgfx-client)
+endif()
 set(${MODULE_PREFIX}_LIBS ${${MODULE_PREFIX}_LIBS} winpr)
 
 target_link_libraries(${MODULE_NAME} ${${MODULE_PREFIX}_LIBS})
Index: b/channels/rdpsnd/client/proxy/CMakeLists.txt
===================================================================
--- a/channels/rdpsnd/client/proxy/CMakeLists.txt
+++ b/channels/rdpsnd/client/proxy/CMakeLists.txt
@@ -25,6 +25,9 @@ include_directories(..)
 
 add_channel_client_subsystem_library(${MODULE_PREFIX} ${MODULE_NAME} ${CHANNEL_NAME} "" TRUE "")
 
+if (NOT BUILTIN_CHANNELS)
+	list(APPEND ${MODULE_PREFIX}_LIBS freerdp-client)
+endif()
 list(APPEND ${MODULE_PREFIX}_LIBS freerdp)
 list(APPEND ${MODULE_PREFIX}_LIBS winpr)
 
Index: b/channels/rdp2tcp/client/CMakeLists.txt
===================================================================
--- a/channels/rdp2tcp/client/CMakeLists.txt
+++ b/channels/rdp2tcp/client/CMakeLists.txt
@@ -22,6 +22,6 @@ set(${MODULE_PREFIX}_SRCS
 
 add_channel_client_library(${MODULE_PREFIX} ${MODULE_NAME} ${CHANNEL_NAME} TRUE "VirtualChannelEntryEx")
 
-target_link_libraries(${MODULE_NAME} freerdp)
+target_link_libraries(${MODULE_NAME} winpr freerdp)
 
 set_property(TARGET ${MODULE_NAME} PROPERTY FOLDER "Channels/${CHANNEL_NAME}/Client")
