From 8cb789af39e93edb07f1eee39c87908e0d7c5ee5 Mon Sep 17 00:00:00 2001
From: Tony Zorman <soliditsallgood@mailbox.org>
Date: Thu, 3 Nov 2022 08:25:00 +0100
Subject: [PATCH] CI: Compatibility with mtl 2.3.1

Libraries like Control.Monad are no longer exported from
Control.Monad.Reader et.al.

Related: https://github.com/xmonad/xmonad/pull/427
         xmonad/xmonad#d170e99bc5e97db96be9a02b72149103e8d419af
---
 XMonad/Actions/GroupNavigation.hs | 6 +++---
 XMonad/Hooks/WorkspaceByPos.hs    | 3 ++-
 XMonad/Util/PureX.hs              | 2 +-
 3 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/XMonad/Actions/GroupNavigation.hs b/XMonad/Actions/GroupNavigation.hs
index ef77f7b38..ff0410d90 100644
--- a/XMonad/Actions/GroupNavigation.hs
+++ b/XMonad/Actions/GroupNavigation.hs
@@ -33,8 +33,8 @@ module XMonad.Actions.GroupNavigation ( -- * Usage
                                       , isOnAnyVisibleWS
                                       ) where
 
-import Control.Monad.Reader
-import Control.Monad.State
+import Control.Monad.Reader (ask, asks)
+import Control.Monad.State (gets)
 import Control.DeepSeq
 import Data.Map ((!))
 import qualified Data.Map as Map
@@ -47,7 +47,7 @@ import Prelude hiding (drop, elem, filter, null, reverse)
 import XMonad.Core
 import XMonad.ManageHook
 import XMonad.Operations (windows, withFocused)
-import XMonad.Prelude (elem, foldl')
+import XMonad.Prelude (elem, foldl', (>=>))
 import qualified XMonad.StackSet as SS
 import qualified XMonad.Util.ExtensibleState as XS
 
diff --git a/XMonad/Hooks/WorkspaceByPos.hs b/XMonad/Hooks/WorkspaceByPos.hs
index 19ef77944..e073af63f 100644
--- a/XMonad/Hooks/WorkspaceByPos.hs
+++ b/XMonad/Hooks/WorkspaceByPos.hs
@@ -26,7 +26,8 @@ import XMonad
 import XMonad.Prelude
 import qualified XMonad.StackSet as W
 
-import Control.Monad.Except (lift, runExceptT, throwError)
+import Control.Monad.Except (runExceptT, throwError)
+import Control.Monad.Trans (lift)
 
 -- $usage
 -- You can use this module with the following in your @~\/.xmonad\/xmonad.hs@:
diff --git a/XMonad/Util/PureX.hs b/XMonad/Util/PureX.hs
index e0bdf133d..2a360ce87 100644
--- a/XMonad/Util/PureX.hs
+++ b/XMonad/Util/PureX.hs
@@ -53,7 +53,7 @@ module XMonad.Util.PureX (
 
 -- xmonad
 import XMonad
-import XMonad.Prelude (Any (..), liftA2, WindowScreen)
+import XMonad.Prelude
 import qualified XMonad.StackSet as W
 import qualified XMonad.Actions.FocusNth
 
