Index: owntone-28.2/owntone.service.in
===================================================================
--- owntone-28.2.orig/owntone.service.in
+++ owntone-28.2/owntone.service.in
@@ -5,6 +5,19 @@ Requires=network.target local-fs.target
 After=network.target sound.target remote-fs.target pulseaudio.service
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+PrivateDevices=true
+ProtectHostname=true
+ProtectClock=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 ExecStart=@sbindir@/owntone -f
 
 # Constrain the upper limit of memory/swap that can be used; this prevents 
