Index: gnome-shell-48.rc/js/ui/endSessionDialog.js
===================================================================
--- gnome-shell-48.rc.orig/js/ui/endSessionDialog.js
+++ gnome-shell-48.rc/js/ui/endSessionDialog.js
@@ -373,7 +373,10 @@ class EndSessionDialog extends ModalDial
 
         let description;
         let displayTime = _roundSecondsToInterval(
-            this._totalSecondsToStayOpen, this._secondsLeft, 10);
+            this._totalSecondsToStayOpen,
+            this._secondsLeft,
+            // larger than any normal value
+            1000000);
 
         if (this._user.is_loaded) {
             let realName = this._user.get_real_name();
