Index: gnome-shell-3.34.5/js/ui/unlockDialog.js
===================================================================
--- gnome-shell-3.34.5.orig/js/ui/unlockDialog.js
+++ gnome-shell-3.34.5/js/ui/unlockDialog.js
@@ -23,6 +23,10 @@ var UnlockDialog = GObject.registerClass
             visible: false,
         });
 
+        let path = '/usr/share/gnome-shell/theme/SLE-background.png';
+        if (GLib.file_test(path, GLib.FileTest.EXISTS))
+            this.style = 'background: url(' + path + ')';
+
         this.add_constraint(new Layout.MonitorConstraint({ primary: true }));
         parentActor.add_child(this);
 
