Index: gnome-shell-3.24.2/js/gdm/authPrompt.js
===================================================================
--- gnome-shell-3.24.2.orig/js/gdm/authPrompt.js
+++ gnome-shell-3.24.2/js/gdm/authPrompt.js
@@ -458,8 +458,10 @@ const AuthPrompt = new Lang.Class({
         this.setUser(null);
         this.stopSpinning();
 
-        if (oldStatus == AuthPromptStatus.VERIFICATION_FAILED)
+        if (oldStatus == AuthPromptStatus.VERIFICATION_FAILED) {
+            this._userVerifier.cancel();
             this.emit('failed');
+        }
 
         let beginRequestType;
 
