diff --git a/js/ui/endSessionDialog.js b/js/ui/endSessionDialog.js
index 42ae0d3f2..3e5eb94a8 100644
--- a/js/ui/endSessionDialog.js
+++ b/js/ui/endSessionDialog.js
@@ -270,7 +270,7 @@ class EndSessionDialog extends ModalDialog.ModalDialog {
         this.contentLayout.add_child(this._applicationSection);
 
         this._sessionSection = new Dialog.ListSection({
-            title: _('Other users are logged in'),
+            title: _('Other users are logged in. You can list these users with the "who" command.'),
         });
         this.contentLayout.add_child(this._sessionSection);
 
@@ -643,8 +643,8 @@ class EndSessionDialog extends ModalDialog.ModalDialog {
             this._applications.push(inhibitor);
         }
 
-        if (dialogContent.showOtherSessions)
-            this._loadSessions();
+        // if (dialogContent.showOtherSessions) // Hide _sessionList for jsc#SLE-9267
+        //    this._loadSessions();
 
         // Only consider updates and upgrades if PackageKit is available.
         let updateTriggered = this._pkOfflineProxy ? this._pkOfflineProxy.UpdateTriggered : false;
