--- gnome-shell-41.orig/js/ui/endSessionDialog.js	2022-08-11 16:16:07.000000000 +0300
+++ gnome-shell-41/js/ui/endSessionDialog.js	2022-10-26 12:49:20.435238071 +0300
@@ -707,19 +703,7 @@
     }
 
     async _getUpdateInfo() {
-        const connection = this._pkOfflineProxy.get_connection();
-        const reply = await connection.call(
-            this._pkOfflineProxy.g_name,
-            this._pkOfflineProxy.g_object_path,
-            'org.freedesktop.DBus.Properties',
-            'GetAll',
-            new GLib.Variant('(s)', [this._pkOfflineProxy.g_interface_name]),
-            null,
-            Gio.DBusCallFlags.NONE,
-            -1,
-            null);
-        const [info] = reply.recursiveUnpack();
-        return info;
+        return [];
     }
 
     async OpenAsync(parameters, invocation) {
         
         