From 4288b191efc7c5b86fb6317cc291d66adcc61588 Mon Sep 17 00:00:00 2001
From: Matthias Kestenholz <mk@feinheit.ch>
Date: Sun, 5 May 2019 11:51:51 +0200
Subject: [PATCH] Django 2.2's admin app checks for SessionMiddleware's
 presence

---
 tests/test_integration.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/test_integration.py b/tests/test_integration.py
index 129e819d..2e5d1c2d 100644
--- a/tests/test_integration.py
+++ b/tests/test_integration.py
@@ -354,6 +354,7 @@ class DebugToolbarSystemChecksTestCase(SimpleTestCase):
     @override_settings(
         MIDDLEWARE=[
             "django.contrib.messages.middleware.MessageMiddleware",
+            "django.contrib.sessions.middleware.SessionMiddleware",
             "django.contrib.auth.middleware.AuthenticationMiddleware",
             "django.middleware.gzip.GZipMiddleware",
             "debug_toolbar.middleware.DebugToolbarMiddleware",
@@ -366,6 +367,7 @@ def test_check_good_configuration(self):
     @override_settings(
         MIDDLEWARE=[
             "django.contrib.messages.middleware.MessageMiddleware",
+            "django.contrib.sessions.middleware.SessionMiddleware",
             "django.contrib.auth.middleware.AuthenticationMiddleware",
         ]
     )
@@ -387,6 +389,7 @@ def test_check_missing_middleware_error(self):
     @override_settings(
         MIDDLEWARE=[
             "django.contrib.messages.middleware.MessageMiddleware",
+            "django.contrib.sessions.middleware.SessionMiddleware",
             "django.contrib.auth.middleware.AuthenticationMiddleware",
             "debug_toolbar.middleware.DebugToolbarMiddleware",
             "django.middleware.gzip.GZipMiddleware",
