diff -rub secrets/tests/test_element.py secrets-patched/tests/test_element.py
--- secrets/tests/test_element.py	2025-04-10 15:02:25.582243334 +0200
+++ secrets-patched/tests/test_element.py	2025-04-10 15:22:17.981158388 +0200
@@ -1,7 +1,5 @@
 # SPDX-License-Identifier: GPL-3.0-only
 import os
-import sys
-sys.path.append(os.environ['G_TEST_BUILDDIR'] + "/../")
 
 import pytest
 from pykeepass import PyKeePass
diff -rub secrets/tests/test_password_generator.py secrets-patched/tests/test_password_generator.py
--- secrets/tests/test_password_generator.py	2025-04-10 15:02:25.582243334 +0200
+++ secrets-patched/tests/test_password_generator.py	2025-04-10 15:22:45.245922013 +0200
@@ -1,8 +1,5 @@
 # SPDX-License-Identifier: GPL-3.0-only
 import os
-import sys
-
-sys.path.append(os.environ["G_TEST_BUILDDIR"] + "/../")
 
 import unittest
 from gsecrets.password_generator import _satisfies_requirements
