---
 configure.ac |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: libaal-1.0.7/configure.ac
===================================================================
--- libaal-1.0.7.orig/configure.ac
+++ libaal-1.0.7/configure.ac
@@ -127,7 +127,7 @@ else
   ALIGN_FLAGS="-falign-jumps=1 -falign-loops=1 -falign-functions=1"
 fi
 
-CFLAGS=""
+CFLAGS="$RPM_OPT_FLAGS"
 MINIMAL_CFLAGS=""
 GENERIC_CFLAGS=""
 
