diff -Naur libspnav-0.2.3.orig/Makefile.in libspnav-0.2.3/Makefile.in
--- libspnav-0.2.3.orig/Makefile.in	2014-08-17 20:38:31.000000000 -0500
+++ libspnav-0.2.3/Makefile.in	2014-08-18 13:02:08.243930849 -0500
@@ -54,9 +54,11 @@
 	cp $(lib_so) $(DESTDIR)$(PREFIX)/$(libdir)/$(lib_so)
 	[ -n "$(soname)" ] && \
 		rm -f $(DESTDIR)$(PREFIX)/$(libdir)/$(soname) $(DESTDIR)$(PREFIX)/$(libdir)/$(devlink) && \
-		ln -s $(DESTDIR)$(PREFIX)/$(libdir)/$(lib_so) $(DESTDIR)$(PREFIX)/$(libdir)/$(soname) && \
-		ln -s $(DESTDIR)$(PREFIX)/$(libdir)/$(soname) $(DESTDIR)$(PREFIX)/$(libdir)/$(devlink) || \
-		true
+		pushd $(DESTDIR)$(PREFIX)/$(libdir) && \
+		ln -s $(lib_so) $(soname) && \
+		ln -s $(soname) $(devlink) || \
+		true && \
+		popd
 	for h in $(hdr); do cp -p $(srcdir)/$$h $(DESTDIR)$(PREFIX)/include/; done
 
 .PHONY: uninstall

