diff -Nur orig-nautilus-sendto-3.8.6/src/meson.build nautilus-sendto-3.8.6/src/meson.build
--- orig-nautilus-sendto-3.8.6/src/meson.build	2022-02-02 20:04:56.691407033 +0100
+++ nautilus-sendto-3.8.6/src/meson.build	2022-02-02 20:05:30.791824431 +0100
@@ -7,7 +7,7 @@
 
 po_dir = join_paths(meson.source_root(), 'po')
 
-i18n.merge_file ('appdata',
+i18n.merge_file (
                  input: 'nautilus-sendto.metainfo.xml.in',
                  output: 'nautilus-sendto.metainfo.xml',
                  install: true,
