Index: nagios_check_drbd9-0.1/check_drbd9.pl
===================================================================
--- nagios_check_drbd9-0.1.orig/check_drbd9.pl
+++ nagios_check_drbd9-0.1/check_drbd9.pl
@@ -631,7 +631,7 @@ sub chk_drbd_walk($)
 
 
    # builds resources
-   $sh_resources = `/usr/sbin/drbdadm sh-resources 2> /dev/null`;
+   $sh_resources = `drbdadm sh-resources 2> /dev/null`;
    if ($? != 0)
    {
       printf("DRBD UNKNOWN: error running drbdadm\n");
@@ -660,7 +660,7 @@ sub chk_drbd_walk($)
 
 
    # read events
-   @lines = `/usr/sbin/drbdsetup events2 --now --statistics all 2> /dev/null`;
+   @lines = `drbdsetup events2 --now --statistics all 2> /dev/null`;
    if ($? != 0)
    {
       printf("DRBD UNKNOWN: error running drbdsetup\n");
Index: nagios_check_drbd9-0.1/dump_drbd9.pl
===================================================================
--- nagios_check_drbd9-0.1.orig/dump_drbd9.pl
+++ nagios_check_drbd9-0.1/dump_drbd9.pl
@@ -162,7 +162,7 @@ sub chk_drbd_walk($)
 
 
    # builds resources
-   $sh_resources = `/usr/sbin/drbdadm sh-resources`;
+   $sh_resources = `drbdadm sh-resources`;
    if ($sh_resources =~ /^([-_.\w\s]+)$/)
    {
       $sh_resources = $1;
@@ -182,7 +182,7 @@ sub chk_drbd_walk($)
 
 
    # read events
-   @lines = `/usr/sbin/drbdsetup events2 --now --statistics all`;
+   @lines = `drbdsetup events2 --now --statistics all`;
    chomp(@lines);
 
 
