diff --git a/tests/standalone/run_all.py b/tests/standalone/run_all.py
index 371ba8a6a..f3c63c6fa 100755
--- a/tests/standalone/run_all.py
+++ b/tests/standalone/run_all.py
@@ -311,6 +311,9 @@ def main():
             if loaded_filename.startswith("/etc/"):
                 continue
 
+            if loaded_filename.startswith("/usr/etc/"):
+                continue
+
             if loaded_filename.startswith("/proc/") or loaded_filename == "/proc":
                 continue
 
