diff --git a/setup.py b/setup.py
index 95338f84ffe43ad13758c7f0d87d96c3af45822d..452df95adb23b562214bcf8ec09f7dca68807c01 100755
--- a/setup.py
+++ b/setup.py
@@ -66,7 +66,7 @@ for dep in info.get('depends', []):
         requires.append(get_require_version('trytond_%s' % dep))
 requires.append(get_require_version('trytond'))
 
-tests_require = [get_require_version('proteus'), 'pycountry<=20.7.3']
+tests_require = [get_require_version('proteus'), 'pycountry']
 dependency_links = []
 if minor_version % 2:
     dependency_links.append(
@@ -142,7 +142,7 @@ setup(name=name,
     python_requires='>=3.6',
     install_requires=requires,
     extras_require={
-        'data': ['pycountry<=20.7.3', get_require_version('proteus')],
+        'data': ['pycountry', get_require_version('proteus')],
         'GeoNames': [get_require_version('proteus')],
         },
     dependency_links=dependency_links,
