Index: qcs-api-client-python-0.23.1/qcs_api_client/client/_configuration/configuration.py
===================================================================
--- qcs-api-client-python-0.23.1.orig/qcs_api_client/client/_configuration/configuration.py
+++ qcs-api-client-python-0.23.1/qcs_api_client/client/_configuration/configuration.py
@@ -2,7 +2,7 @@ import os
 from pathlib import Path
 from typing import Optional
 
-from pydantic.main import BaseModel
+from pydantic.v1 import BaseModel
 
 from .error import QCSClientConfigurationError
 from .secrets import (
Index: qcs-api-client-python-0.23.1/qcs_api_client/client/_configuration/environment.py
===================================================================
--- qcs-api-client-python-0.23.1.orig/qcs_api_client/client/_configuration/environment.py
+++ qcs-api-client-python-0.23.1/qcs_api_client/client/_configuration/environment.py
@@ -1,8 +1,8 @@
 import os
 from typing import Any, Dict
 
-from pydantic import BaseModel
-from pydantic.utils import deep_update
+from pydantic.v1 import BaseModel
+from pydantic.v1.utils import deep_update
 
 
 class _EnvironmentBaseModel(BaseModel):
Index: qcs-api-client-python-0.23.1/qcs_api_client/client/_configuration/secrets.py
===================================================================
--- qcs-api-client-python-0.23.1.orig/qcs_api_client/client/_configuration/secrets.py
+++ qcs-api-client-python-0.23.1/qcs_api_client/client/_configuration/secrets.py
@@ -4,8 +4,8 @@ from typing import Dict, Optional, Union
 
 import toml
 from jwt import decode
-from pydantic import BaseModel, ValidationError
-from pydantic.fields import Field
+from pydantic.v1 import BaseModel, ValidationError
+from pydantic.v1.fields import Field
 
 from .file import QCSClientConfigurationFile
 
Index: qcs-api-client-python-0.23.1/qcs_api_client/client/_configuration/settings.py
===================================================================
--- qcs-api-client-python-0.23.1.orig/qcs_api_client/client/_configuration/settings.py
+++ qcs-api-client-python-0.23.1/qcs_api_client/client/_configuration/settings.py
@@ -1,7 +1,7 @@
 from typing import Dict, Optional
-from pydantic import BaseModel
-from pydantic.fields import Field
-from pydantic.networks import HttpUrl
+from pydantic.v1 import BaseModel
+from pydantic.v1.fields import Field
+from pydantic.v1.networks import HttpUrl
 from enum import Enum
 from .environment import EnvironmentModel
 from .file import QCSClientConfigurationFile
Index: qcs-api-client-python-0.23.1/qcs_api_client/client/auth.py
===================================================================
--- qcs-api-client-python-0.23.1.orig/qcs_api_client/client/auth.py
+++ qcs-api-client-python-0.23.1/qcs_api_client/client/auth.py
@@ -1,7 +1,7 @@
 import asyncio
 import threading
 import httpx
-from pydantic import BaseModel, Field
+from pydantic.v1 import BaseModel, Field
 from http import HTTPStatus
 from typing import Set
 
Index: qcs-api-client-python-0.23.1/qcs_api_client/client/_configuration/file.py
===================================================================
--- qcs-api-client-python-0.23.1.orig/qcs_api_client/client/_configuration/file.py
+++ qcs-api-client-python-0.23.1/qcs_api_client/client/_configuration/file.py
@@ -2,8 +2,8 @@ from pathlib import Path
 from typing import Any, Dict, Optional
 
 import toml
-from pydantic.types import FilePath
-from pydantic.utils import deep_update
+from pydantic.v1.types import FilePath
+from pydantic.v1.utils import deep_update
 
 from .environment import _EnvironmentBaseModel
 
Index: qcs-api-client-python-0.23.1/tests/test_client/test_client_configuration/test_client_configuration_file.py
===================================================================
--- qcs-api-client-python-0.23.1.orig/tests/test_client/test_client_configuration/test_client_configuration_file.py
+++ qcs-api-client-python-0.23.1/tests/test_client/test_client_configuration/test_client_configuration_file.py
@@ -1,7 +1,7 @@
 from pathlib import Path
 
 import toml
-from pydantic.main import BaseConfig
+from pydantic.v1 import BaseConfig
 
 from qcs_api_client.client._configuration.file import QCSClientConfigurationFile
 
