--- a/tools/winegcc/winegcc.c
+++ b/tools/winegcc/winegcc.c
@@ -904,6 +904,7 @@ no_compat_defines:
         const char *incl_dirs[] = { INCLUDEDIR, "/usr/include", "/usr/local/include" };
         const char *root = opts->isysroot ? opts->isysroot : opts->sysroot ? opts->sysroot : "";
         const char *isystem = gcc_defs ? "-isystem" : "-I";
+        const char *idirafter = gcc_defs ? "-idirafter" : "-I";
 
         if (opts->use_msvcrt)
         {
@@ -921,11 +922,11 @@ no_compat_defines:
             if (j && !strcmp( incl_dirs[0], incl_dirs[j] )) continue;
             strarray_add(comp_args, strmake( "%s%s%s/wine/windows", isystem, root, incl_dirs[j] ));
         }
-        if (includedir) strarray_add( comp_args, strmake( "%s%s", isystem, includedir ));
+        if (includedir) strarray_add( comp_args, strmake( "%s%s", idirafter, includedir ));
         for (j = 0; j < ARRAY_SIZE(incl_dirs); j++)
         {
             if (j && !strcmp( incl_dirs[0], incl_dirs[j] )) continue;
-            strarray_add(comp_args, strmake( "%s%s%s", isystem, root, incl_dirs[j] ));
+            strarray_add(comp_args, strmake( "%s%s%s", idirafter, root, incl_dirs[j] ));
         }
     }
     else if (opts->wine_objdir)
