diff -upr django-rest-framework-client-0.1.1.orig/tests/api.py django-rest-framework-client-0.1.1/tests/api.py
--- django-rest-framework-client-0.1.1.orig/tests/api.py	2022-05-12 09:32:27.295052321 +0200
+++ django-rest-framework-client-0.1.1/tests/api.py	2022-05-12 09:32:27.299052347 +0200
@@ -1,6 +1,6 @@
 import sys
 import json
-import mock
+from unittest import mock
 import requests
 import requests_mock
 import unittest
diff -upr django-rest-framework-client-0.1.1.orig/tests/resources.py django-rest-framework-client-0.1.1/tests/resources.py
--- django-rest-framework-client-0.1.1.orig/tests/resources.py	2022-05-12 09:32:27.295052321 +0200
+++ django-rest-framework-client-0.1.1/tests/resources.py	2022-05-12 09:32:27.299052347 +0200
@@ -1,6 +1,6 @@
 import sys
 import json
-import mock
+from unittest import mock
 import requests
 import requests_mock
 import unittest
