diff -rub gxml-0.20.4/docs/meson.build gxml-0.20.4-patched/docs/meson.build
--- gxml-0.20.4/docs/meson.build	2024-04-16 19:27:12.000000000 +0200
+++ gxml-0.20.4-patched/docs/meson.build	2025-07-10 13:03:02.838261456 +0200
@@ -1,11 +1,11 @@
 valadoc = find_program ('valadoc', required: false)
 if valadoc.found()
-outdir ='GXml-'+API_VERSION
-gtkdoc_outdir ='gxml-'+API_VERSION
+outdir ='GXml'
+gtkdoc_outdir ='gxml'
 valacapi = run_command ('valac', '--api-version', check: false)
 driver = '--driver='+valacapi.stdout().strip()
-pkgname = '--package-name='+'GXml-'+API_VERSION
-pkgversion = '--package-version='+PROJECT_VERSION
+pkgname = '--package-name='+'GXml'
+pkgversion = '--package-version='+meson.project_version ()
 vapidirgxml = '--vapidir='+join_paths (meson.current_source_dir (),'gxml')
 vapidirvapi = '--vapidir='+join_paths (meson.current_source_dir (),'vapi')
 docsdir = join_paths (get_option ('datadir'), 'devhelp','books')
diff -rub gxml-0.20.4/gxml/gxml.pc.in gxml-0.20.4-patched/gxml/gxml.pc.in
--- gxml-0.20.4/gxml/gxml.pc.in	2024-04-16 19:27:12.000000000 +0200
+++ gxml-0.20.4-patched/gxml/gxml.pc.in	2025-07-10 13:03:02.838617758 +0200
@@ -8,6 +8,6 @@
 Description: GObject API for XML manipulation and serialization library
 URL: http://live.gnome.org/GXml
 Version: @PROJECT_VERSION@
-Requires: glib-2.0 >= 2.72 gio-2.0 >= 2.72 gobject-2.0 >= 2.72 libxml-2.0 >=2.9.13 gee-0.8 >=0.20.6
-Libs: -L${libdir} -lgxml-@API_VERSION@
-Cflags: -I${includedir}/gxml-@API_VERSION@
+Requires: glib-2.0 >= 2.72 gio-2.0 >= 2.72 gobject-2.0 >= 2.72 libxml-2.0 >= 2.9.13 gee-0.8 >= 0.20.6
+Libs: -L${libdir} -lgxml
+Cflags: -I${includedir}/gxml
diff -rub gxml-0.20.4/gxml/meson.build gxml-0.20.4-patched/gxml/meson.build
--- gxml-0.20.4/gxml/meson.build	2024-04-16 19:27:12.000000000 +0200
+++ gxml-0.20.4-patched/gxml/meson.build	2025-07-10 13:06:14.870377945 +0200
@@ -1,24 +1,23 @@
 vapidir = join_paths (get_option('datadir'),'vala','vapi')
-GIR_NAME= VERSIONED_CAMEL_CASE_NAME+'.gir'
-TYPELIB_NAME= VERSIONED_CAMEL_CASE_NAME+'.typelib'
-VAPI_NAME = VERSIONED_PROJECT_NAME+'.vapi'
+GIR_NAME= CAMEL_CASE_NAME+'-'+meson.project_version ()+'.gir'
+TYPELIB_NAME= CAMEL_CASE_NAME+'-'+meson.project_version ()+'.typelib'
+VAPI_NAME = meson.project_name ()+'.vapi'
 
 conf = configuration_data()
 conf.set('prefix', get_option('prefix'))
 conf.set('libdir', '${exec_prefix}/'+get_option ('libdir'))
 conf.set('PCDEPS', PCDEPS)
-conf.set('PROJECT_NAME', PROJECT_NAME)
-conf.set('PROJECT_VERSION', PROJECT_VERSION)
-conf.set('API_VERSION', API_VERSION)
+conf.set('PROJECT_NAME', meson.project_name ())
+conf.set('PROJECT_VERSION', meson.project_version ())
 
 configure_file(input : 'gxml.pc.in',
-	output : 'gxml-@0@.pc'.format(API_VERSION),
+	output : 'gxml.pc',
 	configuration : conf,
 	install : true,
 	install_dir : join_paths(get_option('libdir'), 'pkgconfig'))
 
 configure_file(input : 'gxml.deps.in',
-	output : 'gxml-@0@.deps'.format(API_VERSION),
+	output : 'gxml.deps',
 	configuration : conf,
 	install : true,
 	install_dir : vapidir)
@@ -123,16 +122,10 @@
 # Increase the age value only if the changes made to the ABI are backward compatible.
 # Set version to the value of subtract age from current
 # Reset current and version to 1 and, age and version to 0 if library's name is changed
-LT_CURRENT='3'
-LT_REVISION='2'
-LT_AGE='0'
-LT_VERSION='2'
-libgxml = library(VERSIONED_PROJECT_NAME,
+libgxml = library(meson.project_name (),
 	valasources+sources+configvapi,
-	version : LT_VERSION,
-	soversion : LT_VERSION+'.'+LT_AGE+'.'+LT_REVISION,
-	darwin_versions : [LT_CURRENT.to_int() + 1, '@0@.@1@'.format(LT_CURRENT.to_int() + 1, LT_REVISION)],
-	vala_header : PROJECT_NAME+'.h',
+	soversion : meson.project_version (),
+	vala_header : meson.project_name ()+'.h',
 	vala_vapi : VAPI_NAME,
 	vala_gir : GIR_NAME,
 	dependencies : [ libgxml_deps, namespaceinfo_dep, inc_libh_dep, inc_rooth_dep ],
@@ -143,7 +136,7 @@
 	install : true,
 	install_dir : [
 		true,
-		join_paths (get_option('includedir'), 'gxml-@0@'.format (API_VERSION), 'gxml'),
+		join_paths (get_option('includedir'), 'gxml'),
 		vapidir,
 		get_option('introspection')
 	])
@@ -154,7 +147,7 @@
 custom_target('libgxml-typelib',
 	command: [
 		g_ir_compiler,
-		'--shared-library', 'lib'+PROJECT_NAME+'-@0@.so'.format (API_VERSION),
+		'--shared-library', 'lib'+meson.project_name ()+'.so.'+meson.project_version (),
 		'--output', '@OUTPUT@',
 		join_paths(meson.current_build_dir(), GIR_NAME)
 	],
diff -rub gxml-0.20.4/gxml/namespace-info.vala.in gxml-0.20.4-patched/gxml/namespace-info.vala.in
--- gxml-0.20.4/gxml/namespace-info.vala.in	2024-04-16 19:27:12.000000000 +0200
+++ gxml-0.20.4-patched/gxml/namespace-info.vala.in	2025-07-10 13:03:02.839308004 +0200
@@ -22,5 +22,5 @@
  *      Richard Schwarting <aquarichy@gmail.com>
  *      Daniel Espinosa <esodan@gmail.com>
  */
-[CCode (gir_namespace = "GXml", gir_version = "@API_VERSION@", cheader_filename = "gxml/@PROJECT_NAME@.h")]
+[CCode (gir_namespace = "GXml", gir_version = "@PROJECT_VERSION@", cheader_filename = "gxml/@PROJECT_NAME@.h")]
 namespace GXml {}
diff -rub gxml-0.20.4/meson.build gxml-0.20.4-patched/meson.build
--- gxml-0.20.4/meson.build	2024-04-16 19:27:12.000000000 +0200
+++ gxml-0.20.4-patched/meson.build	2025-07-10 13:03:02.839596686 +0200
@@ -1,11 +1,7 @@
 project('gxml', [ 'vala', 'c'], version : '0.20.4')
 
 PROJECT_NAME = meson.project_name ()
-API_VERSION = '0.20'
-PROJECT_VERSION = meson.project_version ()
-VERSIONED_PROJECT_NAME = PROJECT_NAME+'-'+API_VERSION
 CAMEL_CASE_NAME = 'GXml'
-VERSIONED_CAMEL_CASE_NAME = CAMEL_CASE_NAME +'-'+ API_VERSION
 
 libgxml_deps = [
 	dependency('libxml-2.0', version: '>=2.9.13'),
diff -rub gxml-0.20.4/po/meson.build gxml-0.20.4-patched/po/meson.build
--- gxml-0.20.4/po/meson.build	2024-04-16 19:27:12.000000000 +0200
+++ gxml-0.20.4-patched/po/meson.build	2025-07-10 13:03:02.839890809 +0200
@@ -1,2 +1,2 @@
 i18n = import('i18n')
-i18n.gettext(VERSIONED_CAMEL_CASE_NAME, preset : 'glib')
+i18n.gettext(CAMEL_CASE_NAME, preset : 'glib')
