Index: meson-0.52.0/setup.py
===================================================================
--- meson-0.52.0.orig/setup.py
+++ meson-0.52.0/setup.py
@@ -21,7 +21,7 @@ if sys.version_info < (3, 5, 2):
                      '\nMeson requires Python 3.5.2 or greater'.format(sys.version))
 
 from mesonbuild.coredata import version
-from setuptools import setup
+from distutils.core import setup
 
 # On windows, will create Scripts/meson.exe and Scripts/meson-script.py
 # Other platforms will create bin/meson
