Index: b/setup.py
===================================================================
--- a/setup.py
+++ b/setup.py
@@ -20,7 +20,7 @@ if sys.version_info < (3, 7):
     raise SystemExit('ERROR: Tried to install Meson with an unsupported Python version: \n{}'
                      '\nMeson requires Python 3.7.0 or greater'.format(sys.version))
 
-from setuptools import setup
+from distutils.core import setup
 
 data_files = []
 if sys.platform != 'win32':
@@ -28,4 +28,16 @@ if sys.platform != 'win32':
     data_files = [('share/man/man1', ['man/meson.1']),
                   ('share/polkit-1/actions', ['data/com.mesonbuild.install.policy'])]
 
-setup(data_files=data_files,)
+# Since distutils does not understand the find: directive in setup.cfg, and
+# therefore doesn't expand it, we need to do it directly
+import os
+packages=[
+    root.replace('/', '.')
+    for root, dirs, files in os.walk('mesonbuild')
+    if '__init__.py' in files]
+
+setup(
+    packages=packages,
+    package_data={'mesonbuild.dependencies.data': ['mesonbuild/dependencies/data/*.txt']},
+    data_files=data_files,
+)
Index: b/MANIFEST.in
===================================================================
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -17,3 +17,5 @@ include run_unittests.py
 include run_meson_command_tests.py
 include run_project_tests.py
 include meson.py
+
+include mesonbuild/dependencies/data/*.txt
