Index: libguile/guile-snarf.in
===================================================================
--- libguile/guile-snarf.in.orig	2010-12-13 19:24:40.000000000 +0200
+++ libguile/guile-snarf.in	2011-09-27 19:28:33.721419831 +0200
@@ -71,8 +71,7 @@ fi
 cpp_ok_p=false
 
 if [ x"$TMPDIR" = x ]; then TMPDIR="/tmp" ; else : ; fi
-tempdir="$TMPDIR/guile-snarf.$$"
-(umask 077 && mkdir $tempdir) || exit 1
+tempdir=$(mktemp -d -q "$TMPDIR/snarf.XXXXXX") || { echo >&2 "guile-snarf: can not create temporary file"; exit 1; }
 temp="$tempdir/tmp"
 
 if [ x"$CPP" = x ] ; then cpp="@CPP@" ; else cpp="$CPP" ; fi
