--- msr_version.c	2023-02-01 04:45:13.000000000 +0800
+++ new-msr_version.c	2023-03-30 21:38:38.000000000 +0800
@@ -53,7 +53,11 @@
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,39)
 static char *msr_version_nodename(struct device *dev, mode_t *mode)
 #else
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(6,2,0)
 static char *msr_version_nodename(struct device *dev, umode_t *mode)
+#else
+static char *msr_version_nodename(const struct device *dev, umode_t *mode)
+#endif
 #endif
 {
     if (mode)
