--- a/project/Sxr.scala	2014-01-20 15:50:18.561978172 -0600
+++ b/project/Sxr.scala	2014-01-20 15:35:04.949702779 -0600
@@ -11,15 +11,10 @@
 	lazy val settings: Seq[Setting[_]] = inTask(sxr)(inSxrSettings) ++ baseSettings
 
 	def baseSettings = Seq(
-		libraryDependencies += "org.scala-sbt.sxr" % "sxr_2.10" % "0.3.0" % sxrConf.name
 	)
 	def inSxrSettings = Seq(
 		managedClasspath := update.value.matching( configurationFilter(sxrConf.name) ).classpath,
-		scalacOptions += "-P:sxr:base-directory:" + sourceDirectories.value.absString,
-		scalacOptions += "-Xplugin:" + managedClasspath.value.files.filter(_.getName.contains("sxr")).absString,
-		scalacOptions += "-Ystop-after:sxr",
-		target := target.in(taskGlobal).value / "browse",
-		sxr in taskGlobal <<= sxrTask
+		scalacOptions += "-P:sxr:base-directory:" + sourceDirectories.value.absString
 	)
 	def taskGlobal = ThisScope.copy(task = Global)
 	def sxrTask = (sources, target, scalacOptions, classpathOptions, scalaInstance, fullClasspath, streams) map { (srcs, out, opts, cpOpts, si, cp, s) =>
