--- ivy/src/main/scala/sbt/IvyLogger.scala~	2015-07-21 12:06:06.883992874 +0200
+++ ivy/src/main/scala/sbt/IvyLogger.scala	2015-07-21 12:06:20.516805104 +0200
@@ -31,7 +31,7 @@
 	def warn(msg: String) = logger.warn(msg)
 	def error(msg: String) = if(SbtIvyLogger.acceptError(msg)) logger.error(msg)
 	
-	private def emptyList = java.util.Collections.emptyList[T forSome { type T}]
+	private def emptyList = java.util.Collections.emptyList[String]
 	def getProblems = emptyList
 	def getWarns = emptyList
 	def getErrors = emptyList
