diff '--color=auto' -rub goverlay-1.8.4.orig/Makefile goverlay-1.8.4/Makefile
--- goverlay-1.8.4.orig/Makefile	2026-06-20 14:54:09.000000000 -0300
+++ goverlay-1.8.4/Makefile	2026-06-27 14:57:48.348232774 -0300
@@ -3,15 +3,11 @@
 libexecdir = /libexec
 datadir = /share
 
-all: goverlay start_goverlay.sh bgmod bgmod-uninstaller pascube_bin
+all: goverlay start_goverlay.sh bgmod bgmod-uninstaller
 
 goverlay: *.pas *.lfm goverlay.lpi goverlay.lpr goverlay.ico
 	lazbuild -B goverlay.lpi --bm=Release $(LAZBUILDOPTS)
 
-pascube_bin:
-	lazbuild -B pascube_src/pascube.lpi $(LAZBUILDOPTS)
-	cp pascube_src/pascube ./pascube
-
 bgmod: bgmod.lpr
 	fpc -O3 bgmod.lpr
 	mkdir -p data/bgmod
@@ -32,11 +28,6 @@
 
 clean:
 	rm -f goverlay
-	rm -f pascube
-	rm -f pascube_src/pascube
-	rm -rf pascube_src/lib/
-	rm -rf pascube_src/backup/
-	rm -f pascube_src/pascube.lps
 	rm -f bgmod
 	rm -f bgmod-uninstaller
 	rm -f data/bgmod/bgmod
@@ -51,7 +42,6 @@
 
 install: goverlay data/goverlay.sh bgmod bgmod-uninstaller
 	install -D -m=755 goverlay $(DESTDIR)$(prefix)$(libexecdir)/goverlay
-	install -D -m=755 pascube $(DESTDIR)$(prefix)$(libexecdir)/pascube
 	install -D -m=755 data/goverlay.sh $(DESTDIR)$(prefix)$(bindir)/goverlay
 	install -D -m=644 data/io.github.benjamimgois.goverlay.desktop $(DESTDIR)$(prefix)$(datadir)/applications/io.github.benjamimgois.goverlay.desktop
 	install -D -m=644 data/io.github.benjamimgois.goverlay.metainfo.xml $(DESTDIR)$(prefix)$(datadir)/metainfo/io.github.benjamimgois.goverlay.metainfo.xml
@@ -68,7 +58,6 @@
 
 uninstall:
 	rm -f $(DESTDIR)$(prefix)$(libexecdir)/goverlay
-	rm -f $(DESTDIR)$(prefix)$(libexecdir)/pascube
 	rm -f $(DESTDIR)$(prefix)$(bindir)/goverlay
 	rm -f $(DESTDIR)$(prefix)$(datadir)/applications/io.github.benjamimgois.goverlay.desktop
 	rm -f $(DESTDIR)$(prefix)$(datadir)/metainfo/io.github.benjamimgois.goverlay.metainfo.xml
