Index: installer.sh
===================================================================
--- installer.sh.orig
+++ installer.sh
@@ -131,7 +131,7 @@ selectTemplate() { # Take input from the
 	/usr|/usr/local|default|custom_*|RPM|DEB|TGZ|TXZ)
 		case "$1" in
 		default)
-			PREFIX="/usr/local"
+			PREFIX="/usr"
 			;;
 		custom_*)
 			PREFIX=`echo "${RKHINST_LAYOUT}" | sed 's|custom_||g'`
@@ -165,9 +165,9 @@ selectTemplate() { # Take input from the
 			if [ -n "${RPM_BUILD_ROOT}" ]; then
 				if [ "${RPM_BUILD_ROOT}" = "/" ]; then
 					RPM_USING_ROOT=1
-					PREFIX="/usr/local"
+					PREFIX="/usr"
 				else
-					PREFIX="${RPM_BUILD_ROOT}/usr/local"
+					PREFIX="${RPM_BUILD_ROOT}/usr"
 				fi
 			else
 				echo "RPM installation chosen but \$RPM_BUILD_ROOT variable not found. Exiting."
@@ -233,7 +233,7 @@ selectTemplate() { # Take input from the
 		esac
 
 		case "$1" in
-		/usr/local|custom_*)
+		/usr|custom_*)
 			SYSCONFIGDIR="${PREFIX}/etc"
 			;;
 		RPM)
@@ -299,10 +299,12 @@ selectTemplate() { # Take input from the
 			fi
 			;;
 		RPM)
-			if [ "${UNAMEM}" = "x86_64" -o "${UNAMEM}" = "ppc64" ]; then
-				LIBDIR="${PREFIX}/lib64"
-			else
-				LIBDIR="${PREFIX}/lib"
+			if [ -z "${LIBDIR}" ]; then
+				if [ "${UNAMEM}" = "x86_64" -o "${UNAMEM}" = "ppc64" -o "${UNAMEM}" = "aarch64" -o "${UNAMEM}" = "s390x" -o "${UNAMEM}" = "ppc64le" ]; then
+					LIBDIR="${PREFIX}/lib64"
+				else
+					LIBDIR="${PREFIX}/lib"
+				fi
 			fi
 
 			BINDIR="${PREFIX}/bin"
@@ -411,7 +413,7 @@ selectTemplate() { # Take input from the
 	else
 		RKHINST_DB_DIR="${VARDIR}/lib/${APPNAME}/db"
 		RKHINST_TMP_DIR="${VARDIR}/lib/${APPNAME}/tmp"
-		RKHINST_DOC_DIR="${SHAREDIR}/doc/${APPNAME}-${APPVERSION}"
+		RKHINST_DOC_DIR="${SHAREDIR}/doc/packages/${APPNAME}-${APPVERSION}"
 	fi
 
 	RKHINST_LANG_DIR="${RKHINST_DB_DIR}/i18n"
