Index: SPARQLWrapper-1.8.5/setup.py
===================================================================
--- SPARQLWrapper-1.8.5.orig/setup.py
+++ SPARQLWrapper-1.8.5/setup.py
@@ -75,8 +75,6 @@ setup(
         'Topic :: Software Development :: Libraries :: Python Modules',
       ],
       keywords = ['python', 'sparql', 'rdf', 'rdflib'],
-      use_2to3 = True,
-      use_2to3_fixers = ['custom_fixers'],
       project_urls={
         'Home': 'https://rdflib.github.io/sparqlwrapper/',
         'Documentation': 'https://rdflib.github.io/sparqlwrapper/doc/',
