Index: src/sphinx_lmtools/sphinx_lm_eval.c
===================================================================
--- src/sphinx_lmtools/sphinx_lm_eval.c.orig
+++ src/sphinx_lmtools/sphinx_lm_eval.c
@@ -193,7 +193,7 @@ evaluate_file(ngram_model_t *lm, logmath
 {
 	FILE *fh;
         lineiter_t *litor;
-	int32 nccs, noovs, nwords, lscr;
+	int32 nccs, noovs, nwords, lscr = 0;
 	float64 ch, log_to_log2;;
 
 	if ((fh = fopen(lsnfn, "r")) == NULL)
