Index: sphinxbase-0.8/src/libsphinxbase/util/info.c
===================================================================
--- sphinxbase-0.8.orig/src/libsphinxbase/util/info.c
+++ sphinxbase-0.8/src/libsphinxbase/util/info.c
@@ -69,7 +69,4 @@ print_appl_info(char *appl_name)
             E_INFO("Directory: '%s'\n", path);
     }
 #endif
-
-    E_INFO("%s Compiled on: %s, AT: %s\n\n", appl_name, __DATE__,
-           __TIME__);
 }
