diff --git a/src/libImaging/TiffDecode.c b/src/libImaging/TiffDecode.c
index f33cbc6bd6..d86a42915b 100644
--- a/src/libImaging/TiffDecode.c
+++ b/src/libImaging/TiffDecode.c
@@ -578,7 +578,7 @@ int ImagingLibTiffMergeFieldInfo(ImagingCodecState state, TIFFDataType field_typ
 
     // custom fields added with ImagingLibTiffMergeFieldInfo are only used for
     // decoding, ignore readcount;
-    int readcount = 0;
+    int readcount = 1;
     // we support writing a single value, or a variable number of values
     int writecount = 1;
     // whether the first value should encode the number of values.

