diff --git a/etc/nodes.conf b/etc/nodes.conf
index c5b9487f..94eeb9ee 100644
--- a/etc/nodes.conf
+++ b/etc/nodes.conf
@@ -23,6 +23,7 @@ nodeprofiles:
       args:
        - quiet
        - crashkernel=no
+       - net.ifnames=1
     init: /sbin/init
     root: initramfs
     ipxe template: default
diff --git a/internal/pkg/upgrade/node.go b/internal/pkg/upgrade/node.go
index 3e8fa26f..dbe69b5b 100644
--- a/internal/pkg/upgrade/node.go
+++ b/internal/pkg/upgrade/node.go
@@ -123,7 +123,7 @@ func (legacy *NodesYaml) Upgrade(addDefaults bool, replaceOverlays bool, warewul
 				defaultProfile.RuntimeOverlay, genericSplitOverlays...)
 		}
 		if len(defaultProfile.Kernel.Args) < 1 {
-			defaultProfile.Kernel.Args = []string{"quiet", "crashkernel=no"}
+			defaultProfile.Kernel.Args = []string{"quiet", "crashkernel=no", "net.ifnames=1"}
 		}
 		if defaultProfile.Init == "" {
 			defaultProfile.Init = "/sbin/init"
diff --git a/internal/pkg/upgrade/node_test.go b/internal/pkg/upgrade/node_test.go
index 132b109b..d32a5275 100644
--- a/internal/pkg/upgrade/node_test.go
+++ b/internal/pkg/upgrade/node_test.go
@@ -576,6 +576,7 @@ nodeprofiles:
       args:
       - quiet
       - crashkernel=no
+      - net.ifnames=1
     init: /sbin/init
     root: initramfs
     resources:
@@ -654,6 +655,7 @@ nodeprofiles:
       args:
       - quiet
       - crashkernel=no
+      - net.ifnames=1
     init: /sbin/init
     root: initramfs
     resources:
@@ -915,6 +917,7 @@ nodeprofiles:
       args:
       - quiet
       - crashkernel=no
+      - net.ifnames=1
     init: /sbin/init
     root: initramfs
     resources:
