diff -u -r mesa-demos-8.5.0.orig/src/egl/opengl/meson.build mesa-demos-8.5.0/src/egl/opengl/meson.build
--- mesa-demos-8.5.0.orig/src/egl/opengl/meson.build	2023-03-24 14:12:08.868037150 +0100
+++ mesa-demos-8.5.0/src/egl/opengl/meson.build	2023-03-24 14:14:13.570480915 +0100
@@ -2,19 +2,23 @@
 
 executable(
   'eglgears_x11', files('eglgears.c'),
-  dependencies: [_deps, idep_eglut_x11]
+  dependencies: [_deps, idep_eglut_x11],
+  install: true
 )
 executable(
   'egltri_x11', files('egltri.c'),
-  dependencies: [_deps, idep_eglut_x11]
+  dependencies: [_deps, idep_eglut_x11],
+  install: true
 )
 executable(
   'xeglgears', files('xeglgears.c'),
-  dependencies: [_deps, dep_egl, dep_x11]
+  dependencies: [_deps, dep_egl, dep_x11],
+  install: true
 )
 executable(
   'xeglthreads', files('xeglthreads.c'),
-  dependencies: [_deps, dep_x11]
+  dependencies: [_deps, dep_x11],
+  install: true
 )
 
 executable(
@@ -39,6 +43,7 @@
 
 executable(
   'peglgears', 'peglgears.c',
-  dependencies: [dep_gl, dep_egl, dep_m, idep_util]
+  dependencies: [dep_gl, dep_egl, dep_m, idep_util],
+  install: true
 )
 
