Index: opensaml-2.5.5/doxygen.cfg
===================================================================
--- opensaml-2.5.5.orig/doxygen.cfg
+++ opensaml-2.5.5/doxygen.cfg
@@ -140,6 +140,7 @@ HTML_OUTPUT            = html
 HTML_FILE_EXTENSION    = .html
 HTML_HEADER            = 
 HTML_FOOTER            = 
+HTML_TIMESTAMP         = NO
 HTML_STYLESHEET        = 
 HTML_ALIGN_MEMBERS     = YES
 GENERATE_HTMLHELP      = $(GENERATE_CHM)
