Index: testresources-2.1.2/tests/test_optimising_test_suite.py
===================================================================
--- testresources-2.1.2.orig/tests/test_optimising_test_suite.py
+++ testresources-2.1.2/tests/test_optimising_test_suite.py
@@ -558,6 +558,7 @@ class TestGraphStuff(testtools.TestCase)
         permutations.append([case4, case1, case3, case2])
         return permutations
 
+    @unittest.skip("too flaky")
     def testBasicSortTests(self):
         self.skipTest("flaky, see LP #1645008")
         # Test every permutation of inputs, with legacy tests.
