diff -Nru efs-utils-1.7.orig/src/mount_efs/__init__.py efs-utils-1.7/src/mount_efs/__init__.py
--- efs-utils-1.7.orig/src/mount_efs/__init__.py	2019-04-09 20:27:34.000000000 +0200
+++ efs-utils-1.7/src/mount_efs/__init__.py	2019-04-09 23:59:43.477327640 +0200
@@ -44,9 +44,9 @@
 from logging.handlers import RotatingFileHandler
 
 try:
-    import ConfigParser
+    import ConfigParser as cp
 except ImportError:
-    from configparser import ConfigParser
+    import configparser as cp
 
 try:
     from urllib2 import urlopen, URLError
@@ -537,7 +537,7 @@
 
 
 def read_config(config_file=CONFIG_FILE):
-    p = ConfigParser.SafeConfigParser()
+    p = cp.ConfigParser()
     p.read(config_file)
     return p
 
diff -Nru efs-utils-1.7.orig/src/watchdog/__init__.py efs-utils-1.7/src/watchdog/__init__.py
--- efs-utils-1.7.orig/src/watchdog/__init__.py	2019-04-09 20:27:34.000000000 +0200
+++ efs-utils-1.7/src/watchdog/__init__.py	2019-04-09 23:59:43.477327640 +0200
@@ -21,9 +21,9 @@
 from signal import SIGTERM
 
 try:
-    import ConfigParser
+    import ConfigParser as cp
 except ImportError:
-    from configparser import ConfigParser
+    import configparser as cp
 
 VERSION = '1.7'
 
@@ -280,7 +280,7 @@
 
 
 def read_config(config_file=CONFIG_FILE):
-    p = ConfigParser.SafeConfigParser()
+    p = cp.ConfigParser()
     p.read(config_file)
     return p
 
diff -Nru efs-utils-1.7.orig/test/mount_efs_test/test_choose_tls_port.py efs-utils-1.7/test/mount_efs_test/test_choose_tls_port.py
--- efs-utils-1.7.orig/test/mount_efs_test/test_choose_tls_port.py	2019-04-09 20:27:34.000000000 +0200
+++ efs-utils-1.7/test/mount_efs_test/test_choose_tls_port.py	2019-04-09 23:59:43.477327640 +0200
@@ -7,9 +7,13 @@
 #
 
 import mount_efs
-import ConfigParser
 import socket
 
+try:
+    import ConfigParser as cp
+except ImportError:
+    import configparser as cp
+
 import pytest
 
 from mock import MagicMock
@@ -19,7 +23,7 @@
 
 
 def _get_config():
-    config = ConfigParser.SafeConfigParser()
+    config = cp.ConfigParser()
     config.add_section(mount_efs.CONFIG_SECTION)
     config.set(mount_efs.CONFIG_SECTION, 'port_range_lower_bound', str(DEFAULT_TLS_PORT_RANGE_LOW))
     config.set(mount_efs.CONFIG_SECTION, 'port_range_upper_bound', str(DEFAULT_TLS_PORT_RANGE_HIGH))
diff -Nru efs-utils-1.7.orig/test/mount_efs_test/test_write_stunnel_config_file.py efs-utils-1.7/test/mount_efs_test/test_write_stunnel_config_file.py
--- efs-utils-1.7.orig/test/mount_efs_test/test_write_stunnel_config_file.py	2019-04-09 20:27:34.000000000 +0200
+++ efs-utils-1.7/test/mount_efs_test/test_write_stunnel_config_file.py	2019-04-09 23:59:43.477327640 +0200
@@ -7,9 +7,13 @@
 #
 
 import mount_efs
-import ConfigParser
 import os
 
+try:
+    import ConfigParser as cp
+except ImportError:
+    import configparser as cp
+
 import pytest
 
 FS_ID = 'fs-deadbeef'
@@ -32,7 +36,7 @@
     if stunnel_check_cert_validity is None:
         stunnel_check_cert_validity = stunnel_check_cert_validity_supported
 
-    config = ConfigParser.SafeConfigParser()
+    config = cp.ConfigParser()
     config.add_section(mount_efs.CONFIG_SECTION)
     config.set(mount_efs.CONFIG_SECTION, 'stunnel_debug_enabled', str(stunnel_debug_enabled))
     config.set(mount_efs.CONFIG_SECTION, 'stunnel_check_cert_hostname', str(stunnel_check_cert_hostname))
