From: Andreas Stieger <astieger@suse.com>
Date: Sun, 20 Jan 2019 11:09:25 +0100
Subject: Fix syntax errors in mongodb 3.68 build scripts
References:
Upstream: no

Index: mongodb-src-r3.6.8/SConstruct
===================================================================
--- mongodb-src-r3.6.8.orig/SConstruct
+++ mongodb-src-r3.6.8/SConstruct
@@ -3117,7 +3117,7 @@ def doConfigure(myenv):
 
         outputIndex = next((idx for idx in [0,1] if conf.CheckAltivecVbpermqOutput(idx)), None)
         if outputIndex is not None:
-	    conf.env.SetConfigHeaderDefine("MONGO_CONFIG_ALTIVEC_VEC_VBPERMQ_OUTPUT_INDEX", outputIndex)
+            conf.env.SetConfigHeaderDefine("MONGO_CONFIG_ALTIVEC_VEC_VBPERMQ_OUTPUT_INDEX", outputIndex)
         else:
             myenv.ConfError("Running on ppc64le, but can't find a correct vec_vbpermq output index.  Compiler or platform not supported")
 
