https://gitlab.freedesktop.org/polkit/polkit/-/merge_requests/131
build: Honour pam_prefix meson option

Make the use of pam_prefix worth its while since, at the moment, its value
is not being used. Instead, a hard-coded path is being deployed when it
shouldn't anymore.

The pam_prefix Meson option was designed to allow us to choose where pam
configuration files should end up. But at the moment it is not being used at
all where it should be.

--- a/meson.build
+++ b/meson.build
@@ -241,7 +241,7 @@ if enable_pam
 
   pam_prefix = get_option('pam_prefix')
   if pam_prefix == ''
-    pam_prefix = pk_sysconfdir
+    pam_prefix = pk_sysconfdir / 'pam.d'
   else
     message('PAM files will be installed in prefix ' + pam_prefix)
   endif
--- a/data/meson.build
+++ b/data/meson.build
@@ -22,7 +22,7 @@ if enable_pam
     output: '@BASENAME@',
     configuration: pam_conf,
     install: true,
-    install_dir: pk_sysconfdir / 'pam.d',
+    install_dir: pam_prefix,
   )
 endif
 
