Index: mgetty-1.1.36/fax/Makefile
===================================================================
--- mgetty-1.1.36.orig/fax/Makefile
+++ mgetty-1.1.36/fax/Makefile
@@ -36,7 +36,7 @@ faxheader: faxheader.in ../sedscript
 	@cd .. ; $(MAKE) sedscript
 
 faxq-helper: faxq-helper.o 
-	$(CC) $(CFLAGS) -o faxq-helper faxq-helper.o
+	$(CC) $(LDFLAGS) -o faxq-helper faxq-helper.o
 
 faxq-helper.o: faxq-helper.c ../sedscript
 	$(CC) $(CFLAGS) -DFAX_SPOOL_OUT=\"$(FAX_SPOOL_OUT)\" \
