diff -Naur lcms2-2.11.orig/src/cmscgats.c lcms2-2.11/src/cmscgats.c
--- lcms2-2.11.orig/src/cmscgats.c	2020-08-05 15:14:07.488229633 +0200
+++ lcms2-2.11/src/cmscgats.c	2020-08-05 15:16:41.214899028 +0200
@@ -930,7 +930,7 @@
                     return;
                 }
 
-                FileNest->Stream = fopen(FileNest->FileName, "rt");
+                FileNest->Stream = fopen(FileNest->FileName, "rte");
                 if (FileNest->Stream == NULL) {
 
                         SynError(it8, "File %s not found", FileNest->FileName);
@@ -1773,7 +1773,7 @@
 
     memset(&sd, 0, sizeof(sd));
 
-    sd.stream = fopen(cFileName, "wt");
+    sd.stream = fopen(cFileName, "wte");
     if (!sd.stream) return FALSE;
 
     for (i=0; i < it8 ->TablesCount; i++) {
@@ -2266,7 +2266,7 @@
    cmsUInt32Number Size;
    cmsUInt8Number Ptr[133];
 
-   fp = fopen(FileName, "rt");
+   fp = fopen(FileName, "rte");
    if (!fp) {
        cmsSignalError(0, cmsERROR_FILE, "File '%s' not found", FileName);
        return FALSE;
@@ -2349,7 +2349,7 @@
      if (!hIT8) return NULL;
 
 
-     it8 ->FileStack[0]->Stream = fopen(cFileName, "rt");
+     it8 ->FileStack[0]->Stream = fopen(cFileName, "rte");
 
      if (!it8 ->FileStack[0]->Stream) {
          cmsIT8Free(hIT8);
