diff --git a/nml/version_info.py b/nml/version_info.py
index d7faec0..32aa3b1 100644
--- a/nml/version_info.py
+++ b/nml/version_info.py
@@ -71,11 +71,11 @@ def get_lib_versions():
     #PIL
     try:
         from PIL import Image
-        versions["PIL"] = Image.VERSION
+        versions["PIL"] = Image.PILLOW_VERSION
     except ImportError:
         try:
             import Image
-            versions["PIL"] = Image.VERSION
+            versions["PIL"] = Image.PILLOW_VERSION
         except ImportError:
             versions["PIL"] = "Not found!"
