From 5adea178fa062c21d3ade671bfc7a6d8eb56d118 Mon Sep 17 00:00:00 2001
From: Christian Goll <cgoll@suse.de>
Date: Wed, 12 Sep 2018 11:37:06 +0200
Subject: [PATCH 2/2] removed check ENOATTR

---
 robinhood-3.1/src/policies/policy_matching.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/robinhood-3.1/src/policies/policy_matching.c b/robinhood-3.1/src/policies/policy_matching.c
index 5145d05..fce50cb 100644
--- a/robinhood-3.1/src/policies/policy_matching.c
+++ b/robinhood-3.1/src/policies/policy_matching.c
@@ -1196,7 +1196,7 @@ static policy_match_t eval_condition(const entry_id_t *p_entry_id,
             /* retrieve xattr value */
             rc = lgetxattr(entry_path, p_triplet->attr_name, value, 1024);
             if (rc < 0) {
-                if (errno == ENOATTR || errno == ENODATA || errno == ENOENT)
+                if (errno == ENODATA || errno == ENOENT)
                     /* empty string == no attribute */
                     strcpy(value, "");
                 else if (errno == ENOTSUP) {
-- 
2.16.4

