--- a/daemon/transmission-daemon.service
+++ b/daemon/transmission-daemon.service
@@ -4,6 +4,18 @@ Wants=network-online.target
 After=network-online.target
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+PrivateDevices=true
+ProtectHostname=true
+ProtectClock=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 User=transmission
 Type=notify
 ExecStart=/usr/bin/transmission-daemon -f --log-level=error
