Index: pacemaker/cts/CTSaudits.py
===================================================================
--- pacemaker.orig/cts/CTSaudits.py
+++ pacemaker/cts/CTSaudits.py
@@ -67,10 +67,8 @@ class LogAudit(ClusterAudit):
 
         for node in nodes:
             if self.CM.Env["have_systemd"]:
-                if self.CM.rsh(node, "systemctl stop systemd-journald.socket") != 0:
-                    self.CM.log ("ERROR: Cannot stop 'systemd-journald' on %s" % node)
-                if self.CM.rsh(node, "systemctl start systemd-journald.service") != 0:
-                    self.CM.log ("ERROR: Cannot start 'systemd-journald' on %s" % node)
+                if self.CM.rsh(node, "systemctl restart systemd-journald.socket") != 0:
+                    self.CM.log ("ERROR: Cannot restart 'systemd-journald' on %s" % node)
 
             if self.CM.rsh(node, "service %s restart" % self.CM.Env["syslogd"]) != 0:
                 self.CM.log ("ERROR: Cannot restart '%s' on %s" % (self.CM.Env["syslogd"], node))
