commit f813880dd1b6d1614393128a7f5f745437bea121
Author: Gao,Yan <ygao@suse.com>
Date:   Wed Mar 27 22:03:56 2013 +0800

    Build: lrmd: Change the default directory for nagios plugins

Index: pacemaker-2.1.4+20220720.3b57f9b58/configure.ac
===================================================================
--- pacemaker-2.1.4+20220720.3b57f9b58.orig/configure.ac
+++ pacemaker-2.1.4+20220720.3b57f9b58/configure.ac
@@ -1523,7 +1523,7 @@ AC_DEFINE_UNQUOTED([SUPPORT_NAGIOS], [$w
 AM_CONDITIONAL([BUILD_NAGIOS], [test $with_nagios -eq $REQUIRED])
 
 AS_IF([test x"$NAGIOS_PLUGIN_DIR" = x""],
-      [NAGIOS_PLUGIN_DIR="${libexecdir}/nagios/plugins"])
+      [NAGIOS_PLUGIN_DIR="${prefix}/lib/nagios/plugins"])
 
 AC_DEFINE_UNQUOTED(NAGIOS_PLUGIN_DIR, "$NAGIOS_PLUGIN_DIR", Directory for nagios plugins)
 AC_SUBST(NAGIOS_PLUGIN_DIR)
