From 79102bf264f8651692fe9435a4d1bb06aa022d55 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Tue, 17 Mar 2020 10:21:10 +0100
Subject: [PATCH 07/16] ucm: clarify the index syntax for the device names

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 include/use-case.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/include/use-case.h b/include/use-case.h
index 134303afaa18..27cb3fe1613f 100644
--- a/include/use-case.h
+++ b/include/use-case.h
@@ -117,7 +117,10 @@ extern "C" {
  *
  * If multiple devices with the same name exists, the number suffixes should
  * be added to these names like HDMI1,HDMI2,HDMI3 etc. No number gaps are
- * allowed. The names with numbers must be continuous.
+ * allowed. The names with numbers must be continuous. It is allowed to put
+ * a whitespace between name and index (like 'Line 1') for the better
+ * readability. The device names 'Line 1' and 'Line1' are equal for
+ * thus purpose.
  *
  * If EnableSequence/DisableSequence controls independent paths in the hardware
  * it is also recommended to split playback and capture UCM devices and use
-- 
2.16.4

