From ed752498522b0a99a9fbec99247d9c73b10abb3e Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Tue, 18 Aug 2020 16:53:23 +0200
Subject: [PATCH 15/32] conf: quote also strings with '*' and '#' characters in
 string_print()

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 src/conf.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/conf.c b/src/conf.c
index 8518f90c5d60..7df2b4e77759 100644
--- a/src/conf.c
+++ b/src/conf.c
@@ -1536,6 +1536,8 @@ static void string_print(char *str, int id, snd_output_t *out)
 	case ']':
 	case '\'':
 	case '"':
+	case '*':
+	case '#':
 		goto quoted;
 	default:
 		if (*p <= 31 || *p >= 127)
-- 
2.16.4

