--- a/Cargo.toml	2026-08-28 07:59:14.964906268 +0200
+++ b/Cargo.toml	2026-08-28 07:59:14.968906717 +0200
@@ -48,7 +48,7 @@
 indoc = { version = "2", default-features = false }
 iroh = { version = "=1.0.3", default-features = false }
 iroh-relay = { version = "1", default-features = false }
-keyring = { version = "3.6.3", default-features = false, features = ["vendored"] }
+keyring = { version = "3.6.3", default-features = false }
 lru = { version = "0.18", default-features = false }
 once_cell = { version = "1.21.3", default-features = false, features = ["std"] }
 rand = "0.10.1"
