From 694e8945db265e00f584565dd1428ab70e37785b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ale=C5=A1=20Mat=C4=9Bj?= <amatej@redhat.com>
Date: Wed, 5 Nov 2025 07:38:43 +0100
Subject: [PATCH 1/2] test_conf.cpp: make comparing `size_type` cross platform

On some platform `size_type` can be different to the previously hard
coded `long unsigned int`.
---
 test/libdnf5/conf/test_conf.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/test/libdnf5/conf/test_conf.cpp b/test/libdnf5/conf/test_conf.cpp
index a031481d..15be8b6d 100644
--- a/test/libdnf5/conf/test_conf.cpp
+++ b/test/libdnf5/conf/test_conf.cpp
@@ -100,11 +100,11 @@ void ConfTest::test_config_load_from_config() {
     CPPUNIT_ASSERT_EQUAL(libdnf5::Option::Priority::RUNTIME, config_copy.get_assumeyes_option().get_priority());
     CPPUNIT_ASSERT_EQUAL(true, config_copy.get_assumeyes_option().get_value());
 
-    CPPUNIT_ASSERT_EQUAL(0ul, config_copy.get_excludepkgs_option().get_value().size());
-    CPPUNIT_ASSERT_EQUAL(0ul, config.get_excludepkgs_option().get_value().size());
+    CPPUNIT_ASSERT_EQUAL(static_cast<std::size_t>(0), config_copy.get_excludepkgs_option().get_value().size());
+    CPPUNIT_ASSERT_EQUAL(static_cast<std::size_t>(0), config.get_excludepkgs_option().get_value().size());
 
     config_copy.get_excludepkgs_option().add_item(libdnf5::Option::Priority::RUNTIME, "abc");
 
-    CPPUNIT_ASSERT_EQUAL(1ul, config_copy.get_excludepkgs_option().get_value().size());
-    CPPUNIT_ASSERT_EQUAL(0ul, config.get_excludepkgs_option().get_value().size());
+    CPPUNIT_ASSERT_EQUAL(static_cast<std::size_t>(1), config_copy.get_excludepkgs_option().get_value().size());
+    CPPUNIT_ASSERT_EQUAL(static_cast<std::size_t>(0), config.get_excludepkgs_option().get_value().size());
 }
-- 
2.51.0

