diff -Nru aws-c-io-0.27.7.orig/CMakeLists.txt aws-c-io-0.27.7/CMakeLists.txt
--- aws-c-io-0.27.7.orig/CMakeLists.txt	2026-08-18 00:32:18.000000000 +0000
+++ aws-c-io-0.27.7/CMakeLists.txt	2026-08-25 17:08:15.087430300 +0000
@@ -189,6 +189,7 @@
 # We are not ABI stable yet
 # TODO: switch to VERSION derived from the VERSION file (via aws_get_version) once SOVERSION is set explicitly, to avoid changing the SONAME.
 set_target_properties(${PROJECT_NAME} PROPERTIES VERSION 1.0.0)
+set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES SOVERSION 0unstable)
 
 if (NOT EVENT_LOOP_DEFINES)
     message(FATAL_ERROR "Event Loop is not setup on the platform.")
