--- ./ext/GDBM_File/t/fatal.t.orig	2019-10-24 20:18:04.000000000 +0000
+++ ./ext/GDBM_File/t/fatal.t	2019-11-19 13:40:06.696330179 +0000
@@ -37,6 +37,7 @@ is((open $fh, "<&=$fileno"), undef,
 umask(0);
 my %h;
 isa_ok(tie(%h, 'GDBM_File', 'fatal_dbmx', GDBM_WRCREAT, 0640), 'GDBM_File');
+tied(%h)->setopt(7, 0, length(pack("i", 0)));	# turn off mmap mode
 
 isnt((open $fh, "<&=$fileno"), undef, "dup fileno $fileno")
     or diag("\$! = $!");
