--- compiler-rt-r215000/lib/tsan/go/buildgo.sh.orig	2021-07-12 09:06:26.349444513 +0200
+++ compiler-rt-r215000/lib/tsan/go/buildgo.sh	2021-07-12 09:06:58.949876578 +0200
@@ -31,7 +31,7 @@
 if [ "`uname -a | grep Linux`" != "" ]; then
 	SUFFIX="linux_amd64"
 	OSCFLAGS="-fPIC -ffreestanding -Wno-maybe-uninitialized -Wno-unused-const-variable -Werror -Wno-unknown-warning-option"
-	OSLDFLAGS="-lpthread -fPIC -fpie"
+	OSLDFLAGS="-lpthread -fPIC -fpie -no-pie"
 	SRCS+="
 		../rtl/tsan_platform_linux.cc
 		../../sanitizer_common/sanitizer_posix.cc
