Index: mc-4.8.33/misc/ext.d/sound.sh
===================================================================
--- mc-4.8.33.orig/misc/ext.d/sound.sh
+++ mc-4.8.33/misc/ext.d/sound.sh
@@ -13,6 +13,11 @@ fi
 do_view_action() {
     filetype=$1
 
+    if type -p mediainfo >/dev/null 2>&1 ; then
+        mediainfo "${MC_EXT_FILENAME}"
+        return $?
+    fi
+
     case "${filetype}" in
     common)
         mediainfo "${MC_EXT_FILENAME}"
@@ -33,7 +38,7 @@ do_view_action() {
             tail +13 || file "${MC_EXT_FILENAME}"
         ;;
     *)
-        cat "${MC_EXT_FILENAME}"
+        file "${MC_EXT_FILENAME}"
         ;;
     esac
 }
