Index: misc/mc.ext.in
===================================================================
--- misc/mc.ext.in.orig
+++ misc/mc.ext.in
@@ -573,17 +573,20 @@ shell/i/.svg
 ### Sound files ###
 
 regex/i/\.(wav|snd|voc|au|smp|aiff|snd|m4a|ape|aac|wv)$
-	Open=@EXTHELPERSDIR@/sound.sh open common
+	Include=audio
 	View=%view{ascii} @EXTHELPERSDIR@/sound.sh view common
 
 regex/i/\.(mod|s3m|xm|it|mtm|669|stm|ult|far)$
 	Open=@EXTHELPERSDIR@/sound.sh open mod
+	View=%view{ascii} @EXTHELPERSDIR@/sound.sh view mod
 
 shell/i/.waw22
 	Open=@EXTHELPERSDIR@/sound.sh open wav22
+	View=%view{ascii} @EXTHELPERSDIR@/sound.sh view wav22
 
 shell/i/.mp3
 	Open=@EXTHELPERSDIR@/sound.sh open mp3
+	Include=audio
 	View=%view{ascii} @EXTHELPERSDIR@/sound.sh view mp3
 
 regex/i/\.og[gax]$
@@ -595,15 +598,22 @@ shell/i/.opus
 	View=%view{ascii} @EXTHELPERSDIR@/sound.sh view opus
 
 regex/i/\.(spx|flac)$
-	Open=@EXTHELPERSDIR@/sound.sh open common
+	Include=audio
 
 regex/i/\.(midi?|rmid?)$
 	Open=@EXTHELPERSDIR@/sound.sh open midi
+	View=%view{ascii} @EXTHELPERSDIR@/sound.sh view midi
 
 shell/i/.wma
 	Open=@EXTHELPERSDIR@/sound.sh open wma
 	View=%view{ascii} @EXTHELPERSDIR@/sound.sh view wma
 
+shell/i/.mka
+	Include=audio
+
+include/audio
+	Open=@EXTHELPERSDIR@/sound.sh open common
+	View=%view{ascii} @EXTHELPERSDIR@/sound.sh view common
 
 ### Play lists ###
 
