Author: Adam Majer
Upstream: http://midnight-commander.org/ticket/3912
Bugzilla: 1087708

Index: mc-4.8.22/misc/ext.d/doc.sh.in
===================================================================
--- mc-4.8.22.orig/misc/ext.d/doc.sh.in
+++ mc-4.8.22/misc/ext.d/doc.sh.in
@@ -74,7 +74,7 @@ do_view_action() {
             elinks -dump "$tmp/page.html"
             rm -rf "$tmp"
         elif xls2csv >/dev/null 2>&1; then
-            xls2csv "${MC_EXT_FILENAME}"
+            xls2csv -c - -x "${MC_EXT_FILENAME}"
         else
             strings "${MC_EXT_FILENAME}"
         fi
