--- configure.in
+++ configure.in
@@ -727,6 +727,11 @@
 dnl end of headers etc.
 dnl --------------------------------------------------
 
+AC_CHECK_LIB(supc++, main, LIBSUPCPP=-lsupc++)
+AC_SUBST(LIBSUPCPP)
+AC_CHECK_LIB(gcc_eh, main, LIBGCC_EH=-lgcc_eh)
+AC_SUBST(LIBGCC_EH)
+
 dnl --------------------------------------------------
 dnl output files
 dnl vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
--- src/engine/Makefile.am
+++ src/engine/Makefile.am
@@ -1,6 +1,6 @@
 EXPATH = @EXPATH@
 
-CXXFLAGS = @CXXFLAGS@ @CXXWARNING_FLAGS@
+AM_CXXFLAGS = @CXXFLAGS@ @CXXWARNING_FLAGS@
 
 lib_LTLIBRARIES = libsablot.la
 
@@ -75,5 +75,6 @@
 	sabdbg.h \
 	sabcfg.h
 
-libsablot_la_LDFLAGS = -version-info 100:3:100
+libsablot_la_LDFLAGS = -L$(libdir) -version-info 100:3:100
 
+libsablot_la_LIBADD = @LIBSUPCPP@ @LIBGCC_EH@ @EXPAT_LIBS@
