From 36d44bf9f9b99cca77dea313eec3bfef435cbd5d Mon Sep 17 00:00:00 2001
From: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
Date: Mon, 8 Jun 2020 20:15:32 -0500
Subject: [PATCH 14/51] broadwell-rt286: add SOF support

Include legacy configuration using SOF driver name and card check

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 ucm2/SOF/HiFi.conf | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/ucm2/SOF/HiFi.conf b/ucm2/SOF/HiFi.conf
index 82a66a067c49..2a6e21e89bcc 100644
--- a/ucm2/SOF/HiFi.conf
+++ b/ucm2/SOF/HiFi.conf
@@ -78,3 +78,12 @@ If.bytcht_max98090 {
 	}
 	True.Include.main.File "/chtmax98090/HiFi.conf"
 }
+
+If.bdw_rt286 {
+	Condition {
+		Type String
+		Haystack "${CardName}"
+		Needle "sof-bdw rt286"
+	}
+	True.Include.main.File "/broadwell-rt286/HiFi.conf"
+}
-- 
2.16.4

