From eed10701463b605ba5b6587d897deff742cbb5a0 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Tue, 4 Feb 2020 17:48:10 +0100
Subject: [PATCH 24/36] VEYRON-I2S: corrections and fixes

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 ucm2/VEYRON-I2S/HiFi.conf | 24 ++++++++++++++++--------
 1 file changed, 16 insertions(+), 8 deletions(-)

diff --git a/ucm2/VEYRON-I2S/HiFi.conf b/ucm2/VEYRON-I2S/HiFi.conf
index 8d07c7337fd7..bed87c54cbd9 100644
--- a/ucm2/VEYRON-I2S/HiFi.conf
+++ b/ucm2/VEYRON-I2S/HiFi.conf
@@ -45,25 +45,31 @@ SectionDevice."Speaker" {
 	Comment "Speaker"
 
 	Value {
-		PlaybackPCM "hw:${CardId},0"
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
 	}
+
 	EnableSequence [
 		cset "name='Speaker Switch' on"
 	]
+
 	DisableSequence [
 		cset "name='Speaker Switch' off"
 	]
 }
 
-SectionDevice."Internal Mic" {
-	Comment "Int Mic"
+SectionDevice."Mic" {
+	Comment "Internal Microphone"
 
 	Value {
-		CapturePCM "hw:${CardId},0"
+		CapturePriority 100
+		CapturePCM "hw:${CardId}"
 	}
+
 	EnableSequence [
 		cset "name='Int Mic Switch' on"
 	]
+
 	DisableSequence [
 		cset "name='Int Mic Switch' off"
 	]
@@ -73,7 +79,8 @@ SectionDevice."Headphones" {
 	Comment "Headphones"
 
 	Value {
-		PlaybackPCM "hw:${CardId},0"
+		PlaybackPriority 200
+		PlaybackPCM "hw:${CardId}"
 	}
 
 	EnableSequence [
@@ -88,11 +95,12 @@ SectionDevice."Headphones" {
 	]
 }
 
-SectionDevice."Mic" {
-	Comment "Headset Mic"
+SectionDevice."Headset" {
+	Comment "Headset Microphone"
 
 	Value {
-		CapturePCM "hw:${CardId},0"
+		CapturePriority 200
+		CapturePCM "hw:${CardId}"
 	}
 
 	EnableSequence [
-- 
2.16.4

