From 2abe227ba1ba8ff4b5b5aedfc4611474be0ff57a Mon Sep 17 00:00:00 2001
From: Lucas Mulling <lucas.mulling@suse.com>
Date: Tue, 13 May 2025 16:36:29 -0300
Subject: [PATCH GnuPG] dirmngr: Don't install expired sks certificate

* dirmngr/Makefile.am (dist_pkgdata_DATA): Remove
  sks-keyservers.netCA.pem.

Signed-off-by: Lucas Mulling <lucas.mulling@suse.com>
---
 dirmngr/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dirmngr/Makefile.am b/dirmngr/Makefile.am
index a0f8d5a79..b13f37d09 100644
--- a/dirmngr/Makefile.am
+++ b/dirmngr/Makefile.am
@@ -29,7 +29,7 @@ EXTRA_DIST = OAUTHORS ONEWS ChangeLog-2011 tls-ca.pem \
 
 
 
-dist_pkgdata_DATA = sks-keyservers.netCA.pem
+dist_pkgdata_DATA =
 
 bin_PROGRAMS = dirmngr dirmngr-client
 
-- 
2.49.0

