From: Philipp Thomas <pth@suse.de>
Date: 2013-04-11 11:25:14+02:00
Subject: Clean up the code

---
 ez-ipupdate.c |    6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

signed-off-by: pth@suse.de

Index: ez-ipupdate.c
===================================================================
--- ez-ipupdate.c.orig	2013-04-11 10:54:02.685445947 +0200
+++ ez-ipupdate.c	2013-04-11 11:21:24.515547623 +0200
@@ -2525,7 +2525,7 @@ int DHS_update_entry(void)
   p += strlen(p);
   limit = BUFFER_SIZE - 1 - strlen(buf);
 
-  snprintf(buf, BUFFER_SIZE, "Content-length: %d\015\012", strlen(putbuf));
+  snprintf(buf, BUFFER_SIZE, "Content-length: %d\015\012", (int)strlen(putbuf));
   output(buf);
   snprintf(buf, BUFFER_SIZE, "\015\012");
   output(buf);
@@ -2662,7 +2662,7 @@ int DHS_update_entry(void)
     p += strlen(p);
     limit = BUFFER_SIZE - 1 - strlen(buf);
 
-    snprintf(buf, BUFFER_SIZE, "Content-length: %d\015\012", strlen(putbuf));
+    snprintf(buf, BUFFER_SIZE, "Content-length: %d\015\012", (int)strlen(putbuf));
     output(buf);
     snprintf(buf, BUFFER_SIZE, "\015\012");
     output(buf);
@@ -4257,8 +4257,6 @@ int HEIPV6TB_update_entry(void)
 
   switch(ret)
   {
-    char *p;
-
     case -1:
       if(!(options & OPT_QUIET))
       {
