diff -rupN a/desktop/build-scripts/common-resources/common-linux/share/applications/tuxguitar.desktop b/desktop/build-scripts/common-resources/common-linux/share/applications/tuxguitar.desktop
--- a/desktop/build-scripts/common-resources/common-linux/share/applications/tuxguitar.desktop	2024-01-26 20:39:47.000000000 +0100
+++ b/desktop/build-scripts/common-resources/common-linux/share/applications/tuxguitar.desktop	2024-01-27 21:41:28.253414136 +0100
@@ -15,6 +15,6 @@ Type=Application
 MimeType=audio/x-tuxguitar;audio/x-gtp;audio/x-ptb
 Categories=AudioVideo;Audio
 Exec=tuxguitar %F
-Icon=/opt/tuxguitar/share/skins/Oxygen/icon.png
+Icon=tuxguitar
 Terminal=false
 StartupNotify=false
