--- perl/Makefile.PL	2002-12-15 20:40:19.000000000 +0100
+++ perl/Makefile.PL	2011-05-26 22:28:06.673024204 +0200
@@ -99,10 +99,7 @@
 # the contents of the Makefile that is written.
 
 # Work around bug in previous versions of MakeMaker
-WriteMakefile(NAME => 'FCGI') 
-    if $ExtUtils::MakeMaker::VERSION <= 5.4302;
-
-$mm = MM->new({
+WriteMakefile(
     'NAME'		=> 'FCGI',
     'VERSION_FROM'	=> 'version.pm',
     'dist'		=> { 'COMPRESS' => 'gzip -9f', 
@@ -123,10 +120,7 @@
     'PL_FILES'		=> $plfiles,
     PM			=> {'FCGI.pm' => '$(INST_ARCHLIBDIR)/FCGI.pm'},
     @extras,
-});
-# don't install oldinterface pod
-delete $mm->{MAN3PODS}{oldinterface.pod};
-$mm->flush;
+);
 
 exit if -f 'fcgi_config.h' or $libfound or $pure;
 
