diff -uprN mono-debugger-2.10.old/classes/StackFrame.cs mono-debugger-2.10/classes/StackFrame.cs
--- mono-debugger-2.10.old/classes/StackFrame.cs	2011-01-14 01:41:31.000000000 +0300
+++ mono-debugger-2.10/classes/StackFrame.cs	2018-03-24 01:46:11.675203718 +0300
@@ -5,6 +5,7 @@ using System.Runtime.InteropServices;
 
 using Mono.Debugger.Languages;
 using Mono.Debugger.Backend;
+using Architecture = Mono.Debugger.Backend.Architecture;
 
 namespace Mono.Debugger
 {
diff -uprN mono-debugger-2.10.old/classes/Thread.cs mono-debugger-2.10/classes/Thread.cs
--- mono-debugger-2.10.old/classes/Thread.cs	2011-01-14 01:41:31.000000000 +0300
+++ mono-debugger-2.10/classes/Thread.cs	2018-03-24 01:49:26.635203718 +0300
@@ -12,6 +12,7 @@ using System.Runtime.Remoting.Messaging;
 
 using Mono.Debugger.Backend;
 using Mono.Debugger.Languages;
+using Architecture = Mono.Debugger.Backend.Architecture;
 
 namespace Mono.Debugger
 {
