Index: sxiv-25/Makefile
===================================================================
--- sxiv-25.orig/Makefile	2019-01-26 14:54:40.000000000 +0100
+++ sxiv-25/Makefile	2019-10-13 16:45:00.384870829 +0200
@@ -16,8 +16,9 @@
 HAVE_LIBEXIF = 1
 
 cflags = -std=c99 -Wall -pedantic $(CFLAGS)
-cppflags = -I. $(CPPFLAGS) -D_XOPEN_SOURCE=700 -DHAVE_GIFLIB=$(HAVE_GIFLIB) \
-  -DHAVE_LIBEXIF=$(HAVE_LIBEXIF) -I/usr/include/freetype2
+cppflags = -I. $(CPPFLAGS) -D_GNU_SOURCE -DVERSION=\"$(VERSION)\" \
+   -DHAVE_GIFLIB=$(HAVE_GIFLIB) -DHAVE_LIBEXIF=$(HAVE_LIBEXIF) \
+   -I/usr/include/freetype2
 
 lib_exif_0 =
 lib_exif_1 = -lexif
@@ -76,6 +77,7 @@
 	mkdir -p $(DESTDIR)$(PREFIX)/share/sxiv/exec
 	cp exec/* $(DESTDIR)$(PREFIX)/share/sxiv/exec/
 	chmod 755 $(DESTDIR)$(PREFIX)/share/sxiv/exec/*
+	install -m 644 sxiv.desktop -D $(DESTDIR)$(PREFIX)/share/applications/sxiv.desktop
 
 uninstall:
 	@echo "REMOVE bin/sxiv"
@@ -84,4 +86,5 @@
 	rm -f $(DESTDIR)$(MANPREFIX)/man1/sxiv.1
 	@echo "REMOVE share/sxiv/"
 	rm -rf $(DESTDIR)$(PREFIX)/share/sxiv
-
+	@echo "REMOVE share/applications/sxiv.desktop"
+	rm -f $(DESTDIR)$(PREFIX)/share/applications/sxiv.desktop
