Index: config/jail.conf
===================================================================
--- config/jail.conf.orig
+++ config/jail.conf
@@ -688,7 +688,7 @@ backend = %(syslog_backend)s
 # filter   = named-refused
 # port     = domain,953
 # protocol = udp
-# logpath  = /var/log/named/security.log
+# logpath  = /var/lib/named/log/security.log
 
 # IMPORTANT: see filter.d/named-refused for instructions to enable logging
 # This jail blocks TCP traffic for DNS requests.
@@ -696,7 +696,7 @@ backend = %(syslog_backend)s
 [named-refused]
 
 port     = domain,953
-logpath  = /var/log/named/security.log
+logpath  = /var/lib/named/log/security.log
 
 
 [nsd]
Index: config/paths-common.conf
===================================================================
--- config/paths-common.conf.orig
+++ config/paths-common.conf
@@ -90,7 +90,7 @@ solidpop3d_log = %(syslog_local0)s
 mysql_log = %(syslog_daemon)s
 mysql_backend = %(default_backend)s
 
-roundcube_errors_log = /var/log/roundcube/errors
+roundcube_errors_log = /srv/www/roundcubemail/logs/errors
 
 # Directory with ignorecommand scripts
 ignorecommands_dir = /etc/fail2ban/filter.d/ignorecommands
