Index: haproxy-3.3/Makefile
===================================================================
--- haproxy-3.3.orig/Makefile
+++ haproxy-3.3/Makefile
@@ -799,7 +799,7 @@ ifneq ($(USE_PCRE:0=)$(USE_STATIC_PCRE:0
   PCREDIR     := $(shell $(PCRE_CONFIG) --prefix 2>/dev/null || echo /usr/local)
   ifneq ($(PCREDIR),)
     PCRE_INC := $(PCREDIR)/include
-    PCRE_LIB := $(PCREDIR)/lib
+    PCRE_LIB := $(PCREDIR)/$(LIB)
   endif
 
   PCRE_CFLAGS := $(if $(PCRE_INC),-I$(PCRE_INC))
@@ -817,7 +817,7 @@ ifneq ($(USE_PCRE2:0=)$(USE_STATIC_PCRE2
   PCRE2DIR     := $(shell $(PCRE2_CONFIG) --prefix 2>/dev/null || echo /usr/local)
   ifneq ($(PCRE2DIR),)
     PCRE2_INC := $(PCRE2DIR)/include
-    PCRE2_LIB := $(PCRE2DIR)/lib
+    PCRE2_LIB := $(PCRE2DIR)/$(LIB)
 
     ifeq ($(PCRE2_WIDTH),)
       PCRE2_WIDTH = 8
