diff --git a/rpcs3/CMakeLists.txt b/rpcs3/CMakeLists.txt
index 727bee8..adb7f5c 100644
--- a/rpcs3/CMakeLists.txt
+++ b/rpcs3/CMakeLists.txt
@@ -181,8 +181,6 @@ if (NOT ANDROID)
             DESTINATION ${CMAKE_INSTALL_DATADIR}/rpcs3)
         install(DIRECTORY ../bin/GuiConfigs
             DESTINATION ${CMAKE_INSTALL_DATADIR}/rpcs3)
-        install(DIRECTORY ../bin/test
-            DESTINATION ${CMAKE_INSTALL_DATADIR}/rpcs3)
     endif()
 endif()
