diff --git a/3rdparty/qt6.cmake b/3rdparty/qt6.cmake
index ef89bda..e15e0ab 100644
--- a/3rdparty/qt6.cmake
+++ b/3rdparty/qt6.cmake
@@ -6,14 +6,15 @@ find_package(Qt6 ${QT_MIN_VER} CONFIG COMPONENTS Widgets Concurrent Multimedia M
 if(WIN32)
 	target_link_libraries(3rdparty_qt6 INTERFACE Qt6::Widgets Qt6::Concurrent Qt6::Multimedia Qt6::MultimediaWidgets Qt6::Svg Qt6::SvgWidgets)
 else()
-	find_package(Qt6 ${QT_MIN_VER} COMPONENTS DBus Gui)
+	set(QT_NO_PRIVATE_MODULE_WARNING ON)
+	find_package(Qt6 ${QT_MIN_VER} COMPONENTS DBus Gui GuiPrivate)
 	if(Qt6DBus_FOUND)
 		target_link_libraries(3rdparty_qt6 INTERFACE Qt6::Widgets Qt6::DBus Qt6::Concurrent Qt6::Multimedia Qt6::MultimediaWidgets Qt6::Svg Qt6::SvgWidgets)
 		target_compile_definitions(3rdparty_qt6 INTERFACE -DHAVE_QTDBUS)
 	else()
 		target_link_libraries(3rdparty_qt6 INTERFACE Qt6::Widgets Qt6::Concurrent Qt6::Multimedia Qt6::MultimediaWidgets Qt6::Svg Qt6::SvgWidgets)
 	endif()
-	target_include_directories(3rdparty_qt6 INTERFACE ${Qt6Gui_PRIVATE_INCLUDE_DIRS})
+	target_link_libraries(3rdparty_qt6 INTERFACE Qt6::GuiPrivate)
 endif()
 
 if(Qt6Widgets_FOUND)
