diff --git a/3rdparty/qt6.cmake b/3rdparty/qt6.cmake
index ef89bda..bda135a 100644
--- a/3rdparty/qt6.cmake
+++ b/3rdparty/qt6.cmake
@@ -6,12 +6,12 @@ find_package(Qt6 ${QT_MIN_VER} CONFIG COMPONENTS Widgets Concurrent Multimedia M
 if(WIN32)
 	target_link_libraries(3rdparty_qt6 INTERFACE Qt6::Widgets Qt6::Concurrent Qt6::Multimedia Qt6::MultimediaWidgets Qt6::Svg Qt6::SvgWidgets)
 else()
-	find_package(Qt6 ${QT_MIN_VER} COMPONENTS DBus Gui)
+	find_package(Qt6 ${QT_MIN_VER} COMPONENTS DBus GuiPrivate)
 	if(Qt6DBus_FOUND)
-		target_link_libraries(3rdparty_qt6 INTERFACE Qt6::Widgets Qt6::DBus Qt6::Concurrent Qt6::Multimedia Qt6::MultimediaWidgets Qt6::Svg Qt6::SvgWidgets)
+		target_link_libraries(3rdparty_qt6 INTERFACE Qt6::Widgets Qt6::DBus Qt6::Concurrent Qt6::GuiPrivate Qt6::Multimedia Qt6::MultimediaWidgets Qt6::Svg Qt6::SvgWidgets)
 		target_compile_definitions(3rdparty_qt6 INTERFACE -DHAVE_QTDBUS)
 	else()
-		target_link_libraries(3rdparty_qt6 INTERFACE Qt6::Widgets Qt6::Concurrent Qt6::Multimedia Qt6::MultimediaWidgets Qt6::Svg Qt6::SvgWidgets)
+		target_link_libraries(3rdparty_qt6 INTERFACE Qt6::Widgets Qt6::Concurrent Qt6::GuiPrivate Qt6::Multimedia Qt6::MultimediaWidgets Qt6::Svg Qt6::SvgWidgets)
 	endif()
 	target_include_directories(3rdparty_qt6 INTERFACE ${Qt6Gui_PRIVATE_INCLUDE_DIRS})
 endif()
