Index: docky-2.2.0/StandardPlugins/WorkspaceSwitcher/src/Desk.cs
===================================================================
--- docky-2.2.0.orig/StandardPlugins/WorkspaceSwitcher/src/Desk.cs
+++ docky-2.2.0/StandardPlugins/WorkspaceSwitcher/src/Desk.cs
@@ -130,7 +130,13 @@ namespace WorkspaceSwitcher
 					grid [x, y] = desk;
 				}
 				if (gridsize.Y - 1 > y) {
-					desk = (grid [0, y] != null ? grid [0, y].GetNeighbor (Wnck.MotionDirection.Down) : null);
+					if (grid [0, y] != null ) {
+						desk = grid [0, y].GetNeighbor (Wnck.MotionDirection.Down);
+					}
+					else {
+						desk = null;
+					}
+					// desk = grid [0, y] != null ? grid [0, y].GetNeighbor (Wnck.MotionDirection.Down) : null;
 					grid [0, y+1] = desk;
 				}
 			}
