From 3203bf12a4bef81f880789eb9cde7ff489ae5115 Mon Sep 17 00:00:00 2001
From: Markus Seidel <markus.seidel@cern.ch>
Date: Fri, 4 Dec 2020 09:23:06 +0100
Subject: [PATCH] Fix compilation of L3_2008_I825820

---
 analyses/pluginLEP/L3_2008_I825820.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/analyses/pluginLEP/L3_2008_I825820.cc
b/analyses/pluginLEP/L3_2008_I825820.cc
index 16de60393..216833e6c 100644
--- a/analyses/pluginLEP/L3_2008_I825820.cc
+++ b/analyses/pluginLEP/L3_2008_I825820.cc
@@ -1,5 +1,7 @@
 // -*- C++ -*-
 #include "Rivet/Analysis.hh"
+#include "Rivet/Projections/Beam.hh"
+#include "Rivet/Projections/ChargedFinalState.hh"
 #include "Rivet/Projections/Thrust.hh"
 #include "Rivet/Projections/Hemispheres.hh"
 #include "Rivet/Projections/ParisiTensor.hh"
-- 
GitLab

From 476f267c46b126fa163a92aa6cbcb7806c4624c3 Mon Sep 17 00:00:00 2001
From: Markus Seidel <markus.seidel@cern.ch>
Date: Mon, 7 Dec 2020 12:40:44 +0100
Subject: [PATCH] Fix analysis compilation errors

---
 analyses/pluginBELLE/BELLE_2019_I1762826.cc | 5 +----
 analyses/pluginBELLE/BELLE_2020_I1796822.cc | 5 +----
 analyses/pluginLEP/L3_2008_I825820.cc       | 1 +
 3 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/analyses/pluginBELLE/BELLE_2019_I1762826.cc
b/analyses/pluginBELLE/BELLE_2019_I1762826.cc
index 4f47368a9..95133c83a 100644
--- a/analyses/pluginBELLE/BELLE_2019_I1762826.cc
+++ b/analyses/pluginBELLE/BELLE_2019_I1762826.cc
@@ -1,10 +1,7 @@
 // -*- C++ -*-
 #include "Rivet/Analysis.hh"
 #include "Rivet/Projections/FinalState.hh"
-#include "Rivet/Projections/FastJets.hh"
-#include "Rivet/Projections/DressedLeptons.hh"
-#include "Rivet/Projections/MissingMomentum.hh"
-#include "Rivet/Projections/PromptFinalState.hh"
+#include "Rivet/Projections/UnstableParticles.hh"
 
 namespace Rivet {
 
diff --git a/analyses/pluginBELLE/BELLE_2020_I1796822.cc
b/analyses/pluginBELLE/BELLE_2020_I1796822.cc
index a35cfff5d..080605b54 100644
--- a/analyses/pluginBELLE/BELLE_2020_I1796822.cc
+++ b/analyses/pluginBELLE/BELLE_2020_I1796822.cc
@@ -1,10 +1,7 @@
 // -*- C++ -*-
 #include "Rivet/Analysis.hh"
 #include "Rivet/Projections/FinalState.hh"
-#include "Rivet/Projections/FastJets.hh"
-#include "Rivet/Projections/DressedLeptons.hh"
-#include "Rivet/Projections/MissingMomentum.hh"
-#include "Rivet/Projections/PromptFinalState.hh"
+#include "Rivet/Projections/UnstableParticles.hh"
 
 namespace Rivet {
 
diff --git a/analyses/pluginLEP/L3_2008_I825820.cc
b/analyses/pluginLEP/L3_2008_I825820.cc
index 216833e6c..584e248ad 100644
--- a/analyses/pluginLEP/L3_2008_I825820.cc
+++ b/analyses/pluginLEP/L3_2008_I825820.cc
@@ -1,5 +1,6 @@
 // -*- C++ -*-
 #include "Rivet/Analysis.hh"
+#include "Rivet/Projections/FinalState.hh"
 #include "Rivet/Projections/Beam.hh"
 #include "Rivet/Projections/ChargedFinalState.hh"
 #include "Rivet/Projections/Thrust.hh"
-- 
GitLab
