Index: Rivet-3.1.7/configure.ac
===================================================================
--- Rivet-3.1.7.orig/configure.ac
+++ Rivet-3.1.7/configure.ac
@@ -358,7 +358,7 @@ if test x$enable_pyext == xyes; then
   AC_SUBST(PYTHON_FULL_VERSION)
   AC_SUBST(PYTHON_MAJOR_VERSION)
   #RIVET_PYTHONPATH=$pythondir
-  RIVET_PYTHONPATH=`$PYTHON -c "from __future__ import print_function; import sysconfig; print(sysconfig.get_path('platlib', 'posix_user', vars={'userbase' : '$prefix'}))"`
+  RIVET_PYTHONPATH=`$PYTHON -c "from __future__ import print_function; import sysconfig; print(sysconfig.get_path('platlib', vars={'userbase' : '$prefix'}))"`
   AC_SUBST(RIVET_PYTHONPATH)
   AC_MSG_NOTICE(Rivet Python lib will be installed to $RIVET_PYTHONPATH)
   if test -z "$PYTHON"; then
