diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
index f78f275..b67ac1e 100644
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -117,7 +117,7 @@ AC_DEFUN([AC_OPENSSL],
               [with_openssl_prefix=/usr])
 
   if test "x$with_openssl_prefix" = "x/usr" ; then
-    AC_CHECK_LIB(crypto, CRYPTO_num_locks, [found=yes], [found=no])
+    AC_CHECK_LIB(crypto, ERR_print_errors_fp, [found=yes], [found=no])
 
     if test "x$found" = "xyes" ; then
 	OPENSSL_LIBS="-lcrypto -lssl"
@@ -128,7 +128,7 @@ AC_DEFUN([AC_OPENSSL],
     LD_LIBRARY_PATH="$with_openssl_prefix/lib"
 
     AC_LANG_PUSH(C)
-    AC_CHECK_LIB(crypto, CRYPTO_num_locks, [found=yes], [found=no])
+    AC_CHECK_LIB(crypto, ERR_print_errors_fp, [found=yes], [found=no])
     AC_LANG_POP(C)  
     NO_GLOBUS_FLAGS="-I$with_openssl_prefix/include"
 
