diff --git a/src/api/ccapi/voms_apic.h b/src/api/ccapi/voms_apic.h
index e613d09..8807803 100644
--- a/src/api/ccapi/voms_apic.h
+++ b/src/api/ccapi/voms_apic.h
@@ -31,18 +31,10 @@ extern "C" {
 #endif
 
 #define NOGLOBUS
-#ifndef GSSAPI_H_
-
-/*
- * Also check against _GSSAPI_H_ as that is what the Kerberos 5 code defines and
- * what header files on some systems look for.
- */
-
-#ifndef _GSSAPI_H
+#if !(defined(GSSAPI_H_) || defined(_GSSAPI_H) || defined(_GSSAPI_H_))
 typedef void * gss_cred_id_t;
 typedef void * gss_ctx_id_t;
 #endif
-#endif
 
 #include <openssl/x509.h>
 #include <time.h>
