From 830e2df6c7938b6ddd515ef4c88d72d4de81bf3e Mon Sep 17 00:00:00 2001
From: Rowan Thorpe <rowan@rowanthorpe.com>
Date: Mon, 9 Dec 2013 11:38:11 +0200
Subject: [PATCH] Fix Makefile typo

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index eca5ccf..60dbd47 100644
--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ all: $(DISTFILES)
 
 $(DISTFILES): jshon.o
 
-strip: $(DISFILES)
+strip: $(DISTFILES)
 	strip --strip-all $(DISTFILES)
 
 clean:
-- 
2.16.4

