diff --git a/Makefile b/Makefile
index 60dbd47..fb99bc2 100644
--- a/Makefile
+++ b/Makefile
@@ -25,8 +25,8 @@ clean:
 	rm -f *.o $(DISTFILES)
 
 install:
-	$(INSTALL) -D $(DISTFILES) $(TARGET_PATH)/$(DISTFILES)
-	$(INSTALL) -D $(MANFILE) $(MANDIR)/$(MANFILE)
+	$(INSTALL) -Dm0755 $(DISTFILES) $(TARGET_PATH)/$(DISTFILES)
+	$(INSTALL) -Dm0644 $(MANFILE) $(MANDIR)/$(MANFILE)
 
 dist: clean
 	sed -i "s/#define JSHONVER .*/#define JSHONVER ${VERSION}/" jshon.c
