Index: libArcus-4.13/CMakeLists.txt
===================================================================
--- libArcus-4.13.orig/CMakeLists.txt
+++ libArcus-4.13/CMakeLists.txt
@@ -120,7 +120,7 @@ endif()
 
 set_target_properties(Arcus PROPERTIES
     FRAMEWORK FALSE
-    VERSION ${ARCUS_VERSION}
+    VERSION ${ARCUS_SOVERSION}.1.0
     SOVERSION ${ARCUS_SOVERSION}
     PUBLIC_HEADER "${arcus_HDRS}"
     DEFINE_SYMBOL MAKE_ARCUS_LIB
