--- man/Makefile
+++ man/Makefile
@@ -41,12 +41,12 @@
 
 install: man-pages
 	$(prn_install)
-	$(ECMD)install -m 755 -d $(MAN4DIR)
-	$(ECMD)install -m 644 $(MANPAGES4) $(MAN4DIR)/
-	$(ECMD)install -m 755 -d $(MAN7DIR)
-	$(ECMD)install -m 644 $(MANPAGES7) $(MAN7DIR)/
-	$(ECMD)install -m 755 -d $(MAN8DIR)
-	$(ECMD)install -m 644 $(MANPAGES8) $(MAN8DIR)/
+	$(ECMD)install -m 755 -d $(ROOT)/$(MAN4DIR)
+	$(ECMD)install -m 644 $(MANPAGES4) $(ROOT)/$(MAN4DIR)/
+	$(ECMD)install -m 755 -d $(ROOT)/$(MAN7DIR)
+	$(ECMD)install -m 644 $(MANPAGES7) $(ROOT)/$(MAN7DIR)/
+	$(ECMD)install -m 755 -d $(ROOT)/$(MAN8DIR)
+	$(ECMD)install -m 644 $(MANPAGES8) $(ROOT)/$(MAN8DIR)/
 
 
 clean:
