Index: irattach/Makefile
===================================================================
--- irattach/Makefile.orig	2009-11-26 19:20:58.000000000 +0100
+++ irattach/Makefile	2009-11-26 19:30:01.626990378 +0100
@@ -31,7 +31,7 @@ RM           = rm -f
 CFLAGS       = $(RPM_OPT_FLAGS) -O2 -W -Wall -Wstrict-prototypes -Wshadow $(SYS_INCLUDES)
 
 SYS_INCLUDES = -I/usr/include
-SYS_LIBPATH  = -L/usr/lib -L/usr/local/lib 
+SYS_LIBPATH  = 
 
 INCLUDES     = $(SYS_INCLUDES) -I../include/
 LIBRARIES    = $(SYS_LIBRARIES)
Index: irdadump/Makefile
===================================================================
--- irdadump/Makefile.orig	2009-11-26 19:20:58.000000000 +0100
+++ irdadump/Makefile	2009-11-26 19:30:33.010374107 +0100
@@ -17,7 +17,7 @@ LIBPATH   = $(SYS_LIBPATH)
 #
 SYS_INCLUDES = -I/usr/include
 SYS_LIBRARIES = 
-SYS_LIBPATH = -L/usr/lib -L/usr/local/lib 
+SYS_LIBPATH = 
 
 CFLAGS= $(RPM_OPT_FLAGS) -O2 -W -Wall -I. `pkg-config --cflags glib-2.0`
 LIBIRDADUMP_OBJS = irdadump.o irlap.o irlmp.o obex.o ircomm.o netbuf.o capture.o
Index: irdaping/Makefile
===================================================================
--- irdaping/Makefile.orig	2009-11-26 19:20:58.000000000 +0100
+++ irdaping/Makefile	2009-11-26 19:30:51.242793418 +0100
@@ -40,7 +40,7 @@ LIBPATH   = $(SYS_LIBPATH)
 #
 SYS_INCLUDES = -I/usr/include
 SYS_LIBRARIES = 
-SYS_LIBPATH = -L/usr/lib -L/usr/local/lib 
+SYS_LIBPATH = 
 
 CFLAGS= $(RPM_OPT_FLAGS) $(SYS_INCLUDES) -Wall -Wstrict-prototypes -fno-strict-aliasing
 
Index: irsockets/Makefile
===================================================================
--- irsockets/Makefile.orig	2006-07-11 08:16:06.000000000 +0200
+++ irsockets/Makefile	2009-11-26 19:31:36.095361594 +0100
@@ -40,7 +40,7 @@ LIBPATH   = $(SYS_LIBPATH)
 #
 SYS_INCLUDES = -I/usr/include -I/usr/src/linux/include
 SYS_LIBRARIES = 
-SYS_LIBPATH = -L/usr/lib -L/usr/local/lib 
+SYS_LIBPATH = 
 
 CFLAGS= $(SYS_INCLUDES) $(RPM_OPT_FLAGS) -O2 -Wall -Wstrict-prototypes
 
Index: smcinit/Makefile
===================================================================
--- smcinit/Makefile.orig	2009-11-26 19:20:58.000000000 +0100
+++ smcinit/Makefile	2009-11-26 19:32:16.598441120 +0100
@@ -14,7 +14,7 @@ LIBPATH   = $(SYS_LIBPATH)
 #
 SYS_INCLUDES = -I/usr/include
 SYS_LIBRARIES = 
-SYS_LIBPATH = -L/usr/lib -L/usr/local/lib 
+SYS_LIBPATH = 
 
 CFLAGS= $(RPM_OPT_FLAGS) -O2 -W -Wall -I.
 
Index: tekram/Makefile
===================================================================
--- tekram/Makefile.orig	2009-11-26 19:20:58.000000000 +0100
+++ tekram/Makefile	2009-11-26 19:34:08.806871746 +0100
@@ -31,16 +31,16 @@ RM = rm -f
 
 RM_CMD = $(RM) *.BAK *.bak *.o ,* *~ *.a
 
-INCLUDES  = $(SYS_INCLUDES)
+INCLUDES  = $(SYS_INCLUDES) -I../include
 LIBRARIES = $(SYS_LIBRARIES)
 LIBPATH   = $(SYS_LIBPATH)
 
 #
 # System stuff
 #
-SYS_INCLUDES = -I/usr/include -I../include
+SYS_INCLUDES = -I/usr/include
 SYS_LIBRARIES = 
-SYS_LIBPATH = -L/usr/lib -L/usr/local/lib 
+SYS_LIBPATH = 
 
 CFLAGS= $(RPM_OPT_FLAGS) -Wall -Wstrict-prototypes
 
