Index: firetools-0.9.62/src/fstats/fstats.pro
===================================================================
--- firetools-0.9.62.orig/src/fstats/fstats.pro
+++ firetools-0.9.62/src/fstats/fstats.pro
@@ -1,6 +1,7 @@
 QMAKE_CXXFLAGS += $$(CXXFLAGS) -fstack-protector-all -D_FORTIFY_SOURCE=2 -fPIE -pie -Wformat -Wformat-security
 QMAKE_CFLAGS += $$(CFLAGS) -fstack-protector-all -D_FORTIFY_SOURCE=2 -fPIE -pie -Wformat -Wformat-security
-QMAKE_LFLAGS += $$(LDFLAGS) -lrt -Wl,-z,relro -Wl,-z,now
+QMAKE_LFLAGS += $$(LDFLAGS) -Wl,-z,relro -Wl,-z,now
+LIBS= -lrt -lc
 QT += widgets
  HEADERS       = ../common/utils.h ../common/pid.h ../common/common.h \
  		  pid_thread.h db.h dbstorage.h dbpid.h stats_dialog.h graph.h fstats.h
