diff -rup a/terminal/terminal-screen.c b/terminal/terminal-screen.c
--- a/terminal/terminal-screen.c	2024-01-29 14:34:45.000000000 +0100
+++ b/terminal/terminal-screen.c	2024-02-07 12:40:58.232372739 +0100
@@ -1892,7 +1892,7 @@ terminal_screen_paste_unsafe_text (Termi
 {
   GtkWidget *dialog;
 
-  g_return_if_fail (original_clipboard != GDK_SELECTION_CLIPBOARD && original_clipboard != GDK_SELECTION_PRIMARY);
+  g_return_if_fail (original_clipboard == GDK_SELECTION_CLIPBOARD || original_clipboard == GDK_SELECTION_PRIMARY);
 
   dialog = terminal_screen_unsafe_paste_dialog_new (screen, text);
   gtk_widget_show_all (dialog);
