Index: bcmatroska2-5.3.95/corec/corec/CMakeLists.txt
===================================================================
--- bcmatroska2-5.3.95.orig/corec/corec/CMakeLists.txt
+++ bcmatroska2-5.3.95/corec/corec/CMakeLists.txt
@@ -103,18 +103,18 @@ install(FILES
 	memalloc.h
 	memheap.h
 	portab.h
-	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/corec
+	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bcmatroska2/corec
 )
 
-install(FILES array/array.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/corec/array)
-install(FILES helpers/charconvert/charconvert.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/corec/helpers/charconvert)
-install(FILES helpers/date/date.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/corec/helpers/date)
+install(FILES array/array.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bcmatroska2/corec/array)
+install(FILES helpers/charconvert/charconvert.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bcmatroska2/corec/helpers/charconvert)
+install(FILES helpers/date/date.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bcmatroska2/corec/helpers/date)
 install(FILES
 	helpers/file/file.h
 	helpers/file/streams.h
-	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/corec/helpers/file
+	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bcmatroska2/corec/helpers/file
 )
-install(FILES helpers/md5/md5.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/corec/helpers/md5)
+install(FILES helpers/md5/md5.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bcmatroska2/corec/helpers/md5)
 install(FILES
 	helpers/parser/buffer.h
 	helpers/parser/dataheap.h
@@ -124,16 +124,16 @@ install(FILES
 	helpers/parser/strtab.h
 	helpers/parser/strtypes.h
 	helpers/parser/urlpart.h
-	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/corec/helpers/parser
+	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bcmatroska2/corec/helpers/parser
 )
-install(FILES helpers/system/ccsystem.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/corec/helpers/system)
-install(FILES multithread/multithread.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/corec/multithread)
+install(FILES helpers/system/ccsystem.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bcmatroska2/corec/helpers/system)
+install(FILES multithread/multithread.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bcmatroska2/corec/multithread)
 install(FILES
 	node/node.h
 	node/node_internal.h
 	node/nodebase.h
 	node/nodetools.h
 	node/nodetree.h
-	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/corec/node
+	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bcmatroska2/corec/node
 )
-install(FILES str/str.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/corec/str)
+install(FILES str/str.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bcmatroska2/corec/str)
Index: bcmatroska2-5.3.95/libebml2/CMakeLists.txt
===================================================================
--- bcmatroska2-5.3.95.orig/libebml2/CMakeLists.txt
+++ bcmatroska2-5.3.95/libebml2/CMakeLists.txt
@@ -34,5 +34,5 @@ add_library(bcmatroska2-ebml-objects OBJ
 set_target_properties(bcmatroska2-ebml-objects PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
 
 install(DIRECTORY ebml
-	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bcmatroska2
 )
Index: bcmatroska2-5.3.95/libmatroska2/CMakeLists.txt
===================================================================
--- bcmatroska2-5.3.95.orig/libmatroska2/CMakeLists.txt
+++ bcmatroska2-5.3.95/libmatroska2/CMakeLists.txt
@@ -72,5 +72,5 @@ install(TARGETS bcmatroska2 EXPORT ${PRO
 )
 
 install(DIRECTORY matroska
-	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/bcmatroska2
 )
