--- bcmatroska2-5.2.1/libmatroska2/CMakeLists.txt.orig	2023-01-04 14:22:06.937952129 +0100
+++ bcmatroska2-5.2.1/libmatroska2/CMakeLists.txt	2023-01-04 14:22:37.018106870 +0100
@@ -49,7 +49,7 @@
 	)
 endif()
 if(ENABLE_SHARED)
-	add_library(bcmatroska2 SHARED $<TARGET_OBJECTS:corec-objects> $<TARGET_OBJECTS:ebml2-objects> ${MATROSKA2_SOURCE_FILES})
+	add_library(bcmatroska2 SHARED $<TARGET_OBJECTS:corec-objects> $<TARGET_OBJECTS:ebml2-objects> $<TARGET_OBJECTS:bctoolbox> ${MATROSKA2_SOURCE_FILES})
 	set_target_properties(bcmatroska2 PROPERTIES VERSION 0.0.0 SOVERSION 0)
 	set_target_properties(bcmatroska2 PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
 	if(NOT "${LINK_FLAGS_STR}" STREQUAL "")
