diff -urpN foot.orig/tests/meson.build foot/tests/meson.build
--- foot.orig/tests/meson.build      2022-02-05 16:09:28.070754663 -0800
+++ foot/tests/meson.build   2022-02-05 16:09:46.739384271 -0800
@@ -3,6 +3,6 @@ config_test = executable(
   'test-config.c', '../tokenize.c',
   wl_proto_headers,
   link_with: [common],
-  dependencies: [pixman, xkb, fontconfig, fcft, tllist])
+  dependencies: [pixman, xkb, fontconfig, fcft, tllist, wayland_client])
 
 test('config', config_test)
