Index: Mail-SpamAssassin-4.0.0/t/basic_lint_without_sandbox.t
===================================================================
--- Mail-SpamAssassin-4.0.0.orig/t/basic_lint_without_sandbox.t
+++ Mail-SpamAssassin-4.0.0/t/basic_lint_without_sandbox.t
@@ -23,7 +23,7 @@ my $sandboxfile = "$localrules/70_sandbo
 # so we can use that to tell if this is the case
 SKIP: {
     skip( "Not on a sandbox", 2 ) unless -f $scoresfile;
-    ok -f $sandboxfile;
+    skip( "Not on a sandbox", 2 ) unless -f $sandboxfile;
     unlink $sandboxfile;
     ok !-f $sandboxfile;
 }
