--- geda-gaf-1.9.2/libgeda/scheme/Makefile.am.orig	2019-03-22 10:14:32.558929363 +0100
+++ geda-gaf-1.9.2/libgeda/scheme/Makefile.am	2019-03-22 10:14:58.758913793 +0100
@@ -41,15 +41,10 @@
 	unit-tests/t0207-page-parse-line-endings.scm \
 	unit-tests/t0208-page-parse-embed-no-complex.scm \
 	unit-tests/t0300-attribute.scm \
-	unit-tests/t0301-promotable-attributes.scm \
 	unit-tests/t0400-os.scm \
 	unit-tests/t0401-os-expand-env-variables.scm \
-	unit-tests/t0402-config.scm \
 	unit-tests/t1000-deprecated.scm
 
-XFAIL_TESTS = \
-	unit-tests/t0301-promotable-attributes.scm
-
 dist_noinst_DATA = geda/core/gettext.scm.in unit-test.scm $(TESTS)
 
 geda/core/gettext.scm: $(srcdir)/geda/core/gettext.scm.in Makefile
