Index: powerman-2.3.26/scripts/powerman.service
===================================================================
--- powerman-2.3.26.orig/scripts/powerman.service	2018-12-29 00:03:25.000000000 +0100
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,14 +0,0 @@
-[Unit]
-Description=PowerMan
-After=syslog.target network.target
-
-[Service]
-Type=forking
-PrivateTmp=yes
-User=daemon
-Group=daemon
-ExecStart=/usr/sbin/powermand
-PIDFile=/var/run/powerman/powermand.pid
-
-[Install]
-WantedBy=multi-user.target
Index: powerman-2.3.26/scripts/powerman.service.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ powerman-2.3.26/scripts/powerman.service.in	2021-05-13 09:07:58.387942345 +0200
@@ -0,0 +1,16 @@
+[Unit]
+Description=PowerMan
+After=syslog.target network.target
+
+[Service]
+Type=forking
+PrivateTmp=yes
+User=@RUN_AS_USER@
+Group=@RUN_AS_GROUP@
+ExecStart=/usr/sbin/powermand
+RuntimeDirectory=powerman
+RuntimeDirectoryMode=0755
+PIDFile=/run/powerman/powermand.pid
+
+[Install]
+WantedBy=multi-user.target
Index: powerman-2.3.26/configure.ac
===================================================================
--- powerman-2.3.26.orig/configure.ac	2020-02-18 20:56:01.000000000 +0100
+++ powerman-2.3.26/configure.ac	2021-05-13 09:07:58.387942345 +0200
@@ -90,7 +90,7 @@ AC_DEFINE(WITH_LSD_NOMEM_ERROR_FUNC, 1,
 # whether to install pkg-config file for API
 AC_PKGCONFIG
 
-# what user to run daemon as
+# what user and group to run daemon as
 AC_RUNAS
 
 ##
@@ -111,6 +111,7 @@ AC_CONFIG_FILES( \
   etc/Makefile \
   scripts/Makefile \
   scripts/powerman \
+  scripts/powerman.service \
   scripts/tmpfiles.d/powerman.conf \
   heartbeat/Makefile \
   man/Makefile \
Index: powerman-2.3.26/config/ac_runas.m4
===================================================================
--- powerman-2.3.26.orig/config/ac_runas.m4	2018-12-29 00:02:57.000000000 +0100
+++ powerman-2.3.26/config/ac_runas.m4	2021-05-13 09:07:58.387942345 +0200
@@ -1,6 +1,7 @@
 AC_DEFUN([AC_RUNAS],
 [
     RUN_AS_USER="daemon"
+    RUN_AS_GROUP="daemon"
     AC_MSG_CHECKING(user to run as)
     AC_ARG_WITH(user,
     AC_HELP_STRING([--with-user=username], [user for powerman daemon (daemon)]),
@@ -16,4 +17,20 @@ AC_DEFUN([AC_RUNAS],
             [Powerman daemon user])
     AC_MSG_RESULT(${RUN_AS_USER})
     AC_SUBST(RUN_AS_USER)
+
+    AC_MSG_CHECKING(group to run as)
+    AC_ARG_WITH(group,
+    AC_HELP_STRING([--with-group=groupname], [group for powerman daemon (daemon)]),
+    [       case "${withval}" in
+            yes|no)
+                    ;;
+            *)
+                    RUN_AS_GROUP="${withval}"
+                    ;;
+            esac],
+    )
+    AC_DEFINE_UNQUOTED(RUN_AS_GROUP, "${RUN_AS_GROUP}",
+            [Powerman daemon group])
+    AC_MSG_RESULT(${RUN_AS_GROUP})
+    AC_SUBST(RUN_AS_GROUP)
 ])
Index: powerman-2.3.26/config/systemd.m4
===================================================================
--- powerman-2.3.26.orig/config/systemd.m4	2020-02-18 20:56:01.000000000 +0100
+++ powerman-2.3.26/config/systemd.m4	2021-05-13 09:27:58.838707064 +0200
@@ -33,7 +33,7 @@ AC_DEFUN([RRA_WITH_SYSTEMD_UNITDIR],
     [AS_HELP_STRING([--with-systemdsystemunitdir=DIR],
         [Directory for systemd service files])],
     [],
-    [with_systemdsystemunitdir=\${prefix}$($PKG_CONFIG --variable=systemdsystemunitdir systemd)])
+    [with_systemdsystemunitdir=$($PKG_CONFIG --variable=systemdsystemunitdir systemd)])
  AS_IF([test x"$with_systemdsystemunitdir" != xno],
     [AC_SUBST([systemdsystemunitdir], [$with_systemdsystemunitdir])])
  AM_CONDITIONAL([HAVE_SYSTEMD],
