--- Makefile.orig	2009-08-23 00:01:10.000000000 +0200
+++ Makefile	2010-03-13 02:45:09.000000000 +0100
@@ -35,8 +35,8 @@
 	${PYTHON} ./setup.py install ; \
 	;; \
 	*) \
-	echo ${PYTHON} ./setup.py install --root=${DESTDIR} ; \
-	${PYTHON} ./setup.py install --root=${DESTDIR} ; \
+	echo ${PYTHON} ./setup.py install --root=${DESTDIR} --prefix=/usr ; \
+	${PYTHON} ./setup.py install --root=${DESTDIR} --prefix=/usr ; \
 	;; \
 	esac
 
