#--- tclpkg/mkpkgindex.sh
#+++ tclpkg/mkpkgindex.sh
#@@ -22,4 +22,4 @@
#     echo "	package require Tk 8.3" >>pkgIndex.tcl
#     ;;
# esac
#-echo "	load [file join \$dir $lib] $2\"" >>pkgIndex.tcl
#+echo "	load $lib $2\"" >>pkgIndex.tcl
--- tclpkg/mkpkgindex.py
+++ tclpkg/mkpkgindex.py
@@ -39,7 +39,7 @@
         f.write(f'package ifneeded {options.name} {version} "\n')
         if "tk" in str(options.file):
             f.write("\tpackage require Tk 8.3\n")
-        f.write(f'\tload [file join $dir {options.file}] {options.name}"\n')
+        f.write(f'\tload {options.file} {options.name}"\n')
 
 
 if __name__ == "__main__":

