diff --git a/tclpkg/gv/gv.i b/tclpkg/gv/gv.i
index ea65ada2f..0bab8817f 100644
--- a/tclpkg/gv/gv.i
+++ b/tclpkg/gv/gv.i
@@ -10,6 +10,13 @@
 
 %module gv
 
+#ifdef SWIGPHP
+%pragma(php) code="/*
+This is a deprecated interface.
+Load graphviz via PHP's C API in php.ini instead.
+*/"
+#endif
+
 #ifdef SWIGTCL
 // A typemap telling SWIG to ignore an argument for input
 // However, we still need to pass a pointer to the C function
