Index: apache-tomcat-11.0.5-src/java/org/apache/coyote/ajp/AbstractAjpProtocol.java
===================================================================
--- apache-tomcat-11.0.5-src.orig/java/org/apache/coyote/ajp/AbstractAjpProtocol.java
+++ apache-tomcat-11.0.5-src/java/org/apache/coyote/ajp/AbstractAjpProtocol.java
@@ -155,7 +155,7 @@ public abstract class AbstractAjpProtoco
     }
 
 
-    private boolean secretRequired = true;
+    private boolean secretRequired = false;
 
     public void setSecretRequired(boolean secretRequired) {
         this.secretRequired = secretRequired;
