--- x0rfbserver/XUpdateScanner.h
+++ x0rfbserver/XUpdateScanner.h
@@ -75,4 +75,4 @@
 
 } // namespace rfb
 
-#endif _hexonet_rfb_XUpdateScanner_h_
+#endif // _hexonet_rfb_XUpdateScanner_h_
--- xrfbviewer/xrfbviewer.cc
+++ xrfbviewer/xrfbviewer.cc
@@ -780,7 +780,7 @@
     connection->send( (unsigned char*) &zlib, 1 );
     connection->send( (unsigned char*) &level, 1 );
   }
-  #endif USE_ZLIB_WARREN
+  #endif //USE_ZLIB_WARREN
 
 
 
