From 5d7d16a4e417c9483d8ff529835498f336c6cc37 Mon Sep 17 00:00:00 2001
From: Andrew Leech <andrew@alelec.net>
Date: Mon, 23 Apr 2018 10:12:16 +1000
Subject: [PATCH] Fix python2 support

---
 pywebdav/server/fshandler.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/pywebdav/server/fshandler.py b/pywebdav/server/fshandler.py
index 84bb333..7b9b0a6 100644
--- a/pywebdav/server/fshandler.py
+++ b/pywebdav/server/fshandler.py
@@ -1,6 +1,5 @@
 from __future__ import absolute_import
 import os
-import html
 import textwrap
 import six
 import logging
@@ -12,6 +11,10 @@
 from pywebdav.lib.errors import *
 from pywebdav.lib.iface import *
 from pywebdav.lib.davcmd import copyone, copytree, moveone, movetree, delone, deltree
+if six.PY2:
+    from cgi import escape
+else:
+    from html import escape
 
 log = logging.getLogger(__name__)
 
@@ -142,7 +145,7 @@ def _get_listing(self, path):
                 </body>
             </html>
             """)
-        escapeditems = (html.escape(i) + ('/' if os.path.isdir(os.path.join(path, i)) else '') for i in os.listdir(path) if not i.startswith('.'))
+        escapeditems = (escape(i) + ('/' if os.path.isdir(os.path.join(path, i)) else '') for i in os.listdir(path) if not i.startswith('.'))
         htmlitems = "\n".join('<li><a href="{i}">{i}</a></li>'.format(i=i) for i in escapeditems)
 
         return template.format(items=htmlitems, path=path)
