Index: bluez-5.71/Makefile.am
===================================================================
--- bluez-5.71.orig/Makefile.am
+++ bluez-5.71/Makefile.am
@@ -679,7 +679,7 @@ unit_test_bass_SOURCES = unit/test-bass.
 unit_test_bass_LDADD = src/libshared-glib.la \
 				lib/libbluetooth-internal.la $(GLIB_LIBS)
 
-unit_tests += unit/test-vcp
+# unit_tests += unit/test-vcp
 
 unit_test_vcp_SOURCES = unit/test-vcp.c $(btio_sources)
 unit_test_vcp_LDADD = src/libshared-glib.la \
@@ -696,7 +696,7 @@ unit_test_midi_LDADD = src/libshared-gli
 endif
 
 if MESH
-unit_tests += unit/test-mesh-crypto
+#unit_tests += unit/test-mesh-crypto
 unit_test_mesh_crypto_CPPFLAGS = $(ell_cflags)
 unit_test_mesh_crypto_SOURCES = unit/test-mesh-crypto.c \
 				mesh/crypto.h ell/internal ell/ell.h
