diff -Nru AusweisApp2-1.20.0.orig/resources/packaging/linux/com.governikus.ausweisapp2.desktop.in AusweisApp2-1.20.0/resources/packaging/linux/com.governikus.ausweisapp2.desktop.in
--- AusweisApp2-1.20.0.orig/resources/packaging/linux/com.governikus.ausweisapp2.desktop.in	2020-01-15 16:59:29.000000000 +0100
+++ AusweisApp2-1.20.0/resources/packaging/linux/com.governikus.ausweisapp2.desktop.in	2020-01-15 22:21:45.043831241 +0100
@@ -2,7 +2,7 @@
 Version=1.0
 Type=Application
 Exec=@CMAKE_INSTALL_FULL_BINDIR@/AusweisApp2
-Icon=@CMAKE_INSTALL_FULL_DATADIR@/@VENDOR@/AusweisApp2/npa.svg
+Icon=AusweisApp2
 StartupNotify=true
 Terminal=false
 Categories=Utility;Accessibility;
