diff -Nru AusweisApp2-1.18.2.orig/src/secure_storage/SecureStorage.cpp AusweisApp2-1.18.2/src/secure_storage/SecureStorage.cpp
--- AusweisApp2-1.18.2.orig/src/secure_storage/SecureStorage.cpp	2019-09-30 17:25:27.000000000 +0200
+++ AusweisApp2-1.18.2/src/secure_storage/SecureStorage.cpp	2019-10-18 23:37:10.684045625 +0200
@@ -95,7 +95,7 @@
 
 void SecureStorage::load()
 {
-	const auto& path = FileDestination::getPath(QStringLiteral("config.json"));
+	const auto& path = QStringLiteral("/usr/share/AusweisApp2/config.json");
 
 	if (!QFile::exists(path))
 	{
