--- ./createrepo/__init__.py.orig	2014-02-24 12:31:40.702205468 +0000
+++ ./createrepo/__init__.py	2014-02-24 12:31:54.565205444 +0000
@@ -82,7 +82,7 @@ class MetaDataConfig(object):
         self.max_delta_rpm_size = 100000000
         self.update_md_path = None
         self.skip_stat = False
-        self.database = True
+        self.database = False
         self.outputdir = None
         self.file_patterns = ['.*bin\/.*', '^\/etc\/.*', '^\/usr\/lib\/sendmail$']
         self.dir_patterns = ['.*bin\/.*', '^\/etc\/.*']
--- ./genpkgmetadata.py.orig	2014-02-24 12:26:37.150206006 +0000
+++ ./genpkgmetadata.py	2014-02-24 12:28:29.876205806 +0000
@@ -71,8 +71,8 @@ def parse_args(args, conf):
     parser.add_option("-C", "--checkts", default=False, action="store_true",
         help="check timestamps on files vs the metadata to see " \
            "if we need to update")
-    parser.add_option("-d", "--database", default=True, action="store_true",
-        help="create sqlite database files: now default, see --no-database to disable")
+    parser.add_option("-d", "--database", default=False, action="store_true",
+        help="create sqlite database files")
     parser.add_option("--no-database", default=False, dest="nodatabase", action="store_true",
         help="do not create sqlite dbs of metadata")
     # temporarily disabled
--- ./mergerepo.py.orig	2014-02-24 12:28:35.926205795 +0000
+++ ./mergerepo.py	2014-02-24 12:30:05.740205636 +0000
@@ -40,7 +40,7 @@ def parse_args(args):
                       help="repo url")
     parser.add_option("-a", "--archlist", default=[], action="append",
                       help="Defaults to all arches - otherwise specify arches")
-    parser.add_option("-d", "--database", default=True, action="store_true")
+    parser.add_option("-d", "--database", default=False, action="store_true")
     parser.add_option( "--no-database", default=False, action="store_true", dest="nodatabase")
     parser.add_option("-o", "--outputdir", default=None,
                       help="Location to create the repository")
@@ -75,6 +75,8 @@ def main(args):
         rmbase.archlist = opts.archlist
     if opts.outputdir:
         rmbase.outputdir = opts.outputdir
+    if opts.database:
+        rmbase.mdconf.database = True
     if opts.nodatabase:
         rmbase.mdconf.database = False
     if opts.nogroups:
