From 337e1a70b1b9a70046f7afde6a59205293d1dac4 Mon Sep 17 00:00:00 2001
From: Javier Llorente <javier@opensuse.org>
Date: Tue, 5 Oct 2021 23:36:26 +0200
Subject: [PATCH] Fix build for Qt versions < 5.14.0


diff --git a/src/gui/monitor/monitorrepositorytab.cpp b/src/gui/monitor/monitorrepositorytab.cpp
index 415d1b2..c8fb962 100644
--- a/src/gui/monitor/monitorrepositorytab.cpp
+++ b/src/gui/monitor/monitorrepositorytab.cpp
@@ -90,8 +90,14 @@ void MonitorRepositoryTab::slotAddResultList(const QList<OBSResult *> &resultLis
 void MonitorRepositoryTab::checkForResultListChanges(const QList<OBSResult *> &resultList)
 {
     // FIXME: intersect always returns 0
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
     QSet<OBSResult *> old_resultSet(m_resultList.begin(), m_resultList.end());
     QSet<OBSResult *> new_resultSet(resultList.begin(), resultList.end());
+#else
+    QSet<OBSResult *> old_resultSet = m_resultList.toSet();
+    QSet<OBSResult *> new_resultSet= resultList.toSet();
+#endif
+
     QList<OBSResult *> commonResults = old_resultSet.intersect(new_resultSet).values();
 
     for (OBSResult *oldResult : commonResults) {
