diff -ruN a/files/usr/share/cinnamon/cinnamon-settings/bin/SettingsWidgets.py b/files/usr/share/cinnamon/cinnamon-settings/bin/SettingsWidgets.py
--- a/files/usr/share/cinnamon/cinnamon-settings/bin/SettingsWidgets.py	2021-01-13 21:22:39.000000000 +0700
+++ b/files/usr/share/cinnamon/cinnamon-settings/bin/SettingsWidgets.py	2021-02-10 02:14:29.516667849 +0700
@@ -213,7 +213,7 @@
             self.module.loaded = True
 
         if self.is_standalone:
-            subprocess.Popen(self.exec_name.split())
+            subprocess.Popen(self.exec_name.split(' ',2))
             return
 
         # Add our own widgets
