Index: Catch2-2.13.10/projects/SelfTest/UsageTests/Message.tests.cpp
===================================================================
--- Catch2-2.13.10.orig/projects/SelfTest/UsageTests/Message.tests.cpp
+++ Catch2-2.13.10/projects/SelfTest/UsageTests/Message.tests.cpp
@@ -242,7 +242,7 @@ std::ostream& operator<<(std::ostream& o
 }
 
 // clang can handle GCC's diagnostic pragma
-#if defined( __GNUG__ ) || defined(__clang__)
+#if defined( __GNUG__ ) && __GNUG__ >= 8 || defined(__clang__)
 #    pragma GCC diagnostic push
 #endif
 // Clang and gcc have different names for this warning, and clang also
@@ -263,7 +263,7 @@ TEST_CASE("CAPTURE can deal with complex
     CAPTURE( (1, 2), (2, 3) );
     SUCCEED();
 }
-#if defined( __GNUG__ ) || defined(__clang__)
+#if defined( __GNUG__ ) && __GNUG__ >= 8 || defined(__clang__)
 #    pragma GCC diagnostic pop
 #endif
 
