--- src/libpocketsphinx/hmm.h.orig	2012-12-12 14:43:56.000000000 -0700
+++ src/libpocketsphinx/hmm.h	2016-07-15 09:16:22.510397933 -0600
@@ -65,12 +65,12 @@ extern "C" {
  * process longer utterances. Due to limitations of FSG search implementation
  * this value needs to be signed.
  */
-typedef int16 frame_idx_t;
+typedef int32 frame_idx_t;
 
 /**
  * Maximum number of frames in index, should be in sync with above.
  */
-#define MAX_N_FRAMES MAX_INT16
+#define MAX_N_FRAMES MAX_INT32
 
 
 /** Shift count for senone scores. */
