diff -Nru google-auth-2.0.1.orig/tests/test__cloud_sdk.py google-auth-2.0.1/tests/test__cloud_sdk.py
--- google-auth-2.0.1.orig/tests/test__cloud_sdk.py	2021-08-19 21:22:12.000000000 +0200
+++ google-auth-2.0.1/tests/test__cloud_sdk.py	2021-08-20 11:09:31.644103701 +0200
@@ -73,7 +73,7 @@
 
 def test__run_subprocess_ignore_stderr():
     command = [
-        "python",
+        "python3",
         "-c",
         "from __future__ import print_function;"
         + "import sys;"
