diff -Nru google-auth-2.26.1.orig/tests/compute_engine/test_credentials.py google-auth-2.26.1/tests/compute_engine/test_credentials.py
--- google-auth-2.26.1.orig/tests/compute_engine/test_credentials.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/compute_engine/test_credentials.py	2024-01-04 10:40:25.973375772 +0100
@@ -14,7 +14,7 @@
 import base64
 import datetime
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 import responses  # type: ignore
 
diff -Nru google-auth-2.26.1.orig/tests/compute_engine/test__metadata.py google-auth-2.26.1/tests/compute_engine/test__metadata.py
--- google-auth-2.26.1.orig/tests/compute_engine/test__metadata.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/compute_engine/test__metadata.py	2024-01-04 10:40:25.976709143 +0100
@@ -18,7 +18,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _helpers
diff -Nru google-auth-2.26.1.orig/tests/conftest.py google-auth-2.26.1/tests/conftest.py
--- google-auth-2.26.1.orig/tests/conftest.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/conftest.py	2024-01-04 10:40:25.873374649 +0100
@@ -15,7 +15,7 @@
 import os
 import sys
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 
diff -Nru google-auth-2.26.1.orig/tests/crypt/test__python_rsa.py google-auth-2.26.1/tests/crypt/test__python_rsa.py
--- google-auth-2.26.1.orig/tests/crypt/test__python_rsa.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/crypt/test__python_rsa.py	2024-01-04 10:40:26.013376222 +0100
@@ -16,7 +16,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 from pyasn1_modules import pem  # type: ignore
 import pytest  # type: ignore
 import rsa  # type: ignore
diff -Nru google-auth-2.26.1.orig/tests/oauth2/test_challenges.py google-auth-2.26.1/tests/oauth2/test_challenges.py
--- google-auth-2.26.1.orig/tests/oauth2/test_challenges.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/oauth2/test_challenges.py	2024-01-04 10:40:25.953375548 +0100
@@ -17,7 +17,7 @@
 import base64
 import sys
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 import pyu2f  # type: ignore
 
diff -Nru google-auth-2.26.1.orig/tests/oauth2/test__client.py google-auth-2.26.1/tests/oauth2/test__client.py
--- google-auth-2.26.1.orig/tests/oauth2/test__client.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/oauth2/test__client.py	2024-01-04 10:40:25.940042064 +0100
@@ -18,7 +18,7 @@
 import os
 import urllib
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _helpers
diff -Nru google-auth-2.26.1.orig/tests/oauth2/test_credentials.py google-auth-2.26.1/tests/oauth2/test_credentials.py
--- google-auth-2.26.1.orig/tests/oauth2/test_credentials.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/oauth2/test_credentials.py	2024-01-04 10:40:25.946708806 +0100
@@ -18,7 +18,7 @@
 import pickle
 import sys
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _helpers
diff -Nru google-auth-2.26.1.orig/tests/oauth2/test_gdch_credentials.py google-auth-2.26.1/tests/oauth2/test_gdch_credentials.py
--- google-auth-2.26.1.orig/tests/oauth2/test_gdch_credentials.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/oauth2/test_gdch_credentials.py	2024-01-04 10:40:25.930041952 +0100
@@ -17,7 +17,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 import requests
 
diff -Nru google-auth-2.26.1.orig/tests/oauth2/test_id_token.py google-auth-2.26.1/tests/oauth2/test_id_token.py
--- google-auth-2.26.1.orig/tests/oauth2/test_id_token.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/oauth2/test_id_token.py	2024-01-04 10:40:25.936708693 +0100
@@ -15,7 +15,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import environment_vars
diff -Nru google-auth-2.26.1.orig/tests/oauth2/test_reauth.py google-auth-2.26.1/tests/oauth2/test_reauth.py
--- google-auth-2.26.1.orig/tests/oauth2/test_reauth.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/oauth2/test_reauth.py	2024-01-04 10:40:25.950042176 +0100
@@ -14,7 +14,7 @@
 
 import copy
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import exceptions
diff -Nru google-auth-2.26.1.orig/tests/oauth2/test_service_account.py google-auth-2.26.1/tests/oauth2/test_service_account.py
--- google-auth-2.26.1.orig/tests/oauth2/test_service_account.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/oauth2/test_service_account.py	2024-01-04 10:40:25.943375435 +0100
@@ -16,7 +16,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _helpers
diff -Nru google-auth-2.26.1.orig/tests/oauth2/test_sts.py google-auth-2.26.1/tests/oauth2/test_sts.py
--- google-auth-2.26.1.orig/tests/oauth2/test_sts.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/oauth2/test_sts.py	2024-01-04 10:40:25.956708918 +0100
@@ -16,7 +16,7 @@
 import json
 import urllib
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import exceptions
diff -Nru google-auth-2.26.1.orig/tests/test_app_engine.py google-auth-2.26.1/tests/test_app_engine.py
--- google-auth-2.26.1.orig/tests/test_app_engine.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test_app_engine.py	2024-01-04 10:40:25.906708356 +0100
@@ -14,7 +14,7 @@
 
 import datetime
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import app_engine
diff -Nru google-auth-2.26.1.orig/tests/test_aws.py google-auth-2.26.1/tests/test_aws.py
--- google-auth-2.26.1.orig/tests/test_aws.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test_aws.py	2024-01-04 10:40:26.040043188 +0100
@@ -18,7 +18,7 @@
 import os
 import urllib.parse
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _helpers
diff -Nru google-auth-2.26.1.orig/tests/test__cloud_sdk.py google-auth-2.26.1/tests/test__cloud_sdk.py
--- google-auth-2.26.1.orig/tests/test__cloud_sdk.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test__cloud_sdk.py	2024-01-04 10:40:25.920041839 +0100
@@ -18,7 +18,7 @@
 import subprocess
 import sys
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _cloud_sdk
diff -Nru google-auth-2.26.1.orig/tests/test_credentials.py google-auth-2.26.1/tests/test_credentials.py
--- google-auth-2.26.1.orig/tests/test_credentials.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test_credentials.py	2024-01-04 10:40:25.960042289 +0100
@@ -14,7 +14,7 @@
 
 import datetime
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _helpers
diff -Nru google-auth-2.26.1.orig/tests/test__default.py google-auth-2.26.1/tests/test__default.py
--- google-auth-2.26.1.orig/tests/test__default.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test__default.py	2024-01-04 10:40:25.963375660 +0100
@@ -15,7 +15,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _default
diff -Nru google-auth-2.26.1.orig/tests/test_downscoped.py google-auth-2.26.1/tests/test_downscoped.py
--- google-auth-2.26.1.orig/tests/test_downscoped.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test_downscoped.py	2024-01-04 10:40:25.966709030 +0100
@@ -17,7 +17,7 @@
 import json
 import urllib
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _helpers
diff -Nru google-auth-2.26.1.orig/tests/test__exponential_backoff.py google-auth-2.26.1/tests/test__exponential_backoff.py
--- google-auth-2.26.1.orig/tests/test__exponential_backoff.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test__exponential_backoff.py	2024-01-04 10:40:25.863374536 +0100
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import mock
+from unittest import mock
 
 from google.auth import _exponential_backoff
 
diff -Nru google-auth-2.26.1.orig/tests/test_external_account_authorized_user.py google-auth-2.26.1/tests/test_external_account_authorized_user.py
--- google-auth-2.26.1.orig/tests/test_external_account_authorized_user.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test_external_account_authorized_user.py	2024-01-04 10:40:25.963375660 +0100
@@ -16,7 +16,7 @@
 import http.client as http_client
 import json
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import exceptions
diff -Nru google-auth-2.26.1.orig/tests/test_external_account.py google-auth-2.26.1/tests/test_external_account.py
--- google-auth-2.26.1.orig/tests/test_external_account.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test_external_account.py	2024-01-04 10:40:25.866707907 +0100
@@ -17,7 +17,7 @@
 import json
 import urllib
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _helpers
diff -Nru google-auth-2.26.1.orig/tests/test_iam.py google-auth-2.26.1/tests/test_iam.py
--- google-auth-2.26.1.orig/tests/test_iam.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test_iam.py	2024-01-04 10:40:25.983375885 +0100
@@ -17,7 +17,7 @@
 import http.client as http_client
 import json
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _helpers
diff -Nru google-auth-2.26.1.orig/tests/test_identity_pool.py google-auth-2.26.1/tests/test_identity_pool.py
--- google-auth-2.26.1.orig/tests/test_identity_pool.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test_identity_pool.py	2024-01-04 10:40:25.990042626 +0100
@@ -18,7 +18,7 @@
 import os
 import urllib
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _helpers
diff -Nru google-auth-2.26.1.orig/tests/test_impersonated_credentials.py google-auth-2.26.1/tests/test_impersonated_credentials.py
--- google-auth-2.26.1.orig/tests/test_impersonated_credentials.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test_impersonated_credentials.py	2024-01-04 10:40:26.030043076 +0100
@@ -17,7 +17,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _helpers
diff -Nru google-auth-2.26.1.orig/tests/test_jwt.py google-auth-2.26.1/tests/test_jwt.py
--- google-auth-2.26.1.orig/tests/test_jwt.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test_jwt.py	2024-01-04 10:40:25.993375997 +0100
@@ -17,7 +17,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _helpers
diff -Nru google-auth-2.26.1.orig/tests/test_metrics.py google-auth-2.26.1/tests/test_metrics.py
--- google-auth-2.26.1.orig/tests/test_metrics.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test_metrics.py	2024-01-04 10:40:25.926708581 +0100
@@ -14,7 +14,7 @@
 
 import platform
 
-import mock
+from unittest import mock
 
 from google.auth import metrics
 from google.auth import version
diff -Nru google-auth-2.26.1.orig/tests/test__oauth2client.py google-auth-2.26.1/tests/test__oauth2client.py
--- google-auth-2.26.1.orig/tests/test__oauth2client.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test__oauth2client.py	2024-01-04 10:40:25.910041727 +0100
@@ -17,7 +17,7 @@
 import os
 import sys
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 try:
diff -Nru google-auth-2.26.1.orig/tests/test_pluggable.py google-auth-2.26.1/tests/test_pluggable.py
--- google-auth-2.26.1.orig/tests/test_pluggable.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test_pluggable.py	2024-01-04 10:40:26.033376446 +0100
@@ -16,7 +16,7 @@
 import os
 import subprocess
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import exceptions
diff -Nru google-auth-2.26.1.orig/tests/test__refresh_worker.py google-auth-2.26.1/tests/test__refresh_worker.py
--- google-auth-2.26.1.orig/tests/test__refresh_worker.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/test__refresh_worker.py	2024-01-04 10:40:26.016709592 +0100
@@ -17,7 +17,7 @@
 import threading
 import time
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _refresh_worker, credentials, exceptions
diff -Nru google-auth-2.26.1.orig/tests/transport/test__custom_tls_signer.py google-auth-2.26.1/tests/transport/test__custom_tls_signer.py
--- google-auth-2.26.1.orig/tests/transport/test__custom_tls_signer.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/transport/test__custom_tls_signer.py	2024-01-04 10:40:25.880041390 +0100
@@ -15,7 +15,7 @@
 import ctypes
 import os
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 from requests.packages.urllib3.util.ssl_ import create_urllib3_context  # type: ignore
 import urllib3.contrib.pyopenssl  # type: ignore
diff -Nru google-auth-2.26.1.orig/tests/transport/test_grpc.py google-auth-2.26.1/tests/transport/test_grpc.py
--- google-auth-2.26.1.orig/tests/transport/test_grpc.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/transport/test_grpc.py	2024-01-04 10:40:25.883374761 +0100
@@ -16,7 +16,7 @@
 import os
 import time
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import _helpers
diff -Nru google-auth-2.26.1.orig/tests/transport/test__mtls_helper.py google-auth-2.26.1/tests/transport/test__mtls_helper.py
--- google-auth-2.26.1.orig/tests/transport/test__mtls_helper.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/transport/test__mtls_helper.py	2024-01-04 10:40:25.890041502 +0100
@@ -15,7 +15,7 @@
 import os
 import re
 
-import mock
+from unittest import mock
 from OpenSSL import crypto
 import pytest  # type: ignore
 
diff -Nru google-auth-2.26.1.orig/tests/transport/test_mtls.py google-auth-2.26.1/tests/transport/test_mtls.py
--- google-auth-2.26.1.orig/tests/transport/test_mtls.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/transport/test_mtls.py	2024-01-04 10:40:25.893374873 +0100
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import mock
+from unittest import mock
 import pytest  # type: ignore
 
 from google.auth import exceptions
diff -Nru google-auth-2.26.1.orig/tests/transport/test_requests.py google-auth-2.26.1/tests/transport/test_requests.py
--- google-auth-2.26.1.orig/tests/transport/test_requests.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/transport/test_requests.py	2024-01-04 10:40:25.896708244 +0100
@@ -19,7 +19,7 @@
 import sys
 
 import freezegun
-import mock
+from unittest import mock
 import OpenSSL
 import pytest  # type: ignore
 import requests
diff -Nru google-auth-2.26.1.orig/tests/transport/test_urllib3.py google-auth-2.26.1/tests/transport/test_urllib3.py
--- google-auth-2.26.1.orig/tests/transport/test_urllib3.py	2024-01-03 23:49:22.000000000 +0100
+++ google-auth-2.26.1/tests/transport/test_urllib3.py	2024-01-04 10:40:25.876708019 +0100
@@ -16,7 +16,7 @@
 import os
 import sys
 
-import mock
+from unittest import mock
 import OpenSSL
 import pytest  # type: ignore
 import urllib3  # type: ignore
