Index: CSXCAD-0.6.0/README
===================================================================
--- CSXCAD-0.6.0.orig/README
+++ CSXCAD-0.6.0/README
@@ -11,8 +11,11 @@ Forum:     http://openems.de/forum/
 Tutorials: http://openems.de/index.php/Tutorials
 
 
-To use CSXCAD from Matlab or Octave, you need to include the <CSXCAD-install-folder>/matlab folder in the respective environment:
-> addpath( '<CSXCAD-install-folder>/matlab' );
+To use CSXCAD from Octave, you need to load CSXCAD package:
+> pkg load csxcad;
+
+To use CSXCAD from Matlab, you need to include the </usr/share/CSXCAD/matlab folder in the respective environment:
+> addpath( '/usr/share/CSXCAD/matlab' );
 
 To verify the correct installation follow the instructions at:
 http://openems.de/index.php/Tutorial:_First_Steps
