From 2b5f416924309f0e0e2e719c57cf07a8b4570e2f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Klaus=20K=C3=A4mpf?= <kkaempf@suse.de>
Date: Thu, 23 Jun 2022 15:48:41 +0200
Subject: [PATCH 3/3] drop duplicate definition of _FORTIFY_SOURCE
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Klaus Kämpf <kkaempf@suse.de>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 8db336e..1dfdd00 100644
--- a/configure.ac
+++ b/configure.ac
@@ -26,7 +26,7 @@ CC_WARNINGS="\
 -Wcast-align \
 -Wno-unused-value"
 
-CFLAGS="$CFLAGS $CC_WARNINGS -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE"
+CFLAGS="$CFLAGS $CC_WARNINGS -D_GNU_SOURCE"
 
 AC_CONFIG_HEADER([config.h])
 
-- 
2.36.1

