Index: uvicorn-0.22.0/tests/protocols/test_websocket.py
===================================================================
--- uvicorn-0.22.0.orig/tests/protocols/test_websocket.py
+++ uvicorn-0.22.0/tests/protocols/test_websocket.py
@@ -584,7 +584,7 @@ async def test_asgi_return_value(
     async with run_server(config):
         with pytest.raises(websockets.exceptions.ConnectionClosed) as exc_info:
             await connect(f"ws://127.0.0.1:{unused_tcp_port}")
-        assert exc_info.value.code == 1006
+        assert exc_info.value.code in {1000, 1006}
 
 
 @pytest.mark.anyio
