Index: blender-5.0.0/intern/cycles/bvh/octree.cpp
===================================================================
--- blender-5.0.0.orig/intern/cycles/bvh/octree.cpp
+++ blender-5.0.0/intern/cycles/bvh/octree.cpp
@@ -16,6 +16,8 @@
 #  include <openvdb/tools/FindActiveValues.h>
 #endif
 
+#include <fstream>
+
 CCL_NAMESPACE_BEGIN
 
 __forceinline int Octree::flatten_index(int x, int y, int z) const
