Index: django-redis-4.12.1/tests/test_backend.py
===================================================================
--- django-redis-4.12.1.orig/tests/test_backend.py
+++ django-redis-4.12.1/tests/test_backend.py
@@ -288,7 +288,7 @@ class DjangoRedisCacheTests(unittest.Tes
 
         res = cache.get("add_key")
         self.assertEqual(res, "Initial value")
-
+        cache.delete("other_key")
         res = self.cache.add("other_key", "New value")
         self.assertIs(res, True)
 
