--- a/build.gradle
+++ b/build.gradle
@@ -18,29 +18,8 @@
  */
 buildscript {
     repositories {
+        xmvn()
         jcenter()
-        maven {
-            name 'Bintray Asciidoctor repo'
-            url 'http://dl.bintray.com/content/aalmiray/asciidoctor'
-        }
-        maven{
-            name 'Bintray Javadoc Hotfix repo'
-            url 'http://dl.bintray.com/melix/gradle-javadoc-hotfix-plugin'
-        }
-        maven {
-            url "https://plugins.gradle.org/m2/"
-        }
-    }
-
-    dependencies {
-        classpath 'org.asciidoctor:asciidoctor-gradle-plugin:1.5.7'
-        classpath 'org.jfrog.buildinfo:build-info-extractor-gradle:3.0.3'
-        classpath 'me.champeau.gradle:gradle-javadoc-hotfix-plugin:0.1'
-        classpath 'me.champeau.gradle:japicmp-gradle-plugin:0.1.1'
-        //classpath 'nl.javadude.gradle.plugins:license-gradle-plugin:0.11.0'
-        classpath "gradle.plugin.org.nosphere.apache:creadur-rat-gradle:0.1.3"
-        classpath 'com.gradle:build-scan-plugin:1.8'
-        classpath 'me.champeau.gradle:buildscan-recipes-plugin:0.2.0'
     }
 }
 
@@ -64,15 +45,13 @@ allprojects {
     group = 'org.codehaus.groovy'
     version = groovyVersion
     repositories {
+        xmvn()
         jcenter()
         maven { url 'http://dl.bintray.com/melix/thirdparty-apache' } // openbeans
     }
 
     apply plugin: 'groovy'
     apply from: "${rootProject.projectDir}/gradle/indy.gradle"
-    if (JavaVersion.current().java7Compatible) {
-        apply from: "${rootProject.projectDir}/gradle/asciidoctor.gradle"
-    }
 }
 
 // todo: use the conventional "resources" directory for classpath resources
@@ -110,6 +89,7 @@ subprojects {
 }
 
 repositories {
+    xmvn()
     // todo Some repos are needed only for some configs. Declare them just for the configuration once Gradle allows this.
     maven { url 'http://repository.jboss.org/nexus/content/groups/m2-release-proxy' } // examples, tools
 }
@@ -468,7 +468,6 @@
 apply from: 'gradle/idea.gradle'
 apply from: 'gradle/eclipse.gradle'
 apply from: 'gradle/codehaus.gradle'
-apply from: 'gradle/quality.gradle'
 
 // If a local configuration file for tweaking the build is present, apply it
 if (file('user.gradle').exists()) {
--- a/gradle/codehaus.gradle
+++ b/gradle/codehaus.gradle
@@ -22,6 +22,7 @@ import groovy.io.*
 
 buildscript {
     repositories {
+        xmvn()
         mavenCentral()
     }
     dependencies {
--- a/gradle/utils.gradle
+++ b/gradle/utils.gradle
@@ -32,6 +32,7 @@ import static org.objectweb.asm.Opcodes.V1_5
 
 buildscript {
     repositories {
+        xmvn()
         mavenCentral()
     }
     dependencies {
