--- a/build.gradle
+++ b/build.gradle
@@ -46,7 +46,6 @@
 apply from: 'gradle/filter.gradle'
 apply from: 'gradle/indy.gradle'
 apply from: 'gradle/publish.gradle'
-apply plugin: 'javadocHotfix'
 
 File javaHome = new File(System.getProperty('java.home'))
 logger.lifecycle "Using Java from $javaHome (version ${System.getProperty('java.version')})"
@@ -186,11 +185,7 @@
     testCompile "com.thoughtworks.qdox:qdox:$qdoxVersion"
 
     tools "com.googlecode.jarjar:jarjar:$jarjarVersion"
-    tools("net.sourceforge.cobertura:cobertura:$coberturaVersion") {
-        exclude(module: 'asm')
-        exclude(module: 'asm')
-        exclude(module: 'ant')
-    }
+    tools "org.ow2.asm:asm-commons:$asmVersion"
     tools "org.ow2.asm:asm:$asmVersion"
     tools "com.thoughtworks.qdox:qdox:$qdoxVersion"
 
@@ -447,9 +441,6 @@
 apply from: 'gradle/docs.gradle'
 apply from: 'gradle/assemble.gradle'
 apply from: 'gradle/upload.gradle'
-apply from: 'gradle/idea.gradle'
-apply from: 'gradle/eclipse.gradle'
-apply from: 'gradle/codehaus.gradle'
 
 // If a local configuration file for tweaking the build is present, apply it
 if (file('user.gradle').exists()) {
--- a/gradle/docs.gradle
+++ b/gradle/docs.gradle
@@ -17,24 +17,10 @@
  *  under the License.
  */
 task doc(dependsOn: ['javadocAll', 'groovydocAll', 'docGDK']) {
-    if (JavaVersion.current().java7Compatible) {
-        dependsOn 'asciidocAll', 'assembleAsciidoc'
-    }
     ext.footer = 'Copyright &copy; 2003-2018 The Apache Software Foundation. All rights reserved.'
     ext.title = "Groovy ${groovyVersion}"
 }
 
-if (JavaVersion.current().java7Compatible) {
-    task assembleAsciidoc(type: Copy, dependsOn: 'asciidocAll') {
-        subprojects {
-            from project.asciidoctor
-        }
-        into "$buildDir/asciidoc"
-    }
-
-    task asciidocAll(dependsOn: 'asciidoctor')
-}
-
 def javadocSpec = {
     maxMemory = javaDoc_mx
     project.configure(options) {
@@ -190,11 +176,6 @@
     }
 }
 
-// apply javadoc fix tool to generated javadocs - ensures no vulnerability even on old JDK versions
-allprojects {
-    apply plugin: 'javadocHotfix'
-}
-
 if (JavaVersion.current().isJava7Compatible()) {
     javadocAll.options.source = '1.7'
 }
--- a/gradle/upload.gradle
+++ b/gradle/upload.gradle
@@ -94,15 +94,12 @@
 
         artifacts {
             archives jar
-            archives sourceJar
-            archives javadocJar
-            archives groovydocJar
         }
 
         [uploadArchives, install]*.with {
             // dependency on jarAllAll should in theory be replaced with jar, jarWithIndy but
             // in practice, it is faster
-            dependsOn([jarAllAll, sourceJar, javadocJar, groovydocJar])
+        dependsOn([jarAllAll])
             doFirst {
                 if (rootProject.useIndy()) {
                     new GradleException('You cannot use uploadArchives or install task with the flag [indy] turned'
@@ -124,9 +121,6 @@
                 // gradle doesn't expect us to mutate configurations like we do here
                 // so signing the configuration won't work and we do it manually here
                 signArchiveTask(jar)
-                signArchiveTask(sourceJar)
-                signArchiveTask(javadocJar)
-                signArchiveTask(groovydocJar)
                 def indyJar = rootProject.ext.deriveFile(jar.archivePath, 'indy')
                 if (indyJar.exists()) {
                     signWithClassifier('indy', indyJar)
