Index: spice-gtk-0.18/data/org.spice-space.lowlevelusbaccess.policy
===================================================================
--- spice-gtk-0.18.orig/data/org.spice-space.lowlevelusbaccess.policy
+++ spice-gtk-0.18/data/org.spice-space.lowlevelusbaccess.policy
@@ -12,8 +12,9 @@
     <description>Low level USB device access</description>
     <message>Privileges are required for low level USB device access (for usb device pass through).</message>
     <defaults>
+      <allow_any>auth_admin</allow_any>
       <allow_inactive>no</allow_inactive>
-      <allow_active>yes</allow_active>
+      <allow_active>auth_admin</allow_active>
     </defaults>
   </action>
 
