diff --git a/test/elf/relocatable.sh b/test/elf/relocatable.sh
index 0065479..c5fe661 100755
--- a/test/elf/relocatable.sh
+++ b/test/elf/relocatable.sh
@@ -14,7 +14,7 @@ t=out/test/elf/$testname
 mkdir -p $t
 
 # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98667
-[ $MACHINE = i386 ] && { echo skipped; exit; }
+[ $MACHINE = i386 -o $MACHINE = i686 ] && { echo skipped; exit; }
 
 cat <<EOF | $CXX -c -o $t/a.o -xc++ -
 int one() { return 1; }
