From cd1ee0e172a10a3f8a420ff20aa1a69d46c7b4ad Mon Sep 17 00:00:00 2001
From: Kirill Primak <vyivel@eclair.cafe>
Date: Sun, 23 Jan 2022 18:16:44 +0300
Subject: [PATCH] swaynag: remove buffer destruction condition

An address of a variable can never be NULL, so checking it doesn't make
sense; and `destroy_buffer()` can operate on already destroyed buffers
anyway.

Fixes #6780
---
 swaynag/swaynag.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/swaynag/swaynag.c b/swaynag/swaynag.c
index 9b57d57856..03ec0411b7 100644
--- a/swaynag/swaynag.c
+++ b/swaynag/swaynag.c
@@ -511,13 +511,8 @@ void swaynag_destroy(struct swaynag *swaynag) {
 		swaynag_seat_destroy(seat);
 	}
 
-	if (&swaynag->buffers[0]) {
-		destroy_buffer(&swaynag->buffers[0]);
-	}
-
-	if (&swaynag->buffers[1]) {
-		destroy_buffer(&swaynag->buffers[1]);
-	}
+	destroy_buffer(&swaynag->buffers[0]);
+	destroy_buffer(&swaynag->buffers[1]);
 
 	if (swaynag->outputs.prev || swaynag->outputs.next) {
 		struct swaynag_output *output, *temp;
