Index: contrib/SpamAssassin/Collectd.pm
===================================================================
--- a/contrib/SpamAssassin/Collectd.pm.orig
+++ b/contrib/SpamAssassin/Collectd.pm
@@ -4,12 +4,6 @@
 
 Collectd - plugin for filling collectd with stats 
 
-=head1 INSTALLATION
-
-Just copy Collectd.pm into your SpamAssassin Plugin path 
-(e.g /usr/share/perl5/Mail/SpamAssassin/Plugin/) and
-add a loadplugin call into your init.pre file. 
-
 =head1 SYNOPSIS
 
   loadplugin    Mail::SpamAssassin::Plugin::Collectd
@@ -47,6 +41,8 @@ often it should try.
 
 =cut
 
+=back
+
 =head1 DESCRIPTION
 
 This modules uses the email plugin of collectd from Sebastian Harl to
