From 3ac9f4e421e4471b08875f8e82c8d367fa9dc243 Mon Sep 17 00:00:00 2001
From: Julius Kriukas <julius@kriukas.lt>
Date: Wed, 28 Mar 2018 21:29:56 +0300
Subject: [PATCH] Mark db_reconnect as static inline, fixes GH-11

---
 src/addrwatch_mysql.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/addrwatch_mysql.c b/src/addrwatch_mysql.c
index 389153f..b4ed760 100644
--- a/src/addrwatch_mysql.c
+++ b/src/addrwatch_mysql.c
@@ -232,7 +232,7 @@ void db_disconnect(struct ctx_s *data)
 	data->dbh = NULL;
 }
 
-inline void db_reconnect(struct ctx_s *data) {
+static inline void db_reconnect(struct ctx_s *data) {
 	while (1) {
 		if (data->dbh)
 			db_disconnect(data);
