diff --git a/README.md b/README.md
index 376461e..4213730 100644
--- a/README.md
+++ b/README.md
@@ -41,7 +41,6 @@ The following files are installed under each module directory.  They are shown h
 MODULE.cpp (source code)
 MODULE.h (header)
 MODULE.ui (Qt Designer file)
-icons.qrc (resource file for icons in 'kmdevices' and 'kminfo')
 ui_MODULE.h (Qt designer header file - for hand coding purposes)
 CMakeLists.txt (build instructions for each module)
 
diff --git a/devices/CMakeLists.txt b/devices/CMakeLists.txt
index c1d3158..fc281cd 100644
--- a/devices/CMakeLists.txt
+++ b/devices/CMakeLists.txt
@@ -3,9 +3,6 @@ INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
 SET(kmdevices_PART_SRCS
       kmdevices.cpp
       )
-SET(kmdevices_RCCS
-      icons.qrc
-      )
 
 QT5_ADD_RESOURCES(kcm_kmdevices_RCC_SRCS ${kmdevices_RCCS})
 ADD_LIBRARY( kcm_kmdevices MODULE ${kmdevices_PART_SRCS} ${kmdevices_RCC_SRCS})
diff --git a/devices/icons.qrc b/devices/icons.qrc
deleted file mode 100644
index 6a40e9e..0000000
--- a/devices/icons.qrc
+++ /dev/null
@@ -1,7 +0,0 @@
-<RCC>
-  <qresource prefix="/resources" >
-    <file alias="printer.png">resources/printer.png</file>
-    <file alias="monitor.png">resources/monitor.png</file>
-    <file alias="scanner.png">resources/scanner.png</file>
-  </qresource>
-</RCC>
diff --git a/information/CMakeLists.txt b/information/CMakeLists.txt
index 837ef6b..1f98fbd 100644
--- a/information/CMakeLists.txt
+++ b/information/CMakeLists.txt
@@ -1,9 +1,6 @@
 set(kminfo_PART_SRCS
       kminfo.cpp
       )
-SET(kminfo_RCCS
-      icons.qrc
-      )
 
 QT5_ADD_RESOURCES(kcm_kminfo_RCC_SRCS ${kminfo_RCCS})
 ADD_LIBRARY( kcm_kminfo MODULE ${kminfo_PART_SRCS} ${kminfo_RCC_SRCS} )
diff --git a/information/icons.qrc b/information/icons.qrc
deleted file mode 100644
index d1f44ab..0000000
--- a/information/icons.qrc
+++ /dev/null
@@ -1,7 +0,0 @@
-<RCC>
-  <qresource prefix="/resources" >
-    <file alias="printer.png">resources/printer1.png</file>
-    <file alias="monitor.png">resources/monitor.png</file>
-    <file alias="scanner.png">resources/scanner.png</file>
-  </qresource>
-</RCC>
diff --git a/settings/CMakeLists.txt b/settings/CMakeLists.txt
index 6f05a41..4880adf 100644
--- a/settings/CMakeLists.txt
+++ b/settings/CMakeLists.txt
@@ -3,9 +3,6 @@ INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
 SET(kmsettings_PART_SRCS
       kmsettings.cpp
       )
-SET(kmsettings_RCCS
-      icons.qrc
-      )
 
 QT5_ADD_RESOURCES(kcm_kmsettings_RCC_SRCS ${kmsettings_RCCS})
 ADD_LIBRARY( kcm_kmsettings MODULE ${kmsettings_PART_SRCS} )
