Index: wire-0.6.0/go.mod
===================================================================
--- wire-0.6.0.orig/go.mod
+++ wire-0.6.0/go.mod
@@ -1,6 +1,6 @@
 module github.com/google/wire
 
-go 1.12
+go 1.22
 
 require (
 	github.com/google/go-cmp v0.2.0
@@ -8,3 +8,5 @@ require (
 	github.com/pmezard/go-difflib v1.0.0
 	golang.org/x/tools v0.17.0
 )
+
+require golang.org/x/mod v0.14.0 // indirect
