diff --git a/creduce/creduce_config.pm.in b/creduce/creduce_config.pm.in
index 1e71112..4578b30 100644
--- a/creduce/creduce_config.pm.in
+++ b/creduce/creduce_config.pm.in
@@ -32,6 +32,8 @@ use constant {
     # External programs invoked.
     #
     CLANG_FORMAT	=> q{@CLANG_FORMAT@},
+    UNIFDEF => q{@UNIFDEF@},
+    TOPFORMFLAT => q{@TOPFORMFLAT@},
 };
 
 our @EXPORT = qw();
diff --git a/creduce/pass_clang.pm b/creduce/pass_clang.pm
index 3bd8314..3b8085b 100644
--- a/creduce/pass_clang.pm
+++ b/creduce/pass_clang.pm
@@ -35,7 +35,7 @@ sub check_prereqs () {
     if ((defined $abs_bindir) && ($FindBin::RealBin eq $abs_bindir)) {
 	# This script is in the installation directory.
 	# Use the installed `clang_delta'.
-	$path = libexecdir . "/clang_delta";
+	$path = $abs_bindir . "/clang_delta";
     } else {
 	# Assume that this script is in the C-Reduce build tree.
 	# Use the `clang_delta' that is also in the build tree.
diff --git a/creduce/pass_clang_binsrch.pm b/creduce/pass_clang_binsrch.pm
index 7470d51..b079214 100644
--- a/creduce/pass_clang_binsrch.pm
+++ b/creduce/pass_clang_binsrch.pm
@@ -47,7 +47,7 @@ sub check_prereqs () {
     if ((defined $abs_bindir) && ($FindBin::RealBin eq $abs_bindir)) {
 	# This script is in the installation directory.
 	# Use the installed `clang_delta'.
-	$path = libexecdir . "/clang_delta";
+	$path = $abs_bindir . "/clang_delta";
     } else {
 	# Assume that this script is in the C-Reduce build tree.
 	# Use the `clang_delta' that is also in the build tree.
diff --git a/creduce/pass_clex.pm b/creduce/pass_clex.pm
index 66707bf..29ea764 100644
--- a/creduce/pass_clex.pm
+++ b/creduce/pass_clex.pm
@@ -35,7 +35,7 @@ sub check_prereqs () {
     if ((defined $abs_bindir) && ($FindBin::RealBin eq $abs_bindir)) {
 	# This script is in the installation directory.
 	# Use the installed `clex'.
-	$path = libexecdir . "/clex";
+	$path = $abs_bindir . "/clex";
     } else {
 	# Assume that this script is in the C-Reduce build tree.
 	# Use the `clex' that is also in the build tree.
diff --git a/creduce/pass_ifs.pm b/creduce/pass_ifs.pm
index c8811a4..238f977 100644
--- a/creduce/pass_ifs.pm
+++ b/creduce/pass_ifs.pm
@@ -25,28 +25,8 @@ my $unifdef;
 my $options = "-B -x 2";
 
 sub check_prereqs () {
-    my $path;
-    my $abs_bindir = abs_path(bindir);
-    if ((defined $abs_bindir) && ($FindBin::RealBin eq $abs_bindir)) {
-        # This script is in the installation directory.
-        # Use the installed `unifdef'.
-        $path = libexecdir . "/unifdef";
-    } else {
-        # Assume that this script is in the C-Reduce build tree.
-        # Use the `unifdef' that is also in the build tree.
-        $path = "$FindBin::Bin/../unifdef/unifdef";
-    }
-    if ((-e $path) && (-x $path)) {
-        $unifdef = $path;
-        return 1;
-    }
-    # Check Windows
-    $path = $path . ".exe";
-    if (($^O eq "MSWin32") && (-e $path) && (-x $path)) {
-        $unifdef = $path;
-        return 1;
-    }
-    return 0;
+	$unifdef = find_external_program(creduce_config::UNIFDEF, "unifdef");
+  return defined ($unifdef);
 }
 
 sub count_ifs($$) {
diff --git a/creduce/pass_lines.pm b/creduce/pass_lines.pm
index 5166453..affa241 100644
--- a/creduce/pass_lines.pm
+++ b/creduce/pass_lines.pm
@@ -24,28 +24,8 @@ use creduce_utils;
 my $topformflat;
 
 sub check_prereqs () {
-    my $path;
-    my $abs_bindir = abs_path(bindir);
-    if ((defined $abs_bindir) && ($FindBin::RealBin eq $abs_bindir)) {
-	# This script is in the installation directory.
-	# Use the installed `topformflat'.
-	$path = libexecdir . "/topformflat";
-    } else {
-	# Assume that this script is in the C-Reduce build tree.
-	# Use the `topformflat' that is also in the build tree.
-	$path = "$FindBin::Bin/../delta/topformflat";
-    }
-    if ((-e $path) && (-x $path)) {
-	$topformflat = $path;
-	return 1;
-    }
-    # Check Windows
-    $path = $path . ".exe";
-    if (($^O eq "MSWin32") && (-e $path) && (-x $path)) {
-	$topformflat = $path;
-	return 1;
-    }
-    return 0;
+	$topformflat = find_external_program(creduce_config::TOPFORMFLAT, "topformflat");
+  return defined ($topformflat);
 }
 
 # unlike the previous version of pass_lines, this one always
diff --git a/creduce/pass_unifdef.pm b/creduce/pass_unifdef.pm
index 34c6d52..d0df854 100644
--- a/creduce/pass_unifdef.pm
+++ b/creduce/pass_unifdef.pm
@@ -24,28 +24,8 @@ my $unifdef;
 my $options = "-B -x 2";
 
 sub check_prereqs () {
-    my $path;
-    my $abs_bindir = abs_path(bindir);
-    if ((defined $abs_bindir) && ($FindBin::RealBin eq $abs_bindir)) {
-	# This script is in the installation directory.
-	# Use the installed `unifdef'.
-	$path = libexecdir . "/unifdef";
-    } else {
-	# Assume that this script is in the C-Reduce build tree.
-	# Use the `unifdef' that is also in the build tree.
-	$path = "$FindBin::Bin/../unifdef/unifdef";
-    }
-    if ((-e $path) && (-x $path)) {
-	$unifdef = $path;
-	return 1;
-    }
-    # Check Windows
-    $path = $path . ".exe";
-    if (($^O eq "MSWin32") && (-e $path) && (-x $path)) {
-	$unifdef = $path;
-	return 1;
-    }
-    return 0;
+	$unifdef = find_external_program(creduce_config::UNIFDEF, "unifdef");
+  return defined ($unifdef);
 }
 
 sub new ($$) {
