--- org.eclipse.ecf-62a51c1a2635c5e17a17c0d8fc46c1f006e24184/protocols/bundles/ch.ethz.iks.r_osgi.remote/META-INF/MANIFEST.MF	2018-07-30 22:51:30.000000000 +0200
+++ org.eclipse.ecf-62a51c1a2635c5e17a17c0d8fc46c1f006e24184/protocols/bundles/ch.ethz.iks.r_osgi.remote/META-INF/MANIFEST.MF	2019-11-26 20:21:36.722413163 +0100
@@ -7,7 +7,7 @@
 Bundle-Vendor: %plugin.provider
 Import-Package: org.eclipse.ecf.remoteservice;version="7.3.0",
  org.eclipse.ecf.remoteservice.asyncproxy;version="[1.0.0,3.0.0)",
- org.objectweb.asm;version="[5.0.1,7.0.0)",
+ org.objectweb.asm;version="[5.0.1,8.0.0)",
  org.osgi.framework;version="[1.0.0,2.0.0)",
  org.osgi.service.event;version="[1.0.0,2.0.0)",
  org.osgi.service.log;version="[1.0.0,2.0.0)",
