Index: xsnow-3.8.1/src/Makefile.am
===================================================================
--- xsnow-3.8.1.orig/src/Makefile.am
+++ xsnow-3.8.1/src/Makefile.am
@@ -23,7 +23,7 @@ SUBDIRS = Pixmaps
 TOCC     = $(top_srcdir)/src/tocc.sh
 TOASCII  = $(top_srcdir)/src/toascii.sh
 
-gamesdir = $(exec_prefix)/games
+gamesdir = $(exec_prefix)/bin
 games_PROGRAMS = xsnow
 
 noinst_PROGRAMS = makeman
Index: xsnow-3.8.1/src/Makefile.in
===================================================================
--- xsnow-3.8.1.orig/src/Makefile.in
+++ xsnow-3.8.1/src/Makefile.in
@@ -668,7 +668,7 @@ top_srcdir = @top_srcdir@
 SUBDIRS = Pixmaps
 TOCC = $(top_srcdir)/src/tocc.sh
 TOASCII = $(top_srcdir)/src/toascii.sh
-gamesdir = $(exec_prefix)/games
+gamesdir = $(exec_prefix)/bin
 xsnow_CPPFLAGS = $(GTK_CFLAGS) $(X11_CFLAGS) $(GSL_CFLAGS) \
 	-DLOCALEDIR=\"$(LOCALEDIR)\" -DLANGUAGES='"$(LANGUAGES)"' \
 	$(am__append_1) $(am__append_2)
