Index: knot-1.6.8/configure.ac
===================================================================
--- knot-1.6.8.orig/configure.ac	2017-06-05 10:55:45.049766182 +0200
+++ knot-1.6.8/configure.ac	2017-06-05 10:56:44.258809504 +0200
@@ -2,8 +2,8 @@
 
 AC_PREREQ([2.60])
 AC_INIT([knot], [1.6.8], [knot-dns@labs.nic.cz])
-AM_INIT_AUTOMAKE([gnits subdir-objects dist-xz -Wall -Werror])
-AM_SILENT_RULES([yes])
+AM_INIT_AUTOMAKE([gnits subdir-objects -Wall -Werror])
+#AM_SILENT_RULES([yes])
 AC_CONFIG_SRCDIR([src/knot/main.c])
 AC_CONFIG_HEADERS([src/config.h])
 AC_CONFIG_MACRO_DIR([m4])
