diff -Naur dovecot-2.4.1-4/src/auth/mech-gssapi.c dovecot-2.4.1-4.new/src/auth/mech-gssapi.c
--- dovecot-2.4.1-4/src/auth/mech-gssapi.c	2025-03-28 12:32:27.000000000 +0100
+++ dovecot-2.4.1-4.new/src/auth/mech-gssapi.c	2025-05-23 14:09:17.765742203 +0200
@@ -672,7 +672,7 @@
 
 	if (data_size == 0) {
 		/* The client should go first */
-		auth_request_handler_reply_continue(request, NULL, 0);
+		auth_request_handler_reply_continue(request, uchar_empty_ptr, 0);
 	} else {
 		mech_gssapi_auth_continue(request, data, data_size);
 	}
