Index: djangorestframework_simplejwt-5.5.1/tests/test_tokens.py
===================================================================
--- djangorestframework_simplejwt-5.5.1.orig/tests/test_tokens.py
+++ djangorestframework_simplejwt-5.5.1/tests/test_tokens.py
@@ -5,7 +5,7 @@ from unittest.mock import patch
 from django.contrib.auth import get_user_model
 from django.test import TestCase
 from freezegun import freeze_time
-from jose import jwt
+import jwt
 
 from rest_framework_simplejwt.exceptions import (
     ExpiredTokenError,
