Index: htdig/ExternalParser.cc
===================================================================
--- htdig/ExternalParser.cc.orig
+++ htdig/ExternalParser.cc
@@ -428,7 +428,7 @@
 		  {
 			metadatetags = new StringMatch();
 			metadatetags->IgnoreCase();
-			metadatetags->Pattern("date|dc.date|dc.date.created|dc.data.modified");
+			metadatetags->Pattern("date|dc.date|dc.date.created|dc.date.modified");
 		  }
     
 		  // <URL:http://www.w3.org/MarkUp/html-spec/html-spec_5.html#SEC5.2.5> 
