---
 lgi/callable.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

Index: lgi-0.9.2/lgi/callable.c
===================================================================
--- lgi-0.9.2.orig/lgi/callable.c	2017-10-09 20:55:55.000000000 +0200
+++ lgi-0.9.2/lgi/callable.c	2025-10-29 12:16:31.099371249 +0100
@@ -1233,7 +1233,10 @@
     }
   else
     {
-#if LUA_VERSION_NUM >= 502
+#if LUA_VERSION_NUM >= 504
+      int nresults;
+      res = lua_resume (L, NULL, npos, &nresults);
+#elif LUA_VERSION_NUM >= 502
       res = lua_resume (L, NULL, npos);
 #else
       res = lua_resume (L, npos);
