Index: argh-0.26.2/setup.py
===================================================================
--- argh-0.26.2.orig/setup.py	2016-05-11 21:05:03.000000000 +0200
+++ argh-0.26.2/setup.py	2022-03-10 11:06:42.410811689 +0100
@@ -82,7 +82,7 @@ setup(
     install_requires = install_requires,
 
     # testing
-    tests_require = ['pytest', 'mock', 'iocapture'],
+    tests_require = ['pytest', 'iocapture'],
     cmdclass = {'test': PyTest},
 
     # copyright
Index: argh-0.26.2/test/test_assembling.py
===================================================================
--- argh-0.26.2.orig/test/test_assembling.py	2014-10-26 22:04:34.000000000 +0100
+++ argh-0.26.2/test/test_assembling.py	2022-03-10 11:06:42.410811689 +0100
@@ -4,7 +4,7 @@ Unit Tests For Assembling Phase
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 """
 import sys
-import mock
+import unittest.mock as mock
 import pytest
 
 import argh
Index: argh-0.26.2/test/test_dispatching.py
===================================================================
--- argh-0.26.2.orig/test/test_dispatching.py	2015-10-13 00:25:40.000000000 +0200
+++ argh-0.26.2/test/test_dispatching.py	2022-03-10 11:06:42.410811689 +0100
@@ -4,7 +4,7 @@ Dispatching tests
 ~~~~~~~~~~~~~~~~~
 """
 import argh
-from mock import Mock, patch
+from unittest.mock import Mock, patch
 import pytest
 
 from .base import make_IO
Index: argh-0.26.2/test/test_integration.py
===================================================================
--- argh-0.26.2.orig/test/test_integration.py	2022-03-10 11:06:42.398811617 +0100
+++ argh-0.26.2/test/test_integration.py	2022-03-10 11:06:42.410811689 +0100
@@ -8,7 +8,7 @@ import re
 import argparse
 
 import iocapture
-import mock
+import unittest.mock as mock
 import pytest
 
 import argh
Index: argh-0.26.2/test/test_interaction.py
===================================================================
--- argh-0.26.2.orig/test/test_interaction.py	2014-01-06 07:20:25.000000000 +0100
+++ argh-0.26.2/test/test_interaction.py	2022-03-10 11:07:05.850951267 +0100
@@ -4,7 +4,7 @@ Interaction Tests
 ~~~~~~~~~~~~~~~~~
 """
 import sys
-import mock
+import unittest.mock as mock
 
 import argh
 
