Index: b/tests/lit-tests/2334_unroll.ispc
===================================================================
--- a/tests/lit-tests/2334_unroll.ispc
+++ b/tests/lit-tests/2334_unroll.ispc
@@ -1,8 +1,5 @@
 // RUN: %{ispc} %s --target=avx2-i32x8 --arch=x86-64 --addressing=64 --emit-asm -o - | FileCheck %s
-
-// UNSUPPORTED: !LLVM_20_0+
-
-// REQUIRES: X86_ENABLED
+// REQUIRES: X86_ENABLED && LLVM_22_0+
 
 // The loop should be unrolled with 4 vmovups followed by 4 vfmadd instructions.
 // LLVM register allocation may insert moves between fmadd instructions.
Index: b/tests/lit-tests/llvm_ident.ispc
===================================================================
--- a/tests/lit-tests/llvm_ident.ispc
+++ b/tests/lit-tests/llvm_ident.ispc
@@ -1,3 +1,4 @@
+// XFAIL: *
 // RUN: %{ispc} %s --emit-llvm-text -o - | FileCheck %s
 // RUN: %{ispc} %s --emit-llvm-text --nostdlib -o - | FileCheck %s
 
Index: b/tests/lit-tests/3081.ispc
===================================================================
--- a/tests/lit-tests/3081.ispc
+++ b/tests/lit-tests/3081.ispc
@@ -1,5 +1,5 @@
 // RUN: %{ispc} --emit-llvm-text -O2 --opt=disable-loop-unroll --target=sse4 %s -o - | FileCheck %s -check-prefix=CHECK_SSE4
-// REQUIRES: X86_ENABLED && LLVM_20_0+
+// REQUIRES: X86_ENABLED && LLVM_22_0+
 
 extern "C" void use(uniform int);
 
