Index: quagga-1.2.4/redhat/zebra.service
===================================================================
--- quagga-1.2.4.orig/redhat/zebra.service
+++ quagga-1.2.4/redhat/zebra.service
@@ -6,6 +6,17 @@ ConditionPathExists=/etc/quagga/zebra.co
 Documentation=man:zebra
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+ProtectHostname=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 Type=forking
 PIDFile=/run/quagga/zebra.pid
 EnvironmentFile=-/etc/sysconfig/quagga
