--- a/share/CMakeLists.txt
+++ b/share/CMakeLists.txt
@@ -35,6 +35,15 @@ install(FILES ${GRAMMAR_FILES}
 	PARENT_SCOPE
 )
 
+set(LINPHONE_CFLAGS ${LINPHONE_CPPFLAGS})
+set(LINPHONE_LIBS ${LINPHONE_LDFLAGS})
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${prefix}/bin)
+set(libdir ${CMAKE_INSTALL_LIBDIR})
+set(includedir ${prefix}/include)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/linphone.pc.in ${CMAKE_CURRENT_BINARY_DIR}/linphone.pc)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/linphone.pc DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+
 if(ENABLE_ASSETS)
 	set(SOUND_FILES
 		"${CMAKE_CURRENT_SOURCE_DIR}/hello16000.wav"
--- a/share/linphone.pc.in
+++ b/share/linphone.pc.in
@@ -3,7 +3,7 @@ exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
 
-Requires: mediastreamer ortp bctoolbox
+Requires: mediastreamer belle-sip belcard
 Name: liblinphone
 Description: All in one linphone libs.
 Version: @VERSION@
