Index: liblinphone-5.2.88/CMakeLists.txt
===================================================================
--- liblinphone-5.2.88.orig/CMakeLists.txt
+++ liblinphone-5.2.88/CMakeLists.txt
@@ -174,12 +174,7 @@ find_package(XML2 REQUIRED)
 
 if (ENABLE_DB_STORAGE)
 	if(NOT DISABLE_SOCI_PACKAGE_SEARCH)
-		#APPLE platform does not use dlopen for soci backend
-		if (APPLE OR ANDROID)
-			find_package(Soci REQUIRED COMPONENTS sqlite3)
-		else()
-			find_package(Soci REQUIRED)
-		endif()
+		find_package(Soci REQUIRED COMPONENTS sqlite3)
 	else()
 		set(SOCI_FOUND YES)
 		set(SOCI_LIBRARIES soci_core)
Index: liblinphone-5.2.88/src/CMakeLists.txt
===================================================================
--- liblinphone-5.2.88.orig/src/CMakeLists.txt
+++ liblinphone-5.2.88/src/CMakeLists.txt
@@ -70,11 +70,8 @@ if(ZLIB_FOUND)
 endif()
 if (ENABLE_DB_STORAGE)
 	if (SOCI_FOUND)
-		#take care of the order  for static link
-		if(APPLE OR ANDROID)
-			list(APPEND INTERFACE_LIBS ${SOCI_sqlite3_PLUGIN})
-			list(APPEND LINK_LIBS ${SOCI_sqlite3_PLUGIN})
-		endif()
+		list(APPEND INTERFACE_LIBS ${SOCI_sqlite3_PLUGIN})
+		list(APPEND LINK_LIBS ${SOCI_sqlite3_PLUGIN})
 		list(APPEND INTERFACE_LIBS soci)
 		list(APPEND LINK_LIBS ${SOCI_LIBRARIES})
 	endif()
