Index: arpack-ng-3.7.0/TESTS/bug_79_double_complex.f
===================================================================
--- arpack-ng-3.7.0.orig/TESTS/bug_79_double_complex.f
+++ arpack-ng-3.7.0/TESTS/bug_79_double_complex.f
@@ -217,7 +217,7 @@ c           %---------------------------
             go to 9000
          end if
  9000 continue
-      if (res1 .ne. res2) then
+      if (abs(res1 - res2) > 0.0D+0) then
          stop 1
       end if
       end
