Index: dkimpy-1.1.5/dkim/__init__.py
===================================================================
--- dkimpy-1.1.5.orig/dkim/__init__.py
+++ dkimpy-1.1.5/dkim/__init__.py
@@ -43,18 +43,9 @@ import binascii
 # Set to False to not use async functions even though aiodns is installed.
 USE_ASYNC = True
 
-# only needed for arc
-try:
-    import authres
-except ImportError:
-    pass
-
-# only needed for ed25519 signing/verification
-try:
-    import nacl.signing
-    import nacl.encoding
-except ImportError:
-    pass
+import authres
+import nacl.signing
+import nacl.encoding
 
 from dkim.canonicalization import (
     CanonicalizationPolicy,
@@ -1181,7 +1172,7 @@ class ARC(DomainSigner):
     if chain_validation_status == CV_Fail:
       self.headers.reverse()
     if b'h' in as_fields:
-        raise ValidationError("h= tag not permitted in ARC-Seal header field")    
+        raise ValidationError("h= tag not permitted in ARC-Seal header field")
     res = self.gen_header(as_fields, as_include_headers, canon_policy,
                            b"ARC-Seal", pk, standardize)
 
