Index: dkimpy-1.0.3/dkim/__init__.py
===================================================================
--- dkimpy-1.0.3.orig/dkim/__init__.py
+++ dkimpy-1.0.3/dkim/__init__.py
@@ -40,18 +40,9 @@ import sys
 import time
 import binascii
 
-# only needed for arc
-try:
-    from authres import AuthenticationResultsHeader
-except ImportError:
-    pass
-
-# only needed for ed25519 signing/verification
-try:
-    import nacl.signing
-    import nacl.encoding
-except ImportError:
-    pass
+from authres import AuthenticationResultsHeader
+import nacl.signing
+import nacl.encoding
 
 from dkim.canonicalization import (
     CanonicalizationPolicy,
