--- Makefile.am
+++ Makefile.am
@@ -26,13 +26,13 @@
 
 # Print out an informative summary.
 all-local:
-	@$(ECHO) "Done."
-	@$(ECHO)
+	@echo "Done."
+	@echo
 	@if test "x$(MAKECMDGOALS)" = "xall-am" -o "x$(.TARGETS)" = "xall-am" -o "x$(MAKECMDGOALS)" = "x" -o "x$(.TARGETS)" = "x" ; then \
-	  $(ECHO) "---" ;\
-	  $(ECHO) "Run 'make install' to begin installation into $(prefix)" ;\
+	  echo "---" ;\
+	  echo "Run 'make install' to begin installation into $(prefix)" ;\
 	fi
-	@$(ECHO)
+	@echo
 
 # Upload documentation
 DOC = docs/html docs/latex/ftgl.pdf
