From 4230bb3dd48d21e48a0812e497aa405d20a3ade9 Mon Sep 17 00:00:00 2001
From: Mingming Cao <mmc@linux.vnet.ibm.com>
Date: Thu, 24 Feb 2022 09:02:05 -0800
Subject: [PATCH] Fix NM HNV setting primary slave

Signed-off-by: Mingming Cao <mmc@linux.vnet.ibm.com>
---
 scripts/hcnmgr | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/hcnmgr b/scripts/hcnmgr
index ce34179..6526b08 100644
--- a/scripts/hcnmgr
+++ b/scripts/hcnmgr
@@ -375,7 +375,7 @@ do_config_vdevice_nm() {
 	# if the device is primary, and link is up, force it as primary se
 	if [[ $MODE == "primary" ]]; then
 		hcnlog INFO "Change bonding primary slave to $DEVNAME"
-		echo "$DEVNAME" >"$BOND_PATH"/active_slave
+		echo "$DEVNAME" >"$BOND_PATH"/primary
 	fi
 
 	hcnlog DEBUG "do_config_vdevice: exit"
-- 
2.35.1

