From faa28462dee0fa9f212b5ea81893c6cc7fb569c7 Mon Sep 17 00:00:00 2001
From: Marius Tomaschewski <mt@suse.de>
Date: Mon, 7 Feb 2022 18:47:55 +0100
Subject: [PATCH 26/27] hcnmgr: move sourcing functions.suse to wicked case

---
 scripts/hcnmgr | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/scripts/hcnmgr b/scripts/hcnmgr
index c7bcd5e..839e882 100644
--- a/scripts/hcnmgr
+++ b/scripts/hcnmgr
@@ -44,7 +44,6 @@ VIO_TYPE=""
 DISTRO=""
 SERVICE=""
 
-source /usr/lib/powerpc-utils/functions.suse
 
 # Usage statements
 usage() {
@@ -953,6 +952,9 @@ check_network_service() {
 			SERVICE=$(systemctl show -P Id network.service 2>/dev/null)
 			case $SERVICE in
 				wicked.service)
+					source /usr/lib/powerpc-utils/functions.suse || \
+					err $E_EPERM "Unable to source SUSE function library"
+
 					if ! wicked --version >/dev/null 2>&1; then
 						err $E_ENETUNREACH "wicked management command not installed"
 					fi
-- 
2.34.1

