From 4e582b71282f67d670f47ee4d3da21cb814b8f30 Mon Sep 17 00:00:00 2001
From: Marius Tomaschewski <mt@suse.de>
Date: Tue, 8 Feb 2022 10:06:19 +0100
Subject: [PATCH 27/27] hcnmgr: extend suse os-release match

Extend to use sles|sled|*suse* pattern catching sled,
suse-microos and opensuse-(leap|tumbleweed|microos).
---
 scripts/hcnmgr | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/hcnmgr b/scripts/hcnmgr
index 839e882..d3c51be 100644
--- a/scripts/hcnmgr
+++ b/scripts/hcnmgr
@@ -948,7 +948,7 @@ check_network_service() {
 
 	#Validate distro service CLI packages is installed to manage networking
 	case $DISTRO in
-		sles)
+		sles|sled|*suse*)
 			SERVICE=$(systemctl show -P Id network.service 2>/dev/null)
 			case $SERVICE in
 				wicked.service)
-- 
2.34.1

