Upstream: TBD
References: lost in the mists of time

Index: powerpc-utils-1.3.7/Makefile.am
===================================================================
--- powerpc-utils-1.3.7.orig/Makefile.am
+++ powerpc-utils-1.3.7/Makefile.am
@@ -181,7 +181,7 @@ src_drmgr_lsslot_LDADD = -lrtas
 
 if WITH_SYSTEMD
 install-data-hook: systemd/smt_off.service
-	$(MKDIR_P) @prefix@${systemd_unit_dir}
-	$(INSTALL_SCRIPT) $< @prefix@${systemd_unit_dir}/
-	sed -i -e 's,$${exec_prefix},@sbindir@,g' @prefix@${systemd_unit_dir}/smt_off.service
+	$(MKDIR_P) ${systemd_unit_dir}
+	$(INSTALL_SCRIPT) $< ${systemd_unit_dir}/
+	sed -i -e 's,$${exec_prefix},@sbindir@,g' ${systemd_unit_dir}/smt_off.service
 endif
