--- libgnomevfs/gnome-vfs-ssl.c.orig
+++ libgnomevfs/gnome-vfs-ssl.c
@@ -364,7 +364,14 @@ gnome_vfs_ssl_create_from_fd (GnomeVFSSS
 	if (ssl_ctx == NULL) {
 		return GNOME_VFS_ERROR_INTERNAL;
 	}
+    /* Never negotiate SSLv2 as it is broken from the security POV. */
+    SSL_CTX_set_options(ssl_ctx, SSL_OP_ALL | SSL_OP_NO_SSLv2);
 
+    /* Make openssl to release internal buffers when no longer needed
+     * to keep memory usage low */
+#ifdef SSL_MODE_RELEASE_BUFFERS
+    SSL_CTX_set_mode(ssl_ctx, SSL_MODE_RELEASE_BUFFERS);
+#endif
         /* FIXME: SSL_CTX_set_verify (ssl_ctx, SSL_VERIFY_PEER, &ssl_verify);*/
         ssl->private->ssl = SSL_new (ssl_ctx);
 
