Index: scripts/bash-based-configure
===================================================================
--- scripts/bash-based-configure.orig	2015-12-07 22:06:50.000000000 +0200
+++ scripts/bash-based-configure	2017-01-06 08:33:06.361854198 +0200
@@ -1,3 +1,4 @@
+#!/bin/bash
 #  Bash based configure - library of test functions
 #  Copyright (C) 2004-2016 Carsten Gnörlich
 #
@@ -313,13 +314,13 @@ function REQUIRE_GMAKE()
 
    echo -n "Checking for gmake: "
 
-   if (gmake -v | grep "GNU Make") > /dev/null 2>&1 ;
+   if gmake -v > /dev/null ;
      then echo "yes"
 	  echo "MAKE = `which gmake`" >>Makefile.config
 	  return 0
    fi;
 
-   if (make -v | grep "GNU Make") > /dev/null 2>&1 ;
+   if make -v > /dev/null ;
      then echo "yes"
 	  echo "MAKE = `which make`" >>Makefile.config
 	  return 0
