Index: php-8.0.0/ext/date/config0.m4
===================================================================
--- php-8.0.0.orig/ext/date/config0.m4	2021-01-07 10:56:05.542685455 +0100
+++ php-8.0.0/ext/date/config0.m4	2021-01-07 10:56:10.130714392 +0100
@@ -23,6 +23,8 @@ timelib_sources="lib/astro.c lib/dow.c l
 
 PHP_NEW_EXTENSION(date, php_date.c $timelib_sources, no,, $PHP_DATE_CFLAGS)
 
+PHP_ADD_MAKEFILE_FRAGMENT()
+
 PHP_ADD_BUILD_DIR([$ext_builddir/lib], 1)
 PHP_ADD_INCLUDE([$ext_builddir/lib])
 PHP_ADD_INCLUDE([$ext_srcdir/lib])
Index: php-8.0.0/ext/date/Makefile.frag
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ php-8.0.0/ext/date/Makefile.frag	2021-01-07 10:56:10.134714417 +0100
@@ -0,0 +1,6 @@
+$(srcdir)/lib/parse_date.c: $(srcdir)/lib/parse_date.re
+	@(cd $(top_srcdir); $(RE2C) $(RE2C_FLAGS) -b --no-generation-date -o ext/date/lib/parse_date.c ext/date/lib/parse_date.re)
+
+$(srcdir)/lib/parse_iso_intervals.c: $(srcdir)/lib/parse_iso_intervals.re
+	@(cd $(top_srcdir); $(RE2C) $(RE2C_FLAGS) -b --no-generation-date -o ext/date/lib/parse_iso_intervals.c    ext/date/lib/parse_iso_intervals.re)
+
