diff -Nur deepin-reader-5.10.23/3rdparty/deepin-pdfium/src/src.pro deepin-reader-5.10.23-new/3rdparty/deepin-pdfium/src/src.pro
--- deepin-reader-5.10.23/3rdparty/deepin-pdfium/src/src.pro	2022-08-02 13:40:01.000000000 +0800
+++ deepin-reader-5.10.23-new/3rdparty/deepin-pdfium/src/src.pro	2022-08-25 21:59:54.426159727 +0800
@@ -30,7 +30,7 @@
 
 INCLUDEPATH += /usr/include/chardet
 
-LIBS += -lchardet
+LIBS += -lchardet -ljpeg -licuuc -lz
 
 public_headers += \
     $$PWD/../include/dpdfglobal.h \
