diff --git a/tests/meson.build b/tests/meson.build
index 5439239..cbb147c 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -7,6 +7,8 @@ libwayland = [
 	dependency('wayland-server'),
 ]
 
+message('Build Machine CPU:' + build_machine.cpu_family())
+
 # Check that each protocol passes through the scanner
 foreach protocol_file : protocol_files
 	protocol_path = join_paths(wayland_protocols_srcdir, protocol_file)
@@ -113,7 +115,12 @@ foreach protocol_file : protocol_files
 			'-Werror' ],
 		install: false,
 	)
-	test(test_name, pedantic_test_executable)
+
+    if test_name == 'test-build-pedantic-unstable_xdg_decoration_xdg_decoration_unstable_v1_xml' and build_machine.cpu_family() == 'ppc64'
+        message('Skip test-build-pedantic-unstable_xdg_decoration_xdg_decoration_unstable_v1_xml on ppc64/ppc64le')
+    else
+        test(test_name, pedantic_test_executable)
+    endif
 
 	# Check that the header
 	if not protocol_file.contains('xdg-foreign-unstable-v1')
