--- a/xmlto.in	2014-06-11 10:16:55.661844817 +0200
+++ b/xmlto.in	2014-06-11 10:18:02.362850985 +0200
@@ -39,6 +39,7 @@
   -o directory    put output in the specified directory instead of
                   the current working directory
   -p postprocopts pass option to postprocessor
+  --xsltopts      use xsltproc command line options
   --extensions    turn on stylesheet extensions for this tool chain
   --noautosize    do not autodetect paper size via locales or paperconf
   --noclean       temp files are not deleted automatically
@@ -282,7 +282,7 @@
 
 # Process any options
 ARGS=$(${GETOPT} \
-	--longoptions=help,version,extensions,searchpath:,skip-validation,stringparam:,noclean,noautosize,noextensions,profile,with-fop,with-dblatex \
+	--longoptions=help,version,extensions,searchpath:,skip-validation,stringparam:,noclean,noautosize,noextensions,profile,with-fop,with-dblatex,xsltopts: \
 	-n xmlto -- x:m:o:p:v "$@")
 [ $? != 0 ] && { usage; exit 1; }
 eval set -- "$ARGS"
@@ -391,6 +391,10 @@
 	PROFILE=1
 	shift
 	;;
+  --xsltopts)
+      XSLTOPTS="$XSLTOPTS $2"
+      shift 2
+      ;;
   --)
 	shift
 	break
