From: Luigi Baldoni <aloisio@gmx.com>
Date: 2018-06-11 14:19:30 +0200
Subject: Add missing include statement for Qt 5.11
Upstream: submitted

Qt 5.11 requires more explicit include statements
---
Index: qnapi-0.2.3/gui/src/forms/subdatawidget.h
===================================================================
--- qnapi-0.2.3.orig/gui/src/forms/subdatawidget.h
+++ qnapi-0.2.3/gui/src/forms/subdatawidget.h
@@ -16,6 +16,7 @@
 #define SUBDATAWIDGET_H
 
 #include <QWidget>
+#include <QIcon>
 #include "ui_subDataWidget.h"
 
 class subDataWidget : public QWidget {
