Index: src/validate.c
===================================================================
--- src/validate.c.orig	2016-06-18 19:00:22.000000000 +0200
+++ src/validate.c	2016-10-15 13:05:26.067464415 +0200
@@ -319,6 +319,10 @@ static DesktopKeyDefinition registered_d
 
   { DESKTOP_BOOLEAN_TYPE,           "DBusActivatable",   FALSE, FALSE, FALSE, handle_dbus_activatable_key },
 
+  /* SuSE specific */
+  { DESKTOP_BOOLEAN_TYPE,      "X-SuSE-translate",  FALSE, FALSE, FALSE, NULL },
+
+
   /* Keys reserved for KDE */
 
   /* since 0.9.4 */
