diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6376b88..62a8991 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,7 +71,7 @@ add_custom_target(uninstall
 ########################################################################
 add_subdirectory(include)
 add_subdirectory(src)
-add_subdirectory(tests)
+#add_subdirectory(tests)
 
 # use space-separation format for the pc file
 STRING(REPLACE ";" " " RTL433_PC_CFLAGS "${RTL433_PC_CFLAGS}")
