Index: tdesktop-4.4.0-full/cmake/external/webrtc/CMakeLists.txt
===================================================================
--- tdesktop-4.4.0-full.orig/cmake/external/webrtc/CMakeLists.txt
+++ tdesktop-4.4.0-full/cmake/external/webrtc/CMakeLists.txt
@@ -94,11 +94,8 @@ INTERFACE
     ${webrtc_libs_list}
     $<LINK_ONLY:desktop-app::external_openssl>
     $<LINK_ONLY:desktop-app::external_jpeg>
-    $<TARGET_FILE:desktop-app::external_jpeg>
     $<LINK_ONLY:desktop-app::external_opus>
-    $<TARGET_FILE:desktop-app::external_opus>
     $<LINK_ONLY:desktop-app::external_vpx>
-    $<TARGET_FILE:desktop-app::external_vpx>
 )
 
 if (WIN32)
