diff -Nur tdesktop-3.7.3-full/cmake/external/webrtc/CMakeLists.txt new/cmake/external/webrtc/CMakeLists.txt
--- tdesktop-3.7.3-full/cmake/external/webrtc/CMakeLists.txt	2022-04-19 14:57:00.000000000 +0200
+++ new/cmake/external/webrtc/CMakeLists.txt	2022-05-02 18:53:40.393350191 +0200
@@ -93,11 +93,8 @@
     ${webrtc_libs_list}
     $<LINK_ONLY:desktop-app::external_openssl>
     $<LINK_ONLY:desktop-app::external_jpeg>
-    $<TARGET_FILE:desktop-app::external_jpeg>
     $<LINK_ONLY:desktop-app::external_opus>
-    $<TARGET_FILE:desktop-app::external_opus>
     $<LINK_ONLY:desktop-app::external_vpx>
-    $<TARGET_FILE:desktop-app::external_vpx>
 )
 
 if (WIN32)
