Index: tdesktop-5.6.3-full/cmake/external/webrtc/CMakeLists.txt
===================================================================
--- tdesktop-5.6.3-full.orig/cmake/external/webrtc/CMakeLists.txt
+++ tdesktop-5.6.3-full/cmake/external/webrtc/CMakeLists.txt
@@ -97,13 +97,10 @@ INTERFACE
     ${webrtc_libs_list}
     $<LINK_ONLY:desktop-app::external_openssl>
     $<LINK_ONLY:desktop-app::external_jpeg>
-    $<TARGET_FILE:desktop-app::external_jpeg>
     $<LINK_ONLY:desktop-app::external_openh264>
     $<TARGET_FILE:desktop-app::external_openh264>
     $<LINK_ONLY:desktop-app::external_opus>
-    $<TARGET_FILE:desktop-app::external_opus>
     $<LINK_ONLY:desktop-app::external_vpx>
-    $<TARGET_FILE:desktop-app::external_vpx>
 )
 
 if (WIN32)
