--- Telegram/lib_ui/ui/widgets/popup_menu.cpp.orig	2022-11-12 11:12:02.000000000 +0000
+++ Telegram/lib_ui/ui/widgets/popup_menu.cpp	2022-12-05 07:53:47.978014760 +0000
@@ -637,14 +637,14 @@ bool PopupMenu::eventFilter(QObject *o,
 					event->device(),
 					event->touchPoints(),
 					event->timestamp()));
-#elif QT_VERSION < QT_VERSION_CHECK(6, 3, 0) // Qt < 6.0.0
+#elif QT_VERSION < QT_VERSION_CHECK(6, 2, 2) // Qt < 6.0.0
 			e->setAccepted(
 				QApplicationPrivate::translateRawTouchEvent(
 					this,
 					event->pointingDevice(),
 					const_cast<QList<QEventPoint> &>(event->points()),
 					event->timestamp()));
-#else // Qt < 6.3.0
+#else // Qt < 6.2.0
 			e->setAccepted(
 				QApplicationPrivate::translateRawTouchEvent(this, event));
 #endif
