Index: ipsec-tools-0.8.0/src/racoon/plainrsa-gen.8
===================================================================
--- ipsec-tools-0.8.0.orig/src/racoon/plainrsa-gen.8
+++ ipsec-tools-0.8.0/src/racoon/plainrsa-gen.8
@@ -74,7 +74,7 @@ Note that generating longer keys takes l
 .It Fl e Ar pubexp
 value of RSA public exponent.
 Default is
-.Li 0x3 .
+.Li 0x10001 .
 Don't change this unless you really know what you are doing!
 .It Fl f Ar outfile
 .Ar outfile
Index: ipsec-tools-0.8.0/src/racoon/plainrsa-gen.c
===================================================================
--- ipsec-tools-0.8.0.orig/src/racoon/plainrsa-gen.c
+++ ipsec-tools-0.8.0/src/racoon/plainrsa-gen.c
@@ -72,7 +72,7 @@ usage (char *argv0)
 	fprintf(stderr, "Usage: %s [options]\n", argv0);
 	fprintf(stderr, "\n");
 	fprintf(stderr, "  -b bits       Generate <bits> long RSA key (default=1024)\n");
-	fprintf(stderr, "  -e pubexp     Public exponent to use (default=0x3)\n");
+	fprintf(stderr, "  -e pubexp     Public exponent to use (default=%d)\n", RSA_F4);
 	fprintf(stderr, "  -f filename   Filename to store the key to (default=stdout)\n");
 	fprintf(stderr, "  -i filename   Input source for format conversion\n");
 	fprintf(stderr, "  -h            Help\n");
@@ -222,7 +222,7 @@ main (int argc, char *argv[])
 {
 	FILE *fp = stdout, *fpin = NULL;
 	size_t bits = 1024;
-	unsigned int pubexp = 0x3;
+	unsigned int pubexp = RSA_F4;
 	struct stat st;
 	extern char *optarg;
 	extern int optind;
