From e452307ba07f5d798edad73631182e137265da7d Mon Sep 17 00:00:00 2001
From: Tom de Vries <tdevries@suse.de>
Date: Tue, 13 Jun 2023 17:58:42 +0200
Subject: [PATCH 9/9] fixup gdb-rhbz1261564-aarch64-hw-watchpoint-test.patch

---
 .../rhbz1261564-aarch64-watchpoint.exp         | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/gdb/testsuite/gdb.base/rhbz1261564-aarch64-watchpoint.exp b/gdb/testsuite/gdb.base/rhbz1261564-aarch64-watchpoint.exp
index b1cf7115663..42ebc25cc49 100644
--- a/gdb/testsuite/gdb.base/rhbz1261564-aarch64-watchpoint.exp
+++ b/gdb/testsuite/gdb.base/rhbz1261564-aarch64-watchpoint.exp
@@ -20,12 +20,22 @@ if { [prepare_for_testing rhbz1261564-aarch64-watchpoint.exp "rhbz1261564-aarch6
 if { ! [ runto main ] } then { return 0 }
 
 set test "rwatch aligned.var4"
-if [istarget "s390*-*-*"] {
-    gdb_test $test {Target does not support this type of hardware watchpoint\.}
-    untested "s390* does not support hw read watchpoint"
+
+set supported 1
+gdb_test_multiple $test "" {
+    -re -wrap "Hardware read watchpoint \[0-9\]+: aligned.var4" {
+	pass $gdb_test_name
+    }
+    -re -wrap "Target does not support this type of hardware watchpoint\\." {
+	set supported 0
+    }
+}
+
+if { !$supported } {
+    unsupported $test
     return
 }
-gdb_test $test "Hardware read watchpoint \[0-9\]+: aligned.var4"
+
 
 proc checkvar { address } {
     global gdb_prompt
-- 
2.35.3

