Index: hyena-0.5/configure.ac
===================================================================
--- hyena-0.5.orig/configure.ac
+++ hyena-0.5/configure.ac
@@ -49,9 +49,9 @@ fi
 
 dnl Mono and gmcs
 SHAMROCK_CHECK_MONO_MODULE(1.9.1)
-SHAMROCK_FIND_MONO_2_0_COMPILER
+SHAMROCK_FIND_MONO_4_5_COMPILER
 SHAMROCK_FIND_MONO_RUNTIME
-SHAMROCK_CHECK_MONO_2_0_GAC_ASSEMBLIES([	
+SHAMROCK_CHECK_MONO_4_5_GAC_ASSEMBLIES([	
 	System.Data
 	Mono.Cairo
 	Mono.Posix
Index: hyena-0.5/build/m4/shamrock/mono.m4
===================================================================
--- hyena-0.5.orig/build/m4/shamrock/mono.m4
+++ hyena-0.5/build/m4/shamrock/mono.m4
@@ -10,7 +10,12 @@ AC_DEFUN([SHAMROCK_FIND_MONO_2_0_COMPILE
 
 AC_DEFUN([SHAMROCK_FIND_MONO_4_0_COMPILER],
 [
-	SHAMROCK_FIND_PROGRAM_OR_BAIL(MCS, dmcs)
+	SHAMROCK_FIND_PROGRAM_OR_BAIL(MCS, gmcs)
+])
+
+AC_DEFUN([SHAMROCK_FIND_MONO_4_5_COMPILER],
+[
+	SHAMROCK_FIND_PROGRAM_OR_BAIL(MCS, mcs)
 ])
 
 AC_DEFUN([SHAMROCK_FIND_MONO_RUNTIME],
@@ -62,4 +67,9 @@ AC_DEFUN([SHAMROCK_CHECK_MONO_4_0_GAC_AS
 	_SHAMROCK_CHECK_MONO_GAC_ASSEMBLIES(4.0, $*)
 ])
 
+AC_DEFUN([SHAMROCK_CHECK_MONO_4_5_GAC_ASSEMBLIES],
+[
+	_SHAMROCK_CHECK_MONO_GAC_ASSEMBLIES(4.5, $*)
+])
+
 
Index: hyena-0.5/Hyena.Gui/Makefile.am
===================================================================
--- hyena-0.5.orig/Hyena.Gui/Makefile.am
+++ hyena-0.5/Hyena.Gui/Makefile.am
@@ -1,5 +1,5 @@
 ASSEMBLY = Hyena.Gui
-ASSEMBLY_BUILD_FLAGS = -unsafe
+ASSEMBLY_BUILD_FLAGS = -unsafe -sdk:4
 TARGET = library
 LINK = -r:ICSharpCode.SharpZipLib -r:Mono.Posix -r:System -r:System.Core -r:Mono.Cairo \
 	   $(GTKSHARP_LIBS) $(GLIBSHARP_LIBS) \
