Index: autoconf-2.73/bin/autoreconf.in
===================================================================
--- autoconf-2.73.orig/bin/autoreconf.in
+++ autoconf-2.73/bin/autoreconf.in
@@ -752,6 +752,8 @@ sub autoreconf_current_directory ($)
                'AC_CONFIG_SUBDIRS',
                'AC_CONFIG_MACRO_DIR_TRACE',
                'AC_CONFIG_MACRO_DIR',
+               'AC_LIBLTDL_CONVENIENCE',
+               'AC_LIBLTDL_INSTALLABLE',
                'AC_INIT',
                'AC_REQUIRE_AUX_FILE',
                'AC_PROG_LIBTOOL',
@@ -798,6 +800,8 @@ sub autoreconf_current_directory ($)
                                        || $macro eq "AM_PROG_LIBTOOL"
                                        || $macro eq "LT_INIT";
       $uses_libltdl = 1             if $macro eq "LT_CONFIG_LTDL_DIR";
+      $uses_libltdl = 1             if $macro eq "AC_LIBLTDL_CONVENIENCE";
+      $uses_libltdl = 1             if $macro eq "AC_LIBLTDL_INSTALLABLE";
       $uses_automake = 1            if $macro eq "AM_INIT_AUTOMAKE";
       $uses_intltool = 1            if $macro eq "IT_PROG_INTLTOOL";
       $uses_gtkdoc = 1              if $macro eq "GTK_DOC_CHECK";
