--- Markdown-3.0/tests/test_legacy.py.orig	2018-10-18 20:59:01.521952070 +0200
+++ Markdown-3.0/tests/test_legacy.py	2018-10-18 20:59:20.746395446 +0200
@@ -29,6 +29,7 @@
 # Except for the warnings that shouldn't
 warnings.filterwarnings('default', category=PendingDeprecationWarning)
 warnings.filterwarnings('default', category=DeprecationWarning, module='markdown')
+warnings.filterwarnings('default', category=DeprecationWarning, module='yaml')
 
 parent_test_dir = os.path.abspath(os.path.dirname(__file__))
 
