--- build.xml	2015-02-15 03:12:23.105593326 +0100
+++ build.xml.java8	2015-02-15 03:16:59.857032487 +0100
@@ -694,6 +694,7 @@
                 <pathelement location="build/classes/configschema"/>
                 <pathelement location="build/classes/toolschema"/>
                 <pathelement location="build/classes/xsdschema"/>
+                <pathelement location="build/classes/typestore"/>
             </classpath>
         </javac>
     </target>
