Index: pytest-profiling-1.8.1/pytest_profiling.py
===================================================================
--- pytest-profiling-1.8.1.orig/pytest_profiling.py
+++ pytest-profiling-1.8.1/pytest_profiling.py
@@ -10,7 +10,6 @@ import errno
 from hashlib import md5
 import subprocess
 
-import six
 import pytest
 
 LARGE_FILENAME_HASH_LEN = 8
@@ -18,7 +17,7 @@ LARGE_FILENAME_HASH_LEN = 8
 
 def clean_filename(s):
     forbidden_chars = set(r'/?<>\:*|"')
-    return six.text_type("".join(c if c not in forbidden_chars and ord(c) < 127 else '_'
+    return str("".join(c if c not in forbidden_chars and ord(c) < 127 else '_'
                                  for c in s))
 
 
Index: pytest-profiling-1.8.1/tests/unit/test_profile.py
===================================================================
--- pytest-profiling-1.8.1.orig/tests/unit/test_profile.py
+++ pytest-profiling-1.8.1/tests/unit/test_profile.py
@@ -3,11 +3,11 @@
 # coverage, so force it to be reloaded within this test unit under coverage
 
 import os.path
-from six.moves import reload_module  # @UnresolvedImport
+from importlib import reload
 
 import pytest_profiling
 
-reload_module(pytest_profiling)
+reload(pytest_profiling)
 
 import os
 import subprocess
Index: pytest-profiling-1.8.1/setup.py
===================================================================
--- pytest-profiling-1.8.1.orig/setup.py
+++ pytest-profiling-1.8.1/setup.py
@@ -24,8 +24,7 @@ classifiers = [
     'Programming Language :: Python :: 3.12',
 ]
 
-install_requires = ['six',
-                    'pytest',
+install_requires = ['pytest',
                     'gprof2dot',
                     ]
 
