---
 blue/__init__.py |    5 +++--
 setup.py         |    2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

--- a/blue/__init__.py
+++ b/blue/__init__.py
@@ -448,8 +448,9 @@ def main():
         'Black', 'Blue'
     )
     # Change the config param callback to support setup.cfg, tox.ini, etc.
-    config_param = black.main.params[25]
-    assert config_param.name == 'config'
+    # the index changed somewhere between black 21.7b and 21.12b
+    cfgidx = {p.name: i for (i, p) in enumerate(black.main.params)}['config']
+    config_param = black.main.params[cfgidx]
     config_param.callback = read_configs
     # Change the version string by adding a redundant Click `version_option`
     # decorator on `black.main`. Fortunately the added `version_option` takes
--- a/setup.py
+++ b/setup.py
@@ -37,7 +37,7 @@ setup(
     packages=['blue'],
     tests_require=['tox'],
     cmdclass={'test': Tox},
-    install_requires=['black==22.1.0', 'flake8>=3.8'],
+    install_requires=['black>=22.1.0', 'flake8>=3.8'],
     project_urls={
         'Documentation': 'https://blue.readthedocs.io/en/latest',
         'Source': 'https://github.com/grantjenks/blue.git',
