Index: pan-0.146/pan/usenet-utils/mime-utils.cc
===================================================================
--- pan-0.146.orig/pan/usenet-utils/mime-utils.cc
+++ pan-0.146/pan/usenet-utils/mime-utils.cc
@@ -1749,9 +1749,9 @@ namespace pan
 #ifdef HAVE_GMIME_30
     GMimeObject *gmo;
     gmo = g_mime_message_get_mime_part (body);
-    if (g_mime_multipart_signed_sign (gpg_ctx, gmo, uid.c_str(), &err) <0)
+    if (g_mime_multipart_signed_sign (gpg_ctx, gmo, uid.c_str(), &err) != 0)
 #else
-    if (g_mime_multipart_signed_sign (mps, GMIME_OBJECT (part), gpg_ctx, uid.c_str(), GMIME_DIGEST_ALGO_SHA1, &err) <0)
+    if (g_mime_multipart_signed_sign (mps, GMIME_OBJECT (part), gpg_ctx, uid.c_str(), GMIME_DIGEST_ALGO_SHA1, &err) != 0)
 #endif
     {
       g_object_unref(mps);
@@ -1784,10 +1784,10 @@ namespace pan
 
 #ifdef HAVE_GMIME_30
     if (g_mime_multipart_encrypted_encrypt(gpg_ctx, GMIME_OBJECT (part), sign, uid.c_str(),
-                                           GMIME_ENCRYPT_NONE, rcp, &err) < 0)
+                                           GMIME_ENCRYPT_NONE, rcp, &err) != 0)
 #else
     if (g_mime_multipart_encrypted_encrypt(mpe, GMIME_OBJECT (part), gpg_ctx, sign,
-                                           uid.c_str(), GMIME_DIGEST_ALGO_SHA1, rcp, &err) < 0)
+                                           uid.c_str(), GMIME_DIGEST_ALGO_SHA1, rcp, &err) != 0)
 #endif
     {
       g_object_unref(mpe);
