--- Makefile.orig	2024-06-30 09:09:15.631522825 +0800
+++ Makefile	2024-06-30 09:14:38.004724948 +0800
@@ -25,13 +25,20 @@
 	asciidoctor -D. -a nofooter -a webfonts! $<
 
 install: galaxis.6 uninstall
-	install -m 755 -o 0 -g 0 -d $(ROOT)/usr/bin/
-	install -m 755 -o 0 -g 0 galaxis $(ROOT)/usr/bin/galaxis
-	install -m 755 -o 0 -g 0 -d $(ROOT)/usr/share/man/man6/
-	install -m 755 -o 0 -g 0 galaxis.6 $(ROOT)/usr/share/man/man6/galaxis.6
+	install -m 0755 -d $(DESTDIR)/usr/bin
+	install -m 0755 -d $(DESTDIR)/usr/share/man/man6
+	install -m 0755 -d $(DESTDIR)/usr/share/applications/
+	install -m 0755 -d $(DESTDIR)/usr/share/pixmaps/
+	install -m 0755 -d $(DESTDIR)/usr/share/appdata
+	install -m 0755 galaxis $(DESTDIR)/usr/bin/
+	install -m 0644 galaxis.6 $(DESTDIR)/usr/share/man/man6/
+	install -m 0644 galaxis.desktop $(DESTDIR)/usr/share/applications/
+	install -m 0644 galaxis.png $(DESTDIR)/usr/share/pixmaps/
 
 uninstall:
-	rm -f ${ROOT}/usr/bin/galaxis ${ROOT}/usr/share/man/man6/galaxis.6
+	rm -f /usr/bin/galaxis /usr/share/man/man6/galaxis.6
+	rm -f /usr/share/applications/galaxis.desktop
+	rm -f /usr/share/pixmaps/galaxis.png
 
 clean:
 	rm -f galaxis galaxis.6 galaxis-*.tar.gz *~ *.html
