Index: b/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in
===================================================================
--- a/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in	2018-04-04 16:21:09.706263465 +0800
+++ b/data/org.gnome.settings-daemon.plugins.power.gschema.xml.in	2018-04-04 16:21:25.782315839 +0800
@@ -41,5 +41,10 @@
       <summary>Power button action</summary>
       <description>The action to take when the system power button is pressed. This action is hard-coded (and the setting ignored) on virtual machines (power off) and tablets (suspend).</description>
     </key>
+    <key name="notify-idle-resumed" type="b">
+      <default>true</default>
+      <_summary>If an information message should be displayed when returning from idle sleep.</_summary>
+      <_description>If an information message should be displayed when returning from idle sleep.</_description>
+    </key>
   </schema>
 </schemalist>
Index: b/plugins/power/gsd-power-manager.c
===================================================================
--- a/plugins/power/gsd-power-manager.c	2018-04-04 16:21:09.710263478 +0800
+++ b/plugins/power/gsd-power-manager.c	2018-04-04 16:21:25.782315839 +0800
@@ -153,6 +153,7 @@
         NotifyNotification      *notification_ups_discharging;
         NotifyNotification      *notification_low;
         NotifyNotification      *notification_sleep_warning;
+        NotifyNotification      *notification_resumed;
         GsdPowerActionType       sleep_action_type;
         gboolean                 battery_is_low; /* laptop battery low, or UPS discharging */
 
@@ -188,6 +189,7 @@
         guint                    inhibit_lid_switch_timer_id;
         gboolean                 is_virtual_machine;
         gboolean                 is_tablet;
+        GsdPowerActionType       last_idle_power_action;
 
         /* Idles */
         GnomeIdleMonitor        *idle_monitor;
@@ -1527,6 +1529,49 @@
 }
 
 static void
+show_resumed_notification (GsdPowerManager *manager)
+{
+        GError *error = NULL;
+        gboolean ret;
+
+        /* don't show when running under GDM */
+        if (g_getenv ("RUNNING_UNDER_GDM") != NULL) {
+                g_debug ("running under gdm, so no notification");
+                return;
+        }
+
+        ret = g_settings_get_boolean (manager->priv->settings, "notify-idle-resumed");
+        if (!ret)
+                return;
+
+        /* close any existing notification of this class */
+        notify_close_if_showing (&manager->priv->notification_resumed);
+
+        /* create a new notification */
+        create_notification (_("System resumed from sleep"),
+                             _("The system automatically went to sleep because the system was idle."),
+                             NULL,
+                             &manager->priv->notification_resumed);
+        notify_notification_set_timeout (manager->priv->notification_resumed,
+                                         GSD_POWER_MANAGER_NOTIFY_TIMEOUT_SHORT);
+        notify_notification_set_urgency (manager->priv->notification_resumed,
+                                         NOTIFY_URGENCY_NORMAL);
+        /* TRANSLATORS: this is the notification application name */
+        notify_notification_set_app_name (manager->priv->notification_resumed, _("Power"));
+        notify_notification_set_hint (manager->priv->notification_resumed,
+                                      "transient", g_variant_new_boolean (TRUE));
+
+        /* try to show */
+        ret = notify_notification_show (manager->priv->notification_resumed,
+                                        &error);
+        if (!ret) {
+                g_warning ("failed to show notification: %s", error->message);
+                g_error_free (error);
+                g_object_unref (manager->priv->notification_resumed);
+        }
+}
+
+static void
 idle_set_mode (GsdPowerManager *manager, GsdPowerIdleMode mode)
 {
         gboolean ret = FALSE;
@@ -1616,6 +1661,7 @@
                         action_type = g_settings_get_enum (manager->priv->settings,
                                                            "sleep-inactive-ac-type");
                 }
+                manager->priv->last_idle_power_action = action_type;
                 do_power_action_type (manager, action_type);
 
         /* turn on screen and restore user-selected brightness level */
@@ -2398,6 +2444,12 @@
 static void
 handle_resume_actions (GsdPowerManager *manager)
 {
+        if (manager->priv->last_idle_power_action == GSD_POWER_ACTION_SUSPEND ||
+            manager->priv->last_idle_power_action == GSD_POWER_ACTION_HIBERNATE)
+                show_resumed_notification (manager);
+
+        manager->priv->last_idle_power_action = GSD_POWER_ACTION_NOTHING;
+
         /* ensure we turn the panel back on after resume */
         backlight_enable (manager);
 
