diff --git a/phalcon.zep.c b/phalcon.zep.c
index 8ce9c9c..283cd6d 100644
--- a/phalcon.zep.c
+++ b/phalcon.zep.c
@@ -58,7 +58,11 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <main/php_ini.h>
 #include <main/SAPI.h>
 #include <ext/standard/php_string.h>
+#if PHP_VERSION_ID < 70200
 #include <ext/standard/php_smart_string.h>
+#else
+#include <Zend/zend_smart_string.h>
+#endif
 #include <ext/standard/info.h>
 #include <ext/standard/file.h>
 #include <ext/standard/php_filestat.h>
