Index: c-ares-1.18.1/test/Makefile.inc
===================================================================
--- c-ares-1.18.1.orig/test/Makefile.inc
+++ c-ares-1.18.1/test/Makefile.inc
@@ -16,7 +16,6 @@ TESTSOURCES = ares-test-main.cc	\
   ares-test-parse-txt.cc		\
   ares-test-parse-uri.cc		\
   ares-test-misc.cc			\
-  ares-test-live.cc			\
   ares-test-mock.cc			\
   ares-test-mock-ai.cc			\
   ares-test-internal.cc		\
