diff -Naur c-ares-1.16.1.orig/test/Makefile.inc c-ares-1.16.1/test/Makefile.inc
--- c-ares-1.16.1.orig/test/Makefile.inc	2020-07-08 22:15:36.667605939 +0200
+++ c-ares-1.16.1/test/Makefile.inc	2020-07-08 22:16:25.407171729 +0200
@@ -14,7 +14,6 @@
   ares-test-parse-srv.cc		\
   ares-test-parse-txt.cc		\
   ares-test-misc.cc			\
-  ares-test-live.cc			\
   ares-test-mock.cc			\
   ares-test-mock-ai.cc			\
   ares-test-internal.cc		\
