Index: c-ares-1.17.0/include/CMakeLists.txt
===================================================================
--- c-ares-1.17.0.orig/include/CMakeLists.txt
+++ c-ares-1.17.0/include/CMakeLists.txt
@@ -3,6 +3,6 @@ CONFIGURE_FILE (ares_build.h.cmake ${PRO
 
 # Headers installation target
 IF (CARES_INSTALL)
-	SET (CARES_HEADERS ares.h ares_version.h "${PROJECT_BINARY_DIR}/ares_build.h" ares_rules.h)
+	SET (CARES_HEADERS ares.h ares_version.h "${PROJECT_BINARY_DIR}/ares_build.h" ares_rules.h ares_dns.h)
 	INSTALL (FILES ${CARES_HEADERS} COMPONENT Devel DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 ENDIF ()
