--- src/sysctl.c
+++ src/sysctl.c	2026-08-28 09:08:46.524799993 +0000
@@ -408,7 +408,7 @@ static int ReadSetting(const char *restr
 			rc = EXIT_FAILURE;
 			break;
 		case EIO:	    /* Ignore stable_secret below /proc/sys/net/ipv6/conf */
-			rc = EXIT_FAILURE;
+			rc = EXIT_SUCCESS;
 			break;
 		default:
 			warn(_("reading key \"%s\""), outname);
@@ -460,7 +460,7 @@ static int ReadSetting(const char *restr
 					goto out;
 				}
 			case EIO:	    /* Ignore stable_secret below /proc/sys/net/ipv6/conf */
-				rc = EXIT_FAILURE;
+				rc = EXIT_SUCCESS;
 				break;
 			default:
 				warnx(_("reading key \"%s\""), outname);
