diff --git a/django_opentracing/middleware.py b/django_opentracing/middleware.py
index bcfba42..718073e 100644
--- a/django_opentracing/middleware.py
+++ b/django_opentracing/middleware.py
@@ -29,6 +29,7 @@ class OpenTracingMiddleware(MiddlewareMixin):
         self._init_tracing()
         self._tracing = settings.OPENTRACING_TRACING
         self.get_response = get_response
+        self._is_coroutine = None
 
     def _init_tracing(self):
         if getattr(settings, 'OPENTRACING_TRACER', None) is not None:
