From 324b84b263c7b7cde857c41bfc0458b071fa903f Mon Sep 17 00:00:00 2001
From: Todd <toddrme2178@gmail.com>
Date: Mon, 20 May 2019 12:16:14 -0400
Subject: [PATCH] Change 'bs4' dependency to 'beautifulsoup4'

'bs4' is just a dummy package to make it easier to install.  'beautifulsoup4' is the real package name.  Downstream packagers like Anaconda and Linux distributions will have 'beautifulsoup4' available, but not 'bs4', making it harder to install this package.
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 899d700..7da398e 100644
--- a/setup.py
+++ b/setup.py
@@ -21,7 +21,7 @@
 
 # What packages are required for this module to be executed?
 REQUIRED = [
-    'requests', 'pyquery', 'fake-useragent', 'parse', 'bs4', 'w3lib', 'pyppeteer>=0.0.14'
+    'requests', 'pyquery', 'fake-useragent', 'parse', 'beautifulsoup4', 'w3lib', 'pyppeteer>=0.0.14'
 ]
 
 # The rest you shouldn't have to touch too much :)
