--- a/doc/latex2man.local	2025-12-14 13:04:34.000000000 +0100
+++ b/doc/latex2man.local	2026-08-21 10:58:34.375214259 +0200
@@ -28,7 +28,7 @@
 # use strict 'vars';
 
 $CMD      = basename ($0);
-$gen_date = localtime();       # date when the output was generated
+$gen_date = gmtime($ENV{SOURCE_DATE_EPOCH} || time);       # date when the output was generated
 
 sub date2str;
 $VERSION = "1.29";
@@ -1619,7 +1619,7 @@ sub date2str
     return $Day . " " . $EnglishMonthName[$Month] . " " . $Year;
 }
 
-$rcs_date=`date '+%Y/%m/%d/'`;             # date of the man-page, overwritten
+$rcs_date=`date -u -d@\$SOURCE_DATE_EPOCH '+%Y/%m/%d/'`;             # date of the man-page, overwritten
                                            # by \rcsInfo
 $date =  date2str ($rcs_date);
 $Macro->{'today'}  = $date;
