diff --git a/tests/vala-dmap-container-db.vala b/tests/vala-dmap-container-db.vala
index 7571e39..8aee03d 100644
--- a/tests/vala-dmap-container-db.vala
+++ b/tests/vala-dmap-container-db.vala
@@ -24,7 +24,7 @@ private class ValaDMAPContainerDb : GLib.Object, DMAP.ContainerDb {
 	// A dumb database that stores everything in an array
 
 	/* FIXME: What is with this unowned? */
-	private Gee.ArrayList<unowned DMAP.Record> db = new Gee.ArrayList<DMAP.Record> ();
+	private Gee.ArrayList<unowned DMAP.Record> db = new Gee.ArrayList<unowned DMAP.Record> ();
 
 	public int64 count () {
 		return db.size;
diff --git a/tests/vala-dmap-db.vala b/tests/vala-dmap-db.vala
index 2d6ef0a..171f548 100644
--- a/tests/vala-dmap-db.vala
+++ b/tests/vala-dmap-db.vala
@@ -24,7 +24,7 @@ private class ValaDMAPDb : GLib.Object, DMAP.Db {
 	// A dumb database that stores everything in an array
 
 	/* FIXME: What is with this unowned? */
-	private Gee.ArrayList<unowned DMAP.Record> db = new Gee.ArrayList<DMAP.Record> ();
+	private Gee.ArrayList<unowned DMAP.Record> db = new Gee.ArrayList<unowned DMAP.Record> ();
 
 	public uint add (DMAP.Record record) {
 		db.add (((DMAP.Record) record));

