From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 29 May 2012 20:44:18 +0100
Subject: configure: Allow user defined C compiler flags.

---
 Makefile     |    1 -
 configure.ac |    3 +++
 2 files changed, 3 insertions(+), 1 deletion(-)

--- a/Makefile
+++ b/Makefile
@@ -942,7 +942,6 @@
 DOC_FILES=\
   Changes \
   README.adoc \
-  README.win32.adoc \
   LICENSE
 
 # Run all tests
--- a/configure.ac
+++ b/configure.ac
@@ -2974,6 +2974,9 @@
 AS_IF([test "$ccomp_type" != "msvc"],
   [internal_cflags="-g $internal_cflags"])
 
+# Store user defined global C Compiler flags
+common_cflags="$common_cflags $EXTRA_CFLAGS"
+
 oc_cflags="$common_cflags $internal_cflags"
 oc_cppflags="$common_cppflags $internal_cppflags"
 
