From 7634845cb013301aa159fd26077289b0444b2732 Mon Sep 17 00:00:00 2001
From: Luca Beltrame <lbeltrame@kde.org>
Date: Wed, 27 Feb 2019 09:41:54 +0100
Subject: [PATCH] Use qdbus-qt5 instead of qdbus

---
 desktop/org.kde.spectacle.desktop | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

Index: spectacle-19.08.1/desktop/org.kde.spectacle.desktop
===================================================================
--- spectacle-19.08.1.orig/desktop/org.kde.spectacle.desktop	2019-09-01 05:30:30.000000000 +0200
+++ spectacle-19.08.1/desktop/org.kde.spectacle.desktop	2019-09-06 08:29:28.667414001 +0200
@@ -128,7 +128,7 @@
 Keywords[x-test]=xxsnapshotxx;xxcapturexx;xxprintxx;xxscreenshotxx;xxsnippingxx;xxsnipxx;
 Keywords[zh_CN]=snapshot;capture;print;screenshot;snipping;snip;快照;截图;抓取;屏幕截图;截屏;抓屏;
 Keywords[zh_TW]=snapshot;capture;print;screenshot;snipping;snip;截圖;擷取;截圖程式
-Exec=qdbus org.kde.Spectacle / StartAgent
+Exec=qdbus-qt5 org.kde.Spectacle / StartAgent
 Icon=spectacle
 Type=Application
 Terminal=false
@@ -178,7 +178,7 @@
 Name[x-test]=xxCapture Entire Desktopxx
 Name[zh_CN]=截取整个桌面
 Name[zh_TW]=擷取整個桌面
-Exec=qdbus org.kde.Spectacle / FullScreen false
+Exec=qdbus-qt5 org.kde.Spectacle / FullScreen false
 X-KDE-Shortcuts=Shift+Print
 
 [Desktop Action CurrentMonitorScreenShot]
@@ -221,7 +221,7 @@
 Name[x-test]=xxCapture Current Monitorxx
 Name[zh_CN]=截取当前显示器
 Name[zh_TW]=擷取目前螢幕
-Exec=qdbus org.kde.Spectacle / CurrentScreen false
+Exec=qdbus-qt5 org.kde.Spectacle / CurrentScreen false
 
 
 [Desktop Action ActiveWindowScreenShot]
@@ -264,7 +264,7 @@
 Name[x-test]=xxCapture Active Windowxx
 Name[zh_CN]=截取活动窗口
 Name[zh_TW]=擷取作用中的視窗
-Exec=qdbus org.kde.Spectacle / ActiveWindow true false
+Exec=qdbus-qt5 org.kde.Spectacle / ActiveWindow true false
 X-KDE-Shortcuts=Meta+Print
 
 [Desktop Action RectangularRegionScreenShot]
@@ -307,5 +307,5 @@
 Name[x-test]=xxCapture Rectangular Regionxx
 Name[zh_CN]=截取矩形区域
 Name[zh_TW]=擷取矩形區域
-Exec=qdbus org.kde.Spectacle / RectangularRegion true
+Exec=qdbus-qt5 org.kde.Spectacle / RectangularRegion true
 X-KDE-Shortcuts=Meta+Shift+Print
