Index: kdm/kfrontend/sessions/CMakeLists.txt
===================================================================
--- kdm/kfrontend/sessions/CMakeLists.txt.orig	2010-11-05 11:47:45.000000000 +0100
+++ kdm/kfrontend/sessions/CMakeLists.txt	2010-11-21 20:47:03.551842096 +0100
@@ -12,19 +12,14 @@ install(FILES
 	amaterus.desktop
 	amiwm.desktop
 	asclassic.desktop
-	blackbox.desktop
 	cde.desktop
 	ctwm.desktop
 	cwwm.desktop
 	e16.desktop
-        enlightenment.desktop
 	evilwm.desktop
 	fluxbox.desktop
 	flwm.desktop
-	fvwm.desktop
-	fvwm95.desktop
 	golem.desktop
-	icewm.desktop
 	ion.desktop
 	larswm.desktop
 	lwm.desktop
@@ -47,9 +42,6 @@ install(FILES
 	w9wm.desktop
 	waimea.desktop
 	wm2.desktop
-	wmaker.desktop
-	xfce.desktop
-	xfce4.desktop
 
 	DESTINATION ${DATA_INSTALL_DIR}/kdm/sessions
 )
