diff -ur kde-workspace-4.8.1.orig/plasma/desktop/applets/kickoff/ui/brandingbutton.cpp kde-workspace-4.8.1/plasma/desktop/applets/kickoff/ui/brandingbutton.cpp
--- kde-workspace-4.8.1.orig/plasma/desktop/applets/kickoff/ui/brandingbutton.cpp	2012-03-20 15:15:39.645598514 +0100
+++ kde-workspace-4.8.1/plasma/desktop/applets/kickoff/ui/brandingbutton.cpp	2012-03-20 15:17:03.369606255 +0100
@@ -68,7 +68,9 @@
 
 void BrandingButton::openHomepage()
 {
-    new KRun(Plasma::Theme::defaultTheme()->homepage(), topLevelWidget(), false, false);
+    KUrl suseurl;
+    suseurl = "http://www.opensuse.org";
+    new KRun(suseurl, topLevelWidget(), false, false);
 }
 
 void BrandingButton::paintEvent(QPaintEvent *event)
