---
 emacs/ocamltags.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/emacs/ocamltags.in
+++ b/emacs/ocamltags.in
@@ -1,4 +1,4 @@
-":" ; @EMACS@ -batch -l $0 "$@" ; status=$? ; : '--*-Emacs-Lisp-*--' <<';'
+":" ; @EMACS@ --batch --no-init-file --no-site-file -l $0 "$@" ; status=$? ; : '--*-Emacs-Lisp-*--' <<';'
 
 ;**************************************************************************
 ;*                                                                        *
