Index: qterm-0.7.3/CMakeLists.txt
===================================================================
--- qterm-0.7.3.orig/CMakeLists.txt
+++ qterm-0.7.3/CMakeLists.txt
@@ -22,9 +22,12 @@ IF(QT5)
     find_package(Qt5ScriptTools 5.3.0)
     find_package(Qt5Multimedia 5.3.0)
     IF(NOT QT_BINARY_DIR)
-        exec_program(${_qt5Core_install_prefix}/bin/qmake ARGS "-query QT_INSTALL_BINS"
+        exec_program(${_qt5Core_install_prefix}/bin/qmake-qt5 ARGS "-query QT_INSTALL_BINS"
             OUTPUT_VARIABLE QT_BINARY_DIR)
     ENDIF()
+    exec_program(${_qt5Core_install_prefix}/bin/qmake-qt5 ARGS "-query QT_VERSION"
+        OUTPUT_VARIABLE QT5VERSION)
+    STRING(REGEX REPLACE "^[0-9]+\\.([0-9]+)\\.[0-9]+.*" "\\1" QT5_VERSION_MINOR "${QT5VERSION}")
 ELSE(QT5)
     find_package(Qt4 4.5.0 REQUIRED)
     find_package(KDE4)
Index: qterm-0.7.3/src/doc/CMakeLists.txt
===================================================================
--- qterm-0.7.3.orig/src/doc/CMakeLists.txt
+++ qterm-0.7.3/src/doc/CMakeLists.txt
@@ -7,11 +7,16 @@ CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DI
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/qterm.qhcp ${CMAKE_CURRENT_BINARY_DIR}/qterm.qhcp COPYONLY)
 
 if (QT5)
+if (${QT5_VERSION_MINOR} GREATER 11)
+  find_program(QT5_QCOLLECTIONGENERATOR_EXECUTABLE NAMES qhelpgenerator-qt5 qhelpgenerator PATHS ${QT_BINARY_DIR} NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH)
+else()
+  find_program(QT5_QCOLLECTIONGENERATOR_EXECUTABLE NAMES qcollectiongenerator-qt5 qcollectiongenerator PATHS ${QT_BINARY_DIR} NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH)
+endif()
 ADD_CUSTOM_COMMAND(
   OUTPUT qterm.qhc
   DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/qterm.qhcp
   DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/qterm.qhp
-  COMMAND Qt5::qcollectiongenerator
+  COMMAND ${QT5_QCOLLECTIONGENERATOR_EXECUTABLE}
   ARGS qterm.qhcp
       -o qterm.qhc
   COMMENT "Compiling Qt help collection project qterm.qhcp"
