Index: GASNet-1.30.0/configure
===================================================================
--- GASNet-1.30.0.orig/configure
+++ GASNet-1.30.0/configure
@@ -6008,6 +6008,9 @@ ac_configure_args="$ac_configure_args --
   fi
 
   SYSTEM_NAME="`hostname`"
+  if [ -n "$SOURCE_DATE_EPOCH" ] ; then
+    SYSTEM_NAME=reproducible
+  fi
 
   case "$target" in
    # Apple gcc has -arch options for cross-compilation, but target binaries may still work due to Rosetta (making our cross-compilation support unnecessary)
