Index: streamz-0.6.3/streamz/tests/test_sources.py
===================================================================
--- streamz-0.6.3.orig/streamz/tests/test_sources.py
+++ streamz-0.6.3/streamz/tests/test_sources.py
@@ -101,7 +101,7 @@ def test_http():
 
 @gen_test(timeout=60)
 def test_process():
-    cmd = ["python", "-c", "for i in range(4): print(i, end='')"]
+    cmd = [sys.executable, "-c", "for i in range(4): print(i, end='')"]
     s = Source.from_process(cmd, with_end=True)
     if sys.platform != "win32":
         # don't know why - something with pytest and new processes
@@ -117,7 +117,7 @@ def test_process():
 
 @gen_test(timeout=60)
 def test_process_str():
-    cmd = 'python -c "for i in range(4): print(i)"'
+    cmd = f'{sys.executable} -c "for i in range(4): print(i)"'
     s = Source.from_process(cmd)
     if sys.platform != "win32":
         # don't know why - something with pytest and new processes
