Index: glucat-0.10.0/Doxyfile.in
===================================================================
--- glucat-0.10.0.orig/Doxyfile.in
+++ glucat-0.10.0/Doxyfile.in
@@ -1252,7 +1252,7 @@ HTML_COLORSTYLE_GAMMA  = 80
 # The default value is: NO.
 # This tag requires that the tag GENERATE_HTML is set to YES.
 
-HTML_TIMESTAMP         = YES
+HTML_TIMESTAMP         =
 
 # If the HTML_DYNAMIC_MENUS tag is set to YES then the generated HTML
 # documentation will contain a main index with vertical navigation menus that
