Index: ristretto-0.3.5/ristretto.desktop.in
===================================================================
--- ristretto-0.3.5.orig/ristretto.desktop.in
+++ ristretto-0.3.5/ristretto.desktop.in
@@ -10,4 +10,4 @@ Type=Application
 X-MultipleArgs=false
 Categories=GTK;Graphics;Viewer;
 StartupNotify=true
-MimeType=image/png;image/gif;image/jpeg;image/bmp;image/x-pixmap;image/tiff;image/svg+xml;image/x-xpixmap;
+MimeType=image/png;image/gif;image/jpeg;image/bmp;image/tiff;image/svg+xml;image/x-xpixmap;image/x-xpm;image/x-cmu-raster;image/x-ico;image/x-portable-anymap;image/x-portable-bitmap;image/x-portable-graymap;image/x-portable-pixmap;image/x-sun-raster;image/x-tga;image/x-win-bitmap;image/x-xbitmap;
