From 7904505ec9330656859decd8837fc597ad708bec Mon Sep 17 00:00:00 2001
From: Mizux Seiha <mizux.dev@gmail.com>
Date: Tue, 14 Mar 2023 16:44:30 +0100
Subject: [PATCH] base: Remove AbslStringify()

---
 ortools/base/logging.h | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/ortools/base/logging.h b/ortools/base/logging.h
index 7f570f969e6..509e8873adf 100644
--- a/ortools/base/logging.h
+++ b/ortools/base/logging.h
@@ -52,15 +52,4 @@ enum LogSeverity {
 };
 }  // namespace google
 
-// Implementation of the `AbslStringify` interface. This adds `DebugString()`
-// to the sink. Do not rely on exact format.
-namespace google {
-namespace protobuf {
-template <typename Sink>
-void AbslStringify(Sink& sink, const Message& msg) {
-  sink.Append(msg.DebugString());
-}
-}  // namespace protobuf
-}  // namespace google
-
 #endif  // OR_TOOLS_BASE_LOGGING_H_
