Index: shaderc-2020.4/CMakeLists.txt
===================================================================
--- shaderc-2020.4.orig/CMakeLists.txt
+++ shaderc-2020.4/CMakeLists.txt
@@ -102,18 +102,12 @@ endif(MSVC)
 
 # Configure subdirectories.
 # We depend on these for later projects, so they should come first.
-add_subdirectory(third_party)
 
 add_subdirectory(libshaderc_util)
 add_subdirectory(libshaderc)
 add_subdirectory(glslc)
 add_subdirectory(examples)
 
-add_custom_target(build-version
-  ${PYTHON_EXECUTABLE}
-  ${CMAKE_CURRENT_SOURCE_DIR}/utils/update_build_version.py
-  ${shaderc_SOURCE_DIR} ${spirv-tools_SOURCE_DIR} ${glslang_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/build-version.inc
-  COMMENT "Update build-version.inc in the Shaderc build directory (if necessary).")
 
 function(define_pkg_config_file NAME LIBS)
   add_custom_target(${NAME}-pkg-config ALL
Index: shaderc-2020.4/libshaderc/CMakeLists.txt
===================================================================
--- shaderc-2020.4.orig/libshaderc/CMakeLists.txt
+++ shaderc-2020.4/libshaderc/CMakeLists.txt
@@ -66,6 +66,8 @@ set(SHADERC_LIBS
   shaderc_util
   SPIRV # from glslang
   SPIRV-Tools
+  SPVRemapper
+  HLSL
 )
 
 target_link_libraries(shaderc PRIVATE ${SHADERC_LIBS})
