Index: freeradius-server-3.0.21/suse/radiusd-logrotate
===================================================================
--- freeradius-server-3.0.21.orig/suse/radiusd-logrotate
+++ freeradius-server-3.0.21/suse/radiusd-logrotate
@@ -11,19 +11,23 @@ missingok
 compress
 delaycompress
 notifempty
-su radiusd radiusd
 
 #
 #  The main server log
 #
 /var/log/radius/radius.log {
+	su radiusd radiusd
 	copytruncate
+	postrotate
+			kill -HUP `cat /run/radiusd/radiusd.pid` || :
+	endscript
 }
 
 #
 #  Session monitoring utilities
 #
 /var/log/radius/checkrad.log /var/log/radius/radwatch.log {
+	su radiusd radiusd
 	nocreate
 	size=+1024k
 }
@@ -32,6 +36,7 @@ su radiusd radiusd
 #  SQL log files
 #
 /var/log/radius/sqllog.sql {
+	su radiusd radiusd
 	nocreate
 	size=+2048k
 }
@@ -44,6 +49,7 @@ su radiusd radiusd
 # second technique, you will need another cron job that removes old
 # detail files.  You do not need to comment out the below for method #2.
 /var/log/radius/radacct/*/detail {
+	su radiusd radiusd
 	nocreate
 }
 
