Index: webalizer.h
===================================================================
--- webalizer.h.orig
+++ webalizer.h
@@ -19,7 +19,7 @@
 #define MAXURLH  512                   /* Max URL field size in htab       */
 #define MAXREF   1024                  /* Max referrer field size          */
 #define MAXREFH  256                   /* Max referrer field size in htab  */
-#define MAXAGENT 128                   /* Max user agent field size        */
+#define MAXAGENT (BUFSIZE - 1)         /* Max user agent field size        */
 #define MAXCTRY  48                    /* Max country name size            */
 #define MAXSRCH  256                   /* Max size of search string buffer */
 #define MAXSRCHH 128                   /* Max size of search str in htab   */
