Index: Makefile.in
===================================================================
--- Makefile.in.orig
+++ Makefile.in
@@ -26,7 +26,7 @@ ETCDIR          = @sysconfdir@
 GEODB           = @GEODB_LOC@
 CC              = @CC@
 CFLAGS          = @CFLAGS@ @CPPFLAGS@
-LIBS            = @LIBS@
+LIBS            = @LIBS@ -ljpeg -ldb -L/usr/X11R6/$(LIBNAME) -lX11 -lXpm
 WCMGR_LIBS      = @WCMGR_LIBS@
 DEFS            = -DETCDIR=\"${ETCDIR}\" -DGEODB_LOC=\"${GEODB}\" @DEFS@ @OPTS@
 LDFLAGS         = @LDFLAGS@
Index: configure.in
===================================================================
--- configure.in.orig
+++ configure.in
@@ -48,9 +48,6 @@ if test "$GCC" = "yes"; then
   [  --enable-static         Build as static executable       [[default=no]]],
   LDFLAGS="--static ${LDFLAGS}")
 
-  if test "$IN_CFLAGS" = ""; then
-    CFLAGS="-Wall -O2"
-  fi
   if test "$ac_cv_c_char_unsigned" = "yes"; then
     CFLAGS="-fsigned-char ${CFLAGS}"
   fi
