From 70c5505a75ad545cb671eb73f29d5e1626aebf78 Mon Sep 17 00:00:00 2001
From: Bernd Hahnebach <bernd@bimstatik.org>
Date: Wed, 22 Sep 2021 08:06:47 +0200
Subject: [PATCH] Test: remove not needed u before py3 unicode string

---
 src/Mod/Test/UnitTests.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Mod/Test/UnitTests.py b/src/Mod/Test/UnitTests.py
index 6584ae6fc5..5c6ffea7e6 100644
--- a/src/Mod/Test/UnitTests.py
+++ b/src/Mod/Test/UnitTests.py
@@ -125,10 +125,10 @@ class UnitBasicCases(unittest.TestCase):
                 try:
                     q2 = FreeCAD.Units.Quantity(t[0])
                     if math.fabs(q1.Value - q2.Value) > 0.01:
-                        print (u" {} : {} : {} : {} : {}".format(q1,q2, t, i, val).encode("utf-8").strip())
+                        print (" {} : {} : {} : {} : {}".format(q1, q2, t, i, val).encode("utf-8").strip())
                 except Exception as e:
                     s = "{}: {}".format(e, t[0])
-                    print (u" ".join(e).encode("utf-8").strip())
+                    print (" ".join(e).encode("utf-8").strip())
 
     def testVoltage(self):
         q1 = FreeCAD.Units.Quantity("1e20 V")
-- 
2.35.1

