From 73307991550f57da9504237581b03faf9bcc0824 Mon Sep 17 00:00:00 2001
From: wmayer <wmayer@users.sourceforge.net>
Date: Tue, 25 Jan 2022 20:20:46 +0100
Subject: [PATCH] Test: fix exception handling in tests for units

---
 src/Mod/Test/UnitTests.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Mod/Test/UnitTests.py b/src/Mod/Test/UnitTests.py
index e9006455a0..9228684250 100644
--- a/src/Mod/Test/UnitTests.py
+++ b/src/Mod/Test/UnitTests.py
@@ -128,7 +128,7 @@ class UnitBasicCases(unittest.TestCase):
                         print (" {} : {} : {} : {} : {}".format(q1, q2, t, i, val).encode("utf-8").strip())
                 except Exception as e:
                     s = "{}: {}".format(e, t[0])
-                    print (" ".join(e).encode("utf-8").strip())
+                    print (" ".join(str(e)).encode("utf-8").strip())
 
     def testVoltage(self):
         q1 = FreeCAD.Units.Quantity("1e20 V")
-- 
2.35.1

