--- syslogd.c
+++ syslogd.c	2003-09-29 08:50:05.000000000 +0000
@@ -1305,6 +1305,7 @@ static int create_inet_socket()
 		close(fd);
 		return -1;
 	}
+#ifdef SO_BSDCOMPAT
 	/* We need to enable BSD compatibility. Otherwise an attacker
 	 * could flood our log files by sending us tons of ICMP errors.
 	 */
@@ -1314,6 +1315,7 @@ static int create_inet_socket()
 		close(fd);
 		return -1;
 	}
+#endif
 #ifdef INET6
 	error = bind(fd, res->ai_addr, res->ai_addrlen);
 	freeaddrinfo(res);
