--- klogd.c
+++ klogd.c	2007-06-18 11:22:06.978362000 +0000
@@ -416,9 +416,11 @@ static void Terminate()
 {
 	CloseLogSrc();
 	Syslog(LOG_INFO, "Kernel log daemon terminating.");
-	sleep(1);
-	if ( output_file != (FILE *) 0 )
+	if ( output_file != (FILE *) 0 ) {
+                fflush(output_file);
+                fsync(fileno(output_file));
 		fclose(output_file);
+        }
 	closelog();
 #ifndef TESTING
 	(void) remove_pid(PidFile);
