--- lxde-common-0.99.0/openbox/rc.xml.in	2014-10-24 22:25:07.018394980 +0200
+++ lxde-common-0.99.0-new/openbox/rc.xml.in	2014-10-24 22:27:36.893342071 +0200
@@ -57,10 +57,10 @@
   <keepBorder>yes</keepBorder>
   <animateIconify>yes</animateIconify>
   <font place="ActiveWindow">
-    <name>sans</name>
+    <name>Cantarell</name>
     <size>10</size>
     <!-- font size in points -->
-    <weight>bold</weight>
+    <weight>normal</weight>
     <!-- 'bold' or 'normal' -->
     <slant>normal</slant>
     <!-- 'italic' or 'normal' -->
@@ -69,7 +69,7 @@
     <name>sans</name>
     <size>10</size>
     <!-- font size in points -->
-    <weight>bold</weight>
+    <weight>Cantarell</weight>
     <!-- 'bold' or 'normal' -->
     <slant>normal</slant>
     <!-- 'italic' or 'normal' -->
