Index: fedora-policy-20211111/policy/modules/contrib/colord.fc
===================================================================
--- fedora-policy-20211111.orig/policy/modules/contrib/colord.fc
+++ fedora-policy-20211111/policy/modules/contrib/colord.fc
@@ -6,6 +6,8 @@
 
 /usr/libexec/colord	--	gen_context(system_u:object_r:colord_exec_t,s0)
 /usr/libexec/colord-sane	--	gen_context(system_u:object_r:colord_exec_t,s0)
+/usr/lib/colord	--	gen_context(system_u:object_r:colord_exec_t,s0)
+/usr/lib/colord-sane	--	gen_context(system_u:object_r:colord_exec_t,s0)
 
 /usr/lib/systemd/system/colord.*  -- gen_context(system_u:object_r:colord_unit_file_t,s0)
 
Index: fedora-policy-20211111/policy/modules/contrib/colord.te
===================================================================
--- fedora-policy-20211111.orig/policy/modules/contrib/colord.te
+++ fedora-policy-20211111/policy/modules/contrib/colord.te
@@ -17,6 +17,7 @@ type colord_t;
 type colord_exec_t;
 dbus_system_domain(colord_t, colord_exec_t)
 init_daemon_domain(colord_t, colord_exec_t)
+init_nnp_daemon_domain(colord_t)
 
 type colord_tmp_t;
 files_tmp_file(colord_tmp_t)
