Index: fedora-policy/policy/modules/contrib/screen.if
===================================================================
--- fedora-policy.orig/policy/modules/contrib/screen.if
+++ fedora-policy/policy/modules/contrib/screen.if
@@ -45,6 +45,7 @@ template(`screen_role_template',`
 
     userdom_list_user_home_dirs($1_screen_t)
 	userdom_home_reader($1_screen_t)
+        userdom_read_user_home_content_symlinks($1_screen_t)
 
 	domtrans_pattern($3, screen_exec_t, $1_screen_t)
 	allow $3 $1_screen_t:process { signal sigchld };
Index: fedora-policy/policy/modules/contrib/screen.fc
===================================================================
--- fedora-policy.orig/policy/modules/contrib/screen.fc
+++ fedora-policy/policy/modules/contrib/screen.fc
@@ -8,4 +8,5 @@ HOME_DIR/\.tmux\.conf	--	gen_context(sys
 /usr/bin/tmux			--	gen_context(system_u:object_r:screen_exec_t,s0)
 
 /var/run/screen(/.*)?			gen_context(system_u:object_r:screen_var_run_t,s0)
+/var/run/uscreens(/.*)?'                gen_context(system_u:object_r:screen_var_run_t,s0)
 /var/run/tmux(/.*)?			gen_context(system_u:object_r:screen_var_run_t,s0)
