Index: fedora-policy-20211111/policy/modules/contrib/firewalld.te
===================================================================
--- fedora-policy-20211111.orig/policy/modules/contrib/firewalld.te
+++ fedora-policy-20211111/policy/modules/contrib/firewalld.te
@@ -131,6 +131,7 @@ optional_policy(`
 ')
 
 optional_policy(`
+	iptables_manage_var_lib_files(firewalld_t)
 	iptables_domtrans(firewalld_t)
 	iptables_read_var_run(firewalld_t)
 ')
Index: fedora-policy-20211111/policy/modules/system/iptables.if
===================================================================
--- fedora-policy-20211111.orig/policy/modules/system/iptables.if
+++ fedora-policy-20211111/policy/modules/system/iptables.if
@@ -2,6 +2,25 @@
 
 ########################################
 ## <summary>
+##	Allow management of iptables_var_lib_t files
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to mange files
+##	</summary>
+## </param>
+#
+interface(`iptables_manage_var_lib_files',`
+	gen_require(`
+		type iptables_var_lib_t;
+	')
+
+	manage_dirs_pattern($1, iptables_var_lib_t, iptables_var_lib_t)
+	manage_files_pattern($1, iptables_var_lib_t, iptables_var_lib_t)
+')
+
+########################################
+## <summary>
 ##	Execute iptables in the iptables domain.
 ## </summary>
 ## <param name="domain">
