Index: fedora-policy-20220714/policy/modules/system/systemd.te
===================================================================
--- fedora-policy-20220714.orig/policy/modules/system/systemd.te
+++ fedora-policy-20220714/policy/modules/system/systemd.te
@@ -1447,6 +1447,12 @@ fstools_rw_swap_files(systemd_sleep_t)
 storage_getattr_fixed_disk_dev(systemd_sleep_t)
 storage_getattr_removable_dev(systemd_sleep_t)
 
+#######################################
+#
+# Allow systemd to watch certificate dir for ca-certificates
+# 
+watch_dirs_pattern(init_t,cert_t,cert_t)
+
 optional_policy(`
 	sysstat_domtrans(systemd_sleep_t)
 ')
