Index: fedora-policy/policy/modules/system/authlogin.fc
===================================================================
--- fedora-policy.orig/policy/modules/system/authlogin.fc
+++ fedora-policy/policy/modules/system/authlogin.fc
@@ -49,6 +49,7 @@ ifdef(`distro_gentoo', `
 /usr/sbin/validate	--	gen_context(system_u:object_r:chkpwd_exec_t,s0)
 
 /usr/libexec/utempter/utempter 	--	gen_context(system_u:object_r:utempter_exec_t,s0)
+/usr/lib/utempter/utempter 	--	gen_context(system_u:object_r:utempter_exec_t,s0)
 
 /var/ace(/.*)?			gen_context(system_u:object_r:var_auth_t,s0)
 
