Index: fedora-policy-20221019/policy/modules/contrib/dbus.te
===================================================================
--- fedora-policy-20221019.orig/policy/modules/contrib/dbus.te
+++ fedora-policy-20221019/policy/modules/contrib/dbus.te
@@ -81,6 +81,7 @@ manage_dirs_pattern(system_dbusd_t, syst
 manage_files_pattern(system_dbusd_t, system_dbusd_tmp_t, system_dbusd_tmp_t)
 manage_sock_files_pattern(system_dbusd_t, system_dbusd_tmp_t, system_dbusd_tmp_t)
 files_tmp_filetrans(system_dbusd_t, system_dbusd_tmp_t, { dir file sock_file })
+allow system_dbusd_t system_dbusd_tmp_t:file execute;
 
 manage_files_pattern(system_dbusd_t, system_dbusd_tmpfs_t, system_dbusd_tmpfs_t)
 manage_dirs_pattern(system_dbusd_t, system_dbusd_tmpfs_t, system_dbusd_tmpfs_t)
@@ -109,6 +110,8 @@ files_read_var_lib_symlinks(system_dbusd
 files_rw_inherited_non_security_files(system_dbusd_t)
 files_watch_usr_dirs(system_dbusd_t)
 files_watch_var_lib_dirs(system_dbusd_t)
+# bsc#1205895
+files_watch_lib_dirs(system_dbusd_t)
 
 fs_getattr_all_fs(system_dbusd_t)
 fs_search_auto_mountpoints(system_dbusd_t)
