Index: fedora-policy-20220428/policy/modules/system/systemd.te
===================================================================
--- fedora-policy-20220428.orig/policy/modules/system/systemd.te
+++ fedora-policy-20220428/policy/modules/system/systemd.te
@@ -355,6 +355,10 @@ userdom_manage_user_tmp_chr_files(system
 xserver_dbus_chat(systemd_logind_t)
 
 optional_policy(`
+	packagekit_dbus_chat(systemd_logind_t)
+')
+
+optional_policy(`
 	apache_read_tmp_files(systemd_logind_t)
 ')
 
@@ -882,6 +886,10 @@ optional_policy(`
 	udev_read_pid_files(systemd_hostnamed_t)
 ')
 
+optional_policy(`
+	nscd_unconfined(systemd_hostnamed_t)
+')
+
 #######################################
 #
 # rfkill policy
