Index: fedora-policy/policy/modules/contrib/irqbalance.te
===================================================================
--- fedora-policy.orig/policy/modules/contrib/irqbalance.te
+++ fedora-policy/policy/modules/contrib/irqbalance.te
@@ -29,8 +29,11 @@ allow irqbalance_t self:udp_socket creat
 manage_dirs_pattern(irqbalance_t, irqbalance_var_run_t, irqbalance_var_run_t)
 manage_files_pattern(irqbalance_t, irqbalance_var_run_t, irqbalance_var_run_t)
 manage_sock_files_pattern(irqbalance_t, irqbalance_var_run_t, irqbalance_var_run_t)
+manage_sock_files_pattern(irqbalance_t, irqbalance_var_run_t, irqbalance_var_run_t)
 files_pid_filetrans(irqbalance_t, irqbalance_var_run_t, { dir file sock_file })
 
+init_nnp_daemon_domain(irqbalance_t)
+
 kernel_read_network_state(irqbalance_t)
 kernel_read_system_state(irqbalance_t)
 kernel_read_kernel_sysctls(irqbalance_t)
