Index: fedora-policy-20210628/policy/modules/contrib/logrotate.te
===================================================================
--- fedora-policy-20210628.orig/policy/modules/contrib/logrotate.te
+++ fedora-policy-20210628/policy/modules/contrib/logrotate.te
@@ -104,6 +104,7 @@ files_var_lib_filetrans(logrotate_t, log
 
 kernel_read_system_state(logrotate_t)
 kernel_read_kernel_sysctls(logrotate_t)
+files_manage_mounttab(logrotate_t)
 
 dev_read_urand(logrotate_t)
 dev_read_sysfs(logrotate_t)
