Index: sanlock-3.9.3/init.d/sanlk-resetd.service
===================================================================
--- sanlock-3.9.3.orig/init.d/sanlk-resetd.service
+++ sanlock-3.9.3/init.d/sanlk-resetd.service
@@ -4,6 +4,17 @@ After=wdmd.service sanlock.service
 Requires=wdmd.service sanlock.service
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+ProtectHostname=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 Type=forking
 ExecStart=/usr/sbin/sanlk-resetd
 
