Index: sanlock-3.9.3/init.d/sanlock.service
===================================================================
--- sanlock-3.9.3.orig/init.d/sanlock.service
+++ sanlock-3.9.3/init.d/sanlock.service
@@ -5,8 +5,10 @@ Wants=wdmd.service
 
 [Service]
 Type=forking
-ExecStart=/lib/systemd/systemd-sanlock start
-ExecStop=/lib/systemd/systemd-sanlock stop
+EnvironmentFile=-/etc/sysconfig/sanlock
+ExecStart=/usr/sbin/sanlock daemon $SANLOCKOPTS
+ExecStop=/bin/kill -TERM $MAINPID
+ExecReload=/bin/kill -HUP $MAINPID
 
 [Install]
 WantedBy=multi-user.target
Index: sanlock-3.9.3/init.d/wdmd.service
===================================================================
--- sanlock-3.9.3.orig/init.d/wdmd.service
+++ sanlock-3.9.3/init.d/wdmd.service
@@ -6,7 +6,10 @@ After=syslog.target
 Type=forking
 ExecStartPre=/lib/systemd/systemd-wdmd watchdog-check
 ExecStart=/usr/sbin/wdmd
-SendSIGKILL=no
+EnvironmentFile=-/etc/sysconfig/wdmd
+ExecStart=/usr/sbin/wdmd $WDMDOPTS
+ExecStop=/bin/kill -TERM $MAINPID
+ExecReload=/bin/kill -HUP $MAINPID
 
 [Install]
 WantedBy=multi-user.target
Index: sanlock-3.9.3/init.d/fence_sanlockd.service
===================================================================
--- sanlock-3.9.3.orig/init.d/fence_sanlockd.service
+++ sanlock-3.9.3/init.d/fence_sanlockd.service
@@ -5,8 +5,8 @@ Before=corosync.service
 
 [Service]
 Type=forking
-ExecStart=/lib/systemd/systemd-fence_sanlockd start
-ExecStop=/lib/systemd/systemd-fence_sanlockd stop
+ExecStart=/usr/lib/systemd/systemd-fence_sanlockd start
+ExecStop=/usr/lib/systemd/systemd-fence_sanlockd stop
 
 [Install]
 WantedBy=multi-user.target
