Index: tuxpaint-config-0.0.13/Makefile
===================================================================
--- tuxpaint-config-0.0.13.orig/Makefile
+++ tuxpaint-config-0.0.13/Makefile
@@ -36,7 +36,7 @@ DATA_PREFIX=$(PREFIX)/share/tuxpaint-con
 
 # Docs and man page:
 
-DOC_PREFIX=$(PREFIX)/share/doc/tuxpaint-config/
+DOC_PREFIX=$(PREFIX)/share/doc/packages/tuxpaint/tuxpaint-config/
 MAN_PREFIX=$(PREFIX)/share/man/
 
 
@@ -154,7 +154,6 @@ install-win32:
 # to do this as superuser ("root"))
 
 install:	install-bin install-data install-man install-doc \
-		install-gnome install-kde install-kde-icons \
 		install-icon install-gettext
 	@echo
 	@echo "All done! Now (preferably NOT as 'root' superuser),"
