--- Makefile.am
+++ Makefile.am
@@ -15,22 +15,15 @@
 VERSION = @VERSION@
 
 # User-definable stuff:
-CC=@CC@
 DATA_PREFIX=$(pkgdatadir)/data/
 JOY=@JOY@
 TARGET_DEF=@TARGET_DEF@
 
 
-CFLAGS  = @CFLAGS@ -Wall -O2 -DDATA_PREFIX=\"$(DATA_PREFIX)\" \
-	  -D$(TARGET_DEF) -DJOY_$(JOY)
-LIBS    = @LIBS@ 
-
-
 bin_PROGRAMS = circuslinux
 
 circuslinux_SOURCES = circuslinux.c
-circuslinux_CFLAGS  = CFLAGS
-circuslinux_LIBS    = LIBS
+circuslinux_CFLAGS  = -DDATA_PREFIX=\"$(DATA_PREFIX)\" -D$(TARGET_DEF) -DJOY_$(JOY)
 
 docdir = $(prefix)/share/doc/$(PACKAGE)-$(VERSION)
 
--- data/Makefile.am
+++ data/Makefile.am
@@ -4,10 +4,10 @@
 
 
 install:
-	$(SHELL) ./../mkinstalldirs $(pkgdatadir)
-	$(SHELL) ./../mkinstalldirs $(pkgdatadir)/data
-	cp -r images music sounds $(pkgdatadir)/data/
-	chmod -R 755 $(pkgdatadir)
+	$(SHELL) ./../mkinstalldirs $(DESTDIR)$(pkgdatadir)
+	$(SHELL) ./../mkinstalldirs $(DESTDIR)$(pkgdatadir)/data
+	cp -r images music sounds $(DESTDIR)$(pkgdatadir)/data/
+	chmod -R 755 $(DESTDIR)$(pkgdatadir)
 
 # Directories in EXTRA_DIST doen't work?
 dist-hook:
