Index: openslp-2.0.0/slpd/slpd_process.c
===================================================================
--- openslp-2.0.0.orig/slpd/slpd_process.c
+++ openslp-2.0.0/slpd/slpd_process.c
@@ -1055,7 +1055,9 @@ RESPOND:
       if (truncate && size > truncate && G_SlpdProperty.hardMTU)
       {
          attrlistlen = 0;
+#ifdef ENABLE_SLPv2_SECURITY
          opaqueauth = 0;
+#endif
          size = message->header.langtaglen + 19; /* 14 bytes for header */
          truncated = 1;
       }
