diff --git a/Cargo.toml b/Cargo.toml
index e265273..2949424 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9,7 +9,7 @@ gtk = { version = "~0.8", package = "gtk4" }
 once_cell = "~1.19"
 qrcode-generator = "~4.1"
 # ncm-api = { git = "https://github.com/gmg137/netease-cloud-music-api.git", branch = "dev", package = "netease-cloud-music-api" }
-ncm-api = { git = "https://gitee.com/gmg137/netease-cloud-music-api.git", tag = "1.3.2", package = "netease-cloud-music-api" }
+ncm-api = { version = "1.3.2", package = "netease-cloud-music-api" }
 anyhow = "~1.0"
 gstreamer = "~0.22"
 gstreamer-play = "~0.22"
