---
 Lib/test/test_compile.py |    5 +++++
 1 file changed, 5 insertions(+)

--- a/Lib/test/test_compile.py
+++ b/Lib/test/test_compile.py
@@ -25,6 +25,9 @@ from test.support import (script_helper,
 from test.support.bytecode_helper import instructions_with_positions
 from test.support.os_helper import FakePath
 
+IS_SLE_15_6 = os.environ.get("SLE_VERSION", "") == "0150700"
+IS_32bit = hasattr(os, "uname") and os.uname().machine in ["i386", "i486", "i586", "i686"]
+
 class TestSpecifics(unittest.TestCase):
 
     def compile_single(self, source):
@@ -121,6 +124,7 @@ class TestSpecifics(unittest.TestCase):
         self.assertEqual(d['z'], 12)
 
     @unittest.skipIf(support.is_wasi, "exhausts limited stack on WASI")
+    @unittest.skipIf(IS_SLE_15_6 and IS_32bit, "fails on 15.6 i586")
     def test_extended_arg(self):
         repeat = int(get_c_recursion_limit() * 0.9)
         longexpr = 'x = x or ' + '-x' * repeat
@@ -709,6 +713,7 @@ class TestSpecifics(unittest.TestCase):
 
     @support.cpython_only
     @unittest.skipIf(support.is_wasi, "exhausts limited stack on WASI")
+    @unittest.skipIf(IS_SLE_15_6 and IS_32bit, "fails on 15.6 i586")
     def test_compiler_recursion_limit(self):
         # Expected limit is Py_C_RECURSION_LIMIT
         limit = get_c_recursion_limit()
