diff --git a/ChangeLog b/ChangeLog
index 86acd7b..9afbb0f 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -145,6 +145,12 @@ crash. This issue was fixed for other kinds of repeat in release 8.37 by change
 
 13. Fix misleading error message in configure.ac.
 
+23. Fixed a unicode properrty matching issue in JIT. The character was not
+fully read in caseless matching.
+
+23. Fixed a unicode properrty matching issue in JIT. The character was not
+fully read in caseless matching.
+
 
 Version 8.41 05-July-2017
 -------------------------
diff --git a/pcre_jit_compile.c b/pcre_jit_compile.c
index 4dcf8fc..5ab5206 100644
--- a/pcre_jit_compile.c
+++ b/pcre_jit_compile.c
@@ -5679,7 +5679,7 @@ while (*cc != XCL_END)
     {
     SLJIT_ASSERT(*cc == XCL_PROP || *cc == XCL_NOTPROP);
     cc++;
-    if (*cc == PT_CLIST)
+    if (*cc == PT_CLIST && cc[-1] == XCL_PROP)
       {
       other_cases = PRIV(ucd_caseless_sets) + cc[1];
       while (*other_cases != NOTACHAR)
diff --git a/pcre_jit_test.c b/pcre_jit_test.c
index 034cb52..cd687c3 100644
--- a/pcre_jit_test.c
+++ b/pcre_jit_test.c
@@ -387,6 +387,7 @@ static struct regression_test_case regression_test_cases[] = {
 	{ MUAP, 0 | F_PROPERTY, "[\xc3\xa2-\xc3\xa6\xc3\x81-\xc3\x84\xe2\x80\xa8-\xe2\x80\xa9\xe6\x92\xad\\p{Zs}]{2,}", "\xe2\x80\xa7\xe2\x80\xa9\xe6\x92\xad \xe6\x92\xae" },
 	{ MUAP, 0 | F_PROPERTY, "[\\P{L&}]{2}[^\xc2\x85-\xc2\x89\\p{Ll}\\p{Lu}]{2}", "\xc3\xa9\xe6\x92\xad.a\xe6\x92\xad|\xc2\x8a#" },
 	{ PCRE_UCP, 0, "[a-b\\s]{2,5}[^a]", "AB  baaa" },
+	{ CMUAP, 0, "[^S]\\B", "\xe2\x80\x8a" },
 
 	/* Possible empty brackets. */
 	{ MUA, 0, "(?:|ab||bc|a)+d", "abcxabcabd" },
