--- mybatis-3-mybatis-3.5.19/src/main/java/org/apache/ibatis/annotations/Select.java	2025-09-18 16:08:19.794480299 +0200
+++ mybatis-3-mybatis-3.5.19/src/main/java/org/apache/ibatis/annotations/Select.java	2025-09-18 16:10:06.140546344 +0200
@@ -29,22 +29,20 @@
  * <ul>
  * <li>Simple:
  *
- * <pre>{@code
+ * <pre>
  * public interface UserMapper {
- *   @Select("SELECT id, name FROM users WHERE id = #{id}")
+ *   &#x40;Select("SELECT id, name FROM users WHERE id = #{id}")
  *   User selectById(int id);
- * }
  * }</pre>
  *
  * </li>
  * <li>Dynamic SQL:
  *
- * <pre>{@code
+ * <pre>
  * public interface UserMapper {
- *   @Select({ "<script>", "select * from users", "where name = #{name}",
- *       "<if test=\"age != null\"> age = #{age} </if>", "</script>" })
- *   User select(@NotNull String name, @Nullable Integer age);
- * }
+ *   &#x40;Select({ "&lt;script&gt;", "select * from users", "where name = #{name}",
+ *       "&lt;if test=\"age != null\"&gt; age = #{age} &lt;/if&gt;", "&lt;/script&gt;" })
+ *   User select(&#x40;NotNull String name, &#x40;Nullable Integer age);
  * }</pre>
  *
  * </li>
