From 31a4b1040e04d711c6863f70561bde234f06f05a Mon Sep 17 00:00:00 2001
From: rcmadhankumar <madhankumar.chellamuthu@suse.com>
Date: Mon, 28 Apr 2025 17:40:28 +0530
Subject: [PATCH] remove appending rw as the default mount option

---
 pkg/util/mount_opts.go | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/pkg/util/mount_opts.go b/pkg/util/mount_opts.go
index c9a773093e..4e37fd74a0 100644
--- a/pkg/util/mount_opts.go
+++ b/pkg/util/mount_opts.go
@@ -191,9 +191,6 @@ func processOptionsInternal(options []string, isTmpfs bool, sourcePath string, g
 		newOptions = append(newOptions, opt)
 	}
 
-	if !foundWrite {
-		newOptions = append(newOptions, "rw")
-	}
 	if !foundProp {
 		if recursiveBind {
 			newOptions = append(newOptions, "rprivate")
-- 
2.49.0

