diff -bruNE prosody-0.9.11_orig/prosody prosody-0.9.11/prosody
--- prosody-0.9.11_orig/prosody	2016-12-14 11:10:56.135550760 +0100
+++ prosody-0.9.11/prosody	2016-12-14 11:11:33.025945126 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/usr/bin/lua5.1
 -- Prosody IM
 -- Copyright (C) 2008-2010 Matthew Wild
 -- Copyright (C) 2008-2010 Waqas Hussain
diff -bruNE prosody-0.9.11_orig/prosodyctl prosody-0.9.11/prosodyctl
--- prosody-0.9.11_orig/prosodyctl	2016-12-14 11:10:55.927559814 +0100
+++ prosody-0.9.11/prosodyctl	2016-12-14 11:11:39.469664663 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/usr/bin/lua5.1
 -- Prosody IM
 -- Copyright (C) 2008-2010 Matthew Wild
 -- Copyright (C) 2008-2010 Waqas Hussain
