Index: cinnamon-control-center-4.2.2/panels/common/Makefile.am
===================================================================
--- cinnamon-control-center-4.2.2.orig/panels/common/Makefile.am
+++ cinnamon-control-center-4.2.2/panels/common/Makefile.am
@@ -38,9 +38,9 @@ uidir = $(pkgdatadir)/ui
 dist_ui_DATA = \
 	language-chooser.ui
 
-rulesdir = $(datadir)/polkit-1/rules.d/
-rules_DATA = cinnamon-control-center.rules
+#rulesdir = $(datadir)/polkit-1/rules.d/
+#rules_DATA = cinnamon-control-center.rules
 
-EXTRA_DIST = $(rules_DATA)
+#EXTRA_DIST = $(rules_DATA)
 
 -include $(top_srcdir)/git.mk
