Index: OpenIPMI-2.0.31/ui/Makefile.am
===================================================================
--- OpenIPMI-2.0.31.orig/ui/Makefile.am
+++ OpenIPMI-2.0.31/ui/Makefile.am
@@ -18,6 +18,6 @@ libOpenIPMIui_la_LIBADD = $(top_builddir
 libOpenIPMIui_la_LDFLAGS = -version-info $(LD_VERSION) -no-undefined
 
 ipmi_ui_SOURCES = basic_ui.c
-ipmi_ui_LDADD = libOpenIPMIui.la $(top_builddir)/utils/libOpenIPMIutils.la $(top_builddir)/lib/libOpenIPMI.la $(top_builddir)/unix/libOpenIPMIposix.la $(TERM_LIBS) $(SNMPLIBS) $(OPENSSLLIBS) $(GDBM_LIB)
+ipmi_ui_LDADD = libOpenIPMIui.la $(top_builddir)/utils/libOpenIPMIutils.la $(top_builddir)/lib/libOpenIPMI.la $(top_builddir)/unix/libOpenIPMIposix.la $(TERM_LIBS) $(SNMPLIBS) $(OPENSSLLIBS) $(GDBM_LIB) -lncurses
 ipmi_ui_CFLAGS = -Wall -Wsign-compare -I$(top_builddir)/include \
 	-I$(top_srcdir)/include -DIPMI_CHECK_LOCKS
