Index: ld/emulparams/elf_i386.sh
===================================================================
--- ld/emulparams/elf_i386.sh.orig	2026-07-26 01:00:00.000000000 +0200
+++ ld/emulparams/elf_i386.sh	2026-09-21 16:25:20.479601643 +0200
@@ -49,6 +49,10 @@ case "$target" in
 	LIBPATH_SUFFIX=32
 	LIBPATH_SUFFIX_SKIP=64
 	;;
+      *64*)
+	LIBPATH_SUFFIX=64
+	LIBPATH_SUFFIX_SKIP=32
+	;;
     esac
     ;;
 esac
