From e484252ee7d39816c10c3a952b2e4fffd2c4967e Mon Sep 17 00:00:00 2001
From: Andreas Stieger <Andreas.Stieger@gmx.de>
Date: Sat, 3 Aug 2024 19:46:32 +0200
Subject: [PATCH] Build with Botan 3
References: https://github.com/solemnwarning/rehex/pull/240

Botan 2 is EOL EOY 2024
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 384902c..084905f 100644
--- a/Makefile
+++ b/Makefile
@@ -16,11 +16,11 @@
 
 LUA          ?= lua
 WX_CONFIG    ?= wx-config
-BOTAN_PKG    ?= botan-2
+BOTAN_PKG    ?= botan-3
 CAPSTONE_PKG ?= capstone
 JANSSON_PKG  ?= jansson
 LUA_PKG      ?= $(shell pkg-config --exists lua5.3 && echo lua5.3 || echo lua)
-CXXSTD       ?= -std=c++11
+CXXSTD       ?= -std=c++20
 
 EXE ?= rehex
 EMBED_EXE ?= ./tools/embed
-- 
2.46.0

