From 522f02e303c3762b2c1a81a001e7000055521a94 Mon Sep 17 00:00:00 2001
From: Patrick Griffis <tingping@tingping.se>
Date: Fri, 6 Apr 2018 03:11:46 -0400
Subject: [PATCH] Fix deprecated PyGObject usage

---
 pithos/plugins/notification_icon.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/pithos/plugins/notification_icon.py b/pithos/plugins/notification_icon.py
index 126eca66..1762b20a 100644
--- a/pithos/plugins/notification_icon.py
+++ b/pithos/plugins/notification_icon.py
@@ -164,10 +164,10 @@ def build_context_menu(self):
 
         def button(text, action, checked=False):
             if checked:
-                item = Gtk.CheckMenuItem(text)
+                item = Gtk.CheckMenuItem.new_with_label(text)
                 item.set_active(True)
             else:
-                item = Gtk.MenuItem(text)
+                item = Gtk.MenuItem.new_with_label(text)
             handler = item.connect('activate', action)
             item.show()
             menu.append(item)
@@ -253,16 +253,16 @@ class NotificationIconPluginPrefsDialog(Gtk.Dialog):
 
     def __init__(self, parent, settings):
         super().__init__(
-            _('Icon Type'),
-            parent,
-            0,
-            ('_Cancel', Gtk.ResponseType.CANCEL, '_Apply', Gtk.ResponseType.APPLY),
+            title=_('Icon Type'),
+            transient_for=parent,
             use_header_bar=1,
             resizable=False,
             default_width=300
         )
         self.settings = settings
 
+        self.add_buttons('_Cancel', Gtk.ResponseType.CANCEL, '_Apply', Gtk.ResponseType.APPLY)
+
         self.connect('delete-event', lambda *ignore: self.response(Gtk.ResponseType.CANCEL) or True)
 
         sub_title = Gtk.Label.new(_('Set the Notification Icon Type'))

