diff -urN tnftp-20130505.old/configure.ac tnftp-20130505/configure.ac
--- tnftp-20130505.old/configure.ac	2014-10-30 10:55:11.376328347 +0100
+++ tnftp-20130505/configure.ac	2014-10-30 10:56:37.813328337 +0100
@@ -71,10 +71,16 @@
 #
 # Checks for programs.
 #
-AC_PROG_CC()
+AC_PROG_CC_STDC
+AC_USE_SYSTEM_EXTENSIONS
+AC_SYS_LARGEFILE
+AM_PROG_AR
+AM_PROG_CC_C_O
 AC_PROG_AWK()
 AC_PROG_LIBTOOL()

+PKG_CHECK_MODULES([LIBEDIT], [libedit])
+
 #
 # Checks for tool features.
 #
@@ -95,13 +101,6 @@
 #
 # Checks for libraries.
 #
-AS_IF([test "$opt_editcomplete" = yes],
-      [AC_SEARCH_LIBS([tgetent],
-                      [termcap termlib curses ncurses tinfo],
-                      [],
-                      [AC_MSG_ERROR(
-                           [no relevant library found containing tgetent])])
-       ])

 AC_SEARCH_LIBS([gethostbyname], [nsl])
 AC_SEARCH_LIBS([socket],
@@ -381,7 +380,7 @@
 # Use local libedit if editcomplete is requested
 #
 AS_IF([test "$opt_editcomplete" = yes],
-      [AC_MSG_NOTICE([--enable-editcomplete; using internal libedit])],
+      [AC_MSG_NOTICE([--enable-editcomplete; using system libedit])],
       [CFLAGS="-DNO_EDITCOMPLETE $CFLAGS"])

 # Replace sl_init() (et al) if it provides the older API.
@@ -420,8 +419,6 @@
 #
 AC_CONFIG_FILES([
         Makefile
-        libedit/Makefile
-        libedit/makelist
         libnetbsd/Makefile
         src/Makefile
 ])
diff -urN tnftp-20130505.old/libnetbsd/Makefile.am tnftp-20130505/libnetbsd/Makefile.am
--- tnftp-20130505.old/libnetbsd/Makefile.am	2014-10-30 10:55:11.375328347 +0100
+++ tnftp-20130505/libnetbsd/Makefile.am	2014-10-30 10:55:40.171328344 +0100
@@ -5,7 +5,7 @@
 libnetbsd_la_SOURCES =


-CPPFLAGS = \
+AM_CPPFLAGS = \
 	-I$(srcdir) \
 	-I$(top_srcdir) \
 	-I$(top_builddir)
diff -urN tnftp-20130505.old/Makefile.am tnftp-20130505/Makefile.am
--- tnftp-20130505.old/Makefile.am	2014-10-30 10:55:11.375328347 +0100
+++ tnftp-20130505/Makefile.am	2014-10-30 10:55:40.171328344 +0100
@@ -4,10 +4,6 @@

 SUBDIRS = libnetbsd

-if USE_LIBEDIT
-SUBDIRS += libedit
-endif
-
 SUBDIRS += src

 EXTRA_DIST = \
diff -urN tnftp-20130505.old/src/Makefile.am tnftp-20130505/src/Makefile.am
--- tnftp-20130505.old/src/Makefile.am	2014-10-30 10:55:11.380328347 +0100
+++ tnftp-20130505/src/Makefile.am	2014-10-30 10:55:40.171328344 +0100
@@ -28,10 +28,10 @@

 if USE_LIBEDIT
 tnftp_CPPFLAGS += \
-	-I$(top_srcdir)/libedit
+	$(LIBEDIT_CFLAGS)

 tnftp_LDADD += \
-	../libedit/libedit.la
+	$(LIBEDIT_LIBS)
 endif


