diff -Nur apptools-5.1.0/apptools/io/h5/file.py new/apptools/io/h5/file.py
--- apptools-5.1.0/apptools/io/h5/file.py	2021-01-13 20:10:47.000000000 +0100
+++ new/apptools/io/h5/file.py	2022-05-02 22:42:46.404485964 +0200
@@ -7,7 +7,7 @@
 # is also available online at http://www.enthought.com/licenses/BSD.txt
 #
 # Thanks for using Enthought open source!
-from collections import Mapping, MutableMapping
+from collections.abc import Mapping, MutableMapping
 from functools import partial
 import inspect
 
