Index: cheroot-10.0.1/cheroot/test/conftest.py
===================================================================
--- cheroot-10.0.1.orig/cheroot/test/conftest.py
+++ cheroot-10.0.1/cheroot/test/conftest.py
@@ -9,7 +9,7 @@ import time
 
 import pytest
 
-from cheroot._compat import IS_MACOS, IS_WINDOWS  # noqa: WPS436
+from cheroot._compat import IS_MACOS, IS_WINDOWS, PLATFORM_ARCH  # noqa: WPS436
 from cheroot.server import Gateway, HTTPServer
 from cheroot.testing import (  # noqa: F401  # pylint: disable=unused-import
     native_server,
@@ -31,6 +31,9 @@ def http_request_timeout():
     if IS_WINDOWS:
         computed_timeout *= 10
 
+    if PLATFORM_ARCH == 's390x':
+        computed_timeout *= 2
+
     return computed_timeout
 
 
