---
 cheroot/test/test_server.py |   13 -------------
 1 file changed, 13 deletions(-)

Index: cheroot-9.0.0/cheroot/test/test_server.py
===================================================================
--- cheroot-9.0.0.orig/cheroot/test/test_server.py
+++ cheroot-9.0.0/cheroot/test/test_server.py
@@ -12,8 +12,6 @@ import pytest
 import requests
 import requests_unixsocket
 
-from pypytools.gc.custom import DefaultGc
-
 from .._compat import bton, ntob
 from .._compat import IS_LINUX, IS_MACOS, IS_WINDOWS, SYS_PLATFORM
 from ..server import IS_UID_GID_RESOLVABLE, Gateway, HTTPServer
@@ -380,13 +378,6 @@ if not IS_WINDOWS and not ISSUE511:
 
 
 @pytest.fixture
-def _garbage_bin():
-    """Disable garbage collection when this fixture is in use."""
-    with DefaultGc().nogc():
-        yield
-
-
-@pytest.fixture
 def resource_limit(request):
     """Set the resource limit two times bigger then requested."""
     resource = pytest.importorskip(
@@ -415,11 +406,6 @@ def resource_limit(request):
 @pytest.fixture
 def many_open_sockets(request, resource_limit):
     """Allocate a lot of file descriptors by opening dummy sockets."""
-    # NOTE: `@pytest.mark.usefixtures` doesn't work on fixtures which
-    # NOTE: forces us to invoke this one dynamically to avoid having an
-    # NOTE: unused argument.
-    request.getfixturevalue('_garbage_bin')
-
     # Hoard a lot of file descriptors by opening and storing a lot of sockets
     test_sockets = []
     # Open a lot of file descriptors, so the next one the server
