https://github.com/mehcode/python-xmlsec/pull/91

Index: xmlsec-1.3.3/xmlsec_setupinfo.py
===================================================================
--- xmlsec-1.3.3.orig/xmlsec_setupinfo.py
+++ xmlsec-1.3.3/xmlsec_setupinfo.py
@@ -34,7 +34,7 @@ def description():
 
 
 def sources():
-    return glob.glob(os.path.join(get_base_dir(), "src", "*.c"))
+    return sorted(glob.glob(os.path.join(get_base_dir(), "src", "*.c")))
 
 
 def define_macros():
