diff -Nru uamqp-1.1.0.orig/setup.py uamqp-1.1.0/setup.py
--- uamqp-1.1.0.orig/setup.py	2018-10-17 22:00:12.000000000 +0200
+++ uamqp-1.1.0/setup.py	2019-04-01 15:33:33.974522866 +0200
@@ -317,7 +317,7 @@
     ext_modules=extensions,
     install_requires=[
         "certifi>=2017.4.17",
-        "six~=1.0"
+        "six>=1.0"
     ],
     extras_require={
         ":python_version<'3.4'": ['enum34>=1.0.4']
