Index: tests/Makefile
===================================================================
--- tests/Makefile.orig
+++ tests/Makefile
@@ -48,7 +48,7 @@ ifeq ($(CUSTOM_LDSCRIPTS),yes)
 TESTS += $(LDSCRIPT_TESTS) $(HUGELINK_TESTS) $(HUGELINK_TESTS:%=xB.%) \
 	$(HUGELINK_TESTS:%=xBDT.%) $(HUGELINK_RW_TESTS)
 else
-TESTS += $(LDSCRIPT_TESTS) $(HUGELINK_TESTS) $(HUGELINK_RW_TESTS)
+TESTS += $(LDSCRIPT_TESTS) $(HUGELINK_TESTS)
 endif
 
 else
@@ -57,7 +57,7 @@ ifeq ($(CUSTOM_LDSCRIPTS),yes)
 TESTS += $(LDSCRIPT_TESTS) $(HUGELINK_TESTS) $(HUGELINK_TESTS:%=xB.%) \
 	$(HUGELINK_TESTS:%=xBDT.%) $(HUGELINK_RW_TESTS)
 else
-TESTS += $(LDSCRIPT_TESTS) $(HUGELINK_TESTS) $(HUGELINK_RW_TESTS)
+TESTS += $(LDSCRIPT_TESTS) $(HUGELINK_TESTS)
 endif
 
 endif
