Index: a2ps-4.14/man/help2man
===================================================================
--- a2ps-4.14.orig/man/help2man
+++ a2ps-4.14/man/help2man
@@ -164,7 +164,7 @@ my ($help_text, $version_text) = map {
 	or die "$this_program: can't get `--$_' info from $ARGV[0]\n"
 } qw(help version);
 
-my $date = strftime "%B %Y", localtime;
+my $date = strftime "%B %Y", gmtime($ENV{SOURCE_DATE_EPOCH} || time);
 (my $program = $ARGV[0]) =~ s!.*/!!;
 my $package = $program;
 my $version;
