diff -urEbwBN hplip-3.20.9/protocol/hp_ipp.h hplip-3.20.9.new/protocol/hp_ipp.h
--- hplip-3.20.9/protocol/hp_ipp.h	2020-09-23 07:24:16.000000000 +0200
+++ hplip-3.20.9.new/protocol/hp_ipp.h	2020-11-23 12:17:09.178020748 +0100
@@ -149,6 +149,7 @@
 
 
 http_t*    acquireCupsInstance();
+void _releaseCupsInstance();
 const char *getCupsErrorString(int status);
 void freePrinterList(printer_t *list);
  
@@ -162,6 +163,11 @@
 ipp_t * networkDoRequest(ipp_t *request, char* device_uri);
 ipp_t * getDeviceStatusAttributes(char* device_uri,char* printer_name, int *count);
 int     getCupsPrinters(printer_t **printer_list);
+int addCupsPrinter(char *name, char *device_uri, char *location, char *ppd_file,
+		   char *model, char *info);
+int delCupsPrinter(char *pr_name);
+int setDefaultCupsPrinter(char *pr_name);
+int controlCupsPrinter(char *pr_name, int op);
 
 HPIPP_RESULT parseResponseHeader(char* header, int *content_length, int *chunked, int* header_size);
 HPIPP_RESULT prepend_http_header(raw_ipp *raw_request);
