From: Egbert Eich <eich@suse.com>
Date: Mon Oct 4 19:48:26 2021 +0200
Subject: common: Fix help text
Patch-mainline: Not yet
Git-commit: 1a1e9ccc7fc19fac49eda100d83bfdb82be2b7fb
References: 

Signed-off-by: Egbert Eich <eich@suse.com>
---
 common/bin/wwconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
diff --git a/common/bin/wwconfig b/common/bin/wwconfig
index 58e98fa..34ab765 100755
--- a/common/bin/wwconfig
+++ b/common/bin/wwconfig
@@ -39,7 +39,7 @@ if ($opt_help) {
     print "\n";
     print "   Options:\n";
     print "      -h, --help     Show this help summary\n";
-    print "      -a, --all      Print all evnironment variables\n";
+    print "      -a, --all      Print all environment variables\n";
     print "      -s, --sh       Print output in Bourne shell format\n";
     print "      -e, --exp      Also print out export lines. For use with Bourne shell format.\n";
     print "      -c, --csh      Print output in C shell format\n";
