diff --git a/capi20/alaw.h b/capi20/alaw.h
index ef062fc..78c3996 100644
--- a/capi20/alaw.h
+++ b/capi20/alaw.h
@@ -2,9 +2,9 @@
 #define _ALAW_H
 #include <stdint.h>
 
-signed char lin2alaw[65536];	// 16bit unsigned index
-signed char *slin2alaw;		// 16bit signed index
-uint16_t alaw2lin[256];		// alaw -> 16bit PCM, Mono, 8000 hz
+extern signed char lin2alaw[65536];	// 16bit unsigned index
+extern signed char *slin2alaw;		// 16bit signed index
+extern uint16_t alaw2lin[256];		// alaw -> 16bit PCM, Mono, 8000 hz
 
 void create_lin2alaw_table(void);
 #endif
