Index: peazip-8.6.0.src/dev/project_peach.lpi
===================================================================
--- peazip-8.6.0.src.orig/dev/project_peach.lpi
+++ peazip-8.6.0.src/dev/project_peach.lpi
@@ -1588,7 +1588,7 @@
     </CodeGeneration>
     <Linking>
       <Debugging>
-        <GenerateDebugInfo Value="False"/>
+        <GenerateDebugInfo Value="True"/>
       </Debugging>
       <Options>
         <PassLinkerOptions Value="True"/>
Index: peazip-8.6.0.src/dev/project_pea.lpi
===================================================================
--- peazip-8.6.0.src.orig/dev/project_pea.lpi
+++ peazip-8.6.0.src/dev/project_pea.lpi
@@ -548,7 +548,7 @@
     </CodeGeneration>
     <Linking>
       <Debugging>
-        <GenerateDebugInfo Value="False"/>
+        <GenerateDebugInfo Value="True"/>
         <DebugInfoType Value="dsDwarf3"/>
       </Debugging>
       <Options>
