--- peazip-8.3.0.src/peach.pas.orig	2021-11-17 08:05:28.276986802 +0100
+++ peazip-8.3.0.src/peach.pas	2021-11-17 08:05:58.097106865 +0100
@@ -41412,7 +41412,7 @@
 
 procedure TForm_peach.baboutlocalhelpClick(Sender: TObject);
 begin
-cp_open(sharepath+'peazip_help.pdf',desk_env);
+cp_open('/usr/share/doc/packages/peazip/peazip_help.pdf',desk_env);
 end;
 
 procedure TForm_peach.baboutremoveadditionalClick(Sender: TObject);
@@ -63622,7 +63622,7 @@
 
 procedure TForm_peach.mpeahelpofflineClick(Sender: TObject);
 begin
-cp_open(sharepath+'peazip_help.pdf',desk_env);
+cp_open('/usr/share/doc/packages/peazip/peazip_help.pdf',desk_env);
 end;
 
 procedure TForm_peach.mRecent9Click(Sender: TObject);
