Index: peazip-10.8.0.src/dev/project_peach.lpi
===================================================================
--- peazip-10.8.0.src.orig/dev/project_peach.lpi
+++ peazip-10.8.0.src/dev/project_peach.lpi
@@ -1753,7 +1753,7 @@
     </CodeGeneration>
     <Linking>
       <Debugging>
-        <GenerateDebugInfo Value="False"/>
+        <GenerateDebugInfo Value="True"/>
         <RunWithoutDebug Value="True"/>
         <DebugInfoType Value="dsDwarf3"/>
       </Debugging>
