Index: peazip-9.9.0.src/dev/project_pea.lpi
===================================================================
--- peazip-9.9.0.src.orig/dev/project_pea.lpi
+++ peazip-9.9.0.src/dev/project_pea.lpi
@@ -566,7 +566,7 @@
     </CodeGeneration>
     <Linking>
       <Debugging>
-        <GenerateDebugInfo Value="False"/>
+        <GenerateDebugInfo Value="True"/>
         <DebugInfoType Value="dsDwarf3"/>
       </Debugging>
       <Options>
Index: peazip-9.9.0.src/dev/project_peach.lpi
===================================================================
--- peazip-9.9.0.src.orig/dev/project_peach.lpi
+++ peazip-9.9.0.src/dev/project_peach.lpi
@@ -1650,7 +1650,7 @@
     </CodeGeneration>
     <Linking>
       <Debugging>
-        <GenerateDebugInfo Value="False"/>
+        <GenerateDebugInfo Value="True"/>
       </Debugging>
       <Options>
         <PassLinkerOptions Value="True"/>
