---
 include/f2fs_fs.h |    1 +
 1 file changed, 1 insertion(+)

Index: f2fs-tools-1.7.0/include/f2fs_fs.h
===================================================================
--- f2fs-tools-1.7.0.orig/include/f2fs_fs.h
+++ f2fs-tools-1.7.0/include/f2fs_fs.h
@@ -95,6 +95,7 @@ static inline uint64_t bswap_64(uint64_t
 #define cpu_to_le32(x)	((__u32)(x))
 #define cpu_to_le64(x)	((__u64)(x))
 #elif __BYTE_ORDER == __BIG_ENDIAN
+#include <byteswap.h>
 #define le16_to_cpu(x)	bswap_16(x)
 #define le32_to_cpu(x)	bswap_32(x)
 #define le64_to_cpu(x)	bswap_64(x)
