diff -urEbwB duplicity-0.8.09.orig/bin/duplicity duplicity-0.8.09/bin/duplicity
--- duplicity-0.8.09.orig/bin/duplicity	2020-01-07 13:31:22.000000000 -0600
+++ duplicity-0.8.09/bin/duplicity	2020-01-07 19:39:41.533829779 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 # -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
 #
 # duplicity -- Encrypted bandwidth efficient backup
diff -urEbwB duplicity-0.8.09.orig/bin/rdiffdir duplicity-0.8.09/bin/rdiffdir
--- duplicity-0.8.09.orig/bin/rdiffdir	2020-01-07 13:31:22.000000000 -0600
+++ duplicity-0.8.09/bin/rdiffdir	2020-01-07 19:39:48.157864751 -0600
@@ -1,2 +1,2 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 # rdiffdir -- Extend rdiff functionality to directories
diff -urEbwB duplicity-0.8.09.orig/compilec.py duplicity-0.8.09/compilec.py
--- duplicity-0.8.09.orig/compilec.py	2019-12-28 15:19:23.000000000 -0600
+++ duplicity-0.8.09/compilec.py	2020-01-07 19:39:24.241738498 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 # -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
 #
 # Copyright 2002 Ben Escoto <ben@emerose.org>
