diff -urp duplicity-1.0.0.orig/bin/duplicity duplicity-1.0.0/bin/duplicity
--- duplicity-1.0.0.orig/bin/duplicity	2022-09-24 10:26:49.000000000 -0500
+++ duplicity-1.0.0/bin/duplicity	2022-09-29 18:37:13.118715845 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 # -*- Mode:Python; indent-tabs-mode:nil; tab-width:4; encoding:utf-8 -*-
 #
 # duplicity -- Encrypted bandwidth efficient backup
diff -urp duplicity-1.0.0.orig/bin/rdiffdir duplicity-1.0.0/bin/rdiffdir
--- duplicity-1.0.0.orig/bin/rdiffdir	2022-09-24 10:26:49.000000000 -0500
+++ duplicity-1.0.0/bin/rdiffdir	2022-09-29 18:37:19.610750394 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 # -*- Mode:Python; indent-tabs-mode:nil; tab-width:4; encoding:utf-8 -*-
 #
 # rdiffdir -- Extend rdiff functionality to directories
