--- common/backintime.orig
+++ common/backintime
@@ -24,4 +24,7 @@ else
 	APP_PATH=$(readlink -m "${CUR_PATH}/../share/backintime/common")
 fi
 
+export RSYNC_OLD_ARGS=1
+export RSYNC_PROTECT_ARGS=0
+
 python3 -Es $APP_PATH/backintime.py "$@"
--- qt/backintime-qt.orig
+++ qt/backintime-qt
@@ -28,4 +28,7 @@ else
 	APP_PATH=$(readlink -m "${CUR_PATH}/../share/backintime/qt")
 fi
 
+export RSYNC_OLD_ARGS=1
+export RSYNC_PROTECT_ARGS=0
+
 python3 -Es ${APP_PATH}/app.py "$@"
--- qt/backintime-qt_polkit.orig
+++ qt/backintime-qt_polkit
@@ -6,4 +6,7 @@ else
     PREFIX=""
 fi
 
+export RSYNC_OLD_ARGS=1
+export RSYNC_PROTECT_ARGS=0
+
 pkexec --disable-internal-agent $PREFIX "/usr/bin/backintime-qt" "$@"
