--- qt/net.launchpad.backintime.policy 2016-01-11 19:07:58.000000000 +0100
+++ qt/net.launchpad.backintime.policy 2017-03-29 13:45:08.384966405 +0200
@@ -14,7 +14,7 @@ 
     <defaults>
       <allow_any>auth_admin</allow_any>
       <allow_inactive>auth_admin</allow_inactive>
-      <allow_active>auth_admin_keep</allow_active>
+      <allow_active>auth_admin</allow_active>
     </defaults>
     <annotate key="org.freedesktop.policykit.exec.path">/usr/bin/backintime-qt</annotate>
     <annotate key="org.freedesktop.policykit.exec.allow_gui">true</annotate>
@@ -25,8 +25,8 @@ 
     <description gettext-domain="backintime">This will install Udev rules which will start Back In Time if a drive get connected.</description>
     <defaults>
       <allow_any>auth_admin</allow_any>
-      <allow_inactive>auth_admin_keep</allow_inactive>
-      <allow_active>auth_admin_keep</allow_active>
+      <allow_inactive>auth_admin</allow_inactive>
+      <allow_active>auth_admin</allow_active>
     </defaults>
   </action>
 
@@ -35,8 +35,8 @@ 
     <description gettext-domain="backintime">This will delete Udev rules.</description>
     <defaults>
       <allow_any>auth_admin</allow_any>
-      <allow_inactive>auth_admin_keep</allow_inactive>
-      <allow_active>auth_admin_keep</allow_active>
+      <allow_inactive>auth_admin</allow_inactive>
+      <allow_active>auth_admin</allow_active>
     </defaults>
   </action>
