Index: tinc-1.0.36/systemd/tinc@.service.in
===================================================================
--- tinc-1.0.36.orig/systemd/tinc@.service.in
+++ tinc-1.0.36/systemd/tinc@.service.in
@@ -7,6 +7,19 @@ PartOf=tinc.service
 ReloadPropagatedFrom=tinc.service
 
 [Service]
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+PrivateDevices=true
+ProtectHostname=true
+ProtectClock=true
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+ProtectControlGroups=true
+RestrictRealtime=true
+# end of automatic additions 
 Type=simple
 WorkingDirectory=@sysconfdir@/tinc/%i
 ExecStart=@sbindir@/tincd -n %i -D
