diff -Nur module-ssh2-release-0.9.15/CMakeLists.txt new/CMakeLists.txt
--- module-ssh2-release-0.9.15/CMakeLists.txt	2021-05-06 11:23:18.000000000 +0200
+++ new/CMakeLists.txt	2021-07-29 10:05:54.288042931 +0200
@@ -84,6 +84,7 @@
 )
 
 add_library(${module_name} SHARED ${CPP_SRC} ${QPP_SOURCES})
+target_link_libraries(${module_name} ${CMAKE_THREAD_LIBS_INIT})
 set(MODULE_DOX_INPUT ${CMAKE_BINARY_DIR}/mainpage.dox ${_dox_src})
 qore_external_binary_module(${module_name} ${PROJECT_VERSION} ${LIBSSH2_LDFLAGS})
 
