--- clients-desktop-v2024.11.0/apps/desktop/src/main/native-messaging.main.ts.orig	2024-11-14 21:24:39.217775953 +0100
+++ clients-desktop-v2024.11.0/apps/desktop/src/main/native-messaging.main.ts	2024-11-14 21:37:48.932452741 +0100
@@ -399,26 +399,7 @@ export class NativeMessagingMain {
   }
 
   private binaryPath() {
-    const ext = process.platform === "win32" ? ".exe" : "";
-
-    if (isDev()) {
-      const devPath = path.join(
-        this.appPath,
-        "..",
-        "desktop_native",
-        "target",
-        "debug",
-        `desktop_proxy${ext}`,
-      );
-
-      // isDev() returns true when using a production build with ELECTRON_IS_DEV=1,
-      // so we need to fall back to the prod binary if the dev binary doesn't exist.
-      if (existsSync(devPath)) {
-        return devPath;
-      }
-    }
-
-    return path.join(path.dirname(this.exePath), `desktop_proxy${ext}`);
+    return 'XXXLIBDIRXXX/bitwarden/desktop_proxy'
   }
 
   private homedir() {
