--- clients/tsconfig.base.json	2025-07-02 20:03:23.998679344 +0200
+++ bitwarden-2025.6.1/tsconfig.base.json	2025-07-02 21:23:52.860744260 +0200
@@ -15,7 +15,8 @@
     "baseUrl": ".",
     "resolveJsonModule": true,
     "allowJs": true,
-    "sourceMap": true,
+    "sourceMap": false,
+    "removeComments": true,
     "skipLibCheck": true,
     "paths": {
       "@bitwarden/admin-console/common": ["./libs/admin-console/src/common"],
--- clients/apps/desktop/webpack.renderer.js	2025-07-02 20:03:23.070679456 +0200
+++ bitwarden-2025.6.1/apps/desktop/webpack.renderer.js	2025-07-02 21:27:05.376723459 +0200
@@ -66,7 +61,7 @@ const common = {
 
 const renderer = {
   mode: NODE_ENV,
-  devtool: "source-map",
+  devtool: false,
   target: "web",
   node: {
     __dirname: false,
@@ -169,7 +159,7 @@ const renderer = {
     new AngularWebpackPlugin({
       tsConfigPath: "tsconfig.renderer.json",
       entryModule: "src/app/app.module#AppModule",
-      sourceMap: true,
+      sourceMap: false,
     }),
     // ref: https://github.com/angular/angular/issues/20357
     new webpack.ContextReplacementPlugin(
@@ -181,9 +171,6 @@ const renderer = {
       filename: "index.html",
       chunks: ["app/vendor", "app/main"],
     }),
-    new webpack.SourceMapDevToolPlugin({
-      include: ["app/main.js"],
-    }),
     new MiniCssExtractPlugin({
       filename: "[name].[contenthash].css",
       chunkFilename: "[id].[contenthash].css",
