diff --git a/mjpg-streamer-experimental/mjpg_streamer@.service b/mjpg-streamer-experimental/mjpg_streamer@.service
index 1632969..6cef6d0 100644
--- a/mjpg-streamer-experimental/mjpg_streamer@.service
+++ b/mjpg-streamer-experimental/mjpg_streamer@.service
@@ -4,7 +4,8 @@ After=network.target
 
 [Service]
 User=mjpg_streamer
-ExecStart=/usr/bin/mjpg_streamer -i 'input_uvc.so -d /dev/%I' -o 'output_http.so -w /usr/share/mjpg_streamer/www'
+Group=video
+ExecStart=/usr/bin/mjpg_streamer -i 'input_uvc.so -d /dev/video%I' -o 'output_http.so -w /usr/share/mjpg-streamer/www -p 808%I'
 
 [Install]
 WantedBy=multi-user.target
