# HG changeset patch
# Parent 20a1a6ad46d5a4ec83d9800614fc288bf79e14a8

diff --git a/build/autoconf/icu.m4 b/build/autoconf/icu.m4
--- a/build/autoconf/icu.m4
+++ b/build/autoconf/icu.m4
@@ -73,17 +73,17 @@ if test -n "$USE_ICU"; then
     if test x"$version" = x; then
        AC_MSG_ERROR([cannot determine icu version number from uvernum.h header file $lineno])
     fi
     MOZ_ICU_VERSION="$version"
 
     # TODO: the l is actually endian-dependent
     # We could make this set as 'l' or 'b' for little or big, respectively,
     # but we'd need to check in a big-endian version of the file.
-    ICU_DATA_FILE="icudt${version}l.dat"
+    ICU_DATA_FILE="icudt${version}b.dat"
 
     MOZ_ICU_DATA_ARCHIVE=
 fi
 
 AC_SUBST(MOZ_ICU_VERSION)
 AC_SUBST(ENABLE_INTL_API)
 AC_SUBST(USE_ICU)
 AC_SUBST(ICU_DATA_FILE)
