From fe24ac16eadd0cf61d8d868f9475eea059f9056b Mon Sep 17 00:00:00 2001
From: Michael Vetter <jubalh@iodoru.org>
Date: Tue, 16 Nov 2021 15:19:14 +0100
Subject: [PATCH] Mention state of drivers

---
 Drivers/README | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Drivers/README b/Drivers/README
index 9b59fa5..dde0c71 100644
--- a/Drivers/README
+++ b/Drivers/README
@@ -8,6 +8,13 @@
 | Drivers                                                     |
 +-------------------------------------------------------------+
 
+The drivers in here are historical ones or example drivers.
+They are not supported anymore. But can be used as a starting
+point to write your own.
+
+A list of some third party drivers can be found at:
+http://www.unixodbc.org/drivers.html
+
 Some unixODBC drivers have been coded from scratch while others
 have been integerated (tweeked to compile and tested) into
 unixODBC.
