From f1712186b932581ca8010713565ecdf6f09a2496 Mon Sep 17 00:00:00 2001
From: Michael Vetter <jubalh@iodoru.org>
Date: Tue, 16 Nov 2021 15:08:15 +0100
Subject: [PATCH] readme: add github and website links

---
 README | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/README b/README
index 39a259f..07b8fd6 100644
--- a/README
+++ b/README
@@ -52,3 +52,5 @@ Config Files:
 Resources:
 
         http://sourceforge.net/projects/unixodbc/
+        https://github.com/lurcher/unixODBC/
+        http://www.unixodbc.org/
