Index: unixODBC-2.3.13/Drivers/nn/yyparse.y
===================================================================
--- unixODBC-2.3.13.orig/Drivers/nn/yyparse.y
+++ unixODBC-2.3.13/Drivers/nn/yyparse.y
@@ -40,7 +40,7 @@ static	char	sccsid[]
 # define yylex()		nnsql_yylex(&yylval, pyyenv)
 #endif
 union YYSTYPE;
-
+int nnsql_yylex(union YYSTYPE* pyylval, yyenv_t* penv);
 
 #define yyparse(x)		nnsql_yyparse	(yyenv_t* pyyenv)
 #define yyerror(msg)		nnsql_yyerror (pyyenv, msg)
