--- src/wdatutil.c
+++ src/wdatutil.c
@@ -269,13 +269,13 @@
 /*
  * Fixed data and tables
  */
- 	fprintf (wdat_fp, "unsigned long ref_count = 0;\n");
-	fprintf (wdat_fp, "unsigned long first_sample_rate = 0;\n");
-	fprintf (wdat_fp, "unsigned long table_count = %ld;\n", table_count);
-	fprintf (wdat_fp, "Wavetable tables[%ld];\n", table_count);
-	fprintf (wdat_fp, "Wavetable * ptables[%ld];\n", table_count);
-	fprintf (wdat_fp, "unsigned long lookup[%ld];\n", w->lookup_max + 1);
-	fprintf (wdat_fp, "unsigned long lookup_max = %ld;\n", w->lookup_max);
+ 	fprintf (wdat_fp, "static unsigned long ref_count = 0;\n");
+	fprintf (wdat_fp, "static unsigned long first_sample_rate = 0;\n");
+	fprintf (wdat_fp, "static unsigned long table_count = %ld;\n", table_count);
+	fprintf (wdat_fp, "static Wavetable tables[%ld];\n", table_count);
+	fprintf (wdat_fp, "static Wavetable * ptables[%ld];\n", table_count);
+	fprintf (wdat_fp, "static unsigned long lookup[%ld];\n", w->lookup_max + 1);
+	fprintf (wdat_fp, "static unsigned long lookup_max = %ld;\n", w->lookup_max);
 	fprintf (wdat_fp, "\n");
 /*
  * Sample data
@@ -326,7 +326,7 @@
 		fprintf (wdat_fp, "\n");
 	}
 
-	fprintf (wdat_fp, "LADSPA_Data samples_zero[%ld];\n", t->sample_count + 3);
+	fprintf (wdat_fp, "static LADSPA_Data samples_zero[%ld];\n", t->sample_count + 3);
 	fprintf (wdat_fp, "\n");
 /*
  * Function to get Wavedata - the sample rate is needed to calculate
