Index: system-config-printer-1.4.5/print-applet.desktop.in
===================================================================
--- system-config-printer-1.4.5.orig/print-applet.desktop.in
+++ system-config-printer-1.4.5/print-applet.desktop.in
@@ -5,6 +5,6 @@ Exec=system-config-printer-applet
 Terminal=false
 Type=Application
 Icon=system-config-printer
-NotShowIn=KDE;
+NotShowIn=KDE;GNOME;
 StartupNotify=false
 X-GNOME-Autostart-Delay=30
