Index: policycoreutils-3.4-rc3/run_init/run_init.pamd
===================================================================
--- policycoreutils-3.4-rc3.orig/run_init/run_init.pamd
+++ policycoreutils-3.4-rc3/run_init/run_init.pamd
@@ -1,8 +1,9 @@
 #%PAM-1.0
 # Uncomment the next line if you do not want to enter your passwd every time
 #auth       sufficient   pam_rootok.so
-auth       include	system-auth
-account    include	system-auth
-password   include	system-auth
-session    include	system-auth
+auth       include      common-auth
+account    include      common-account
+password   include      common-password
+session    include      common-session
 session    optional	pam_xauth.so
+session    optional     pam_keyinit.so revoke [force]
