---
 Lib/site.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: Python-3.15.0a7/Lib/site.py
===================================================================
--- Python-3.15.0a7.orig/Lib/site.py	2026-03-10 21:55:34.774030935 +0100
+++ Python-3.15.0a7/Lib/site.py	2026-03-10 21:55:45.838165991 +0100
@@ -80,7 +80,7 @@
 import errno
 
 # Prefixes for site-packages; add additional prefixes like /usr/local here
-PREFIXES = [sys.prefix, sys.exec_prefix]
+PREFIXES = [sys.prefix, sys.exec_prefix, '/usr/local']
 # Enable per user site-packages directory
 # set it to False to disable the feature or True to force the feature
 ENABLE_USER_SITE = None
