---
 Lib/test/test_subprocess.py |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

Index: Python-3.15.0b4/Lib/test/test_subprocess.py
===================================================================
--- Python-3.15.0b4.orig/Lib/test/test_subprocess.py	2026-07-30 17:01:51.784137045 +0200
+++ Python-3.15.0b4/Lib/test/test_subprocess.py	2026-07-30 17:02:03.865301397 +0200
@@ -161,7 +161,11 @@
         # child.
         self.assertRaises(subprocess.TimeoutExpired, subprocess.call,
                           [sys.executable, "-c", "while True: pass"],
-                          timeout=0.1)
+                          # Some heavily loaded buildbots (sparc Debian 3.x) require
+                          # this much time to start and print.
+                          # timeout=3)
+                          # OBS might require even more
+                          timeout=10)
 
     def test_timeout_exception(self):
         try:
