#! /bin/sh /usr/share/dpatch/dpatch-run
## detect-imagick.dpatch by Matej Vela <vela@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: wvDVI.in, wvPDF.in, wvPS.in: Fix ImageMagick check (#160925).

@DPATCH@
diff -urNad wv-1.2.1~/wvDVI.in wv-1.2.1/wvDVI.in
--- wv-1.2.1~/wvDVI.in	2002-02-18 09:08:35.000000000 +0100
+++ wv-1.2.1/wvDVI.in	2006-03-18 00:45:10.000000000 +0100
@@ -153,7 +153,7 @@
 # auto_jpg=`"$wv_exec" --auto-eps=jpg`
 # auto_pict=`"$wv_exec" --auto-eps=pict`
 
-im_chk=`convert --help 2>&1 | grep -v ImageMagick`
+im_chk=`convert --help 2>&1 | grep ImageMagick`
 if test "x$im_chk" = "x"; then
   im_chk="no"
   s_img=""
diff -urNad wv-1.2.1~/wvPDF.in wv-1.2.1/wvPDF.in
--- wv-1.2.1~/wvPDF.in	2004-08-07 22:49:28.000000000 +0200
+++ wv-1.2.1/wvPDF.in	2006-03-18 00:45:07.000000000 +0100
@@ -177,7 +177,7 @@
 # auto_jpg=`"$wv_exec" --auto-eps=jpg`
 # auto_pict=`"$wv_exec" --auto-eps=pict`
 
-im_chk=`convert --help 2>&1 | grep -v ImageMagick`
+im_chk=`convert --help 2>&1 | grep ImageMagick`
 if test "x$im_chk" = "x"; then
   im_chk="no"
   s_img=""
diff -urNad wv-1.2.1~/wvPS.in wv-1.2.1/wvPS.in
--- wv-1.2.1~/wvPS.in	2004-08-07 22:49:28.000000000 +0200
+++ wv-1.2.1/wvPS.in	2006-03-18 00:44:53.000000000 +0100
@@ -159,7 +159,7 @@
 # auto_jpg=`"$wv_exec" --auto-eps=jpg`
 # auto_pict=`"$wv_exec" --auto-eps=pict`
 
-im_chk=`convert --help 2>&1 | grep -v ImageMagick`
+im_chk=`convert --help 2>&1 | grep ImageMagick`
 if test "x$im_chk" = "x"; then
   im_chk="no"
   s_img=""
