Index: scmon-0.4/init/scmon.service
===================================================================
--- /dev/null
+++ scmon-0.4/init/scmon.service
@@ -0,0 +1,16 @@
+[Unit]
+Description = Smart Card Monitor
+Documentation = http://code.google.com/p/loolixbodes/wiki/scmon
+Requires=pcscd.socket
+Requires=dbus.socket
+Requires=remote-fs.target
+
+[Service]
+Type=forking
+PIDFile=/var/run/scmon.pid
+ExecStart=/usr/sbin/scmon
+ExecStop=/sbin/killproc -TERM /usr/sbin/scmon
+ExecReload=/sbin/killproc -TERM /usr/sbin/scmon && /usr/sbin/scmon
+
+[Install]
+WantedBy=multi-user.target
