From: Jan Engelhardt <jengelh@medozas.de>
Date: 2012-03-25 16:17:03.379573514 +0200

---
 BlockOut/Makefile.am |    1 +
 BlockOut/Utils.cpp   |    2 ++
 2 files changed, 3 insertions(+)

Index: BL_SRC/BlockOut/Makefile.am
===================================================================
--- BL_SRC.orig/BlockOut/Makefile.am
+++ BL_SRC/BlockOut/Makefile.am
@@ -1,6 +1,7 @@
 # -*- Makefile -*-
 
 AM_CPPFLAGS = -D_DEBUG -Dlinux -I../ImageLib/src -I. \
+              -DBL2_HOME=\"${pkgdatadir}\" \
               ${libSDL_CFLAGS} ${libSDL_mixer_CFLAGS}
 
 bin_PROGRAMS = blockout
Index: BL_SRC/BlockOut/Utils.cpp
===================================================================
--- BL_SRC.orig/BlockOut/Utils.cpp
+++ BL_SRC/BlockOut/Utils.cpp
@@ -208,6 +208,8 @@ BOOL CheckEnv() {
   }
     
   char *blockoutHome = getenv("BL2_HOME");
+  if (blockoutHome == NULL)
+	blockoutHome = BL2_HOME;
   if( blockoutHome==NULL ) {
     printf("BL2_HOME environement variable if not defined !\n");
     printf("Please set the BL2_HOME to the BlockOut II installation directory (ex: BL2_HOME=/usr/local/bl2).\n");
