Index: b/acinclude/ac_with_cclient.m4
===================================================================
--- a/acinclude/ac_with_cclient.m4
+++ b/acinclude/ac_with_cclient.m4
@@ -126,7 +126,7 @@ AC_DEFUN(AC_WITH_CCLIENT,[
 	 void mm_nocritical(MAILSTREAM*a){}
 	 void mm_notify(MAILSTREAM*a,char*b,long c){}
 	 void mm_login(NETMBX*a,char*b,char*c,long d){}
-	 long mm_diskerror(MAILSTREAM*a,long b,long c){}
+	 long mm_diskerror(MAILSTREAM*a,long b,long c){ return 0; }
 	 void mm_status(MAILSTREAM*a,char*b,MAILSTATUS*c){}
 	 void mm_lsub(MAILSTREAM*a,int b,char*c,long d){}
 	 void mm_list(MAILSTREAM*a,int b,char*c,long d){}
@@ -166,7 +166,7 @@ AC_DEFUN(AC_WITH_CCLIENT,[
 	   void mm_nocritical(MAILSTREAM*a){}
 	   void mm_notify(MAILSTREAM*a,char*b,long c){}
 	   void mm_login(NETMBX*a,char*b,char*c,long d){}
-	   long mm_diskerror(MAILSTREAM*a,long b,long c){}
+	   long mm_diskerror(MAILSTREAM*a,long b,long c){ return 0; }
 	   void mm_status(MAILSTREAM*a,char*b,MAILSTATUS*c){}
 	   void mm_lsub(MAILSTREAM*a,int b,char*c,long d){}
 	   void mm_list(MAILSTREAM*a,int b,char*c,long d){}
Index: b/acinclude/ac_with_md5.m4
===================================================================
--- a/acinclude/ac_with_md5.m4
+++ b/acinclude/ac_with_md5.m4
@@ -38,7 +38,7 @@ AC_DEFUN(AC_WITH_MD5,[
     void mm_nocritical(MAILSTREAM*a){}
     void mm_notify(MAILSTREAM*a,char*b,long c){}
     void mm_login(NETMBX*a,char*b,char*c,long d){}
-    long mm_diskerror(MAILSTREAM*a,long b,long c){}
+    long mm_diskerror(MAILSTREAM*a,long b,long c){ return 0; }
     void mm_status(MAILSTREAM*a,char*b,MAILSTATUS*c){}
     void mm_lsub(MAILSTREAM*a,int b,char*c,long d){}
     void mm_list(MAILSTREAM*a,int b,char*c,long d){}
