Index: qimgv-0.9/qimgv/3rdparty/QtOpenCV/cvmatandqimage.cpp
===================================================================
--- qimgv-0.9.orig/qimgv/3rdparty/QtOpenCV/cvmatandqimage.cpp
+++ qimgv-0.9/qimgv/3rdparty/QtOpenCV/cvmatandqimage.cpp
@@ -28,7 +28,7 @@
 #include <QSysInfo>
 #include <QDebug>
 #include <cstring>
-#include <opencv4/opencv2/imgproc.hpp>
+#include <opencv2/imgproc.hpp>
 
 namespace QtOcv {
 namespace {
Index: qimgv-0.9/qimgv/3rdparty/QtOpenCV/cvmatandqimage.h
===================================================================
--- qimgv-0.9.orig/qimgv/3rdparty/QtOpenCV/cvmatandqimage.h
+++ qimgv-0.9/qimgv/3rdparty/QtOpenCV/cvmatandqimage.h
@@ -27,7 +27,7 @@
 #define CVMATANDQIMAGE_H
 
 #include <QtGui/qimage.h>
-#include <opencv4/opencv2/core.hpp>
+#include <opencv2/core.hpp>
 
 namespace QtOcv {
 
