Index: newt-0.52.20/dialogboxes.c
===================================================================
--- newt-0.52.20.orig/dialogboxes.c
+++ newt-0.52.20/dialogboxes.c
@@ -1,4 +1,5 @@
 /* simple dialog boxes, used by both whiptail and tcl dialog bindings */
+#define _XOPEN_SOURCE 700
 
 #include "config.h"
 #include <fcntl.h>
@@ -8,6 +9,11 @@
 #include <unistd.h>
 #include <wchar.h>
 #include <slang.h>
+#include <strings.h>
+
+#ifdef HAVE_ALLOCA_H
+#include <alloca.h>
+#endif
 
 #include "nls.h"
 #include "dialogboxes.h"
Index: newt-0.52.20/newt.c
===================================================================
--- newt-0.52.20.orig/newt.c
+++ newt-0.52.20/newt.c
@@ -1,3 +1,4 @@
+#define _XOPEN_SOURCE 700
 #include "config.h"
 
 #include <slang.h>
Index: newt-0.52.20/textbox.c
===================================================================
--- newt-0.52.20.orig/textbox.c
+++ newt-0.52.20/textbox.c
@@ -1,3 +1,4 @@
+#define _XOPEN_SOURCE 700
 
 #include <ctype.h>
 #include <slang.h>
