diff -upr magic-wormhole-mailbox-server-0.4.1.orig/src/wormhole_mailbox_server/test/test_rlimits.py magic-wormhole-mailbox-server-0.4.1/src/wormhole_mailbox_server/test/test_rlimits.py
--- magic-wormhole-mailbox-server-0.4.1.orig/src/wormhole_mailbox_server/test/test_rlimits.py	2022-05-17 17:46:58.549457994 +0200
+++ magic-wormhole-mailbox-server-0.4.1/src/wormhole_mailbox_server/test/test_rlimits.py	2022-05-17 17:46:58.553458020 +0200
@@ -1,5 +1,5 @@
 from __future__ import print_function, unicode_literals
-import mock
+from unittest import mock
 from twisted.trial import unittest
 from ..increase_rlimits import increase_rlimits
 
diff -upr magic-wormhole-mailbox-server-0.4.1.orig/src/wormhole_mailbox_server/test/test_server.py magic-wormhole-mailbox-server-0.4.1/src/wormhole_mailbox_server/test/test_server.py
--- magic-wormhole-mailbox-server-0.4.1.orig/src/wormhole_mailbox_server/test/test_server.py	2022-05-17 17:46:58.549457994 +0200
+++ magic-wormhole-mailbox-server-0.4.1/src/wormhole_mailbox_server/test/test_server.py	2022-05-17 17:46:58.553458020 +0200
@@ -1,5 +1,5 @@
 from __future__ import print_function, unicode_literals
-import mock
+from unittest import mock
 from twisted.trial import unittest
 from twisted.python import log
 from .common import ServerBase, _Util
diff -upr magic-wormhole-mailbox-server-0.4.1.orig/src/wormhole_mailbox_server/test/test_service.py magic-wormhole-mailbox-server-0.4.1/src/wormhole_mailbox_server/test/test_service.py
--- magic-wormhole-mailbox-server-0.4.1.orig/src/wormhole_mailbox_server/test/test_service.py	2022-05-17 17:46:58.549457994 +0200
+++ magic-wormhole-mailbox-server-0.4.1/src/wormhole_mailbox_server/test/test_service.py	2022-05-17 17:46:58.553458020 +0200
@@ -1,6 +1,6 @@
 from __future__ import unicode_literals, print_function
 from twisted.trial import unittest
-import mock
+from unittest import mock
 from twisted.application.service import MultiService
 from .. import server_tap
 
diff -upr magic-wormhole-mailbox-server-0.4.1.orig/src/wormhole_mailbox_server/test/test_web.py magic-wormhole-mailbox-server-0.4.1/src/wormhole_mailbox_server/test/test_web.py
--- magic-wormhole-mailbox-server-0.4.1.orig/src/wormhole_mailbox_server/test/test_web.py	2022-05-17 17:46:58.549457994 +0200
+++ magic-wormhole-mailbox-server-0.4.1/src/wormhole_mailbox_server/test/test_web.py	2022-05-17 17:46:58.553458020 +0200
@@ -1,6 +1,6 @@
 from __future__ import print_function, unicode_literals
 import io, time
-import mock
+from unittest import mock
 import treq
 from twisted.trial import unittest
 from twisted.internet import defer, reactor
