Index: caps-0.9.26/Makefile
===================================================================
--- caps-0.9.26.orig/Makefile
+++ caps-0.9.26/Makefile
@@ -18,7 +18,7 @@ LDFLAGS += $(_LDFLAGS) $(CFLAGS)
 
 PLUG = caps
 
-SOURCES = $(wildcard *.cc) $(wildcard dsp/*.cc)
+SOURCES = $(sort $(wildcard *.cc)) $(sort($(wildcard dsp/*.cc))
 OBJECTS	= $(SOURCES:.cc=.o) 
 HEADERS = $(wildcard *.h) $(wildcard dsp/*.h) $(wildcard util/*.h) $(wildcard dsp/tonestack/*.h)
 
