diff -Nru msal_extensions-1.2.0.orig/setup.py msal_extensions-1.2.0/setup.py
--- msal_extensions-1.2.0.orig/setup.py	2024-06-23 04:15:27.000000000 +0200
+++ msal_extensions-1.2.0/setup.py	2024-11-22 11:05:40.219183804 +0100
@@ -20,7 +20,7 @@
     python_requires=">=3.7",
     install_requires=[
         'msal>=1.29,<2',  # Use TokenCache.search() from MSAL Python 1.29+
-        'portalocker<3,>=1.4',
+        'portalocker<4,>=1.4',
 
         ## We choose to NOT define a hard dependency on this.
         # "pygobject>=3,<4;platform_system=='Linux'",
