diff -Nru sbcl-2.0.2.orig/src/runtime/Config.arm-linux sbcl-2.0.2/src/runtime/Config.arm-linux
--- sbcl-2.0.2.orig/src/runtime/Config.arm-linux	2020-02-29 21:25:50.000000000 +0100
+++ sbcl-2.0.2/src/runtime/Config.arm-linux	2020-03-13 13:53:38.156922974 +0100
@@ -9,7 +9,7 @@
 # provided with absolutely no warranty. See the COPYING and CREDITS
 # files for more information.
 
-CFLAGS += -marm -march=armv5
+CFLAGS += -marm
 NM = ./linux-nm
 
 ASSEM_SRC = arm-assem.S
