diff -Nru sbcl-2.0.10.orig/src/runtime/Config.arm-linux sbcl-2.0.10/src/runtime/Config.arm-linux
--- sbcl-2.0.10.orig/src/runtime/Config.arm-linux	2020-10-27 14:44:50.000000000 +0100
+++ sbcl-2.0.10/src/runtime/Config.arm-linux	2020-10-29 12:42:20.394163187 +0100
@@ -9,7 +9,7 @@
 # provided with absolutely no warranty. See the COPYING and CREDITS
 # files for more information.
 
-CFLAGS += -marm -march=armv5
+CFLAGS += -marm
 
 ASSEM_SRC = arm-assem.S
 ARCH_SRC = arm-arch.c
