diff -Nur startdde-5.9.49/main.go startdde-5.9.49-new/main.go
--- startdde-5.9.49/main.go	2022-08-17 12:21:36.000000000 +0800
+++ startdde-5.9.49-new/main.go	2022-10-27 21:30:23.915940930 +0800
@@ -91,12 +91,12 @@
 }
 
 func shouldUseDDEKWin() bool {
-	_, err := os.Stat("/usr/bin/kwin_no_scale")
+	_, err := os.Stat("/usr/bin/deepin-kwin_x11")
 	return err == nil
 }
 
 const (
-	cmdKWin                = "/usr/bin/kwin_no_scale"
+	cmdKWin                = "/usr/bin/deepin-kwin_x11"
 	cmdDdeSessionDaemon    = "/usr/lib/deepin-daemon/dde-session-daemon"
 	cmdDdeDock             = "/usr/bin/dde-dock"
 	cmdDdeDesktop          = "/usr/bin/dde-desktop"
diff -Nur startdde-5.9.49/main_test.go startdde-5.9.49-new/main_test.go
--- startdde-5.9.49/main_test.go	2022-08-17 12:21:36.000000000 +0800
+++ startdde-5.9.49-new/main_test.go	2022-10-27 21:30:23.915940930 +0800
@@ -28,7 +28,7 @@
 func Test_ShouldUseDDEKWin(t *testing.T) {
 	t.Run("Test is should use DDE KWin", func(t *testing.T) {
 		should := shouldUseDDEKWin()
-		exist := Exist("/usr/bin/kwin_no_scale")
+		exist := Exist("/usr/bin/deepin-kwin_x11")
 		assert.Equal(t, exist, should)
 	})
 }
diff -Nur startdde-5.9.49/watchdog/dde_kwin.go startdde-5.9.49-new/watchdog/dde_kwin.go
--- startdde-5.9.49/watchdog/dde_kwin.go	2022-08-17 12:21:36.000000000 +0800
+++ startdde-5.9.49-new/watchdog/dde_kwin.go	2022-10-27 21:30:23.915940930 +0800
@@ -6,7 +6,7 @@
 
 const (
 	kWinServiceName = "org.kde.KWin"
-	ddeKWinCommand  = "kwin_no_scale"
+	ddeKWinCommand  = "deepin-kwin_x11"
 )
 
 func isDdeKWinRunning() (bool, error) {
diff -Nur startdde-5.9.49/watchdog/watchdog_test.go startdde-5.9.49-new/watchdog/watchdog_test.go
--- startdde-5.9.49/watchdog/watchdog_test.go	2022-08-17 12:21:36.000000000 +0800
+++ startdde-5.9.49-new/watchdog/watchdog_test.go	2022-10-27 21:30:23.915940930 +0800
@@ -28,7 +28,7 @@
 )
 
 func isUseKwin() bool {
-	_, err := os.Stat("/usr/bin/kwin_no_scale")
+	_, err := os.Stat("/usr/bin/deepin-kwin_x11")
 	return err == nil
 }
 
