--- configure.ac.orig
+++ configure.ac
@@ -16,9 +16,9 @@ dnl with this program; if not, write to
 dnl 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 dnl
 
-AC_PREREQ(2.50)
+AC_PREREQ([2.68])
 
-AC_INIT([heroes], [0.21], [heroes-bugs@lists.sourceforge.net])
+AC_INIT([heroes],[0.21],[heroes-bugs@lists.sourceforge.net])
 AC_CONFIG_AUX_DIR(tools)
 AC_CONFIG_SRCDIR([src/heroes.c])
 AM_INIT_AUTOMAKE([heroes], [0.21])
@@ -30,17 +30,13 @@ heroes_CHECK_OBSOLETE_OPTIONS
 AC_CANONICAL_HOST
 AC_DEFINE_UNQUOTED([HOST],["$host"],[The canonical host])
 
-# If the user didn't supplied a CFLAGS value,
-# set an empty one to prevent autoconf to stick -O2 -g here.
-test -z "$CFLAGS" && CFLAGS=""
-
 # ===================
 # Check for programs.
 # ===================
 
-AC_PROG_CC
-AC_ISC_POSIX
 AC_PROG_CC_STDC
+AC_USE_SYSTEM_EXTENSIONS
+AC_SYS_LARGEFILE
 AM_PROG_CC_C_O
 AC_PROG_CPP
 AC_PROG_AWK
@@ -57,16 +53,10 @@ AM_CONDITIONAL(HAVE_GPERF, [test \! -z "
 
 AC_CACHE_SAVE
 
-# Set _GNU_SOURCE before any other tests.
-AC__GNU_SOURCE
-
 # =====================
 # Checks for libraries.
 # =====================
 
-dnl pass a third argument to AC_CHECK_LIB to
-dnl prevent HAVE_LIBM from being defined (we don't need it).
-AC_CHECK_LIB([m], [sin], [LIBS="-lm $LIBS"])
 heroes_MEDIALIB_SELECTION
 
 # ================================
@@ -200,7 +190,7 @@ heroes_DEBUG
 # as clean as possible.
 dnl Don't quote the following argument, it is parsed by automake.
 AM_CONFIG_HEADER(config.h:src/config.h.in)
-
+AC_SEARCH_LIBS(sin, m)
 AC_CONFIG_FILES([
  Makefile
  doc/Makefile
