From ce9f6e1187867c4fbb7935e08a9ab4d9d8dea8c3 Mon Sep 17 00:00:00 2001
From: Landry Breuil <landry@xfce.org>
Date: Sat, 13 Apr 2024 19:12:32 +0200
Subject: [PATCH] add missing break to fix use-after-free when alt-tabbing
 (#762)

---
 src/tabwin.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/tabwin.c b/src/tabwin.c
index 34a117d32..a5ab85d36 100644
--- a/src/tabwin.c
+++ b/src/tabwin.c
@@ -1000,6 +1000,7 @@ tabwinRemoveClient (Tabwin *tabwin, Client *c)
             {
                 gtk_container_remove (GTK_CONTAINER (tabwin_widget->container), icon);
                 tabwin_widget->widgets = g_list_delete_link (tabwin_widget->widgets, widgets);
+                break;
             }
         }
     }
-- 
GitLab

