diff -Pdpru rp-pppoe-3.14.orig/src/Makefile.in rp-pppoe-3.14/src/Makefile.in
--- rp-pppoe-3.14.orig/src/Makefile.in	2020-05-27 03:29:37.000000000 +0300
+++ rp-pppoe-3.14/src/Makefile.in	2020-05-28 03:07:40.557172960 +0300
@@ -210,11 +210,9 @@ install: all
 		mkdir -p $(DESTDIR)/etc/rc.d/init.d ;\
 		$(install) -m 755 ../scripts/pppoe-init $(DESTDIR)/etc/rc.d/init.d/pppoe ; \
 	fi
-	@if [ -f /etc/SuSE-release ] ; then \
-		echo "Looks like a SuSE Linux system; installing $(DESTDIR)/etc/rc.d/init.d/pppoe" ; \
-		mkdir -p $(DESTDIR)/etc/rc.d/init.d ;\
-		$(install) -m 755 ../scripts/pppoe-init-suse $(DESTDIR)/etc/rc.d/init.d/pppoe ; \
-	fi
+	echo "Looks like a SuSE Linux system; installing $(DESTDIR)/etc/init.d/pppoe" ; \
+	mkdir -p $(DESTDIR)/etc/init.d ;\
+	$(install) -m 755 ../scripts/pppoe-init-suse $(DESTDIR)/etc/init.d/pppoe ; \
 
 	@if [ -f l2tp/handlers/sync-pppd.so ] ; then \
 		mkdir -p $(DESTDIR)/usr/lib/l2tp/plugins ; \
