Index: grimoirelab-sortinghat-0.11.1/config/settings/testing.py
===================================================================
--- grimoirelab-sortinghat-0.11.1.orig/config/settings/testing.py
+++ grimoirelab-sortinghat-0.11.1/config/settings/testing.py
@@ -1,3 +1,4 @@
+import os
 import sys
 import logging
 
@@ -38,8 +39,8 @@ SQL_MODE = [
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.mysql',
-        'USER': 'root',
-        'PASSWORD': 'root',
+        'USER': os.environ.get('TEST_SORTINGHAT_DB_USER', 'root'),
+        'PASSWORD': os.environ.get('TEST_SORTINGHAT_DB_PASSWORD', 'root'),
         'NAME': 'sortinghat_db',
         'OPTIONS': {
             'charset': 'utf8mb4',
@@ -49,9 +50,10 @@ DATABASES = {
             'NAME': 'testhat',
             'CHARSET': 'utf8mb4',
             'COLLATION': 'utf8mb4_unicode_520_ci',
+            'MIRROR': False
         },
         'HOST': '127.0.0.1',
-        'PORT': 3306
+        'PORT': os.environ.get('TEST_SORTINGHAT_DB_PORT', 3306)
     }
 }
 
Index: grimoirelab-sortinghat-0.11.1/config/settings/testing_tenant.py
===================================================================
--- grimoirelab-sortinghat-0.11.1.orig/config/settings/testing_tenant.py
+++ grimoirelab-sortinghat-0.11.1/config/settings/testing_tenant.py
@@ -5,8 +5,8 @@ from .testing import SQL_MODE, DATABASES
 DATABASES.update({
     tenant: {
         'ENGINE': 'django.db.backends.mysql',
-        'USER': 'root',
-        'PASSWORD': 'root',
+        'USER': os.environ.get('TEST_SORTINGHAT_DB_USER', 'root'),
+        'PASSWORD': os.environ.get('TEST_SORTINGHAT_DB_PASSWORD', 'root'),
         'NAME': tenant,
         'OPTIONS': {
             'charset': 'utf8mb4',
@@ -16,9 +16,10 @@ DATABASES.update({
             'NAME': tenant,
             'CHARSET': 'utf8mb4',
             'COLLATION': 'utf8mb4_unicode_520_ci',
+            'MIRROR': False
         },
         'HOST': '127.0.0.1',
-        'PORT': 3306
+        'PORT': os.environ.get('TEST_SORTINGHAT_DB_PORT', 3306)
     }
     for tenant in ['tenant_1', 'tenant_2']
 })
