diff -Nru s3transfer-0.2.0.orig/s3transfer/__init__.py s3transfer-0.2.0/s3transfer/__init__.py
--- s3transfer-0.2.0.orig/s3transfer/__init__.py	2019-02-07 00:14:54.000000000 +0100
+++ s3transfer-0.2.0/s3transfer/__init__.py	2019-02-13 13:29:44.840939556 +0100
@@ -134,8 +134,7 @@
 import concurrent.futures
 
 from botocore.compat import six
-from botocore.vendored.requests.packages.urllib3.exceptions import \
-    ReadTimeoutError
+from urllib3.exceptions import ReadTimeoutError
 from botocore.exceptions import IncompleteReadError
 
 import s3transfer.compat
diff -Nru s3transfer-0.2.0.orig/tests/unit/test_s3transfer.py s3transfer-0.2.0/tests/unit/test_s3transfer.py
--- s3transfer-0.2.0.orig/tests/unit/test_s3transfer.py	2019-02-07 00:14:54.000000000 +0100
+++ s3transfer-0.2.0/tests/unit/test_s3transfer.py	2019-02-13 13:30:41.309433514 +0100
@@ -18,7 +18,7 @@
 from contextlib import closing
 
 import mock
-from botocore.vendored import six
+import six
 from concurrent import futures
 
 from s3transfer.exceptions import RetriesExceededError
