Index: powerpc-utils-python-1.2.1/scripts/amsvis/Makefile
===================================================================
--- powerpc-utils-python-1.2.1.orig/scripts/amsvis/Makefile
+++ powerpc-utils-python-1.2.1/scripts/amsvis/Makefile
@@ -28,7 +28,7 @@ install: all
 	@echo "Installing amsvis"
 	@$(INSTALL) -d -m 755 $(DESTDIR)/usr/bin
 	@$(INSTALL) -m 755 amsvis $(DESTDIR)/usr/bin/
-	@python setup.py install --prefix=$(DESTDIR)/usr
+	@python setup.py install --prefix=$(PREFIX) --root=${DESTDIR}
 
 uninstall:
 	@echo "Un-installing amsvis"
