diff --git a/test/test-list.h b/test/test-list.h
index a6cfc6b..eb514db 100644
--- a/test/test-list.h
+++ b/test/test-list.h
@@ -678,8 +678,6 @@ TASK_LIST_START
   TEST_ENTRY  (udp_no_autobind)
   TEST_ENTRY  (udp_multicast_interface)
   TEST_ENTRY  (udp_multicast_interface6)
-  TEST_ENTRY  (udp_multicast_join)
-  TEST_ENTRY  (udp_multicast_join6)
   TEST_ENTRY  (udp_multicast_ttl)
   TEST_ENTRY  (udp_try_send)
 
@@ -817,7 +815,6 @@ TASK_LIST_START
 
   TEST_ENTRY  (getnameinfo_basic_ip4)
   TEST_ENTRY  (getnameinfo_basic_ip4_sync)
-  TEST_ENTRY  (getnameinfo_basic_ip6)
 
   TEST_ENTRY  (getsockname_tcp)
   TEST_ENTRY  (getsockname_udp)
