Index: openshot-qt-3.5.1/src/settings/_default.settings
===================================================================
--- openshot-qt-3.5.1.orig/src/settings/_default.settings
+++ openshot-qt-3.5.1/src/settings/_default.settings
@@ -515,7 +515,7 @@
     "restart": false
   },
   {
-    "value": true,
+    "value": false,
     "title": "Send Anonymous Metrics and Errors",
     "type": "bool",
     "category": "Advanced",
