Index: xournal-0.4.8.2016/src/ttsubset/sft.c
===================================================================
--- xournal-0.4.8.2016.orig/src/ttsubset/sft.c
+++ xournal-0.4.8.2016/src/ttsubset/sft.c
@@ -42,11 +42,14 @@
  * @author Alexander Gelfenbain <adg@sun.com>
  * @version 1.0
  */
-
+#include <config.h>
 #include <assert.h>
 #include <stdlib.h>
 #include <string.h>
 #include <fcntl.h>
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
 #include "sft.h"
 #ifdef USE_GSUB
 #include "gsub.h"
Index: xournal-0.4.8.2016/src/ttsubset/ttcr.c
===================================================================
--- xournal-0.4.8.2016.orig/src/ttsubset/ttcr.c
+++ xournal-0.4.8.2016/src/ttsubset/ttcr.c
@@ -43,7 +43,7 @@
  * @version 1.3
  *
  */
-
+#include <config.h>
 #include <sys/types.h>
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
