Index: numpy-1.16.2/numpy/lib/tests/test_mixins.py
===================================================================
--- numpy-1.16.2.orig/numpy/lib/tests/test_mixins.py
+++ numpy-1.16.2/numpy/lib/tests/test_mixins.py
@@ -4,6 +4,8 @@ import sys
 import numpy as np
 from numpy.testing import assert_, assert_equal, assert_raises
 
+import pytest
+import sys
 
 # NOTE: This class should be kept as an exact copy of the example from the
 # docstring for NDArrayOperatorsMixin.
@@ -193,6 +195,7 @@ class TestNDArrayOperatorsMixin(object):
             err_msg = 'failed for operator {}'.format(op)
             _assert_equal_type_and_value(expected, actual, err_msg=err_msg)
 
+    @pytest.mark.skipif(sys.version_info < (3, 5), reason="requires python >= 3.5")
     def test_matmul(self):
         array = np.array([1, 2], dtype=np.float64)
         array_like = ArrayLike(array)
