Index: numpy-1.24.2/numpy/conftest.py
===================================================================
--- numpy-1.24.2.orig/numpy/conftest.py
+++ numpy-1.24.2/numpy/conftest.py
@@ -30,7 +30,7 @@ hypothesis.settings.register_profile(
 hypothesis.settings.register_profile(
     name="np.test() profile",
     deadline=None, print_blob=True, database=None, derandomize=True,
-    suppress_health_check=hypothesis.HealthCheck.all(),
+    suppress_health_check=list(hypothesis.HealthCheck),
 )
 # Note that the default profile is chosen based on the presence 
 # of pytest.ini, but can be overridden by passing the 
