---
 src/scim_uim_imengine.cpp |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/src/scim_uim_imengine.cpp
+++ b/src/scim_uim_imengine.cpp
@@ -36,7 +36,8 @@
 
 #include <scim.h>
 #include "scim_uim_imengine.h"
-#include <uim/uim-compat-scm.h>
+#include <uim/uim-scm.h>
+#include <uim/uim-util.h>
 
 #define scim_module_init uim_LTX_scim_module_init
 #define scim_module_exit uim_LTX_scim_module_exit
