From c5ea1d8b25ab1958ece6d6f3652ca879bd406fac Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fridrich=20=C5=A0trba?= <fridrich.strba@bluewin.ch>
Date: Mon, 30 Sep 2024 19:53:57 +0200
Subject: [PATCH 3/3] reproducible from environment

---
 .../apache/maven/plugins/javadoc/AbstractJavadocMojo.java   | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java b/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java
index 2c258c7e..ddcb6ede 100644
--- a/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java
+++ b/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java
@@ -2671,6 +2671,12 @@ public abstract class AbstractJavadocMojo extends AbstractMojo {
     private String getBottomText() {
         final String inceptionYear = project.getInceptionYear();
 
+        if (outputTimestamp == null
+                || outputTimestamp.length() < 1
+                || ((outputTimestamp.length() == 1) && !Character.isDigit(outputTimestamp.charAt(0)))) {
+            outputTimestamp = System.getenv("SOURCE_DATE_EPOCH");
+        }
+
         // get Reproducible Builds outputTimestamp date value or the current local date.
         final LocalDate localDate = MavenArchiver.parseBuildOutputTimestamp(outputTimestamp)
                 .map(instant -> instant.atZone(ZoneOffset.UTC).toLocalDate())
-- 
2.50.0

