Index: boost_1_69_0/libs/test/build/Jamfile.v2
===================================================================
--- boost_1_69_0.orig/libs/test/build/Jamfile.v2
+++ boost_1_69_0/libs/test/build/Jamfile.v2
@@ -102,7 +102,7 @@ lib boost_test_exec_monitor
   : # sources
     $(TEST_EXEC_MON_SOURCES).cpp
   : # requirements
-    <link>static
+    # <link>static
   : # default build
   : # usage-requirements
     <link>shared:<define>BOOST_TEST_DYN_LINK=1
Index: boost_1_69_0/libs/exception/build/Jamfile.v2
===================================================================
--- boost_1_69_0.orig/libs/exception/build/Jamfile.v2
+++ boost_1_69_0/libs/exception/build/Jamfile.v2
@@ -7,7 +7,7 @@
 
 project boost/exception
     : source-location ../src
-    : requirements <link>static
+    : # requirements <link>static
     ;
 
 lib boost_exception : clone_current_exception_non_intrusive.cpp ;
Index: boost_1_69_0/libs/chrono/build/Jamfile.v2
===================================================================
--- boost_1_69_0.orig/libs/chrono/build/Jamfile.v2
+++ boost_1_69_0/libs/chrono/build/Jamfile.v2
@@ -97,7 +97,7 @@ project boost/chrono
         <toolset>sun:<define>__typeof__=__typeof__
 
         <link>shared:<define>BOOST_CHRONO_DYN_LINK=1
-        <link>static:<define>BOOST_CHRONO_STATIC_LINK=1
+        # <link>static:<define>BOOST_CHRONO_STATIC_LINK=1
         <toolset>gcc-3.4.4:<linkflags>--enable-auto-import
         <toolset>gcc-4.3.4:<linkflags>--enable-auto-import
         <toolset>gcc-4.4.0,<target-os>windows:<linkflags>--enable-auto-import 
@@ -111,6 +111,6 @@ lib boost_chrono
     : $(SOURCES).cpp
     :
     <link>shared:<define>BOOST_ALL_DYN_LINK=1 # tell source we're building dll's
-    <link>static:<define>BOOST_All_STATIC_LINK=1 # tell source we're building static lib's
+    # <link>static:<define>BOOST_All_STATIC_LINK=1 # tell source we're building static lib's
     ;
 
Index: boost_1_69_0/libs/system/build/Jamfile.v2
===================================================================
--- boost_1_69_0.orig/libs/system/build/Jamfile.v2
+++ boost_1_69_0/libs/system/build/Jamfile.v2
@@ -11,7 +11,7 @@ project boost/system
     : source-location ../src
     : usage-requirements  # pass these requirement to dependents (i.e. users)
       <link>shared:<define>BOOST_SYSTEM_DYN_LINK=1
-      <link>static:<define>BOOST_SYSTEM_STATIC_LINK=1
+#      <link>static:<define>BOOST_SYSTEM_STATIC_LINK=1
     ;
 
 SOURCES = error_code ;
@@ -19,7 +19,7 @@ SOURCES = error_code ;
 lib boost_system
    : $(SOURCES).cpp
    : <link>shared:<define>BOOST_SYSTEM_DYN_LINK=1
-     <link>static:<define>BOOST_SYSTEM_STATIC_LINK=1
+#     <link>static:<define>BOOST_SYSTEM_STATIC_LINK=1
    ;
 
 boost-install boost_system ;
Index: boost_1_69_0/libs/timer/build/Jamfile.v2
===================================================================
--- boost_1_69_0.orig/libs/timer/build/Jamfile.v2
+++ boost_1_69_0/libs/timer/build/Jamfile.v2
@@ -14,7 +14,7 @@ project boost/timer
       <library>/boost/system//boost_system
     : usage-requirements  # pass these requirement to dependants (i.e. users)
       <link>shared:<define>BOOST_TIMER_DYN_LINK=1
-      <link>static:<define>BOOST_TIMER_STATIC_LINK=1
+#      <link>static:<define>BOOST_TIMER_STATIC_LINK=1
     ;
 
 SOURCES = auto_timers_construction cpu_timer ;
@@ -22,7 +22,7 @@ SOURCES = auto_timers_construction cpu_t
 lib boost_timer
    : $(SOURCES).cpp
    : <link>shared:<define>BOOST_TIMER_DYN_LINK=1
-     <link>static:<define>BOOST_TIMER_STATIC_LINK=1
+#     <link>static:<define>BOOST_TIMER_STATIC_LINK=1
    ;
 
 boost-install boost_timer ;
Index: boost_1_69_0/libs/filesystem/build/Jamfile.v2
===================================================================
--- boost_1_69_0.orig/libs/filesystem/build/Jamfile.v2
+++ boost_1_69_0/libs/filesystem/build/Jamfile.v2
@@ -11,7 +11,7 @@ project boost/filesystem
     : source-location ../src
     : usage-requirements # pass these requirement to dependents (i.e. users)
       <link>shared:<define>BOOST_FILESYSTEM_DYN_LINK=1
-      <link>static:<define>BOOST_FILESYSTEM_STATIC_LINK=1
+#      <link>static:<define>BOOST_FILESYSTEM_STATIC_LINK=1
     ;
 
 SOURCES =
@@ -28,7 +28,7 @@ SOURCES =
 lib boost_filesystem
     : $(SOURCES).cpp ../../system/build//boost_system
     : <link>shared:<define>BOOST_FILESYSTEM_DYN_LINK=1
-      <link>static:<define>BOOST_FILESYSTEM_STATIC_LINK=1
+#      <link>static:<define>BOOST_FILESYSTEM_STATIC_LINK=1
     :
     : # Boost.Filesystem uses some of Boost.System functions in inlined/templated
       # functions, so clients that use Boost.Filesystem will have direct references 
