Index: boost_1_80_0/libs/test/build/Jamfile.v2
===================================================================
--- boost_1_80_0.orig/libs/test/build/Jamfile.v2
+++ boost_1_80_0/libs/test/build/Jamfile.v2
@@ -95,7 +95,7 @@ lib boost_test_exec_monitor
   : # sources
     $(TEST_EXEC_MON_SOURCES).cpp
   : # requirements
-    <link>static
+    # <link>static
   : # default build
   : # usage-requirements
     <link>shared:<define>BOOST_TEST_DYN_LINK=1
Index: boost_1_80_0/libs/exception/build/Jamfile.v2
===================================================================
--- boost_1_80_0.orig/libs/exception/build/Jamfile.v2
+++ boost_1_80_0/libs/exception/build/Jamfile.v2
@@ -7,7 +7,7 @@
 
 project boost/exception
     : source-location ../src
-    : requirements <link>static
+    : # requirements <link>static
     ;
 
 lib boost_exception : clone_current_exception_non_intrusive.cpp ;
Index: boost_1_80_0/libs/chrono/build/Jamfile.v2
===================================================================
--- boost_1_80_0.orig/libs/chrono/build/Jamfile.v2
+++ boost_1_80_0/libs/chrono/build/Jamfile.v2
@@ -90,7 +90,7 @@ project boost/chrono
         <toolset>sun:<define>__typeof__=__typeof__
 
         <link>shared:<define>BOOST_CHRONO_DYN_LINK=1
-        <link>static:<define>BOOST_CHRONO_STATIC_LINK=1
+        # <link>static:<define>BOOST_CHRONO_STATIC_LINK=1
         <define>BOOST_CHRONO_NO_LIB=1
         <toolset>gcc-3.4.4:<linkflags>--enable-auto-import
         <toolset>gcc-4.3.4:<linkflags>--enable-auto-import
@@ -104,6 +104,6 @@ lib boost_chrono
     : $(SOURCES).cpp
     :
     $(cxx11-requirements)
     <link>shared:<define>BOOST_CHRONO_DYN_LINK=1 # tell source we're building dll's
-    <link>static:<define>BOOST_CHRONO_STATIC_LINK=1 # tell source we're building static lib's
+    # <link>static:<define>BOOST_CHRONO_STATIC_LINK=1 # tell source we're building static lib's
     ;
Index: boost_1_80_0/libs/timer/build/Jamfile.v2
===================================================================
--- boost_1_83_0.orig/libs/timer/build/Jamfile.v2
+++ boost_1_83_0/libs/timer/build/Jamfile.v2
@@ -20,7 +20,7 @@
       <library>$(boost_dependencies)
 
       <link>shared:<define>BOOST_TIMER_DYN_LINK=1
-      <link>static:<define>BOOST_TIMER_STATIC_LINK=1
+      # <link>static:<define>BOOST_TIMER_STATIC_LINK=1
 
       <define>BOOST_TIMER_NO_LIB=1
 
Index: boost_1_80_0/libs/filesystem/build/Jamfile.v2
===================================================================
--- boost_1_80_0.orig/libs/filesystem/build/Jamfile.v2
+++ boost_1_80_0/libs/filesystem/build/Jamfile.v2
@@ -130,8 +130,8 @@ project boost/filesystem
     : source-location ../src
     : usage-requirements # pass these requirement to dependents (i.e. users)
       <link>shared:<define>BOOST_FILESYSTEM_DYN_LINK=1
-      <link>static:<define>BOOST_FILESYSTEM_STATIC_LINK=1
+      # <link>static:<define>BOOST_FILESYSTEM_STATIC_LINK=1
       <define>BOOST_FILESYSTEM_NO_LIB=1
     ;
 
 SOURCES =
@@ -163,7 +163,7 @@ lib boost_filesystem
       <conditional>@select-platform-specific-sources
       <include>../src
       <link>shared:<define>BOOST_FILESYSTEM_DYN_LINK=1
-      <link>static:<define>BOOST_FILESYSTEM_STATIC_LINK=1
+#      <link>static:<define>BOOST_FILESYSTEM_STATIC_LINK=1
       $(cxx_requirements)
     : usage-requirements
       $(cxx_requirements)
