Index: boost_1_74_0/boost/serialization/set.hpp
===================================================================
--- boost_1_74_0.orig/boost/serialization/set.hpp
+++ boost_1_74_0/boost/serialization/set.hpp
@@ -25,6 +25,7 @@
 #include <boost/serialization/detail/stack_constructor.hpp>
 #include <boost/serialization/collection_size_type.hpp>
 #include <boost/serialization/item_version_type.hpp>
+#include <boost/serialization/library_version_type.hpp>
 
 #include <boost/serialization/collections_save_imp.hpp>
 #include <boost/serialization/split_free.hpp>
