Index: mod_auth_gssapi-1.6.3/tests/magtests.py
===================================================================
--- mod_auth_gssapi-1.6.3.orig/tests/magtests.py	2020-08-05 18:56:18.944535412 +0200
+++ mod_auth_gssapi-1.6.3/tests/magtests.py	2020-11-25 10:34:58.170058037 +0100
@@ -413,6 +413,11 @@ def setup_http(testdir, so_dir, wrapenv)
         distro = "Debian"
         moddir = "/usr/lib/apache2/modules"
     if not os.path.exists(moddir):
+        distro = "SUSE"
+        moddir = "/usr/lib64/apache2"
+        if not os.path.exists(moddir):
+            moddir = "/usr/lib/apache2"
+    if not os.path.exists(moddir):
         raise ValueError("Could not find Apache module directory!")
     os.symlink(moddir, os.path.join(httpdir, 'modules'))
 
@@ -436,7 +441,7 @@ def setup_http(testdir, so_dir, wrapenv)
                'MALLOC_PERTURB_': str(random.randint(0, 32767) % 255 + 1)}
     httpenv.update(wrapenv)
 
-    httpd = "httpd" if distro == "Fedora" else "apache2"
+    httpd = "httpd" if (distro == "Fedora") or (distro == "SUSE") else "apache2"
     httpproc = subprocess.Popen([httpd, '-DFOREGROUND', '-f', config],
                                 env=httpenv, preexec_fn=os.setsid)
     return httpproc
@@ -449,7 +454,7 @@ def kinit_user(testdir, kdcenv):
     testenv.update(kdcenv)
 
     with (open(testlog, 'a')) as logfile:
-        kinit = subprocess.Popen(["kinit", USR_NAME],
+        kinit = subprocess.Popen(["kinit", "-V", USR_NAME],
                                  stdin=subprocess.PIPE,
                                  stdout=logfile, stderr=logfile,
                                  env=testenv, preexec_fn=os.setsid)
Index: mod_auth_gssapi-1.6.3/tests/httpd.conf
===================================================================
--- mod_auth_gssapi-1.6.3.orig/tests/httpd.conf	2020-08-05 18:56:18.943535397 +0200
+++ mod_auth_gssapi-1.6.3/tests/httpd.conf	2020-11-25 10:31:32.600854885 +0100
@@ -71,7 +71,9 @@ LoadModule userdir_module modules/mod_us
     LoadModule version_module modules/mod_version.so
 </IfModule>
 LoadModule vhost_alias_module modules/mod_vhost_alias.so
+<IfModule !prefork.c>
 LoadModule mpm_prefork_module modules/mod_mpm_prefork.so
+</IfModule>
 LoadModule proxy_module modules/mod_proxy.so
 LoadModule proxy_http_module modules/mod_proxy_http.so
 
