--- a/lib/aio/reader.cpp	2024/01/10 10:23:55	11064
+++ b/lib/aio/reader.cpp	2024/02/13 18:08:57	11095
@@ -319,7 +319,7 @@
 			if (remaining_ != nosize && to_read > remaining_) {
 				to_read = remaining_;
 			}
-			rwresult r = to_read ? file_.read2(b->get(to_read), to_read) : rwresult{rwresult::none, 0};
+			rwresult r = to_read ? file_.read2(b->get(to_read), to_read) : rwresult{size_t(0)};
 			l.lock();
 			if (quit_ || error_) {
 				return;

