Index: ART-1.26.4/rtengine/CMakeLists.txt
===================================================================
--- ART-1.26.4.orig/rtengine/CMakeLists.txt
+++ ART-1.26.4/rtengine/CMakeLists.txt
@@ -234,6 +234,7 @@ target_link_libraries(rtengine
     ${LENSFUN_LIBRARIES}
     ${RSVG_LIBRARIES}
     ${EXIV2_LIBRARIES}
+    m 
     )
 if(LCMS_FAST_FLOAT AND NOT LCMS_FAST_FLOAT_SYSTEM)
     target_link_libraries(rtengine lcms2_fast_float)
