diff -ru lv451.orig/src/command.c lv451/src/command.c
--- lv451.orig/src/command.c	2004-01-05 08:42:31.000000000 +0100
+++ lv451/src/command.c	2006-07-17 16:50:36.000000000 +0200
@@ -551,7 +551,8 @@
 #define COM_SIZE 128
 #define ARG_SIZE 64
   int argc;
-  byte *ptr, *nptr, *argv[ ARG_SIZE ];
+  byte *ptr, *nptr;
+  char *argv[ ARG_SIZE ];
   byte com[ COM_SIZE ];
   byte num[ COM_SIZE ];
 #ifndef MSDOS /* NOT DEFINED */
@@ -599,7 +600,7 @@
   return spawnvp( 0, argv[ 0 ], argv );
 #else
   if( 0 == (pid = fork()) ){
-    execvp( argv[ 0 ], (char **)argv );
+    execvp( argv[ 0 ], argv );
     exit( 1 );
   } else if( -1 == pid ){
     return 1;
diff -ru lv451.orig/src/raw.c lv451/src/raw.c
--- lv451.orig/src/raw.c	2004-01-05 08:23:29.000000000 +0100
+++ lv451/src/raw.c	2006-07-17 17:00:02.000000000 +0200
@@ -42,6 +42,7 @@
       DecodeAddControl( ch );
     } else {
       c[ 0 ] = ch;
+      c[ 1 ] = 0;
       DecodeAddChar( ASCII, c, 0 );
     }
   }
