--- cmd/kubeadm/app/phases/controlplane/volumes.go
+++ cmd/kubeadm/app/phases/controlplane/volumes.go	2020-01-13 16:27:04.267070673 +0100
@@ -33,7 +33,7 @@
 	caCertsVolumeName              = "ca-certs"
 	caCertsVolumePath              = "/etc/ssl/certs"
 	flexvolumeDirVolumeName        = "flexvolume-dir"
-	defaultFlexvolumeDirVolumePath = "/usr/libexec/kubernetes/kubelet-plugins/volume/exec"
+	defaultFlexvolumeDirVolumePath = "/var/lib/kubelet/volume-plugin"
 )
 
 // caCertsExtraVolumePaths specifies the paths that can be conditionally mounted into the apiserver and controller-manager containers
