Index: automake-1.16.5/lib/config.guess
===================================================================
--- automake-1.16.5.orig/lib/config.guess
+++ automake-1.16.5/lib/config.guess
@@ -188,6 +188,17 @@ Linux|GNU|GNU/*)
 	;;
 esac
 
+case "${UNAME_MACHINE}" in
+  i?86)
+     test -z "$VENDOR" && VENDOR=pc
+     ;;
+  *)
+     test -z "$VENDOR" && VENDOR=unknown
+     ;;
+esac
+test -f /etc/SuSE-release -o -f /.buildenv && VENDOR=suse
+test -f /etc/os-release && grep -q suse /etc/os-release && VENDOR=suse
+
 # Note: order is significant - the case branches are not exclusive.
 
 case $UNAME_MACHINE:$UNAME_SYSTEM:$UNAME_RELEASE:$UNAME_VERSION in
@@ -269,46 +280,46 @@ case $UNAME_MACHINE:$UNAME_SYSTEM:$UNAME
 	;;
     *:Bitrig:*:*)
 	UNAME_MACHINE_ARCH=`arch | sed 's/Bitrig.//'`
-	GUESS=$UNAME_MACHINE_ARCH-unknown-bitrig$UNAME_RELEASE
+	GUESS=$UNAME_MACHINE_ARCH-${VENDOR}-bitrig$UNAME_RELEASE
 	;;
     *:OpenBSD:*:*)
 	UNAME_MACHINE_ARCH=`arch | sed 's/OpenBSD.//'`
-	GUESS=$UNAME_MACHINE_ARCH-unknown-openbsd$UNAME_RELEASE
+	GUESS=$UNAME_MACHINE_ARCH-${VENDOR}-openbsd$UNAME_RELEASE
 	;;
     *:SecBSD:*:*)
 	UNAME_MACHINE_ARCH=`arch | sed 's/SecBSD.//'`
-	GUESS=$UNAME_MACHINE_ARCH-unknown-secbsd$UNAME_RELEASE
+	GUESS=$UNAME_MACHINE_ARCH-${VENDOR}-secbsd$UNAME_RELEASE
 	;;
     *:LibertyBSD:*:*)
 	UNAME_MACHINE_ARCH=`arch | sed 's/^.*BSD\.//'`
-	GUESS=$UNAME_MACHINE_ARCH-unknown-libertybsd$UNAME_RELEASE
+	GUESS=$UNAME_MACHINE_ARCH-${VENDOR}-libertybsd$UNAME_RELEASE
 	;;
     *:MidnightBSD:*:*)
-	GUESS=$UNAME_MACHINE-unknown-midnightbsd$UNAME_RELEASE
+	GUESS=$UNAME_MACHINE-${VENDOR}-midnightbsd$UNAME_RELEASE
 	;;
     *:ekkoBSD:*:*)
-	GUESS=$UNAME_MACHINE-unknown-ekkobsd$UNAME_RELEASE
+	GUESS=$UNAME_MACHINE-${VENDOR}-ekkobsd$UNAME_RELEASE
 	;;
     *:SolidBSD:*:*)
-	GUESS=$UNAME_MACHINE-unknown-solidbsd$UNAME_RELEASE
+	GUESS=$UNAME_MACHINE-${VENDOR}-solidbsd$UNAME_RELEASE
 	;;
     *:OS108:*:*)
-	GUESS=$UNAME_MACHINE-unknown-os108_$UNAME_RELEASE
+	GUESS=$UNAME_MACHINE-${VENDOR}-os108_$UNAME_RELEASE
 	;;
     macppc:MirBSD:*:*)
-	GUESS=powerpc-unknown-mirbsd$UNAME_RELEASE
+	GUESS=powerpc-${VENDOR}-mirbsd$UNAME_RELEASE
 	;;
     *:MirBSD:*:*)
-	GUESS=$UNAME_MACHINE-unknown-mirbsd$UNAME_RELEASE
+	GUESS=$UNAME_MACHINE-${VENDOR}-mirbsd$UNAME_RELEASE
 	;;
     *:Sortix:*:*)
-	GUESS=$UNAME_MACHINE-unknown-sortix
+	GUESS=$UNAME_MACHINE-${VENDOR}-sortix
 	;;
     *:Twizzler:*:*)
-	GUESS=$UNAME_MACHINE-unknown-twizzler
+	GUESS=$UNAME_MACHINE-${VENDOR}-twizzler
 	;;
     *:Redox:*:*)
-	GUESS=$UNAME_MACHINE-unknown-redox
+	GUESS=$UNAME_MACHINE-${VENDOR}-redox
 	;;
     mips:OSF1:*.*)
 	GUESS=mips-dec-osf1
@@ -370,13 +381,13 @@ case $UNAME_MACHINE:$UNAME_SYSTEM:$UNAME
 	GUESS=$UNAME_MACHINE-dec-osf$OSF_REL
 	;;
     Amiga*:UNIX_System_V:4.0:*)
-	GUESS=m68k-unknown-sysv4
+	GUESS=m68k-${VENDOR}-sysv4
 	;;
     *:[Aa]miga[Oo][Ss]:*:*)
-	GUESS=$UNAME_MACHINE-unknown-amigaos
+	GUESS=$UNAME_MACHINE-${VENDOR}-amigaos
 	;;
     *:[Mm]orph[Oo][Ss]:*:*)
-	GUESS=$UNAME_MACHINE-unknown-morphos
+	GUESS=$UNAME_MACHINE-${VENDOR}-morphos
 	;;
     *:OS/390:*:*)
 	GUESS=i370-ibm-openedition
@@ -391,7 +402,7 @@ case $UNAME_MACHINE:$UNAME_SYSTEM:$UNAME
 	GUESS=arm-acorn-riscix$UNAME_RELEASE
 	;;
     arm*:riscos:*:*|arm*:RISCOS:*:*)
-	GUESS=arm-unknown-riscos
+	GUESS=arm-${VENDOR}-riscos
 	;;
     SR2?01:HI-UX/MPP:*:* | SR8000:HI-UX/MPP:*:*)
 	GUESS=hppa1.1-hitachi-hiuxmpp
@@ -505,7 +516,7 @@ case $UNAME_MACHINE:$UNAME_SYSTEM:$UNAME
 	GUESS=m68k-hades-mint$UNAME_RELEASE
 	;;
     *:*MiNT:*:* | *:*mint:*:* | *:*TOS:*:*)
-	GUESS=m68k-unknown-mint$UNAME_RELEASE
+	GUESS=m68k-${VENDOR}-mint$UNAME_RELEASE
 	;;
     m68k:machten:*:*)
 	GUESS=m68k-apple-machten$UNAME_RELEASE
@@ -818,9 +829,9 @@ EOF
 	;;
     i*86:OSF1:*:*)
 	if test -x /usr/sbin/sysversion ; then
-	    GUESS=$UNAME_MACHINE-unknown-osf1mk
+	    GUESS=$UNAME_MACHINE-${VENDOR}-osf1mk
 	else
-	    GUESS=$UNAME_MACHINE-unknown-osf1
+	    GUESS=$UNAME_MACHINE-${VENDOR}-osf1
 	fi
 	;;
     parisc*:Lites*:*:*)
@@ -885,10 +896,10 @@ EOF
 	GUESS=$UNAME_MACHINE-pc-bsdi$UNAME_RELEASE
 	;;
     sparc*:BSD/OS:*:*)
-	GUESS=sparc-unknown-bsdi$UNAME_RELEASE
+	GUESS=sparc-${VENDOR}-bsdi$UNAME_RELEASE
 	;;
     *:BSD/OS:*:*)
-	GUESS=$UNAME_MACHINE-unknown-bsdi$UNAME_RELEASE
+	GUESS=$UNAME_MACHINE-${VENDOR}-bsdi$UNAME_RELEASE
 	;;
     arm:FreeBSD:*:*)
 	UNAME_PROCESSOR=`uname -p`
@@ -897,10 +908,10 @@ EOF
 	    | grep -q __ARM_PCS_VFP
 	then
 	    FREEBSD_REL=`echo "$UNAME_RELEASE" | sed -e 's/[-(].*//'`
-	    GUESS=$UNAME_PROCESSOR-unknown-freebsd$FREEBSD_REL-gnueabi
+	    GUESS=$UNAME_PROCESSOR-${VENDOR}-freebsd$FREEBSD_REL-gnueabi
 	else
 	    FREEBSD_REL=`echo "$UNAME_RELEASE" | sed -e 's/[-(].*//'`
-	    GUESS=$UNAME_PROCESSOR-unknown-freebsd$FREEBSD_REL-gnueabihf
+	    GUESS=$UNAME_PROCESSOR-${VENDOR}-freebsd$FREEBSD_REL-gnueabihf
 	fi
 	;;
     *:FreeBSD:*:*)
@@ -912,7 +923,7 @@ EOF
 		UNAME_PROCESSOR=i586 ;;
 	esac
 	FREEBSD_REL=`echo "$UNAME_RELEASE" | sed -e 's/[-(].*//'`
-	GUESS=$UNAME_PROCESSOR-unknown-freebsd$FREEBSD_REL
+	GUESS=$UNAME_PROCESSOR-${VENDOR}-freebsd$FREEBSD_REL
 	;;
     i*:CYGWIN*:*)
 	GUESS=$UNAME_MACHINE-pc-cygwin
@@ -935,10 +946,10 @@ EOF
 		GUESS=i586-pc-interix$UNAME_RELEASE
 		;;
 	    authenticamd | genuineintel | EM64T)
-		GUESS=x86_64-unknown-interix$UNAME_RELEASE
+		GUESS=x86_64-${VENDOR}-interix$UNAME_RELEASE
 		;;
 	    IA64)
-		GUESS=ia64-unknown-interix$UNAME_RELEASE
+		GUESS=ia64-${VENDOR}-interix$UNAME_RELEASE
 		;;
 	esac ;;
     i*:UWIN*:*)
@@ -949,29 +960,29 @@ EOF
 	;;
     prep*:SunOS:5.*:*)
 	SUN_REL=`echo "$UNAME_RELEASE" | sed -e 's/[^.]*//'`
-	GUESS=powerpcle-unknown-solaris2$SUN_REL
+	GUESS=powerpcle-${VENDOR}-solaris2$SUN_REL
 	;;
     *:GNU:*:*)
 	# the GNU system
 	GNU_ARCH=`echo "$UNAME_MACHINE" | sed -e 's,[-/].*$,,'`
 	GNU_REL=`echo "$UNAME_RELEASE" | sed -e 's,/.*$,,'`
-	GUESS=$GNU_ARCH-unknown-$LIBC$GNU_REL
+	GUESS=$GNU_ARCH-${VENDOR}-$LIBC$GNU_REL
 	;;
     *:GNU/*:*:*)
 	# other systems with GNU libc and userland
 	GNU_SYS=`echo "$UNAME_SYSTEM" | sed 's,^[^/]*/,,' | tr "[:upper:]" "[:lower:]"`
 	GNU_REL=`echo "$UNAME_RELEASE" | sed -e 's/[-(].*//'`
-	GUESS=$UNAME_MACHINE-unknown-$GNU_SYS$GNU_REL-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-$GNU_SYS$GNU_REL-$LIBC
 	;;
     *:Minix:*:*)
-	GUESS=$UNAME_MACHINE-unknown-minix
+	GUESS=$UNAME_MACHINE-${VENDOR}-minix
 	;;
     aarch64:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     aarch64_be:Linux:*:*)
 	UNAME_MACHINE=aarch64_be
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     alpha:Linux:*:*)
 	case `sed -n '/^cpu model/s/^.*: \(.*\)/\1/p' /proc/cpuinfo 2>/dev/null` in
@@ -985,29 +996,29 @@ EOF
 	esac
 	objdump --private-headers /bin/sh | grep -q ld.so.1
 	if test "$?" = 0 ; then LIBC=gnulibc1 ; fi
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     arc:Linux:*:* | arceb:Linux:*:* | arc32:Linux:*:* | arc64:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     arm*:Linux:*:*)
 	set_cc_for_build
 	if echo __ARM_EABI__ | $CC_FOR_BUILD -E - 2>/dev/null \
 	    | grep -q __ARM_EABI__
 	then
-	    GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	    GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	else
 	    if echo __ARM_PCS_VFP | $CC_FOR_BUILD -E - 2>/dev/null \
 		| grep -q __ARM_PCS_VFP
 	    then
-		GUESS=$UNAME_MACHINE-unknown-linux-${LIBC}eabi
+		GUESS=$UNAME_MACHINE-${VENDOR}-linux-${LIBC}eabi
 	    else
-		GUESS=$UNAME_MACHINE-unknown-linux-${LIBC}eabihf
+		GUESS=$UNAME_MACHINE-${VENDOR}-linux-${LIBC}eabihf
 	    fi
 	fi
 	;;
     avr32*:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     cris:Linux:*:*)
 	GUESS=$UNAME_MACHINE-axis-linux-$LIBC
@@ -1016,31 +1027,31 @@ EOF
 	GUESS=$UNAME_MACHINE-axis-linux-$LIBC
 	;;
     e2k:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     frv:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     hexagon:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     i*86:Linux:*:*)
-	GUESS=$UNAME_MACHINE-pc-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     ia64:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     k1om:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     loongarch32:Linux:*:* | loongarch64:Linux:*:* | loongarchx32:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     m32r*:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     m68*:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     mips:Linux:*:* | mips64:Linux:*:*)
 	set_cc_for_build
@@ -1088,60 +1099,60 @@ EOF
 EOF
 	cc_set_vars=`$CC_FOR_BUILD -E "$dummy.c" 2>/dev/null | grep '^CPU\|^MIPS_ENDIAN\|^LIBCABI'`
 	eval "$cc_set_vars"
-	test "x$CPU" != x && { echo "$CPU${MIPS_ENDIAN}-unknown-linux-$LIBCABI"; exit; }
+	test "x$CPU" != x && { echo "$CPU${MIPS_ENDIAN}-${VENDOR}-linux-$LIBCABI"; exit; }
 	;;
     mips64el:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     openrisc*:Linux:*:*)
-	GUESS=or1k-unknown-linux-$LIBC
+	GUESS=or1k-${VENDOR}-linux-$LIBC
 	;;
     or32:Linux:*:* | or1k*:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     padre:Linux:*:*)
-	GUESS=sparc-unknown-linux-$LIBC
+	GUESS=sparc-${VENDOR}-linux-$LIBC
 	;;
     parisc64:Linux:*:* | hppa64:Linux:*:*)
-	GUESS=hppa64-unknown-linux-$LIBC
+	GUESS=hppa64-${VENDOR}-linux-$LIBC
 	;;
     parisc:Linux:*:* | hppa:Linux:*:*)
 	# Look for CPU level
 	case `grep '^cpu[^a-z]*:' /proc/cpuinfo 2>/dev/null | cut -d' ' -f2` in
-	  PA7*) GUESS=hppa1.1-unknown-linux-$LIBC ;;
-	  PA8*) GUESS=hppa2.0-unknown-linux-$LIBC ;;
-	  *)    GUESS=hppa-unknown-linux-$LIBC ;;
+	  PA7*) GUESS=hppa1.1-${VENDOR}-linux-$LIBC ;;
+	  PA8*) GUESS=hppa2.0-${VENDOR}-linux-$LIBC ;;
+	  *)    GUESS=hppa-${VENDOR}-linux-$LIBC ;;
 	esac
 	;;
     ppc64:Linux:*:*)
-	GUESS=powerpc64-unknown-linux-$LIBC
+	GUESS=powerpc64-${VENDOR}-linux-$LIBC
 	;;
     ppc:Linux:*:*)
-	GUESS=powerpc-unknown-linux-$LIBC
+	GUESS=powerpc-${VENDOR}-linux-$LIBC
 	;;
     ppc64le:Linux:*:*)
-	GUESS=powerpc64le-unknown-linux-$LIBC
+	GUESS=powerpc64le-${VENDOR}-linux-$LIBC
 	;;
     ppcle:Linux:*:*)
-	GUESS=powerpcle-unknown-linux-$LIBC
+	GUESS=powerpcle-${VENDOR}-linux-$LIBC
 	;;
     riscv32:Linux:*:* | riscv32be:Linux:*:* | riscv64:Linux:*:* | riscv64be:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     s390:Linux:*:* | s390x:Linux:*:*)
 	GUESS=$UNAME_MACHINE-ibm-linux-$LIBC
 	;;
     sh64*:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     sh*:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     sparc:Linux:*:* | sparc64:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     tile*:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     vax:Linux:*:*)
 	GUESS=$UNAME_MACHINE-dec-linux-$LIBC
@@ -1157,10 +1168,10 @@ EOF
 		LIBCABI=${LIBC}x32
 	    fi
 	fi
-	GUESS=$UNAME_MACHINE-pc-linux-$LIBCABI
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBCABI
 	;;
     xtensa*:Linux:*:*)
-	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+	GUESS=$UNAME_MACHINE-${VENDOR}-linux-$LIBC
 	;;
     i*86:DYNIX/ptx:4*:*)
 	# ptx 4.0 does uname -s correctly, with DYNIX/ptx in there.
@@ -1182,16 +1193,16 @@ EOF
 	GUESS=$UNAME_MACHINE-pc-os2-emx
 	;;
     i*86:XTS-300:*:STOP)
-	GUESS=$UNAME_MACHINE-unknown-stop
+	GUESS=$UNAME_MACHINE-${VENDOR}-stop
 	;;
     i*86:atheos:*:*)
-	GUESS=$UNAME_MACHINE-unknown-atheos
+	GUESS=$UNAME_MACHINE-${VENDOR}-atheos
 	;;
     i*86:syllable:*:*)
 	GUESS=$UNAME_MACHINE-pc-syllable
 	;;
     i*86:LynxOS:2.*:* | i*86:LynxOS:3.[01]*:* | i*86:LynxOS:4.[02]*:*)
-	GUESS=i386-unknown-lynxos$UNAME_RELEASE
+	GUESS=i386-${VENDOR}-lynxos$UNAME_RELEASE
 	;;
     i*86:*DOS:*:*)
 	GUESS=$UNAME_MACHINE-pc-msdosdjgpp
@@ -1211,7 +1222,7 @@ EOF
 	    *Pentium)	     UNAME_MACHINE=i586 ;;
 	    *Pent*|*Celeron) UNAME_MACHINE=i686 ;;
 	esac
-	GUESS=$UNAME_MACHINE-unknown-sysv${UNAME_RELEASE}${UNAME_SYSTEM}${UNAME_VERSION}
+	GUESS=$UNAME_MACHINE-${VENDOR}-sysv${UNAME_RELEASE}${UNAME_SYSTEM}${UNAME_VERSION}
 	;;
     i*86:*:3.2:*)
 	if test -f /usr/options/cb.name; then
@@ -1250,7 +1261,7 @@ EOF
 	if grep Stardent /usr/include/sys/uadmin.h >/dev/null 2>&1 ; then
 	  GUESS=i860-stardent-sysv$UNAME_RELEASE    # Stardent Vistra i860-SVR4
 	else # Add other i860-SVR4 vendors below as they are discovered.
-	  GUESS=i860-unknown-sysv$UNAME_RELEASE     # Unknown i860-SVR4
+	  GUESS=i860-${VENDOR}-sysv$UNAME_RELEASE     # Unknown i860-SVR4
 	fi
 	;;
     mini*:CTIX:SYS*5:*)
@@ -1287,19 +1298,19 @@ EOF
 	/bin/uname -p 2>/dev/null | /bin/grep pteron >/dev/null \
 	    && { echo i586-ncr-sysv4.3"$OS_REL"; exit; } ;;
     m68*:LynxOS:2.*:* | m68*:LynxOS:3.0*:*)
-	GUESS=m68k-unknown-lynxos$UNAME_RELEASE
+	GUESS=m68k-${VENDOR}-lynxos$UNAME_RELEASE
 	;;
     mc68030:UNIX_System_V:4.*:*)
 	GUESS=m68k-atari-sysv4
 	;;
     TSUNAMI:LynxOS:2.*:*)
-	GUESS=sparc-unknown-lynxos$UNAME_RELEASE
+	GUESS=sparc-${VENDOR}-lynxos$UNAME_RELEASE
 	;;
     rs6000:LynxOS:2.*:*)
-	GUESS=rs6000-unknown-lynxos$UNAME_RELEASE
+	GUESS=rs6000-${VENDOR}-lynxos$UNAME_RELEASE
 	;;
     PowerPC:LynxOS:2.*:* | PowerPC:LynxOS:3.[01]*:* | PowerPC:LynxOS:4.[02]*:*)
-	GUESS=powerpc-unknown-lynxos$UNAME_RELEASE
+	GUESS=powerpc-${VENDOR}-lynxos$UNAME_RELEASE
 	;;
     SM[BE]S:UNIX_SV:*:*)
 	GUESS=mips-dde-sysv$UNAME_RELEASE
@@ -1349,7 +1360,7 @@ EOF
 	if test -d /usr/nec; then
 		GUESS=mips-nec-sysv$UNAME_RELEASE
 	else
-		GUESS=mips-unknown-sysv$UNAME_RELEASE
+		GUESS=mips-${VENDOR}-sysv$UNAME_RELEASE
 	fi
 	;;
     BeBox:BeOS:*:*)	# BeOS running on hardware made by Be, PPC only.
@@ -1365,7 +1376,7 @@ EOF
 	GUESS=i586-pc-haiku
 	;;
     x86_64:Haiku:*:*)
-	GUESS=x86_64-unknown-haiku
+	GUESS=x86_64-${VENDOR}-haiku
 	;;
     SX-4:SUPER-UX:*:*)
 	GUESS=sx4-nec-superux$UNAME_RELEASE
@@ -1478,13 +1489,13 @@ EOF
 	elif test "x${cputype-}" != x; then
 	    UNAME_MACHINE=$cputype
 	fi
-	GUESS=$UNAME_MACHINE-unknown-plan9
+	GUESS=$UNAME_MACHINE-${VENDOR}-plan9
 	;;
     *:TOPS-10:*:*)
-	GUESS=pdp10-unknown-tops10
+	GUESS=pdp10-${VENDOR}-tops10
 	;;
     *:TENEX:*:*)
-	GUESS=pdp10-unknown-tenex
+	GUESS=pdp10-${VENDOR}-tenex
 	;;
     KS10:TOPS-20:*:* | KL10:TOPS-20:*:* | TYPE4:TOPS-20:*:*)
 	GUESS=pdp10-dec-tops20
@@ -1493,17 +1504,17 @@ EOF
 	GUESS=pdp10-xkl-tops20
 	;;
     *:TOPS-20:*:*)
-	GUESS=pdp10-unknown-tops20
+	GUESS=pdp10-${VENDOR}-tops20
 	;;
     *:ITS:*:*)
-	GUESS=pdp10-unknown-its
+	GUESS=pdp10-${VENDOR}-its
 	;;
     SEI:*:*:SEIUX)
 	GUESS=mips-sei-seiux$UNAME_RELEASE
 	;;
     *:DragonFly:*:*)
 	DRAGONFLY_REL=`echo "$UNAME_RELEASE" | sed -e 's/[-(].*//'`
-	GUESS=$UNAME_MACHINE-unknown-dragonfly$DRAGONFLY_REL
+	GUESS=$UNAME_MACHINE-${VENDOR}-dragonfly$DRAGONFLY_REL
 	;;
     *:*VMS:*:*)
 	UNAME_MACHINE=`(uname -p) 2>/dev/null`
@@ -1523,16 +1534,16 @@ EOF
 	GUESS=$UNAME_MACHINE-pc-rdos
 	;;
     *:AROS:*:*)
-	GUESS=$UNAME_MACHINE-unknown-aros
+	GUESS=$UNAME_MACHINE-${VENDOR}-aros
 	;;
     x86_64:VMkernel:*:*)
-	GUESS=$UNAME_MACHINE-unknown-esx
+	GUESS=$UNAME_MACHINE-${VENDOR}-esx
 	;;
     amd64:Isilon\ OneFS:*:*)
-	GUESS=x86_64-unknown-onefs
+	GUESS=x86_64-${VENDOR}-onefs
 	;;
     *:Unleashed:*:*)
-	GUESS=$UNAME_MACHINE-unknown-unleashed$UNAME_RELEASE
+	GUESS=$UNAME_MACHINE-${VENDOR}-unleashed$UNAME_RELEASE
 	;;
 esac
 
