diff -rup a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-07-31 09:00:01.000000000 +0200
+++ b/CMakeLists.txt	2025-11-08 09:44:27.749158974 +0100
@@ -134,7 +134,8 @@ else()
 endif()
 
 #-- minimalist build allowed with boost version older than 1.48
-set( SFCGAL_Boost_COMPONENTS thread system serialization )
+#-- removed system which don't exist anymore in 1.89
+set( SFCGAL_Boost_COMPONENTS thread serialization )
 #-- program_options
 if ( SFCGAL_BUILD_TESTS OR SFCGAL_BUILD_EXAMPLES OR SFCGAL_BUILD_OSG )
 	set( SFCGAL_Boost_COMPONENTS chrono unit_test_framework filesystem program_options timer ${SFCGAL_Boost_COMPONENTS} )
