---
 CMakeLists.txt |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

Index: compile-time-regular-expressions-3.11.0/CMakeLists.txt
===================================================================
--- compile-time-regular-expressions-3.11.0.orig/CMakeLists.txt
+++ compile-time-regular-expressions-3.11.0/CMakeLists.txt
@@ -10,9 +10,14 @@ if(NOT CMAKE_EXPERIMENTAL_CXX_STD AND NO
     elseif(CMAKE_VERSION VERSION_GREATER_EQUAL "4.0.3" AND CMAKE_VERSION VERSION_LESS "4.3")
         set(CMAKE_EXPERIMENTAL_CXX_IMPORT_STD "d0edc3af-4c50-42ea-a356-e2862fe7a444")
         set(CMAKE_CXX_MODULE_STD 1)
-    elseif(CMAKE_VERSION VERSION_GREATER_EQUAL "4.3")
+		elseif(CMAKE_VERSION VERSION_GREATER_EQUAL "4.3" AND CMAKE_VERSION VERSION_LESS "4.4")
         set(CMAKE_EXPERIMENTAL_CXX_IMPORT_STD "451f2fe2-a8a2-47c3-bc32-94786d8fc91b")
         set(CMAKE_CXX_MODULE_STD 1)
+		elseif(CMAKE_VERSION VERSION_GREATER_EQUAL "4.4" AND CMAKE_VERSION VERSION_LESS "4.5")
+        set(CMAKE_EXPERIMENTAL_CXX_IMPORT_STD "f35a9ac6-8463-4d38-8eec-5d6008153e7d")
+        set(CMAKE_CXX_MODULE_STD 1)
+		else()
+        set(CMAKE_CXX_MODULE_STD 0)
     endif()
 endif()
 
