---
 samples/auto.master |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/samples/auto.master
+++ b/samples/auto.master
@@ -4,13 +4,13 @@
 # mount-point [map-type[,format]:]map [options]
 # For details of the format look at auto.master(5).
 #
-/misc	/etc/auto.misc
+#/misc	/etc/auto.misc
 #
 # NOTE: mounts done from a hosts map will be mounted with the
 #	"nosuid" and "nodev" options unless the "suid" and "dev"
 #	options are explicitly given.
 #
-/net	-hosts
+#/net	-hosts
 #
 # Include /etc/auto.master.d/*.autofs
 # To add an extra map using this mechanism you will need to add
@@ -19,7 +19,7 @@
 # and a separate mount map (e.g. /etc/auto.extra or an auto.extra NIS map)
 # that is referred to by the extra.autofs file.
 #
-+dir:/etc/auto.master.d
+#+dir:/etc/auto.master.d
 #
 # If you have fedfs set up and the related binaries, either
 # built as part of autofs or installed from another package,
