References: bsc#1207881 - Obsolete and incorrect manual page details for autofs(8)

Index: autofs-5.1.8/man/autofs.8.in
===================================================================
--- autofs-5.1.8.orig/man/autofs.8.in
+++ autofs-5.1.8/man/autofs.8.in
@@ -2,13 +2,6 @@
 .SH NAME
 autofs \- Service control for the automounter
 .SH SYNOPSIS
-If a SysV init script system is being used:
-.br
-.B @@initdir@@/autofs
-.I start|stop|restart|reload|status
-.P
-or if the systemd init system is being used:
-.br
 .B systemctl
 .I start|stop|restart|reload|status
 .B autofs.service
@@ -37,8 +30,6 @@ by default to find mount points on the s
 will mount and start a thread, with the appropriate parameters, to
 manage the mount point.
 .P
-.B @@initdir@@/autofs reload
-or
 .B systemctl autofs.service reload
 will check the current auto.master map against the current automount managed
 mounts. It will terminate those daemons or threads (depending on
@@ -61,8 +52,6 @@ mounts cannot be updated due to potentia
 case the map entry offsets will not be updated until after the map
 entry has expired.
 .P
-.B @@initdir@@/autofs status
-or
 .B systemctl status autofs.service
 will display the status of,
 .BR automount (8),
