diff -ruN CoreFreq/corefreqk.c CoreFreq.patch/corefreqk.c
--- CoreFreq/corefreqk.c	2021-04-15 11:52:47.775838212 +0200
+++ CoreFreq.patch/corefreqk.c	2021-04-15 12:47:48.725250138 +0200
@@ -13753,7 +13753,6 @@
 #ifdef CONFIG_CPU_IDLE
 static void CoreFreqK_Idle_State_Withdraw(int idx, bool disable)
 {
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0)
 	struct cpuidle_device *device;
 	unsigned int cpu;
   for (cpu = 0; cpu < PUBLIC(RO(Proc))->CPU.Count; cpu++)
@@ -13768,9 +13767,6 @@
       }
     }
   }
-#else
-	CoreFreqK.IdleDriver.states[idx].disabled = disable;
-#endif /* 5.5.0 */
 }
 #endif /* CONFIG_CPU_IDLE */
 
@@ -13848,13 +13844,7 @@
 	return (0);
 }
 
-#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 19))	\
-  && (LINUX_VERSION_CODE <= KERNEL_VERSION(5, 5, 0)))	\
-  || (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 3))
 static int CoreFreqK_Policy_Verify(struct cpufreq_policy_data *policy)
-#else
-static int CoreFreqK_Policy_Verify(struct cpufreq_policy *policy)
-#endif
 {
 	if (policy != NULL) {
 		cpufreq_verify_within_cpu_limits(policy);
@@ -13890,14 +13880,10 @@
     }
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 8, 0)
 static int CoreFreqK_SetBoost(struct cpufreq_policy *policy, int state)
 {
 	UNUSED(policy);
-#else
-static int CoreFreqK_SetBoost(int state)
-{
-#endif /* 5.8.0 */
+
 	Controller_Stop(1);
 	TurboBoost_Enable = (state != 0);
 	Controller_Start(1);
diff -ruN CoreFreq/corefreqk.h CoreFreq.patch/corefreqk.h
--- CoreFreq/corefreqk.h	2021-04-15 11:52:47.779838223 +0200
+++ CoreFreq.patch/corefreqk.h	2021-04-15 12:35:09.479093783 +0200
@@ -4536,20 +4536,10 @@
 #endif
 static int CoreFreqK_Policy_Exit(struct cpufreq_policy*) ;
 static int CoreFreqK_Policy_Init(struct cpufreq_policy*) ;
-#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 19))   \
-  && (LINUX_VERSION_CODE <= KERNEL_VERSION(5, 5, 0)))  \
-  || (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 3))
 static int CoreFreqK_Policy_Verify(struct cpufreq_policy_data*) ;
-#else
-static int CoreFreqK_Policy_Verify(struct cpufreq_policy*) ;
-#endif
 static int CoreFreqK_SetPolicy(struct cpufreq_policy*) ;
 static int CoreFreqK_Bios_Limit(int, unsigned int*) ;
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 8, 0)
 static int CoreFreqK_SetBoost(struct cpufreq_policy*, int) ;
-#else
-static int CoreFreqK_SetBoost(int) ;
-#endif
 static ssize_t CoreFreqK_Show_SetSpeed(struct cpufreq_policy*, char*);
 static int CoreFreqK_Store_SetSpeed(struct cpufreq_policy*, unsigned int) ;
 #endif /* CONFIG_CPU_FREQ */
