--- corefreqd.service	2021-10-10 16:23:31.324358421 +0200
+++ corefreqd.service.new	2021-10-10 16:24:56.504466281 +0200
@@ -13,8 +13,10 @@
 ProtectHostname=true
 ProtectKernelLogs=true
 Type=simple
+ExecStartPre=/sbin/modprobe corefreqk
 ExecStart=corefreqd -q
 ExecStop=/bin/kill -QUIT $MAINPID
+ExecStopPost=-/sbin/modprobe -r corefreqk
 RemainAfterExit=no
 SuccessExitStatus=SIGQUIT SIGUSR1 SIGTERM
 
