Index: lxml-4.9.2/src/lxml/tests/test_io.py
===================================================================
--- lxml-4.9.2.orig/src/lxml/tests/test_io.py
+++ lxml-4.9.2/src/lxml/tests/test_io.py
@@ -15,6 +15,7 @@ from .common_imports import (
     read_file, write_to_file, BytesIO, tmpfile
 )
 
+import lxml
 
 class _IOTestCaseBase(HelperTestCase):
     """(c)ElementTree compatibility for IO functions/methods
@@ -304,6 +305,7 @@ class _IOTestCaseBase(HelperTestCase):
             os.unlink(f.name)
         self.assertEqual(utext, root.text)
 
+    @unittest.skipIf(lxml.etree.LIBXML_VERSION >= (2, 11, 1), "libxml2 regression has issues with utf16")
     def test_iterparse_utf16_bom(self):
         utext = _str('Søk på nettet')
         uxml = '<?xml version="1.0" encoding="UTF-16"?><p>%s</p>' % utext
