--- a/bindings/mono/ayatana-appindicator-sharp-0.1.pc.in
+++ b/bindings/mono/ayatana-appindicator-sharp-0.1.pc.in
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-assemblies_dir=${prefix}/lib/cli/ayatana-appindicator-sharp-0.1
+assemblies_dir=${libdir}/ayatana-appindicator-sharp-0.1
 
 Name: ayatana-appindicator-sharp
 Description: application indicators for .NET
--- a/bindings/mono/Makefile.am
+++ b/bindings/mono/Makefile.am
@@ -31,7 +31,7 @@ TARGET = \
 	$(DLLPOLICY1) \
 	$(POLICY1).config
 
-assemblydir = $(libdir)/cli/ayatana-appindicator-sharp-0.1
+assemblydir = $(libdir)/ayatana-appindicator-sharp-0.1
 assembly_DATA = $(TARGET)
 
 CLEANFILES = \
