diff -Pdpru lsyncd-release-2.2.3.orig/cmake/FindLua.cmake lsyncd-release-2.2.3/cmake/FindLua.cmake
--- lsyncd-release-2.2.3.orig/cmake/FindLua.cmake	2018-03-09 13:39:11.000000000 +0100
+++ lsyncd-release-2.2.3/cmake/FindLua.cmake	2018-04-26 14:39:46.042589696 +0200
@@ -43,7 +43,7 @@ FOREACH(_SUFFIX ${_POSSIBLE_SUFFIXES})
   LIST(APPEND _POSSIBLE_LUA_INCLUDE "include/lua${_SUFFIX}")
   LIST(APPEND _POSSIBLE_LUA_EXECUTABLE "lua${_SUFFIX}")
   LIST(APPEND _POSSIBLE_LUA_COMPILER "luac${_SUFFIX}")
-  LIST(APPEND _POSSIBLE_LUA_LIBRARY "lua${_SUFFIX}")
+  LIST(APPEND _POSSIBLE_LUA_LIBRARY "liblua.so.${_SUFFIX}")
 ENDFOREACH(_SUFFIX)
 
 # Find the lua executable
