--- a/src/main/java/com/jcraft/jsch/SSHAgentConnector.java
+++ b/src/main/java/com/jcraft/jsch/SSHAgentConnector.java
@@ -93,21 +93,7 @@ public class SSHAgentConnector implements AgentConnector {
   }
 
   private static USocketFactory getUSocketFactory() throws AgentProxyException {
-    try {
-      return new UnixDomainSocketFactory();
-    } catch (AgentProxyException e) {
-      try {
-        return new JUnixSocketFactory();
-      } catch (NoClassDefFoundError ee) {
-        AgentProxyException eee = new AgentProxyException("junixsocket library unavailable");
-        eee.addSuppressed(e);
-        eee.addSuppressed(ee);
-        throw eee;
-      } catch (AgentProxyException ee) {
-        ee.addSuppressed(e);
-        throw e;
-      }
-    }
+    return new UnixDomainSocketFactory();
   }
 
   private static Path getSshAuthSocket() throws AgentProxyException {
--- a/src/main/java9/module-info.java
+++ b/src/main/java9/module-info.java
@@ -7,6 +7,5 @@ module com.jcraft.jsch {
   requires static java.security.jgss;
   requires static org.apache.logging.log4j;
   requires static org.bouncycastle.provider;
-  requires static org.newsclub.net.unix;
   requires static org.slf4j;
 }
