Index: ocfs2-tools/include/ocfs2-kernel/ocfs2_fs.h
===================================================================
--- ocfs2-tools.orig/include/ocfs2-kernel/ocfs2_fs.h	2012-08-24 10:22:41.000000000 -0500
+++ ocfs2-tools/include/ocfs2-kernel/ocfs2_fs.h	2012-08-24 10:23:30.000000000 -0500
@@ -329,6 +329,9 @@
  */
 #define OCFS2_CLUSTER_O2CB_GLOBAL_HEARTBEAT	(0x01)
 
+// Reintroduce umode_t, which disappeared in kernel 3.3
+typedef unsigned short          umode_t;
+
 struct ocfs2_system_inode_info {
 	char	*si_name;
 	int	si_iflags;
