Index: b/Modules/FindRuby.cmake
===================================================================
--- a/Modules/FindRuby.cmake
+++ b/Modules/FindRuby.cmake
@@ -290,7 +290,7 @@
   _RUBY_CONFIG_VAR("sitelibdir" Ruby_SITELIB_DIR)
 
   # vendor_ruby available ?
-  execute_process(COMMAND ${Ruby_EXECUTABLE} -r vendor-specific -e "print 'true'"
+  execute_process(COMMAND ${Ruby_EXECUTABLE} -r rbconfig -e "print 'true' unless RbConfig::CONFIG['vendorarchdir'].nil?"
     OUTPUT_VARIABLE Ruby_HAS_VENDOR_RUBY  ERROR_QUIET)
 
   if(Ruby_HAS_VENDOR_RUBY)
