Index: cmake-3.26.2/Modules/FindPNG.cmake
===================================================================
--- cmake-3.26.2.orig/Modules/FindPNG.cmake
+++ cmake-3.26.2/Modules/FindPNG.cmake
@@ -54,12 +54,16 @@ endif()
 find_package(ZLIB ${_FIND_ZLIB_ARG})
 
 if(ZLIB_FOUND)
-  find_path(PNG_PNG_INCLUDE_DIR png.h PATH_SUFFIXES include/libpng)
+  set(_PNG_VERSION_SUFFIXES 17 16 15 14 12)
+  foreach(v IN LISTS _PNG_VERSION_SUFFIXES)
+    list(APPEND _PNG_INCLUDE_SUFFIXES libpng${v})
+  endforeach()
+  find_path(PNG_PNG_INCLUDE_DIR png.h PATH_SUFFIXES include/libpng ${_PNG_INCLUDE_SUFFIXES})
+  unset(_PNG_INCLUDE_SUFFIXES)
   mark_as_advanced(PNG_PNG_INCLUDE_DIR)
 
   list(APPEND PNG_NAMES png libpng)
   unset(PNG_NAMES_DEBUG)
-  set(_PNG_VERSION_SUFFIXES 17 16 15 14 12)
   if (PNG_FIND_VERSION MATCHES "^([0-9]+)\\.([0-9]+)(\\..*)?$")
     set(_PNG_VERSION_SUFFIX_MIN "${CMAKE_MATCH_1}${CMAKE_MATCH_2}")
     if (PNG_FIND_VERSION_EXACT)
