--- a/build.xml
+++ b/build.xml
@@ -185,6 +185,7 @@
         <javac srcdir="${src.dir}"
                destdir="${core.classes.build.dir}"
                sourcepath=""
+               classpathref="lib.classpath"
                source="${ivy.minimum.javaversion}"
                target="${ivy.minimum.javaversion}"
                debug="${debug.mode}"
--- a/src/java/org/apache/ivy/util/FileUtil.java
+++ b/src/java/org/apache/ivy/util/FileUtil.java
@@ -46,7 +46,7 @@ import java.util.jar.JarOutputStream;
 import java.util.zip.GZIPInputStream;
 import java.util.zip.ZipInputStream;
 
-import static java.util.jar.Pack200.newUnpacker;
+import static io.pack200.Pack200.newUnpacker;
 
 /**
  * Utility class used to deal with file related operations, like copy, full reading, symlink, ...
