From: Thomas Andrejak <thomas.andrejak@gmail.com>
Date: 2017-02-27 23:25:00 +0100
References: Specific
Upstream: no
Subject: Add default syslog format to work out of the box

--- prelude-lml.conf	2016-09-15 09:03:39.835000884 +0200
+++ prelude-lml.conf	2017-02-18 00:46:18.061525071 +0100
@@ -88,6 +88,13 @@
 #
 
 
+[format=rsyslog]
+time-format = "%Y-%m-%dT%H:%M:%S" 
+prefix-regex = "^(?P<timestamp>.{20}).{12} (?P<hostname>\S+) (?:(?P<process>\S+?)(?:\[(?P<pid>[0-9]+)\])?: )?"
+file = /var/log/messages
+file = /var/log/secure
+
+
 [format=syslog]
 time-format = "%b %d %H:%M:%S"
 prefix-regex = "^(?P<timestamp>.{15}) (?P<hostname>\S+) (?:(?P<process>\S+?)(?:\[(?P<pid>[0-9]+)\])?: )?"
--- prelude-lml.conf.in	2016-09-15 08:49:20.170000884 +0200
+++ prelude-lml.conf.in	2017-02-18 00:46:47.168897144 +0100
@@ -88,6 +88,13 @@
 #
 
 
+[format=rsyslog]
+time-format = "%Y-%m-%dT%H:%M:%S" 
+prefix-regex = "^(?P<timestamp>.{20}).{12} (?P<hostname>\S+) (?:(?P<process>\S+?)(?:\[(?P<pid>[0-9]+)\])?: )?"
+file = /var/log/messages
+file = /var/log/secure
+
+
 [format=syslog]
 time-format = "%b %d %H:%M:%S"
 prefix-regex = "^(?P<timestamp>.{15}) (?P<hostname>\S+) (?:(?P<process>\S+?)(?:\[(?P<pid>[0-9]+)\])?: )?"
