--- a/blueman/main/indicators/AppIndicator.py
+++ b/blueman/main/indicators/AppIndicator.py
@@ -2,8 +2,12 @@ from typing import Iterable, TYPE_CHECKI
 
 import gi
 
-gi.require_version('AppIndicator3', '0.1')
-from gi.repository import AppIndicator3
+try:
+    gi.require_version('AyatanaAppIndicator3', '0.1')
+    from gi.repository import AyatanaAppIndicator3 as AppIndicator3
+except (ValueError, ImportError):
+    gi.require_version('AppIndicator3', '0.1')
+    from gi.repository import AppIndicator3 as AppIndicator3
 from blueman.main.indicators.GtkStatusIcon import build_menu
 
 if TYPE_CHECKING:
