---
 dpme.h |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/dpme.h
+++ b/dpme.h
@@ -63,7 +63,7 @@
 //
 typedef	unsigned char	u8;
 typedef	unsigned short	u16;
-typedef	unsigned long	u32;
+typedef	unsigned int	u32;
 
 
 // Physical block zero of the disk has this format
@@ -115,10 +115,10 @@ struct dpme {
 #endif
     u32     dpme_boot_block         ;
     u32     dpme_boot_bytes         ;
-    u8     *dpme_load_addr          ;
-    u8     *dpme_load_addr_2        ;
-    u8     *dpme_goto_addr          ;
-    u8     *dpme_goto_addr_2        ;
+    u32 dpme_load_addr          ;
+    u32 dpme_load_addr_2        ;
+    u32 dpme_goto_addr          ;
+    u32 dpme_goto_addr_2        ;
     u32     dpme_checksum           ;
     char    dpme_process_id[16]     ;
     u32     dpme_boot_args[32]      ;
