diff -Nur dde-kwin-5.4.26/deepin-wm-dbus/CMakeLists.txt dde-kwin-5.4.26-new/deepin-wm-dbus/CMakeLists.txt
--- dde-kwin-5.4.26/deepin-wm-dbus/CMakeLists.txt	2022-05-16 21:17:05.695018935 +0800
+++ dde-kwin-5.4.26-new/deepin-wm-dbus/CMakeLists.txt	2022-05-16 21:17:49.254878145 +0800
@@ -8,6 +8,7 @@
 find_package(KF5WindowSystem REQUIRED)
 find_package(KF5GlobalAccel REQUIRED)
 find_package(DtkCore REQUIRED)
+find_package(X11 REQUIRED)
 find_package(PkgConfig REQUIRED)
 
 option(DISABLE_DEEPIN_WM "disable compatibility with deepin-wm" OFF)
diff -Nur dde-kwin-5.4.26/plugins/kdecoration/CMakeLists.txt dde-kwin-5.4.26-new/plugins/kdecoration/CMakeLists.txt
--- dde-kwin-5.4.26/plugins/kdecoration/CMakeLists.txt	2022-05-16 21:17:05.699018922 +0800
+++ dde-kwin-5.4.26-new/plugins/kdecoration/CMakeLists.txt	2022-05-16 21:18:14.570781127 +0800
@@ -10,6 +10,7 @@
 find_package(KF5WindowSystem REQUIRED)
 find_package(KDecoration2 REQUIRED)
 find_package(Qt5LinguistTools REQUIRED)
+find_package(X11 REQUIRED)
 include_directories(${Qt5Core_PRIVATE_INCLUDE_DIRS})
 include_directories(${Qt5Gui_PRIVATE_INCLUDE_DIRS})
 
diff -Nur dde-kwin-5.4.26/plugins/kwineffects/multitasking/CMakeLists.txt dde-kwin-5.4.26-new/plugins/kwineffects/multitasking/CMakeLists.txt
--- dde-kwin-5.4.26/plugins/kwineffects/multitasking/CMakeLists.txt	2022-02-22 19:37:10.000000000 +0800
+++ dde-kwin-5.4.26-new/plugins/kwineffects/multitasking/CMakeLists.txt	2022-05-16 21:18:54.506618901 +0800
@@ -74,6 +74,7 @@
 find_package(KF5I18n REQUIRED)
 find_package(PkgConfig REQUIRED)
 find_package(Qt5LinguistTools REQUIRED)
+find_package(X11 REQUIRED)
 
 pkg_search_module(QGSETTINGS REQUIRED IMPORTED_TARGET gsettings-qt)
 
diff -Nur dde-kwin-5.4.26/plugins/platforms/lib/CMakeLists.txt dde-kwin-5.4.26-new/plugins/platforms/lib/CMakeLists.txt
--- dde-kwin-5.4.26/plugins/platforms/lib/CMakeLists.txt	2022-05-16 21:17:05.699018922 +0800
+++ dde-kwin-5.4.26-new/plugins/platforms/lib/CMakeLists.txt	2022-05-16 21:19:21.982470960 +0800
@@ -7,6 +7,7 @@
 find_package(KF5Config REQUIRED)
 find_package(KF5WindowSystem REQUIRED)
 find_package(KF5CoreAddons REQUIRED)
+find_package(X11 REQUIRED)
 
 if (${KF5WindowSystem_VERSION_MINOR} GREATER 45)
     add_definitions(-DKWINDOWSYSTEM_NO_QWIDGET)
diff -Nur dde-kwin-5.4.26/plugins/platforms/plugin/CMakeLists.txt dde-kwin-5.4.26-new/plugins/platforms/plugin/CMakeLists.txt
--- dde-kwin-5.4.26/plugins/platforms/plugin/CMakeLists.txt	2022-02-22 19:37:10.000000000 +0800
+++ dde-kwin-5.4.26-new/plugins/platforms/plugin/CMakeLists.txt	2022-05-16 21:19:37.386388010 +0800
@@ -10,6 +10,7 @@
 find_package(Qt5DBus REQUIRED)
 find_package(Qt5Widgets REQUIRED)
 find_package(DtkGui REQUIRED)
+find_package(X11 REQUIRED)
 
 include_directories(${Qt5Core_PRIVATE_INCLUDE_DIRS})
 include_directories(${Qt5Gui_PRIVATE_INCLUDE_DIRS})
diff -Nur dde-kwin-5.4.26/tests/CMakeLists.txt dde-kwin-5.4.26-new/tests/CMakeLists.txt
--- dde-kwin-5.4.26/tests/CMakeLists.txt	2022-02-22 19:37:10.000000000 +0800
+++ dde-kwin-5.4.26-new/tests/CMakeLists.txt	2022-05-16 21:20:11.890222459 +0800
@@ -29,6 +29,7 @@
 find_package(PkgConfig REQUIRED)
 find_package(Qt5LinguistTools REQUIRED)
 find_package(Qt5X11Extras REQUIRED)
+find_package(X11 REQUIRED)
 
 pkg_search_module(QGSETTINGS REQUIRED IMPORTED_TARGET gsettings-qt)
 
