Index: libcap/cap_extint.c
===================================================================
--- libcap/cap_extint.c.orig
+++ libcap/cap_extint.c
@@ -102,7 +102,7 @@ cap_t cap_copy_int(const void *cap_ext)
 
     to = (__u32 *) &cap_d->set;
     blen = export->length_of_capset;
-    for (set=0; set<=NUMBER_OF_CAP_SETS; ++set) {
+    for (set=0; set<NUMBER_OF_CAP_SETS; ++set) {
 	int blk;
 	int bno = 0;
 	for (blk=0; blk<(CAP_SET_SIZE/4); ++blk) {
