From 1ab9fcb7361898f117057280225ac9132abfa55b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Klaus=20K=C3=A4mpf?= <kkaempf@suse.de>
Date: Tue, 28 Nov 2017 15:40:03 +0100
Subject: [PATCH 23/25] Alias sblim-sfcb service to sfcb and sfcbd

---
 sblim-sfcb.service.pre.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sblim-sfcb.service.pre.in b/sblim-sfcb.service.pre.in
index 3b49da3472f1..11e77f1cd022 100644
--- a/sblim-sfcb.service.pre.in
+++ b/sblim-sfcb.service.pre.in
@@ -7,3 +7,5 @@ ExecStart=${exec_prefix}/sbin/sfcbd
 
 [Install]
 WantedBy=multi-user.target
+Alias=sfcbd.service
+Alias=sfcb.service
-- 
2.15.0

