diff -Nur deepin-compressor-5.11.7/CMakeLists.txt deepin-compressor-5.11.7-new/CMakeLists.txt
--- deepin-compressor-5.11.7/CMakeLists.txt	2022-01-31 11:12:43.045264000 +0800
+++ deepin-compressor-5.11.7-new/CMakeLists.txt	2022-01-31 11:13:12.181082651 +0800
@@ -87,6 +87,7 @@
 
 set(CMAKE_INSTALL_PREFIX /usr)
 
+install(TARGETS compressor-ChardetDetector DESTINATION ${CMAKE_INSTALL_LIBDIR}/deepin-compressor/plugins/)
 install(TARGETS cli7zplugin DESTINATION ${CMAKE_INSTALL_LIBDIR}/deepin-compressor/plugins/)
 install(TARGETS clirarplugin DESTINATION ${CMAKE_INSTALL_LIBDIR}/deepin-compressor/plugins/)
 install(TARGETS libzipplugin DESTINATION ${CMAKE_INSTALL_LIBDIR}/deepin-compressor/plugins/)
