Index: icewm-config-openSUSE-icewm-config-openSUSE-1.2.4/preferences
===================================================================
--- icewm-config-openSUSE-icewm-config-openSUSE-1.2.4.orig/preferences
+++ icewm-config-openSUSE-icewm-config-openSUSE-1.2.4/preferences
@@ -539,7 +539,7 @@ MenuMouseTracking=1 # 0/1
 # NetWorkAreaBehaviour=0 # [0-2]
 
 #  Icon search path (colon separated)
-IconPath="/usr/share/icons/Adwaita/16x16/apps:/usr/share/icons/hicolor/16x16/apps:/usr/share/pixmaps"
+IconPath="/usr/share/icons/Adwaita/16x16/legacy:/usr/share/icons/Adwaita/16x16/apps:/usr/share/icons/hicolor/16x16/apps:/usr/share/pixmaps"
 
 #  Mailbox path (use $MAIL instead)
 # MailBoxPath=""
