Index: icewm-config-openSUSE-icewm-config-openSUSE-1.2.5/startup
===================================================================
--- icewm-config-openSUSE-icewm-config-openSUSE-1.2.5.orig/startup
+++ icewm-config-openSUSE-icewm-config-openSUSE-1.2.5/startup
@@ -14,3 +14,8 @@ done;
 # Start xscreensaver daemon
 
 [ -x /usr/bin/xscreensaver ] && /usr/bin/xscreensaver > /dev/null &
+
+# Pass environment variables to systemd --user session (to fix bsc#1179237)
+if command -v dbus-update-activation-environment >/dev/null 2>&1; then
+            dbus-update-activation-environment --systemd --all
+fi
