Index: b/src/roff/nroff/nroff.sh
===================================================================
--- a/src/roff/nroff/nroff.sh
+++ b/src/roff/nroff/nroff.sh
@@ -131,7 +131,7 @@ then
     *)
       # Some old shells don't support ${FOO:-bar} expansion syntax.  We
       # should switch to it when it is safe to abandon support for them.
-      case "${LC_ALL-${LC_CTYPE-${LANG}}}" in
+      case "${LC_ALL:-${LC_CTYPE:-${LANG}}}" in
         *.UTF-8)
           Tloc=utf8 ;;
         iso_8859_1 | *.ISO-8859-1 | *.ISO8859-1 | \
