Index: squashfs-tools-4.6.1/squashfs-tools/mksquashfs.c
===================================================================
--- squashfs-tools-4.6.1.orig/squashfs-tools/mksquashfs.c
+++ squashfs-tools-4.6.1/squashfs-tools/mksquashfs.c
@@ -6378,7 +6378,7 @@ static void print_options(FILE *stream,
 	fprintf(stream, "file on\n\t\t\tencountering them. This option makes ");
 	fprintf(stream, "Mksquashfs ignore\n\t\t\tthe zero filled blocks\n");
 	fprintf(stream, "\nExpert options (these may make the filesystem unmountable):\n");
-	fprintf(stream, "-nopad\t\t\tdo not pad filesystem to a multiple of 4K\n");
+	fprintf(stream, "-nopad\t\t\tdo not pad filesystem to a multiple of 64K\n");
 	fprintf(stream, "-offset <offset>\tskip <offset> bytes at the beginning of ");
 	fprintf(stream, "FILESYSTEM.\n\t\t\tOptionally a suffix of K, M or G can be given ");
 	fprintf(stream, "to specify\n\t\t\tKbytes, Mbytes or Gbytes respectively.\n");
@@ -6606,7 +6606,7 @@ static void print_sqfstar_options(FILE *
 	fprintf(stream, "memory for caches.  Default 25%%\n");
 	fprintf(stream, "-mem-default\t\tprint default memory usage in Mbytes\n");
 	fprintf(stream, "\nExpert options (these may make the filesystem unmountable):\n");
-	fprintf(stream, "-nopad\t\t\tdo not pad filesystem to a multiple of 4K\n");
+	fprintf(stream, "-nopad\t\t\tdo not pad filesystem to a multiple of 64K\n");
 	fprintf(stream, "-offset <offset>\tskip <offset> bytes at the beginning of ");
 	fprintf(stream, "FILESYSTEM.\n\t\t\tOptionally a suffix of K, M or G can be given ");
 	fprintf(stream, "to specify\n\t\t\tKbytes, Mbytes or Gbytes respectively.\n");
@@ -7605,9 +7605,9 @@ print_sqfstar_compressor_options:
 	set_progressbar_state(FALSE);
 	write_filesystem_tables(&sBlk);
 
-	if(!nopad && (i = bytes & (4096 - 1))) {
-		char temp[4096] = {0};
-		write_destination(fd, bytes, 4096 - i, temp);
+	if(!nopad && (i = bytes & (65536 - 1))) {
+		char temp[65536] = {0};
+		write_destination(fd, bytes, 65536 - i, temp);
 	}
 
 	res = close(fd);
@@ -8877,10 +8877,10 @@ print_compressor_options:
 
 	set_progressbar_state(FALSE);
 	write_filesystem_tables(&sBlk);
-
-	if(!nopad && (i = bytes & (4096 - 1))) {
-		char temp[4096] = {0};
-		write_destination(fd, bytes, 4096 - i, temp);
+	
+	if(!nopad && (i = bytes & (65536 - 1))) {
+		char temp[65536] = {0};
+		write_destination(fd, bytes, 65536 - i, temp);
 	}
 
 	res = close(fd);
