--- squashfs4.3/squashfs-tools/mksquashfs.c.orig	2014-05-13 00:18:20.000000000 +0200
+++ squashfs4.3/squashfs-tools/mksquashfs.c	2014-05-16 14:59:30.802144932 +0200
@@ -4674,9 +4674,9 @@ void write_filesystem_tables(struct squa
 	SQUASHFS_INSWAP_SUPER_BLOCK(sBlk); 
 	write_destination(fd, SQUASHFS_START, sizeof(*sBlk), sBlk);
 
-	if(!nopad && (i = bytes & (4096 - 1))) {
-		char temp[4096] = {0};
-		write_destination(fd, bytes, 4096 - i, temp);
+	if(!nopad && (i = bytes & (65536 - 1))) {
+		char temp[65536] = {0};
+		write_destination(fd, bytes, 65536 - i, temp);
 	}
 
 	close(fd);
@@ -5309,7 +5309,7 @@ printOptions:
 			ERROR("-force-uid uid\t\tset all file uids to uid\n");
 			ERROR("-force-gid gid\t\tset all file gids to gid\n");
 			ERROR("-nopad\t\t\tdo not pad filesystem to a multiple "
-				"of 4K\n");
+				"of 64K\n");
 			ERROR("-keep-as-directory\tif one source directory is "
 				"specified, create a root\n");
 			ERROR("\t\t\tdirectory containing that directory, "
