diff -ruNp a/src/AV/Output/BaseEncoder.cpp b/src/AV/Output/BaseEncoder.cpp
--- a/src/AV/Output/BaseEncoder.cpp	2025-10-13 07:28:54.284175639 +0200
+++ b/src/AV/Output/BaseEncoder.cpp	2025-10-13 07:26:21.043996527 +0200
@@ -176,7 +176,9 @@ void BaseEncoder::Init(AVCodec* codec, A
 
 void BaseEncoder::Free() {
 	if(m_codec_opened) {
+#if !SSR_USE_AVCODEC_CLOSE_DEPRECATED
 		avcodec_close(m_codec_context);
+#endif
 		m_codec_opened = false;
 	}
 }
diff -ruNp a/src/Global.h b/src/Global.h
--- a/src/Global.h	2025-10-13 07:29:04.995166923 +0200
+++ b/src/Global.h	2025-10-13 07:27:33.455932767 +0200
@@ -257,6 +257,9 @@ inline void atomic_thread_fence_replacem
 // avformat_free_context: lavf 52.96.0 / 52.96.0
 #define SSR_USE_AVFORMAT_FREE_CONTEXT              TEST_AV_VERSION(LIBAVFORMAT, 52, 96, 52, 96)
 
+// avcodec_close deprecated: lavc 60.39.100 / ???
+// - ffmpeg: missing, commit: https://github.com/FFmpeg/FFmpeg/commit/1cc24d749569a42510399a29b034f7a77bdec34e
+#define SSR_USE_AVCODEC_CLOSE_DEPRECATED           TEST_AV_VERSION(LIBAVCODEC, 60, 39, 999, 999)
 // av_codec_iterate: lavc 58.10.100 / ???
 #define SSR_USE_AV_CODEC_ITERATE                   TEST_AV_VERSION(LIBAVCODEC, 58, 10, 999, 999)
 // av_lockmgr_register deprecated: lavc 58.9.100 / ???
