From 30b1223b4979a18132d599792c995264c299ba4e Mon Sep 17 00:00:00 2001
From: "Joshua J. Cogliati" <jrincayc@yahoo.com>
Date: Wed, 10 Jan 2024 20:06:12 -0700
Subject: [PATCH] Define SIG_TAKES_ARG in unix.

---
 logo.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/logo.h b/logo.h
index 314173f..024c6be 100644
--- a/logo.h
+++ b/logo.h
@@ -69,6 +69,7 @@
 #ifndef unix
 #define unix
 #endif
+#define SIG_TAKES_ARG
 #include "config.h"
 #ifndef X_DISPLAY_MISSING
 #define x_window
From d300824c6eaf5f98faded7c51ea799b32be7a50b Mon Sep 17 00:00:00 2001
From: "Joshua J. Cogliati" <jrincayc@yahoo.com>
Date: Fri, 12 Jan 2024 10:05:45 -0700
Subject: [PATCH] Fixing incorrect return and wrong type warnings.

---
 term.c      |  4 ++--
 xgraphics.c | 10 +++++-----
 xgraphics.h | 12 +++++++++---
 3 files changed, 16 insertions(+), 10 deletions(-)

diff --git a/term.c b/term.c
index 48ddc425..17c697bd 100644
--- a/term.c
+++ b/term.c
@@ -95,8 +95,8 @@ extern char **environ, *tgoto(), *tgetstr();
 
 char *termcap_ptr;
 
-int termcap_putter(char ch) {
-    *termcap_ptr++ = ch;
+int termcap_putter(int ch) {
+    *termcap_ptr++ = (char)ch;
     return 0;
 }
 
diff --git a/xgraphics.c b/xgraphics.c
index 5d1c8d48..afc750a9 100644
--- a/xgraphics.c
+++ b/xgraphics.c
@@ -268,7 +268,7 @@ void placate_x()
   XConfigureEvent *xce;
   XMotionEvent    *xme;
   XButtonEvent    *xbe;
-    checkX;
+  checkX();
 
   while(XCheckWindowEvent(dpy, win, EVENT_MASK, (XEvent *)&event))
 
@@ -310,14 +310,14 @@ void check_X11_stop() {
 
     if (--count == 0) {
 	count = 300;
-	checkX;
+	checkX();
 	placate_x();
     }
 }
 
 int get_button()
 {
-  checkX;
+  checkX(0);
 
   placate_x();
 
@@ -326,7 +326,7 @@ int get_button()
 
 int get_mouse_x()
 {
-  checkX;
+  checkX(-1);
 
   placate_x();
 
@@ -336,7 +336,7 @@ int get_mouse_x()
 
 int get_mouse_y()
 {
-  checkX;
+  checkX(-1);
 
   placate_x();
 
diff --git a/xgraphics.h b/xgraphics.h
index 64b01fea..8d4aeae3 100644
--- a/xgraphics.h
+++ b/xgraphics.h
@@ -22,15 +22,21 @@ void logofill(void);
 
 #define GR_SIZE         60000
 
-#define checkX { \
+#define checkX(ret) {			\
+    if (have_x < 0) real_window_init(); \
+    if (!have_x) { \
+	err_logo(BAD_GRAPH_INIT,NIL); \
+	return ret; \
+    } \
+}
+#define checkXnoreturn { \
     if (have_x < 0) real_window_init(); \
     if (!have_x) { \
 	err_logo(BAD_GRAPH_INIT,NIL); \
-	return; \
     } \
 }
 
-#define prepare_to_draw          {checkX; placate_x();}
+#define prepare_to_draw          {checkXnoreturn; placate_x();}
 #define done_drawing             XFlush(dpy)
 extern void placate_x();
 
