From: Antonio Larrosa <alarrosa@suse.com>
Subject: Fix undefined reference to XSetTransientForHint

Index: QGnomePlatform-0.9.0/CMakeLists.txt
===================================================================
--- QGnomePlatform-0.9.0.orig/CMakeLists.txt
+++ QGnomePlatform-0.9.0/CMakeLists.txt
@@ -106,6 +106,8 @@
     endif()
 endif()

+find_package(X11 REQUIRED)
+
 add_subdirectory(src)

 feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)

Index: QGnomePlatform-0.9.0/src/theme/CMakeLists.txt
===================================================================
--- QGnomePlatform-0.9.0.orig/src/theme/CMakeLists.txt
+++ QGnomePlatform-0.9.0/src/theme/CMakeLists.txt
@@ -19,6 +19,7 @@
     Qt${QT_VERSION_MAJOR}::QuickControls2
     ${ADWAITAQT_LIBRARIES}
     PkgConfig::GTK+3
+    X11::X11
 )

 if (NOT USE_QT6)
