From 8b9a2cb2879012dfa2e666a6c33be183da532d47 Mon Sep 17 00:00:00 2001
From: Mikolaj Izdebski <mizdebsk@redhat.com>
Date: Fri, 22 Apr 2022 11:15:38 +0200
Subject: [PATCH 6/7] Restore DefaultModelValidator compatibility with Maven
 3.5.4

---
 .../maven/model/validation/DefaultModelValidator.java       | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java b/maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java
index f77321c16..4ed22f3ed 100644
--- a/maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java
+++ b/maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java
@@ -44,6 +44,7 @@
 import org.apache.maven.model.building.ModelProblem.Version;
 import org.apache.maven.model.building.ModelProblemCollector;
 import org.apache.maven.model.building.ModelProblemCollectorRequest;
+import org.apache.maven.model.interpolation.DefaultModelVersionProcessor;
 import org.apache.maven.model.interpolation.ModelVersionProcessor;
 import org.codehaus.plexus.util.StringUtils;
 
@@ -85,6 +86,11 @@
 
     private ModelVersionProcessor versionProcessor;
 
+    public DefaultModelValidator()
+    {
+        this( new DefaultModelVersionProcessor() );
+    }
+
     @Inject
     public DefaultModelValidator(  ModelVersionProcessor versionProcessor )
     {
-- 
2.36.1

