From 26c805f5b38bed987461ff51a0f6461827c43236 Mon Sep 17 00:00:00 2001
From: Eugenio Paolantonio <eugenio.paolantonio@suse.com>
Date: Fri, 20 Jun 2025 18:18:02 +0200
Subject: [PATCH] commands: build: ensure supportstatus_override is passed to
 parse_supportstatus()

The function signature changed after the recent rework.

Signed-off-by: Eugenio Paolantonio <eugenio.paolantonio@suse.com>
---
 src/productcomposer/commands/build.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/productcomposer/commands/build.py b/src/productcomposer/commands/build.py
index 6c5318c..4cd8789 100644
--- a/src/productcomposer/commands/build.py
+++ b/src/productcomposer/commands/build.py
@@ -69,7 +69,7 @@ def build(self, args):
 
         supportstatus_fn = os.path.join(directory, 'supportstatus.txt')
         if os.path.isfile(supportstatus_fn):
-            parse_supportstatus(supportstatus_fn)
+            parse_supportstatus(supportstatus_fn, supportstatus_override)
 
         if args.euladir and os.path.isdir(args.euladir):
             parse_eulas(args.euladir, eulas)
