Index: ltrace-ltrace-0.8.0/filter.c
===================================================================
--- ltrace-ltrace-0.8.0.orig/filter.c
+++ ltrace-ltrace-0.8.0/filter.c
@@ -18,6 +18,8 @@
  * 02110-1301 USA
  */
 
+#include "config.h"
+
 #include <stdlib.h>
 #include <assert.h>
 #include <stdio.h>
Index: ltrace-ltrace-0.8.0/options.h
===================================================================
--- ltrace-ltrace-0.8.0.orig/options.h
+++ ltrace-ltrace-0.8.0/options.h
@@ -25,6 +25,8 @@
 #ifndef OPTIONS_H
 #define OPTIONS_H
 
+#include "config.h"
+
 #include <stdio.h>
 #include <stdbool.h>
 #include <sys/types.h>
