--- dirvish.pl	2005-02-19 02:07:53.000000000 +0100
+++ dirvish.pl	2011-04-05 15:57:06.274127906 +0200
@@ -632,6 +632,9 @@
 	open (STDOUT, ">$log_temp");
 
 	$status{code} = (system(@cmd) >> 8) & 255;
+		# set the status code to 0 when the status code has been
+		# Status: warning (24) -- file vanished on sender 
+	$status{code} = 0 if ($status{code} == 24);
 
 	open (STDERR, ">&ERRHOLD");
 	open (STDOUT, ">&OUTHOLD");
