diff -Naur dirvish-1.2.1.original/install.sh dirvish-1.2.1/install.sh
--- dirvish-1.2.1.original/install.sh	2005-02-18 20:56:13.000000000 -0400
+++ dirvish-1.2.1/install.sh	2006-09-22 22:56:04.000000000 -0400
@@ -20,12 +20,12 @@
 	then
 		PERL=/usr/bin/perl
 	fi
-	echo $ECHO_N "perl to use ($PERL) $ECHO_C"
-	read ans
-	if [ -n "$ans" ] 
-	then
-		PERL="$ans"
-	fi
+#	echo $ECHO_N "perl to use ($PERL) $ECHO_C"
+#	read ans
+#	if [ -n "$ans" ] 
+#	then
+#		PERL="$ans"
+#	fi
 
 	until [ -n "$PREFIX_OK" ]
 	do
@@ -33,101 +33,101 @@
 		CONFDIR="/etc/dirvish"
 		MANDIR="/usr/share/man"
 
-		echo $ECHO_N "What installation prefix should be used? ($PREFIX) $ECHO_C"
-		read ans
-		if [ -n "$ans" ] 
-		then
-			PREFIX="$ans"
-                        # KHL 2005-02-18  == changed to = in "if" below
-			if [ "$PREFIX" = "/" ]
-			then
-				PREFIX=""
-			fi
-		fi
-		if [ -n "$PREFIX" -a  ! -d "$PREFIX" ]
-		then
-			echo $ECHO_N "$PREFIX doesn't exist, create it? (n) $ECHO_C"
-			read ans
-			if [ `expr "$ans" : '[yY]'` -ne 0 ]
-			then
+#		echo $ECHO_N "What installation prefix should be used? ($PREFIX) $ECHO_C"
+#		read ans
+#		if [ -n "$ans" ] 
+#		then
+#			PREFIX="$ans"
+#                        # KHL 2005-02-18  == changed to = in "if" below
+#			if [ "$PREFIX" = "/" ]
+#			then
+#				PREFIX=""
+#			fi
+#		fi
+#		if [ -n "$PREFIX" -a  ! -d "$PREFIX" ]
+#		then
+#			echo $ECHO_N "$PREFIX doesn't exist, create it? (n) $ECHO_C"
+#			read ans
+#			if [ `expr "$ans" : '[yY]'` -ne 0 ]
+#			then
 				CREATE_PREFIX="$PREFIX directory will be created"
 				PREFIX_OK="yes"
-			else
-				continue
-			fi
-		else
-			PREFIX_OK="yes"
-		fi
-
-		if [ -d "$PREFIX/sbin" ]
-		then
-			BINDIR=$PREFIX/sbin
-		else
+#			else
+#				continue
+#			fi
+#		else
+#			PREFIX_OK="yes"
+#		fi
+
+#		if [ -d "$PREFIX/sbin" ]
+#		then
+#			BINDIR=$PREFIX/sbin
+#		else
 			BINDIR=$PREFIX/bin
-		fi
+#		fi
 
-		if [ -d "$PREFIX/share/man" ]
-		then
+#		if [ -d "$PREFIX/share/man" ]
+#		then
 			MANDIR=$PREFIX/share/man
-		elif [ -d "$PREFIX/usr/share/man" ]
-		then
-			MANDIR=$PREFIX/usr/share/man
-		elif [ -d "$PREFIX/usr/man" ]
-		then
-			MANDIR="$PREFIX/usr/man" ]
-		else
-			MANDIR=$PREFIX/man
-		fi
-		if [ `expr "$PREFIX" : '.*dirvish.*'` -gt 0 ]
-		then
-			CONFDIR="$PREFIX/etc"
-		else
+#		elif [ -d "$PREFIX/usr/share/man" ]
+#		then
+#			MANDIR=$PREFIX/usr/share/man
+#		elif [ -d "$PREFIX/usr/man" ]
+#		then
+#			MANDIR="$PREFIX/usr/man" ]
+#		else
+#			MANDIR=$PREFIX/man
+#		fi
+#		if [ `expr "$PREFIX" : '.*dirvish.*'` -gt 0 ]
+#		then
+#			CONFDIR="$PREFIX/etc"
+#		else
 			CONFDIR="/etc/dirvish"
-		fi
+#		fi
 	done
 
 
-	echo $ECHO_N "Directory to install executables? ($BINDIR) $ECHO_C"
-	read ans
-	if [ -n "$ans" ] 
-	then
-		BINDIR="$ans"
-	fi
-
-	echo $ECHO_N "Directory to install MANPAGES? ($MANDIR) $ECHO_C"
-	read ans
-	if [ -n "$ans" ] 
-	then
-		MANDIR="$ans"
-	fi
-
-	echo $ECHO_N "Configuration directory ($CONFDIR) $ECHO_C"
-	read ans
-	if [ -n "$ans" ] 
-	then
-		CONFDIR="$ans"
-	fi
-
-	cat <<EOSTAT
-
-Perl executable to use is $PERL
-Dirvish executables to be installed in $BINDIR
-Dirvish manpages to be installed in $MANDIR
-Dirvish will expect its configuration files in $CONFDIR
-
-$CREATE_PREFIX
-
-EOSTAT
-
-	echo $ECHO_N "Is this correct? (no/yes/quit) $ECHO_C"
-	read ans
-	if [ `expr "$ans" : '[qQ]'` -ne 0 ]
-	then
-		exit
-	elif [ `expr "$ans" : '[yY]'` -ne 0 ]
-	then
+#	echo $ECHO_N "Directory to install executables? ($BINDIR) $ECHO_C"
+#	read ans
+#	if [ -n "$ans" ] 
+#	then
+#		BINDIR="$ans"
+#	fi
+#
+#	echo $ECHO_N "Directory to install MANPAGES? ($MANDIR) $ECHO_C"
+#	read ans
+#	if [ -n "$ans" ] 
+#	then
+#		MANDIR="$ans"
+#	fi
+#
+#	echo $ECHO_N "Configuration directory ($CONFDIR) $ECHO_C"
+#	read ans
+#	if [ -n "$ans" ] 
+#	then
+#		CONFDIR="$ans"
+#	fi
+#
+#	cat <<EOSTAT
+#
+#Perl executable to use is $PERL
+#Dirvish executables to be installed in $BINDIR
+#Dirvish manpages to be installed in $MANDIR
+#Dirvish will expect its configuration files in $CONFDIR
+#
+#$CREATE_PREFIX
+#
+#EOSTAT
+#
+#	echo $ECHO_N "Is this correct? (no/yes/quit) $ECHO_C"
+#	read ans
+#	if [ `expr "$ans" : '[qQ]'` -ne 0 ]
+#	then
+#		exit
+#	elif [ `expr "$ans" : '[yY]'` -ne 0 ]
+#	then
 		break
-	fi
+#	fi
 done
 
 HEADER="#!$PERL
@@ -148,10 +148,10 @@
 echo "Executables created."
 echo
 
-echo $ECHO_N "Install executables and manpages? (no/yes) $ECHO_C"
-read ans
-if [ `expr "$ans" : '[yY]'` -ne 0 ]
-then
+#echo $ECHO_N "Install executables and manpages? (no/yes) $ECHO_C"
+#read ans
+#if [ `expr "$ans" : '[yY]'` -ne 0 ]
+#then
 	echo
 	if [ -n "$CREATE_PREFIX" ]
 	then
@@ -174,10 +174,10 @@
 		mkdir -p "$MANDIR"
 	fi
 
-	if [ ! -d "$CONFDIR" ]
-	then
-		mkdir -p "$CONFDIR"
-	fi
+#	if [ ! -d "$CONFDIR" ]
+#	then
+#		mkdir -p "$CONFDIR"
+#	fi
 
 	for f in $EXECUTABLES
 	do
@@ -198,15 +198,15 @@
 	done
 	echo
 	echo "Installation complete"
-fi
+#fi
 
-echo $ECHO_N "Clean installation directory? (no/yes) $ECHO_C"
-read ans
-if [ `expr "$ans" : '[yY]'` -ne 0 ]
-then
+#echo $ECHO_N "Clean installation directory? (no/yes) $ECHO_C"
+#read ans
+#if [ `expr "$ans" : '[yY]'` -ne 0 ]
+#then
 	for f in $EXECUTABLES
 	do
 		rm $f
 	done
 	echo "Install directory cleaned."
-fi
+#fi
