From b5f67968332fa680e8af51be41d26aa47cfd3a67 Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Thu, 19 Sep 2019 10:43:03 +0200
Subject: [PATCH] Use qdbus-qt5

---
 startkde/startplasma-wayland.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: plasma-workspace-5.17.80git.20210531T011335~40d302afe/startkde/startplasma-wayland.cpp
===================================================================
--- plasma-workspace-5.17.80git.20210531T011335~40d302afe.orig/startkde/startplasma-wayland.cpp	2021-05-31 03:13:35.000000000 +0200
+++ plasma-workspace-5.17.80git.20210531T011335~40d302afe/startkde/startplasma-wayland.cpp	2021-05-31 06:59:35.188774369 +0200
@@ -71,7 +71,7 @@
     runEnvironmentScripts();
 
     if (!qEnvironmentVariableIsSet("DBUS_SESSION_BUS_ADDRESS")) {
-        out << "startplasmacompositor: Could not start D-Bus. Can you call qdbus?\n";
+        out << "startplasmacompositor: Could not start D-Bus. Can you call qdbus-qt5?\n";
         return 1;
     }
     setupPlasmaEnvironment();
