---
 crypto/fips/fips_sli.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/crypto/fips/fips_sli.c
+++ b/crypto/fips/fips_sli.c
@@ -328,14 +328,14 @@ static FIPS_STATUS get_fips_keygen_ecdsa
     BN_set_bit(pwr24, 24);
     BN_set_bit(pwr32, 32);
 
-    if (224 < n && n <= 255) {
+    if (224 <= n && n <= 255) {
         if (BN_cmp(cofactor, pwr14) != 1)
             ret = FIPS_APPROVED;
-    } else if (256 < n && n <= 383) {
+    } else if (256 <= n && n <= 383) {
         if (BN_cmp(cofactor, pwr16) != 1)
             ret = FIPS_APPROVED;
 
-    } else if (384 < n && n <= 511) {
+    } else if (384 <= n && n <= 511) {
         if (BN_cmp(cofactor, pwr24) != 1)
             ret = FIPS_APPROVED;
 
