diff -Nur pbuilder-0.231/Makefile new/Makefile
--- pbuilder-0.231/Makefile	2018-11-23 11:51:49.000000000 +0100
+++ new/Makefile	2022-05-04 11:08:27.273634062 +0200
@@ -154,7 +154,6 @@
 
 all:
 	$(MAKE) -C pbuildd $@
-	$(MAKE) -C Documentation $@
 
 check: export PBUILDER_CHECKOUT := $(CURDIR)
 check:
@@ -175,7 +174,6 @@
 		testsuite/testbuild2 \
 		testsuite/regression/work
 	$(MAKE) -C pbuildd $@
-	$(MAKE) -C Documentation $@
 
 TAGS:
 	etags pbuilder-* pbuilder
@@ -192,5 +190,4 @@
 	# install -aptitude flavour as the default satisfydepends
 	ln -sf pbuilder-satisfydepends-aptitude $(PKGLIBDIR)/pbuilder-satisfydepends
 	$(MAKE) -C pbuildd $@
-	$(MAKE) -C Documentation $@
 
