diff -ur pbuilder-0.199+nmu1.orig/pbuilderrc pbuilder-0.199+nmu1/pbuilderrc
--- pbuilder-0.199+nmu1.orig/pbuilderrc	2010-07-05 15:06:08.000000000 +0200
+++ pbuilder-0.199+nmu1/pbuilderrc	2011-01-26 17:30:14.000000000 +0100
@@ -77,7 +77,6 @@
 # Set the debootstrap variant to 'buildd' type.
 DEBOOTSTRAPOPTS=(
     '--variant=buildd'
-    '--keyring' '/usr/share/keyrings/debian-archive-keyring.gpg'
     )
 # or unset it to make it not a buildd type.
 # unset DEBOOTSTRAPOPTS
