diff --git a/CMakeLists.txt b/CMakeLists.txt
index c58e5c5..6a24b7d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -255,12 +255,12 @@ INSTALL(FILES
 	${CMAKE_SOURCE_DIR}/README.md
 	DESTINATION ${DATA_INSTALL_DIR})
 IF(UNIX AND NOT APPLE AND NOT BEOS AND NOT HAIKU)
-	INSTALL(FILES ${CMAKE_SOURCE_DIR}/build/linux/io.github.cxong.cdogs-sdl.desktop DESTINATION ${INSTALL_PREFIX}/share/applications)
-	INSTALL(FILES ${CMAKE_SOURCE_DIR}/build/linux/io.github.cxong.cdogs-sdl.appdata.xml DESTINATION ${INSTALL_PREFIX}/share/appdata)
+	INSTALL(FILES ${CMAKE_SOURCE_DIR}/build/linux/io.github.cxong.cdogs-sdl.desktop DESTINATION ${INSTALL_PREFIX}/share/applications/ RENAME cdogs-sdl.desktop)
+	INSTALL(FILES ${CMAKE_SOURCE_DIR}/build/linux/io.github.cxong.cdogs-sdl.appdata.xml DESTINATION ${INSTALL_PREFIX}/share/appdata/ RENAME cdogs-sdl.appdata.xml)
 	foreach(RES 16 22 32 48 128)
 		INSTALL(FILES ${CMAKE_SOURCE_DIR}/build/linux/cdogs-icon.${RES}.png
 			DESTINATION ${INSTALL_PREFIX}/share/icons/hicolor/${RES}x${RES}/apps
-			RENAME io.github.cxong.cdogs-sdl.png)
+			RENAME cdogs-sdl.png)
 	endforeach(RES)
 elseif(WIN32)
 	# Package for Windows
diff --git a/build/linux/io.github.cxong.cdogs-sdl.desktop.cmake b/build/linux/io.github.cxong.cdogs-sdl.desktop.cmake
index 3d40fbf..567315f 100644
--- a/build/linux/io.github.cxong.cdogs-sdl.desktop.cmake
+++ b/build/linux/io.github.cxong.cdogs-sdl.desktop.cmake
@@ -1,7 +1,6 @@
 [Desktop Entry]
 Name=C-Dogs SDL
 GenericName=Action/Arcade Game
-Version=@VERSION@
 Type=Application
 Categories=Game;ActionGame;
 Keywords=Game;Arcade;Shooter;Retro;
@@ -9,4 +8,4 @@ Comment=Classic overhead run-and-gun game
 TryExec=cdogs-sdl
 Exec=cdogs-sdl
 Terminal=false
-Icon=io.github.cxong.cdogs-sdl
+Icon=cdogs-sdl
