diff -Nur old/src/proto/CMakeLists.txt cdogs-sdl-1.2.0/src/proto/CMakeLists.txt
--- old/src/proto/CMakeLists.txt	2021-12-02 11:16:04.000000000 +0100
+++ cdogs-sdl-1.2.0/src/proto/CMakeLists.txt	2021-12-03 18:43:18.315210954 +0100
@@ -4,11 +4,12 @@
 add_definitions(-DPB_FIELD_16BIT)	# Allow larger field numbers in nanopb
 
 # Note: uncomment below line to build new proto files
-# nanopb_generate_cpp(PROTO_SRCS PROTO_HDRS msg.proto)
+nanopb_generate_cpp(PROTO_SRCS PROTO_HDRS msg.proto)
 set(PROTO_SRCS "msg.pb.c" "nanopb/pb_decode.c" "nanopb/pb_encode.c" "nanopb/pb_common.c")
 set(PROTO_HDRS "msg.pb.h" "nanopb/pb_decode.h" "nanopb/pb_encode.c" "nanopb/pb_common.c" "nanopb/pb.h")
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 set_source_files_properties(${PROTO_SRCS} ${PROTO_HDRS}
+    DIRECTORY .
     PROPERTIES GENERATED TRUE)
 add_library(cdogs_proto STATIC ${PROTO_SRCS} ${PROTO_HDRS} msg.options msg.proto)
