diff --git a/src/formatter.c b/src/formatter.c
index 7a68bcd..32d06bf 100644
--- a/src/formatter.c
+++ b/src/formatter.c
@@ -628,7 +628,7 @@ static bool timestamp_unix(struct keyval_list *kv_list, bool frac)
     status = gettimeofday(&tv, NULL);
     if (status != 0) {
         log_error("Failed to get current time: %s\n", strerror(errno));
-        return;
+        return false;
     }
 
     ts = tv.tv_sec + ((double) tv.tv_usec / 1000000.0);
