---
 Makefile |    1 -
 1 file changed, 1 deletion(-)

Index: webdot-2.30/Makefile
===================================================================
--- webdot-2.30.orig/Makefile	2013-01-14 04:18:52.000000000 +0100
+++ webdot-2.30/Makefile	2013-12-06 13:37:25.364668601 +0100
@@ -62,7 +62,6 @@ install:
 	rm -rf $(DESTDIR)$(CACHE_DIR)/webdot
 	mkdir -p $(DESTDIR)$(CACHE_DIR)/webdot
 	chmod 700 $(DESTDIR)$(CACHE_DIR)/webdot
-	chown $(HTTPD-USER-GROUP) $(DESTDIR)$(CACHE_DIR)/webdot
 
 uninstall:
 	rm -f $(DESTDIR)$(CGI-BIN_DIR)/webdot
