diff --git a/CMakeLists.txt b/CMakeLists.txt
index ac0432c34..bff6f289b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,7 +38,13 @@ if(POLICY CMP0148)
   cmake_policy(SET CMP0148 OLD)
 endif()
+if(POLICY CMP0167)
+  # use BoostConfig.cmake shipped with Boost 1.70+ instead of the one in CMake
+  cmake_policy(SET CMP0167 NEW)
+endif()
 # CMake modules/macros are in a subdirectory to keep this file cleaner
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
+project(ESPResSo)
+
 # C++ standard
 enable_language(CXX)
@@ -49,5 +55,4 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 include(FeatureSummary)
 include(GNUInstallDirs)
-project(ESPResSo)
 include(option_enum)
 if(POLICY CMP0074)
@@ -268,7 +273,5 @@ if(WITH_STOKESIAN_DYNAMICS)
   set(STOKESIAN_DYNAMICS 1)
   if(NOT stokesian_dynamics_POPULATED)
-    FetchContent_Populate(stokesian_dynamics)
-    add_subdirectory(${stokesian_dynamics_SOURCE_DIR}
-                     ${stokesian_dynamics_BINARY_DIR})
+    FetchContent_MakeAvailable(stokesian_dynamics)
   endif()
 endif(WITH_STOKESIAN_DYNAMICS)
