From: Jan Engelhardt <jengelh@inai.de>
Date: 2011-09-27 16:28:00 +0000

---
 Makefile.am  |    9 +++++++++
 configure.ac |   10 ++++++++++
 2 files changed, 19 insertions(+)

Index: zn_poly-0.9.1/Makefile.am
===================================================================
--- /dev/null
+++ zn_poly-0.9.1/Makefile.am
@@ -0,0 +1,10 @@
+# -*- Makefile -*-
+AM_CPPFLAGS = -I${top_srcdir}/include
+pkginclude_HEADERS = include/zn_poly.h include/wide_arith.h
+noinst_HEADERS = include/support.h include/profiler.h include/zn_poly_internal.h
+libsrcs = src/array.c src/invert.c src/ks_support.c src/mulmid.c src/mulmid_ks.c src/misc.c src/mpn_mulmid.c src/mul.c src/mul_fft.c src/mul_fft_dft.c src/mul_ks.c src/nuss.c src/pack.c src/pmf.c src/pmfvec_fft.c tune/tuning.c src/zn_mod.c
+lib_LTLIBRARIES = libzn_poly.la
+libzn_poly_la_SOURCES = ${libsrcs}
+ABI_VERSION = $(shell cat ABI_VERSION)
+libzn_poly_la_LDFLAGS = -release ${ABI_VERSION}
+libzn_poly_la_LIBADD  = -lgmp
Index: zn_poly-0.9.1/configure.ac
===================================================================
--- /dev/null
+++ zn_poly-0.9.1/configure.ac
@@ -0,0 +1,9 @@
+AC_INIT([zn_poly], [0.9.1])
+AC_CONFIG_AUX_DIR([build-aux])
+AM_INIT_AUTOMAKE([-Wall foreign])
+AC_PROG_INSTALL
+AC_PROG_CC
+m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
+LT_INIT([disable-static])
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
