From 2508f3f453bd5aeff89c43af5333a1a44c0e9510 Mon Sep 17 00:00:00 2001
From: VaiTon <eyadlorenzo@gmail.com>
Date: Wed, 21 Jan 2026 23:41:34 +0100
Subject: [PATCH 1/2] Link to distro provided mbedtls

---
 vlib/net/mbedtls/mbedtls.c.v | 125 +----------------------------------
 1 file changed, 2 insertions(+), 123 deletions(-)

diff --git a/vlib/net/mbedtls/mbedtls.c.v b/vlib/net/mbedtls/mbedtls.c.v
index d18689e8c..61fc82b9b 100644
--- a/vlib/net/mbedtls/mbedtls.c.v
+++ b/vlib/net/mbedtls/mbedtls.c.v
@@ -1,128 +1,7 @@
 module mbedtls
 
-#flag -I @VEXEROOT/thirdparty/mbedtls/library
-#flag -I @VEXEROOT/thirdparty/mbedtls/include
-// #flag -D _FILE_OFFSET_BITS=64
-#flag -I @VEXEROOT/thirdparty/mbedtls/3rdparty/everest/include
-#flag -I @VEXEROOT/thirdparty/mbedtls/3rdparty/everest/include/everest
-#flag -I @VEXEROOT/thirdparty/mbedtls/3rdparty/everest/include/everest/kremlib
-
-// TODO: this should be built-in to the compiler
-$if prod && opt_size ? {
-	#flag -Os
-}
-
-#flag @VEXEROOT/thirdparty/mbedtls/library/aes.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/aesce.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/aesni.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/aria.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/asn1parse.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/asn1write.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/base64.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/bignum.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/bignum_core.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/bignum_mod.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/bignum_mod_raw.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/block_cipher.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/camellia.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ccm.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/chacha20.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/chachapoly.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/cipher.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/cipher_wrap.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/cmac.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/constant_time.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ctr_drbg.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/debug.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/des.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/dhm.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ecdh.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ecdsa.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ecjpake.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ecp.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ecp_curves.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ecp_curves_new.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/entropy.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/entropy_poll.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/error.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/gcm.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/hkdf.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/hmac_drbg.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/lmots.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/lms.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/md5.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/md.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/memory_buffer_alloc.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/mps_reader.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/mps_trace.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/net_sockets.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/nist_kw.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/oid.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/padlock.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/pem.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/pk.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/pkcs12.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/pkcs5.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/pkcs7.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/pk_ecc.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/pkparse.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/pk_wrap.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/pkwrite.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/platform.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/platform_util.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/poly1305.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto_aead.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto_cipher.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto_client.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto_driver_wrappers_no_static.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto_ecp.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto_ffdh.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto_hash.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto_mac.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto_pake.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto_rsa.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto_se.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto_slot_management.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_crypto_storage.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_its_file.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/psa_util.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ripemd160.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/rsa_alt_helpers.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/rsa.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/sha1.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/sha256.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/sha3.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/sha512.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_cache.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_ciphersuites.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_client.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_cookie.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_debug_helpers_generated.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_msg.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_ticket.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_tls12_client.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_tls12_server.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_tls13_client.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_tls13_generic.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_tls13_keys.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_tls13_server.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/ssl_tls.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/threading.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/timing.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/version.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/version_features.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/x509.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/x509_create.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/x509_crl.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/x509_crt.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/x509_csr.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/x509write.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/x509write_crt.o
-#flag @VEXEROOT/thirdparty/mbedtls/library/x509write_csr.o
-#flag @VEXEROOT/thirdparty/mbedtls/3rdparty/everest/library/Hacl_Curve25519_joined.o
-#flag @VEXEROOT/thirdparty/mbedtls/3rdparty/everest/library/everest.o
-#flag @VEXEROOT/thirdparty/mbedtls/3rdparty/everest/library/x25519.o
+#flag -I /usr/include
+#flag -lmbedtls -lmbedx509 -lmbedcrypto
 
 #include <mbedtls/net_sockets.h>
 #include <mbedtls/ssl.h>
-- 
2.52.0

