Index: v-0.5.2/vlib/json/json_primitives.c.v
===================================================================
--- v-0.5.2.orig/vlib/json/json_primitives.c.v
+++ v-0.5.2/vlib/json/json_primitives.c.v
@@ -7,8 +7,7 @@ import math
 import strconv
 import time
 
-#flag -I @VEXEROOT/thirdparty/cJSON
-#flag @VEXEROOT/thirdparty/cJSON/cJSON.o
+#pkgconfig libcjson
 #include "cJSON.h"
 #define js_get(object, key) cJSON_GetObjectItemCaseSensitive((object), (key))
 
Index: v-0.5.2/vlib/json/cjson/cjson_wrapper.c.v
===================================================================
--- v-0.5.2.orig/vlib/json/cjson/cjson_wrapper.c.v
+++ v-0.5.2/vlib/json/cjson/cjson_wrapper.c.v
@@ -8,8 +8,7 @@ module cjson
 // that has NULL leaves for example, which is currently not convenient/easy to do with just the
 // high level `json.encode(value)` API that V has.
 
-#flag -I @VEXEROOT/thirdparty/cJSON
-#flag @VEXEROOT/thirdparty/cJSON/cJSON.o
+#pkgconfig libcjson
 #include "cJSON.h"
 
 // As cJSON use `libm`, we need to link it.
Index: v-0.5.2/vlib/compress/zstd/zstd.c.v
===================================================================
--- v-0.5.2.orig/vlib/compress/zstd/zstd.c.v
+++ v-0.5.2/vlib/compress/zstd/zstd.c.v
@@ -5,8 +5,8 @@ module zstd
 import os
 import encoding.binary
 
-#flag -I @VMODROOT/thirdparty/zstd
-#include "zstd.c"	// msvc can't compile multiple source files, so included
+#pkgconfig libzstd
+#include <zstd.h>
 
 const frame_header_size_max = 18
 const content_size_unknown = u64(-1)
