Index: bear-project/bear-0.0.0.git20200220/bear-engine/core/src/engine/code/game_network.cpp
===================================================================
--- bear-project.orig/bear-0.0.0.git20200220/bear-engine/core/src/engine/code/game_network.cpp
+++ bear-project/bear-0.0.0.git20200220/bear-engine/core/src/engine/code/game_network.cpp
@@ -148,7 +148,7 @@ void bear::engine::game_network::create_
     {
       net::server* s = new net::server(port);
       s->on_new_client.connect
-        ( boost::bind( &game_network::on_new_client, this, s, _1 ) );
+        ( boost::bind( &game_network::on_new_client, this, s, boost::placeholders::_1 ) );
       m_server[name] = s;
     }
 } // game_network::create_service()
Index: bear-project/bear-0.0.0.git20200220/bear-engine/core/src/engine/layer/code/transition_layer.cpp
===================================================================
--- bear-project.orig/bear-0.0.0.git20200220/bear-engine/core/src/engine/layer/code/transition_layer.cpp
+++ bear-project/bear-0.0.0.git20200220/bear-engine/core/src/engine/layer/code/transition_layer.cpp
@@ -113,7 +113,7 @@ void bear::engine::transition_layer::ren
 bool bear::engine::transition_layer::key_pressed( const input::key_info& key )
 {
   return diffuse_call
-    ( boost::bind( &transition_effect::key_pressed, _1, key) );
+    ( boost::bind( &transition_effect::key_pressed, boost::placeholders::_1, key) );
 } // transition_layer::key_pressed()
 
 /*----------------------------------------------------------------------------*/
@@ -125,7 +125,7 @@ bool bear::engine::transition_layer::key
 ( const input::key_info& key )
 {
   return diffuse_call
-    ( boost::bind( &transition_effect::key_maintained, _1, key) );
+    ( boost::bind( &transition_effect::key_maintained, boost::placeholders::_1, key) );
 } // transition_layer::key_maintained()
 
 /*----------------------------------------------------------------------------*/
@@ -136,7 +136,7 @@ bool bear::engine::transition_layer::key
 bool bear::engine::transition_layer::key_released( const input::key_info& key )
 {
   return diffuse_call
-    ( boost::bind( &transition_effect::key_released, _1, key) );
+    ( boost::bind( &transition_effect::key_released, boost::placeholders::_1, key) );
 } // transition_layer::key_released()
 
 /*----------------------------------------------------------------------------*/
@@ -147,7 +147,7 @@ bool bear::engine::transition_layer::key
 bool bear::engine::transition_layer::char_pressed( const input::key_info& key )
 {
   return diffuse_call
-    ( boost::bind( &transition_effect::char_pressed, _1, key) );
+    ( boost::bind( &transition_effect::char_pressed, boost::placeholders::_1, key) );
 } // transition_layer::char_pressed()
 
 /*----------------------------------------------------------------------------*/
@@ -160,7 +160,7 @@ bool bear::engine::transition_layer::but
 ( input::joystick::joy_code button, unsigned int joy_index )
 {
   return diffuse_call
-    ( boost::bind( &transition_effect::button_pressed, _1, button, joy_index) );
+    ( boost::bind( &transition_effect::button_pressed, boost::placeholders::_1, button, joy_index) );
 } // transition_layer::button_pressed()
 
 /*----------------------------------------------------------------------------*/
@@ -174,7 +174,7 @@ bool bear::engine::transition_layer::but
 {
   return diffuse_call
     ( boost::bind
-      ( &transition_effect::button_maintained, _1, button, joy_index) );
+      ( &transition_effect::button_maintained, boost::placeholders::_1, button, joy_index) );
 } // transition_layer::button_maintained()
 
 /*----------------------------------------------------------------------------*/
@@ -188,7 +188,7 @@ bool bear::engine::transition_layer::but
 {
   return diffuse_call
     ( boost::bind
-      ( &transition_effect::button_released, _1, button, joy_index) );
+      ( &transition_effect::button_released, boost::placeholders::_1, button, joy_index) );
 } // transition_layer::button_released()
 
 /*----------------------------------------------------------------------------*/
@@ -202,7 +202,7 @@ bool bear::engine::transition_layer::mou
   const claw::math::coordinate_2d<unsigned int>& pos )
 {
   return diffuse_call
-    ( boost::bind( &transition_effect::mouse_pressed, _1, key, pos) );
+    ( boost::bind( &transition_effect::mouse_pressed, boost::placeholders::_1, key, pos) );
 } // transition_layer::mouse_pressed()
 
 /*----------------------------------------------------------------------------*/
@@ -216,7 +216,7 @@ bool bear::engine::transition_layer::mou
   const claw::math::coordinate_2d<unsigned int>& pos )
 {
   return diffuse_call
-    ( boost::bind( &transition_effect::mouse_maintained, _1, key, pos) );
+    ( boost::bind( &transition_effect::mouse_maintained, boost::placeholders::_1, key, pos) );
 } // transition_layer::mouse_maintained()
 
 /*----------------------------------------------------------------------------*/
@@ -230,7 +230,7 @@ bool bear::engine::transition_layer::mou
   const claw::math::coordinate_2d<unsigned int>& pos )
 {
   return diffuse_call
-    ( boost::bind( &transition_effect::mouse_released, _1, key, pos) );
+    ( boost::bind( &transition_effect::mouse_released, boost::placeholders::_1, key, pos) );
 } // transition_layer::mouse_released()
 
 /*----------------------------------------------------------------------------*/
@@ -241,7 +241,7 @@ bool bear::engine::transition_layer::mou
 bool bear::engine::transition_layer::mouse_move
 ( const claw::math::coordinate_2d<unsigned int>& pos )
 {
-  return diffuse_call( boost::bind( &transition_effect::mouse_move, _1, pos) );
+  return diffuse_call( boost::bind( &transition_effect::mouse_move, boost::placeholders::_1, pos) );
 } // transition_layer::mouse_move()
 
 /*----------------------------------------------------------------------------*/
@@ -253,7 +253,7 @@ bool bear::engine::transition_layer::fin
 ( const bear::input::finger_event& event )
 {
   return diffuse_call
-    ( boost::bind( &transition_effect::finger_action, _1, event) );
+    ( boost::bind( &transition_effect::finger_action, boost::placeholders::_1, event) );
 } // transition_layer::finger_action()
 
 /*----------------------------------------------------------------------------*/
Index: bear-project/bear-0.0.0.git20200220/bear-engine/core/src/engine/model/code/model_snapshot_tweener.cpp
===================================================================
--- bear-project.orig/bear-0.0.0.git20200220/bear-engine/core/src/engine/model/code/model_snapshot_tweener.cpp
+++ bear-project/bear-0.0.0.git20200220/bear-engine/core/src/engine/model/code/model_snapshot_tweener.cpp
@@ -183,7 +183,7 @@ void bear::engine::model_snapshot_tweene
     m_tweeners.insert
       ( claw::tween::single_tweener
         (m_placement[id].get_angle(), end.get_angle(), d,
-         boost::bind( &model_mark_placement::set_angle, &m_placement[id], _1 ),
+         boost::bind( &model_mark_placement::set_angle, &m_placement[id], boost::placeholders::_1 ),
          m_placement[id].get_angle_easing() ) );
 
   if ( m_placement[id].get_position().x != end.get_position().x )
@@ -191,7 +191,7 @@ void bear::engine::model_snapshot_tweene
       ( claw::tween::single_tweener
         (m_placement[id].get_position().x, end.get_position().x, d,
          boost::bind
-         ( &model_mark_placement::set_x_position, &m_placement[id], _1 ),
+         ( &model_mark_placement::set_x_position, &m_placement[id], boost::placeholders::_1 ),
          m_placement[id].get_x_position_easing() ) );
 
   if ( m_placement[id].get_position().y != end.get_position().y )
@@ -199,7 +199,7 @@ void bear::engine::model_snapshot_tweene
       ( claw::tween::single_tweener
         (m_placement[id].get_position().y, end.get_position().y, d,
          boost::bind
-         ( &model_mark_placement::set_y_position, &m_placement[id], _1 ),
+         ( &model_mark_placement::set_y_position, &m_placement[id], boost::placeholders::_1 ),
          m_placement[id].get_y_position_easing() ) );
 
   if ( m_placement[id].get_size().x != end.get_size().x )
@@ -207,7 +207,7 @@ void bear::engine::model_snapshot_tweene
       ( claw::tween::single_tweener
         (m_placement[id].get_size().x, end.get_size().x, d,
          boost::bind
-         ( &model_mark_placement::set_width, &m_placement[id], _1 ),
+         ( &model_mark_placement::set_width, &m_placement[id], boost::placeholders::_1 ),
          m_placement[id].get_width_easing() ) );
 
   if ( m_placement[id].get_size().y != end.get_size().y )
@@ -215,6 +215,6 @@ void bear::engine::model_snapshot_tweene
       ( claw::tween::single_tweener
         (m_placement[id].get_size().y, end.get_size().y, d,
          boost::bind
-         ( &model_mark_placement::set_height, &m_placement[id], _1 ),
+         ( &model_mark_placement::set_height, &m_placement[id], boost::placeholders::_1 ),
          m_placement[id].get_height_easing() ) );
 } // model_snapshot_tweener::insert_tweener()
Index: bear-project/bear-0.0.0.git20200220/bear-engine/core/src/net/code/client.cpp
===================================================================
--- bear-project.orig/bear-0.0.0.git20200220/bear-engine/core/src/net/code/client.cpp
+++ bear-project/bear-0.0.0.git20200220/bear-engine/core/src/net/code/client.cpp
@@ -101,7 +101,7 @@ bear::net::message_handle bear::net::cli
 void bear::net::client::connect()
 {
   const connection_task connection
-    ( boost::bind( &client::set_stream, this, _1 ),
+    ( boost::bind( &client::set_stream, this, boost::placeholders::_1 ),
       m_host, m_port, m_read_time_limit );
   //m_connection = new boost::thread(connection);
   connection();
Index: bear-project/bear-0.0.0.git20200220/bear-factory/model-editor/src/bf/code/model_snapshot_tweener.cpp
===================================================================
--- bear-project.orig/bear-0.0.0.git20200220/bear-factory/model-editor/src/bf/code/model_snapshot_tweener.cpp
+++ bear-project/bear-0.0.0.git20200220/bear-factory/model-editor/src/bf/code/model_snapshot_tweener.cpp
@@ -146,26 +146,26 @@ void bf::model_snapshot_tweener::insert_
   m_tweeners.insert
   ( claw::tween::single_tweener
     (m.get_angle(), end.get_angle(), d,
-      boost::bind( &mark_placement::set_angle, &m, _1 ),
+      boost::bind( &mark_placement::set_angle, &m, boost::placeholders::_1 ),
       m.get_angle_easing().to_claw_easing_function() ) );
   m_tweeners.insert
   ( claw::tween::single_tweener
     ( m.get_x_position(), end.get_x_position(), d,
-      boost::bind( &mark_placement::set_x_position, &m, _1 ),
+      boost::bind( &mark_placement::set_x_position, &m, boost::placeholders::_1 ),
       m.get_x_easing().to_claw_easing_function() ) );
   m_tweeners.insert
   ( claw::tween::single_tweener
     ( m.get_y_position(), end.get_y_position(), d,
-      boost::bind( &mark_placement::set_y_position, &m, _1 ),
+      boost::bind( &mark_placement::set_y_position, &m, boost::placeholders::_1 ),
       m.get_y_easing().to_claw_easing_function() ) );
   m_tweeners.insert
   ( claw::tween::single_tweener
     ( m.get_width(), end.get_width(), d,
-      boost::bind( &mark_placement::set_width, &m, _1 ),
+      boost::bind( &mark_placement::set_width, &m, boost::placeholders::_1 ),
       m.get_width_easing().to_claw_easing_function() ) );
   m_tweeners.insert
   ( claw::tween::single_tweener
     ( m.get_height(), end.get_height(), d,
-      boost::bind( &mark_placement::set_height, &m, _1 ),
+      boost::bind( &mark_placement::set_height, &m, boost::placeholders::_1 ),
       m.get_height_easing().to_claw_easing_function() ) );
 } // model_snapshot_tweener::insert_tweener()
Index: bear-project/bear-0.0.0.git20200220/bear-factory/model-editor/src/bf/history/code/action_modify_placement.cpp
===================================================================
--- bear-project.orig/bear-0.0.0.git20200220/bear-factory/model-editor/src/bf/history/code/action_modify_placement.cpp
+++ bear-project/bear-0.0.0.git20200220/bear-factory/model-editor/src/bf/history/code/action_modify_placement.cpp
@@ -29,8 +29,8 @@ bf::make_action_modify_placement_x( snap
 {
   return new action_modify_placement<double>
     ( s, m,
-      boost::bind( &mark_placement::set_x_position, _1, _2 ),
-      boost::bind( &mark_placement::get_x_position, _1 ),
+      boost::bind( &mark_placement::set_x_position, boost::placeholders::_1, _2 ),
+      boost::bind( &mark_placement::get_x_position, boost::placeholders::_1 ),
       value );
 } // make_action_modify_placement_x()
 
@@ -46,8 +46,8 @@ bf::make_action_modify_placement_y( snap
 {
   return new action_modify_placement<double>
     ( s, m,
-      boost::bind( &mark_placement::set_y_position, _1, _2 ),
-      boost::bind( &mark_placement::get_y_position, _1 ),
+      boost::bind( &mark_placement::set_y_position, boost::placeholders::_1, _2 ),
+      boost::bind( &mark_placement::get_y_position, boost::placeholders::_1 ),
       value );
 } // make_action_modify_placement_y()
 
@@ -64,8 +64,8 @@ bf::make_action_modify_placement_angle
 {
   return new action_modify_placement<double>
     ( s, m,
-      boost::bind( &mark_placement::set_angle, _1, _2 ),
-      boost::bind( &mark_placement::get_angle, _1 ),
+      boost::bind( &mark_placement::set_angle, boost::placeholders::_1, _2 ),
+      boost::bind( &mark_placement::get_angle, boost::placeholders::_1 ),
       value );
 } // make_action_modify_placement_angle()
 
@@ -82,8 +82,8 @@ bf::make_action_modify_placement_depth
 {
   return new action_modify_placement<int>
     ( s, m,
-      boost::bind( &mark_placement::set_depth_position, _1, _2 ),
-      boost::bind( &mark_placement::get_depth_position, _1 ),
+      boost::bind( &mark_placement::set_depth_position, boost::placeholders::_1, _2 ),
+      boost::bind( &mark_placement::get_depth_position, boost::placeholders::_1 ),
       value );
 } // make_action_modify_placement_depth()
 
@@ -100,8 +100,8 @@ bf::make_action_modify_placement_visibil
 {
   return new action_modify_placement<bool>
     ( s, m,
-      boost::bind( &mark_placement::set_visibility, _1, _2 ),
-      boost::bind( &mark_placement::is_visible, _1 ),
+      boost::bind( &mark_placement::set_visibility, boost::placeholders::_1, _2 ),
+      boost::bind( &mark_placement::is_visible, boost::placeholders::_1 ),
       value );
 } // make_action_modify_placement_visibility()
 
@@ -118,8 +118,8 @@ bf::make_action_modify_placement_width
 {
   return new action_modify_placement<double>
     ( s, m,
-      boost::bind( &mark_placement::set_width, _1, _2 ),
-      boost::bind( &mark_placement::get_width, _1 ),
+      boost::bind( &mark_placement::set_width, boost::placeholders::_1, _2 ),
+      boost::bind( &mark_placement::get_width, boost::placeholders::_1 ),
       value );
 } // make_action_modify_placement_width()
 
@@ -136,7 +136,7 @@ bf::make_action_modify_placement_height
 {
   return new action_modify_placement<double>
     ( s, m,
-      boost::bind( &mark_placement::set_height, _1, _2 ),
-      boost::bind( &mark_placement::get_height, _1 ),
+      boost::bind( &mark_placement::set_height, boost::placeholders::_1, _2 ),
+      boost::bind( &mark_placement::get_height, boost::placeholders::_1 ),
       value );
 } // make_action_modify_placement_height()
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/code/player.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/code/player.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/code/player.cpp
@@ -3828,7 +3828,7 @@ void ptb::player::create_invincible_star
     ( 1, 0, 0.4,
       boost::bind
       ( &bear::visual::bitmap_rendering_attributes::set_opacity,
-        &(s->get_rendering_attributes()), _1 ), 
+        &(s->get_rendering_attributes()), boost::placeholders::_1 ), 
       &claw::tween::easing_linear::ease_in );
 
   bear::tweener_item* t = new bear::tweener_item(tweener, s, true);
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/item/castle/code/catapult.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/item/castle/code/catapult.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/item/castle/code/catapult.cpp
@@ -253,7 +253,7 @@ void ptb::catapult::start_throw()
      (m_arm_angle, -1.57+m_stop_angle, 0.1,
        boost::bind
        ( &catapult::on_arm_angle_update,
-	 this, _1 ), &claw::tween::easing_linear::ease_out ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_linear::ease_out ) );
   m_arm_angle_tweener.insert
     (claw::tween::single_tweener
      (m_arm_angle, 0, 0.5,
@@ -282,7 +282,7 @@ void ptb::catapult::cancel()
      (m_arm_angle, m_arm_angle-0.2, 0.05,
        boost::bind
        ( &catapult::on_arm_angle_update,
-	 this, _1 ), &claw::tween::easing_linear::ease_out ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_linear::ease_out ) );
   m_arm_angle_tweener.insert
     (claw::tween::single_tweener
      (m_arm_angle, 0, 0.5,
@@ -351,25 +351,25 @@ void ptb::catapult::create_stop_angle_tw
      (0, -s_initial_stop_angle, 0.3,
        boost::bind
        ( &catapult::on_stop_angle_update,
-	 this, _1 ), &claw::tween::easing_cubic::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_cubic::ease_in ) );
   m_stop_angle_tweener.insert
     (claw::tween::single_tweener
      (-s_initial_stop_angle, 0, 0.3,
        boost::bind
        ( &catapult::on_stop_angle_update,
-	 this, _1 ), &claw::tween::easing_cubic::ease_out ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_cubic::ease_out ) );
   m_stop_angle_tweener.insert
     (claw::tween::single_tweener
      (0, s_initial_stop_angle, 0.3,
        boost::bind
        ( &catapult::on_stop_angle_update,
-	 this, _1 ), &claw::tween::easing_cubic::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_cubic::ease_in ) );
   m_stop_angle_tweener.insert
     (claw::tween::single_tweener
      (s_initial_stop_angle, 0, 0.3,
        boost::bind
        ( &catapult::on_stop_angle_update,
-	 this, _1 ), &claw::tween::easing_cubic::ease_out ) ); 
+	 this, boost::placeholders::_1 ), &claw::tween::easing_cubic::ease_out ) ); 
 } // catapult::create_stop_angle_tweener()
 
 /*----------------------------------------------------------------------------*/
@@ -384,13 +384,13 @@ void ptb::catapult::create_arm_angle_twe
      (m_arm_angle, s_minimal_arm_angle, 1,
        boost::bind
        ( &catapult::on_arm_angle_update,
-	 this, _1 ), &claw::tween::easing_cubic::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_cubic::ease_in ) );
   m_arm_angle_tweener.insert
     (claw::tween::single_tweener
      (s_minimal_arm_angle, s_maximal_arm_angle, 1,
        boost::bind
        ( &catapult::on_arm_angle_update,
-	 this, _1 ), &claw::tween::easing_cubic::ease_out ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_cubic::ease_out ) );
 } // catapult::create_arm_angle_tweener()
 
 /*----------------------------------------------------------------------------*/
@@ -405,7 +405,7 @@ void ptb::catapult::init_angle()
      (m_arm_angle, s_maximal_arm_angle, 0.5,
        boost::bind
        ( &catapult::on_arm_angle_update,
-	 this, _1 ), &claw::tween::easing_linear::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_linear::ease_in ) );
    m_arm_angle_tweener.insert
     (claw::tween::single_tweener
      (m_arm_angle, s_initial_arm_angle, 1,
@@ -419,7 +419,7 @@ void ptb::catapult::init_angle()
      (m_stop_angle, 0, 0.5,
        boost::bind
        ( &catapult::on_stop_angle_update,
-	 this, _1 ), &claw::tween::easing_linear::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_linear::ease_in ) );
 } // catapult::init_angle()
 
 /*----------------------------------------------------------------------------*/
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/item/code/invincibility_effect.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/item/code/invincibility_effect.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/item/code/invincibility_effect.cpp
@@ -74,7 +74,7 @@ void ptb::invincibility_effect::create_s
     ( 1, 0, 0.4,
       boost::bind
       ( &bear::visual::bitmap_rendering_attributes::set_opacity,
-        &(s->get_rendering_attributes()), _1 ), 
+        &(s->get_rendering_attributes()), boost::placeholders::_1 ), 
       &claw::tween::easing_linear::ease_in );
 
   bear::tweener_item* t = new bear::tweener_item(tweener, s, true);
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/item/code/power_effect.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/item/code/power_effect.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/item/code/power_effect.cpp
@@ -79,17 +79,17 @@ void ptb::power_effect::set_player_index
   connect
     ( bear::engine::game::get_instance().listen_bool_variable_change
       ( game_variables::get_air_power_variable_name(i),
-        boost::bind( &power_effect::on_air_power_changed, this, _1) ) );
+        boost::bind( &power_effect::on_air_power_changed, this, boost::placeholders::_1) ) );
 
   connect
     ( bear::engine::game::get_instance().listen_bool_variable_change
       ( game_variables::get_fire_power_variable_name(i),
-        boost::bind( &power_effect::on_fire_power_changed, this, _1) ) );
+        boost::bind( &power_effect::on_fire_power_changed, this, boost::placeholders::_1) ) );
 
   connect
     ( bear::engine::game::get_instance().listen_bool_variable_change
       ( game_variables::get_water_power_variable_name(i),
-        boost::bind( &power_effect::on_water_power_changed, this, _1) ) );
+        boost::bind( &power_effect::on_water_power_changed, this, boost::placeholders::_1) ) );
 } // power_effect::set_player_index()
 
 /*----------------------------------------------------------------------------*/
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/code/status_layer.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/layer/code/status_layer.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/code/status_layer.cpp
@@ -209,7 +209,7 @@ void ptb::status_layer::player_status::p
              (notification.height()+s_margin, -1, 0.5,
               boost::bind
               ( &player_status::on_notification_position_update,
-                this, _1 ), &claw::tween::easing_back::ease_in ) );
+                this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
       }
     }
 } // player_status::progress_notification
@@ -234,14 +234,14 @@ void ptb::status_layer::player_status::c
        (0, notification.height()+s_margin, 0.5,
         boost::bind
         (&player_status::on_notification_position_update,
-         this, _1 ), &claw::tween::easing_back::ease_out ) );
+         this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_out ) );
       
    m_notification_tweener.insert
       (claw::tween::single_tweener
        (notification.height()+s_margin, notification.height()+s_margin,1,
         boost::bind
         (&player_status::on_notification_position_update,
-         this, _1 ), &claw::tween::easing_linear::ease_out ) );     
+         this, boost::placeholders::_1 ), &claw::tween::easing_linear::ease_out ) );     
 } // player_status::create_notification()
 
 /*----------------------------------------------------------------------------*/
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/code/windows_layer.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/layer/code/windows_layer.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/code/windows_layer.cpp
@@ -390,7 +390,7 @@ void ptb::windows_layer::apply_show_effe
 
   const claw::tween::single_tweener t
     ( get_size().y, m_ref_bottom[wnd], s_effect_duration,
-      boost::bind( &frame::set_bottom, wnd, _1 ),
+      boost::bind( &frame::set_bottom, wnd, boost::placeholders::_1 ),
       &claw::tween::easing_back::ease_out );
 
   m_tweener.insert(t);
@@ -415,7 +415,7 @@ void ptb::windows_layer::apply_hide_effe
   // updating the position of the window once it was deleted by the end of the
   // hide effect.
   claw::tween::single_tweener t
-    ( wnd->top(), 0, s_effect_duration, boost::bind( &frame::set_top, wnd, _1 ),
+    ( wnd->top(), 0, s_effect_duration, boost::bind( &frame::set_top, wnd, boost::placeholders::_1 ),
       &claw::tween::easing_quad::ease_out );
 
   if ( d )
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/corrupting_bonus_component.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/corrupting_bonus_component.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/corrupting_bonus_component.cpp
@@ -175,7 +175,7 @@ void ptb::corrupting_bonus_component::in
       ( game_variables::get_corrupting_bonus_count_variable_name(),
         boost::bind
         (&ptb::corrupting_bonus_component::on_corrupting_bonus_updated,
-         this, _1) ) );
+         this, boost::placeholders::_1) ) );
 } // corrupting_bonus_component::init_signals()
 
 /*----------------------------------------------------------------------------*/
@@ -239,21 +239,21 @@ void ptb::corrupting_bonus_component::mo
       (get_position().y, get_active_position().y, 0.3,
        boost::bind
        ( &ptb::status_component::on_y_position_update,
-         this, _1 ), &claw::tween::easing_back::ease_out ) );
+         this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_out ) );
   
   tween.insert
 	( claw::tween::single_tweener
 	  (get_active_position().y, get_active_position().y, 1,
 	   boost::bind
 	   ( &ptb::status_component::on_y_position_update,
-	     this, _1 ), &claw::tween::easing_back::ease_in ) );
+	     this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
       
   tween.insert
     ( claw::tween::single_tweener
       (get_active_position().y, get_inactive_position().y, 0.5,
        boost::bind
        ( &ptb::status_component::on_y_position_update,
-	 this, _1 ), &claw::tween::easing_back::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
 
   add_tweener( tween );
 } // corrupting_bonus_component::move()
@@ -271,21 +271,21 @@ void ptb::corrupting_bonus_component::ch
       (1, 1, 0.3,
        boost::bind
        (&ptb::corrupting_bonus_component::on_corrupting_bonus_scale_update,
-	this, _1 ), &claw::tween::easing_back::ease_out ) );
+	this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_out ) );
   
   tween.insert
     ( claw::tween::single_tweener
       (1, 1.5, 0.3,
        boost::bind
        (&ptb::corrupting_bonus_component::on_corrupting_bonus_scale_update,
-	this, _1 ), &claw::tween::easing_back::ease_out ) );
+	this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_out ) );
   
   tween.insert
     ( claw::tween::single_tweener
       (1.5, 1, 0.3,
        boost::bind
        (&ptb::corrupting_bonus_component::on_corrupting_bonus_scale_update,
-	this, _1 ), &claw::tween::easing_back::ease_in ) );
+	this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
   
   add_tweener( tween );
 } // corrupting_bonus_component::change_scale()
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/energy_component.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/energy_component.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/energy_component.cpp
@@ -118,19 +118,19 @@ void ptb::energy_component::init_signals
   add_signal
     ( get_player().get_signals().energy_added.connect
       ( boost::bind
-        (&ptb::energy_component::on_energy_added, this, _1) ) );
+        (&ptb::energy_component::on_energy_added, this, boost::placeholders::_1) ) );
   
   add_signal
     ( get_player().get_signals().energy_removed.connect
       ( boost::bind
-        (&ptb::energy_component::on_energy_removed, this, _1) ) );
+        (&ptb::energy_component::on_energy_removed, this, boost::placeholders::_1) ) );
 
   add_signal
     ( bear::engine::game::get_instance().listen_double_variable_change
       ( game_variables::get_max_energy_variable_name(get_player().get_index()),
         boost::bind
         (&ptb::energy_component::on_max_energy_added,
-         this, _1) ) );
+         this, boost::placeholders::_1) ) );
 } // energy_component::init_signals()
 
 /*----------------------------------------------------------------------------*/
@@ -146,7 +146,7 @@ void ptb::energy_component::on_energy_ch
       (get_position().y, get_active_position().y, 0.3,
        boost::bind
        ( &ptb::status_component::on_y_position_update,
-         this, _1 ), &claw::tween::easing_back::ease_out ) );
+         this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_out ) );
 
   if ( ! m_energy.is_critical() )
     {
@@ -155,14 +155,14 @@ void ptb::energy_component::on_energy_ch
 	  (get_active_position().y, get_active_position().y, 1,
 	   boost::bind
 	   ( &ptb::status_component::on_y_position_update,
-	     this, _1 ), &claw::tween::easing_back::ease_in ) );
+	     this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
       
       tween.insert
 	( claw::tween::single_tweener
 	  (get_active_position().y, get_inactive_position().y, 0.5,
 	   boost::bind
 	   ( &ptb::status_component::on_y_position_update,
-	     this, _1 ), &claw::tween::easing_back::ease_in ) );
+	     this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
     }
 
   add_tweener( tween );
@@ -203,7 +203,7 @@ void ptb::energy_component::on_max_energ
       (m_energy.length(), e, 1,
        boost::bind
        ( &ptb::energy_component::on_max_energy_length_update,
-         this, _1 ), &claw::tween::easing_elastic::ease_out ) );
+         this, boost::placeholders::_1 ), &claw::tween::easing_elastic::ease_out ) );
 } // energy_component::on_max_energy_added()
 
 /*----------------------------------------------------------------------------*/
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/floating_bonus.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/floating_bonus.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/floating_bonus.cpp
@@ -110,7 +110,7 @@ void ptb::floating_bonus::set_position
     ( init_position.x, end_position.x, 1.0,
       boost::bind
       (&floating_bonus::on_x_position_update,
-       this, _1 ), &claw::tween::easing_back::ease_in );
+       this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in );
   
   m_tweeners.insert(tween_x);
 
@@ -118,7 +118,7 @@ void ptb::floating_bonus::set_position
     ( init_position.y, end_position.y, 1.0,
       boost::bind
       (&floating_bonus::on_y_position_update,
-       this, _1 ), &claw::tween::easing_back::ease_in );
+       this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in );
   
   m_tweeners.insert(tween_y);
 } // floating_bonus::set_position()
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/floating_corrupting_bonus.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/floating_corrupting_bonus.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/floating_corrupting_bonus.cpp
@@ -88,7 +88,7 @@ void ptb::floating_corrupting_bonus::set
     ( init_position.x, end_position.x, 0.5,
       boost::bind
       (&floating_corrupting_bonus::on_x_position_update,
-       this, _1 ), &easing_back_large::ease_in );
+       this, boost::placeholders::_1 ), &easing_back_large::ease_in );
   
        m_tweeners.insert(tween_x);
 
@@ -96,7 +96,7 @@ void ptb::floating_corrupting_bonus::set
     ( init_position.y, end_position.y, 0.5,
       boost::bind
       (&floating_corrupting_bonus::on_y_position_update,
-       this, _1 ), &easing_back_large::ease_in );
+       this, boost::placeholders::_1 ), &easing_back_large::ease_in );
   
   m_tweeners.insert(tween_y);
 } // floating_corrupting_bonus::set_position()
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/gauge_component.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/gauge_component.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/gauge_component.cpp
@@ -102,7 +102,7 @@ void ptb::gauge_component::on_enters_zon
       (get_position().x, get_active_position().x, 1,
        boost::bind
        ( &ptb::gauge_component::on_x_position_update,
-	 this, _1 ), &claw::tween::easing_elastic::ease_out ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_elastic::ease_out ) );
 } // gauge_component::on_enters_zone()
 
 /*----------------------------------------------------------------------------*/
@@ -116,13 +116,13 @@ void ptb::gauge_component::on_leaves_zon
       (get_position().x, get_position().x, 0.5,
        boost::bind
        ( &ptb::gauge_component::on_x_position_update,
-	 this, _1 ), &claw::tween::easing_linear::ease_in_out ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_linear::ease_in_out ) );
 
   tween.insert( claw::tween::single_tweener
       (get_position().x, get_inactive_position().x, 1,
        boost::bind
        ( &ptb::gauge_component::on_x_position_update,
-	 this, _1 ), &claw::tween::easing_linear::ease_in_out ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_linear::ease_in_out ) );
 
   add_tweener( tween );
 } // gauge_component::on_leaves_zone()
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/honeypots_component.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/honeypots_component.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/honeypots_component.cpp
@@ -127,21 +127,21 @@ void ptb::honeypots_component::on_count_
       (get_position().y, get_active_position().y, 0.5,
        boost::bind
        ( &ptb::status_component::on_y_position_update,
-         this, _1 ), &claw::tween::easing_back::ease_out ) );
+         this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_out ) );
 
   tween.insert
     ( claw::tween::single_tweener
       (get_active_position().y, get_active_position().y, 2,
        boost::bind
        ( &ptb::status_component::on_y_position_update,
-	 this, _1 ), &claw::tween::easing_back::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
   
   tween.insert
     ( claw::tween::single_tweener
       (get_active_position().y, get_inactive_position().y, 0.5,
        boost::bind
        ( &ptb::status_component::on_y_position_update,
-	 this, _1 ), &claw::tween::easing_back::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
   
   add_tweener( tween );
 } // honeypots_component::on_count_change()
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/lives_component.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/lives_component.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/lives_component.cpp
@@ -120,7 +120,7 @@ void ptb::lives_component::init_signals(
       ( game_variables::get_lives_count_variable_name
 	(get_player().get_index()),
         boost::bind
-        (&ptb::lives_component::on_lives_changed,this, _1) ) );
+        (&ptb::lives_component::on_lives_changed,this, boost::placeholders::_1) ) );
 } // lives_component::init_signals()
 
 /*----------------------------------------------------------------------------*/
@@ -159,19 +159,19 @@ void ptb::lives_component::on_lives_chan
       (1, 1, 0.5,
        boost::bind
        ( &ptb::lives_component::on_lives_scale_update,
-         this, _1 ), &claw::tween::easing_back::ease_out ) );
+         this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_out ) );
   tween.insert
     ( claw::tween::single_tweener
       (1, 1.5, 0.5,
        boost::bind
        ( &ptb::lives_component::on_lives_scale_update,
-         this, _1 ), &claw::tween::easing_back::ease_out ) );
+         this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_out ) );
   tween.insert
     ( claw::tween::single_tweener
       (1.5, 1, 0.5,
        boost::bind
        ( &ptb::lives_component::on_lives_scale_update,
-         this, _1 ), &claw::tween::easing_back::ease_in ) );
+         this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
 
   add_tweener( tween );
 
@@ -182,21 +182,21 @@ void ptb::lives_component::on_lives_chan
       (get_position().x, get_active_position().x, 0.3,
        boost::bind
        ( &ptb::status_component::on_x_position_update,
-         this, _1 ), &claw::tween::easing_back::ease_out ) );
+         this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_out ) );
 
   tween2.insert
     ( claw::tween::single_tweener
       (get_active_position().x, get_active_position().x, 2,
        boost::bind
        ( &ptb::status_component::on_x_position_update,
-	 this, _1 ), &claw::tween::easing_back::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
   
   tween2.insert
     ( claw::tween::single_tweener
       (get_active_position().x, get_inactive_position().x, 0.5,
        boost::bind
        ( &ptb::status_component::on_x_position_update,
-	 this, _1 ), &claw::tween::easing_back::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
   
   add_tweener( tween2 );
 } // lives_component::on_lives_changed()
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/score_component.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/score_component.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/score_component.cpp
@@ -109,7 +109,7 @@ void ptb::score_component::init_signals(
       ( game_variables::get_score_variable_name(get_player().get_index()),
         boost::bind
         (&ptb::score_component::on_score_changed,
-         this, _1) ) );
+         this, boost::placeholders::_1) ) );
 } // score_component::init_signals()
 
 /*----------------------------------------------------------------------------*/
@@ -133,21 +133,21 @@ void ptb::score_component::on_score_chan
       (get_position().x, get_active_position().x, 0.3,
        boost::bind
        ( &ptb::status_component::on_x_position_update,
-         this, _1 ), &claw::tween::easing_back::ease_out ) );
+         this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_out ) );
 
   tween.insert
     ( claw::tween::single_tweener
       (get_active_position().x, get_active_position().x, 1,
        boost::bind
        ( &ptb::status_component::on_x_position_update,
-	 this, _1 ), &claw::tween::easing_back::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
   
   tween.insert
     ( claw::tween::single_tweener
       (get_active_position().x, get_inactive_position().x, 0.5,
        boost::bind
        ( &ptb::status_component::on_x_position_update,
-	 this, _1 ), &claw::tween::easing_back::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
   
   add_tweener( tween );
 } // score_component::on_score_changed()
Index: bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/throwable_item_component.cpp
===================================================================
--- bear-project.orig/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/throwable_item_component.cpp
+++ bear-project/plee-the-bear-0.7.1/lib/src/ptb/layer/status/code/throwable_item_component.cpp
@@ -175,13 +175,13 @@ void ptb::throwable_item_component::init
     ( get_player().get_throwable_items().throwable_item_changed.connect
       ( boost::bind
         (&ptb::throwable_item_component::on_throwable_item_changed,
-         this, _1) ) );
+         this, boost::placeholders::_1) ) );
   
   add_signal
     ( get_player().get_throwable_items().throwable_item_stock_changed.connect
       ( boost::bind
         (&ptb::throwable_item_component::on_throwable_item_stock_changed,
-         this, _1) ) );
+         this, boost::placeholders::_1) ) );
 
   add_signal
     ( get_player().get_throwable_items().throwable_item_no_stock.connect
@@ -192,26 +192,26 @@ void ptb::throwable_item_component::init
     ( bear::engine::game::get_instance().listen_bool_variable_change
       ( game_variables::get_air_power_variable_name(get_player().get_index()),
         boost::bind
-        ( &throwable_item_component::on_power_changed, this, _1, "air") ) );
+        ( &throwable_item_component::on_power_changed, this, boost::placeholders::_1, "air") ) );
 
   add_signal
     ( bear::engine::game::get_instance().listen_bool_variable_change
       ( game_variables::get_fire_power_variable_name(get_player().get_index()),
         boost::bind
-        ( &throwable_item_component::on_power_changed, this, _1, "fire") ) );
+        ( &throwable_item_component::on_power_changed, this, boost::placeholders::_1, "fire") ) );
 
   add_signal
     ( bear::engine::game::get_instance().listen_bool_variable_change
       ( game_variables::get_water_power_variable_name(get_player().get_index()),
         boost::bind
-        ( &throwable_item_component::on_power_changed, this, _1, "water") ) );
+        ( &throwable_item_component::on_power_changed, this, boost::placeholders::_1, "water") ) );
 
   add_signal
     ( bear::engine::game::get_instance().listen_uint_variable_change
       ( game_variables::get_stones_count_variable_name
         (get_player().get_index()),
         boost::bind
-        ( &throwable_item_component::on_stones_stock_changed, this, _1) ) );
+        ( &throwable_item_component::on_stones_stock_changed, this, boost::placeholders::_1) ) );
 } // throwable_item_component::init_signals()
 
 /*----------------------------------------------------------------------------*/
@@ -324,21 +324,21 @@ void ptb::throwable_item_component::on_t
       (get_position().x, get_active_position().x, 0.3,
        boost::bind
        ( &ptb::status_component::on_x_position_update,
-         this, _1 ), &claw::tween::easing_back::ease_out ) );
+         this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_out ) );
 
   tween.insert
     ( claw::tween::single_tweener
       (get_active_position().x, get_active_position().x, 1,
        boost::bind
        ( &ptb::status_component::on_x_position_update,
-	 this, _1 ), &claw::tween::easing_back::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
   
   tween.insert
     ( claw::tween::single_tweener
       (get_active_position().x, get_inactive_position().x, 0.5,
        boost::bind
        ( &ptb::status_component::on_x_position_update,
-	 this, _1 ), &claw::tween::easing_back::ease_in ) );
+	 this, boost::placeholders::_1 ), &claw::tween::easing_back::ease_in ) );
 
   add_tweener( tween ); 
 } // throwable_item_component::on_throwable_item_changed()
