Index: capi4hylafax-01.03.00/setupconffile
===================================================================
--- capi4hylafax-01.03.00.orig/setupconffile
+++ capi4hylafax-01.03.00/setupconffile
@@ -12,6 +12,9 @@ InDDIs=""
 FaxNumber="+49.00.00000"
 Identifier="AVM CAPI4HylaFAX"
 NumberPrefix=""
+AcceptSpeech=0
+AcceptGlobalCall=0
+
 DDISamples="             DDI Params\n"
 DDISamples="$DDISamples\nSample for DDI Offset   : 11223344"
 DDISamples="$DDISamples\nSample for DDI Length   : 3"
@@ -36,6 +39,26 @@ function test_for_number {
     return 0
 }
 
+ReadConfigFile() {
+    while read item value; do
+        case $item in
+            "#");;
+            "FAXNumber:") FaxNumber=$value;;
+            "LocalIdentifier:") eval Identifier=$value ;;
+            "OutgoingMSN:") OutMSN=$value;;
+            "NumberPrefix:") NumberPrefix=$value;;
+            "UseDDI:") UseDDI=$value;;
+            "AcceptSpeech:") AcceptSpeech=$value;;
+            "AcceptGlobalCall:") AcceptGlobalCall=$value;;
+            "DDIOffset:") DDIOffset=$value;;
+            "DDILength:") DDILen=$value;;
+            "IncomingDDIs:") InDDIs=$value;;
+            "IncomingMSNs:") InMSNs=$value;;
+            *) ;;
+        esac
+    done < $1
+}
+
 function WriteToConfigFile {
 sed "
 /^ *FAXNumber *:/c\\
@@ -48,6 +71,8 @@ sed "
     NumberPrefix:               $NumberPrefix
 /^ *UseDDI *:/c\\
         UseDDI:                 $UseDDI
+/^ *AcceptSpeech *:/c\\
+        AcceptSpeech:           $AcceptSpeech
 /^ *DDIOffset *:/c\\
         DDIOffset:              $DDIOffset
 /^ *DDILength *:/c\\
@@ -55,7 +80,9 @@ sed "
 /^ *IncomingDDIs *:/c\\
         IncomingDDIs:           $InDDIs
 /^ *IncomingMSNs *:/c\\
-        IncomingMSNs:           $InMSNs" "$ConfigFileName" > "$ConfigFileName.c2fax"
+        IncomingMSNs:           $InMSNs
+/^ *AcceptGlobalCall *:/c\\
+        AcceptGlobalCall:           $AcceptGlobalCall" "$ConfigFileName" > "$ConfigFileName.c2fax"
     mv -f "$ConfigFileName.c2fax" "$ConfigFileName"
 }
 
@@ -101,6 +128,8 @@ if [ ! -e "$ConfigFileName" ] ; then
     return 1
 fi
 
+ReadConfigFile $ConfigFileName
+
 #
 #   show info dialog
 #
@@ -131,6 +160,16 @@ while [ $terminate -eq 0 ] ; do
     if [ -n "$InMSNs" ] ; then
         dialogInMSN="-- is set ---"
     fi
+    if [ $AcceptSpeech -ne 0 ]; then
+	dialogAcceptSpeech="yes"
+    else
+	dialogAcceptSpeech="no"
+    fi
+    if [ $AcceptGlobalCall -ne 0 ]; then
+	dialogAcceptGlobalCall="yes"
+    else
+        dialogAcceptGlobalCall="no"
+    fi
     if [ $UseDDI -ne 0 ] ; then
         dialogDDI="-- use ddi --"
         dialogInMSN="-- disabled -"
@@ -148,32 +187,52 @@ while [ $terminate -eq 0 ] ; do
         dialogNumPre="[   empty   ]"
     fi
 
-    dialog --backtitle "$BT" --clear            \
-           --menu "Main menu" 15 50 7           \
-           "Outgoing MSN  "   "$dialogOutMSN"   \
-           "Incoming MSNs "   "$dialogInMSN"    \
-           "DDI Params    "   "$dialogDDI"      \
-           "Fax Number    "   "$dialogFaxNum"   \
-           "Fax Identifier"   "$dialogId"       \
-           "Number Prefix "   "$dialogNumPre"   \
-           "Save & Exit   "   "" 2> res
+    dialog --backtitle "$BT" --clear                         \
+           --menu "Main menu" 17 60 9                        \
+           "Outgoing MSN       "   "$dialogOutMSN"           \
+           "Incoming MSNs      "   "$dialogInMSN"            \
+           "Accept Speech      "   "$dialogAcceptSpeech"     \
+           "Accept Global Calls"   "$dialogAcceptGlobalCall" \
+           "DDI Params         "   "$dialogDDI"              \
+           "Fax Number         "   "$dialogFaxNum"           \
+           "Fax Identifier     "   "$dialogId"               \
+           "Number Prefix      "   "$dialogNumPre"           \
+           "Save & Exit        "   "" 2> res
     if [ $? -eq 0 ] ; then
         case "`cat res`" in
-        "Outgoing MSN  ")
+        "Outgoing MSN       ")
             dialog --backtitle "$BT" --clear \
                    --inputbox "Outgoing MSN" 9 50 "$OutMSN" 2> res
             if [ $? -eq 0 ] ; then
                 OutMSN="`cat res`"
             fi
             ;;
-        "Incoming MSNs ")
+        "Incoming MSNs      ")
             dialog --backtitle "$BT" --clear \
                    --inputbox "List of Incomming MSNs (comma seperated)" 9 70 "$InMSNs" 2> res
             if [ $? -eq 0 ] ; then
                 InMSNs="`cat res`"
             fi
             ;;
-        "DDI Params    ")
+        "Accept Speech      ")
+            dialog --backtitle "$BT" --clear \
+                   --yesno "Accept also calls which a service id for speech ?" 7 30
+            if [ $? -eq 0 ] ; then
+                AcceptSpeech=1
+            else
+                AcceptSpeech=0
+            fi
+            ;;
+        "Accept Global Calls")
+            dialog --backtitle "$BT" --clear \
+                   --yesno "Accept also global calls (calls without a MSN) ?" 7 30
+            if [ $? -eq 0 ] ; then
+                AcceptGlobalCall=1
+            else
+                AcceptGlobalCall=0
+            fi
+            ;;
+        "DDI Params         ")
             tmpDDIOffset=$DDIOffset
             tmpDDILen=$DDILen
             tmpInDDIs=$InDDIs
@@ -240,28 +299,28 @@ while [ $terminate -eq 0 ] ; do
             done
             terminate=0
             ;;
-        "Fax Number    ")
+        "Fax Number         ")
             dialog --backtitle "$BT" --clear \
                    --inputbox "Fax Number (use dots instead of spaces)" 9 50 "$FaxNumber" 2> res
             if [ $? -eq 0 ] ; then
                 FaxNumber="`cat res`"
             fi
             ;;
-        "Fax Identifier")
+        "Fax Identifier     ")
             dialog --backtitle "$BT" --clear \
                    --inputbox "Fax Identifier" 9 50 "$Identifier" 2> res
             if [ $? -eq 0 ] ; then
                 Identifier="`cat res`"
             fi
             ;;
-        "Number Prefix ")
+        "Number Prefix      ")
             dialog --backtitle "$BT" --clear \
                    --inputbox "Number Prefix" 9 50 "$NumberPrefix" 2> res
             if [ $? -eq 0 ] ; then
                 NumberPrefix="`cat res`"
             fi
             ;;
-        "Save & Exit   ")
+        "Save & Exit        ")
             WriteToConfigFile
             terminate=1
             ;;
Index: capi4hylafax-01.03.00/faxaddmodem.capi
===================================================================
--- /dev/null
+++ capi4hylafax-01.03.00/faxaddmodem.capi
@@ -0,0 +1,84 @@
+#!/bin/bash
+#
+#
+#
+HylafaxSpoolDir="/var/spool/fax"
+ConfigFileName=config.faxCAPI
+DestDirectory=/usr/bin
+FAXUSER=fax
+FAXGROUP=dialout
+CFGFILEMODE=0644
+#
+#
+#
+function SetConfigLine {
+    if [ -z "$3" ] ; then
+        grep -v "$2.*:.*/c2fax" $1 > $1.new
+    else
+        gawk 'BEGIN {repl = 0} END {if (repl == 0) {print SEARCHIT":"REPLACEIT}}
+             {if ($1 != SEARCHIT) {print $0} else {repl = 1; print $1":"REPLACEIT}}
+             ' FS=':' IGNORECASE=1 SEARCHIT="$2" REPLACEIT="$3" $1 > $1.new
+    fi
+    if [ "$4" != "no" ] ; then
+        mv $1 $1.c2fax
+    fi
+    mv $1.new $1
+}
+
+
+#
+#
+#
+if [ ! -e $HylafaxSpoolDir/etc/config ] ; then
+    echo ""
+    echo "Can't find Hylafax's config file: $HylafaxSpoolDir/etc/config."
+    echo "Please run faxsetup before setting up capi modems"
+    exit 1
+fi
+
+#
+#
+#
+if [ -e "/etc/$ConfigFileName" ] ; then
+    echo "Backup /etc/$ConfigFileName to /etc/$ConfigFileName.c2fax"
+    cp "/etc/$ConfigFileName" "/etc/$ConfigFileName.c2fax"
+elif [ -e "/etc/$ConfigFileName.sample" ] ; then
+    cp "/etc/$ConfigFileName.sample" "/etc/$ConfigFileName"
+elif [ -e "/usr/share/doc/packages/capi4hylafax/$ConfigFileName" ] ; then
+    cp "/usr/share/doc/packages/capi4hylafax/$ConfigFileName" "/etc/$ConfigFileName"
+else
+    echo "No sample configfile for /etc/$ConfigFileName"
+    echo "found, cannot create one, exit"
+    exit
+fi
+
+ConfigFileName="/etc/$ConfigFileName"
+SetConfigLine "$ConfigFileName" SpoolDir "\t\t$HylafaxSpoolDir" no
+SetConfigLine "$ConfigFileName" FaxRcvdCmd "\t\t$HylafaxSpoolDir/bin/faxrcvd" no
+
+#
+# capi devices should owned by uucp
+#
+chown uucp /dev/capi*
+
+# adjust hylafax config file and inittab
+echo "Change $HylafaxSpoolDir/etc/config and /etc/inittab."
+SetConfigLine $HylafaxSpoolDir/etc/config SendFaxCmd "\t\t$HylafaxSpoolDir/bin/faxsend"
+SetConfigLine /etc/inittab fr "35:respawn:$DestDirectory/c2faxrecv -q \"$HylafaxSpoolDir\""
+
+. /usr/sbin/faxaddmodem.capi_dia $HylafaxSpoolDir
+
+main_config_dialog "$ConfigFileName"
+
+# fixup permissions
+chmod $CFGFILEMODE $HylafaxSpoolDir/etc/config*
+chmod $CFGFILEMODE /etc/config.faxCAPI*
+chmod $CFGFILEMODE /etc/inittab
+chown ${FAXUSER}:${FAXGROUP} $HylafaxSpoolDir/etc/config*
+chown ${FAXUSER}:${FAXGROUP} /etc/config.faxCAPI*
+
+# that was it
+
+echo -e "\nInstall was successfully!\n"
+echo "The Fax Receive Daemon c2faxrecv was added to /etc/inittab and will be started"
+echo "automatically on system start. You could start it manually now with telinit q."
Index: capi4hylafax-01.03.00/README.SuSE
===================================================================
--- /dev/null
+++ capi4hylafax-01.03.00/README.SuSE
@@ -0,0 +1,12 @@
+Short hints for using capi4hylafax under SuSE Linux
+
+1. call faxsetup and follow the dialog
+2. call faxaddmodem, answer the first question
+   "Do you want setup a CAPI 2.0 based ISDN modem"
+   with yes, follow the dialog.
+3. start c2faxrecv via telinit -q
+
+After this fax receiving and sending should work you can test it for example
+with:
+sendfax -d <FaxPhoneNumber> <this file>
+
Index: capi4hylafax-01.03.00/LIESMICH.SuSE
===================================================================
--- /dev/null
+++ capi4hylafax-01.03.00/LIESMICH.SuSE
@@ -0,0 +1,16 @@
+Einige kurze Hinweise zur Benutzung von capi4hylafax unter SuSE Linux
+
+1. faxsetup starten und entsprechend configurieren
+2. faxaddmodem starten und die Frage
+   "Do you want setup a CAPI 2.0 based ISDN modem"
+   mit yes beantworten.
+3. Starten des c2faxrecv daemons durch telinit -q
+
+Danach sollte der Faxempfang und das Faxsenden funktionieren.
+Sie können das z.B. mit
+sendfax -d <Rufnummer> <dieses file>
+
+ein Testfax verschicken.
+
+Um aus anderen Anwendungen Faxe zu verschicken siehe
+http://sdb.suse.de/de/sdb/html/ghillie_hylafax_einrichten.html
