From 0c9ea3c00cc0a267b3ef9dd7a0c0874f466e9cf6 Mon Sep 17 00:00:00 2001
From: Marcelo Trylesinski <marcelotryle@gmail.com>
Date: Thu, 16 Sep 2021 13:48:07 +0200
Subject: [PATCH] Update setup.cfg

---
 setup.cfg | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/setup.cfg b/setup.cfg
index f59a7202..1266d95c 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -29,6 +29,8 @@ filterwarnings=
     ignore: Using or importing the ABCs from 'collections' instead of from 'collections\.abc' is deprecated.*:DeprecationWarning
     ignore: The 'context' alias has been deprecated. Please use 'context_value' instead\.:DeprecationWarning
     ignore: The 'variables' alias has been deprecated. Please use 'variable_values' instead\.:DeprecationWarning
+    # Workaround for Python 3.9.7 (see https://bugs.python.org/issue45097)
+    ignore:The loop argument is deprecated since Python 3\.8, and scheduled for removal in Python 3\.10\.:DeprecationWarning:asyncio
 
 [coverage:run]
 source_pkgs = starlette, tests
