Subject: virt-install: Use minutes instead of seconds on get_time_string()
From: Fabiano Fidêncio fidencio@redhat.com Thu Jul 25 14:09:53 2019 +0200
Date: Thu Jul 25 19:13:13 2019 -0400:
Git: 414ffa5ef82548d502a4de98d40228a68bdb29c1

get_time_string() currently uses self._wait_secs, while it should use
self._wait_mins, resulting on confusing messages when using `--wait`
option, as shown below:

fidencio@laerte ~/src/upstream/virt-manager $ ./virt-install --install fedora30 --unattended --wait 20
 ...
Waiting 1200 minutes for installation to complete.

Reviewed-by: Cole Robinson <crobinso@redhat.com>
Signed-off-by: Fabiano Fidêncio <fidencio@redhat.com>

diff --git a/virt-install b/virt-install
index af6c75bf..8efa4960 100755
--- a/virt-install
+++ b/virt-install
@@ -610,7 +610,7 @@ class WaitHandler:
         self._start_time = time.time()
 
     def get_time_string(self):
-        timestr = _(" %d minutes") % self._wait_secs
+        timestr = _(" %d minutes") % self._wait_mins
         if self._wait_forever:
             timestr = ""
         ret = _("Waiting%(time_string)s for installation to complete.") % {
