Index: virt-manager-2.2.0/virtconv/vmx.py
===================================================================
--- virt-manager-2.2.0.orig/virtconv/vmx.py
+++ virt-manager-2.2.0/virtconv/vmx.py
@@ -259,7 +259,7 @@ class vmx_parser(parser_class):
 
         def _find_keys(prefixes):
             ret = []
-            for key, value in config.items():
+            for key, value in list(config.items()):
                 for p in xmlutil.listify(prefixes):
                     if key.startswith(p):
                         ret.append((key, value))
