diff --color -ur zcfan-1.4.0.orig/Makefile zcfan-1.4.0/Makefile
--- zcfan-1.4.0.orig/Makefile	2024-12-18 01:36:27.000000000 +0100
+++ zcfan-1.4.0/Makefile	2024-12-31 09:28:33.344935876 +0100
@@ -5,8 +5,8 @@
 EXECUTABLES=$(patsubst %.c,%,$(SOURCES))
 
 INSTALL:=install
-prefix:=/usr/local
-bindir:=$(prefix)/bin
+prefix:=/usr
+bindir:=$(prefix)/sbin
 datarootdir:=$(prefix)/share
 mandir:=$(datarootdir)/man
 
