diff -ur gnome-keysign-0.9.6.orig/setup.py gnome-keysign-0.9.6/setup.py
--- gnome-keysign-0.9.6.orig/setup.py	2017-12-15 08:40:54.000000000 -0600
+++ gnome-keysign-0.9.6/setup.py	2018-03-27 12:04:16.803946822 -0500
@@ -114,7 +114,10 @@
     catalogs = {}
 
     for pofile in pofiles:
-        catalog = read_po(open(pofile, 'r'))
+        if sys.version_info.major > 2:
+            catalog = read_po(open(pofile, 'r', encoding='utf-8'))
+        else:
+            catalog = read_po(open(pofile, 'r'))
         catalogs[catalog.locale] = catalog
         logging.info("Found %d strings for %s", len(catalog), catalog.locale)
         # logging.debug("Strings for %r", catalog, catalog.values())
