--- scala-2.10.7/src/compiler/scala/tools/ant/antlib.xml	2023-09-28 18:35:06.147897841 +0200
+++ scala-2.10.7/src/compiler/scala/tools/ant/antlib.xml	2023-09-28 18:47:47.202949642 +0200
@@ -13,6 +13,4 @@
              classname="scala.tools.ant.ScalaTool"/>
     <taskdef name="same"
              classname="scala.tools.ant.Same"/>
-    <taskdef name="pack200"
-             classname="scala.tools.ant.Pack200Task"/>
 </antlib>
Only in scala-2.10.7/src/compiler/scala/tools/ant: Pack200Task.scala
--- scala-2.10.7/src/library/scala/collection/mutable/StringBuilder.scala	2023-09-28 18:35:06.187898108 +0200
+++ scala-2.10.7/src/library/scala/collection/mutable/StringBuilder.scala	2023-09-28 18:49:06.536809546 +0200
@@ -437,6 +437,14 @@
    *  @return  the string assembled by this StringBuilder
    */
   def result(): String = toString
+
+  /** Tests whether this builder is empty.
+   *
+   *  This method is required for JDK15+ compatibility
+   *
+   *  @return  `true` if this builder contains nothing, `false` otherwise.
+   */
+  override def isEmpty: Boolean = underlying.length() == 0
 }
 
 object StringBuilder {
