Index: pocketsphinx-5prealpha+git~20181130T131256~600fe3e/src/programs/continuous.c
===================================================================
--- pocketsphinx-5prealpha+git~20181130T131256~600fe3e.orig/src/programs/continuous.c
+++ pocketsphinx-5prealpha+git~20181130T131256~600fe3e/src/programs/continuous.c
@@ -304,7 +304,7 @@ main(int argc, char *argv[])
         return 1;
     }
 
-    E_INFO("%s COMPILED ON: %s, AT: %s\n\n", argv[0], __DATE__, __TIME__);
+    E_INFO("%s\n\n", argv[0]);
 
     if (cmd_ln_str_r(config, "-infile") != NULL) {
         recognize_from_file();
