--- a/llvm/test/CMakeLists.txt
+++ b/llvm/test/CMakeLists.txt
@@ -81,7 +81,6 @@ set(LLVM_TEST_DEPENDS
           llvm-dwarfdump
           llvm-dwarfutil
           llvm-dwp
-          llvm-exegesis
           llvm-extract
           llvm-gsymutil
           llvm-isel-fuzzer
--- a/llvm/test/tools/llvm-exegesis/PowerPC/lit.local.cfg
+++ b/llvm/test/tools/llvm-exegesis/PowerPC/lit.local.cfg
@@ -1,3 +1 @@
-if not ('PowerPC' in config.root.targets):
-    # We need support for PowerPC.
-    config.unsupported = True
+config.unsupported = True
--- a/llvm/test/tools/llvm-exegesis/X86/lit.local.cfg
+++ b/llvm/test/tools/llvm-exegesis/X86/lit.local.cfg
@@ -1,3 +1 @@
-if not ('X86' in config.root.targets):
-    # We need support for X86.
-    config.unsupported = True
+config.unsupported = True
