Index: uproot-4.2.1/src/uproot/_util.py
===================================================================
--- uproot-4.2.1.orig/src/uproot/_util.py
+++ uproot-4.2.1/src/uproot/_util.py
@@ -16,7 +16,7 @@ from collections.abc import Iterable
 from urllib.parse import unquote, urlparse
 
 import numpy
-import setuptools
+import packaging
 
 win = platform.system().lower().startswith("win")
 
@@ -92,10 +92,10 @@ def parse_version(version):
     Converts a semver string into a Version object that can be compared with
     ``<``, ``>=``, etc.
 
-    Currently implemented using ``setuptools.extern.packaging.version.parse``
+    Currently implemented using ``packaging.version.parse``
     (exposing that library in the return type).
     """
-    return setuptools.extern.packaging.version.parse(version)
+    return packaging.version.parse(version)
 
 
 def from_module(obj, module_name):
