diff -PdpruN a/gtests/CMakeLists.txt b/gtests/CMakeLists.txt
--- a/gtests/CMakeLists.txt	2013-07-06 05:11:56.000000000 -0400
+++ b/gtests/CMakeLists.txt	2013-10-17 18:18:57.093590757 -0400
@@ -25,11 +25,11 @@
 #
 # -------------------------------------------------------------------------
 
-file ( GLOB ITPP_TEST_SRCS *.cpp )
+file ( GLOB ITPP_TEST_SRCS *.cpp gtest_main.cc )
 
 include_directories ( ${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR} ${GTEST_DIR} ${GTEST_DIR}/include )
 
-add_executable ( itpp_gtests ${ITPP_TEST_SRCS} ${GTEST_DIR}/src/gtest-all.cc ${GTEST_DIR}/src/gtest_main.cc )
+add_executable ( itpp_gtests ${ITPP_TEST_SRCS} )
 foreach (FILE ${ITPP_TEST_SRCS})
   get_filename_component ( target ${FILE} NAME_WE )
   if (target MATCHES "^fastica")
@@ -51,7 +51,7 @@ foreach (FILE ${ITPP_TEST_SRCS})
 endforeach (FILE)
 
 if (NOT WIN32)
-  target_link_libraries ( itpp_gtests ${ITPP_LIBS} ${libitpp_target} pthread )
+  target_link_libraries ( itpp_gtests ${ITPP_LIBS} ${libitpp_target} pthread gtest )
 else()
-  target_link_libraries ( itpp_gtests ${ITPP_LIBS} ${libitpp_target} )
+  target_link_libraries ( itpp_gtests ${ITPP_LIBS} ${libitpp_target} gtest )
 endif()
diff -PdpruN a/gtests/gtest_main.cc b/gtests/gtest_main.cc
--- a/gtests/gtest_main.cc	1969-12-31 19:00:00.000000000 -0500
+++ b/gtests/gtest_main.cc	2011-04-15 15:49:11.000000000 -0400
@@ -0,0 +1,39 @@
+// Copyright 2006, Google Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+// copyright notice, this list of conditions and the following disclaimer
+// in the documentation and/or other materials provided with the
+// distribution.
+//     * Neither the name of Google Inc. nor the names of its
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include <iostream>
+
+#include "gtest/gtest.h"
+
+GTEST_API_ int main(int argc, char **argv) {
+  std::cout << "Running main() from gtest_main.cc\n";
+
+  testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
