Index: respx-0.20.1/tests/test_patterns.py
===================================================================
--- respx-0.20.1.orig/tests/test_patterns.py
+++ respx-0.20.1/tests/test_patterns.py
@@ -280,9 +280,9 @@ def test_params_pattern_hash():
         (Lookup.STARTS_WITH, "http://a.b/baz/", {}, "https://a.b/baz/", False),
         (
             Lookup.EQUAL,
-            (b"https", b"fake:ipv6", None, b""),
+            (b"https", b"2001:db8::1", None, b""),
             {},
-            "https://[fake:ipv6]",
+            "https://[2001:db8::1]",
             True,
         ),
     ],
