Index: glibc-2.27/nscd/nscd.h
===================================================================
--- glibc-2.27.orig/nscd/nscd.h
+++ glibc-2.27/nscd/nscd.h
@@ -161,11 +161,11 @@ struct database_dyn
 
 
 /* Paths of the file for the persistent storage.  */
-#define _PATH_NSCD_PASSWD_DB	"/var/db/nscd/passwd"
-#define _PATH_NSCD_GROUP_DB	"/var/db/nscd/group"
-#define _PATH_NSCD_HOSTS_DB	"/var/db/nscd/hosts"
-#define _PATH_NSCD_SERVICES_DB	"/var/db/nscd/services"
-#define _PATH_NSCD_NETGROUP_DB	"/var/db/nscd/netgroup"
+#define _PATH_NSCD_PASSWD_DB	"/var/lib/nscd/passwd"
+#define _PATH_NSCD_GROUP_DB	"/var/lib/nscd/group"
+#define _PATH_NSCD_HOSTS_DB	"/var/lib/nscd/hosts"
+#define _PATH_NSCD_SERVICES_DB	"/var/lib/nscd/services"
+#define _PATH_NSCD_NETGROUP_DB	"/var/lib/nscd/netgroup"
 
 /* Path used when not using persistent storage.  */
 #define _PATH_NSCD_XYZ_DB_TMP	"/var/run/nscd/dbXXXXXX"
