--- loook.desktop.orig	2016-02-13 17:04:49.000000000 +0100
+++ loook.desktop	2016-03-08 18:40:47.412530493 +0100
@@ -5,6 +5,7 @@
 
 Version=1.0
 Name=Loook
+GenericName=Loook
 Comment=Search strings in ODF documents
 Comment[de]=Sucht Begriffe in ODF Dokumenten
 
@@ -12,6 +13,6 @@
 Icon=loook
 Terminal=false
 Type=Application
-Categories=Office;Utility
-Keywords=Office;Utility;ODF Files; Search
-StartupNotify=true
+Categories=Office;WordProcessor;Utility;
+Keywords=Office;Utility;ODF Files;Search;
+StartupNotify=false
