diff -ur librtas-2.0.4~/Makefile.am librtas-2.0.4/Makefile.am
--- librtas-2.0.4~/Makefile.am	2023-01-24 19:08:58.000000000 +0100
+++ librtas-2.0.4/Makefile.am	2023-01-25 12:35:10.011935185 +0100
@@ -3,7 +3,7 @@
 EXTRA_DIST = README COPYING.LESSER
 EXTRA_DIST += doc/doxygen.rtas doc/doxygen.rtasevent Changelog
 
-docdir = $(datadir)/doc/@PACKAGE@
+docdir = $(datadir)/doc/packages/@PACKAGE@
 doc_DATA = README COPYING.LESSER
 
 bin_PROGRAMS=
