diff --git a/pytest-verbose-parametrize/tests/unit/test_verbose_parametrize.py b/pytest-verbose-parametrize/tests/unit/test_verbose_parametrize.py
index fcce01d..bba9aa6 100644
--- pytest-verbose-parametrize/tests/unit/test_verbose_parametrize.py
+++ pytest-verbose-parametrize.patched/tests/unit/test_verbose_parametrize.py
@@ -1,5 +1,8 @@
 # -*- coding: utf-8 -*-
-from mock import Mock
+try:
+    from unittest.mock import Mock
+except ImportError:
+    from mock import Mock
 
 from pytest_verbose_parametrize import pytest_generate_tests
 
