From 4ed6ad8ef3c5e2bb97ba6c16c0d368d476b4862d Mon Sep 17 00:00:00 2001
From: escribana <escriban@bluewin.ch>
Date: Mon, 15 Aug 2022 17:42:00 +0200
Subject: [PATCH] Corrections for Pegasus Falcon Rotator: (#1721)

---
 drivers/rotator/pegasus_falcon.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/rotator/pegasus_falcon.cpp b/drivers/rotator/pegasus_falcon.cpp
index c78f0e8fdf..4bb787f60c 100644
--- a/drivers/rotator/pegasus_falcon.cpp
+++ b/drivers/rotator/pegasus_falcon.cpp
@@ -191,15 +191,15 @@ bool PegasusFalcon::AbortRotator()
 }
 
 //////////////////////////////////////////////////////////////////////
-///
+/// Command for reverse action ("FN:0" disabled, "FN:1" enabled)
 //////////////////////////////////////////////////////////////////////
 bool PegasusFalcon::ReverseRotator(bool enabled)
 {
     char cmd[DRIVER_LEN] = {0}, res[DRIVER_LEN] = {0};
-    snprintf(cmd, DRIVER_LEN, "FR:%d", enabled ? 1 : 0);
+    snprintf(cmd, DRIVER_LEN, "FN:%d", enabled ? 1 : 0);
     if (sendCommand(cmd, res))
     {
-        return (!strcmp(res, cmd));
+        return (!strncmp(res, cmd, 4)); //Restrict length to 4 chars!
     }
 
     return false;

