--- a/tests/test-thread.lua
+++ b/tests/test-thread.lua
@@ -58,28 +58,4 @@ return require('lib/tap')(function (test
     assert(elapsed >= delay, "elapsed should be at least delay ")
   end)
 
-  test("test thread create with options table", function(print, p, expect, uv)
-    local delay = 100
-    uv.update_time()
-    local before = uv.now()
-    local args = {delay, 'string', nil, false, 5, "helloworld"}
-    local unpack = unpack or table.unpack
-    uv.new_thread({stack_size=0}, function(delay,s,null,bool,five,hw)
-      assert(type(delay) == "number")
-      assert(type(s) == "string")
-      assert(null == nil)
-      assert(bool == false)
-      assert(five == 5)
-      assert(hw == 'helloworld')
-      require('luv').sleep(delay)
-    end, unpack(args)):join()
-    uv.update_time()
-    local elapsed = uv.now() - before
-    p({
-      delay = delay,
-      elapsed = elapsed
-    })
-    assert(elapsed >= 100, "elapsed should be at least delay ")
-  end)
-
 end)
