---
 zipl/boot/menu.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

--- a/zipl/boot/menu.c
+++ b/zipl/boot/menu.c
@@ -168,8 +168,11 @@
 	/* print config list */
 	menu_list();
 
-	if (is_zvm())
-		printf("Note: VM users please use '#cp vi vmsg <input>'\n");
+	if (is_zvm()) {
+		printf(" \n");
+		printf("Note: VM users please use '#cp vi vmsg <input> <kernel-parameters>'\n");
+		printf(" \n");
+	}
 
 	value = menu_read();
 
