diff -Naur a/etc/modules-load.d/s390-pkey.conf b/etc/modules-load.d/s390-pkey.conf
--- a/etc/modules-load.d/s390-pkey.conf	2026-09-21 19:36:47.000000000 +0200
+++ b/etc/modules-load.d/s390-pkey.conf	2026-09-22 10:04:20.104771268 +0200
@@ -4,4 +4,3 @@
 pkey_ep11
 pkey_cca
 paes_s390
-phmac_s390
diff -Naur a/zkey/dracut/99-pkey.conf.in b/zkey/dracut/99-pkey.conf.in
--- a/zkey/dracut/99-pkey.conf.in	2026-09-21 19:36:47.000000000 +0200
+++ b/zkey/dracut/99-pkey.conf.in	2026-09-22 10:02:17.647392941 +0200
@@ -4,6 +4,6 @@
 # it under the terms of the MIT license. See LICENSE for details.
 #
 
-add_drivers+=" uvdevice pkey pkey_cca pkey_ep11 pkey_pckmo pkey_uv paes_s390 phmac_s390 zcrypt zcrypt_cex4 "
+add_drivers+=" uvdevice pkey pkey_cca pkey_ep11 pkey_pckmo pkey_uv paes_s390 zcrypt zcrypt_cex4 "
 install_items+=" chzcrypt lszcrypt zkey zkey-cryptsetup "
 install_optional_items+=" %ZKEYKMSPLUGINDIR%/*.so  /etc/zkey/* /etc/zkey/repository/* /etc/zkey/kmip/profiles/* "
diff -Naur a/zkey/initramfs/hooks/s390-tools-zkey.in b/zkey/initramfs/hooks/s390-tools-zkey.in
--- a/zkey/initramfs/hooks/s390-tools-zkey.in	2026-09-21 19:36:47.000000000 +0200
+++ b/zkey/initramfs/hooks/s390-tools-zkey.in	2026-09-22 10:02:57.271515356 +0200
@@ -29,7 +29,7 @@
 . /usr/share/initramfs-tools/hook-functions
 
 # Add zcrypt modules
-zdev_modules="uvdevice pkey pkey_cca pkey_ep11 pkey_pckmo pkey_uv paes_s390 phmac_s390 zcrypt zcrypt_cex4"
+zdev_modules="uvdevice pkey pkey_cca pkey_ep11 pkey_pckmo pkey_uv paes_s390 zcrypt zcrypt_cex4"
 
 for x in $zdev_modules ; do
 	manual_add_modules ${x}
