Index: invoke-1.6.0/invoke/config.py
===================================================================
--- invoke-1.6.0.orig/invoke/config.py
+++ invoke-1.6.0/invoke/config.py
@@ -906,7 +906,7 @@ class Config(DataProxy):
 
     def _load_yaml(self, path):
         with open(path) as fd:
-            return yaml.load(fd)
+            return yaml.load(fd, Loader=yaml.SafeLoader)
 
     def _load_yml(self, path):
         return self._load_yaml(path)
