Index: lalapps-9.2.1/src/string/StringSearch.c
===================================================================
--- lalapps-9.2.1.orig/src/string/StringSearch.c
+++ lalapps-9.2.1/src/string/StringSearch.c
@@ -200,7 +200,7 @@ int main(int argc,char *argv[])
   unsigned seg_length;
   StringTemplate strtemplate[MAXTEMPLATES];
   int NTemplates;
-  int NTemplates_fix; /* number of template given by the template bank file */
+  int NTemplates_fix = 0; /* number of template given by the template bank file */
   REAL8 fcutoff_fix[MAXTEMPLATES]; /* high frequency cutoffs given by the template bank file */
   SnglBurst *events=NULL;
   ProcessTable *process;
