From 15f80c05b7aa3e62067c6424df805bb43881e05d Mon Sep 17 00:00:00 2001
From: Atri Bhattacharya <badshah400@gmail.org>
Date: Sat, 12 Apr 2025 16:40:31 +0530
Subject: [PATCH] Rename library when building with EGL.

---
 build/cmake/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

Index: glew-2.3.0/build/cmake/CMakeLists.txt
===================================================================
--- glew-2.3.0.orig/build/cmake/CMakeLists.txt
+++ glew-2.3.0/build/cmake/CMakeLists.txt
@@ -57,6 +57,9 @@ else ()
   set (GLEW_LIB_NAME GLEW)
   set (DLL_PREFIX lib)
 endif ()
+if (GLEW_EGL)
+  set (GLEW_LIB_NAME ${GLEW_LIB_NAME}_EGL)
+endif()
 
 add_definitions (-DGLEW_NO_GLU)
 
