--- qclib-2.2.0/Makefile	2020-09-19 00:27:40.249451093 +0200
+++ qclib-2.2.99/Makefile	2020-09-19 00:21:59.292489845 +0200
@@ -74,50 +75,6 @@ html: $(CFILES) query_capacity.h query_c
 		echo "Error: 'doxygen' not installed"; \
 	fi
 
-		cppcheck query_capacity*.[ch] 2>&1 | sed 's/^/   /'; \
-	else \
-		echo "cppcheck not available"; \
-	fi
-	@echo;
-	-@if which smatch >/dev/null 2>&1; then \
-		echo "Running smatch"; \
-		smatch query_capacity*.[ch] 2>&1 | sed 's/^/   /'; \
-	else \
-		echo "smatch not available"; \
-	fi
-	@echo;
-	-@if which sparse >/dev/null 2>&1; then \
-		echo "Running sparse"; \
-		sparse -Wsparse-all query_capacity*.[ch] 2>&1 | sed 's/^/   /'; \
-	else \
-		echo "sparse not available"; \
-	fi
-	@echo;
-	-@if which valgrind >/dev/null 2>&1; then \
-		echo "Running valgrind: qc_test"; \
-		LD_LIBRARY_PATH=. valgrind --leak-check=full --show-leak-kinds=all ./qc_test-sh `find regtest_data -maxdepth 1 -mindepth 1 -type d | \
-			grep -ve "^regtest_data/extras_" | sort` 2>&1 | grep -A 100 "HEAP SUMMARY:" | sed 's/^/   /'; \
-		echo; \
-		echo "zname: Running valgrind: zname -a"; \
-		QC_USE_DUMP=regtest_data/dump_r3545038 LD_LIBRARY_PATH=. valgrind --leak-check=full --show-leak-kinds=all ./zname -a | \
-			grep -A 100 "HEAP SUMMARY:" | sed 's/^/   /'; \
-		echo; \
-		echo "zname: Running valgrind: zhypinfo"; \
-		QC_USE_DUMP=regtest_data/dump_r3545038 LD_LIBRARY_PATH=. valgrind --leak-check=full --show-leak-kinds=all ./zhypinfo | \
-			grep -A 100 "HEAP SUMMARY:" | sed 's/^/   /'; \
-		echo; \
-		echo "zname: Running valgrind: zhypinfo -L"; \
-		QC_USE_DUMP=regtest_data/dump_r3545038 LD_LIBRARY_PATH=. valgrind --leak-check=full --show-leak-kinds=all ./zhypinfo -L | \
-			grep -A 100 "HEAP SUMMARY:" | sed 's/^/   /'; \
-		echo; \
-		echo "Running callgrind: qctest"; \
-		LD_LIBRARY_PATH=. valgrind --callgrind-out-file=/dev/null --tool=callgrind ./zhypinfo -l `find regtest_data -maxdepth 1 -mindepth 1 -type d | \
-			grep -ve "^regtest_data/extras_" | sort` 2>&1 | grep "refs:" | sed 's/^/   /'; \
-	else \
-		echo "valgrind not available"; \
-	fi
-	@echo;
-
 install: libqc.a libqc.so.$(VERSION)
 	echo "  INSTALL"
 	install -Dm 644 libqc.a $(DESTDIR)/usr/lib64/libqc.a
