Index: xzgv-0.9.2/src/Makefile
===================================================================
--- xzgv-0.9.2.orig/src/Makefile
+++ xzgv-0.9.2/src/Makefile
@@ -28,16 +28,14 @@ xzgv: $(OBJS)
 	$(CC) -o xzgv $(OBJS) $(LDFLAGS)
 
 installdirs:
-	/bin/sh ../mkinstalldirs $(BINDIR)
+	/bin/sh ../mkinstalldirs $(BINDIR) $(DESKTOPDIR1) $(PIXMAPDIR)
 
 install: xzgv installdirs
 	install xzgv $(BINDIR)
 	chmod 755 $(BINDIR)/xzgv
 	install xzgv.desktop $(DESKTOPDIR1)
-	install xzgv.desktop $(DESKTOPDIR2)
 	install icon-64.xpm $(PIXMAPDIR)/xzgv.xpm
 	chmod 644 $(DESKTOPDIR1)/xzgv.desktop
-	chmod 644 $(DESKTOPDIR2)/xzgv.desktop
 	chmod 644 $(PIXMAPDIR)/xzgv.xpm
 
 uninstall:
