---
 build/CMakeLists.txt |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: ZeroBraneStudio-2.01/build/CMakeLists.txt
===================================================================
--- ZeroBraneStudio-2.01.orig/build/CMakeLists.txt	2023-09-27 03:26:58.000000000 +0200
+++ ZeroBraneStudio-2.01/build/CMakeLists.txt	2025-10-16 22:38:21.281284874 +0200
@@ -102,7 +102,7 @@
   install_from_manifest(PROGRAMS ${TOPDIR}/zbstudio/MANIFEST-bin-macos ${DATADIR} "^$")
 else()
   # check the available Lua modules
-  check_lua_module(wx TRUE)
+  find_package(wxLua REQUIRED)
   check_lua_module(socket TRUE)
   check_lua_module(copas ${USE_SYSTEM_COPAS})
   check_lua_module(luainspect ${USE_SYSTEM_LUAINSPECT})
