From 74a4eff529e6b71f6a822841d9112d9b71b44d24 Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Sat, 13 Jan 2018 15:16:39 +0100
Subject: [PATCH] Warning for FlatHub

---
 .../backends/FlatpakBackend/FlatpakSourcesBackend.cpp      | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

Index: discover-5.17.80git.20210404T011831~f9348542/libdiscover/backends/FlatpakBackend/FlatpakSourcesBackend.cpp
===================================================================
--- discover-5.17.80git.20210404T011831~f9348542.orig/libdiscover/backends/FlatpakBackend/FlatpakSourcesBackend.cpp	2021-04-05 10:10:15.785178715 +0200
+++ discover-5.17.80git.20210404T011831~f9348542/libdiscover/backends/FlatpakBackend/FlatpakSourcesBackend.cpp	2021-04-05 10:12:16.117507702 +0200
@@ -12,6 +12,8 @@
 #include <KLocalizedString>
 #include <KSharedConfig>
 #include <QDebug>
+#include <QAbstractButton>
+#include <QMessageBox>
 #include <QNetworkAccessManager>
 #include <QNetworkReply>
 
@@ -53,8 +55,18 @@
     m_flathubAction->setObjectName(QStringLiteral("flathub"));
     m_flathubAction->setToolTip(i18n("Makes it possible to easily install the applications listed in https://flathub.org"));
     connect(m_flathubAction, &DiscoverAction::triggered, this, [this]() {
+        auto *warnBox = new QMessageBox(QMessageBox::Information, i18n("Adding a third-party source"),
+                            i18n("FlatHub is not part of the openSUSE distribution.\n"
+                                 "It contains packages of unknown quality which might not be tested to work correctly "
+                                 "on openSUSE. Prefer to install software from the official openSUSE repository."),
+                            QMessageBox::Ok | QMessageBox::Abort);
+        connect(warnBox, &QMessageBox::buttonClicked, this, [this, warnBox](QAbstractButton *button) {
+            if (warnBox->standardButton(button) == QMessageBox::Ok)
         addSource(QStringLiteral("https://flathub.org/repo/flathub.flatpakrepo"));
     });
+        warnBox->setAttribute(Qt::WA_DeleteOnClose);
+        warnBox->show();
+    });
     for (auto installation : installations) {
         if (!listRepositories(installation)) {
             qWarning() << "Failed to list repositories from installation" << installation;
