diff -uprN xsp-3.0.11.old/packaging/opensuse/xsp2.init xsp-3.0.11.new/packaging/opensuse/xsp2.init
--- xsp-3.0.11.old/packaging/opensuse/xsp2.init	2013-05-14 15:46:56.000000000 +0400
+++ xsp-3.0.11.new/packaging/opensuse/xsp2.init	2016-06-08 15:07:47.954947643 +0300
@@ -111,7 +111,7 @@
 
 # Check for missing binaries (stale symlinks should not happen)
 # Note: Special treatment of stop for LSB conformance
-XSP2_BIN=/usr/lib/mono/2.0/xsp2.exe
+XSP2_BIN=/usr/lib/mono/xsp2/xsp2.exe
 XSP2_WRAPPER=/usr/bin/xsp2
 MONO_BIN=/usr/bin/mono
 test -x $XSP2_BIN || { echo "$XSP2_BIN not installed"; 
diff -uprN xsp-3.0.11.old/scripts/Makefile.am xsp-3.0.11.new/scripts/Makefile.am
--- xsp-3.0.11.old/scripts/Makefile.am	2013-05-14 15:46:56.000000000 +0400
+++ xsp-3.0.11.new/scripts/Makefile.am	2016-06-08 15:08:27.210947559 +0300
@@ -16,11 +16,11 @@ bin_SCRIPTS = $(bin2_scripts) $(tool2_sc
 
 CLEANFILES = $(bin2_scripts_real) $(tool2_scripts) $(bin_scripts) $(tool_scripts) $(bin4_scripts) $(tool4_scripts)
 
-plat_bindir2 = $(prefix)/lib/mono/2.0
+plat_bindir2 = $(prefix)/lib/mono/xsp2
 plat_tooldir2 = $(prefix)/lib/xsp/2.0
 
 if NET_4_0
-plat_bindir4 = $(prefix)/lib/mono/4.5
+plat_bindir4 = $(prefix)/lib/mono/xsp4
 plat_tooldir4 = $(prefix)/lib/xsp/4.0
 endif
 
diff -uprN xsp-3.0.11.old/src/Mono.WebServer/Makefile.am xsp-3.0.11.new/src/Mono.WebServer/Makefile.am
--- xsp-3.0.11.old/src/Mono.WebServer/Makefile.am	2013-05-14 15:46:56.000000000 +0400
+++ xsp-3.0.11.new/src/Mono.WebServer/Makefile.am	2016-06-08 15:10:08.520628986 +0300
@@ -6,13 +6,13 @@ SYSTEMCORE_DEP=@SYSTEMCORE_DEP@
 monowebserver2dir = $(prefix)/lib/xsp/2.0
 monowebserver2_install = 2.0/Mono.WebServer2.dll
 monowebserver2_references = -r:System.Web.dll -r:System.Configuration.dll -r:Mono.Security.dll -r:System.Core.dll
-GACUTIL2=$(GACUTIL) -package 2.0
+GACUTIL2=$(GACUTIL) -package xsp2
 
 if NET_4_0
 monowebserver4dir = $(prefix)/lib/xsp/4.5
 monowebserver4_install = 4.0/Mono.WebServer2.dll
 monowebserver4_references = $(monowebserver2_references)
-GACUTIL4=$(GACUTIL) -package 4.5
+GACUTIL4=$(GACUTIL) -package xsp4
 endif
 
 noinst_SCRIPTS= $(monowebserver2_install) $(monowebserver4_install)
diff -uprN xsp-3.0.11.old/src/Mono.WebServer.Apache/Makefile.am xsp-3.0.11.new/src/Mono.WebServer.Apache/Makefile.am
--- xsp-3.0.11.old/src/Mono.WebServer.Apache/Makefile.am	2013-05-14 15:46:56.000000000 +0400
+++ xsp-3.0.11.new/src/Mono.WebServer.Apache/Makefile.am	2016-06-08 15:11:06.756628861 +0300
@@ -3,11 +3,11 @@ builddir=$(top_builddir)/src/Mono.WebSer
 MCSFLAGS= -unsafe -debug+ -debug:full -nologo -nowarn:618 $(WEBTRACING) -keyfile:$(top_srcdir)/src/mono.snk
 
 MODMONOSERVER2_EXE = mod-mono-server2.exe
-GACUTIL2=$(GACUTIL) -package 2.0
+GACUTIL2=$(GACUTIL) -package xsp2
 
 if NET_4_0
 MODMONOSERVER4_EXE = mod-mono-server4.exe
-GACUTIL4=$(GACUTIL) -package 4.5
+GACUTIL4=$(GACUTIL) -package xsp4
 endif
 
 noinst_SCRIPTS = $(MODMONOSERVER2_EXE) $(MODMONOSERVER4_EXE)
diff -uprN xsp-3.0.11.old/src/Mono.WebServer.FastCgi/Makefile.am xsp-3.0.11.new/src/Mono.WebServer.FastCgi/Makefile.am
--- xsp-3.0.11.old/src/Mono.WebServer.FastCgi/Makefile.am	2013-05-14 15:46:56.000000000 +0400
+++ xsp-3.0.11.new/src/Mono.WebServer.FastCgi/Makefile.am	2016-06-08 15:11:23.980628824 +0300
@@ -3,11 +3,11 @@ builddir=$(top_builddir)/src/Mono.WebSer
 MCSFLAGS= -debug+ -debug:full -nologo -nowarn:618 $(WEBTRACING) -unsafe+ -keyfile:$(top_srcdir)/src/mono.snk
 
 XSP2_EXE = fastcgi-mono-server2.exe
-GACUTIL2=$(GACUTIL) -package 2.0
+GACUTIL2=$(GACUTIL) -package xsp2
 
 if NET_4_0
 XSP4_EXE = fastcgi-mono-server4.exe
-GACUTIL4=$(GACUTIL) -package 4.5
+GACUTIL4=$(GACUTIL) -package xsp4
 endif
 
 noinst_SCRIPTS = $(XSP2_EXE) $(XSP4_EXE)
diff -uprN xsp-3.0.11.old/src/Mono.WebServer.XSP/Makefile.am xsp-3.0.11.new/src/Mono.WebServer.XSP/Makefile.am
--- xsp-3.0.11.old/src/Mono.WebServer.XSP/Makefile.am	2013-05-14 15:46:56.000000000 +0400
+++ xsp-3.0.11.new/src/Mono.WebServer.XSP/Makefile.am	2016-06-08 15:11:40.280628788 +0300
@@ -4,11 +4,11 @@ pkgconfigdir=$(libdir)/pkgconfig
 MCSFLAGS= -debug+ -debug:full -nologo -nowarn:618 -unsafe $(WEBTRACING) -keyfile:$(top_srcdir)/src/mono.snk
 
 XSP2_EXE = xsp2.exe
-GACUTIL2=$(GACUTIL) -package 2.0
+GACUTIL2=$(GACUTIL) -package xsp2
 
 if NET_4_0
 XSP4_EXE = xsp4.exe
-GACUTIL4=$(GACUTIL) -package 4.5
+GACUTIL4=$(GACUTIL) -package xsp4
 endif
 
 noinst_SCRIPTS = $(XSP2_EXE) $(XSP4_EXE)
diff -uprN xsp-3.0.11.old/src/Mono.WebServer.XSP/xsp-2.pc.in xsp-3.0.11.new/src/Mono.WebServer.XSP/xsp-2.pc.in
--- xsp-3.0.11.old/src/Mono.WebServer.XSP/xsp-2.pc.in	2013-05-14 15:46:56.000000000 +0400
+++ xsp-3.0.11.new/src/Mono.WebServer.XSP/xsp-2.pc.in	2016-06-08 15:07:47.954947643 +0300
@@ -1,5 +1,5 @@
 prefix=@prefix@
-assemblies_dir=${prefix}/lib/mono/2.0
+assemblies_dir=${prefix}/lib/mono/xsp2
 Libraries=${assemblies_dir}/Mono.WebServer2.dll
 Executable=${assemblies_dir}/xsp2.exe
  
diff -uprN xsp-3.0.11.old/src/Mono.WebServer.XSP/xsp-4.pc.in xsp-3.0.11.new/src/Mono.WebServer.XSP/xsp-4.pc.in
--- xsp-3.0.11.old/src/Mono.WebServer.XSP/xsp-4.pc.in	2013-05-14 15:46:56.000000000 +0400
+++ xsp-3.0.11.new/src/Mono.WebServer.XSP/xsp-4.pc.in	2016-06-08 15:24:35.764508410 +0300
@@ -1,5 +1,5 @@
 prefix=@prefix@
-assemblies_dir=${prefix}/lib/mono/4.5
+assemblies_dir=${prefix}/lib/mono/xsp4
 Libraries=${assemblies_dir}/Mono.WebServer2.dll
 Executable=${assemblies_dir}/xsp4.exe
  
