From 689b4adefaf0302a9338bb28bd3c3f0ef1c95ce7 Mon Sep 17 00:00:00 2001
From: Antony Lee <anntzer.lee@gmail.com>
Date: Thu, 6 Apr 2023 19:56:52 +0200
Subject: [PATCH] Fix tests re: stem API changes.

---
 tests/test_mplcursors.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/tests/test_mplcursors.py b/tests/test_mplcursors.py
index 7ac6659..56fe85d 100644
--- a/tests/test_mplcursors.py
+++ b/tests/test_mplcursors.py
@@ -387,8 +387,11 @@ def test_dataless_errorbar(ax):
 
 
 def test_stem(ax):
-    with pytest.warns(None):  # stem use_line_collection API change.
-        ax.stem([1, 2, 3], use_line_collection=True)
+    try:  # stem use_line_collection API change.
+        with pytest.warns(None):
+            ax.stem([1, 2, 3], use_line_collection=True)
+    except TypeError:
+        ax.stem([1, 2, 3])
     cursor = mplcursors.cursor()
     assert len(cursor.artists) == 1
     _process_event("__mouse_click__", ax, (.5, .5), 1)
