Index: SuperTuxKart-1.4-src/CMakeLists.txt
===================================================================
--- SuperTuxKart-1.4-src.orig/CMakeLists.txt
+++ SuperTuxKart-1.4-src/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.4)
+cmake_minimum_required(VERSION 3.10)
 
 # root CMakeLists for the SuperTuxKart project
 project(SuperTuxKart)
@@ -6,10 +6,6 @@ set(PROJECT_VERSION "1.4")
 
 add_definitions( -DSUPERTUXKART_VERSION="${PROJECT_VERSION}" )
 
-if(NOT (CMAKE_MAJOR_VERSION VERSION_LESS 3))
-  cmake_policy(SET CMP0043 OLD)
-endif()
-
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake")
 include(CMakeDependentOption)
 
Index: SuperTuxKart-1.4-src/lib/enet/CMakeLists.txt
===================================================================
--- SuperTuxKart-1.4-src.orig/lib/enet/CMakeLists.txt
+++ SuperTuxKart-1.4-src/lib/enet/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.10)
 
 project(enet)
 
Index: SuperTuxKart-1.4-src/lib/graphics_utils/CMakeLists.txt
===================================================================
--- SuperTuxKart-1.4-src.orig/lib/graphics_utils/CMakeLists.txt
+++ SuperTuxKart-1.4-src/lib/graphics_utils/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.10)
 include_directories("${PROJECT_SOURCE_DIR}/lib/simd_wrapper")
 if (UNIX OR MINGW)
     add_definitions(-O3)
Index: SuperTuxKart-1.4-src/lib/mcpp/CMakeLists.txt
===================================================================
--- SuperTuxKart-1.4-src.orig/lib/mcpp/CMakeLists.txt
+++ SuperTuxKart-1.4-src/lib/mcpp/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.10)
 if (UNIX OR MINGW)
     add_definitions(-O3)
     if (APPLE)
Index: SuperTuxKart-1.4-src/lib/shaderc/CMakeLists.txt
===================================================================
--- SuperTuxKart-1.4-src.orig/lib/shaderc/CMakeLists.txt
+++ SuperTuxKart-1.4-src/lib/shaderc/CMakeLists.txt
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.10)
 project(shaderc)
 enable_testing()
 
Index: SuperTuxKart-1.4-src/lib/shaderc/third_party/glslang/CMakeLists.txt
===================================================================
--- SuperTuxKart-1.4-src.orig/lib/shaderc/third_party/glslang/CMakeLists.txt
+++ SuperTuxKart-1.4-src/lib/shaderc/third_party/glslang/CMakeLists.txt
@@ -33,7 +33,7 @@
 
 # increase to 3.1 once all major distributions
 # include a version of CMake >= 3.1
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.10)
 if (POLICY CMP0048)
   cmake_policy(SET CMP0048 NEW)
 endif()
Index: SuperTuxKart-1.4-src/lib/shaderc/third_party/spirv-headers/CMakeLists.txt
===================================================================
--- SuperTuxKart-1.4-src.orig/lib/shaderc/third_party/spirv-headers/CMakeLists.txt
+++ SuperTuxKart-1.4-src/lib/shaderc/third_party/spirv-headers/CMakeLists.txt
@@ -28,7 +28,7 @@
 # The SPIR-V headers from the SPIR-V Registry
 # https://www.khronos.org/registry/spir-v/
 #
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.10)
 project(SPIRV-Headers VERSION 1.5.5)
 
 # There are two ways to use this project.
Index: SuperTuxKart-1.4-src/lib/shaderc/third_party/spirv-headers/tools/buildHeaders/CMakeLists.txt
===================================================================
--- SuperTuxKart-1.4-src.orig/lib/shaderc/third_party/spirv-headers/tools/buildHeaders/CMakeLists.txt
+++ SuperTuxKart-1.4-src/lib/shaderc/third_party/spirv-headers/tools/buildHeaders/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.10)
 
 set(CMAKE_INSTALL_PREFIX "install" CACHE STRING "prefix" FORCE)
 
Index: SuperTuxKart-1.4-src/lib/shaderc/third_party/spirv-tools/CMakeLists.txt
===================================================================
--- SuperTuxKart-1.4-src.orig/lib/shaderc/third_party/spirv-tools/CMakeLists.txt
+++ SuperTuxKart-1.4-src/lib/shaderc/third_party/spirv-tools/CMakeLists.txt
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.10)
 if (POLICY CMP0048)
   cmake_policy(SET CMP0048 NEW)
 endif()
Index: SuperTuxKart-1.4-src/lib/tinygettext/CMakeLists.txt
===================================================================
--- SuperTuxKart-1.4-src.orig/lib/tinygettext/CMakeLists.txt
+++ SuperTuxKart-1.4-src/lib/tinygettext/CMakeLists.txt
@@ -39,7 +39,7 @@ set(VERSION "0.1.0")
 
 ### CMake configuration
 
-cmake_minimum_required(VERSION 2.4)
+cmake_minimum_required(VERSION 3.10)
 if(COMMAND cmake_policy)
 	CMAKE_POLICY(SET CMP0003 NEW)
 endif(COMMAND cmake_policy)
Index: SuperTuxKart-1.4-src/lib/wiiuse/CMakeLists.txt
===================================================================
--- SuperTuxKart-1.4-src.orig/lib/wiiuse/CMakeLists.txt
+++ SuperTuxKart-1.4-src/lib/wiiuse/CMakeLists.txt
@@ -2,7 +2,7 @@
 # http://academic.cleardefinition.com/
 # Iowa State University HCI Graduate Program/VRAC
 
-cmake_minimum_required(VERSION 2.8.0)
+cmake_minimum_required(VERSION 3.10)
 
 # Added for STK, silences CMake warning
 # See https://github.com/supertuxkart/stk-code/commit/b0ff15873ee7fa8901672672b47def9039a5534b#diff-a3e272598233e89e4b577a434cc2a89d
Index: SuperTuxKart-1.4-src/lib/wiiuse/cmake/DashboardScript.cmake.in
===================================================================
--- SuperTuxKart-1.4-src.orig/lib/wiiuse/cmake/DashboardScript.cmake.in
+++ SuperTuxKart-1.4-src/lib/wiiuse/cmake/DashboardScript.cmake.in
@@ -17,7 +17,7 @@
 # http://academic.cleardefinition.com
 # Iowa State University HCI Graduate Program/VRAC
 
-cmake_minimum_required(VERSION 2.8 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
 
 set(CTEST_CMAKE_GENERATOR "@CMAKE_GENERATOR@")
 set(CTEST_PROJECT_NAME "@PROJECT_NAME@")
Index: SuperTuxKart-1.4-src/lib/wiiuse/cmake/FindOpenHaptics.cmake
===================================================================
--- SuperTuxKart-1.4-src.orig/lib/wiiuse/cmake/FindOpenHaptics.cmake
+++ SuperTuxKart-1.4-src/lib/wiiuse/cmake/FindOpenHaptics.cmake
@@ -53,7 +53,7 @@
 # (See accompanying file LICENSE_1_0.txt or copy at
 # http://www.boost.org/LICENSE_1_0.txt)
 
-cmake_minimum_required(VERSION 2.6.3)
+cmake_minimum_required(VERSION 3.10)
 
 set(OPENHAPTICS_ROOT_DIR
 	"${OPENHAPTICS_ROOT_DIR}"
Index: SuperTuxKart-1.4-src/switch/pkgbuild-scripts/devkita64.cmake
===================================================================
--- SuperTuxKart-1.4-src.orig/switch/pkgbuild-scripts/devkita64.cmake
+++ SuperTuxKart-1.4-src/switch/pkgbuild-scripts/devkita64.cmake
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 3.10)
 
 set(CMAKE_SYSTEM_NAME Generic)
 set(CMAKE_SYSTEM_VERSION 1)
Index: SuperTuxKart-1.4-src/lib/angelscript/projects/cmake/CMakeLists.txt
===================================================================
--- SuperTuxKart-1.4-src.orig/lib/angelscript/projects/cmake/CMakeLists.txt
+++ SuperTuxKart-1.4-src/lib/angelscript/projects/cmake/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.10)
 
 # STK Fix llvm mingw crashes
 if (MINGW AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_SIZEOF_VOID_P EQUAL 4)
