diff -ur distlib-0.2.8-orig/distlib/_backport/shutil.py distlib-0.2.8/distlib/_backport/shutil.py
--- distlib-0.2.8-orig/distlib/_backport/shutil.py	2019-03-08 21:07:05.882089921 +0700
+++ distlib-0.2.8/distlib/_backport/shutil.py	2019-03-08 21:07:30.806286099 +0700
@@ -16,7 +16,7 @@
 import fnmatch
 import collections
 import errno
-from . import tarfile
+import tarfile
 
 try:
     import bz2
diff -ur distlib-0.2.8-orig/distlib/compat.py distlib-0.2.8/distlib/compat.py
--- distlib-0.2.8-orig/distlib/compat.py	2019-03-08 21:07:05.882089921 +0700
+++ distlib-0.2.8/distlib/compat.py	2019-03-08 21:08:56.770962447 +0700
@@ -311,10 +311,7 @@
             return 'IronPython'
         return 'CPython'
 
-try:
-    import sysconfig
-except ImportError: # pragma: no cover
-    from ._backport import sysconfig
+import sysconfig
 
 try:
     callable = callable
diff -ur distlib-0.2.8-orig/setup.py distlib-0.2.8/setup.py
--- distlib-0.2.8-orig/setup.py	2019-03-08 21:07:05.894090015 +0700
+++ distlib-0.2.8/setup.py	2019-03-08 21:07:30.794286005 +0700
@@ -6,7 +6,6 @@
 #
 
 import distutils.core
-from distutils.sysconfig import get_python_lib
 from os.path import join, dirname, abspath
 import re
 import sys
@@ -68,7 +67,6 @@
         'distlib._backport',
     ],
     package_data={
-        'distlib._backport': ['sysconfig.cfg'],
         'distlib': ['t32.exe', 't64.exe', 'w32.exe', 'w64.exe'],
     },
     cmdclass={
diff -ur distlib-0.2.8-orig/tests/distlib_tests.py distlib-0.2.8/tests/distlib_tests.py
--- distlib-0.2.8-orig/tests/distlib_tests.py	2019-03-08 21:07:05.890089984 +0700
+++ distlib-0.2.8/tests/distlib_tests.py	2019-03-08 21:07:30.782285911 +0700
@@ -10,7 +10,6 @@
 from test_database import (DataFilesTestCase, TestDatabase, TestDistribution,
                            TestEggInfoDistribution, DepGraphTestCase)
 from test_index import PackageIndexTestCase
-from test_locators import LocatorTestCase
 from test_manifest import ManifestTestCase
 from test_markers import MarkersTestCase
 from test_metadata import MetadataTestCase, LegacyMetadataTestCase
@@ -23,6 +22,5 @@
 
 if _ver == (2, 6):
     from test_shutil import TestCopyFile, TestMove, TestShutil
-    from test_sysconfig import TestSysConfig, MakefileTests
 from test_util import (UtilTestCase, ProgressTestCase, FileOpsTestCase,
                        GlobTestCase)
diff -ur distlib-0.2.8-orig/tests/test_shutil.py distlib-0.2.8/tests/test_shutil.py
--- distlib-0.2.8-orig/tests/test_shutil.py	2019-03-08 21:07:05.890089984 +0700
+++ distlib-0.2.8/tests/test_shutil.py	2019-03-08 21:07:30.818286194 +0700
@@ -20,7 +20,7 @@
 from compat import unittest
 
 
-from distlib._backport import shutil, tarfile
+from distlib._backport import shutil; import tarfile
 from distlib._backport.shutil import (
     _make_tarball, _make_zipfile, make_archive, unpack_archive,
     register_archive_format, unregister_archive_format, get_archive_formats,
