Index: asynctest-0.12.2/test/test_selector.py
===================================================================
--- asynctest-0.12.2.orig/test/test_selector.py
+++ asynctest-0.12.2/test/test_selector.py
@@ -286,6 +286,7 @@ class Test_fail_on_active_selector_callb
 
         TestCase().debug()
 
+    @unittest.skip('needs root perms')
     def test_events_watched_outside_test_are_ignored(self):
         loop = asyncio.new_event_loop()
         asyncio.set_event_loop(loop)
