--- mv-ddr-marvell-armada-atf-master/Makefile.orig	2022-05-19 13:46:15.391043106 +0200
+++ mv-ddr-marvell-armada-atf-master/Makefile	2022-05-19 14:19:44.503213296 +0200
@@ -328,7 +328,7 @@ ECHO     = @echo
 # OBJ_DIR set in ble/ble.mk
 OBJ_DIR ?= $(MV_DDR_ROOT)
 
-CFLAGS = -DMV_DDR_ATF -DCONFIG_DDR4
+CFLAGS = -DMV_DDR_ATF -DCONFIG_DDR4 --param=min-pagesize=0
 CFLAGS += -Wall -Werror -Os -ffreestanding -mlittle-endian -g -gdwarf-2 -nostdinc
 CFLAGS += -march=armv8-a -fpie
 
