From f1b2adb5b76c2ba80ddfb84a9929ef7bbb9768ef Mon Sep 17 00:00:00 2001
From: David Teigland <teigland@redhat.com>
Date: Fri, 17 Feb 2023 10:07:12 -0600
Subject: [PATCH 05/24] libdlm_lt: fix pc file

---
 libdlm/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libdlm/Makefile b/libdlm/Makefile
index 5069ccf1f7f9..918fb46eeb0c 100644
--- a/libdlm/Makefile
+++ b/libdlm/Makefile
@@ -108,7 +108,7 @@ $(LIB_PC): $(LIB_PCIN)
 	cat $(LIB_PCIN) | sed -e 's#@PREFIX@#$(PREFIX)#g;s#@LIBDIR@#$(LIBDIR)#g' > $@
 
 $(LLT_PC): $(LLT_PCIN)
-	cat $(LIB_PCIN) | sed -e 's#@PREFIX@#$(PREFIX)#g;s#@LIBDIR@#$(LIBDIR)#g' > $@
+	cat $(LLT_PCIN) | sed -e 's#@PREFIX@#$(PREFIX)#g;s#@LIBDIR@#$(LIBDIR)#g' > $@
 
 clean:
 	rm -f *.o *.so *.so.* *.a *.pc
-- 
2.35.3

