Index: lrzsz-0.12.21rc/configure.in
===================================================================
--- lrzsz-0.12.21rc.orig/configure.in
+++ lrzsz-0.12.21rc/configure.in
@@ -1,7 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(src/crctab.c)
 AM_INIT_AUTOMAKE(lrzsz, 0.12.21rc)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
 
 dnl AC_C_CROSS is in AC_PROG_CC since 2.12
