--- fop-2.5/fop-core/src/main/java/org/apache/fop/cli/Main.java
+++ fop-2.5/fop-core/src/main/java/org/apache/fop/cli/Main.java
@@ -210,11 +210,7 @@
      * @param args the command line parameters
      */
     public static void main(String[] args) {
-        if (checkDependencies()) {
-            startFOP(args);
-        } else {
-            startFOPWithDynamicClasspath(args);
-        }
+        startFOP(args);
     }
 
 }
