Index: MotionBox-1.5.0/Sky-1.5.0/src/SkBackend/SkBackend.pro
===================================================================
--- MotionBox-1.5.0.orig/Sky-1.5.0/src/SkBackend/SkBackend.pro
+++ MotionBox-1.5.0/Sky-1.5.0/src/SkBackend/SkBackend.pro
@@ -22,7 +22,7 @@ contains(QT_MAJOR_VERSION, 4) {
     QT += quick network xml
 }
 
-CONFIG       += plugin
+CONFIG       += unversioned_libname unversioned_soname
 win32:CONFIG += dll
 
 DEFINES += SK_BACKEND_LIBRARY
Index: MotionBox-1.5.0/Sky-1.5.0/src/SkCore/SkCore.pro
===================================================================
--- MotionBox-1.5.0.orig/Sky-1.5.0/src/SkCore/SkCore.pro
+++ MotionBox-1.5.0/Sky-1.5.0/src/SkCore/SkCore.pro
@@ -22,7 +22,7 @@ contains(QT_MAJOR_VERSION, 4) {
     QT += quick widgets network xml xmlpatterns
 }
 
-CONFIG       += plugin
+CONFIG       += unversioned_libname unversioned_soname
 win32:CONFIG += dll
 
 DEFINES += SK_CORE_LIBRARY
Index: MotionBox-1.5.0/Sky-1.5.0/src/SkGui/SkGui.pro
===================================================================
--- MotionBox-1.5.0.orig/Sky-1.5.0/src/SkGui/SkGui.pro
+++ MotionBox-1.5.0/Sky-1.5.0/src/SkGui/SkGui.pro
@@ -28,7 +28,7 @@ contains(QT_MAJOR_VERSION, 5) {
     unix:QT += x11extras
 }
 
-CONFIG       += plugin
+CONFIG       += unversioned_libname unversioned_soname
 win32:CONFIG += dll
 
 DEFINES += QT_QTLOCKEDFILE_IMPORT
Index: MotionBox-1.5.0/Sky-1.5.0/src/SkMedia/SkMedia.pro
===================================================================
--- MotionBox-1.5.0.orig/Sky-1.5.0/src/SkMedia/SkMedia.pro
+++ MotionBox-1.5.0/Sky-1.5.0/src/SkMedia/SkMedia.pro
@@ -22,7 +22,7 @@ contains(QT_MAJOR_VERSION, 4) {
     QT += opengl quick network
 }
 
-CONFIG       += plugin
+CONFIG       += unversioned_libname unversioned_soname
 win32:CONFIG += dll
 
 DEFINES += SK_MEDIA_LIBRARY CAN_COMPILE_SSE2
Index: MotionBox-1.5.0/Sky-1.5.0/src/SkTorrent/SkTorrent.pro
===================================================================
--- MotionBox-1.5.0.orig/Sky-1.5.0/src/SkTorrent/SkTorrent.pro
+++ MotionBox-1.5.0/Sky-1.5.0/src/SkTorrent/SkTorrent.pro
@@ -22,7 +22,7 @@ contains(QT_MAJOR_VERSION, 4) {
     QT += quick network
 }
 
-CONFIG       += plugin
+CONFIG       += unversioned_libname unversioned_soname
 win32:CONFIG += dll
 
 DEFINES += SK_TORRENT_LIBRARY
Index: MotionBox-1.5.0/Sky-1.5.0/src/SkWeb/SkWeb.pro
===================================================================
--- MotionBox-1.5.0.orig/Sky-1.5.0/src/SkWeb/SkWeb.pro
+++ MotionBox-1.5.0/Sky-1.5.0/src/SkWeb/SkWeb.pro
@@ -18,7 +18,7 @@ TEMPLATE = lib
 
 QT += network webkit webkitwidgets
 
-CONFIG       += plugin
+CONFIG       += unversioned_libname unversioned_soname
 win32:CONFIG += dll
 
 DEFINES += SK_WEB_LIBRARY
