diff --git a/autogen.sh b/autogen.sh
index 6d8c4969..da2f79f0 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -48,4 +48,4 @@ which acinclude >/dev/null 2>&1 && acinclude
 which aclocal >/dev/null 2>&1 && aclocal
 autoconf
 
-./configure ${CFGARGS}  $*
+test -n "$NOCONFIGURE" || ./configure ${CFGARGS}  $*
diff --git a/runqmake.sh b/runqmake.sh
index 7db4571b..8fb8958e 100755
--- a/runqmake.sh
+++ b/runqmake.sh
@@ -13,5 +13,10 @@ IFS="
 
 test -z "$EXTRA_CXXFLAGS" || FLAGS="QMAKE_CXXFLAGS += $EXTRA_CXXFLAGS"
 
-$QMAKE "$FLAGS" $C
+if [ -z "$FLAGS" ]; then
+  # in case there are no FLAGS defined, we can't pass an empty "" parameter to qmake
+  $QMAKE $C
+else
+  $QMAKE "$FLAGS" $C
+fi
 
