Index: xtl-0.7.2/CMakeLists.txt
===================================================================
--- xtl-0.7.2.orig/CMakeLists.txt
+++ xtl-0.7.2/CMakeLists.txt
@@ -122,9 +122,9 @@ install(FILES ${XTL_HEADERS}
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/xtl)
 endif()
 
-set(XTL_CMAKECONFIG_INSTALL_DIR "${CMAKE_INSTALL_DATAROOTDIR}/cmake/${PROJECT_NAME}" CACHE
-    STRING "install path for xtlConfig.cmake")
 
+set(XTL_CMAKECONFIG_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}" CACHE STRING "install path for xtlConfig.cmake")
+  
 configure_package_config_file(${PROJECT_NAME}Config.cmake.in
                               "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
                               INSTALL_DESTINATION ${XTL_CMAKECONFIG_INSTALL_DIR})
@@ -157,4 +157,4 @@ configure_file(${PROJECT_NAME}.pc.in
                "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc"
                 @ONLY)
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc"
-        DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig/")
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig/")
