--- a/base/time/time.cc	2022-06-21 21:52:33.367987695 +0200
+++ b/base/time/time.cc	2022-06-30 13:15:58.619513336 +0200
@@ -21,7 +21,7 @@
 
 #include "base/check.h"
 #include "base/strings/stringprintf.h"
-#include "base/third_party/nspr/prtime.h"
+#include <prtime.h>
 #include "base/time/time_override.h"
 #include "build/build_config.h"
 #include "third_party/abseil-cpp/absl/types/optional.h"
--- a/base/BUILD.gn	2022-06-30 13:13:20.315537859 +0200
+++ b/base/BUILD.gn	2022-06-30 13:25:39.947410797 +0200
@@ -1,3 +1,8 @@
+import("//build/config/linux/pkg_config.gni")
+
+pkg_config("system_nspr") {
+  packages = [ "nspr" ]
+}
 # Copyright 2013 The Chromium Authors
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
@@ -826,8 +831,6 @@
     "third_party/cityhash_v103/src/city_v103.cc",
     "third_party/cityhash_v103/src/city_v103.h",
     "third_party/icu/icu_utf.h",
-    "third_party/nspr/prtime.cc",
-    "third_party/nspr/prtime.h",
     "third_party/superfasthash/superfasthash.c",
     "thread_annotations.h",
     "threading/hang_watcher.cc",
@@ -1414,6 +1417,7 @@
     "//build/config/compiler:prevent_unsafe_narrowing",
     "//build/config/compiler:wexit_time_destructors",
     "//build/config/compiler:wglobal_constructors",
+    ":system_nspr",
   ]
 
   deps = [
