--- src/content/public/browser/content_browser_client.cc.orig	2025-04-16 14:45:09.657704679 +0200
+++ src/content/public/browser/content_browser_client.cc	2025-04-17 16:18:19.395831947 +0200
@@ -53,6 +53,7 @@
 #include "content/public/browser/vpn_service_proxy.h"
 #include "content/public/browser/web_contents.h"
 #include "content/public/browser/web_contents_view_delegate.h"
+#include "content/public/browser/web_ui_controller.h"
 #include "content/public/common/alternative_error_page_override_info.mojom.h"
 #include "content/public/common/content_features.h"
 #include "content/public/common/url_utils.h"
