--- src/media/ffmpeg/ffmpeg_common.cc.orig	2025-04-11 12:05:21.289638236 +0200
+++ src/media/ffmpeg/ffmpeg_common.cc	2025-04-12 12:23:36.647784415 +0200
@@ -76,7 +76,7 @@ const char* GetAllowedVideoDecoders() {
 void ApplyCodecContextSecuritySettings(AVCodecContext* codec_context) {
   // Future versions of ffmpeg may copy the allow list from the format
   // context.
-  if (!codec_context->codec_whitelist) {
+  if (false) {
     // Note: FFmpeg will try to free this string, so we must duplicate it.
     codec_context->codec_whitelist =
         av_strdup(codec_context->codec_type == AVMEDIA_TYPE_AUDIO
