Index: electron-17.1.2/third_party/breakpad/breakpad/src/client/linux/handler/exception_handler.cc
===================================================================
--- electron-17.1.2.orig/third_party/breakpad/breakpad/src/client/linux/handler/exception_handler.cc	2022-03-11 11:17:39.716354439 +0100
+++ electron-17.1.2/third_party/breakpad/breakpad/src/client/linux/handler/exception_handler.cc	2022-03-15 07:51:37.994251537 +0100
@@ -138,7 +138,7 @@ void InstallAlternateStackLocked() {
   // SIGSTKSZ may be too small to prevent the signal handlers from overrunning
   // the alternative stack. Ensure that the size of the alternative stack is
   // large enough.
-  static const unsigned kSigStackSize = std::max(16384, SIGSTKSZ);
+  static const unsigned kSigStackSize = std::max(16384L, (long)SIGSTKSZ);
 
   // Only set an alternative stack if there isn't already one, or if the current
   // one is too small.
