--- src/chrome/test/BUILD.gn.orig	2024-09-12 13:54:44.642624995 +0200
+++ src/chrome/test/BUILD.gn	2024-09-12 15:25:29.792388388 +0200
@@ -56,7 +56,6 @@ import("//v8/gni/v8.gni")
 
 assert(!is_ios, "Chromium/iOS shouldn't use anything in //chrome")
 assert(!is_fuchsia, "Fuchsia shouldn't use anything in //chrome")
-assert(enable_supervised_users)
 
 if (is_android) {
   import("//build/config/android/rules.gni")
--- src/chrome/common/BUILD.gn.orig	2024-09-12 13:46:51.785984196 +0200
+++ src/chrome/common/BUILD.gn	2024-09-12 15:27:58.398867501 +0200
@@ -22,7 +22,6 @@ import("//testing/libfuzzer/fuzzer_test.
 import("//third_party/widevine/cdm/widevine.gni")
 import("//tools/grit/grit_rule.gni")
 
-assert(enable_supervised_users)
 assert(!is_fuchsia, "Fuchsia shouldn't use anything in //chrome")
 
 fuzzer_test("ini_parser_fuzzer") {
--- src/chrome/renderer/BUILD.gn.orig	2024-09-12 13:46:51.849317526 +0200
+++ src/chrome/renderer/BUILD.gn	2024-09-12 15:29:17.915843739 +0200
@@ -18,7 +18,6 @@ import("//testing/libfuzzer/fuzzer_test.
 import("//third_party/widevine/cdm/widevine.gni")
 import("//tools/grit/grit_rule.gni")
 
-assert(enable_supervised_users)
 
 grit("resources") {
   source = "resources/renderer_resources.grd"
--- src/chrome/browser/BUILD.gn.orig	2024-09-12 13:54:44.639291662 +0200
+++ src/chrome/browser/BUILD.gn	2024-09-12 15:30:42.759505691 +0200
@@ -45,7 +45,6 @@ import("//third_party/widevine/cdm/widev
 import("//tools/grit/grit_rule.gni")
 import("//ui/base/ui_features.gni")
 
-assert(enable_supervised_users)
 
 # //build/config/android/rules.gni imports //tools/grit/grit_rule.gni, which
 # produces a conflict for the "grit" template so we have to only include one.
--- src/chrome/browser/ui/BUILD.gn.orig	2024-09-12 13:46:51.212650894 +0200
+++ src/chrome/browser/ui/BUILD.gn	2024-09-12 15:32:36.489733448 +0200
@@ -32,7 +32,6 @@ import("//third_party/protobuf/proto_lib
 import("//ui/base/ui_features.gni")
 import("//ui/views/features.gni")
 
-assert(enable_supervised_users)
 assert(!is_fuchsia, "Fuchsia shouldn't use anything in //chrome")
 
 generate_histograms_variants_allowlist("webui_name_variants") {
--- src/chrome/browser/extensions/BUILD.gn.orig	2024-09-12 13:46:50.369317607 +0200
+++ src/chrome/browser/extensions/BUILD.gn	2024-09-12 15:33:39.046452725 +0200
@@ -20,7 +20,6 @@ import("//testing/libfuzzer/fuzzer_test.
 import("//third_party/protobuf/proto_library.gni")
 
 assert(enable_extensions)
-assert(enable_supervised_users)
 assert(!is_fuchsia, "Fuchsia shouldn't use anything in //chrome")
 
 static_library("extensions") {
