--- src/third_party/webrtc/modules/video_coding/codecs/av1/libaom_av1_encoder.cc.orig	2024-05-29 12:13:50.205359130 +0200
+++ src/third_party/webrtc/modules/video_coding/codecs/av1/libaom_av1_encoder.cc	2024-05-29 23:55:59.239008160 +0200
@@ -311,8 +311,10 @@ int LibaomAv1Encoder::InitEncode(const V
 
   if (codec_settings->mode == VideoCodecMode::kRealtimeVideo &&
       encoder_settings_.GetFrameDropEnabled() && max_consec_frame_drop_ > 0) {
+#ifdef AOM_CTRL_AV1E_SET_MAX_CONSEC_FRAME_DROP_CBR
     SET_ENCODER_PARAM_OR_RETURN_ERROR(AV1E_SET_MAX_CONSEC_FRAME_DROP_CBR,
                                       max_consec_frame_drop_);
+#endif
   }
 
   if (cfg_.g_threads == 8) {
