--- src/components/browsing_data/core/counters/passwords_counter.cc.old	2022-10-20 19:00:23.009865700 +0200
+++ src/components/browsing_data/core/counters/passwords_counter.cc	2022-10-29 22:29:38.660589600 +0200
@@ -34,6 +34,8 @@
   }
 }
 
+}  // namespace
+
 // PasswordStoreFetcher ----------------------------------
 
 // Fetches passswords and observes a PasswordStoreInterface.
@@ -168,8 +170,6 @@
   weak_ptr_factory_.InvalidateWeakPtrs();
 }
 
-}  // namespace
-
 // PasswordsCounter::PasswordsResult ----------------------------------
 PasswordsCounter::PasswordsResult::PasswordsResult(
     const BrowsingDataCounter* source,
--- src/components/browsing_data/core/counters/passwords_counter.h.old	2022-10-20 19:00:23.009865700 +0200
+++ src/components/browsing_data/core/counters/passwords_counter.h	2022-10-29 22:28:42.288402800 +0200
@@ -18,9 +18,7 @@
 }
 
 namespace browsing_data {
-namespace {
 class PasswordStoreFetcher;
-}
 class PasswordsCounter : public browsing_data::BrowsingDataCounter {
  public:
   // A subclass of SyncResult that stores the result value, a boolean
