--- src/content/public/browser/preview_cancel_reason.h.orig	2024-05-30 22:55:47.180621500 +0000
+++ src/content/public/browser/preview_cancel_reason.h	2024-05-31 07:46:06.285116500 +0000
@@ -5,6 +5,8 @@
 #ifndef CONTENT_PUBLIC_BROWSER_PREVIEW_CANCEL_REASON_H_
 #define CONTENT_PUBLIC_BROWSER_PREVIEW_CANCEL_REASON_H_
 
+#include <string>
+
 #include "content/common/content_export.h"
 #include "third_party/abseil-cpp/absl/types/variant.h"
 
