--- src/third_party/blink/renderer/modules/ai/BUILD.gn.orig	2025-08-15 17:32:29.337405097 +0200
+++ src/third_party/blink/renderer/modules/ai/BUILD.gn	2025-08-16 17:57:26.885671869 +0200
@@ -7,45 +7,19 @@ import("//third_party/blink/renderer/mod
 blink_modules_sources("ai") {
   sources = [
     "ai_context_observer.h",
-    "ai_interface_proxy.cc",
-    "ai_interface_proxy.h",
     "ai_metrics.cc",
     "ai_metrics.h",
-    "ai_utils.cc",
-    "ai_utils.h",
     "ai_writing_assistance_base.h",
     "ai_writing_assistance_create_client.h",
     "availability.cc",
     "availability.h",
-    "create_monitor.cc",
-    "create_monitor.h",
     "exception_helpers.cc",
     "exception_helpers.h",
-    "language_model.cc",
-    "language_model.h",
-    "language_model_create_client.cc",
-    "language_model_create_client.h",
     "language_model_params.cc",
     "language_model_params.h",
-    "language_model_prompt_builder.cc",
-    "language_model_prompt_builder.h",
     "model_execution_responder.cc",
     "model_execution_responder.h",
-    "on_device_translation/create_translator_client.cc",
-    "on_device_translation/create_translator_client.h",
-    "on_device_translation/language_detector.cc",
-    "on_device_translation/language_detector.h",
     "on_device_translation/resolver_with_abort_signal.h",
-    "on_device_translation/translator.cc",
-    "on_device_translation/translator.h",
-    "proofreader.cc",
-    "proofreader.h",
-    "rewriter.cc",
-    "rewriter.h",
-    "summarizer.cc",
-    "summarizer.h",
-    "writer.cc",
-    "writer.h",
   ]
 
   public_deps = [ "//third_party/blink/public/mojom:mojom_modules_blink" ]
@@ -53,8 +27,6 @@ blink_modules_sources("ai") {
   deps = [
     "//components/language_detection/content/common",
     "//components/language_detection/content/common:common_blink",
-    "//components/language_detection/content/renderer",
-    "//components/language_detection/core",
     "//third_party/blink/renderer/modules/canvas",
     "//third_party/blink/renderer/modules/webaudio",
   ]
--- src/third_party/blink/renderer/platform/BUILD.gn.orig	2025-04-12 14:33:45.435739667 +0200
+++ src/third_party/blink/renderer/platform/BUILD.gn	2025-04-12 15:58:59.809440918 +0200
@@ -1239,8 +1239,6 @@ component("platform") {
     "keyboard_codes.h",
     "language.cc",
     "language.h",
-    "language_detection/language_detection_model.cc",
-    "language_detection/language_detection_model.h",
     "link_hash.cc",
     "link_hash.h",
     "media/media_player_client.h",
@@ -1757,11 +1755,8 @@ component("platform") {
     "//cc/mojo_embedder",
     "//components/language_detection/content/common",
     "//components/language_detection/content/common:common_blink",
-    "//components/language_detection/content/renderer",
-    "//components/language_detection/core",
     "//components/paint_preview/common",
     "//components/search_engines:search_engine_utils",
-    "//components/translate/core/language_detection:language_detection",
     "//components/viz/client",
     "//components/viz/common",
     "//components/webrtc:net_address_utils",
--- src/components/translate/core/common/BUILD.gn.orig	2025-04-12 14:33:35.535738956 +0200
+++ src/components/translate/core/common/BUILD.gn	2025-04-12 16:04:56.897440792 +0200
@@ -23,7 +23,6 @@ static_library("common") {
   deps = [
     "//base",
     "//components/language/core/common",
-    "//components/language_detection/core:core",
     "//third_party/metrics_proto",
     "//url",
   ]
--- src/third_party/blink/renderer/bindings/idl_in_modules.gni.orig	2025-08-15 17:32:29.049405150 +0200
+++ src/third_party/blink/renderer/bindings/idl_in_modules.gni	2025-08-16 14:55:26.743869183 +0200
@@ -19,15 +19,8 @@ static_idl_files_in_modules = [
   "//third_party/blink/renderer/modules/ad_auction/storage_interest_group.idl",
   "//third_party/blink/renderer/modules/ai/availability.idl",
   "//third_party/blink/renderer/modules/ai/create_monitor.idl",
-  "//third_party/blink/renderer/modules/ai/language_model.idl",
   "//third_party/blink/renderer/modules/ai/language_model_create_options.idl",
   "//third_party/blink/renderer/modules/ai/language_model_params.idl",
-  "//third_party/blink/renderer/modules/ai/on_device_translation/language_detector.idl",
-  "//third_party/blink/renderer/modules/ai/on_device_translation/translator.idl",
-  "//third_party/blink/renderer/modules/ai/proofreader.idl",
-  "//third_party/blink/renderer/modules/ai/rewriter.idl",
-  "//third_party/blink/renderer/modules/ai/summarizer.idl",
-  "//third_party/blink/renderer/modules/ai/writer.idl",
   "//third_party/blink/renderer/modules/animationworklet/animation_worklet_global_scope.idl",
   "//third_party/blink/renderer/modules/animationworklet/css_animation_worklet.idl",
   "//third_party/blink/renderer/modules/animationworklet/worklet_animation.idl",
--- src/third_party/blink/renderer/bindings/generated_in_modules.gni.orig	2025-08-15 17:32:29.049405150 +0200
+++ src/third_party/blink/renderer/bindings/generated_in_modules.gni	2025-08-16 15:26:16.723512975 +0200
@@ -682,18 +682,6 @@ generated_dictionary_sources_in_modules
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_key_system_track_configuration.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_landmark.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_landmark.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_detection_result.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_detection_result.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_detector_create_core_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_detector_create_core_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_detector_create_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_detector_create_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_detector_detect_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_detector_detect_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_append_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_append_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_clone_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_clone_options.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_create_core_options.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_create_core_options.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_create_options.cc",
@@ -704,8 +692,6 @@ generated_dictionary_sources_in_modules
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_message.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_message_content.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_message_content.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_prompt_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_prompt_options.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_lock_info.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_lock_info.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_lock_manager_snapshot.cc",
@@ -1008,14 +994,6 @@ generated_dictionary_sources_in_modules
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_private_attribution_network.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_private_attribution_options.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_private_attribution_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_proofreader_create_core_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_proofreader_create_core_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_proofreader_create_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_proofreader_create_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_proofread_correction.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_proofread_correction.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_proofread_result.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_proofread_result.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_protected_audience_private_aggregation_config.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_protected_audience_private_aggregation_config.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_public_key_credential_creation_options.cc",
@@ -1064,12 +1042,6 @@ generated_dictionary_sources_in_modules
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_remote_desktop_client_override.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_request_device_options.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_request_device_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter_rewrite_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter_rewrite_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter_create_core_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter_create_core_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter_create_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter_create_options.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_router_condition.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_router_condition.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_router_rule.cc",
@@ -1278,12 +1250,6 @@ generated_dictionary_sources_in_modules
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_sub_apps_list_result.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_subsample_entry.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_subsample_entry.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer_create_core_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer_create_core_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer_create_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer_create_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer_summarize_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer_summarize_options.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_svc_output_metadata.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_svc_output_metadata.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_sync_event_init.cc",
@@ -1296,12 +1262,6 @@ generated_dictionary_sources_in_modules
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_text_encoder_encode_into_result.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_top_level_storage_access_permission_descriptor.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_top_level_storage_access_permission_descriptor.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_translator_create_core_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_translator_create_core_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_translator_create_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_translator_create_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_translator_translate_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_translator_translate_options.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_usb_connection_event_init.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_usb_connection_event_init.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_usb_control_transfer_parameters.cc",
@@ -1380,12 +1340,6 @@ generated_dictionary_sources_in_modules
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_websocket_stream_options.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_write_params.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_write_params.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer_write_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer_write_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer_create_core_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer_create_core_options.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer_create_options.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer_create_options.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_xr_depth_state_init.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_xr_depth_state_init.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_xr_dom_overlay_init.cc",
@@ -1772,8 +1726,6 @@ generated_enumeration_sources_in_modules
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_presentation_connection_state.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_private_attribution_event.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_private_attribution_event.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_correction_type.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_correction_type.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_push_encryption_key_name.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_push_encryption_key_name.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_recording_state.cc",
@@ -1782,12 +1734,6 @@ generated_enumeration_sources_in_modules
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_red_eye_reduction.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_remote_playback_state.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_remote_playback_state.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter_format.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter_format.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter_length.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter_length.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter_tone.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter_tone.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_router_source_enum.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_router_source_enum.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rtc_bundle_policy.cc",
@@ -1868,12 +1814,6 @@ generated_enumeration_sources_in_modules
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_storage_bucket_durability.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_sub_apps_result_code.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_sub_apps_result_code.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer_format.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer_format.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer_length.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer_length.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer_type.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer_type.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_display_media_include_or_exclude.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_display_media_include_or_exclude.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_track_default_type.cc",
@@ -1912,12 +1852,6 @@ generated_enumeration_sources_in_modules
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_well_known_directory.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_write_command_type.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_write_command_type.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer_format.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer_format.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer_length.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer_length.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer_tone.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer_tone.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_xr_depth_data_format.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_xr_depth_data_format.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_xr_depth_type.cc",
@@ -2464,12 +2398,8 @@ generated_interface_sources_in_modules =
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_keyboard_layout_map.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_khr_parallel_shader_compile.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_khr_parallel_shader_compile.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_detector.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_detector.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_params.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model_params.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_language_model.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_launch_params.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_launch_params.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_launch_queue.cc",
@@ -2708,8 +2638,6 @@ generated_interface_sources_in_modules =
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_private_aggregation.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_private_attribution.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_private_attribution.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_proofreader.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_proofreader.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_protected_audience.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_protected_audience.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_public_key_credential.cc",
@@ -2732,8 +2660,6 @@ generated_interface_sources_in_modules =
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_remote_playback.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_restriction_target.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_restriction_target.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rewriter.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rtc_certificate.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rtc_certificate.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rtc_data_channel.cc",
@@ -2902,8 +2828,6 @@ generated_interface_sources_in_modules =
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_sub_apps.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_subtle_crypto.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_subtle_crypto.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_summarizer.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_svg_element.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_svg_element.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_sync_event.cc",
@@ -2926,8 +2850,6 @@ generated_interface_sources_in_modules =
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_track_default.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_track_default_list.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_track_default_list.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_translator.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_translator.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_url.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_url.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_usb.cc",
@@ -3100,8 +3022,6 @@ generated_interface_sources_in_modules =
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_worklet_animation_effect.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_worklet_group_effect.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_worklet_group_effect.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_writer.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_xr_anchor.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_xr_anchor.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_xr_anchor_set.cc",
--- src/content/public/browser/content_browser_client.cc	2025-04-17 16:18:19.395831947 +0200
+++ src/content/public/browser/content_browser_client.cc	2025-04-18 22:48:04.297289260 +0200
@@ -1882,15 +1882,6 @@ GetContentLanguageDetectionDriver() {
 }
 }  // namespace
 
-void ContentBrowserClient::BindLanguageDetectionDriver(
-    content::BrowserContext* browser_context,
-    base::SupportsUserData* context_user_data,
-    mojo::PendingReceiver<
-        language_detection::mojom::ContentLanguageDetectionDriver> receiver) {
-  if (base::FeatureList::IsEnabled(blink::features::kLanguageDetectionAPI)) {
-    GetContentLanguageDetectionDriver().AddReceiver(std::move(receiver));
-  }
-}
 
 #if !BUILDFLAG(IS_ANDROID)
 void ContentBrowserClient::QueryInstalledWebAppsByManifestId(
--- src/content/browser/browser_interface_binders.cc.orig	2025-07-20 14:51:05.789212478 +0200
+++ src/content/browser/browser_interface_binders.cc	2025-08-15 20:15:02.320284607 +0200
@@ -1161,19 +1161,6 @@ void PopulateFrameBinders(RenderFrameHos
         base::Unretained(host)));
   }
 
-  if (base::FeatureList::IsEnabled(blink::features::kLanguageDetectionAPI)) {
-    map->Add<language_detection::mojom::ContentLanguageDetectionDriver>(
-        base::BindRepeating(
-            [](RenderFrameHostImpl* host,
-               mojo::PendingReceiver<
-                   language_detection::mojom::ContentLanguageDetectionDriver>
-                   receiver) {
-              GetContentClient()->browser()->BindLanguageDetectionDriver(
-                  host->GetBrowserContext(), &host->document_associated_data(),
-                  std::move(receiver));
-            },
-            base::Unretained(host)));
-  }
 }
 
 void PopulateBinderMapWithContext(
@@ -1465,19 +1452,6 @@ void PopulateDedicatedWorkerBinders(Dedi
         },
         base::Unretained(host)));
   }
-  if (base::FeatureList::IsEnabled(blink::features::kLanguageDetectionAPI)) {
-    map->Add<language_detection::mojom::ContentLanguageDetectionDriver>(
-        base::BindRepeating(
-            [](DedicatedWorkerHost* host,
-               mojo::PendingReceiver<
-                   language_detection::mojom::ContentLanguageDetectionDriver>
-                   receiver) {
-              GetContentClient()->browser()->BindLanguageDetectionDriver(
-                  host->GetProcessHost()->GetBrowserContext(), host,
-                  std::move(receiver));
-            },
-            base::Unretained(host)));
-  }
 }
 
 void PopulateBinderMapWithContext(
@@ -1575,19 +1549,6 @@ void PopulateSharedWorkerBinders(SharedW
         },
         base::Unretained(host)));
   }
-  if (base::FeatureList::IsEnabled(blink::features::kLanguageDetectionAPI)) {
-    map->Add<language_detection::mojom::ContentLanguageDetectionDriver>(
-        base::BindRepeating(
-            [](SharedWorkerHost* host,
-               mojo::PendingReceiver<
-                   language_detection::mojom::ContentLanguageDetectionDriver>
-                   receiver) {
-              GetContentClient()->browser()->BindLanguageDetectionDriver(
-                  host->GetProcessHost()->GetBrowserContext(), host,
-                  std::move(receiver));
-            },
-            base::Unretained(host)));
-  }
 
 #if !BUILDFLAG(IS_ANDROID)
   map->Add<blink::mojom::DirectSocketsService>(base::BindRepeating(
@@ -1753,22 +1714,6 @@ void PopulateServiceWorkerBinders(Servic
         },
         base::Unretained(host)));
   }
-  if (base::FeatureList::IsEnabled(blink::features::kLanguageDetectionAPI)) {
-    map->Add<language_detection::mojom::ContentLanguageDetectionDriver>(
-        base::BindRepeating(
-            [](ServiceWorkerHost* host,
-               mojo::PendingReceiver<
-                   language_detection::mojom::ContentLanguageDetectionDriver>
-                   receiver) {
-              if (auto* process_host = static_cast<RenderProcessHostImpl*>(
-                      RenderProcessHost::FromID(host->worker_process_id()))) {
-                GetContentClient()->browser()->BindLanguageDetectionDriver(
-                    process_host->GetBrowserContext(), host,
-                    std::move(receiver));
-              }
-            },
-            base::Unretained(host)));
-  }
 
   // RenderProcessHost binders
   map->Add<media::mojom::VideoDecodePerfHistory>(BindServiceWorkerReceiver(
--- src/content/public/browser/content_browser_client.h	2025-04-16 14:45:28.636752215 +0200
+++ src/content/public/browser/content_browser_client.h	2025-04-18 23:30:15.399851141 +0200
@@ -3144,11 +3144,6 @@ class CONTENT_EXPORT ContentBrowserClien
   // Binds to a singleton new instance of
   // `language_detection::ContentLanguageDetectionDriver` which receives the
   // model from a local file specified by a flag param..
-  virtual void BindLanguageDetectionDriver(
-      BrowserContext* browser_context,
-      base::SupportsUserData* context_user_data,
-      mojo::PendingReceiver<
-          language_detection::mojom::ContentLanguageDetectionDriver> receiver);
 
 #if !BUILDFLAG(IS_ANDROID)
   // Given the last committed URL of the RenderFrameHost, |frame_url|, and the
