--- src/third_party/blink/renderer/platform/loader/fetch/resource_response.h.orig	2025-04-16 14:36:33.089070014 +0200
+++ src/third_party/blink/renderer/platform/loader/fetch/resource_response.h	2025-04-16 21:43:33.972794371 +0200
@@ -174,7 +174,7 @@ class PLATFORM_EXPORT ResourceResponse f
   std::optional<base::Time> LastModified(UseCounter&) const;
   // Will always return values >= 0.
   base::TimeDelta CacheControlStaleWhileRevalidate() const;
-  std::optional<UnencodedDigest> UnencodedDigest() const;
+  std::optional<blink::UnencodedDigest> UnencodedDigest() const;
 
   unsigned ConnectionID() const;
   void SetConnectionID(unsigned);
