https://github.com/llvm/llvm-project/pull/74261

--- src/third_party/swiftshader/src/Reactor/LLVMJIT.cpp.orig	2024-03-13 17:28:56.716633400 +0100
+++ src/third_party/swiftshader/src/Reactor/LLVMJIT.cpp	2024-03-16 23:41:43.769435900 +0100
@@ -47,7 +47,7 @@ __pragma(warning(push))
 #include "llvm/IR/DiagnosticInfo.h"
 #include "llvm/IR/Verifier.h"
 #include "llvm/Support/CommandLine.h"
-#include "llvm/Support/Host.h"
+#include "llvm/TargetParser/Host.h"
 #include "llvm/Support/TargetSelect.h"
 #include "llvm/Transforms/InstCombine/InstCombine.h"
 #include "llvm/Transforms/Instrumentation/AddressSanitizer.h"
