--- src/third_party/blink/renderer/modules/xr/xr_webgl_swap_chain.h.orig	2025-08-15 17:32:29.469405073 +0200
+++ src/third_party/blink/renderer/modules/xr/xr_webgl_swap_chain.h	2025-08-16 00:08:44.433036412 +0200
@@ -8,6 +8,7 @@
 #include "third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.h"
 #include "third_party/blink/renderer/modules/webgl/webgl_unowned_texture.h"
 #include "third_party/blink/renderer/modules/xr/xr_swap_chain.h"
+#include "third_party/blink/renderer/platform/graphics/static_bitmap_image.h"
 #include "third_party/blink/renderer/platform/heap/garbage_collected.h"
 #include "third_party/blink/renderer/platform/heap/member.h"
 
