--- maven-deploy-plugin-3.2.0/src/main/java/org/apache/maven/plugins/deploy/DeployFileMojo.java	2026-09-15 12:14:09.893659577 +0200
+++ maven-deploy-plugin-3.2.0/src/main/java/org/apache/maven/plugins/deploy/DeployFileMojo.java	2026-09-15 12:14:45.577790046 +0200
@@ -47,8 +47,8 @@
 import org.codehaus.plexus.util.FileUtils;
 import org.codehaus.plexus.util.IOUtil;
 import org.codehaus.plexus.util.StringUtils;
-import org.codehaus.plexus.util.xml.ReaderFactory;
-import org.codehaus.plexus.util.xml.WriterFactory;
+import org.codehaus.plexus.util.xml.XmlStreamReader;
+import org.codehaus.plexus.util.xml.XmlStreamWriter;
 import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
 import org.eclipse.aether.RepositorySystem;
 import org.eclipse.aether.RepositorySystemSession;
@@ -475,7 +475,7 @@
      * @throws MojoExecutionException If the file doesn't exist or cannot be read.
      */
     Model readModel(File pomFile) throws MojoExecutionException {
-        try (Reader reader = ReaderFactory.newXmlReader(pomFile)) {
+        try (Reader reader = new XmlStreamReader(pomFile)) {
             return new MavenXpp3Reader().read(reader);
         } catch (FileNotFoundException e) {
             throw new MojoExecutionException("POM not found " + pomFile, e);
@@ -499,7 +499,7 @@
             File tempFile = File.createTempFile("mvndeploy", ".pom");
             tempFile.deleteOnExit();
 
-            try (Writer fw = WriterFactory.newXmlWriter(tempFile)) {
+            try (Writer fw = new XmlStreamWriter(tempFile)) {
                 new MavenXpp3Writer().write(fw, model);
             }
 
