diff -ruN gcad3d-2.35+git20170419.c50f21f.orig/src/xa/xa_undo.c gcad3d-2.35+git20170419.c50f21f/src/xa/xa_undo.c
--- gcad3d-2.35+git20170419.c50f21f.orig/src/xa/xa_undo.c	2017-04-19 15:28:44.000000000 +0300
+++ gcad3d-2.35+git20170419.c50f21f/src/xa/xa_undo.c	2017-04-19 18:48:37.328686341 +0300
@@ -1124,7 +1124,7 @@
 /// UNDO_upd_bt             activate/disactivate  delete/restore-buttons
 
   // int    iRec;
-  int    iUndo, iRedo;    // 1=TRUE, 0=FALSE
+  int    iUndo, tiUndo, iRedo, tiRedo;    // 1=TRUE, 0=FALSE
 
 
   // printf("UNDO_upd_bt\n");
@@ -1141,12 +1141,14 @@
   // check if one of the previous groups can be deleted
   iUndo = UNDO_del_active ();  // 1=delete-icon-active; 0=not.
     // printf(" iUndo=%d\n",iUndo);
-  iUndo = ILIM01(++iUndo);
+  tiUndo = iUndo;
+  iUndo = ILIM01(++tiUndo);
 
   // check if one of the next groups can be restored
   iRedo = UNDO_res_active ();  // 1=restore-icon-active; 0=not.
     // printf(" iRedo=%d\n",iRedo);
-  iRedo = ILIM01(++iRedo);
+  tiRedo = iRedo;
+  iRedo = ILIM01(++tiRedo);
 
 
 
