Index: influxdb-5.2.2/influxdb/tests/server_tests/influxdb_instance.py
===================================================================
--- influxdb-5.2.2.orig/influxdb/tests/server_tests/influxdb_instance.py	2019-03-14 11:55:41.000000000 +0100
+++ influxdb-5.2.2/influxdb/tests/server_tests/influxdb_instance.py	2019-05-10 12:53:54.133491138 +0200
@@ -7,7 +7,7 @@ from __future__ import print_function
 from __future__ import unicode_literals
 
 import datetime
-import distutils
+import distutils.util
 import os
 import tempfile
 import shutil
