Index: docutils/test/test_parsers/test_rst/test_directives/test_code.py
===================================================================
--- docutils/test/test_parsers/test_rst/test_directives/test_code.py	(revision 8494)
+++ docutils/test/test_parsers/test_rst/test_directives/test_code.py	(revision 8495)
@@ -98,7 +98,7 @@
 
 totest['code-parsing'] = [
 ["""\
-.. code:: python
+.. code:: python3
   :class: testclass
 
    print('hello world') # to stdout
@@ -105,9 +105,9 @@
 """,
 """\
 <document source="test data">
-    <literal_block classes="code python testclass" xml:space="preserve">
+    <literal_block classes="code python3 testclass" xml:space="preserve">
          \n\
-        <inline classes="keyword">
+        <inline classes="name builtin">
             print
         <inline classes="punctuation">
             (
@@ -120,7 +120,7 @@
             # to stdout
 """],
 ["""\
-.. code:: python
+.. code:: python3
   :class: testclass
   :name: my_function
   :number-lines: 7
@@ -134,7 +134,7 @@
 """,
 """\
 <document source="test data">
-    <literal_block classes="code python testclass" ids="my-function" names="my_function" xml:space="preserve">
+    <literal_block classes="code python3 testclass" ids="my-function" names="my_function" xml:space="preserve">
         <inline classes="ln">
              7 \n\
         <inline classes="keyword">
@@ -167,7 +167,7 @@
         <inline classes="ln">
             12 \n\
             \n\
-        <inline classes="keyword">
+        <inline classes="name builtin">
             print
         <inline classes="punctuation">
             (
Index: docutils/test/test_parsers/test_rst/test_directives/test_code_long.py
===================================================================
--- docutils/test/test_parsers/test_rst/test_directives/test_code_long.py	(revision 8494)
+++ docutils/test/test_parsers/test_rst/test_directives/test_code_long.py	(revision 8495)
@@ -24,7 +24,7 @@
 
 totest['code-parsing-long'] = [
 ["""\
-.. code:: python
+.. code:: python3
   :number-lines: 7
 
   def my_function():
@@ -36,7 +36,7 @@
 """,
 """\
 <document source="test data">
-    <literal_block classes="code python" xml:space="preserve">
+    <literal_block classes="code python3" xml:space="preserve">
         <inline classes="ln">
              7 \n\
         <inline classes="keyword">
@@ -69,7 +69,7 @@
         <inline classes="ln">
             12 \n\
             \n\
-        <inline classes="keyword">
+        <inline classes="name builtin">
             print
         <inline classes="punctuation">
             (
Index: docutils/test/test_parsers/test_rst/test_interpreted.py
===================================================================
--- docutils/test/test_parsers/test_rst/test_interpreted.py	(revision 8494)
+++ docutils/test/test_parsers/test_rst/test_interpreted.py	(revision 8495)
@@ -271,7 +271,7 @@
 Custom role based on code role:
 
 .. role:: python(code)
-   :language: python
+   :language: python3
    :class: testclass
 
 Python code :python:`print("The end")`.
@@ -282,8 +282,8 @@
         Custom role based on code role:
     <paragraph>
         Python code \n\
-        <literal classes="code testclass python">
-            <inline classes="keyword">
+        <literal classes="code testclass python3">
+            <inline classes="name builtin">
                 print
             <inline classes="punctuation">
                 (
