--- inkscape-0.92.3/CMakeScripts/HelperMacros.cmake_orig	2018-04-24 21:28:19.299976294 +0200
+++ inkscape-0.92.3/CMakeScripts/HelperMacros.cmake	2018-04-24 21:00:30.300534272 +0200
@@ -37,7 +37,7 @@
 
     # static libraries are probably not useful on Windows
     # (if we ever build shared libraries those would use the RUNTIME target and we might have to revisit this)
-    if(NOT WIN32)
+    if(0)
         install(TARGETS ${name} 
             LIBRARY DESTINATION lib/inkscape
             ARCHIVE DESTINATION lib/inkscape
--- inkscape-0.92.3/src/CMakeLists.txt_orig	2018-04-24 21:21:17.069106797 +0200
+++ inkscape-0.92.3/src/CMakeLists.txt	2018-04-24 21:26:42.844223886 +0200
@@ -575,9 +575,8 @@
     inkscape_base
     inkscape
     inkview 
-    RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib/inkscape
-    ARCHIVE DESTINATION lib/inkscape)
+    RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 else()
   install(TARGETS
     inkscape_base
