diff -ur frozen-bubble-2.212/server/log.c frozen-bubble-2.212_fix/server/log.c
--- frozen-bubble-2.212/server/log.c	2012-10-24 19:53:38.000000000 +0200
+++ frozen-bubble-2.212_fix/server/log.c	2018-06-18 16:44:51.963979828 +0200
@@ -56,7 +56,7 @@
 char* get_current_date(void) 
 {
     struct tm * lt;
-    char buf[50];
+    char buf[30];
     double time = get_current_time_exact();
     time_t seconds = (time_t)time;
     lt = localtime(&seconds);
