Index: ekiga/plugins/avahi/avahi-heap.cpp
===================================================================
--- ekiga.orig/plugins/avahi/avahi-heap.cpp	2019-04-02 15:17:51.999315057 +0200
+++ ekiga/plugins/avahi/avahi-heap.cpp	2020-06-10 14:40:26.738949351 +0200
@@ -36,6 +36,7 @@
  */
 
 #include <cstdlib>
+#include <iostream>
 #include <glib/gi18n.h>
 
 #include "avahi-heap.h"
Index: ekiga/plugins/resource-list/rl-presentity.cpp
===================================================================
--- ekiga.orig/plugins/resource-list/rl-presentity.cpp	2019-04-02 15:17:52.003315050 +0200
+++ ekiga/plugins/resource-list/rl-presentity.cpp	2020-06-10 14:40:26.738949351 +0200
@@ -34,6 +34,7 @@
  */
 
 #include <algorithm>
+#include <iostream>
 #include <set>
 
 #include <glib/gi18n.h>
Index: ekiga/plugins/resource-list/rl-entry-ref.cpp
===================================================================
--- ekiga.orig/plugins/resource-list/rl-entry-ref.cpp	2019-04-02 15:17:52.003315050 +0200
+++ ekiga/plugins/resource-list/rl-entry-ref.cpp	2020-06-10 14:40:26.738949351 +0200
@@ -35,6 +35,8 @@
  *
  */
 
+#include <iostream>
+
 #include "config.h"
 
 #include <glib/gi18n-lib.h>
Index: ekiga/plugins/resource-list/rl-heap.cpp
===================================================================
--- ekiga.orig/plugins/resource-list/rl-heap.cpp	2019-04-02 15:17:52.003315050 +0200
+++ ekiga/plugins/resource-list/rl-heap.cpp	2020-06-10 14:40:26.738949351 +0200
@@ -35,6 +35,8 @@
  *
  */
 
+#include <iostream>
+
 #include <glib/gi18n.h>
 
 #include "robust-xml.h"
Index: ekiga/lib/engine/framework/services.cpp
===================================================================
--- ekiga.orig/lib/engine/framework/services.cpp	2019-04-02 15:17:51.979315092 +0200
+++ ekiga/lib/engine/framework/services.cpp	2020-06-10 14:41:09.398950440 +0200
@@ -44,6 +44,7 @@
 #include <signal.h>
 #endif
 
+#include <ostream>
 #include "services.h"
 
 boost::optional<bool>
