From: Antonio Larrosa <alarrosa@suse.com>
Subject: Fix build with SLE 15-SP1 / Leap 15.1

The net_device.ndo_select_queue signature in SLE-15-SP1's 4.12 kernel
was backported from kernel 4.19 so we must actually support it from 4.12

Index: rtl8812au-5.2.20.2/os_dep/linux/os_intfs.c
===================================================================
--- rtl8812au-5.2.20.2.orig/os_dep/linux/os_intfs.c
+++ rtl8812au-5.2.20.2/os_dep/linux/os_intfs.c
@@ -1281,7 +1281,7 @@ unsigned int rtw_classify8021d(struct sk
 
 static u16 rtw_select_queue(struct net_device *dev, struct sk_buff *skb
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 13, 0)
-  #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 19, 0)
+  #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 12, 0)
 	  , void *accel_priv
   #else
     , struct net_device *sb_dev
