diff --git a/gui/map.cc b/gui/map.cc
index a55c752e..d8656c97 100644
--- a/gui/map.cc
+++ b/gui/map.cc
@@ -44,6 +44,10 @@
 #include <vector>
 #include "appname.h"
 
+#ifndef PKGDATADIR
+#define PKGDATADIR "/usr/share/gpsbabel"
+#endif
+
 using std::string;
 using std::vector;
 
@@ -90,7 +94,8 @@ Map::Map(QWidget* parent,
   connect(mclicker, SIGNAL(logTime(QString)), this, SLOT(logTime(QString)));
 #endif
 
-  QString baseFile =  QApplication::applicationDirPath() + "/gmapbase.html";
+  QString baseFile =  PKGDATADIR;
+  baseFile += "/gmapbase.html";
   if (!QFile(baseFile).exists()) {
     QMessageBox::critical(nullptr, appName,
                           tr("Missing \"gmapbase.html\" file.  Check installation"));
