diff -rU 3 a/Framework/Inputs/DicomPyramidLevel.cpp b/Framework/Inputs/DicomPyramidLevel.cpp
--- a/Framework/Inputs/DicomPyramidLevel.cpp	2025-11-06 19:33:17.000000000 +0100
+++ b/Framework/Inputs/DicomPyramidLevel.cpp	2026-04-16 13:46:23.158878865 +0200
@@ -31,6 +31,8 @@
 
 #include <boost/lexical_cast.hpp>
 
+#include <cassert>
+
 namespace OrthancWSI
 {
   DicomPyramidLevel::TileContent& DicomPyramidLevel::GetTileContent(unsigned int tileX,
