Index: upower-0.99.8/etc/UPower.conf
===================================================================
--- upower-0.99.8.orig/etc/UPower.conf	2017-09-08 15:29:32.000000000 +0200
+++ upower-0.99.8/etc/UPower.conf	2018-07-29 00:45:16.176796927 +0200
@@ -91,4 +91,4 @@ TimeAction=120
 #
 # If HybridSleep isn't available, Hibernate will be used
 # If Hibernate isn't available, PowerOff will be used
-CriticalPowerAction=HybridSleep
+CriticalPowerAction=Hibernate
