--- ./include/sphinxbase/matrix.h.orig	2012-12-05 07:06:01.000000000 -0700
+++ ./include/sphinxbase/matrix.h	2013-03-29 08:06:21.648426061 -0600
@@ -82,7 +82,7 @@ SPHINXBASE_EXPORT void norm_3d(float32 *
 /**
  * Floor 3-d array
  * @param out output array
- * @para in input array
+ * @param in input array
  * @param d1 dimension
  * @param d2 dimension
  * @param d3 dimension
@@ -109,7 +109,7 @@ SPHINXBASE_EXPORT void floor_nz_3d(float
 
 /**
  * Floor 1-d array
- * @param m array
+ * @param v array
  * @param d1 dimension
  * @param floor floor value
  **/
--- ./src/libsphinxbase/util/heap.c.orig	2012-12-05 07:06:08.000000000 -0700
+++ ./src/libsphinxbase/util/heap.c	2013-03-29 08:03:34.889007411 -0600
@@ -78,7 +78,7 @@ typedef struct heapnode_s {
     void *data;                 /**< Application data at this node */
     int32 val;                  /**< Associated with above application data; according to which
                                    heap is sorted (in ascending order) */
-    int32 nl, nr;               /**< #left/right descendants of this node (for balancing heap) */
+    int32 nl, nr;               /**< \#left/right descendants of this node (for balancing heap) */
     struct heapnode_s *l;       /**< Root of left descendant heap */
     struct heapnode_s *r;       /**< Root of right descendant heap */
 } heapnode_t;
