Index: lalinference-4.1.6/bin/cbcBayesMCMC2pos.py
===================================================================
--- lalinference-4.1.6.orig/bin/cbcBayesMCMC2pos.py
+++ lalinference-4.1.6/bin/cbcBayesMCMC2pos.py
@@ -156,8 +156,8 @@ def downsample_and_evidence(data_hdf5, d
 	ebetas2 = np.concatenate((betas[::2], [0.0]))
 	elogls2 = np.concatenate((logls[::2], [logls[::2][-1]]))
 
-	evidence = -si.trapz(elogls, ebetas)
-	evidence2 = -si.trapz(elogls2, ebetas2)
+	evidence = -si.trapezoid(elogls, ebetas)
+	evidence2 = -si.trapezoid(elogls2, ebetas2)
 
 	posterior_samples['chain_log_evidence'] = evidence
 	posterior_samples['chain_delta_log_evidence'] = np.absolute(evidence - evidence2)
Index: lalinference-4.1.6/bin/cbcBayesThermoInt.py
===================================================================
--- lalinference-4.1.6.orig/bin/cbcBayesThermoInt.py
+++ lalinference-4.1.6/bin/cbcBayesThermoInt.py
@@ -86,8 +86,8 @@ positional arguments:
     ebetas2 = np.concatenate((betas[::2], [0.0]))
     elogls2 = np.concatenate((logls[::2], [logls[::2][-1]]))
 
-    evidence = -si.trapz(elogls, ebetas)
-    evidence2 = -si.trapz(elogls2, ebetas2)
+    evidence = -si.trapezoid(elogls, ebetas)
+    evidence2 = -si.trapezoid(elogls2, ebetas2)
 
     devidence = np.abs(evidence - evidence2)
 
