--- xf86-video-fbturbo-0.4.0/src/fbdev.c.orig	2018-05-23 14:42:50.647069133 +0000
+++ xf86-video-fbturbo-0.4.0/src/fbdev.c	2018-05-23 14:43:27.607237003 +0000
@@ -675,7 +675,7 @@ FBDevCreateScreenResources(ScreenPtr pSc
     pPixmap = pScreen->GetScreenPixmap(pScreen);
 
     if (!shadowAdd(pScreen, pPixmap, fPtr->rotate ?
-		   shadowUpdateRotatePackedWeak() : shadowUpdatePackedWeak(),
+		   shadowUpdateRotatePacked : shadowUpdatePacked,
 		   FBDevWindowLinear, fPtr->rotate, NULL)) {
 	return FALSE;
     }
