Index: tasty.cabal
===================================================================
--- tasty.cabal.orig	2023-11-07 17:48:13.765306814 +0000
+++ tasty.cabal	2023-11-07 17:48:44.757109342 +0000
@@ -69,7 +69,7 @@ library
     ansi-terminal        >= 0.9  && < 1.1
 
   -- No reason to depend on unbounded-delays on 64-bit architecture
-  if(!arch(x86_64) && !arch(aarch64))
+  if(!arch(x86_64) && !arch(aarch64) && !arch(riscv64))
     build-depends:
       unbounded-delays >= 0.1 && < 0.2
 
