Index: tpm2-tools-4.1/lib/tpm2_attr_util.c
===================================================================
--- tpm2-tools-4.1.orig/lib/tpm2_attr_util.c
+++ tpm2-tools-4.1/lib/tpm2_attr_util.c
@@ -202,7 +202,7 @@ static bool lookup_nt_friendly_name(cons
 
 static bool nt(TPMA_NV *nv, char *arg) {
 
-    uint16_t value;
+    uint16_t value = 0;
     bool result = tpm2_util_string_to_uint16(arg, &value);
     if (!result) {
         result = lookup_nt_friendly_name(arg, &value);
