Index: kded/vfolder_menu.cpp
===================================================================
--- kded/vfolder_menu.cpp.orig
+++ kded/vfolder_menu.cpp
@@ -1448,6 +1448,7 @@ static QString parseAttribute( const QDo
         else
             kDebug()<<" Error in parsing show_empty attribute :"<<str;
     }
+#if 0
     if ( e.hasAttribute( "inline" ) )
     {
         QString str = e.attribute( "inline" );
@@ -1486,6 +1487,7 @@ static QString parseAttribute( const QDo
         else
             kDebug()<<" Error in parsing inline_alias attribute :"<<str;
     }
+#endif
     if( !option.isEmpty())
     {
         option = option.prepend(":O");
