Index: py-junos-eznc-2.5.3/lib/jnpr/junos/command/__init__.py
===================================================================
--- py-junos-eznc-2.5.3.orig/lib/jnpr/junos/command/__init__.py	2020-08-25 11:35:29.000000000 +0200
+++ py-junos-eznc-2.5.3/lib/jnpr/junos/command/__init__.py	2020-09-15 09:47:59.613429201 +0200
@@ -5,7 +5,7 @@ import types
 
 from jnpr.junos.factory.factory_loader import FactoryLoader
 
-import yamlordereddictloader
+import yamlloader
 
 __all__ = []
 
@@ -30,7 +30,7 @@ class MetaPathLoader(object):
         with open(os.path.join(os.path.dirname(__file__), mod + ".yml"), "r") as stream:
             try:
                 modules = FactoryLoader().load(
-                    yaml.load(stream, Loader=yamlordereddictloader.Loader)
+                    yaml.load(stream, Loader=yamlloader.ordereddict.Loader)
                 )
             except yaml.YAMLError as exc:
                 raise ImportError("%s is not loaded" % mod)
Index: py-junos-eznc-2.5.3/tests/unit/factory/test_cmdtable.py
===================================================================
--- py-junos-eznc-2.5.3.orig/tests/unit/factory/test_cmdtable.py	2020-09-15 09:47:59.509428602 +0200
+++ py-junos-eznc-2.5.3/tests/unit/factory/test_cmdtable.py	2020-09-15 09:50:10.066179959 +0200
@@ -11,7 +11,7 @@ from jnpr.junos.exception import RpcErro
 from ncclient.manager import Manager, make_device_handler
 from ncclient.transport import SSHSession
 from mock import MagicMock, patch
-import yamlordereddictloader
+import yamlloader
 from jnpr.junos.factory.factory_loader import FactoryLoader
 import yaml
 import json
@@ -50,7 +50,7 @@ CMErrorView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = CMErrorTable(self.dev)
@@ -85,7 +85,7 @@ sysctlView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = sysctlVeriexecTable(self.dev)
@@ -126,7 +126,7 @@ CMErrorView:
     """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = CMErrorTable(self.dev)
@@ -188,7 +188,7 @@ CMErrorView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = CMErrorTable(self.dev)
@@ -212,7 +212,7 @@ CMErrorView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = CMErrorTable(self.dev)
@@ -240,7 +240,7 @@ CMErrorView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = CMErrorTable(self.dev)
@@ -267,7 +267,7 @@ CMErrorView:
     """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = CMErrorTable(self.dev)
@@ -293,7 +293,7 @@ CMErrorView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = CMErrorTable(self.dev).get()
@@ -327,7 +327,7 @@ FPCLinkStatTable:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = FPCLinkStatTable(self.dev)
@@ -371,7 +371,7 @@ ShowLuchipView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = ShowLuchipTable(self.dev)
@@ -474,7 +474,7 @@ FPCLinkStatTable:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = FPCLinkStatTable(self.dev)
@@ -518,7 +518,7 @@ XMChipStatsView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = XMChipStatsTable(self.dev)
@@ -537,7 +537,7 @@ FPCLinkStatTable:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = FPCLinkStatTable(self.dev)
@@ -636,7 +636,7 @@ FPCTTPReceiveStatsView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = FPCTTPStatsTable(self.dev)
@@ -710,7 +710,7 @@ MtipCgeStatisticsTable:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = MtipCgeSummaryTable(self.dev)
@@ -804,7 +804,7 @@ _ICMPRateView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = ICMPStatsTable(self.dev)
@@ -905,7 +905,7 @@ _ThrottleStatsTable:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = IthrottleIDTable(self.dev).get(target="fpc2")
@@ -946,7 +946,7 @@ ShowPciErrorsView:
     """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = ShowPciErrorsTable(self.dev).get()
@@ -988,7 +988,7 @@ FPCMemoryView:
         """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = FPCMemory(self.dev).get()
@@ -1046,7 +1046,7 @@ PQ3PCI:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = PQ3PCITable(self.dev)
@@ -1172,7 +1172,7 @@ _TopThreadTable:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = SchedulerTable(self.dev)
@@ -1209,7 +1209,7 @@ HostlbStatusSummaryView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = HostlbStatusSummaryTable(self.dev)
@@ -1235,7 +1235,7 @@ HostlbStatusSummaryView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = HostlbStatusSummaryTable(
@@ -1311,7 +1311,7 @@ _TransmitPerQueueView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = DevicesLocalTable(self.dev)
@@ -1382,7 +1382,7 @@ _ReceiveView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = DevicesLocalTable(self.dev)
@@ -1402,7 +1402,7 @@ EthernetSwitchStatisticsIterTable:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = EthernetSwitchStatisticsIterTable(self.dev)
@@ -1532,7 +1532,7 @@ _EthSwitchStatsFpc5Table:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = EthernetSwitchStatistics(self.dev)
@@ -1689,7 +1689,7 @@ _ShowToePfePacketStatsStream_rx_errors:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = ShowToePfePacketStatsTable(self.dev)
@@ -1757,7 +1757,7 @@ _ShowToePfePacketStatsStream_rx_errors:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = ShowToePfePacketStatsTable(self.dev)
@@ -2094,7 +2094,7 @@ XMChipInterruptStatsView:
     """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = XMChipInterruptStatsTable(self.dev)
@@ -2127,7 +2127,7 @@ XMChipInterruptStatsView:
         """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = XMChipInterruptStatsTable(self.dev)
@@ -2161,7 +2161,7 @@ FPCThreadView:
             """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = FPCThreads(self.dev)
@@ -2328,7 +2328,7 @@ CChipLoStatsView:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = CChipLoStatsTable(self.dev)
@@ -2356,7 +2356,7 @@ ARPview:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = ARPtable(self.dev)
@@ -2387,7 +2387,7 @@ ARPview:
 """
         globals().update(
             FactoryLoader().load(
-                yaml.load(yaml_data, Loader=yamlordereddictloader.Loader)
+                yaml.load(yaml_data, Loader=yamlloader.ordereddict.Loader)
             )
         )
         stats = ARPtable(self.dev)
