From: Chow Loong Jin <hyperair@debian.org>
Subject: Fix gdk_threads_init crash
Bug-Debian: http://bugs.debian.org/763724
Forwarded: no
Index: nautilus-dropbox/dropbox.in
===================================================================
-- nautilus-dropbox.orig/dropbox.in	2014-10-03 11:51:01.472492588 +0800
+++ nautilus-dropbox/dropbox.in	2014-10-03 11:55:57.429265924 +0800
@@ -297,7 +297,7 @@
         import pango
         import webbrowser
 
-        gtk.gdk.threads_init()
+        gobject.threads_init()
 
         load_serialized_images()
 
