diff --unified --recursive --text --new-file --color libwnck-43.2/libwnck/tasklist.c libwnck-43.2.new/libwnck/tasklist.c
--- libwnck-43.2/libwnck/tasklist.c	2025-01-15 02:48:27.000000000 +0800
+++ libwnck-43.2.new/libwnck/tasklist.c	2025-05-20 15:36:04.326697704 +0800
@@ -3985,6 +3985,7 @@
   GdkPixbuf *pixbuf;
   WnckHandle *handle;
   gsize mini_icon_size;
+  const char *icon_name;
 
   pixbuf = NULL;
 
@@ -4009,8 +4010,6 @@
 
     case WNCK_TASK_STARTUP_SEQUENCE:
 #ifdef HAVE_STARTUP_NOTIFICATION
-      const char *icon_name;
-
       icon_name = sn_startup_sequence_get_icon_name (task->startup_sequence);
       if (icon_name != NULL)
         {
