---
 include/builddefs.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- include/builddefs.in.orig
+++ include/builddefs.in
@@ -37,7 +37,7 @@ PKG_ROOT_SBIN_DIR = @root_sbindir@
 PKG_ROOT_LIB_DIR= @root_libdir@@libdirsuffix@
 PKG_INC_DIR	= @includedir@
 PKG_MAN_DIR	= @mandir@
-PKG_DOC_DIR	= @datadir@/doc/@pkg_name@
+PKG_DOC_DIR	= @datadir@/doc/packages/@pkg_name@
 PKG_LOCALE_DIR	= @datadir@/locale
 
 CC		= @cc@
