--- a/configure.ac
+++ b/configure.ac
@@ -46,7 +46,7 @@ dnl ************************************
 dnl Initialize automake
 dnl **************************************************************************
 
-AM_INIT_AUTOMAKE([1.14 no-dist-gzip dist-xz tar-ustar -Wno-portability subdir-objects])
+AM_INIT_AUTOMAKE([1.13 no-dist-gzip dist-xz tar-ustar -Wno-portability subdir-objects])
 AM_MAINTAINER_MODE([enable])
 AM_SILENT_RULES([yes])
 
@@ -69,7 +69,7 @@ dnl Internationalization
 dnl **************************************************************************
 
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.19.4])
+AM_GNU_GETTEXT_VERSION([0.19.2])
 
 GETTEXT_PACKAGE=metacity
 AC_SUBST([GETTEXT_PACKAGE])
