--- a/Cargo.toml.orig	2025-07-18 13:10:15.182788420 +0800
+++ b/Cargo.toml	2025-07-18 13:11:10.652425382 +0800
@@ -402,7 +402,6 @@
 [profile.release]
 codegen-units = 1
 incremental = true
-lto = true
 opt-level = 'z' # Optimize for size
 split-debuginfo = "packed"
 debug = "line-tables-only"
@@ -416,7 +415,6 @@
 [profile.release-lite]
 inherits = "release"
 codegen-units = 128
-lto = "thin"
 
 # Key generation is too slow on `debug`
 [profile.dev.package.num-bigint-dig]
