diff --git a/vendor/v8/build/config/compiler/BUILD.gn b/vendor/v8/build/config/compiler/BUILD.gn
index 9cdafb2..211684c 100644
--- a/vendor/v8/build/config/compiler/BUILD.gn
+++ b/vendor/v8/build/config/compiler/BUILD.gn
@@ -1548,15 +1548,6 @@ config("default_warnings") {
         cflags += [ "-Wno-unqualified-std-cast-call" ]
       }
 
-      if (!is_nacl && !(is_chromeos ||
-                        default_toolchain == "//build/toolchain/cros:target")) {
-        # TODO(https://crbug.com/1316298): Re-enable once test failure is figured out
-        cflags += [
-          "-Xclang",
-          "-no-opaque-pointers",
-        ]
-      }
-
       if (is_fuchsia) {
         # TODO(https://bugs.chromium.org/p/fuchsia/issues/detail?id=77383)
         cflags += [ "-Wno-deprecated-copy" ]
