Index: unix/unix.c
===================================================================
--- unix/unix.c.orig
+++ unix/unix.c
@@ -835,11 +835,7 @@ void version_local()
 
 
 /* Define the compile date string */
-#ifdef __DATE__
-#  define COMPILE_DATE " on " __DATE__
-#else
-#  define COMPILE_DATE ""
-#endif
+#define COMPILE_DATE ""
 
     printf("Compiled with %s for Unix (%s)%s.\n\n",
            COMPILER_NAME, OS_NAME, COMPILE_DATE);
