From 0f5427bc66b9c5d4372676fa3c845da710688e37 Mon Sep 17 00:00:00 2001
From: Vlad Zahorodnii <vlad.zahorodnii@kde.org>
Date: Mon, 18 Oct 2021 12:42:19 +0300
Subject: [PATCH] foo

---
 src/composite.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/composite.cpp b/src/composite.cpp
index 8d8edb323..2317ad9ff 100644
--- a/src/composite.cpp
+++ b/src/composite.cpp
@@ -251,6 +251,10 @@ bool Compositor::setupStart()
         delete m_scene;
         m_scene = nullptr;
 
+        if (auto *con = kwinApp()->x11Connection()) {
+            xcb_composite_unredirect_subwindows(con, kwinApp()->x11RootWindow(),
+                                                XCB_COMPOSITE_REDIRECT_MANUAL);
+        }
         if (m_selectionOwner) {
             m_selectionOwner->setOwning(false);
             m_selectionOwner->release();
-- 
2.33.1

