Index: aiohttp-3.13.3/Makefile
===================================================================
--- aiohttp-3.13.3.orig/Makefile
+++ aiohttp-3.13.3/Makefile
@@ -47,10 +47,8 @@ endif
 .SECONDARY: $(call to-hash,$(ALLS))
 
 .update-pip:
-	@python -m pip install --upgrade pip
 
 .install-cython: .update-pip $(call to-hash,requirements/cython.txt)
-	@python -m pip install -r requirements/cython.in -c requirements/cython.txt
 	@touch .install-cython
 
 aiohttp/_find_header.c: $(call to-hash,aiohttp/hdrs.py ./tools/gen.py)
@@ -85,7 +83,6 @@ cythonize: .install-cython $(PYXS:.pyx=.
 cythonize-nodeps: $(PYXS:.pyx=.c) aiohttp/_websocket/reader_c.c
 
 .install-deps: .install-cython $(PYXS:.pyx=.c) aiohttp/_websocket/reader_c.c $(call to-hash,$(CYS) $(REQS))
-	@python -m pip install -r requirements/dev.in -c requirements/dev.txt
 	@touch .install-deps
 
 .PHONY: lint
@@ -100,7 +97,6 @@ mypy:
 	mypy
 
 .develop: .install-deps generate-llhttp $(call to-hash,$(PYS) $(CYS) $(CS))
-	python -m pip install -e . -c requirements/runtime-deps.txt
 	@touch .develop
 
 .PHONY: test
@@ -110,12 +106,12 @@ test: .develop
 .PHONY: vtest
 vtest: .develop
 	@pytest -s -v
-	@python -X dev -m pytest -s -v -m dev_mode
+	python3 -X dev -m pytest -s -v -m dev_mode
 
 .PHONY: vvtest
 vvtest: .develop
 	@pytest -vv
-	@python -X dev -m pytest -s -v -m dev_mode
+	python3 -X dev -m pytest -s -v -m dev_mode
 
 
 define run_tests_in_docker
@@ -151,7 +147,7 @@ clean:
 	@rm -rf build
 	@rm -rf cover
 	@make -C docs clean
-	@python setup.py clean
+	python3 setup.py clean
 	@rm -f aiohttp/*.so
 	@rm -f aiohttp/*.pyd
 	@rm -f aiohttp/*.html
@@ -182,7 +178,6 @@ doc-spelling:
 
 .PHONY: install
 install: .update-pip
-	@python -m pip install -r requirements/dev.in -c requirements/dev.txt
 
 .PHONY: install-dev
 install-dev: .develop
@@ -190,4 +185,4 @@ install-dev: .develop
 .PHONY: sync-direct-runtime-deps
 sync-direct-runtime-deps:
 	@echo Updating 'requirements/runtime-deps.in' from 'pyproject.toml'... >&2
-	@python requirements/sync-direct-runtime-deps.py
+	python3 requirements/sync-direct-runtime-deps.py
Index: aiohttp-3.13.3/tools/gen.py
===================================================================
--- aiohttp-3.13.3.orig/tools/gen.py
+++ aiohttp-3.13.3/tools/gen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import io
 import pathlib
