Index: aiohttp-3.8.3/setup.cfg
===================================================================
--- aiohttp-3.8.3.orig/setup.cfg
+++ aiohttp-3.8.3/setup.cfg
@@ -125,6 +125,7 @@ filterwarnings =
 	ignore:Creating a LegacyVersion has been deprecated and will be removed in the next major release:DeprecationWarning::
 	ignore:module 'sre_constants' is deprecated:DeprecationWarning:pkg_resources._vendor.pyparsing
 	ignore:path is deprecated. Use files.. instead. Refer to https.//importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.:DeprecationWarning:certifi.core
+	ignore:Passing None has been deprecated:pytest.PytestDeprecationWarning
 junit_suite_name = aiohttp_test_suite
 norecursedirs = dist docs build .tox .eggs
 minversion = 3.8.2
