diff -Nru aiohttp-3.10.9.orig/tests/test_imports.py aiohttp-3.10.9/tests/test_imports.py
--- aiohttp-3.10.9.orig/tests/test_imports.py	2024-10-04 18:47:06.000000000 +0200
+++ aiohttp-3.10.9/tests/test_imports.py	2024-10-09 09:57:53.705087969 +0200
@@ -29,7 +29,7 @@
 
 
 _TARGET_TIMINGS_BY_PYTHON_VERSION = {
-    "3.12": 250,  # 3.12 is expected to be a bit slower due to performance trade-offs
+    "3.12": 500,  # 3.12 is expected to be a bit slower due to performance trade-offs
 }
 
 
@@ -66,6 +66,6 @@
             os.environ["PYTHONPATH"] = old_path
 
     expected_time = _TARGET_TIMINGS_BY_PYTHON_VERSION.get(
-        f"{sys.version_info.major}.{sys.version_info.minor}", 200
+        f"{sys.version_info.major}.{sys.version_info.minor}", 500
     )
     assert best_time_ms < expected_time
