Index: aspell-0.60.8.2/modules/speller/default/writable.cpp
===================================================================
--- aspell-0.60.8.2.orig/modules/speller/default/writable.cpp
+++ aspell-0.60.8.2/modules/speller/default/writable.cpp
@@ -298,7 +298,7 @@ static inline void set_sl(WordEntry & re
 
 static void soundslike_next(WordEntry * w)
 {
-  const Str * & i   = (const Str * &)(w->intr[0]);
+  const Str * i   = (const Str * )(w->intr[0]);
   const Str *   end = (const Str *  )(w->intr[1]);
   set_word(*w, *i);
   ++i;
@@ -741,7 +741,7 @@ WritableReplDict::Enum * WritableReplDic
 
 static void repl_next(WordEntry * w)
 {
-  const Str * & i   = (const Str * &)(w->intr[0]);
+  const Str * i   = (const Str * )(w->intr[0]);
   const Str *   end = (const Str *  )(w->intr[1]);
   set_word(*w, *i);
   ++i;
