diff -Naur ois.orig/demos/FFConsoleDemo.cpp ois/demos/FFConsoleDemo.cpp
--- ois.orig/demos/FFConsoleDemo.cpp	2010-04-01 05:38:04.000000000 +0200
+++ ois/demos/FFConsoleDemo.cpp	2012-04-25 22:01:01.000000000 +0200
@@ -7,6 +7,7 @@
 #include <ios>
 #include <sstream>
 #include <vector>
+#include <unistd.h>
 
 using namespace std;
 
diff -Naur ois.orig/demos/OISConsole.cpp ois/demos/OISConsole.cpp
--- ois.orig/demos/OISConsole.cpp	2010-08-16 00:51:34.000000000 +0200
+++ ois/demos/OISConsole.cpp	2012-04-25 21:56:14.000000000 +0200
@@ -12,6 +12,7 @@
 #include <iostream>
 #include <vector>
 #include <sstream>
+#include <unistd.h>
 
 ////////////////////////////////////Needed Windows Headers////////////
 #if defined OIS_WIN32_PLATFORM
diff -Naur ois.orig/src/linux/LinuxForceFeedback.cpp ois/src/linux/LinuxForceFeedback.cpp
--- ois.orig/src/linux/LinuxForceFeedback.cpp	2010-04-01 05:06:55.000000000 +0200
+++ ois/src/linux/LinuxForceFeedback.cpp	2012-04-25 21:50:57.000000000 +0200
@@ -26,6 +26,7 @@
 #include <cstdlib>
 #include <errno.h>
 #include <memory.h>
+#include <unistd.h>
 
 using namespace OIS;
 
diff -Naur ois.orig/src/linux/LinuxJoyStickEvents.cpp ois/src/linux/LinuxJoyStickEvents.cpp
--- ois.orig/src/linux/LinuxJoyStickEvents.cpp	2010-07-26 03:28:41.000000000 +0200
+++ ois/src/linux/LinuxJoyStickEvents.cpp	2012-04-25 21:44:17.000000000 +0200
@@ -36,7 +36,8 @@
 
 
 #include <sstream>
-# include <iostream>
+#include <iostream>
+#include <unistd.h>
 using namespace std;
 
 using namespace OIS;
