diff -rup pagure-5.0/files/gitolite3.rc pagure-5.0.cfg-defs/files/gitolite3.rc
--- pagure-5.0/files/gitolite3.rc	2018-08-20 05:28:38.000000000 -0400
+++ pagure-5.0.cfg-defs/files/gitolite3.rc	2018-09-24 17:29:01.927780974 -0400
@@ -16,7 +16,7 @@
 
     # ------------------------------------------------------------------
 
-    GL_REPO_BASE                    => '/path/to/git/repositories',
+    GL_REPO_BASE                    => '/srv/gitolite/repositories',
 
     # default umask gives you perms of '0700'; see the rc file docs for
     # how/why you might change this
diff -rup pagure-5.0/files/pagure.cfg.sample pagure-5.0.cfg-defs/files/pagure.cfg.sample
--- pagure-5.0/files/pagure.cfg.sample	2018-09-24 15:57:50.000000000 -0400
+++ pagure-5.0.cfg-defs/files/pagure.cfg.sample	2018-09-24 17:29:01.928780964 -0400
@@ -68,21 +68,21 @@ GIT_URL_GIT = 'git://localhost.localdoma
 
 ### Folder containing to the git repos
 GIT_FOLDER = os.path.join(
-    os.path.abspath(os.path.dirname(__file__)),
-    '..',
-    'repos'
+    '/srv',
+    'gitolite',
+    'repositories'
 )
 
 REPOSPANNER_PSEUDO_FOLDER = os.path.join(
-    os.path.abspath(os.path.dirname(__file__)),
-    '..',
+    '/srv',
+    'gitolite',
     'pseudo'
 )
 
 ### Folder containing the clones for the remote pull-requests
 REMOTE_GIT_FOLDER = os.path.join(
-    os.path.abspath(os.path.dirname(__file__)),
-    '..',
+    '/srv',
+    'gitolite',
     'remotes'
 )
 
@@ -92,21 +92,23 @@ VIRUS_SCAN_ATTACHMENTS = False
 
 ### Configuration file for gitolite
 GITOLITE_CONFIG = os.path.join(
-    os.path.abspath(os.path.dirname(__file__)),
-    '..',
+    '/srv',
+    'gitolite',
+    '.gitolite',
+    'conf',
     'gitolite.conf'
 )
 
 
 ### Home folder of the gitolite user
 ### Folder where to run gl-compile-conf from
-GITOLITE_HOME = None
+GITOLITE_HOME = '/srv/gitolite'
 
 ### Version of gitolite used: 2 or 3?
 GITOLITE_VERSION = 3
 
 ### Folder containing all the public ssh keys for gitolite
-GITOLITE_KEYDIR = None
+GITOLITE_KEYDIR = os.path.join(GITOLITE_HOME, '.gitolite', 'keydir')
 
 ### Path to the gitolite.rc file
 GL_RC = None
diff -rup pagure-5.0/files/pagure.conf pagure-5.0.cfg-defs/files/pagure.conf
--- pagure-5.0/files/pagure.conf	2018-08-23 15:08:03.000000000 -0400
+++ pagure-5.0.cfg-defs/files/pagure.conf	2018-09-24 17:29:01.928780964 -0400
@@ -29,9 +29,9 @@
   ## Use secure TLSv1.1 and TLSv1.2 ciphers
   #Header always add Strict-Transport-Security "max-age=15768000; includeSubDomains; preload"
 
-  #SSLCertificateFile /etc/pki/tls/....crt
-  #SSLCertificateChainFile /etc/pki/tls/....intermediate.crt
-  #SSLCertificateKeyFile /etc/pki/tls/....key
+  #SSLCertificateFile /etc/ssl/....crt
+  #SSLCertificateChainFile /etc/ssl/....intermediate.crt
+  #SSLCertificateKeyFile /etc/ssl/....key
 
   #Alias /static /usr/lib/pythonX.Y/site-packages/pagure/static/
 
@@ -60,24 +60,24 @@
   ## Use secure TLSv1.1 and TLSv1.2 ciphers
   #Header always add Strict-Transport-Security "max-age=15768000; includeSubDomains; preload"
 
-  #SSLCertificateFile /etc/pki/tls/....crt
-  #SSLCertificateChainFile /etc/pki/tls/....intermediate.crt
-  #SSLCertificateKeyFile /etc/pki/tls/....key
+  #SSLCertificateFile /etc/ssl/....crt
+  #SSLCertificateChainFile /etc/ssl/....intermediate.crt
+  #SSLCertificateKeyFile /etc/ssl/....key
 
   #Alias /static /usr/lib/pythonX.Y/site-packages/pagure/static/
-  #Alias /releases /var/www/releases
+  #Alias /releases /srv/www/pagure-releases
 
   ## Section used to support cloning git repo over http (https in this case)
-  #SetEnv GIT_PROJECT_ROOT /path/to/git/repositories
+  #SetEnv GIT_PROJECT_ROOT /srv/gitolite/repositories
 
-  #AliasMatch ^/(.*/objects/[0-9a-f]{2}/[0-9a-f]{38})$          /path/to/git/repositories/$1
-  #AliasMatch ^/(.*/objects/pack/pack-[0-9a-f]{40}.(pack|idx))$ /path/to/git/repositories/$1
+  #AliasMatch ^/(.*/objects/[0-9a-f]{2}/[0-9a-f]{38})$          /srv/gitolite/repositories/$1
+  #AliasMatch ^/(.*/objects/pack/pack-[0-9a-f]{40}.(pack|idx))$ /srv/gitolite/repositories/$1
   #ScriptAliasMatch \
     #"(?x)^/(.*/(HEAD | \
     #info/refs | \
     #objects/info/[^/]+ | \
     #git-(upload|receive)-pack))$" \
-    #/usr/libexec/git-core/git-http-backend/$1
+    #/usr/lib/git/git-http-backend/$1
 
   #<Location />
    #WSGIProcessGroup pagure
@@ -106,7 +106,7 @@
    #</IfModule>
   #</Location>
 
-  #<Directory /var/www/releases>
+  #<Directory /srv/www/pagure-releases>
     #Options +Indexes
   #</Directory>
 
