diff -Naur corosync-2.4.5/exec/ipc_glue.c corosync-2.4.5-new/exec/ipc_glue.c
--- corosync-2.4.5/exec/ipc_glue.c	2019-07-30 20:26:27.000000000 +0800
+++ corosync-2.4.5-new/exec/ipc_glue.c	2019-08-13 13:54:38.187443967 +0800
@@ -272,7 +272,7 @@
 	struct cs_ipcs_conn_context *context;
 	char proc_name[32];
 	struct qb_ipcs_connection_stats stats;
-	int32_t size = sizeof(struct cs_ipcs_conn_context);
+	size_t size = sizeof(struct cs_ipcs_conn_context);
 	char key_name[ICMAP_KEYNAME_MAXLEN];
 	int set_client_pid = 0;
 	int set_proc_name = 0;
diff -Naur corosync-2.4.5/include/corosync/coroapi.h corosync-2.4.5-new/include/corosync/coroapi.h
--- corosync-2.4.5/include/corosync/coroapi.h	2019-07-30 20:26:27.000000000 +0800
+++ corosync-2.4.5-new/include/corosync/coroapi.h	2019-08-13 13:55:26.031711198 +0800
@@ -494,7 +494,7 @@
 	unsigned short priority; /* Lower priority are loaded first, unloaded last.
 				  * 0 is a special case which always loaded _and_ unloaded last
 				  */
-	unsigned int private_data_size;
+	size_t private_data_size;
 	enum cs_lib_flow_control flow_control;
 	enum cs_lib_allow_inquorate allow_inquorate;
 	char *(*exec_init_fn) (struct corosync_api_v1 *);
