diff -purN gdal-2.4.0.orig/swig/perl/GNUmakefile gdal-2.4.0/swig/perl/GNUmakefile
--- gdal-2.4.0.orig/swig/perl/GNUmakefile	2018-12-14 22:34:20.000000000 +0100
+++ gdal-2.4.0/swig/perl/GNUmakefile	2018-12-21 17:37:59.897136465 +0100
@@ -28,7 +28,7 @@ build: Makefile_Geo__GDAL
 	if [ -f Makefile_Geo__GNM ]; then $(MAKE) -f Makefile_Geo__GNM; fi
 
 Makefile_Geo__GDAL: ${WRAPPERS} Makefile.PL
-	perl Makefile.PL INSTALL_BASE=$(INST_PREFIX)
+	perl Makefile.PL INSTALLDIRS=vendor
 
 test: build
 	$(MAKE) -f Makefile_Geo__GDAL test
diff -purN gdal-2.4.0.orig/swig/perl/Makefile gdal-2.4.0/swig/perl/Makefile
--- gdal-2.4.0.orig/swig/perl/Makefile	2018-12-14 22:34:20.000000000 +0100
+++ gdal-2.4.0/swig/perl/Makefile	2018-12-21 17:37:45.553036952 +0100
@@ -7,7 +7,7 @@ build: Makefile_Geo__GDAL
 	$(MAKE) -f Makefile_Geo__OSR
 
 Makefile_Geo__GDAL:
-	perl Makefile.PL INSTALL_BASE=$(INST_PREFIX)
+	perl Makefile.PL INSTALLDIRS=vendor
 
 test: build
 	$(MAKE) -f Makefile_Geo__GDAL test
