Index: file-roller-3.31.92/src/fr-init.c
===================================================================
--- file-roller-3.31.92.orig/src/fr-init.c
+++ file-roller-3.31.92/src/fr-init.c
@@ -374,6 +374,7 @@ register_archives (void)
 
 	register_archive (fr_command_tar_get_type ());
 	register_archive (fr_command_cfile_get_type ());
+	register_archive (fr_command_zip_get_type ());
 	register_archive (fr_command_7z_get_type ());
 	register_archive (fr_command_dpkg_get_type ());
 
@@ -389,7 +390,6 @@ register_archives (void)
 	register_archive (fr_command_rpm_get_type ());
 	register_archive (fr_command_unsquashfs_get_type ());
 	register_archive (fr_command_unstuff_get_type ());
-	register_archive (fr_command_zip_get_type ());
 	register_archive (fr_command_lrzip_get_type ());
 	register_archive (fr_command_zoo_get_type ());
 #if HAVE_JSON_GLIB
