--- src-tarball/CMakeLists.txt	
+++ src-tarball/CMakeLists.txt	
@@ -209,12 +209,7 @@ find_package(Boost ${MINIMUM_BOOST_VERSI
         system
         thread
     REQUIRED)
-if(${Boost_VERSION} GREATER 106699) # boost >= 1.67
-    find_package(Boost COMPONENTS python27 REQUIRED)
-    set(Boost_PYTHON_LIBRARY ${Boost_PYTHON27_LIBRARY})
-else()
     find_package(Boost COMPONENTS python REQUIRED)
-endif()
 
 find_package(ZLIB REQUIRED)
 if(NOT BUILD_HEADLESS)
