From: Rafael Cunha de Almeida <rafael@kontesti.me>
Forwarded: Yes
Last-Update: 2011-01-17
Description: Warns about HTML syntax checker returning an error
 If the HTML syntax checker returns an error (if the command doesn't
 exist or if it fails), the original behaviour is to quietly continue.
 This patch makes html2ps louder. It warns that the HTML syntax checker
 program returned an error.

diff --git a/html2ps b/html2ps
index ba6bf3b..1908139 100755
--- a/html2ps
+++ b/html2ps
@@ -2430,6 +2430,7 @@ sub h2p {
       $file="$scr";
     }
     &dbg(`$package{'check'} $file`);
+    &dbg("warning: HTML syntax checker returned in error.\n") if ($?);
   }
 
   if(!$latin1) {
