diff -Nuar a/waynergy.desktop b/waynergy.desktop
--- a/waynergy.desktop	2023-05-29 16:55:35.000000000 -0500
+++ b/waynergy.desktop	2023-07-08 13:06:37.359067024 -0500
@@ -1,11 +1,10 @@
 [Desktop Entry]
 Comment=A Barrier/Synergy client for Wayland compositors
-Exec=/usr/bin/waynergy
+Exec=waynergy
 Icon=system-run
 Name=Waynergy
 StartupNotify=true
 Terminal=true
-TerminalOptions=\s--noclose
 Type=Application
 X-KDE-Wayland-Interfaces=org_kde_kwin_fake_input
 X-Desktop-File-Install-Version=0.26
