Index: cpprestsdk-2.10.16/Release/tests/functional/streams/istream_tests.cpp
===================================================================
--- cpprestsdk-2.10.16.orig/Release/tests/functional/streams/istream_tests.cpp
+++ cpprestsdk-2.10.16/Release/tests/functional/streams/istream_tests.cpp
@@ -1305,7 +1305,7 @@ SUITE(istream_tests)
         test_string.append(" 6E+4.5"); // two numbers merged in exponent
         test_string.append(" 6E-4.5"); // two numbers merged in exponent
         test_string.append(" 3.14 -10 +42.0 -1234.567 .01 +0 -0");
-#ifndef __APPLE__
+#if FLT_EVAL_METHOD > 1
         test_string.append(" 12345678901234567890123456789012345678901234567890"); // a big number
 #endif
         test_string.append(" 9.81E05 6.0221413e+23 1.6e-14"); // numbers with exponent
