Index: jedi-0.14.0/setup.py
===================================================================
--- jedi-0.14.0.orig/setup.py
+++ jedi-0.14.0/setup.py
@@ -2,7 +2,6 @@
 
 from setuptools import setup, find_packages
 
-import os
 import ast
 
 __AUTHOR__ = 'David Halter'
@@ -17,8 +16,6 @@ readme = open('README.rst').read() + '\n
 with open('requirements.txt') as f:
     install_requires = f.read().splitlines()
 
-assert os.path.isfile("jedi/third_party/typeshed/LICENSE"), \
-    "Please download the typeshed submodule first (Hint: git submodule update --init)"
 
 setup(name='jedi',
       version=version,
Index: jedi-0.14.0/jedi/evaluate/gradual/typeshed.py
===================================================================
--- jedi-0.14.0.orig/jedi/evaluate/gradual/typeshed.py
+++ jedi-0.14.0/jedi/evaluate/gradual/typeshed.py
@@ -8,7 +8,7 @@ from jedi.evaluate.base_context import C
 from jedi.evaluate.gradual.stub_context import TypingModuleWrapper, StubModuleContext
 
 _jedi_path = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
-TYPESHED_PATH = os.path.join(_jedi_path, 'third_party', 'typeshed')
+TYPESHED_PATH = "/usr/share/typeshed"
 
 _IMPORT_MAP = dict(
     _collections='collections',
