Index: lalpulsar-3.0.1/test/LineRobustStatsTest.c
===================================================================
--- lalpulsar-3.0.1.orig/test/LineRobustStatsTest.c
+++ lalpulsar-3.0.1/test/LineRobustStatsTest.c
@@ -410,7 +410,7 @@ XLALCheckBSGLVectorFunctions ( void )
 
   REAL4 cohFstar0 = 4.37; // 10% false-alarm: invFalseAlarm_chi2 ( 0.1, 4 * 3 ) / 3
   UINT4 numDet = 3;
-  REAL4 oLtLGX[3] = {0.1, 0.8, 1.5}; /* per-IFO prior odds ratio for line vs. Gaussian noise */
+  REAL4 oLtLGX[PULSAR_MAX_DETECTORS] = {0.1, 0.8, 1.5}; /* per-IFO prior odds ratio for line vs. Gaussian noise */
 
   BSGLSetup *setup_noLogCorr;
   BSGLSetup *setup_withLogCorr;
