Index: django-threadedcomments-1.2/runtests.py
===================================================================
--- django-threadedcomments-1.2.orig/runtests.py
+++ django-threadedcomments-1.2/runtests.py
@@ -50,6 +50,7 @@ if not settings.configured:
             MIDDLEWARE = (
                 'django.middleware.common.CommonMiddleware',
                 'django.contrib.sessions.middleware.SessionMiddleware',
+                'django.contrib.messages.middleware.MessageMiddleware',
                 'django.middleware.csrf.CsrfViewMiddleware',
                 'django.contrib.auth.middleware.AuthenticationMiddleware',
             )
@@ -59,6 +60,7 @@ if not settings.configured:
             MIDDLEWARE_CLASSES = (
                 'django.middleware.common.CommonMiddleware',
                 'django.contrib.sessions.middleware.SessionMiddleware',
+                'django.contrib.messages.middleware.MessageMiddleware',
                 'django.middleware.csrf.CsrfViewMiddleware',
                 'django.contrib.auth.middleware.AuthenticationMiddleware',
             )
@@ -73,6 +75,7 @@ if not settings.configured:
         INSTALLED_APPS = (
             'django.contrib.auth',
             'django.contrib.contenttypes',
+            'django.contrib.sessions',
             'django.contrib.messages',
             'django.contrib.sites',
             'django.contrib.admin',
