--- m4/qt5.m4.orig	2015-03-16 15:29:47.800729769 +0100
+++ m4/qt5.m4	2015-03-16 15:41:35.044747937 +0100
@@ -114,8 +114,8 @@
 
 dnl If $1 is Qt 5's moc command then $2 else $3
 AC_DEFUN([FC_QT5_IF_QT5_MOC],
-  AS_IF([test "`$1 -v 2<&1 | grep -o 'Qt [[[0-9]]]\+'`" = "Qt 5" ||
-         test "`$1 -v 2<&1 | grep -o 'moc [[[0-9]]]\+'`" = "moc 5" ],
+  AS_IF([test "`$1 -v 2<&1 | grep -o 'Qt 5\+'`" = "Qt 5" ||
+         test "`$1 -v 2<&1 | grep -o $1' 5\+'`" = "$1 5" ],
     [$2], [$3]))
 
 dnl Set MOCCMD to $1 if it is the Qt 5 "moc". If not run $2 parameter.
