Index: freeciv-3.2.0/dependencies/tolua-5.2/src/bin/lua/package.lua
===================================================================
--- freeciv-3.2.0.orig/dependencies/tolua-5.2/src/bin/lua/package.lua
+++ freeciv-3.2.0/dependencies/tolua-5.2/src/bin/lua/package.lua
@@ -92,7 +92,6 @@ function classPackage:preamble ()
  output('/*\n')
  output('** Lua binding: '..self.name..'\n')
 -- fred: removido porque gera trabalho na hora de fazer diffs
--- output('** Generated automatically by '..TOLUA_VERSION..' on '..date()..'.\n')
  output('*/\n\n')
 
  output('#include "tolua.h"\n\n')
@@ -125,7 +124,7 @@ function classPackage:preamble ()
 		output('\n')
 		output('/* function to release collected object via destructor */')
 		output('#ifdef __cplusplus\n')
-		for i,v in pairs(_collect) do
+		for i,v in ipairs(_collect) do
 		 output('\nstatic int '..v..' (lua_State* tolua_S)')
 			output('{')
 			output(' '..i..'* self = ('..i..'*) tolua_tousertype(tolua_S,1,0);')
@@ -182,7 +181,6 @@ end
 -- write header file
 function classPackage:header ()
  output('/*\n') output('** Lua binding: '..self.name..'\n')
- output('** Generated automatically by '..TOLUA_VERSION..' on '..date()..'.\n')
  output('*/\n\n')
 
  if not flags.h then
