diff -ru gjiten-2.5.20060907.orig/help/C/Makefile.am gjiten-2.5.20060907/help/C/Makefile.am
--- gjiten-2.5.20060907.orig/help/C/Makefile.am	2006-04-09 21:19:01.000000000 +0200
+++ gjiten-2.5.20060907/help/C/Makefile.am	2006-09-08 11:14:36.000000000 +0200
@@ -9,6 +9,6 @@
 dist-hook: app-dist-hook
 
 $(top_srcdir)/README: gjiten.xml
-	xmlto txt gjiten.xml
+	xmlto --skip-validation txt gjiten.xml
 	-mv gjiten.txt $(top_srcdir)/README
 
