Index: FreeFileSync-14.3/FreeFileSync/Source/ffs_paths.cpp
===================================================================
--- FreeFileSync-14.3.orig/FreeFileSync/Source/ffs_paths.cpp
+++ FreeFileSync-14.3/FreeFileSync/Source/ffs_paths.cpp
@@ -53,7 +53,8 @@ Zstring fff::getInstallDirPath()
 
 Zstring fff::getResourceDirPath()
 {
-    return appendPath(getProcessParentFolderPath(), Zstr("Resources"));
+    // for opensuse install to /usr/share/FreeFileSync specificly
+    return Zstr("/usr/share/FreeFileSync/");
 }
 
 
