diff -ruN a/FreeFileSync/Source/localization.cpp b/FreeFileSync/Source/localization.cpp
--- a/FreeFileSync/Source/localization.cpp	2026-10-05 11:23:23.983134803 +0200
+++ b/FreeFileSync/Source/localization.cpp	2026-10-05 11:23:24.003523570 +0200
@@ -8,7 +8,6 @@
 #include <clocale> //setlocale
 #include <zen/file_traverser.h>
 #include <zen/file_io.h>
-#include <wx/log.h>
 #include <wx/zipstrm.h>
 #include <wx/mstream.h>
 #include <wx/uilocale.h>
@@ -129,8 +128,6 @@
         }
         //-------------------------------------------------------------
 
-        wxLogCollector zipLog; //wxWidgets shows modal error dialog by default => "no, wxWidgets, NO!"
-
         wxMemoryInputStream byteStream(rawStream.c_str(), rawStream.size()); //does not take ownership
         wxZipInputStream zipStream(byteStream, wxConvUTF8);
 
@@ -151,11 +148,8 @@
                 streams.emplace_back(zipPath + Zstr(':') + utfTo<Zstring>(entry->GetName()), std::move(stream));
             }
             else
-                break; //error
+                assert(false);
         }
-
-        if (zipStream.GetLastError() != wxSTREAM_EOF || !zipLog.GetMessages().empty())
-            throw FileError(replaceCpy(_("Cannot read file %x."), L"%x", fmtPath(zipPath)), utfTo<std::wstring>(zipLog.GetMessages()));
     }();
     //--------------------------------------------------------------------
 
diff -ruN a/FreeFileSync/Source/ui/batch_status_handler.cpp b/FreeFileSync/Source/ui/batch_status_handler.cpp
--- a/FreeFileSync/Source/ui/batch_status_handler.cpp	2026-10-05 11:23:23.980777739 +0200
+++ b/FreeFileSync/Source/ui/batch_status_handler.cpp	2026-10-05 11:23:24.002664094 +0200
@@ -190,12 +190,12 @@
     else if (!suspend && !autoClose && //only play when actually showing results dialog
              !soundFileSyncComplete_.empty())
     {
-        wxLogCollector soundLog; //wxWidgets shows modal error dialog by default => "no, wxWidgets, NO!"
+        //wxWidgets shows modal error dialog by default => "no, wxWidgets, NO!"
+        wxLog* oldLogTarget = wxLog::SetActiveTarget(new wxLogStderr); //transfer and receive ownership!
+        ZEN_ON_SCOPE_EXIT(delete wxLog::SetActiveTarget(oldLogTarget));
 
         wxSound::Play(utfTo<wxString>(soundFileSyncComplete_), wxSOUND_ASYNC);
 
-        if (!soundLog.GetMessages().empty())
-            logMsg(errorLog_.ref(), utfTo<std::wstring>(soundLog.GetMessages()), MSG_TYPE_INFO);
     }
     //if (::GetForegroundWindow() != GetHWND())
     //    RequestUserAttention(); -> probably too much since task bar is already colorized with Taskbar::Status::error or Status::normal
diff -ruN a/FreeFileSync/Source/ui/gui_status_handler.cpp b/FreeFileSync/Source/ui/gui_status_handler.cpp
--- a/FreeFileSync/Source/ui/gui_status_handler.cpp	2026-10-05 11:23:23.981591100 +0200
+++ b/FreeFileSync/Source/ui/gui_status_handler.cpp	2026-10-05 11:23:24.001916292 +0200
@@ -508,12 +508,11 @@
     if (!taskCancelled() && !suspend && !autoClose && //only play when actually showing results dialog
         !soundFileSyncComplete_.empty())
     {
-        wxLogCollector soundLog; //wxWidgets shows modal error dialog by default => "no, wxWidgets, NO!"
+        //wxWidgets shows modal error dialog by default => "no, wxWidgets, NO!"
+        wxLog* oldLogTarget = wxLog::SetActiveTarget(new wxLogStderr); //transfer and receive ownership!
+        ZEN_ON_SCOPE_EXIT(delete wxLog::SetActiveTarget(oldLogTarget));
 
         wxSound::Play(utfTo<wxString>(soundFileSyncComplete_), wxSOUND_ASYNC);
-
-        if (!soundLog.GetMessages().empty())
-            logMsg(errorLog_.ref(), utfTo<std::wstring>(soundLog.GetMessages()), MSG_TYPE_INFO);
     }
     //if (::GetForegroundWindow() != GetHWND())
     //    RequestUserAttention(); -> probably too much since task bar is already colorized with Taskbar::Status::error or Status::normal
diff -ruN a/FreeFileSync/Source/ui/main_dlg.cpp b/FreeFileSync/Source/ui/main_dlg.cpp
--- a/FreeFileSync/Source/ui/main_dlg.cpp	2026-10-05 11:23:23.981832066 +0200
+++ b/FreeFileSync/Source/ui/main_dlg.cpp	2026-10-05 11:23:24.002832207 +0200
@@ -4614,12 +4614,11 @@
         //play (optional) sound notification
         if (!globalCfg_.soundFileCompareFinished.empty())
         {
-            wxLogCollector soundLog; //wxWidgets shows modal error dialog by default => "no, wxWidgets, NO!"
+            //wxWidgets shows modal error dialog by default => "no, wxWidgets, NO!"
+            wxLog* oldLogTarget = wxLog::SetActiveTarget(new wxLogStderr); //transfer and receive ownership!
+            ZEN_ON_SCOPE_EXIT(delete wxLog::SetActiveTarget(oldLogTarget));
 
             wxSound::Play(utfTo<wxString>(globalCfg_.soundFileCompareFinished), wxSOUND_ASYNC);
-
-            if (!soundLog.GetMessages().empty())
-                statusHandler.logMessage(utfTo<std::wstring>(soundLog.GetMessages()), PhaseCallback::MsgType::info);
         }
     }
     catch (CancelProcess&) {}
diff -ruN a/wx+/image_resources.cpp b/wx+/image_resources.cpp
--- a/wx+/image_resources.cpp	2026-10-05 11:23:23.987369636 +0200
+++ b/wx+/image_resources.cpp	2026-10-05 11:23:24.007936155 +0200
@@ -9,7 +9,6 @@
 #include <zen/thread.h>
 #include <zen/file_io.h>
 #include <zen/file_traverser.h>
-#include <wx/log.h>
 #include <wx/mstream.h>
 #include <wx/zipstrm.h>
 #include <xBRZ/src/xbrz.h>
@@ -204,8 +203,6 @@
         }
     //--------------------------------------------------------------------
 
-        wxLogCollector zipLog; //wxWidgets shows modal error dialog by default => "no, wxWidgets, NO!"
-
         //wxFFileInputStream/wxZipInputStream loads in junks of 512 bytes => WTF!!! => implement sane file loading:
         wxMemoryInputStream memStream(rawStream.c_str(), rawStream.size()); //does not take ownership
         wxZipInputStream zipStream(memStream, wxConvUTF8);
@@ -222,10 +219,7 @@
                 streams.emplace_back(utfTo<Zstring>(entry->GetName()), std::move(stream));
             }
             else
-                break; //error
-
-        if (zipStream.GetLastError() != wxSTREAM_EOF || !zipLog.GetMessages().empty())
-            throw FileError(replaceCpy(_("Cannot read file %x."), L"%x", fmtPath(zipPath)), utfTo<std::wstring>(zipLog.GetMessages()));
+                assert(false);
     }();
     //--------------------------------------------------------------------
 
