diff -Nurd fritzing-0.9.1b.source.orig/fritzing-app-master/src/program/programtab.cpp fritzing-0.9.1b.source/fritzing-app-master/src/program/programtab.cpp
--- fritzing-0.9.1b.source.orig/fritzing-app-master/src/program/programtab.cpp	2014-12-02 03:46:32.000000000 +0100
+++ fritzing-0.9.1b.source/fritzing-app-master/src/program/programtab.cpp	2015-01-05 15:45:23.508333593 +0100
@@ -751,7 +751,7 @@
 	m_programButton->setEnabled(false);
     m_console->setPlainText("");
     m_platform->upload(this,
-                       m_portComboBox->currentData().toString(),
+                       m_portComboBox->currentText(),
                        m_boardComboBox->currentText(),
                        m_filename);
 }
