Index: invocations-1.4.0/tests/packaging/release.py
===================================================================
--- invocations-1.4.0.orig/tests/packaging/release.py
+++ invocations-1.4.0/tests/packaging/release.py
@@ -5,8 +5,12 @@ from os import path
 import re
 import sys
 
-from invoke.vendor.six import PY2
-from invoke.vendor.lexicon import Lexicon
+try:
+    from invoke.vendor.six import PY2
+    from invoke.vendor.lexicon import Lexicon
+except ImportError:
+    from six import PY2
+    from lexicon import Lexicon
 from invoke import MockContext, Result, Config
 from mock import Mock, patch
 from pytest import skip
Index: invocations-1.4.0/invocations/console.py
===================================================================
--- invocations-1.4.0.orig/invocations/console.py
+++ invocations-1.4.0/invocations/console.py
@@ -6,7 +6,10 @@ from __future__ import unicode_literals,
 
 import sys
 
-from invoke.vendor.six.moves import input
+try:
+    from invoke.vendor.six.moves import input
+except ImportError:
+    from six.moves import input
 
 
 # NOTE: originally cribbed from fab 1's contrib.console.confirm
Index: invocations-1.4.0/invocations/packaging/release.py
===================================================================
--- invocations-1.4.0.orig/invocations/packaging/release.py
+++ invocations-1.4.0/invocations/packaging/release.py
@@ -19,10 +19,14 @@ import sys
 from glob import glob
 from shutil import rmtree
 
-from invoke.vendor.six import StringIO
-
-from invoke.vendor.six import text_type, binary_type, PY2
-from invoke.vendor.lexicon import Lexicon
+try:
+    from invoke.vendor.six import StringIO
+    from invoke.vendor.six import text_type, binary_type, PY2
+    from invoke.vendor.lexicon import Lexicon
+except ImportError:
+    from six import StringIO
+    from six import text_type, binary_type, PY2
+    from lexicon import Lexicon
 
 from blessings import Terminal
 from enum import Enum
Index: invocations-1.4.0/invocations/packaging/semantic_version_monkey.py
===================================================================
--- invocations-1.4.0.orig/invocations/packaging/semantic_version_monkey.py
+++ invocations-1.4.0/invocations/packaging/semantic_version_monkey.py
@@ -5,7 +5,10 @@ We never like monkey-patching, but for n
 or distributing our own fork.
 """
 
-from invoke.vendor.six import text_type
+try:
+    from invoke.vendor.six import text_type
+except ImportError:
+    from six import text_type
 
 from semantic_version import Version
 
Index: invocations-1.4.0/invocations/testing.py
===================================================================
--- invocations-1.4.0.orig/invocations/testing.py
+++ invocations-1.4.0/invocations/testing.py
@@ -1,8 +1,12 @@
 import sys
 import time
 from collections import defaultdict
-from invoke.vendor.six import iteritems
-from invoke.vendor.six.moves import range
+try:
+    from invoke.vendor.six import iteritems
+    from invoke.vendor.six.moves import range
+except ImportError:
+    from six import iteritems
+    from six.moves import range
 
 from invoke import task
 from tqdm import tqdm
