Index: invocations-4.0.2/tests/packaging/release.py
===================================================================
--- invocations-4.0.2.orig/tests/packaging/release.py
+++ invocations-4.0.2/tests/packaging/release.py
@@ -4,7 +4,10 @@ from pathlib import Path
 import re
 import sys
 
-from invoke.vendor.lexicon import Lexicon
+try:
+    from invoke.vendor.lexicon import Lexicon
+except ImportError:
+    from lexicon import Lexicon
 from invoke import MockContext, Result, Config, Exit
 from docutils.utils import Reporter
 from unittest.mock import patch, call
Index: invocations-4.0.2/invocations/packaging/release.py
===================================================================
--- invocations-4.0.2.orig/invocations/packaging/release.py
+++ invocations-4.0.2/invocations/packaging/release.py
@@ -19,7 +19,10 @@ from shutil import rmtree
 from typing import Union
 
 from build._builder import _read_pyproject_toml
-from invoke.vendor.lexicon import Lexicon
+try:
+    from invoke.vendor.lexicon import Lexicon
+except ImportError:
+    from lexicon import Lexicon
 
 from blessings import Terminal
 from docutils.utils import Reporter
