Index: invocations-3.3.0/tests/packaging/release.py
===================================================================
--- invocations-3.3.0.orig/tests/packaging/release.py
+++ invocations-3.3.0/tests/packaging/release.py
@@ -3,7 +3,10 @@ from os import path
 import re
 import sys
 
-from invoke.vendor.lexicon import Lexicon
+try:
+    from invoke.vendor.lexicon import Lexicon
+except ImportError:
+    from lexicon import Lexicon
 from invoke import MockContext, Result, Config, Exit
 from docutils.utils import Reporter
 from unittest.mock import Mock, patch, call
Index: invocations-3.3.0/invocations/packaging/release.py
===================================================================
--- invocations-3.3.0.orig/invocations/packaging/release.py
+++ invocations-3.3.0/invocations/packaging/release.py
@@ -21,7 +21,10 @@ from io import StringIO
 from pathlib import Path
 from shutil import rmtree
 
-from invoke.vendor.lexicon import Lexicon
+try:
+    from invoke.vendor.lexicon import Lexicon
+except ImportError:
+    from lexicon import Lexicon
 
 from blessings import Terminal
 from docutils.utils import Reporter
