diff -upr invocations-2.3.0.orig/tests/autodoc/base.py invocations-2.3.0/tests/autodoc/base.py
--- invocations-2.3.0.orig/tests/autodoc/base.py	2022-05-04 14:22:40.135321552 +0200
+++ invocations-2.3.0/tests/autodoc/base.py	2022-05-04 14:22:40.143321605 +0200
@@ -3,7 +3,7 @@ import codecs
 import re
 import shutil
 
-from mock import Mock, patch
+from unittest.mock import Mock, patch
 
 from invoke import Context
 from invocations.autodoc import setup, TaskDocumenter
diff -upr invocations-2.3.0.orig/tests/console.py invocations-2.3.0/tests/console.py
--- invocations-2.3.0.orig/tests/console.py	2022-05-04 14:22:40.135321552 +0200
+++ invocations-2.3.0/tests/console.py	2022-05-04 14:22:40.143321605 +0200
@@ -2,7 +2,7 @@ from __future__ import unicode_literals
 
 import sys
 
-from mock import patch
+from unittest.mock import patch
 from pytest_relaxed import trap
 
 from invocations.console import confirm
diff -upr invocations-2.3.0.orig/tests/environment.py invocations-2.3.0/tests/environment.py
--- invocations-2.3.0.orig/tests/environment.py	2022-05-04 14:22:40.135321552 +0200
+++ invocations-2.3.0/tests/environment.py	2022-05-04 14:22:40.143321605 +0200
@@ -1,4 +1,4 @@
-from mock import patch
+from unittest.mock import patch
 from pytest import mark
 
 from invocations.environment import in_ci
diff -upr invocations-2.3.0.orig/tests/packaging/release.py invocations-2.3.0/tests/packaging/release.py
--- invocations-2.3.0.orig/tests/packaging/release.py	2022-05-04 14:22:40.135321552 +0200
+++ invocations-2.3.0/tests/packaging/release.py	2022-05-04 14:22:40.143321605 +0200
@@ -13,7 +13,7 @@ except ImportError:
     from lexicon import Lexicon
 from invoke import MockContext, Result, Config, Exit
 from docutils.utils import Reporter
-from mock import Mock, patch, call
+from unittest.mock import Mock, patch, call
 import pytest
 from pytest import skip
 from pytest_relaxed import trap, raises
diff -upr invocations-2.3.0.orig/tests/pytest_.py invocations-2.3.0/tests/pytest_.py
--- invocations-2.3.0.orig/tests/pytest_.py	2022-05-04 14:22:40.135321552 +0200
+++ invocations-2.3.0/tests/pytest_.py	2022-05-04 14:22:40.143321605 +0200
@@ -2,7 +2,7 @@ from contextlib import contextmanager
 
 from invoke import MockContext
 from invocations.pytest import test, coverage
-from mock import Mock, call
+from unittest.mock import Mock, call
 
 
 @contextmanager
