diff -Pdpru eiskaltdcpp-2.4.2.orig/extra/upnpc.cpp eiskaltdcpp-2.4.2/extra/upnpc.cpp
--- eiskaltdcpp-2.4.2.orig/extra/upnpc.cpp	2021-03-03 00:07:33.000000000 +0300
+++ eiskaltdcpp-2.4.2/extra/upnpc.cpp	2026-04-05 17:20:40.421074583 +0300
@@ -53,7 +53,11 @@ bool UPnPc::init()
     if (!devices)
         return false;
 
+#if (MINIUPNPC_API_VERSION >= 18)
+    bool ret = UPNP_GetValidIGD(devices, &urls, &data, nullptr, 0, nullptr, 0);
+#else
     bool ret = UPNP_GetValidIGD(devices, &urls, &data, nullptr, 0);
+#endif
 
     freeUPNPDevlist(devices);
 
