---
 test_settings.py |    2 ++
 1 file changed, 2 insertions(+)

--- a/test_settings.py
+++ b/test_settings.py
@@ -9,6 +9,7 @@ DATABASES = {
 }
 
 ROOT_URLCONF = "test_urls"
+STATIC_URL = '/static/'
 
 TEMPLATES = [
     {
@@ -34,6 +35,7 @@ MIDDLEWARE = [
     "django.contrib.auth.middleware.AuthenticationMiddleware",
     "django.contrib.messages.middleware.MessageMiddleware",
     "django.middleware.clickjacking.XFrameOptionsMiddleware",
+    "allauth.account.middleware.AccountMiddleware",
 ]
 
 EMAIL_BACKEND = "django.core.mail.backends.locmem.EmailBackend"
