From aea2bd6c725eb06da51fd99abcac232cbf857f06 Mon Sep 17 00:00:00 2001
From: saschahofmann <sascha.kelevra@gmail.com>
Date: Wed, 9 Feb 2022 09:02:30 +0100
Subject: [PATCH 1/4] Update import for django 4.0

---
 invitations/adapters.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

Index: django-invitations-1.9.3/invitations/adapters.py
===================================================================
--- django-invitations-1.9.3.orig/invitations/adapters.py
+++ django-invitations-1.9.3/invitations/adapters.py
@@ -9,9 +9,12 @@ from .app_settings import app_settings
 from .utils import import_attribute
 
 try:
-    from django.utils.encoding import force_text
+    from django.utils.encoding import force_str as force_text
 except ImportError:
-    from django.utils.encoding import force_unicode as force_text
+    try:
+        from django.utils.encoding import force_text
+    except ImportError:
+        from django.utils.encoding import force_unicode as force_text
 
 
 # Code credits here to django-allauth
Index: django-invitations-1.9.3/invitations/base_invitation.py
===================================================================
--- django-invitations-1.9.3.orig/invitations/base_invitation.py
+++ django-invitations-1.9.3/invitations/base_invitation.py
@@ -1,6 +1,6 @@
 from django.conf import settings
 from django.db import models
-from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import gettext_lazy as _
 
 from .managers import BaseInvitationManager
 
Index: django-invitations-1.9.3/invitations/forms.py
===================================================================
--- django-invitations-1.9.3.orig/invitations/forms.py
+++ django-invitations-1.9.3/invitations/forms.py
@@ -1,6 +1,6 @@
 from django import forms
 from django.contrib.auth import get_user_model
-from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import gettext_lazy as _
 
 from .adapters import get_invitations_adapter
 from .exceptions import AlreadyAccepted, AlreadyInvited, UserRegisteredEmail
Index: django-invitations-1.9.3/invitations/models.py
===================================================================
--- django-invitations-1.9.3.orig/invitations/models.py
+++ django-invitations-1.9.3/invitations/models.py
@@ -9,7 +9,7 @@ except ImportError:
 from django.db import models
 from django.utils import timezone
 from django.utils.crypto import get_random_string
-from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import gettext_lazy as _
 
 from . import signals
 from .adapters import get_invitations_adapter
Index: django-invitations-1.9.3/invitations/views.py
===================================================================
--- django-invitations-1.9.3.orig/invitations/views.py
+++ django-invitations-1.9.3/invitations/views.py
@@ -7,7 +7,7 @@ from django.core.validators import valid
 from django.http import Http404, HttpResponse
 from django.shortcuts import redirect
 from django.utils.decorators import method_decorator
-from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import gettext_lazy as _
 from django.views.generic import FormView, View
 from django.views.generic.detail import SingleObjectMixin
 
Index: django-invitations-1.9.3/invitations/signals.py
===================================================================
--- django-invitations-1.9.3.orig/invitations/signals.py
+++ django-invitations-1.9.3/invitations/signals.py
@@ -1,7 +1,7 @@
 from django.dispatch import Signal
 
-invite_url_sent = Signal(providing_args=['invite_url_sent', 'inviter'])
-invite_accepted = Signal(providing_args=['email'])
+invite_url_sent = Signal()
+invite_accepted = Signal()
 
 """
 @receiver(invite_url_sent, sender=Invitation)
Index: django-invitations-1.9.3/invitations/urls.py
===================================================================
--- django-invitations-1.9.3.orig/invitations/urls.py
+++ django-invitations-1.9.3/invitations/urls.py
@@ -1,15 +1,15 @@
-from django.conf.urls import url
+from django.urls import re_path
 
 from . import views
 
 app_name = 'invitations'
 urlpatterns = [
-    url(r'^send-invite/$', views.SendInvite.as_view(),
+    re_path(r'^send-invite/$', views.SendInvite.as_view(),
         name='send-invite'),
 
-    url(r'^send-json-invite/$', views.SendJSONInvite.as_view(),
+    re_path(r'^send-json-invite/$', views.SendJSONInvite.as_view(),
         name='send-json-invite'),
 
-    url(r'^accept-invite/(?P<key>\w+)/?$', views.AcceptInvite.as_view(),
+    re_path(r'^accept-invite/(?P<key>\w+)/?$', views.AcceptInvite.as_view(),
         name='accept-invite'),
 ]
Index: django-invitations-1.9.3/test_urls.py
===================================================================
--- django-invitations-1.9.3.orig/test_urls.py
+++ django-invitations-1.9.3/test_urls.py
@@ -1,15 +1,15 @@
-from django.conf.urls import url, include
 from django.contrib import admin
 from django.conf import settings
+from django.urls import include, re_path
 
 admin.autodiscover()
 
 urlpatterns = [
-    url(r'^invitations/', include('invitations.urls')),
-    url(r'^admin/', admin.site.urls),
+    re_path(r'^invitations/', include('invitations.urls')),
+    re_path(r'^admin/', admin.site.urls),
 ]
 
 if 'allauth' in settings.INSTALLED_APPS:
     urlpatterns.append(
-        url(r'^accounts/', include('allauth.urls'))
+        re_path(r'^accounts/', include('allauth.urls'))
     )
