diff -upr django-invitations-1.9.3.orig/tests/basic/tests.py django-invitations-1.9.3/tests/basic/tests.py
--- django-invitations-1.9.3.orig/tests/basic/tests.py	2022-05-09 12:25:28.995331381 +0200
+++ django-invitations-1.9.3/tests/basic/tests.py	2022-05-09 12:25:28.999331406 +0200
@@ -1,7 +1,7 @@
 import datetime
 import re
 import json
-from mock import patch
+from unittest.mock import patch
 
 from django.test import Client
 from django.test.client import RequestFactory
