--- plette-0.2.2/tests/test_lockfiles.py.orig	2019-03-02 18:27:26.997699977 +0700
+++ plette-0.2.2/tests/test_lockfiles.py	2019-03-02 18:56:31.174560822 +0700
@@ -111,7 +111,10 @@
         "verify_ssl": True,
     })
 
-    assert lockfile.meta.hash._data == {"sha256": pipfile_hash_value}
+    assert lockfile.meta.hash._data in [
+        {"sha256": pipfile_hash_value},
+        {"SHA256": pipfile_hash_value},
+    ]
     assert lockfile.meta.requires._data == {"python_version": "3.7"}
     assert lockfile.meta.sources._data == [
         {
--- plette-0.2.2/tests/test_models.py.orig	2019-03-02 18:55:33.342133583 +0700
+++ plette-0.2.2/tests/test_models.py	2019-03-02 18:56:01.902344580 +0700
@@ -13,7 +13,7 @@
 def test_hash_from_hash():
     v = hashlib.md5(b"foo")
     h = models.Hash.from_hash(v)
-    assert h.name == "md5"
+    assert h.name in ["md5", "MD5"]
     assert h.value == "acbd18db4cc2f85cedef654fccc4a4d8"
 
 
