--- Makefile.smb	2024-09-20 13:30:10.759178569 -0400
+++ Makefile	2024-09-20 13:30:57.670165519 -0400
@@ -7,7 +7,7 @@
 # This must remain the first target in this file, which is what 'make' with no
 # arguments will run.
 build:
-	go build github.com/johnkerl/miller/v6/cmd/mlr
+	go build -buildmode=pie github.com/johnkerl/miller/v6/cmd/mlr
 	@echo "Build complete. The Miller executable is ./mlr (or .\mlr.exe on Windows)."
 	@echo "You can use 'make check' to run tests".
 
