diff -urB ksshaskpass/CMakeLists.txt new/CMakeLists.txt
--- ksshaskpass/CMakeLists.txt	2014-11-14 14:43:03.239674828 +0100
+++ new/CMakeLists.txt	2014-11-14 14:55:55.783702339 +0100
@@ -44,7 +44,7 @@ target_link_libraries(ksshaskpass
     KF5::WidgetsAddons
 )
 
-install(TARGETS ksshaskpass DESTINATION ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS ksshaskpass DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/ssh)
 
 if (KF5DocTools_FOUND)
     add_subdirectory(doc)
