Index: libraries/lib.inc.php
===================================================================
--- libraries/lib.inc.php.orig
+++ libraries/lib.inc.php
@@ -27,12 +27,12 @@
 		exit(sprintf('Version of PHP not supported. Please upgrade to version %s or later.', $phpMinVer));
 
 	// Check to see if the configuration file exists, if not, explain
-	if (file_exists('conf/config.inc.php')) {
+	if (file_exists('/etc/phpPgAdmin/config.inc.php')) {
 		$conf = array();
-		include('./conf/config.inc.php');
+		include('/etc/phpPgAdmin/config.inc.php');
 	}
 	else {
-		echo 'Configuration error: Copy conf/config.inc.php-dist to conf/config.inc.php and edit appropriately.';
+		echo 'Configuration error: Can not include config /etc/phpPgAdmin/config.inc.php.';
 		exit;
 	}
 
