--- openjdk/hotspot/src/os/linux/vm/os_linux.cpp	Wed Oct 23 15:44:12 2013 -0700
+++ openjdk/hotspot/src/os/linux/vm/os_linux.cpp	Thu Dec 19 16:03:33 2013 +0000
@@ -4797,9 +4797,19 @@ 
   // size.  Add a page for compiler2 recursion in main thread.
   // Add in 2*BytesPerWord times page size to account for VM stack during
   // class initialization depending on 32 or 64 bit VM.
+
+
   os::Linux::min_stack_allowed = MAX2(os::Linux::min_stack_allowed,
             (size_t)(StackYellowPages+StackRedPages+StackShadowPages) * Linux::page_size() +
-                    (2*BytesPerWord COMPILER2_PRESENT(+1)) * Linux::vm_default_page_size());
+                    (2*BytesPerWord COMPILER2_PRESENT(+1)) 
+                       * 
+#ifdef PPC
+                      NOT_ZERO ( Linux::vm_default_page_size() ) 
+                       ZERO_ONLY ( Linux::page_size() )
+#else                   
+                       ( Linux::vm_default_page_size() )
+#endif
+                     );
 
   size_t threadStackSizeInBytes = ThreadStackSize * K;
   if (threadStackSizeInBytes != 0 &&
