Index: pysnmp-5.0.28/pysnmp/proto/api/verdec.py
===================================================================
--- pysnmp-5.0.28.orig/pysnmp/proto/api/verdec.py
+++ pysnmp-5.0.28/pysnmp/proto/api/verdec.py
@@ -6,19 +6,20 @@
 #
 from pyasn1.type import univ
 from pyasn1.codec.ber import decoder, eoo
+from pyasn1.codec.streaming import readFromStream
 from pyasn1.error import PyAsn1Error
 from pysnmp.proto.error import ProtocolError
 
 
 def decodeMessageVersion(wholeMsg):
     try:
-        seq, wholeMsg = decoder.decode(
+        wholeMsg, seq = decoder.decode(
             wholeMsg, asn1Spec=univ.Sequence(),
-            recursiveFlag=False, substrateFun=lambda a, b, c: (a, b[:c])
+            recursiveFlag=False, substrateFun=lambda a, b, c, d: readFromStream(b, c)
         )
         ver, wholeMsg = decoder.decode(
             wholeMsg, asn1Spec=univ.Integer(),
-            recursiveFlag=False, substrateFun=lambda a, b, c: (a, b[:c])
+            recursiveFlag=False, substrateFun=lambda a, b, c, d: readFromStream(b, c)
         )
         if eoo.endOfOctets.isSameTypeWith(ver):
             raise ProtocolError('EOO at SNMP version component')
Index: pysnmp-5.0.28/pyproject.toml
===================================================================
--- pysnmp-5.0.28.orig/pyproject.toml
+++ pysnmp-5.0.28/pyproject.toml
@@ -27,7 +27,7 @@ include = ["docs", "tests", "examples"]
 
 [tool.poetry.dependencies]
 python = "^3.7"
-pyasn1 = ">=0.4.8, <0.5.0"
+pyasn1 = "^0.5.0"
 pysnmpcrypto = "^0.0.4"
 pysmi-lextudio = "^1.0.4"
 
