Index: Clementine-1.3.92/src/CMakeLists.txt
===================================================================
--- Clementine-1.3.92.orig/src/CMakeLists.txt	2019-12-30 22:30:22.000000000 +0200
+++ Clementine-1.3.92/src/CMakeLists.txt	2020-06-02 11:29:58.834577729 +0200
@@ -976,6 +976,8 @@ if(UNIX AND HAVE_DBUS)
   list(APPEND HEADERS ${CMAKE_CURRENT_BINARY_DIR}/dbus/avahientrygroup.h)
   list(APPEND SOURCES ${CMAKE_CURRENT_BINARY_DIR}/dbus/avahientrygroup.cpp)
 
+  set_source_files_properties( dbus/org.freedesktop.UDisks.xml dbus/org.freedesktop.UDisks.Device.xml PROPERTIES NO_NAMESPACE TRUE)
+
   # DeviceKit DBUS interfaces
   if(HAVE_DEVICEKIT)
     set_source_files_properties(dbus/org.freedesktop.UDisks.xml
