Index: b/dist/clementine.desktop
===================================================================
--- a/dist/clementine.desktop	2021-04-07 16:29:25.956516313 +0200
+++ b/dist/clementine.desktop	2021-04-07 16:29:43.073173522 +0200
@@ -36,7 +36,7 @@ TryExec=clementine
 Icon=clementine
 Terminal=false
 Categories=AudioVideo;Player;Qt;Audio;
-StartupNotify=false
+StartupNotify=true
 MimeType=application/ogg;application/x-ogg;application/x-ogm-audio;audio/aac;audio/mp4;audio/mpeg;audio/mpegurl;audio/ogg;audio/vnd.rn-realaudio;audio/vorbis;audio/x-flac;audio/x-mp3;audio/x-mpeg;audio/x-mpegurl;audio/x-ms-wma;audio/x-musepack;audio/x-oggflac;audio/x-pn-realaudio;audio/x-scpls;audio/x-speex;audio/x-vorbis;audio/x-vorbis+ogg;audio/x-wav;video/x-ms-asf;x-content/audio-player;x-scheme-handler/zune;x-scheme-handler/itpc;x-scheme-handler/itms;x-scheme-handler/feed;
 Actions=Play;Pause;Stop;StopAfterCurrent;Previous;Next;
 
