diff -uNr mysql-workbench-community-8.0.19-org/backend/wbprivate/CMakeLists.txt mysql-workbench-community-8.0.19-src/backend/wbprivate/CMakeLists.txt
--- mysql-workbench-community-8.0.19-org/backend/wbprivate/CMakeLists.txt	2019-12-15 21:40:32.000000000 +0200
+++ mysql-workbench-community-8.0.19-src/backend/wbprivate/CMakeLists.txt	2020-04-05 14:48:36.731862692 +0300
@@ -112,6 +112,7 @@
  SYSTEM INTERFACE
     ${VSQLITE_INCLUDE_DIRS}
     ${GDAL_INCLUDE_DIRS}
+    ${SCINTILLA_INCLUDE_DIRS}
 )
 
 add_dependencies(wbprivate generate_parser_mysql)
diff -uNr mysql-workbench-community-8.0.19-org/backend/wbpublic/CMakeLists.txt mysql-workbench-community-8.0.19-src/backend/wbpublic/CMakeLists.txt
--- mysql-workbench-community-8.0.19-org/backend/wbpublic/CMakeLists.txt	2019-12-15 21:40:32.000000000 +0200
+++ mysql-workbench-community-8.0.19-src/backend/wbpublic/CMakeLists.txt	2020-04-05 14:48:36.735862665 +0300
@@ -157,6 +157,7 @@
     ${GLIB_INCLUDE_DIRS}
     ${MySQLCppConn_INCLUDE_DIRS}
     ${GDAL_INCLUDE_DIRS}
+    ${SCINTILLA_INCLUDE_DIRS}
 )
 
 
@@ -181,6 +182,7 @@
 target_include_directories(wbpublic_Iface
  SYSTEM INTERFACE
     $<BUILD_INTERFACE:${VSQLITE_INCLUDE_DIR}>
+    scintilla
 )
 
 
diff -uNr mysql-workbench-community-8.0.19-org/CMakeLists.txt mysql-workbench-community-8.0.19-src/CMakeLists.txt
--- mysql-workbench-community-8.0.19-org/CMakeLists.txt	2019-12-15 21:40:32.000000000 +0200
+++ mysql-workbench-community-8.0.19-src/CMakeLists.txt	2020-04-05 14:48:36.735862665 +0300
@@ -143,7 +143,7 @@
 check_function_exists (strtoimax HAVE_STRTOIMAX)
 check_function_exists (strtoumax HAVE_STRTOUMAX)
 
-set(ANTLR_JAR_FILENAME "antlr-4.9.1-complete.jar")
+set(ANTLR_JAR_FILENAME "antlr4.jar")
 get_filename_component(SOURCE_PARENT_DIR ${CMAKE_SOURCE_DIR} DIRECTORY)
 set(LINUX_RES_BIN_DIR ${SOURCE_PARENT_DIR}/linux-res/bin)
 message("WITH_ANTLR_JAR: ${WITH_ANTLR_JAR}")
@@ -161,6 +161,9 @@
   message(WARNING ${ANTLR_JAR_FILENAME} " or java runtime wasn't found which is required if you'd like to rebuild Lexer and Parser files")
 endif()
 
+if (UNIX)
+  set(SCINTILLA_INCLUDE_DIRS /usr/include/scintilla)
+endif()
 
 if (UNIX)
 	if (NOT LIB_INSTALL_DIR)
diff -uNr mysql-workbench-community-8.0.19-org/ext/CMakeLists.txt mysql-workbench-community-8.0.19-src/ext/CMakeLists.txt
--- mysql-workbench-community-8.0.19-org/ext/CMakeLists.txt	2019-12-15 21:40:32.000000000 +0200
+++ mysql-workbench-community-8.0.19-src/ext/CMakeLists.txt	2020-04-05 14:48:36.735862665 +0300
@@ -1 +0,0 @@
-add_subdirectory(scintilla/gtk)
diff -uNr mysql-workbench-community-8.0.19-org/frontend/linux/linux_utilities/CMakeLists.txt mysql-workbench-community-8.0.19-src/frontend/linux/linux_utilities/CMakeLists.txt
--- mysql-workbench-community-8.0.19-org/frontend/linux/linux_utilities/CMakeLists.txt	2019-12-15 21:40:32.000000000 +0200
+++ mysql-workbench-community-8.0.19-src/frontend/linux/linux_utilities/CMakeLists.txt	2020-04-05 14:49:01.407717175 +0300
@@ -19,6 +19,11 @@
     ${PROJECT_SOURCE_DIR}/generated
 )
 
+target_include_directories(linux_utilities
+ SYSTEM
+  PRIVATE
+    ${SCINTILLA_INCLUDE_DIRS}
+)
 
 
 target_compile_options(linux_utilities PRIVATE $<$<CXX_COMPILER_ID:GNU>:-rdynamic>)
@@ -32,7 +37,7 @@
     wbpublic::wbpublic
     mforms::mforms
     parsers::parsers
-    wbscintilla::wbscintilla
+    scintilla
 )
 
 add_library(linux_utilities_Iface INTERFACE)
diff -uNr mysql-workbench-community-8.0.19-org/frontend/linux/sqlide/CMakeLists.txt mysql-workbench-community-8.0.19-src/frontend/linux/sqlide/CMakeLists.txt
--- mysql-workbench-community-8.0.19-org/frontend/linux/sqlide/CMakeLists.txt	2019-12-15 21:40:32.000000000 +0200
+++ mysql-workbench-community-8.0.19-src/frontend/linux/sqlide/CMakeLists.txt	2020-04-05 14:48:36.735862665 +0300
@@ -18,6 +18,7 @@
   PRIVATE
     ${VSQLITE_INCLUDE_DIRS}
     ${GTKMM_INCLUDE_DIRS}
+    ${SCINTILLA_INCLUDE_DIRS}
 )
 
 
diff -uNr mysql-workbench-community-8.0.19-org/library/forms/CMakeLists.txt mysql-workbench-community-8.0.19-src/library/forms/CMakeLists.txt
--- mysql-workbench-community-8.0.19-org/library/forms/CMakeLists.txt	2019-12-15 21:40:32.000000000 +0200
+++ mysql-workbench-community-8.0.19-src/library/forms/CMakeLists.txt	2020-04-05 14:48:36.735862665 +0300
@@ -103,6 +103,7 @@
  SYSTEM
   PRIVATE
     ${LIBSECRET_INCLUDE_DIRS}
+    ${SCINTILLA_INCLUDE_DIRS}
 )
 
 
@@ -140,7 +141,7 @@
 target_link_libraries(mforms_Iface INTERFACE
     ${LIBSECRET_LIBRARIES}
     wbbase
-    wbscintilla
+    scintilla
 )
 
 
diff -uNr mysql-workbench-community-8.0.19-org/library/forms/swig/CMakeLists.txt mysql-workbench-community-8.0.19-src/library/forms/swig/CMakeLists.txt
--- mysql-workbench-community-8.0.19-org/library/forms/swig/CMakeLists.txt	2019-12-15 21:40:32.000000000 +0200
+++ mysql-workbench-community-8.0.19-src/library/forms/swig/CMakeLists.txt	2020-04-05 14:48:36.735862665 +0300
@@ -95,7 +95,7 @@
 else()
   swig_add_library(mforms LANGUAGE python SOURCES mforms.i)
 endif()
-swig_link_libraries(mforms Boost::boost mforms wbscintilla wbbase::wbbase wbpublic::wbpublic grt::grt ${GTK3_LIBRARIES} ${LIBXML2_LIBRARIES})
+swig_link_libraries(mforms Boost::boost mforms scintilla wbbase::wbbase wbpublic::wbpublic grt::grt ${GTK3_LIBRARIES} ${LIBXML2_LIBRARIES})
 set_source_files_properties(${swig_generated_file_fullname} PROPERTIES COMPILE_FLAGS ${SOURCE_FILE_COMPILE_FLAGS})
 
 target_compile_options(_mforms PRIVATE ${WB_CXXFLAGS})
@@ -110,6 +110,7 @@
  SYSTEM
   PRIVATE
     ${PYTHON_INCLUDE_DIRS}
+    ${SCINTILLA_INCLUDE_DIRS}
 )
 
 
diff -uNr mysql-workbench-community-8.0.19-org/plugins/db.mysql.diff.reporting/CMakeLists.txt mysql-workbench-community-8.0.19-src/plugins/db.mysql.diff.reporting/CMakeLists.txt
--- mysql-workbench-community-8.0.19-org/plugins/db.mysql.diff.reporting/CMakeLists.txt	2019-12-15 21:40:32.000000000 +0200
+++ mysql-workbench-community-8.0.19-src/plugins/db.mysql.diff.reporting/CMakeLists.txt	2020-04-05 14:48:36.735862665 +0300
@@ -19,7 +19,7 @@
 )
 
 target_link_libraries(db.mysql.diff.reporting.wbp 
-  PRIVATE wbpublic grt mforms wbscintilla cdbc db.mysql.wbp)
+  PRIVATE wbpublic grt mforms scintilla cdbc db.mysql.wbp)
 
 if(BUILD_FOR_GCOV)
   target_link_libraries(db.mysql.diff.reporting.wbp PRIVATE gcov)
diff -uNr mysql-workbench-community-8.0.19-org/plugins/wb.model.editors/linux/CMakeLists.txt mysql-workbench-community-8.0.19-src/plugins/wb.model.editors/linux/CMakeLists.txt
--- mysql-workbench-community-8.0.19-org/plugins/wb.model.editors/linux/CMakeLists.txt	2019-12-15 21:40:32.000000000 +0200
+++ mysql-workbench-community-8.0.19-src/plugins/wb.model.editors/linux/CMakeLists.txt	2020-04-05 14:48:36.735862665 +0300
@@ -25,6 +25,11 @@
     ${PROJECT_SOURCE_DIR}/modules
 )
 
+target_include_directories(wb.model.editors.wbp
+ SYSTEM
+  PRIVATE
+    ${SCINTILLA_INCLUDE_DIRS}
+)
 
 target_compile_options(wb.model.editors.wbp PRIVATE ${WB_CXXFLAGS})
 
