diff --git a/configure b/configure
index 442b0d0..0b359f4 100755
--- a/configure
+++ b/configure
@@ -742,8 +742,7 @@ echo "pixmapdir  = $PIXMAPDIR" >> configmk.tmp
 echo "desktopdir = $DESKTOPDIR" >> configmk.tmp
 if [ $OS = unix ]; then
 	echo >> configmk.tmp
-	echo 'datadir = $(datarootdir)/allegro' >> configmk.tmp
-	echo '#datadir = $(datarootdir)/allegro/vitetris' >> configmk.tmp
+	echo 'datadir = $(datarootdir)/vitetris' >> configmk.tmp
 else
 	echo 'datadir = $(prefix)' >> configmk.tmp
 fi
diff --git a/src/textgfx/allegro.c b/src/textgfx/allegro.c
index 68066bc..9dd75ed 100644
--- a/src/textgfx/allegro.c
+++ b/src/textgfx/allegro.c
@@ -105,10 +105,7 @@ static void load_pc8x16_font()
 		set_allegro_resource_path(5, fname);
 	}
 #ifdef UNIX
-	set_allegro_resource_path(4, "/usr/share/allegro/vitetris");
-	set_allegro_resource_path(3, "/usr/share/allegro");
-	set_allegro_resource_path(2, "/usr/local/share/allegro/vitetris");
-	set_allegro_resource_path(1, "/usr/local/share/allegro");
+	set_allegro_resource_path(1, "/usr/share/vitetris");
 #endif
 	if (find_allegro_resource(fname, "pc8x16.fnt", 0,0,0,0,0,
 							sizeof(fname)) == 0)
