Index: include/osdetection
===================================================================
--- include/osdetection.orig
+++ include/osdetection
@@ -400,7 +400,7 @@
                 OS_NAME="${LINUX_VERSION}"
             fi
             # If Linux version (full name) is unknown, use uname value
-            if [ "${OS_FULLNAME}" = "" ]; then OS_FULLNAME=$(uname -s -r); fi
+            #if [ "${OS_FULLNAME}" = "" ]; then OS_FULLNAME=$(uname -s -r); fi
             SYSCTL_READKEY="sysctl -n"
         ;;
 
