diff --git a/src/gui/libcloudproviders/libcloudproviders.cmake b/src/gui/libcloudproviders/libcloudproviders.cmake
index cfb86b8..30e73da 100644
--- a/src/gui/libcloudproviders/libcloudproviders.cmake
+++ b/src/gui/libcloudproviders/libcloudproviders.cmake
@@ -32,7 +32,7 @@ if(WITH_LIBCLOUDPROVIDERS)
     configure_file(libcloudproviders/cloud-provider.ini.in ${CMAKE_CURRENT_BINARY_DIR}/${APPLICATION_CLOUDPROVIDERS_DBUS_NAME}.ini)
     install(
         FILES ${CMAKE_CURRENT_BINARY_DIR}/${APPLICATION_CLOUDPROVIDERS_DBUS_NAME}.ini
-        DESTINATION "${DATADIR}/cloud-providers")
+        DESTINATION "${DATA_INSTALL_DIR}/cloud-providers")
 
     message("Building with libcloudproviders")
 elseif(UNIX AND NOT APPLE)
