diff -upr kakasi-2.3.6.orig/src/jj2.c kakasi-2.3.6/src/jj2.c
--- kakasi-2.3.6.orig/src/jj2.c	2025-06-19 08:17:47.991893099 +0000
+++ kakasi-2.3.6/src/jj2.c	2025-06-19 09:28:56.547581582 +0000
@@ -201,7 +201,7 @@ static void
 J2convert(m, n, proc)
      Character *m;
      Character *n;
-     int (*proc)();
+     int (*proc)(Character *c, Character *n);
 {
     int mp=0, np=0;
     int ret;
diff -upr kakasi-2.3.6.orig/src/kakasi.c kakasi-2.3.6/src/kakasi.c
--- kakasi-2.3.6.orig/src/kakasi.c	2025-06-19 08:17:47.992576362 +0000
+++ kakasi-2.3.6/src/kakasi.c	2025-06-19 09:27:47.585376192 +0000
@@ -97,7 +97,7 @@ Character right_paren[KAKASIBUF];
 
 #ifdef LIBRARY
 extern FILE *kanwadict;
-static int (*proc[8])()={NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL};
+static int (*proc[8])(Character *c, Character *n)={NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL};
 /* ASCII, JISROMAN, KATAKANA, GRAPHIC, ZENKAKU-KIGOU, ZENKAKU-KATAKANA, ZENKAKU-HIRAGANA, KANJI, */
 #endif
 
@@ -121,7 +121,7 @@ KAKASI_ATTR void digest_start_copy PARAM
 KAKASI_ATTR void put_separator PARAMS((void));
 KAKASI_ATTR void putchars PARAMS((Character *results));
 KAKASI_ATTR void digest_out PARAMS((Character *c, int ret));
-KAKASI_ATTR int digest PARAMS((Character *c, int clen, Character *r, int rlen, int type, int (*proc)(void)));
+KAKASI_ATTR int digest PARAMS((Character *c, int clen, Character *r, int rlen, int type, int (*proc)(Character *c, Character *n)));
 KAKASI_ATTR void digest_shift PARAMS((Character *c, int s));
 #ifdef LIBRARY
 static void free_jisyo PARAMS((void));
@@ -240,7 +240,7 @@ digest(c, clen, r, rlen, type, proc)
      Character *r;
      int rlen;
      int type;
-     int (*proc)();
+     int (*proc)(Character *c, Character *n);
 {
     int ret, i, j, k;
     Character new;
@@ -337,7 +337,7 @@ kakasi_getopt_argv(argc, argv)
 #ifndef LIBRARY
     Character c[KAKASIBUF], r[KAKASIBUF];
     int clen, ptype, pctype;
-    static int (*proc[8])()={NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL};
+    static int (*proc[8])(Character *c, Character *n)={NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL};
     /* ASCII, JISROMAN, KATAKANA, GRAPHIC, ZENKAKU-KIGOU, ZENKAKU-KATAKANA, ZENKAKU-HIRAGANA, KANJI, */
 #endif

diff -upr kakasi-2.3.6.orig/lib/libkakasi.h kakasi-2.3.6/lib/libkakasi.h
--- kakasi-2.3.6.orig/lib/libkakasi.h	2025-06-19 09:38:26.229459429 +0000
+++ kakasi-2.3.6/lib/libkakasi.h	2025-06-19 09:43:30.076519441 +0000
@@ -38,12 +38,12 @@ int kakasi_free PARAMS((char *p));
 /*
  * NOTE: these are internal functions. Do not use.
  */
-void digest_start_copy();
-void put_separator();
-void putchars();
-void digest_out();
-int digest();
-void digest_shift();
+void digest_start_copy PARAMS((Character *c, Character *r));
+void put_separator PARAMS((void));
+void putchars PARAMS((Character *results));
+void digest_out PARAMS((Character *c, int ret));
+int digest PARAMS((Character *c, int clen, Character *r, int rlen, int type, int (*proc)(Character *c, Character *n)));
+void digest_shift PARAMS((Character *c, int s));
 
 void setcharbuffer PARAMS((unsigned char *s));
 char *getpbstr PARAMS((void));
 
