diff --git a/lib/libkakasi.h b/lib/libkakasi.h
index 54cebe0..68685ad 100644
--- a/lib/libkakasi.h
+++ b/lib/libkakasi.h
@@ -38,6 +38,11 @@ int kakasi_free PARAMS((char *p));
 /*
  * NOTE: these are internal functions. Do not use.
  */
+typedef struct character {
+    char type;
+    unsigned char c1, c2;
+} Character;
+
 void digest_start_copy PARAMS((Character *c, Character *r));
 void put_separator PARAMS((void));
 void putchars PARAMS((Character *results));
