--- hmconv.c	2001-01-11 15:59:43.000000000 +0100
+++ hmconv.c.new	2005-04-12 17:40:34.000000000 +0200
@@ -137,7 +137,7 @@
 int argc;
 char **argv;
 #else
-main (int argc, char **argv)
+int main (int argc, char **argv)
 #endif
 
 {
@@ -219,7 +219,7 @@
 
       if ( !ishangul) {     /* KSC 5601 doesn't appear, yet */
          fputs((char *) line,out);   /* no conversion */
-         return;
+         return (0);
       }
 
 
