Index: gettext-0.21.1/libtextstyle/configure.ac
===================================================================
--- gettext-0.21.1.orig/libtextstyle/configure.ac
+++ gettext-0.21.1/libtextstyle/configure.ac
@@ -17,11 +17,10 @@ dnl along with this program.  If not, se
 dnl Process this file with autoconf to produce a configure script.
 
 AC_PREREQ([2.64])
-AC_INIT
+AC_INIT([libtextstyle],
+    m4_esyscmd([source ./version.sh; echo $VERSION_NUMBER | tr -d '\n']))
 AC_CONFIG_SRCDIR([version.sh])
 AC_CONFIG_AUX_DIR([build-aux])
-. $srcdir/version.sh
-gl_INIT_PACKAGE([libtextstyle], [$VERSION_NUMBER])
 AM_INIT_AUTOMAKE([1.13 silent-rules])
 AC_CONFIG_HEADERS([config.h])
 
