diff -uprN mono-4.2.3.old/scripts/mono-find-provides.in mono-4.2.3/scripts/mono-find-provides.in
--- mono-4.2.3.old/scripts/mono-find-provides.in	2016-03-15 14:31:53.000000000 +0300
+++ mono-4.2.3/scripts/mono-find-provides.in	2016-06-01 09:21:13.028703810 +0300
@@ -24,7 +24,7 @@ libdir=$prefix/@reloc_libdir@
 bindir=$prefix/bin
 
 # Bail out if monodis or libmono is missing
-if [ ! -x $bindir/monodis ] || [ ! -f $libdir/libmono-2.0.so.1 ] ; then
+if [ ! -x $bindir/monodis ] ; then
 	echo "monodis missing or unusable, exiting..." 1>&2
 	exit 1
 fi
diff -uprN mono-4.2.3.old/scripts/mono-find-requires.in mono-4.2.3/scripts/mono-find-requires.in
--- mono-4.2.3.old/scripts/mono-find-requires.in	2016-03-15 14:31:53.000000000 +0300
+++ mono-4.2.3/scripts/mono-find-requires.in	2016-06-01 09:21:25.876703594 +0300
@@ -28,7 +28,7 @@ libdir=$prefix/@reloc_libdir@
 bindir=$prefix/bin
 
 # Bail out if monodis or libmono is missing
-if [ ! -x $bindir/monodis ] || [ ! -f $libdir/libmono-2.0.so.1 ] ; then
+if [ ! -x $bindir/monodis ] ; then
 	echo "monodis missing or unusable, exiting..." 1>&2
 	exit 1
 fi
