diff -uprN mono.bak/Makefile.am mono-6.4.0.198/Makefile.am
--- mono.bak/Makefile.am	2019-11-20 21:36:02.812573696 +0100
+++ mono-6.4.0.198/Makefile.am	2019-11-20 21:36:28.188845471 +0100
@@ -178,3 +178,6 @@ update-solution-files:
 
 update-solution-files-with-tests:
 	$(MAKE) "GENPROJ_ARGS=2012 true true" update-solution-files
+
+confighfixdir = $(includedir)/mono-2.0
+dist_confighfix_DATA = config.h
diff -uprN mono.bak/mono/metadata/Makefile.am mono-6.4.0.198/mono/metadata/Makefile.am
--- mono.bak/mono/metadata/Makefile.am	2019-11-20 21:35:47.512409834 +0100
+++ mono-6.4.0.198/mono/metadata/Makefile.am	2019-11-20 21:36:28.192845514 +0100
@@ -461,6 +461,7 @@ libmonoruntimeinclude_HEADERS = \
 	attrdefs.h		\
 	blob.h			\
 	class.h			\
+	debug-internals.h	\
 	debug-helpers.h		\
 	debug-mono-symfile.h	\
 	environment.h		\
diff -uprN mono.bak/mono/utils/Makefile.am mono-6.4.0.198/mono/utils/Makefile.am
--- mono.bak/mono/utils/Makefile.am	2019-11-20 21:35:47.552410263 +0100
+++ mono-6.4.0.198/mono/utils/Makefile.am	2019-11-20 21:36:28.192845514 +0100
@@ -328,6 +328,7 @@ libmonoutilsinclude_HEADERS = 	\
 	mono-publib.h		\
 	mono-jemalloc.h		\
 	mono-dl-fallback.h	\
+	mono-compiler.h		\
 	mono-counters.h
 
 EXTRA_DIST = mono-errno.h mono-embed.h mono-embed.c ../../support/libm/complex.c mono-experiments.def
