diff -uprN mono-5.0.0.old/mcs/class/Microsoft.Build.Tasks/Microsoft.Build.Tasks/Csc.cs mono-5.0.0/mcs/class/Microsoft.Build.Tasks/Microsoft.Build.Tasks/Csc.cs
--- mono-5.0.0.old/mcs/class/Microsoft.Build.Tasks/Microsoft.Build.Tasks/Csc.cs	2017-05-03 18:45:38.000000000 +0300
+++ mono-5.0.0/mcs/class/Microsoft.Build.Tasks/Microsoft.Build.Tasks/Csc.cs	2017-05-05 05:05:48.901776264 +0300
@@ -219,7 +219,7 @@ namespace Microsoft.Build.Tasks {
 
 		protected override string ToolName {
 			get {
-				return MSBuildUtils.RunningOnWindows ? "csc.bat" : "csc.exe";
+				return MSBuildUtils.RunningOnWindows ? "mcs.bat" : "mcs.exe";
 			}
 		}
 
