---
 src/xrdb.c |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

--- src/xrdb.c
+++ src/xrdb.c	2016-09-19 09:10:35.504833294 +0000
@@ -39,6 +39,9 @@ along with GNU Emacs.  If not, see <http
 #include <X11/X.h>
 #include <X11/Xutil.h>
 #include <X11/Xresource.h>
+#ifdef USE_X_TOOLKIT
+#include <X11/Intrinsic.h>
+#endif
 #ifdef HAVE_PWD_H
 #include <pwd.h>
 #endif
@@ -502,6 +505,15 @@ x_load_resources (Display *display, cons
   XrmPutLineResource (&rdb, line);
 
 #endif /* not USE_MOTIF */
+#ifdef USE_X_TOOLKIT
+  if ((db = XtScreenDatabase(DefaultScreenOfDisplay (display))))
+    {
+      XrmCombineDatabase (rdb, &db, FALSE);
+      rdb = db;
+    }
+  else
+    {
+#endif /* not USE_X_TOOLKIT */
 
   user_database = get_user_db (display);
 
@@ -543,6 +555,10 @@ x_load_resources (Display *display, cons
 	XrmMergeDatabases (db, &rdb);
     }
 
+#ifdef USE_X_TOOLKIT
+    } /* (db != XtScreenDatabase()) */
+#endif /* not USE_X_TOOLKIT */
+
   return rdb;
 }
 
