Index: libsoup-2.70.0/tests/meson.build
===================================================================
--- libsoup-2.70.0.orig/tests/meson.build
+++ libsoup-2.70.0/tests/meson.build
@@ -35,8 +35,6 @@ tests = [
   ['date', true, []],
   ['forms', true, []],
   ['header-parsing', true, []],
-  ['hsts', true, []],
-  ['hsts-db', true, []],
   ['misc', true, []],
   ['multipart', true, []],
   ['no-ssl', true, []],
