Index: xen-4.12.0-testing/xen/arch/Kconfig
===================================================================
--- xen-4.12.0-testing.orig/xen/arch/Kconfig
+++ xen-4.12.0-testing/xen/arch/Kconfig
@@ -2,7 +2,7 @@
 config NR_CPUS
 	int "Maximum number of physical CPUs"
 	range 1 4095
-	default "256" if X86
+	default "1024" if X86
 	default "8" if ARM && RCAR3
 	default "4" if ARM && QEMU
 	default "4" if ARM && MPSOC
