Index: numcodecs-0.16.5/numcodecs/tests/test_blosc.py
===================================================================
--- numcodecs-0.16.5.orig/numcodecs/tests/test_blosc.py
+++ numcodecs-0.16.5/numcodecs/tests/test_blosc.py
@@ -138,6 +138,7 @@ def test_compress_complib(use_threads):
         'blosclz': 'BloscLZ',
         'zlib': 'Zlib',
         'zstd': 'Zstd',
+        'snappy': 'Snappy'
     }
     blosc.use_threads = use_threads
     for cname in blosc.list_compressors():
