diff --git a/src/kdesud/CMakeLists.txt b/src/kdesud/CMakeLists.txt
index 90bd3f7..1118755 100644
--- a/src/kdesud/CMakeLists.txt
+++ b/src/kdesud/CMakeLists.txt
@@ -20,10 +20,9 @@ if(HAVE_X11)
   target_include_directories(kdesud PRIVATE ${X11_X11_INCLUDE_PATH})
 endif()
 
-if(KDE4_ENABLE_FPIE)
-    macro_add_compile_flags(kdesud ${KDE4_CXX_FPIE_FLAGS})
-    macro_add_link_flags(kdesud ${KDE4_PIE_LDFLAGS})
-endif()
+set_property(TARGET kdesud APPEND_STRING PROPERTY COMPILE_FLAGS " -fPIE")
+
+set_property(TARGET kdesud APPEND_STRING PROPERTY LINK_FLAGS " -pie")
 
 ########### install files ###############
 
