From 8e2fd311778b990240847e3e9ce3840ff312f5a7 Mon Sep 17 00:00:00 2001
From: Gayatri Kammela <gayatri.kammela@intel.com>
Date: Thu, 29 Oct 2020 12:09:06 -0700
Subject: [PATCH] intel_cpus: Enable Rocket Lake desktop platform support in
 PowerTop

Signed-off-by: Gayatri Kammela <gayatri.kammela@intel.com>
---
 src/cpu/intel_cpus.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/cpu/intel_cpus.cpp b/src/cpu/intel_cpus.cpp
index 13923b2..3d4464c 100644
--- a/src/cpu/intel_cpus.cpp
+++ b/src/cpu/intel_cpus.cpp
@@ -83,6 +83,7 @@ static int intel_cpu_models[] = {
 	0x9E,	/* KBL */
 	0xA5,   /* CML_DESKTOP */
 	0xA6,   /* CML_MOBILE */
+	0xA7,	/* RKL_DESKTOP */
 	0	/* last entry must be zero */
 };
 
@@ -196,6 +197,7 @@ nhm_core::nhm_core(int model)
 		case 0x9E:	/* KBL */
 		case 0xA5:      /* CML_DESKTOP */
 		case 0xA6:      /* CML_MOBILE */
+		case 0xA7:	/* RKL_DESKTOP */
 			has_c7_res = 1;
 	}
 
@@ -385,6 +387,7 @@ nhm_package::nhm_package(int model)
 		case 0x9E:	/* KBL */
 		case 0xA5:      /* CML_DESKTOP */
 		case 0xA6:      /* CML_MOBILE */
+		case 0xA7:	/* RKL_DESKTOP */
 			has_c2c6_res=1;
 			has_c7_res = 1;
 	}
@@ -428,6 +431,7 @@ nhm_package::nhm_package(int model)
 		case 0x9E:	/* KBL */
 		case 0xA5:      /* CML_DESKTOP */
 		case 0xA6:      /* CML_MOBILE */
+		case 0xA7:	/* RKL_DESKTOP */
 			has_c8c9c10_res = 1;
 			break;
 	}
