--- Makefile.am
+++ Makefile.am
@@ -1,10 +1,10 @@
 AUTOMAKE_OPTIONS = dist-bzip2
 SUBDIRS=cursors 
 
-themedir = $(prefix)/X11R6/lib/X11/icons/default/
+themedir = $(prefix)/share/icons/default/
 theme_DATA = index.theme.blueprint
 
-indexdir = $(prefix)/X11R6/lib/X11/icons/$(PACKAGE)/
+indexdir = $(prefix)/share/icons/$(PACKAGE)/
 index_DATA = index.theme
 
 EXTRA_DIST= autogen.sh blueprint-cursor.spec.in index.theme.blueprint.in index.theme.in
--- cursors/Makefile.am
+++ cursors/Makefile.am
@@ -1,4 +1,4 @@
-cursordir=	$(prefix)/X11R6/lib/X11/icons/$(PACKAGE)/cursors
+cursordir=	$(prefix)/share/icons/$(PACKAGE)/cursors
 
 cursor_DATA=    arrow \
 		bottom_left_corner \
