diff -Naurp scsirastools-1.6.4.orig/files/sgdisk scsirastools-1.6.4/files/sgdisk
--- scsirastools-1.6.4.orig/files/sgdisk	2012-10-26 22:42:37.149405573 +0200
+++ scsirastools-1.6.4/files/sgdisk	2012-10-26 22:43:14.591209370 +0200
@@ -52,7 +52,6 @@ case "$1" in
 	fi
 	sgdiskmon -b 
 	rc_status -v
-	touch /var/lock/subsys/sgdiskmon 
 	;;
 "stop")
 	echo -n "Stopping sgdiskmon "
@@ -62,7 +61,6 @@ case "$1" in
            echo -n "(pid $rpid) "
 	   kill $rpid
 	fi
-	rm -f /var/lock/subsys/sgdiskmon 
 	rc_status -v
 	;;
 "status")
diff -Naurp scsirastools-1.6.4.orig/files/sgraid scsirastools-1.6.4/files/sgraid
--- scsirastools-1.6.4.orig/files/sgraid	2012-10-26 22:42:37.149405573 +0200
+++ scsirastools-1.6.4/files/sgraid	2012-10-26 22:43:08.884086862 +0200
@@ -68,7 +68,6 @@ case "$1" in
 	fi
 	sgraidmon -b 
 	rc_status -v
-	touch /var/lock/subsys/sgraidmon 
 	;;
 "stop")
 	echo -n "Stopping sgraidmon "
@@ -77,7 +76,6 @@ case "$1" in
 	then
 	   kill $rpid
 	fi
-	rm -f /var/lock/subsys/sgraidmon 
 	rc_status -v
 	;;
 "status")
