--- Pixmap.h	Fri Mar 10 09:42:00 2000
+++ Pixmap.h	Fri Mar 10 09:36:16 2000
@@ -69,7 +69,7 @@
 #include <X11/Xmu/Converters.h>
 #include <X11/Xos.h>
 #include <X11/Xfuncproto.h>
-#include "xpm.h"
+#include <X11/xpm.h>
 
 /* Resources:
 
--- SelFile/SelFile.c	Mon Dec 27 13:14:20 1993
+++ SelFile/SelFile.c	Fri Mar 10 09:40:40 2000
@@ -46,7 +46,9 @@
 /* BSD 4.3 errno.h does not declare errno */
 extern int errno;
 extern int sys_nerr;
+#if !(defined __GLIBC__ && __GLIBC__ >=2)
 extern char *sys_errlist[];
+#endif
 
 #include <sys/param.h>
 #include <X11/cursorfont.h>
--- Imakefile	Fri May 13 13:04:38 1994
+++ Imakefile	Fri Mar 10 12:11:13 2000
@@ -9,7 +9,7 @@
 
 XCOMM def/undef USE_MOTIF there if you want Pixmap to use Motif or not.
 #undef USE_MOTIF
-#ifdef HPArchitecture
+#if defined(HPArchitecture) && !defined(LinuxArchitecture)
 CCOPTIONS = -Aa -D_HPUX_SOURCE -DNO_EDITRES
 #if ProjectX > 4
 MOTIFVER = Motif1.2
@@ -70,8 +70,8 @@
 
 ComplexProgramTarget(pixmap)
 InstallAppDefaults(Pixmap)
-MakeDirectories(install, $(PIXMAPDIR))
-InstallMultipleFlags($(BITMAPS), $(PIXMAPDIR), $(INSTDATFLAGS))
+MakeDirectories(install,$(PIXMAPDIR))
+InstallMultipleFlags($(BITMAPS),$(PIXMAPDIR),$(INSTDATFLAGS))
 
 AllTarget(dotpixmap)
