diff -rup a/configure.ac b/configure.ac
--- a/configure.ac	2024-12-24 11:40:02.000000000 +0100
+++ b/configure.ac	2024-12-29 10:04:31.028508599 +0100
@@ -18,7 +18,7 @@ m4_define([libxfce4ui_min_version], [4.1
 m4_define([libxfce4util_min_version], [4.16.0])
 m4_define([libxfce4panel_min_version], [4.16.0])
 m4_define([xfconf_minimum_version], [4.16.0])
-m4_define([libx11_min_version], [1.6.7])
+m4_define([libx11_min_version], [1.6.5])
 m4_define([libxproto_min_version], [7.0.0])
 m4_define([libxtst_min_version], [1.0.0])
 m4_define([wayland_min_version], [1.15.0])
diff -rup a/configure b/configure
--- a/configure	2024-12-24 11:41:19.000000000 +0100
+++ b/configure	2024-12-29 10:05:18.251853225 +0100
@@ -16680,11 +16680,11 @@ fi
 
 
   if test x"$xdt_feature_deps_check_only" = x"yes"; then
-    if ! $PKG_CONFIG --exists "x11 >= 1.6.7" >/dev/null 2>&1; then
+    if ! $PKG_CONFIG --exists "x11 >= 1.6.5" >/dev/null 2>&1; then
       if test x"$xdt_feature_deps_missing" = x""; then
-        xdt_feature_deps_missing="x11 >= 1.6.7"
+        xdt_feature_deps_missing="x11 >= 1.6.5"
       else
-        xdt_feature_deps_missing="$xdt_feature_deps_missing, x11 >= 1.6.7"
+        xdt_feature_deps_missing="$xdt_feature_deps_missing, x11 >= 1.6.5"
       fi
     fi
   else
@@ -16843,9 +16843,9 @@ fi
 
 
 
-  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for x11 >= 1.6.7" >&5
-printf %s "checking for x11 >= 1.6.7... " >&6; }
-  if $PKG_CONFIG "--atleast-version=1.6.7" "x11" >/dev/null 2>&1; then
+  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for x11 >= 1.6.5" >&5
+printf %s "checking for x11 >= 1.6.5... " >&6; }
+  if $PKG_CONFIG "--atleast-version=1.6.5" "x11" >/dev/null 2>&1; then
     LIBX11_VERSION=`$PKG_CONFIG --modversion "x11"`
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $LIBX11_VERSION" >&5
 printf "%s\n" "$LIBX11_VERSION" >&6; }
@@ -16862,7 +16862,7 @@ printf %s "checking LIBX11_LIBS... " >&6
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $LIBX11_LIBS" >&5
 printf "%s\n" "$LIBX11_LIBS" >&6; }
 
-    LIBX11_REQUIRED_VERSION=1.6.7
+    LIBX11_REQUIRED_VERSION=1.6.5
 
 
 
@@ -16891,7 +16891,7 @@ printf "%s\n" "found, but $xdt_cv_versio
 
       echo "*** The required package x11 was found on your system,"
       echo "*** but the installed version ($xdt_cv_version) is too old."
-      echo "*** Please upgrade x11 to atleast version 1.6.7, or adjust"
+      echo "*** Please upgrade x11 to atleast version 1.6.5, or adjust"
       echo "*** the PKG_CONFIG_PATH environment variable if you installed"
       echo "*** the new version of the package in a nonstandard prefix so"
       echo "*** pkg-config is able to find it."
@@ -16903,7 +16903,7 @@ printf "%s\n" "not found" >&6; }
 
 
       echo "*** The required package x11 was not found on your system."
-      echo "*** Please install x11 (atleast version 1.6.7) or adjust"
+      echo "*** Please install x11 (atleast version 1.6.5) or adjust"
       echo "*** the PKG_CONFIG_PATH environment variable if you"
       echo "*** installed the package in a nonstandard prefix so that"
       echo "*** pkg-config is able to find it."
@@ -17395,11 +17395,11 @@ printf "%s\n" "not found" >&6; }
 
 
   if test x"$xdt_feature_deps_check_only" = x"yes"; then
-    if ! $PKG_CONFIG --exists "x11 >= 1.6.7" >/dev/null 2>&1; then
+    if ! $PKG_CONFIG --exists "x11 >= 1.6.5" >/dev/null 2>&1; then
       if test x"$xdt_feature_deps_missing" = x""; then
-        xdt_feature_deps_missing="x11 >= 1.6.7"
+        xdt_feature_deps_missing="x11 >= 1.6.5"
       else
-        xdt_feature_deps_missing="$xdt_feature_deps_missing, x11 >= 1.6.7"
+        xdt_feature_deps_missing="$xdt_feature_deps_missing, x11 >= 1.6.5"
       fi
     fi
   else
@@ -17558,9 +17558,9 @@ fi
 
 
 
-  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for x11 >= 1.6.7" >&5
-printf %s "checking for x11 >= 1.6.7... " >&6; }
-  if $PKG_CONFIG "--atleast-version=1.6.7" "x11" >/dev/null 2>&1; then
+  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for x11 >= 1.6.5" >&5
+printf %s "checking for x11 >= 1.6.5... " >&6; }
+  if $PKG_CONFIG "--atleast-version=1.6.5" "x11" >/dev/null 2>&1; then
     LIBX11_VERSION=`$PKG_CONFIG --modversion "x11"`
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $LIBX11_VERSION" >&5
 printf "%s\n" "$LIBX11_VERSION" >&6; }
@@ -17577,7 +17577,7 @@ printf %s "checking LIBX11_LIBS... " >&6
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $LIBX11_LIBS" >&5
 printf "%s\n" "$LIBX11_LIBS" >&6; }
 
-    LIBX11_REQUIRED_VERSION=1.6.7
+    LIBX11_REQUIRED_VERSION=1.6.5
 
 
 
@@ -17606,7 +17606,7 @@ printf "%s\n" "found, but $xdt_cv_versio
 
       echo "*** The required package x11 was found on your system,"
       echo "*** but the installed version ($xdt_cv_version) is too old."
-      echo "*** Please upgrade x11 to atleast version 1.6.7, or adjust"
+      echo "*** Please upgrade x11 to atleast version 1.6.5, or adjust"
       echo "*** the PKG_CONFIG_PATH environment variable if you installed"
       echo "*** the new version of the package in a nonstandard prefix so"
       echo "*** pkg-config is able to find it."
@@ -17618,7 +17618,7 @@ printf "%s\n" "not found" >&6; }
 
 
       echo "*** The required package x11 was not found on your system."
-      echo "*** Please install x11 (atleast version 1.6.7) or adjust"
+      echo "*** Please install x11 (atleast version 1.6.5) or adjust"
       echo "*** the PKG_CONFIG_PATH environment variable if you"
       echo "*** installed the package in a nonstandard prefix so that"
       echo "*** pkg-config is able to find it."
