Index: configure.ac
===================================================================
--- configure.ac	2010-08-01 01:17:41.000000000 -0300
+++ configure.ac	2016-06-17 18:35:11.499234579 -0300
@@ -122,7 +122,7 @@
 AC_HEADER_DIRENT
 AC_HEADER_STDC
 AC_HEADER_SYS_WAIT
-AC_CHECK_HEADERS([OS.h arpa/inet.h fcntl.h fenv.h float.h inttypes.h limits.h malloc.h memory.h netdb.h netinet/in.h stddef.h stdint.h stdlib.h string.h sys/file.h sys/ioctl.h sys/param.h sys/socket.h sys/time.h unistd.h values.h wchar.h wctype.h])
+AC_CHECK_HEADERS([OS.h arpa/inet.h fcntl.h fenv.h float.h inttypes.h limits.h malloc.h memory.h netdb.h netinet/in.h stddef.h stdint.h stdlib.h string.h sys/file.h sys/ioctl.h sys/param.h sys/socket.h sys/time.h unistd.h values.h wchar.h wctype.h math.h])
 
 
 if test "x$slabcheck" = "xyes"; then
Index: config.h.in
===================================================================
--- config.h.in	2012-03-31 22:53:54.000000000 -0300
+++ config.h.in	2016-06-17 18:37:29.000000000 -0300
@@ -139,6 +139,9 @@
 /* Define to 1 if you have the <malloc.h> header file. */
 #undef HAVE_MALLOC_H
 
+/* Define to 1 if you have the <math.h> header file. */
+#undef HAVE_MATH_H
+
 /* Define to 1 if you have the `mblen' function. */
 #undef HAVE_MBLEN
 
