Index: configure.ac
===================================================================
--- configure.ac.orig	2014-03-17 17:29:16.000000000 +0200
+++ configure.ac	2015-07-18 13:29:33.709776183 +0200
@@ -226,14 +226,14 @@ fi
 
 STEPMAKE_PATH_PROG(GHOSTSCRIPT, gs, $DOCUMENTATION_REQUIRED, 8.60)
 
-STEPMAKE_PROGS(MAKEINFO, makeinfo, REQUIRED, 4.11)
-STEPMAKE_PROGS(TEXI2HTML, texi2html, $DOCUMENTATION_REQUIRED, 1.82)
+STEPMAKE_PROGS(MAKEINFO, makeinfo4, REQUIRED, 4.11)
+STEPMAKE_PROGS(TEXI2HTML, texi2html4, $DOCUMENTATION_REQUIRED, 1.82)
 STEPMAKE_PROGS(DBLATEX, dblatex, $DOCUMENTATION_REQUIRED, 0.1.4)
 STEPMAKE_PROGS(BIBTEX, bibtex, $DOCUMENTATION_REQUIRED)
 STEPMAKE_PROGS(PDFLATEX, pdflatex, $DOCUMENTATION_REQUIRED)
 STEPMAKE_PROGS(PDFTEX, pdfetex pdftex etex, $DOCUMENTATION_REQUIRED)
-STEPMAKE_PROGS(TEXI2PDF, texi2pdf, $DOCUMENTATION_REQUIRED)
-STEPMAKE_PROGS(TEXINDEX, texindex, $DOCUMENTATION_REQUIRED)
+STEPMAKE_PROGS(TEXI2PDF, texi2pdf4, $DOCUMENTATION_REQUIRED)
+STEPMAKE_PROGS(TEXINDEX, texindex4, $DOCUMENTATION_REQUIRED)
 AC_MSG_CHECKING(for epsf.tex)
 TEX_EPSF=`kpsewhich tex epsf`
 if test "$TEX_EPSF" = ""; then
Index: scripts/lilypond-book.py
===================================================================
--- scripts/lilypond-book.py.orig	2014-03-17 17:29:16.000000000 +0200
+++ scripts/lilypond-book.py	2015-07-18 13:29:33.709776183 +0200
@@ -245,7 +245,7 @@ case --pdf option is set instead of pdfl
               help=_ ("run executable PROG instead of texi2pdf"),
               metavar=_ ("PROG"),
               action='store', dest='texinfo_program',
-              default='texi2pdf')
+              default='texi2pdf4')
     group.add_option ('--pdf',
               action="store_true",
               dest="create_pdf",
Index: stepmake/stepmake/texinfo-rules.make
===================================================================
--- stepmake/stepmake/texinfo-rules.make.orig	2014-03-17 17:29:16.000000000 +0200
+++ stepmake/stepmake/texinfo-rules.make	2015-07-18 13:43:46.163460072 +0200
@@ -68,9 +68,9 @@ endif
 
 $(outdir)/%.pdf: $(outdir)/%.texi $(outdir)/version.itexi $(outdir)/%.pdf.omf $(outdir)/weblinks.itexi | $(OUT_TEXINFO_MANUALS)
 ifeq ($(WEB_VERSION),yes)
-	$(buildscript-dir)/run-and-check "cd $(outdir); texi2pdf $(TEXI2PDF_FLAGS) -D web_version -I $(abs-src-dir) $(TEXINFO_PAPERSIZE_OPTION) $(<F) < /dev/null" "$*.texi2pdf.log"
+	$(buildscript-dir)/run-and-check "cd $(outdir); texi2pdf4 $(TEXI2PDF_FLAGS) -D web_version -I $(abs-src-dir) $(TEXINFO_PAPERSIZE_OPTION) $(<F) < /dev/null" "$*.texi2pdf.log"
 else
-	$(buildscript-dir)/run-and-check "cd $(outdir); texi2pdf $(TEXI2PDF_FLAGS) -I $(abs-src-dir) $(TEXINFO_PAPERSIZE_OPTION) $(<F) < /dev/null" "$*.texi2pdf.log"
+	$(buildscript-dir)/run-and-check "cd $(outdir); texi2pdf4 $(TEXI2PDF_FLAGS) -I $(abs-src-dir) $(TEXINFO_PAPERSIZE_OPTION) $(<F) < /dev/null" "$*.texi2pdf.log"
 endif
 
 $(outdir)/%.txt: $(outdir)/%.texi $(outdir)/version.itexi $(outdir)/weblinks.itexi | $(OUT_TEXINFO_MANUALS)
Index: python/book_texinfo.py
===================================================================
--- python/book_texinfo.py.orig	2014-03-17 17:29:16.000000000 +0200
+++ python/book_texinfo.py	2015-07-18 13:54:21.809900693 +0200
@@ -203,7 +203,7 @@ def get_texinfo_width_indent (source, gl
     # Work around a texi2pdf bug: if LANG=C is not given, a broken regexp is
     # used to detect relative/absolute paths, so the absolute path is not
     # detected as such and this command fails:
-    progress (_ ("Running texi2pdf on file %s to detect default page settings.\n") % tmpfile);
+    progress (_ ("Running texi2pdf4 on file %s to detect default page settings.\n") % tmpfile);
 
     # execute the command and pipe stdout to the parameter_string:
     cmd = '%s -c -o %s %s' % (global_options.texinfo_program, outfile, tmpfile);
