Subject: Use system* instead of system when invoking browser
Author: Don Armstrong <don@debian.org>
Bug-Debian: https://bugs.debian.org/884136
Bug: https://sourceforge.net/p/testlilyissues/issues/5243
Index: scripts/lilypond-invoke-editor.scm
===================================================================
--- scripts/lilypond-invoke-editor.scm.orig	2018-05-15 10:49:41.446667535 +0200
+++ scripts/lilypond-invoke-editor.scm	2018-05-15 10:51:33.710227632 +0200
@@ -127,11 +127,14 @@ Options:
 	  (exit 1)))))
 
 (define (run-browser uri)
-  (system
-   (if (getenv "BROWSER")
-       (format #f "~a ~a" (getenv "BROWSER") uri)
-       (format #f "firefox -remote 'OpenURL(~a,new-tab)'" uri))))
-
+  (if (getenv "BROWSER")
+      (system*
+       (getenv "BROWSER")
+       uri)
+      (system*
+       "firefox"
+       "-remote"
+       (format #f "OpenUrl(~a,new-tab)" uri))))
 
 (define (strip-framework-path var)
   (define pat "lilypond/usr")
