Use /etc/pam.d/xdm

================================================================================
--- src/wdm/verify.c
+++ src/wdm/verify.c
@@ -446,7 +446,7 @@
 	if (pam_error != PAM_SUCCESS) goto pam_failed;
 
 	PAM_password = greet->password;
-	pam_error = pam_start("wdm", greet->name, &PAM_conversation, pamhp);
+	pam_error = pam_start("xdm", greet->name, &PAM_conversation, pamhp);
 	PAM_BAIL;
 	pam_error = pam_set_item(*pamhp, PAM_TTY, d->name);
 	PAM_BAIL;
