Index: b/dnscrypt-proxy.conf
===================================================================
--- a/dnscrypt-proxy.conf
+++ b/dnscrypt-proxy.conf
@@ -39,19 +39,21 @@ ResolverName random
 ## Unless you are using systemd, you probably want to change this to "yes"
 ## after having verified that the rest of the configuration works as expected.
 
-Daemonize no
+Daemonize yes
 
 
 ## Write the PID number to a file
+## in openSUSE, the pidfile should be dnscrypt-proxy@[configfile name].pid
+## inside /var/run/dnscrypt-proxy/ to make systemd instantiated service work
 
-# PidFile /var/run/dnscrypt-proxy.pid
+PidFile /var/run/dnscrypt-proxy/dnscrypt-proxy@default.pid
 
 
 ## [NOT AVAILABLE ON WINDOWS] Start the process, bind the required ports, and
 ## run the server as a less-privileged system user.
 ## The value for this parameter is a user name.
 
-# User _dnscrypt-proxy
+User dnscrypt
 
 
 
@@ -66,7 +68,7 @@ Daemonize no
 ## using this option. You should edit systemd's dnscrypt-proxy.socket file
 ## instead.
 
-# LocalAddress 127.0.0.1:53
+LocalAddress 127.0.0.1:53
 
 
 ## Cache DNS responses to avoid outgoing traffic when the same queries
@@ -136,15 +138,16 @@ EphemeralKeys off
 ## Log file to write server errors and information to.
 ## If you use this tool for privacy, keeping logs of any kind is usually not
 ## a good idea.
+## in openSUSE, logfile should be /var/log/dnscrypt-proxy/[config filename].log
 
-# LogFile /var/log/dnscrypt-proxy.log
+LogFile /var/log/dnscrypt-proxy/default.log
 
 
 ## Don't log events with priority above this log level after the service has
 ## been started up. Default is 6.
 ## Valid values are between 0 (critical) to 7 (debug-level messages).
 
-# LogLevel 6
+LogLevel 7
 
 
 ## [NOT AVAILABLE ON WINDOWS] Send server logs to the syslog daemon
