--- src/events.c.orig	2001-01-14 17:14:33.000000000 +0100
+++ src/events.c	2014-10-22 15:34:41.549543325 +0200
@@ -135,6 +135,13 @@
       HandleClientMessageEvent((ClientMessageEvent *) event);
       break;
 
+#ifdef TARGET_SDL // hack SDL returns this when toggling fullscreen
+    case EVENT_FULLSCREENTOGGLE:
+      if (game_status == SETUP) // redraw the setup screen to update fs status
+        DrawSetupScreen();
+      break;
+#endif
+
     default:
       break;
   }
