Index: mesa-24.0.3/include/meson.build
===================================================================
--- mesa-24.0.3.orig/include/meson.build
+++ mesa-24.0.3/include/meson.build
@@ -28,7 +28,6 @@ if with_android_stub
   inc_include += [include_directories('android_stub')]
 endif
 
-if not with_glvnd and host_machine.system() != 'windows'
   if with_gles1 or with_gles2 or with_opengl or with_egl
     install_headers('KHR/khrplatform.h', subdir : 'KHR')
   endif
@@ -84,7 +83,6 @@ if not with_glvnd and host_machine.syste
       subdir : 'EGL',
     )
   endif
-endif
 
 # Non-upstream headers
 if with_egl
