Index: mesa-25.3.0/src/util/00-mesa-defaults.conf
===================================================================
--- mesa-25.3.0.orig/src/util/00-mesa-defaults.conf
+++ mesa-25.3.0/src/util/00-mesa-defaults.conf
@@ -822,6 +822,14 @@ TODO: document the other workarounds.
         <application name="Rocket League" executable="RocketLeague">
             <option name="radeonsi_zerovram" value="true" />
         </application>
+        <!-- This is workaround to prevent distorted colors on hardware accelerated video in Chromium. https://bugzilla.opensuse.org/show_bug.cgi?id=1113211 -->
+        <application name="Chromium" executable="chromium">
+            <option name="allow_rgb10_configs" value="false" />
+        </application>
+        <!-- This is workaround to prevent distorted colors on hardware accelerated video in Totem. https://bugzilla.opensuse.org/show_bug.cgi?id=1142270 -->
+        <application name="Totem" executable="totem">
+            <option name="allow_rgb10_configs" value="false" />
+        </application>
         <!-- YoYo game engine workarounds -->
         <application name="Monolith demo5" executable="runner" sha1="0c40d509a74e357f0280cb1bd882e9cd94b91bdf">
             <option name="radeonsi_sync_compile" value="true" />
