diff -Nru amazon-ecs-agent-1.102.1.orig/ecs-init/config/config_generic_rpm.go amazon-ecs-agent-1.102.1/ecs-init/config/config_generic_rpm.go
--- amazon-ecs-agent-1.102.1.orig/ecs-init/config/config_generic_rpm.go	2026-03-06 22:46:03.000000000 +0100
+++ amazon-ecs-agent-1.102.1/ecs-init/config/config_generic_rpm.go	2026-03-31 13:10:12.587131159 +0200
@@ -1,5 +1,5 @@
 //go:build generic_rpm
-// +build generic_rpm
+// +build generic_rpm suse
 
 // Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 //
diff -Nru amazon-ecs-agent-1.102.1.orig/ecs-init/config/config_suse_ubuntu_debian_chainguard.go amazon-ecs-agent-1.102.1/ecs-init/config/config_suse_ubuntu_debian_chainguard.go
--- amazon-ecs-agent-1.102.1.orig/ecs-init/config/config_suse_ubuntu_debian_chainguard.go	2026-03-06 22:46:03.000000000 +0100
+++ amazon-ecs-agent-1.102.1/ecs-init/config/config_suse_ubuntu_debian_chainguard.go	2026-03-31 13:11:39.190335214 +0200
@@ -1,5 +1,5 @@
 //go:build suse || ubuntu || debian || chainguard
-// +build suse ubuntu debian chainguard
+// +build ubuntu debian chainguard
 
 // Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 //
