--- lib/libxclass/OXHtmlParse.cc
+++ lib/libxclass/OXHtmlParse.cc
@@ -1486,12 +1486,12 @@
         n = 1;
         z = "";
       }
-      printf("Block 0x%08x flags=%02x cnt=%d x=%d..%d y=%d..%d z=\"%.*s\"\n",
-        (int) p, p->flags, p->count, block->left, block->right,
+      printf("Block %p flags=%02x cnt=%d x=%d..%d y=%d..%d z=\"%.*s\"\n",
+        p, p->flags, p->count, block->left, block->right,
         block->top, block->bottom, n, z);
     } else {
-      printf("Token 0x%08x font=%2d color=%2d align=%d flags=0x%04x name=%s\n",
-        (int) p, p->style.font, p->style.color,
+      printf("Token %p font=%2d color=%2d align=%d flags=0x%04x name=%s\n",
+        p, p->style.font, p->style.color,
         p->style.align, p->style.flags, DumpToken(p));
     }
   }
--- lib/libxclass/OXHtmlSizer.cc
+++ lib/libxclass/OXHtmlSizer.cc
@@ -980,9 +980,9 @@
     }
 
     TRACE(HtmlTrace_Style,
-          ("Style of 0x%08x font=%02d color=%02d bg=%02d "
+          ("Style of %p font=%02d color=%02d bg=%02d "
            "align=%d flags=0x%04x token=%s\n",
-           (int)p, p->style.font, p->style.color, p->style.bgcolor,
+           p, p->style.font, p->style.color, p->style.bgcolor,
            p->style.align, p->style.flags, DumpToken(p)));
 
     p = p->pNext;
