--- a/surefire-providers/surefire-testng/pom.xml
+++ b/surefire-providers/surefire-testng/pom.xml
@@ -51,8 +51,7 @@
     <dependency>
       <groupId>org.testng</groupId>
       <artifactId>testng</artifactId>
-      <version>5.10</version>
-      <classifier>jdk15</classifier>
+      <version>7.4.0</version>
       <scope>provided</scope>
     </dependency>
   </dependencies>
--- a/surefire-providers/surefire-testng/src/main/java/org/apache/maven/surefire/testng/conf/AbstractDirectConfigurator.java
+++ b/surefire-providers/surefire-testng/src/main/java/org/apache/maven/surefire/testng/conf/AbstractDirectConfigurator.java
@@ -63,7 +63,7 @@ public void configure( TestNG testng, Map<String, String> options )
         testng.setUseDefaultListeners( false );
         configureInstance( testng, options );
         // TODO: we should have the Profile so that we can decide if this is needed or not
-        testng.setListenerClasses( loadListenerClasses( listeners ) );
+        testng.setListenerClasses( ( List ) loadListenerClasses( listeners ) );
     }
 
     @Override
--- a/surefire-providers/surefire-testng/src/main/java/org/apache/maven/surefire/testng/conf/TestNGMapConfigurator.java
+++ b/surefire-providers/surefire-testng/src/main/java/org/apache/maven/surefire/testng/conf/TestNGMapConfigurator.java
@@ -68,7 +68,7 @@ public void configure( XmlSuite suite, Map<String, String> options )
         String parallel = options.get( PARALLEL_PROP );
         if ( parallel != null )
         {
-            suite.setParallel( parallel );
+            suite.setParallel( XmlSuite.ParallelMode.getValidParallel( parallel ) );
         }
     }
 
