Index: ImageMagick-7.1.0-57/configure
===================================================================
--- ImageMagick-7.1.0-57.orig/configure
+++ ImageMagick-7.1.0-57/configure
@@ -35038,7 +35038,9 @@ fi
 
 
 # Subdirectory to place architecture-dependent configuration files
-CONFIGURE_RELATIVE_PATH="${PACKAGE_NAME}-${MAGICK_MAJOR_VERSION}"
+if test "x$CONFIGURE_RELATIVE_PATH" = "x"; then
+   CONFIGURE_RELATIVE_PATH="${PACKAGE_NAME}-${MAGICK_MAJOR_VERSION}"
+fi
 
 printf "%s\n" "#define CONFIGURE_RELATIVE_PATH \"$CONFIGURE_RELATIVE_PATH\"" >>confdefs.h
 
