Index: ImageMagick-7.1.1-21/MagickCore/resource.c
===================================================================
--- ImageMagick-7.1.1-21.orig/MagickCore/resource.c
+++ ImageMagick-7.1.1-21/MagickCore/resource.c
@@ -136,7 +136,7 @@ static ResourceInfo
     MagickULLConstant(768),            /* file limit */
     MagickULLConstant(1),              /* thread limit */
     MagickULLConstant(0),              /* throttle limit */
-    MagickResourceInfinity             /* time limit */
+    INT_MAX                            /* time limit */
   };
 
 static SemaphoreInfo
@@ -971,7 +971,7 @@ MagickExport MagickBooleanType ListMagic
     (void) FormatMagickSize(resource_info.disk_limit,MagickTrue,"B",
       MagickFormatExtent,disk_limit);
   (void) CopyMagickString(time_limit,"unlimited",MagickFormatExtent);
-  if (resource_info.time_limit != MagickResourceInfinity)
+  if (resource_info.time_limit != INT_MAX)
     FormatTimeToLive(resource_info.time_limit,time_limit);
   (void) FormatLocaleFile(file,"Resource limits:\n");
   (void) FormatLocaleFile(file,"  Width: %s\n",width_limit);
@@ -1333,7 +1333,7 @@ MagickPrivate MagickBooleanType Resource
         limit,100.0));
       limit=DestroyString(limit);
     }
-  (void) SetMagickResourceLimit(TimeResource,MagickResourceInfinity);
+  (void) SetMagickResourceLimit(TimeResource,INT_MAX);
   limit=GetEnvironmentValue("MAGICK_TIME_LIMIT");
   if (limit != (char *) NULL)
     {
