diff -Nur ipadic-2.7.0/Makefile.am new/Makefile.am
--- ipadic-2.7.0/Makefile.am	2003-09-13 07:37:17.000000000 +0200
+++ new/Makefile.am	2021-05-12 21:48:10.407751866 +0200
@@ -18,10 +18,10 @@
 	$(MKDICDIR)/makeda chadic *.dic
 
 install-data-local:
-	if test -f @CHASEN_CHASENRC_PATH@  ; then \
+	if test -f $(DESTDIR)@CHASEN_CHASENRC_PATH@  ; then \
 		echo "Please modify @CHASEN_CHASENRC_PATH@"; \
 	else \
-		cp chasenrc @CHASEN_CHASENRC_PATH@ ;\
+		install -D -m 0644 chasenrc $(DESTDIR)@CHASEN_CHASENRC_PATH@/chasenrc ;\
 	fi
 
 sjis-dist: dist
