diff -Nru amazon-ssm-agent-2.3.978.0.orig/agent/version/version.go amazon-ssm-agent-2.3.978.0/agent/version/version.go
--- amazon-ssm-agent-2.3.978.0.orig/agent/version/version.go	2020-04-08 21:04:26.000000000 +0200
+++ amazon-ssm-agent-2.3.978.0/agent/version/version.go	2020-04-30 13:40:52.228599625 +0200
@@ -19,4 +19,4 @@
 package version
 
 // Version is the version of the Agent
-const Version = "2.3.0.0"
+const Version = "2.3.978.0"
diff -Nru amazon-ssm-agent-2.3.978.0.orig/VERSION amazon-ssm-agent-2.3.978.0/VERSION
--- amazon-ssm-agent-2.3.978.0.orig/VERSION	2020-04-08 21:04:26.000000000 +0200
+++ amazon-ssm-agent-2.3.978.0/VERSION	2020-04-30 13:40:20.676516447 +0200
@@ -1 +1 @@
-2.3.0.0
\ No newline at end of file
+2.3.978.0
\ No newline at end of file
