diff -Nru amazon-ssm-agent-2.3.978.0.orig/agent/plugins/inventory/gatherers/application/dataProvider.go amazon-ssm-agent-2.3.978.0/agent/plugins/inventory/gatherers/application/dataProvider.go
--- amazon-ssm-agent-2.3.978.0.orig/agent/plugins/inventory/gatherers/application/dataProvider.go	2020-04-08 21:04:26.000000000 +0200
+++ amazon-ssm-agent-2.3.978.0/agent/plugins/inventory/gatherers/application/dataProvider.go	2020-04-30 14:03:04.956300230 +0200
@@ -7,7 +7,6 @@
 	"fmt"
 
 	"github.com/aws/amazon-ssm-agent/agent/context"
-	"github.com/aws/amazon-ssm-agent/agent/plugins/configurepackage/localpackages"
 	"github.com/aws/amazon-ssm-agent/agent/plugins/inventory/model"
 )
 
