From 0e576f3f922f8659c6b36184480c9db8e4d3b802 Mon Sep 17 00:00:00 2001
From: Ben Greiner <code@bnavigator.de>
Date: Sat, 22 Oct 2022 12:18:54 +0200
Subject: [PATCH 1/2] restrict setuptools.packages.find

---
 pyproject.toml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pyproject.toml b/pyproject.toml
index 00587b43..576ac7e2 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -69,6 +69,7 @@ build-backend = 'setuptools.build_meta'
 write_to = "asdf/_version.py"
 
 [tool.setuptools.packages.find]
+include = ['asdf*', 'pytest_asdf*']
 
 [tool.setuptools.package-data]
 'asdf.commands.tests.data' = ["*"]


