--- xsltConf.sh.in.orig	2012-12-06 08:43:13.843408831 +0100
+++ xsltConf.sh.in	2012-12-06 08:43:22.617391851 +0100
@@ -2,6 +2,6 @@
 # Configuration file for using the xslt library
 #
 XSLT_LIBDIR="@XSLT_LIBDIR@"
-XSLT_LIBS="@XSLT_LIBS@"
+XSLT_LIBS="-lxslt"
 XSLT_INCLUDEDIR="@XSLT_INCLUDEDIR@"
 MODULE_VERSION="xslt-@VERSION@"
--- xslt-config.in.orig	2012-12-06 08:42:48.797457330 +0100
+++ xslt-config.in	2012-12-06 08:43:10.402415492 +0100
@@ -89,7 +89,7 @@
     shift
 done
 
-the_libs="@XSLT_LIBDIR@ @XSLT_LIBS@ @EXTRA_LIBS@"
+the_libs="-lxslt"
 if test "$includedir" != "/usr/include"; then
     the_flags="$the_flags -I$includedir `@XML_CONFIG@ --cflags`"
 else
