Index: lime-5.2.73/src/lime_double_ratchet.cpp
===================================================================
--- lime-5.2.73.orig/src/lime_double_ratchet.cpp
+++ lime-5.2.73/src/lime_double_ratchet.cpp
@@ -25,6 +25,7 @@
 #include "bctoolbox/exception.hh"
 
 #include <algorithm> //copy_n
+#include <cstdint>
 
 
 using namespace::std;
Index: lime-5.2.73/src/lime_settings.hpp
===================================================================
--- lime-5.2.73.orig/src/lime_settings.hpp
+++ lime-5.2.73/src/lime_settings.hpp
@@ -21,6 +21,8 @@
 #define lime_settings_hpp
 
 
+#include <cstdint>
+
 namespace lime {
 /** @brief Hold constants definition used as settings in all components of the lime library
  *
