--- ./latex2html.pin.orig	2013-06-25 14:57:10.000000000 +0000
+++ ./latex2html.pin	2013-06-25 14:57:27.000000000 +0000
@@ -16557,7 +16557,7 @@ sub make_isolatin1_rx {
      $caption_suffixes = "lof|lot";
 #    $caption_rx = "\\\\caption\\s*([[]\\s*((($any_next_pair_rx5)|([[][^]]*[]])|[^]])*)[]])?$O(\\d+)$C([\\s\\S]*)$O\\8$C$EOL";
 
-    $caption_rx = "\\\\(top|bottom|table)?caption\\s*\\\*?\\s*([[]\\s*((($any_next_pair_rx6)|([[][^]]*[]])|[^]])*)[]])?$O(\\d+)$C([\\s\\S]*)$O\\9$C$EOL";
+    $caption_rx = "\\\\(top|bottom|table)?caption\\s*\\\*?\\s*([[]\\s*((($any_next_pair_rx6)|([[][^]]*[]])|[^]])*)[]])?$O(\\d+)$C([\\s\\S]*)$O\\g9$C$EOL";
     $caption_width_rx = "\\\\setlength\\s*(($O|$OP)\\d+($C|$CP))\\\\captionwidth\\1\\s*(($O|$OP)\\d+($C|$CP))([^>]*)\\4";
 
     # Matches the \htmlimage command
@@ -16622,7 +16622,7 @@ sub make_isolatin1_rx {
 
     # Matches a label command and its argument
     $labels_rx = "\\\\label\\s*$O(\\d+)$C([\\s\\S]*)$O\\1$C$EOL";
-    $labels_rx8 = "\\\\label\\s*$O(\\d+)$C([\\s\\S]*)$O\\8$C$EOL";
+    $labels_rx8 = "\\\\label\\s*$O(\\d+)$C([\\s\\S]*)$O\\g8$C$EOL";
 
     # Matches environments that should not be touched during the translation
 #   $verbatim_env_rx = "\\s*{(verbatim|rawhtml|LVerbatim)[*]?}";
