 mi/miscrinit.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)
Index: xorg-server-21.1.4/mi/miscrinit.c
===================================================================
--- xorg-server-21.1.4.orig/mi/miscrinit.c
+++ xorg-server-21.1.4/mi/miscrinit.c
@@ -121,7 +121,9 @@ miModifyPixmapHeader(PixmapPtr pPixmap,
 static Bool
 miCloseScreen(ScreenPtr pScreen)
 {
-    return ((*pScreen->DestroyPixmap) ((PixmapPtr) pScreen->devPrivate));
+    if (pScreen->devPrivate)
+        return ((*pScreen->DestroyPixmap) ((PixmapPtr) pScreen->devPrivate));
+    return TRUE;
 }
 
 static Bool
