From f28005325922c47d59f9735acd5aa87a3e14a53d Mon Sep 17 00:00:00 2001
From: Thomas Zimmermann <tzimmermann@suse.de>
Date: Tue, 30 Nov 2021 14:12:55 +0100
Subject: Revert "xf86: Accept devices with the 'simpledrm'
 driver."

This reverts commit b89fdd523e2c9e9b0cdf37b263833c4b0a8868b8. Using
simpledrm, or any other platform driver, can now be achieved via udev-
generated configuration files. A driver-specific workaround is no
longer needed.

Signed-off-by: Thomas Zimmermann <tzimmermann@suse.de>
---
 hw/xfree86/common/xf86platformBus.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/hw/xfree86/common/xf86platformBus.c b/hw/xfree86/common/xf86platformBus.c
index 45028f7a6..0e0a995ac 100644
--- a/hw/xfree86/common/xf86platformBus.c
+++ b/hw/xfree86/common/xf86platformBus.c
@@ -557,13 +557,8 @@ xf86platformProbeDev(DriverPtr drvp)
             }
             else {
                 /* for non-seat0 servers assume first device is the master */
-                if (ServerIsNotSeat0()) {
+                if (ServerIsNotSeat0())
                     break;
-                } else {
-                    /* Accept the device if the driver is simpledrm */
-                    if (strcmp(xf86_platform_devices[j].attribs->driver, "simpledrm") == 0)
-                        break;
-                }

                 if (xf86IsPrimaryPlatform(&xf86_platform_devices[j]))
                     break;
--
2.34.0

