Index: configure.in
===================================================================
--- configure.in.orig
+++ configure.in
@@ -546,7 +546,9 @@ AC_TRY_COMPILE([
 #endif /* HAVE_UNISTD_H */
 ],
 [struct timeval tv;
-gettimeofday(&tv, NULL)], AC_MSG_RESULT(no), AC_MSG_RESULT(maybe) AC_TRY_COMPILE([
+gettimeofday(&tv, NULL)],[AC_MSG_RESULT(no)],[
+AC_MSG_RESULT(maybe)
+AC_TRY_COMPILE([
 #if HAVE_SYS_TIME_H
 #include <sys/time.h>
 #endif /* HAVE_SYS_TIME_H */
@@ -555,8 +557,13 @@ gettimeofday(&tv, NULL)], AC_MSG_RESULT(
 #endif /* HAVE_UNISTD_H */
 ],
 [struct timeval tv;
-gettimeofday(&tv)], AC_DEFINE(HAVE_ONE_ARG_GETTIMEOFDAY) AC_MSG_RESULT(yes), AC_MSG_RESULT(no)))
+gettimeofday(&tv)],
+[
+	AC_DEFINE(HAVE_ONE_ARG_GETTIMEOFDAY)
+	AC_MSG_RESULT(yes)
+],[AC_MSG_RESULT(no)])])
 
 # Munge out LOCALBIN and LOCALMAN in canonical (no bletch) form
-AC_OUTPUT(configure.munger libmissing/Makefile libopie/Makefile Makefile.munge:Makefile.in)
+AC_CONFIG_FILES(configure.munger libmissing/Makefile libopie/Makefile Makefile.munge:Makefile.in)
+AC_OUTPUT
 sh configure.munger
