--- opie-2.4/Makefile.in.xx	2007-02-09 07:52:13.000000000 +0100
+++ opie-2.4/Makefile.in	2007-02-09 07:53:49.000000000 +0100
@@ -142,26 +142,18 @@
 	@cp opielogin $(OPIEAUTO) $(DESTDIR)$(LOCALBIN)
 	@cp opieftpd $(OPIEAUTO) $(DESTDIR)$(LOCALBIN)
 	@cp opiegen $(OPIEAUTO) $(DESTDIR)$(LOCALBIN)
-	@$(CHOWN) $(OWNER) $(DESTDIR)$(LOCALBIN)/opiekey
-	@$(CHOWN) $(OWNER) $(DESTDIR)$(LOCALBIN)/opiesu
-	@$(CHOWN) $(OWNER) $(DESTDIR)$(LOCALBIN)/opielogin
-	@$(CHOWN) $(OWNER) $(DESTDIR)$(LOCALBIN)/opieftpd
-	@$(CHOWN) $(OWNER) $(DESTDIR)$(LOCALBIN)/opiegen
-	@if test ! -z "$(OPIEAUTO)"; then $(CHOWN) $(OWNER) $(DESTDIR)$(LOCALBIN)/opieauto; fi
-	@chgrp $(GROUP) $(DESTDIR)$(LOCALBIN)/opiekey
 	@echo "Changing file permissions"
 	@chmod 0755 $(DESTDIR)$(LOCALBIN)/opiekey
 	@chmod 0755 $(DESTDIR)$(LOCALBIN)/opieftpd
 	@chmod 0755 $(DESTDIR)$(LOCALBIN)/opiegen
 	@chmod 0755 $(DESTDIR)$(LOCALBIN)/opielogin
-	@chmod 4755 $(DESTDIR)$(LOCALBIN)/opiesu
+	@chmod 0755 $(DESTDIR)$(LOCALBIN)/opiesu
 	@if test ! -z "$(OPIEAUTO)"; then chmod 0511 $(DESTDIR)$(LOCALBIN)/opieauto; fi
 	@echo "Symlinking aliases to opiekey"
 	@-ln -s opiekey $(DESTDIR)$(LOCALBIN)/otp-md4
 	@-ln -s opiekey $(DESTDIR)$(LOCALBIN)/otp-md5
 	@echo "Installing manual pages"
 	@-for i in otp-md4 otp-md5; do ln -s opiekey.1 $(DESTDIR)$(LOCALMAN)/man1/$$i.1; done
-	@if test ! -d $(DESTDIR)$(LOCALMAN)/man1; then $(MKDIR) $(DESTDIR)$(LOCALMAN)/man1; chmod 755 $(DESTDIR)$(LOCALMAN)/man1; fi; cp opiekey.1 $(DESTDIR)$(LOCALMAN)/man1/opiekey.1; $(CHOWN) $(OWNER) $(DESTDIR)$(LOCALMAN)/man1/opiekey.1; chgrp $(GROUP) $(DESTDIR)$(LOCALMAN)/man1/opiekey.1; chmod 644 $(DESTDIR)$(LOCALMAN)/man1/opiekey.1
 
 server:	libopie/libopie.a libmissing/libmissing.a opietest-passed opielogin opiesu opiepasswd opieinfo opieftpd opieserv
 
@@ -170,9 +162,6 @@
 	@echo "Copying OPIE user programs"
 	@if test ! -d $(DESTDIR)$(LOCALBIN); then $(MKDIR) $(DESTDIR)$(LOCALBIN); chmod 755 $(DESTDIR)$(LOCALBIN); fi 
 	@cp opiepasswd opieinfo $(DESTDIR)$(LOCALBIN)
-	@echo "Changing ownership"
-	@$(CHOWN) $(OWNER) $(DESTDIR)$(LOCALBIN)/opiepasswd $(DESTDIR)$(LOCALBIN)/opieinfo
-	@chgrp $(GROUP) $(DESTDIR)$(LOCALBIN)/opiepasswd $(DESTDIR)$(LOCALBIN)/opieinfo
 	@echo "Changing file permissions"
 	@chmod 0755 $(DESTDIR)$(LOCALBIN)/opieinfo
 	@chmod 4755 $(DESTDIR)$(LOCALBIN)/opiepasswd
@@ -183,13 +172,11 @@
 	@echo "Changing permissions of OPIE database file"
 	@chmod 0600 $(DESTDIR)$(KEY_FILE)
 	@echo "Changing ownership of OPIE database file"
-	@$(CHOWN) $(OWNER) $(DESTDIR)$(KEY_FILE)
-	@chgrp $(GROUP) $(DESTDIR)$(KEY_FILE)
-	@-if test ! -z "$(DESTDIR)$(LOCK_DIR)"; then echo "Creating OPIE lock directory"; mkdir $(DESTDIR)$(LOCK_DIR); $(CHOWN) 0 $(DESTDIR)$(LOCK_DIR); chgrp 0 $(DESTDIR)$(LOCK_DIR); chmod 0700 $(DESTDIR)$(LOCK_DIR); fi;
-	@-if test ! -z "$(ACCESS_FILE)"; then echo "Creating OPIE access file (don't say we didn't warn you)"; touch $(ACCESS_FILE); $(CHOWN) 0 $(ACCESS_FILE); chgrp 0 $(ACCESS_FILE); chmod 0444 $(ACCESS_FILE); fi;
+	@-if test ! -z "$(DESTDIR)$(LOCK_DIR)"; then echo "Creating OPIE lock directory"; mkdir $(DESTDIR)$(LOCK_DIR);  fi;
+	@-if test ! -z "$(ACCESS_FILE)"; then echo "Creating OPIE access file (don't say we didn't warn you)"; touch $(ACCESS_FILE);  chmod 0444 $(ACCESS_FILE); fi;
 	@echo "Installing manual pages"
 	@if test ! -d $(DESTDIR)$(LOCALMAN); then $(MKDIR) $(DESTDIR)$(LOCALMAN); chmod 755 $(DESTDIR)$(LOCALMAN); fi
-	@for i in 1 4 5 8; do for j in *.$$i; do if test ! -d $(DESTDIR)$(LOCALMAN)/man$$i; then $(MKDIR) $(DESTDIR)$(LOCALMAN)/man$$i; chmod 755 $(DESTDIR)$(LOCALMAN)/man$$i; fi; cp $$j $(DESTDIR)$(LOCALMAN)/man$$i/$$j; $(CHOWN) $(OWNER) $(DESTDIR)$(LOCALMAN)/man$$i/$$j; chgrp $(GROUP) $(DESTDIR)$(LOCALMAN)/man$$i/$$j; chmod 644 $(DESTDIR)$(LOCALMAN)/man$$i/$$j; done; done
+	@for i in 1 4 5 8; do for j in *.$$i; do if test ! -d $(DESTDIR)$(LOCALMAN)/man$$i; then $(MKDIR) $(DESTDIR)$(LOCALMAN)/man$$i; chmod 755 $(DESTDIR)$(LOCALMAN)/man$$i; fi; cp $$j $(DESTDIR)$(LOCALMAN)/man$$i/$$j;  chmod 644 $(DESTDIR)$(LOCALMAN)/man$$i/$$j; done; done
 	@echo "REMEMBER to run opiepasswd on your users immediately."
 
 install: client-install server-install
