Index: openssl-3.1.4/apps/openssl.cnf
===================================================================
--- openssl-3.1.4.orig/apps/openssl.cnf
+++ openssl-3.1.4/apps/openssl.cnf
@@ -70,11 +70,11 @@ engines = engine_section
 # to side-channel attacks and as such have been deprecated.
 
 [provider_sect]
-default = default_sect
+##default = default_sect
 ##legacy = legacy_sect
 
-[default_sect]
-activate = 1
+##[default_sect]
+##activate = 1
 
 ##[legacy_sect]
 ##activate = 1
