Index: pythia8245/examples/Makefile
===================================================================
--- pythia8245.orig/examples/Makefile
+++ pythia8245/examples/Makefile
@@ -114,7 +114,7 @@ endif
 main71 main72 main75: $$@.cc $(PREFIX_LIB)/libpythia8.a
 ifeq ($(FASTJET3_USE),true)
 	$(CXX) $< -o $@ -I$(FASTJET3_INCLUDE) $(CXX_COMMON)\
-	 -L$(FASTJET3_LIB) -Wl,-rpath,$(FASTJET3_LIB) -lfastjet\
+	 -L$(FASTJET3_LIB) -Wl,-rpath,$(FASTJET3_LIB) -lfastjet -lgmp\
 	  $(GZIP_INC) $(GZIP_FLAGS)
 else
 	@echo "Error: $@ requires FASTJET3"
