Index: pythia8306/Makefile
===================================================================
--- pythia8306.orig/Makefile
+++ pythia8306/Makefile
@@ -132,12 +132,12 @@ python: $(LOCAL_LIB)/libpythia8$(LIB_SUF
 
 # Install.
 install: all
-	mkdir -p $(PREFIX_BIN) $(PREFIX_INCLUDE) $(PREFIX_LIB) $(PREFIX_SHARE)
-	rsync -a $(LOCAL_BIN)/* $(PREFIX_BIN)
-	rsync -a $(LOCAL_INCLUDE)/* $(PREFIX_INCLUDE)
-	rsync -a $(LOCAL_LIB)/* $(PREFIX_LIB)
-	rsync -a $(LOCAL_SHARE)/* $(PREFIX_SHARE)
-	rsync -a $(LOCAL_EXAMPLE) $(PREFIX_SHARE)
+	mkdir -p $(DESTDIR)$(PREFIX_BIN) $(DESTDIR)$(PREFIX_INCLUDE) $(DESTDIR)$(PREFIX_LIB) $(DESTDIR)$(PREFIX_SHARE)
+	rsync -a $(LOCAL_BIN)/* $(DESTDIR)$(PREFIX_BIN)
+	rsync -a $(LOCAL_INCLUDE)/* $(DESTDIR)$(PREFIX_INCLUDE)
+	rsync -a $(LOCAL_LIB)/* $(DESTDIR)$(PREFIX_LIB)
+	rsync -a $(LOCAL_SHARE)/* $(DESTDIR)$(PREFIX_SHARE)
+	rsync -a $(LOCAL_EXAMPLE) $(DESTDIR)$(PREFIX_SHARE)
 
 # Clean.
 clean:
