Index: sphinx-autodoc-typehints-1.10.3/tests/conftest.py
===================================================================
--- sphinx-autodoc-typehints-1.10.3.orig/tests/conftest.py	2019-11-11 20:10:08.000000000 +0100
+++ sphinx-autodoc-typehints-1.10.3/tests/conftest.py	2020-03-06 15:49:28.411760937 +0100
@@ -13,10 +13,8 @@ collect_ignore = ['roots']
 
 @pytest.fixture(scope='session')
 def inv(pytestconfig):
-    cache_path = 'python{v.major}.{v.minor}/objects.inv'.format(v=sys.version_info)
-    inv_dict = pytestconfig.cache.get(cache_path, None)
-    if inv_dict is not None:
-        return Inventory(inv_dict)
+    inv_dict = '/usr/share/doc/packages/python3/html/objects.inv'
+    return Inventory(inv_dict)
 
     print("Downloading objects.inv")
     url = 'https://docs.python.org/{v.major}.{v.minor}/objects.inv'.format(v=sys.version_info)
