Index: pytest_reorder-0.1.1/tests/test_pytest_reorder.py
===================================================================
--- pytest_reorder-0.1.1.orig/tests/test_pytest_reorder.py
+++ pytest_reorder-0.1.1/tests/test_pytest_reorder.py
@@ -3,7 +3,7 @@ import re
 
 import pytest
 import subprocess
-from mock import Mock
+from unittest.mock import Mock
 from pytest_reorder import (
     default_reordering_hook, unpack_test_ordering, make_reordering_hook,
     DEFAULT_ORDER
