Index: glbinding-3.5.0/CMakeLists.txt
===================================================================
--- glbinding-3.5.0.orig/CMakeLists.txt
+++ glbinding-3.5.0/CMakeLists.txt
@@ -165,13 +165,14 @@ endif()
 # Installation paths
 if(UNIX AND SYSTEM_DIR_INSTALL)
     # Install into the system (/usr/bin or /usr/local/bin)
+    include(GNUInstallDirs)
     set(INSTALL_ROOT      "share/${project}")       # /usr/[local]/share/<project>
     set(INSTALL_CMAKE     "share/${project}/cmake") # /usr/[local]/share/<project>/cmake
     set(INSTALL_EXAMPLES  "share/${project}")       # /usr/[local]/share/<project>
     set(INSTALL_DATA      "share/${project}")       # /usr/[local]/share/<project>
     set(INSTALL_BIN       "bin")                    # /usr/[local]/bin
-    set(INSTALL_SHARED    "lib")                    # /usr/[local]/lib
-    set(INSTALL_LIB       "lib")                    # /usr/[local]/lib
+    set(INSTALL_SHARED    "${CMAKE_INSTALL_LIBDIR}")# /usr/[local]/lib/<arch>
+    set(INSTALL_LIB       "${CMAKE_INSTALL_LIBDIR}")# /usr/[local]/lib/<arch>
     set(INSTALL_INCLUDE   "include")                # /usr/[local]/include
     set(INSTALL_DOC       "share/doc/${project}")   # /usr/[local]/share/doc/<project>
     set(INSTALL_SHORTCUTS "share/applications")     # /usr/[local]/share/applications
