Index: percona-xtrabackup-2.4.11/storage/innobase/xtrabackup/src/backup_copy.cc
===================================================================
--- percona-xtrabackup-2.4.11.orig/storage/innobase/xtrabackup/src/backup_copy.cc	2018-06-06 11:24:11.008685160 +0200
+++ percona-xtrabackup-2.4.11/storage/innobase/xtrabackup/src/backup_copy.cc	2018-06-06 11:25:48.437090110 +0200
@@ -53,7 +53,6 @@ Place, Suite 330, Boston, MA 02111-1307
 #include <string>
 #include <mysqld.h>
 #include <my_default.h>
-#include <version_check_pl.h>
 #include <sstream>
 #include <algorithm>
 #include "fil_cur.h"
@@ -2298,33 +2297,3 @@ decrypt_decompress()
 	return(ret);
 }
 
-void
-version_check()
-{
-	if (opt_password != NULL) {
-		setenv("option_mysql_password", opt_password, 1);
-	}
-	if (opt_user != NULL) {
-		setenv("option_mysql_user", opt_user, 1);
-	}
-	if (opt_host != NULL) {
-		setenv("option_mysql_host", opt_host, 1);
-	}
-	if (opt_socket != NULL) {
-		setenv("option_mysql_socket", opt_socket, 1);
-	}
-	if (opt_port != 0) {
-		char port[20];
-		snprintf(port, sizeof(port), "%u", opt_port);
-		setenv("option_mysql_port", port, 1);
-	}
-
-	FILE *pipe = popen("perl", "w");
-	if (pipe == NULL) {
-		return;
-	}
-
-	fwrite((const char *) version_check_pl, version_check_pl_len, 1, pipe);
-
-	pclose(pipe);
-}
Index: percona-xtrabackup-2.4.11/storage/innobase/xtrabackup/src/backup_copy.h
===================================================================
--- percona-xtrabackup-2.4.11.orig/storage/innobase/xtrabackup/src/backup_copy.h	2018-06-06 11:24:11.008685160 +0200
+++ percona-xtrabackup-2.4.11/storage/innobase/xtrabackup/src/backup_copy.h	2018-06-06 11:26:10.249180770 +0200
@@ -42,8 +42,6 @@ bool
 copy_back(int argc, char **argv);
 bool
 decrypt_decompress();
-void
-version_check();
 bool
 is_path_separator(char);
 bool
Index: percona-xtrabackup-2.4.11/storage/innobase/xtrabackup/src/CMakeLists.txt
===================================================================
--- percona-xtrabackup-2.4.11.orig/storage/innobase/xtrabackup/src/CMakeLists.txt	2018-06-06 11:24:11.008685160 +0200
+++ percona-xtrabackup-2.4.11/storage/innobase/xtrabackup/src/CMakeLists.txt	2018-06-06 11:27:24.561489639 +0200
@@ -28,13 +28,6 @@ FIND_GCRYPT()
 MYSQL_CHECK_CURL()
 FIND_EV()
 
-# xxd is needed to embed version_check script
-FIND_PROGRAM(XXD_PATH xxd)
-
-IF(NOT XXD_PATH)
-  MESSAGE(FATAL_ERROR "xxd not found. Try to install vim-common.")
-ENDIF(NOT XXD_PATH)
-
 INCLUDE_DIRECTORIES(
   ${CMAKE_SOURCE_DIR}/include
   ${CMAKE_SOURCE_DIR}/storage/innobase/include
@@ -57,14 +50,6 @@ ADD_DEFINITIONS(${SSL_DEFINES})
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/xtrabackup_version.h.in
                ${CMAKE_CURRENT_BINARY_DIR}/xtrabackup_version.h )
 
-ADD_CUSTOM_COMMAND(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/version_check_pl.h
-                   COMMAND ${XXD_PATH} --include version_check.pl
-                   ${CMAKE_CURRENT_BINARY_DIR}/version_check_pl.h
-                   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-
-ADD_CUSTOM_TARGET(GenVersionCheck
-                  DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/version_check_pl.h)
-
 INCLUDE_DIRECTORIES(
   ${CMAKE_SOURCE_DIR}/plugin/keyring
   ${CMAKE_SOURCE_DIR}/plugin/keyring_vault
Index: percona-xtrabackup-2.4.11/storage/innobase/xtrabackup/src/xtrabackup.cc
===================================================================
--- percona-xtrabackup-2.4.11.orig/storage/innobase/xtrabackup/src/xtrabackup.cc	2018-06-06 11:24:11.008685160 +0200
+++ percona-xtrabackup-2.4.11/storage/innobase/xtrabackup/src/xtrabackup.cc	2018-06-06 11:27:42.497564191 +0200
@@ -8218,7 +8218,6 @@ xb_init()
 	if (xtrabackup_backup) {
 
 		if (!opt_noversioncheck) {
-			version_check();
 		}
 
 		if ((mysql_connection = xb_mysql_connect()) == NULL) {
