Index: fipscheck-1.5.0/man/Makefile.in
===================================================================
--- fipscheck-1.5.0.orig/man/Makefile.in	2017-02-23 15:49:16.000000000 +0100
+++ fipscheck-1.5.0/man/Makefile.in	2019-01-03 12:03:26.003429620 +0100
@@ -567,7 +567,7 @@ uninstall-man: uninstall-man3 uninstall-
 
 
 $(dist_man_MANS): $(manpages_XMLSOURCES)
-	for i in $^ ; do xmlto man $$i ; done
+	#for i in $^ ; do xmlto man $$i ; done
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
