--- Makefile.orig
+++ Makefile
@@ -11,7 +11,7 @@ PIP_INSTALL := pip install
 
 ifeq ($(PYVER),python3)
   pyflakes = pyflakes3
-  unittests = unittest3
+  unittests = unittest
   yaml = yaml
 else
 ifeq ($(PYVER),python2)
@@ -19,7 +19,7 @@ ifeq ($(PYVER),python2)
   unittests = unittest
 else
   pyflakes = pyflakes pyflakes3
-  unittests = unittest unittest3
+  unittests = unittest unittest
 endif
 endif
 
