--- cloudinit/net/sysconfig.py.orig
+++ cloudinit/net/sysconfig.py
@@ -998,6 +998,11 @@ class Renderer(renderer.Renderer):
                 network_state, existing_dns_path=dns_path
             )
             if resolv_content:
+                # netconfig checks if /etc/resolv.conf is a symlink and if
+                # that is true will write it's version of the file clobbering
+                # our changes.
+                if os.path.islink(dns_path):
+                    os.unlink(dns_path)
                 util.write_file(dns_path, resolv_content, file_mode)
         if self.networkmanager_conf_path:
             nm_conf_path = subp.target_path(
