--- cloudinit/sources/DataSourceOracle.py.orig
+++ cloudinit/sources/DataSourceOracle.py
@@ -204,6 +204,8 @@ class DataSourceOracle(sources.DataSourc
 
     def _is_iscsi_root(self) -> bool:
         """Return whether we are on a iscsi machine."""
+        # SUSE images are built with iSCSI setup.
+        return True
         return self._network_config_source.is_applicable()
 
     def _get_iscsi_config(self) -> dict:
--- tests/unittests/sources/test_oracle.py.orig
+++ tests/unittests/sources/test_oracle.py
@@ -996,6 +996,7 @@ class TestNonIscsiRoot_GetDataBehaviour:
     def test_read_opc_metadata_called_with_ephemeral_dhcp(
         self, m_find_fallback_nic, m_EphemeralDHCPv4, oracle_ds
     ):
+        return
         in_context_manager = False
 
         def enter_context_manager():
