Index: keylime-6.2.0/keylime.conf
===================================================================
--- keylime-6.2.0.orig/keylime.conf
+++ keylime-6.2.0/keylime.conf
@@ -12,11 +12,13 @@ tls_check_hostnames = False
 # Valid values are "cfssl" or "openssl". For cfssl to work, you must have the
 # go binary installed in your path or in /usr/local/.
 # Note: Revocation list generation is only supported by "cfssl".
-ca_implementation = openssl
+# ca_implementation = openssl
+ca_implementation = cfssl
 
 # Revocation IP & Port used by either the cloud_agent or keylime_ca to receive
 # revocation events from the verifier.
-receive_revocation_ip = 127.0.0.1
+# receive_revocation_ip = 127.0.0.1
+receive_revocation_ip = <REMOTE_IP>
 receive_revocation_port = 8992
 
 #=============================================================================
@@ -24,7 +26,8 @@ receive_revocation_port = 8992
 #=============================================================================
 
 # The binding address and port for the agent server
-cloudagent_ip = 127.0.0.1
+# cloudagent_ip = 127.0.0.1
+cloudagent_ip = 0.0.0.0
 cloudagent_port = 9002
 
 # Address and port where the verifier and tenant can connect to reach the agent.
@@ -33,7 +36,8 @@ agent_contact_ip = 127.0.0.1
 agent_contact_port = 9002
 
 # The address and port of registrar server which agent communicate with
-registrar_ip = 127.0.0.1
+# registrar_ip = 127.0.0.1
+registrar_ip = <REMOTE_IP>
 registrar_port = 8890
 
 # The name of the RSA key that Keylime should use for protecting shares of U/V.
@@ -62,7 +66,8 @@ tpm_ownerpassword = keylime
 # After decryption, the archive will be unzipped to a directory in /var/lib/keylime/secure.
 # Note: the limits on the size of the tmpfs partition set above with the 'secure_size'
 # option will affect this.
-extract_payload_zip = True
+# extract_payload_zip = True
+extract_payload_zip = False
 
 # The agent's UUID.
 # Set to "openstack", it will try to get the UUID from the metadata service.
@@ -73,7 +78,8 @@ extract_payload_zip = True
 # 'dmidecode -s system-uuid'.
 # If you set this to "hostname", Keylime will use the full qualified domain
 # name of current host as the agent id.
-agent_uuid = d432fbb3-d2f1-4a97-9ef7-75bd81c00000
+# agent_uuid = d432fbb3-d2f1-4a97-9ef7-75bd81c00000
+agent_uuid = generate
 
 # Whether to listen for revocation notifications from the verifier or not.
 listen_notfications = True
@@ -137,7 +143,8 @@ ek_handle = generate
 cloudverifier_id = default
 
 # The IP address and port of verifier server binds to
-cloudverifier_ip = 127.0.0.1
+# cloudverifier_ip = 127.0.0.1
+cloudverifier_ip = 0.0.0.0
 cloudverifier_port = 8881
 
 # The address and port of registrar server that verifier communicates with
@@ -250,7 +257,8 @@ revocation_notifier = True
 # The revocation notifier IP address and port used to start the revocation service.
 # If the 'revocation_notifier' option is set to "true", then the verifier
 # automatically starts the revocation service.
-revocation_notifier_ip = 127.0.0.1
+# revocation_notifier_ip = 127.0.0.1
+revocation_notifier_ip = 0.0.0.0
 revocation_notifier_port = 8992
 
 # The verifier limits the size of upload payloads (allowlists) which defaults to
@@ -354,10 +362,12 @@ max_payload_size = 1048576
 # and SHA-512).
 # Note that you can't set a policy on PCR10 and PCR16 because Keylime uses
 # them internally.
-tpm_policy = {"22":["0000000000000000000000000000000000000001","0000000000000000000000000000000000000000000000000000000000000001","000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001","ffffffffffffffffffffffffffffffffffffffff","ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff","ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"],"15":["0000000000000000000000000000000000000000","0000000000000000000000000000000000000000000000000000000000000000","000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"]}
+# tpm_policy = {"22":["0000000000000000000000000000000000000001","0000000000000000000000000000000000000000000000000000000000000001","000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001","ffffffffffffffffffffffffffffffffffffffff","ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff","ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"],"15":["0000000000000000000000000000000000000000","0000000000000000000000000000000000000000000000000000000000000000","000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"]}
+tpm_policy = {}
 
 # Same as 'tpm_policy' but for virtual PCRs.
-vtpm_policy = {"23":["ffffffffffffffffffffffffffffffffffffffff","0000000000000000000000000000000000000000"],"15":"0000000000000000000000000000000000000000"}
+# vtpm_policy = {"23":["ffffffffffffffffffffffffffffffffffffffff","0000000000000000000000000000000000000000"],"15":"0000000000000000000000000000000000000000"}
+vtpm_policy = {}
 
 # Specify the file containing allowlists for processing Linux IMA measurements
 # this file is used if tenant provides "default" as the allowlist file
@@ -409,7 +419,8 @@ max_retries = 10
 # might provide a signed list of EK public key hashes.  Then you could write
 # an ek_check_script that checks the signature of the allowlist and then
 # compares the hash of the given EK with the allowlist.
-require_ek_cert = True
+# require_ek_cert = True
+require_ek_cert = False
 
 # Optional script to execute to check the EK and/or EK certificate against a
 # allowlist or any other additional EK processing you want to do. Runs in
@@ -435,7 +446,8 @@ ek_check_script=
 
 # The registrar's IP address and port used to communicate with other services
 # as well as the bind address for the registrar server.
-registrar_ip = 127.0.0.1
+# registrar_ip = 127.0.0.1
+registrar_ip = 0.0.0.0
 registrar_port = 8890
 registrar_tls_port = 8891
 
