--- verifier.conf.ORIG	2023-01-23 09:36:14.684727116 +0100
+++ verifier.conf	2023-01-23 09:45:13.585042153 +0100
@@ -8,7 +8,8 @@
 uuid = default
 
 # The verifier server IP address and port
-ip = 127.0.0.1
+# ip = 127.0.0.1
+ip = 0.0.0.0
 port = 8881
 
 # The address and port of registrar server that the verifier communicates with
@@ -233,7 +234,8 @@
 enabled_revocation_notifications = ['agent']
 
 # The binding address and port of the revocation notifier service via ZeroMQ.
-zmq_ip = 127.0.0.1
+# zmq_ip = 127.0.0.1
+zmp_ip = 0.0.0.0
 zmq_port = 8992
 
 # Webhook url for revocation notifications.
