---
 Makefile.in |   10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

--- a/Makefile.in
+++ b/Makefile.in
@@ -59,7 +59,7 @@ VERSION = @VERSION@
 # Build procedure
 ###################################################################
 
-.PHONY : all byte opt tools toolslib test test.opt unimaps chamap_data unidata dist depend clean distclean allclean
+.PHONY : do_byte do_opt all byte opt tools toolslib test test.opt unimaps chamap_data unidata dist depend clean distclean allclean
 
 
 SUBDIR = -I internal -I public -I toolslib -I .
@@ -70,12 +70,16 @@ OFLAGS =  $(OOPTIONS) $(INCLUDES) -g
 # Generic targets
 #################
 
-all : byte $(OCAMLBEST) opt unidata unimaps charmap_data locale_data
+all : lib unidata unimaps charmap_data locale_data
 
 # Build the library
 ###################
 
-lib : byte $(OCAMLBEST) opt
+lib : byte $(OCAMLBEST)
+
+do_opt: opt do_byte
+
+do_byte: all
 
 NOOBJ_INT= \
 public/unicodeString.cmi configInt.cmi internal/uReStrParserType.cmi
--- a/internal/unimap.ml
+++ b/internal/unimap.ml
@@ -58,7 +58,7 @@ val of_name : string -> t
 end
 
 
-module Make (Config : ConfigInt.Type) = struct
+module Make (Config : ConfigInt.Type) : Type = struct
 
 type mapping = {no_char : int; tbl : Tbl31.Bytes.t}
 
--- a/public/uCharInfo.ml
+++ b/public/uCharInfo.ml
@@ -298,7 +298,7 @@ val load_composition_exclusion_tbl : uni
 
 end
 
-module Make (Config : ConfigInt.Type) = struct
+module Make (Config : ConfigInt.Type) : Type = struct
 include Unidata.Make(Config)
 
 (* General category *)
