Index: transmission-2.94/gtk/CMakeLists.txt
===================================================================
--- transmission-2.94.orig/gtk/CMakeLists.txt
+++ transmission-2.94/gtk/CMakeLists.txt
@@ -51,10 +51,16 @@ add_custom_command(
 if(ENABLE_NLS)
     find_program(INTLTOOL_MERGE_EXECUTABLE intltool-merge REQUIRED)
     set(${PROJECT_NAME}_DESKTOP_FILE "${PROJECT_BINARY_DIR}/${TR_NAME}-gtk.desktop")
+    set(${PROJECT_NAME}_APPDATA_FILE "${PROJECT_BINARY_DIR}/${TR_NAME}-gtk.appdata.xml")
     add_custom_command(
-        OUTPUT ${${PROJECT_NAME}_DESKTOP_FILE}
+        OUTPUT
+          ${${PROJECT_NAME}_DESKTOP_FILE}
+          ${${PROJECT_NAME}_APPDATA_FILE}
         COMMAND ${INTLTOOL_MERGE_EXECUTABLE} --desktop-style --utf8 ${CMAKE_SOURCE_DIR}/po ${PROJECT_SOURCE_DIR}/transmission-gtk.desktop.in ${${PROJECT_NAME}_DESKTOP_FILE}
-        DEPENDS ${PROJECT_SOURCE_DIR}/transmission-gtk.desktop.in
+        COMMAND ${INTLTOOL_MERGE_EXECUTABLE} --xml-style --utf8 ${CMAKE_SOURCE_DIR}/po ${PROJECT_SOURCE_DIR}/transmission-gtk.appdata.xml.in ${${PROJECT_NAME}_APPDATA_FILE}
+        DEPENDS
+          ${PROJECT_SOURCE_DIR}/transmission-gtk.desktop.in
+          ${PROJECT_SOURCE_DIR}/transmission-gtk.appdata.xml.in
         VERBATIM
     )
 endif()
@@ -188,6 +194,8 @@ endif()
 
 if(ENABLE_NLS)
     install(FILES ${${PROJECT_NAME}_DESKTOP_FILE} DESTINATION ${CMAKE_INSTALL_DATADIR}/applications)
+    install(FILES ${${PROJECT_NAME}_APPDATA_FILE} DESTINATION ${CMAKE_INSTALL_DATADIR}/appdata)
 else()
     install(FILES transmission-gtk.desktop.in DESTINATION ${CMAKE_INSTALL_DATADIR}/applications RENAME ${TR_NAME}-gtk.desktop)
+    install(FILES transmission-gtk.appdata.xml.in DESTINATION ${CMAKE_INSTALL_DATADIR}/appdata RENAME ${TR_NAME}-gtk.appdata.xml)
 endif()
Index: transmission-2.94/gtk/transmission-gtk.appdata.xml.in
===================================================================
--- /dev/null
+++ transmission-2.94/gtk/transmission-gtk.appdata.xml.in
@@ -0,0 +1,26 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<component>
+  <id type="desktop">transmission-gtk.desktop</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <_name>Transmission BitTorrent Client</_name>
+  <project_license>(GPL-2.0 or GPL-3.0) and MIT</project_license>
+  <_summary>Lightweight, yet powerful BitTorrent client</_summary>
+  <url type="homepage">http://www.transmissionbt.com/</url>
+  <description>
+    <_p>Transmission is a fast, easy, and free multi-platform BitTorrent client
+with a focus on being lightweight yet feature-filled. Its simple,
+intuitive interface is designed to integrate tightly with whatever
+computing environment you choose to use. Transmission strikes a balance
+between providing useful functionality without feature bloat.
+Furthermore, it is free for anyone to use or modify.</_p>
+    <_p>This installs the GTK interface for transmission.</_p>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image height="805" width="1252">https://www.transmissionbt.com/images/screenshots/GTK-Large.jpg</image>
+      <caption>Transmission in action</caption>
+    </screenshot>
+  </screenshots>
+  <update_contact>dev@transmissionbt.com</update_contact>
+  <translation type="gettext">transmission-gtk</translation>
+</component>
Index: transmission-2.94/qt/CMakeLists.txt
===================================================================
--- transmission-2.94.orig/qt/CMakeLists.txt
+++ transmission-2.94/qt/CMakeLists.txt
@@ -250,6 +250,8 @@ endif()
 
 install(FILES transmission-qt.desktop DESTINATION ${CMAKE_INSTALL_DATADIR}/applications)
 
+install(FILES transmission-qt.appdata.xml.in DESTINATION ${CMAKE_INSTALL_DATADIR}/appdata RENAME ${TR_NAME}-qt.appdata.xml)
+
 if(ENABLE_NLS)
     install(FILES ${${PROJECT_NAME}_QM_FILES} DESTINATION ${CMAKE_INSTALL_DATADIR}/${TR_NAME}/translations)
 endif()
Index: transmission-2.94/qt/transmission-qt.appdata.xml.in
===================================================================
--- /dev/null
+++ transmission-2.94/qt/transmission-qt.appdata.xml.in
@@ -0,0 +1,26 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<component>
+  <id type="desktop">transmission-qt.desktop</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <name>Transmission (Qt) BitTorrent Client</name>
+  <project_license>(GPL-2.0 or GPL-3.0) and MIT</project_license>
+  <summary>Lightweight, yet powerful BitTorrent client</summary>
+  <url type="homepage">http://www.transmissionbt.com/</url>
+  <description>
+    <p>Transmission is a fast, easy, and free multi-platform BitTorrent client
+with a focus on being lightweight yet feature-filled. Its simple,
+intuitive interface is designed to integrate tightly with whatever
+computing environment you choose to use. Transmission strikes a balance
+between providing useful functionality without feature bloat.
+Furthermore, it is free for anyone to use or modify.</p>
+    <p>
+This installs the QT graphical user interface to transmission.</p>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image height="785" width="1165">https://www.transmissionbt.com/images/screenshots/Qt-Large.jpg</image>
+      <caption>QT interface for transmission</caption>
+    </screenshot>
+  </screenshots>
+  <update_contact>dev@transmissionbt.com</update_contact>
+</component>
