---
 src/module.mk |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

Index: octave-11.1.0/src/module.mk
===================================================================
--- octave-11.1.0.orig/src/module.mk	2026-02-18 18:54:40.000000000 +0100
+++ octave-11.1.0/src/module.mk	2026-03-15 16:11:28.035314785 +0100
@@ -80,6 +80,7 @@ nodist_%canon_reldir%_octave_SOURCES = %
   $(AM_LDFLAGS) \
   $(NO_UNDEFINED_LDFLAG) \
   $(OCTAVE_LINK_OPTS) \
+  -pie \
   $(WARN_LDFLAGS) \
   $(OCTAVE_UNICODE_EXE_LDFLAGS) \
   $(WAYLAND_CLIENT_LDFLAGS)
@@ -105,6 +106,7 @@ nodist_%canon_reldir%_octave_cli_SOURCES
   $(NO_UNDEFINED_LDFLAG) \
   $(OCTAVE_LINK_OPTS) \
   $(OCTAVE_CLI_LTLDFLAGS) \
+  -pie \
   $(WARN_LDFLAGS) \
   $(OCTAVE_UNICODE_EXE_LDFLAGS)
 
@@ -132,6 +134,7 @@ endif
   $(NO_UNDEFINED_LDFLAG) \
   $(OCTAVE_GUI_LINK_OPTS) \
   $(OCTAVE_GUI_LTLDFLAGS) \
+  -pie \
   $(WARN_LDFLAGS) \
   $(OCTAVE_UNICODE_EXE_LDFLAGS)
 
@@ -154,7 +157,8 @@ nodist_%canon_reldir%_mkoctfile_SOURCES
   libgnu/libgnu.la $(LIBS)
 
 %canon_reldir%_mkoctfile_LDFLAGS = \
-  $(OCTAVE_UNICODE_EXE_LDFLAGS)
+  $(OCTAVE_UNICODE_EXE_LDFLAGS) \
+  -pie
 
 %canon_reldir%_mkoctfile_CPPFLAGS = \
   $(SRC_DIR_CPPFLAGS) \
@@ -171,7 +175,8 @@ nodist_%canon_reldir%_octave_config_SOUR
   $(LIBS)
 
 %canon_reldir%_octave_config_LDFLAGS = \
-  $(OCTAVE_UNICODE_EXE_LDFLAGS)
+  $(OCTAVE_UNICODE_EXE_LDFLAGS) \
+  -pie
 
 %canon_reldir%_octave_config_CPPFLAGS = \
   $(SRC_DIR_CPPFLAGS) \
