From 147a49013a2f72f20edd058d9df7175d1283e2b1 Mon Sep 17 00:00:00 2001
From: Chris Sewell <chrisj_sewell@hotmail.com>
Date: Thu, 3 Sep 2020 19:51:14 +0100
Subject: [PATCH] Fix collections deprecation warning

---
 scss/types.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/scss/types.py b/scss/types.py
index c35905a..acaf72c 100644
--- a/scss/types.py
+++ b/scss/types.py
@@ -3,7 +3,10 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
-from collections import Iterable
+try:
+    from collections.abc import Iterable
+except ImportError:
+    from collections import Iterable
 import colorsys
 from fractions import Fraction
 import operator
