From 39e7b6d0144800e2f1c8824b38863f9f1cc52e82 Mon Sep 17 00:00:00 2001
From: Sergey Vlasov <sergey@vlasov.me>
Date: Tue, 7 May 2019 11:54:58 +0000
Subject: [PATCH] pkg_check_lib fix

---
 configure | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure b/configure
index 645d280..cabd0c3 100755
--- a/configure
+++ b/configure
@@ -117,9 +117,9 @@ echo "PKG_CONFIG = $PKG_CONFIG" >> $QMAKE_CACHE
 [[ -n "$LDFLAGS" ]]  && echo "QMAKE_LFLAGS += ${LDFLAGS}"    >> $QMAKE_CACHE
 
 pkg_check_lib() {
-    for P in $@; do
-        if $PKG_CONFIG --print-errors $1; then
-            echo "Found $1"
+    for LIB in $@; do
+        if $PKG_CONFIG --print-errors $LIB; then
+            echo "Found '$LIB'"
         else
             echo
             exit 1
-- 
2.16.4

