Index: equalx-0.7.1/resources/equalx.desktop
===================================================================
--- equalx-0.7.1.orig/resources/equalx.desktop
+++ equalx-0.7.1/resources/equalx.desktop
@@ -1,12 +1,11 @@
 [Desktop Entry]
-Version=0.7.0
 Exec=equalx
 Terminal=false
 Type=Application
 StartupNotify=false
 MimeType=image/png;image/svg+xml;image/jpg;
-Icon=equalx-64x64.png
-Categories=Qt;KDE;Office;Math;
+Icon=equalx
+Categories=Qt;KDE;Office;WordProcessor;
 X-DBUS-StartupType=Unique
 Name=EqualX
 Name[ro]=EgalX
