diff --git a/node_modules/@patternfly/react-console/dist/esm/components/DesktopViewer/MoreInformationDefaultContent.js b/node_modules/@patternfly/react-console/dist/esm/components/DesktopViewer/MoreInformationDefaultContent.js
index fef60fe..47f02b6 100644
--- a/node_modules/@patternfly/react-console/dist/esm/components/DesktopViewer/MoreInformationDefaultContent.js
+++ b/node_modules/@patternfly/react-console/dist/esm/components/DesktopViewer/MoreInformationDefaultContent.js
@@ -12,6 +12,7 @@ export const MoreInformationDefaultContent = ({ textMoreInfoContent = (React.cre
     React.createElement(DescriptionList, { isHorizontal: true },
         React.createElement(MoreInformationInstallVariant, { os: "RHEL, CentOS", content: "sudo yum install virt-viewer" }),
         React.createElement(MoreInformationInstallVariant, { os: "Fedora", content: "sudo dnf install virt-viewer" }),
+        React.createElement(MoreInformationInstallVariant, { os: "SLE, openSUSE", content: "sudo zypper install virt-viewer" }),
         React.createElement(MoreInformationInstallVariant, { os: "Ubuntu, Debian", content: "sudo apt-get install virt-viewer" }),
         React.createElement(MoreInformationInstallVariant, { os: "Windows" },
             React.createElement("div", null,
diff --git a/node_modules/@patternfly/react-console/dist/js/components/DesktopViewer/MoreInformationDefaultContent.js b/node_modules/@patternfly/react-console/dist/js/components/DesktopViewer/MoreInformationDefaultContent.js
index d74617f..30e58aa 100644
--- a/node_modules/@patternfly/react-console/dist/js/components/DesktopViewer/MoreInformationDefaultContent.js
+++ b/node_modules/@patternfly/react-console/dist/js/components/DesktopViewer/MoreInformationDefaultContent.js
@@ -16,6 +16,7 @@ exports.MoreInformationDefaultContent = ({ textMoreInfoContent = (react_1.defaul
     react_1.default.createElement(react_core_1.DescriptionList, { isHorizontal: true },
         react_1.default.createElement(MoreInformationInstallVariant_1.MoreInformationInstallVariant, { os: "RHEL, CentOS", content: "sudo yum install virt-viewer" }),
         react_1.default.createElement(MoreInformationInstallVariant_1.MoreInformationInstallVariant, { os: "Fedora", content: "sudo dnf install virt-viewer" }),
+        react_1.default.createElement(MoreInformationInstallVariant_1.MoreInformationInstallVariant, { os: "SLE, openSUSE", content: "sudo zypper install virt-viewer" }),
         react_1.default.createElement(MoreInformationInstallVariant_1.MoreInformationInstallVariant, { os: "Ubuntu, Debian", content: "sudo apt-get install virt-viewer" }),
         react_1.default.createElement(MoreInformationInstallVariant_1.MoreInformationInstallVariant, { os: "Windows" },
             react_1.default.createElement("div", null,
diff --git a/node_modules/@patternfly/react-console/src/components/DesktopViewer/MoreInformationDefaultContent.tsx b/node_modules/@patternfly/react-console/src/components/DesktopViewer/MoreInformationDefaultContent.tsx
index 02db247..68fb507 100644
--- a/node_modules/@patternfly/react-console/src/components/DesktopViewer/MoreInformationDefaultContent.tsx
+++ b/node_modules/@patternfly/react-console/src/components/DesktopViewer/MoreInformationDefaultContent.tsx
@@ -24,6 +24,7 @@ export const MoreInformationDefaultContent: React.FunctionComponent<MoreInformat
     <DescriptionList isHorizontal>
       <MoreInformationInstallVariant os="RHEL, CentOS" content="sudo yum install virt-viewer" />
       <MoreInformationInstallVariant os="Fedora" content="sudo dnf install virt-viewer" />
+      <MoreInformationInstallVariant os="SLE, openSUSE" content="sudo zypper install virt-viewer" />
       <MoreInformationInstallVariant os="Ubuntu, Debian" content="sudo apt-get install virt-viewer" />
       <MoreInformationInstallVariant os="Windows">
         <div>
