https://github.com/backtracking/ptmap/issues/16
--- ptmap-2.0.4.old/ptmap.ml	2018-07-23 08:03:49.000000000 +0100
+++ ptmap-2.0.4.new/ptmap.ml	2020-04-22 16:17:50.755378816 +0100
@@ -188,6 +188,11 @@
   | Leaf (k, v) as t -> if pr k v then t else Empty
   | Branch (p,m,t0,t1) -> branch (p, m, filter pr t0, filter pr t1)
 
+let rec filter_map pr = function
+  | Empty -> Empty
+  | Leaf (k, v) -> (match pr k v with Some v' -> Leaf (k, v') | None -> Empty)
+  | Branch (p,m,t0,t1) -> branch (p, m, filter_map pr t0, filter_map pr t1)
+
 let partition p s =
   let rec part (t,f as acc) = function
     | Empty -> acc
