Index: Python-3.7.0/Lib/test/test_subprocess.py
===================================================================
--- Python-3.7.0.orig/Lib/test/test_subprocess.py
+++ Python-3.7.0/Lib/test/test_subprocess.py
@@ -1088,7 +1088,8 @@ class ProcessTestCase(BaseTestCase):
         self.assertIn("0.0001", str(c.exception))  # For coverage of __str__.
         # Some heavily loaded buildbots (sparc Debian 3.x) require this much
         # time to start.
-        self.assertEqual(p.wait(timeout=3), 0)
+        # OBS might require even more
+        self.assertEqual(p.wait(timeout=10), 0)
 
     def test_invalid_bufsize(self):
         # an invalid type of the bufsize argument should raise
