Index: tuned-2.8.0/dbus.conf
===================================================================
--- tuned-2.8.0.orig/dbus.conf
+++ tuned-2.8.0/dbus.conf
@@ -4,7 +4,6 @@
 
 <busconfig>
 	<policy context="default">
-		<allow receive_sender="com.redhat.tuned" />
 		<allow send_destination="com.redhat.tuned" send_interface="org.freedesktop.DBus.Introspectable" />
 		<allow send_destination="com.redhat.tuned" send_interface="com.redhat.tuned.control" />
 	</policy>
