From c8bcd716800b1e842301fbef8a0adf16d9042eb1 Mon Sep 17 00:00:00 2001
From: David Redondo <kde@david-redondo.de>
Date: Wed, 15 Dec 2021 17:19:15 +0100
Subject: [PATCH] SwitcherBackend: Only create the workaround window on wayland

Not needed on X
BUG:443968


(cherry picked from commit cad50cb7961b7ebe87b08334ea11ec2f83979680)
---
 imports/activitymanager/switcherbackend.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/imports/activitymanager/switcherbackend.cpp b/imports/activitymanager/switcherbackend.cpp
index d97ef9b94..8a12da5e6 100644
--- a/imports/activitymanager/switcherbackend.cpp
+++ b/imports/activitymanager/switcherbackend.cpp
@@ -229,7 +229,7 @@ void SwitcherBackend::switchToActivity(Direction direction)
 void SwitcherBackend::keybdSwitchedToAnotherActivity()
 {
     m_lastInvokedAction = dynamic_cast<QAction *>(sender());
-    if (!qGuiApp->focusWindow() && !m_inputWindow) {
+    if (KWindowSystem::isPlatformWayland() && !qGuiApp->focusWindow() && !m_inputWindow) {
         // create a new Window so the compositor sends us modifier info
         m_inputWindow = new QRasterWindow();
         m_inputWindow->setGeometry(0, 0, 1, 1);
-- 
2.33.1

