--- openssl-ibmpkcs11-1.0.0/openssl.cnf.sample.in	2011-05-11 17:57:02.000000000 -0400
+++ openssl-ibmpkcs11-1.0.0-key/openssl.cnf.sample.in	2011-08-05 19:42:30.000000000 -0400
@@ -17,9 +17,9 @@
 ibmpkcs11 = ibmpkcs11_section
 
 [ibmpkcs11_section]
-SLOT_ID=0
-dynamic_path = @LIBDIR@/libibmpkcs11.so
+dynamic_path = @LIBDIR@/libibmpkcs11.so.0.0.0
 engine_id = ibmpkcs11
+SLOT_ID=0
 #
 # The following algorithms will be enabled by these parameters
 # to the default_algorithms line. Any combination of these is valid,
