From: Petr Uzel <petr.uzel@suse.cz>
Subject: parted: Add an "resize" alias for "resizepart"
Patch-mainline: no, compatibility for parted < 3.0 syntax
---
 parted/parted.c |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

Index: parted-3.2/parted/parted.c
===================================================================
--- parted-3.2.orig/parted/parted.c
+++ parted-3.2/parted/parted.c
@@ -2055,9 +2055,11 @@ NULL),
 
 command_register (commands, command_create (
         str_list_create_unique ("resize", _("resize"), NULL),
-        do_resize,
-        NULL,
-        str_list_create (_(N_("The resize command was removed in parted 3.0\n")), NULL), 1));
+        do_resizepart,
+        str_list_create (
+_("resizepart NUMBER END                    resize partition NUMBER"),
+NULL),
+        str_list_create (_(number_msg), _(end_msg), NULL), 1));
 
 command_register (commands, command_create (
         str_list_create_unique ("resizepart", _("resizepart"), NULL),
