--- a/Makefile.am
+++ b/Makefile.am
@@ -9,8 +9,8 @@ ACLOCAL_AMFLAGS = -I m4
 if CPLUFFXX
 DIR_LIBCPLUFFXX = libcpluffxx
 endif
-SUBDIRS = libcpluff $(DIR_LIBCPLUFFXX) loader console po test docsrc doc
-DIST_SUBDIRS = libcpluff libcpluffxx loader console po test docsrc doc examples
+SUBDIRS = libcpluff $(DIR_LIBCPLUFFXX) loader console
+DIST_SUBDIRS = libcpluff libcpluffxx loader console
 DOC_SUBDIRS = libcpluff $(DIR_LIBCPLUFFXX)
 
 EXTRA_DIST = COPYRIGHT.txt INSTALL.txt ChangeLog.txt autogen.sh plugin.xsd
--- a/configure.ac
+++ b/configure.ac
@@ -32,6 +32,8 @@ AC_CONFIG_SRCDIR([libcpluff/cpluff.h])
 AC_CONFIG_AUX_DIR([auxliary])
 AC_CONFIG_HEADERS([config.h])
 
+PKG_PROG_PKG_CONFIG
+PKG_INSTALLDIR
 
 # Version information
 # -------------------
@@ -386,11 +388,13 @@ AC_SUBST(CPLUFF_LOADER)
 # Output Makefiles
 # ----------------
 AC_CONFIG_FILES([Makefile
+libcpluff/libcpluff.pc
 libcpluff/Makefile
 libcpluff/cpluffdef.h
 libcpluff/docsrc/Makefile
 libcpluff/docsrc/Doxyfile-ref
 libcpluff/docsrc/Doxyfile-impl
+libcpluffxx/libcpluffxx.pc
 libcpluffxx/Makefile
 libcpluffxx/cpluffxx/Makefile
 libcpluffxx/cpluffxx/sharedptr.h
--- /dev/null
+++ b/libcpluff/libcpluff.pc.in
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=${prefix}
+includedir=@includedir@
+libdir=@libdir@
+
+Name: libcpluff
+Description: C-Pluff, a plug-in framework for C
+URL: https://github.com/jlehtine/c-pluff
+Version: @PACKAGE_VERSION@
+Cflags: -I${includedir}
+Libs: -L${libdir} -lcpluff
--- /dev/null
+++ b/libcpluffxx/libcpluffxx.pc.in
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=${prefix}
+includedir=@includedir@
+libdir=@libdir@
+
+Name: libcpluffxx
+Description: C-Pluff, a plug-in framework for C++
+URL: https://github.com/jlehtine/c-pluff
+Version: @PACKAGE_VERSION@
+Cflags: -I${includedir}
+Libs: -L${libdir} -lcpluffxx
+Requires.private: libcpluff
--- a/libcpluff/Makefile.am
+++ b/libcpluff/Makefile.am
@@ -27,6 +27,8 @@ libcpluff_la_LDFLAGS = -no-undefined -ve
 
 include_HEADERS = cpluff.h cpluffdef.h
 
+pkgconfig_DATA = libcpluff.pc
+
 doc: refdoc
 
 refdoc: doc/reference/c-api/index.html
--- a/libcpluffxx/Makefile.am
+++ b/libcpluffxx/Makefile.am
@@ -26,6 +26,8 @@ libcpluffxx_la_LDFLAGS = -no-undefined -
 
 include_HEADERS = cpluffxx.h
 
+pkgconfig_DATA = libcpluffxx.pc
+
 doc: refdoc
 
 refdoc: doc/reference/cxx-api/index.html
