--- nfs4-acl-tools-0.3.3-git4cb4187e83.orig/configure.ac
+++ nfs4-acl-tools-0.3.3-git4cb4187e83/configure.ac
@@ -23,12 +23,8 @@ AC_FUNC_MALLOC
 AC_FUNC_STAT
 AC_CHECK_FUNCS([memset strchr])
 
-AC_CHECK_LIB([attr], [getxattr])
-
 AC_PACKAGE_GLOBALS([nfs4acl])
 AC_PACKAGE_UTILITIES([nfs4acl])
-AC_PACKAGE_NEED_ATTR_XATTR_H
-AC_PACKAGE_NEED_GETXATTR_LIBATTR
 AC_MANUAL_FORMAT
 
 AC_FUNC_GCC_VISIBILITY
--- nfs4-acl-tools-0.3.3-git4cb4187e83.orig/libnfs4acl/nfs4_acl_for_path.c
+++ nfs4-acl-tools-0.3.3-git4cb4187e83/libnfs4acl/nfs4_acl_for_path.c
@@ -32,13 +32,17 @@
  *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-
+#include <errno.h>
 #include <sys/types.h>
-#include <attr/xattr.h>
+#include <sys/xattr.h>
 #include <sys/stat.h>
 #include <stdio.h>
 #include "libacl_nfs4.h"
 
+#ifndef ENOATTR
+#define ENOATTR ENODATA        /* No such attribute */
+#endif
+
 static int nfs4_getxattr(const char *, void *, size_t);
 
 /* returns a newly-allocated struct nfs4_acl for `path', or NULL on error. */
--- nfs4-acl-tools-0.3.3-git4cb4187e83.orig/libnfs4acl/nfs4_set_acl.c
+++ nfs4-acl-tools-0.3.3-git4cb4187e83/libnfs4acl/nfs4_set_acl.c
@@ -29,14 +29,19 @@
  *  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+#include <errno.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/errno.h>
-#include <attr/xattr.h>
+#include <sys/xattr.h>
 #include <unistd.h>
 #include <stdio.h>
 #include "libacl_nfs4.h"
 
+#ifndef ENOATTR
+#define ENOATTR ENODATA        /* No such attribute */
+#endif
+
 int nfs4_set_acl(struct nfs4_acl *acl, const char *path)
 {
 	int res = 0;
--- nfs4-acl-tools-0.3.3-git4cb4187e83.orig/nfs4_setfacl/nfs4_setfacl.c
+++ nfs4-acl-tools-0.3.3-git4cb4187e83/nfs4_setfacl/nfs4_setfacl.c
@@ -34,11 +34,12 @@
  */
 
 #define _XOPEN_SOURCE 500
+#include <errno.h>
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <sys/errno.h>
-#include <attr/xattr.h>
+#include <sys/xattr.h>
 #include <stdlib.h>
 #include <unistd.h>
 #include <ctype.h>
--- nfs4-acl-tools-0.3.3-git4cb4187e83.orig/libnfs4acl/Makefile
+++ nfs4-acl-tools-0.3.3-git4cb4187e83/libnfs4acl/Makefile
@@ -37,7 +37,7 @@ LTLDFLAGS += -Wl,--version-script,$(TOPD
 include $(TOPDIR)/include/builddefs
 
 LTLIBRARY = libnfs4acl.la
-LTLIBS = -lattr
+LTLIBS =
 LTDEPENDENCIES = $(TOPDIR)/include/nfs4.h
 
 # 3 2 1  ->  .so.2.1.2
--- nfs4-acl-tools-0.3.3-git4cb4187e83.orig/include/builddefs.in
+++ nfs4-acl-tools-0.3.3-git4cb4187e83/include/builddefs.in
@@ -40,7 +40,7 @@ OPTIMIZER = @opt_build@
 MALLOCLIB = @malloc_lib@
 
 LIBNFS4ACL = $(TOPDIR)/libnfs4acl/libnfs4acl.la
-LIBATTR = @libattr@
+LIBATTR =
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
