Index: GemPC410/GemPC410Utils.c
===================================================================
--- GemPC410/GemPC410Utils.c.orig
+++ GemPC410/GemPC410Utils.c
@@ -71,7 +71,7 @@ RESPONSECODE OpenGemPC410(DWORD lun, DWO
 				if (channel == 0x0102E8)
 					channel = 4;
 
-	sprintf(dev_name, "/dev/pcsc/%d", (int) channel);
+	sprintf(dev_name, "/dev/ttyS%d", (int) channel - 1);
 
 	return OpenGemPC410ByName(lun, dev_name);
 }	/* OpenGemPC410 */
