--- yazi-plugin/Cargo.toml
+++ yazi-plugin/Cargo.toml
@@ -21,7 +21,7 @@ ansi-to-tui   = "3.1.0"
 anyhow        = "1.0.82"
 futures       = "0.3.30"
 md-5          = "0.10.6"
-mlua          = { version = "0.9.7", features = [ "lua54", "vendored", "serialize", "macros", "async" ] }
+mlua          = { version = "0.9.7", features = [ "lua54", "serialize", "macros", "async" ] }
 parking_lot   = "0.12.1"
 ratatui       = "=0.26.1"
 serde         = "1.0.198"
