Index: tiff-4.0.3/tools/tiff2pdf.c
===================================================================
--- tiff-4.0.3.orig/tools/tiff2pdf.c
+++ tiff-4.0.3/tools/tiff2pdf.c
@@ -1264,6 +1264,15 @@ void t2p_read_tiff_data(T2P* t2p, TIFF*
                 return;
 
         }
+	if(t2p->tiff_compression != COMPRESSION_LZW && 
+	   t2p->tiff_compression != COMPRESSION_NONE){
+		TIFFWarning(
+			  TIFF2PDF_MODULE,
+			  "%s is not compressed with LZW or NONE.\n"
+			  "tiff2pdf result may be incorrect in that case.\n"
+			  "Consider to use tiffcp(1) to change compress algorithm first.",
+			  TIFFFileName(input)    );
+	}
         if( TIFFIsCODECConfigured(t2p->tiff_compression) == 0){
 		TIFFError(
 			TIFF2PDF_MODULE, 
