diff -Naur a/configure b/configure
--- a/configure	2012-03-04 05:37:42.000000000 -0600
+++ b/configure	2012-03-22 11:07:04.622010969 -0500
@@ -17192,7 +17192,7 @@
   MOO_LIB_DIR="${libdir}/$MOO_PACKAGE_NAME"
 
 
-  MOO_DOC_DIR="${datadir}/doc/$MOO_PACKAGE_NAME"
+  MOO_DOC_DIR="${datadir}/doc/packages/$MOO_PACKAGE_NAME"
 
   MOO_HELP_DIR="${MOO_DOC_DIR}/help"
 
@@ -17580,7 +17580,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_GTHREAD_LIBS=`$PKG_CONFIG --libs "gthread-2.0" 2>/dev/null`
+  pkg_cv_GTHREAD_LIBS=`$PKG_CONFIG --libs "gthread-2.0 gmodule-no-export-2.0" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
