Index: liblightgrep-1.2.1/m4/ax_boost_base.m4
===================================================================
--- liblightgrep-1.2.1.orig/m4/ax_boost_base.m4
+++ liblightgrep-1.2.1/m4/ax_boost_base.m4
@@ -91,7 +91,7 @@ if test "x$want_boost" = "xyes"; then
     dnl are found, e.g. when only header-only libraries are installed!
     libsubdirs="lib"
     ax_arch=`uname -m`
-    if test $ax_arch = x86_64 -o $ax_arch = ppc64 -o $ax_arch = s390x -o $ax_arch = sparc64; then
+    if test $ax_arch = x86_64 -o $ax_arch = aarch64 -o $ax_arch = ppc64 -o $ax_arch = ppc64le -o $ax_arch = s390x -o $ax_arch = sparc64 -o $ax_arch = riscv64; then
         libsubdirs="lib64 lib lib64"
     fi
 
