diff -ur pyqt5_sip-12.19.0/PKG-INFO pyqt5_sip-12.19.0_fix/PKG-INFO
--- pyqt5_sip-12.19.0/PKG-INFO	2026-08-02 13:20:00.472665800 +0200
+++ pyqt5_sip-12.19.0_fix/PKG-INFO	2026-09-26 17:48:51.297267271 +0200
@@ -9,7 +9,7 @@
 Platform: X11
 Platform: macOS
 Platform: Windows
-Requires-Python: >=3.10
+Requires-Python: >=3.6
 License-File: LICENSE
 Dynamic: license
 Dynamic: license-file
diff -ur pyqt5_sip-12.19.0/PyQt5_sip.egg-info/PKG-INFO pyqt5_sip-12.19.0_fix/PyQt5_sip.egg-info/PKG-INFO
--- pyqt5_sip-12.19.0/PyQt5_sip.egg-info/PKG-INFO	2026-08-02 13:20:00.000000000 +0200
+++ pyqt5_sip-12.19.0_fix/PyQt5_sip.egg-info/PKG-INFO	2026-09-26 17:48:51.297319972 +0200
@@ -9,7 +9,7 @@
 Platform: X11
 Platform: macOS
 Platform: Windows
-Requires-Python: >=3.10
+Requires-Python: >=3.6
 License-File: LICENSE
 Dynamic: license
 Dynamic: license-file
diff -ur pyqt5_sip-12.19.0/setup.py pyqt5_sip-12.19.0_fix/setup.py
--- pyqt5_sip-12.19.0/setup.py	2026-08-02 13:20:00.000000000 +0200
+++ pyqt5_sip-12.19.0_fix/setup.py	2026-09-26 17:48:51.297488860 +0200
@@ -18,6 +18,6 @@
         name='PyQt5_sip',
         version='12.19.0',
         license='BSD-2-Clause',
-        python_requires='>=3.10',
+        python_requires='>=3.6',
         ext_modules=[module]
      )
diff -ur pyqt5_sip-12.19.0/sip.h pyqt5_sip-12.19.0_fix/sip.h
--- pyqt5_sip-12.19.0/sip.h	2026-08-02 13:20:00.000000000 +0200
+++ pyqt5_sip-12.19.0_fix/sip.h	2026-09-26 17:48:51.300637452 +0200
@@ -14,8 +14,8 @@
 #include <Python.h>
 
 /* Sanity check on the Python version. */
-#if PY_VERSION_HEX < 0x030a0000
-#error "This version of PyQt5.sip requires Python v3.10 or later"
+#if PY_VERSION_HEX < 0x03060000
+#error "This version of PyQt5.sip requires Python v3.6 or later"
 #endif
 
 
diff -ur pyqt5_sip-12.19.0/siplib.c pyqt5_sip-12.19.0_fix/siplib.c
--- pyqt5_sip-12.19.0/siplib.c	2026-08-02 13:20:00.000000000 +0200
+++ pyqt5_sip-12.19.0_fix/siplib.c	2026-09-26 17:49:17.398305059 +0200
@@ -1022,6 +1022,10 @@
     PyObject *obj;
     PyMethodDef *md;
 
+#if PY_VERSION_HEX < 0x03070000 && defined(WITH_THREAD)
+    PyEval_InitThreads();
+#endif
+
     /* Add the SIP version number. */
     obj = PyLong_FromLong(SIP_VERSION);
 
@@ -13305,12 +13309,16 @@
         Py_ssize_t *start, Py_ssize_t *stop, Py_ssize_t *step,
         Py_ssize_t *slicelength)
 {
+#if PY_VERSION_HEX >= 0x03070000
     if (PySlice_Unpack(slice, start, stop, step) < 0)
         return -1;
 
     *slicelength = PySlice_AdjustIndices(length, start, stop, *step);
 
     return 0;
+#else
+    return PySlice_GetIndicesEx(slice, length, start, stop, step, slicelength);
+#endif
 }
 
 
