--- xf86-video-openchrome-0.6.0/src/via_driver.c.orig	2018-05-23 15:05:21.167506179 +0000
+++ xf86-video-openchrome-0.6.0/src/via_driver.c	2018-05-23 15:05:48.931619056 +0000
@@ -1675,7 +1675,7 @@ VIACreateScreenResources(ScreenPtr pScre
         return FALSE;
 
     if (pVia->shadowFB) {
-        if (!shadowAdd(pScreen, rootPixmap, shadowUpdatePackedWeak(),
+        if (!shadowAdd(pScreen, rootPixmap, shadowUpdatePacked,
                         viaShadowWindow, 0, NULL))
             return FALSE;
     }
