--- rinetd.conf.sample
+++ rinetd.conf.sample	2001/03/09 17:34:48
@@ -0,0 +1,68 @@
+# This is a sample configuration file for rinetd
+# Please refer to the rinetd manual page, adjust it to your personal needs
+# and copy it to /etc/rinetd.conf
+
+# FORWARDING RULES
+
+# A simple forwarding rule
+#
+# Format: <bindaddress> <bindport> <connectaddress> <connectport>
+#
+# This would redirect all connections to port 80 of the "real" IP address
+# 206.125.69.81, which could be a virtual interface, through rinetd to port
+# 80 of the address 10.1.1.2, which would typically be a machine on the
+# inside of a firewall which has no direct routing to the outside world.
+# Service names can be specified instead of port numbers. On most systems,
+# service names are defined in the file /etc/services.
+# Both IP addresses and hostnames are accepted for bindaddress and
+# connectaddress.
+# Optionally another column <sourceaddress> can be given to specify
+# a local address to bind to specifically on outgoing connections.
+
+206.125.69.81 80 10.1.1.2 80
+
+# ALLOW AND DENY RULES
+
+# Allow rules which appear before the first forwarding rule are applied
+# globally: if at least one global allow rule exists, and the address of a
+# new connection does not satisfy at least one of the global allow rules,
+# that connection is immediately rejected, regardless of any other rules.
+# 
+# Allow rules which appear after a specific forwarding rule apply to that
+# forwarding rule only. If at least one allow rule exists for a particular
+# forwarding rule, and the address of a new connection does not satisfy at
+# least one of the allow rules for that forwarding rule, that connection
+# is immediately rejected, regardless of any other rules.
+#
+# Format: [allow|deny] <pattern>
+#
+# Patterns can contain the following characters: 0, 1, 2, 3, 4, 5, 6, 7, 8,
+# 9, . (period), ?, and *. The ? wildcard matches any one character. The *
+# wildcard matches any number of characters, including zero.
+#
+# Host names are NOT permitted in allow and deny rules. The performance
+# cost of looking up IP addresses to find their corresponding names is
+# prohibitive. Since rinetd is a single process server, all other connections
+# would be forced to pause during the address lookup.
+#
+# This allow rule matches all IP addresses in the 206.125.69 class C domain.
+
+allow 206.125.69.*
+
+# LOGGING
+
+# rinetd is able to produce a log file in either of two formats:
+# tab-delimited and web server-style "common log format."
+# By default, rinetd does not produce a log file.
+#
+# Format: logfile <log-file-location>
+#
+# By default, rinetd logs in a simple tab-delimited format.
+# To activate web server-style "common log format" logging, add the following
+# line to the configuration file:
+#
+# logcommon
+#
+
+logfile /var/log/rinetd.log
+logcommon
