Index: neatvnc-0.9.2/meson.build
===================================================================
--- neatvnc-0.9.2.orig/meson.build
+++ neatvnc-0.9.2/meson.build
@@ -54,7 +54,7 @@ libturbojpeg = dependency('libturbojpeg'
 gnutls = dependency('gnutls', required: get_option('tls'))
 nettle = dependency('nettle', required: get_option('nettle'))
 hogweed = dependency('hogweed', required: get_option('nettle'))
-gmp = dependency('gmp', required: get_option('nettle'))
+gmp = cc.find_library('gmp')
 zlib = dependency('zlib')
 gbm = dependency('gbm', required: get_option('gbm'))
 libdrm = dependency('libdrm', required: get_option('h264'))
