Index: tornado-6.0.4/tornado/util.py
===================================================================
--- tornado-6.0.4.orig/tornado/util.py	2020-03-11 11:42:49.610254636 +0100
+++ tornado-6.0.4/tornado/util.py	2020-03-11 11:43:51.470603323 +0100
@@ -468,5 +468,7 @@ else:
 def doctests():
     # type: () -> unittest.TestSuite
     import doctest
+    import warnings
+    warnings.simplefilter("ignore", ResourceWarning)
 
     return doctest.DocTestSuite()
Index: tornado-6.0.4/tornado/httputil.py
===================================================================
--- tornado-6.0.4.orig/tornado/httputil.py	2020-03-11 11:42:49.610254636 +0100
+++ tornado-6.0.4/tornado/httputil.py	2020-03-11 11:44:46.178911693 +0100
@@ -1032,6 +1032,8 @@ def encode_username_password(
 def doctests():
     # type: () -> unittest.TestSuite
     import doctest
+    import warnings
+    warnings.simplefilter("ignore", ResourceWarning)
 
     return doctest.DocTestSuite()
 
Index: tornado-6.0.4/tornado/iostream.py
===================================================================
--- tornado-6.0.4.orig/tornado/iostream.py	2020-03-11 11:42:49.610254636 +0100
+++ tornado-6.0.4/tornado/iostream.py	2020-03-11 11:45:31.015164413 +0100
@@ -1677,5 +1677,7 @@ class PipeIOStream(BaseIOStream):
 
 def doctests() -> Any:
     import doctest
+    import warnings
+    warnings.simplefilter("ignore", ResourceWarning)
 
     return doctest.DocTestSuite()
