Index: tornado-6.0.4/tornado/test/testing_test.py
===================================================================
--- tornado-6.0.4.orig/tornado/test/testing_test.py	2020-03-11 11:37:45.016537755 +0100
+++ tornado-6.0.4/tornado/test/testing_test.py	2020-03-11 11:38:38.740840572 +0100
@@ -54,7 +54,7 @@ class AsyncTestCaseTest(AsyncTestCase):
         self.io_loop.add_timeout(self.io_loop.time() + 0.00, self.stop)
         self.wait(timeout=0.02)
         self.io_loop.add_timeout(self.io_loop.time() + 0.03, self.stop)
-        self.wait(timeout=0.15)
+        self.wait(timeout=0.25)
 
 
 class LeakTest(AsyncTestCase):
@@ -262,7 +262,7 @@ class GenTest(AsyncTestCase):
         self.finished = True
 
     def test_timeout_environment_variable(self):
-        @gen_test(timeout=0.5)
+        @gen_test(timeout=0.9)
         def test_long_timeout(self):
             yield gen.sleep(0.25)
 
