--- orig/makefile	2021-02-08 16:10:42.938826333 +0100
+++ sunwait/makefile	2021-02-08 16:11:22.119084413 +0100
@@ -7,7 +7,7 @@
 
 
 C=gcc
-CFLAGS=-c -Wall 
+CFLAGS=-c -Wall $(RPM_OPT_FLAGS)
 LDFLAGS= -lm -lstdc++
 SOURCES=sunwait.cpp sunriset.cpp print.cpp
 HEADERS=sunwait.h sunriset.h print.h
@@ -15,7 +15,7 @@
 EXECUTABLE=sunwait
 
 all: $(SOURCES) $(EXECUTABLE)
-	
+
 $(EXECUTABLE): $(OBJECTS)
 	$(C) $(OBJECTS) -o $@ $(LDFLAGS)
 
@@ -23,6 +23,6 @@
 	$(C) $(CFLAGS) $< -o $@
 
 clean:
-	rm -f *.o sunwait
+	$(RM) *.o sunwait
 
 
