Index: persistent-4.6.3/persistent/tests/test_picklecache.py
===================================================================
--- persistent-4.6.3.orig/persistent/tests/test_picklecache.py
+++ persistent-4.6.3/persistent/tests/test_picklecache.py
@@ -1124,13 +1124,12 @@ class CPickleCacheTests(PickleCacheTestM
         # interface declaration to the C implementation.
         from persistent.interfaces import IExtendedPickleCache
         from zope.interface.verify import verifyObject
-        from zope.interface.exceptions import DoesNotImplement
-        from zope.interface.exceptions import BrokenImplementation
+        from zope.interface.exceptions import MultipleInvalid
         # We don't claim to implement it.
-        with self.assertRaises(DoesNotImplement):
+        with self.assertRaises(MultipleInvalid):
             verifyObject(IExtendedPickleCache, self._makeOne())
         # And we don't even provide everything it asks for.
-        with self.assertRaises(BrokenImplementation):
+        with self.assertRaises(MultipleInvalid):
             verifyObject(IExtendedPickleCache, self._makeOne(), tentative=True)
 
     def test___setitem___persistent_class(self):
