From 1d821ca3f5ba3a750e4760b08ea078bf162adbb4 Mon Sep 17 00:00:00 2001
From: Todd <toddrme2178@gmail.com>
Date: Mon, 26 Nov 2018 18:30:56 -0500
Subject: [PATCH] Also catch exceptions in generating documentation

---
 setup.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 6adde0c1..69a88b2d 100644
--- a/setup.py
+++ b/setup.py
@@ -267,7 +267,11 @@ class MyBuild(build.build):
     def run(self):
         build_tvtk_classes_zip()
         build.build.run(self)
-        self.run_command('gen_docs')
+        try:
+            self.run_command('gen_docs')
+        except:
+            log.warn("Couldn't generate documentation:\n%s" %
+                     traceback.format_exception(*sys.exc_info()))
         try:
             self.run_command('build_docs')
         except:
