--- check_bind.sh.orig	2013-01-07 15:40:23.000000000 +0200
+++ check_bind.sh	2014-11-16 02:22:42.000000000 +0200
@@ -203,49 +203,49 @@
             ;;
     esac
 
-    if [ "$succ_1st" == '' ]
+    if [ -z "$succ_1st" ]
     then
         success=0
     else
         success=`expr $succ_1st - $succ_2nd`
     fi
-    if [ "$ref_1st" == '' ]
+    if [ -z "$ref_1st" ]
     then
         referral=0
     else
         referral=`expr $ref_1st - $ref_2nd`
     fi
-    if [ "$nxrr_1st" == '' ]
+    if [ -z "$nxrr_1st" ]
     then
         nxrrset=0
     else
         nxrrset=`expr $nxrr_1st - $nxrr_2nd`
     fi
-    if [ "$nxdom_1st" == '' ]
+    if [ -z "$nxdom_1st" ]
     then
         nxdomain=0
     else
         nxdomain=`expr $nxdom_1st - $nxdom_2nd`
     fi
-    if [ "$rec_1st" == '' ]
+    if [ -z "$rec_1st" ]
     then
         recursion=0
     else
         recursion=`expr $rec_1st - $rec_2nd`
     fi
-    if [ "$fail_1st" == '' ]
+    if [ -z "$fail_1st" ]
     then
         failure=0
     else
         failure=`expr $fail_1st - $fail_2nd`
     fi
-    if [ "$dup_1st" == '' ]
+    if [ -z "$dup_1st" ]
     then
         duplicate=0
     else
         duplicate=`expr $dup_1st - $dup_2nd`
     fi
-    if [ "$drop_1st" == '' ]
+    if [ -z "$drop_1st" ]
     then
         dropped=0
     else
@@ -264,7 +264,7 @@
     esac
 }
 
-if [ ${pid_check} == 1 ]
+if [ ${pid_check} = 1 ]
 then
     check_pid
     if [ "$retval" = 1 ]
