Index: support/mixer-qt4/SConscript
===================================================================
--- support/mixer-qt4/SConscript.orig	2015-04-12 13:18:15.000000000 +0200
+++ support/mixer-qt4/SConscript	2019-01-03 18:41:11.712952139 +0200
@@ -38,7 +38,7 @@ if env['BUILD_MIXER'] == 'true':
                 arg.append( os.path.join( dirname, name ) )
 
     pythonfiles = [ 'ffado/config.py' ]
-    os.path.walk( "ffado", findfiles, pythonfiles )
+    os.walk( "ffado", findfiles, pythonfiles )
 
     e.ScanReplace( "ffado/config.py.in" )
     e.Depends( "ffado/config.py", "#/SConstruct" )
