diff -Nur dde-api-5.3.2/adjust-grub-theme/main.go dde-api-5.3.2-new/adjust-grub-theme/main.go
--- dde-api-5.3.2/adjust-grub-theme/main.go	2020-12-11 13:06:50.000000000 +0800
+++ dde-api-5.3.2-new/adjust-grub-theme/main.go	2021-02-21 16:23:04.589407239 +0800
@@ -46,11 +46,11 @@
 )
 
 const (
-	defaultThemeOutputDir = "/boot/grub/themes/"
-	defaultThemeInputDir  = "/usr/share/dde-api/data/grub-themes/"
+	defaultThemeOutputDir = "/boot/grub2/themes/openSUSE"
+	defaultThemeInputDir  = "/usr/share/grub2/themes/openSUSE"
 
-	themeNameNormal   = "deepin"
-	themeNameFallback = "deepin-fallback"
+	themeNameNormal   = "openSUSE"
+	themeNameFallback = "openSUSE"
 )
 
 var optScreenHeight int
