From f2955af017f604003e3c8c3efe0fb0fb85584cea Mon Sep 17 00:00:00 2001
From: Pavel Shilovsky <pshilov@microsoft.com>
Date: Thu, 18 Apr 2019 12:32:02 -0700
Subject: [PATCH] getcifsacl: Fix usage message to include multiple files

Signed-off-by: Pavel Shilovsky <pshilov@microsoft.com>
---
 getcifsacl.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/getcifsacl.c b/getcifsacl.c
index 556178a..bea81ee 100644
--- a/getcifsacl.c
+++ b/getcifsacl.c
@@ -330,7 +330,7 @@ getcifsacl_usage(const char *prog)
 	fprintf(stderr,
 	"%s: Display CIFS/NTFS ACL in a security descriptor of a file object\n",
 		prog);
-	fprintf(stderr, "Usage: %s [option] <file_name>\n", prog);
+	fprintf(stderr, "Usage: %s [option] <file_name1> [<file_name2>,<file_name3>,...]\n", prog);
 	fprintf(stderr, "Valid options:\n");
 	fprintf(stderr, "\t-h	Display this help text\n");
 	fprintf(stderr, "\n");
@@ -410,7 +410,7 @@ main(const int argc, char *const argv[])
 
 	if (optind >= argc) {
 		printf("you must specify a filename after options.\n");
-		printf("Usage: getcifsacl [option] <file_name>\n");
+		printf("Usage: getcifsacl [option] <file_name1> [<file_name2>,<file_name3>,...]\n");
 		goto out;
 	}
 
-- 
2.16.4

