Index: fastparquet-0.3.3/fastparquet/test/test_api.py
===================================================================
--- fastparquet-0.3.3.orig/fastparquet/test/test_api.py
+++ fastparquet-0.3.3/fastparquet/test/test_api.py
@@ -4,6 +4,7 @@ from __future__ import unicode_literals
 import io
 import os
 import subprocess
+import sys
 from distutils.version import LooseVersion
 
 import numba
@@ -26,7 +27,7 @@ TEST_DATA = "test-data"
 @pytest.mark.skipif(numba.__version__ <= LooseVersion("0.39.0"), reason="Warning from numba.")
 def test_import_without_warning():
     # in a subprocess to avoid import chacing issues.
-    subprocess.check_call(["python", "-Werror", "-c", "import fastparquet"])
+    subprocess.check_call([sys.executable, "-Werror", "-c", "import fastparquet"])
 
 
 def test_statistics(tempdir):
