Index: internal.h
===================================================================
RCS file: /home/mjl/cvsroot/scamper/internal.h,v
retrieving revision 1.42
diff -u -p -r1.42 internal.h
--- internal.h	17 Sep 2016 01:38:13 -0000	1.42
+++ internal.h	19 Aug 2019 23:44:05 -0000
@@ -173,6 +173,7 @@ typedef unsigned short sa_family_t;
 #include <linux/types.h>
 #include <linux/filter.h>
 #include <linux/netfilter_ipv4/ip_tables.h>
+#include <linux/sockios.h>
 #include <limits.h>
 #ifndef SOL_PACKET
 #define SOL_PACKET 263
