From 95ca5c3a64e4e01070df143ca99246303fe09176 Mon Sep 17 00:00:00 2001
From: Luigi Baldoni <lgbaldoni@github.com>
Date: Fri, 4 Jun 2021 07:20:41 +0200
Subject: [PATCH] Fix build with gcc11

Index: hidviz-0.1.5/libhidx/libhidx/include/libhidx/Interface.hh
===================================================================
--- hidviz-0.1.5.orig/libhidx/libhidx/include/libhidx/Interface.hh
+++ hidviz-0.1.5/libhidx/libhidx/include/libhidx/Interface.hh
@@ -26,6 +26,7 @@
 #include <functional>
 #include <vector>
 #include <string>
+#include <memory>
 
 namespace libhidx {
 
