---
 pytest.ini                 |    2 ++
 tests/test_pluginsystem.py |    1 +
 2 files changed, 3 insertions(+)

--- a/pytest.ini
+++ b/pytest.ini
@@ -1,2 +1,4 @@
 [pytest]
 norecursedirs = venv
+markers =
+    network: tests requiring network connection
--- a/tests/test_pluginsystem.py
+++ b/tests/test_pluginsystem.py
@@ -155,6 +155,7 @@ class TestPlugin:
     def test_path(self, dummy_plugin):
         assert dummy_plugin.path == str(Path(__file__).parent)
 
+    @pytest.mark.network
     @pytest.mark.usefixtures("save_sys_path")
     def test_path_installed_plugin_is_none(self, scratch_project):
         # XXX: this test is slow and fragile. (It won't run
