--- a/test-suite/standalone/Makefile.am	2025-11-11 10:56:38.000000000 +0100
+++ b/test-suite/standalone/Makefile.am	2026-07-28 16:35:19.933192522 +0200
@@ -312,9 +312,6 @@
 check_SCRIPTS += test-stack-overflow
 TESTS += test-stack-overflow
 
-check_SCRIPTS += test-out-of-memory
-TESTS += test-out-of-memory
-
 check_SCRIPTS += test-close-on-exec
 TESTS += test-close-on-exec
 
--- a/test-suite/standalone/Makefile.in	2025-11-30 23:00:51.000000000 +0100
+++ b/test-suite/standalone/Makefile.in	2026-07-28 16:35:19.951789377 +0200
@@ -109,7 +109,7 @@
 	test-scm-c-bind-keyword-arguments$(EXEEXT) \
 	test-srfi-4$(EXEEXT) $(am__append_6) $(am__EXEEXT_1) \
 	test-smob-mark$(EXEEXT) test-smob-mark-race$(EXEEXT) \
-	test-stack-overflow test-out-of-memory test-close-on-exec
+	test-stack-overflow test-close-on-exec
 check_PROGRAMS = test-num2integral$(EXEEXT) test-round$(EXEEXT) \
 	test-foreign-object-c$(EXEEXT) test-list$(EXEEXT) \
 	test-unwind$(EXEEXT) test-conversion$(EXEEXT) \
@@ -2466,8 +2466,7 @@
 	test-language test-guild-compile test-sigaction-fork \
 	test-signal-fork $(am__append_3) test-foreign-object-scm \
 	test-fast-slot-ref test-mb-regexp test-use-srfi \
-	$(am__append_5) test-stack-overflow test-out-of-memory \
-	test-close-on-exec
+	$(am__append_5) test-stack-overflow test-close-on-exec
 BUILT_SOURCES = $(am__append_2)
 EXTRA_DIST = test-import-order-a.scm test-import-order-b.scm \
 	test-import-order-c.scm test-import-order-d.scm \
