Index: whois-5.5.16/mkpasswd.c
===================================================================
--- whois-5.5.16.orig/mkpasswd.c
+++ whois-5.5.16/mkpasswd.c
@@ -17,10 +17,7 @@
  */
 
 /* for crypt, snprintf and strcasecmp */
-#define _XOPEN_SOURCE 500
-#define _BSD_SOURCE 1
 #define _DEFAULT_SOURCE 1
-#define __EXTENSIONS__ 1
 
 /* System library */
 #include <stdio.h>
Index: whois-5.5.16/whois.c
===================================================================
--- whois-5.5.16.orig/whois.c
+++ whois-5.5.16/whois.c
@@ -7,9 +7,6 @@
  * (at your option) any later version.
  */
 
-/* for AI_IDN */
-#define _GNU_SOURCE
-
 /* System library */
 #include <stdio.h>
 #include <stdlib.h>
Index: whois-5.5.16/Makefile
===================================================================
--- whois-5.5.16.orig/Makefile
+++ whois-5.5.16/Makefile
@@ -69,7 +69,7 @@ else
 mkpasswd_LDADD += -lcrypt
 endif
 
-CPPFLAGS += $(DEFS) $(INCLUDES)
+CPPFLAGS += -D_GNU_SOURCE $(DEFS) $(INCLUDES)
 
 BASHCOMPDIR ?= $(shell $(PKG_CONFIG) --variable=completionsdir bash-completion 2>/dev/null || echo /etc/bash_completion.d)
 
