Index: whois-5.2.12/mkpasswd.c
===================================================================
--- whois-5.2.12.orig/mkpasswd.c
+++ whois-5.2.12/mkpasswd.c
@@ -17,10 +17,7 @@
  */
 
 /* for crypt, snprintf and strcasecmp */
-#define _XOPEN_SOURCE 500
-#define _BSD_SOURCE 1
 #define _DEFAULT_SOURCE 1
-#define __EXTENSIONS__ 1
 
 /* System library */
 #include <stdio.h>
Index: whois-5.2.12/whois.c
===================================================================
--- whois-5.2.12.orig/whois.c
+++ whois-5.2.12/whois.c
@@ -7,9 +7,6 @@
  * (at your option) any later version.
  */
 
-/* for AI_IDN */
-#define _GNU_SOURCE
-
 /* System library */
 #include <stdio.h>
 #include <stdlib.h>
Index: whois-5.2.12/Makefile
===================================================================
--- whois-5.2.12.orig/Makefile
+++ whois-5.2.12/Makefile
@@ -53,7 +53,7 @@ endif
 mkpasswd_LDADD += -lcrypt
 endif
 
-CPPFLAGS += $(DEFS) $(INCLUDES)
+CPPFLAGS += -D_GNU_SOURCE $(DEFS) $(INCLUDES)
 
 ##############################################################################
 all: Makefile.depend whois mkpasswd pos
