diff -Nru azure_cli_core-2.62.0.orig/setup.py azure_cli_core-2.62.0/setup.py
--- azure_cli_core-2.62.0.orig/setup.py	2024-07-04 04:23:33.000000000 +0200
+++ azure_cli_core-2.62.0/setup.py	2024-07-19 16:27:28.767852656 +0200
@@ -59,7 +59,7 @@
     'paramiko>=2.0.8,<4.0.0',
     'pkginfo>=1.5.0.1',
     # psutil can't install on cygwin: https://github.com/Azure/azure-cli/issues/9399
-    'psutil~=5.9; sys_platform != "cygwin"',
+    'psutil>=5.9; sys_platform != "cygwin"',
     'PyJWT>=2.1.0',
     'pyopenssl>=17.1.0',  # https://github.com/pyca/pyopenssl/pull/612
     'requests[socks]'
