Index: cryptography-43.0.0/pyproject.toml
===================================================================
--- cryptography-43.0.0.orig/pyproject.toml
+++ cryptography-43.0.0/pyproject.toml
@@ -84,7 +84,8 @@ python-source = "src"
 python-packages = ["cryptography"]
 manifest-path = "src/rust/Cargo.toml"
 module-name = "cryptography.hazmat.bindings._rust"
-locked = true
+locked = false
+offline = true
 sdist-generator = "git"
 features = ["pyo3/abi3-py37"]
 include = [
