From e854a4a68181114b0fc0be6d966d941e0b33db33 Mon Sep 17 00:00:00 2001
From: Nick ODell <nickodell@gmail.com>
Date: Wed, 5 Nov 2025 17:36:08 -0700
Subject: [PATCH] BUG: sparse.csgraph: remove rel import to fix Cython 3.2
 (#23940)

---
 scipy/sparse/csgraph/_reordering.pyx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scipy/sparse/csgraph/_reordering.pyx b/scipy/sparse/csgraph/_reordering.pyx
index e58c6a9d2d12..e951f410c246 100644
--- a/scipy/sparse/csgraph/_reordering.pyx
+++ b/scipy/sparse/csgraph/_reordering.pyx
@@ -7,7 +7,7 @@ cimport numpy as np
 from warnings import warn
 from scipy.sparse import csr_array, issparse, SparseEfficiencyWarning
 from scipy.sparse._sputils import convert_pydata_sparse_to_scipy
-from . import maximum_bipartite_matching
+from scipy.sparse.csgraph import maximum_bipartite_matching
 
 np.import_array()
 
