diff -urEbw minigalaxy-1.0.0/data/io.github.sharkwouter.Minigalaxy.desktop minigalaxy-1.0.0.new/data/io.github.sharkwouter.Minigalaxy.desktop
--- minigalaxy-1.0.0/data/io.github.sharkwouter.Minigalaxy.desktop	2020-11-29 13:25:24.000000000 +0100
+++ minigalaxy-1.0.0.new/data/io.github.sharkwouter.Minigalaxy.desktop	2020-11-30 09:42:35.690421038 +0100
@@ -6,5 +6,5 @@
 Icon=io.github.sharkwouter.Minigalaxy
 Type=Application
 Keywords=galaxy;gaming;games;game;old;good;gog;
-Categories=Game;
+Categories=Game;Amusement;
 
