diff -ruN qwt-6.1.3.orig/designer/designer.pro qwt-6.1.3/designer/designer.pro
--- qwt-6.1.3.orig/designer/designer.pro	2016-06-13 10:14:23.019600853 +0300
+++ qwt-6.1.3/designer/designer.pro	2017-04-21 02:01:26.358076198 +0300
@@ -83,7 +83,7 @@
         # compile the path for finding the Qwt library
         # into the plugin. Not supported on Windows !
 
-        QMAKE_RPATHDIR *= $${QWT_INSTALL_LIBS}
+#        QMAKE_RPATHDIR *= $${QWT_INSTALL_LIBS}
         qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt)
 
         contains(QWT_CONFIG, QwtDll) {
diff -ruN qwt-6.1.3.orig/examples/examples.pri qwt-6.1.3/examples/examples.pri
--- qwt-6.1.3.orig/examples/examples.pri	2016-06-13 10:14:23.023600934 +0300
+++ qwt-6.1.3/examples/examples.pri	2017-04-21 02:01:43.778128834 +0300
@@ -34,7 +34,7 @@
     }
 }
 
-QMAKE_RPATHDIR *= $${QWT_OUT_ROOT}/lib
+#QMAKE_RPATHDIR *= $${QWT_OUT_ROOT}/lib
 qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt)
 
 greaterThan(QT_MAJOR_VERSION, 4) {
