---
 virt-top/Makefile.in |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/virt-top/Makefile.in
+++ b/virt-top/Makefile.in
@@ -64,11 +64,11 @@ OBJS		+= virt_top_main.cmo
 
 XOBJS		:= $(OBJS:.cmo=.cmx)
 
-OCAMLCFLAGS	:= -g -warn-error A
+OCAMLCFLAGS	:= -g -warn-error A-3-52
 OCAMLCLIBS	:= -linkpkg
 
 OCAMLOPTPACKAGES := $(OCAMLCPACKAGES)
-OCAMLOPTFLAGS	:= -warn-error A
+OCAMLOPTFLAGS	:= -warn-error A-3-52
 OCAMLOPTLIBS	:= $(OCAMLCLIBS)
 
 BYTE_TARGETS	:= virt-top
