--- a/virt-top/Makefile.in
+++ b/virt-top/Makefile.in
@@ -102,9 +102,13 @@ virt-top.txt: virt-top.pod
 endif
 
 install:
+	set -x ; \
+	exe=virt-top ; \
 	if [ -x virt-top.opt ]; then \
-	  mkdir -p $(DESTDIR)$(bindir); \
-	  $(INSTALL) -m 0755 virt-top.opt $(DESTDIR)$(bindir)/virt-top; \
-	fi
+	exe=virt-top.opt ; \
+	fi ; \
+	mkdir -p $(DESTDIR)$(bindir); \
+	$(INSTALL) -m 0755 $${exe} $(DESTDIR)$(bindir)/virt-top; \
+	: done
 
 include ../Make.rules
