diff -Nru azure-mgmt-cdn-3.0.0.orig/setup.py azure-mgmt-cdn-3.0.0/setup.py
--- azure-mgmt-cdn-3.0.0.orig/setup.py	2018-05-25 20:50:54.000000000 +0200
+++ azure-mgmt-cdn-3.0.0/setup.py	2018-09-03 16:13:58.604696959 +0200
@@ -78,7 +78,7 @@
     packages=find_packages(exclude=["tests"]),
     install_requires=[
         'msrestazure>=0.4.27,<2.0.0',
-        'azure-common~=1.1',
+        'azure-common>=1.1',
     ],
     cmdclass=cmdclass
 )
