diff --git a/compiler/control/OMROptions.hpp b/compiler/control/OMROptions.hpp
index ea99d9405..e39d367e6 100644
--- a/omr/compiler/control/OMROptions.hpp
+++ b/omr/compiler/control/OMROptions.hpp
@@ -1235,19 +1235,18 @@ class TR_FilterBST {
 public:
     void *operator new(size_t size, TR::PersistentAllocator &allocator);
 
-    TR_FilterBST(int32_t type, int32_t optionSet)
+    TR_FilterBST(int32_t type, int32_t optionSet, int32_t lineNumber = 0)
+        : _name(NULL)
+        , _class(NULL)
+        , _signature(NULL)
+        , _regex(NULL)
+        , _optionSet(optionSet)
+        , _lineNumber(lineNumber)
+        , _nameLen(0)
+        , _filterType((char)type)
+        , subGroup(NULL)
     {
-        memset(this, 0, sizeof(TR_FilterBST));
-        _filterType = (char)type;
-        _optionSet = optionSet;
-    }
-
-    TR_FilterBST(int32_t type, int32_t optionSet, int32_t lineNumber)
-    {
-        memset(this, 0, sizeof(TR_FilterBST));
-        _filterType = (char)type;
-        _optionSet = optionSet;
-        _lineNumber = lineNumber;
+        memset(_child, 0, sizeof(_child)); ///< in C++11 notation: _child = {0};
     }
 
     bool isExclude() { return _filterType <= TR_FILTER_EXCLUDE_MAX; }
diff --git a/compiler/infra/Statistics.hpp b/compiler/infra/Statistics.hpp
index d880666d1..a0bdeb2a3 100644
--- a/omr/compiler/infra/Statistics.hpp
+++ b/omr/compiler/infra/Statistics.hpp
@@ -303,7 +303,15 @@ public:
         NAME_LEN = 31
     };
 
-    TR_StatsEvents() {}
+    TR_StatsEvents()
+        : _binNames(NULL)
+        , _minEventId(0)
+        , _numSamples(0)
+        , _numInvalidSamples(0)
+    {
+        memset(_name, 0, sizeof(_name)); ///< in C++11 notation: _name = {0};
+        memset(_bins, 0, sizeof(_bins)); ///< in C++11 notation: _bins = {0};
+    }
 
     TR_StatsEvents(const char *name, const char **binNames, int minEventId) { init(name, binNames, minEventId); }
 
diff --git a/gc/base/MemorySubSpaceGenerational.hpp b/gc/base/MemorySubSpaceGenerational.hpp
index 53fe31f9f..3691f7bbe 100644
--- a/omr/gc/base/MemorySubSpaceGenerational.hpp
+++ b/omr/gc/base/MemorySubSpaceGenerational.hpp
@@ -107,7 +107,7 @@ public:
 
 	virtual uintptr_t releaseFreeMemoryPages(MM_EnvironmentBase* env, uintptr_t memoryType);
 
-	virtual MMINLINE uintptr_t getContractionSize() const { return _memorySubSpaceOld->getContractionSize(); }
+	virtual uintptr_t getContractionSize() const { return _memorySubSpaceOld->getContractionSize(); }
 
 	/**
 	 * Create a MemorySubSpaceGenerational object.
