Index: wxmp3gain-4.0/CMakeLists.txt
===================================================================
--- wxmp3gain-4.0.orig/CMakeLists.txt
+++ wxmp3gain-4.0/CMakeLists.txt
@@ -35,7 +35,6 @@ target_link_libraries(wxmp3gain ${wxWidg
 if(UNIX)
     install(TARGETS "wxmp3gain" DESTINATION "bin")
     install(DIRECTORY "resource/" DESTINATION "share/${PROJECT_NAME}/resource/")
-    install(DIRECTORY "docs/" DESTINATION "share/doc/${PROJECT_NAME}/")
     install(FILES "debian/menu/icon32x32.png" DESTINATION "share/icons/hicolor/32x32/apps/" RENAME "${PROJECT_NAME}.png")
     install(FILES "debian/menu/icon48x48.png" DESTINATION "share/icons/hicolor/48x48/apps/" RENAME "${PROJECT_NAME}.png")
     install(FILES "debian/menu/icon64x64.png" DESTINATION "share/icons/hicolor/64x64/apps/" RENAME "${PROJECT_NAME}.png")
