--- a/src/main/java/com/github/luben/zstd/util/Native.java
+++ b/src/main/java/com/github/luben/zstd/util/Native.java
@@ -103,79 +103,11 @@ public enum Native {
         String resourceName = resourceName();
 
         String overridePath = System.getProperty(nativePathOverride);
-        if (overridePath != null) {
+        if (overridePath != null)
             // Do not fall back to auto-discovery - consumers know better
             loadLibraryFile(overridePath);
-            loaded.set(true);
-            return;
-        }
-
-        // try to load the shared library directly from the JAR or system path
-        try {
-            loadLibrary(libnameShort);
-            loaded.set(true);
-            return;
-        } catch (Throwable e) {
-            // ignore UnsatisfiedLinkError, and try other methods
-        }
-
-        InputStream is = Native.class.getResourceAsStream(resourceName);
-        if (is == null) {
-            UnsatisfiedLinkError err = new UnsatisfiedLinkError("Failed to open resource " + resourceName + " as stream:\n" + errorMsg);
-            throw err;
-        }
-        File tempLib = null;
-        FileOutputStream out = null;
-        try {
-            tempLib = File.createTempFile(libname, "." + libExtension(), tempFolder);
-            // try to delete on exit, does not work on Windows
-            tempLib.deleteOnExit();
-            // copy to tempLib
-            out = new FileOutputStream(tempLib);
-            byte[] buf = new byte[4096];
-            while (true) {
-                int read = is.read(buf);
-                if (read == -1) {
-                    break;
-                }
-                out.write(buf, 0, read);
-            }
-            try {
-                out.flush();
-                out.close();
-                out = null;
-            } catch (IOException e) {
-                // ignore
-            }
-            try {
-                loadLibraryFile(tempLib.getAbsolutePath());
-            } catch (UnsatisfiedLinkError e) {
-                // display error in case problem with loading from temp folder
-                UnsatisfiedLinkError err = new UnsatisfiedLinkError(
-                        e.getMessage() + "\n" +
-                        errorMsg);
-                err.setStackTrace(e.getStackTrace());
-                throw err;
-            }
-            loaded.set(true);
-        } catch (IOException e) {
-            // IO errors in extracting and writing the shared object in the temp dir
-            ExceptionInInitializerError err = new ExceptionInInitializerError(
-                    "Cannot unpack " + libname + ": " + e.getMessage());
-            err.setStackTrace(e.getStackTrace());
-            throw err;
-        } finally {
-            try {
-                is.close();
-                if (out != null) {
-                    out.close();
-                }
-                if (tempLib != null && tempLib.exists()) {
-                    tempLib.delete();
-                }
-            } catch (IOException e) {
-                // ignore
-            }
-        }
+		else
+			loadLibraryFile("@SYS_LIBRARY_PREFIX@/"+libname+".so");
+		loaded.set(true);
     }
 }
