https://github.com/greenbone/openvas-smb/pull/40
commit 195440b141e198740d06d1391a9d4de44521855a
Author: Bernhard M. Wiedemann <bwiedemann@suse.de>
Date:   Sat Jun 12 05:42:28 2021 +0200

    Omit timestamp from PE executable
    
    to make binaries reproducible.
    See https://reproducible-builds.org/ for why this is good.
    
    This PR was done while working on reproducible builds for openSUSE.
    
    Co-authored-by: Jaspar L. <jaspar.loechte@greenbone.net>

Index: openvas-smb-1.0.5/winexe/CMakeLists.txt
===================================================================
--- openvas-smb-1.0.5.orig/winexe/CMakeLists.txt
+++ openvas-smb-1.0.5/winexe/CMakeLists.txt
@@ -71,7 +71,7 @@ set (BINTOC_SOURCES
 
 add_custom_command(
    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/winexesvc.exe
-   COMMAND ${MINGW_GCC} ${WINEXESVC_SOURCES} -o winexesvc.exe
+   COMMAND ${MINGW_GCC} -Xlinker --no-insert-timestamp ${WINEXESVC_SOURCES} -o winexesvc.exe
    DEPENDS ${WINEXESVC_SOURCES})
 
 add_custom_command(
