--- eclipse-platform-sources-I20180906-0745/rt.equinox.bundles/bundles/org.eclipse.equinox.security.linux.x86_64/keystorelinuxnative/keystoreLinuxNative.c	2020-06-04 09:40:51.525778305 +0200
+++ eclipse-platform-sources-I20180906-0745/rt.equinox.bundles/bundles/org.eclipse.equinox.security.linux.x86_64/keystorelinuxnative/keystoreLinuxNative.c	2020-07-17 00:04:15.481020726 +0200
@@ -17,6 +17,7 @@
 #include <sys/types.h>
 #include <stdlib.h>
 #include <string.h>
+#include <gio/gio.h>
 
 const SecretSchema *
 equinox_get_schema (void)
--- eclipse-platform-sources-I20180906-0745/rt.equinox.bundles/bundles/org.eclipse.equinox.security.linux.x86_64/keystorelinuxnative/makefile	2020-07-17 00:17:39.178284379 +0200
+++ eclipse-platform-sources-I20180906-0745/rt.equinox.bundles/bundles/org.eclipse.equinox.security.linux.x86_64/keystorelinuxnative/makefile	2020-07-17 00:47:40.882104621 +0200
@@ -2,7 +2,7 @@
 LIBSECRET_LIBS = `pkg-config --libs libsecret-1`
 
 CFLAGS := $(CFLAGS) -fPIC -I"${JAVA_HOME}/include" -I"${JAVA_HOME}/include/linux" $(LIBSECRET_CFLAGS)
-LDFLAGS := $(LDFLAGS) -shared $(LIBSECRET_LIBS)
+LDFLAGS := $(LDFLAGS) $(LIBSECRET_LIBS)
 
 all: ../libkeystorelinuxnative.so
 
@@ -10,7 +10,7 @@
 	rm ../libkeystorelinuxnative.so ../target/keystoreLinuxNative.o
 
 ../libkeystorelinuxnative.so: ../target/keystoreLinuxNative.o
-	$(CC) $(LDFLAGS) -o $@ $<
+	$(CC) -shared -o $@ $< $(LDFLAGS) -Wl,--no-undefined -Wl,--as-needed
 
 ../target/keystoreLinuxNative.o: ../target keystoreLinuxNative.c
 	$(CC) -c $(CFLAGS) -o $@ $(TESTFLAGS) keystoreLinuxNative.c
