--- eclipse-platform-sources-I20180906-0745/dependencies/replace_platform_plugins_with_symlinks.sh	2019-01-21 00:06:48.212694105 +0100
+++ eclipse-platform-sources-I20180906-0745/dependencies/replace_platform_plugins_with_symlinks.sh	2019-01-21 00:09:22.629481027 +0100
@@ -35,9 +35,12 @@
 		_symlink javax.servlet.jsp_ glassfish-jsp-api/javax.servlet.jsp-api.jar
 		_symlink javax.servlet-api_ glassfish-servlet-api.jar
 		_symlink javax.xml_ xml-commons-apis.jar
-		_symlink org.apache.batik.css_ batik/batik-css.jar
-		_symlink org.apache.batik.util_ batik/batik-util.jar
-		_symlink org.apache.batik.util.gui_ batik/batik-gui-util.jar
+		_symlink org.apache.batik.css_ xmlgraphics-batik/css.jar
+		_symlink org.apache.batik.util_ xmlgraphics-batik/util.jar
+		_symlink org.apache.batik.i18n_ xmlgraphics-batik/i18n.jar
+		_symlink org.apache.batik.constants_ xmlgraphics-batik/constants.jar
+		_symlink org.apache.xmlgraphics_ xmlgraphics-commons.jar
+		_symlink org.apache.commons.io_ commons-io.jar
 		_symlink org.apache.commons.codec_ commons-codec.jar
 		_symlink org.apache.commons.jxpath_ commons-jxpath.jar
 		_symlink org.apache.commons.logging_ commons-logging.jar
