--- eclipse-platform-sources-I20180906-0745/eclipse.platform.releng.tychoeclipsebuilder/platform/platform.product	2018-09-06 13:46:56.000000000 +0200
+++ eclipse-platform-sources-I20180906-0745/eclipse.platform.releng.tychoeclipsebuilder/platform/platform.product	2020-03-26 18:08:15.516714532 +0100
@@ -7,7 +7,7 @@
    </configIni>
 
    <launcherArgs>
-      <programArgs>--launcher.defaultAction openFile --launcher.appendVmargs
+      <programArgs>--launcher.defaultAction openFile --launcher.GTK_version 2 --launcher.appendVmargs
       </programArgs>
       <vmArgs>-Dosgi.requiredJavaVersion=1.8 -Dosgi.dataAreaRequiresExplicitInit=true -Xms40m -Xmx512m --add-modules=ALL-SYSTEM
       </vmArgs>
