--- asedriveiiie-usb-3.7.orig/50-pcscd-asedriveiiie.rules	2010-09-04 17:53:53.000000000 +0300
+++ asedriveiiie-usb-3.7/50-pcscd-asedriveiiie.rules	2016-06-12 12:17:27.649100240 +0300
@@ -6,8 +6,8 @@
 SUBSYSTEM!="usb", GOTO="asedriveiiie_rules_end"
 ENV{DEVTYPE}!="usb_device", GOTO="asedriveiiie_rules_end"
 
-ATTRS{idVendor}=="0dc3", ATTRS{idProduct}=="0802", GROUP="pcscd"
-ATTRS{idVendor}=="0dc3", ATTRS{idProduct}=="1104", GROUP="pcscd"
+ATTRS{idVendor}=="0dc3", ATTRS{idProduct}=="0802"
+ATTRS{idVendor}=="0dc3", ATTRS{idProduct}=="1104"
 
 # All done
 LABEL="asedriveiiie_rules_end"
