Index: libgda-5.2.4/m4/java.m4
===================================================================
--- libgda-5.2.4.orig/m4/java.m4
+++ libgda-5.2.4/m4/java.m4
@@ -185,6 +185,11 @@ m4_define([_JAVA_CHECK_INTERNAL],
 		    JTYPE="Sun JRE 9"
 		    JFLAGS="-Xlint:unchecked -Xlint:deprecation"
 		    ;;
+                JRE1[[0-9]]*)
+		    try_java=true
+		    JTYPE="Sun JRE 10+"
+		    JFLAGS="-Xlint:unchecked -Xlint:deprecation"
+		    ;;
                 JREgcj-4*)
 		    try_java=true
 		    JTYPE="GCJ"
