Index: libgda-5.0.3/getsp.java
===================================================================
--- libgda-5.0.3.orig/getsp.java
+++ libgda-5.0.3/getsp.java
@@ -25,7 +25,7 @@ public class getsp {
 				while (i<=j) {
 					if (i==j || lp.charAt(i)==ps) {
 						String lib=lp.substring(k,i);
-						String suffix="/lib/amd64/server";
+						String suffix="/lib/--ARCH--/server";
 						k=i+1;
 						if (lib.compareTo(".")!=0)
 							r=(r==null)?(prefix+lib+suffix):(r+" "+prefix+lib+suffix);
@@ -50,7 +50,7 @@ public class getsp {
 
 				if (r!=null) System.out.println(r);
 			} else if (args[0].compareTo("-ldpath")==0) {
-				String lp1=System.getProperty("java.home")+"/lib/amd64/server";
+				String lp1=System.getProperty("java.home")+"/lib/--ARCH--/server";
 				String lp2=System.getProperty("java.library.path");
 				System.out.println(lp1+":"+lp2);
 			}
