diff -Naur xfce4-power-manager-4.16.0/src/xfpm-power.c xfce4-power-manager-4.16.0.new/src/xfpm-power.c
--- xfce4-power-manager-4.16.0/src/xfpm-power.c	2020-11-06 23:13:15.000000000 +0100
+++ xfce4-power-manager-4.16.0.new/src/xfpm-power.c	2022-01-03 22:05:53.664020771 +0100
@@ -952,6 +952,8 @@
                 power->priv->screensaver_inhibited ? "TRUE" : "FALSE",
                 power->priv->presentation_mode ? "TRUE" : "FALSE");
 
+    xfpm_dpms_inhibit (power->priv->dpms, is_inhibit);
+
     /* If we are inhibited make sure we inhibit the screensaver too */
     if (is_inhibit)
     {
@@ -976,6 +978,8 @@
   power->priv->inhibited ? "TRUE" : "FALSE",
   power->priv->screensaver_inhibited ? "TRUE" : "FALSE",
   power->priv->presentation_mode ? "TRUE" : "FALSE");
+
+  xfpm_update_blank_time (power);
 }
 
 static void
