diff -Nru azure-storage-blob-1.5.0.orig/setup.py azure-storage-blob-1.5.0/setup.py
--- azure-storage-blob-1.5.0.orig/setup.py	2019-02-16 00:11:37.000000000 +0100
+++ azure-storage-blob-1.5.0/setup.py	2019-04-02 12:16:35.116776872 +0200
@@ -72,7 +72,7 @@
     ]),
     install_requires=[
         'azure-common>=1.1.5',
-        'azure-storage-common~=1.4'
+        'azure-storage-common>=1.4'
     ],
     extras_require={
         ":python_version<'3.0'": ['futures'],
