diff -Nur xf86-video-vmware/vmwgfx/vmwgfx_tex_video.c patched-xf86-video-vmware/vmwgfx/vmwgfx_tex_video.c
--- xf86-video-vmware/vmwgfx/vmwgfx_tex_video.c	2022-02-23 10:26:07.678383367 +0100
+++ patched-xf86-video-vmware/vmwgfx/vmwgfx_tex_video.c	2022-02-23 11:03:54.104420733 +0100
@@ -237,7 +237,7 @@
 
        for (i=0; i<3; ++i) {
 	   for (j=0; j<2; ++j) {
-	       if (priv->yuv[i]) {
+	       if (priv->yuv[j][i]) {
 		   xa_surface_destroy(priv->yuv[j][i]);
 		   priv->yuv[j][i] = NULL;
 	       }
