Index: scikit-hep-testdata-0.4.26/src/skhep_testdata/local_files.py
===================================================================
--- scikit-hep-testdata-0.4.26.orig/src/skhep_testdata/local_files.py
+++ scikit-hep-testdata-0.4.26/src/skhep_testdata/local_files.py
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+import os
 import sys
 import tempfile
 import zipfile
@@ -17,7 +18,7 @@ else:
 __all__ = ("data_path", "download_all", "known_files")
 
 
-DIR = Path(__file__).parent.resolve()
+DIR = Path(os.environ.get("SKHEP_DATA_DIR", "/usr/share/scikit-hep-testdata/"))
 
 baseurl = "https://raw.githubusercontent.com/scikit-hep/scikit-hep-testdata/main/src/skhep_testdata/data/"
 zipurl = "https://github.com/scikit-hep/scikit-hep-testdata/zipball/main"
Index: scikit-hep-testdata-0.4.26/tests/test_local_files.py
===================================================================
--- scikit-hep-testdata-0.4.26.orig/tests/test_local_files.py
+++ scikit-hep-testdata-0.4.26/tests/test_local_files.py
@@ -4,7 +4,7 @@ import pytest
 
 import skhep_testdata as skhtd
 
-data_dir = os.path.dirname(skhtd.__file__)
+data_dir = str(skhtd.local_files.DIR)
 data_dir = os.path.join(data_dir, "data")
 
 
