Index: VirtualBox-6.0.2/configure
===================================================================
--- VirtualBox-6.0.2.orig/configure
+++ VirtualBox-6.0.2/configure
@@ -1538,14 +1538,14 @@ EOF
         INCQT5="$q/include $q/include/QtCore"
         FLGQT5="-DQT_SHARED"
         I_INCQT5=`prefix_I "$INCQT5"`
-        LIBQT5="-L$q/lib -lQt5CoreVBox"
+        LIBQT5="-L$q/lib -L$q/usr/lib64 -lQt5CoreVBox"
         TOOLQT5="$q"
         if test_compile "$LIBQT5 $LIBPTHREAD $I_INCQT5 $FLGQT5" qt5 qt5 nofatal &&
             test_execute_path "`strip_L "$LIBQT5"`" nofatal; then
           foundqt5=2 # internal
           break;
         fi
-        LIBQT5="-L$q/lib -lQt5Core"
+        LIBQT5="-L$q/lib -L$q/usr/lib64 -lQt5Core"
         if test_compile "$LIBQT5 $LIBPTHREAD $I_INCQT5 $FLGQT5" qt5 qt5 nofatal &&
             test_execute_path "`strip_L "$LIBQT5"`" nofatal; then
           foundqt5=1 # no pkg-config, Qt directory
Index: VirtualBox-6.0.2/src/VBox/Additions/common/crOpenGL/Makefile.kmk
===================================================================
--- VirtualBox-6.0.2.orig/src/VBox/Additions/common/crOpenGL/Makefile.kmk
+++ VirtualBox-6.0.2/src/VBox/Additions/common/crOpenGL/Makefile.kmk
@@ -217,15 +217,7 @@ VBoxOGL_LIBS.win += \
 	$(PATH_STAGE_LIB)/additions/VBoxCrHgsmi$(VBOX_SUFF_LIB)
 
 if1of ($(KBUILD_TARGET), linux solaris freebsd)
- ifdef VBOX_USE_SYSTEM_GL_HEADERS
   VBoxOGL_LIBS += Xcomposite Xdamage Xfixes Xext
- else
-  VBoxOGL_LIBS += \
- 	$(PATH_STAGE_LIB)/libXcomposite.so \
- 	$(PATH_STAGE_LIB)/libXdamage.so \
- 	$(PATH_STAGE_LIB)/libXfixes.so \
- 	$(PATH_STAGE_LIB)/libXext.so
- endif
  ifdef VBoxOGL_FAKEDRI
   ifeq ($(KBUILD_TARGET), freebsd)
     VBoxOGL_LIBS += \
