diff --git a/go.mod b/go.mod
index 26a19495..7614c8d2 100644
--- a/go.mod
+++ b/go.mod
@@ -114,7 +114,6 @@ require (
 	github.com/ti-mo/conntrack v0.5.1
 	github.com/ti-mo/netfilter v0.5.2
 	github.com/vmihailenco/msgpack/v5 v5.4.1
-	github.com/wailsapp/wails/v3 v3.0.0-alpha2.117
 	github.com/yusufpapurcu/wmi v1.2.4
 	github.com/zcalusic/sysinfo v1.1.3
 	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.67.0
@@ -339,3 +338,12 @@ replace github.com/dexidp/dex => github.com/netbirdio/dex v0.244.1-0.20260716205
 replace github.com/dexidp/dex/api/v2 => github.com/netbirdio/dex/api/v2 v2.0.0-20260512110716-8d70ad8647c1
 
 replace github.com/mailru/easyjson => github.com/netbirdio/easyjson v0.9.0
+
+ignore (
+	github.com/netbirdio/netbird/client/ui
+	github.com/wailsapp/wails/v3
+	github.com/wailsapp/wails/v3/pkg/application
+	github.com/wailsapp/wails/v3/pkg/events
+	github.com/wailsapp/wails/v3/pkg/services/notifications
+	github.com/wailsapp/wails/v3/pkg/w32
+)
