Index: addons/allegrogl/CMakeLists.txt
===================================================================
--- addons/allegrogl/CMakeLists.txt.orig
+++ addons/allegrogl/CMakeLists.txt
@@ -19,11 +19,11 @@ if(ALLEGRO_XWINDOWS_WITH_XF86VIDMODE)
     set(ALLEGROGL_HAVE_XF86VIDMODE 1)
 endif()
 
-find_library(DLOPEN_LIB dl)
-mark_as_advanced(DLOPEN_LIB)
-if(DLOPEN_LIB)
-    set(ALLEGROGL_HAVE_DYNAMIC_LINK 1)
-endif()
+#find_library(DLOPEN_LIB dl)
+#mark_as_advanced(DLOPEN_LIB)
+#if(DLOPEN_LIB)
+#    set(ALLEGROGL_HAVE_DYNAMIC_LINK 1)
+#endif()
 
 if(ALLEGRO_WITH_XWINDOWS)
     set(CMAKE_REQUIRED_LIBRARIES ${OPENGL_gl_LIBRARY})
