Index: sitar-1.0.6/Makefile
===================================================================
--- sitar-1.0.6.orig/Makefile
+++ sitar-1.0.6/Makefile
@@ -26,7 +26,7 @@ PERL		= /usr/bin/perl
 RELEASE		= 1.0.6
 
 SVNVERSION     := $(shell svnversion -n . | sed "s/:/./" )
-RELDATE        := $(shell date +"%Y%m%d")
+RELDATE        := $(shell date --date=@$$SOURCE_DATE_EPOCH +"%Y%m%d")
 CFG_PFRAGS	= pfrags/cfg2scm.header.pl				\
 		  pfrags/cfg2scm.manpage.pl 				\
 		  pfrags/cfg2scm.globals.pl 				\
