Index: pure-ftpd-1.0.36/src/caps_p.h
===================================================================
--- pure-ftpd-1.0.36.orig/src/caps_p.h
+++ pure-ftpd-1.0.36/src/caps_p.h
@@ -7,7 +7,8 @@
 # endif
 
 cap_value_t cap_keep_startup[] = {
-# if defined(USE_PAM) && defined(CAP_AUDIT_WRITE)
+# if defined(USE_PAM) && defined(CAP_AUDIT_CONTROL) && defined(CAP_AUDIT_WRITE)
+    CAP_AUDIT_CONTROL,
     CAP_AUDIT_WRITE,
 # endif
     CAP_SETGID,
