From 71a726ca84535a6bfb2ad4795c6d499085df9594 Mon Sep 17 00:00:00 2001
From: Jim Fehlig <jfehlig@suse.com>
Date: Tue, 5 Jul 2022 11:24:54 -0600
Subject: Canonicalize hostarch name ppc64le to ppc64

See bnc#894956

Signed-off-by: Jim Fehlig <jfehlig@suse.com>
---
 src/util/virarch.c | 2 ++
 1 file changed, 2 insertions(+)

Index: libvirt-8.10.0/src/util/virarch.c
===================================================================
--- libvirt-8.10.0.orig/src/util/virarch.c
+++ libvirt-8.10.0/src/util/virarch.c
@@ -222,6 +222,8 @@ virArch virArchFromHost(void)
         arch = VIR_ARCH_X86_64;
     } else if (STREQ(ut.machine, "arm64")) {
         arch = VIR_ARCH_AARCH64;
+    } else if (STREQ(ut.machine, "ppc64le")) {
+        arch = VIR_ARCH_PPC64;
     } else {
         /* Otherwise assume the canonical name */
         if ((arch = virArchFromString(ut.machine)) == VIR_ARCH_NONE) {
