Adjust libvirtd sysconfig file to conform to SUSE standards

Index: libvirt-6.7.0/src/remote/libvirtd.sysconf
===================================================================
--- libvirt-6.7.0.orig/src/remote/libvirtd.sysconf
+++ libvirt-6.7.0/src/remote/libvirtd.sysconf
@@ -1,5 +1,9 @@
+## Path: System/Virtualization/libvirt
 # Customizations for the libvirtd.service systemd unit
 
+## Type: string
+## Default: "--timeout 120"
+# Arguments to pass to libvirtd. Not required if using systemd socket activation.
 # Default behaviour is for libvirtd.service to start on boot
 # so that VM autostart can be performed. We then want it to
 # shutdown again if nothing was started and rely on systemd
@@ -11,6 +15,8 @@ LIBVIRTD_ARGS="--timeout 120"
 # can be used to listen on TCP/TLS sockets
 #LIBVIRTD_ARGS="--listen"
 
+## Type: string
+## Default: none
 # Override the QEMU/SDL default audio driver probing when
 # starting virtual machines using SDL graphics
 #
