From d46371e4dbef932332b2b2809c1fcd61a0617c03 Mon Sep 17 00:00:00 2001
From: Alex Kapranoff <alex@kapranoff.ru>
Date: Sun, 23 Jan 2022 13:44:44 -0800
Subject: [PATCH] Support GNOME Shell 42 in Shell extension

Clutter.get_default_frame_rate() was removed in
https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2002/diffs?commit_id=33cdb45c8fdd9aa6f6da334a139a4a0fd1afe361

It effectively was always equal to 60, so let's hardcode the value.
---
 plugins/gnome/extension/dialogs.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/gnome/extension/dialogs.js b/plugins/gnome/extension/dialogs.js
index be4f15a..0f06448 100644
--- a/plugins/gnome/extension/dialogs.js
+++ b/plugins/gnome/extension/dialogs.js
@@ -46,7 +46,7 @@ const ngettext = Gettext.ngettext;
  */
 const IDLE_TIME_TO_PUSH_MODAL = 600;
 const PUSH_MODAL_TIME_LIMIT = 1000;
-const PUSH_MODAL_RATE = Clutter.get_default_frame_rate();
+const PUSH_MODAL_RATE = 60;
 const MOTION_DISTANCE_TO_CLOSE = 20;
 
 const IDLE_TIME_TO_OPEN = 60000;
