commit fc0e3304732a9aaff1487833342d5fc8ea26ce04
Author: Ludwig Nussel <ludwig.nussel@suse.de>
Date:   Fri Aug 6 15:11:23 2021 +0200

    selinux: allow login to read motd file

Index: cockpit/selinux/cockpit.te
===================================================================
diff --git a/selinux/cockpit.te b/selinux/cockpit.te
index 931cbd775..4a439c8d5 100644
--- a/selinux/cockpit.te
+++ b/selinux/cockpit.te
@@ -232,6 +232,14 @@ optional_policy(`
     allow cockpit_session_t pidfs_t:filesystem getattr;
 ')
 
+# login may read motd file through pam
+optional_policy(`
+    gen_require(`
+        type local_login_t;
+    ')
+    cockpit_read_pid_files(local_login_t)
+')
+
 #########################################################
 #
 #  Misc
