diff -Ndur audit-visualize/mkbar audit-visualize-fix-bashisms/mkbar
--- audit-visualize/mkbar	2005-11-20 22:23:04.000000000 +0200
+++ audit-visualize-fix-bashisms/mkbar	2014-11-16 03:21:16.593781158 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # Copyright 2005 Red Hat Inc., Durham, North Carolina.
 # All Rights Reserved.
 # This software may be freely redistributed and/or modified under the
diff -Ndur audit-visualize/mkgraph audit-visualize-fix-bashisms/mkgraph
--- audit-visualize/mkgraph	2005-11-18 00:17:59.000000000 +0200
+++ audit-visualize-fix-bashisms/mkgraph	2014-11-16 03:21:21.775780807 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 # Copyright 2005 Red Hat Inc., Durham, North Carolina.
 # All Rights Reserved.
