Index: xmorph-20060817/gtkmorph/callbacks_subimg.c
===================================================================
--- xmorph-20060817.orig/gtkmorph/callbacks_subimg.c
+++ xmorph-20060817/gtkmorph/callbacks_subimg.c
@@ -197,7 +197,6 @@ void on_hscale_mesh_changed    (GtkRange
   if(spin) {
     gtk_spin_button_set_value (spin, range->adjustment->value);
   } else g_critical("no adjustment");
-  return FALSE;
 }
 
 /**************/
@@ -220,7 +219,6 @@ void on_hscale_image_changed    (GtkRang
   if(spin) {  
     gtk_spin_button_set_value (spin, range->adjustment->value);
   } else g_critical("no adjustment");
-  return FALSE;
 }
 
 
