diff --git a/build-data/platforms/vpp.mk b/build-data/platforms/vpp.mk
index 1018496..4f775c7 100644
--- a/build-data/platforms/vpp.mk
+++ b/build-data/platforms/vpp.mk
@@ -43,3 +43,8 @@ vpp_gcov_TAG_LDFLAGS = -g -O0 -DCLIB_DEBUG -fPIC -Werror -coverage
 vpp_coverity_TAG_CFLAGS = -g -O2 -fPIC -Werror -D__COVERITY__
 vpp_coverity_TAG_LDFLAGS = -g -O2 -fPIC -Werror -D__COVERITY__
 
+vpp_uses_external_dpdk = yes
+vpp_dpdk_inc_dir = /usr/include/dpdk
+vpp_dpdk_lib_dir = /usr/lib
+vpp_dpdk_shared_lib = yes
+
