---
 ulogd.conf.in | 21 +++++++++++----------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/ulogd.conf.in b/ulogd.conf.in
index 6e0b7c7..1239745 100644
--- a/ulogd.conf.in
+++ b/ulogd.conf.in
@@ -6,9 +6,10 @@
 # GLOBAL OPTIONS
 ######################################################################
 
+user="ulogd"
 
 # logfile for status messages
-logfile="/var/log/ulogd.log"
+logfile="/var/log/ulogd/ulogd.log"
 
 # loglevel: debug(1), info(3), notice(5), error(7) or fatal(8) (default 5)
 # loglevel=1
@@ -175,25 +176,25 @@ numeric_label=1 # you can label the log info based on the packet verdict
 socket_path="/tmp/nuauth_ulogd2.sock"
 
 [emu1]
-file="/var/log/ulogd_syslogemu.log"
+file="/var/log/ulogd/syslogemu.log"
 sync=1
 
 [op1]
-file="/var/log/ulogd_oprint.log"
+file="/var/log/ulogd/oprint.log"
 sync=1
 
 [gp1]
-file="/var/log/ulogd_gprint.log"
+file="/var/log/ulogd/gprint.log"
 sync=1
 timestamp=1
 
 [xml1]
-directory="/var/log/"
+directory="/var/log/ulogd/"
 sync=1
 
 [json1]
 sync=1
-#file="/var/log/ulogd.json"
+#file="/var/log/ulogd/ulogd.json"
 #timestamp=0
 # device name to be used in JSON message
 #device="My awesome Netfilter firewall"
@@ -218,7 +219,7 @@ sync=1
 
 [pcap1]
 #default file is /var/log/ulogd.pcap
-#file="/var/log/ulogd.pcap"
+#file="/var/log/ulogd/ulogd.pcap"
 sync=1
 
 [mysql1]
@@ -303,18 +304,18 @@ procedure="INSERT_PACKET_FULL"
 
 [sqlite3_ct]
 table="ulog_ct"
-db="/var/log/ulogd.sqlite3db"
+db="/var/log/ulogd/ulogd.sqlite3db"
 
 [sqlite3_pkt]
 table="ulog_pkt"
-db="/var/log/ulogd.sqlite3db"
+db="/var/log/ulogd/ulogd.sqlite3db"
 
 [sys2]
 facility=LOG_LOCAL2
 
 [nacct1]
 sync = 1
-#file = /var/log/ulogd_nacct.log
+#file = /var/log/ulogd/nacct.log
 
 [mark1]
 mark = 1
-- 
2.50.1

