Index: mpv/player/main.c
===================================================================
--- mpv.orig/player/main.c
+++ mpv/player/main.c
@@ -384,7 +384,7 @@ int mp_initialize(struct MPContext *mpct
     if (handle_help_options(mpctx))
         return 1; // help
 
-    if (!print_libav_versions(mp_null_log, 0)) {
+    if (0 && !print_libav_versions(mp_null_log, 0)) {
         print_libav_versions(mpctx->log, MSGL_FATAL);
         MP_FATAL(mpctx, "\nmpv was compiled against an incompatible version of "
                  "FFmpeg/Libav than the shared\nlibrary it is linked against. "
