Index: mpv/common/av_log.c
===================================================================
--- mpv.orig/common/av_log.c
+++ mpv/common/av_log.c
@@ -200,7 +200,7 @@ void check_library_versions(struct mp_lo
         {
             fprintf(stderr, "%s: %d.%d.%d -> %d.%d.%d\n",
                     l->name, V(l->buildv), V(l->runv));
-            abort();
+            //abort();
         }
     }
 
