Index: mpv/player/main.c
===================================================================
--- mpv.orig/player/main.c
+++ mpv/player/main.c
@@ -382,7 +382,7 @@ int mp_initialize(struct MPContext *mpct
     if (handle_help_options(mpctx))
         return 1; // help
 
-    if (!print_libav_versions(mp_null_log, 0)) {
+    if (0 && !print_libav_versions(mp_null_log, 0)) {
         // Using mismatched libraries can be legitimate, but even then it's
         // a bad idea. We don't acknowledge its usefulness and stability.
         // Distro maintainers who patch this out should be aware that mpv
