diff --git a/logback-core/src/main/java/module-info.java b/logback-core/src/main/java/module-info.java
index 1cac38433..dcc91438a 100644
--- a/logback-core/src/main/java/module-info.java
+++ b/logback-core/src/main/java/module-info.java
@@ -5,8 +5,8 @@ module ch.qos.logback.core {
     // required by the optional SMTPAppenderBase component
     requires static java.naming;
 
-    requires static janino;
-    requires static commons.compiler;
+    requires static org.codehaus.janino;
+    requires static org.codehaus.commons.compiler;
 
     // transitive _imposes_ the presence of jakarta.mail on downstream users,
     // let them declare it if they need it
