diff -u -r intel-media-driver-20.1.1.old/media-driver/cmrtlib/linux/hardware/cm_device_os.cpp intel-media-driver-20.1.1/media-driver/cmrtlib/linux/hardware/cm_device_os.cpp
--- intel-media-driver-20.1.1.old/media-driver/cmrtlib/linux/hardware/cm_device_os.cpp	2020-04-14 10:52:29.330074894 +0000
+++ intel-media-driver-20.1.1/media-driver/cmrtlib/linux/hardware/cm_device_os.cpp	2020-04-14 10:58:15.500863916 +0000
@@ -63,7 +63,6 @@
 {
     INSERT_PROFILER_RECORD();
     int32_t result = CM_SUCCESS;
-    uint32_t i = 0;
     uint32_t k = 0;
     char* driver_name;
 
@@ -534,7 +533,7 @@
         return CM_INVALID_LIBVA_INITIALIZE;
     }
 
-    if (m_drmIndex < g_supportedAdapterCount)
+    if ((int32_t)m_drmIndex < g_supportedAdapterCount)
     {
         m_driFileDescriptor = GetRendererFileDescriptor(g_AdapterList[m_drmIndex]->nodes[2]);
     }
diff -u -r intel-media-driver-20.1.1.old/media-driver/cmrtlib/linux/hardware/drm_device.h intel-media-driver-20.1.1/media-driver/cmrtlib/linux/hardware/drm_device.h
--- intel-media-driver-20.1.1.old/media-driver/cmrtlib/linux/hardware/drm_device.h	2020-04-14 10:52:29.330074894 +0000
+++ intel-media-driver-20.1.1/media-driver/cmrtlib/linux/hardware/drm_device.h	2020-04-14 11:16:44.474619431 +0000
@@ -546,9 +546,10 @@
     memset(buffer, 0, sizeof(buffer));
     if (fd >= 0)
     {
-        int count = 0;
-        count = read(fd, buffer, sizeof(buffer));
+        int count = read(fd, buffer, sizeof(buffer));
         close(fd);
+        if (count < 0)
+            return -errno;
     }
 
     snprintf(resourcename, sizeof(resourcename), "%s/resource", pci_path);
