diff -u -r media_driver.orig/linux/common/codec/ddi/media_ddi_encode_av1.cpp media_driver/linux/common/codec/ddi/media_ddi_encode_av1.cpp
--- media_driver.orig/linux/common/codec/ddi/media_ddi_encode_av1.cpp	2022-09-30 21:14:38.593042000 +0200
+++ media_driver/linux/common/codec/ddi/media_ddi_encode_av1.cpp	2022-09-30 21:18:34.333014000 +0200
@@ -434,7 +434,7 @@
     av1SeqParams->CodingToolFlags.fields.enable_restoration   = seqParams->seq_fields.bits.enable_restoration;
 
     av1SeqParams->order_hint_bits_minus_1 = seqParams->order_hint_bits_minus_1;
-    av1SeqParams->SeqFlags.fields.HierarchicalFlag = seqParams->reserved8b;
+    av1SeqParams->SeqFlags.fields.HierarchicalFlag = seqParams->hierarchical_flag;
 
     return VA_STATUS_SUCCESS;
 }
@@ -547,7 +547,7 @@
     av1PicParams->PicFlags.fields.DisableFrameRecon            = picParams->picture_flags.bits.disable_frame_recon;
     av1PicParams->PicFlags.fields.PaletteModeEnable            = picParams->picture_flags.bits.palette_mode_enable;
     av1PicParams->PicFlags.fields.SegIdBlockSize               = picParams->seg_id_block_size;
-    av1PicParams->HierarchLevelPlus1                           = picParams->reserved8bits0;
+    av1PicParams->HierarchLevelPlus1                           = picParams->hierarchical_level_plus1;
 
     DDI_CHK_RET(
         MOS_SecureMemcpy(av1PicParams->filter_level,
