https://github.com/deater/linux_logo/pull/17
commit c3f23ac89189139e1b295a4fef50af336917b5fe
Author: Bernhard M. Wiedemann <bwiedemann@suse.de>
Date:   Fri May 22 12:21:36 2020 +0200

    Sort input file list
    
    so that logo_config and load_logos.h build in a reproducible way
    in spite of indeterministic filesystem readdir order
    
    See https://reproducible-builds.org/ for why this is good.

Index: linux_logo-6.0/Makefile
===================================================================
--- linux_logo-6.0.orig/Makefile
+++ linux_logo-6.0/Makefile
@@ -42,7 +42,7 @@ translations:
 	cd po && $(MAKE)
 
 logos-all:	logo_config
-	find ./logos -type f -a ! -name banner.logo -a ! -name classic.logo >> logo_config
+	find ./logos -type f -a ! -name banner.logo -a ! -name classic.logo | sort >> logo_config
 	$(MAKE) all
 
 logo_config:
