Index: ladish-1.3-gfcd24852/meson.build
===================================================================
--- ladish-1.3-gfcd24852.orig/meson.build
+++ ladish-1.3-gfcd24852/meson.build
@@ -12,11 +12,7 @@ fs = import('fs')
 if fs.is_dir('.git')
   # This block handles building from git source tree
   # version.h is generated during build
-  git_version = run_command(
-    ['git', 'rev-parse', '--short', 'HEAD'],
-    check : true,
-    capture : true,
-  ).stdout().strip()
+  git_version = "-"
 
   configure_file(output : 'version.h',
                  configuration : {
