--- mysql-connector-c++-8.0.19-org/jdbc/driver/CMakeLists.txt	2019-11-28 01:08:32.000000000 +0200
+++ mysql-connector-c++-8.0.19-src/jdbc/driver/CMakeLists.txt	2020-04-03 00:04:30.255859754 +0300
@@ -201,6 +201,10 @@
   target_include_directories(jdbc PRIVATE ${MYSQL_INCLUDE_DIR})
 endif()
 
+if(CMAKE_DL_LIBS)
+  target_link_libraries(jdbc PUBLIC ${CMAKE_DL_LIBS})
+endif()
+
 if(NOT BUILD_SHARED_LIBS)
   target_compile_definitions(jdbc PRIVATE -DCPPCONN_LIB_BUILD)
 else()
