diff -uNr mysql-connector-c++-8.0.25-src/jdbc/driver/nativeapi/libmysql_dynamic_proxy.cpp mysql-connector-c++-8.0.25-mod/jdbc/driver/nativeapi/libmysql_dynamic_proxy.cpp
--- mysql-connector-c++-8.0.25-src/jdbc/driver/nativeapi/libmysql_dynamic_proxy.cpp	2021-03-15 17:21:15.000000000 +0200
+++ mysql-connector-c++-8.0.25-mod/jdbc/driver/nativeapi/libmysql_dynamic_proxy.cpp	2021-05-12 17:00:33.080944723 +0300
@@ -440,7 +440,7 @@
 int
 LibmysqlDynamicProxy::get_option(MYSQL * mysql, enum mysql_option option, const void *arg)
 {
-  static ptr2mysql_get_option ptr2_get_option = symbol_safe_cast<ptr2mysql_options>(GetProcAddr("mysql_get_option"));
+  static ptr2mysql_get_option ptr2_get_option = symbol_safe_cast<ptr2mysql_get_option>(GetProcAddr("mysql_get_option"));
   if (ptr2_get_option != NULL) {
     if (((*ptr2_get_option)(mysql, option, arg))) {
       throw sql::InvalidArgumentException("Unsupported option provided to mysql_get_option()");
