Index: curl-7.63.0/lib/url.c
===================================================================
--- curl-7.63.0.orig/lib/url.c
+++ curl-7.63.0/lib/url.c
@@ -1976,9 +1976,13 @@ static CURLcode findprotocol(struct Curl
     /* it is allowed for "normal" request, now do an extra check if this is
        the result of a redirect */
     if(data->state.this_is_a_follow &&
-       !(data->set.redir_protocols & p->protocol))
+       !(data->set.redir_protocols & p->protocol)) {
       /* nope, get out */
-      ;
+       failf(data, "Redirect to protocol \"%s\" not supported or disabled in " LIBCURL_NAME,
+	     protostr);
+
+       return CURLE_UNSUPPORTED_PROTOCOL;
+    }
     else {
       /* Perform setup complement if some. */
       conn->handler = conn->given = p;
