--- libguestfs-1.43.2/python/setup.py.in.orig	2020-10-07 11:50:58.255281701 -0600
+++ libguestfs-1.43.2/python/setup.py.in	2020-10-07 11:51:19.023282602 -0600
@@ -64,7 +64,7 @@ This package contains the Python binding
                'structs.c',
                'stringlists-utils.c'],
 
-              include_dirs=['.', '../lib'],
+              include_dirs=['.', '../lib', '../include'],
               libraries=['guestfs'],
               define_macros=[('GUESTFS_PRIVATE', '1')],
           )
