Subject: customize: Suppress OCaml warning
From: Richard W.M. Jones rjones@redhat.com Tue Oct 5 20:53:25 2021 +0100
Date: Tue Oct 5 20:53:25 2021 +0100:
Git: a4930f5fad82e5358d565b8cf3610970e9646259

In OCaml 4.13:

File "perl_edit.ml", line 30, characters 2-13:
30 |   c_edit_file (verbose ()) g (Guestfs.c_pointer g) file expr
       ^^^^^^^^^^^
Error (warning 6 [labels-omitted]): label verbose was omitted in the application of this function.

--- a/m4/guestfs-ocaml.m4
+++ b/m4/guestfs-ocaml.m4
@@ -232,7 +232,7 @@ EOF
 ])
 
 dnl Flags we want to pass to every OCaml compiler call.
-OCAML_WARN_ERROR="-warn-error +C+D+E+F+L+M+P+S+U+V+Y+Z+X+52-3"
+OCAML_WARN_ERROR="-warn-error +C+D+E+F+L+M+P+S+U+V+Y+Z+X+52-3-6"
 AC_SUBST([OCAML_WARN_ERROR])
 OCAML_FLAGS="-g -annot $safe_string_option"
 AC_SUBST([OCAML_FLAGS])
