--- a/tools/virt-list-filesystems
+++ b/tools/virt-list-filesystems
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 # virt-list-filesystems
 # Copyright (C) 2009-2020 Red Hat Inc.
 #
--- a/tools/virt-list-partitions
+++ b/tools/virt-list-partitions
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 # virt-list-partitions
 # Copyright (C) 2010 Red Hat Inc.
 #
--- a/tools/virt-tar
+++ b/tools/virt-tar
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 # virt-tar
 # Copyright (C) 2009-2020 Red Hat Inc.
 #
--- a/tools/virt-win-reg
+++ b/tools/virt-win-reg
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 # virt-win-reg
 # Copyright (C) 2010 Red Hat Inc.
 #
