Description: Disable the version check for new upstream versions.
 This check can mainly serve certain closed source binary distributions,
 but is not adequate for free distributions managing their packages with
 their respective package manager. Users of those free distributions will
 only be confused by the announce of a new version and will in the worst
 case spoil their system by trying to install a version provided by the
 upstream link.
 The unfriendlyness of this 'feature' was extensivly discussed to no avail
 on https://discuss.tryton.org/t/notify-user-about-new-client-version/
Author: Mathias Behrle <mbehrle@debian.org>
Bug: https://bugs.tryton.org/issue7033
Forwarded: not-needed
Adaption for Tryton 6: Axel Braun <DocB@opensuse.org>

diff -U 3 -dHrN a/tryton/config.py b/tryton/config.py
--- a/tryton/config.py	2021-09-27 23:40:00.000000000 +0200
+++ b/tryton/config.py	2021-10-07 21:24:31.560694731 +0200
@@ -53,7 +53,8 @@
             'client.language_direction': 'ltr',
             'client.email': '',
             'client.limit': 1000,
-            'client.check_version': True,
+# Check version not required for *nix systems            
+            'client.check_version': False,
             'client.bus_timeout': 10 * 60,
             'icon.colors': '#3465a4,#555753,#cc0000',
             'tree.colors': '#777,#198754,#ffc107,#dc3545',
diff -U 3 -dHrN a/tryton/gui/main.py b/tryton/gui/main.py
--- a/tryton/gui/main.py	2021-09-27 23:40:00.000000000 +0200
+++ b/tryton/gui/main.py	2021-10-07 21:25:41.532349966 +0200
@@ -301,7 +301,8 @@
 
         section.append(_("PDA Mode"), 'app.mode-pda')
         section.append(_("Search Limit..."), 'app.search-limit')
-        section.append(_("Check Version"), 'app.check-version')
+# Check version not required for *nix systems   
+#        section.append(_("Check Version"), 'app.check-version')
 
         menu.append_section(_("Options"), section)
 
