--- configure.in.old	2013-05-31 17:45:03.395109000 +0200
+++ configure.in	2013-05-31 17:45:31.887586000 +0200
@@ -71,6 +71,7 @@
 AC_CHECK_HEADERS(sys/time.h)
 AC_CHECK_HEADERS(sys/signal.h)
 AC_CHECK_HEADERS(signal.h)
+AC_CHECK_HEADERS(sys/io.h)
 AC_CHECK_HEADERS(termios.h)
 AC_CHECK_HEADERS(grp.h)
 AC_CHECK_HEADERS(pwd.h)
--- src/RawImp.c.old	2013-05-31 17:46:32.595659000 +0200
+++ src/RawImp.c	2013-05-31 17:46:49.886950000 +0200
@@ -99,11 +99,9 @@
 #	include <linux/serial.h>
 #	include <linux/version.h>
 #endif
-#ifndef __APPLE__  /* dima */
-#ifndef PPC
-#include <sys/io.h>
-#endif /* PPC */
-#endif /* dima */
+#ifdef HAVE_SYS_IO_H
+#   include <sys/io.h>
+#endif
 
 extern int errno;
 #include "I2CImp.h"
