diff --git a/applets/install.sh b/applets/install.sh
index 415896893..7b51fd8f3 100755
--- a/applets/install.sh
+++ b/applets/install.sh
@@ -1,11 +1,11 @@
-#!/bin/sh
+#!/usr/bin/busybox sh
 
 export LC_ALL=POSIX
 export LC_CTYPE=POSIX
 
 prefix=$1
 if [ -z "$prefix" ]; then
-	echo "usage: applets/install.sh DESTINATION TYPE [OPTS ...]"
+	echo "usage: busybox.install DESTINATION TYPE [OPTS ...]"
 	echo "  TYPE is one of: --symlinks --hardlinks --binaries --scriptwrapper --none"
 	echo "  OPTS is one or more of: --cleanup --noclobber"
 	exit 1
@@ -13,9 +13,9 @@ fi
 shift # Keep only remaining options
 
 # Source the configuration
-. ./.config
+#. ./.config
 
-h=`sort busybox.links | uniq`
+h=`sort /usr/share/busybox/busybox.links | uniq`
 
 sharedlib_dir="0_lib"
 
@@ -77,8 +77,8 @@ if [ x"$cleanup" = x"1" ] && [ -e "$prefix/bin/busybox" ]; then
 fi
 
 rm -f "$prefix/bin/busybox" || exit 1
-mkdir -p "$prefix/bin" || exit 1
-install -m 755 busybox "$prefix/bin/busybox" || exit 1
+mkdir -p "$prefix/usr/bin" || exit 1
+install -m 755 /usr/bin/busybox "$prefix/bin/busybox" || exit 1
 
 for i in $h; do
 	appdir=`dirname "$i"`
