--- busybox-1.36.1/networking/udhcp/d6_dhcpc.c.old	2023-01-03 15:14:43.000000000 +0100
+++ busybox-1.36.1/networking/udhcp/d6_dhcpc.c	2023-12-08 11:46:16.570298842 +0100
@@ -45,7 +45,7 @@
 //config:	You can request bootfile-url with "-O bootfile_url" and
 //config:	bootfile-params with "-O bootfile_params".
 
-//applet:IF_UDHCPC6(APPLET(udhcpc6, BB_DIR_USR_BIN, BB_SUID_DROP))
+//applet:IF_UDHCPC6(APPLET(udhcpc6, BB_DIR_USR_SBIN, BB_SUID_DROP))
 
 //kbuild:lib-$(CONFIG_UDHCPC6) += d6_dhcpc.o d6_packet.o d6_socket.o common.o socket.o signalpipe.o
 //kbuild:lib-$(CONFIG_FEATURE_UDHCPC6_RFC3646) += domain_codec.o
