From: Some One <nobody@opensuse.org>
Date: Thu, 9 Apr 2015 14:55:39 +0200
Subject: [PATCH] no-badness-return.diff

===================================================================
---
 Filter.py | 2 +-
 rpmlint   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: rpmlint-rpmlint-1.11/Filter.py
===================================================================
--- rpmlint-rpmlint-1.11.orig/Filter.py
+++ rpmlint-rpmlint-1.11/Filter.py
@@ -130,7 +130,7 @@ def printAllReasons():
                 if len(last_reason):
                     printDescriptions(last_reason)
                 last_reason = reason
-        __print(diag)
+        __print(diag[:-1])
     if Config.info and len(last_reason):
         printDescriptions(last_reason)
     _diagnostic = list()
Index: rpmlint-rpmlint-1.11/rpmlint
===================================================================
--- rpmlint-rpmlint-1.11.orig/rpmlint
+++ rpmlint-rpmlint-1.11/rpmlint
@@ -207,7 +207,7 @@ def main():
               % (packages_checked, specfiles_checked,
                  printed_messages["E"], printed_messages["W"]))
 
-    if printed_messages["E"] > 0:
+    if badnessThreshold() < 0 and printed_messages["E"] > 0:
         sys.exit(64)
     sys.exit(0)
 
