From: Some One <nobody@opensuse.org>
Date: Thu, 9 Apr 2015 14:55:39 +0200
Subject: [PATCH] rpmgroup-checks.diff

===================================================================
---
 TagsCheck.py | 6 ++++++
 1 file changed, 6 insertions(+)

Index: rpmlint-rpmlint-1.11/TagsCheck.py
===================================================================
--- rpmlint-rpmlint-1.11.orig/TagsCheck.py
+++ rpmlint-rpmlint-1.11/TagsCheck.py
@@ -750,6 +750,8 @@ class TagsCheck(AbstractCheck.AbstractCh
         self._unexpanded_macros(pkg, 'Group', group)
         if not group:
             printError(pkg, 'no-group-tag')
+        elif pkg.name.endswith('-devel') and not group.startswith('Development/'):
+            printWarning(pkg, 'devel-package-with-non-devel-group', group)
         elif VALID_GROUPS and group not in VALID_GROUPS:
             printWarning(pkg, 'non-standard-group', group)
 
@@ -1085,6 +1087,10 @@ won't fool the specfile parser, and rebu
 '''There is no Group tag in your package. You have to specify a valid group
 in your spec file using the Group tag.''',
 
+'devel-package-with-non-devel-group',
+'''The package ends with -devel but does not have a RPM group starting with
+Development/''',
+
 'non-standard-group',
 '''The value of the Group tag in the package is not valid.  Valid groups are:
 "%s".''' % '", "'.join(VALID_GROUPS),
