--- proximity.py	2008-02-14 12:57:16.000000000 -0200
+++ blueproximity-1.py	2008-02-14 23:00:14.000000000 -0200
@@ -25,8 +25,8 @@
 ## This value gives us the base directory for language files and icons.
 # Set this value to './' for svn version
 # or to '/usr/share/blueproximity/' for packaged version
-dist_path = './' 
-
+dist_path = '/usr/share/blueproximity/' 
+pixmaps_path = 'pixmaps/'
 
 # system includes
 import os
@@ -42,7 +42,7 @@
 import gettext
 
 #Get the local directory since we are not installing anything
-local_path = dist_path + 'LANG/'
+local_path = '/usr/share/locale/'
 # Init the list of languages to support
 langs = []
 #Check the default locale
@@ -81,8 +81,8 @@
 except:
     print _("The program cannot import the module gobject.")
     print _("Please make sure the GObject bindings for python are installed.")
-    print _("e.g. with Ubuntu Linux, type")
-    print _(" sudo apt-get install python-gobject")
+    print _("e.g. with openSUSE, type as root")
+    print _("zypper install python-gobject2")
     sys.exit(1)
 try:
     from configobj import ConfigObj
@@ -90,8 +90,8 @@
 except:
     print _("The program cannot import the module ConfigObj or Validator.")
     print _("Please make sure the ConfigObject package for python is installed.")
-    print _("e.g. with Ubuntu Linux, type")
-    print _(" sudo apt-get install python-configobj")
+    print _("e.g. with openSUSE, type as root")
+    print _("zypper install python-configobj")
     sys.exit(1)
 IMPORT_BT=0
 try:
@@ -112,8 +112,8 @@
 if (IMPORT_BT!=2):
     print _("The program cannot import the module bluetooth.")
     print _("Please make sure the bluetooth bindings for python as well as bluez are installed.")
-    print _("e.g. with Ubuntu Linux, type")
-    print _(" sudo apt-get install python-bluez")
+    print _("e.g. with openSUSE, type as root")
+    print _("zypper install python-pybluez")
     sys.exit(1)
 try:
     import pygtk
@@ -122,16 +122,16 @@
 except:
     print _("The program cannot import the module pygtk.")
     print _("Please make sure the GTK2 bindings for python are installed.")
-    print _("e.g. with Ubuntu Linux, type")
-    print _(" sudo apt-get install python-gtk2")
+    print _("e.g. with openSUSE, type as root")
+    print _("zypper install python-gtk")
     sys.exit(1)
 try:
     import gtk.glade
 except:
     print _("The program cannot import the module glade.")
     print _("Please make sure the Glade2 bindings for python are installed.")
-    print _("e.g. with Ubuntu Linux, type")
-    print _(" sudo apt-get install python-glade2")
+    print _("e.g. with openSUSE type as root")
+    print _("zypper install pygtk2-libglade")
     sys.exit(1)
 
 
@@ -157,15 +157,15 @@
     
 
 ## The icon used at normal operation and in the info dialog.
-icon_base = 'blueproximity_base.svg'
+icon_base = pixmaps_path + 'blueproximity_base.svg'
 ## The icon used at distances greater than the unlock distance.
-icon_att = 'blueproximity_attention.svg'
+icon_att = pixmaps_path + 'blueproximity_attention.svg'
 ## The icon used if no proximity is detected.
-icon_away = 'blueproximity_nocon.svg'
+icon_away = pixmaps_path  + 'blueproximity_nocon.svg'
 ## The icon used during connection processes and with connection errors.
-icon_con = 'blueproximity_error.svg'
+icon_con = pixmaps_path + 'blueproximity_error.svg'
 ## The icon shown if we are in pause mode.
-icon_pause = 'blueproximity_pause.svg'
+icon_pause = pixmaps_path + 'blueproximity_pause.svg'
 
 
 ## This class represents the main configuration window and
