Index: low-memory-monitor-2.1/data/low-memory-monitor.service.in
===================================================================
--- low-memory-monitor-2.1.orig/data/low-memory-monitor.service.in
+++ low-memory-monitor-2.1/data/low-memory-monitor.service.in
@@ -31,6 +31,12 @@ ProtectKernelModules=true
 
 # Real-time
 RestrictRealtime=true
+# added automatically, for details please see
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectHostname=true
+ProtectKernelTunables=true
+ProtectKernelLogs=true
+# end of automatic additions 
 
 # Privilege escalation
 NoNewPrivileges=true
