From 91a9e41fed6269637da52d5f88699782a940b198 Mon Sep 17 00:00:00 2001
From: Rico Tzschichholz <ricotz@ubuntu.com>
Date: Mon, 22 Jun 2020 18:17:06 +0200
Subject: [PATCH] g-i: Annotate champlain_layer_set_view to allow NULL as view

---
 champlain/champlain-layer.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/champlain/champlain-layer.c b/champlain/champlain-layer.c
index 2c973de5..ec279010 100644
--- a/champlain/champlain-layer.c
+++ b/champlain/champlain-layer.c
@@ -58,11 +58,11 @@ champlain_layer_class_init (ChamplainLayerClass *klass)
 /**
  * champlain_layer_set_view:
  * @layer: a #ChamplainLayer
- * @view: a #ChamplainView
+ * @view: (nullable): a #ChamplainView
  *
  * #ChamplainView calls this method to pass a reference to itself to the layer
  * when the layer is added to the view. When the layer is removed from the
- * view, it passes NULL to the layer. Custom layers can implement this method
+ * view, it passes %NULL to the layer. Custom layers can implement this method
  * and perform the necessary initialization. This method should not be called
  * by user code.
  *
-- 
GitLab

