Index: libsndfile-1.0.31/programs/sndfile-deinterleave.c
===================================================================
--- libsndfile-1.0.31.orig/programs/sndfile-deinterleave.c
+++ libsndfile-1.0.31/programs/sndfile-deinterleave.c
@@ -96,6 +96,11 @@ main (int argc, char **argv)
 		} ;
 
 
+	if (sfinfo.channels > MAX_CHANNELS)
+	{	printf ("\nError : Too many channels %d in input file '%s'.\n", sfinfo.channels, argv[1]) ;
+		exit (1) ;
+		}
+
 	state.channels = sfinfo.channels ;
 	sfinfo.channels = 1 ;
 
