Index: xsp-4.7.1/packaging/opensuse/xsp2.init
===================================================================
--- xsp-4.7.1.orig/packaging/opensuse/xsp2.init
+++ xsp-4.7.1/packaging/opensuse/xsp2.init
@@ -111,7 +111,7 @@
 
 # Check for missing binaries (stale symlinks should not happen)
 # Note: Special treatment of stop for LSB conformance
-XSP2_BIN=/usr/lib/mono/2.0/xsp2.exe
+XSP2_BIN=/usr/lib/mono/xsp2/xsp2.exe
 XSP2_WRAPPER=/usr/bin/xsp2
 MONO_BIN=/usr/bin/mono
 test -x $XSP2_BIN || { echo "$XSP2_BIN not installed"; 
Index: xsp-4.7.1/scripts/Makefile.am
===================================================================
--- xsp-4.7.1.orig/scripts/Makefile.am
+++ xsp-4.7.1/scripts/Makefile.am
@@ -16,7 +16,7 @@ bin_SCRIPTS = $(bin2_scripts) $(tool2_sc
 
 CLEANFILES = $(bin2_scripts_real) $(tool2_scripts) $(bin_scripts) $(tool_scripts) $(bin4_scripts) $(tool4_scripts) $(binplain_scripts)
 
-plat_bindir4 = $(prefix)/lib/mono/4.5
+plat_bindir4 = $(prefix)/lib/mono/xsp4
 plat_tooldir4 = $(prefix)/lib/xsp/4.0
 
 REWRITE2 = sed \
Index: xsp-4.7.1/src/Mono.WebServer.Apache/Makefile.am
===================================================================
--- xsp-4.7.1.orig/src/Mono.WebServer.Apache/Makefile.am
+++ xsp-4.7.1/src/Mono.WebServer.Apache/Makefile.am
@@ -3,7 +3,7 @@ builddir=$(top_builddir)/src/Mono.WebSer
 MCSFLAGS= -unsafe -debug+ -debug:full -nologo -nowarn:618 $(WEBTRACING) -keyfile:$(top_srcdir)/src/mono.snk
 
 MODMONOSERVER4_EXE = mod-mono-server4.exe
-GACUTIL4=$(GACUTIL) -package 4.5
+GACUTIL4=$(GACUTIL) -package xsp4
 
 noinst_SCRIPTS = $(MODMONOSERVER4_EXE)
 
Index: xsp-4.7.1/src/Mono.WebServer.FastCgi/Makefile.am
===================================================================
--- xsp-4.7.1.orig/src/Mono.WebServer.FastCgi/Makefile.am
+++ xsp-4.7.1/src/Mono.WebServer.FastCgi/Makefile.am
@@ -1,9 +1,9 @@
-builddir=$(top_builddir)/src/Mono.WebServer.FastCgi
+bilddir=$(top_builddir)/src/Mono.WebServer.FastCgi
 
 MCSFLAGS= -debug+ -debug:full -nologo -nowarn:618 $(WEBTRACING) -unsafe+ -keyfile:$(top_srcdir)/src/mono.snk
 
 XSP4_EXE = fastcgi-mono-server4.exe
-GACUTIL4=$(GACUTIL) -package 4.5
+GACUTIL4=$(GACUTIL) -package xsp4
 
 noinst_SCRIPTS = $(XSP4_EXE)
 
Index: xsp-4.7.1/src/Mono.WebServer.XSP/Makefile.am
===================================================================
--- xsp-4.7.1.orig/src/Mono.WebServer.XSP/Makefile.am
+++ xsp-4.7.1/src/Mono.WebServer.XSP/Makefile.am
@@ -4,7 +4,7 @@ pkgconfigdir=$(libdir)/pkgconfig
 MCSFLAGS= -debug+ -debug:full -nologo -nowarn:618 -unsafe $(WEBTRACING) -keyfile:$(top_srcdir)/src/mono.snk
 
 XSP4_EXE = xsp4.exe
-GACUTIL4=$(GACUTIL) -package 4.5
+GACUTIL4=$(GACUTIL) -package xsp4
 
 noinst_SCRIPTS = $(XSP4_EXE)
 
Index: xsp-4.7.1/src/Mono.WebServer.XSP/xsp-2.pc.in
===================================================================
--- xsp-4.7.1.orig/src/Mono.WebServer.XSP/xsp-2.pc.in
+++ xsp-4.7.1/src/Mono.WebServer.XSP/xsp-2.pc.in
@@ -1,5 +1,5 @@
 prefix=@prefix@
-assemblies_dir=${prefix}/lib/mono/2.0
+assemblies_dir=${prefix}/lib/mono/xsp2
 Libraries=${assemblies_dir}/Mono.WebServer2.dll
 Executable=${assemblies_dir}/xsp2.exe
  
Index: xsp-4.7.1/src/Mono.WebServer.XSP/xsp-4.pc.in
===================================================================
--- xsp-4.7.1.orig/src/Mono.WebServer.XSP/xsp-4.pc.in
+++ xsp-4.7.1/src/Mono.WebServer.XSP/xsp-4.pc.in
@@ -1,5 +1,5 @@
 prefix=@prefix@
-assemblies_dir=${prefix}/lib/mono/4.5
+assemblies_dir=${prefix}/lib/mono/xsp4
 Libraries=${assemblies_dir}/Mono.WebServer2.dll
 Executable=${assemblies_dir}/xsp4.exe
  
Index: xsp-4.7.1/src/Mono.WebServer/Makefile.am
===================================================================
--- xsp-4.7.1.orig/src/Mono.WebServer/Makefile.am
+++ xsp-4.7.1/src/Mono.WebServer/Makefile.am
@@ -6,7 +6,7 @@ SYSTEMCORE_DEP=@SYSTEMCORE_DEP@
 monowebserver4dir = $(prefix)/lib/xsp/4.5
 monowebserver4_install = 4.0/Mono.WebServer2.dll
 monowebserver4_references = -r:System.Web.dll -r:System.Configuration.dll -r:Mono.Security.dll -r:System.Core.dll -r:Mono.Posix.dll
-GACUTIL4=$(GACUTIL) -package 4.5
+GACUTIL4=$(GACUTIL) -package xsp4
 
 noinst_SCRIPTS= $(monowebserver4_install)
 
Index: xsp-4.7.1/src/Mono.WebServer.Fpm/Makefile.am
===================================================================
--- xsp-4.7.1.orig/src/Mono.WebServer.Fpm/Makefile.am
+++ xsp-4.7.1/src/Mono.WebServer.Fpm/Makefile.am
@@ -3,7 +3,7 @@ builddir=$(top_builddir)/src/Mono.WebSer
 MCSFLAGS= -debug+ -debug:full -nologo -nowarn:618 $(WEBTRACING) -unsafe+ -keyfile:$(top_srcdir)/src/mono.snk
 
 XSP_EXE = mono-fpm.exe
-GACUTIL4=$(GACUTIL) -package 4.5
+GACUTIL4=$(GACUTIL) -package xsp4
 
 noinst_SCRIPTS = $(XSP_EXE) 
 
