Index: src/polkitgnomeauthenticationdialog.c
===================================================================
--- src/polkitgnomeauthenticationdialog.c.orig
+++ src/polkitgnomeauthenticationdialog.c
@@ -531,6 +531,7 @@ polkit_gnome_authentication_dialog_const
                                                           _("_Authenticate"),
                                                           GTK_RESPONSE_OK);
   gtk_dialog_set_default_response (GTK_DIALOG (dialog), GTK_RESPONSE_OK);
+  gtk_dialog_set_alternative_button_order (GTK_DIALOG (dialog), GTK_RESPONSE_OK, GTK_RESPONSE_CANCEL, -1);
 
   content_area = gtk_dialog_get_content_area (GTK_DIALOG (dialog));
   action_area = gtk_dialog_get_action_area (GTK_DIALOG (dialog));
