diff --git a/setup.py b/setup.py
index dfa79a58..4cf9073b 100755
--- a/setup.py
+++ b/setup.py
@@ -27,7 +27,6 @@
 tests_require = [
     "coverage[toml]==5.2.1",
     "freezegun==0.3.15",
-    "mock==2.0.0",
     "pretend==1.0.9",
     "pytest-cov==2.8.1",
     "pytest-httpx",
diff --git a/tests/conftest.py b/tests/conftest.py
index 64c9bf95..4bd41de0 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -2,10 +2,6 @@
 
 import pytest
 
-# Don't try to test asyncio since it is py3 only syntax
-if sys.version_info < (3, 5):
-    collect_ignore = ["test_asyncio_transport.py"]
-
 pytest.register_assert_rewrite("tests.utils")
 
 
diff --git a/tests/test_main.py b/tests/test_main.py
index bffc7230..f4504820 100644
--- a/tests/test_main.py
+++ b/tests/test_main.py
@@ -1,4 +1,5 @@
-from mock import patch
+from unittest.mock import patch
+
 from pretend import stub
 
 from zeep import __main__, client

