Index: pyee-7.0.1/tests/test_async.py
===================================================================
--- pyee-7.0.1.orig/tests/test_async.py	2020-01-30 21:22:49.000000000 +0100
+++ pyee-7.0.1/tests/test_async.py	2020-03-27 10:10:34.734403451 +0100
@@ -4,7 +4,7 @@ import pytest
 import pytest_asyncio.plugin  # noqa
 
 from asyncio import Future, wait_for
-from concurrent.futures import TimeoutError
+from asyncio.exceptions import TimeoutError
 from mock import Mock
 
 from twisted.internet.defer import ensureDeferred, succeed
