--- Canna37p3/Imakefile
+++ Canna37p3/Imakefile
@@ -18,7 +18,7 @@
         INCLUDE = ./include/canna
      RKIINCLUDE = ./include/RKindep
         DEFINES = $(CANNA_DEFINES)
-CONFIGURE_CMD = IN_MAKE=yes $(CANNAROOT)/confwrapper run ./configure --prefix=$(cannaPrefix) ConfigureArgs
+CONFIGURE_CMD = IN_MAKE=yes ./configure --prefix=$(cannaPrefix) ConfigureArgs
 PURE_CFLAGS = $(CDEBUGFLAGS) $(CCOPTIONS) /* $(THREADS_CFLAGS) $(MODULE_CFLAGS) */
 supportOldWchar = SupportOldWchar
 
--- Canna37p3/canuum/Imakefile
+++ Canna37p3/canuum/Imakefile
@@ -20,7 +20,7 @@
 
             TARGET = canuum
 XCOMM --prefix is not used but add just in case
-CONFIGURE_CMD = IN_MAKE=yes $(CANNAROOT)/confwrapper run ./configure --prefix=$(cannaPrefix) CanuumConfigureArgs
+CONFIGURE_CMD = IN_MAKE=yes ./configure --prefix=$(cannaPrefix) CanuumConfigureArgs
 PURE_CFLAGS = $(CDEBUGFLAGS) $(CCOPTIONS) /* $(THREADS_CFLAGS) $(MODULE_CFLAGS) */
 
 AllTarget(ProgramTargetName($(TARGET)))
