--- orig-gnome-music-40.1/data/org.gnome.Music.appdata.xml.in.in	2021-06-08 10:09:00.671407200 +0200
+++ gnome-music-40.1/data/org.gnome.Music.appdata.xml.in.in	2021-06-08 10:58:28.967854786 +0200
@@ -37,7 +37,6 @@
         <p>This release also updates translations.</p>
       </description>
     </release>
-  <releases>
     <release version="40.0" date="2021-03-21">
       <description>
         <p>
