--- a/instrumentation/events/lttng-module/block.h
+++ b/instrumentation/events/lttng-module/block.h
@@ -619,7 +619,8 @@ LTTNG_TRACEPOINT_EVENT(block_bio_bounce,
 	TP_ARGS(q, bio),
 
 	TP_FIELDS(
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 		ctf_integer(dev_t, dev, bio_dev(bio))
 #else
 		ctf_integer(dev_t, dev, bio->bi_bdev ? bio->bi_bdev->bd_dev : 0)
@@ -664,7 +665,8 @@ LTTNG_TRACEPOINT_EVENT(block_bio_complet
 #endif
 
 	TP_FIELDS(
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 		ctf_integer(dev_t, dev, bio_dev(bio))
 #else
 		ctf_integer(dev_t, dev, bio->bi_bdev->bd_dev)
@@ -698,7 +700,8 @@ LTTNG_TRACEPOINT_EVENT_CLASS(block_bio_m
 	TP_ARGS(q, rq, bio),
 
 	TP_FIELDS(
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 		ctf_integer(dev_t, dev, bio_dev(bio))
 #else
 		ctf_integer(dev_t, dev, bio->bi_bdev->bd_dev)
@@ -764,7 +767,8 @@ LTTNG_TRACEPOINT_EVENT(block_bio_queue,
 	TP_ARGS(q, bio),
 
 	TP_FIELDS(
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 		ctf_integer(dev_t, dev, bio_dev(bio))
 #else
 		ctf_integer(dev_t, dev, bio->bi_bdev->bd_dev)
@@ -855,7 +859,8 @@ LTTNG_TRACEPOINT_EVENT_CLASS(block_get_r
 	TP_ARGS(q, bio, rw),
 
 	TP_FIELDS(
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 		ctf_integer(dev_t, dev, bio ? bio_dev(bio) : 0)
 #else
 		ctf_integer(dev_t, dev, bio ? bio->bi_bdev->bd_dev : 0)
@@ -1020,7 +1025,8 @@ LTTNG_TRACEPOINT_EVENT(block_split,
 	TP_ARGS(q, bio, new_sector),
 
 	TP_FIELDS(
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 		ctf_integer(dev_t, dev, bio_dev(bio))
 #else
 		ctf_integer(dev_t, dev, bio->bi_bdev->bd_dev)
@@ -1063,7 +1069,8 @@ LTTNG_TRACEPOINT_EVENT(block_remap,
 	TP_ARGS(q, bio, dev, from),
 
 	TP_FIELDS(
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 		ctf_integer(dev_t, dev, bio_dev(bio))
 #else
 		ctf_integer(dev_t, dev, bio->bi_bdev->bd_dev)
--- a/instrumentation/events/lttng-module/btrfs.h
+++ b/instrumentation/events/lttng-module/btrfs.h
@@ -44,10 +44,12 @@ struct extent_state;
 #endif
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0) /* openSUSE */  || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,73,5,0,0, 4,4,73,6,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,82,6,0,0, 4,4,82,7,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,92,6,0,0, 4,4,92,7,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,103,6,0,0, 4,5,0,0,0,0))
+
 LTTNG_TRACEPOINT_EVENT(btrfs_transaction_commit,
 
 	TP_PROTO(const struct btrfs_root *root),
@@ -151,7 +153,8 @@ LTTNG_TRACEPOINT_EVENT_INSTANCE(btrfs__i
 )
 #endif
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 
 LTTNG_TRACEPOINT_EVENT(btrfs_get_extent,
 
@@ -368,7 +371,8 @@ LTTNG_TRACEPOINT_EVENT_CLASS(btrfs__orde
 				BTRFS_I(inode)->root->root_key.objectid)
 	)
 )
-#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0))
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 LTTNG_TRACEPOINT_EVENT_CLASS(btrfs__ordered_extent,
 
 	TP_PROTO(const struct inode *inode,
@@ -412,6 +416,7 @@ LTTNG_TRACEPOINT_EVENT_CLASS(btrfs__orde
 	)
 )
 #elif (LTTNG_SLE_KERNEL_RANGE(4,4,73,5,0,0, 4,4,73,6,0,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0) /* openSUSE */  || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,82,6,0,0, 4,4,82,7,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,92,6,0,0, 4,4,92,7,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,103,6,0,0, 4,5,0,0,0,0))
@@ -459,6 +464,7 @@ LTTNG_TRACEPOINT_EVENT_CLASS(btrfs__orde
 #endif
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0) /* openSUSE */  || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,73,5,0,0, 4,4,73,6,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,82,6,0,0, 4,4,82,7,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,92,6,0,0, 4,4,92,7,0,0) || \
@@ -664,6 +670,7 @@ LTTNG_TRACEPOINT_EVENT(btrfs_sync_file,
 #endif
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0) /* openSUSE */  || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,73,5,0,0, 4,4,73,6,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,82,6,0,0, 4,4,82,7,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,92,6,0,0, 4,4,92,7,0,0) || \
@@ -766,6 +773,7 @@ LTTNG_TRACEPOINT_EVENT(btrfs_add_block_g
 #endif
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0) /* openSUSE */  || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,73,5,0,0, 4,4,73,6,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,82,6,0,0, 4,4,82,7,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,92,6,0,0, 4,4,92,7,0,0) || \
@@ -1006,6 +1014,7 @@ LTTNG_TRACEPOINT_EVENT(btrfs_delayed_tre
 #endif
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0) /* openSUSE */  || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,73,5,0,0, 4,4,73,6,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,82,6,0,0, 4,4,82,7,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,92,6,0,0, 4,4,92,7,0,0) || \
@@ -1295,6 +1304,7 @@ LTTNG_TRACEPOINT_EVENT_INSTANCE_MAP(btrf
 )
 
 #elif (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0) /* openSUSE */  || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,73,5,0,0, 4,4,73,6,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,82,6,0,0, 4,4,82,7,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,92,6,0,0, 4,4,92,7,0,0) || \
@@ -1498,7 +1508,8 @@ LTTNG_TRACEPOINT_EVENT(btrfs_delayed_ref
 )
 #endif
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 
 LTTNG_TRACEPOINT_EVENT_CLASS(btrfs__chunk,
 
@@ -1568,7 +1579,8 @@ LTTNG_TRACEPOINT_EVENT_INSTANCE(btrfs__c
 	TP_ARGS(fs_info, map, offset, size)
 )
 
-#elif (LTTNG_SLE_KERNEL_RANGE(4,4,73,5,0,0, 4,4,73,6,0,0) || \
+#elif (LTTNG_KERNEL_RANGE(4,12,0, 4,13,0) /* openSUSE */  || \
+	LTTNG_SLE_KERNEL_RANGE(4,4,73,5,0,0, 4,4,73,6,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,82,6,0,0, 4,4,82,7,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,92,6,0,0, 4,4,92,7,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,103,6,0,0, 4,5,0,0,0,0))
@@ -1644,6 +1656,7 @@ LTTNG_TRACEPOINT_EVENT_INSTANCE(btrfs__c
 #endif /* #else #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,10,0)) */
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0) /* openSUSE */  || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,73,5,0,0, 4,4,73,6,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,82,6,0,0, 4,4,82,7,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,92,6,0,0, 4,4,92,7,0,0) || \
@@ -1737,7 +1750,8 @@ LTTNG_TRACEPOINT_EVENT(btrfs_space_reser
 )
 #endif
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 
 LTTNG_TRACEPOINT_EVENT_CLASS(btrfs__reserved_extent,
 
@@ -1956,7 +1970,8 @@ LTTNG_TRACEPOINT_EVENT_INSTANCE(btrfs__r
 	TP_ARGS(block_group, start, len)
 )
 
-#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0))
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 LTTNG_TRACEPOINT_EVENT_MAP(find_free_extent,
 
 	btrfs_find_free_extent,
@@ -2316,6 +2331,7 @@ LTTNG_TRACEPOINT_EVENT(btrfs_setup_clust
 #endif
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,14,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0) /* openSUSE */  || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,73,5,0,0, 4,4,73,6,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,82,6,0,0, 4,4,82,7,0,0) || \
 	LTTNG_SLE_KERNEL_RANGE(4,4,92,6,0,0, 4,4,92,7,0,0) || \
--- a/instrumentation/events/lttng-module/kmem.h
+++ b/instrumentation/events/lttng-module/kmem.h
@@ -133,7 +133,8 @@ LTTNG_TRACEPOINT_EVENT_MAP(mm_page_free_
 	)
 )
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0) || \
+	LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 LTTNG_TRACEPOINT_EVENT_MAP(mm_page_free_batched, kmem_mm_page_free_batched,
 
 	TP_PROTO(struct page *page),
--- a/instrumentation/events/lttng-module/kvm.h
+++ b/instrumentation/events/lttng-module/kvm.h
@@ -112,6 +112,7 @@ LTTNG_TRACEPOINT_EVENT(kvm_mmio,
 	|| LTTNG_KERNEL_RANGE(3,2,97, 3,3,0) \
 	|| LTTNG_UBUNTU_KERNEL_RANGE(4,13,16,38, 4,14,0,0) \
 	|| LTTNG_DEBIAN_KERNEL_RANGE(4,9,65,0,3,0, 4,10,0,0,0,0) \
+	|| LTTNG_KERNEL_RANGE(4,12,0, 4,13,0) /* openSUSE */ \
 	|| LTTNG_FEDORA_KERNEL_RANGE(4,14,13,300, 4,15,0,0))
 
 LTTNG_TRACEPOINT_EVENT(kvm_mmio,
--- a/instrumentation/events/lttng-module/mm_vmscan.h
+++ b/instrumentation/events/lttng-module/mm_vmscan.h
@@ -94,7 +94,8 @@ LTTNG_TRACEPOINT_EVENT(mm_vmscan_kswapd_
 
 #endif
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,17,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,17,0) \
+	|| LTTNG_KERNEL_RANGE(4,12,0, 4,13,0)) /* openSUSE */
 LTTNG_TRACEPOINT_EVENT(mm_vmscan_wakeup_kswapd,
 
 	TP_PROTO(int nid, int zid, int order, gfp_t gfp_flags),
