diff -Pdpru mpd-0.23.3.orig/systemd/system/mpd.service.in mpd-0.23.3/systemd/system/mpd.service.in
--- mpd-0.23.3.orig/systemd/system/mpd.service.in	2021-10-31 20:13:10.000000000 +0300
+++ mpd-0.23.3/systemd/system/mpd.service.in	2021-11-06 16:16:11.928539482 +0300
@@ -9,7 +9,7 @@ ExecStart=@prefix@/bin/mpd --no-daemon
 
 # Create /run/mpd (if MPD is launched without the socket unit and is
 # configured to bind listener sockets there).
-RuntimeDirectory=mpd
+#RuntimeDirectory=mpd
 
 # Enable this setting to ask systemd to watch over MPD, see
 # systemd.service(5).  This is disabled by default because it causes
diff -Pdpru mpd-0.23.3.orig/systemd/user/mpd.service.in mpd-0.23.3/systemd/user/mpd.service.in
--- mpd-0.23.3.orig/systemd/user/mpd.service.in	2021-10-31 20:13:10.000000000 +0300
+++ mpd-0.23.3/systemd/user/mpd.service.in	2021-11-06 16:16:32.508576374 +0300
@@ -9,7 +9,7 @@ ExecStart=@prefix@/bin/mpd --no-daemon
 
 # Create /run/user/$UID/mpd (if MPD is launched without the socket
 # unit and is configured to bind listener sockets there).
-RuntimeDirectory=mpd
+#RuntimeDirectory=mpd
 
 # Enable this setting to ask systemd to watch over MPD, see
 # systemd.service(5).  This is disabled by default because it causes
