Index: dolphin-25.03.70git.20260124T015104~32f41bd69/src/dolphinui.rc
===================================================================
--- dolphin-25.03.70git.20260124T015104~32f41bd69.orig/src/dolphinui.rc	2026-01-24 02:51:04.000000000 +0100
+++ dolphin-25.03.70git.20260124T015104~32f41bd69/src/dolphinui.rc	2026-01-24 08:10:22.418071207 +0100
@@ -90,6 +90,7 @@
             <Action name="invert_selection" />
             <Separator/>
             <Action name="go_back" />
+            <Action name="go_up" />
             <Action name="go_forward" />
         </disable>
     </State>
