Index: dolphin-19.11.70git.20230101T031501~0c1157f3/src/dolphinui.rc
===================================================================
--- dolphin-19.11.70git.20230101T031501~0c1157f3.orig/src/dolphinui.rc
+++ dolphin-19.11.70git.20230101T031501~0c1157f3/src/dolphinui.rc
@@ -105,6 +105,7 @@
         <text context="@title:menu">Main Toolbar</text>
         <Action name="go_back" />
         <Action name="go_forward" />
+        <Action name="go_up" />
         <Separator name="separator_1" />
         <Action name="icons" />
         <Action name="compact" />
