--- a/tensorflow-2.6.0/tensorflow/python/util/BUILD
+++ b/tensorflow-2.6.0/tensorflow/python/util/BUILD
@@ -458,7 +458,6 @@ py_library(
         "//tensorflow/python:global_test_configuration",
         ":tf_decorator",
         ":tf_export",
-        "@org_python_pypi_backports_weakref",
         "@com_google_protobuf//:protobuf_python",
         "//third_party/py/numpy",
         "@six_archive//:six",
--- a/tensorflow-2.6.0/tensorflow/tools/pip_package/BUILD
+++ b/tensorflow-2.6.0/tensorflow/tools/pip_package/BUILD
@@ -215,7 +215,6 @@ filegroup(
         "@nasm//:LICENSE",
         "@nsync//:LICENSE",
         "@opt_einsum_archive//:LICENSE",
-        "@org_python_pypi_backports_weakref//:LICENSE",
         "@pasta//:LICENSE",
         "@png//:LICENSE",
         "@six_archive//:LICENSE",
--- a/tensorflow-2.6.0/tensorflow/workspace2.bzl
+++ b/tensorflow-2.6.0/tensorflow/workspace2.bzl
@@ -517,17 +517,6 @@ def _tf_repositories():
     )
 
     tf_http_archive(
-        name = "org_python_pypi_backports_weakref",
-        build_file = "//third_party:backports_weakref.BUILD",
-        sha256 = "8813bf712a66b3d8b85dc289e1104ed220f1878cf981e2fe756dfaabe9a82892",
-        strip_prefix = "backports.weakref-1.0rc1/src",
-        urls = [
-            "https://storage.googleapis.com/mirror.tensorflow.org/pypi.python.org/packages/bc/cc/3cdb0a02e7e96f6c70bd971bc8a90b8463fda83e264fa9c5c1c98ceabd81/backports.weakref-1.0rc1.tar.gz",
-            "https://pypi.python.org/packages/bc/cc/3cdb0a02e7e96f6c70bd971bc8a90b8463fda83e264fa9c5c1c98ceabd81/backports.weakref-1.0rc1.tar.gz",
-        ],
-    )
-
-    tf_http_archive(
         name = "dill_archive",
         build_file = "//third_party:dill.BUILD",
         system_build_file = "//third_party/systemlibs:dill.BUILD",
--- a/tensorflow-2.6.0/third_party/backports_weakref.BUILD
+++ /dev/null
@@ -1,22 +0,0 @@
-# Description:
-#   Backport of new features in Python's weakref module.
-
-package(default_visibility = ["//visibility:public"])
-
-licenses(["notice"])  # Python 2.0
-
-py_library(
-    name = "org_python_pypi_backports_weakref",
-    srcs = [
-        "backports/__init__.py",
-        "backports/weakref.py",
-    ],
-    srcs_version = "PY3",
-)
-
-genrule(
-    name = "license",
-    srcs = ["@org_python_license"],
-    outs = ["LICENSE"],
-    cmd = "cp $< $@",
-)
