Index: tensorflow2-2.4.0/tensorflow/python/keras/saving/hdf5_format.py
===================================================================
--- tensorflow2-2.4.0.orig/tensorflow/python/keras/saving/hdf5_format.py
+++ tensorflow2-2.4.0/tensorflow/python/keras/saving/hdf5_format.py
@@ -179,7 +179,7 @@ def load_model_from_hdf5(filepath, custo
     model_config = f.attrs.get('model_config')
     if model_config is None:
       raise ValueError('No model found in config file.')
-    model_config = json_utils.decode(model_config.decode('utf-8'))
+    model_config = json_utils.decode(model_config)
     model = model_config_lib.model_from_config(model_config,
                                                custom_objects=custom_objects)
 
@@ -193,7 +193,7 @@ def load_model_from_hdf5(filepath, custo
         logging.warning('No training configuration found in the save file, so '
                         'the model was *not* compiled. Compile it manually.')
         return model
-      training_config = json_utils.decode(training_config.decode('utf-8'))
+      training_config = json_utils.decode(training_config)
 
       # Compile model.
       model.compile(**saving_utils.compile_args_from_training_config(
@@ -659,11 +659,11 @@ def load_weights_from_hdf5_group(f, laye
           and weights file.
   """
   if 'keras_version' in f.attrs:
-    original_keras_version = f.attrs['keras_version'].decode('utf8')
+    original_keras_version = f.attrs['keras_version']
   else:
     original_keras_version = '1'
   if 'backend' in f.attrs:
-    original_backend = f.attrs['backend'].decode('utf8')
+    original_backend = f.attrs['backend']
   else:
     original_backend = None
 
@@ -678,7 +678,7 @@ def load_weights_from_hdf5_group(f, laye
   for name in layer_names:
     g = f[name]
     weight_names = load_attributes_from_hdf5_group(g, 'weight_names')
-    if weight_names:
+    if np.any(weight_names):
       filtered_layer_names.append(name)
   layer_names = filtered_layer_names
   if len(layer_names) != len(filtered_layers):
@@ -730,11 +730,11 @@ def load_weights_from_hdf5_group_by_name
           and weights file and skip_match=False.
   """
   if 'keras_version' in f.attrs:
-    original_keras_version = f.attrs['keras_version'].decode('utf8')
+    original_keras_version = f.attrs['keras_version']
   else:
     original_keras_version = '1'
   if 'backend' in f.attrs:
-    original_backend = f.attrs['backend'].decode('utf8')
+    original_backend = f.attrs['backend']
   else:
     original_backend = None
 
@@ -849,13 +849,13 @@ def load_attributes_from_hdf5_group(grou
       data: Attributes data.
   """
   if name in group.attrs:
-    data = [n.decode('utf8') for n in group.attrs[name]]
+    data = group.attrs[name]
   else:
     data = []
     chunk_id = 0
     while '%s%d' % (name, chunk_id) in group.attrs:
       data.extend(
-          [n.decode('utf8') for n in group.attrs['%s%d' % (name, chunk_id)]])
+          group.attrs['%s%d' % (name, chunk_id)])
       chunk_id += 1
   return data
 
