--- a/profiles/apparmor.d/lsusb	2026-04-19 21:31:46.468630454 +0200
+++ b/profiles/apparmor.d/lsusb	2026-05-16 20:39:39.074351473 +0200
@@ -15,6 +15,7 @@
 
 profile lsusb /usr/bin/lsusb {
   include <abstractions/base>
+  include <abstractions/consoles>
 
   /usr/bin/lsusb mr,
 
@@ -43,8 +44,11 @@
   # can silence the denial.
   deny capability sys_admin,
 
+  /etc/udev/hwdb.bin r,
+
   /dev/bus/usb/@{d}@{d}@{d}/@{d}@{d}@{d} rw,
 
+  / r,
   /dev/ r,
   /dev/bus/usb/ r,
   @{run}/udev/data/*usb:* r,
@@ -54,6 +58,9 @@
   @{sys}/bus/usb/devices/ r,
   @{sys}/class/ r,
 
+  @{sys}/devices/@{pci_bus}/uevent r,
+  @{sys}/devices/@{pci_bus}/**/uevent r,
+
   @{sys}/devices/**/usb[0-9]**/uevent r,
   @{sys}/devices/**/usb[0-9]**/busnum r,
   @{sys}/devices/**/usb[0-9]**/descriptors r,


