Index: libraries/libapparmor/swig/ruby/extconf.rb
===================================================================
--- a/libraries/libapparmor/swig/ruby/extconf.rb.orig	2022-02-10 17:54:05.008544807 +0100
+++ b/libraries/libapparmor/swig/ruby/extconf.rb	2022-02-10 17:54:21.792506325 +0100
@@ -20,7 +20,14 @@ if find_library('apparmor', 'parse_recor
   # hack 2: strip all rpath references
   open('Makefile.ruby', 'w') do |out|
     IO.foreach('Makefile') do |line|
-      out.puts line.gsub(/-Wl,-R'[^']*'/, '')
+      l = line.gsub(/-Wl,-R'[^']*'/, '')
+      # oldincludedir = $(DESTDIR)/usr/include
+      # -> oldincludedir = /usr/include
+      l = l.gsub(/(oldincludedir)\s+=\s+\$\(DESTDIR\)(.*)/) {  |m| "#{$1} = #{$2}" }
+      # hdrdir = $(includedir)/$(RUBY_VERSION_NAME)
+      #  -> hdrdir = $(oldincludedir)/$(RUBY_VERSION_NAME)
+      l = l.gsub(/(hdrdir)\s+=\s+\$\(includedir\)(.*)/) {  |m| "#{$1} = $(oldincludedir)#{$2}" }
+      out.puts l
     end
   end
 else
