--- a/profiles/apparmor.d/who	2026-02-27 00:21:24.123695275 +0100
+++ b/profiles/apparmor.d/who	2026-05-25 21:38:06.473530856 +0200
@@ -23,14 +23,22 @@
   #   man page
   # Overall result: the above calls are not mediated by LSMs
 
+  # Nevertheless, /dev/pts/ is needed
+  /dev/pts/ r,
+
   /usr/bin/who mr,
 
+  # uutils ships Fluent localization files used for translated output
+  /usr/share/coreutils/locales/** r,
+
+  /run/systemd/sessions/@{int} r,
+
   # who sends the "exists" signal via kill(pid, 0)
   capability kill,
 
   # Deny the writes allowed by abstractions/wutmp
   audit deny /var/** w,
   audit deny @{run}/utmp w,
-  
+
   include if exists <local/who>
 }


