Patch-mainline: To be upstreamed
References: bnc#1084865
Author: Michal Srb <msrb@suse.com>
Subject: Change button layout in ServerDialog.

To fit strings in languages with longer words...
---
 vncviewer/ServerDialog.cxx | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/vncviewer/ServerDialog.cxx b/vncviewer/ServerDialog.cxx
index de67f87b..0a8aa775 100644
--- a/vncviewer/ServerDialog.cxx
+++ b/vncviewer/ServerDialog.cxx
@@ -53,7 +53,7 @@ ServerDialog::ServerDialog()
   
   serverName = new Fl_Input(x, y, w() - margin*2 - server_label_width, INPUT_HEIGHT, _("VNC server:"));
 
-  int adjust = (w() - 20) / 4;
+  int adjust = (w() - 20) / 3;
   int button_width = adjust - margin/2;
 
   x = margin;
@@ -76,6 +76,8 @@ ServerDialog::ServerDialog()
   
   x = 0;
   y += margin/2 + BUTTON_HEIGHT;
+  adjust = (w() - 20) / 4;
+  button_width = adjust - margin/2;
 
   divider = new Fl_Box(x, y, w(), 2);
   divider->box(FL_THIN_DOWN_FRAME);
-- 
2.13.6

