--- a/Makefile	2015-11-22 00:46:57.000000000 +0100
+++ b/Makefile	2016-05-24 12:23:43.091977974 +0200
@@ -11,8 +11,7 @@
 # everything needed to run, just short of installation
 #
 all: test
-	make -C src all
-	make -C images all
+	$(MAKE) -C src all
 
 #
 #  make help
@@ -45,7 +44,6 @@
 #
 alles: test
 	$(MAKE) -C src alles
-	$(MAKE) -C images all
 
 #
 # documentation files
@@ -104,9 +102,6 @@
 #
 install:  all
 	$(MAKE) -C src install
-	$(MAKE) -C images install
-	$(MAKE) -C hooks install
-	$(MAKE) -C sample install
 	$(MAKE) -C scripts install
 	$(MAKE) -C man install
 	$(MAKE) -C dos install
@@ -131,8 +126,5 @@
 
 uninstall:
 	$(MAKE) -C src uninstall
-	$(MAKE) -C images uninstall
-	$(MAKE) -C hooks uninstall
-	$(MAKE) -C sample uninstall
 	$(MAKE) -C scripts uninstall
 	$(MAKE) -C man uninstall
