Index: Makefile.in
===================================================================
--- a/Makefile.in
+++ b/Makefile.in
@@ -4051,7 +4051,7 @@ version.mac: genversion$(EXEEXT)
 	$(top_builddir)/genversion$(EXEEXT) $@
 
 genversion.$(OBJEXT): modules/preprocs/nasm/genversion.c
-	$(CC_FOR_BUILD) $(DEFAULT_INCLUDES) $(INCLUDES) -c -o $@ `test -f modules/preprocs/nasm/genversion.c || echo '$(srcdir)/'`modules/preprocs/nasm/genversion.c
+	$(CC_FOR_BUILD) $(CFLAGS) $(DEFAULT_INCLUDES) $(INCLUDES) -c -o $@ `test -f modules/preprocs/nasm/genversion.c || echo '$(srcdir)/'`modules/preprocs/nasm/genversion.c
 
 #EXTRA_DIST += modules/preprocs/gas/tests/rawpp_test.sh
 #EXTRA_DIST += modules/preprocs/gas/tests/longline.asm
@@ -4078,7 +4078,7 @@ module.c: $(top_srcdir)/libyasm/module.i
 	$(top_builddir)/genmodule$(EXEEXT) $(top_srcdir)/libyasm/module.in Makefile
 
 genmodule.$(OBJEXT): libyasm/genmodule.c
-	$(CC_FOR_BUILD) $(DEFAULT_INCLUDES) $(INCLUDES) -c -o $@ `test -f libyasm/genmodule.c || echo '$(srcdir)/'`libyasm/genmodule.c
+	$(CC_FOR_BUILD) $(CFLAGS) $(DEFAULT_INCLUDES) $(INCLUDES) -c -o $@ `test -f libyasm/genmodule.c || echo '$(srcdir)/'`libyasm/genmodule.c
 
 @BUILD_MAN_TRUE@yasm.1: frontends/yasm/yasm.xml
 @BUILD_MAN_TRUE@	$(XMLTO) -o $(top_builddir) man $(srcdir)/frontends/yasm/yasm.xml
@@ -4110,7 +4110,7 @@ install-exec-hook: python-install
 uninstall-hook: python-uninstall
 
 genstring.$(OBJEXT): genstring.c
-	$(CC_FOR_BUILD) $(DEFAULT_INCLUDES) $(INCLUDES) -c -o $@ `test -f genstring.c || echo '$(srcdir)/'`genstring.c
+	$(CC_FOR_BUILD) $(CFLAGS) $(DEFAULT_INCLUDES) $(INCLUDES) -c -o $@ `test -f genstring.c || echo '$(srcdir)/'`genstring.c
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
Index: Makefile.am
===================================================================
--- a/Makefile.am
+++ b/Makefile.am
@@ -158,5 +158,5 @@ genstring_LDADD = genstring.$(OBJEXT)
 genstring_LINK = $(CCLD_FOR_BUILD) -o $@
 
 genstring.$(OBJEXT): genstring.c
-	$(CC_FOR_BUILD) $(DEFAULT_INCLUDES) $(INCLUDES) -c -o $@ `test -f genstring.c || echo '$(srcdir)/'`genstring.c
+	$(CC_FOR_BUILD) $(CFLAGS) $(DEFAULT_INCLUDES) $(INCLUDES) -c -o $@ `test -f genstring.c || echo '$(srcdir)/'`genstring.c
 
Index: libyasm/Makefile.inc
===================================================================
--- a/libyasm/Makefile.inc
+++ b/libyasm/Makefile.inc
@@ -41,7 +41,7 @@ genmodule_LDADD = genmodule.$(OBJEXT)
 genmodule_LINK = $(CCLD_FOR_BUILD) -o $@
 
 genmodule.$(OBJEXT): libyasm/genmodule.c
-	$(CC_FOR_BUILD) $(DEFAULT_INCLUDES) $(INCLUDES) -c -o $@ `test -f libyasm/genmodule.c || echo '$(srcdir)/'`libyasm/genmodule.c
+	$(CC_FOR_BUILD) $(CFLAGS) $(DEFAULT_INCLUDES) $(INCLUDES) -c -o $@ `test -f libyasm/genmodule.c || echo '$(srcdir)/'`libyasm/genmodule.c
 
 EXTRA_DIST += libyasm/module.in
 
Index: modules/preprocs/nasm/Makefile.inc
===================================================================
--- a/modules/preprocs/nasm/Makefile.inc.orig
+++ b/modules/preprocs/nasm/Makefile.inc
@@ -31,7 +31,7 @@ genversion_LDADD = genversion.$(OBJEXT)
 genversion_LINK = $(CCLD_FOR_BUILD) -o $@
 
 genversion.$(OBJEXT): modules/preprocs/nasm/genversion.c
-	$(CC_FOR_BUILD) $(DEFAULT_INCLUDES) $(INCLUDES) -c -o $@ `test -f modules/preprocs/nasm/genversion.c || echo '$(srcdir)/'`modules/preprocs/nasm/genversion.c
+	$(CC_FOR_BUILD) $(CFLAGS) $(DEFAULT_INCLUDES) $(INCLUDES) -c -o $@ `test -f modules/preprocs/nasm/genversion.c || echo '$(srcdir)/'`modules/preprocs/nasm/genversion.c
 
 EXTRA_DIST += modules/preprocs/nasm/tests/Makefile.inc
 
