diff --git a/sfd/Makefile b/sfd/Makefile
index fdbc61f..67e2dc9 100644
--- a/sfd/Makefile
+++ b/sfd/Makefile
@@ -11,9 +11,9 @@ FF=fontforge -lang=ff -script
 FFPY=fontforge -lang=py -script 
 FFPY=fontforge -lang=py -script 
 
-ffversion=`fontforge --version 2> /dev/null | grep '^fontforge' | sed -e 's/^fontforge //'`
+ffversion:=`fontforge -version 2> /dev/null | grep '^libfontforge' | sed -e 's/^libfontforge //'`
 
-TESTFF=if test -z ${ffversion}; then echo FontForge program is required to build FreeFont; exit 1; fi; if test `fontforge --version 2> /dev/null | grep '^fontforge' | sed -e 's/^fontforge //'` -lt 20080429 ; then echo FontForge version too old; exit 1; fi
+TESTFF:=if test -z $(ffversion); then echo FontForge program is required to build FreeFont; exit 1; fi; if test $(ffversion) -lt 20080429 ; then echo FontForge version too old; exit 1; fi
 
 
 .SUFFIXES: $(SUFFIXES) .sfd .ttf .otf .woff
