--- /dev/null
+++ b/dune-project
@@ -0,0 +1,2 @@
+(lang dune 1.0)
+(name rresult)
--- /dev/null
+++ b/src/dune
@@ -0,0 +1,14 @@
+(library
+ (name rresult)
+ (public_name rresult)
+ (libraries result)
+ (modules rresult)
+ (flags :standard -w -27)
+ (wrapped false))
+
+(library
+ (name rresult_top)
+ (public_name rresult.top)
+ (libraries compiler-libs.toplevel rresult)
+ (modules rresult_top)
+ (wrapped false))
