diff -ruN a/data/audio-recorder.desktop b/data/audio-recorder.desktop
--- a/data/audio-recorder.desktop	2017-11-04 16:48:34.754392449 -0400
+++ b/data/audio-recorder.desktop	2017-11-04 16:49:48.220546146 -0400
@@ -1,6 +1,6 @@
 [Desktop Entry]
 GenericName=Audio Recorder
-Categories=GNOME;AudioVideo;Recorder
+Categories=GNOME;AudioVideo;Recorder;
 Exec=audio-recorder
 Icon=audio-recorder
 StartupNotify=true
@@ -51,36 +51,3 @@
 Comment[uk]=Легкий в застосуванні інструмент для аудіо запису
 Comment[vi]=Trình ghi âm
 
-# Right-click menu items for Unity's launcher icon.
-# Ref: https://wiki.ubuntu.com/Unity/LauncherAPI
-Actions=StartRecording;StopRecording;PauseRecording;ShowWindow
-
-[Desktop Action StartRecording]
-Name=Start
-Name[fi_FI]=Aloita
-Name[nb_NO]=Start
-Exec=audio-recorder --command=start
-OnlyShowIn=Unity;GNOME;
-
-[Desktop Action StopRecording]
-Name=Stop
-Name[fi_FI]=Lopeta
-Name[nb_NO]=Stopp
-Exec=audio-recorder --command=stop
-OnlyShowIn=Unity;GNOME;
-
-[Desktop Action PauseRecording]
-Name=Pause
-Name[fi_FI]=Keskeytä väliaikaisesti
-Name[nb_NO]=Pause
-Exec=audio-recorder --command=pause
-OnlyShowIn=Unity;GNOME;
-
-[Desktop Action ShowWindow]
-Name=Show window
-Name[fi_FI]=Näytä ikkuna
-Name[nb_NO]=Vis applikasjon
-Exec=audio-recorder --command=show
-OnlyShowIn=Unity;GNOME;
-
-
