Index: django-json-widget-1.1.1/setup.py
===================================================================
--- django-json-widget-1.1.1.orig/setup.py
+++ django-json-widget-1.1.1/setup.py
@@ -56,7 +56,7 @@ setup(
         'django_json_widget',
     ],
     include_package_data=True,
-    install_requires=['future'], # Only required for Django < 3.1
+    install_requires=[],
     license="MIT",
     zip_safe=False,
     keywords='django-json-widget',
