diff -ur siproxd-0.8.1dev.orig/src/plugin_fix_bogus_via.c siproxd-0.8.1dev/src/plugin_fix_bogus_via.c
--- siproxd-0.8.1dev.orig/src/plugin_fix_bogus_via.c	2010-01-07 12:44:38.000000000 +0100
+++ siproxd-0.8.1dev/src/plugin_fix_bogus_via.c	2010-11-01 23:07:12.142684601 +0100
@@ -103,7 +103,7 @@
 
    /* Incoming SIP message? */
 DEBUGC(DBCLASS_PLUGIN, "plugin_fix_bogus_via: type=%i", type);
-   if (type == REQTYP_INCOMING) {
+   if (type == REQTYP_INCOMING ) {
 
       if((via = osip_list_get(&(ticket->sipmsg->vias), 0)) == NULL) {
          WARN("no Via header found in incoming SIP message");
@@ -121,6 +121,9 @@
          if (sip_patch_topvia(ticket) == STS_FAILURE) {
             ERROR("patching inbound Via failed!");
          }
+      } else {
+	 char *in=inet_ntoa(from.sin_addr);
+	 DEBUGC(DBCLASS_PLUGIN, "plugin_fix_bogus_via: addr %s non in network acl %s", in, plugin_cfg.networks);
       }
    }
    return STS_SUCCESS;
Solo in siproxd-0.8.1dev/src: tags
