--- check_rsync.orig	2011-05-03 12:15:11.465040210 +0200
+++ check_rsync	2011-05-03 12:15:46.924579509 +0200
@@ -101,7 +101,7 @@
 alarm($TIMEOUT);
 
 # Get a list of modules to see if rsync is up
-my $command = "/usr/bin/rsync --port=$port $source";
+my $command = "/usr/bin/rsync --timeout=$TIMEOUT --contimeout=$TIMEOUT --port=$port $source";
 
 # Workaround to kill stale rsync processes
 $cpid = open(RSYNC, "$command|") or report_error("Unable to execute rsync: $!");
