--- xgboost-2.0.0/jvm-packages/create_jni.py	2023-10-07 11:33:17.581438398 +0200
+++ xgboost-2.0.0/jvm-packages/create_jni.py	2023-10-07 11:57:33.488402381 +0200
@@ -122,8 +122,8 @@
             run("cmake --build . --config Release" + maybe_parallel_build)
 
         with cd("demo/CLI/regression"):
-            run(f'"{sys.executable}" mapfeat.py')
-            run(f'"{sys.executable}" mknfold.py machine.txt 1')
+            run("{} mapfeat.py".format(sys.executable))
+            run("{} mknfold.py machine.txt 1".format(sys.executable))
 
     xgboost4j = 'xgboost4j-gpu' if cli_args.use_cuda == 'ON' else 'xgboost4j'
     xgboost4j_spark = 'xgboost4j-spark-gpu' if cli_args.use_cuda == 'ON' else 'xgboost4j-spark'
@@ -153,8 +153,8 @@
     print("copying train/test files")
     maybe_makedirs("{}/src/test/resources".format(xgboost4j_spark))
     with cd("../demo/CLI/regression"):
-        run(f'"{sys.executable}" mapfeat.py')
-        run(f'"{sys.executable}" mknfold.py machine.txt 1')
+        run("{} mapfeat.py".format(sys.executable))
+        run("{} mknfold.py machine.txt 1".format(sys.executable))
 
     for file in glob.glob("../demo/CLI/regression/machine.txt.t*"):
         cp(file, "{}/src/test/resources".format(xgboost4j_spark))
