diff -ru old/src/bc/configure new/src/bc/configure
--- old/src/bc/configure	2021-01-11 23:44:10.000000000 +0100
+++ new/src/bc/configure	2021-04-27 13:40:11.656962958 +0200
@@ -7052,7 +7052,7 @@
   LIBSFX=la
   WXLIBS=WXLIBSDYN
   ICP="${LIBTOOLPROG} --mode=install cp"
-  ICP_LIB="${LIBTOOLPROG} --mode=install install -s"
+  ICP_LIB="${LIBTOOLPROG} --mode=install install"
   ICP_STATIC_LIB='$(NOOP)'
   MRLIBINSTALL="install-lib"
   LIBFINISH="${LIBTOOLPROG} --mode=finish"
diff -ru old/src/bc/configure.ac new/src/bc/configure.ac
--- old/src/bc/configure.ac	2021-01-11 23:44:10.000000000 +0100
+++ new/src/bc/configure.ac	2021-04-27 13:40:12.884967341 +0200
@@ -1358,7 +1358,7 @@
   LIBSFX=la
   WXLIBS=WXLIBSDYN
   ICP="${LIBTOOLPROG} --mode=install cp"
-  ICP_LIB="${LIBTOOLPROG} --mode=install install -s"
+  ICP_LIB="${LIBTOOLPROG} --mode=install install"
   ICP_STATIC_LIB='$(NOOP)'
   MRLIBINSTALL="install-lib"
   LIBFINISH="${LIBTOOLPROG} --mode=finish"
diff -ru old/src/cs/c/configure new/src/cs/c/configure
--- old/src/cs/c/configure	2021-01-11 23:44:14.000000000 +0100
+++ new/src/cs/c/configure	2021-04-27 13:40:17.672984418 +0200
@@ -5570,7 +5570,7 @@
   LTA="la"
   STRIP_LIB_DEBUG=":"
   LIBSFX=la
-  ICP_LIB="${LIBTOOLPROG} --mode=install install -s"
+  ICP_LIB="${LIBTOOLPROG} --mode=install install"
 
   CONFIGURE_RACKET_SO_COMPILE="env PLT_CS_MAKE_LINK_SHARED=y"
 else
diff -ru old/src/cs/c/configure.ac new/src/cs/c/configure.ac
--- old/src/cs/c/configure.ac	2021-01-11 23:44:14.000000000 +0100
+++ new/src/cs/c/configure.ac	2021-04-27 13:39:05.168699229 +0200
@@ -688,7 +688,7 @@
   LTA="la"
   STRIP_LIB_DEBUG=":"
   LIBSFX=la
-  ICP_LIB="${LIBTOOLPROG} --mode=install install -s"
+  ICP_LIB="${LIBTOOLPROG} --mode=install install"
 
   CONFIGURE_RACKET_SO_COMPILE="env PLT_CS_MAKE_LINK_SHARED=y"
 else
diff -ru old/src/rktio/configure new/src/rktio/configure
--- old/src/rktio/configure	2021-01-11 23:44:18.000000000 +0100
+++ new/src/rktio/configure	2021-04-27 13:40:21.736998921 +0200
@@ -4624,7 +4624,7 @@
   LTA="la"
   STRIP_LIB_DEBUG=":"
   LIBSFX=la
-  ICP_LIB="${LIBTOOLPROG} --mode=install install -s"
+  ICP_LIB="${LIBTOOLPROG} --mode=install install"
 else
   ICP=cp
   LTO="o"
diff -ru old/src/rktio/configure.ac new/src/rktio/configure.ac
--- old/src/rktio/configure.ac	2021-01-11 23:44:18.000000000 +0100
+++ new/src/rktio/configure.ac	2021-04-27 13:40:22.965003300 +0200
@@ -425,7 +425,7 @@
   LTA="la"
   STRIP_LIB_DEBUG=":"
   LIBSFX=la
-  ICP_LIB="${LIBTOOLPROG} --mode=install install -s"
+  ICP_LIB="${LIBTOOLPROG} --mode=install install"
 else
   ICP=cp
   LTO="o"
