Index: racket-7.2/src/cfg-racket
===================================================================
--- racket-7.2.orig/src/cfg-racket
+++ racket-7.2/src/cfg-racket
@@ -6833,7 +6833,7 @@ if test "${enable_shared}" = "yes" ; the
   LIBSFX=la
   WXLIBS=WXLIBSDYN
   ICP="${LIBTOOLPROG} --mode=install cp"
-  ICP_LIB="${LIBTOOLPROG} --mode=install install -s"
+  ICP_LIB="${LIBTOOLPROG} --mode=install install"
   MRLIBINSTALL="install-lib"
   LIBFINISH="${LIBTOOLPROG} --mode=finish"
   LTO="lo"
Index: racket-7.2/src/cs/c/configure
===================================================================
--- racket-7.2.orig/src/cs/c/configure
+++ racket-7.2/src/cs/c/configure
@@ -4615,7 +4615,7 @@ if test "${enable_shared}" = "yes" ; the
   LTA="la"
   STRIP_LIB_DEBUG=":"
   LIBSFX=la
-  ICP_LIB="${LIBTOOLPROG} --mode=install install -s"
+  ICP_LIB="${LIBTOOLPROG} --mode=install install"
 else
   ICP=cp
   LTO="o"
Index: racket-7.2/src/cs/c/configure.ac
===================================================================
--- racket-7.2.orig/src/cs/c/configure.ac
+++ racket-7.2/src/cs/c/configure.ac
@@ -415,7 +415,7 @@ if test "${enable_shared}" = "yes" ; the
   LTA="la"
   STRIP_LIB_DEBUG=":"
   LIBSFX=la
-  ICP_LIB="${LIBTOOLPROG} --mode=install install -s"
+  ICP_LIB="${LIBTOOLPROG} --mode=install install"
 else
   ICP=cp
   LTO="o"
Index: racket-7.2/src/racket/configure.ac
===================================================================
--- racket-7.2.orig/src/racket/configure.ac
+++ racket-7.2/src/racket/configure.ac
@@ -1382,7 +1382,7 @@ if test "${enable_shared}" = "yes" ; the
   LIBSFX=la
   WXLIBS=WXLIBSDYN
   ICP="${LIBTOOLPROG} --mode=install cp"
-  ICP_LIB="${LIBTOOLPROG} --mode=install install -s"
+  ICP_LIB="${LIBTOOLPROG} --mode=install install"
   MRLIBINSTALL="install-lib"
   LIBFINISH="${LIBTOOLPROG} --mode=finish"
   LTO="lo"
Index: racket-7.2/src/rktio/configure
===================================================================
--- racket-7.2.orig/src/rktio/configure
+++ racket-7.2/src/rktio/configure
@@ -4490,7 +4490,7 @@ if test "${enable_shared}" = "yes" ; the
   LTA="la"
   STRIP_LIB_DEBUG=":"
   LIBSFX=la
-  ICP_LIB="${LIBTOOLPROG} --mode=install install -s"
+  ICP_LIB="${LIBTOOLPROG} --mode=install install"
 else
   ICP=cp
   LTO="o"
Index: racket-7.2/src/rktio/configure.ac
===================================================================
--- racket-7.2.orig/src/rktio/configure.ac
+++ racket-7.2/src/rktio/configure.ac
@@ -384,7 +384,7 @@ if test "${enable_shared}" = "yes" ; the
   LTA="la"
   STRIP_LIB_DEBUG=":"
   LIBSFX=la
-  ICP_LIB="${LIBTOOLPROG} --mode=install install -s"
+  ICP_LIB="${LIBTOOLPROG} --mode=install install"
 else
   ICP=cp
   LTO="o"
Index: racket-7.2/src/start/configure.ac
===================================================================
--- racket-7.2.orig/src/start/configure.ac
+++ racket-7.2/src/start/configure.ac
@@ -1360,7 +1360,7 @@ if test "${enable_shared}" = "yes" ; the
   LIBSFX=la
   WXLIBS=WXLIBSDYN
   ICP="${LIBTOOLPROG} --mode=install cp"
-  ICP_LIB="${LIBTOOLPROG} --mode=install install -s"
+  ICP_LIB="${LIBTOOLPROG} --mode=install install"
   MRLIBINSTALL="install-lib"
   LIBFINISH="${LIBTOOLPROG} --mode=finish"
   LTO="lo"
