--- evolution-rss-0.3.95+git.20171129-orig/configure.ac	2017-11-29 10:58:57.000000000 +0100
+++ evolution-rss-0.3.95+git.20171129/configure.ac	2018-06-13 23:39:43.646407501 +0200
@@ -321,12 +321,6 @@
 AC_SUBST(RENDER)
 AC_SUBST(RENDER_N)
 
-dnl ***********
-dnl GConf stuff
-dnl ***********
-AC_PATH_PROG(GCONFTOOL, gconftool-2, no)
-AM_GCONF_SOURCE_2
-
 dnl ***************
 dnl Gsettings stuff
 dnl ***************
