Workaround

[ 3445s] I: Program returns random data in a function
[ 3445s] E: gcc5 no-return-in-nonvoid-function ../../gcc/cp/constexpr.c:3676

during stage1 when compiling with a GCC 4.3 host compiler

--- gcc/cp/constexpr.c.orig	2015-06-02 09:33:15.085636193 +0200
+++ gcc/cp/constexpr.c	2015-06-02 09:34:04.637080196 +0200
@@ -3668,11 +3668,10 @@
 	{
 	  r = get_target_expr (r);
 	  TREE_CONSTANT (r) = true;
-	  return r;
 	}
     }
-  else
-    return r;
+
+  return r;
 }
 
 /* Returns true if T is a valid subexpression of a constant expression,
