diff -rup a/configure.ac b/configure.ac
--- a/configure.ac	2024-12-15 10:50:18.000000000 +0100
+++ b/configure.ac	2024-12-15 15:54:46.636204292 +0100
@@ -6,7 +6,7 @@ dnl xfdesktop - xfce4's desktop manager
 dnl
 
 dnl minimum required versions
-m4_define([libx11_minimum_version], [1.6.7])
+m4_define([libx11_minimum_version], [1.6.5])
 m4_define([glib_minimum_version], [2.72.0])
 m4_define([gtk_minimum_version], [3.24.0])
 m4_define([libxfce4util_minimum_version], [4.13.0])
diff -rup a/configure b/configure
--- a/configure	2024-12-15 10:51:59.000000000 +0100
+++ b/configure	2024-12-15 15:55:12.989531245 +0100
@@ -19610,11 +19610,11 @@ fi
 
 
   if test x"$xdt_feature_deps_check_only" = x"yes"; then
-    if ! $PKG_CONFIG --exists "x11 >= 1.6.7" >/dev/null 2>&1; then
+    if ! $PKG_CONFIG --exists "x11 >= 1.6.5" >/dev/null 2>&1; then
       if test x"$xdt_feature_deps_missing" = x""; then
-        xdt_feature_deps_missing="x11 >= 1.6.7"
+        xdt_feature_deps_missing="x11 >= 1.6.5"
       else
-        xdt_feature_deps_missing="$xdt_feature_deps_missing, x11 >= 1.6.7"
+        xdt_feature_deps_missing="$xdt_feature_deps_missing, x11 >= 1.6.5"
       fi
     fi
   else
@@ -19768,9 +19768,9 @@ fi
 
 
 
-  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for x11 >= 1.6.7" >&5
-printf %s "checking for x11 >= 1.6.7... " >&6; }
-  if $PKG_CONFIG "--atleast-version=1.6.7" "x11" >/dev/null 2>&1; then
+  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for x11 >= 1.6.5" >&5
+printf %s "checking for x11 >= 1.6.5... " >&6; }
+  if $PKG_CONFIG "--atleast-version=1.6.5" "x11" >/dev/null 2>&1; then
     LIBX11_VERSION=`$PKG_CONFIG --modversion "x11"`
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $LIBX11_VERSION" >&5
 printf "%s\n" "$LIBX11_VERSION" >&6; }
@@ -19787,7 +19787,7 @@ printf %s "checking LIBX11_LIBS... " >&6
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $LIBX11_LIBS" >&5
 printf "%s\n" "$LIBX11_LIBS" >&6; }
 
-    LIBX11_REQUIRED_VERSION=1.6.7
+    LIBX11_REQUIRED_VERSION=1.6.5
 
 
 
@@ -19816,7 +19816,7 @@ printf "%s\n" "found, but $xdt_cv_versio
 
       echo "*** The required package x11 was found on your system,"
       echo "*** but the installed version ($xdt_cv_version) is too old."
-      echo "*** Please upgrade x11 to atleast version 1.6.7, or adjust"
+      echo "*** Please upgrade x11 to atleast version 1.6.5, or adjust"
       echo "*** the PKG_CONFIG_PATH environment variable if you installed"
       echo "*** the new version of the package in a nonstandard prefix so"
       echo "*** pkg-config is able to find it."
@@ -19828,7 +19828,7 @@ printf "%s\n" "not found" >&6; }
 
 
       echo "*** The required package x11 was not found on your system."
-      echo "*** Please install x11 (atleast version 1.6.7) or adjust"
+      echo "*** Please install x11 (atleast version 1.6.5) or adjust"
       echo "*** the PKG_CONFIG_PATH environment variable if you"
       echo "*** installed the package in a nonstandard prefix so that"
       echo "*** pkg-config is able to find it."
@@ -20310,11 +20310,11 @@ printf "%s\n" "not found" >&6; }
 
 
   if test x"$xdt_feature_deps_check_only" = x"yes"; then
-    if ! $PKG_CONFIG --exists "x11 >= 1.6.7" >/dev/null 2>&1; then
+    if ! $PKG_CONFIG --exists "x11 >= 1.6.5" >/dev/null 2>&1; then
       if test x"$xdt_feature_deps_missing" = x""; then
-        xdt_feature_deps_missing="x11 >= 1.6.7"
+        xdt_feature_deps_missing="x11 >= 1.6.5"
       else
-        xdt_feature_deps_missing="$xdt_feature_deps_missing, x11 >= 1.6.7"
+        xdt_feature_deps_missing="$xdt_feature_deps_missing, x11 >= 1.6.5"
       fi
     fi
   else
@@ -20468,9 +20468,9 @@ fi
 
 
 
-  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for x11 >= 1.6.7" >&5
-printf %s "checking for x11 >= 1.6.7... " >&6; }
-  if $PKG_CONFIG "--atleast-version=1.6.7" "x11" >/dev/null 2>&1; then
+  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for x11 >= 1.6.5" >&5
+printf %s "checking for x11 >= 1.6.5... " >&6; }
+  if $PKG_CONFIG "--atleast-version=1.6.5" "x11" >/dev/null 2>&1; then
     LIBX11_VERSION=`$PKG_CONFIG --modversion "x11"`
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $LIBX11_VERSION" >&5
 printf "%s\n" "$LIBX11_VERSION" >&6; }
@@ -20487,7 +20487,7 @@ printf %s "checking LIBX11_LIBS... " >&6
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $LIBX11_LIBS" >&5
 printf "%s\n" "$LIBX11_LIBS" >&6; }
 
-    LIBX11_REQUIRED_VERSION=1.6.7
+    LIBX11_REQUIRED_VERSION=1.6.5
 
 
 
@@ -20516,7 +20516,7 @@ printf "%s\n" "found, but $xdt_cv_versio
 
       echo "*** The required package x11 was found on your system,"
       echo "*** but the installed version ($xdt_cv_version) is too old."
-      echo "*** Please upgrade x11 to atleast version 1.6.7, or adjust"
+      echo "*** Please upgrade x11 to atleast version 1.6.5, or adjust"
       echo "*** the PKG_CONFIG_PATH environment variable if you installed"
       echo "*** the new version of the package in a nonstandard prefix so"
       echo "*** pkg-config is able to find it."
@@ -20528,7 +20528,7 @@ printf "%s\n" "not found" >&6; }
 
 
       echo "*** The required package x11 was not found on your system."
-      echo "*** Please install x11 (atleast version 1.6.7) or adjust"
+      echo "*** Please install x11 (atleast version 1.6.5) or adjust"
       echo "*** the PKG_CONFIG_PATH environment variable if you"
       echo "*** installed the package in a nonstandard prefix so that"
       echo "*** pkg-config is able to find it."
