diff -rup a/meson.build b/meson.build
--- a/meson.build	2025-02-17 09:57:17.000000000 +0100
+++ b/meson.build	2025-10-22 13:15:38.919268643 +0200
@@ -21,7 +21,7 @@ dependency_versions = {
   'libnotify': '>= 0.4.0',
   'libx11': '>= 1.6.7',
   'libxfce4windowing': '>= 4.19.8',
-  'libyaml': '>= 0.2.5',
+  'libyaml': '>= 0.1.7',
   'xfce4': '>= 4.18.0',
   'xfconf': '>= 4.19.3',
   'gtk-layer-shell': '>= 0.7.0',
