diff --git a/Makefile b/Makefile
index 1daa431..c4525db 100644
--- a/Makefile
+++ b/Makefile
@@ -1,12 +1,12 @@
 VERSION	= 0.9.1
 
-PREFIX	= /usr
+PREFIX	?= /usr
 
 CC	= gcc
-CFLAGS	= -W -Wall -Wextra -Werror \
+CFLAGS	+= -W -Wall -Wextra -Werror \
 	-DVERSION=\"$(VERSION)\" \
 	-g
-LDFLAGS =
+LDFLAGS ?=
 
 all: atinout
 
diff --git a/atinout.c b/atinout.c
index 4801547..ae0444e 100644
--- a/atinout.c
+++ b/atinout.c
@@ -141,7 +141,7 @@ static bool is_final_result(const char * const response)
 		if (strcmp(&response[1], "K\r\n") == 0) {
 			return true;
 		}
-		/* no break */
+		// fallthrough
 	default:
 		return false;
 	}
@@ -170,7 +170,7 @@ int main(int argc, char *argv[])
 			help(argv[0]);
 			return EXIT_SUCCESS;
 		case 'V':
-			printf("atinout version " VERSION "\n");
+			printf("atinout version %s \n", VERSION);
 			if (argc == 2) {
 				printf("Copyright (C) 2013 Håkon Løvdal <hlovdal@users.sourceforge.net>\n"
 				       "This program comes with ABSOLUTELY NO WARRANTY.\n"
