From 0cd471dcfd46e6cb8b71eceddb20cc02eadabf61 Mon Sep 17 00:00:00 2001
From: Robin Ebert <ebertrobin2002@gmail.com>
Date: Tue, 15 Oct 2024 16:26:05 +0200
Subject: [PATCH 2/3] egl-wayland: Fix use after free in library teardown

---
 src/wayland-egldisplay.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/wayland-egldisplay.c b/src/wayland-egldisplay.c
index 7089271..edd79c4 100644
--- a/src/wayland-egldisplay.c
+++ b/src/wayland-egldisplay.c
@@ -729,10 +729,10 @@ static EGLBoolean terminateDisplay(WlEglDisplay *display, EGLBoolean globalTeard
      * destroy the display connection itself */
     wlEglDestroyAllSurfaces(display);
 
-    wlEglDestroyFormatSet(&display->formatSet);
-    wlEglDestroyFeedback(&display->defaultFeedback);
-
     if (!globalTeardown || display->ownNativeDpy) {
+        wlEglDestroyFormatSet(&display->formatSet);
+        wlEglDestroyFeedback(&display->defaultFeedback);
+
         if (display->wlRegistry) {
             wl_registry_destroy(display->wlRegistry);
             display->wlRegistry = NULL;
-- 
2.43.0

