Index: bc-java-r1rv85/pg/src/main/java/org/bouncycastle/bcpg/ArmoredOutputStream.java
===================================================================
--- bc-java-r1rv85.orig/pg/src/main/java/org/bouncycastle/bcpg/ArmoredOutputStream.java
+++ bc-java-r1rv85/pg/src/main/java/org/bouncycastle/bcpg/ArmoredOutputStream.java
@@ -633,7 +633,7 @@ public class ArmoredOutputStream
 
             if (comment.length() > availableCommentCharsPerLine)
             {
-                comment = comment.substring(0, availableCommentCharsPerLine - 1) + '…';
+                comment = comment.substring(0, availableCommentCharsPerLine - "…".length()) + "…";
             }
             return addComment(comment);
         }
