Index: sphinxcontrib-blockdiag-3.0.0/sphinxcontrib/__init__.py
===================================================================
--- sphinxcontrib-blockdiag-3.0.0.orig/sphinxcontrib/__init__.py
+++ /dev/null
@@ -1,13 +0,0 @@
-# -*- coding: utf-8 -*-
-"""
-    sphinxcontrib
-    ~~~~~~~~~~~~~
-
-    This package is a namespace package that contains all extensions
-    distributed in the ``sphinx-contrib`` distribution.
-
-    :copyright: Copyright 2007-2009 by the Sphinx team, see AUTHORS.
-    :license: BSD, see LICENSE for details.
-"""
-
-__import__('pkg_resources').declare_namespace(__name__)
Index: sphinxcontrib-blockdiag-3.0.0/sphinxcontrib/blockdiag.py
===================================================================
--- sphinxcontrib-blockdiag-3.0.0.orig/sphinxcontrib/blockdiag.py
+++ sphinxcontrib-blockdiag-3.0.0/sphinxcontrib/blockdiag.py
@@ -14,9 +14,9 @@ import os
 import re
 import posixpath
 import traceback
-import pkg_resources
 from collections import namedtuple
 from docutils import nodes
+from importlib.metadata import version
 from sphinx import addnodes
 from sphinx.util import logging
 from sphinx.util.osutil import ensuredir
@@ -312,7 +312,7 @@ def setup(app):
     app.connect("doctree-resolved", on_doctree_resolved)
 
     return {
-        'version': pkg_resources.require('blockdiag')[0].version,
+        'version': version('blockdiag'),
         'parallel_read_safe': True,
         'parallel_write_safe': True,
     }
Index: sphinxcontrib-blockdiag-3.0.0/setup.py
===================================================================
--- sphinxcontrib-blockdiag-3.0.0.orig/setup.py
+++ sphinxcontrib-blockdiag-3.0.0/setup.py
@@ -1,6 +1,6 @@
 # -*- coding: utf-8 -*-
 
-from setuptools import setup, find_packages
+from setuptools import setup, find_namespace_packages
 
 requires = ['blockdiag>=1.5.0', 'Sphinx>=2.0']
 
@@ -33,9 +33,8 @@ setup(
         'Topic :: Utilities',
     ],
     platforms='any',
-    packages=find_packages(exclude=['tests']),
+    packages=find_namespace_packages(where=".", exclude=['tests*']),
     include_package_data=True,
     python_requires=">=3.7",
     install_requires=requires,
-    namespace_packages=['sphinxcontrib'],
 )
