diff -upr Flask-SQLAlchemy-2.5.1.orig/tests/test_config.py Flask-SQLAlchemy-2.5.1/tests/test_config.py
--- Flask-SQLAlchemy-2.5.1.orig/tests/test_config.py	2022-05-17 13:45:23.513344508 +0200
+++ Flask-SQLAlchemy-2.5.1/tests/test_config.py	2022-05-17 13:45:23.521344560 +0200
@@ -1,6 +1,6 @@
 import os
 
-import mock
+from unittest import mock
 import pytest
 from sqlalchemy.pool import NullPool
 
diff -upr Flask-SQLAlchemy-2.5.1.orig/tests/test_utils.py Flask-SQLAlchemy-2.5.1/tests/test_utils.py
--- Flask-SQLAlchemy-2.5.1.orig/tests/test_utils.py	2022-05-17 13:45:23.513344508 +0200
+++ Flask-SQLAlchemy-2.5.1/tests/test_utils.py	2022-05-17 13:45:23.521344560 +0200
@@ -1,4 +1,4 @@
-import mock
+from unittest import mock
 
 from flask_sqlalchemy import utils
 
