Index: libsamplerate-0.1.9/tests/Makefile.in
===================================================================
--- libsamplerate-0.1.9.orig/tests/Makefile.in
+++ libsamplerate-0.1.9/tests/Makefile.in
@@ -1136,7 +1136,7 @@ check: $(noinst_PROGRAMS)
 	./varispeed_test
 	./float_short_test
 	./snr_bw_test
-	./throughput_test
+	#./throughput_test # disabled to have reproducible builds with PGO - see https://github.com/bmwiedemann/theunreproduciblepackage/tree/master/pgo
 	@echo "-----------------------------------------------------------------"
 	@echo "  ${PACKAGE}-${VERSION} passed all tests."
 	@echo "-----------------------------------------------------------------"
