Index: rstudio-2024.04.0-735/src/cpp/CMakeLists.txt
===================================================================
--- rstudio-2024.04.0-735.orig/src/cpp/CMakeLists.txt
+++ rstudio-2024.04.0-735/src/cpp/CMakeLists.txt
@@ -281,7 +281,7 @@
       set(Boost_USE_STATIC_LIBS ON)
       set(BOOST_INCLUDEDIR  ${RSTUDIO_TOOLS_BOOST}/include)
       find_package(Boost ${RSTUDIO_BOOST_REQUESTED_VERSION} HINTS "${RSTUDIO_TOOLS_BOOST}" EXACT REQUIRED)
-      if(NOT Boost_VERSION LESS 106900)
+      if(NOT Boost_VERSION LESS 106600)
          list(REMOVE_ITEM BOOST_LIBS signals)
       endif()
 
