Index: ghc-9.4.4/libraries/containers/containers/include/containers.h
===================================================================
--- ghc-9.4.4.orig/libraries/containers/containers/include/containers.h
+++ ghc-9.4.4/libraries/containers/containers/include/containers.h
@@ -23,7 +23,6 @@
  * Unboxed arrays are broken on big-endian architectures.
  * See https://gitlab.haskell.org/ghc/ghc/-/issues/16998
  */
-# define USE_UNBOXED_ARRAYS 1
 #endif
 #endif
 
