Index: ghc-9.4.4/rts/package.conf.in
===================================================================
--- ghc-9.4.4.orig/rts/package.conf.in
+++ ghc-9.4.4/rts/package.conf.in
@@ -66,6 +66,7 @@ extra-libraries:
 #if NEED_ATOMIC_LIB
                              , "atomic"
 #endif
+                             , "atomic"
 
 #if defined(INSTALLING)
 include-dirs:           INCLUDE_DIR FFI_INCLUDE_DIR
Index: ghc-9.4.4/rts/rts.cabal.in
===================================================================
--- ghc-9.4.4.orig/rts/rts.cabal.in
+++ ghc-9.4.4/rts/rts.cabal.in
@@ -152,6 +152,7 @@ library
        extra-libraries: elf dw
     if flag(libnuma)
        extra-libraries: numa
+    extra-libraries: atomic
     if !flag(smp)
        cpp-options: -DNOSMP
 
