Index: Twisted-21.2.0/src/twisted/test/test_failure.py
===================================================================
--- Twisted-21.2.0.orig/src/twisted/test/test_failure.py
+++ Twisted-21.2.0/src/twisted/test/test_failure.py
@@ -22,7 +22,8 @@ from twisted.python import failure
 from twisted.trial.unittest import SynchronousTestCase
 
 
-from cython_test_exception_raiser import raiser
+#from cython_test_exception_raiser import raiser
+raiser = None
 
 
 def getDivisionFailure(*args, **kwargs):
