Index: abseil-cpp-20240722.0/CMakeLists.txt
===================================================================
--- abseil-cpp-20240722.0.orig/CMakeLists.txt
+++ abseil-cpp-20240722.0/CMakeLists.txt
@@ -16,7 +16,7 @@
 
 # https://github.com/google/oss-policies-info/blob/main/foundational-cxx-support-matrix.md
 # As of 2024-07-01, CMake 3.16 is the minimum supported version.
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.5)
 
 # Allow the user to specify the CMAKE_MSVC_DEBUG_INFORMATION_FORMAT
 if (POLICY CMP0141)
