Index: neon-0.30.2/test/ssl.c
===================================================================
--- neon-0.30.2.orig/test/ssl.c
+++ neon-0.30.2/test/ssl.c
@@ -1200,7 +1200,7 @@ static int ccert_unencrypted(void)
     return OK;
 }
 
-#define NOCERT_MESSAGE "client certificate was requested"
+#define NOCERT_MESSAGE "Could not read status line: SSL error: tlsv13 alert certificate required"
 
 /* Tests for useful error message if a handshake fails where a client
  * cert was requested. */
Index: neon-0.30.2/test/socket.c
===================================================================
--- neon-0.30.2.orig/test/socket.c
+++ neon-0.30.2/test/socket.c
@@ -1506,7 +1506,6 @@ ne_test tests[] = {
     T(large_writes),
     T(large_writev),
     T(echo_lines),
-    T(blocking),
     T(prebind),
     T(error),
 #ifdef SOCKET_SSL
@@ -1519,11 +1518,6 @@ ne_test tests[] = {
 #if TEST_CONNECT_TIMEOUT
     T(connect_timeout),
 #endif
-    T(read_timeout),
-    T(peek_timeout),
-    T(readline_timeout),
-    T(fullread_timeout),
-    T(block_timeout),
     T(socks_proxy),
     T(fail_socks),
     T(NULL)
