diff -rub cosmic-session-1.0.0~alpha5+1/Justfile cosmic-session-1.0.0~alpha5+1-patched/Justfile
--- cosmic-session-1.0.0~alpha5+1/Justfile	2025-01-10 15:28:23.000000000 +0100
+++ cosmic-session-1.0.0~alpha5+1-patched/Justfile	2025-01-11 01:44:34.364203789 +0100
@@ -1,6 +1,6 @@
 rootdir := ''
 etcdir := rootdir + '/etc'
-prefix := rootdir + '/usr'
+prefix := '/usr'
 clean := '0'
 debug := '0'
 vendor := '0'
@@ -13,10 +13,10 @@
 orca := '/usr/bin/orca'
 cosmic_dconf_profile := '/usr/share/dconf/profile/cosmic'
 
-bindir := prefix + '/bin'
-systemddir := prefix + '/lib/systemd/user'
-sessiondir := prefix + '/share/wayland-sessions'
-applicationdir := prefix + '/share/applications'
+bindir := rootdir + prefix + '/bin'
+systemddir := rootdir + prefix + '/lib/systemd/user'
+sessiondir := rootdir + prefix + '/share/wayland-sessions'
+applicationdir := rootdir + prefix + '/share/applications'
 
 all: _extract_vendor build
 
